From 7b77a9288e728ae3fbc45e06c2f6a21501875444 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20Dutkiewicz?= <emdeck@gmail.com>
Date: Tue, 2 Jul 2019 22:54:56 +0200
Subject: [PATCH] Bump required CMake version to 3.7.0

---
 CMakeLists.txt | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ae4843c865..4999e2699b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,12 +1,4 @@
-cmake_minimum_required(VERSION 3.1.0)
-
-if (POLICY CMP0020)
-	cmake_policy(SET CMP0020 NEW)
-endif ()
-
-if (POLICY CMP0043)
-	cmake_policy(SET CMP0043 NEW)
-endif ()
+cmake_minimum_required(VERSION 3.7.0)
 
 project(otter-browser)
 
