--- a/libcperciva/cpusupport/Build/cpusupport.sh	2022-04-16 05:30:41.748083177 +0000
+++ b/libcperciva/cpusupport/Build/cpusupport.sh	2022-04-16 05:31:47.356988582 +0000
@@ -81,14 +81,3 @@
     "-msse4.2 -Wno-cast-align -fno-strict-aliasing -Wno-cast-qual"
 feature X86 SSSE3 "" "-mssse3"						\
     "-mssse3 -Wno-cast-align"
-
-# Detect specific ARM features
-feature ARM AES "-march=armv8.1-a+crypto"				\
-    "-march=armv8.1-a+crypto -D__ARM_ACLE=200"
-feature ARM CRC32_64 "-march=armv8.1-a"					\
-    "-march=armv8.1-a+crc"						\
-    "-march=armv8.1-a+crc -Wno-cast-align"				\
-    "-march=armv8.1-a -D__ARM_ACLE=200"
-feature ARM SHA256 "-march=armv8.1-a+crypto"				\
-    "-march=armv8.1-a+crypto -Wno-cast-align"				\
-    "-march=armv8.1-a+crypto -D__ARM_ACLE=200"
