--- a/src/gpsim/protocol.cc
+++ b/src/gpsim/protocol.cc
@@ -55,9 +55,9 @@
   return ret;
 }
 
-static uint ascii2uint64(char *buffer, int digits)
+static uint64_t ascii2uint64(char *buffer, int digits)
 {
-  uint ret = 0;
+  uint64_t ret = 0;
   char *b = buffer;
 
   for(int i=0; i<digits; i++) ret = (ret << 4) + a2i(*b++);
@@ -67,7 +67,7 @@
 
 static uint ascii2uint(char *buffer, int digits)
 {
-  uint long long i = ascii2uint64(buffer,digits);
+  uint64_t i = ascii2uint64(buffer,digits);
   return (uint) i;
 }
 
