diff -uNr zig-bootstrap-0.11.0/zig/lib/libc/musl/src/stat/fstatat.c zig-bootstrap-0.11.0.mod/zig/lib/libc/musl/src/stat/fstatat.c
--- zig-bootstrap-0.11.0/zig/lib/libc/musl/src/stat/fstatat.c	2023-08-11 09:21:14.000000000 +0800
+++ zig-bootstrap-0.11.0.mod/zig/lib/libc/musl/src/stat/fstatat.c	2023-09-06 21:34:13.166214431 +0800
@@ -7,6 +7,9 @@
 #include <sys/sysmacros.h>
 #include "syscall.h"
 
+#undef SYS_lstat
+#undef SYS_stat
+
 struct statx {
 	uint32_t stx_mask;
 	uint32_t stx_blksize;
