diff --git a/zig/lib/std/zig/system.zig b/zig/lib/std/zig/system.zig
index 83c798c3..69aef66e 100644
--- a/zig/lib/std/zig/system.zig
+++ b/zig/lib/std/zig/system.zig
@@ -996,6 +996,8 @@ fn detectAbiAndDynamicLinker(
             .haiku => "/bin/env",
         };
 
+        file_name = "@TERMUX_PREFIX@/bin/env";
+
         // According to `man 2 execve`:
         //
         // The kernel imposes a maximum length on the text
