diff -u -r ../gdb-7.10.1/gdb/inflow.c ./gdb/inflow.c
--- ../gdb-7.10.1/gdb/inflow.c	2015-12-05 10:16:45.000000000 -0500
+++ ./gdb/inflow.c	2016-02-07 19:28:38.979181296 -0500
@@ -862,10 +862,10 @@
     {
 #if defined (HAVE_TERMIOS) || defined (TIOCGPGRP)
 #ifdef HAVE_SETPGID
-      /* The call setpgid (0, 0) is supposed to work and mean the same
-         thing as this, but on Ultrix 4.2A it fails with EPERM (and
-         setpgid (getpid (), getpid ()) succeeds).  */
-      retval = setpgid (getpid (), getpid ());
+      /* Termux change: setpgid(getpid(), getpid()), which is used upstream
+	 for working around Ultrix 4.2A issue, does not work on Android but
+	 fails with ESRCH. So revert back to simple setpgid(0,0). */
+      retval = setpgid (0, 0);
 #else
 #ifdef HAVE_SETPGRP
 #ifdef SETPGRP_VOID 
