diff -u -r ../gdb-7.11/gdb/gdbserver/configure ./gdb/gdbserver/configure
--- ../gdb-7.11/gdb/gdbserver/configure	2016-02-09 22:19:39.000000000 -0500
+++ ./gdb/gdbserver/configure	2016-03-10 10:17:40.111988028 -0500
@@ -6735,17 +6735,6 @@
 
 
 case "${target}" in
-  *-android*)
-    # Starting with NDK version 9, <elf.h> actually includes definitions
-    # of Elf32_auxv_t and Elf64_auxv_t. But sadly, <elf.h> includes
-    # <sys/exec_elf.h> which defines some of the ELF types incorrectly,
-    # leading to conflicts with the defintions from <linux/elf.h>.
-    # This makes it impossible for us to include both <elf.h> and
-    # <linux/elf.h>, which means that, in practice, we do not have
-    # access to Elf32_auxv_t and Elf64_auxv_t on this platform.
-    # Therefore, do not try to auto-detect availability, as it would
-    # get it wrong on this platform.
-    ;;
   *)
     ac_fn_c_check_type "$LINENO" "Elf32_auxv_t" "ac_cv_type_Elf32_auxv_t" "#include <elf.h>
 
