diff -u -r ../gdb-8.1/gdb/gnulib/import/string.in.h ./gdb/gnulib/import/string.in.h
--- ../gdb-8.1/gdb/gnulib/import/string.in.h	2017-06-04 17:51:26.000000000 +0200
+++ ./gdb/gnulib/import/string.in.h	2018-02-01 22:09:38.398023253 +0100
@@ -183,9 +183,8 @@
 #if @GNULIB_MEMPCPY@
 # if ! @HAVE_MEMPCPY@
 _GL_FUNCDECL_SYS (mempcpy, void *,
-                  (void *restrict __dest, void const *restrict __src,
-                   size_t __n)
-                  _GL_ARG_NONNULL ((1, 2)));
+                  (void * __dest, void const * __src,
+                   size_t __n));
 # endif
 _GL_CXXALIAS_SYS (mempcpy, void *,
                   (void *restrict __dest, void const *restrict __src,
