diff -u -r ../gdb-8.1/gdb/nat/fork-inferior.c ./gdb/nat/fork-inferior.c
--- ../gdb-8.1/gdb/nat/fork-inferior.c	2018-01-05 05:07:23.000000000 +0100
+++ ./gdb/nat/fork-inferior.c	2018-02-01 21:49:06.148349274 +0100
@@ -32,7 +32,7 @@
 
 /* Default shell file to be used if 'startup-with-shell' is set but
    $SHELL is not.  */
-#define SHELL_FILE "/bin/sh"
+#define SHELL_FILE "@TERMUX_PREFIX@/bin/sh"
 
 /* Build the argument vector for execv(3).  */
 
