diff -u -r ../gdb-8.1/gdb/common/signals-state-save-restore.c ./gdb/common/signals-state-save-restore.c
--- ../gdb-8.1/gdb/common/signals-state-save-restore.c	2018-01-31 03:58:50.000000000 +0100
+++ ./gdb/common/signals-state-save-restore.c	2018-02-01 21:38:47.791540495 +0100
@@ -69,6 +69,7 @@
 	  && oldact->sa_handler != SIG_DFL
 	  && oldact->sa_handler != SIG_IGN)
 	{
+#ifndef __ANDROID__
 	  found_preinstalled = true;
 
 	  /* Use raw fprintf here because we're being called in early
@@ -78,6 +79,7 @@
 		   _("warning: Found custom handler for signal "
 		     "%d (%s) preinstalled.\n"), i,
 		   strsignal (i));
+#endif
 	}
     }
 
