--- a/gnulib/configure
+++ b/gnulib/configure
@@ -3772,9 +3772,9 @@
 as_fn_append ac_func_list " getrandom"
 as_fn_append ac_header_list " sys/time.h"
 as_fn_append ac_func_list " gettimeofday"
-as_fn_append ac_header_list " glob.h"
-as_fn_append ac_func_list " glob"
-as_fn_append ac_func_list " glob_pattern_p"
+ac_cv_header_glob_h=no
+ac_cv_func_glob=no
+ac_cv_func_glob_pattern_p=no
 as_fn_append ac_func_list " getpwnam_r"
 as_fn_append ac_header_list " threads.h"
 as_fn_append ac_header_list " netdb.h"
