diff -u -r ../gdb-7.8.1/gdb/gregset.h ./gdb/gregset.h
--- ../gdb-7.8.1/gdb/gregset.h	2014-06-11 12:34:41.000000000 -0400
+++ ./gdb/gregset.h	2014-12-22 07:09:39.011700414 -0500
@@ -23,13 +23,7 @@
 #include <sys/procfs.h>
 #endif
 
-#ifndef GDB_GREGSET_T
-#define GDB_GREGSET_T gregset_t
-#endif
-
-#ifndef GDB_FPREGSET_T
-#define GDB_FPREGSET_T fpregset_t
-#endif
+#include "nm.h"
 
 typedef GDB_GREGSET_T gdb_gregset_t;
 typedef GDB_FPREGSET_T gdb_fpregset_t;
