diff -u -r ../gdb-7.9/gdb/gdbserver/linux-x86-low.c ./gdb/gdbserver/linux-x86-low.c
--- ../gdb-7.9/gdb/gdbserver/linux-x86-low.c	2015-02-19 06:58:07.000000000 -0500
+++ ./gdb/gdbserver/linux-x86-low.c	2015-03-17 18:29:50.412805866 -0400
@@ -38,6 +38,11 @@
 #include "tracepoint.h"
 #include "ax.h"
 
+#ifdef __ANDROID__
+#include <sys/ucontext.h>
+typedef fpregset_t elf_fpxregset_t;
+#endif
+
 #ifdef __x86_64__
 /* Defined in auto-generated file amd64-linux.c.  */
 void init_registers_amd64_linux (void);
@@ -113,7 +118,6 @@
 #endif
 
 #include <sys/reg.h>
-#include <sys/procfs.h>
 #include <sys/ptrace.h>
 #include <sys/uio.h>
 
