diff -u -r ../gdb-7.8.1/gdb/gdbserver/linux-low.c ./gdb/gdbserver/linux-low.c
--- ../gdb-7.8.1/gdb/gdbserver/linux-low.c	2014-10-29 15:45:50.000000000 -0400
+++ ./gdb/gdbserver/linux-low.c	2014-12-22 07:51:39.355702157 -0500
@@ -47,6 +47,7 @@
 #include "filestuff.h"
 #include "tracepoint.h"
 #include "hostio.h"
+#include <linux/auxvec.h> /* For AT_PHDR and AT_PHNUM defines */
 #ifndef ELFMAG0
 /* Don't include <linux/elf.h> here.  If it got included by gdb_proc_service.h
    then ELFMAG0 will have been defined.  If it didn't get included by
@@ -108,7 +109,6 @@
 # include "linux-btrace.h"
 #endif
 
-#ifndef HAVE_ELF32_AUXV_T
 /* Copied from glibc's elf.h.  */
 typedef struct
 {
@@ -121,9 +121,7 @@
 	 on 64-bit platforms and vice versa.  */
     } a_un;
 } Elf32_auxv_t;
-#endif
 
-#ifndef HAVE_ELF64_AUXV_T
 /* Copied from glibc's elf.h.  */
 typedef struct
 {
@@ -136,7 +134,6 @@
 	 on 64-bit platforms and vice versa.  */
     } a_un;
 } Elf64_auxv_t;
-#endif
 
 /* A list of all unknown processes which receive stop signals.  Some
    other process will presumably claim each of these as forked
