diff -u -r ../gdb-7.8.1/gdb/config/nm-linux.h ./gdb/config/nm-linux.h
--- ../gdb-7.8.1/gdb/config/nm-linux.h	2014-06-11 12:34:41.000000000 -0400
+++ ./gdb/config/nm-linux.h	2014-12-22 07:45:14.127701891 -0500
@@ -20,5 +20,20 @@
 /* Use elf_gregset_t and elf_fpregset_t, rather than
    gregset_t and fpregset_t.  */
 
+#ifndef NM_CONFIG_LINUX_H
+#define NM_CONFIG_LINUX_H
+
+#include <sys/ucontext.h>
+
+#ifdef __arm__
+/* Structure to describe FPU registers.  */
+typedef struct fpregset { } fpregset_t;
+#endif
+
+typedef gregset_t elf_gregset_t;
+typedef fpregset_t elf_fpregset_t;
+
 #define GDB_GREGSET_T  elf_gregset_t
 #define GDB_FPREGSET_T elf_fpregset_t
+
+#endif
