diff -u -r ../gdb-7.9/gdb/x86-linux-nat.c ./gdb/x86-linux-nat.c
--- ../gdb-7.9/gdb/x86-linux-nat.c	2015-02-20 12:11:44.000000000 -0500
+++ ./gdb/x86-linux-nat.c	2015-03-17 18:20:50.044805492 -0400
@@ -23,7 +23,6 @@
 #include "gdb_proc_service.h"
 #include <sys/ptrace.h>
 #include <sys/user.h>
-#include <sys/procfs.h>
 #include <sys/uio.h>
 
 #include "x86-nat.h"
@@ -39,6 +38,11 @@
 #include "x86-xstate.h"
 #include "nat/linux-btrace.h"
 
+#ifdef __ANDROID__
+#include <sys/ucontext.h>
+typedef fpregset_t elf_fpxregset_t;
+#endif
+
 /* Per-thread arch-specific data we want to keep.  */
 
 struct arch_lwp_info
