diff -u -r ../orig/jill/src/com/android/jill/frontend/java/JavaTransformer.java ./jill/src/com/android/jill/frontend/java/JavaTransformer.java
--- ../orig/jill/src/com/android/jill/frontend/java/JavaTransformer.java	2016-10-03 17:38:30.000000000 -0400
+++ ./jill/src/com/android/jill/frontend/java/JavaTransformer.java	2016-10-05 19:44:23.446481214 -0400
@@ -197,7 +197,7 @@
   @Nonnull
   private OutputVFS wrapOutputVFS(@Nonnull VFS baseVFS) {
     MessageDigestCodec mdCodec = new MessageDigestCodec();
-    Service service = mdCodec.parseString(new CodecContext(), "SHA");
+    Service service = mdCodec.parseString(new CodecContext(), "MD5");
 
     try {
       return new GenericOutputVFS(new DeflateFS(new MessageDigestFS(
