diff -u -r ../pristine/sched/src/com/android/sched/build/SchedAnnotationProcessor.java ./sched/src/com/android/sched/build/SchedAnnotationProcessor.java
--- ../pristine/sched/src/com/android/sched/build/SchedAnnotationProcessor.java	2017-02-17 23:55:50.000000000 +0100
+++ ./sched/src/com/android/sched/build/SchedAnnotationProcessor.java	2017-02-18 00:00:44.231965307 +0100
@@ -16,7 +16,6 @@
 
 package com.android.sched.build;
 
-import com.android.sched.util.findbugs.SuppressFBWarnings;
 
 import java.io.BufferedReader;
 import java.io.IOException;
@@ -101,7 +100,6 @@
     FILTER("com.android.sched.schedulable.ComponentFilter") {
       @Override
       @Nonnull
-      @SuppressFBWarnings("NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE") // probably a FindBugs bug
       public TypeMirror getTypeMirror() throws AbortException {
         return env.getTypeUtils().getDeclaredType(getTypeElement(),
             env.getTypeUtils().getWildcardType(null, null));
