--- a/meson.build
+++ b/meson.build
@@ -105,13 +105,13 @@
   libsixel_deps += [curl_found]
 endif
 
-if get_option('jpeg').enabled() and jpeg_found
-  conf_data.set('HAVE_JPEG', true)
+if get_option('jpeg').enabled() and jpeg_found.found()
+  conf_data.set('HAVE_JPEG', 1)
   libsixel_deps += [jpeg_found]
 endif
 
-if get_option('png').enabled() and png_found
-  conf_data.set('HAVE_PNG', true)
+if get_option('png').enabled() and png_found.found()
+  conf_data.set('HAVE_LIBPNG', 1)
   libsixel_deps += [png_found]
 endif
 
