diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index b8dfdd7..63ce251 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -17,10 +17,6 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
 # USA.
 
-if("${CMAKE_SOURCE_DIR}" MATCHES "src/?$")
-	message(FATAL_ERROR "\nPlease execute cmake from the directory above, not the src/ directory.")
-	return()
-endif()
 
 # On OSX, frameworks are found before installed libraries (cf. https://gitlab.kitware.com/cmake/cmake/-/issues/16427)
 if("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
