diff -uNr CUnit-2.1-3/doc/Makefile.am CUnit-2.1-3.mod/doc/Makefile.am
--- CUnit-2.1-3/doc/Makefile.am	2014-04-24 03:54:45.000000000 +0300
+++ CUnit-2.1-3.mod/doc/Makefile.am	2021-01-28 21:15:41.452347087 +0200
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-docdir = $(prefix)/doc/@PACKAGE@
+docdir = $(prefix)/share/doc/@PACKAGE@
 
 doc_DATA = \
 	CUnit_doc.css \
diff -uNr CUnit-2.1-3/doc/headers/Makefile.am CUnit-2.1-3.mod/doc/headers/Makefile.am
--- CUnit-2.1-3/doc/headers/Makefile.am	2014-04-24 03:54:45.000000000 +0300
+++ CUnit-2.1-3.mod/doc/headers/Makefile.am	2021-01-28 21:10:42.601482549 +0200
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-dochdrdir = $(prefix)/doc/@PACKAGE@/headers
+dochdrdir = $(prefix)/share/doc/@PACKAGE@/headers
 
 INCLUDE_FILES = \
 	Automated.h \
