--- a/meson.build
+++ b/meson.build
@@ -583,9 +583,6 @@
 pango_conf.set('HAVE_SYSPROF', libsysprof_capture_dep.found())
 pango_deps += libsysprof_capture_dep
 
-gidocgen_dep = dependency('gi-docgen', version: '>= 2021.1',
-                          fallback: ['gi-docgen', 'dummy_dep'],
-                          required: get_option('gtk_doc'))
 
 gnome = import('gnome')
 pkgconfig = import('pkgconfig')
