diff --color -uNr zorin-appgrid-lite/src/DesktopEntries.vala zorin-appgrid-lite-mod/src/DesktopEntries.vala
--- zorin-appgrid-lite/src/DesktopEntries.vala	2021-11-12 11:51:31.000000000 -0500
+++ zorin-appgrid-lite-mod/src/DesktopEntries.vala	2023-12-19 15:15:48.358989794 -0500
@@ -133,7 +133,7 @@
                     if (!icons.has_key (app_to_add["command"])) {
                         var app_icon = app.get_icon ().to_string ();
                         // This is not related to snap package... is related to the system itself
-                        var icon_prefix = "/usr/share/pixmaps/";
+                        var icon_prefix = "@TERMUX_PREFIX@/share/pixmaps/";
                         try {
                             if (icon_theme.has_icon (app_icon)) {
                                 /* Attention: the icons inside the icon_theme can tell lies about
