diff --git a/squashfuse-0.1.103/Makefile.am b/Makefile.am
index 1507525..8fbc17c 100644
--- a/squashfuse-0.1.103/Makefile.am
+++ b/Makefile.am
@@ -59,11 +59,11 @@ endif
 
 if SQ_WANT_DEMO
 # Sample program squashfuse_ls
-noinst_PROGRAMS += squashfuse_ls
+bin_PROGRAMS += squashfuse_ls
 squashfuse_ls_SOURCES = ls.c
 squashfuse_ls_LDADD = libsquashfuse.la $(COMPRESSION_LIBS)
 # Sample program squashfuse_extract
-noinst_PROGRAMS += squashfuse_extract
+bin_PROGRAMS += squashfuse_extract
 squashfuse_extract_CPPFLAGS = $(FUSE_CPPFLAGS)
 squashfuse_extract_SOURCES = extract.c
 squashfuse_extract_LDADD = libsquashfuse.la libfuseprivate.la $(COMPRESSION_LIBS) \
