--- a/configure.ac	2026-07-15 01:41:48.000000000 +0300
+++ b/configure.ac	2026-09-18 12:29:54.868567386 +0300
@@ -149,7 +149,7 @@
 AC_CHECK_FUNCS([mkfifo], [], [AC_MSG_ERROR([cannot find required function])])
 AC_CHECK_FUNCS([munmap], [], [AC_MSG_ERROR([cannot find required function])])
 AC_CHECK_FUNCS([openat], [], [AC_MSG_ERROR([cannot find required function])])
-AC_CHECK_FUNCS([reallocarray], [], [AC_MSG_ERROR([cannot find required function])])
+AC_CHECK_FUNCS([reallocarray])
 AC_CHECK_FUNCS([realpath], [], [AC_MSG_ERROR([cannot find required function])])
 AC_CHECK_FUNCS([regcomp], [], [AC_MSG_ERROR([cannot find required function])])
 AC_CHECK_FUNCS([rmdir], [], [AC_MSG_ERROR([cannot find required function])])
