diff -u -r ../vala-0.38.7/configure ./configure
--- ../vala-0.38.7/configure	2018-02-04 10:28:51.000000000 +0000
+++ ./configure	2018-02-05 11:39:59.451002816 +0000
@@ -12793,10 +12793,7 @@
 LIBADD="$LIBADD $LIBGVC_LIBS"
 CFLAGS="$CFLAGS $LIBGVC_CFLAGS"
 if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
+  have_cgraph=yes
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
