--- a/zellij-utils/Cargo.toml
+++ b/zellij-utils/Cargo.toml
@@ -57,7 +57,7 @@
 futures = "0.3.28"
 openssl-sys = { version = "0.9.93", features = ["vendored"], optional = true }
 isahc = { version = "1.7.2", default-features = false, features = ["http2", "text-decoding"] }
-curl-sys = { version = "0.4", features = ["force-system-lib-on-osx"], optional = true }
+curl-sys = { version = "0.4.78", features = ["force-system-lib-on-osx"], optional = true }
 
 [dev-dependencies]
 insta = { version = "1.6.0", features = ["backtrace"] }
