--- ../cache/ecl-16.1.3/src/configure	2016-12-19 10:25:00.000000000 +0000
+++ ./src/configure	2017-12-13 03:29:04.711246563 +0000
@@ -4880,7 +4880,7 @@
 SONAME=''
 SONAME_LDFLAGS=''
 case "${host_os}" in
-        linux-androideabi)
+        linux-android*)
                 thehost='android'
                 THREAD_CFLAGS='-D_THREAD_SAFE'
 #               THREAD_LIBS='-lpthread'
@@ -9500,7 +9500,7 @@
 
     CORE_LIBS="-leclatomic ${CORE_LIBS}"
   else
-    CORE_LIBS="-latomic_ops ${CORE_LIBS}"
+    CORE_LIBS="-lgc ${CORE_LIBS}"
   fi
 fi
 
