diff -uNr memcached-1.6.3/configure memcached-1.6.3.mod/configure
--- memcached-1.6.3/configure	2020-03-29 00:35:06.000000000 +0200
+++ memcached-1.6.3.mod/configure	2020-03-29 22:50:09.571008923 +0300
@@ -6687,9 +6687,6 @@
 
 fi
 
-if test "x$ac_cv_search_pthread_create" = "xno"; then
-  as_fn_error $? "Can't enable threads without the POSIX thread library." "$LINENO" 5
-fi
 
 for ac_func in mlockall
 do :
