diff --git a/configure.ac b/configure.ac
index 597a44b331a..fa78e09c192 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1134,6 +1134,7 @@ dnl architecture. PLATFORM_TRIPLET will be a pair or single value for these
 dnl platforms.
 AC_MSG_CHECKING([for multiarch])
 AS_CASE([$ac_sys_system],
+  [Linux-android], [MULTIARCH=""],
   [Darwin*], [MULTIARCH=""],
   [iOS], [MULTIARCH=""],
   [FreeBSD*], [MULTIARCH=""],
