--- Python-3.11.5/configure	2023-08-30 00:48:09.783767381 +0100
+++ Python-3.11.5.mod/configure	2023-08-30 00:52:09.383767498 +0100
@@ -6202,6 +6202,8 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for multiarch" >&5
 $as_echo_n "checking for multiarch... " >&6; }
 case $ac_sys_system in #(
+  Linux-android) :
+    MULTIARCH="" ;; #(
   Darwin*) :
     MULTIARCH="" ;; #(
   FreeBSD*) :
@@ -11027,7 +11029,7 @@
 		 then CCSHARED="-fPIC";
 		 else CCSHARED="+z";
 		 fi;;
-	Linux-android*) ;;
+	Linux-android*) CCSHARED="-fPIC";;
 	Linux*|GNU*) CCSHARED="-fPIC";;
 	Emscripten*|WASI*)
 		if test "x$enable_wasm_dynamic_linking" = xyes; then :
@@ -15331,7 +15333,6 @@
 $as_echo "$ac_cv_func_link" >&6; }
   if test "x$ac_cv_func_link" = xyes; then :
 
-$as_echo "#define HAVE_LINK 1" >>confdefs.h
 
 fi
 
