--- Python-3.11.0/configure	2022-10-24 23:05:39.000000000 +0530
+++ Python-3.11.0.mod/configure	2022-10-25 10:25:57.074049746 +0530
@@ -6189,6 +6189,8 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for multiarch" >&5
 $as_echo_n "checking for multiarch... " >&6; }
 case $ac_sys_system in #(
+  Linux-android) :
+    MULTIARCH="" ;; #(
   Darwin*) :
     MULTIARCH="" ;; #(
   FreeBSD*) :
@@ -10796,7 +10798,7 @@
 		 then CCSHARED="-fPIC";
 		 else CCSHARED="+z";
 		 fi;;
-	Linux-android*) ;;
+	Linux-android*) CCSHARED="-fPIC";;
 	Linux*|GNU*) CCSHARED="-fPIC";;
 	Emscripten*|WASI*)
 		if test "x$enable_wasm_dynamic_linking" = xyes; then :
@@ -15099,7 +15101,6 @@
 $as_echo "$ac_cv_func_link" >&6; }
   if test "x$ac_cv_func_link" = xyes; then :
 
-$as_echo "#define HAVE_LINK 1" >>confdefs.h
 
 fi
 
