--- ./Lib/distutils/command/build.py.orig	2019-11-19 23:19:08.878782120 +0000
+++ ./Lib/distutils/command/build.py	2019-11-19 23:19:18.410915201 +0000
@@ -118,7 +118,7 @@
 
         if self.executable is None and sys.executable:
             self.executable = os.path.normpath(sys.executable)
-
+        self.parallel = 1
         if isinstance(self.parallel, str):
             try:
                 self.parallel = int(self.parallel)
