diff -u -r ../Python-3.4.0rc1/Modules/pwdmodule.c ./Modules/pwdmodule.c
--- ../Python-3.4.0rc1/Modules/pwdmodule.c	2014-02-10 23:51:50.000000000 +0100
+++ ./Modules/pwdmodule.c	2014-02-13 02:16:12.000000000 +0100
@@ -72,7 +72,12 @@
     SETS(setIndex++, p->pw_passwd);
     PyStructSequence_SET_ITEM(v, setIndex++, _PyLong_FromUid(p->pw_uid));
     PyStructSequence_SET_ITEM(v, setIndex++, _PyLong_FromGid(p->pw_gid));
+#ifdef __ANDROID__
+    SETS(setIndex++, Py_None);
+    Py_INCREF(Py_None);
+#else
     SETS(setIndex++, p->pw_gecos);
+#endif
     SETS(setIndex++, p->pw_dir);
     SETS(setIndex++, p->pw_shell);
 
