--- ../cache/Python-3.6.5/pyconfig.h.in	2018-03-28 09:19:31.000000000 +0000
+++ ./pyconfig.h.in	2018-04-02 23:09:11.381919940 +0000
@@ -25,7 +25,7 @@
 
 /* Define if C doubles are 64-bit IEEE 754 binary format, stored with the
    least significant byte first */
-#undef DOUBLE_IS_LITTLE_ENDIAN_IEEE754
+#define DOUBLE_IS_LITTLE_ENDIAN_IEEE754
 
 /* Define if --enable-ipv6 is specified */
 #undef ENABLE_IPV6
