diff -u -r ../Python-3.4.1/Lib/subprocess.py ./Lib/subprocess.py
--- ../Python-3.4.1/Lib/subprocess.py	2014-05-19 07:19:38.000000000 +0200
+++ ./Lib/subprocess.py	2014-06-04 11:31:48.708843737 +0200
@@ -1344,7 +1344,7 @@
                 args = list(args)
 
             if shell:
-                args = ["/bin/sh", "-c"] + args
+                args = ["/system/bin/sh", "-c"] + args
                 if executable:
                     args[0] = executable
 
