--- src/src/tools/clangpchmanagerbackend/source/generatepchactionfactory.h	2020-07-07 06:44:20.000000000 +0000
+++ src.mod//src/tools/clangpchmanagerbackend/source/generatepchactionfactory.h	2021-05-24 12:46:08.430407000 +0000
@@ -44,7 +44,7 @@
 
     bool BeginInvocation(clang::CompilerInstance &compilerInstance) override
     {
-        compilerInstance.getPreprocessorOpts().DisablePCHValidation = true;
+        // compilerInstance.getPreprocessorOpts().DisablePCHValidation = true;
         compilerInstance.getPreprocessorOpts().AllowPCHWithCompilerErrors = true;
         compilerInstance.getDiagnosticOpts().ErrorLimit = 0;
         compilerInstance.getFrontendOpts().SkipFunctionBodies = true;
