--- a/src/plugins/cppeditor/cppquickfixes.cpp
+++ b/src/plugins/cppeditor/cppquickfixes.cpp
@@ -4371,7 +4371,14 @@
     };
     using Flag = GenerateGetterSetterOp::GenerateFlag;
     constexpr static Flag ColumnFlag[] = {
+#if defined(__clang__) && __clang_major__ >= 16
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wenum-constexpr-conversion"
+#endif
         static_cast<Flag>(-1),
+#if defined(__clang__) && __clang_major__ >= 16
+#pragma GCC diagnostic pop
+#endif
         Flag::GenerateGetter,
         Flag::GenerateSetter,
         Flag::GenerateSignal,
