--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -201,11 +201,12 @@
 add_library(platform_wsi_defines INTERFACE)
 if(WIN32)
     target_compile_definitions(platform_wsi_defines INTERFACE VK_USE_PLATFORM_WIN32_KHR)
-elseif(ANDROID)
-    target_compile_definitions(platform_wsi_defines INTERFACE VK_USE_PLATFORM_ANDROID_KHR)
 elseif(APPLE)
     target_compile_definitions(platform_wsi_defines INTERFACE VK_USE_PLATFORM_MACOS_MVK VK_USE_PLATFORM_METAL_EXT)
 elseif(UNIX AND NOT APPLE) # i.e.: Linux
+    if(ANDROID)
+        target_compile_definitions(platform_wsi_defines INTERFACE VK_USE_PLATFORM_ANDROID_KHR)
+    endif()
     if(BUILD_WSI_XCB_SUPPORT)
         target_compile_definitions(platform_wsi_defines INTERFACE VK_USE_PLATFORM_XCB_KHR)
     endif()
