diff -u -r ../timew-1.0.0/cmake/CXXSniffer.cmake ./cmake/CXXSniffer.cmake
--- ../timew-1.0.0/cmake/CXXSniffer.cmake	2016-08-17 23:22:31.000000000 +0200
+++ ./cmake/CXXSniffer.cmake	2017-01-15 18:34:29.424793652 +0100
@@ -22,6 +22,8 @@
 
 if (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
   set (LINUX true)
+elseif (${CMAKE_SYSTEM_NAME} MATCHES "Android")
+  set (LINUX true)
 elseif (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
   set (DARWIN true)
   set (_CXX11_FLAGS "${_CXX11_FLAGS} -stdlib=libc++")
