diff -u -r ../timew-1.0.0/src/libshared/src/Pig.cpp ./src/libshared/src/Pig.cpp
--- ../timew-1.0.0/src/libshared/src/Pig.cpp	2016-08-17 17:22:34.000000000 -0400
+++ ./src/libshared/src/Pig.cpp	2016-09-05 17:58:52.282716956 -0400
@@ -251,7 +251,7 @@
 
   if (_cursor > save)
   {
-    result = std::strtoimax (_text.substr (save, _cursor - save).c_str (), NULL, 10);
+    result = strtoimax (_text.substr (save, _cursor - save).c_str (), NULL, 10);
     return true;
   }
 
