--- a/usr/include/c++/v1/module.modulemap	2024-11-08 20:03:22
+++ b/usr/include/c++/v1/module.modulemap	2024-11-08 21:27:52
@@ -1537,6 +1537,7 @@
 module std_private_memory_uninitialized_algorithms        [system] {
   header "__memory/uninitialized_algorithms.h"
   export std_private_algorithm_copy
+  export std_private_memory_voidify
 }
 module std_private_memory_unique_ptr                      [system] {
   header "__memory/unique_ptr.h"
