--- orc-0.4.32/orc/orcprogram-neon.c	2020-09-11 09:33:15.000000000 +0000
+++ orc-0.4.32-fix/orc/orcprogram-neon.c	2021-07-04 09:42:01.803161969 +0000
@@ -126,7 +126,7 @@
 void
 orc_neon_init (void)
 {
-#if defined(HAVE_ARM)
+#if defined(HAVE_ARM) && defined(__TARGET_ARCH_ARM)
   if (!(orc_arm_get_cpu_flags () & ORC_TARGET_NEON_NEON)) {
     ORC_INFO("marking neon backend non-executable");
     neon_target.executable = FALSE;

