diff -u -r ../openssl-1.1.1-pre8/apps/ocsp.c ./apps/ocsp.c
--- ../openssl-1.1.1-pre8/apps/ocsp.c	2018-06-20 14:48:09.000000000 +0000
+++ ./apps/ocsp.c	2018-08-17 21:17:37.915758081 +0000
@@ -37,7 +37,7 @@
 # include <openssl/rand.h>
 
 # if defined(OPENSSL_SYS_UNIX) && !defined(OPENSSL_NO_SOCK) \
-     && !defined(OPENSSL_NO_POSIX_IO)
+     && !defined(OPENSSL_NO_POSIX_IO) && !defined(__ANDROID__)
 #  define OCSP_DAEMON
 #  include <sys/types.h>
 #  include <sys/wait.h>
