diff --git a/include/crypto/rand.h b/include/crypto/rand.h
index 4845c84a94..1c67f9d6e4 100644
--- a/include/crypto/rand.h
+++ b/include/crypto/rand.h
@@ -82,7 +82,7 @@
  * sockets will be tried in the order listed in case accessing the device
  * files listed in DEVRANDOM did not return enough randomness.
  */
-#define DEVRANDOM_EGD "/var/run/egd-pool", "/dev/egd-pool", "/etc/egd-pool", "/etc/entropy"
+#define DEVRANDOM_EGD "@TERMUX_PREFIX@/var/run/egd-pool", "/dev/egd-pool", "@TERMUX_PREFIX@/etc/egd-pool", "@TERMUX_PREFIX@/etc/entropy"
 #endif
 
 void ossl_rand_cleanup_int(void);
