--- xfce4-screenshooter-1.9.7/Makefile.am	2019-05-02 15:12:33.000000000 -0500
+++ mod/Makefile.am	2021-12-02 14:52:23.680819349 -0500
@@ -129,12 +129,6 @@
 
 src_xfce4_screenshooter_SOURCES = src/main.c
 
-# Manual file for the application
-if HAVE_HELP2MAN
-xfce4-screenshooter.1: src/xfce4-screenshooter
-	$(HELP2MAN) -o $@ --no-info $<
-endif
-
 # Desktop file for the application
 app_desktopdir = $(datadir)/applications
 app_desktop_in_in_files = src/xfce4-screenshooter.desktop.in.in
