--- xfce4-screenshooter-1.9.7/Makefile.in	2019-11-03 10:57:23.000000000 -0500
+++ mod/Makefile.in	2021-12-02 14:51:47.360815731 -0500
@@ -382,7 +382,6 @@
 GTK_LIBS = @GTK_LIBS@
 GTK_REQUIRED_VERSION = @GTK_REQUIRED_VERSION@
 GTK_VERSION = @GTK_VERSION@
-HELP2MAN = @HELP2MAN@
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
@@ -1748,8 +1747,7 @@
 	$(AM_V_GEN) exo-csource --static --strip-comments --strip-content --name=screenshooter_imgur_dialog_ui $< >$@
 
 # Manual file for the application
-@HAVE_HELP2MAN_TRUE@xfce4-screenshooter.1: src/xfce4-screenshooter
-@HAVE_HELP2MAN_TRUE@	$(HELP2MAN) -o $@ --no-info $<
+
 @INTLTOOL_DESKTOP_RULE@
 
 src/xfce4-screenshooter.desktop.in: src/xfce4-screenshooter.desktop.in.in
