--- a/configure.ac
+++ b/configure.ac
@@ -46,19 +46,6 @@
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GETTEXT_PACKAGE", [The gettext package name])
 AM_GLIB_GNU_GETTEXT
 
-dnl check for GTK 3 Python bindings
-AC_MSG_CHECKING(GTK 3 python bindings)
-gtk_prog="
-import gi
-gi.require_version('Gtk', '3.0')
-from gi.repository import Gtk
-"
-if $PYTHON -c "$gtk_prog"; then
-    AC_MSG_RESULT(yes)
-else
-    AC_MSG_ERROR(Missing Gtk 3 Python bindings, try sudo apt install python3-gi python3-gi-cairo gir1.2-gtk-3.0)
-fi
-
 export PYTHONPATH=$PYGTK_DIR:$PYTHONPATH
 
 saved_PYTHONPATH=$PYTHONPATH
