--- pip-25.0.1/docs/html/conf.py	2025-02-09 22:25:30.000000000 +0530
+++ pip-25.0.1.mod/docs/html/conf.py	2025-02-22 02:45:23.168211229 +0530
@@ -20,12 +20,6 @@
     "sphinx.ext.intersphinx",
     # our extensions
     "pip_sphinxext",
-    # third-party extensions
-    "myst_parser",
-    "sphinx_copybutton",
-    "sphinx_inline_tabs",
-    "sphinxcontrib.towncrier",
-    "sphinx_issues",
 ]
 
 # General information about the project.
