--- a/src/pip/_internal/utils/appdirs.py
+++ b/src/pip/_internal/utils/appdirs.py
@@ -49,4 +49,4 @@ def site_config_dirs(appname: str) -> List[str]:
         return [dirval]
 
     # Unix-y system. Look in /etc as well.
-    return dirval.split(os.pathsep) + ["/etc"]
+    return dirval.split(os.pathsep) + ["@TERMUX_PREFIX@/etc"]
--- a/src/pip/_vendor/platformdirs/unix.py
+++ b/src/pip/_vendor/platformdirs/unix.py
@@ -53,7 +53,7 @@ class _UnixDefaults(PlatformDirsABC):  # noqa: PLR0904
 
     @property
     def _site_config_dirs(self) -> list[str]:
-        return [self._append_app_name_and_version("/etc/xdg")]
+        return [self._append_app_name_and_version("@TERMUX_PREFIX@/etc/xdg")]
 
     @property
     def user_cache_dir(self) -> str:
