diff -uNr rethinkdb-2.4.0/mk/support/pkg/v8.sh rethinkdb-2.4.0.mod/mk/support/pkg/v8.sh
--- rethinkdb-2.4.0/mk/support/pkg/v8.sh	2019-12-19 09:45:41.000000000 +0200
+++ rethinkdb-2.4.0.mod/mk/support/pkg/v8.sh	2020-06-22 11:01:04.940331222 +0300
@@ -135,8 +135,7 @@
     case ${host%%-*} in
         i?86)    arch=ia32 ;;
         x86_64)  arch=x64 ;;
-        arm64)   arch=arm64; arch_gypflags=$arm_gypflags ;;
-        aarch64) arch=aarch64; arch_gypflags=$arm_gypflags ;;
+        aarch64|arm64)   arch=arm64; arch_gypflags=$arm_gypflags ;;
         arm*)    arch=arm; arch_gypflags=$arm_gypflags ;;
         s390x)   arch=s390x ;;
 	ppc64le*|powerpc*) arch=ppc64 ;;
