--- a/mk/support/pkg/v8.sh
+++ b/mk/support/pkg/v8.sh
@@ -155,7 +155,7 @@ pkg_install () {
         if [[ "$arch" = "ppc64" ]]; then
             arch_gypflags="$arch_gypflags -Duse_system_icu=1"
         fi
-        pkg_make $arch.$mode CXX=$CXX LINK=$CXX LINK.target=$CXX GYPFLAGS="-Dwerror= $arch_gypflags" V=1
+        pkg_make -f Makefile.android $arch.$mode CXX=$CXX LINK=$CXX LINK.target=$CXX GYPFLAGS="-Dwerror= $arch_gypflags" V=1
         for lib in `find "$build_dir/out/$arch.$mode" -maxdepth 1 -name \*.a` `find "$build_dir/out/$arch.$mode/obj.target" -name \*.a`; do
             name=`basename $lib`
             cp $lib "$install_dir/lib/${name/.$arch/}"
