From 5f25f643949c855eed24b8816f7e39bc9ecd8f67 Mon Sep 17 00:00:00 2001
From: cnjhb <hikariz@qq.com>
Date: Sat, 24 Jan 2026 20:13:36 +0800
Subject: [PATCH] fix i18n

---
 setup/main2.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup/main2.py b/setup/main2.py
index d9cfad1..c614d42 100644
--- a/setup/main2.py
+++ b/setup/main2.py
@@ -58,7 +58,7 @@ pkgdatadir = os.getenv("IBUS_PKGDATADIR") or "."
 try:
     locale.bindtextdomain(DOMAINNAME, localedir)
     locale.bind_textdomain_codeset(DOMAINNAME, 'UTF-8')
-except AttributeError:
+except (AttributeError,OSError):
     pass
 
 # Python's gettext module doesn't provide all methods in
@@ -66,7 +66,7 @@ except AttributeError:
 try:
     gettext.bindtextdomain(DOMAINNAME, localedir)
     gettext.bind_textdomain_codeset(DOMAINNAME, 'UTF-8')
-except AttributeError:
+except (AttributeError,OSError):
     pass
 
 gettext.install(DOMAINNAME, localedir)
-- 
2.52.0

