diff -uNr libandroid-support-24/include/iconv.h libandroid-support-24.mod/include/iconv.h
--- libandroid-support-24/include/iconv.h	1970-01-01 03:00:00.000000000 +0300
+++ libandroid-support-24.mod/include/iconv.h	2019-04-28 22:49:00.107252621 +0300
@@ -0,0 +1,12 @@
+#ifndef _ICONV_H
+#define _ICONV_H
+
+#include <stddef.h>
+
+typedef void* iconv_t;
+
+iconv_t iconv_open(const char*, const char*);
+size_t  iconv(iconv_t, char**, size_t*, char**, size_t*);
+int     iconv_close(iconv_t);
+
+#endif
