diff -u -r /home/fornwall/lib/android-ndk/sources/android/support/include/wchar.h ./include/wchar.h
--- /home/fornwall/lib/android-ndk/sources/android/support/include/wchar.h	2014-08-03 23:06:22.000000000 -0400
+++ ./include/wchar.h	2015-07-13 03:18:31.440649551 -0400
@@ -119,6 +119,7 @@
 wchar_t *wcstok (wchar_t *__restrict__, const wchar_t *__restrict__, wchar_t **__restrict__);
 
 size_t wcslen (const wchar_t *);
+size_t wcslcpy(wchar_t *dst, const wchar_t *src, size_t siz);
 
 wchar_t *wcsstr (const wchar_t *__restrict__, const wchar_t *__restrict__);
 wchar_t *wcswcs (const wchar_t *, const wchar_t *);
