Include <xlocale.h> from langinfo.h so that locale_t gets defined.

diff -u -r /home/fornwall/lib/android-ndk/sources/android/support/include/langinfo.h ./include/langinfo.h
--- /home/fornwall/lib/android-ndk/sources/android/support/include/langinfo.h	2014-07-29 04:23:14.000000000 -0400
+++ ./include/langinfo.h	2014-11-27 12:46:17.606220352 -0500
@@ -31,6 +31,7 @@
 // __LP64__
 
 #include <nl_types.h>
+#include <xlocale.h>
 
 #define _NL_ITEM(category,index)  (((category) << 10) | (index))
 
