diff -u -r ../lzip-1.18-rc2/configure ./configure
--- ../lzip-1.18-rc2/configure	2016-03-07 12:24:54.000000000 -0500
+++ ./configure	2016-04-13 20:06:04.652007406 -0400
@@ -14,23 +14,12 @@
 LC_ALL=C
 export LC_ALL
 srcdir=
-prefix=/usr/local
+prefix=@TERMUX_PREFIX@
 exec_prefix='$(prefix)'
 bindir='$(exec_prefix)/bin'
 datarootdir='$(prefix)/share'
 infodir='$(datarootdir)/info'
 mandir='$(datarootdir)/man'
-CXX=g++
-CPPFLAGS=
-CXXFLAGS='-Wall -W -O2'
-LDFLAGS=
-
-# checking whether we are using GNU C++.
-if /bin/sh -c "${CXX} --version" > /dev/null 2>&1 ; then true
-else
-	CXX=c++
-	CXXFLAGS='-W -O2'
-fi
 
 # Loop over all args
 args=
