diff -uNr lzip-1.22/configure lzip-1.22.mod/configure
--- lzip-1.22/configure	2021-01-01 13:06:33.000000000 +0000
+++ lzip-1.22.mod/configure	2021-02-05 22:12:08.991335490 +0000
@@ -23,10 +23,10 @@
 build=no
 check=no
 installdir=
-CXX=g++
-CPPFLAGS=
-CXXFLAGS='-Wall -W -O2'
-LDFLAGS=
+CXX?=g++
+CPPFLAGS?=
+CXXFLAGS?='-Wall -W -O2'
+LDFLAGS?=
 
 # checking whether we are using GNU C++.
 /bin/sh -c "${CXX} --version" > /dev/null 2>&1 || { CXX=c++ ; CXXFLAGS=-O2 ; }
