diff -u -r ../stag-1.0.0/stag.c ./stag.c
--- ../stag-1.0.0/stag.c	2014-10-22 05:12:36.000000000 +0000
+++ ./stag.c	2024-05-15 09:59:45.934188475 +0000
@@ -47,7 +47,7 @@
 "\n";
 
 // Display usage info
-void usage(){
+void usage(void){
   printf(stag_usage_string);
   exit(0);
 }
@@ -87,7 +87,7 @@
   
 
   // Read options
-  char opt;
+  signed char opt;
   int option_index = 0;
   struct option long_options[] =
   {
