diff -u -r ../libvpx-1.6.1/vpx_ports/arm_cpudetect.c ./vpx_ports/arm_cpudetect.c
--- ../libvpx-1.6.1/vpx_ports/arm_cpudetect.c	2017-01-12 21:27:27.000000000 +0100
+++ ./vpx_ports/arm_cpudetect.c	2017-01-29 23:55:05.399283897 +0100
@@ -92,20 +92,17 @@
 }
 
 #elif defined(__ANDROID__) /* end _MSC_VER */
-#include <cpu-features.h>
 
 int arm_cpu_caps(void) {
   int flags;
   int mask;
-  uint64_t features;
   if (!arm_cpu_env_flags(&flags)) {
     return flags;
   }
   mask = arm_cpu_env_mask();
-  features = android_getCpuFeatures();
 
 #if HAVE_NEON || HAVE_NEON_ASM
-  if (features & ANDROID_CPU_ARM_FEATURE_NEON) flags |= HAS_NEON;
+  flags |= HAS_NEON;
 #endif /* HAVE_NEON || HAVE_NEON_ASM */
   return flags & mask;
 }
