diff -u -r ../libvpx-1.8.0/build/make/configure.sh ./build/make/configure.sh
--- ../libvpx-1.8.0/build/make/configure.sh	2019-02-04 17:02:33.000000000 +0000
+++ ./build/make/configure.sh	2019-02-09 21:10:52.112075640 +0000
@@ -1017,7 +1017,6 @@
               enable_feature thumb
               ;;
             *)
-              check_add_asflags --defsym ARCHITECTURE=${arch_int}
               ;;
           esac
 
@@ -1147,7 +1146,6 @@
           else
             echo "Assuming standalone build with NDK toolchain."
             echo "See build/make/Android.mk for details."
-            check_add_ldflags -static
             soft_enable unit_tests
           fi
           ;;
