diff -uNr libvpx-1.8.1/build/make/configure.sh libvpx-1.8.1.mod/build/make/configure.sh
--- libvpx-1.8.1/build/make/configure.sh	2019-07-16 00:55:33.000000000 +0300
+++ libvpx-1.8.1.mod/build/make/configure.sh	2019-07-21 18:30:38.097784114 +0300
@@ -1100,7 +1100,6 @@
         android*)
           echo "Assuming standalone build with NDK toolchain."
           echo "See build/make/Android.mk for details."
-          check_add_ldflags -static
           soft_enable unit_tests
           ;;
 
