diff -u -r ../libvpx-1.6.0/configure ./configure
--- ../libvpx-1.6.0/configure	2016-07-20 21:15:41.000000000 -0400
+++ ./configure	2016-08-10 19:36:59.873847660 -0400
@@ -509,7 +509,7 @@
             # Supported platforms
             ;;
         *)
-            if enabled gnu; then
+            if enabled gnu || enabled android; then
                 echo "--enable-shared is only supported on ELF; assuming this is OK"
             else
                 die "--enable-shared only supported on ELF, OS/2, and Darwin for now"
