--- a/share/cmake-3.28/Modules/Platform/Android/Determine-Compiler.cmake
+++ b/share/cmake-3.28/Modules/Platform/Android/Determine-Compiler.cmake
@@ -71,6 +71,10 @@
   set(_ANDROID_TOOL_CXX_TOOLCHAIN_SUFFIX "")
 endif()
 
+if(CMAKE_TERMUX_TOOL_C_COMPILER)
+  set(_ANDROID_TOOL_C_COMPILER ${CMAKE_TERMUX_TOOL_C_COMPILER})
+endif()
+
 unset(_ANDROID_HOST_EXT)
 
 macro(__android_determine_compiler lang)
