diff -u -r ../tt/meson-1.3.2/mesonbuild/scripts/python_info.py ./mesonbuild/scripts/python_info.py
--- ../tt/meson-1.3.2/mesonbuild/scripts/python_info.py	2023-10-26 16:04:24.000000000 +0000
+++ ./mesonbuild/scripts/python_info.py	2024-05-24 06:23:34.550181626 +0000
@@ -47,7 +47,7 @@
         scheme = sysconfig._get_default_scheme()
 
     if sys.version_info >= (3, 10, 3):
-        if 'deb_system' in sysconfig.get_scheme_names():
+        if False:
             scheme = 'deb_system'
     else:
         import distutils.command.install
