--- a/mesonbuild/modules/python.py
+++ b/mesonbuild/modules/python.py
@@ -305,7 +305,7 @@
 # See https://github.com/mesonbuild/meson/issues/8739.
 # XXX: We should be using sysconfig, but Debian only patches distutils.
 
-if 'deb_system' in distutils.command.install.INSTALL_SCHEMES:
+if False:
     paths = get_distutils_paths(scheme='deb_system')
     install_paths = get_distutils_paths(scheme='deb_system', prefix='')
 else:
