--- a/mesonbuild/build.py
+++ b/mesonbuild/build.py
@@ -2181,7 +2181,7 @@
             else:
                 # libfoo.dylib
                 self.filename_tpl = '{0.prefix}{0.name}.{0.suffix}'
-        elif self.environment.machines[self.for_machine].is_android():
+        elif self.environment.machines[self.for_machine].is_android() and not os.environ.get('TERMUX_MESON_ENABLE_SOVERSION'):
             prefix = 'lib'
             suffix = 'so'
             # Android doesn't support shared_library versioning
@@ -2250,7 +2250,7 @@
     def process_kwargs(self, kwargs):
         super().process_kwargs(kwargs)
 
-        if not self.environment.machines[self.for_machine].is_android():
+        if not self.environment.machines[self.for_machine].is_android() or os.environ.get('TERMUX_MESON_ENABLE_SOVERSION'):
             # Shared library version
             if 'version' in kwargs:
                 self.ltversion = kwargs['version']
