https://github.com/benfogle/crossenv/issues/103

--- a/crossenv/__init__.py
+++ b/crossenv/__init__.py
@@ -245,7 +245,13 @@
         else:
             self.host_project_base = os.path.dirname(host)
 
-        if sysconfig._is_python_source_dir(self.host_project_base):
+        def _is_python_source_dir(d):
+            for fn in ("Setup", "Setup.local"):
+                if os.path.isfile(os.path.join(d, "Modules", fn)):
+                    return True
+            return False
+
+        if _is_python_source_dir(self.host_project_base):
             self.host_makefile = os.path.join(self.host_project_base, 'Makefile')
             pybuilddir = os.path.join(self.host_project_base, 'pybuilddir.txt')
             try:
