diff -uNr blink-8b17fedd4d089fa2166267acc8487ed2d1f91fe5/blink/logcpu.c blink-8b17fedd4d089fa2166267acc8487ed2d1f91fe5.mod/blink/logcpu.c
--- blink-8b17fedd4d089fa2166267acc8487ed2d1f91fe5/blink/logcpu.c	2023-04-20 08:56:39.000000000 +0300
+++ blink-8b17fedd4d089fa2166267acc8487ed2d1f91fe5.mod/blink/logcpu.c	2023-04-21 22:45:38.202964172 +0300
@@ -26,7 +26,7 @@
 // use cosmopolitan/tool/build/fastdiff.c
 void LogCpu(struct Machine *m) {
   static FILE *f;
-  if (!f) f = fopen("/tmp/cpu.log", "w");
+  if (!f) f = fopen("@TERMUX_PREFIX@/tmp/cpu.log", "w");
   fprintf(f,
           "\n"
           "IP %" PRIx64 "\n"
diff -uNr blink-8b17fedd4d089fa2166267acc8487ed2d1f91fe5/blink/map.c blink-8b17fedd4d089fa2166267acc8487ed2d1f91fe5.mod/blink/map.c
--- blink-8b17fedd4d089fa2166267acc8487ed2d1f91fe5/blink/map.c	2023-04-20 08:56:39.000000000 +0300
+++ blink-8b17fedd4d089fa2166267acc8487ed2d1f91fe5.mod/blink/map.c	2023-04-21 22:44:52.589560916 +0300
@@ -64,7 +64,7 @@
   _Static_assert(!(MAP_ANONYMOUS_ & MAP_SHARED), "");
   _Static_assert(!(MAP_ANONYMOUS_ & MAP_PRIVATE), "");
   int tfd;
-  char path[] = "/tmp/blink.dat.XXXXXX";
+  char path[] = "@TERMUX_PREFIX@/tmp/blink.dat.XXXXXX";
   if (~flags & MAP_ANONYMOUS_) {
     res = VfsMmap(addr, length, prot, flags, fd, offset);
   } else if ((tfd = mkstemp(path)) != -1) {
diff -uNr blink-8b17fedd4d089fa2166267acc8487ed2d1f91fe5/blink/path.c blink-8b17fedd4d089fa2166267acc8487ed2d1f91fe5.mod/blink/path.c
--- blink-8b17fedd4d089fa2166267acc8487ed2d1f91fe5/blink/path.c	2023-04-20 08:56:39.000000000 +0300
+++ blink-8b17fedd4d089fa2166267acc8487ed2d1f91fe5.mod/blink/path.c	2023-04-21 22:45:14.310229138 +0300
@@ -154,7 +154,7 @@
   m->system->dis = &g_dis;
   LoadDebugSymbols(m->system);
   DisLoadElf(&g_dis, &m->system->elf);
-  g_cod = VfsOpen(AT_FDCWD_LINUX, "/tmp/blink.s",
+  g_cod = VfsOpen(AT_FDCWD_LINUX, "@TERMUX_PREFIX@/tmp/blink.s",
                   O_WRONLY | O_CREAT | O_TRUNC | O_CLOEXEC, 0644);
   g_cod = VfsFcntl(g_cod, F_DUPFD_CLOEXEC, kMinBlinkFd);
 #endif
