__unused is a reserved keyword. Use __qemu_unused instead.

diff -ur qemu-2.9.0-orig/linux-user/syscall_defs.h qemu-2.9.0/linux-user/syscall_defs.h
--- qemu-2.9.0-orig/linux-user/syscall_defs.h   2017-07-31 04:13:25.535179284 +0200
+++ qemu-2.9.0/linux-user/syscall_defs.h        2017-07-31 05:46:49.349822521 +0200
@@ -2048,7 +2048,7 @@
     abi_ulong  target_st_mtime_nsec;
     abi_long  target_st_ctime;
     abi_ulong  target_st_ctime_nsec;
-    unsigned int __unused[2];
+    unsigned int __qemu_unused[2];
 };
 #elif defined(TARGET_OPENRISC) || defined(TARGET_TILEGX) || \
       defined(TARGET_NIOS2)
