From 7ba719a7a53d6025cd82f8b1c86e765285ed802b Mon Sep 17 00:00:00 2001
From: Michael Cho <michael@michaelcho.dev>
Date: Sun, 10 Aug 2025 23:50:21 -0400
Subject: [PATCH] CMake: Fix build with Boost 1.89.0

---
 CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index de4779f..6c7f0d2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,12 +13,13 @@ endif(USE_QT_GUI)
 find_package(Boost
   CONFIG
   REQUIRED
-  system
   filesystem
   regex
   program_options
   date_time
   iostreams
+  OPTIONAL_COMPONENTS
+  system
   )
 find_package(CURL 7.55.0 REQUIRED)
 find_package(Jsoncpp REQUIRED)
