diff -u -r ../radare2-0.10.5/libr/debug/p/debug_native.c ./libr/debug/p/debug_native.c
--- ../radare2-0.10.5/libr/debug/p/debug_native.c	2016-08-16 19:28:34.000000000 -0400
+++ ./libr/debug/p/debug_native.c	2016-09-16 17:37:57.879695115 -0400
@@ -75,7 +75,7 @@
 #   define WIFCONTINUED(s) ((s) == 0xffff)
 #  endif
 # endif
-#if __x86_64__
+#if __x86_64__ && !defined(__ANDROID__)
 #include "native/linux/linux_coredump.h"
 #endif
 #else // OS
