diff -Nuar radare2-5.5.0.orig/libr/anal/p/anal_arm_cs.c radare2-5.5.0.mod/libr/anal/p/anal_arm_cs.c
--- radare2-5.5.0.orig/libr/anal/p/anal_arm_cs.c	2021-11-15 01:18:10.653378506 -0300
+++ radare2-5.5.0.mod/libr/anal/p/anal_arm_cs.c	2021-11-15 01:18:58.165378472 -0300
@@ -1361,9 +1361,6 @@
 	case ARM64_INS_AND:
 		OPCALL("&");
 		break;
-	case ARM64_INS_NAND:
-		OPCALL_NEG("&");
-		break;
 	case ARM64_INS_ORR:
 		OPCALL("|");
 		break;
