diff -uNr radare2-2.0.0/libr/core/cconfig.c radare2-2.0.0.mod/libr/core/cconfig.c
--- radare2-2.0.0/libr/core/cconfig.c	2017-10-09 18:05:53.000000000 +0300
+++ radare2-2.0.0.mod/libr/core/cconfig.c	2017-10-11 16:22:54.521006895 +0300
@@ -2383,10 +2383,8 @@
 	SETPREF ("dir.plugins", R2_LIBDIR"/radare2/"R2_VERSION"/", "Path to plugin files to be loaded at startup");
 #endif
 	SETCB ("dir.source", "", &cb_dirsrc, "Path to find source files");
-	SETPREF ("dir.types", "/usr/include", "Default path to look for cparse type files");
-#if __ANDROID__
-	SETPREF ("dir.projects", "/data/data/org.radare.radare2installer/radare2/projects", "Default path for projects");
-#elif __WINDOWS__
+	SETPREF ("dir.types", "@TERMUX_PREFIX@/include", "Default path to look for cparse type files");
+#if __WINDOWS__
 	SETPREF ("dir.projects", "~\\"R2_HOMEDIR"\\projects", "Default path for projects");
 #else
 	SETPREF ("dir.projects", "~/"R2_HOMEDIR"/projects", "Default path for projects");
@@ -2528,6 +2526,9 @@
 #if __WINDOWS__
 	r_config_set (cfg, "http.browser", "start");
 #else
+#ifdef __ANDROID__
+	r_config_set (cfg, "http.browser", "xdg-open");
+#else
 	if (r_file_exists ("/usr/bin/openURL")) { // iOS ericautils
 		r_config_set (cfg, "http.browser", "/usr/bin/openURL");
 	} else if (r_file_exists ("/system/bin/toolbox")) {
@@ -2540,14 +2541,13 @@
 	} else {
 		r_config_set (cfg, "http.browser", "firefox");
 	}
+#endif
 	r_config_desc (cfg, "http.browser", "Command to open HTTP URLs");
 #endif
 	SETI ("http.maxsize", 0, "Maximum file size for upload");
 	SETPREF ("http.bind", "localhost", "Server address");
 	SETPREF ("http.homeroot", "~/.config/radare2/www", "http home root directory");
-#if __ANDROID__
-	SETPREF ("http.root", "/data/data/org.radare.radare2installer/www", "http root directory");
-#elif __WINDOWS__
+#if __WINDOWS__
 	SETPREF ("http.root", "www", "http root directory");
 #else
 	SETPREF ("http.root", R2_WWWROOT, "http root directory");
diff -uNr radare2-2.0.0/libr/debug/p/debug_rap.c radare2-2.0.0.mod/libr/debug/p/debug_rap.c
--- radare2-2.0.0/libr/debug/p/debug_rap.c	2017-10-09 18:05:53.000000000 +0300
+++ radare2-2.0.0.mod/libr/debug/p/debug_rap.c	2017-10-11 16:24:45.152199241 +0300
@@ -49,7 +49,7 @@
 }
 
 static char *r_debug_rap_reg_profile(RDebug *dbg) {
-	char *out, *tf = r_file_temp ("/tmp/rap.XXXXXX");
+	char *out, *tf = r_file_temp ("@TERMUX_PREFIX@/tmp/rap.XXXXXX");
 	int fd = r_cons_pipe_open (tf, 1, 0);
 	r_io_system (dbg->iob.io, "drp");
 	r_cons_pipe_close (fd);
diff -uNr radare2-2.0.0/libr/include/r_magic.h radare2-2.0.0.mod/libr/include/r_magic.h
--- radare2-2.0.0/libr/include/r_magic.h	2017-10-09 18:05:53.000000000 +0300
+++ radare2-2.0.0.mod/libr/include/r_magic.h	2017-10-11 16:23:00.721073712 +0300
@@ -12,7 +12,7 @@
 R_LIB_VERSION_HEADER(r_magic);
 
 #ifndef MAGICFILE
-#define MAGICFILE "/etc/magic"
+#define MAGICFILE "@TERMUX_PREFIX@/etc/magic"
 #endif
 
 #define R_MAGIC_PATH R2_PREFIX "/share/radare2/" R2_VERSION "/magic"
