diff -ruN radare2-5.4.2/libr/include/r_util/r_file.h radare2-5.4.2.mod/libr/include/r_util/r_file.h
--- radare2-5.4.2/libr/include/r_util/r_file.h	2021-09-21 19:21:18.729361958 -0300
+++ radare2-5.4.2.mod/libr/include/r_util/r_file.h	2021-09-21 19:21:55.053361932 -0300
@@ -8,7 +8,7 @@
 #include <r_util/r_mem.h>
 
 #ifndef SHELL_PATH
-#define SHELL_PATH "/bin/sh"
+#define SHELL_PATH "@TERMUX_PREFIX@/bin/sh"
 #endif
 
 #ifndef TERMUX_PREFIX
