diff -u -r ../radare2-3.6.0/libr/hash/Makefile ./libr/hash/Makefile
--- ../radare2-3.6.0/libr/hash/Makefile	2019-06-24 12:09:20.000000000 +0000
+++ ./libr/hash/Makefile	2019-06-27 22:15:18.439380740 +0000
@@ -12,7 +12,7 @@
 
 DEPS=r_util
 OBJS=state.o hash.o hamdist.o crca.o fletcher.o
-OBJS+=entropy.o calc.o adler32.o luhn.o
+OBJS+=entropy.o calc.o adler32.o luhn.o mem.o
 
 ifeq ($(HAVE_LIB_SSL),1)
 CFLAGS+=${SSL_CFLAGS}
