diff -uNr radare2-2.0.0/libr/include/r_magic.h radare2-2.0.0.mod/libr/include/r_magic.h
--- radare2-2.0.0/libr/include/r_magic.h	2017-10-09 18:05:53.000000000 +0300
+++ radare2-2.0.0.mod/libr/include/r_magic.h	2017-10-11 16:23:00.721073712 +0300
@@ -12,7 +12,7 @@
 R_LIB_VERSION_HEADER(r_magic);
 
 #ifndef MAGICFILE
-#define MAGICFILE "/etc/magic"
+#define MAGICFILE "@TERMUX_PREFIX@/etc/magic"
 #endif
 
 #define R_MAGIC_PATH R2_PREFIX "/share/radare2/" R2_VERSION "/magic"
