diff -u -r ../radare2-3.0.0/libr/include/r_io.h ./libr/include/r_io.h
--- ../radare2-3.0.0/libr/include/r_io.h	2018-10-15 22:29:51.000000000 +0000
+++ ./libr/include/r_io.h	2018-10-18 04:14:28.601218194 +0000
@@ -26,7 +26,7 @@
 #include <sys/ptrace.h>
 #endif
 
-#if defined(__APPLE__) || __sun || __FreeBSD__ || __OpenBSD__
+#if defined(__APPLE__) || __sun || __FreeBSD__ || __OpenBSD__ || defined(__ANDROID__)
 typedef int r_ptrace_request_t;
 typedef int r_ptrace_data_t;
 #else
