diff -u -r ../radare2-1.1.0/libr/core/cconfig.c ./libr/core/cconfig.c
--- ../radare2-1.1.0/libr/core/cconfig.c	2016-12-19 21:13:05.000000000 -0500
+++ ./libr/core/cconfig.c	2016-12-20 20:28:14.811403257 -0500
@@ -1900,9 +1900,7 @@
 #endif
 	SETCB("dir.source", "", &cb_dirsrc, "Path to find source files");
 	SETPREF("dir.types", "/usr/include", "Default path to look for cparse type files");
-#if __ANDROID__
-	SETPREF("dir.projects", "/data/data/org.radare.radare2installer/radare2/projects", "Default path for projects");
-#elif __WINDOWS__
+#if __WINDOWS__
 	SETPREF("dir.projects", "~\\"R2_HOMEDIR"\\projects", "Default path for projects");
 #else
 	SETPREF("dir.projects", "~/"R2_HOMEDIR"/projects", "Default path for projects");
@@ -2036,9 +2034,7 @@
 	SETI("http.maxsize", 0, "Maximum file size for upload");
 	SETPREF("http.bind", "localhost", "Server address");
 	SETPREF("http.homeroot", "~/.config/radare2/www", "http home root directory");
-#if __ANDROID__
-	SETPREF("http.root", "/data/data/org.radare.radare2installer/www", "http root directory");
-#elif __WINDOWS__
+#if __WINDOWS__
 	SETPREF("http.root", "www", "http root directory");
 #else
 	SETPREF("http.root", R2_WWWROOT, "http root directory");
