diff -uNr radare2-3.2.1/libr/util/diff.c radare2-3.2.1.mod/libr/util/diff.c
--- radare2-3.2.1/libr/util/diff.c	2019-01-08 18:14:43.000000000 +0200
+++ radare2-3.2.1.mod/libr/util/diff.c	2019-03-02 03:17:18.566391827 +0200
@@ -145,7 +145,7 @@
 	char* err = NULL;
 	char* out = NULL;
 	int out_len;
-	(void)r_sys_cmd_str_full ("/usr/bin/diff -u .a .b", NULL, &out, &out_len, &err);
+	(void)r_sys_cmd_str_full ("@TERMUX_PREFIX@/bin/diff -u .a .b", NULL, &out, &out_len, &err);
 	r_file_rm (".a");
 	r_file_rm (".b");
 	free (err);
