diff -u -r ../radare2-2.1.0/libr/util/file.c ./libr/util/file.c
--- ../radare2-2.1.0/libr/util/file.c	2017-11-27 13:31:04.000000000 +0100
+++ ./libr/util/file.c	2017-11-27 20:11:09.415987429 +0100
@@ -986,7 +986,7 @@
 	}
 	if (!path) {
 #if __ANDROID__
-		path = strdup ("/data/data/org.radare.radare2installer/radare2/tmp");
+		path = strdup ("@TERMUX_PREFIX@/tmp");
 #else
 		path = strdup ("/tmp");
 #endif
