diff -u -r ../radare2-0.10.5/libr/debug/p/native/linux/linux_debug.c ./libr/debug/p/native/linux/linux_debug.c
--- ../radare2-0.10.5/libr/debug/p/native/linux/linux_debug.c	2016-08-16 19:28:34.000000000 -0400
+++ ./libr/debug/p/native/linux/linux_debug.c	2016-09-16 17:53:45.280118204 -0400
@@ -426,7 +426,7 @@
 			if (ret1 != 0) return false;
 			if (sizeof(fpregs) < size) size = sizeof(fpregs);
 			memcpy (buf, &fpregs, size);
-			return sizeof(fpregs)
+			return sizeof(fpregs);
 #endif // !__ANDROID__
 #elif __i386__
 #if !__ANDROID__
