diff -u -r ../radare2-0.10.5/mk/termux-host.mk ./mk/termux-host.mk
--- ../radare2-0.10.5/mk/termux-host.mk	2016-08-16 19:28:34.000000000 -0400
+++ ./mk/termux-host.mk	2016-09-16 16:58:13.067871972 -0400
@@ -2,10 +2,7 @@
 #CC=gcc -fPIC -fPIE
 #RANLIB=ndk-ranlib
 USERCC=gcc -fPIC -fPIE
-ARCH=arm
 
-RANLIB=aarch64-linux-android-ranlib
-AR=aarch64-linux-android-ar
 CC_AR=${AR} -r ${LIBAR}
 
 ONELIB=0
