diff -u -r ../radare2-0.10.5/libr/core/config.c ./libr/core/config.c
--- ../radare2-0.10.5/libr/core/config.c	2016-08-16 19:28:34.000000000 -0400
+++ ./libr/core/config.c	2016-09-16 08:20:42.072720023 -0400
@@ -1656,9 +1656,7 @@
 #endif
 	SETCB("dir.source", "", &cb_dirsrc, "Path to find source files");
 	SETPREF("dir.types", "/usr/include", "Default path to look for cparse type files");
-#if __ANDROID__
-	SETPREF("dir.projects", "/data/data/org.radare2.installer/radare2/projects", "Default path for projects");
-#elif __WINDOWS__
+#if __WINDOWS__
 	SETPREF("dir.projects", "~\\"R2_HOMEDIR"\\projects", "Default path for projects");
 #else
 	SETPREF("dir.projects", "~/"R2_HOMEDIR"/projects", "Default path for projects");
@@ -1778,9 +1776,7 @@
 	SETI("http.maxsize", 0, "Maximum file size for upload");
 	SETPREF("http.bind", "localhost", "Server address");
 	SETPREF("http.homeroot", "~/.config/radare2/www", "http home root directory");
-#if __ANDROID__
-	SETPREF("http.root", "/data/data/org.radare2.installer/www", "http root directory");
-#elif __WINDOWS__
+#if __WINDOWS__
 	SETPREF("http.root", "www", "http root directory");
 #else
 	SETPREF("http.root", R2_WWWROOT, "http root directory");
diff -u -r ../radare2-0.10.5/libr/util/file.c ./libr/util/file.c
--- ../radare2-0.10.5/libr/util/file.c	2016-08-16 19:28:34.000000000 -0400
+++ ./libr/util/file.c	2016-09-16 08:18:35.302578773 -0400
@@ -770,7 +770,7 @@
 	char *path = r_sys_getenv ("TEMP");
 	if (!path) path = strdup ("C:\\WINDOWS\\Temp\\");
 #elif __ANDROID__
-	char *path = strdup ("/data/data/org.radare2.installer/radare2/tmp");
+	char *path = strdup ("@TERMUX_PREFIX@/tmp");
 #else
 	char *path = r_sys_getenv ("TMPDIR");
 	if (path && !*path) {
