commit e3580f273a8dd83c71bb4ad10b393585dde02c46
Author: Kornel <kornel@geekhood.net>
Date:   Wed Sep 15 09:40:59 2021 +0100

    Update for latest rav1e

diff --git a/ravif/src/av1encoder.rs b/ravif/src/av1encoder.rs
index edcc1c1..2afcb85 100644
--- a/ravif/src/av1encoder.rs
+++ b/ravif/src/av1encoder.rs
@@ -285,7 +285,6 @@ impl SpeedTweaks {
         let mut speed_settings = SpeedSettings::from_preset(self.speed_preset.into());
 
         speed_settings.multiref = false;
-        speed_settings.fast_scene_detection = true;
         speed_settings.no_scene_detection = true;
         speed_settings.include_near_mvs = false;
 
