--- ./cmake/FindBacktrace.cmake.orig	2022-09-02 08:47:14.092006586 +0200
+++ ./cmake/FindBacktrace.cmake	2022-09-02 08:47:32.510108150 +0200
@@ -69,15 +69,9 @@
   endif()
   set(Backtrace_LIBRARY "" CACHE FILEPATH "Library providing backtrace(3), empty for default set of libraries")
 else()
-  # Check for external library, for non-glibc systems
-  if(Backtrace_INCLUDE_DIR)
-    # OpenBSD has libbacktrace renamed to libexecinfo
-    find_library(Backtrace_LIBRARY "execinfo")
-  elseif()     # respect user wishes
-    set(_Backtrace_HEADER_TRY "backtrace.h")
-    find_path(Backtrace_INCLUDE_DIR ${_Backtrace_HEADER_TRY})
-    find_library(Backtrace_LIBRARY "backtrace")
-  endif()
+  set(_Backtrace_HEADER_TRY "backtrace.h")
+  find_path(Backtrace_INCLUDE_DIR ${_Backtrace_HEADER_TRY})
+  find_library(Backtrace_LIBRARY "backtrace")
 
   # Prepend list with library path as it's more common practice
   set(_Backtrace_STD_ARGS Backtrace_LIBRARY ${_Backtrace_STD_ARGS})
