diff --git a/modules/https_proxy/https_proxy.go b/modules/https_proxy/https_proxy.go
index 06bc1e0..127bb39 100644
--- a/modules/https_proxy/https_proxy.go
+++ b/modules/https_proxy/https_proxy.go
@@ -47,12 +47,12 @@ func NewHttpsProxy(s *session.Session) *HttpsProxy {
 		"URL, path or javascript code to inject into every HTML page."))
 
 	mod.AddParam(session.NewStringParameter("https.proxy.certificate",
-		"~/.bettercap-ca.cert.pem",
+		"@TERMUX_HOME@/.bettercap-ca.cert.pem",
 		"",
 		"HTTPS proxy certification authority TLS certificate file."))
 
 	mod.AddParam(session.NewStringParameter("https.proxy.key",
-		"~/.bettercap-ca.key.pem",
+		"@TERMUX_HOME@/.bettercap-ca.key.pem",
 		"",
 		"HTTPS proxy certification authority TLS key file."))
 
diff --git a/modules/https_server/https_server.go b/modules/https_server/https_server.go
index 0d9ff80..3e49996 100644
--- a/modules/https_server/https_server.go
+++ b/modules/https_server/https_server.go
@@ -42,12 +42,12 @@ func NewHttpsServer(s *session.Session) *HttpsServer {
 		"Port to bind the http server to."))
 
 	mod.AddParam(session.NewStringParameter("https.server.certificate",
-		"~/.bettercap-httpd.cert.pem",
+		"@TERMUX_HOME@/.bettercap-httpd.cert.pem",
 		"",
 		"TLS certificate file (will be auto generated if filled but not existing)."))
 
 	mod.AddParam(session.NewStringParameter("https.server.key",
-		"~/.bettercap-httpd.key.pem",
+		"@TERMUX_HOME@/.bettercap-httpd.key.pem",
 		"",
 		"TLS key file (will be auto generated if filled but not existing)."))
 
