--- a/src/plugins/filters/mpxj/import/CMakeLists.txt
+++ b/src/plugins/filters/mpxj/import/CMakeLists.txt
@@ -14,5 +14,4 @@ install(TARGETS planmpxjimport DESTINATION ${PLAN_FILTER_INSTALL_DIR})
 
 if (SharedMimeInfo_FOUND)
     install(FILES plan_mpxj_mimetype.xml DESTINATION ${KDE_INSTALL_MIMEDIR})
-    update_xdg_mimetypes(${KDE_INSTALL_MIMEDIR})
 endif ()
--- a/src/portfolio/CMakeLists.txt
+++ b/src/portfolio/CMakeLists.txt
@@ -124,7 +124,6 @@ install( FILES
 
 if (SharedMimeInfo_FOUND)
     install(FILES calligraplanportfolio_mimetype.xml DESTINATION ${KDE_INSTALL_MIMEDIR})
-    update_xdg_mimetypes(${KDE_INSTALL_MIMEDIR})
 endif ()
 
 install( PROGRAMS  org.kde.calligraplanportfolio.desktop  DESTINATION ${KDE_INSTALL_APPDIR})
