--- ../../build/megatools/cache/megatools-1.9.97/configure.ac	2016-02-02 18:26:06.000000000 +0100
+++ ./configure.ac	2016-10-12 23:00:37.626524369 +0200
@@ -42,11 +42,7 @@
 AC_SUBST(GLIB_MAKEFILE)
 
 # check gobject-introspection
-AS_IF([test "x$enable_shared" = "xyes"], [
-  GOBJECT_INTROSPECTION_CHECK([$GOBJECT_INTROSPECTION_VERSION])
-], [
-  AM_CONDITIONAL([HAVE_INTROSPECTION], [false])
-])
+AM_CONDITIONAL([HAVE_INTROSPECTION], [true])
 
 # check openssl
 PKG_CHECK_MODULES(OPENSSL, [$OPENSSL_REQUIRES])
