diff -uNr megatools-1.9.98/lib/mega.c megatools-1.9.98.mod/lib/mega.c
--- megatools-1.9.98/lib/mega.c	2016-11-03 15:47:43.000000000 +0200
+++ megatools-1.9.98.mod/lib/mega.c	2017-09-14 14:39:19.422860139 +0300
@@ -2859,7 +2859,7 @@
 
   if (has_ffmpegthumbnailer && g_regex_match_simple("\\.(mpg|mpeg|avi|mkv|flv|rm|mp4|wmv|asf|ram|mov)$", local_path, G_REGEX_CASELESS, 0))
   {
-    gchar buf[50] = "/tmp/megatools.XXXXXX";
+    gchar buf[50] = "@TERMUX_PREFIX@/tmp/megatools.XXXXXX";
     gchar* dir = g_mkdtemp(buf);
     if (dir)
     {
@@ -2895,7 +2895,7 @@
   }
   else if (has_convert && g_regex_match_simple("\\.(jpe?g|png|gif|bmp|tiff|svg|pnm|eps|ico|pdf)$", local_path, G_REGEX_CASELESS, 0))
   {
-    gchar buf[50] = "/tmp/megatools.XXXXXX";
+    gchar buf[50] = "@TERMUX_PREFIX@/tmp/megatools.XXXXXX";
     gchar* dir = g_mkdtemp(buf);
     if (dir)
     {
