--- a/src/netxs/desktopio/system.hpp
+++ b/src/netxs/desktopio/system.hpp
@@ -60,7 +60,9 @@
         #endif
         #include <linux/keyboard.h> // ::keyb_ioctl()
         #include <linux/input.h>    // mouse button codes: BTN_LEFT ...
+        #if !defined(__ANDROID__)
         #include <libinput.h>       // libinput API
+        #endif
         #include <dlfcn.h>          // ::dlopen()
     #endif
 
@@ -4718,7 +4720,7 @@
                 }
             });
         }
-        #if defined(__linux__)
+        #if defined(__linux__) && !defined(__ANDROID__)
         namespace libinput
         {
             struct libinput;
@@ -5254,7 +5256,7 @@
                     #endif
                     return state;
                 };
-                #if defined(__linux__)
+                #if defined(__linux__) && !defined(__ANDROID__)
                 if (dtvt::vtmode & ui::console::mouse) // Trying to get direct mouse access.
                 {
                     if (auto li = os::tty::libinput::initialize())
@@ -5829,7 +5831,7 @@
                                   whlacc = fp2d{},
                                   dev_map = std::unordered_map<arch, si32>{}]() mutable
                 {
-                    #if defined(__linux__)
+                    #if defined(__linux__) && !defined(__ANDROID__)
                     while (true)
                     {
                         if (auto e = os::tty::libinput::next_event())
@@ -5986,7 +5988,7 @@
                                micefd,       m_proc,
                                alarm,        f_proc);
                 }
-                #if defined(__linux__)
+                #if defined(__linux__) && !defined(__ANDROID__)
                 os::tty::libinput::uninitialize();
                 #endif
 
--- a/src/vtm.cpp
+++ b/src/vtm.cpp
@@ -125,7 +125,7 @@
                 break;
             }
         }
-        #if defined(__linux__)
+        #if defined(__linux__) && !defined(__ANDROID__)
         else if (getopt.match("--SetMouseAccess"))
         {
             os::dtvt::initialize();
@@ -169,7 +169,7 @@
                 #if defined(WIN32)
                 "\n    -0, --session0       Use Session 0 to run Desktop Server in background."
                 #endif
-                #if defined(__linux__)
+                #if defined(__linux__) && !defined(__ANDROID__)
                 "\n    --SetMouseAccess     Set mouse device access for all users in Linux. Elevated privileges required."
                 #endif
                 "\n    -q, --quiet          Disable logging."

