--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -14,7 +14,7 @@
     #set (CMAKE_EXE_LINKER_FLAGS "/MANIFEST:NO")
     set (CMAKE_CXX_FLAGS "/DWIN32 /D_WINDOWS /W3 /GR /EHsc /bigobj")
 elseif (NOT CMAKE_HOST_SYSTEM_NAME MATCHES "Android")
-    set (CMAKE_CXX_FLAGS "-pthread")
+    #set (CMAKE_CXX_FLAGS "-pthread")
     if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
         #static linkage
         #set (CMAKE_CXX_FLAGS "-static-libstdc++ -static-libgcc -pthread -s")
