diff -uNr vtm-0.5.15/src/netxs/os/system.hpp vtm-0.5.15.mod/src/netxs/os/system.hpp
--- vtm-0.5.15/src/netxs/os/system.hpp	2021-08-04 21:50:44.000000000 +0000
+++ vtm-0.5.15.mod/src/netxs/os/system.hpp	2021-10-04 12:37:37.069884293 +0000
@@ -81,7 +81,7 @@
 
     #if defined(__linux__)
         #include <sys/vt.h> // ::console_ioctl()
-        #include <sys/kd.h> // ::console_ioctl()
+        #include <linux/kd.h> // ::console_ioctl()
         #include <linux/keyboard.h> // ::keyb_ioctl()
     #endif
 #endif
@@ -2634,4 +2634,4 @@
     };
 }
 
-#endif // NETXS_SYSTEM_HPP
\ No newline at end of file
+#endif // NETXS_SYSTEM_HPP
