diff -uNr vtm-0.8.0/src/netxs/os/system.hpp vtm-0.8.0.mod/src/netxs/os/system.hpp
--- vtm-0.8.0/src/netxs/os/system.hpp	2022-08-03 05:13:37.000000000 +0800
+++ vtm-0.8.0.mod/src/netxs/os/system.hpp	2022-08-07 11:10:39.985818809 +0800
@@ -2361,7 +2361,7 @@
                 #elif defined(__linux__)
 
                     auto cred = ucred{};
-                    #ifndef __ANDROID__
+                    #ifdef __ANDROID__
                         auto size = socklen_t{ sizeof(cred) };
                     #else
                         auto size = unsigned{ sizeof(cred) };
