https://github.com/termux/termux-packages/issues/13594

--- a/src/netxs/console/ansi.hpp
+++ b/src/netxs/console/ansi.hpp
@@ -1966,7 +1966,9 @@
                             log("dtvt: corrupted integer data");
                             return D{};
                         }
-                        auto crop = netxs::letoh(*reinterpret_cast<D const*>(data.data()));
+                        auto data_aligned = D{};
+                        ::memcpy(&data_aligned, data.data(), sizeof(D));
+                        auto crop = netxs::letoh(data_aligned);
                         if constexpr (!PeekOnly)
                         {
                             data.remove_prefix(sizeof(D));
