--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -35,44 +35,26 @@ add_executable(vtm "src/vtm.cpp" ${WIN32_RESOURCES})
 # add_executable(term "src/netxs/apps/term.cpp")
 # add_executable(calc "src/netxs/apps/calc.cpp")
 
-# Set BUILD_SHARED_LIBS to OFF globally for fetched content.
-set(BUILD_SHARED_LIBS OFF CACHE BOOL "Build static libraries" FORCE)
-
 # FreeType dependency
-set(FT_DISABLE_BROTLI   ON CACHE BOOL "Disable brotli"   FORCE)
-set(FT_DISABLE_ZLIB     ON CACHE BOOL "Disable zlib"     FORCE)
-set(FT_DISABLE_BZIP2    ON CACHE BOOL "Disable bzip2"    FORCE)
-set(FT_DISABLE_PNG      ON CACHE BOOL "Disable libpng"   FORCE)
-set(FT_DISABLE_HARFBUZZ ON CACHE BOOL "Disable harfbuzz" FORCE)
-FetchContent_Declare(freetype
-                     GIT_REPOSITORY https://github.com/freetype/freetype.git
-                     GIT_TAG        VER-2-14-1)
-FetchContent_MakeAvailable(freetype)
+find_package(Freetype REQUIRED)
+find_package(Threads REQUIRED)
 
 # HarfBuzz dependency
-set(HB_NO_MT                          ON  CACHE BOOL "Single threaded"        FORCE)
-set(HB_NO_PRAGMA_GCC_DIAGNOSTIC_ERROR ON  CACHE BOOL "Disable pragma warning" FORCE)
-set(HB_HAVE_FREETYPE                  OFF CACHE BOOL "Disable freetype"       FORCE)
-set(HB_HAVE_GLIB                      OFF CACHE BOOL "Disable glib"           FORCE)
-set(HB_HAVE_GOBJECT                   OFF CACHE BOOL "Disable gobject"        FORCE)
-set(HB_HAVE_ICU                       OFF CACHE BOOL "Disable icu"            FORCE)
-set(HB_HAVE_CAIRO                     OFF CACHE BOOL "Disable cairo"          FORCE)
-FetchContent_Declare(harfbuzz
-                     URL      https://github.com/harfbuzz/harfbuzz/archive/refs/tags/12.1.0.tar.gz
-                     URL_HASH SHA256=0238bf7ada6b1fb92984f69f8b9cd66518af83cf24f7db1cfe60c772c42312d3)
-FetchContent_MakeAvailable(harfbuzz)
+find_package(harfbuzz CONFIG REQUIRED)
 
 # Lua dependency
-FetchContent_Declare(lua
-                     URL      https://www.lua.org/ftp/lua-5.4.7.tar.gz
-                     URL_HASH SHA256=9fbf5e28ef86c69858f6d3d34eccc32e911c1a28b4120ff3e84aaa70cfbf1e30)
-FetchContent_MakeAvailable(lua)
-file(GLOB lua_src CONFIGURE_DEPENDS ${lua_SOURCE_DIR}/src/*.c)
-list(REMOVE_ITEM lua_src ${lua_SOURCE_DIR}/src/lua.c ${lua_SOURCE_DIR}/src/luac.c)
-add_library(lua EXCLUDE_FROM_ALL ${lua_src})
-target_include_directories(vtm PRIVATE ${lua_SOURCE_DIR}/src)
+find_package(Lua REQUIRED)
+if(Lua_FOUND AND NOT TARGET Lua::Lua)
+  add_library(Lua::Lua INTERFACE IMPORTED)
+  set_target_properties(
+    Lua::Lua
+    PROPERTIES
+      INTERFACE_INCLUDE_DIRECTORIES "${LUA_INCLUDE_DIR}"
+      INTERFACE_LINK_LIBRARIES "${LUA_LIBRARIES}"
+  )
+endif()
 
-target_link_libraries(vtm PRIVATE lua freetype harfbuzz)
+target_link_libraries(vtm PRIVATE Lua::Lua Freetype::Freetype harfbuzz::harfbuzz)
 # target_link_libraries(term PRIVATE lua freetype harfbuzz)
 # target_link_libraries(calc PRIVATE lua freetype harfbuzz)
 
--- a/src/netxs/desktopio/gui.hpp
+++ b/src/netxs/desktopio/gui.hpp
@@ -6,7 +6,7 @@
 #include <ft2build.h>
 #include FT_FREETYPE_H
 
-#include <hb.hh>
+#include <hb-cplusplus.hh>
 
 #if defined(_WIN32)
 
