diff -u -r ../libwebp-1.0.3/CMakeLists.txt ./CMakeLists.txt
--- ../libwebp-1.0.3/CMakeLists.txt	2019-07-04 19:32:37.000000000 +0000
+++ ./CMakeLists.txt	2019-07-14 17:21:56.050221574 +0000
@@ -75,20 +75,7 @@
 
 # ##############################################################################
 # Android only.
-if(ANDROID)
-  include_directories(${ANDROID_NDK}/sources/android/cpufeatures)
-  add_library(cpufeatures STATIC
-              ${ANDROID_NDK}/sources/android/cpufeatures/cpu-features.c)
-  list(APPEND INSTALLED_LIBRARIES cpufeatures)
-  target_link_libraries(cpufeatures dl)
-  set(WEBP_DEP_LIBRARIES ${WEBP_DEP_LIBRARIES} cpufeatures)
-  set(WEBP_DEP_INCLUDE_DIRS ${WEBP_DEP_INCLUDE_DIRS}
-      ${ANDROID_NDK}/sources/android/cpufeatures)
-  add_definitions(-DHAVE_CPU_FEATURES_H=1)
-  set(HAVE_CPU_FEATURES_H 1)
-else()
-  set(HAVE_CPU_FEATURES_H 0)
-endif()
+set(HAVE_CPU_FEATURES_H 0)
 
 function(configure_pkg_config FILE)
   configure_file("${CMAKE_CURRENT_SOURCE_DIR}/${FILE}.in"
