diff -u -r ../libwebp-1.0.0/cmake/cpu.cmake ./cmake/cpu.cmake
--- ../libwebp-1.0.0/cmake/cpu.cmake	2018-04-21 03:04:55.000000000 +0000
+++ ./cmake/cpu.cmake	2018-08-26 00:25:11.905018914 +0000
@@ -45,16 +45,6 @@
 set(WEBP_SIMD_FILES_TO_INCLUDE)
 set(WEBP_SIMD_FLAGS_TO_INCLUDE)
 
-if(${ANDROID})
-  if(${ANDROID_ABI} STREQUAL "armeabi-v7a")
-    # This is because Android studio uses the configuration
-    # "-march=armv7-a -mfloat-abi=softfp -mfpu=vfpv3-d16"
-    # that does not trigger neon optimizations but should
-    # (as this configuration does not exist anymore).
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mfpu=neon ")
-  endif()
-endif()
-
 list(LENGTH WEBP_SIMD_FLAGS WEBP_SIMD_FLAGS_LENGTH)
 math(EXPR WEBP_SIMD_FLAGS_RANGE "${WEBP_SIMD_FLAGS_LENGTH} - 1")
 
