diff -ru a/C/7zip_gcc_c.mak b/C/7zip_gcc_c.mak
--- a/C/7zip_gcc_c.mak	2022-07-15 10:00:00.000000000 +0200
+++ b/C/7zip_gcc_c.mak	2022-11-10 22:05:33.101451700 +0100
@@ -108,7 +108,7 @@
 
 # LOCAL_LIBS=-lpthread
 # LOCAL_LIBS_DLL=$(LOCAL_LIBS) -ldl
-LIB2 = -lpthread -ldl
+LIB2 = -ldl
 
 DEL_OBJ_EXE = -$(RM) $(PROGPATH) $(PROGPATH_STATIC) $(OBJS)
 
diff -ru a/CPP/7zip/7zip_gcc.mak b/CPP/7zip/7zip_gcc.mak
--- a/CPP/7zip/7zip_gcc.mak	2022-07-15 14:00:00.000000000 +0200
+++ b/CPP/7zip/7zip_gcc.mak	2022-11-10 22:05:39.648159700 +0100
@@ -131,7 +131,7 @@
 
 # LOCAL_LIBS=-lpthread
 # LOCAL_LIBS_DLL=$(LOCAL_LIBS) -ldl
-LIB2 = -lpthread -ldl
+LIB2 = -ldl
 
 
 endif
