diff -u -r ../newsbeuter-2.9/src/utils.cpp ./src/utils.cpp
--- ../newsbeuter-2.9/src/utils.cpp	2015-02-19 11:56:59.000000000 +0100
+++ ./src/utils.cpp	2017-07-01 00:13:42.783350794 +0200
@@ -37,6 +41,13 @@
 #include <openssl/crypto.h>
 #endif
 
+#ifdef __ANDROID__
+#include <sys/file.h>
+inline int lockf(int fd, int cmd, off_t ignored_len) {
+    return flock(fd, cmd);
+}
+#endif
+
 namespace newsbeuter {
 
 std::vector<std::string> utils::tokenize_quoted(const std::string& str, std::string delimiters) {
