diff --git a/src/rlimit.rs b/src/rlimit.rs
index f30d8dc..b844aa9 100644
--- a/src/rlimit.rs
+++ b/src/rlimit.rs
@@ -8,7 +8,7 @@ use std::sync::Arc;
 #[cfg(unix)]
 // Get the maximum number of open file descriptors for this process, and if
 // the hard limit is larger than the soft limit increase it.
-fn rlimit_nofile() -> u64 {
+fn rlimit_nofile() -> libc::rlim_t {
     let mut file_limit = libc::rlimit {
         rlim_cur: 0,
         rlim_max: 0,
