--- ./client/log_msg.h.orig	2022-02-09 21:41:08.522248377 +0530
+++ ./client/log_msg.h	2022-02-09 21:41:23.412248372 +0530
@@ -38,7 +38,7 @@
     LOG_VERBOSITY_INFO,         /*!< Constant to define a INFO message */
     LOG_VERBOSITY_DEBUG,        /*!< Constant to define a DEBUG message */
     LOG_LAST_VERBOSITY
-} log_level_t;
+};
 
 #define LOG_DEFAULT_VERBOSITY   LOG_VERBOSITY_NORMAL    /*!< Default verbosity to use */
 
