diff --git a/build/gen.py b/build/gen.py
index 533df589..55470885 100755
--- a/build/gen.py
+++ b/build/gen.py
@@ -968,6 +968,8 @@ def WriteGNNinja(path, platform, host, options, args_list):
   executables['gn']['libs'].extend(static_libraries.keys())
   executables['gn_unittests']['libs'].extend(static_libraries.keys())
 
+  del executables['gn_unittests']
+
   WriteGenericNinja(path, static_libraries, executables, cxx, ar, ld,
                     platform, host, options, args_list,
                     cflags, ldflags, libflags, include_dirs, libs)
