--- ./build/gen.py.orig	2022-05-05 17:13:19.709381636 +0530
+++ ./build/gen.py	2022-05-05 17:13:56.479381622 +0530
@@ -893,6 +893,8 @@
   executables['gn']['libs'].extend(static_libraries.keys())
   executables['gn_unittests']['libs'].extend(static_libraries.keys())
 
+  del executables['gn_unittests']
+
   WriteGenericNinja(path, static_libraries, executables, cxx, ar, ld,
                     platform, host, options, args_list,
                     cflags, ldflags, libflags, include_dirs, libs)
