--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -41,8 +41,8 @@
 
   include(CheckIncludeFile)
   Check_Include_File(sys/auxv.h HAVE_SYS_AUXV)
-  if(EXISTS "/lib/systemd/system")
-    set(CMAKE_INSTALL_SYSTEMD_UNITDIR "/lib/systemd/system" CACHE STRING "Where to install systemd unit files")
+  if(TRUE)
+    set(CMAKE_INSTALL_SYSTEMD_UNITDIR "@TERMUX_PREFIX@/lib/systemd/system" CACHE STRING "Where to install systemd unit files")
   endif()
 endif()
 
