--- a/src/Mayaqua/FileIO.c
+++ b/src/Mayaqua/FileIO.c
@@ -20,8 +20,8 @@
 
 #include <Hamcore.h>
 
-static char exe_file_name[MAX_SIZE] = "/tmp/a.out";
-static wchar_t exe_file_name_w[MAX_SIZE] = L"/tmp/a.out";
+static char exe_file_name[MAX_SIZE] = "@TERMUX_PREFIX@/tmp/a.out";
+static wchar_t exe_file_name_w[MAX_SIZE] = L"@TERMUX_PREFIX@/tmp/a.out";
 static LIST *hamcore = NULL;
 static HAMCORE *hamcore_io = NULL;
 
