diff --git a/morphosis/__init__.py b/morphosis/__init__.py
index 7b43a71..b6af177 100644
--- a/morphosis/__init__.py
+++ b/morphosis/__init__.py
@@ -18,10 +18,6 @@ _RESOURCES = ("data", "ui")
 
 signal.signal(signal.SIGINT, signal.SIG_DFL)
 
-locale.bindtextdomain("morphosis", LOCALEDIR)
-locale.textdomain("morphosis")
-gettext.bindtextdomain("morphosis", LOCALEDIR)
-gettext.textdomain("morphosis")
 
 for name in _RESOURCES:
     path = Path(PKGDATADIR, name).with_suffix(".gresource")
