+++ ./hw/xfree86/os-support/linux/Makefile.am	2018-11-21 22:50:02.949575248 +0200
@@ -18,7 +18,7 @@
 XORG_CFLAGS += $(DBUS_CFLAGS)
 endif
 
-liblinux_la_SOURCES = linux.h lnx_init.c lnx_video.c \
+liblinux_la_SOURCES = linux.h lnx_init.c $(srcdir)/../shared/ioperm_noop.c $(srcdir)/../shared/platform_noop.c \
                      lnx_agp.c lnx_kmod.c lnx_bell.c lnx_platform.c \
 		     $(srcdir)/../shared/VTsw_usl.c \
 		     $(srcdir)/../shared/posix_tty.c \
