--- a/os/backtrace.c
+++ b/os/backtrace.c
@@ -249,14 +249,13 @@
         /* CHILD */
         char parent[16];
 
-        seteuid(0);
         close(STDIN_FILENO);
         close(STDOUT_FILENO);
         dup2(pipefd[1], STDOUT_FILENO);
         closefrom(STDERR_FILENO);
 
         snprintf(parent, sizeof(parent), "%d", getppid());
-        execle("/usr/bin/pstack", "pstack", parent, NULL);
+        execle("@TERMUX_PREFIX@/bin/pstack", "pstack", parent, NULL);
         exit(1);
     }
     else {
