diff -uNr xorg-server-1.20.11/test/xtest.c xorg-server-1.20.11.mod/test/xtest.c
--- xorg-server-1.20.11/test/xtest.c	2021-04-13 14:11:40.000000000 +0000
+++ xorg-server-1.20.11.mod/test/xtest.c	2021-06-18 06:42:16.318605204 +0000
@@ -61,7 +61,7 @@
 {
     ScreenRec screen = {0};
     ClientRec server_client = {0};
-    WindowRec root = {0};
+    WindowRec root = {{0}};
     WindowOptRec optional = {0};
 
     /* random stuff that needs initialization */
