--- a/components/permissions/permission_request_manager.cc
+++ b/components/permissions/permission_request_manager.cc
@@ -437,7 +438,9 @@
 bool PermissionRequestManager::
     HasActiveSourceFrameOrDisallowActivationOtherwise(
         const PermissionRequest& request) const {
-  const auto iter = request_sources_map_.find(request);
+  const auto iter = request_sources_map_.find(
+      base::raw_ref<PermissionRequest>::from_ptr(
+          const_cast<PermissionRequest*>(&request)));
   if (iter != request_sources_map_.end()) {
     return !iter->second.IsSourceFrameInactiveAndDisallowActivation();
   }
