diff --git a/src/lib/baloosettings.kcfg b/src/lib/baloosettings.kcfg
index faa5303..0b1c2d2 100644
--- a/src/lib/baloosettings.kcfg
+++ b/src/lib/baloosettings.kcfg
@@ -9,7 +9,7 @@
   <group name="Basic Settings">
     <entry name="indexingEnabled" key="Indexing-Enabled" type="Bool">
       <label>Indexing-Enabled</label>
-      <default>true</default>
+      <default>false</default>
     </entry>
   </group>
   <group name="General">
