frida-version.h is normally generated from git
--- /dev/null	2018-03-05 09:14:44.324251389 +0000
+++ ./build/frida-version.h	2018-03-05 10:37:42.372271356 +0000
@@ -0,0 +1,12 @@
+/* @TERMUX_PKG_VERSION@.0 */
+#ifndef __FRIDA_VERSION_H__
+#define __FRIDA_VERSION_H__
+
+#define FRIDA_VERSION "@TERMUX_PKG_VERSION@"
+
+#define FRIDA_MAJOR_VERSION @_MAJOR_VERSION@
+#define FRIDA_MINOR_VERSION @_MINOR_VERSION@
+#define FRIDA_MICRO_VERSION @_MICRO_VERSION@
+#define FRIDA_NANO_VERSION 0
+
+#endif
