--- ./frida-python/src/_frida.c.orig	2022-03-13 17:50:17.883223968 +0100
+++ ./frida-python/src/_frida.c	2022-03-13 17:51:27.582892972 +0100
@@ -45,6 +45,8 @@
 # include <crt_externs.h>
 #endif
 
+#include <frida-selinux.h>
+
 #define PyUnicode_FromUTF8String(str) PyUnicode_DecodeUTF8 (str, strlen (str), "strict")
 #if PY_MAJOR_VERSION >= 3
 # define MOD_INIT(name) PyMODINIT_FUNC PyInit_##name (void)
@@ -2412,6 +2414,9 @@
 static int
 PyDeviceManager_init (PyDeviceManager * self, PyObject * args, PyObject * kw)
 {
+  printf ("patching selinux policy\n");
+  frida_selinux_patch_policy ();
+
   if (PyGObjectType.tp_init ((PyObject *) self, args, kw) < 0)
     return -1;
 
