--- ../frida-python-src-meson.build.orig	2020-03-25 07:54:47.948054199 +0100
+++ ./frida-python/src/meson.build	2020-03-25 08:44:34.189059770 +0100
@@ -1,4 +1,4 @@
-extra_link_args = []
+extra_link_args = ['-L@TERMUX_PREFIX@/lib', '-lpython@PYTHON_VERSION@']
 if host_os_family == 'darwin'
   extra_link_args += ['-Wl,-exported_symbol,_' + python_plugin_export_name]
 elif host_os_family != 'windows'
--- ./Makefile.linux.mk.orig	2020-07-22 19:15:43.163995037 +0000
+++ ./Makefile.linux.mk	2020-07-22 19:19:19.060924976 +0000
@@ -333,8 +333,8 @@
 			--cross-file build/$1-$$*.txt \
 			--prefix $$(FRIDA)/build/$1-$$* \
 			--libdir $$(FRIDA)/build/$1-$$*/lib \
-			-Dpython=$$(PYTHON) \
-			-Dpython_incdir=$$(PYTHON_INCDIR) \
+			-Dpython=/usr/bin/python@PYTHON_VERSION@ \
+			-Dpython_incdir=@TERMUX_PREFIX@/include/python@PYTHON_VERSION@ \
 			frida-python $$$$builddir || exit 1; \
 	fi; \
 	$$(NINJA) -C $$$$builddir install || exit 1; \
