--- ./releng/setup-env.sh.orig	2021-03-21 14:30:27.958066126 +0000
+++ ./releng/setup-env.sh	2021-03-28 13:17:51.665477424 +0000
@@ -106,7 +106,7 @@
 fi
 
 if [ "$host_os" == "android" ]; then
-  ndk_required=22
+  ndk_required=21
   if [ -n "$ANDROID_NDK_ROOT" ]; then
     if [ -f "$ANDROID_NDK_ROOT/source.properties" ]; then
       ndk_installed_version=$(grep Pkg.Revision "$ANDROID_NDK_ROOT/source.properties" | awk '{ split($NF, v, "."); print v[1]; }')
