--- ../setup-env.sh.orig	2019-03-04 22:56:56.693240356 +0100
+++ ./releng/setup-env.sh	2019-03-04 22:57:18.379906864 +0100
@@ -97,8 +97,8 @@
 fi
 
 if [ $host_platform = android ]; then
-  ndk_required_name=r17b
-  ndk_required_version=17.1.4828580
+  ndk_required_name=r19b
+  ndk_required_version=19.1.5304403
   if [ -n "$ANDROID_NDK_ROOT" ]; then
     if [ -f "$ANDROID_NDK_ROOT/source.properties" ]; then
       ndk_installed_version=$(grep Pkg.Revision "$ANDROID_NDK_ROOT/source.properties" | awk '{ print $NF; }')
@@ -414,7 +414,7 @@
 
     case $host_arch in
       x86)
-        android_api=14
+        android_api=21
         host_triplet="i686-linux-android"
         host_arch_flags="-march=i686"
         host_ldflags="-fuse-ld=gold"
@@ -426,7 +426,7 @@
         host_ldflags="-fuse-ld=gold"
         ;;
       arm)
-        android_api=14
+        android_api=21
         host_triplet="arm-linux-androideabi"
         host_arch_flags="-march=armv7-a -mfloat-abi=softfp -mfpu=vfpv3-d16"
         host_ldflags="-fuse-ld=gold -Wl,--fix-cortex-a8 -Wl,--icf=safe"
