--- ./releng/setup-env.sh.orig	2021-03-21 14:30:27.958066126 +0000
+++ ./releng/setup-env.sh	2021-03-28 13:17:51.665477424 +0000
@@ -628,26 +628,26 @@
     host_cflags=""
     case $host_arch in
       x86)
-        android_api=18
+        android_api=24
         host_compiler_triplet="i686-linux-android"
         host_arch_flags="-march=pentium4"
         host_cflags="-mfpmath=sse -mstackrealign"
         host_ldflags="-fuse-ld=gold"
         ;;
       x86_64)
-        android_api=21
+        android_api=24
         host_compiler_triplet="x86_64-linux-android"
         host_ldflags="-fuse-ld=gold -Wl,--icf=all"
         ;;
       arm)
-        android_api=18
+        android_api=24
         host_compiler_triplet="armv7a-linux-androideabi"
         host_tooltriplet="arm-linux-androideabi"
         host_arch_flags="-march=armv7-a -mfloat-abi=softfp -mfpu=vfpv3-d16"
         host_ldflags="-fuse-ld=gold -Wl,--icf=all -Wl,--fix-cortex-a8"
         ;;
       arm64)
-        android_api=21
+        android_api=24
         host_compiler_triplet="aarch64-linux-android"
         host_ldflags="-fuse-ld=gold -Wl,--icf=all"
         ;;
