--- ./releng/setup-env.sh.orig	2021-12-05 14:17:02.109846250 +0100
+++ ./releng/setup-env.sh	2021-12-05 14:21:05.179490094 +0100
@@ -668,7 +668,7 @@
     host_cxxlibs="c++_static c++abi"
     case $host_arch in
       x86)
-        android_api=19
+        android_api=24
         android_abi="x86"
         android_target="i686-none-linux-android${android_api}"
         android_clang_arch="i386"
@@ -679,7 +679,7 @@
         host_cxxlibs="$host_cxxlibs android_support"
         ;;
       x86_64)
-        android_api=21
+        android_api=24
         android_abi="x86_64"
         android_target="x86_64-none-linux-android${android_api}"
         android_clang_arch="x86_64"
@@ -687,7 +687,7 @@
         host_ldflags=""
         ;;
       arm)
-        android_api=19
+        android_api=24
         android_abi="armeabi-v7a"
         android_target="armv7-none-linux-androideabi${android_api}"
         android_clang_arch="arm"
@@ -698,7 +698,7 @@
         host_cxxlibs="$host_cxxlibs android_support"
         ;;
       arm64)
-        android_api=21
+        android_api=24
         android_abi="arm64-v8a"
         android_target="aarch64-none-linux-android${android_api}"
         android_clang_arch="aarch64"
