--- ./releng/setup-env.sh.orig	2021-12-05 14:17:02.109846250 +0100
+++ ./releng/setup-env.sh	2021-12-05 14:21:05.179490094 +0100
@@ -458,7 +458,7 @@
     NM="${NM:-${host_toolprefix}nm}"
     RANLIB="${RANLIB:-${host_toolprefix}ranlib}"
     STRIP="${STRIP:-${host_toolprefix}strip}"
-    STRIP_FLAGS="--strip-all"
+    STRIP_FLAGS="--strip-unneeded --preserve-dates"
     READELF="${READELF:-${host_toolprefix}readelf}"
     OBJCOPY="${OBJCOPY:-${host_toolprefix}objcopy}"
     OBJDUMP="${OBJDUMP:-${host_toolprefix}objdump}"
@@ -695,7 +695,7 @@
     host_cxxlibs="c++_static c++abi"
     case $host_arch in
       x86)
-        android_api=19
+        android_api=24
         android_abi="x86"
         android_target="i686-none-linux-android${android_api}"
         android_clang_arch="i386"
@@ -706,7 +706,7 @@
         host_cxxlibs="$host_cxxlibs android_support"
         ;;
       x86_64)
-        android_api=21
+        android_api=24
         android_abi="x86_64"
         android_target="x86_64-none-linux-android${android_api}"
         android_clang_arch="x86_64"
@@ -714,7 +714,7 @@
         host_ldflags=""
         ;;
       arm)
-        android_api=19
+        android_api=24
         android_abi="armeabi-v7a"
         android_target="armv7-none-linux-androideabi${android_api}"
         android_clang_arch="arm"
@@ -725,7 +725,7 @@
         host_cxxlibs="$host_cxxlibs android_support"
         ;;
       arm64)
-        android_api=21
+        android_api=24
         android_abi="arm64-v8a"
         android_target="aarch64-none-linux-android${android_api}"
         android_clang_arch="aarch64"
