Use the NDK we normally use for termux
--- ./releng/setup-env.sh	2018-02-26 19:22:57.000000000 +0000
+++ ../setup-env.sh	2018-03-05 12:49:07.583020888 +0000
@@ -92,8 +92,8 @@
 fi
 
 if [ $host_platform = android ]; then
-  ndk_required_name=r15c
-  ndk_required_version=15.2.4203891
+  ndk_required_name=r16b
+  ndk_required_version=16.0.4442984
   if [ -n "$ANDROID_NDK_ROOT" ]; then
     if [ -f "$ANDROID_NDK_ROOT/source.properties" ]; then
       ndk_installed_version=$(grep Pkg.Revision "$ANDROID_NDK_ROOT/source.properties" | awk '{ print $NF; }')
