--- a/src/CuSE.cxx
+++ b/src/CuSE.cxx
@@ -56,7 +56,7 @@
 #include <pthread.h>
 #include <curses.h>
 #include <math.h>
-#include <cdk/cdk.h>
+#include <cdk.h>
 #include <netinet/in.h>
 #include <SDL/SDL.h>
 #include <SDL/SDL_mixer.h>
@@ -79,7 +79,7 @@
   /* GUI-Components */
 
 Track       *p_first_displayed_track, *p_last_displayed_track;
-static char *menulist[MAX_MENU_ITEMS][MAX_SUB_ITEMS];
+static const char *menulist[MAX_MENU_ITEMS][MAX_SUB_ITEMS];
 char        status_text[LENGTH_DISPLAY_TEXT];
 bool        display_top_line, popup_active;
 short       cursor_column; // current cursor column in track display
