diff -uNr cairo-1.14.12/util/cairo-script/cairo-script-operators.c cairo-1.14.12.mod/util/cairo-script/cairo-script-operators.c
--- cairo-1.14.12/util/cairo-script/cairo-script-operators.c	2017-12-05 00:26:25.000000000 +0200
+++ cairo-1.14.12.mod/util/cairo-script/cairo-script-operators.c	2018-06-19 17:48:57.767824702 +0300
@@ -1719,7 +1719,7 @@
 static void *
 _mmap_bytes (const struct mmap_vec *vec, int count)
 {
-    char template[] = "/tmp/csi-font.XXXXXX";
+    char template[] = "@TERMUX_PREFIX@/tmp/csi-font.XXXXXX";
     void *ptr;
     int fd;
     int num_bytes;
