diff -uNr sc-im-0.7.0/src/clipboard.c sc-im-0.7.0.mod/src/clipboard.c
--- sc-im-0.7.0/src/clipboard.c	2017-12-13 19:48:59.000000000 +0200
+++ sc-im-0.7.0.mod/src/clipboard.c	2019-12-23 14:13:09.827265417 +0200
@@ -68,7 +68,7 @@
     if (! strlen(get_conf_value("default_paste_from_clipboard_cmd"))) return -1;
 
     // create tmp file
-    char template[] = "/tmp/sc-im-clipboardXXXXXX";
+    char template[] = "@TERMUX_PREFIX@/tmp/sc-im-clipboardXXXXXX";
     int fd = mkstemp(template);
     if (fd == -1) {
         sc_error("Error while pasting from clipboard");
