diff -uNr sc-im-0.7.0/src/cmds_visual.c sc-im-0.7.0.mod/src/cmds_visual.c
--- sc-im-0.7.0/src/cmds_visual.c	2017-12-13 19:48:59.000000000 +0200
+++ sc-im-0.7.0.mod/src/cmds_visual.c	2019-01-30 22:00:04.705760754 +0200
@@ -141,6 +141,10 @@
  * \return none
  */
 
+#ifdef USELOCALE
+#include <locale.h>
+#include <langinfo.h>
+#endif
 void do_visualmode(struct block * buf) {
     // we are moving (previous to a 'C-o' keypress)
     if (moving == TRUE) {
@@ -400,8 +404,6 @@
     // datefmt with locale D_FMT format
     } else if (buf->value == ctl('d')) {
         #ifdef USELOCALE
-            #include <locale.h>
-            #include <langinfo.h>
             char * loc = NULL;
             char * f = NULL;
             loc = setlocale(LC_TIME, "");
