--- ./src/socket.c.orig	2022-02-21 16:29:08.658684366 +0530
+++ ./src/socket.c	2022-02-21 16:30:50.978684327 +0530
@@ -460,6 +460,7 @@
 char mccp1_subneg[] = { TN_IAC, TN_SB, TN_COMPRESS, TN_WILL, TN_SE };
 #endif
 
+struct World *world_decl;
 Sock *xsock = NULL;		/* current (transmission) socket */
 int quit_flag = FALSE;		/* Are we all done? */
 int active_count = 0;		/* # of (non-current) active sockets */
