diff --git a/llvm-project/clang/lib/Driver/ToolChain.cpp b/llvm-project/clang/lib/Driver/ToolChain.cpp
index 469ef62f1a9d..218f5af748f4 100644
--- a/llvm-project/clang/lib/Driver/ToolChain.cpp
+++ b/llvm-project/clang/lib/Driver/ToolChain.cpp
@@ -632,7 +632,7 @@
 
   switch (Type) {
   case ToolChain::CST_Libcxx:
-    CmdArgs.push_back("-lc++");
+    CmdArgs.push_back("-lc++_shared");
     if (Args.hasArg(options::OPT_fexperimental_library))
       CmdArgs.push_back("-lc++experimental");
     break;
