diff --git a/llbuild/lib/BuildSystem/BuildSystem.cpp b/llbuild/lib/BuildSystem/BuildSystem.cpp
index 297dee34..40df58ca 100644
--- a/llbuild/lib/BuildSystem/BuildSystem.cpp
+++ b/llbuild/lib/BuildSystem/BuildSystem.cpp
@@ -3269,7 +3269,7 @@ class ArchiveShellCommand : public ExternalCommand {
     if (const char *ar = std::getenv("AR"))
       args.push_back(std::string(ar));
     else
-      args.push_back("ar");
+      args.push_back("llvm-ar");
     args.push_back("cr");
     args.push_back(archiveName);
     args.insert(args.end(), archiveInputs.begin(), archiveInputs.end());
