diff --git a/swift/include/swift/AST/Attr.h b/swift/include/swift/AST/Attr.h
index 0255b38af1b..3949e62e750 100644
--- a/swift/include/swift/AST/Attr.h
+++ b/swift/include/swift/AST/Attr.h
@@ -2661,7 +2661,7 @@ public:
 
   ArrayRef<Identifier> getSuppressedFeatures() const {
     return {getTrailingObjects<Identifier>(),
-            Bits.AllowFeatureSuppressionAttr.NumFeatures};
+            static_cast<size_t>(Bits.AllowFeatureSuppressionAttr.NumFeatures)};
   }
 
   static bool classof(const DeclAttribute *DA) {
