diff --git a/swift/lib/ClangImporter/ImportType.cpp b/swift/lib/ClangImporter/ImportType.cpp
index 08238b491c2..7bafbbac38b 100644
--- a/swift/lib/ClangImporter/ImportType.cpp
+++ b/swift/lib/ClangImporter/ImportType.cpp
@@ -506,7 +506,7 @@ namespace {
       if (size > 4096)
         return Type();
       
-      SmallVector<TupleTypeElt, 8> elts{size, elementType};
+      SmallVector<TupleTypeElt, 8> elts{static_cast<size_t>(size), elementType};
       return TupleType::get(elts, elementType->getASTContext());
     }
 
