From d1a88126f7d1e23afa1a1bab4c4abf0a9bce8a8d
From: Jeremy Schonfeld <jschonfeld@apple.com>
Date: Tue, 3 Sep 2024 09:07:03 -0700
Subject: [PATCH] Add new Foundation libraries to SwiftRuntimeLibsOrdered
 (#76019)

diff --git a/swift/lib/DriverTool/autolink_extract_main.cpp b/swift/lib/DriverTool/autolink_extract_main.cpp
index c74a473fce0e9..8c4381dfddd6c 100644
--- a/swift/lib/DriverTool/autolink_extract_main.cpp
+++ b/swift/lib/DriverTool/autolink_extract_main.cpp
@@ -231,6 +231,7 @@ int autolink_extract_main(ArrayRef<const char *> Args, const char *Argv0,
       "-lswift_RegexBuilder",
       "-lswift_RegexParser",
       "-lswift_Backtracing",
+      "-lswiftSynchronization",
       "-lswiftGlibc",
       "-lBlocksRuntime",
       // Dispatch-specific Swift runtime libs
@@ -238,8 +239,11 @@ int autolink_extract_main(ArrayRef<const char *> Args, const char *Argv0,
       "-lDispatchStubs",
       "-lswiftDispatch",
       // CoreFoundation and Foundation Swift runtime libs
+      "-l_FoundationICU",
       "-lCoreFoundation",
       "-lFoundation",
+      "-lFoundationEssentials",
+      "-lFoundationInternationalization",
       "-lFoundationNetworking",
       "-lFoundationXML",
       // Foundation support libs
