--- a/cmake/wasi-sdk-sysroot.cmake
+++ b/cmake/wasi-sdk-sysroot.cmake
@@ -345,7 +345,7 @@ if(WASI_SDK_INSTALL_TO_CLANG_RESOURCE_DIR)
 else()
   install(DIRECTORY ${wasi_resource_dir}/lib
           USE_SOURCE_PERMISSIONS
-          DESTINATION ${CMAKE_INSTALL_PREFIX}/clang-resource-dir)
+          DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/clang/@LLVM_MAJOR_VERSION@)
 endif()
 
 # Add a top-level `build` target as well as `build-$target` targets.
