--- ./configure	2017-11-27 20:45:47.000000000 +0530
+++ ../configure	2017-12-02 10:42:46.176674068 +0530
@@ -358,12 +358,6 @@
     DIRS+=("$LIQSRCDIR" "$LIQSRCDIR") # local libimagequant
 fi
 
-DIRS+=(
-      "/usr/local/include /usr/local/lib"
-      "/usr/include /usr/lib64"
-      "/usr/include /usr/lib"
-      "/opt/local/include /opt/local/lib" # macports
-      )
 
 if [[ "$OSTYPE" =~ "darwin" ]]; then
     SOLIBSUFFIX=dylib
