diff -u -r ../pngquant-2.12.1/configure ./configure
--- ../pngquant-2.12.1/configure	2018-06-04 11:30:47.000000000 +0000
+++ ./configure	2018-07-02 11:53:11.547417517 +0000
@@ -360,12 +361,6 @@
     DIRS+=("$LIQSRCDIR" "$LIQSRCDIR") # local libimagequant
 fi
 
-DIRS+=(
-      "/usr/local/include /usr/local/lib"
-      "/usr/include /usr/lib64"
-      "/usr/include /usr/lib"
-      "/opt/local/include /opt/local/lib" # macports
-      )
 
 if [[ "$OSTYPE" =~ "darwin" ]]; then
     SOLIBSUFFIX=dylib
