diff -uNr jasper-version-2.0.16/CMakeLists.txt jasper-version-2.0.16.mod/CMakeLists.txt
--- jasper-version-2.0.16/CMakeLists.txt	2019-03-12 20:04:35.000000000 +0200
+++ jasper-version-2.0.16.mod/CMakeLists.txt	2019-08-16 21:46:51.751881755 +0300
@@ -323,7 +323,7 @@
 
 find_library(MATH_LIBRARY m)
 if (NOT MATH_LIBRARY)
-	set(MATH_LIBRARY "")
+	set(MATH_LIBRARY "-lm")
 endif()
 
 ################################################################################
