diff -u -r ../openjpeg-2.1.2/CMakeLists.txt ./CMakeLists.txt
--- ../openjpeg-2.1.2/CMakeLists.txt	2016-09-28 20:18:16.000000000 +0200
+++ ./CMakeLists.txt	2017-01-19 05:15:37.168652776 +0100
@@ -148,7 +148,7 @@
   # We could install *.cmake files in share/ however those files contains
   # hardcoded path to libraries on a multi-arch system (fedora/debian) those
   # path will be different (lib/i386-linux-gnu vs lib/x86_64-linux-gnu)
-  set(OPENJPEG_INSTALL_PACKAGE_DIR "${OPENJPEG_INSTALL_LIB_DIR}/${OPENJPEG_INSTALL_SUBDIR}")
+  set(OPENJPEG_INSTALL_PACKAGE_DIR "${OPENJPEG_INSTALL_LIB_DIR}/cmake/${OPENJPEG_INSTALL_SUBDIR}")
 endif()
 
 if (APPLE)
