diff --git a/src/Makefile b/src/Makefile
index c5279ef..44d3554 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -17,7 +17,7 @@ LIBSSL = -lssl
 LIBCRYPTO = -lcrypto

 CFLAGS = -Wall -Wextra -O -DCONFIG_SHAREDIR='"$(SHAREDIR)"' $(INCDIRS) $(MYCFLAGS)
-LDFLAGS = $(LIBDIRS) $(MYLDFLAGS)
+LDFLAGS += $(LIBDIRS) $(MYLDFLAGS)
 LIBS = -lm -ldl $(LIBLUA) $(LIBPCRE) $(LIBSSL) $(LIBCRYPTO) $(MYLIBS)

 MAN1 = imapfilter.1
