diff --git a/js/ui/main.js b/js/ui/main.js
index 83783d1..2151345 100644
--- a/js/ui/main.js
+++ b/js/ui/main.js
@@ -112,7 +112,9 @@ const KeyringPrompt = imports.ui.keyringPrompt;
 const RunDialog = imports.ui.runDialog;
 const Layout = imports.ui.layout;
 const LookingGlass = imports.ui.lookingGlass;
+if (Config.HAVE_NETWORKMANAGER && global.settings.get_boolean("enable-nm-agent")) {
 const NetworkAgent = imports.ui.networkAgent;
+}
 const NotificationDaemon = imports.ui.notificationDaemon;
 const WindowAttentionHandler = imports.ui.windowAttentionHandler;
 const CinnamonDBus = imports.ui.cinnamonDBus;
