diff --git a/files/usr/share/cinnamon/cinnamon-settings/modules/cs_backgrounds.py b/files/usr/share/cinnamon/cinnamon-settings/modules/cs_backgrounds.py
index 104268f..77c95ac 100755
--- a/files/usr/share/cinnamon/cinnamon-settings/modules/cs_backgrounds.py
+++ b/files/usr/share/cinnamon/cinnamon-settings/modules/cs_backgrounds.py
@@ -319,7 +319,7 @@ class Module:
     def get_system_backgrounds(self):
         picture_list = []
         folder_list = []
-        properties_dir = "/usr/share/cinnamon-background-properties"
+        properties_dir = "@TERMUX_PREFIX@/share/gnome-background-properties"
         backgrounds = []
         if os.path.exists(properties_dir):
             for i in os.listdir(properties_dir):
