diff --git a/cinnamon.session.in b/cinnamon.session.in
index 9191df0..7999775 100644
--- a/cinnamon.session.in
+++ b/cinnamon.session.in
@@ -1,6 +1,6 @@
 [Cinnamon Session]
 Name=Cinnamon
-RequiredComponents=cinnamon;org.cinnamon.ScreenSaver;nemo-autostart;@REQUIRED@cinnamon-killer-daemon;
+RequiredComponents=cinnamon;nemo-autostart;@REQUIRED@cinnamon-killer-daemon;
 DesktopName=X-Cinnamon
 
 
diff --git a/cinnamon2d.session.in b/cinnamon2d.session.in
index 8910f99..e15e849 100644
--- a/cinnamon2d.session.in
+++ b/cinnamon2d.session.in
@@ -1,6 +1,6 @@
 [Cinnamon Session]
 Name=Cinnamon (Software Rendering)
-RequiredComponents=cinnamon2d;org.cinnamon.ScreenSaver;nemo-autostart;@REQUIRED@cinnamon-killer-daemon;
+RequiredComponents=cinnamon2d;nemo-autostart;@REQUIRED@cinnamon-killer-daemon;
 DesktopName=X-Cinnamon
 
 
