--- a/cmake/external/td/CMakeLists.txt
+++ b/cmake/external/td/CMakeLists.txt
@@ -4,6 +4,19 @@
 # For license and copyright information please follow this link:
 # https://github.com/desktop-app/legal/blob/master/LEGAL
 
+if (DESKTOP_APP_USE_PACKAGED)
+    add_library(external_td INTERFACE IMPORTED GLOBAL)
+    add_library(desktop-app::external_td ALIAS external_td)
+
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(TDE2E REQUIRED IMPORTED_TARGET tde2e)
+    pkg_check_modules(TDUTILS REQUIRED IMPORTED_TARGET tdutils)
+
+    target_link_libraries(external_td INTERFACE PkgConfig::TDE2E)
+    target_link_libraries(external_td INTERFACE PkgConfig::TDUTILS)
+    return()
+endif()
+
 add_library(external_td INTERFACE IMPORTED GLOBAL)
 add_library(desktop-app::external_td ALIAS external_td)
 
--- a/Telegram/SourceFiles/tde2e/tde2e_api.cpp
+++ b/Telegram/SourceFiles/tde2e/tde2e_api.cpp
@@ -10,7 +10,11 @@
 #include "base/assertion.h"
 #include "base/debug_log.h"
 
+#if defined(__has_include) && !__has_include(<tde2e/td/e2e/e2e_api.h>)
+#include <td/e2e/e2e_api.h>
+#else
 #include <tde2e/td/e2e/e2e_api.h>
+#endif
 
 #define LOG_ERROR(error) \
 	LOG(("TdE2E Error %1: %2").arg(int(error.code)).arg(error.message.c_str()))
