--- a/cmake/external/lz4/CMakeLists.txt
+++ b/cmake/external/lz4/CMakeLists.txt
@@ -8,12 +8,6 @@
 add_library(desktop-app::external_lz4 ALIAS external_lz4)
 
 if (DESKTOP_APP_USE_PACKAGED)
-    find_package(lz4 QUIET)
-    if (lz4_FOUND)
-        target_link_libraries(external_lz4 INTERFACE lz4::lz4)
-        return()
-    endif()
-
     find_package(PkgConfig)
     if (PkgConfig_FOUND)
         pkg_check_modules(LZ4 IMPORTED_TARGET liblz4)
