diff -uNr pacman-6.0.0/doc/meson.build pacman-termux/doc/meson.build
--- pacman-6.0.0/doc/meson.build	2021-05-20 06:38:45.938120000 +0300
+++ pacman-termux/doc/meson.build	2021-10-04 16:00:39.000000000 +0300
@@ -101,7 +101,7 @@
 endforeach
 
 run_target('html',
-           command : ['/bin/true'],
+           command : ['@TERMUX_PREFIX@/bin/true'],
            depends : html_targets)
 
 custom_target(
