diff -uNr pacman-6.0.0/scripts/libmakepkg/executable/meson.build pacman-termux/scripts/libmakepkg/executable/meson.build
--- pacman-6.0.0/scripts/libmakepkg/executable/meson.build	2021-05-20 06:38:45.948119900 +0300
+++ pacman-termux/scripts/libmakepkg/executable/meson.build	2021-10-04 16:00:39.000000000 +0300
@@ -4,12 +4,10 @@
   'ccache.sh.in',
   'checksum.sh.in',
   'distcc.sh.in',
-  'fakeroot.sh.in',
   'gpg.sh.in',
   'gzip.sh.in',
   'pacman.sh.in',
   'strip.sh.in',
-  'sudo.sh.in',
   'vcs.sh.in',
 ]
 
