diff -bur pacman-6.0.0/scripts/libmakepkg/buildenv/buildflags.sh.in pacman-termux/scripts/libmakepkg/buildenv/buildflags.sh.in
--- pacman-6.0.0/scripts/libmakepkg/buildenv/buildflags.sh.in	2021-05-20 06:38:45.948119900 +0300
+++ pacman-termux/scripts/libmakepkg/buildenv/buildflags.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   buildflags.sh - Clear user-specified buildflags if requested
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/buildenv/compiler.sh.in pacman-termux/scripts/libmakepkg/buildenv/compiler.sh.in
--- pacman-6.0.0/scripts/libmakepkg/buildenv/compiler.sh.in	2021-05-20 06:38:45.948119900 +0300
+++ pacman-termux/scripts/libmakepkg/buildenv/compiler.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   compiler.sh - CCache and DistCC compilation
 #   ccache - Cache compilations and reuse them to save time on repetitions
diff -bur pacman-6.0.0/scripts/libmakepkg/buildenv/debugflags.sh.in pacman-termux/scripts/libmakepkg/buildenv/debugflags.sh.in
--- pacman-6.0.0/scripts/libmakepkg/buildenv/debugflags.sh.in	2021-05-20 06:38:45.948119900 +0300
+++ pacman-termux/scripts/libmakepkg/buildenv/debugflags.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   debugflags.sh - Specify flags for building a package with debugging
 #   symbols
diff -bur pacman-6.0.0/scripts/libmakepkg/buildenv/lto.sh.in pacman-termux/scripts/libmakepkg/buildenv/lto.sh.in
--- pacman-6.0.0/scripts/libmakepkg/buildenv/lto.sh.in	2021-05-20 06:38:45.948119900 +0300
+++ pacman-termux/scripts/libmakepkg/buildenv/lto.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   lto.sh - Specify flags for building a package with link-time
 #            optimisation
diff -bur pacman-6.0.0/scripts/libmakepkg/buildenv/makeflags.sh.in pacman-termux/scripts/libmakepkg/buildenv/makeflags.sh.in
--- pacman-6.0.0/scripts/libmakepkg/buildenv/makeflags.sh.in	2021-05-20 06:38:45.948119900 +0300
+++ pacman-termux/scripts/libmakepkg/buildenv/makeflags.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   makeflags.sh - Clear user-specified makeflags if requested
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/buildenv.sh.in pacman-termux/scripts/libmakepkg/buildenv.sh.in
--- pacman-6.0.0/scripts/libmakepkg/buildenv.sh.in	2021-05-20 06:38:45.948119900 +0300
+++ pacman-termux/scripts/libmakepkg/buildenv.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   buildenv.sh - functions for altering the build environment before
 #   compilation
diff -bur pacman-6.0.0/scripts/libmakepkg/executable/ccache.sh.in pacman-termux/scripts/libmakepkg/executable/ccache.sh.in
--- pacman-6.0.0/scripts/libmakepkg/executable/ccache.sh.in	2021-05-20 06:38:45.948119900 +0300
+++ pacman-termux/scripts/libmakepkg/executable/ccache.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   ccache.sh - Confirm presence of ccache binary
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/executable/checksum.sh.in pacman-termux/scripts/libmakepkg/executable/checksum.sh.in
--- pacman-6.0.0/scripts/libmakepkg/executable/checksum.sh.in	2021-05-20 06:38:45.948119900 +0300
+++ pacman-termux/scripts/libmakepkg/executable/checksum.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   checksum.sh - Confirm presence of binaries for checksum operations
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/executable/distcc.sh.in pacman-termux/scripts/libmakepkg/executable/distcc.sh.in
--- pacman-6.0.0/scripts/libmakepkg/executable/distcc.sh.in	2021-05-20 06:38:45.948119900 +0300
+++ pacman-termux/scripts/libmakepkg/executable/distcc.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   distcc.sh - Confirm presence of distcc binary
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/executable/gpg.sh.in pacman-termux/scripts/libmakepkg/executable/gpg.sh.in
--- pacman-6.0.0/scripts/libmakepkg/executable/gpg.sh.in	2021-05-20 06:38:45.948119900 +0300
+++ pacman-termux/scripts/libmakepkg/executable/gpg.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   gpg.sh - Confirm presence of gpg binary
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/executable/gzip.sh.in pacman-termux/scripts/libmakepkg/executable/gzip.sh.in
--- pacman-6.0.0/scripts/libmakepkg/executable/gzip.sh.in	2021-05-20 06:38:45.948119900 +0300
+++ pacman-termux/scripts/libmakepkg/executable/gzip.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   gzip.sh - Confirm presence of gzip binary
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/executable/meson.build pacman-termux/scripts/libmakepkg/executable/meson.build
--- pacman-6.0.0/scripts/libmakepkg/executable/meson.build	2021-05-20 06:38:45.948119900 +0300
+++ pacman-termux/scripts/libmakepkg/executable/meson.build	2021-10-04 16:00:39.000000000 +0300
@@ -4,12 +4,10 @@
   'ccache.sh.in',
   'checksum.sh.in',
   'distcc.sh.in',
-  'fakeroot.sh.in',
   'gpg.sh.in',
   'gzip.sh.in',
   'pacman.sh.in',
   'strip.sh.in',
-  'sudo.sh.in',
   'vcs.sh.in',
 ]
 
diff -bur pacman-6.0.0/scripts/libmakepkg/executable/pacman.sh.in pacman-termux/scripts/libmakepkg/executable/pacman.sh.in
--- pacman-6.0.0/scripts/libmakepkg/executable/pacman.sh.in	2021-05-20 06:38:45.948119900 +0300
+++ pacman-termux/scripts/libmakepkg/executable/pacman.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   pacman.sh - Confirm presence of pacman binary
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/executable/strip.sh.in pacman-termux/scripts/libmakepkg/executable/strip.sh.in
--- pacman-6.0.0/scripts/libmakepkg/executable/strip.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/executable/strip.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   strip.sh - Confirm presence of strip binary
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/executable/vcs.sh.in pacman-termux/scripts/libmakepkg/executable/vcs.sh.in
--- pacman-6.0.0/scripts/libmakepkg/executable/vcs.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/executable/vcs.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   vcs.sh - Confirm presence of binaries for VCS operations
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/executable.sh.in pacman-termux/scripts/libmakepkg/executable.sh.in
--- pacman-6.0.0/scripts/libmakepkg/executable.sh.in	2021-05-20 06:38:45.948119900 +0300
+++ pacman-termux/scripts/libmakepkg/executable.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   executable.sh - confirm presence of dependent executables
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/integrity/generate_checksum.sh.in pacman-termux/scripts/libmakepkg/integrity/generate_checksum.sh.in
--- pacman-6.0.0/scripts/libmakepkg/integrity/generate_checksum.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/integrity/generate_checksum.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   generate_checksum.sh - functions for generating source checksums
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/integrity/generate_signature.sh.in pacman-termux/scripts/libmakepkg/integrity/generate_signature.sh.in
--- pacman-6.0.0/scripts/libmakepkg/integrity/generate_signature.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/integrity/generate_signature.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   generate_signature.sh - functions for generating PGP signatures
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/integrity/verify_checksum.sh.in pacman-termux/scripts/libmakepkg/integrity/verify_checksum.sh.in
--- pacman-6.0.0/scripts/libmakepkg/integrity/verify_checksum.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/integrity/verify_checksum.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   verify_checksum.sh - functions for checking source checksums
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/integrity/verify_signature.sh.in pacman-termux/scripts/libmakepkg/integrity/verify_signature.sh.in
--- pacman-6.0.0/scripts/libmakepkg/integrity/verify_signature.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/integrity/verify_signature.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   verify_signature.sh - functions for checking PGP signatures
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/integrity.sh.in pacman-termux/scripts/libmakepkg/integrity.sh.in
--- pacman-6.0.0/scripts/libmakepkg/integrity.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/integrity.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   integrity.sh - functions relating to source integrity checking
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_config/ext.sh.in pacman-termux/scripts/libmakepkg/lint_config/ext.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_config/ext.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_config/ext.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   ext.sh - Check that source/package extensions have valid prefixes
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_config/paths.sh.in pacman-termux/scripts/libmakepkg/lint_config/paths.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_config/paths.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_config/paths.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   paths.sh - Check that pathname components do not contain odd characters
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_config/source_date_epoch.sh.in pacman-termux/scripts/libmakepkg/lint_config/source_date_epoch.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_config/source_date_epoch.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_config/source_date_epoch.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   source_date_epoch.sh - Check that reproducible builds timestamp is valid
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_config/variable.sh.in pacman-termux/scripts/libmakepkg/lint_config/variable.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_config/variable.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_config/variable.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   variable.sh - Check that variables are or are not arrays as appropriate
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_config.sh.in pacman-termux/scripts/libmakepkg/lint_config.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_config.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_config.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   lint_config.sh - functions for checking for makepkg.conf errors
 #
@@ -21,7 +21,7 @@
 [[ -n "$LIBMAKEPKG_LINT_CONFIG_SH" ]] && return
 LIBMAKEPKG_LINT_CONFIG_SH=1
 
-LIBRARY=${LIBRARY:-'/usr/share/makepkg'}
+LIBRARY=${LIBRARY:-'/data/data/com.termux/files/usr/share/makepkg'}
 
 source "$LIBRARY/util/message.sh"
 source "$LIBRARY/util/util.sh"
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_package/build_references.sh.in pacman-termux/scripts/libmakepkg/lint_package/build_references.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_package/build_references.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_package/build_references.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   build_references.sh - Warn about files containing references to build directories
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_package/dotfiles.sh.in pacman-termux/scripts/libmakepkg/lint_package/dotfiles.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_package/dotfiles.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_package/dotfiles.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   dotfiles.sh - check for dotfiles in the package root
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_package/file_names.sh.in pacman-termux/scripts/libmakepkg/lint_package/file_names.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_package/file_names.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_package/file_names.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   file_names.sh - check package file names
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_package/missing_backup.sh.in pacman-termux/scripts/libmakepkg/lint_package/missing_backup.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_package/missing_backup.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_package/missing_backup.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   missing_backup.sh - Warn about missing files in the backup array
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_package.sh.in pacman-termux/scripts/libmakepkg/lint_package.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_package.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_package.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   lint_package.sh - functions for checking for packaging errors
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/arch.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/arch.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/arch.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/arch.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   arch.sh - Check the 'arch' array conforms to requirements.
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/arch_specific.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/arch_specific.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/arch_specific.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/arch_specific.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   arch_specific.sh - Check that arch specific variables can be arch specific.
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/backup.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/backup.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/backup.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/backup.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   backup.sh - Check the 'backup' array conforms to requirements.
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/changelog.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/changelog.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/changelog.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/changelog.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   changelog.sh - Check the files in the 'changelog' array exist.
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/checkdepends.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/checkdepends.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/checkdepends.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/checkdepends.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   checkdepends.sh - Check the 'checkdepends' array conforms to requirements.
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/conflicts.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/conflicts.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/conflicts.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/conflicts.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   conflicts.sh - Check the 'conflicts' array conforms to requirements.
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/depends.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/depends.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/depends.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/depends.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   depends.sh - Check the 'depends' array conforms to requirements.
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/epoch.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/epoch.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/epoch.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/epoch.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   epoch.sh - Check the 'epoch' variable conforms to requirements.
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/fullpkgver.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/fullpkgver.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/fullpkgver.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/fullpkgver.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   fullpkgver.sh - Check whether a full version conforms to requirements.
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/install.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/install.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/install.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/install.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   install.sh - Check the files in the 'install' array exist.
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/makedepends.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/makedepends.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/makedepends.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/makedepends.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   makedepends.sh - Check the 'makedepends' array conforms to requirements.
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/optdepends.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/optdepends.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/optdepends.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/optdepends.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   optdepends.sh - Check the 'optdepends' array conforms to requirements.
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/options.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/options.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/options.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/options.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   options.sh - Check the 'options' array conforms to requirements.
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/package_function.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/package_function.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/package_function.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/package_function.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   package_function.sh - Check that required package functions exist.
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/package_function_variable.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/package_function_variable.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/package_function_variable.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/package_function_variable.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   package_function_variable.sh - Check variables inside the package function.
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/pkgbase.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/pkgbase.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/pkgbase.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/pkgbase.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   pkgbase.sh - Check the 'pkgbase' variable conforms to requirements.
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/pkglist.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/pkglist.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/pkglist.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/pkglist.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   pkglist.sh - Check the packages selected to build exist.
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/pkgname.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/pkgname.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/pkgname.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/pkgname.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   pkgname.sh - Check the 'pkgname' variable conforms to requirements.
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/pkgrel.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/pkgrel.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/pkgrel.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/pkgrel.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   pkgrel.sh - Check the 'pkgrel' variable conforms to requirements.
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/pkgver.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/pkgver.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/pkgver.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/pkgver.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   pkgver.sh - Check the 'pkgver' variable conforms to requirements.
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/provides.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/provides.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/provides.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/provides.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   provides.sh - Check the 'provides' array conforms to requirements.
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/source.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/source.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/source.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/source.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   source.sh - Check the 'source' array is not sparse.
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/util.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/util.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/util.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/util.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   util.sh - utility functions for pkgbuild linting.
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/variable.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild/variable.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild/variable.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild/variable.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   variable.sh - Check that variables are or are not arrays as appropriate
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild.sh.in pacman-termux/scripts/libmakepkg/lint_pkgbuild.sh.in
--- pacman-6.0.0/scripts/libmakepkg/lint_pkgbuild.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/lint_pkgbuild.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   lint_pkgbuild.sh - functions for detecting PKGBUILD errors
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/source/bzr.sh.in pacman-termux/scripts/libmakepkg/source/bzr.sh.in
--- pacman-6.0.0/scripts/libmakepkg/source/bzr.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/source/bzr.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   bzr.sh - function for handling the download and "extraction" of Bazaar sources
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/source/file.sh.in pacman-termux/scripts/libmakepkg/source/file.sh.in
--- pacman-6.0.0/scripts/libmakepkg/source/file.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/source/file.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   file.sh - function for handling the download and extraction of source files
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/source/fossil.sh.in pacman-termux/scripts/libmakepkg/source/fossil.sh.in
--- pacman-6.0.0/scripts/libmakepkg/source/fossil.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/source/fossil.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   fossil.sh - function for handling the download and extraction of Fossil sources
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/source/git.sh.in pacman-termux/scripts/libmakepkg/source/git.sh.in
--- pacman-6.0.0/scripts/libmakepkg/source/git.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/source/git.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   git.sh - function for handling the download and "extraction" of Git sources
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/source/hg.sh.in pacman-termux/scripts/libmakepkg/source/hg.sh.in
--- pacman-6.0.0/scripts/libmakepkg/source/hg.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/source/hg.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   hg.sh - function for handling the download and "extraction" of Mercurial sources
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/source/local.sh.in pacman-termux/scripts/libmakepkg/source/local.sh.in
--- pacman-6.0.0/scripts/libmakepkg/source/local.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/source/local.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   local.sh - function for handling the "download" of local sources
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/source/svn.sh.in pacman-termux/scripts/libmakepkg/source/svn.sh.in
--- pacman-6.0.0/scripts/libmakepkg/source/svn.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/source/svn.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   svn.sh - function for handling the download and "extraction" of Subversion sources
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/source.sh.in pacman-termux/scripts/libmakepkg/source.sh.in
--- pacman-6.0.0/scripts/libmakepkg/source.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/source.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   source.sh - functions for downloading and extracting sources
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/srcinfo.sh.in pacman-termux/scripts/libmakepkg/srcinfo.sh.in
--- pacman-6.0.0/scripts/libmakepkg/srcinfo.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/srcinfo.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   srcinfo.sh - functions for writing .SRCINFO files
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/tidy/docs.sh.in pacman-termux/scripts/libmakepkg/tidy/docs.sh.in
--- pacman-6.0.0/scripts/libmakepkg/tidy/docs.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/tidy/docs.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   docs.sh - Remove documentation files from the package
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/tidy/emptydirs.sh.in pacman-termux/scripts/libmakepkg/tidy/emptydirs.sh.in
--- pacman-6.0.0/scripts/libmakepkg/tidy/emptydirs.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/tidy/emptydirs.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   emptydirs.sh - Remove empty directories from the package
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/tidy/libtool.sh.in pacman-termux/scripts/libmakepkg/tidy/libtool.sh.in
--- pacman-6.0.0/scripts/libmakepkg/tidy/libtool.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/tidy/libtool.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   libtool.sh - Remove libtool files from the package
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/tidy/purge.sh.in pacman-termux/scripts/libmakepkg/tidy/purge.sh.in
--- pacman-6.0.0/scripts/libmakepkg/tidy/purge.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/tidy/purge.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   purge.sh - Remove unwanted files from the package
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/tidy/staticlibs.sh.in pacman-termux/scripts/libmakepkg/tidy/staticlibs.sh.in
--- pacman-6.0.0/scripts/libmakepkg/tidy/staticlibs.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/tidy/staticlibs.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   staticlibs.sh - Remove static library files from the package
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/tidy/strip.sh.in pacman-termux/scripts/libmakepkg/tidy/strip.sh.in
--- pacman-6.0.0/scripts/libmakepkg/tidy/strip.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/tidy/strip.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   strip.sh - Strip debugging symbols from binary files
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/tidy/zipman.sh.in pacman-termux/scripts/libmakepkg/tidy/zipman.sh.in
--- pacman-6.0.0/scripts/libmakepkg/tidy/zipman.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/tidy/zipman.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   zipman.sh - Compress man and info pages
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/tidy.sh.in pacman-termux/scripts/libmakepkg/tidy.sh.in
--- pacman-6.0.0/scripts/libmakepkg/tidy.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/tidy.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   tidy.sh - functions for modifying/removing installed files before
 #   package creation
diff -bur pacman-6.0.0/scripts/libmakepkg/util/compress.sh.in pacman-termux/scripts/libmakepkg/util/compress.sh.in
--- pacman-6.0.0/scripts/libmakepkg/util/compress.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/util/compress.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   compress.sh - functions to compress archives in a uniform manner
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/util/config.sh.in pacman-termux/scripts/libmakepkg/util/config.sh.in
--- pacman-6.0.0/scripts/libmakepkg/util/config.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/util/config.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   config.sh - functions for handling makepkg config files
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/util/error.sh.in pacman-termux/scripts/libmakepkg/util/error.sh.in
--- pacman-6.0.0/scripts/libmakepkg/util/error.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/util/error.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   error.sh.in - error variable definitions for makepkg
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/util/message.sh.in pacman-termux/scripts/libmakepkg/util/message.sh.in
--- pacman-6.0.0/scripts/libmakepkg/util/message.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/util/message.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   message.sh - functions for outputting messages in makepkg
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/util/option.sh.in pacman-termux/scripts/libmakepkg/util/option.sh.in
--- pacman-6.0.0/scripts/libmakepkg/util/option.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/util/option.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   option.sh - functions to test if build/packaging options are enabled
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/util/parseopts.sh.in pacman-termux/scripts/libmakepkg/util/parseopts.sh.in
--- pacman-6.0.0/scripts/libmakepkg/util/parseopts.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/util/parseopts.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   parseopts.sh - getopt_long-like parser
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/util/pkgbuild.sh.in pacman-termux/scripts/libmakepkg/util/pkgbuild.sh.in
--- pacman-6.0.0/scripts/libmakepkg/util/pkgbuild.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/util/pkgbuild.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   pkgbuild.sh - functions to extract information from PKGBUILD files
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/util/schema.sh.in pacman-termux/scripts/libmakepkg/util/schema.sh.in
--- pacman-6.0.0/scripts/libmakepkg/util/schema.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/util/schema.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   schema.sh - declare specific groups of pkgbuild variables
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/util/source.sh.in pacman-termux/scripts/libmakepkg/util/source.sh.in
--- pacman-6.0.0/scripts/libmakepkg/util/source.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/util/source.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   source.sh - functions to extract information from source URLs
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/util/util.sh.in pacman-termux/scripts/libmakepkg/util/util.sh.in
--- pacman-6.0.0/scripts/libmakepkg/util/util.sh.in	2021-05-20 06:38:45.954786500 +0300
+++ pacman-termux/scripts/libmakepkg/util/util.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   util.sh - general utility functions
 #
diff -bur pacman-6.0.0/scripts/libmakepkg/util.sh.in pacman-termux/scripts/libmakepkg/util.sh.in
--- pacman-6.0.0/scripts/libmakepkg/util.sh.in	2021-05-20 06:38:45.951453200 +0300
+++ pacman-termux/scripts/libmakepkg/util.sh.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/data/data/com.termux/files/usr/bin/bash
 #
 #   util.sh - utility functions for makepkg
 #
