diff -uNr pacman-6.0.0/scripts/makepkg-template.pl.in pacman-termux/scripts/makepkg-template.pl.in
--- pacman-6.0.0/scripts/makepkg-template.pl.in	2021-05-20 06:38:45.954786500 +0300
+++ pacman-termux/scripts/makepkg-template.pl.in	2021-10-04 16:00:39.000000000 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/data/data/com.termux/files/usr/bin/perl
 #   makepkg-template - template system for makepkg
 #
 #   Copyright (c) 2013-2021 Pacman Development Team <pacman-dev@archlinux.org>
