diff -uNr pacman-6.0.0/etc/pacman.conf.in pacman-termux/etc/pacman.conf.in
--- pacman-6.0.0/etc/pacman.conf.in	2021-05-20 06:38:45.938120000 +0300
+++ pacman-termux/etc/pacman.conf.in	2021-11-05 17:39:44.301346062 +0300
@@ -9,20 +9,20 @@
 [options]
 # The following paths are commented out with their default values listed.
 # If you wish to use different paths, uncomment and update the paths.
-#RootDir     = @ROOTDIR@
+#RootDir     = /
 #DBPath      = @localstatedir@/lib/pacman/
 #CacheDir    = @localstatedir@/cache/pacman/pkg/
 #LogFile     = @localstatedir@/log/pacman.log
 #GPGDir      = @sysconfdir@/pacman.d/gnupg/
 #HookDir     = @sysconfdir@/pacman.d/hooks/
-HoldPkg     = pacman glibc
-#XferCommand = /usr/bin/curl -L -C - -f -o %o %u
-#XferCommand = /usr/bin/wget --passive-ftp -c -O %o %u
+#XferCommand = @TERMUX_PREFIX@/bin/curl -L -C - -f -o %o %u
+#XferCommand = @TERMUX_PREFIX@/bin/wget --passive-ftp -c -O %o %u
+HoldPkg      = pacman
 #CleanMethod = KeepInstalled
 Architecture = auto
 
 # Pacman won't upgrade packages listed in IgnorePkg and members of IgnoreGroup
 #IgnorePkg   =
 #IgnoreGroup =
 
 #NoUpgrade   =
@@ -30,16 +30,21 @@
 
 # Misc options
 #UseSyslog
-#Color
+Color
 #NoProgressBar
 CheckSpace
 #VerbosePkgLists
 ParallelDownloads = 5
 
-# PGP signature checking
-#SigLevel = Optional
-#LocalFileSigLevel = Optional
-#RemoteFileSigLevel = Optional
+# By default, pacman accepts packages signed by keys that its local keyring
+# trusts (see pacman-key and its man page), as well as unsigned packages.
+SigLevel    = Never
+LocalFileSigLevel = Optional
+#RemoteFileSigLevel = Required
+
+# NOTE: You must run `pacman-key --init` before first using pacman; the local
+# keyring can then be populated with the keys of all official Arch Linux ARM
+# packagers with `pacman-key --populate archlinuxarm`.
 
 #
 # REPOSITORIES
@@ -60,18 +65,12 @@
 # uncommented to enable the repo.
 #
 
-# An example of a disabled remote package repository with multiple servers
-# available. To enable, uncomment the following lines. You can add preferred
-# servers immediately after the header and they will be used before the
-# default mirrors.
-#[core]
-#SigLevel = Required
-#Server = ftp://ftp.example.com/foobar/$repo/os/$arch/
-# The file referenced here should contain a list of 'Server = ' lines.
-#Include = @sysconfdir@/pacman.d/mirrorlist
-
-# An example of a custom package repository.  See the pacman manpage for
-# tips on creating your own repositories.
-#[custom]
-#SigLevel = Optional TrustAll
-#Server = file:///home/custompkgs
+# The testing repositories are disabled by default. To enable, uncomment the
+# repo name header and Include lines. You can add preferred servers immediately
+# after the header, and they will be used before the default mirrors.
+
+[main]
+Server = https://maxython.github.io/termux-packages-in-pacman-format/main/$arch
+
+[x11]
+Server = https://maxython.github.io/termux-packages-in-pacman-format/x11/$arch
