diff --git a/xmake/packages/m/moebius/xmake.lua b/xmake/packages/m/moebius/xmake.lua
index 233b876..8c30731 100644
--- a/xmake/packages/m/moebius/xmake.lua
+++ b/xmake/packages/m/moebius/xmake.lua
@@ -25,6 +25,7 @@ package("moebius")
     add_urls("https://github.com/XmacsLabs/moebius.git")
     add_urls("https://gitee.com/XmacsLabs/moebius.git")
     add_versions("0.1.21", "v0.1.21")
+    add_patches("0.1.21", "moebius.diff", "6ddd295110dd2c224f9ffe3a94ba16209df1a86f9d3522d047b847090b0456f8")
 
     add_deps("lolly")
 
