--- a/xmake/packages/l/lolly/xmake.lua
+++ b/xmake/packages/l/lolly/xmake.lua
@@ -25,6 +25,7 @@
     add_urls("https://github.com/XmacsLabs/lolly.git")
     add_urls("https://gitee.com/XmacsLabs/lolly.git")
     add_versions("1.4.21", "v1.4.21")
+    add_patches("1.4.21", "lolly.diff", "5c46292e7c4b3d5420951b477c200915412a8f0675850e3bf593230c5fabfa39")
 
     add_deps("tbox")
     if not is_plat("wasm") then
