--- a/xmake/packages/l/lolly/xmake.lua
+++ b/xmake/packages/l/lolly/xmake.lua
@@ -25,6 +25,7 @@
     add_urls("https://github.com/XmacsLabs/lolly.git")
     add_urls("https://gitee.com/XmacsLabs/lolly.git")
     add_versions("1.4.28", "v1.4.28")
+    add_patches("1.4.28", "lolly.diff", "d58775888f5ec760422bcdc61fd84424204e7a5297b42ceedcdbb93a9b08530d")
 
     add_deps("tbox")
     if not is_plat("wasm") then
