diff --git a/bleak/backends/p4android/utils.py b/bleak/backends/p4android/utils.py
index xxxxxxx..yyyyyyy 100644
--- a/bleak/backends/p4android/utils.py
+++ b/bleak/backends/p4android/utils.py
@@ -8,7 +8,7 @@ from ...exc import BleakError
 logger = logging.getLogger(__name__)
 
 class AsyncJavaCallbacks(PythonJavaClass):
-    __javacontext__ = "app"
+    __javacontext__ = "dex"
 
     def __init__(self, loop: asyncio.AbstractEventLoop):
         self._loop = loop