diff -uNr libXt-1.2.0/include/X11/InitialI.h libXt-1.2.0.mod/include/X11/InitialI.h
--- libXt-1.2.0/include/X11/InitialI.h	2019-06-22 02:29:53.000000000 +0300
+++ libXt-1.2.0.mod/include/X11/InitialI.h	2019-06-29 23:39:02.027977057 +0300
@@ -271,9 +271,7 @@
     _XtBoolean 		/* ignoreInputs */,
     _XtBoolean		/* ignoreSignals */,
     _XtBoolean 		/* block */,
-#ifdef XTHREADS
     _XtBoolean		/* drop_lock */,
-#endif
     unsigned long*	/* howlong */
 );
 
