diff -uNr gtk-2.24.32/gtk/tests/filechooser.c gtk-2.24.32.mod/gtk/tests/filechooser.c
--- gtk-2.24.32/gtk/tests/filechooser.c	2018-01-08 23:35:39.000000000 +0200
+++ gtk-2.24.32.mod/gtk/tests/filechooser.c	2018-06-23 18:07:24.157696216 +0300
@@ -258,13 +258,13 @@
 #if 0
 #define FILE_NAME "/nonexistent"
 #define FILE_NAME_2 "/nonexistent2"
-#define FOLDER_NAME "/etc"
-#define FOLDER_NAME_2 "/usr"
+#define FOLDER_NAME "@TERMUX_PREFIX@/etc"
+#define FOLDER_NAME_2 "@TERMUX_PREFIX@"
 #else
-#define FILE_NAME "/etc/passwd"
-#define FILE_NAME_2 "/etc/group"
-#define FOLDER_NAME "/etc"
-#define FOLDER_NAME_2 "/usr"
+#define FILE_NAME "@TERMUX_PREFIX@/etc/passwd"
+#define FILE_NAME_2 "@TERMUX_PREFIX@/etc/group"
+#define FOLDER_NAME "@TERMUX_PREFIX@/etc"
+#define FOLDER_NAME_2 "@TERMUX_PREFIX@"
 #endif
 
 #define CURRENT_NAME "parangaricutirimicuaro.txt"
@@ -1859,9 +1859,9 @@
   gboolean passed = TRUE;
 
   /* first test for a file we know will always exist */
-  passed = passed && test_confirm_overwrite_for_path ("/etc/passwd", FALSE);
+  passed = passed && test_confirm_overwrite_for_path ("@TERMUX_PREFIX@/etc/passwd", FALSE);
   g_assert (passed);
-  passed = passed && test_confirm_overwrite_for_path ("/etc/resolv.conf", TRUE);
+  passed = passed && test_confirm_overwrite_for_path ("@TERMUX_PREFIX@/etc/resolv.conf", TRUE);
   g_assert (passed);
 }
 #endif
