--- ./data/meson.build.orig	2022-08-31 08:30:04.875909208 +0200
+++ ./data/meson.build	2022-08-31 08:30:10.886196504 +0200
@@ -2,7 +2,6 @@
 desktop_conf.set('VERSION', glade_version)
 
 i18n.merge_file(
-  'desktop',
   type: 'desktop',
   input: configure_file(
     input: 'org.gnome.Glade.desktop.in.in',
@@ -18,7 +17,6 @@
 appdata = glade_name + '.appdata.xml'
 
 i18n.merge_file(
-  'appdata',
   input: 'org.gnome.Glade.appdata.xml.in',
   output: '@BASENAME@',
   po_dir: po_dir,
