From 04ba6f969f716fbfe3c7feb7e4bab8678cc1e9eb Mon Sep 17 00:00:00 2001
From: Mike Gilbert <floppym@gentoo.org>
Date: Thu, 20 Jan 2022 14:19:03 -0500
Subject: [PATCH] Remove duplicate entry for 'da' from help/LINGUAS

---
 help/LINGUAS | 1 -
 1 file changed, 1 deletion(-)

diff --git a/help/LINGUAS b/help/LINGUAS
index fc3cd4c3..d67e42c8 100644
--- a/help/LINGUAS
+++ b/help/LINGUAS
@@ -1,6 +1,5 @@
 # please keep this list sorted alphabetically
 #
-da
 bg
 ca
 cs
-- 
GitLab
