diff -uNr glade-GLADE_3_38_2/meson.build glade-GLADE_3_38_2.mod/meson.build
--- glade-GLADE_3_38_2/meson.build	2020-11-20 22:06:30.000000000 +0200
+++ glade-GLADE_3_38_2.mod/meson.build	2021-08-16 00:30:10.241414718 +0300
@@ -224,7 +224,7 @@
   return 0;
 }
 '''
-tls_backend_support = cc.run(tls_backend_support_src, dependencies: gio_dep).stdout()
+tls_backend_support = 'false'
 
 output = '\nConfiguration:\n\n'
 output += '\tCompiler:                ' + cc.get_id() + '\n'
