--- a/tests/util/lib/consts.rs
+++ b/tests/util/lib/consts.rs
@@ -21,8 +21,7 @@
   ));
 }
 
-pub const TSGO_PLATFORM: &str = tsgo_platform();
-const fn tsgo_platform() -> &'static str {
+pub fn tsgo_platform() -> &'static str {
   match (
     std::env::consts::OS.as_bytes(),
     std::env::consts::ARCH.as_bytes(),
@@ -52,6 +51,6 @@
   crate::prebuilt_path().join(folder).join(format!(
     "tsgo-{}-{}",
     tsgo::VERSION,
-    TSGO_PLATFORM
+    tsgo_platform()
   ))
 }
--- a/tests/util/server/servers/mod.rs
+++ b/tests/util/server/servers/mod.rs
@@ -1482,7 +1482,7 @@
   let archive_name = format!(
     "typescript-go-{}-{}.zip",
     crate::consts::tsgo::VERSION,
-    crate::consts::TSGO_PLATFORM
+    crate::consts::tsgo_platform()
   );
 
   let url =
