--- a/cli/tsc/mod.rs
+++ b/cli/tsc/mod.rs
@@ -64,7 +64,7 @@
   #[inline(never)]
   || {
     static COMPRESSED_COMPILER_SNAPSHOT: &[u8] =
-      include_bytes!(concat!(env!("OUT_DIR"), "/COMPILER_SNAPSHOT.bin"));
+      include_bytes!(env!("DENO_PREBUILT_COMPILER_SNAPSHOT"));
 
     // NOTE(bartlomieju): Compressing the TSC snapshot in debug build took
     // ~45s on M1 MacBook Pro; without compression it took ~1s.
--- a/cli/js.rs
+++ b/cli/js.rs
@@ -4,7 +4,7 @@
 
 #[cfg(not(feature = "hmr"))]
 static CLI_SNAPSHOT: &[u8] =
-  include_bytes!(concat!(env!("OUT_DIR"), "/CLI_SNAPSHOT.bin"));
+  include_bytes!(env!("DENO_PREBUILT_CLI_SNAPSHOT"));
 
 pub fn deno_isolate_init() -> Option<&'static [u8]> {
   debug!("Deno isolate init with snapshots.");
