--- a/Cargo.toml
+++ b/Cargo.toml
@@ -107,7 +107,7 @@
 deno_terminal = "=0.2.3"
 deno_unsync = { version = "0.4.4", default-features = false }
 deno_whoami = "0.1.0"
-v8 = { package = "deno_v8", version = "0.2.0", path = "./libs/deno_v8", default-features = false, features = ["simdutf"] }
+v8 = { package = "deno_v8", version = "0.2.0", path = "./libs/deno_v8", features = ["simdutf"] }
 
 denokv_proto = "0.14.0"
 denokv_remote = "0.14.0"
