--- a/tests/util/server/src/pty.rs
+++ b/tests/util/server/src/pty.rs
@@ -407,7 +407,7 @@
           return Err(std::io::Error::last_os_error());
         }
         // Set the controlling terminal
-        if libc::ioctl(fds, libc::TIOCSCTTY as libc::c_ulong, 0) == -1 {
+        if libc::ioctl(fds, libc::TIOCSCTTY, 0) == -1 {
           return Err(std::io::Error::last_os_error());
         }
         set_winsize(fds, PTY_ROWS_COLS.0, PTY_ROWS_COLS.1)?;
