diff -uNr hexchat-2.14.2/plugins/lua/meson.build hexchat-2.14.2.mod/plugins/lua/meson.build
--- hexchat-2.14.2/plugins/lua/meson.build	2018-08-29 19:41:08.000000000 +0300
+++ hexchat-2.14.2.mod/plugins/lua/meson.build	2018-10-09 22:01:54.019863275 +0300
@@ -1,8 +1,4 @@
-if cc.get_id() == 'msvc'
-  lua_dep = cc.find_library('lua51')
-else
-  lua_dep = dependency(get_option('with-lua'))
-endif
+lua_dep = cc.find_library('lua',  dirs : '@TERMUX_PREFIX@/lib')
 
 shared_module('lua', 'lua.c',
   dependencies: [libgio_dep, hexchat_plugin_dep, lua_dep],
