--- a/slirp/slirp.c
+++ b/slirp/slirp.c
@@ -121,7 +121,7 @@
             return 0;
         }
         old_stat = dns_addr_stat;
-        if (stat("/etc/resolv.conf", &dns_addr_stat) != 0)
+        if (stat("@TERMUX_PREFIX@/etc/resolv.conf", &dns_addr_stat) != 0)
             return -1;
         if ((dns_addr_stat.st_dev == old_stat.st_dev)
             && (dns_addr_stat.st_ino == old_stat.st_ino)
@@ -132,7 +132,7 @@
         }
     }
 
-    f = fopen("/etc/resolv.conf", "r");
+    f = fopen("@TERMUX_PREFIX@/etc/resolv.conf", "r");
     if (!f)
         return -1;
 
