diff --git a/native/src/integrations/utils.rs b/native/src/integrations/utils.rs
index 06c1eab..49dd9f3 100644
--- a/native/src/integrations/utils.rs
+++ b/native/src/integrations/utils.rs
@@ -47,6 +47,7 @@ pub fn sanitize_name<'a>(name: &'a str, id: &'a str) -> String {
     platform_linux,
     platform_macos,
     platform_bsd,
+    platform_termux,
     all(platform_windows, feature = "portable")
 ))]
 #[inline]
