diff --git a/native/src/connector/process.rs b/native/src/connector/process.rs
index 11a14d4..3d2635e 100644
--- a/native/src/connector/process.rs
+++ b/native/src/connector/process.rs
@@ -83,11 +63,11 @@ impl Process for SetConfig {

 impl Process for InstallRuntime {
     fn process(&self, _connection: &Connection) -> Result<ConnectorResponse> {
-        #[cfg(platform_linux)]
+        #[cfg(any(platform_linux, platform_termux))]
         let options = self.clone().unwrap_or_default();

         let command = RuntimeInstallCommand {
-            #[cfg(platform_linux)]
+            #[cfg(any(platform_linux, platform_termux))]
             link: options.link,
         };
         command.run()?;
