diff --git a/native/Cargo.toml b/native/Cargo.toml
index 47315bf..48ccefd 100644
--- a/native/Cargo.toml
+++ b/native/Cargo.toml
@@ -73,11 +73,11 @@ features = [
     "Win32_UI_Shell_PropertiesSystem"
 ]
 
-[target.'cfg(any(target_os = "linux", target_os = "dragonfly", target_os = "freebsd", target_os = "openbsd", target_os = "netbsd"))'.dependencies]
+[target.'cfg(any(target_os = "linux", target_os = "dragonfly", target_os = "freebsd", target_os = "openbsd", target_os = "netbsd", target_os = "android"))'.dependencies]
 glob = "0.3.2"
 phf = { version = "0.11.3", features = ["macros"] }
 
-[target.'cfg(target_os = "linux")'.dependencies]
+[target.'cfg(any(target_os = "linux", target_os = "android"))'.dependencies]
 blake3 = "1.5.5"
 bzip2 = "0.5.1"
 xz2 = "0.1.7"
