diff --git a/native/src/integrations/categories.rs b/native/src/integrations/categories.rs
index 8200485..81209df 100644
--- a/native/src/integrations/categories.rs
+++ b/native/src/integrations/categories.rs
@@ -10,7 +10,7 @@
 /// - W3C: https://github.com/w3c/manifest/wiki/Categories#standardized-categories
 /// - FreeDesktop: https://specifications.freedesktop.org/menu-spec/menu-spec-1.0.html#category-registry
 ///
-#[cfg(any(platform_linux, platform_bsd))]
+#[cfg(any(platform_linux, platform_termux, platform_bsd))]
 pub static XDG_CATEGORIES: phf::Map<&'static str, &'static [&'static str]> = phf::phf_map! {
     // Main Categories
     "audiovideo" => &["AudioVideo"],
