--- a/native/src/storage.rs
+++ b/native/src/storage.rs
@@ -49,7 +49,7 @@ pub struct Config {
     /// May be overwritten with a system environment variable.
     pub runtime_use_portals: bool,
 
-    #[cfg(platform_linux)]
+    #[cfg(any(platform_linux, platform_termux))]
     /// Experimental: Using the system runtime to save some disk space.
     /// This might not work on your system.
     pub use_linked_runtime: bool,
