https://github.com/termux/termux-packages/issues/15852

--- a/lib/xalloc.h
+++ b/lib/xalloc.h
@@ -48,7 +48,7 @@
    or by using gnulib's xalloc-die module.  This is the
    function to call when one wants the program to die because of a
    memory allocation failure.  */
-/*extern*/ _Noreturn void xalloc_die (void);
+/*extern*/ __attribute__ ((__noreturn__)) void xalloc_die (void);
 
 #endif /* GNULIB_XALLOC_DIE */
 
