diff -uNr wireshark-2.6.1/configure wireshark-2.6.1.mod/configure
--- wireshark-2.6.1/configure	2018-05-22 21:53:32.000000000 +0300
+++ wireshark-2.6.1.mod/configure	2018-06-26 18:49:28.811638451 +0300
@@ -45400,7 +45400,7 @@
 fi
 
 
-LD_OPTION="-Wl,--as-needed"
+LD_OPTION=""
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether we can add $LD_OPTION to LDFLAGS" >&5
 $as_echo_n "checking whether we can add $LD_OPTION to LDFLAGS... " >&6; }
 
