diff -uNr wireshark-2.6.1/plugins/epan/profinet/Makefile.in wireshark-2.6.1.mod/plugins/epan/profinet/Makefile.in
--- wireshark-2.6.1/plugins/epan/profinet/Makefile.in	2018-05-22 21:53:29.000000000 +0300
+++ wireshark-2.6.1.mod/plugins/epan/profinet/Makefile.in	2018-06-26 18:52:49.191648512 +0300
@@ -866,7 +866,7 @@
 	}
 
 profinet.la: $(profinet_la_OBJECTS) $(profinet_la_DEPENDENCIES) $(EXTRA_profinet_la_DEPENDENCIES) 
-	$(AM_V_CCLD)$(profinet_la_LINK) -rpath $(epan_plugindir) $(profinet_la_OBJECTS) $(profinet_la_LIBADD) $(LIBS)
+	$(AM_V_CCLD)$(profinet_la_LINK) -rpath $(epan_plugindir) $(profinet_la_OBJECTS) $(profinet_la_LIBADD) -L$(top_srcdir)/epan/.libs -L$(top_srcdir)/wsutil/.libs -L$(top_srcdir)/wiretap/.libs -lwsutil -lwireshark -lglib-2.0 $(LIBS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
