diff -uNr wireshark-2.6.1/plugins/epan/ethercat/Makefile.in wireshark-2.6.1.mod/plugins/epan/ethercat/Makefile.in
--- wireshark-2.6.1/plugins/epan/ethercat/Makefile.in	2018-05-22 21:53:29.000000000 +0300
+++ wireshark-2.6.1.mod/plugins/epan/ethercat/Makefile.in	2018-06-26 18:52:57.078315575 +0300
@@ -866,7 +866,7 @@
 	}
 
 ethercat.la: $(ethercat_la_OBJECTS) $(ethercat_la_DEPENDENCIES) $(EXTRA_ethercat_la_DEPENDENCIES) 
-	$(AM_V_CCLD)$(ethercat_la_LINK) -rpath $(epan_plugindir) $(ethercat_la_OBJECTS) $(ethercat_la_LIBADD) $(LIBS)
+	$(AM_V_CCLD)$(ethercat_la_LINK) -rpath $(epan_plugindir) $(ethercat_la_OBJECTS) $(ethercat_la_LIBADD) -L$(top_srcdir)/epan/.libs -L$(top_srcdir)/wsutil/.libs -L$(top_srcdir)/wiretap/.libs -lwireshark -lwsutil -lglib-2.0 $(LIBS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
