diff --git a/configure b/configure
index c5e62d5..02250f4 100755
--- a/configure
+++ b/configure
@@ -1288,7 +1288,7 @@ if ! [ $darwin_target ] ; then
   if [ "$($make conftest.android)" = "yes" ] ; then
     printf "yes\n"
     android_target=y
-    lang_flags="$lang_flags --target=aarch64-unknown-linux-android26 -D_BSD_SOURCE"
+    lang_flags="$lang_flags -D_BSD_SOURCE"
     printf "Regenerating config.make ... "
     gen_config_make
     printf "done\n"
