--- a/setup.py
+++ b/setup.py
@@ -27,7 +27,7 @@
                   'extras/css': glob.glob('extras/css/*.css'),
                   'extras/dsssl': glob.glob('extras/dsssl/*.dsl'),
                   },
-    data_files=[('/etc/ldptool', ['etc/ldptool.ini']), ],
+    data_files=[('@TERMUX_PREFIX@/etc/ldptool', ['etc/ldptool.ini']), ],
     entry_points={
         'console_scripts': ['ldptool = tldp.driver:main', ],
     },
