--- a/build/config/c++/c++.gni
+++ b/build/config/c++/c++.gni
@@ -6,6 +6,8 @@
 import("//build/config/sanitizers/sanitizers.gni")
 import("//build_overrides/build.gni")
 
+import("//v8/gni/snapshot_toolchain.gni")
+
 declare_args() {
   # Use in-tree libc++ (buildtools/third_party/libc++ and
   # buildtools/third_party/libc++abi) instead of the system C++ library for C++
@@ -42,6 +44,8 @@
 
 use_custom_libcxx =
     use_custom_libcxx || (use_custom_libcxx_for_host && !is_a_target_toolchain)
+use_custom_libcxx =
+    use_custom_libcxx || (use_custom_libcxx_for_host && current_toolchain == v8_snapshot_toolchain)
 use_custom_libcxx = use_custom_libcxx && !is_nacl
 
 declare_args() {
