--- a/headless/lib/headless_content_main_delegate.cc
+++ b/headless/lib/headless_content_main_delegate.cc
@@ -176,6 +176,11 @@
   if (!command_line->HasSwitch(::switches::kHeadless))
     command_line->AppendSwitch(::switches::kHeadless);
 
+#ifdef __TERMUX__
+  // Always pass no sandbox on Termux.
+  command_line->AppendSwitch(sandbox::policy::switches::kNoSandbox);
+#endif
+
   // Use software rendering by default, but don't mess with gl and angle
   // switches if user is overriding them.
   if (!command_line->HasSwitch(::switches::kUseGL) &&
