--- a/pdf/pdfium/pdfium_api_wrappers.cc
+++ b/pdf/pdfium/pdfium_api_wrappers.cc
@@ -147,7 +147,7 @@
 
   // Number of characters, including the NUL.
   const size_t expected_size = base::checked_cast<size_t>(buflen_bytes / 2);
-  PDFiumAPIStringBufferAdapter adapter(&name, expected_size,
+  PDFiumAPIStringBufferAdapter<std::u16string> adapter(&name, expected_size,
                                        /*check_expected_size=*/true);
   unsigned long actual_buflen_bytes = 0;  // NOLINT(runtime/int)
   bool result =
