--- a/chrome/browser/compose/compose_session.cc
+++ b/chrome/browser/compose/compose_session.cc
@@ -543,7 +543,7 @@
   // execution in case request fails.
   compose::LogComposeRequestReason(request_reason);
 
-  optimization_guide::ModelExecutionSessionCallbackWithLogging callback =
+  optimization_guide::ModelExecutionSessionCallbackWithLogging<optimization_guide::proto::ComposeLoggingData> callback =
       base::BindRepeating(&ComposeSession::ModelExecutionCallback,
                           weak_ptr_factory_.GetWeakPtr(),
                           std::move(request_timer), request_id_, request_reason,
