diff -u -r ../torsocks-2.2.0/tests/Makefile.am ./tests/Makefile.am
--- ../torsocks-2.2.0/tests/Makefile.am	2016-10-18 12:44:01.000000000 -0400
+++ ./tests/Makefile.am	2016-12-20 21:23:41.960184652 -0500
@@ -29,7 +29,7 @@
 test_connect_LDADD = $(LIBTAP) $(LIBTORSOCKS)
 
 test_fd_passing_SOURCES = test_fd_passing.c
-test_fd_passing_LDADD = $(LIBTAP) $(LIBTORSOCKS) -lpthread
+test_fd_passing_LDADD = $(LIBTAP) $(LIBTORSOCKS)
 
 test_getpeername_SOURCES = test_getpeername.c
 test_getpeername_LDADD = $(LIBTAP) $(LIBTORSOCKS)
