--- a/src/gui/debugger/dbg_pclog.c	2009-05-02 15:46:04.000000000 -0400
+++ /bsrc/gui/debugger/dbg_pclog.c	2026-07-30 20:46:46.690347313 -0400
@@ -163,7 +163,7 @@
 		gtk_window_iconify(GTK_WINDOW(dbgw.pclog));
 #endif
 
-	if(!GTK_WIDGET_VISIBLE(dbgw.pclog) && !options3.pclog.closed)
+	if(!gtk_widget_get_visible(dbgw.pclog) && !options3.pclog.closed)
 		gtk_widget_show(dbgw.pclog);
 
 	return dbgw.pclog;
