--- a/src/gui/debugger/dbg_stack.c	2009-05-06 15:48:47.000000000 -0400
+++ b/src/gui/debugger/dbg_stack.c	2026-07-30 20:50:27.506347313 -0400
@@ -197,7 +197,7 @@
 		gtk_window_iconify(GTK_WINDOW(dbgw.stack));
 #endif
 
-	if(!GTK_WIDGET_VISIBLE(dbgw.stack) && !options3.stack.closed)
+	if(!gtk_widget_get_visible(dbgw.stack) && !options3.stack.closed)
 		gtk_widget_show(dbgw.stack);
 
 	return dbgw.stack;
