--- a/pedrom/Makefile.in	2009-04-30 16:45:57.000000000 -0400
+++ b/pedrom/Makefile.in	2026-08-01 13:14:45.594347313 -0400
@@ -292,7 +292,7 @@
 x_includes = @x_includes@
 x_libraries = @x_libraries@
 pedromdir = $(pkgdatadir)/pedrom
-dist_pedrom_DATA = pedrom*.tib pedrom.txt
+dist_pedrom_DATA =
 EXTRA_DIST = pedrom-src.tar.bz2
 all: all-am
 
@@ -482,7 +482,7 @@
 
 
 uninstall:
-	rm -f $(pkgdatadir)/pedrom/*.tib
+
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
--- a/pedrom/Makefile.am	2007-12-08 05:26:22.000000000 -0500
+++ b/pedrom/Makefile.am	2026-08-01 13:15:49.014347313 -0400
@@ -1,7 +1,6 @@
 pedromdir = $(pkgdatadir)/pedrom
-dist_pedrom_DATA = pedrom*.tib pedrom.txt
+dist_pedrom_DATA = 
 
 EXTRA_DIST = pedrom-src.tar.bz2
 
 uninstall:
-	rm -f $(pkgdatadir)/pedrom/*.tib
