--- a/data/Makefile.in
+++ b/data/Makefile.in
@@ -57,9 +57,9 @@ $(tablefiles) table.conf: interpolation2.text

 bigram.db: $(textual_model_data)
 	$(RM) $(binary_model_data)
-	../utils/storage/gen_binary_files --gen-punct-table --table-dir $(top_srcdir)/data
-	../utils/storage/import_interpolation --table-dir $(top_srcdir)/data < $(top_srcdir)/data/interpolation2.text
-	../utils/training/gen_unigram --table-dir $(top_srcdir)/data
+	@HOSTBUILD_ROOTFS@/usr/bin/gen_binary_files --gen-punct-table --table-dir $(top_srcdir)/data
+	@HOSTBUILD_ROOTFS@/usr/bin/import_interpolation --table-dir $(top_srcdir)/data < $(top_srcdir)/data/interpolation2.text
+	@HOSTBUILD_ROOTFS@/usr/bin/gen_unigram --table-dir $(top_srcdir)/data

 addon_phrase_index.bin phrase_index.bin addon_pinyin_index.bin pinyin_index.bin $(binfiles): bigram.db

