--- recutils-1.8/lib/getpass.c	2019-01-03 09:19:29.000000000 +0000
+++ ../recutils-1.8-patched/lib/getpass.c	2020-02-15 18:35:52.025960001 +0000
@@ -80,6 +80,7 @@
     fclose (arg);
 }
 
+#ifndef HAVE_GETPASS
 char *
 getpass (const char *prompt)
 {
@@ -170,6 +171,7 @@
 
   return buf;
 }
+#endif
 
 #else /* W32 native */
 
@@ -187,6 +189,7 @@
 #  define PASS_MAX 512
 # endif
 
+#ifndef HAVE_GETPASS
 char *
 getpass (const char *prompt)
 {
@@ -229,3 +232,4 @@
   return strdup (getpassbuf);
 }
 #endif
+#endif
