diff -u -r ../tcsh-TCSH6_21_00/Makefile.in ./Makefile.in
--- ../tcsh-TCSH6_21_00/Makefile.in	2019-05-08 18:39:28.000000000 +0000
+++ ./Makefile.in	2019-05-11 21:18:26.577655692 +0000
@@ -471,11 +471,11 @@
 #	emxbind tcsh
 
 gethost.${SUF}: gethost.c sh.err.h tc.const.h sh.h
-	${CC_FOR_GETHOST} ${CF} ${CPPFLAGS} ${CFLAGS} $(srcdir)/gethost.c
+	${CC_FOR_GETHOST} ${CF} -I. -I$(srcdir) $(srcdir)/gethost.c
 
 gethost:  gethost.${SUF}
 	rm -f gethost
-	${CC_FOR_GETHOST} -o gethost ${CPPFLAGS} ${CFLAGS} ${LDFLAGS} gethost.${SUF}
+	${CC_FOR_GETHOST} -o gethost gethost.${SUF}
 
 tc.defs.c:	gethost host.defs
 	@rm -f $@.tmp
@@ -628,7 +628,7 @@
 
 install.man: tcsh.man
 	$(MKDIR_P) ${DESTMAN}
-	$(INSTALL_DATA) $(srcdir)/tcsh.man ${DESTMAN}/tcsh.${MANSECT}
+	$(INSTALL_DATA) tcsh.man ${DESTMAN}/tcsh.${MANSECT}
 
 # Amiga Unix
 #install.man: tcsh.man
