--- a/meson.build
+++ b/meson.build
@@ -63,7 +63,7 @@
     configure_file(
         input : pc + '.pc.in',
         output : pc + '.pc',
-        install_dir : get_option('datadir') + '/pkgconfig',
+        install_dir : '@TERMUX_PREFIX@/lib/pkgconfig',
         configuration : pc_data,
     )
 endforeach
@@ -102,7 +102,7 @@
         configure_file(
             input : pc + '.pc.in',
             output : pc + '.pc',
-            install_dir : get_option('datadir') + '/pkgconfig',
+            install_dir : '@TERMUX_PREFIX@/lib/pkgconfig',
             configuration : pc_data,
         )
     endforeach
