--- ../meson.build.orig	2021-11-15 20:27:03.284851141 +0000
+++ ./meson.build	2021-11-15 20:31:39.475430037 +0000
@@ -65,7 +65,7 @@
         filebase : pc[0],
         description : pc[0] + ' headers',
         version : pc[1],
-        install_dir : get_option('datadir') + '/pkgconfig',
+        install_dir : get_option('prefix') + '/lib/pkgconfig',
         subdirs : subdir,
     )
 endforeach
@@ -89,7 +89,7 @@
             filebase : pc[0],
             description : pc[0] + ' headers',
             version : pc[1],
-            install_dir : get_option('datadir') + '/pkgconfig'
+            install_dir : get_option('prefix') + '/lib/pkgconfig'
         )
     endforeach
 endif
