diff -uNr libepoxy-1.5.3/meson.build libepoxy-1.5.3.mod/meson.build
--- libepoxy-1.5.3/meson.build	2018-10-05 02:40:48.000000000 +0300
+++ libepoxy-1.5.3.mod/meson.build	2018-10-06 21:13:18.854247131 +0300
@@ -36,7 +36,7 @@
 # support on Linux and Unix
 enable_glx = get_option('glx')
 if enable_glx == 'auto'
-  build_glx = not ['windows', 'darwin', 'android', 'haiku'].contains(host_system)
+  build_glx = not ['windows', 'darwin', 'haiku'].contains(host_system)
 else
   build_glx = enable_glx == 'yes'
 endif
@@ -211,6 +211,7 @@
 wgl_registry = files('registry/wgl.xml')
 
 libepoxy_inc = [
+  include_directories('@TERMUX_PREFIX@/include'),
   include_directories('include'),
   include_directories('src'),
 ]
