diff -uNr libepoxy-1.5.2/src/dispatch_common.c libepoxy-1.5.2.mod/src/dispatch_common.c
--- libepoxy-1.5.2/src/dispatch_common.c	2018-05-14 14:49:17.000000000 +0300
+++ libepoxy-1.5.2.mod/src/dispatch_common.c	2018-06-22 19:50:14.587757143 +0300
@@ -177,7 +177,7 @@
 #define GLX_LIB "/opt/X11/lib/libGL.1.dylib"
 #define OPENGL_LIB "/System/Library/Frameworks/OpenGL.framework/Versions/Current/OpenGL"
 #elif defined(__ANDROID__)
-#define GLX_LIB "libGLESv2.so"
+#define GLX_LIB "libGL.so"
 #else
 #define GLVND_GLX_LIB "libGLX.so.1"
 #define GLX_LIB "libGL.so.1"
