--- a/tasks/toolchains/android.rake
+++ b/tasks/toolchains/android.rake
@@ -252,7 +252,7 @@
   def ar
     case toolchain
     when :gcc   then bin_gcc('ar')
-    when :clang then bin_gcc('ar')
+    when :clang then bin('llvm-ar')
     end
   end
 
@@ -271,7 +271,7 @@
       when /mips/         then flags += %W(-march=mips32)
       end
     when :clang
-      case arch
+      case nil
       when /armeabi-v7a/  then flags += %W(-target armv7-none-linux-androideabi#{v})
       when /armeabi/      then flags += %W(-target armv5te-none-linux-androideabi#{v})
       when /arm64-v8a/    then flags += %W(-target aarch64-none-linux-android#{v})
@@ -282,7 +282,7 @@
       end
     end
 
-    case arch
+    case nil
     when /armeabi-v7a/  then flags += %W(-mfpu=#{armeabi_v7a_mfpu} -mfloat-abi=#{armeabi_v7a_mfloat_abi})
     when /armeabi/      then flags += %W(-mtune=xscale -msoft-float)
     when /arm64-v8a/    then flags += %W()
@@ -254,7 +254,7 @@
     flags += ctarget
     case toolchain
     when :gcc
-    when :clang
+    when nil
       flags += %W(-gcc-toolchain "#{gcc_toolchain_path}" -Wno-invalid-command-line-argument -Wno-unused-command-line-argument)
     end
     flags += %W(-fpic -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes)
@@ -333,8 +331,7 @@
       when /armeabi-v7a/  then flags += %W(-Wl#{no_warn_mismatch})
       end
     when :clang
-      flags += %W(-gcc-toolchain "#{gcc_toolchain_path.to_s}")
-      case arch
+      case nil
       when /armeabi-v7a/  then flags += %W(-target armv7-none-linux-androideabi#{v} -Wl,--fix-cortex-a8#{no_warn_mismatch})
       when /armeabi/      then flags += %W(-target armv5te-none-linux-androideabi#{v})
       when /arm64-v8a/    then flags += %W(-target aarch64-none-linux-android#{v})
