diff -uNr ncurses-6.1-20181117/progs/tic.c ncurses-6.1-20181117.mod/progs/tic.c
--- ncurses-6.1-20181117/progs/tic.c	2018-03-18 02:05:10.000000000 +0200
+++ ncurses-6.1-20181117.mod/progs/tic.c	2019-03-01 20:40:56.193173489 +0200
@@ -386,7 +386,7 @@
 {
     FILE *result = 0;
 
-    _nc_STRCPY(filename, "/tmp/XXXXXX", PATH_MAX);
+    _nc_STRCPY(filename, "@TERMUX_PREFIX@/tmp/XXXXXX", PATH_MAX);
 #if HAVE_MKSTEMP
     {
 	int oldmask = (int) umask(077);
@@ -900,7 +900,7 @@
     } else {
 	if (infodump == TRUE) {
 	    /* captoinfo's no-argument case */
-	    source_file = "/etc/termcap";
+	    source_file = "@TERMUX_PREFIX@/etc/termcap";
 	    if ((termcap = getenv("TERMCAP")) != 0
 		&& (namelst = make_namelist(getenv("TERM"))) != 0) {
 		if (access(termcap, F_OK) == 0) {
diff -uNr ncurses-6.1-20181117/progs/tset.c ncurses-6.1-20181117.mod/progs/tset.c
--- ncurses-6.1-20181117/progs/tset.c	2017-10-08 03:01:29.000000000 +0300
+++ ncurses-6.1-20181117.mod/progs/tset.c	2019-03-01 20:40:15.049476720 +0200
@@ -566,8 +566,8 @@
 	    goto map;
 	}
 #else
-	if ((fp = fopen("/etc/ttytype", "r")) != 0
-	    || (fp = fopen("/etc/ttys", "r")) != 0) {
+	if ((fp = fopen("@TERMUX_PREFIX@/etc/ttytype", "r")) != 0
+	    || (fp = fopen("@TERMUX_PREFIX@/etc/ttys", "r")) != 0) {
 	    char buffer[BUFSIZ];
 	    char *s, *t, *d;
 
