diff -u -r ../ncurses-snapshots-a480458efb0662531287f0c75116c0e91fe235cb/ncurses/tinfo/tinfo_driver.c ./ncurses/tinfo/tinfo_driver.c
--- ../ncurses-snapshots-a480458efb0662531287f0c75116c0e91fe235cb/ncurses/tinfo/tinfo_driver.c	2024-08-31 22:38:42.000000000 +0000
+++ ./ncurses/tinfo/tinfo_driver.c	2024-09-08 13:17:49.112088024 +0000
@@ -497,22 +497,6 @@
 		}
 	    }
 
-	    /*
-	     * Finally, look for environment variables.
-	     *
-	     * Solaris lets users override either dimension with an environment
-	     * variable.
-	     */
-	    if ((value = _nc_getenv_num("LINES")) > 0) {
-		*linep = Min(value, MAX_ENV_LINES);
-		T(("screen size: environment LINES = %d", *linep));
-	    }
-	    if ((value = _nc_getenv_num("COLUMNS")) > 0) {
-		*colp = Min(value, MAX_ENV_COLUMNS);
-		T(("screen size: environment COLUMNS = %d", *colp));
-	    }
-	}
-
 	/* if we can't get dynamic info about the size, use static */
 	if (*linep <= 0) {
 	    *linep = (int) lines;
