diff -u -r ../ncurses-snapshots-a480458efb0662531287f0c75116c0e91fe235cb/ncurses/tinfo/lib_setup.c ./ncurses/tinfo/lib_setup.c
--- ../ncurses-snapshots-a480458efb0662531287f0c75116c0e91fe235cb/ncurses/tinfo/lib_setup.c	2024-08-31 22:38:42.000000000 +0000
+++ ./ncurses/tinfo/lib_setup.c	2024-09-08 13:16:55.592141493 +0000
@@ -561,21 +561,6 @@
 		}
 	    }
 
-	    /*
-	     * Finally, look for environment variables.
-	     *
-	     * Solaris lets users override either dimension with an environment
-	     * variable.
-	     */
-	    if ((value = _nc_getenv_num("LINES")) > 0) {
-		*linep = Min(value, MAX_ENV_LINES);
-		T(("screen size: environment LINES = %d", *linep));
-	    }
-	    if ((value = _nc_getenv_num("COLUMNS")) > 0) {
-		*colp = Min(value, MAX_ENV_COLUMNS);
-		T(("screen size: environment COLUMNS = %d", *colp));
-	    }
-
 	    _nc_default_screensize(termp, linep, colp);
 	} else {
 	    _nc_check_screensize(sp, termp, linep, colp);
