--- a/src/network/getaddrinfo.c
+++ b/src/network/getaddrinfo.c
@@ -121,7 +121,7 @@
 }
 
 #if defined (_WIN32) || defined (__OS2__) \
- || defined (__ANDROID__) || defined (__APPLE__) \
+ || (defined (__ANDROID__) && !defined (__TERMUX__)) || defined (__APPLE__) \
  || defined (__native_client__)
 #warning vlc_getaddr_info_i11e() not implemented!
 int vlc_getaddrinfo_i11e(const char *node, unsigned port,
