diff -u -r ../gcc-6.1.0/gcc/config/arm/arm.opt ./gcc/config/arm/arm.opt
--- ../gcc-6.1.0/gcc/config/arm/arm.opt	2016-01-04 09:30:50.000000000 -0500
+++ ./gcc/config/arm/arm.opt	2016-04-29 04:51:45.773163392 -0400
@@ -193,6 +193,10 @@
 Target Report Mask(INTERWORK)
 Support calls between Thumb and ARM instruction sets.
 
+minline-thumb1-jumptable
+Target Report Var(inline_thumb1_jump_table)
+Inline Thumb1 Jump table code
+
 mtls-dialect=
 Target RejectNegative Joined Enum(tls_type) Var(target_tls_dialect) Init(TLS_GNU)
 Specify thread local storage scheme.
