diff -uNr qt-everywhere-src-5.11.3/qtbase/src/corelib/global/qrandom.cpp qt-everywhere-src-5.11.3.mod/qtbase/src/corelib/global/qrandom.cpp
--- qt-everywhere-src-5.11.3/qtbase/src/corelib/global/qrandom.cpp	2018-11-25 14:51:11.000000000 +0200
+++ qt-everywhere-src-5.11.3.mod/qtbase/src/corelib/global/qrandom.cpp	2020-02-05 19:37:27.660672838 +0200
@@ -218,6 +218,7 @@
 #endif // Q_OS_WINRT
 
     static SystemGenerator &self();
+    typedef quint32 result_type;
     void generate(quint32 *begin, quint32 *end) Q_DECL_NOEXCEPT_EXPR(FillBufferNoexcept);
 
     // For std::mersenne_twister_engine implementations that use something
