diff -uNr qt-everywhere-src-5.11.2/qtlocation/src/plugins/geoservices/geoservices.pro qt-everywhere-src-5.11.2.mod/qtlocation/src/plugins/geoservices/geoservices.pro
--- qt-everywhere-src-5.11.2/qtlocation/src/plugins/geoservices/geoservices.pro	2018-08-31 15:50:06.000000000 +0300
+++ qt-everywhere-src-5.11.2.mod/qtlocation/src/plugins/geoservices/geoservices.pro	2018-09-21 16:25:01.061996160 +0300
@@ -8,11 +8,3 @@
 qtConfig(geoservices_itemsoverlay): SUBDIRS += itemsoverlay
 qtConfig(geoservices_osm): SUBDIRS += osm
 
-qtConfig(geoservices_mapboxgl) {
-    !exists(../../3rdparty/mapbox-gl-native/mapbox-gl-native.pro) {
-        warning("Submodule mapbox-gl-native does not exist. Run 'git submodule update --init' on qtlocation.")
-    } else {
-        SUBDIRS += mapboxgl ../../3rdparty/mapbox-gl-native
-        mapboxgl.depends = ../../3rdparty/mapbox-gl-native
-    }
-}
