diff -uNr qt-everywhere-src-5.11.2/qtsensors/src/plugins/sensors/linux/linux.pro qt-everywhere-src-5.11.2.mod/qtsensors/src/plugins/sensors/linux/linux.pro
--- qt-everywhere-src-5.11.2/qtsensors/src/plugins/sensors/linux/linux.pro	2018-08-28 16:10:03.000000000 +0300
+++ qt-everywhere-src-5.11.2.mod/qtsensors/src/plugins/sensors/linux/linux.pro	2018-09-21 16:25:01.061996160 +0300
@@ -3,7 +3,6 @@
 
 OTHER_FILES = plugin.json
 
-!android:LIBS += -lrt
 HEADERS += linuxsysaccelerometer.h
 SOURCES += linuxsysaccelerometer.cpp \
 main.cpp
