diff -uNr qt-everywhere-src-5.11.1/qtserialport/src/serialport/qserialport_unix.cpp qt-everywhere-src-5.11.1.mod/qtserialport/src/serialport/qserialport_unix.cpp
--- qt-everywhere-src-5.11.1/qtserialport/src/serialport/qserialport_unix.cpp	2018-06-09 14:09:14.000000000 +0300
+++ qt-everywhere-src-5.11.1.mod/qtserialport/src/serialport/qserialport_unix.cpp	2018-09-15 15:10:50.031336089 +0300
@@ -73,7 +73,7 @@
 #  define __ANDROID_API__ 21
 # endif
 
-# if !defined(Q_OS_ANDROID) || (!defined(Q_PROCESSOR_X86) && __ANDROID_API__ < 21)
+# if 0
 struct termios2 {
     tcflag_t c_iflag;       /* input mode flags */
     tcflag_t c_oflag;       /* output mode flags */
