diff -uNr qt-everywhere-src-5.11.2/qtserialport/src/serialport/qserialport_unix.cpp qt-everywhere-src-5.11.2.mod/qtserialport/src/serialport/qserialport_unix.cpp
--- qt-everywhere-src-5.11.2/qtserialport/src/serialport/qserialport_unix.cpp	2018-08-29 13:18:41.000000000 +0300
+++ qt-everywhere-src-5.11.2.mod/qtserialport/src/serialport/qserialport_unix.cpp	2018-09-21 16:25:01.061996160 +0300
@@ -73,7 +73,7 @@
 #  define __ANDROID_API__ 21
 # endif
 
-# if !defined(Q_OS_ANDROID) || (!defined(Q_PROCESSOR_X86) && __ANDROID_API__ < 21)
+# if 0
 struct termios2 {
     tcflag_t c_iflag;       /* input mode flags */
     tcflag_t c_oflag;       /* output mode flags */
