diff -uNr qt-everywhere-src-5.11.1/qtsensors/src/plugins/sensors/linux/linux.pro qt-everywhere-src-5.11.1.mod/qtsensors/src/plugins/sensors/linux/linux.pro
--- qt-everywhere-src-5.11.1/qtsensors/src/plugins/sensors/linux/linux.pro	2018-06-12 12:10:20.000000000 +0300
+++ qt-everywhere-src-5.11.1.mod/qtsensors/src/plugins/sensors/linux/linux.pro	2018-09-13 18:07:43.505306179 +0300
@@ -3,7 +3,6 @@
 
 OTHER_FILES = plugin.json
 
-!android:LIBS += -lrt
 HEADERS += linuxsysaccelerometer.h
 SOURCES += linuxsysaccelerometer.cpp \
 main.cpp
