--- a/bindings/sled-native/Cargo.toml
+++ b/bindings/sled-native/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "sled-native"
-version = "0.31.0"
+version = "@TERMUX_PKG_VERSION@"
 authors = ["Tyler Neely <t@jujit.su>"]
 description = "a C-compatible API for sled"
 license = "Apache-2.0"
@@ -16,4 +16,4 @@
 
 [dependencies]
 libc = "0.2.62"
-sled = {version = "0.31.0", path = "../.."}
+sled = {version = "@TERMUX_PKG_VERSION@", path = "../.."}
