--- ../ec_ui.c.orig	2019-08-27 20:07:21.158532026 +0200
+++ ./src/ec_ui.c	2019-08-27 20:07:36.117841510 +0200
@@ -256,9 +256,6 @@
    if (STAILQ_EMPTY(&messages_queue))
 	return 0; 
 
-   // don't allow the thread to cancel while holding the ui mutex
-   pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, &old);
-
    /* the queue is updated by other threads */
    UI_MSG_LOCK;
    
@@ -280,8 +277,6 @@
    
    UI_MSG_UNLOCK;
 
-   pthread_setcancelstate(PTHREAD_CANCEL_ENABLE, &old);
-
    /* returns the number of displayed messages */
    return i;
    
