--- ./share/etter.conf.v4   2024-05-11 12:53:00.000000000 +0800
+++ ./share/etter.conf.v4   2024-05-11 12:53:00.000000000 +0800
@@ -15,3 +15,3 @@
 [privs]
-ec_uid = 65534                # nobody is the default
-ec_gid = 65534                # nobody is the default
+ #ec_uid = 65534                # nobody is the default
+ #ec_gid = 65534                # nobody is the default
--- ./share/etter.conf.v6   2024-05-11 12:53:00.000000000 +0800
+++ ./share/etter.conf.v6   2024-05-11 12:53:00.000000000 +0800
@@ -15,3 +15,3 @@
 [privs]
-ec_uid = 65534                # nobody is the default
-ec_gid = 65534                # nobody is the default
+ #ec_uid = 65534                # nobody is the default
+ #ec_gid = 65534                # nobody is the default
