--- a/configure.ac	2016-03-23 07:38:40.000000000 -0400
+++ b/configure.ac	2026-07-29 21:57:56.223235725 -0400
@@ -35,7 +35,6 @@
   build/mingw/Makefile
   docs/Makefile
   src/Makefile
-  tests/Makefile
   ticonv.pc
 ])
 
--- a/Makefile.am	2013-04-13 09:26:51.000000000 -0400
+++ b/Makefile.am	2026-07-29 22:06:10.503235725 -0400
@@ -3,7 +3,7 @@
 ACLOCAL_AMFLAGS=-I m4
 
 # Subdirectories to scan
-SUBDIRS = build/mingw src tests
+SUBDIRS = build/mingw src
 
 if USE_DOCGEN
   SUBDIRS += docs
