--- a/configure
+++ b/configure
@@ -16092,7 +16092,7 @@
 fi
 
 
-GLIB_COMPILE_RESOURCES=`$PKG_CONFIG --variable=glib_compile_resources gio-2.0`
+GLIB_COMPILE_RESOURCES=glib-compile-resources
 
 
 
@@ -17237,7 +17237,7 @@
 			 -Werror=missing-prototypes -Werror=nested-externs \
 			 -Werror=implicit-function-declaration \
 			 -Wmissing-declarations -Wno-pointer-sign \
-			 -Werror=format-security -Wbitwise -Wcast-to-as \
+			 -Wno-error=format-security -Wbitwise -Wcast-to-as \
 			 -Wdefault-bitfield-sign -Wdo-while -Wparen-string \
 			 -Wptr-subtraction-blows -Wreturn-void -Wtypesign \
 			 -Wstrict-prototypes -Wno-error=format-nonliteral "
