--- a/introspection/Makefile.in
+++ b/introspection/Makefile.in
@@ -568,7 +568,7 @@
 
 
 @HAVE_INTROSPECTION_TRUE@@HAVE_PYTHON_GI_TRUE@install-data-local:
-@HAVE_INTROSPECTION_TRUE@@HAVE_PYTHON_GI_TRUE@	install -D -m 0444 -t $(DESTDIR)@GIOVERRIDESDIR@ gi/overrides/Gnm.py
+@HAVE_INTROSPECTION_TRUE@@HAVE_PYTHON_GI_TRUE@	install -D -m 0444 -t $(DESTDIR)@GIOVERRIDESDIR@ $(srcdir)/gi/overrides/Gnm.py
 
 @HAVE_INTROSPECTION_TRUE@@HAVE_PYTHON_GI_TRUE@uninstall-local:
 @HAVE_INTROSPECTION_TRUE@@HAVE_PYTHON_GI_TRUE@	rm -f $(DESTDIR)@GIOVERRIDESDIR@/Gnm.py
