--- a/src/grep_proc.cc
+++ b/src/grep_proc.cc
@@ -172,7 +172,7 @@
         perror("setvbuf");
     }
     lnav_log_file
-        = make_optional_from_nullable(fopen("/tmp/lnav.grep.err", "a"));
+        = make_optional_from_nullable(fopen("@TERMUX_PREFIX@/tmp/lnav.grep.err", "a"));
     line_value.reserve(BUFSIZ * 2);
     while (!this->gp_queue.empty()) {
         LineType start_line = this->gp_queue.front().first;
