--- a/src/lnav.cc
+++ b/src/lnav.cc
@@ -149,7 +149,7 @@
 #include "archive_manager.hh"
 
 #ifndef SYSCONFDIR
-#define SYSCONFDIR "/usr/etc"
+#define SYSCONFDIR "@TERMUX_PREFIX@/etc"
 #endif
 
 using namespace std;
@@ -2098,7 +2098,7 @@
 #endif
 
     lnav_data.ld_debug_log_name = "/dev/null";
-    lnav_data.ld_config_paths.emplace_back("/etc/lnav");
+    lnav_data.ld_config_paths.emplace_back("@TERMUX_PREFIX@/etc/lnav");
     lnav_data.ld_config_paths.emplace_back(SYSCONFDIR "/lnav");
     lnav_data.ld_config_paths.emplace_back(lnav::paths::dotlnav());
     while ((c = getopt(argc, argv, "hHarRCc:I:iuf:d:nNqtw:vVW")) != -1) {
@@ -2363,7 +2363,7 @@
      * so that it will try the default path.
      */
     setenv("TERMINFO_DIRS",
-           "/usr/share/terminfo:/lib/terminfo:/usr/share/lib/terminfo",
+           "@TERMUX_PREFIX@/share/terminfo:@TERMUX_PREFIX@/lib/terminfo:@TERMUX_PREFIX@/etc/terminfo",
            0);
 
     {
