--- ./config-files/kdc.conf	2017-03-03 03:36:02.000000000 +0530
+++ ../kdc.conf	2017-04-02 20:29:55.296108766 +0530
@@ -1,14 +1,14 @@
 [kdcdefaults]
-	kdc_listen = 88
-	kdc_tcp_listen = 88
+	kdc_listen = 1088
+	kdc_tcp_listen = 1088
 
 [realms]
 	ATHENA.MIT.EDU = {
-		database_name = /usr/local/var/krb5kdc/principal
-		acl_file = /usr/local/var/krb5kdc/kadm5.acl
-		key_stash_file = /usr/local/var/krb5kdc/.k5.ATHENA.MIT.EDU
-		kdc_listen = 88
-		kdc_tcp_listen = 88
+		database_name = @TERMUX_PREFIX@/var/krb5kdc/principal
+		acl_file = @TERMUX_PREFIX@/var/krb5kdc/kadm5.acl
+		key_stash_file = @TERMUX_PREFIX@/var/krb5kdc/.k5.ATHENA.MIT.EDU
+		kdc_listen = 1088
+		kdc_tcp_listen = 1088
 		max_life = 10h 0m 0s
 		max_renewable_life = 7d 0h 0m 0s
 	}
