--- ./include/osconf.hin	2017-03-03 03:36:02.000000000 +0530
+++ ../osconf.hin	2017-04-03 10:45:03.475924421 +0530
@@ -81,12 +81,12 @@
 
 #define KDC_PORTNAME            "kerberos" /* for /etc/services or equiv. */
 
-#define KRB5_DEFAULT_PORT       88
+#define KRB5_DEFAULT_PORT       1088
 
-#define DEFAULT_KPASSWD_PORT    464
+#define DEFAULT_KPASSWD_PORT    1464
 
-#define DEFAULT_KDC_UDP_PORTLIST "88"
-#define DEFAULT_KDC_TCP_PORTLIST "88"
+#define DEFAULT_KDC_UDP_PORTLIST "1088"
+#define DEFAULT_KDC_TCP_PORTLIST "1088"
 #define DEFAULT_TCP_LISTEN_BACKLOG 5
 
 /*
@@ -94,7 +94,7 @@
  */
 #define DEFAULT_KADM5_KEYTAB    KDC_DIR "/kadm5.keytab"
 #define DEFAULT_KADM5_ACL_FILE  KDC_DIR "/kadm5.acl"
-#define DEFAULT_KADM5_PORT      749 /* assigned by IANA */
+#define DEFAULT_KADM5_PORT      1749 /* assigned by IANA */
 
 #define KRB5_DEFAULT_SUPPORTED_ENCTYPES                 \
     "aes256-cts-hmac-sha1-96:normal "                   \
