https://bugs.gentoo.org/show_bug.cgi?id=448778
--- ./build-tools/krb5-config.in	2012-12-18 02:47:04.000000000 +0000
+++ ./build-tools/krb5-config.in	2012-12-28 07:13:16.582693363 +0000
@@ -217,7 +217,7 @@
 	    -e 's#\$(PROG_RPATH)#'$libdir'#' \
 	    -e 's#\$(PROG_LIBPATH)#'$libdirarg'#' \
 	    -e 's#\$(RPATH_FLAG)#'"$RPATH_FLAG"'#' \
-	    -e 's#\$(LDFLAGS)#'"$LDFLAGS"'#' \
+	    -e 's#\$(LDFLAGS)##' \
 	    -e 's#\$(PTHREAD_CFLAGS)#'"$PTHREAD_CFLAGS"'#' \
 	    -e 's#\$(CFLAGS)##'`
 
