--- ./go/pkg/mod/github.com/!drake!w/corgi@v0.2.4/snippet/history.go	2022-07-16 17:04:11.818280758 +0800
+++ ./go/pkg/mod/github.com/!drake!w/corgi@v0.2.4/snippet/history.go	2022-07-16 17:20:06.508280393 +0800
@@ -22,7 +22,7 @@
 )
 
 var shellType string
-var TempHistFile = "/tmp/corgi.hist"
+var TempHistFile = "@TERMUX_PREFIX@/tmp/corgi.hist"
 
 func getFishHistoryPath(homeDir string) string {
 	var (
