diff -u -r ../libical-3.0.1/src/libical/CMakeLists.txt ./src/libical/CMakeLists.txt
--- ../libical-3.0.1/src/libical/CMakeLists.txt	2017-11-19 15:32:53.000000000 +0100
+++ ./src/libical/CMakeLists.txt	2017-12-09 22:14:42.254898817 +0100
@@ -17,7 +17,7 @@
   include_directories(${ICU_INCLUDE_DIR})
 endif()
 
-set(PACKAGE_DATA_DIR "\\\"${CMAKE_INSTALL_PREFIX}/share/libical\\\"")
+set(PACKAGE_DATA_DIR "${CMAKE_INSTALL_PREFIX}/share/libical")
 add_definitions(-DPACKAGE_DATA_DIR="${PACKAGE_DATA_DIR}")
 
 set(ICALSCRIPTS ${CMAKE_SOURCE_DIR}/scripts/)
