--- a/configure
+++ b/configure
@@ -6033,13 +6033,9 @@ case $target_os in
         striptype=""
         ;;
     android)
-        disable symver
         enable section_data_rel_ro
         add_allcflags -fPIE
         add_ldexeflags -fPIE -pie
-        SLIB_INSTALL_NAME='$(SLIBNAME)'
-        SLIB_INSTALL_LINKS=
-        SHFLAGS='-shared -Wl,-soname,$(SLIBNAME)'
         ;;
     haiku)
         prefix_default="/boot/common"
