diff --git a/CMakeLists.txt b/CMakeLists.txt
index ee4e579..bb12f41 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,7 +9,7 @@ set(KF_MIN_VERSION "6.22.0")
 set(QT_REQUIRED_VERSION "6.9.0")
 set(CMAKE_CXX_SCAN_FOR_MODULES OFF)
 find_package(ECM ${KF_MIN_VERSION} CONFIG REQUIRED)
-set(CMAKE_MODULE_PATH
+list(APPEND CMAKE_MODULE_PATH
     ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules/
     ${ECM_MODULE_PATH}
 )
