diff -uNr xrdp-0.9.8/common/Makefile.am xrdp-0.9.8.mod/common/Makefile.am
--- xrdp-0.9.8/common/Makefile.am	2018-03-27 17:15:16.000000000 +0300
+++ xrdp-0.9.8.mod/common/Makefile.am	2018-12-27 15:41:13.808966307 +0200
@@ -58,6 +58,5 @@
   $(PIXMAN_SOURCES)
 
 libcommon_la_LIBADD = \
-  -lpthread \
   $(OPENSSL_LIBS) \
   $(DLOPEN_LIBS)
diff -uNr xrdp-0.9.8/common/Makefile.in xrdp-0.9.8.mod/common/Makefile.in
--- xrdp-0.9.8/common/Makefile.in	2018-09-25 09:19:19.000000000 +0300
+++ xrdp-0.9.8.mod/common/Makefile.in	2018-12-27 15:41:13.815632972 +0200
@@ -417,7 +417,6 @@
   $(PIXMAN_SOURCES)
 
 libcommon_la_LIBADD = \
-  -lpthread \
   $(OPENSSL_LIBS) \
   $(DLOPEN_LIBS)
 
diff -uNr xrdp-0.9.8/sesman/libscp/Makefile.am xrdp-0.9.8.mod/sesman/libscp/Makefile.am
--- xrdp-0.9.8/sesman/libscp/Makefile.am	2018-03-26 09:14:14.000000000 +0300
+++ xrdp-0.9.8.mod/sesman/libscp/Makefile.am	2018-12-27 15:41:13.815632972 +0200
@@ -42,5 +42,4 @@
   libscp_vX.h
 
 libscp_la_LIBADD = \
-  $(top_builddir)/common/libcommon.la \
-  -lpthread
+  $(top_builddir)/common/libcommon.la
diff -uNr xrdp-0.9.8/sesman/libscp/Makefile.in xrdp-0.9.8.mod/sesman/libscp/Makefile.in
--- xrdp-0.9.8/sesman/libscp/Makefile.in	2018-09-25 09:19:20.000000000 +0300
+++ xrdp-0.9.8.mod/sesman/libscp/Makefile.in	2018-12-27 15:41:13.822299639 +0200
@@ -397,8 +397,7 @@
   libscp_vX.h
 
 libscp_la_LIBADD = \
-  $(top_builddir)/common/libcommon.la \
-  -lpthread
+  $(top_builddir)/common/libcommon.la
 
 all: all-am
 
diff -uNr xrdp-0.9.8/sesman/Makefile.am xrdp-0.9.8.mod/sesman/Makefile.am
--- xrdp-0.9.8/sesman/Makefile.am	2018-09-18 07:21:13.000000000 +0300
+++ xrdp-0.9.8.mod/sesman/Makefile.am	2018-12-27 15:41:13.825632971 +0200
@@ -70,8 +70,7 @@
 xrdp_sesman_LDADD = \
   $(top_builddir)/common/libcommon.la \
   $(top_builddir)/sesman/libscp/libscp.la \
-  $(AUTH_LIB) \
-  -lpthread
+  $(AUTH_LIB)
 
 sesmansysconfdir=$(sysconfdir)/xrdp
 
diff -uNr xrdp-0.9.8/sesman/Makefile.in xrdp-0.9.8.mod/sesman/Makefile.in
--- xrdp-0.9.8/sesman/Makefile.in	2018-09-25 09:19:20.000000000 +0300
+++ xrdp-0.9.8.mod/sesman/Makefile.in	2018-12-27 15:41:13.828966305 +0200
@@ -474,8 +474,7 @@
 xrdp_sesman_LDADD = \
   $(top_builddir)/common/libcommon.la \
   $(top_builddir)/sesman/libscp/libscp.la \
-  $(AUTH_LIB) \
-  -lpthread
+  $(AUTH_LIB)
 
 sesmansysconfdir = $(sysconfdir)/xrdp
 SUBST_VARS = sed \
