Fixes https://github.com/termux/termux-packages/issues/10220.

diff --git a/sesman/sesexec/session.c b/sesman/sesexec/session.c
index a495097..b9a3b5d 100644
--- a/sesman/sesexec/session.c
+++ b/sesman/sesexec/session.c
@@ -419,8 +419,13 @@ prepare_xvnc_xserver_params(const struct session_parameters *s,
                               "-auth", authfile,
                               "-geometry", geometry,
                               "-depth", depth,
-                              "-rfbauth", passwd_file,
                               NULL);
+        if (passwd_file != 0 && *passwd_file != '\0')
+        {
+            list_add_strdup_multi(params,
+                                  "-rfbauth", passwd_file,
+                                  NULL);
+        }
 
         /* additional parameters from sesman.ini file */
         //config_read_xserver_params(SCP_SESSION_TYPE_XVNC,
