diff -uNr xrdp-0.9.8/sesman/Makefile.in xrdp-0.9.8.mod/sesman/Makefile.in
--- xrdp-0.9.8/sesman/Makefile.in	2018-09-25 09:19:20.000000000 +0300
+++ xrdp-0.9.8.mod/sesman/Makefile.in	2018-12-26 18:16:52.382458449 +0200
@@ -474,8 +474,7 @@
 xrdp_sesman_LDADD = \
   $(top_builddir)/common/libcommon.la \
   $(top_builddir)/sesman/libscp/libscp.la \
-  $(AUTH_LIB) \
-  -lpthread
+  $(AUTH_LIB)
 
 sesmansysconfdir = $(sysconfdir)/xrdp
 SUBST_VARS = sed \
