diff -uNr xrdp-0.9.8/Makefile.in xrdp-0.9.8.mod/Makefile.in
--- xrdp-0.9.8/Makefile.in	2018-09-25 09:19:19.000000000 +0300
+++ xrdp-0.9.8.mod/Makefile.in	2018-12-26 19:04:16.010272502 +0200
@@ -167,7 +167,7 @@
 CTAGS = ctags
 CSCOPE = cscope
 DIST_SUBDIRS = common vnc xup mc neutrinordp libxrdp libpainter \
-	librfxcodec xrdp sesman keygen docs instfiles genkeymap \
+	librfxcodec xrdp keygen docs instfiles genkeymap \
 	xrdpapi pkgconfig xrdpvr
 am__DIST_COMMON = $(srcdir)/Makefile.in $(srcdir)/config_ac-h.in \
 	COPYING compile config.guess config.sub install-sh ltmain.sh \
@@ -400,7 +400,6 @@
   $(PAINTERDIR) \
   $(RFXCODECDIR) \
   xrdp \
-  sesman \
   keygen \
   docs \
   instfiles \
