diff -uNr xrdp-0.9.8/sesman/Makefile.am xrdp-0.9.8.mod/sesman/Makefile.am
--- xrdp-0.9.8/sesman/Makefile.am	2018-09-18 07:21:13.000000000 +0300
+++ xrdp-0.9.8.mod/sesman/Makefile.am	2018-12-26 18:16:46.679099072 +0200
@@ -70,8 +70,7 @@
 xrdp_sesman_LDADD = \
   $(top_builddir)/common/libcommon.la \
   $(top_builddir)/sesman/libscp/libscp.la \
-  $(AUTH_LIB) \
-  -lpthread
+  $(AUTH_LIB)
 
 sesmansysconfdir=$(sysconfdir)/xrdp
 
