diff -uNr lastpass-cli-1.3.3/clipboard.c lastpass-cli-1.3.3.mod/clipboard.c
--- lastpass-cli-1.3.3/clipboard.c	2019-04-15 17:15:52.000000000 +0300
+++ lastpass-cli-1.3.3.mod/clipboard.c	2020-02-09 15:52:53.783902088 +0200
@@ -89,13 +89,14 @@
 		char *clipboard_command = getenv("LPASS_CLIPBOARD_COMMAND");
 		if (clipboard_command) {
 			exec_command(clipboard_command);
-			die("Unable to copy contents to clipboard. Please make sure you have `xclip`, `xsel`, `pbcopy`, or `putclip` installed.");
+			die("Unable to copy contents to clipboard. Please make sure you have `termux-api`, `xclip`, `xsel`, `pbcopy`, or `putclip` installed.");
 		} else {
+			execlp("termux-clipboard-set", "termux-clipboard-set", NULL);
 			execlp("xclip", "xclip", "-selection", "clipboard", "-in", NULL);
 			execlp("xsel", "xsel", "--clipboard", "--input", NULL);
 			execlp("pbcopy", "pbcopy", NULL);
 			execlp("putclip", "putclip", "--dos", NULL);
-			die("Unable to copy contents to clipboard. Please make sure you have `xclip`, `xsel`, `pbcopy`, or `putclip` installed.");
+			die("Unable to copy contents to clipboard. Please make sure you have `termux-api`, `xclip`, `xsel`, `pbcopy`, or `putclip` installed.");
 		}
 	}
 	close(pipefd[0]);
diff -uNr lastpass-cli-1.3.3/lpass.1.txt lastpass-cli-1.3.3.mod/lpass.1.txt
--- lastpass-cli-1.3.3/lpass.1.txt	2019-04-15 17:15:52.000000000 +0300
+++ lastpass-cli-1.3.3.mod/lpass.1.txt	2020-02-09 15:54:25.264257729 +0200
@@ -224,9 +224,10 @@
 Clipboard
 ~~~~~~~~~
 Commands that take a '-c' or '--clip' option will copy the output to the
-clipboard, using *xclip*(1) or *xsel*(1) on X11-based systems, *pbcopy*(1)
-on OSX, or *putclip* on Cygwin. The command to be used can be overridden by
-specifying the `LPASS_CLIPBOARD_COMMAND` environment variable.
+clipboard, using *termux-clipboard-set* on Termux, *xclip*(1) or *xsel*(1) on
+X11-based systems, *pbcopy*(1) on OSX, or *putclip* on Cygwin. The command to
+be used can be overridden by specifying the `LPASS_CLIPBOARD_COMMAND` environment
+variable.
 
 Color Output
 ~~~~~~~~~~~~
