diff -u -r ../node-v4.4.3/deps/npm/lib/build.js ./deps/npm/lib/build.js > deps-npm-lib-build.js.patch
--- ../node-v4.4.3/deps/npm/lib/build.js	2016-04-12 19:42:04.000000000 +0000
+++ ./deps/npm/lib/build.js	2016-04-19 09:24:57.262100886 +0000
@@ -208,6 +208,8 @@
 
 function linkBin (from, to, gently, cb) {
   if (process.platform !== "win32") {
+    // Fix shebang paths in binary scripts:
+    require('child_process').spawn('termux-fix-shebang', [from])
     return linkIfExists(from, to, gently, cb)
   } else {
     return cmdShimIfExists(from, to, cb)
