diff -u -r ../node-v8.12.0/deps/uv/uv.gyp ./deps/uv/uv.gyp
--- ../node-v8.12.0/deps/uv/uv.gyp	2018-09-10 18:46:54.000000000 +0000
+++ ./deps/uv/uv.gyp	2018-09-11 20:52:40.320690510 +0000
@@ -39,7 +39,7 @@
     {
       'target_name': 'libuv',
       'type': '<(uv_library)',
-      'include_dirs': [
+      'include_dirs+': [
         'include',
         'src/',
       ],
@@ -54,7 +54,7 @@
           '<@(shared_unix_defines)',
           '<@(shared_zos_defines)',
         ],
-        'include_dirs': [ 'include' ],
+        'include_dirs+': [ 'include' ],
         'conditions': [
           ['OS == "linux"', {
             'defines': [ '_POSIX_C_SOURCE=200112' ],
