diff -u -r ../node-v4.4.3/lib/os.js ./lib/os.js > lib-os.js.patch
--- ../node-v4.4.3/lib/os.js	2016-04-12 19:42:14.000000000 +0000
+++ ./lib/os.js	2016-04-19 09:27:26.330140690 +0000
@@ -36,7 +36,7 @@
     path = process.env.TMPDIR ||
            process.env.TMP ||
            process.env.TEMP ||
-           '/tmp';
+           '@TERMUX_PREFIX@/tmp';
     if (path.length > 1 && path.endsWith('/'))
       path = path.slice(0, -1);
   }
