diff -u -r ../node-v4.4.3/deps/npm/node_modules/osenv/node_modules/os-tmpdir/index.js ./deps/npm/node_modules/osenv/node_modules/os-tmpdir/index.js > npm-tmp-patch.patch
--- ../node-v4.4.3/deps/npm/node_modules/osenv/node_modules/os-tmpdir/index.js	2016-04-12 19:42:05.000000000 +0000
+++ ./deps/npm/node_modules/osenv/node_modules/os-tmpdir/index.js	2016-04-19 09:33:06.883412631 +0000
@@ -14,7 +14,7 @@
 		path = process.env.TMPDIR ||
 			process.env.TMP ||
 			process.env.TEMP ||
-			'/tmp';
+			'@TERMUX_PREFIX@/tmp';
 	}
 
 	if (trailingSlashRe.test(path)) {
