https://github.com/groonga/groonga/commit/89e46ff9437c0ca98e417f1c87372224321b0081

--- a/cmake/FindGroongalz4.cmake
+++ b/cmake/FindGroongalz4.cmake
@@ -29,7 +29,7 @@ endif()
 find_package(lz4 ${find_package_args})
 set(Groongalz4_FOUND ${lz4_FOUND})
 if(Groongalz4_FOUND)
-  if(TARGET LZ4::liblz4_shared)
+  if(TARGET LZ4::lz4_shared)
     set(GRN_LZ4_TARGET LZ4::lz4_shared)
   else()
     set(GRN_LZ4_TARGET LZ4::lz4_static)
