diff -uNr libsearpc-3.1-latest/lib/Makefile.am libsearpc-3.1-latest.mod/lib/Makefile.am
--- libsearpc-3.1-latest/lib/Makefile.am	2019-07-19 10:31:10.000000000 +0300
+++ libsearpc-3.1-latest.mod/lib/Makefile.am	2019-07-25 18:03:05.629491489 +0300
@@ -12,6 +12,6 @@
 
 libsearpc_la_LDFLAGS = -version-info 1:2:0  -no-undefined
 
-libsearpc_la_LIBADD = @GLIB_LIBS@ @JANSSON_LIBS@ -lpthread
+libsearpc_la_LIBADD = @GLIB_LIBS@ @JANSSON_LIBS@
 
 dist_bin_SCRIPTS = searpc-codegen.py
