diff --git a/route_info_android.go b/route_info_android.go
index 9885915a..94c25789 100644
--- a/route_info_android.go
+++ b/route_info_android.go
@@ -5,10 +5,6 @@ import (
 	"os/exec"
 )

-type routeInfo struct {
-	cmds map[string][]string
-}
-
 // NewRouteInfo returns a Android-specific implementation of the RouteInfo
 // interface.
 func NewRouteInfo() (routeInfo, error) {
diff --git a/route_info_default.go b/route_info_default.go
index 6df864ba..db0052fa 100644
--- a/route_info_default.go
+++ b/route_info_default.go
@@ -1,5 +1,5 @@
-//go:build android || nacl || plan9 || js
-// +build android nacl plan9 js
+//go:build nacl || plan9 || js
+// +build nacl plan9 js

 package sockaddr

@@ -8,12 +8,3 @@ func getDefaultIfName() (string, error) {
 	return "", ErrNoInterface
 }

-func NewRouteInfo() (routeInfo, error) {
-	return routeInfo{}, ErrNoRoute
-}
-
-// GetDefaultInterfaceName returns the interface name attached to the default
-// route on the default interface.
-func (ri routeInfo) GetDefaultInterfaceName() (string, error) {
-	return "", ErrNoInterface
-}
