diff -u -r ../elfutils-0.166/libebl/eblwstrtab.c ./libebl/eblwstrtab.c
--- ../elfutils-0.166/libebl/eblwstrtab.c	2015-08-21 08:22:37.000000000 -0400
+++ ./libebl/eblwstrtab.c	2016-05-04 13:11:26.834934000 -0400
@@ -294,6 +294,11 @@
   return newstr;
 }
 
+static wchar_t* wmempcpy(wchar_t* s1, const wchar_t* s2, size_t n)
+{
+    size_t bytes = n * sizeof(wchar_t);
+    return (wchar_t *) (memcpy((char *) s1, (char *) s2, bytes) + bytes);
+}
 
 static void
 copystrings (struct Ebl_WStrent *nodep, wchar_t **freep, size_t *offsetp)
