diff -u -r ../elfutils-0.166/src/elfcompress.c ./src/elfcompress.c
--- ../elfutils-0.166/src/elfcompress.c	2016-03-02 11:25:38.000000000 -0500
+++ ./src/elfcompress.c	2016-05-04 14:33:03.812183632 -0400
@@ -166,7 +166,7 @@
   struct section_pattern *pattern = patterns;
   while (pattern != NULL)
     {
-      if (fnmatch (pattern->pattern, name, FNM_EXTMATCH) == 0)
+      if (fnmatch (pattern->pattern, name, 0) == 0)
 	return true;
       pattern = pattern->next;
     }
