diff -u -r ../elfutils-0.166/libelf/elf_getdata_rawchunk.c ./libelf/elf_getdata_rawchunk.c
--- ../elfutils-0.166/libelf/elf_getdata_rawchunk.c	2015-11-27 08:36:29.000000000 -0500
+++ ./libelf/elf_getdata_rawchunk.c	2016-05-04 13:03:35.578844420 -0400
@@ -41,7 +41,7 @@
 #include "common.h"
 
 Elf_Data *
-elf_getdata_rawchunk (Elf *elf, off_t offset, size_t size, Elf_Type type)
+elf_getdata_rawchunk (Elf *elf, loff_t offset, size_t size, Elf_Type type)
 {
   if (unlikely (elf == NULL))
     return NULL;
