diff -u -r ../elfutils-0.166/src/readelf.c ./src/readelf.c
--- ../elfutils-0.166/src/readelf.c	2016-03-02 11:25:38.000000000 -0500
+++ ./src/readelf.c	2016-05-04 14:13:42.967553723 -0400
@@ -43,6 +43,7 @@
 #include <sys/param.h>
 #include <sys/stat.h>
 #include <signal.h>
+#include <qsort_r.h>
 
 #include <system.h>
 #include "../libelf/libelfP.h"
@@ -6128,7 +6129,7 @@
 
       printf ("           %*s%-20s (%s) %s\n",
 	      (int) (level * 2), "", dwarf_attr_name (attr),
-	      dwarf_form_name (form), nl_langinfo (flag ? YESSTR : NOSTR));
+	      dwarf_form_name (form), flag ? "yes" : "no");
       break;
 
     case DW_FORM_flag_present:
@@ -6136,7 +6137,7 @@
 	break;
       printf ("           %*s%-20s (%s) %s\n",
 	      (int) (level * 2), "", dwarf_attr_name (attr),
-	      dwarf_form_name (form), nl_langinfo (YESSTR));
+	      dwarf_form_name (form), "yes");
       break;
 
     case DW_FORM_exprloc:
@@ -7646,7 +7647,7 @@
 		      if (readp + 1 > readendp)
 			goto invalid_data;
 		      val = *readp++;
-		      printf (" %s", nl_langinfo (val != 0 ? YESSTR : NOSTR));
+		      printf (" %s", val != 0 ? "yes" : "no");
 		      break;
 
 		    case DW_FORM_string:
