See https://github.com/libffi/libffi/pull/265

diff -u -r ../libffi-60e4250a77eb3fde500bfd68ec40519fe34b21bd/configure.ac ./configure.ac
--- ../libffi-60e4250a77eb3fde500bfd68ec40519fe34b21bd/configure.ac	2016-09-04 09:34:18.000000000 -0400
+++ ./configure.ac	2016-09-14 06:43:08.916379800 -0400
@@ -197,7 +197,7 @@
                  [Cannot use PROT_EXEC on this target, so, we revert to
                    alternative means])
      ;;
-     *-apple-darwin* | *-*-freebsd* | *-*-kfreebsd* | *-*-openbsd* | *-pc-solaris*)
+     *-apple-darwin* | *-*-freebsd* | *-*-kfreebsd* | *-*-openbsd* | *-pc-solaris* | *-linux-android*)
        AC_DEFINE(FFI_MMAP_EXEC_WRIT, 1,
                  [Cannot use malloc on this target, so, we revert to
                    alternative means])
