diff -u -r ../libffi-3.1/libffi.pc.in ./libffi.pc.in
--- ../libffi-3.1/libffi.pc.in	2014-04-25 19:45:13.000000000 +0200
+++ ./libffi.pc.in	2014-07-13 22:37:09.818674529 +0200
@@ -2,7 +2,7 @@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 toolexeclibdir=@toolexeclibdir@
-includedir=${libdir}/@PACKAGE_NAME@-@PACKAGE_VERSION@/include
+includedir=@includedir@
 
 Name: @PACKAGE_NAME@
 Description: Library supporting Foreign Function Interfaces
