See https://github.com/libffi/libffi/pull/265

diff -u -r ../libffi-60e4250a77eb3fde500bfd68ec40519fe34b21bd/src/closures.c ./src/closures.c
--- ../libffi-60e4250a77eb3fde500bfd68ec40519fe34b21bd/src/closures.c	2016-09-04 09:34:18.000000000 -0400
+++ ./src/closures.c	2016-09-14 16:51:05.956548790 -0400
@@ -55,7 +55,7 @@
 #endif
 
 #if FFI_MMAP_EXEC_WRIT && !defined FFI_MMAP_EXEC_SELINUX
-# ifdef __linux__
+# if defined(__linux__) && !defined(__ANDROID__)
 /* When defined to 1 check for SELinux and if SELinux is active,
    don't attempt PROT_EXEC|PROT_WRITE mapping at all, as that
    might cause audit messages.  */
