diff -uNr newsboat-2.17/rust/libnewsboat/Cargo.toml newsboat-2.17.mod/rust/libnewsboat/Cargo.toml
--- newsboat-2.17/rust/libnewsboat/Cargo.toml	2019-09-22 17:19:19.000000000 +0300
+++ newsboat-2.17.mod/rust/libnewsboat/Cargo.toml	2019-09-24 20:57:10.680900795 +0300
@@ -19,7 +19,6 @@
 unicode-segmentation = "1"
 curl-sys = "0.4.5"
 libc = "0.2"
-gettext-rs = "0.4.1"
 natord = "1.0.9"
 
 # We don't use the following crates, but we pin their versions to make sure
@@ -37,12 +36,6 @@
 #   I want as little dependencies as practically possible.
 default-features = false
 
-[dependencies.gettext-sys]
-version = "0.19.8"
-# Don't let the crate build its own copy of gettext; force it to use the one
-# built into glibc.
-features = [ "gettext-system" ]
-
 [dev-dependencies]
 tempfile = "3"
 proptest = "0.7"
diff -uNr newsboat-2.17/rust/libnewsboat/src/cliargsparser.rs newsboat-2.17.mod/rust/libnewsboat/src/cliargsparser.rs
--- newsboat-2.17/rust/libnewsboat/src/cliargsparser.rs	2019-09-22 17:19:19.000000000 +0300
+++ newsboat-2.17.mod/rust/libnewsboat/src/cliargsparser.rs	2019-09-24 20:56:44.030818719 +0300
@@ -1,5 +1,4 @@
 use clap::{App, Arg};
-use gettextrs::gettext;
 use libc::{EXIT_FAILURE, EXIT_SUCCESS};
 use std::path::PathBuf;
 
@@ -282,7 +281,7 @@
                 }
                 _ => {
                     args.display_msg = fmt!(
-                        &gettext("%s: %s: invalid loglevel value"),
+                        "%s: %s: invalid loglevel value",
                         &opts[0],
                         log_level_str
                     );
diff -ruw newsboat-2.20.1/rust/regex-rs/Cargo.toml src/rust/regex-rs/Cargo.toml
--- newsboat-2.20.1/rust/regex-rs/Cargo.toml	2020-06-24 09:51:43.000000000 +0000
+++ src/rust/regex-rs/Cargo.toml	2020-08-09 15:53:33.917513569 +0000
@@ -12,4 +12,3 @@
 
 bitflags = "1.0"
 libc = ">=0.2.69"
-gettext-rs = "0.4.1"
diff -ruw newsboat-2.20.1/rust/regex-rs/src/lib.rs src/rust/regex-rs/src/lib.rs
--- newsboat-2.20.1/rust/regex-rs/src/lib.rs	2020-06-24 09:51:43.000000000 +0000
+++ src/rust/regex-rs/src/lib.rs	2020-08-09 15:55:06.893777550 +0000
@@ -29,7 +29,6 @@
 //! ```
 
 use bitflags::bitflags;
-use gettextrs::gettext;
 use libc::{regcomp, regerror, regex_t, regexec, regfree, regmatch_t};
 use std::ffi::{CString, OsString};
 use std::mem;
@@ -155,12 +154,12 @@
             } else {
                 match regex_error_to_str(errcode, &regex) {
                     Some(regcomp_errmsg) => {
-                        let msg = fmt!(&gettext("regcomp returned code %i"), errcode);
+                        let msg = fmt!("regcomp returned code %i", errcode);
                         let msg = format!("{}: {}", msg, regcomp_errmsg);
                         Err(msg)
                     }
 
-                    None => Err(fmt!(&gettext("regcomp returned code %i"), errcode)),
+                    None => Err(fmt!("regcomp returned code %i", errcode)),
                 }
             }
         }
@@ -240,11 +239,11 @@
             _ => unsafe {
                 match regex_error_to_str(errcode, &self.regex) {
                     Some(regexec_errmsg) => {
-                        let msg = fmt!(&gettext("regexec returned code %i"), errcode);
+                        let msg = fmt!("regexec returned code %i", errcode);
                         let msg = format!("{}: {}", msg, regexec_errmsg);
                         Err(msg)
                     }
-                    None => Err(fmt!(&gettext("regexec returned code %i"), errcode)),
+                    None => Err(fmt!("regexec returned code %i", errcode)),
                 }
             },
         }
