diff -u -r ../newsboat-2.16.1/Makefile ./Makefile
--- ../newsboat-2.16.1/Makefile	2019-06-26 17:55:21.000000000 +0000
+++ ./Makefile	2019-08-06 17:59:51.574712056 +0000
@@ -18,7 +18,7 @@
 
 WARNFLAGS=-Werror -Wall -Wextra -Wunreachable-code
 INCLUDES=-Iinclude -Istfl -Ifilter -I. -Irss
-BARE_CXXFLAGS=-std=c++11 -O2 -ggdb $(INCLUDES)
+BARE_CXXFLAGS=-std=c++11 -O2 $(INCLUDES)
 LDFLAGS+=-L.
 
 PACKAGE=newsboat
@@ -32,7 +32,7 @@
 LDFLAGS+=-fprofile-arcs -ftest-coverage
 endif
 
-CXXFLAGS:=$(BARE_CXXFLAGS) $(WARNFLAGS) $(DEFINES) $(CXXFLAGS)
+CXXFLAGS:=$(BARE_CXXFLAGS) $(WARNFLAGS) $(DEFINES) $(CXXFLAGS) $(CPPFLAGS)
 CXXFLAGS_FOR_BUILD?=$(CXXFLAGS)
 
 LIB_SOURCES:=$(shell cat mk/libboat.deps)
@@ -46,7 +46,7 @@
 NEWSBOAT=newsboat
 NEWSBOAT_SOURCES:=$(shell cat mk/newsboat.deps)
 NEWSBOAT_OBJS:=$(patsubst %.cpp,%.o,$(NEWSBOAT_SOURCES))
-NEWSBOAT_LIBS=-lboat -lnewsboat -lfilter -lpthread -lrsspp -ldl
+NEWSBOAT_LIBS=-lboat -lnewsboat -lfilter -lrsspp -landroid-glob
 
 RSSPPLIB_SOURCES=$(sort $(wildcard rss/*.cpp))
 RSSPPLIB_OBJS=$(patsubst rss/%.cpp,rss/%.o,$(RSSPPLIB_SOURCES))
@@ -75,7 +75,7 @@
 PODBOAT=podboat
 PODBOAT_SOURCES:=$(shell cat mk/podboat.deps)
 PODBOAT_OBJS:=$(patsubst %.cpp,%.o,$(PODBOAT_SOURCES))
-PODBOAT_LIBS=-lboat -lnewsboat -lpthread -ldl
+PODBOAT_LIBS=-lboat -lnewsboat -ldl
 
 ifeq (, $(filter Linux GNU GNU/%, $(shell uname -s)))
 NEWSBOAT_LIBS+=-liconv -lintl
