diff -uNr newsboat-2.20/Makefile newsboat-2.20.mod/Makefile
--- newsboat-2.20/Makefile	2020-06-20 18:54:45.000000000 +0300
+++ newsboat-2.20.mod/Makefile	2020-06-21 19:53:36.104143404 +0300
@@ -18,7 +18,7 @@
 
 WARNFLAGS=-Werror -Wall -Wextra -Wunreachable-code
 INCLUDES=-Iinclude -Istfl -Ifilter -I. -Irss
-BARE_CXXFLAGS=-std=c++11 -O2 -ggdb $(INCLUDES)
+BARE_CXXFLAGS=-std=c++11 -O2 $(INCLUDES)
 LDFLAGS+=-L.
 
 PACKAGE=newsboat
@@ -46,7 +46,7 @@
 NEWSBOAT=newsboat
 NEWSBOAT_SOURCES:=$(shell cat mk/newsboat.deps)
 NEWSBOAT_OBJS:=$(patsubst %.cpp,%.o,$(NEWSBOAT_SOURCES))
-NEWSBOAT_LIBS=-lboat -lnewsboat -lfilter -lpthread -lrsspp -ldl
+NEWSBOAT_LIBS=-lboat -lnewsboat -lfilter -lrsspp -ldl
 
 RSSPPLIB_SOURCES=$(sort $(wildcard rss/*.cpp))
 RSSPPLIB_OBJS=$(patsubst rss/%.cpp,rss/%.o,$(RSSPPLIB_SOURCES))
@@ -75,7 +75,7 @@
 PODBOAT=podboat
 PODBOAT_SOURCES:=$(shell cat mk/podboat.deps)
 PODBOAT_OBJS:=$(patsubst %.cpp,%.o,$(PODBOAT_SOURCES))
-PODBOAT_LIBS=-lboat -lnewsboat -lfilter -lpthread -ldl
+PODBOAT_LIBS=-lboat -lnewsboat -lfilter -ldl
 
 ifeq (, $(filter Linux GNU GNU/%, $(shell uname -s)))
 NEWSBOAT_LIBS+=-liconv -lintl
