--- ./rust/libnewsboat/Cargo.toml.orig	2021-05-17 16:44:12.665188034 +0000
+++ ./rust/libnewsboat/Cargo.toml	2021-05-17 16:44:22.377378220 +0000
@@ -17,7 +17,7 @@
 backtrace = "= 0.3"
 unicode-width = "0.1.8"
 nom = "6"
-libc = "0.2"
+libc = "0.2.94"
 natord = "1.0.9"
 
 [dependencies.clap]
@@ -31,12 +31,6 @@
 #   I want as little dependencies as practically possible.
 default-features = false
 
-[dependencies.gettext-rs]
-version = "0.6.0"
-# Don't let the crate build its own copy of gettext; force it to use the one
-# built into glibc.
-features = [ "gettext-system" ]
-
 [dependencies.curl-sys]
 version = "0.4.5"
 # This disables the only default feature, "ssl". It doesn't affect the
