--- a/rust/libnewsboat/Cargo.toml
+++ b/rust/libnewsboat/Cargo.toml
@@ -23,12 +23,6 @@
 unicode-segmentation = "1.13.3"
 regex = "1.12.3"
 
-[dependencies.gettext-rs]
-version = "0.8.0"
-# Don't let the crate build its own copy of gettext; force it to use the one
-# built into glibc.
-features = [ "gettext-system" ]
-
 [dependencies.curl-sys]
 version = "0.4.88"
 # This disables the only default feature, "ssl". It doesn't affect the
