diff -u -r ../newsboat-2.21/rust/libnewsboat/Cargo.toml ./rust/libnewsboat/Cargo.toml
--- ../newsboat-2.21/rust/libnewsboat/Cargo.toml	2020-09-20 19:14:02.000000000 +0000
+++ ./rust/libnewsboat/Cargo.toml	2020-10-21 22:26:52.328755304 +0000
@@ -18,8 +18,7 @@
 unicode-width = "0.1.8"
 nom = "5"
 curl-sys = "0.4.5"
-libc = "0.2"
-gettext-rs = "0.5.0"
+libc = { git = "https://github.com/fornwall/libc", branch = "android-lockf-operations" }
 natord = "1.0.9"
 lazy_static = "1.4.0"
 
@@ -34,12 +33,6 @@
 #   I want as little dependencies as practically possible.
 default-features = false
 
-[dependencies.gettext-sys]
-version = "0.19.9"
-# Don't let the crate build its own copy of gettext; force it to use the one
-# built into glibc.
-features = [ "gettext-system" ]
-
 [dev-dependencies]
 tempfile = "3"
 # 0.9.6 fixes build failures on Nightly >=2020-04-08: https://github.com/newsboat/newsboat/issues/870
