--- a/rust/libnewsboat/Cargo.toml
+++ b/rust/libnewsboat/Cargo.toml
@@ -17,18 +17,12 @@
 backtrace = "= 0.3"
 unicode-width = "0.1.10"
 nom = "7"
-libc = "0.2"
+libc = "0.2.94"
 natord = "1.0.9"
 md5 = "0.7.0"
 
 lexopt = "0.2.1"
 
-[dependencies.gettext-rs]
-version = "0.7.0"
-# Don't let the crate build its own copy of gettext; force it to use the one
-# built into glibc.
-features = [ "gettext-system" ]
-
 [dependencies.curl-sys]
 version = "0.4.5"
 # This disables the only default feature, "ssl". It doesn't affect the
