diff -u -r ../newsboat-2.37/rust/libnewsboat/Cargo.toml ./rust/libnewsboat/Cargo.toml
--- ../newsboat-2.37/rust/libnewsboat/Cargo.toml	2024-09-22 15:05:50.000000000 +0000
+++ ./rust/libnewsboat/Cargo.toml	2024-09-24 19:24:27.988264342 +0000
@@ -22,12 +22,6 @@
 chrono = "0.4.34"
 unicode-segmentation = "1.12.0"
 
-[dependencies.gettext-rs]
-version = "0.7.1"
-# Don't let the crate build its own copy of gettext; force it to use the one
-# built into glibc.
-features = [ "gettext-system" ]
-
 [dependencies.curl-sys]
 version = "0.4.75"
 # This disables the only default feature, "ssl". It doesn't affect the
