diff -u -r ../newsboat-2.20.1/rust/regex-rs/Cargo.toml ./rust/regex-rs/Cargo.toml
--- ../newsboat-2.20.1/rust/regex-rs/Cargo.toml	2020-06-24 09:51:43.000000000 +0000
+++ ./rust/regex-rs/Cargo.toml	2020-08-12 07:44:18.635260000 +0000
@@ -11,5 +11,4 @@
 strprintf = { path="../strprintf" }
 
 bitflags = "1.0"
-libc = ">=0.2.69"
-gettext-rs = "0.4.1"
+libc = { git = "https://github.com/fornwall/libc", branch = "android-regex" }
