diff -uNr newsboat-2.24/rust/libnewsboat/Cargo.toml newsboat-2.24.mod/rust/libnewsboat/Cargo.toml
--- newsboat-2.24/rust/libnewsboat/Cargo.toml	2021-06-20 20:40:09.000000000 +0300
+++ newsboat-2.24.mod/rust/libnewsboat/Cargo.toml	2021-07-17 22:24:18.930766657 +0300
@@ -17,7 +17,7 @@
 backtrace = "= 0.3"
 unicode-width = "0.1.8"
 nom = "6"
-libc = "0.2"
+libc = "0.2.94"
 natord = "1.0.9"
 
 [dependencies.clap]
@@ -31,12 +31,6 @@
 #   I want as little dependencies as practically possible.
 default-features = false
 
-[dependencies.gettext-rs]
-version = "0.7.0"
-# Don't let the crate build its own copy of gettext; force it to use the one
-# built into glibc.
-features = [ "gettext-system" ]
-
 [dependencies.curl-sys]
 version = "0.4.5"
 # This disables the only default feature, "ssl". It doesn't affect the
