diff -u -r ../newsboat-2.10.2/src/oldreader_api.cpp ./src/oldreader_api.cpp
--- ../newsboat-2.10.2/src/oldreader_api.cpp	2017-12-25 18:49:35.000000000 +0100
+++ ./src/oldreader_api.cpp	2017-12-26 23:50:14.802426368 +0100
@@ -134,7 +134,7 @@
 
 			json_object_object_get_ex(sub, "categories", &node);
 			struct array_list * categories = json_object_get_array(node);
-			for (int i = 0; i < array_list_length(categories); i++) {
+			for (size_t i = 0; i < array_list_length(categories); i++) {
 				json_object* cat = json_object_array_get_idx(node, i);
 				json_object* label_node {};
 				json_object_object_get_ex(cat, "label", &label_node);
