diff -u -r ../newsboat-2.18/src/rssparser.cpp ./src/rssparser.cpp
--- ../newsboat-2.18/src/rssparser.cpp	2019-12-22 11:56:20.000000000 +0000
+++ ./src/rssparser.cpp	2020-01-02 20:37:50.905020000 +0000
@@ -305,7 +305,7 @@
 {
 	std::string buf = utils::retrieve_url(uri, cfgcont);
 
-	char* argv[4] = {const_cast<char*>("/bin/sh"),
+	char* argv[4] = {const_cast<char*>("@TERMUX_PREFIX@/bin/sh"),
 			const_cast<char*>("-c"),
 			const_cast<char*>(filter.c_str()),
 			nullptr
