diff -u -r ../newsboat-2.10.2/src/inoreader_api.cpp ./src/inoreader_api.cpp
--- ../newsboat-2.10.2/src/inoreader_api.cpp	2017-12-25 18:49:35.000000000 +0100
+++ ./src/inoreader_api.cpp	2017-12-27 00:18:36.167450052 +0100
@@ -136,7 +136,7 @@
 
 		json_object_object_get_ex(sub, "categories", &node);
 		struct array_list * categories = json_object_get_array(node);
-		for (int i = 0; i < array_list_length(categories); i++) {
+		for (size_t i = 0; i < array_list_length(categories); i++) {
 			json_object* cat = json_object_array_get_idx(node, i);
 			json_object* label_node {};
 			json_object_object_get_ex(cat, "label", &label_node);
