--- a/setup.py
+++ b/setup.py
@@ -15,10 +15,10 @@
 
 MACOS_FLAG = ['-mmacosx-version-min=10.7']
 FLAGS_POSIX = [
-    '-fPIC', '-std=gnu++0x', '-Wall', '-Wno-parentheses', '-Werror=switch',
+    '-fPIC', '-Wall', '-Wno-parentheses', '-Werror=switch',
 ]
 FLAGS_CLANG = ['-c', '-O3'] + FLAGS_POSIX + ['-stdlib=libc++']
-LFLAGS_POSIX = ['-fPIC',  '-lstdc++']
+LFLAGS_POSIX = ['-fPIC']
 LFLAGS_CLANG = ['-fPIC', '-stdlib=libc++']
 
 sources = ['_sass.c']
