--- a/src/x11_init.c
+++ b/src/x11_init.c
@@ -606,7 +606,7 @@
 //
 static GLFWbool initExtensions(void)
 {
-#if defined(__OpenBSD__) || defined(__NetBSD__)
+#if defined(__OpenBSD__) || defined(__NetBSD__) || defined(__ANDROID__)
     _glfw.x11.vidmode.handle = _glfwPlatformLoadModule("libXxf86vm.so");
 #else
     _glfw.x11.vidmode.handle = _glfwPlatformLoadModule("libXxf86vm.so.1");
@@ -630,7 +630,7 @@
 
 #if defined(__CYGWIN__)
     _glfw.x11.xi.handle = _glfwPlatformLoadModule("libXi-6.so");
-#elif defined(__OpenBSD__) || defined(__NetBSD__)
+#elif defined(__OpenBSD__) || defined(__NetBSD__) || defined(__ANDROID__)
     _glfw.x11.xi.handle = _glfwPlatformLoadModule("libXi.so");
 #else
     _glfw.x11.xi.handle = _glfwPlatformLoadModule("libXi.so.6");
@@ -662,7 +662,7 @@
 
 #if defined(__CYGWIN__)
     _glfw.x11.randr.handle = _glfwPlatformLoadModule("libXrandr-2.so");
-#elif defined(__OpenBSD__) || defined(__NetBSD__)
+#elif defined(__OpenBSD__) || defined(__NetBSD__) || defined(__ANDROID__)
     _glfw.x11.randr.handle = _glfwPlatformLoadModule("libXrandr.so");
 #else
     _glfw.x11.randr.handle = _glfwPlatformLoadModule("libXrandr.so.2");
@@ -756,7 +756,7 @@
 
 #if defined(__CYGWIN__)
     _glfw.x11.xcursor.handle = _glfwPlatformLoadModule("libXcursor-1.so");
-#elif defined(__OpenBSD__) || defined(__NetBSD__)
+#elif defined(__OpenBSD__) || defined(__NetBSD__) || defined(__ANDROID__)
     _glfw.x11.xcursor.handle = _glfwPlatformLoadModule("libXcursor.so");
 #else
     _glfw.x11.xcursor.handle = _glfwPlatformLoadModule("libXcursor.so.1");
@@ -779,7 +779,7 @@
 
 #if defined(__CYGWIN__)
     _glfw.x11.xinerama.handle = _glfwPlatformLoadModule("libXinerama-1.so");
-#elif defined(__OpenBSD__) || defined(__NetBSD__)
+#elif defined(__OpenBSD__) || defined(__NetBSD__) || defined(__ANDROID__)
     _glfw.x11.xinerama.handle = _glfwPlatformLoadModule("libXinerama.so");
 #else
     _glfw.x11.xinerama.handle = _glfwPlatformLoadModule("libXinerama.so.1");
@@ -834,7 +834,7 @@
     {
 #if defined(__CYGWIN__)
         _glfw.x11.x11xcb.handle = _glfwPlatformLoadModule("libX11-xcb-1.so");
-#elif defined(__OpenBSD__) || defined(__NetBSD__)
+#elif defined(__OpenBSD__) || defined(__NetBSD__) || defined(__ANDROID__)
         _glfw.x11.x11xcb.handle = _glfwPlatformLoadModule("libX11-xcb.so");
 #else
         _glfw.x11.x11xcb.handle = _glfwPlatformLoadModule("libX11-xcb.so.1");
@@ -849,7 +849,7 @@
 
 #if defined(__CYGWIN__)
     _glfw.x11.xrender.handle = _glfwPlatformLoadModule("libXrender-1.so");
-#elif defined(__OpenBSD__) || defined(__NetBSD__)
+#elif defined(__OpenBSD__) || defined(__NetBSD__) || defined(__ANDROID__)
     _glfw.x11.xrender.handle = _glfwPlatformLoadModule("libXrender.so");
 #else
     _glfw.x11.xrender.handle = _glfwPlatformLoadModule("libXrender.so.1");
@@ -878,7 +878,7 @@
 
 #if defined(__CYGWIN__)
     _glfw.x11.xshape.handle = _glfwPlatformLoadModule("libXext-6.so");
-#elif defined(__OpenBSD__) || defined(__NetBSD__)
+#elif defined(__OpenBSD__) || defined(__NetBSD__) || defined(__ANDROID__)
     _glfw.x11.xshape.handle = _glfwPlatformLoadModule("libXext.so");
 #else
     _glfw.x11.xshape.handle = _glfwPlatformLoadModule("libXext.so.6");
