diff --git a/lang/LangSource/Bison/lang11d_tab.cpp b/lang/LangSource/Bison/lang11d_tab.cpp
index ebafb0d65..76340a9cb 100644
--- a/lang/LangSource/Bison/lang11d_tab.cpp
+++ b/lang/LangSource/Bison/lang11d_tab.cpp
@@ -81,7 +81,9 @@
 #include "PredefinedSymbols.h"
 #include "SimpleStack.h"
 
+#if !defined(__ANDROID__)
 void bcopy(void *src, void *dst, size_t size) ;
+#endif
 int yyparse();
 extern bool compilingCmdLine;
 extern LongStack generatorStack;
