diff --git a/lang/LangPrimSource/PyrPrimitive.cpp b/lang/LangPrimSource/PyrPrimitive.cpp
index 527369718..7167eaea2 100644
--- a/lang/LangPrimSource/PyrPrimitive.cpp
+++ b/lang/LangPrimSource/PyrPrimitive.cpp
@@ -4016,7 +4016,7 @@ void initPrimitives() {
     initMIDIPrimitives();
 #endif
 
-#if defined __linux__
+#if defined __linux__ && defined(HAVE_LID)
     void initLIDPrimitives();
     initLIDPrimitives();
 #endif
