diff --git a/lang/LangSource/Bison/lang11d b/lang/LangSource/Bison/lang11d
index b441b7a13..fd03d0cd9 100644
--- a/lang/LangSource/Bison/lang11d
+++ b/lang/LangSource/Bison/lang11d
@@ -26,7 +26,9 @@
 #include "PredefinedSymbols.h"
 #include "SimpleStack.h"
 
+#if !defined(__ANDROID__)
 void bcopy(void *src, void *dst, size_t size) ;
+#endif
 int yyparse();
 extern bool compilingCmdLine;
 extern LongStack generatorStack;
