diff -u -r ../mlocate-0.26/src/conf.c ./src/conf.c
--- ../mlocate-0.26/src/conf.c	2011-12-12 00:13:18.000000000 +0100
+++ ./src/conf.c	2014-07-02 03:34:47.000000000 +0200
@@ -646,7 +646,7 @@
     error (EXIT_FAILURE, 0, _("unexpected operand on command line"));
   if (conf_scan_root == NULL)
     {
-      static char root[] = "/";
+      static char root[] = "@TERMUX_BASE_DIR@/";
 
       conf_scan_root = root;
     }
