diff -uNr libXt-1.2.0/src/Shell.c libXt-1.2.0.mod/src/Shell.c
--- libXt-1.2.0/src/Shell.c	2019-06-22 02:29:55.000000000 +0300
+++ libXt-1.2.0.mod/src/Shell.c	2019-06-29 23:41:30.634530801 +0300
@@ -1961,9 +1961,7 @@
 	    if (_XtWaitForSomething (app,
 				     FALSE, TRUE, TRUE, TRUE,
 				     TRUE,
-#ifdef XTHREADS
 				     FALSE,
-#endif
 				     &timeout) != -1) {
 		while (XCheckIfEvent(XtDisplay(w),event,isMine,(char*)&q)) {
 		    if (q.done) return TRUE;
