--- a/Makethird
+++ b/Makethird
@@ -242,7 +242,7 @@
 THIRD_LIBS += $(EXTRACT_LIBS)
 THIRD_SRC += $(EXTRACT_SRC)
 $(OUT)/thirdparty/extract/%.o: thirdparty/extract/%.c
-	$(CC_CMD) $(WARNING_CFLAGS) $(LIB_CFLAGS) $(EXTRACT_CFLAGS) $(ZLIB_CFLAGS) $(EXTRACT_BUILD_CFLAGS)
+	$(CC_CMD) $(WARNING_CFLAGS) $(LIB_CFLAGS) $(EXTRACT_CFLAGS) $(SYS_ZLIB_CFLAGS) $(EXTRACT_BUILD_CFLAGS)
 
 # The first python3 command in the next two rules returns 0 (success) iff
 # python3 exists and is version 3.6 or later.
