diff -u -r ../libxslt-1.1.29/xslt-config.in ./xslt-config.in
--- ../libxslt-1.1.29/xslt-config.in	2015-05-10 10:11:30.000000000 -0400
+++ ./xslt-config.in	2016-07-19 08:16:16.959043088 -0400
@@ -89,9 +89,9 @@
     shift
 done
 
-the_libs="@XSLT_LIBDIR@ @XSLT_LIBS@ @EXTRA_LIBS@"
+the_libs="@XSLT_LIBS@ @EXTRA_LIBS@"
 if test "$includedir" != "/usr/include"; then
-    the_flags="$the_flags -I$includedir `@XML_CONFIG@ --cflags`"
+    the_flags="$the_flags -I$includedir `@TERMUX_PREFIX@/bin/xml2-config --cflags`"
 else
     the_flags="$the_flags `@XML_CONFIG@ --cflags`"
 fi
