--- a/src/libusbmuxd.c
+++ b/src/libusbmuxd.c
@@ -78,7 +78,7 @@ extern int _NSGetExecutablePath(char* buf, uint32_t* bufsize);
 #include <pthread.h>
 #define EVENT_SIZE  (sizeof (struct inotify_event))
 #define EVENT_BUF_LEN (1024 * (EVENT_SIZE + 16))
-#define USBMUXD_DIRNAME "/var/run"
+#define USBMUXD_DIRNAME "@TERMUX_PREFIX@/var/run"
 #define USBMUXD_SOCKET_NAME "usbmuxd"
 static int use_inotify = 1;
 #endif /* HAVE_INOTIFY */
