--- ./toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/cmath.orig	2023-01-04 23:26:10.402017181 +0100
+++ ./toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/cmath	2023-01-04 23:27:42.228693072 +0100
@@ -301,7 +301,7 @@
 */
 
 #include <__config>
-#include <math.h>
+#include <c++/v1/math.h>
 #include "version"
 #include <type_traits>
 
--- ./toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/math.h.orig	2023-01-04 22:39:35.231736330 +0100
+++ ./toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/math.h	2023-01-04 22:41:40.911748958 +0100
@@ -300,7 +300,7 @@
 #define _LIBCPP_STDLIB_INCLUDE_NEXT
 #include <stdlib.h>
 
-#include_next <math.h>
+#include "../../math.h"
 
 #ifdef __cplusplus
 
