--- data/data/com.termux/files/usr/include/c++/v1/cmath.orig	2019-03-24 00:31:49.512145195 +0000
+++ ./cmath	2019-03-24 00:32:19.392703442 +0000
@@ -301,7 +301,7 @@
 */
 
 #include <__config>
-#include <math.h>
+#include <c++/v1/math.h>
 #include "version"
 #include <type_traits>
 
--- data/data/com.termux/files/usr/include/c++/v1/math.h.orig	2019-03-24 01:10:28.881807799 +0000
+++ ./math.h	2019-03-24 01:11:01.078384393 +0000
@@ -300,7 +300,7 @@
 #define _LIBCPP_STDLIB_INCLUDE_NEXT
 #include <stdlib.h>
 
-#include_next <math.h>
+#include "../../math.h"
 
 #ifdef __cplusplus
 
