diff -uNr PyPanel-2.4/pypanelrc PyPanel-2.4.mod/pypanelrc
--- PyPanel-2.4/pypanelrc	2005-06-27 02:27:37.000000000 +0300
+++ PyPanel-2.4.mod/pypanelrc	2017-12-12 17:52:16.997058670 +0200
@@ -11,17 +11,17 @@
 #------------------------------------------------------------------------------
 # Colors: Format is hex triplet - 0xrrggbb
 #------------------------------------------------------------------------------
-BG_COLOR        = "0xd6d6d6"    # Panel background and tinting color
-TASK_COLOR      = "0x000000"    # Normal task name color 
-FOCUSED_COLOR   = "0x1826de"    # Focused task name color
-SHADED_COLOR    = "0x808080"    # Shaded task name color 
-MINIMIZED_COLOR = "0x808080"    # Minimized task name color 
-DESKTOP_COLOR   = "0x000000"    # Desktop name color
-CLOCK_COLOR     = "0x000000"    # Clock text color
-LINE_COLOR      = "0x606060"    # Vertical line color
+BG_COLOR        = "0x0a0a0a"    # Panel background and tinting color
+TASK_COLOR      = "0xa0a0a0"    # Normal task name color 
+FOCUSED_COLOR   = "0xffffff"    # Focused task name color
+SHADED_COLOR    = "0xa0a0a0"    # Shaded task name color 
+MINIMIZED_COLOR = "0xa0a0a0"    # Minimized task name color 
+DESKTOP_COLOR   = "0xffffff"    # Desktop name color
+CLOCK_COLOR     = "0xf0f0f0"    # Clock text color
+LINE_COLOR      = "0x555555"    # Vertical line color
 
 # Text Shadow Colors
-TASK_SHADOW_COLOR      = "0xffffff"
+TASK_SHADOW_COLOR      = "0xa0a0a0"
 FOCUSED_SHADOW_COLOR   = "0xffffff"
 SHADED_SHADOW_COLOR    = "0xffffff"
 MINIMIZED_SHADOW_COLOR = "0xffffff" 
@@ -123,7 +123,7 @@
 # Background Alpha/Shade Level: 0 (Fully Translucent) -> 255 (Fully Opaque)
 # BG_COLOR is used for tinting
 #------------------------------------------------------------------------------
-SHADE           = 50
+SHADE           = 200
 
 #------------------------------------------------------------------------------
 # Misc. Options: 1 = Enabled/Yes, 0 = Disabled/No
@@ -132,7 +132,7 @@
 APPICONS        = 1             # Show application icons
 AUTOHIDE        = 0             # Autohide uses the CLOCK_DELAY timer above
 SHADOWS         = 0             # Show text shadows
-SHOWLINES       = 0             # Show object seperation lines
+SHOWLINES       = 1             # Show object seperation lines
 SHOWBORDER      = 0             # Show a border around the panel
 
 #------------------------------------------------------------------------------
@@ -141,7 +141,7 @@
 # by the XServer, if that fails it will use the desktop number as its name
 # Ex. ["One", "Two", "Three", "Four", "Five", "Six", "Seven", "Eight"]
 #------------------------------------------------------------------------------
-DESKTOP_NAMES   = []
+DESKTOP_NAMES   = ["Desktop 1", "Desktop 2", "Desktop 3", "Desktop 4"]
 
 #------------------------------------------------------------------------------
 # Panel Layout:       -----------------------------------
@@ -152,7 +152,7 @@
 # in the diagram above.  Each of the following objects can be enabled by
 # assigning it a section number or disabled by assigning it 0:
 #------------------------------------------------------------------------------
-DESKTOP         = 1             # Desktop name section
+DESKTOP         = 0             # Desktop name section
 TASKS           = 2             # Task names section 
 TRAY            = 3             # System tray section
 CLOCK           = 4             # Clock section
@@ -262,4 +262,3 @@
         pp.taskRaise(task, focus=1)
     elif button == 5:
         pp.taskLower(task, focus=0)
-        
