--- a/lib/Xm/Makefile.in
+++ b/lib/Xm/Makefile.in
@@ -1340,7 +1340,7 @@
 	test -f XmStrDefs23.ht || $(LN_S) $(srcdir)/XmStrDefs23.ht XmStrDefs23.ht
 	test -f XmStrDefsI.ht || $(LN_S) $(srcdir)/XmStrDefsI.ht XmStrDefsI.ht
 	test -f XmStrDefs.ct || $(LN_S) $(srcdir)/XmStrDefs.ct XmStrDefs.ct
-	$(top_builddir)/config/util/makestrs -f $(XMSTRING_LIST) >XmStrDefs.c
+	makestrs -f $(XMSTRING_LIST) >XmStrDefs.c
 
 @MessageCatalog_TRUE@$(CAT_FILES): Xm.msg
 @MessageCatalog_TRUE@	$(mkinstalldirs) $(top_builddir)/localized/C/msg
