diff -uNr cairo-dock-core-6c569e67a2a366e7634224a0133ede51755629cb/src/gldit/CMakeLists.txt cairo-dock-core-6c569e67a2a366e7634224a0133ede51755629cb.mod/src/gldit/CMakeLists.txt
--- cairo-dock-core-6c569e67a2a366e7634224a0133ede51755629cb/src/gldit/CMakeLists.txt	2021-03-27 06:23:53.000000000 +0200
+++ cairo-dock-core-6c569e67a2a366e7634224a0133ede51755629cb.mod/src/gldit/CMakeLists.txt	2021-12-05 11:40:06.757366142 +0200
@@ -122,8 +122,6 @@
 configure_file (${CMAKE_CURRENT_SOURCE_DIR}/gldi.pc.in ${CMAKE_CURRENT_BINARY_DIR}/gldi.pc)
 install (FILES  ${CMAKE_CURRENT_BINARY_DIR}/gldi.pc DESTINATION ${install-pc-path})
 install (FILES  ${CMAKE_CURRENT_BINARY_DIR}/libgldi.so DESTINATION ${libdir})
-install (FILES  ${CMAKE_CURRENT_BINARY_DIR}/libgldi.so.${VERSION} DESTINATION ${libdir})
-install (FILES  ${CMAKE_CURRENT_BINARY_DIR}/libgldi.so.${SOVERSION} DESTINATION ${libdir})
 
 ########### install files ###############
 
