--- a/Makefile
+++ b/Makefile
@@ -2,10 +2,10 @@
 .PHONY: all install clean
 
 OS=posix
-PREFIX=/usr/local
+PREFIX=@TERMUX_PREFIX@
 BINDIR=$(PREFIX)/bin
 MANDIR=$(PREFIX)/share/man
-ALL_CFLAGS=$(CFLAGS) -std=c99 -Wall -Wextra -Wshadow -Wmissing-prototypes -Wpedantic -Wno-unused-parameter
+ALL_CFLAGS=$(CFLAGS) -std=c99 -Wall -Wextra -Wshadow -Wmissing-prototypes -Wpedantic -Wno-unused-parameter -I@TERMUX_PREFIX@/include
 LDLIBS=-lrt
 OBJ=\
 	build.o\
