diff -uNr libfuse-fuse-3.9.1/util/install_helper.sh libfuse-fuse-3.9.1.mod/util/install_helper.sh
--- libfuse-fuse-3.9.1/util/install_helper.sh	2020-03-19 22:00:41.000000000 +0200
+++ libfuse-fuse-3.9.1.mod/util/install_helper.sh	2020-08-11 20:45:23.479917234 +0300
@@ -26,6 +26,8 @@
 install -D -m 644 "${MESON_SOURCE_ROOT}/util/fuse.conf" \
 	"${DESTDIR}${sysconfdir}/fuse.conf"
 
+exit 0
+
 if $useroot; then
     chown root:root "${DESTDIR}${bindir}/fusermount3"
     chmod u+s "${DESTDIR}${bindir}/fusermount3"
