--- ../configure.orig	2019-01-30 20:21:43.973270891 +0000
+++ ./configure	2019-01-30 20:22:28.683270592 +0000
@@ -5478,30 +5478,7 @@
 $as_echo "no, using $LN_S" >&6; }
 fi
 
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking whether ln works" >&5
-$as_echo_n "checking whether ln works... " >&6; }
-rm -f conf$$ conf$$.exe conf$$.file
-echo >conf$$.file
-if ln conf$$.file conf$$ 2>/dev/null; then
-  # We could just check for DJGPP; but this test works, is more generic
-  # and will remain valid once DJGPP supports symlinks (DJGPP 2.04).
-  if test -f conf$$.exe; then
-    # Don't use ln at all; we don't have any links
-    LN='cp -p'
-  else
-    LN='ln'
-  fi
-else
   LN=$as_ln_s
-fi
-rm -f conf$$ conf$$.exe conf$$.file
-if test "$LN" = ln; then
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-$as_echo "yes" >&6; }
-else
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: no, using $LN" >&5
-$as_echo "no, using $LN" >&6; }
-fi
 
 
 
