diff --git a/docklets/CPUMonitor/meson.build b/docklets/CPUMonitor/meson.build
index 3a33a86..92f5239 100644
--- a/docklets/CPUMonitor/meson.build
+++ b/docklets/CPUMonitor/meson.build
@@ -6,7 +6,7 @@ docklet_cpumonitor_sources = [
 shared_module(
   'docklet-cpumonitor',
   docklet_cpumonitor_sources,
-  dependencies: [plank_dep, plank_base_dep],
+  dependencies: [plank_dep, plank_base_dep, m_dep],
   install: true,
   install_dir: docklets_dir,
 )
diff --git a/docklets/Workspaces/meson.build b/docklets/Workspaces/meson.build
index ac1f15a..2cf5d14 100644
--- a/docklets/Workspaces/meson.build
+++ b/docklets/Workspaces/meson.build
@@ -7,7 +7,7 @@ docklet_workspaces_sources = [
 shared_module(
   'docklet-workspaces',
   docklet_workspaces_sources,
-  dependencies: [plank_dep, plank_base_dep, gnomemenu_dep, wnck_x11_dep],
+  dependencies: [plank_dep, plank_base_dep, gnomemenu_dep, wnck_x11_dep, m_dep],
   install: true,
   install_dir: docklets_dir,
 )
