diff -u -r ../libarchive-3.2.0/libarchive/archive.h ./libarchive/archive.h
--- ../libarchive-3.2.0/libarchive/archive.h	2016-04-30 00:44:07.000000000 -0400
+++ ./libarchive/archive.h	2016-05-02 19:58:33.742590137 -0400
@@ -96,11 +96,6 @@
 # endif
 #endif
 
-/* Large file support for Android */
-#ifdef __ANDROID__
-#include "android_lf.h"
-#endif
-
 /*
  * On Windows, define LIBARCHIVE_STATIC if you're building or using a
  * .lib.  The default here assumes you're building a DLL.  Only
