diff -u -r ../libarchive-3.2.0/libarchive/archive_entry.h ./libarchive/archive_entry.h
--- ../libarchive-3.2.0/libarchive/archive_entry.h	2016-04-30 00:44:07.000000000 -0400
+++ ./libarchive/archive_entry.h	2016-05-02 19:58:22.878678982 -0400
@@ -75,11 +75,6 @@
 # define	__LA_MODE_T	mode_t
 #endif
 
-/* Large file support for Android */
-#ifdef __ANDROID__
-#include "android_lf.h"
-#endif
-
 /*
  * On Windows, define LIBARCHIVE_STATIC if you're building or using a
  * .lib.  The default here assumes you're building a DLL.  Only
