diff -uNr libarchive-3.3.3/libarchive/archive_util.c libarchive-3.3.3.mod/libarchive/archive_util.c
--- libarchive-3.3.3/libarchive/archive_util.c	2018-09-02 09:05:18.000000000 +0300
+++ libarchive-3.3.3.mod/libarchive/archive_util.c	2019-03-02 02:14:39.252942170 +0200
@@ -374,7 +374,7 @@
 #ifdef _PATH_TMP
 		tmp = _PATH_TMP;
 #else
-                tmp = "/tmp";
+                tmp = "@TERMUX_PREFIX@/tmp";
 #endif
 	archive_strcpy(temppath, tmp);
 	if (temppath->s[temppath->length-1] != '/')
