diff -u -r ../pick-1.3.0/src/compat.h ./src/compat.h
--- ../pick-1.3.0/src/compat.h	2016-02-12 03:48:41.000000000 -0500
+++ ./src/compat.h	2016-02-17 22:47:09.825782253 -0500
@@ -10,7 +10,8 @@
 #endif /* __FreeBSD__ */
 
 #if defined(__linux__) || defined(__CYGWIN__)
-#ifdef __GNUC__
+#if defined(__GNUC__) || defined(__ANDROID__)
+#undef __dead
 #define __dead		__attribute__((__noreturn__))
 #else
 #define __dead
