--- a/Pillow/setup.py
+++ b/Pillow/setup.py
@@ -561,8 +561,6 @@
             or sys.platform.startswith("gnu")
             or sys.platform.startswith("freebsd")
         ):
-            for dirname in _find_library_dirs_ldconfig():
-                _add_directory(library_dirs, dirname)
             if sys.platform.startswith("linux") and os.environ.get(
                 "ANDROID_ROOT", None
             ):
@@ -588,16 +586,6 @@
 
         # FIXME: check /opt/stuff directories here?
 
-        # standard locations
-        if not self.disable_platform_guessing:
-            _add_directory(library_dirs, "/usr/local/lib")
-            _add_directory(include_dirs, "/usr/local/include")
-
-            _add_directory(library_dirs, "/usr/lib")
-            _add_directory(include_dirs, "/usr/include")
-            # alpine, at least
-            _add_directory(library_dirs, "/lib")
-
         if sys.platform == "win32":
             # on Windows, look for the OpenJPEG libraries in the location that
             # the official installer puts them
