--- a/setup.py
+++ b/setup.py
@@ -316,13 +316,13 @@
     setup_requires=[
         "certifi>=2020.06.20",
         "numpy>=1.17",
-        "setuptools_scm>=4",
+        "setuptools_scm<5,>=4",
         "setuptools_scm_git_archive",
     ],
     install_requires=[
         "cycler>=0.10",
         "fonttools>=4.22.0",
-        "kiwisolver>=1.0.1",
+        "kiwisolver<1.4.0,>=1.0.1",
         "numpy>=1.17",
         "packaging>=20.0",
         "pillow>=6.2.0",
