diff -u -r ../libgnt-2.14.3/meson.build ./meson.build
--- ../libgnt-2.14.3/meson.build	2021-09-26 03:56:33.000000000 +0000
+++ ./meson.build	2024-10-02 20:30:13.929310165 +0000
@@ -85,7 +85,7 @@
 	# FIXME: $host ?
 	ncurses_sys_prefix = '/usr/$host/sys-root/mingw'
 else
-	ncurses_sys_prefix = '/usr'
+	ncurses_sys_prefix = '@TERMUX_PREFIX@'
 endif
 
 ncurses_sys_dirs = [ncurses_sys_prefix + '/include/ncursesw',
