--- a/base/tools/aapt2/util/Util.cpp
+++ b/base/tools/aapt2/util/Util.cpp
@@ -24,7 +24,6 @@
 #include "android-base/stringprintf.h"
 #include "android-base/strings.h"
 #include "androidfw/StringPiece.h"
-#include "build/version.h"
 #include "text/Unicode.h"
 #include "text/Utf8Iterator.h"
 #include "util/BigBuffer.h"
@@ -230,17 +229,7 @@
   // Update minor version whenever a feature or flag is added.
   static const char* const sMinorVersion = "19";
 
-  // The build id of aapt2 binary.
-  static std::string sBuildId = android::build::GetBuildNumber();
-
-  if (android::base::StartsWith(sBuildId, "eng.")) {
-    time_t now = time(0);
-    tm* ltm = localtime(&now);
-
-    sBuildId = android::base::StringPrintf("eng.%d%d", 1900 + ltm->tm_year, 1 + ltm->tm_mon);
-  }
-
-  return android::base::StringPrintf("%s.%s-%s", sMajorVersion, sMinorVersion, sBuildId.c_str());
+  return android::base::StringPrintf("%s.%s", sMajorVersion, sMinorVersion);
 }
 
 static size_t ConsumeDigits(const char* start, const char* end) {
