--- a/base/tools/aapt2/cmd/Compile.h
+++ b/base/tools/aapt2/cmd/Compile.h
@@ -35,6 +35,7 @@
   bool pseudolocalize = false;
   bool no_png_crunch = false;
   bool legacy_mode = false;
+  bool source_path = false;
   // See comments on aapt::ResourceParserOptions.
   bool preserve_visibility_of_styleables = false;
   bool verbose = false;
@@ -58,6 +59,8 @@
     AddOptionalSwitch("--no-crunch", "Disables PNG processing", &options_.no_png_crunch);
     AddOptionalSwitch("--legacy", "Treat errors that used to be valid in AAPT as warnings",
         &options_.legacy_mode);
+    AddOptionalSwitch("--source-path", "Ignored for compatibility",
+        &options_.source_path);
     AddOptionalSwitch("--preserve-visibility-of-styleables",
                       "If specified, apply the same visibility rules for\n"
                       "styleables as are used for all other resources.\n"
