--- a/aidl/options.h
+++ b/aidl/options.h
@@ -21,7 +21,6 @@
 #include <vector>
 
 #include <android-base/macros.h>
-#include <gtest/gtest_prod.h>
 
 namespace android {
 namespace aidl {
@@ -59,13 +58,6 @@
  private:
   JavaOptions() = default;
 
-  FRIEND_TEST(EndToEndTest, IExampleInterface);
-  FRIEND_TEST(AidlTest, FailOnParcelable);
-  FRIEND_TEST(AidlTest, WritePreprocessedFile);
-  FRIEND_TEST(AidlTest, WritesCorrectDependencyFile);
-  FRIEND_TEST(AidlTest, WritesCorrectDependencyFileNinja);
-  FRIEND_TEST(AidlTest, WritesTrivialDependencyFileForParcelable);
-
   DISALLOW_COPY_AND_ASSIGN(JavaOptions);
 };
 
@@ -97,8 +89,6 @@
   std::string dep_file_name_;
   bool dep_file_ninja_{false};
 
-  FRIEND_TEST(CppOptionsTests, ParsesCompileCpp);
-  FRIEND_TEST(CppOptionsTests, ParsesCompileCppNinja);
   DISALLOW_COPY_AND_ASSIGN(CppOptions);
 };
 
