--- a/base/libs/androidfw/Asset.cpp
+++ b/base/libs/androidfw/Asset.cpp
@@ -455,7 +455,6 @@
 {
     assert(mFp == NULL);    // no reopen
     assert(!mMap.has_value());
-    assert(dataMap != NULL);
 
     mMap = std::move(dataMap);
     mStart = -1;            // not used
@@ -800,7 +799,6 @@
 {
     assert(mFd < 0);        // no re-open
     assert(!mMap.has_value());
-    assert(dataMap != NULL);
 
     mMap = std::move(dataMap);
     mStart = -1;        // not used
