--- a/vendor/base/libs/androidfw/Asset.cpp
+++ b/vendor/base/libs/androidfw/Asset.cpp
@@ -452,7 +452,6 @@
 {
     assert(mFp == NULL);    // no reopen
     assert(!mMap.has_value());
-    assert(dataMap.data());
 
     mMap = std::move(dataMap);
     mStart = -1;            // not used
@@ -797,7 +796,6 @@
 {
     assert(mFd < 0);        // no re-open
     assert(!mMap.has_value());
-    assert(dataMap.data());
 
     mMap = std::move(dataMap);
     mStart = -1;        // not used
