--- a/core/base/liblog_symbols.h
+++ b/core/base/liblog_symbols.h
@@ -36,6 +36,7 @@
   int32_t (*__android_log_set_minimum_priority)(int32_t priority);
   int32_t (*__android_log_get_minimum_priority)();
   void (*__android_log_set_default_tag)(const char* tag);
+  int (*__android_log_is_loggable)(int prio, const char* tag, int default_prio);
 };
 
 const std::optional<LibLogFunctions>& GetLibLogFunctions();
