--- a/core/base/logging.cpp
+++ b/core/base/logging.cpp
@@ -555,7 +555,7 @@
   // take into consideration the value from SetMinimumLogSeverity().
   if (liblog_functions) {
     int32_t priority = LogSeverityToPriority(severity);
-    return __android_log_is_loggable(priority, tag, ANDROID_LOG_INFO);
+    return liblog_functions->__android_log_is_loggable(priority, tag, ANDROID_LOG_INFO);
   } else {
     return severity >= gMinimumLogSeverity;
   }
