--- a/core/base/liblog_symbols.cpp
+++ b/core/base/liblog_symbols.cpp
@@ -55,6 +55,7 @@
     DLSYM(__android_log_set_minimum_priority);
     DLSYM(__android_log_get_minimum_priority);
     DLSYM(__android_log_set_default_tag);
+    DLSYM(__android_log_is_loggable);
 #undef DLSYM
 
     return real_liblog_functions;
@@ -78,6 +79,7 @@
         .__android_log_set_minimum_priority = __android_log_set_minimum_priority,
         .__android_log_get_minimum_priority = __android_log_get_minimum_priority,
         .__android_log_set_default_tag = __android_log_set_default_tag,
+        .__android_log_is_loggable = __android_log_is_loggable,
     };
   }();
   return liblog_functions;
