--- a/libbase/properties.cpp
+++ b/libbase/properties.cpp
@@ -16,7 +16,7 @@
 
 #include "android-base/properties.h"
 
-#if defined(__BIONIC__)
+#if defined(__BIONIC__) && !defined(__TERMUX__)
 #include <sys/system_properties.h>
 #endif
 
@@ -31,7 +31,7 @@
 #include <android-base/strings.h>
 #include <android-base/thread_annotations.h>
 
-#if !defined(__BIONIC__)
+#if !defined(__BIONIC__) || defined(__TERMUX__)
 
 // Here lies a rudimentary implementation of system properties for non-Bionic
 // platforms. We are using weak symbols here because we want to allow
@@ -178,7 +178,7 @@ bool SetProperty(const std::string& key, const std::string& value) {
   return (__system_property_set(key.c_str(), value.c_str()) == 0);
 }
 
-#if defined(__BIONIC__)
+#if defined(__BIONIC__) && !defined(__TERMUX__)
 
 struct WaitForPropertyData {
   bool done;
