--- a/vendor/libbase/properties.cpp
+++ b/vendor/libbase/properties.cpp
@@ -27,0 +28 @@
+#include <thread>
@@ -163,0 +165,5 @@
+#if defined(__BIONIC__) && __ANDROID_API__ < 26
+  char value[PROP_VALUE_MAX];
+  if (__system_property_read(pi, nullptr, value) < 0) return default_value;
+  property_value = value;
+#else
@@ -170,0 +177 @@
+#endif
@@ -182,0 +190,2 @@
+#if __ANDROID_API__ >= 26
+
@@ -239,0 +249,2 @@
+#endif
+
@@ -241,0 +253,13 @@
+#if __ANDROID_API__ < 26
+  auto start_time = std::chrono::steady_clock::now();
+  while (true) {
+    if (GetProperty(key, "") == expected_value) return true;
+
+    auto now = std::chrono::steady_clock::now();
+    auto elapsed = std::chrono::duration_cast<std::chrono::milliseconds>(now - start_time);
+    if (elapsed >= relative_timeout) return false;
+
+    auto remaining_timeout = relative_timeout - elapsed;
+    std::this_thread::sleep_for(std::min(std::chrono::milliseconds(50), remaining_timeout));
+  }
+#else
@@ -259,0 +284 @@
+#endif
@@ -263,0 +289,13 @@
+#if __ANDROID_API__ < 26
+  auto start_time = std::chrono::steady_clock::now();
+  while (true) {
+    if (__system_property_find(key.c_str()) != nullptr) return true;
+
+    auto now = std::chrono::steady_clock::now();
+    auto elapsed = std::chrono::duration_cast<std::chrono::milliseconds>(now - start_time);
+    if (elapsed >= relative_timeout) return false;
+
+    auto remaining_timeout = relative_timeout - elapsed;
+    std::this_thread::sleep_for(std::min(std::chrono::milliseconds(50), remaining_timeout));
+  }
+#else
@@ -265,0 +304 @@
+#endif
@@ -298,0 +338,8 @@
+#if __ANDROID_API__ < 26
+      char value[PROP_VALUE_MAX];
+      if (__system_property_read(prop_info_, nullptr, value) >= 0) {
+        cached_property_serial_ = property_serial;
+        strlcpy(cached_value_, value, PROP_VALUE_MAX);
+        read_only_property_ = cached_value_;
+      }
+#else
@@ -312,0 +360 @@
+#endif
@@ -327,0 +376,15 @@
+#if __ANDROID_API__ < 26
+  auto start_time = std::chrono::steady_clock::now();
+  uint32_t old_serial = cached_property_serial_.value_or(0);
+  while (true) {
+    const char* value = Get(nullptr);
+    if (prop_info_ != nullptr && cached_property_serial_.value_or(0) != old_serial) return value;
+
+    auto now = std::chrono::steady_clock::now();
+    auto elapsed = std::chrono::duration_cast<std::chrono::milliseconds>(now - start_time);
+    if (elapsed >= relative_timeout) return nullptr;
+
+    auto remaining_timeout = relative_timeout - elapsed;
+    std::this_thread::sleep_for(std::min(std::chrono::milliseconds(50), remaining_timeout));
+  }
+#else
@@ -343,0 +407 @@
+#endif
