--- a/vendor/logging/liblog/logger_write.cpp
+++ b/vendor/logging/liblog/logger_write.cpp
@@ -43 +43 @@
-#ifdef __ANDROID__
+#if defined(__ANDROID__) && !defined(__TERMUX__)
@@ -59 +59 @@
-#if defined(__ANDROID__)
+#if defined(__ANDROID__) && !defined(__TERMUX__)
@@ -116 +116 @@
-#ifdef __ANDROID__
+#if defined(__ANDROID__) && !defined(__TERMUX__)
@@ -163 +163 @@
-#ifdef __ANDROID__
+#if defined(__ANDROID__) && !defined(__TERMUX__)
@@ -198 +198 @@
-#ifdef __ANDROID__
+#if defined(__ANDROID__) && !defined(__TERMUX__)
@@ -275 +275 @@
-#ifdef __ANDROID__
+#if defined(__ANDROID__) && !defined(__TERMUX__)
