--- a/core/libziparchive/include/ziparchive/zip_writer.h
+++ b/core/libziparchive/include/ziparchive/zip_writer.h
@@ -19,7 +19,6 @@
 #include <cstdio>
 #include <ctime>
 
-#include <gtest/gtest_prod.h>
 #include <memory>
 #include <string>
 #include <string_view>
@@ -184,6 +183,4 @@
 
   std::unique_ptr<z_stream, void (*)(z_stream*)> z_stream_;
   std::vector<uint8_t> buffer_;
-
-  FRIEND_TEST(zipwriter, WriteToUnseekableFile);
 };
