diff --git a/submodules/protobuf/cmake/CMakeLists.txt b/submodules/protobuf/cmake/CMakeLists.txt
index 04cb330..95882a2 100644
--- a/submodules/protobuf/CMakeLists.txt
+++ b/submodules/protobuf/CMakeLists.txt
@@ -298,7 +298,8 @@ endif (MSVC)
 include_directories(
   ${ZLIB_INCLUDE_DIRECTORIES}
   ${protobuf_BINARY_DIR}
-  ${protobuf_SOURCE_DIR}/src)
+  ${protobuf_SOURCE_DIR}/src
+  ${protobuf_SOURCE_DIR}/android)

 if (protobuf_UNICODE)
   add_definitions(-DUNICODE -D_UNICODE)
diff --git a/submodules/protobuf/cmake/libprotobuf.cmake b/submodules/protobuf/cmake/libprotobuf.cmake
index 07e4bcf..f77815c 100644
--- a/submodules/protobuf/cmake/libprotobuf.cmake
+++ b/submodules/protobuf/cmake/libprotobuf.cmake
@@ -118,7 +118,8 @@ if(protobuf_LINK_LIBATOMIC)
   target_link_libraries(libprotobuf PRIVATE atomic)
 endif()
 if(${CMAKE_SYSTEM_NAME} STREQUAL "Android")
-  target_link_libraries(libprotobuf PRIVATE log)
+  # static lib compiled and linked by aapt(2)'s dedicated cmake
+  # target_link_libraries(libprotobuf PRIVATE log)
 endif()
 target_include_directories(libprotobuf PUBLIC ${protobuf_SOURCE_DIR}/src)
 if(protobuf_BUILD_SHARED_LIBS)
