--- a/netconf/src/ncx/ncxmod.h
+++ b/netconf/src/ncx/ncxmod.h
@@ -168,7 +168,7 @@
 #define NCXMOD_YUMA_DIR (const xmlChar *)"~/.yuma"
 
 /* per user yangcli internal data home when $HOME not defined */
-#define NCXMOD_TEMP_YUMA_DIR (const xmlChar *)"/tmp/yuma"
+#define NCXMOD_TEMP_YUMA_DIR (const xmlChar *)"@TERMUX_PREFIX@/tmp/yuma"
 
 /* Yuma work directory name */
 #define NCXMOD_YUMA_DIRNAME    (const xmlChar *)".yuma"
@@ -176,7 +176,7 @@
 /* sub-directory name yangcli uses to store local per-session workdirs
  * appended to ncxmod_yumadir_path
  */
-#define NCXMOD_TEMP_DIR (const xmlChar *)"/tmp"
+#define NCXMOD_TEMP_DIR (const xmlChar *)"@TERMUX_PREFIX@/tmp"
 
 /********************************************************************
 *								    *
