--- a/netconf/src/agt/agt_ncxserver.h
+++ b/netconf/src/agt/agt_ncxserver.h
@@ -45,7 +45,7 @@
 *			 C O N S T A N T S			    *
 *								    *
 *********************************************************************/
-#define NCXSERVER_SOCKNAME   "/tmp/ncxserver.sock"
+#define NCXSERVER_SOCKNAME   "@TERMUX_PREFIX@/tmp/ncxserver.sock"
 
 /********************************************************************
 *								    *
