--- a/netconf/src/yangdiff/yangdiff.h
+++ b/netconf/src/yangdiff/yangdiff.h
@@ -71,7 +71,7 @@
 #define YANGDIFF_DEF_OUTPUT        (const xmlChar *)"stdout"
 #define YANGDIFF_DEF_DIFFTYPE      (const xmlChar *)"summary"
 #define YANGDIFF_DEF_DT            YANGDIFF_DT_NORMAL
-#define YANGDIFF_DEF_CONFIG        (const xmlChar *)"/etc/yuma/yangdiff.conf"
+#define YANGDIFF_DEF_CONFIG        (const xmlChar *)"@TERMUX_PREFIX@/etc/yuma/yangdiff.conf"
 #define YANGDIFF_DEF_FILENAME      (const xmlChar *)"yangdiff.log"
 
 #define YANGDIFF_MOD               (const xmlChar *)"yangdiff"
