https://github.com/google/quiche/commit/c1af894e0f5c4f732a983e7c93227854e203570e

--- a/chromium/src/net/third_party/quiche/src/quiche/quic/core/quic_interval_deque.h
+++ b/chromium/src/net/third_party/quiche/src/quiche/quic/core/quic_interval_deque.h
@@ -198,12 +198,12 @@
     Iterator operator+(difference_type amount) const {
       Iterator copy = *this;
       copy.index_ += amount;
-      QUICHE_DCHECK(copy.index_ < copy.deque_->size());
+      QUICHE_DCHECK(copy.index_ < copy.deque_->Size());
       return copy;
     }
     Iterator& operator+=(difference_type amount) {
       index_ += amount;
-      QUICHE_DCHECK(index_ < deque_->size());
+      QUICHE_DCHECK(index_ < deque_->Size());
       return *this;
     }
     difference_type operator-(const Iterator& rhs) const {
