diff -uNr rest-server-0.9.8/cmd/rest-server/main.go rest-server-0.9.8.mod/cmd/rest-server/main.go
--- rest-server-0.9.8/cmd/rest-server/main.go	2018-06-15 00:53:29.000000000 +0300
+++ rest-server-0.9.8.mod/cmd/rest-server/main.go	2020-02-18 14:56:01.369738081 +0200
@@ -26,7 +26,7 @@
 }
 
 var server = restserver.Server{
-	Path:   "/tmp/restic",
+	Path:   "@TERMUX_PREFIX@/tmp/restic",
 	Listen: ":8000",
 }
 
