diff -uNr SDL2-2.0.9/include/SDL_stdinc.h SDL2-2.0.9.mod/include/SDL_stdinc.h
--- SDL2-2.0.9/include/SDL_stdinc.h	2018-10-31 17:07:22.000000000 +0200
+++ SDL2-2.0.9.mod/include/SDL_stdinc.h	2018-11-09 16:54:52.125408951 +0200
@@ -332,7 +332,6 @@
 
 /** \cond */
 #ifndef DOXYGEN_SHOULD_IGNORE_THIS
-#if !defined(__ANDROID__)
    /* TODO: include/SDL_stdinc.h:174: error: size of array 'SDL_dummy_enum' is negative */
 typedef enum
 {
@@ -340,7 +339,6 @@
 } SDL_DUMMY_ENUM;
 
 SDL_COMPILE_TIME_ASSERT(enum, sizeof(SDL_DUMMY_ENUM) == sizeof(int));
-#endif
 #endif /* DOXYGEN_SHOULD_IGNORE_THIS */
 /** \endcond */
 
