diff -uNr SDL2-2.0.9/src/joystick/SDL_sysjoystick.h SDL2-2.0.9.mod/src/joystick/SDL_sysjoystick.h
--- SDL2-2.0.9/src/joystick/SDL_sysjoystick.h	2018-10-31 17:07:22.000000000 +0200
+++ SDL2-2.0.9.mod/src/joystick/SDL_sysjoystick.h	2018-11-09 17:04:46.285771392 +0200
@@ -74,7 +74,7 @@
     struct _SDL_Joystick *next; /* pointer to next joystick we have allocated */
 };
 
-#if defined(__IPHONEOS__) || defined(__ANDROID__)
+#if defined(__IPHONEOS__)
 #define HAVE_STEAMCONTROLLERS
 #define USE_STEAMCONTROLLER_HIDAPI
 #elif defined(__LINUX__)
