diff -uNr SDL2-2.0.8/include/SDL_stdinc.h SDL2-2.0.8.mod/include/SDL_stdinc.h
--- SDL2-2.0.8/include/SDL_stdinc.h	2018-03-01 18:34:41.000000000 +0200
+++ SDL2-2.0.8.mod/include/SDL_stdinc.h	2018-09-07 16:13:56.456904701 +0300
@@ -310,7 +310,6 @@
 
 /** \cond */
 #ifndef DOXYGEN_SHOULD_IGNORE_THIS
-#if !defined(__ANDROID__)
    /* TODO: include/SDL_stdinc.h:174: error: size of array 'SDL_dummy_enum' is negative */
 typedef enum
 {
@@ -318,7 +317,6 @@
 } SDL_DUMMY_ENUM;
 
 SDL_COMPILE_TIME_ASSERT(enum, sizeof(SDL_DUMMY_ENUM) == sizeof(int));
-#endif
 #endif /* DOXYGEN_SHOULD_IGNORE_THIS */
 /** \endcond */
 
