diff -uNr SDL2-2.0.8/include/SDL_config.h SDL2-2.0.8.mod/include/SDL_config.h
--- SDL2-2.0.8/include/SDL_config.h	2018-03-01 18:34:41.000000000 +0200
+++ SDL2-2.0.8.mod/include/SDL_config.h	2018-09-07 16:11:53.382758975 +0300
@@ -37,8 +37,6 @@
 #include "SDL_config_macosx.h"
 #elif defined(__IPHONEOS__)
 #include "SDL_config_iphoneos.h"
-#elif defined(__ANDROID__)
-#include "SDL_config_android.h"
 #elif defined(__PSP__)
 #include "SDL_config_psp.h"
 #else
