diff -uNr SDL2-2.0.9/include/SDL_egl.h SDL2-2.0.9.mod/include/SDL_egl.h
--- SDL2-2.0.9/include/SDL_egl.h	2018-10-31 17:07:22.000000000 +0200
+++ SDL2-2.0.9.mod/include/SDL_egl.h	2018-11-09 16:54:15.701628001 +0200
@@ -24,7 +24,7 @@
  *
  *  This is a simple file to encapsulate the EGL API headers.
  */
-#if !defined(_MSC_VER) && !defined(__ANDROID__)
+#if !defined(_MSC_VER)
 
 #include <EGL/egl.h>
 #include <EGL/eglext.h>
@@ -421,15 +421,6 @@
 typedef struct gbm_bo      *EGLNativePixmapType;
 typedef void               *EGLNativeWindowType;
 
-#elif defined(__ANDROID__) /* Android */
-
-struct ANativeWindow;
-struct egl_native_pixmap_t;
-
-typedef struct ANativeWindow        *EGLNativeWindowType;
-typedef struct egl_native_pixmap_t  *EGLNativePixmapType;
-typedef void                        *EGLNativeDisplayType;
-
 #elif defined(MIR_EGL_PLATFORM)
 
 #include <mir_toolkit/mir_client_library.h>
