diff -uNr SDL2-2.0.9/src/SDL.c SDL2-2.0.9.mod/src/SDL.c
--- SDL2-2.0.9/src/SDL.c	2018-10-31 17:07:22.000000000 +0200
+++ SDL2-2.0.9.mod/src/SDL.c	2018-11-09 17:09:44.315855024 +0200
@@ -420,8 +420,6 @@
 {
 #if __AIX__
     return "AIX";
-#elif __ANDROID__
-    return "Android";
 #elif __BSDI__
     return "BSDI";
 #elif __DREAMCAST__
@@ -478,10 +476,7 @@
 SDL_bool
 SDL_IsTablet()
 {
-#if __ANDROID__
-    extern SDL_bool SDL_IsAndroidTablet(void);
-    return SDL_IsAndroidTablet();
-#elif __IPHONEOS__
+#if __IPHONEOS__
     extern SDL_bool SDL_IsIPad(void);
     return SDL_IsIPad();
 #else
