diff -uNr SDL2-2.0.9/src/video/SDL_video.c SDL2-2.0.9.mod/src/video/SDL_video.c
--- SDL2-2.0.9/src/video/SDL_video.c	2018-10-31 17:07:22.000000000 +0200
+++ SDL2-2.0.9.mod/src/video/SDL_video.c	2018-11-09 16:58:16.844546616 +0200
@@ -1435,12 +1435,6 @@
         return NULL;
     }
 
-    /* Some platforms have OpenGL enabled by default */
-#if (SDL_VIDEO_OPENGL && __MACOSX__) || __IPHONEOS__ || __ANDROID__ || __NACL__
-    if (!_this->is_dummy && !(flags & SDL_WINDOW_VULKAN)) {
-        flags |= SDL_WINDOW_OPENGL;
-    }
-#endif
     if (flags & SDL_WINDOW_OPENGL) {
         if (!_this->GL_CreateContext) {
             SDL_SetError("OpenGL support is either not configured in SDL "
