diff -uNr SDL2-2.0.9/include/SDL_rwops.h SDL2-2.0.9.mod/include/SDL_rwops.h
--- SDL2-2.0.9/include/SDL_rwops.h	2018-10-31 17:07:22.000000000 +0200
+++ SDL2-2.0.9.mod/include/SDL_rwops.h	2018-11-09 16:54:40.285263749 +0200
@@ -93,20 +93,7 @@
     Uint32 type;
     union
     {
-#if defined(__ANDROID__)
-        struct
-        {
-            void *fileNameRef;
-            void *inputStreamRef;
-            void *readableByteChannelRef;
-            void *readMethod;
-            void *assetFileDescriptorRef;
-            long position;
-            long size;
-            long offset;
-            int fd;
-        } androidio;
-#elif defined(__WIN32__)
+#if   defined(__WIN32__)
         struct
         {
             SDL_bool append;
