diff -uNr SDL2-2.0.9/src/thread/pthread/SDL_systhread.c SDL2-2.0.9.mod/src/thread/pthread/SDL_systhread.c
--- SDL2-2.0.9/src/thread/pthread/SDL_systhread.c	2018-10-31 17:07:22.000000000 +0200
+++ SDL2-2.0.9.mod/src/thread/pthread/SDL_systhread.c	2018-11-09 16:57:44.624162105 +0200
@@ -51,9 +51,6 @@
 #include "SDL_thread.h"
 #include "../SDL_thread_c.h"
 #include "../SDL_systhread.h"
-#ifdef __ANDROID__
-#include "../../core/android/SDL_android.h"
-#endif
 
 #ifdef __HAIKU__
 #include <kernel/OS.h>
@@ -72,9 +69,6 @@
 static void *
 RunThread(void *data)
 {
-#ifdef __ANDROID__
-    Android_JNI_SetupThread();
-#endif
     SDL_RunThread(data);
     return NULL;
 }
