diff -uNr SDL2-2.0.9/src/stdlib/SDL_getenv.c SDL2-2.0.9.mod/src/stdlib/SDL_getenv.c
--- SDL2-2.0.9/src/stdlib/SDL_getenv.c	2018-10-31 17:07:22.000000000 +0200
+++ SDL2-2.0.9.mod/src/stdlib/SDL_getenv.c	2018-11-09 16:57:33.934034207 +0200
@@ -29,9 +29,6 @@
 #include "../core/windows/SDL_windows.h"
 #endif
 
-#if defined(__ANDROID__)
-#include "../core/android/SDL_android.h"
-#endif
 
 #include "SDL_stdinc.h"
 
@@ -175,10 +172,6 @@
 char *
 SDL_getenv(const char *name)
 {
-#if defined(__ANDROID__)
-    /* Make sure variables from the application manifest are available */
-    Android_JNI_GetManifestEnvironmentVariables();
-#endif
 
     /* Input validation */
     if (!name || !*name) {
