From 67729f54c2e7a24b2bebfa23b75afe607c09a67a Mon Sep 17 00:00:00 2001
From: Sebastian Pipping <sebastian@pipping.org>
Date: Thu, 26 Nov 2020 19:28:03 +0100
Subject: [PATCH] Fix build with -fno-common or GCC 10

---
 io_internal.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/io_internal.h b/io_internal.h
index 6d37c11..3e378d9 100644
--- a/io_internal.h
+++ b/io_internal.h
@@ -83,7 +83,7 @@ my_extern array io_pollfds;
 my_extern long first_readable;
 my_extern long first_writeable;
 
-my_extern long first_deferred;
+extern long first_deferred;
 
 my_extern enum __io_waitmode {
   UNDECIDED,
-- 
2.27.0

