diff -u -r ../pidgin-2.10.12/libpurple/protocols/bonjour/Makefile.in ./libpurple/protocols/bonjour/Makefile.in
--- ../pidgin-2.10.12/libpurple/protocols/bonjour/Makefile.in	2015-12-31 18:19:51.000000000 -0500
+++ ./libpurple/protocols/bonjour/Makefile.in	2016-05-05 04:28:23.297133517 -0400
@@ -592,7 +592,7 @@
 	}
 
 libbonjour.la: $(libbonjour_la_OBJECTS) $(libbonjour_la_DEPENDENCIES) $(EXTRA_libbonjour_la_DEPENDENCIES) 
-	$(AM_V_CCLD)$(libbonjour_la_LINK) $(am_libbonjour_la_rpath) $(libbonjour_la_OBJECTS) $(libbonjour_la_LIBADD) $(LIBS)
+	$(AM_V_CCLD)$(libbonjour_la_LINK) $(am_libbonjour_la_rpath) $(libbonjour_la_OBJECTS) $(libbonjour_la_LIBADD) $(LIBS) -lpurple
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
