diff -u -r ../pidgin-2.10.11/configure ./configure
--- ../pidgin-2.10.11/configure	2014-11-23 11:42:00.000000000 -0500
+++ ./configure	2014-12-28 16:09:30.872081387 -0500
@@ -19876,7 +19876,6 @@
 			"-Wcast-align" \
 			"-Wdeclaration-after-statement" \
 			"-Wendif-labels" \
-			"-Werror-implicit-function-declaration" \
 			"-Wextra -Wno-unused-parameter" \
 			"-Wformat-security" \
 				"-Werror=format-security" \
