diff -u -r ../pidgin-2.10.11/libpurple/plugin.c ./libpurple/plugin.c
--- ../pidgin-2.10.11/libpurple/plugin.c	2014-11-23 11:41:25.000000000 -0500
+++ ./libpurple/plugin.c	2015-12-17 17:31:41.038378834 -0500
@@ -254,7 +254,7 @@
 		 *
 		 * G_MODULE_BIND_LOCAL was added in glib 2.3.3.
 		 */
-		plugin->handle = g_module_open(filename, G_MODULE_BIND_LOCAL);
+		plugin->handle = g_module_open(filename, G_MODULE_BIND_LOCAL/*test*/);
 
 		if (plugin->handle == NULL)
 		{
