diff -u -r ../pidgin-2.10.9/libpurple/example/Makefile.in ./libpurple/example/Makefile.in
--- ../pidgin-2.10.9/libpurple/example/Makefile.in	2014-02-02 22:30:21.000000000 +0100
+++ ./libpurple/example/Makefile.in	2014-03-09 22:35:03.000000000 +0100
@@ -51,7 +51,6 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-noinst_PROGRAMS = nullclient$(EXEEXT)
 subdir = libpurple/example
 DIST_COMMON = $(srcdir)/Makefile.am $(srcdir)/Makefile.in
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
