diff -u -r ../pidgin-2.10.12/libpurple/protocols/gg/Makefile.in ./libpurple/protocols/gg/Makefile.in
--- ../pidgin-2.10.12/libpurple/protocols/gg/Makefile.in	2015-12-31 18:19:51.000000000 -0500
+++ ./libpurple/protocols/gg/Makefile.in	2016-05-05 04:29:27.412205753 -0400
@@ -697,7 +697,7 @@
 	}
 
 libgg.la: $(libgg_la_OBJECTS) $(libgg_la_DEPENDENCIES) $(EXTRA_libgg_la_DEPENDENCIES) 
-	$(AM_V_CCLD)$(libgg_la_LINK) $(am_libgg_la_rpath) $(libgg_la_OBJECTS) $(libgg_la_LIBADD) $(LIBS)
+	$(AM_V_CCLD)$(libgg_la_LINK) $(am_libgg_la_rpath) $(libgg_la_OBJECTS) $(libgg_la_LIBADD) $(LIBS) -lpurple
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
