diff -u -r ../pidgin-2.10.12/libpurple/protocols/jabber/Makefile.in ./libpurple/protocols/jabber/Makefile.in
--- ../pidgin-2.10.12/libpurple/protocols/jabber/Makefile.in	2015-12-31 18:19:52.000000000 -0500
+++ ./libpurple/protocols/jabber/Makefile.in	2016-05-05 04:17:05.462969023 -0400
@@ -649,10 +649,10 @@
 	}
 
 libjabber.la: $(libjabber_la_OBJECTS) $(libjabber_la_DEPENDENCIES) $(EXTRA_libjabber_la_DEPENDENCIES) 
-	$(AM_V_CCLD)$(libjabber_la_LINK) $(am_libjabber_la_rpath) $(libjabber_la_OBJECTS) $(libjabber_la_LIBADD) $(LIBS)
+	$(AM_V_CCLD)$(libjabber_la_LINK) $(am_libjabber_la_rpath) $(libjabber_la_OBJECTS) $(libjabber_la_LIBADD) $(LIBS) -lpurple
 
 libxmpp.la: $(libxmpp_la_OBJECTS) $(libxmpp_la_DEPENDENCIES) $(EXTRA_libxmpp_la_DEPENDENCIES) 
-	$(AM_V_CCLD)$(libxmpp_la_LINK) $(am_libxmpp_la_rpath) $(libxmpp_la_OBJECTS) $(libxmpp_la_LIBADD) $(LIBS)
+	$(AM_V_CCLD)$(libxmpp_la_LINK) $(am_libxmpp_la_rpath) $(libxmpp_la_OBJECTS) $(libxmpp_la_LIBADD) $(LIBS) -lpurple
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
