diff -u -r ../pidgin-2.10.12/libpurple/protocols/oscar/Makefile.in ./libpurple/protocols/oscar/Makefile.in
--- ../pidgin-2.10.12/libpurple/protocols/oscar/Makefile.in	2015-12-31 18:19:52.000000000 -0500
+++ ./libpurple/protocols/oscar/Makefile.in	2016-05-05 04:27:46.529665679 -0400
@@ -652,13 +652,13 @@
 	}
 
 libaim.la: $(libaim_la_OBJECTS) $(libaim_la_DEPENDENCIES) $(EXTRA_libaim_la_DEPENDENCIES) 
-	$(AM_V_CCLD)$(libaim_la_LINK) $(am_libaim_la_rpath) $(libaim_la_OBJECTS) $(libaim_la_LIBADD) $(LIBS)
+	$(AM_V_CCLD)$(libaim_la_LINK) $(am_libaim_la_rpath) $(libaim_la_OBJECTS) $(libaim_la_LIBADD) $(LIBS) -lpurple
 
 libicq.la: $(libicq_la_OBJECTS) $(libicq_la_DEPENDENCIES) $(EXTRA_libicq_la_DEPENDENCIES) 
-	$(AM_V_CCLD)$(libicq_la_LINK) $(am_libicq_la_rpath) $(libicq_la_OBJECTS) $(libicq_la_LIBADD) $(LIBS)
+	$(AM_V_CCLD)$(libicq_la_LINK) $(am_libicq_la_rpath) $(libicq_la_OBJECTS) $(libicq_la_LIBADD) $(LIBS) -lpurple
 
 liboscar.la: $(liboscar_la_OBJECTS) $(liboscar_la_DEPENDENCIES) $(EXTRA_liboscar_la_DEPENDENCIES) 
-	$(AM_V_CCLD)$(liboscar_la_LINK) $(am_liboscar_la_rpath) $(liboscar_la_OBJECTS) $(liboscar_la_LIBADD) $(LIBS)
+	$(AM_V_CCLD)$(liboscar_la_LINK) $(am_liboscar_la_rpath) $(liboscar_la_OBJECTS) $(liboscar_la_LIBADD) $(LIBS) -lpurple
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
