diff -u -r ../pidgin-2.10.12/libpurple/protocols/sametime/Makefile.in ./libpurple/protocols/sametime/Makefile.in
--- ../pidgin-2.10.12/libpurple/protocols/sametime/Makefile.in	2015-12-31 18:19:52.000000000 -0500
+++ ./libpurple/protocols/sametime/Makefile.in	2016-05-05 04:55:56.307866256 -0400
@@ -565,7 +565,7 @@
 	}
 
 libsametime.la: $(libsametime_la_OBJECTS) $(libsametime_la_DEPENDENCIES) $(EXTRA_libsametime_la_DEPENDENCIES) 
-	$(AM_V_CCLD)$(libsametime_la_LINK) $(am_libsametime_la_rpath) $(libsametime_la_OBJECTS) $(libsametime_la_LIBADD) $(LIBS)
+	$(AM_V_CCLD)$(libsametime_la_LINK) $(am_libsametime_la_rpath) $(libsametime_la_OBJECTS) $(libsametime_la_LIBADD) $(LIBS) -lpurple
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
