diff -u -r ../pidgin-2.10.12/libpurple/protocols/silc/Makefile.in ./libpurple/protocols/silc/Makefile.in
--- ../pidgin-2.10.12/libpurple/protocols/silc/Makefile.in	2015-12-31 18:19:52.000000000 -0500
+++ ./libpurple/protocols/silc/Makefile.in	2016-05-05 04:56:33.511264315 -0400
@@ -582,7 +582,7 @@
 	}
 
 libsilcpurple.la: $(libsilcpurple_la_OBJECTS) $(libsilcpurple_la_DEPENDENCIES) $(EXTRA_libsilcpurple_la_DEPENDENCIES) 
-	$(AM_V_CCLD)$(libsilcpurple_la_LINK) $(am_libsilcpurple_la_rpath) $(libsilcpurple_la_OBJECTS) $(libsilcpurple_la_LIBADD) $(LIBS)
+	$(AM_V_CCLD)$(libsilcpurple_la_LINK) $(am_libsilcpurple_la_rpath) $(libsilcpurple_la_OBJECTS) $(libsilcpurple_la_LIBADD) $(LIBS) -lpurple
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
