--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -355,8 +355,8 @@ if (ASCIIDOC_PY_FOUND)
     # does not recognize it.
     add_custom_command(
       OUTPUT "${PAGE}.1"
-      COMMAND "${ASCIIDOC_PY}"
-      ARGS "${ASCIIDOC}"
+      COMMAND "${ASCIIDOC}"
+      ARGS
         --format manpage --destination-dir="${CMAKE_CURRENT_BINARY_DIR}"
         "${CMAKE_CURRENT_SOURCE_DIR}/doc/man/${PAGE}.txt"
       MAIN_DEPENDENCY "${CMAKE_CURRENT_SOURCE_DIR}/doc/man/${PAGE}.txt")
