diff -uNr lua-5.3.4/src/loslib.c lua-5.3.4.mod/src/loslib.c
--- lua-5.3.4/src/loslib.c	2016-07-18 20:58:58.000000000 +0300
+++ lua-5.3.4.mod/src/loslib.c	2017-09-14 13:42:35.922862573 +0300
@@ -115,7 +115,7 @@
 #define LUA_TMPNAMBUFSIZE	32
 
 #if !defined(LUA_TMPNAMTEMPLATE)
-#define LUA_TMPNAMTEMPLATE	"/tmp/lua_XXXXXX"
+#define LUA_TMPNAMTEMPLATE	"@TERMUX_PREFIX@/tmp/lua_XXXXXX"
 #endif
 
 #define lua_tmpnam(b,e) { \
