diff -r -u ../lua-5.3.3/src/luaconf.h ./src/luaconf.h
--- ../lua-5.3.3/src/luaconf.h	2016-05-01 22:06:09.000000000 +0200
+++ ./src/luaconf.h	2017-01-17 23:47:26.867840450 +0100
@@ -188,7 +188,7 @@
 
 #else			/* }{ */
 
-#define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT	"@TERMUX_PREFIX@/"
 #define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR "/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR "/"
 #define LUA_PATH_DEFAULT  \
