diff -uNr lua-5.1.5/etc/lua.pc lua-5.1.5.mod/etc/lua.pc
--- lua-5.1.5/etc/lua.pc	2012-02-10 11:50:30.000000000 +0200
+++ lua-5.1.5.mod/etc/lua.pc	2018-06-26 16:08:00.271151971 +0300
@@ -8,9 +8,9 @@
 R= 5.1.5
 
 # grep '^INSTALL_.*=' ../Makefile | sed 's/INSTALL_TOP/prefix/'
-prefix= /usr/local
+prefix= @TERMUX_PREFIX@
 INSTALL_BIN= ${prefix}/bin
-INSTALL_INC= ${prefix}/include
+INSTALL_INC= ${prefix}/include/lua5.1
 INSTALL_LIB= ${prefix}/lib
 INSTALL_MAN= ${prefix}/man/man1
 INSTALL_LMOD= ${prefix}/share/lua/${V}
@@ -19,13 +19,13 @@
 # canonical vars
 exec_prefix=${prefix}
 libdir=${exec_prefix}/lib
-includedir=${prefix}/include
+includedir=${prefix}/include/lua5.1
 
 Name: Lua
 Description: An Extensible Extension Language
 Version: ${R}
 Requires: 
-Libs: -L${libdir} -llua -lm
+Libs: -L${libdir} -llua5.1 -lm
 Cflags: -I${includedir}
 
 # (end of lua.pc)
