--- a/app/app.pro
+++ b/app/app.pro
@@ -19,20 +19,20 @@
 ##              INTALLS
 #########################################
 
-target.path += /usr/bin/
+target.path += @TERMUX_PREFIX@/bin/
 
 INSTALLS += target
 
 # Install icons
 unix {
     icon32.files = icons/32x32/cool-retro-term.png
-    icon32.path = /usr/share/icons/hicolor/32x32/apps
+    icon32.path = @TERMUX_PREFIX@/share/icons/hicolor/32x32/apps
     icon64.files = icons/64x64/cool-retro-term.png
-    icon64.path = /usr/share/icons/hicolor/64x64/apps
+    icon64.path = @TERMUX_PREFIX@/share/icons/hicolor/64x64/apps
     icon128.files = icons/128x128/cool-retro-term.png
-    icon128.path = /usr/share/icons/hicolor/128x128/apps
+    icon128.path = @TERMUX_PREFIX@/share/icons/hicolor/128x128/apps
     icon256.files = icons/256x256/cool-retro-term.png
-    icon256.path = /usr/share/icons/hicolor/256x256/apps
+    icon256.path = @TERMUX_PREFIX@/share/icons/hicolor/256x256/apps
 
     INSTALLS += icon32 icon64 icon128 icon256
 }
