diff -uNr e2fsprogs-1.44.4/lib/blkid/blkidP.h e2fsprogs-1.44.4.mod/lib/blkid/blkidP.h
--- e2fsprogs-1.44.4/lib/blkid/blkidP.h	2018-08-19 05:26:58.000000000 +0300
+++ e2fsprogs-1.44.4.mod/lib/blkid/blkidP.h	2018-09-22 15:22:14.795917878 +0300
@@ -110,7 +110,7 @@
 extern char *blkid_strdup(const char *s);
 extern char *blkid_strndup(const char *s, const int length);
 
-#define BLKID_CACHE_FILE "/etc/blkid.tab"
+#define BLKID_CACHE_FILE "@TERMUX_PREFIX@/etc/blkid.tab"
 
 #define BLKID_ERR_IO	 5
 #define BLKID_ERR_PROC	 9
