diff -uNr e2fsprogs-1.44.4/util/android_config.h e2fsprogs-1.44.4.mod/util/android_config.h
--- e2fsprogs-1.44.4/util/android_config.h	2018-08-19 05:26:58.000000000 +0300
+++ e2fsprogs-1.44.4.mod/util/android_config.h	2018-09-22 15:22:14.845918064 +0300
@@ -2,7 +2,7 @@
 #define HAVE_MALLOC_H 1
 #endif
 
-#define ROOT_SYSCONFDIR "/etc"
+#define ROOT_SYSCONFDIR "@TERMUX_PREFIX@/etc"
 
 #define ENABLE_LIBSPARSE 1
 
