diff -uNr e2fsprogs-1.44.4/doc/Makefile.in e2fsprogs-1.44.4.mod/doc/Makefile.in
--- e2fsprogs-1.44.4/doc/Makefile.in	2018-08-19 05:26:58.000000000 +0300
+++ e2fsprogs-1.44.4.mod/doc/Makefile.in	2018-09-22 15:22:14.779251149 +0300
@@ -17,7 +17,7 @@
 HTML=makeinfo --html --no-split
 PS2PDF=ps2pdf
 
-all:: libext2fs.info libext2fs.dvi libext2fs.html
+all:: libext2fs.info
 
 install-doc-libs: libext2fs.info libext2fs.dvi
 	$(Q) $(RM) -rf $(DESTDIR)$(infodir)/libext2fs.info*
