diff -uNr e2fsprogs-1.44.4/lib/support/nls-enable.h e2fsprogs-1.44.4.mod/lib/support/nls-enable.h
--- e2fsprogs-1.44.4/lib/support/nls-enable.h	2018-08-19 05:26:58.000000000 +0300
+++ e2fsprogs-1.44.4.mod/lib/support/nls-enable.h	2018-09-22 15:22:14.799251223 +0300
@@ -12,7 +12,7 @@
 #define NLS_CAT_NAME "e2fsprogs"
 #endif
 #ifndef LOCALEDIR
-#define LOCALEDIR "/usr/share/locale"
+#define LOCALEDIR "@TERMUX_PREFIX@/share/locale"
 #endif
 #else
 #define _(a) (a)
