diff -uNr e2fsprogs-1.44.4/lib/ext2fs/hashmap.h e2fsprogs-1.44.4.mod/lib/ext2fs/hashmap.h
--- e2fsprogs-1.44.4/lib/ext2fs/hashmap.h	2018-08-19 05:26:58.000000000 +0300
+++ e2fsprogs-1.44.4.mod/lib/ext2fs/hashmap.h	2018-09-22 15:26:16.196805586 +0300
@@ -17,14 +17,18 @@
 		struct ext2fs_hashmap_entry *next;
 		struct ext2fs_hashmap_entry *list_next;
 		struct ext2fs_hashmap_entry *list_prev;
+#ifndef __ANDROID__
 #if __GNUC_PREREQ (4, 8)
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wpedantic"
 #endif
+#endif
 	} *entries[0];
+#ifndef __ANDROID__
 #if __GNUC_PREREQ (4, 8)
 #pragma GCC diagnostic pop
 #endif
+#endif
 };
 
 struct ext2fs_hashmap *ext2fs_hashmap_create(
