diff -uNr e2fsprogs-1.44.4/misc/e4defrag.c e2fsprogs-1.44.4.mod/misc/e4defrag.c
--- e2fsprogs-1.44.4/misc/e4defrag.c	2018-08-19 05:26:58.000000000 +0300
+++ e2fsprogs-1.44.4.mod/misc/e4defrag.c	2018-09-22 15:22:14.812584606 +0300
@@ -223,7 +223,7 @@
 
 	fp = setmntent(mtab, "r");
 	if (fp == NULL) {
-		perror("Couldn't access /etc/mtab");
+		perror("Couldn't access @TERMUX_PREFIX@/etc/mtab");
 		return -1;
 	}
 
@@ -290,7 +290,7 @@
 
 	fp = setmntent(mtab, "r");
 	if (fp == NULL) {
-		perror("Couldn't access /etc/mtab");
+		perror("Couldn't access @TERMUX_PREFIX@/etc/mtab");
 		return -1;
 	}
 
