diff -uNr e2fsprogs-1.44.4/misc/util.c e2fsprogs-1.44.4.mod/misc/util.c
--- e2fsprogs-1.44.4/misc/util.c	2018-08-19 05:26:58.000000000 +0300
+++ e2fsprogs-1.44.4.mod/misc/util.c	2018-09-22 15:22:14.832584681 +0300
@@ -130,7 +130,7 @@
 	if (mount_flags & EXT2_MF_MOUNTED) {
 		fprintf(stderr, _("%s is mounted; "), device);
 		if (force >= 2) {
-			fputs(_("mke2fs forced anyway.  Hope /etc/mtab is "
+			fputs(_("mke2fs forced anyway.  Hope @TERMUX_PREFIX@/etc/mtab is "
 				"incorrect.\n"), stderr);
 			return;
 		}
