diff -u -r ../cyrus-sasl-2.1.27/saslauthd/auth_getpwent.c ./saslauthd/auth_getpwent.c
--- ../cyrus-sasl-2.1.27/saslauthd/auth_getpwent.c	2016-12-10 16:17:11.000000000 +0000
+++ ./saslauthd/auth_getpwent.c	2018-12-27 22:51:44.851604495 +0000
@@ -72,6 +72,9 @@
   /* END PARAMETERS */
   )
 {
+#ifdef __ANDROID__
+   RETURN("NO Invalid username");
+#else
     /* VARIABLES */
     struct passwd *pw;			/* pointer to passwd file entry */
     char *crpt_passwd;			/* encrypted password */
@@ -115,6 +118,7 @@
 	syslog(LOG_DEBUG, "DEBUG: auth_getpwent: OK: %s", login);
     }
     RETURN("OK");
+#endif
 }
 
 /* END FUNCTION: auth_getpwent */
