--- a/Cargo.toml
+++ b/Cargo.toml
@@ -68,7 +68,7 @@ pipewire = []
 ffmpeg = []
 
 [target.'cfg(target_os = "linux")'.dependencies]
-cpal = { git = "https://github.com/RustAudio/cpal.git", rev = "1c40a85", features = ["pipewire"] }
+cpal = { git = "https://github.com/RustAudio/cpal.git", rev = "1c40a85" }
 pulsectl-rs = { version = "0.3.2", optional = true }
 ksni = { version = "0.3", default-features = false, features = ["async-io"], optional = true }
 libpulse-binding = { version = "2.24.0", optional = true }
