diff -r -u ../coreutils-8.22/configure ./configure
--- ../coreutils-8.22/configure	2013-12-13 16:05:30.000000000 +0100
+++ ./configure	2014-02-12 18:21:28.000000000 +0100
@@ -39537,7 +39537,7 @@
 
 
 # Determine how to get the list of mounted file systems.
-ac_list_mounted_fs=
+ac_list_mounted_fs=found
 
 # If the getmntent function is available but not in the standard library,
 # make sure LIBS contains the appropriate -l option.
@@ -40077,12 +40077,6 @@
   esac
 fi
 
-if test -z "$ac_list_mounted_fs"; then
-  as_fn_error $? "could not determine how to read list of mounted file systems" "$LINENO" 5
-  # FIXME -- no need to abort building the whole package
-  # Can't build mountlist.c or anything that needs its functions
-fi
-
 if test $ac_list_mounted_fs = found; then :
   gl_cv_list_mounted_fs=yes
 else
