diff -u -r ../coreutils-8.22/lib/mountlist.c ./lib/mountlist.c
--- ../coreutils-8.22/lib/mountlist.c	2013-12-04 15:53:33.000000000 +0100
+++ ./lib/mountlist.c	2014-01-07 19:12:03.794066905 +0100
@@ -428,6 +428,9 @@
 
 #ifdef MOUNTED_GETMNTENT1 /* GNU/Linux, 4.3BSD, SunOS, HP-UX, Dynix, Irix.  */
   {
+# ifdef __ANDROID__
+    return NULL;
+# else
     struct mntent *mnt;
     char const *table = MOUNTED;
     FILE *fp;
@@ -454,6 +457,7 @@
 
     if (endmntent (fp) == 0)
       goto free_then_fail;
+# endif
   }
 #endif /* MOUNTED_GETMNTENT1. */
 
