diff -u -r ../coreutils-8.22/src/ls.c ./src/ls.c
--- ../coreutils-8.22/src/ls.c	2013-12-04 15:48:30.000000000 +0100
+++ ./src/ls.c	2014-02-14 16:11:48.000000000 +0100
@@ -1641,6 +1641,15 @@
       }
   }
 
+  print_with_color = isatty (STDOUT_FILENO);
+  if (print_with_color)
+  {
+	  /* Don't use TAB characters in output.  Some terminal
+	     emulators can't handle the combination of tabs and
+	     color codes on the same line.  */
+	  tabsize = 0;
+  }
+
   while (true)
     {
       int oi = -1;
@@ -1889,25 +1898,7 @@
 
         case COLOR_OPTION:
           {
-            int i;
-            if (optarg)
-              i = XARGMATCH ("--color", optarg, color_args, color_types);
-            else
-              /* Using --color with no argument is equivalent to using
-                 --color=always.  */
-              i = color_always;
-
-            print_with_color = (i == color_always
-                                || (i == color_if_tty
-                                    && isatty (STDOUT_FILENO)));
-
-            if (print_with_color)
-              {
-                /* Don't use TAB characters in output.  Some terminal
-                   emulators can't handle the combination of tabs and
-                   color codes on the same line.  */
-                tabsize = 0;
-              }
+	    // Do nothing - color has been patched to color_if_tty.
             break;
           }
 
@@ -2098,11 +2089,13 @@
             }
         }
 
+#ifndef __ANDROID__
       /* Note we leave %5b etc. alone so user widths/flags are honored.  */
       if (strstr (long_time_format[0], "%b")
           || strstr (long_time_format[1], "%b"))
         if (!abmon_init ())
           error (0, 0, _("error initializing month strings"));
+#endif
     }
 
   return optind;
