diff -u -r ../coreutils-8.22/src/pinky.c ./src/pinky.c
--- ../coreutils-8.22/src/pinky.c	2013-12-04 15:48:30.000000000 +0100
+++ ./src/pinky.c	2014-02-07 01:18:15.000000000 +0100
@@ -242,13 +242,13 @@
         printf (" %19s", _("        ???"));
       else
         {
-          char *const comma = strchr (pw->pw_gecos, ',');
+          char *const comma = strchr ("", ',');
           char *result;
 
           if (comma)
             *comma = '\0';
 
-          result = create_fullname (pw->pw_gecos, pw->pw_name);
+          result = create_fullname ("", pw->pw_name);
           printf (" %-19.19s", result);
           free (result);
         }
@@ -323,13 +323,13 @@
     }
   else
     {
-      char *const comma = strchr (pw->pw_gecos, ',');
+      char *const comma = strchr ("", ',');
       char *result;
 
       if (comma)
         *comma = '\0';
 
-      result = create_fullname (pw->pw_gecos, pw->pw_name);
+      result = create_fullname ("", pw->pw_name);
       printf (" %s", result);
       free (result);
     }
