--- a/ytfzf
+++ b/ytfzf
@@ -20,9 +20,9 @@
 : "${YTFZF_EXTENSIONS_DIR:=$YTFZF_CONFIG_DIR/extensions}"
 : "${YTFZF_CUSTOM_SCRAPERS_DIR:=$YTFZF_CONFIG_DIR/scrapers}"
 
-: "${YTFZF_SYSTEM_ADDON_DIR:=/usr/local/share/ytfzf/addons}"
+: "${YTFZF_SYSTEM_ADDON_DIR:=@TERMUX_PREFIX@/share/ytfzf/addons}"
 
-: "${YTFZF_TEMP_DIR:="${TMPDIR:-/tmp}"/ytfzf-$(id -u)}"
+: "${YTFZF_TEMP_DIR:="${TMPDIR:-@TERMUX_PREFIX@/tmp}"/ytfzf-$(id -u)}"
 
 : "${YTFZF_LOGFILE:="/dev/stderr"}"
 
