--- a/ytfzf
+++ b/ytfzf
@@ -189,7 +189,7 @@
 : "${YTFZF_URL_HANDLERS_DIR:=$YTFZF_CONFIG_DIR/url-handlers}"
 : "${YTFZF_CUSTOM_THUMBNAILS_DIR:=$YTFZF_CONFIG_DIR/thumbnails}"
 
-: "${YTFZF_SYSTEM_ADDON_DIR:=/usr/local/share/ytfzf/addons}"
+: "${YTFZF_SYSTEM_ADDON_DIR:=@TERMUX_PREFIX@/share/ytfzf/addons}"
 
 [ -f "$YTFZF_CONFIG_FILE" ] && . "$YTFZF_CONFIG_FILE"
 #}}}
@@ -249,7 +249,7 @@
 
 : "${thumbnail_viewer:=ueberzug}"
 
-: "${w3mimgdisplay_path:=/usr/lib/w3m/w3mimgdisplay}"
+: "${w3mimgdisplay_path:=@TERMUX_PREFIX@/lib/w3m/w3mimgdisplay}"
 
 # shortcuts
 : "${download_shortcut:=alt-d}"
