--- a/config/apache2/z-push-autodiscover.conf
+++ b/config/apache2/z-push-autodiscover.conf
@@ -1,6 +1,6 @@
 # Z-Push AutoDiscover - ActiveSync over-the-air - default Apache configuration
 <IfModule mod_alias.c>
-    Alias /AutoDiscover/AutoDiscover.xml "/usr/share/z-push/autodiscover/autodiscover.php"
-    Alias /Autodiscover/Autodiscover.xml "/usr/share/z-push/autodiscover/autodiscover.php"
-    Alias /autodiscover/autodiscover.xml "/usr/share/z-push/autodiscover/autodiscover.php"
+    Alias /AutoDiscover/AutoDiscover.xml "@TERMUX_PREFIX@/share/z-push/autodiscover/autodiscover.php"
+    Alias /Autodiscover/Autodiscover.xml "@TERMUX_PREFIX@/share/z-push/autodiscover/autodiscover.php"
+    Alias /autodiscover/autodiscover.xml "@TERMUX_PREFIX@/share/z-push/autodiscover/autodiscover.php"
 </IfModule>
