--- a/src/backend/kopano/config.php
+++ b/src/backend/kopano/config.php
@@ -31,17 +31,17 @@
 //
 // Depending on your setup, it might be advisable to change the lines below to one defined with your
 // default socket location.
-// Normally "default:" points to the default setting ("file:///var/run/kopano/server.sock")
+// Normally "default:" points to the default setting ("file://@TERMUX_PREFIX@/var/run/kopano/server.sock")
 // Examples: define("MAPI_SERVER", "default:");
 //           define("MAPI_SERVER", "http://localhost:236/kopano");
 //           define("MAPI_SERVER", "https://localhost:237/kopano");
-//           define("MAPI_SERVER", "file:///var/run/kopano/server.sock");
+//           define("MAPI_SERVER", "file://@TERMUX_PREFIX@/var/run/kopano/server.sock");
 // If you are using ZCP >= 7.2.0, set it to the zarafa location, e.g.
 //           define("MAPI_SERVER", "http://localhost:236/zarafa");
 //           define("MAPI_SERVER", "https://localhost:237/zarafa");
-//           define("MAPI_SERVER", "file:///var/run/zarafad/server.sock");
+//           define("MAPI_SERVER", "file://@TERMUX_PREFIX@/var/run/zarafad/server.sock");
 // For ZCP versions prior to 7.2.0 the socket location is different (http(s) sockets are the same):
-//           define("MAPI_SERVER", "file:///var/run/zarafa");
+//           define("MAPI_SERVER", "file://@TERMUX_PREFIX@/var/run/zarafa");
 
 define('MAPI_SERVER', 'default:');
 
