--- a/src/config.php
+++ b/src/config.php
@@ -68,7 +68,7 @@
  *           State migration script is available, more informations: https://wiki.z-hub.io/x/xIAa
  */
     define('STATE_MACHINE', 'FILE');
-    define('STATE_DIR', '/var/lib/z-push/');
+    define('STATE_DIR', '@TERMUX_PREFIX@/var/lib/z-push/');
 
 /**********************************************************************************
  *  IPC - InterProcessCommunication
@@ -123,7 +123,7 @@
     $specialLogUsers = array();
 
     // Filelog settings
-    define('LOGFILEDIR', '/var/log/z-push/');
+    define('LOGFILEDIR', '@TERMUX_PREFIX@/var/log/z-push/');
     define('LOGFILE', LOGFILEDIR . 'z-push.log');
     define('LOGERRORFILE', LOGFILEDIR . 'z-push-error.log');
 
