--- a/src/backend/imap/config.php
+++ b/src/backend/imap/config.php
@@ -204,7 +204,7 @@
 
 // A file containing file mime types->extension mappings.
 //  SELINUX users: make sure the file has a security context accesible by your apache/php-fpm process
-define('SYSTEM_MIME_TYPES_MAPPING', '/etc/mime.types');
+define('SYSTEM_MIME_TYPES_MAPPING', '@TERMUX_PREFIX@/etc/mime.types');
 
 
 // Use BackendCalDAV for Meetings. You cannot hope to get that functionality working without a caldav backend.
