--- a/src/autodiscover/config.php
+++ b/src/autodiscover/config.php
@@ -87,7 +87,7 @@
 
     define('LOGBACKEND', 'filelog');
 
-    define('LOGFILEDIR', '/var/log/z-push/');
+    define('LOGFILEDIR', '@TERMUX_PREFIX@/var/log/z-push/');
     define('LOGFILE', LOGFILEDIR . 'autodiscover.log');
     define('LOGERRORFILE', LOGFILEDIR . 'autodiscover-error.log');
     define('LOGLEVEL', LOGLEVEL_INFO);
