--- ../config.cpp.orig	2020-06-06 14:29:46.809335434 +0200
+++ ./src/config.cpp	2020-06-06 09:57:26.580039189 +0200
@@ -13,18 +13,18 @@
     // Data files used by Umoria
     // NOTE: use relative paths to the executable binary.
     namespace files {
-        const std::string splash_screen = "data/splash.txt";
-        const std::string welcome_screen = "data/welcome.txt";
-        const std::string license = "LICENSE";
-        const std::string versions_history = "data/versions.txt";
-        const std::string help = "data/help.txt";
-        const std::string help_wizard = "data/help_wizard.txt";
-        const std::string help_roguelike = "data/rl_help.txt";
-        const std::string help_roguelike_wizard = "data/rl_help_wizard.txt";
-        const std::string death_tomb = "data/death_tomb.txt";
-        const std::string death_royal = "data/death_royal.txt";
-        const std::string scores = "scores.dat";
-        std::string save_game = "game.sav";
+        const std::string splash_screen = "@TERMUX_PREFIX@/lib/games/moria/data/splash.txt";
+        const std::string welcome_screen = "@TERMUX_PREFIX@/lib/games/moria/data/welcome.txt";
+        const std::string license = "@TERMUX_PREFIX@/share/doc/moria/LICENSE";
+        const std::string versions_history = "@TERMUX_PREFIX@/lib/games/moria/data/versions.txt";
+        const std::string help = "@TERMUX_PREFIX@/lib/games/moria/data/help.txt";
+        const std::string help_wizard = "@TERMUX_PREFIX@/lib/games/moria/data/help_wizard.txt";
+        const std::string help_roguelike = "@TERMUX_PREFIX@/lib/games/moria/data/rl_help.txt";
+        const std::string help_roguelike_wizard = "@TERMUX_PREFIX@/lib/games/moria/data/rl_help_wizard.txt";
+        const std::string death_tomb = "@TERMUX_PREFIX@/lib/games/moria/data/death_tomb.txt";
+        const std::string death_royal = "@TERMUX_PREFIX@/lib/games/moria/data/death_royal.txt";
+        const std::string scores = "@TERMUX_PREFIX@/lib/games/moria/scores.dat";
+        std::string save_game = "@TERMUX_PREFIX@/lib/games/moria/game.sav";
     } // namespace files
 
     // Game options as set on startup and with `=` set options command -CJS-
