diff --git a/source/death.c b/source/death.c
index 87102eb..e5c951b 100644
--- a/source/death.c
+++ b/source/death.c
@@ -214,9 +214,9 @@ int f, l;
 		  sbuf.st_ino);
 #else
 #ifdef __linux__
-  (void) sprintf (lockname, "/tmp/moria.%ld", sbuf.st_ino);
+  (void) sprintf (lockname, "@TERMUX_PREFIX@/tmp/moria.%ld", sbuf.st_ino);
 #else
-  (void) sprintf (lockname, "/tmp/moria.%d", sbuf.st_ino);
+  (void) sprintf (lockname, "@TERMUX_PREFIX@/tmp/moria.%d", sbuf.st_ino);
 #endif
 #endif
   if (l & LOCK_UN)
