diff --git a/source/config.h b/source/config.h
index bd64c8f..b0b7911 100644
--- a/source/config.h
+++ b/source/config.h
@@ -41,7 +41,7 @@ some of the system defines set up here.
 
 /* If compiling on Debian (also works on other versions of Linux), 
    define this. */
-#define DEBIAN_LINUX
+//#define DEBIAN_LINUX
 
 /* If you are compiling on an ultrix/4.2BSD/Dynix/etc. version of UNIX,
    define this.  It is not needed for SUNs.  */
@@ -221,20 +221,20 @@ some of the system defines set up here.
 
 #else
 
-#if 0
+#if 1
 /* Debian standards for file location */
 /* This must be unix; change file names as appropriate.  */
 #define MORIA_SAV	".moria-save"
-#define MORIA_HOU	"/etc/moria-hours"
-#define MORIA_MOR	"/usr/lib/games/moria/news"
-#define MORIA_GPL	"/usr/lib/games/moria/COPYING"
-#define MORIA_TOP	"/var/games/moria/scores"
-#define MORIA_HELP	"/usr/lib/games/moria/roglcmds.hlp"
-#define MORIA_ORIG_HELP	"/usr/lib/games/moria/origcmds.hlp"
-#define MORIA_WIZ_HELP	"/usr/lib/games/moria/rwizcmds.hlp"
-#define MORIA_OWIZ_HELP	"/usr/lib/games/moria/owizcmds.hlp"
-#define MORIA_WELCOME	"/usr/lib/games/moria/welcome.hlp"
-#define MORIA_VER	"/usr/lib/games/moria/version.hlp"
+#define MORIA_HOU	"@TERMUX_PREFIX@/etc/moria-hours"
+#define MORIA_MOR	"@TERMUX_PREFIX@/lib/games/moria/news"
+#define MORIA_GPL	"@TERMUX_PREFIX@lib/games/moria/COPYING"
+#define MORIA_TOP	"@TERMUX_PREFIX@/var/games/moria/scores"
+#define MORIA_HELP	"@TERMUX_PREFIX@/lib/games/moria/roglcmds.hlp"
+#define MORIA_ORIG_HELP	"@TERMUX_PREFIX@/lib/games/moria/origcmds.hlp"
+#define MORIA_WIZ_HELP	"@TERMUX_PREFIX@/lib/games/moria/rwizcmds.hlp"
+#define MORIA_OWIZ_HELP	"@TERMUX_PREFIX@/lib/games/moria/owizcmds.hlp"
+#define MORIA_WELCOME	"@TERMUX_PREFIX@/lib/games/moria/welcome.hlp"
+#define MORIA_VER	"@TERMUX_PREFIX@/lib/games/moria/version.hlp"
 
 #else
 
