--- a/src/app/CMakeLists.txt
+++ b/src/app/CMakeLists.txt
@@ -9,7 +9,7 @@ foreach(widget_file ${widget_files})
 
   add_custom_command(
     OUTPUT ${output_fn}
-    COMMAND ${CMAKE_BINARY_DIR}/bin/gen --input ${widget_file} --widgetid ${widget_name} > ${output_fn}.tmp
+    COMMAND gen --input ${widget_file} --widgetid ${widget_name} > ${output_fn}.tmp
     COMMAND ${CMAKE_COMMAND} -E copy_if_different ${output_fn}.tmp ${output_fn}
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
     MAIN_DEPENDENCY ${widget_file}
@@ -24,7 +24,7 @@ set(pref_xml ${CMAKE_SOURCE_DIR}/data/pref.xml)
 set(output_fn ${CMAKE_CURRENT_BINARY_DIR}/pref.xml.h)
 add_custom_command(
   OUTPUT ${output_fn}
-  COMMAND ${CMAKE_BINARY_DIR}/bin/gen --input ${pref_xml} --pref-h > ${output_fn}.tmp
+  COMMAND gen --input ${pref_xml} --pref-h > ${output_fn}.tmp
   COMMAND ${CMAKE_COMMAND} -E copy_if_different ${output_fn}.tmp ${output_fn}
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
   MAIN_DEPENDENCY ${pref_xml}
@@ -34,7 +34,7 @@ list(APPEND generated_files ${output_fn})
 set(output_fn ${CMAKE_CURRENT_BINARY_DIR}/pref.xml.cpp)
 add_custom_command(
   OUTPUT ${output_fn}
-  COMMAND ${CMAKE_BINARY_DIR}/bin/gen --input ${pref_xml} --pref-cpp > ${output_fn}.tmp
+  COMMAND gen --input ${pref_xml} --pref-cpp > ${output_fn}.tmp
   COMMAND ${CMAKE_COMMAND} -E copy_if_different ${output_fn}.tmp ${output_fn}
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
   MAIN_DEPENDENCY ${pref_xml}
@@ -46,7 +46,7 @@ set(skin_xml ${CMAKE_SOURCE_DIR}/data/skins/default/skin.xml)
 set(output_fn ${CMAKE_CURRENT_BINARY_DIR}/skin.xml.h)
 add_custom_command(
   OUTPUT ${output_fn}
-  COMMAND ${CMAKE_BINARY_DIR}/bin/gen --input ${skin_xml} --skin > ${output_fn}.tmp
+  COMMAND gen --input ${skin_xml} --skin > ${output_fn}.tmp
   COMMAND ${CMAKE_COMMAND} -E copy_if_different ${output_fn}.tmp ${output_fn}
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
   MAIN_DEPENDENCY ${skin_xml}
