--- a/src/base/launcher.cpp
+++ b/src/base/launcher.cpp
@@ -137,7 +137,7 @@ bool open_file(const std::string& file)
 
 #else
 
-  ret = spawn_and_wait({"xdg-open", file});
+  ret = spawn_and_wait({"xdg-utils-xdg-open", file});
 
 #endif
 
@@ -181,7 +181,7 @@ bool open_folder(const std::string& _file)
   if (!base::is_directory(file))
     file = base::get_file_path(file);
 
-  int ret = spawn_and_wait({"xdg-open", file});
+  int ret = spawn_and_wait({"xdg-utils-xdg-open", file});
   return (ret == 0);
 
 #endif
