diff -uNr inetutils-2.3/configure inetutils-2.3.mod/configure
--- inetutils-2.3/configure	2022-07-08 09:14:46.000000000 +0100
+++ inetutils-2.3.mod/configure	2022-07-22 22:00:47.478678000 +0100
@@ -20452,7 +20452,7 @@
           else
             if case "$gl_cv_have_weak" in *yes) true;; *) false;; esac; then
 
-printf "%s\n" "#define USE_POSIX_THREADS_WEAK 1" >>confdefs.h
+#printf "%s\n" "#define USE_POSIX_THREADS_WEAK 1" >>confdefs.h
 
               LIBTHREAD= LTLIBTHREAD=
             else
