diff --git a/src/transport.cpp b/src/transport.cpp
index 0fa9afc..c733853 100644
--- a/src/transport.cpp
+++ b/src/transport.cpp
@@ -271,9 +271,13 @@ void Transport::readTransportPasswordFinished(QKeychain::Job *baseJob)
     Q_ASSERT(job);
     if (job->error()) {
         d->password.clear();
+#ifdef __TERMUX__
+        d->passwordLoaded = true;
+#else
         d->passwordLoaded = false;
         qCWarning(MAILTRANSPORT_LOG) << "We have an error during reading password for" << id() << job->errorString();
         Q_EMIT passwordChanged();
+#endif
     } else {
         setPassword(job->textData());
     }
