diff -uNr gtk-2.24.32/docs/tools/widgets.c gtk-2.24.32.mod/docs/tools/widgets.c
--- gtk-2.24.32/docs/tools/widgets.c	2018-01-08 23:35:39.000000000 +0200
+++ gtk-2.24.32.mod/docs/tools/widgets.c	2018-06-23 17:46:59.147633140 +0300
@@ -492,7 +492,7 @@
   picker = gtk_file_chooser_button_new ("File Chooser Button",
 		  			GTK_FILE_CHOOSER_ACTION_OPEN);
   gtk_widget_set_size_request (picker, 150, -1);
-  gtk_file_chooser_set_filename (GTK_FILE_CHOOSER (picker), "/etc/yum.conf");
+  gtk_file_chooser_set_filename (GTK_FILE_CHOOSER (picker), "@TERMUX_PREFIX@/etc/yum.conf");
   gtk_container_add (GTK_CONTAINER (align), picker);
   gtk_box_pack_start (GTK_BOX (vbox2), align, FALSE, FALSE, 0);
   gtk_box_pack_start (GTK_BOX (vbox2),
