diff -uNr gtk-2.24.32/modules/printbackends/cups/gtkprintbackendcups.c gtk-2.24.32.mod/modules/printbackends/cups/gtkprintbackendcups.c
--- gtk-2.24.32/modules/printbackends/cups/gtkprintbackendcups.c	2018-01-08 23:35:39.000000000 +0200
+++ gtk-2.24.32.mod/modules/printbackends/cups/gtkprintbackendcups.c	2018-06-23 18:15:19.251054007 +0300
@@ -3519,7 +3519,7 @@
 
 /* Ordering matters for default preference */
 static const char *lpoptions_locations[] = {
-  "/etc/cups/lpoptions",
+  "@TERMUX_PREFIX@/etc/cups/lpoptions",
   ".lpoptions", 
   ".cups/lpoptions"
 };
