diff -uNr gtk-2.24.32/demos/testpixbuf-color.c gtk-2.24.32.mod/demos/testpixbuf-color.c
--- gtk-2.24.32/demos/testpixbuf-color.c	2018-01-08 23:35:39.000000000 +0200
+++ gtk-2.24.32.mod/demos/testpixbuf-color.c	2018-06-23 17:45:29.427628521 +0300
@@ -6,7 +6,7 @@
 
 #include <gtk/gtk.h>
 
-#define ICC_PROFILE             "/usr/share/color/icc/bluish.icc"
+#define ICC_PROFILE             "@TERMUX_PREFIX@/share/color/icc/bluish.icc"
 #define ICC_PROFILE_SIZE        3966
 
 static gboolean
