diff -u -r ../kiwix-tools-3.7.0/meson.build ./meson.build
--- ../kiwix-tools-3.7.0/meson.build	2024-03-13 14:17:32.000000000 +0000
+++ ./meson.build	2024-10-12 11:18:32.796310063 +0000
@@ -18,8 +18,8 @@
 thread_dep = dependency('threads')
 libzim_dep = dependency('libzim', version:'>=9.0.0', static:static_linkage)
 libzim_dep = dependency('libzim', version:'<10.0.0', static:static_linkage)
-kiwixlib_dep = dependency('kiwix', version:'>=13.0.0', static:static_linkage)
-kiwixlib_dep = dependency('kiwix', version:'<14.0.0', static:static_linkage)
+kiwixlib_dep = dependency('libkiwix', version:'>=13.0.0', static:static_linkage)
+kiwixlib_dep = dependency('libkiwix', version:'<15.0.0', static:static_linkage)
 
 all_deps = [thread_dep, kiwixlib_dep, libzim_dep]
 
