diff -uNr gobject-introspection.mod/giscanner/giscannermodule.c gobject-introspection/giscanner/giscannermodule.c
--- gobject-introspection.mod/giscanner/giscannermodule.c	2021-06-03 06:05:53.443619406 +0000
+++ gobject-introspection/giscanner/giscannermodule.c	2021-06-03 08:11:21.258099293 +0000
@@ -23,7 +23,7 @@
 #  include "config.h"
 #endif
 #define PY_SSIZE_T_CLEAN
-#include <Python.h>
+#include <python3.9/Python.h>
 #include "sourcescanner.h"
 
 #include <glib-object.h>
diff -uNr gobject-introspection.mod/giscanner/meson.build gobject-introspection/giscanner/meson.build
--- gobject-introspection.mod/giscanner/meson.build	2021-06-03 06:05:53.444619492 +0000
+++ gobject-introspection/giscanner/meson.build	2021-06-03 06:28:13.917426670 +0000
@@ -105,24 +105,12 @@
   dependencies: [glib_dep, gobject_dep, gio_dep, gmodule_dep],
 )
 
-# https://github.com/mesonbuild/meson/issues/4117
-if host_machine.system() == 'windows'
-  python_ext_dep = python.dependency()
-else
-  python_ext_dep = python.dependency().partial_dependency(compile_args: true)
-endif
-
-if not python_ext_dep.found()
-  # For example if Python is 32bit but we require a 64bit variant
-  error('Python installation not useable')
-endif
 
 giscanner_pymod = python.extension_module('_giscanner', ['giscannermodule.c'],
   link_with: giscanner_lib,
   c_args: gi_hidden_visibility_cflags + custom_c_args,
   include_directories : configinc,
-  dependencies: [glib_dep, gobject_dep, gio_dep, gmodule_dep,
-                 python_ext_dep],
+  dependencies: [glib_dep, gobject_dep, gio_dep, gmodule_dep],
   install: true,
   install_dir: giscannerdir,
 )
diff -uNr gobject-introspection.mod/meson.build gobject-introspection/meson.build
--- gobject-introspection.mod/meson.build	2021-06-03 06:05:53.446619665 +0000
+++ gobject-introspection/meson.build	2021-06-03 06:22:51.080531636 +0000
@@ -3,6 +3,7 @@
   meson_version: '>= 0.50.1',
   default_options: [
     'warning_level=1',
+    'c_link_args=-lpython3.9',
     'buildtype=debugoptimized',
   ],
 )
@@ -158,7 +159,7 @@
   fallback : ['libffi', 'ffi_dep'])
 
 # python headers
-cc.check_header('Python.h', dependencies: [python.dependency()], required: true)
+
 
 # cairo
 cairo_option = get_option('cairo')
diff -uNr gobject-introspection.mod/meson_options.txt gobject-introspection/meson_options.txt
--- gobject-introspection.mod/meson_options.txt	2021-06-03 06:05:53.446619665 +0000
+++ gobject-introspection/meson_options.txt	2021-06-03 06:35:51.235935954 +0000
@@ -26,7 +26,7 @@
   description: 'Intermediate prefix for gir installation under ${prefix}'
 )
 
-option('gi_cross_use_prebuilt_gi', type: 'boolean', value : false,
+option('gi_cross_use_prebuilt_gi', type: 'boolean', value : true,
   description: 'Use gobject introspection tools installed in the build system (useful when cross-compiling)'
 )
 
@@ -38,7 +38,7 @@
   description: 'Use a ldd wrapper instead of system ldd command in giscanner (useful when cross-compiling)'
 )
 
-option('build_introspection_data', type: 'boolean', value : true,
+option('build_introspection_data', type: 'boolean', value : false,
   description: 'Build introspection data (.gir and .typelib files) in addition to library and tools'
 )
 
