--- a/meson.build
+++ b/meson.build
@@ -251,9 +249,9 @@
 pkgconfig_variables = [
   'datadir=' + '${prefix}' / get_option('datadir'),
   'bindir=' + '${prefix}' / get_option('bindir'),
-  'g_ir_scanner=${bindir}/g-ir-scanner',
-  'g_ir_compiler=${bindir}/g-ir-compiler@0@'.format(exe_ext),
-  'g_ir_generate=${bindir}/g-ir-generate@0@'.format(exe_ext),
+  'g_ir_scanner=g-ir-scanner',
+  'g_ir_compiler=g-ir-compiler@0@'.format(exe_ext),
+  'g_ir_generate=g-ir-generate@0@'.format(exe_ext),
   'gidatadir=${datadir}/gobject-introspection-1.0',
   'girdir=' + gir_dir_pc_prefix / 'gir-1.0',
   'typelibdir=${libdir}/girepository-1.0',
