diff -u -r ../cboard-0.7.3/libchess/pgn.c ./libchess/pgn.c
--- ../cboard-0.7.3/libchess/pgn.c	2015-01-14 21:18:45.000000000 -0500
+++ ./libchess/pgn.c	2015-07-11 06:55:57.087877710 -0400
@@ -813,7 +813,6 @@
     time_t now;
     char tbuf[11] = {0};
     struct tm *tp;
-    struct passwd *pw = getpwuid(getuid());
     char host[64] = { 0 };
 
     time(&now);
@@ -834,7 +833,7 @@
     if (pgn_tag_add(&g->tag, "Round", "-") != E_PGN_OK)
 	warn("pgn_tag_add()");
 
-    if (pgn_tag_add(&g->tag, "White", pw->pw_gecos) != E_PGN_OK)
+    if (pgn_tag_add(&g->tag, "White", "You") != E_PGN_OK)
 	warn("pgn_tag_add()");
 
     if (pgn_tag_add(&g->tag, "Black", "?") != E_PGN_OK)
