diff -u -r ../raw/sched/src/com/android/sched/vfs/CaseInsensitiveFS.java ./sched/src/com/android/sched/vfs/CaseInsensitiveFS.java
--- ../raw/sched/src/com/android/sched/vfs/CaseInsensitiveFS.java	2016-10-03 10:53:49.000000000 -0400
+++ ./sched/src/com/android/sched/vfs/CaseInsensitiveFS.java	2016-10-03 17:21:25.053176313 -0400
@@ -78,7 +78,7 @@
   @Nonnull
   public static final MessageDigestPropertyId ALGO = MessageDigestPropertyId.create(
       "sched.vfs.case-insensitive.algo", "Algorithm used to encode a path name").addDefaultValue(
-      "SHA");
+      "MD5");
 
   @Nonnull
   public static final BooleanPropertyId DEBUG = BooleanPropertyId.create(
