diff -uNr libX11-1.7.2/configure libX11-1.7.2.mod/configure
--- libX11-1.7.2/configure	2021-06-06 19:49:01.000000000 +0300
+++ libX11-1.7.2.mod/configure	2021-07-17 02:12:57.003231884 +0300
@@ -20605,6 +20605,7 @@
 fi
 
 
+if false; then
 # Check for flag to avoid builtin definitions - assumes unix is predefined,
 # which is not the best choice for supporting other OS'es, but covers most
 # of the ones we need for now.
@@ -20653,6 +20654,7 @@
 	fi
 fi
 rm -f conftest.$ac_ext
+fi
 
 
 
