diff -r -u old/Project/CMake/libzen.pc.in ./Project/CMake/libzen.pc.in
--- old/Project/CMake/libzen.pc.in	2021-03-25 15:29:18.000000000 +0100
+++ ./Project/CMake/libzen.pc.in	2021-03-26 12:34:30.820891100 +0100
@@ -3,10 +3,10 @@
 libdir=@LIB_INSTALL_DIR@
 includedir=@INCLUDE_INSTALL_DIR@
 Unicode=@ZenLib_Unicode@
-Libs_Static=@LIB_INSTALL_DIR@/libzen.a -lpthread
+Libs_Static=@LIB_INSTALL_DIR@/libzen.a
 
 Name: libzen
 Version: @ZenLib_VERSION@
 Description: ZenLib
-Libs: -L${libdir} -lzen -lpthread
+Libs: -L${libdir} -lzen
 Cflags: -I${includedir} @ZenLib_CXXFLAGS@
Only in ./Project: CodeBlocks
Only in ./Project: Coverity
Only in ./Project/GNU/Library: Makefile.am
Only in ./Project/GNU/Library: autogen.sh
diff -r -u old/Project/GNU/Library/configure.ac ./Project/GNU/Library/configure.ac
--- old/Project/GNU/Library/configure.ac	2021-03-25 15:29:19.000000000 +0100
+++ ./Project/GNU/Library/configure.ac	2021-03-26 12:34:30.841028100 +0100
@@ -144,10 +144,10 @@
 dnl Common
 dnl
 if test "$enable_shared" = "yes"; then
-    ZenLib_LIBS="-L$(pwd)/.libs -lzen -lpthread $ZenLib_LIBS"
+    ZenLib_LIBS="-L$(pwd)/.libs -lzen $ZenLib_LIBS"
 fi
 if test "$enable_static" = "yes"; then
-    ZenLib_LIBS_Static="$(pwd)/.libs/libzen.a -lpthread $ZenLib_LIBS_Static"
+    ZenLib_LIBS_Static="$(pwd)/.libs/libzen.a $ZenLib_LIBS_Static"
 fi
 
 dnl #########################################################################
@@ -252,7 +252,7 @@
 dnl -------------------------------------------------------------------------
 dnl External libs
 dnl
-LDFLAGS="$LDFLAGS -lpthread"
+LDFLAGS="$LDFLAGS"
 
 dnl -------------------------------------------------------------------------
 dnl Check if thread safe variants of time functions are available
Only in ./Project/GNU/Library: libzen-config.in
diff -r -u old/Project/GNU/Library/libzen.pc.in ./Project/GNU/Library/libzen.pc.in
--- old/Project/GNU/Library/libzen.pc.in	2021-03-25 15:29:18.000000000 +0100
+++ ./Project/GNU/Library/libzen.pc.in	2021-03-26 12:34:52.385894900 +0100
@@ -4,10 +4,10 @@
 includedir=@includedir@
 Unicode=@ZenLib_Unicode@
 WstringMissing=@ZenLib_wstring_missing@
-Libs_Static=@libdir@/libzen.a -lpthread
+Libs_Static=@libdir@/libzen.a
 
 Name: libzen
 Version: @PACKAGE_VERSION@
 Description: ZenLib
-Libs: -L${libdir} -lzen -lpthread
+Libs: -L${libdir} -lzen
 Cflags: -I${includedir} @ZenLib_CXXFLAGS@
