diff -u -r ../cache/tmp-checkout/go.mod ./go.mod
--- ../cache/tmp-checkout/go.mod	2024-09-11 19:37:06.886258923 +0000
+++ ./go.mod	2024-09-11 19:38:40.638038867 +0000
@@ -493,5 +493,7 @@
 // Use our fork xorm. go.work currently overrides this and points to the local ./pkg/util/xorm directory.
 replace xorm.io/xorm => github.com/grafana/grafana/pkg/util/xorm v0.0.1
 
+replace github.com/hashicorp/go-sockaddr v1.0.6 => ./go-sockaddr
+
 // lock for mysql tsdb compat
 replace github.com/go-sql-driver/mysql => github.com/go-sql-driver/mysql v1.7.1
