diff --git a/go.mod b/go.mod
index bc186b5e..e387a798 100644
--- a/go.mod
+++ b/go.mod
@@ -580,6 +580,9 @@ exclude github.com/mattn/go-sqlite3 v2.0.3+incompatible
 // Use our fork xorm. go.work currently overrides this and points to the local ./pkg/util/xorm directory.
 replace xorm.io/xorm => github.com/grafana/grafana/pkg/util/xorm v0.0.1
 
+replace github.com/hashicorp/go-sockaddr v1.0.6 => ./go-sockaddr
+replace github.com/dolthub/go-mysql-server v0.19.1-0.20250206012855-c216e59c21a7 => ./go-mysql-server
+
 // lock for mysql tsdb compat
 replace github.com/go-sql-driver/mysql => github.com/go-sql-driver/mysql v1.7.1
 
