--- ./vendor/github.com/dolthub/go-mysql-server/sql/rowexec/proc.go.orig	2025-05-21 05:13:24.414802891 +0000
+++ ./vendor/github.com/dolthub/go-mysql-server/sql/rowexec/proc.go	2025-05-21 05:14:16.421110130 +0000
@@ -237,9 +237,9 @@
 	selectSeen := false
 
 	// It's technically valid to make an infinite loop, but we don't want to actually allow that
-	const maxIterationCount = 10_000_000_000
+	const maxIterationCount uint64 = 10_000_000_000
 
-	for loopIteration := 0; loopIteration <= maxIterationCount; loopIteration++ {
+	for loopIteration := uint64(0); loopIteration <= maxIterationCount; loopIteration++ {
 		if loopIteration >= maxIterationCount {
 			return nil, fmt.Errorf("infinite LOOP detected")
 		}
