diff -Naur aspell-0.60.6.1/interfaces/cc/aspell.h ../aspell-0.60.6.1/interfaces/cc/aspell.h
--- aspell-0.60.6.1/interfaces/cc/aspell.h      2011-07-02 14:53:27.000000000 -0700
+++ ../aspell-0.60.6.1/interfaces/cc/aspell.h   2017-07-31 08:37:34.905244087 -0700
@@ -236,7 +236,7 @@

 /******************************** errors ********************************/

-
+#ifndef __cplusplus
 extern const struct AspellErrorInfo * const aerror_other;
 extern const struct AspellErrorInfo * const aerror_operation_not_supported;
 extern const struct AspellErrorInfo * const   aerror_cant_copy;
@@ -322,7 +322,7 @@
 extern const struct AspellErrorInfo * const   aerror_bad_magic;
 extern const struct AspellErrorInfo * const aerror_expression;
 extern const struct AspellErrorInfo * const   aerror_invalid_expression;
-
+#endif //__cplusplus

 /******************************* speller *******************************/
