diff -u -r ../asciinema-2.0.2/asciinema/__main__.py ./asciinema/__main__.py
--- ../asciinema-2.0.2/asciinema/__main__.py	2019-01-12 18:58:29.000000000 +0000
+++ ./asciinema/__main__.py	2019-01-16 21:07:50.834574609 +0000
@@ -50,10 +50,6 @@
 
 
 def main():
-    if locale.nl_langinfo(locale.CODESET).upper() not in ['US-ASCII', 'UTF-8', 'UTF8']:
-        print("asciinema needs an ASCII or UTF-8 character encoding to run. Check the output of `locale` command.")
-        sys.exit(1)
-
     try:
         cfg = config.load()
     except config.ConfigError as e:
