diff -N -u -r ../asciinema-1.1.1/Godeps/_workspace/src/github.com/kr/pty/ztypes_arm64.go ./Godeps/_workspace/src/github.com/kr/pty/ztypes_arm64.go
--- ../asciinema-1.1.1/Godeps/_workspace/src/github.com/kr/pty/ztypes_arm64.go	1969-12-31 19:00:00.000000000 -0500
+++ ./Godeps/_workspace/src/github.com/kr/pty/ztypes_arm64.go	2015-12-20 17:45:50.332987015 -0500
@@ -0,0 +1,12 @@
+// From upstream pty package - https://github.com/kr/pty/blob/master/ztypes_arm64.go
+// Created by cgo -godefs - DO NOT EDIT
+// cgo -godefs types.go
+
+// +build arm64
+
+package pty
+
+type (
+	_C_int  int32
+	_C_uint uint32
+)
