diff -Naur stfl-0.24.orig/Makefile stfl-0.24/Makefile
--- stfl-0.24.orig/Makefile	2015-02-12 14:14:33.000000000 +0100
+++ stfl-0.24/Makefile	2016-05-31 21:26:11.162048447 +0200
@@ -20,10 +20,6 @@
 
 include Makefile.cfg
 
-export CC = gcc -pthread
-export CFLAGS += -I. -Wall -Os -ggdb -D_GNU_SOURCE -fPIC
-export LDLIBS += -lncursesw
-
 SONAME  := libstfl.so.0
 VERSION := 0.24
 
@@ -34,12 +30,12 @@
 libstfl.a: public.o base.o parser.o dump.o style.o binding.o iconv.o \
            $(patsubst %.c,%.o,$(wildcard widgets/*.c))
 	rm -f $@
-	ar qc $@ $^
-	ranlib $@
+	$(AR) qc $@ $^
+	$(RANLIB) $@
 
 libstfl.so.$(VERSION): public.o base.o parser.o dump.o style.o binding.o iconv.o \
                        $(patsubst %.c,%.o,$(wildcard widgets/*.c))
-	$(CC) -shared -Wl,-soname,$(SONAME) -o $@ $(LDLIBS) $^
+	$(CC) $(LDFLAGS) -shared -Wl,-soname,$(SONAME) -o $@ $(LDLIBS) $^
 
 clean:
 	rm -f libstfl.a example core core.* *.o Makefile.deps
@@ -65,6 +61,7 @@
 	install -m 644 stfl.pc $(DESTDIR)$(prefix)/$(libdir)/pkgconfig/
 	install -m 644 libstfl.so.$(VERSION) $(DESTDIR)$(prefix)/$(libdir)
 	ln -fs libstfl.so.$(VERSION) $(DESTDIR)$(prefix)/$(libdir)/libstfl.so
+	ln -fs libstfl.so.$(VERSION) $(DESTDIR)$(prefix)/$(libdir)/libstfl.so.0
 
 stfl.pc: stfl.pc.in
 	sed 's,@VERSION@,$(VERSION),g' < $< | sed 's,@PREFIX@,$(prefix),g' > $@
diff -Naur stfl-0.24.orig/Makefile.cfg stfl-0.24/Makefile.cfg
--- stfl-0.24.orig/Makefile.cfg	2009-05-31 20:20:39.000000000 +0200
+++ stfl-0.24/Makefile.cfg	2016-05-30 05:50:59.051375377 +0200
@@ -20,7 +20,6 @@
 
 export libdir ?= lib
 export prefix ?= /usr/local
-export DESTDIR ?= /
 
 ifneq ($(shell spl-config --cflags 2>/dev/null),)
 FOUND_SPL = 1
