diff -u -r ../protobuf-3.2.0/src/google/protobuf/stubs/common.cc ./src/google/protobuf/stubs/common.cc
--- ../protobuf-3.2.0/src/google/protobuf/stubs/common.cc	2017-01-28 00:49:29.000000000 +0100
+++ ./src/google/protobuf/stubs/common.cc	2017-01-29 14:43:35.021889040 +0100
@@ -116,7 +116,7 @@
 
 namespace internal {
 
-#if defined(__ANDROID__)
+#if defined(__ANDROID__) && 0
 inline void DefaultLogHandler(LogLevel level, const char* filename, int line,
                               const string& message) {
   if (level < GOOGLE_PROTOBUF_MIN_LOG_LEVEL) {
