diff -u -r ../mg-20180927/GNUmakefile ./GNUmakefile
--- ../mg-20180927/GNUmakefile	2018-09-27 20:00:17.000000000 +0000
+++ ./GNUmakefile	2018-11-17 23:55:15.804955616 +0000
@@ -11,7 +11,7 @@
 bindir=		$(prefix)/bin
 libdir=		$(prefix)/lib
 includedir=	$(prefix)/include
-mandir=		$(prefix)/man
+mandir=		$(prefix)/share/man
 
 PKG_CONFIG=	/usr/bin/pkg-config --silence-errors
 INSTALL=	/usr/bin/install
