diff -u -r ../wireshark-4.4.1/tools/lemon/CMakeLists.txt ./tools/lemon/CMakeLists.txt
--- ../wireshark-4.4.1/tools/lemon/CMakeLists.txt	2024-10-09 18:08:09.000000000 +0000
+++ ./tools/lemon/CMakeLists.txt	2024-10-22 21:01:45.426873365 +0000
@@ -22,11 +22,6 @@
 else()
 	target_compile_options(lemon PRIVATE -w)
 endif()
-if(CMAKE_C_COMPILER_ID MATCHES "Clang")
-	# Disable static analysis for lemon source code. These issues don't
-	# affect Wireshark at runtime.
-	target_compile_options(lemon PRIVATE -Xclang -analyzer-disable-all-checks)
-endif()
 if(DEFINED NO_SANITIZE_CFLAGS)
 	target_compile_options(lemon PRIVATE ${NO_SANITIZE_CFLAGS})
 endif()
