diff -uNr calc-2.12.7.6/Makefile calc-2.12.7.6.mod/Makefile
--- calc-2.12.7.6/Makefile	2021-02-03 20:52:43.000000000 +0000
+++ calc-2.12.7.6.mod/Makefile	2021-02-04 20:00:47.075344477 +0000
@@ -1122,7 +1122,7 @@
 # CCMISC and preceed EXTRA_CFLAGS.
 #
 ARCH_CFLAGS= -march=native
-#ARCH_CFLAGS=
+ARCH_CFLAGS=
 
 # COMMON_CFLAGS are the common ${CC} flags used for all programs, both
 #	    intermediate and final calc and calc related programs
