--- ./Makefile	2021-02-13 06:35:46.000000000 +0000
+++ ./Makefile	2021-02-14 07:20:08.457009951 +0000
@@ -1125,8 +1125,8 @@
 # The ARCH_CFLAGS are ${CC} when compiling C files. They follow
 # CCMISC and precede EXTRA_CFLAGS.
 #
-ARCH_CFLAGS= -march=native
-#ARCH_CFLAGS=
+#ARCH_CFLAGS= -march=native
+ARCH_CFLAGS=
 
 # COMMON_CFLAGS are the common ${CC} flags used for all programs, both
 #	    intermediate and final calc and calc related programs
