--- a/lib/Makefile.in
+++ b/lib/Makefile.in
@@ -1031,7 +1031,7 @@
 noinst_LIBRARIES = libmuscript.a libmutcpwrap.a
 libmuaux_la_SOURCES = mailcap.c manlock.c mdecode.c signal.c strexit.c \
 	mu_umaxtostr.c mu_umaxtostr.h $(am__append_3) $(am__append_4)
-libmuaux_la_LIBADD = gnu/libgnu.la
+libmuaux_la_LIBADD = gnu/libgnu.la $(UNISTRING_LIBS)
 libmuaux_la_LDFLAGS = -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
 libmutcpwrap_a_SOURCES = tcpwrap.c
 noinst_HEADERS = \
