--- a/build/config/BUILD.gn
+++ b/build/config/BUILD.gn
@@ -59,7 +59,7 @@
   if (use_glib) {
     defines += [ "USE_GLIB=1" ]
   }
-  if (use_ozone && !is_android) {
+  if (false) {
     # Note that some Chrome OS builds unconditionally set |use_ozone| to true,
     # but they also build some targets with the Android toolchain. This ensures
     # that Android targets still build with USE_OZONE=0 in such cases.
