--- a/CMake/dcmtk.pc.in
+++ b/CMake/dcmtk.pc.in
@@ -1,6 +1,6 @@
  prefix="@CMAKE_INSTALL_PREFIX@"
  exec_prefix="${prefix}"
- libdir="${prefix}/@CMAKE_INSTALL_LIBDIR@"
+ libdir="@CMAKE_INSTALL_LIBDIR@"
  includedir="${prefix}/include/"
 
  Name: DCMTK
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -208,5 +208,5 @@
 )
 
 install(FILES "${DCMTK_BINARY_DIR}/dcmtk.pc"
-    DESTINATION ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/pkgconfig
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig
 )
