--- a/src/System/Boot/init_texmacs.cpp
+++ b/src/System/Boot/init_texmacs.cpp
@@ -235,7 +235,7 @@
 static void
 clean_temp_dirs () {
   bool          err         = false;
-  url           main_tmp_dir= url_system ("$TMP") * url (".lolly");
+  url           main_tmp_dir= url_system ("@TERMUX_PREFIX@/tmp") * url (".lolly");
   array<string> a           = read_directory (main_tmp_dir, err);
   for (int i= 0; i < N (a); i++)
     if (is_int (a[i]))
