--- a/xmake/packages/s/s7/xmake.lua
+++ b/xmake/packages/s/s7/xmake.lua
@@ -27,6 +27,7 @@
     add_urls("https://github.com/XmacsLabs/s7.git")
 
     add_versions("20241122", "20241122")
+    add_patches("20241122", "s7.diff", "4852e322970c0b22c8f32e68bb89c74520e7731c0a5623111d942fa68e77eb33")
 
     add_configs("gmp", {description = "enable gmp support", default = false, type = "boolean"})
 
@@ -38,7 +39,7 @@
     end)
 
     if is_plat("linux") then
-        add_syslinks("pthread", "dl", "m")
+        add_syslinks("dl", "m")
     end
 
     on_install("bsd", "cross", "cygwin", "linux", "macosx", "mingw", "msys", "wasm", "windows", function (package)
