--- a/xmake/packages/l/lolly/xmake.lua
+++ b/xmake/packages/l/lolly/xmake.lua
@@ -25,6 +25,7 @@
     add_urls("https://github.com/XmacsLabs/lolly.git")
     add_urls("https://gitee.com/XmacsLabs/lolly.git")
     add_versions("1.4.28", "v1.4.28")
+    add_patches("1.4.28", "lolly.diff", "b5253738e894c2f5d6633da8c33959f0e64f8daf71f9a147651b03f50f2ded1a")
 
     add_deps("tbox")
     if not is_plat("wasm") then
