--- a/xmake/packages/c/cpr/xmake.lua
+++ b/xmake/packages/c/cpr/xmake.lua
@@ -17,17 +17,17 @@
     add_configs("ssl", {description = "Enable SSL.", default = false, type = "boolean"})
 
     add_deps("cmake")
-    if is_plat("mingw", "linux") then
+    if false and is_plat("mingw", "linux") then
         add_syslinks("pthread")
     end
     add_links("cpr")
 
     on_load(function (package)
         if package:config("ssl") then
-            package:add("deps", "libcurl", {configs = {libssh2 = true, zlib = true}})
-            package:add("deps", "libssh2")
+            package:add("deps", "pkgconfig::libcurl", {configs = {libssh2 = true, zlib = true}})
+            package:add("deps", "pkgconfig::libssh2")
         else
-            package:add("deps", "libcurl")
+            package:add("deps", "pkgconfig::libcurl")
         end
     end)
 
@@ -42,9 +42,9 @@
         if package:config("shared") and package:is_plat("macosx") then
             shflags = {"-framework", "CoreFoundation", "-framework", "Security", "-framework", "SystemConfiguration"}
         end
-        local packagedeps = {"libcurl"}
+        local packagedeps = {"pkgconfig::libcurl"}
         if package:config("ssl") then
-            table.insert(packagedeps, "libssh2")
+            table.insert(packagedeps, "pkgconfig::libssh2")
         end
         if package:is_plat("windows") then
             -- fix find_package issue on windows
