diff --git a/Cargo.toml b/Cargo.toml
index d00bbbf..8bd9775 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -34,6 +34,6 @@ objc2-app-kit = { version = "0.2.0", default-features = false, features = [
     "NSPasteboard",
 ] }

-[target.'cfg(all(unix, not(any(target_os="macos", target_os="android", target_os="ios", target_os="emscripten"))))'.dependencies]
+[target.'cfg(all(unix, not(any(target_os="macos", target_os="ios", target_os="emscripten"))))'.dependencies]
 x11-clipboard = { version = "0.9.1", optional = true }
 smithay-clipboard = { version = "0.7.0", default-features = false, optional = true }
diff --git a/src/lib.rs b/src/lib.rs
index 092a23c..e449535 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -20,7 +20,6 @@ pub use crate::common::ClipboardProvider;
     unix,
     not(any(
         target_os = "macos",
-        target_os = "android",
         target_os = "ios",
         target_os = "emscripten"
     ))
@@ -31,7 +30,6 @@ pub mod wayland_clipboard;
     unix,
     not(any(
         target_os = "macos",
-        target_os = "android",
         target_os = "ios",
         target_os = "emscripten"
     ))
@@ -51,7 +49,6 @@ pub mod nop_clipboard;
     unix,
     not(any(
         target_os = "macos",
-        target_os = "android",
         target_os = "ios",
         target_os = "emscripten"
     ))
@@ -62,15 +59,12 @@ pub type ClipboardContext = x11_clipboard::X11ClipboardContext;
 pub type ClipboardContext = windows_clipboard::WindowsClipboardContext;
 #[cfg(target_os = "macos")]
 pub type ClipboardContext = osx_clipboard::OSXClipboardContext;
-#[cfg(target_os = "android")]
-pub type ClipboardContext = nop_clipboard::NopClipboardContext; // TODO: implement AndroidClipboardContext
 #[cfg(target_os = "ios")]
 pub type ClipboardContext = nop_clipboard::NopClipboardContext; // TODO: implement IOSClipboardContext
 #[cfg(not(any(
     unix,
     windows,
     target_os = "macos",
-    target_os = "android",
     target_os = "ios",
     target_os = "emscripten"
 )))]
