diff --git a/build.rs b/build.rs
index 4948cf1..363299d 100644
--- a/build.rs
+++ b/build.rs
@@ -13,7 +13,7 @@ fn add_sources(build: &mut cc::Build, root: &str, files: &[&str]) {
 
 fn main() {
     let target = env::var("TARGET").unwrap();
-    if !target.contains("android")
+    if true
         && !target.contains("ohos")
         && pkg_config::Config::new()
             .atleast_version("24.3.18")
