diff -u -r ../cache/vim-hg/src/vimtutor ./src/vimtutor
--- ../cache/vim-hg/src/vimtutor	2014-05-30 17:16:01.677978089 +0200
+++ ./src/vimtutor	2014-06-15 23:36:24.545163147 +0200
@@ -24,7 +24,7 @@
 export xx
 
 # We need a temp file for the copy.  First try using a standard command.
-tmp="${TMPDIR-/tmp}"
+tmp="${TMPDIR-@TERMUX_PREFIX@/tmp}"
 TUTORCOPY=`mktemp $tmp/tutorXXXXXX || tempfile -p tutor || echo none`
 
 # If the standard commands failed then create a directory to put the copy in.
