diff --git a/CMakeLists.txt b/CMakeLists.txt
index f8e9583..b025115 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -126,10 +126,8 @@ if(with_zlib)
   find_package(ZLIB)
 endif()
 
-if(UNIX)
-  find_library(MATH_LIB m)
-  link_libraries(${MATH_LIB})
-endif()
+set(MATH_LIB -lm)
+link_libraries(${MATH_LIB})
 
 if(WIN32)
   # Find Windows specific dependencies
diff --git a/lib/gvc/CMakeLists.txt b/lib/gvc/CMakeLists.txt
index b437bde..34ebade 100644
--- a/lib/gvc/CMakeLists.txt
+++ b/lib/gvc/CMakeLists.txt
@@ -75,6 +75,8 @@ if(ZLIB_FOUND)
   target_link_libraries(gvc PUBLIC ${ZLIB_LIBRARIES})
 endif()
 
+target_link_libraries(gvc PUBLIC -landroid-glob)
+
 if(${with_ortho})
   target_link_libraries(gvc PRIVATE
     $<TARGET_OBJECTS:ortho_obj>
