diff -u -r ../graphviz-2.40.1/cmd/edgepaint/Makefile.in ./cmd/edgepaint/Makefile.in
--- ../graphviz-2.40.1/cmd/edgepaint/Makefile.in	2016-12-25 03:04:55.000000000 +0000
+++ ./cmd/edgepaint/Makefile.in	2019-04-26 20:26:13.197184745 +0000
@@ -585,7 +585,7 @@
 	$(top_builddir)/lib/edgepaint/liblab_gamut.la \
 	$(top_builddir)/lib/cgraph/libcgraph.la \
 	$(top_builddir)/lib/cdt/libcdt.la \
-	$(ANN_LIBS) -lstdc++ -lm
+	$(ANN_LIBS) -lc++_shared -lm
 
 EXTRA_DIST = $(man_MANS) $(pdf_DATA) edgepaint.vcxproj*
 DISTCLEANFILES = $(pdf_DATA)
