--- ../cmd-edgepaint-Makefile.am.orig	2020-03-28 14:17:47.357033245 +0100
+++ ./cmd/edgepaint/Makefile.am	2020-03-28 14:18:27.756954983 +0100
@@ -31,7 +31,7 @@
 	$(top_builddir)/lib/edgepaint/liblab_gamut.la \
 	$(top_builddir)/lib/cgraph/libcgraph.la \
 	$(top_builddir)/lib/cdt/libcdt.la \
-	$(ANN_LIBS) -lstdc++ -lm
+	$(ANN_LIBS) -lc++_shared -lm
 
 edgepaint.1.pdf: $(srcdir)/edgepaint.1
 	- @GROFF@ -e -Tps -man -t $(srcdir)/edgepaint.1 | @PS2PDF@ - - >edgepaint.1.pdf
