--- ./configure.ac.orig	2021-11-21 17:16:59.713007065 +0000
+++ ./configure.ac	2021-11-21 17:17:39.776340523 +0000
@@ -279,9 +279,9 @@
 AC_CHECK_PROGS(SED,gsed sed,false)
 AC_CHECK_PROGS(SORT,gsort sort,false)
 
-AC_CHECK_PROG(EGREP,egrep,egrep,false)
-AC_CHECK_PROG(GROFF,groff,groff,false)
-AC_CHECK_PROGS(PS2PDF,ps2pdf pstopdf,false)
+AC_CHECK_PROG(EGREP,egrep,egrep,echo)
+AC_CHECK_PROG(GROFF,groff,groff,echo)
+AC_CHECK_PROGS(PS2PDF,ps2pdf pstopdf,echo)
 AM_CONDITIONAL(HAVE_PS2PDF,[test "x$PS2PDF" != xfalse])
 
 AC_ARG_ENABLE([man-pdfs],
