Do not require libltdl.la for building.

diff -u -r ../graphviz-2.40.1/configure ./configure
--- ../graphviz-2.40.1/configure	2016-12-25 04:04:52.000000000 +0100
+++ ./configure	2017-08-11 14:35:43.459379008 +0200
@@ -19556,15 +19556,6 @@
 fi
 
 
-if test -n "$with_ltdl_lib"; then
-  if test -f "$with_ltdl_lib/libltdl.la"; then :
-  else
-    as_fn_error $? "invalid ltdl library directory: \`$with_ltdl_lib'" "$LINENO" 5
-  fi
-else
-  with_ltdl_lib=no
-fi
-
 case ,$with_included_ltdl,$with_ltdl_include,$with_ltdl_lib, in
   ,yes,no,no,)
 	case $enable_ltdl_convenience in
