--- ../gvconfig.c.orig	2020-04-18 19:07:49.625114827 +0200
+++ ./lib/gvc/gvconfig.c	2020-04-18 19:16:04.460802577 +0200
@@ -393,7 +393,7 @@
     char *plugin_re_end = "$"; 
 #else
     /* Everyone else */
-    char *plugin_re_beg = "\\.so\\.";
+    char *plugin_re_beg = "\\.so";
     char *plugin_re_end= "$";
 #endif
 
@@ -417,10 +417,8 @@
 
 #if defined(_WIN32) && !defined(__MINGW32__) && !defined(__CYGWIN__)
     sprintf(config_re,"%s%s", plugin_re_beg, plugin_re_end);
-#elif defined(GVPLUGIN_VERSION)
-    sprintf(config_re,"%s%d%s", plugin_re_beg, GVPLUGIN_VERSION, plugin_re_end);
 #else
-    sprintf(config_re,"%s[0-9]+%s", plugin_re_beg, plugin_re_end);
+    sprintf(config_re,"%s%s", plugin_re_beg, plugin_re_end);
 #endif
 
     if (regcomp(&re, config_re, REG_EXTENDED|REG_NOSUB) != 0) {
