--- ../blocks.cpp.orig	2020-03-24 12:30:32.334785560 +0100
+++ ./lib/vpsc/blocks.cpp	2020-03-24 12:34:15.597529253 +0100
@@ -155,8 +155,8 @@
 	//erase(doomed);
 }
 void Blocks::cleanup() {
-	vector<Block*> bcopy(begin(),end());
-	for(vector<Block*>::iterator i=bcopy.begin();i!=bcopy.end();i++) {
+	vector<Block*> b_copy(begin(),end());
+	for(vector<Block*>::iterator i=b_copy.begin();i!=b_copy.end();i++) {
 		Block *b=*i;
 		if(b->deleted) {
 			erase(b);
