--- ./cmd/dot/Makefile.am.orig	2021-12-02 18:35:30.719043970 +0530
+++ ./cmd/dot/Makefile.am	2021-12-02 18:36:09.779043955 +0530
@@ -76,7 +76,6 @@
 # run "dot -c", if possible, to create plugin config
 install-exec-hook:
 	(cd $(DESTDIR)$(bindir); if test -x dot$(EXEEXT); then for i in $(linkedprogram); do rm -f $$i; $(LN_S) dot$(EXEEXT) $$i; done; fi;)
-	if test "x$(DESTDIR)" = "x"; then if test -x $(bindir)/dot$(EXEEXT); then if test -x /sbin/ldconfig; then /sbin/ldconfig 2>/dev/null; fi; cd $(bindir); ./dot$(EXEEXT) -c; else cd $(bindir); ./dot_static$(EXEEXT) -c; fi; fi
 
 uninstall-hook:
 	(cd $(DESTDIR)$(man1dir); for i in $(linkedman); do rm -f $$i; done;)
