http://www.graphviz.org/content/cant-cross-compile-graphviz

diff -u -r ../graphviz-2.40.1/lib/gvpr/Makefile.am ./lib/gvpr/Makefile.am
--- ../graphviz-2.40.1/lib/gvpr/Makefile.am	2016-08-09 23:02:10.000000000 +0200
+++ ./lib/gvpr/Makefile.am	2017-01-19 20:01:22.745725697 +0100
@@ -57,7 +57,7 @@
 mkdefs_SOURCES = mkdefs.c
 
 gdefs.h: $(top_srcdir)/lib/gvpr/gprdata mkdefs$(EXEEXT)
-	./mkdefs$(EXEEXT) gdefs.h < $(top_srcdir)/lib/gvpr/gprdata
+	touch gdefs.h
 
 EXTRA_DIST = $(man_MANS) $(pdf_DATA) gprdata mkdefs.c gdefs.h trie.c gvprlib.vcxproj*
 
diff -u -r ../graphviz-2.40.1/lib/gvpr/Makefile.in ./lib/gvpr/Makefile.in
--- ../graphviz-2.40.1/lib/gvpr/Makefile.in	2016-12-25 04:04:59.000000000 +0100
+++ ./lib/gvpr/Makefile.in	2017-01-19 20:02:07.293214993 +0100
@@ -1084,7 +1084,7 @@
 	- @GROFF@ -Tps -man $(srcdir)/gvpr.3 | @PS2PDF@ - - >gvpr.3.pdf
 
 gdefs.h: $(top_srcdir)/lib/gvpr/gprdata mkdefs$(EXEEXT)
-	./mkdefs$(EXEEXT) gdefs.h < $(top_srcdir)/lib/gvpr/gprdata
+	touch gdefs.h
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
