diff -uNr graphviz-2.44.1/plugin/webp/Makefile.am graphviz-2.44.1.mod/plugin/webp/Makefile.am
--- graphviz-2.44.1/plugin/webp/Makefile.am	2020-06-29 11:47:18.000000000 +0300
+++ graphviz-2.44.1.mod/plugin/webp/Makefile.am	2020-08-09 22:58:32.615146633 +0300
@@ -26,7 +26,7 @@
 
 libgvplugin_webp_la_LDFLAGS = -version-info @GVPLUGIN_VERSION_INFO@
 libgvplugin_webp_la_SOURCES = $(libgvplugin_webp_C_la_SOURCES)
-libgvplugin_webp_la_LIBADD = @WEBP_LIBS@ @PANGOCAIRO_LIBS@
+libgvplugin_webp_la_LIBADD = $(top_builddir)/lib/gvc/libgvc.la @WEBP_LIBS@ @PANGOCAIRO_LIBS@
 
 if WITH_WIN32
 libgvplugin_webp_la_LDFLAGS += -no-undefined
