--- src/lib/sfdpgen/PriorityQueue.c.orig	2016-04-01 08:29:58.788223689 -0700
+++ src/lib/sfdpgen/PriorityQueue.c	2016-04-01 08:30:44.507731384 -0700
@@ -11,6 +11,8 @@
  * Contributors: See CVS logs. Details at http://www.graphviz.org/
  *************************************************************************/
 
+void free(void *ptr);
+
 #include "LinkedList.h"
 #include "PriorityQueue.h"
 #include "memory.h"
