--- ./configure.ac.orig	2009-11-12 15:57:39.000000000 +0200
+++ ./configure.ac	2019-08-11 01:22:37.503246795 +0300
@@ -178,7 +178,7 @@
 AS_IF([test "x$enable_maintainer_flags" = "xyes" &&  test "x$GCC" = "xyes"],
       [
         AS_COMPILER_FLAGS([MAINTAINER_CFLAGS],
-                          ["-Werror -Wall -Wshadow -Wcast-align
+                          ["-Wall -Wshadow -Wcast-align
                             -Wno-uninitialized -Wempty-body -Wformat-security
                             -Winit-self"])
       ]
