diff -u -r ../mc-4.8.23/configure ./configure
--- ../mc-4.8.23/configure	2019-06-22 11:44:11.000000000 +0000
+++ ./configure	2019-06-23 17:51:24.487569137 +0000
@@ -20600,12 +20600,6 @@
     esac
   fi
 
-  if test -z "$ac_list_mounted_fs"; then
-    as_fn_error $? "could not determine how to read list of mounted file systems" "$LINENO" 5
-    # FIXME -- no need to abort building the whole package
-    # Can't build mountlist.c or anything that needs its functions
-  fi
-
   if test $ac_list_mounted_fs = found; then
     gl_cv_list_mounted_fs=yes
   else
