diff -u -r ../mc-4.8.11/configure ./configure
--- ../mc-4.8.11/configure	2013-11-29 19:45:45.000000000 +0100
+++ ./configure	2014-01-08 12:02:35.217251791 +0100
@@ -19036,12 +19036,6 @@
   esac
 fi
 
-if test -z "$ac_list_mounted_fs"; then
-  as_fn_error $? "could not determine how to read list of mounted file systems" "$LINENO" 5
-  # FIXME -- no need to abort building the whole package
-  # Can't build mountlist.c or anything that needs its functions
-fi
-
 if test $ac_list_mounted_fs = found; then :
   gl_cv_list_mounted_fs=yes
 else
@@ -19737,12 +19731,6 @@
   esac
 fi
 
-if test -z "$ac_list_mounted_fs"; then
-  as_fn_error $? "could not determine how to read list of mounted file systems" "$LINENO" 5
-  # FIXME -- no need to abort building the whole package
-  # Can't build mountlist.c or anything that needs its functions
-fi
-
 if test $ac_list_mounted_fs = found; then :
 
 
