diff -u -r ../mc-4.8.16/src/filemanager/usermenu.c ./src/filemanager/usermenu.c
--- ../mc-4.8.16/src/filemanager/usermenu.c	2016-03-12 10:45:48.000000000 -0500
+++ ./src/filemanager/usermenu.c	2016-03-14 20:34:12.523712672 -0400
@@ -448,7 +448,7 @@
         return;
     }
     cmd_file = fdopen (cmd_file_fd, "w");
-    fputs ("#! /bin/sh\n", cmd_file);
+    fputs ("#! @TERMUX_PREFIX@/bin/sh\n", cmd_file);
     commands++;
 
     for (col = 0; *commands != '\0'; commands++)
@@ -554,7 +554,7 @@
          * on no-exec filesystems. */
         char *cmd;
 
-        cmd = g_strconcat ("/bin/sh ", vfs_path_as_str (file_name_vpath), (char *) NULL);
+        cmd = g_strconcat ("@TERMUX_PREFIX@/bin/sh ", vfs_path_as_str (file_name_vpath), (char *) NULL);
         if (!show_prompt)
         {
             if (system (cmd) == -1)
