diff -u -r ../mc-4.8.12/src/filemanager/usermenu.c ./src/filemanager/usermenu.c
--- ../mc-4.8.12/src/filemanager/usermenu.c	2014-04-01 12:54:01.000000000 +0200
+++ ./src/filemanager/usermenu.c	2014-06-05 16:27:09.221997381 +0200
@@ -453,7 +453,7 @@
         return;
     }
     cmd_file = fdopen (cmd_file_fd, "w");
-    fputs ("#! /bin/sh\n", cmd_file);
+    fputs ("#! @TERMUX_PREFIX@/bin/sh\n", cmd_file);
     commands++;
 
     for (col = 0; *commands; commands++)
