diff -u -r ../mc-4.8.12/src/main.c ./src/main.c
--- ../mc-4.8.12/src/main.c	2014-04-01 12:54:01.000000000 +0200
+++ ./src/main.c	2014-06-05 16:29:02.609994281 +0200
@@ -140,7 +140,7 @@
     if ((mc_global.tty.shell == NULL) || (mc_global.tty.shell[0] == '\0'))
     {
         g_free (mc_global.tty.shell);
-        mc_global.tty.shell = g_strdup ("/bin/sh");
+        mc_global.tty.shell = g_strdup ("/system/bin/sh");
     }
 
     /* This is the directory, where MC was installed, on Unix this is DATADIR */
