diff -u -r ../mc-4.8.15/src/filemanager/achown.c ./src/filemanager/achown.c
--- ../mc-4.8.15/src/filemanager/achown.c	2015-11-06 04:11:09.000000000 -0500
+++ ./src/filemanager/achown.c	2015-11-20 18:00:58.613280615 -0500
@@ -347,21 +347,25 @@
         if (is_owner)
         {
             /* get and put user names in the listbox */
+#ifndef __ANDROID__
             setpwent ();
             while ((chl_pass = getpwent ()) != NULL)
                 listbox_add_item (chl_list, LISTBOX_APPEND_SORTED, 0, chl_pass->pw_name, NULL,
                                   FALSE);
+#endif
             endpwent ();
             fe = listbox_search_text (chl_list, get_owner (sf_stat->st_uid));
         }
         else
         {
             /* get and put group names in the listbox */
+#ifndef __ANDROID__
             setgrent ();
             while ((chl_grp = getgrent ()) != NULL)
                 listbox_add_item (chl_list, LISTBOX_APPEND_SORTED, 0, chl_grp->gr_name, NULL,
                                   FALSE);
             endgrent ();
+#endif
             fe = listbox_search_text (chl_list, get_group (sf_stat->st_gid));
         }
 
