diff -uNr mc-4.8.27/src/usermenu.c mc-4.8.27.mod/src/usermenu.c
--- mc-4.8.27/src/usermenu.c	2021-08-01 19:45:58.000000000 +0300
+++ mc-4.8.27.mod/src/usermenu.c	2021-08-18 00:18:39.918253182 +0300
@@ -450,7 +450,7 @@
     }
 
     cmd_file = fdopen (cmd_file_fd, "w");
-    fputs ("#! /bin/sh\n", cmd_file);
+    fputs ("#! @TERMUX_PREFIX@/bin/sh\n", cmd_file);
     commands++;
 
     for (col = 0; *commands != '\0'; commands++)
@@ -546,7 +546,7 @@
     mc_chmod (file_name_vpath, S_IRWXU);
 
     /* Execute the command indirectly to allow execution even on no-exec filesystems. */
-    cmd = g_strconcat ("/bin/sh ", vfs_path_as_str (file_name_vpath), (char *) NULL);
+    cmd = g_strconcat ("@TERMUX_PREFIX@/bin/sh ", vfs_path_as_str (file_name_vpath), (char *) NULL);
 
     if (run_view)
     {
