diff --git a/src/vfs/extfs/helpers/s3+.in b/src/vfs/extfs/helpers/s3+.in
index 2ab596a..65b709b 100644
--- a/src/vfs/extfs/helpers/s3+.in
+++ b/src/vfs/extfs/helpers/s3+.in
@@ -1,4 +1,4 @@
-#! @PYTHON@
+#! @PYTHON@2
 # -*- coding: utf-8 -*-
 
 #
diff --git a/src/vfs/extfs/helpers/uc1541.in b/src/vfs/extfs/helpers/uc1541.in
index dd7b7c4..8da6bfb 100644
--- a/src/vfs/extfs/helpers/uc1541.in
+++ b/src/vfs/extfs/helpers/uc1541.in
@@ -1,4 +1,4 @@
-#! @PYTHON@
+#! @PYTHON@2
 """
 UC1541 Virtual filesystem
 
