--- a/cmake/init-global-vars.cmake
+++ b/cmake/init-global-vars.cmake
@@ -7,12 +7,9 @@
 set(GENERATED_DIR "${OBJS_DIR}/generated")
 set(AUTO_DIR "${GENERATED_DIR}/auto")
 
-if(APPLE)
+if(TRUE)
     set(CURL_LIB curl)
     set(ICONV_LIB iconv)
-else()
-    set(CURL_LIB /opt/curl7600/lib/libcurl.a)
-    set(RT_LIB rt)
 endif()
 
 find_package(Git REQUIRED)
@@ -72,7 +69,7 @@
     set(CMAKE_INSTALL_PREFIX "/." CACHE PATH "install prefix" FORCE)
 endif()
 
-set(VK_INSTALL_DIR usr/share/vkontakte)
+set(VK_INSTALL_DIR share/vkontakte)
 set(INSTALL_KPHP_SOURCE ${VK_INSTALL_DIR}/kphp_source)
 
 set(CPACK_PACKAGING_INSTALL_PREFIX ${CMAKE_INSTALL_PREFIX})
