From ca5ceb5264c880315f18ba56417e8c66c9fc3cf3 Mon Sep 17 00:00:00 2001
From: Chongyun Lee <licy183@termux.dev>
Date: Sun, 24 Aug 2025 00:06:38 +0800
Subject: [PATCH] Treat `macro redefined` as error in jumbo build

---
 build/config/compiler/BUILD.gn | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index c9a67656..01eb5811 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -2285,6 +2285,12 @@ config("default_warnings") {
   if (is_apple) {
     ldflags += [ "-Wl,-no_warn_duplicate_libraries" ]
   }
+
+  cflags += [
+    # Treat macro redefined as error
+    # In jumbo build, it may cause undefined behavior
+    "-Werror=macro-redefined",
+  ]
 }
 
 # prevent_unsafe_narrowing ----------------------------------------------------
