--- a/lexilla/src/makefile
+++ b/lexilla/src/makefile
@@ -43,7 +43,7 @@
 BASE_FLAGS += --std=c++17
 
 ifdef CLANG
-CXX = clang++
+CXX ?= clang++
 ifdef windir
 # Clang on Win32 uses MSVC headers so will complain about strcpy without this
 DEFINES += -D_CRT_SECURE_NO_DEPRECATE=1
@@ -78,7 +78,7 @@
 vpath %.cxx ../src ../lexlib ../lexers
 
 DEFINES += -D$(if $(DEBUG),DEBUG,NDEBUG)
-BASE_FLAGS += $(if $(DEBUG),-g,-O3)
+BASE_FLAGS += $(if $(DEBUG),-g,)
 
 INCLUDES = -I ../include -I $(SCINTILLA_INCLUDE) -I ../src -I ../lexlib
 LDFLAGS += -shared
