--- a/bin/9c
+++ b/bin/9c
@@ -29,7 +29,12 @@ usegcc()
 	# want to put -fno-optimize-sibling-calls here but
 	# that option only works with gcc3+ it seems
 	cflags="$cflags -ggdb"
-	cflags="$cflags $CC9FLAGS"
+	if [ "$bootstrap" = 'true' ]
+	then
+		cflags="$cflags $CC9FLAGS"
+	else
+		cflags="$cflags $CFLAGS"
+	fi
 }
 
 quiet()
@@ -88,7 +93,12 @@ useclang()
 		-fno-common \
 	"
 	cflags="$cflags -g"
-	cflags="$cflags $CC9FLAGS"
+	if [ "$bootstrap" = 'true' ]
+	then
+		cflags="$cflags $CC9FLAGS"
+	else
+		cflags="$cflags $CFLAGS"
+	fi
 }
 
 usexlc()
@@ -118,14 +128,7 @@ case "$tag" in
 		cflags="$cflags -g3 -m64"
 		;;
 *HP-UX*)	cc=${CC9:-cc}; cflags="-g -O -c -Ae" ;;
-*Linux*)	usegcc
-		case "${CC9:-gcc}" in
-		tcc)
-			cc=tcc
-			cflags="-c -g"
-			;;
-		esac
-		;;
+*Linux*)	useclang ;;
 *OSF1*)		cc=${CC9:-cc}; cflags="-g -O -c" ;;
 *SunOS*-cc)	cc=cc;
 		cflags="-mt -g -O -c -xCC -D__sun__"
@@ -150,7 +153,7 @@ case "$tag" in
 esac
 
 # Must use temp file to avoid pipe; pipe loses status.
-xtmp=${TMPDIR-/tmp}/9c.$$.$USER.out
+xtmp=${TMPDIR-@TERMUX_PREFIX@/tmp}/9c.$$.$USER.out
 $cc -DPLAN9PORT -I"$PLAN9/include" $cflags "$@" 2>"$xtmp"
 status=$?
 quiet "$xtmp"
