--- a/INSTALL
+++ b/INSTALL
@@ -150,7 +150,8 @@ unset MKSHELL
 
 cd src
 if $dobuild; then
-	echo "* Building mk..."
+	echo "* Building native mk..."
+	export bootstrap=true
 	../dist/buildmk 2>&1 | sed 's/^[+] //'
 
 	if [ ! -x ../bin/mk ]; then
@@ -175,6 +176,9 @@ if $dobuild; then
 		cp $PLAN9/bin/quote1 $PLAN9/bin/'"'
 		cp $PLAN9/bin/quote2 $PLAN9/bin/'""'
 	fi
+	echo "* Building cross mk..."
+	export bootstrap=false
+	../dist/buildmk 2>&1 | sed 's/^[+] //'
 	echo "* Cleaning up..."
 	mk clean
 fi
