--- a/src/cmd/diff/main.c
+++ b/src/cmd/diff/main.c
@@ -14,7 +14,7 @@ char rflag;			/* recurse down directory trees */
 char mflag;			/* pseudo flag: doing multiple files, one dir */
 int anychange;
 
-static char *tmp[] = {"/tmp/diff1XXXXXXXXXXX", "/tmp/diff2XXXXXXXXXXX"};
+static char *tmp[] = {"@TERMUX_PREFIX@/tmp/diff1XXXXXXXXXXX", "/tmp/diff2XXXXXXXXXXX"};
 static int whichtmp;
 static char *progname;
 static char usage[] = "diff [ -acefmnbwr ] file1 ... file2\n";
