--- redir-2.2.1/Makefile	1999-12-26 12:51:55.000000000 -0800
+++ src/Makefile	2017-02-15 13:07:10.577481896 -0800
@@ -15,7 +15,8 @@
 # if your system needs any additional libraries (solaris, for example, 
 # needs the ones commented out below), edit this line.
 
-EXTRA_LIBS = #-lnsl -lsocket
+#EXTRA_LIBS = #-lnsl -lsocket
+EXTRA_LIBS = -llog
 
 # add additional compiler flags here.  Some useful ones are:
 #
@@ -27,7 +28,7 @@
 ### end of user configuration section
 
 # redir requires gcc.  if you're lucky, another compiler might work.
-CC = gcc
+#CC = gcc
 
 # if your system lacks getopt_long, remove the comment from this line
 OBJS = redir.o $(GETOPT_OBJS)
@@ -51,6 +52,9 @@
 redir:		${OBJS}
 	${CC} ${LDFLAGS} -o redir ${OBJS} ${LIBS}
 
+install: all
+	install redir $(PREFIX)/bin
+	install redir.man $(PREFIX)/share/man/man1/redir.1
 
 
 
