--- a/src/vde_switch/hash.c
+++ b/src/vde_switch/hash.c
@@ -345,7 +345,7 @@
 	int rv=-1;
 	int vlan=0;
 	struct hash_entry *e;
-	if (index(strmac,':') != NULL)
+	if (strchr(strmac,':') != NULL)
 		rv=sscanf(strmac,"%x:%x:%x:%x:%x:%x %d", maci+0, maci+1, maci+2, maci+3, maci+4, maci+5, &vlan);
 	else
 		rv=sscanf(strmac,"%x.%x.%x.%x.%x.%x %d", maci+0, maci+1, maci+2, maci+3, maci+4, maci+5, &vlan);

--- a/src/vde_switch/vde_switch.c
+++ b/src/vde_switch/vde_switch.c
@@ -390,7 +390,7 @@
 #endif
 		case MACADDR_ARG:
 			{int maci[ETH_ALEN],rv;
-				if (index(optarg,':') != NULL)
+				if (strchr(optarg,':') != NULL)
 					rv=sscanf(optarg,"%x:%x:%x:%x:%x:%x", maci+0, maci+1, maci+2, maci+3, maci+4, maci+5);
 				else
 					rv=sscanf(optarg,"%x.%x.%x.%x.%x.%x", maci+0, maci+1, maci+2, maci+3, maci+4, maci+5);
