diff -uNr src/sdk/pkg/_fe_analyzer_shared/lib/src/util/colors.dart src.mod/sdk/pkg/_fe_analyzer_shared/lib/src/util/colors.dart
--- src/sdk/pkg/_fe_analyzer_shared/lib/src/util/colors.dart	2021-07-21 10:21:26.170021583 +0000
+++ src.mod/sdk/pkg/_fe_analyzer_shared/lib/src/util/colors.dart	2021-07-21 10:45:50.756764582 +0000
@@ -166,7 +166,7 @@
   // The `-S` option of `tput` allows us to query multiple capabilities at
   // once.
   ProcessResult result = Process.runSync(
-      "/bin/sh", ["-c", "printf '%s' '$TERMINAL_CAPABILITIES' | tput -S"]);
+      "@TERMUX_PREFIX@/bin/sh", ["-c", "printf '%s' '$TERMINAL_CAPABILITIES' | tput -S"]);
 
   if (result.exitCode != 0) {
     printEnableColorsReason("Not enabling colors, running tput failed.");
diff -uNr src/sdk/sdk/lib/_internal/vm/bin/process_patch.dart src.mod/sdk/sdk/lib/_internal/vm/bin/process_patch.dart
--- src/sdk/sdk/lib/_internal/vm/bin/process_patch.dart	2021-07-21 10:21:29.378043843 +0000
+++ src.mod/sdk/sdk/lib/_internal/vm/bin/process_patch.dart	2021-07-21 10:46:23.677037513 +0000
@@ -301,7 +301,7 @@
     if (Platform.isWindows) {
       return 'cmd.exe';
     }
-    return '/bin/sh';
+    return '@TERMUX_PREFIX@/bin/sh';
   }
 
   static List<String> _getShellArguments(
