--- ./sdk/pkg/_fe_analyzer_shared/lib/src/util/colors.dart   2023-08-18 18:30:44.021019202 +0530
+++ ./sdk/pkg/_fe_analyzer_shared/lib/src/util/colors.dart   2023-08-18 20:24:26.037025590 +0530
@@ -163,7 +163,7 @@
   // The `-S` option of `tput` allows us to query multiple capabilities at
   // once.
   ProcessResult result = Process.runSync(
-      "/bin/sh", ["-c", "printf '%s' '$TERMINAL_CAPABILITIES' | tput -S"]);
+      "@TERMUX_PREFIX@/bin/sh", ["-c", "printf '%s' '$TERMINAL_CAPABILITIES' | tput -S"]);

   if (result.exitCode != 0) {
     printEnableColorsReason("Not enabling colors, running tput failed.");
--- ./sdk/sdk/lib/_internal/vm/bin/process_patch.dart   2023-08-18 18:31:28.061019185 +0530
+++ ./sdk/sdk/lib/_internal/vm/bin/process_patch.dart   2023-08-18 20:36:02.387025324 +0530
@@ -312,7 +312,7 @@
     if (Platform.isWindows) {
       return 'cmd.exe';
     }
-    return '/bin/sh';
+    return '@TERMUX_PREFIX@/bin/sh';
   }

   static List<String> _getShellArguments(
