--- ./sdk/runtime/bin/directory_linux.cc
+++ ./sdk/runtime/bin/directory_linux.cc
@@ -425,7 +425,7 @@ const char* Directory::SystemTemp(Namespace* namespc) {
   }
   if (temp_dir == nullptr) {
 #if defined(DART_HOST_OS_ANDROID)
-    temp_dir = "/data/local/tmp";
+    temp_dir = "@TERMUX_PREFIX@/tmp";
 #else
     temp_dir = "/tmp";
 #endif
--- ./sdk/runtime/vm/os_android.cc
+++ ./sdk/runtime/vm/os_android.cc
@@ -52,7 +52,7 @@ class PerfCodeObserver : public CodeObserver {
       return;
     }
     intptr_t pid = getpid();
-    char* filename = OS::SCreate(nullptr, "/tmp/perf-%" Pd ".map", pid);
+    char* filename = OS::SCreate(nullptr, "@TERMUX_PREFIX@/tmp/perf-%" Pd ".map", pid);
     out_file_ = (*file_open)(filename, true);
     free(filename);
   }
