--- ./sdk/runtime/bin/directory_android.cc	2019-06-14 00:27:54.559512261 +0530
+++ ./sdk/runtime/bin/directory_android.cc	2019-06-14 00:32:21.470848945 +0530
@@ -429,7 +429,7 @@
   // TODO(4413): In the long run, when running in an application we should
   // probably use the appropriate directory from the Android API,
   // probably what File.createTempFile uses.
-  const char* kAndroidTempDir = "/data/local/tmp";
+  const char* kAndroidTempDir = "@TERMUX_PREFIX@/tmp";
   struct stat st;
   if (stat(kAndroidTempDir, &st) != 0) {
     mkdir(kAndroidTempDir, 0777);
--- ./sdk/runtime/vm/os_android.cc	2019-06-14 23:47:25.713639563 +0530
+++ ./sdk/runtime/vm/os_android.cc	2019-06-14 23:47:19.743694949 +0530
@@ -42,7 +42,7 @@
       return;
     }
     intptr_t pid = getpid();
-    char* filename = OS::SCreate(NULL, "/tmp/perf-%" Pd ".map", pid);
+    char* filename = OS::SCreate(NULL, "@TERMUX_PREFIX@/tmp/perf-%" Pd ".map", pid);
     out_file_ = (*file_open)(filename, true);
     free(filename);
   }
