--- ./sdk/pkg/_fe_analyzer_shared/lib/src/util/colors.dart
+++ ./sdk/pkg/_fe_analyzer_shared/lib/src/util/colors.dart
@@ -143,7 +143,7 @@ bool _computeEnableColors() {
 
   // The `-S` option of `tput` allows us to query multiple capabilities at
   // once.
-  ProcessResult result = Process.runSync("/bin/sh", [
+  ProcessResult result = Process.runSync("@TERMUX_PREFIX@/bin/sh", [
     "-c",
     "printf '%s' '$TERMINAL_CAPABILITIES' | tput -S",
   ]);
