--- ./sdk/third_party/pkg/pub/lib/src/io.dart.orig	2021-04-14 05:29:00.606101795 +0000
+++ ./sdk/third_party/pkg/pub/lib/src/io.dart	2021-04-14 05:29:41.712685487 +0000
@@ -797,13 +797,13 @@
 ///
 /// [1]: https://refspecs.linuxfoundation.org/FHS_2.3/fhs-2.3.pdf
 String _findTarPath() {
-  for (final file in ['/bin/tar', '/usr/bin/tar']) {
+  for (final file in ['@TERMUX_PREFIX@/bin/tar', '@TERMUX_PREFIX@/local/bin/tar']) {
     if (fileExists(file)) {
       return file;
     }
   }
   log.warning(
-      'Could not find a system `tar` installed in /bin/tar or /usr/bin/tar, '
+      'Could not find a system `tar` installed in @TERMUX_PREFIX@/tar or @TERMUX_PREFIX@/local/bin/tar, '
       'attempting to use tar from PATH');
   return 'tar';
 }
