--- ghc-8.10.1/compiler/main/DynFlags.hs.orig	2020-06-13 08:56:50.595808526 +0100
+++ ghc-8.10.1/compiler/main/DynFlags.hs	2020-06-13 08:57:24.770000253 +0100
@@ -4624,6 +4624,14 @@
 default_PIC platform =
   case (platformOS platform, platformArch platform) of
     (OSDarwin, ArchX86_64) -> [Opt_PIC]
+    (OSLinux, ArchARM{}) -> [Opt_PIC]
+    (OSLinux, ArchARM64) -> [Opt_PIC]
+    (OSLinux, ArchX86) -> [Opt_PIC]
+    (OSLinux, ArchX86_64) -> [Opt_PIC]
+    (OSUnknown, ArchARM{}) -> [Opt_PIC]
+    (OSUnknown, ArchARM64) -> [Opt_PIC]
+    (OSUnknown, ArchX86) -> [Opt_PIC]
+    (OSUnknown, ArchX86_64) -> [Opt_PIC]
     (OSOpenBSD, ArchX86_64) -> [Opt_PIC] -- Due to PIE support in
                                          -- OpenBSD since 5.3 release
                                          -- (1 May 2013) we need to
