--- a/libraries/Cabal/Cabal/src/Distribution/Simple/InstallDirs.hs
+++ b/libraries/Cabal/Cabal/src/Distribution/Simple/InstallDirs.hs
@@ -204,7 +204,7 @@
           windowsProgramFilesDir <- getWindowsProgramFilesDir
           return (windowsProgramFilesDir </> "Haskell")
         Haiku -> return "/boot/system/non-packaged"
-        _ -> return "/usr/local"
+        _ -> return "@TERMUX_PREFIX@"
   installLibDir <-
     case buildOS of
       Windows -> return "$prefix"
