--- ghc-8.10.7/mk/config.mk.in	2021-06-21 12:24:36.000000000 +0530
+++ ghc-8.10.7-patch/mk/config.mk.in	2022-02-04 13:18:15.667333665 +0530
@@ -105,7 +105,7 @@
 WITH_TERMINFO=YES
 endif

-ifeq "$(findstring $(TargetOS_CPP),linux freebsd dragonfly openbsd netbsd solaris2 kfreebsdgnu haiku linux-android)" ""
+ifeq "$(findstring $(TargetOS_CPP),linux freebsd dragonfly openbsd netbsd solaris2 kfreebsdgnu haiku linux_android)" ""
 TargetElf = NO
 else
 TargetElf = YES
