--- a/hadrian/src/Settings/Packages.hs
+++ b/hadrian/src/Settings/Packages.hs
@@ -84,8 +84,8 @@
             -- (#14335) and completely untested in CI for cross
             -- backends at the moment, so we might as well disable it
             -- for cross GHC.
-            [ andM [expr (ghcWithInterpreter stage), notCross] `cabalFlag` "internal-interpreter"
-            , notM cross `cabalFlag` "terminfo"
+            [ andM [expr (ghcWithInterpreter stage), notStage0] `cabalFlag` "internal-interpreter"
+            , notStage0 `cabalFlag` "terminfo"
             , arg "-build-tool-depends"
             , flag UseLibzstd `cabalFlag` "with-libzstd"
             -- ROMES: While the boot compiler is not updated wrt -this-unit-id
@@ -123 +123 @@
-          builder (Cabal Flags) ? notM cross `cabalFlag` "terminfo"
+          builder (Cabal Flags) ? notStage0 `cabalFlag` "terminfo"
@@ -205,7 +205,7 @@
         -- dependencies.
         -- TODO: Perhaps the user should rather be responsible for this?
         , package haskeline ?
-          builder (Cabal Flags) ? notM cross `cabalFlag` "terminfo"
+          builder (Cabal Flags) ? notStage0 `cabalFlag` "terminfo"
 
         -------------------------------- terminfo ------------------------------
         , package terminfo ?
