--- a/hadrian/bindist/config.mk.in
+++ b/hadrian/bindist/config.mk.in
@@ -130,9 +130,9 @@
 #-----------------------------------------------------------------------------
 # Build configuration
 
-CrossCompiling        = @CrossCompiling@
-CrossCompilePrefix    = @CrossCompilePrefix@
+CrossCompiling        = NO
+CrossCompilePrefix    =
 GhcUnregisterised     = @Unregisterised@
 EnableDistroToolchain = @SettingsUseDistroMINGW@
 BaseUnitId            = @BaseUnitId@
 
