--- ghc-9.12.1/hadrian/src/Packages.hs	2025-02-24 22:46:43.665442228 +0530
+++ ghc-9.12.1.mod/hadrian/src/Packages.hs	2025-02-26 14:41:51.865132206 +0530
@@ -171,7 +171,7 @@
 crossPrefix = do
     cross <- flag CrossCompiling
     targetPlatform <- setting TargetPlatformFull
-    return $ if cross then targetPlatform ++ "-" else ""
+    return ""
 
 -- | Given a 'Context', compute the name of the program that is built in it
 -- assuming that the corresponding package's type is 'Program'. For example, GHC
