--- ghc-9.12.1/hadrian/bindist/config.mk.in	2025-02-24 22:46:43.665077469 +0530
+++ ghc-9.12.1.mod/hadrian/bindist/config.mk.in	2025-02-26 02:09:44.451291933 +0530
@@ -130,8 +130,8 @@
 #-----------------------------------------------------------------------------
 # Build configuration
 
-CrossCompiling        = @CrossCompiling@
-CrossCompilePrefix    = @CrossCompilePrefix@
+CrossCompiling        = NO
+CrossCompilePrefix    =
 GhcUnregisterised     = @Unregisterised@
 EnableDistroToolchain = @SettingsUseDistroMINGW@
 
