--- a/hadrian/src/Packages.hs
+++ b/hadrian/src/Packages.hs
@@ -172,7 +172,7 @@
 crossPrefix = do
     cross <- flag CrossCompiling
     targetPlatform <- setting TargetPlatformFull
-    return $ if cross then targetPlatform ++ "-" else ""
+    return ""
 
 -- | Given a 'Context', compute the name of the program that is built in it
 -- assuming that the corresponding package's type is 'Program'. For example, GHC
