--- a/hadrian/src/Settings/Default.hs
+++ b/hadrian/src/Settings/Default.hs
@@ -175,12 +175,10 @@
         , unlit
         , xhtml
         , if winTarget then win32 else unix
-        ]
-      , when (not cross)
-        [ hpcBin
+        , hpcBin
         , iserv
         , runGhc
-        , ghcToolchainBin
+        , terminfo
         ]
       , when (winTarget && not cross)
         [ -- See Note [Hadrian's ghci-wrapper package]
