--- ./Source/cmLocalUnixMakefileGenerator3.cxx.orig	2025-06-10 12:19:50.329169553 +0000
+++ ./Source/cmLocalUnixMakefileGenerator3.cxx	2025-06-10 12:20:37.802501872 +0000
@@ -667,7 +667,7 @@
   } else {
 #if !defined(__VMS)
     makefileStream << "# The shell in which to execute make rules.\n"
-                      "SHELL = /bin/sh\n"
+                      "SHELL = @TERMUX_PREFIX@/bin/sh\n"
                       "\n";
 #endif
   }
