diff -u -r ../cmake-3.2.2/Source/cmLocalUnixMakefileGenerator3.cxx ./Source/cmLocalUnixMakefileGenerator3.cxx
--- ../cmake-3.2.2/Source/cmLocalUnixMakefileGenerator3.cxx	2015-04-13 13:09:00.000000000 -0400
+++ ./Source/cmLocalUnixMakefileGenerator3.cxx	2015-05-24 16:04:48.182697757 -0400
@@ -743,7 +743,7 @@
 #if !defined(__VMS)
       makefileStream
         << "# The shell in which to execute make rules.\n"
-        << "SHELL = /bin/sh\n"
+        << "SHELL = @TERMUX_PREFIX@/bin/sh\n"
         << "\n";
 #endif
     }
