+++ ./src/util/detect_os.h
@@ -24,7 +24,7 @@
  * Android defines __linux__, so DETECT_OS_LINUX and DETECT_OS_POSIX will
  * also be defined.
  */
-#if defined(__ANDROID__)
+#if 0
 #define DETECT_OS_ANDROID 1
 #endif

--- mesa-24.3.1/include/vulkan/vk_android_native_buffer.h	2024-12-05 03:43:54.000000000 +0530
+++ mesa-24.3.1.mod/include/vulkan/vk_android_native_buffer.h	2025-01-06 16:39:32.986592254 +0530
@@ -17,7 +17,7 @@
  * system rather the build target.
  */
 
-#if defined(__ANDROID__) || defined(ANDROID)
+#if 0
 
 #include <cutils/native_handle.h>
 #if ANDROID_API_LEVEL < 28
