+++ ./src/util/detect_os.h
@@ -24,7 +24,7 @@
  * Android defines __linux__, so DETECT_OS_LINUX and DETECT_OS_POSIX will
  * also be defined.
  */
-#if defined(__ANDROID__)
+#if 0
 #define DETECT_OS_ANDROID 1
 #endif
 
+++ ./include/vulkan/vk_android_native_buffer.h	
@@ -1,5 +1,5 @@
 /* MESA: A hack to avoid #ifdefs in driver code. */
-#ifdef __ANDROID__
+#if 0
 
 #include <cutils/native_handle.h>
 #if ANDROID_API_LEVEL < 28
