+++ ./src/util/detect_os.h
@@ -24,7 +24,7 @@
  * Android defines __linux__, so DETECT_OS_LINUX and DETECT_OS_UNIX will
  * also be defined.
  */
-#if defined(__ANDROID__)
+#if 0
 #define DETECT_OS_ANDROID 1
 #endif
 
+++ ./include/vulkan/vk_android_native_buffer.h	
@@ -18,7 +18,7 @@
 #define __VK_ANDROID_NATIVE_BUFFER_H__
 
 /* MESA: A hack to avoid #ifdefs in driver code. */
-#ifdef __ANDROID__
+#if 0
 #include <cutils/native_handle.h>
 #include <vulkan/vulkan.h>
 
