--- a/src/gallium/auxiliary/gallivm/lp_bld_misc.cpp
+++ b/src/gallium/auxiliary/gallivm/lp_bld_misc.cpp
@@ -402,8 +402,12 @@
     * which allows us to enable/disable code generation based
     * on the results of cpuid on these architectures.
     */
+#if LLVM_VERSION_MAJOR >= 19 && LLVM_VERSION_MINOR >= 1
+   llvm::StringMap<bool> features = llvm::sys::getHostCPUFeatures();
+#else
    llvm::StringMap<bool> features;
    llvm::sys::getHostCPUFeatures(features);
+#endif
 
    for (StringMapIterator<bool> f = features.begin();
         f != features.end();
@@ -443,8 +447,10 @@
    /* All avx512 have avx512f */
    MAttrs.push_back(util_get_cpu_caps()->has_avx512f ? "+avx512f"  : "-avx512f");
    MAttrs.push_back(util_get_cpu_caps()->has_avx512cd ? "+avx512cd"  : "-avx512cd");
+#if LLVM_VERSION_MAJOR < 19
    MAttrs.push_back(util_get_cpu_caps()->has_avx512er ? "+avx512er"  : "-avx512er");
    MAttrs.push_back(util_get_cpu_caps()->has_avx512pf ? "+avx512pf"  : "-avx512pf");
+#endif
    MAttrs.push_back(util_get_cpu_caps()->has_avx512bw ? "+avx512bw"  : "-avx512bw");
    MAttrs.push_back(util_get_cpu_caps()->has_avx512dq ? "+avx512dq"  : "-avx512dq");
    MAttrs.push_back(util_get_cpu_caps()->has_avx512vl ? "+avx512vl"  : "-avx512vl");
