Submitted upstream at https://github.com/visit1985/mdp/pull/128

diff -u -r ../mdp-1.0.12/src/viewer.c ./src/viewer.c
--- ../mdp-1.0.12/src/viewer.c	2018-01-14 13:30:39.000000000 +0000
+++ ./src/viewer.c	2018-06-07 23:03:32.846968101 +0000
@@ -982,7 +982,7 @@
 
 int get_slide_number(char init) {
     int retval = init - '0';
-    char c;
+    int c;
     // block for tenths of a second when using getch, ERR if no input
     halfdelay(GOTO_SLIDE_DELAY);
     while((c = getch()) != ERR) {
