diff -u -r ../node-v5.1.0/deps/npm/lib/build.js ./deps/npm/lib/build.js
--- ../node-v5.1.0/deps/npm/lib/build.js	2015-11-17 15:55:27.000000000 -0500
+++ ./deps/npm/lib/build.js	2015-11-17 19:01:33.061204729 -0500
@@ -223,6 +223,8 @@
 
 function linkBin (from, to, gently, cb) {
   if (process.platform !== 'win32') {
+    // Fix shebang paths in binary scripts:
+    require('child_process').spawn('termux-fix-shebang', [from])
     return linkIfExists(from, to, gently, cb)
   } else {
     return cmdShimIfExists(from, to, cb)
