--- a/entr.c
+++ b/entr.c
@@ -181,11 +181,11 @@
 		fprintf(stderr, "open_max: %d\n", open_max);
 
 	/* prevent interactive utilities from paging output */
-	setenv("PAGER", "/bin/cat", 0);
+	setenv("PAGER", "@TERMUX_PREFIX@/bin/cat", 0);
 
 	/* ensure a shell is available to use */
 	if ((shell = getenv("SHELL")) == NULL)
-		shell = "/bin/sh";
+		shell = "@TERMUX_PREFIX@/bin/sh";
 	shell_base = strdup(shell);
 	shell_base = basename(shell_base);
 
@@ -571,7 +571,7 @@
 		new_argv = calloc(argc + 1, sizeof(char *));
 		if (new_argv == NULL)
 			err(1, "calloc");
-		new_argv[0] = "/bin/false";
+		new_argv[0] = "@TERMUX_PREFIX@/bin/false";
 		for (m = 0, i = 0, p = arg_buf; i < argc; i++) {
 			remaining = ARG_MAX - (p - arg_buf);
 			new_argv[i] = p;
