diff -u -r ../mosquitto-1.4.10/lib/CMakeLists.txt ./lib/CMakeLists.txt
--- ../mosquitto-1.4.10/lib/CMakeLists.txt	2016-08-17 13:05:14.000000000 +0200
+++ ./lib/CMakeLists.txt	2017-01-15 21:09:21.548573341 +0100
@@ -7,7 +7,12 @@
 		set (PTHREAD_LIBRARIES C:\\pthreads\\Pre-built.2\\lib\\x86\\pthreadVC2.lib)
 		set (PTHREAD_INCLUDE_DIR C:\\pthreads\\Pre-built.2\\include)
 	else (WIN32)
-		set (PTHREAD_LIBRARIES pthread)
+		find_library(LIBPTHREAD pthread)
+		if (LIBPTHREAD)
+			set (PTHREAD_LIBRARIES pthread)
+		else (LIBPTHREAD)
+			set (PTHREAD_LIBRARIES "")
+		endif()
 		set (PTHREAD_INCLUDE_DIR "")
 	endif (WIN32)
 else (${WITH_THREADING} STREQUAL ON)
@@ -44,7 +49,10 @@
 set (LIBRARIES ${OPENSSL_LIBRARIES} ${PTHREAD_LIBRARIES})
 
 if (UNIX AND NOT APPLE)
-	set (LIBRARIES ${LIBRARIES} rt)
+	find_library(LIBRT rt)
+	if (LIBRT)
+		set (LIBRARIES ${LIBRARIES} rt)
+	endif (LIBRT)
 endif (UNIX AND NOT APPLE)
 
 if (WIN32)
