diff --git a/pkg/config/config_unix.go b/pkg/config/config_unix.go
index 55a2712..e99b1b7 100644
--- a/pkg/config/config_unix.go
+++ b/pkg/config/config_unix.go
@@ -9,7 +9,7 @@ import (
 )
 
 // DefaultResolvConfPath specifies path to default resolv config file on UNIX.
-const DefaultResolvConfPath = "/etc/resolv.conf"
+const DefaultResolvConfPath = "@TERMUX_PREFIX@/etc/resolv.conf"
 
 // GetDefaultServers get system default nameserver
 func GetDefaultServers() ([]string, int, []string, error) {
