diff -uNr libpcap-1.8.1/pcap/namedb.h libpcap-1.8.1.mod/pcap/namedb.h
--- libpcap-1.8.1/pcap/namedb.h	2016-10-26 03:07:59.000000000 +0300
+++ libpcap-1.8.1.mod/pcap/namedb.h	2017-09-12 11:12:26.672992613 +0300
@@ -53,7 +53,7 @@
 	char name[122];
 };
 #ifndef PCAP_ETHERS_FILE
-#define PCAP_ETHERS_FILE "/etc/ethers"
+#define PCAP_ETHERS_FILE "@TERMUX_PREFIX@/etc/ethers"
 #endif
 PCAP_API struct	pcap_etherent *pcap_next_etherent(FILE *);
 PCAP_API u_char *pcap_ether_hostton(const char*);
