diff -u -r ../libpcap-1.8.1/optimize.c ./optimize.c
--- ../libpcap-1.8.1/optimize.c	2016-10-26 00:07:59.000000000 +0000
+++ ./optimize.c	2017-07-06 23:14:39.055350334 +0000
@@ -66,7 +66,7 @@
 /*
  * So is the check for _MSC_VER done because MinGW has this?
  */
-#if defined(_WIN32) && defined (_MSC_VER)
+#if (defined(_WIN32) && defined (_MSC_VER)) || defined(__ANDROID__)
 /*
  * ffs -- vax ffs instruction
  *
