diff -u -r ../nmap-7.70/libssh2/src/openssl.h ./libssh2/src/openssl.h
--- ../nmap-7.70/libssh2/src/openssl.h	2017-11-11 17:43:24.000000000 +0000
+++ ./libssh2/src/openssl.h	2018-08-24 08:59:09.372633959 +0000
@@ -227,14 +227,18 @@
 #endif
 
 #if OPENSSL_VERSION_NUMBER >= 0x10100000L
-#define libssh2_crypto_init() \
-  ENGINE_load_builtin_engines(); \
-  ENGINE_register_all_complete()
+# ifdef OPENSSL_NO_ENGINE
+#  define libssh2_crypto_init()
+# else
+# define libssh2_crypto_init() \
+   ENGINE_load_builtin_engines(); \
+   ENGINE_register_all_complete()
+# endif
 #else
-#define libssh2_crypto_init() \
-  OpenSSL_add_all_algorithms(); \
-  ENGINE_load_builtin_engines(); \
-  ENGINE_register_all_complete()
+# define libssh2_crypto_init() \
+   OpenSSL_add_all_algorithms(); \
+   ENGINE_load_builtin_engines(); \
+   ENGINE_register_all_complete()
 #endif
 
 #define libssh2_crypto_exit()
