diff --git a/CMakeLists.txt b/CMakeLists.txt
index 807ccb4af..daf559ac9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -146,9 +146,6 @@ set_package_properties(Threads PROPERTIES TYPE REQUIRED)
 # platform-specific logics
 if(WIN32)
   set(LIBRT wsock32 ws2_32 secur32)
-elseif(NOT APPLE)
-  find_library(LIBM m REQUIRED)
-  find_library(LIBRT rt REQUIRED)
 endif()
 if(NOT ${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
   include_directories(/usr/local/include)
