--- ./libgc/pthread_stop_world.c	2017-01-03 15:49:34.000000000 +0530
+++ ../pthread_stop_world.c	2017-01-05 18:54:28.314868754 +0530
@@ -379,11 +379,7 @@
 	      GC_printf1("Sending suspend signal to 0x%lx\n", p -> id);
 	    #endif
 
-#ifndef PLATFORM_ANDROID
         result = pthread_kill(p -> id, SIG_SUSPEND);
-#else
-        result = android_thread_kill(p -> kernel_id, SIG_SUSPEND);
-#endif
 	    switch(result) {
                 case ESRCH:
                     /* Not really there anymore.  Possible? */
--- ./mono/utils/mono-threads-posix.c	2017-01-03 15:49:38.000000000 +0530
+++ ../mono-threads-posix.c	2017-01-05 18:57:48.313245586 +0530
@@ -21,10 +21,6 @@
 
 #include <errno.h>
 
-#if defined(PLATFORM_ANDROID) && !defined(TARGET_ARM64) && !defined(TARGET_AMD64)
-#define USE_TKILL_ON_ANDROID 1
-#endif
-
 #ifdef USE_TKILL_ON_ANDROID
 extern int tkill (pid_t tid, int signal);
 #endif
