Only in .: install.pl
diff -u -r ../cowsay-3.03+dfsg1/install.sh ./install.sh
--- ../cowsay-3.03+dfsg1/install.sh	1999-11-01 15:19:21.000000000 -0500
+++ ./install.sh	2016-04-14 21:14:44.715683710 -0400
@@ -14,55 +14,8 @@
 
 filelist='cows'
 
-cat <<DOG
-===================
-cowsay Installation
-===================
-
-Searching for useful perl executables...
-DOG
-
-backdoor=$1
-
-pathdirs=`echo $PATH | tr : " "`
-for p in $pathdirs; do
-	set $p/perl $p/perl5*
-	while [ ! -z "$1" ]; do
-		if [ -x "$1" ]; then
-			echo Found perl in $1
-			perls="$perls $1"
-		fi
-		shift
-	done
-done
-for perl in $perls; do
-	if $perl -MText::Wrap -e0 >/dev/null 2>&1; then
-		echo Found a good perl in $perl
-		goodperls="$goodperls $perl"
-	fi
-done
-echo The following perl executables will run cowsay:
-echo $goodperls
-echo I recommend the latest stable perl you can find.
-set $goodperls
-if [ -z "$1" ]; then
-	echo Ack!  You do not have Perl 5 installed correctly!
-	echo Get thee to CPAN!
-	exit 1
-fi
-usethisperl=$1
-echo I will be using $1 because I know it will work.
-
-echo Now I need an installation prefix.  I will use /usr/local unless
-printf "you give me a better idea here: "
-if [ -n "$backdoor" ]; then
-	prefix=$backdoor
-	printf "%s (specified on command line)\n" $prefix
-else
-	read prefix
-fi
-
-PREFIX=${prefix:-/usr/local}
+usethisperl=@TERMUX_PREFIX@/bin/perl
+PREFIX=@TERMUX_PREFIX@
 
 echo Okay, time to install this puppy.
 
@@ -70,13 +23,13 @@
 echo s,%PREFIX%,$PREFIX,\; >> install.pl
 set -x
 mkdir -p $PREFIX/bin || (mkdir $PREFIX; mkdir $PREFIX/bin)
-$usethisperl -p install.pl cowsay > $PREFIX/bin/cowsay
+perl -p install.pl cowsay > $PREFIX/bin/cowsay
 chmod a+x $PREFIX/bin/cowsay
 ln -s cowsay $PREFIX/bin/cowthink
-mkdir -p $PREFIX/man/man1 || ($mkdir $PREFIX; mkdir $PREFIX/man; mkdir $PREFIX/man/man1)
-$usethisperl -p install.pl cowsay.1 > $PREFIX/man/man1/cowsay.1
-chmod a+r $PREFIX/man/man1/cowsay.1
-ln -s cowsay.1 $PREFIX/man/man1/cowthink.1
+mkdir -p $PREFIX/share/man/man1
+perl -p install.pl cowsay.1 > $PREFIX/share/man/man1/cowsay.1
+chmod a+r $PREFIX/share/man/man1/cowsay.1
+(cd $PREFIX/share/man/man1; ln -f -s cowsay.1 cowthink.1 )
 mkdir -p $PREFIX/share/cows || (mkdir $PREFIX; mkdir $PREFIX/share; mkdir $PREFIX/share/cows)
 tar -cf - $filelist | (cd $PREFIX/share && tar -xvf -)
 set +x
