diff -uNr ugrep-3.9.2/configure.ac ugrep-3.9.2.mod/configure.ac
--- ugrep-3.9.2/configure.ac	2022-08-21 19:20:26.000000000 +0300
+++ ugrep-3.9.2.mod/configure.ac	2022-10-29 15:09:34.749719948 +0300
@@ -292,7 +292,7 @@
 AC_SUBST(SIMD_AVX512BW_FLAGS)
 
 AC_SUBST([PTHREAD_CFLAGS], [""])
-AC_SUBST([PTHREAD_LIBS], ["-lpthread"])
+AC_SUBST([PTHREAD_LIBS], [""])
 
 AC_CONFIG_FILES([Makefile lib/Makefile src/Makefile])
 
