diff --git ./pkg/edim/src/compstat_with_ediv.c b/pkg/edim/src/compstat_with_ediv.c
index 0b335b25..010122d1 100644
--- ./pkg/edim/src/compstat_with_ediv.c
+++ ./pkg/edim/src/compstat_with_ediv.c
@@ -1,5 +1,5 @@
 /* made by 'gac', can be thrown away */
-#include "src/compiled.h" 
+#include "compiled.h" 
 #ifndef AVOID_PRECOMPILED
 extern StructInitInfo * Init__methsel1 ( void );
 extern StructInitInfo * Init__type1 ( void );
diff --git ./pkg/edim/src/ediv.c b/pkg/edim/src/ediv.c
index 23203d6b..d50a1fc7 100644
--- ./pkg/edim/src/ediv.c
+++ ./pkg/edim/src/ediv.c
@@ -12,7 +12,7 @@
 
 /* read GAP source header files with a combined header file */
 
-#include        "src/compiled.h"          /* GAP headers                */
+#include        "compiled.h"          /* GAP headers                */
 
 
 #include        <stdio.h>
diff --git ./pkg/hap/lib/CompiledGAP/absint.c b/pkg/hap/lib/CompiledGAP/absint.c
index 362dc0c2..e88fc4fd 100644
--- ./pkg/hap/lib/CompiledGAP/absint.c
+++ ./pkg/hap/lib/CompiledGAP/absint.c
@@ -1,6 +1,6 @@
 /*Written by Paul Smith*/
 
-#include "src/compiled.h"  /* the GAP headers */
+#include "compiled.h"  /* the GAP headers */
 #include <stdlib.h>        /* for abs */
 
 
diff --git ./pkg/hap/lib/CompiledGAP/signint.c b/pkg/hap/lib/CompiledGAP/signint.c
index 68310f4c..e8508863 100644
--- ./pkg/hap/lib/CompiledGAP/signint.c
+++ ./pkg/hap/lib/CompiledGAP/signint.c
@@ -1,6 +1,6 @@
 /*Based on a template by Paul Smith*/
 
-#include "src/compiled.h"  /* the GAP headers */
+#include "compiled.h"  /* the GAP headers */
 #include <stdlib.h>        /* for SignInt */
 
 
