--- ../gap-4.12.1/Makefile.rules	2022-10-20 19:53:39.000000000 +0000
+++ ./Makefile.rules	2022-10-26 08:09:47.012253589 +0000
@@ -586,7 +586,7 @@
 install-bin: gap
 	# install the real GAP executable as $(libdir)/gap/gap
 	$(INSTALL) -d -m 0755 $(DESTDIR)$(libdir)/gap
-	$(LTINSTALL) -s gap $(DESTDIR)$(libdir)/gap
+	$(LTINSTALL) gap $(DESTDIR)$(libdir)/gap
 	
 	# install a wrapper shell script invoking the real GAP executable as $(bindir)/gap
 	$(INSTALL) -d -m 0755 $(DESTDIR)$(bindir)
@@ -658,8 +658,6 @@
 	$(INSTALL) -m 0644 $(builddir)/build/version.h $(DESTDIR)$(includedir)/gap
 	$(INSTALL) -d -m 0755 $(DESTDIR)$(includedir)/gap/hpc
 	$(INSTALL) -m 0644 $(srcdir)/src/hpc/*.h $(DESTDIR)$(includedir)/gap/hpc
-	# Create a symlink to support packages using `#include "src/compiled.h"`
-	ln -sfn . $(DESTDIR)$(includedir)/gap/src
 
 install-libgap: libgap.la
 	$(INSTALL) -d -m 0755 $(DESTDIR)$(libdir)
