--- iproute2-6.0.0/bridge/vni.c.orig	2022-10-05 00:17:15.000000000 +0900
+++ iproute2-6.0.0/bridge/vni.c	2022-10-15 20:17:02.509331351 +0900
@@ -41,6 +41,9 @@
 	exit(-1);
 }
 
+#undef strdupa
+#define strdupa(s) strcpy(alloca(strlen(s) + 1), s)
+
 static int parse_vni_filter(const char *argv, struct nlmsghdr *n, int reqsize,
 			    inet_prefix *group)
 {
--- iproute2-5.15.0/ip/iplink_bond.c.orig
+++ iproute2-5.15.0/ip/iplink_bond.c
@@ -171,6 +171,9 @@
 	print_explain(stderr);
 }
 
+#undef strdupa
+#define strdupa(s) strcpy(alloca(strlen(s) + 1), s)
+
 static int bond_parse_opt(struct link_util *lu, int argc, char **argv,
 			  struct nlmsghdr *n)
 {
--- iproute2-v6.1.0/ip/iproute_lwtunnel.c.orig
+++ iproute2-v6.1.0/ip/iproute_lwtunnel.c
@@ -185,6 +185,9 @@
 	return -1;
 }
 
+#undef strdupa
+#define strdupa(s) strcpy(alloca(strlen(s) + 1), s)
+
 static int parse_seg6local_flavors(const char *buf, __u32 *flv_mask)
 {
 	unsigned char flavor_ok[SEG6_LOCAL_FLV_OP_MAX + 1] = { 0, };
--- iproute2-5.15.0/lib/color.c.orig
+++ iproute2-5.15.0/lib/color.c
@@ -93,6 +93,9 @@
 	return false;
 }
 
+#undef strdupa
+#define strdupa(s) strcpy(alloca(strlen(s) + 1), s)
+
 bool matches_color(const char *arg, int *val)
 {
 	char *dup, *p;
