diff -u -r ../wget-1.20.2/lib/stdio-impl.h ./lib/stdio-impl.h
--- ../wget-1.20.2/lib/stdio-impl.h	2019-04-01 18:26:23.000000000 +0000
+++ ./lib/stdio-impl.h	2019-04-03 21:54:29.141765465 +0000
@@ -60,7 +60,7 @@
 #  define _flags pub._flags
 #  define _r pub._r
 #  define _w pub._w
-# elif defined __ANDROID__ /* Android */
+# elif defined __ANDROID_BUT_THIS_IS_BROKEN__ /* Android */
 #  ifdef __LP64__
 #   define _gl_flags_file_t int
 #  else
