diff -uNr libgpg-error-1.36/m4/threadlib.m4 libgpg-error-1.36.mod/m4/threadlib.m4
--- libgpg-error-1.36/m4/threadlib.m4	2016-11-12 21:02:56.000000000 +0200
+++ libgpg-error-1.36.mod/m4/threadlib.m4	2019-08-16 21:23:04.646225294 +0300
@@ -213,8 +213,6 @@
             [Define if the POSIX multithreading library can be used.])
           if test -n "$LIBMULTITHREAD" || test -n "$LTLIBMULTITHREAD"; then
             if case "$gl_cv_have_weak" in *yes) true;; *) false;; esac; then
-              AC_DEFINE([USE_POSIX_THREADS_WEAK], [1],
-                [Define if references to the POSIX multithreading library should be made weak.])
               LIBTHREAD=
               LTLIBTHREAD=
             fi
