--- a/src/main.c
+++ b/src/main.c
@@ -53,8 +53,8 @@
 #include "client.h"
 #include "conf.h"
 
-static const char *socket_path = "/var/run/usbmuxd";
-#define DEFAULT_LOCKFILE "/var/run/usbmuxd.pid"
+static const char *socket_path = "@TERMUX_PREFIX@/var/run/usbmuxd";
+#define DEFAULT_LOCKFILE "@TERMUX_PREFIX@/var/run/usbmuxd.pid"
 static const char *lockfile = DEFAULT_LOCKFILE;
 
 // Global state used in other files
