--- a/src/os_unix.c
+++ b/src/os_unix.c
@@ -6264,9 +6264,7 @@
       case 0:  zDir = sqlite3_temp_directory;   break;
       case 1:  zDir = getenv("SQLITE_TMPDIR");  break;
       case 2:  zDir = getenv("TMPDIR");         break;
-      case 3:  zDir = "/var/tmp";               break;
-      case 4:  zDir = "/usr/tmp";               break;
-      case 5:  zDir = "/tmp";                   break;
+      case 3:  zDir = "@TERMUX_PREFIX@/tmp";    break;
       default: zDir = ".";                      break;
     }
     if( zDir!=0
