diff --git a/libltrace.c b/libltrace.c
index 97fcb2b..ecd501c 100644
--- a/libltrace.c
+++ b/libltrace.c
@@ -38,6 +38,7 @@
 #include "prototype.h"
 #include "read_config_file.h"
 #include "summary.h"
+#include "reallocarray.c"
 
 char *command = NULL;
 
diff --git a/options.c b/options.c
index 7f725a0..f769ec0 100644
--- a/options.c
+++ b/options.c
@@ -42,6 +42,7 @@
 #include "filter.h"
 #include "glob.h"
 #include "demangle.h"
+#include "reallocarray.c"
 
 struct options_t options = {
 	.align    = DEFAULT_ALIGN,    /* alignment column for results */
diff --git a/output.c b/output.c
index 9b20e09..85f4824 100644
--- a/output.c
+++ b/output.c
@@ -51,6 +51,7 @@
 #include "value_dict.h"
 #include "filter.h"
 #include "debug.h"
+#include "reallocarray.c"
 
 #if defined(HAVE_LIBDW)
 #include "dwarf_prototypes.h"
