--- procps-3.3.17.orig/configure	2021-02-09 17:11:25.000000000 +0700
+++ procps-3.3.17/configure	2021-05-12 00:46:28.000000000 +0700
@@ -17904,7 +17904,7 @@
 fi
 
 
-for ac_func in __fpending alarm atexit dup2 gethostname getpagesize gettimeofday iswprint memchr memmove memset mkdir nl_langinfo putenv regcomp rpmatch select setlocale strcasecmp strchr strcspn strdup strerror strncasecmp strndup strpbrk strrchr strspn strstr strtol strtoul strtoull strverscmp utmpname wcwidth
+for ac_func in __fpending alarm atexit dup2 gethostname getpagesize gettimeofday iswprint memchr memmove memset mkdir nl_langinfo putenv regcomp rpmatch select setlocale strcasecmp strchr strcspn strdup strerror strncasecmp strndup strpbrk strrchr strspn strstr strtol strtoul strtoull utmpname wcwidth
 do :
   as_ac_var=`$as_echo "ac_cv_func_$ac_func" | $as_tr_sh`
 ac_fn_c_check_func "$LINENO" "$ac_func" "$as_ac_var"
diff -ru procps-3.3.17.orig/configure.ac procps-3.3.17/configure.ac
