--- procps-3.3.17.orig/configure.ac	2021-02-09 17:11:25.000000000 +0700
+++ procps-3.3.17/configure.ac	2021-05-12 00:45:18.000000000 +0700
@@ -76,10 +76,8 @@
 # Checks for library functions.
 AC_FUNC_ERROR_AT_LINE
 AC_FUNC_FORK
-AC_FUNC_MALLOC
 AC_FUNC_MBRTOWC
 AC_FUNC_MMAP
-AC_FUNC_REALLOC
 AC_FUNC_STRTOD
 
 usrbin_execdir='${exec_prefix}/bin'
@@ -319,7 +317,7 @@
 fi
 AC_SUBST(DEJAGNU)
 
-AC_CHECK_FUNCS([__fpending alarm atexit dup2 gethostname getpagesize gettimeofday iswprint memchr memmove memset mkdir nl_langinfo putenv regcomp rpmatch select setlocale strcasecmp strchr strcspn strdup strerror strncasecmp strndup strpbrk strrchr strspn strstr strtol strtoul strtoull strverscmp utmpname wcwidth])
+AC_CHECK_FUNCS([__fpending alarm atexit dup2 gethostname getpagesize gettimeofday iswprint memchr memmove memset mkdir nl_langinfo putenv regcomp rpmatch select setlocale strcasecmp strchr strcspn strdup strerror strncasecmp strndup strpbrk strrchr strspn strstr strtol strtoul strtoull utmpname wcwidth])
 
 AC_CONFIG_FILES([Makefile
                  include/Makefile
diff -ru procps-3.3.17.orig/pgrep.c procps-3.3.17/pgrep.c
