--- a/library/meminfo.c
+++ b/library/meminfo.c
@@ -21,7 +21,7 @@
 
 #include <errno.h>
 #include <fcntl.h>
-#include <search.h>
+#include "search.h"
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
--- a/library/vmstat.c
+++ b/library/vmstat.c
@@ -24,7 +24,7 @@
 
 #include <errno.h>
 #include <fcntl.h>
-#include <search.h>
+#include "search.h"
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
