Bionic lacks strverscmp as well.

diff -u -r ../procps-ng-3.3.14/Makefile.in ./Makefile.in
--- ../procps-ng-3.3.14/Makefile.in	2018-04-10 11:40:19.331497997 +0000
+++ ./Makefile.in	2018-05-19 00:15:36.379445979 +0000
@@ -136,7 +136,7 @@
 @WITH_NCURSES_TRUE@	watch.1 \
 @WITH_NCURSES_TRUE@	top/top.1
 
-@CYGWIN_TRUE@@WITH_NCURSES_TRUE@am__append_18 = lib/strverscmp.c
+@WITH_NCURSES_TRUE@am__append_18 = lib/strverscmp.c
 @CYGWIN_FALSE@@WITH_NCURSES_TRUE@am__append_19 = \
 @CYGWIN_FALSE@@WITH_NCURSES_TRUE@	slabtop.1
 
@@ -356,8 +356,8 @@
 tload_DEPENDENCIES = ./proc/libprocps.la $(am__DEPENDENCIES_2)
 am__top_top_SOURCES_DIST = top/top.h top/top.c top/top_nls.h \
 	top/top_nls.c lib/fileutils.c lib/strverscmp.c
-@CYGWIN_TRUE@@WITH_NCURSES_TRUE@am__objects_1 =  \
-@CYGWIN_TRUE@@WITH_NCURSES_TRUE@	lib/strverscmp.$(OBJEXT)
+@WITH_NCURSES_TRUE@am__objects_1 =  \
+@WITH_NCURSES_TRUE@	lib/strverscmp.$(OBJEXT)
 @WITH_NCURSES_TRUE@am_top_top_OBJECTS = top/top.$(OBJEXT) \
 @WITH_NCURSES_TRUE@	top/top_nls.$(OBJEXT) \
 @WITH_NCURSES_TRUE@	lib/fileutils.$(OBJEXT) $(am__objects_1)
