diff -u -r ../procps-ng-3.3.14/procio.c ./procio.c
--- ../procps-ng-3.3.14/procio.c	2018-04-10 13:26:34.695269068 +0200
+++ ./procio.c	2018-05-19 03:03:48.858822656 +0200
@@ -46,6 +46,7 @@
 static ssize_t proc_write(void *, const char *, size_t);
 static int proc_close(void *);
 
+#ifndef __ANDROID__
 __extension__
 static cookie_io_functions_t procio = {
     .read  = proc_read,
@@ -53,9 +54,13 @@
     .seek  = NULL,
     .close = proc_close,
 };
+#endif
 
 FILE *fprocopen(const char *path, const char *mode)
 {
+#ifdef __ANDROID__
+	return fopen(path, mode);
+#else
 	pcookie_t *cookie = NULL;
 	FILE *handle = NULL;
 	mode_t flags = 0;
@@ -144,6 +149,7 @@
 	}
 out:
 	return handle;
+#endif
 }
 
 static
