--- procps-3.3.17.orig/proc/sysinfo.c	2021-02-09 17:11:25.000000000 +0700
+++ procps-3.3.17/proc/sysinfo.c	2021-05-11 23:59:48.000000000 +0700
@@ -33,6 +33,9 @@
 #ifdef __CYGWIN__
 #include <sys/param.h>
 #endif
+#ifdef __ANDROID__
+#include <sys/sysinfo.h>
+#endif
 #include "alloc.h"
 #include "version.h"
 #include "sysinfo.h" /* include self to verify prototypes */
@@ -104,6 +107,18 @@
 
 /***********************************************************************/
 int uptime(double *restrict uptime_secs, double *restrict idle_secs) {
+#ifdef __ANDROID__
+    /* Android does not allow read access to /proc/uptime */
+    SET_IF_DESIRED(idle_secs, 0.);
+    struct sysinfo system_information;
+    if (sysinfo(&system_information) == 0) {
+        SET_IF_DESIRED(uptime_secs, (double) system_information.uptime);
+        return (double) system_information.uptime;
+    } else {
+        SET_IF_DESIRED(uptime_secs, 0.);
+        return 0;
+    }
+#else
     double up=0, idle=0;
     char *savelocale;
 
@@ -121,6 +136,7 @@
     SET_IF_DESIRED(uptime_secs, up);
     SET_IF_DESIRED(idle_secs, idle);
     return up;	/* assume never be zero seconds in practice */
+#endif
 }
 
 unsigned long getbtime(void) {
diff -ru procps-3.3.17.orig/proc/whattime.c procps-3.3.17/proc/whattime.c
