diff -u -r ../procps-ng-3.3.9/top/Makefile.in ./top/Makefile.in
--- ../procps-ng-3.3.9/top/Makefile.in	2013-12-03 12:23:23.293820882 +0100
+++ ./top/Makefile.in	2014-07-06 07:50:12.794907542 +0200
@@ -103,8 +103,10 @@
 	"$(DESTDIR)$(procpsngdir)"
 PROGRAMS = $(usrbin_exec_PROGRAMS)
 am__top_SOURCES_DIST = top.h top.c top_nls.h top_nls.c \
+        strverscmp.c \
 	$(top_srcdir)/lib/fileutils.c
 @WITH_NCURSES_TRUE@am_top_OBJECTS = top.$(OBJEXT) top_nls.$(OBJEXT) \
+@WITH_NCURSES_TRUE@	strverscmp.$(OBJEXT) \
 @WITH_NCURSES_TRUE@	fileutils.$(OBJEXT)
 top_OBJECTS = $(am_top_OBJECTS)
 top_DEPENDENCIES =
@@ -359,6 +361,7 @@
 @WITH_NCURSES_TRUE@top_SOURCES = \
 @WITH_NCURSES_TRUE@	top.h \
 @WITH_NCURSES_TRUE@	top.c \
+@WITH_NCURSES_TRUE@	strverscmp.c \
 @WITH_NCURSES_TRUE@	top_nls.h \
 @WITH_NCURSES_TRUE@	top_nls.c \
 @WITH_NCURSES_TRUE@	$(top_srcdir)/lib/fileutils.c
