diff -u -r ../libevent-2.0.22-stable/evutil_rand.c ./evutil_rand.c
--- ../libevent-2.0.22-stable/evutil_rand.c	2013-11-01 14:18:57.000000000 -0400
+++ ./evutil_rand.c	2016-01-03 20:05:10.168762009 -0500
@@ -171,10 +171,15 @@
 	ev_arc4random_buf(buf, n);
 }
 
+#ifndef __ANDROID__
+/* The arc4random_addrandom() has been removed from 64-bit Android libc,
+   see https://bugzilla.mozilla.org/show_bug.cgi?id=931354 for motivation
+   about just removing this function completely. */
 void
 evutil_secure_rng_add_bytes(const char *buf, size_t n)
 {
 	arc4random_addrandom((unsigned char*)buf,
 	    n>(size_t)INT_MAX ? INT_MAX : (int)n);
 }
+#endif
 
