diff --git a/deps/lpeg.cmake b/deps/lpeg.cmake
index a95a207..7361b37 100644
--- a/deps/lpeg.cmake
+++ b/deps/lpeg.cmake
@@ -1,13 +1,18 @@
 set(LPEGLIB_DIR "${CMAKE_CURRENT_SOURCE_DIR}/deps/lpeg" CACHE PATH "Path to lpeg")
 
-add_library(lpeglib STATIC
-  ${LPEGLIB_DIR}/lpcap.c
-  ${LPEGLIB_DIR}/lpcode.c
-  ${LPEGLIB_DIR}/lpcset.c
-  ${LPEGLIB_DIR}/lpprint.c
-  ${LPEGLIB_DIR}/lptree.c
-  ${LPEGLIB_DIR}/lpvm.c
-)
+if (WithSharedLPEG)
+  list(APPEND LUVI_LIBRARIES ${LPEG_LIBRARIES})
+else ()
+  add_library(lpeglib STATIC
+    ${LPEGLIB_DIR}/lpcap.c
+    ${LPEGLIB_DIR}/lpcode.c
+    ${LPEGLIB_DIR}/lpcset.c
+    ${LPEGLIB_DIR}/lpprint.c
+    ${LPEGLIB_DIR}/lptree.c
+    ${LPEGLIB_DIR}/lpvm.c
+  )
+
+  list(APPEND LUVI_LIBRARIES lpeglib)
+endif ()
 
-list(APPEND LUVI_LIBRARIES lpeglib)
 list(APPEND LUVI_DEFINITIONS WITH_LPEG=1)
