diff --git a/cmake/Modules/LuaJITAddExecutable.cmake b/cmake/Modules/LuaJITAddExecutable.cmake
index e250d08..22d53ae 100644
--- a/cmake/Modules/LuaJITAddExecutable.cmake
+++ b/cmake/Modules/LuaJITAddExecutable.cmake
@@ -24,7 +24,7 @@ MACRO(LUAJIT_add_custom_commands luajit_target)
       math(EXPR _stripped_file_length "${_luajit_file_length} - ${_luajit_source_dir_length} - 1")
       string(SUBSTRING ${file} ${_begin} ${_stripped_file_length} stripped_file)
 
-      set(generated_file "${CMAKE_BINARY_DIR}/jitted_tmp/${stripped_file}_${luajit_target}_generated${CMAKE_C_OUTPUT_EXTENSION}")
+      set(generated_file "${CMAKE_BINARY_DIR}/jitted_tmp/${stripped_file}_${luajit_target}_generated.c")
 
       # Only use LUA_PATH if it's set
       IF(LUA_PATH)
