--- ../cache/libbsd-0.9.1/include/bsd/sys/endian.h	2017-06-06 02:21:05.000000000 +0000
+++ ./include/bsd/sys/endian.h	2018-10-16 02:59:14.582440259 +0000
@@ -25,12 +25,13 @@
  * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
  * SUCH DAMAGE.
  */
-
+#if !defined(__ANDROID__)
 #ifdef LIBBSD_OVERLAY
 #include_next <endian.h>
 #else
 #include <endian.h>
 #endif
+#endif
 
 #ifndef LIBBSD_SYS_ENDIAN_H
 #define LIBBSD_SYS_ENDIAN_H
