--- ../cache/libbsd-0.9.1/src/vis.c	2018-05-22 14:07:42.000000000 +0000
+++ ./src/vis.c	2018-10-16 03:14:02.997723611 +0000
@@ -305,7 +305,7 @@
 	/* See comment in istrsenvisx() output loop, below. */
 	wmsk = 0;
 	for (i = sizeof(wmsk) - 1; i >= 0; i--) {
-		shft = i * NBBY;
+		shft = i * CHAR_BIT;
 		bmsk = (uint64_t)0xffLL << shft;
 		wmsk |= bmsk;
 		if ((c & wmsk) || i == 0)
@@ -536,7 +536,7 @@
 			clen = 0;
 			wmsk = 0;
 			for (i = sizeof(wmsk) - 1; i >= 0; i--) {
-				shft = i * NBBY;
+				shft = i * CHAR_BIT;
 				bmsk = (uint64_t)0xffLL << shft;
 				wmsk |= bmsk;
 				if ((*dst & wmsk) || i == 0)
