diff -uNr node-v13.0.0/deps/uv/src/unix/fs.c node-v13.0.0.mod/deps/uv/src/unix/fs.c
--- node-v13.0.0/deps/uv/src/unix/fs.c	2019-10-22 17:23:28.000000000 +0300
+++ node-v13.0.0.mod/deps/uv/src/unix/fs.c	2020-01-16 02:27:44.944603915 +0200
@@ -1048,6 +1048,7 @@
     goto out;
   }
 
+#ifndef __ANDROID__
 #ifdef FICLONE
   if (req->flags & UV_FS_COPYFILE_FICLONE ||
       req->flags & UV_FS_COPYFILE_FICLONE_FORCE) {
@@ -1072,7 +1073,7 @@
     goto out;
   }
 #endif
-
+#endif
   bytes_to_send = src_statsbuf.st_size;
   in_offset = 0;
   while (bytes_to_send != 0) {
