diff -u -r ../node-v12.9.0/deps/v8/src/logging/log.cc ./deps/v8/src/logging/log.cc
--- ../node-v12.9.0/deps/v8/src/logging/log.cc	2019-08-20 17:14:50.000000000 +0000
+++ ./deps/v8/src/logging/log.cc	2019-08-24 22:39:52.426124955 +0000
@@ -280,7 +280,7 @@
   FILE* perf_output_handle_;
 };
 
-const char PerfBasicLogger::kFilenameFormatString[] = "/tmp/perf-%d.map";
+const char PerfBasicLogger::kFilenameFormatString[] = "@TERMUX_PREFIX@/tmp/perf-%d.map";
 // Extra space for the PID in the filename
 const int PerfBasicLogger::kFilenameBufferPadding = 16;
 
