--- ./lib/os.js.orig	2021-09-14 11:38:18.226906330 +0530
+++ ./lib/os.js	2021-09-14 11:39:16.286906308 +0530
@@ -183,7 +183,7 @@
     path = safeGetenv('TMPDIR') ||
            safeGetenv('TMP') ||
            safeGetenv('TEMP') ||
-           '/tmp';
+           '@TERMUX_PREFIX@/tmp';
     if (path.length > 1 && StringPrototypeEndsWith(path, '/'))
       path = StringPrototypeSlice(path, 0, -1);
   }
