--- ./tools/v8_gypfiles/v8.gyp	2021-06-03 07:15:33.000000000 +0530
+++ ./tools/v8_gypfiles/v8.gyp.mod	2021-06-18 21:18:09.722327631 +0530
@@ -928,6 +928,7 @@
             '<(V8_ROOT)/src/base/platform/platform-posix.h',
             '<(V8_ROOT)/src/base/platform/platform-posix-time.cc',
             '<(V8_ROOT)/src/base/platform/platform-posix-time.h',
+	    '<(V8_ROOT)/src/base/platform/platform-linux.h',
           ],
           'link_settings': {
             'target_conditions': [
