--- a/CMakeLists.txt      2026-01-07 17:43:48.515232213 +0530
+++ b/CMakeLists.txt        2026-01-07 18:48:23.927232444 +0530
@@ -167,7 +167,7 @@
         TYPE REQUIRED
         PURPOSE "Required for building Tablet input KCM"
     )
-    pkg_check_modules(LIBWACOM libwacom REQUIRED IMPORTED_TARGET)
+#    pkg_check_modules(LIBWACOM libwacom REQUIRED IMPORTED_TARGET)
 endif()

 find_package(SDL2 2.0.16)
 
