--- a/kcms/CMakeLists.txt 2026-01-07 17:43:48.483232213 +0530
+++ b/kcms/CMakeLists.txt    2026-01-07 19:18:04.487232550 +0530
@@ -1,6 +1,6 @@
 remove_definitions(-DQT_NO_CAST_FROM_ASCII -DQT_STRICT_ITERATORS -DQT_NO_CAST_FROM_BYTEARRAY -DQT_NO_KEYWORDS)

-add_subdirectory(libkwindevices)
+#add_subdirectory(libkwindevices)

 if(X11_Xkb_FOUND AND XCB_XKB_FOUND)
     add_subdirectory( keyboard )
@@ -25,7 +25,7 @@
 add_subdirectory(runners)
 add_subdirectory(spellchecking)
 add_subdirectory(qtquicksettings)
-add_subdirectory(touchscreen)
+#add_subdirectory(touchscreen)

 add_subdirectory(workspaceoptions)
 if (KF6Baloo_FOUND)
@@ -42,5 +42,5 @@
 endif()

 if(BUILD_KCM_TABLET)
-    add_subdirectory(tablet)
+#    add_subdirectory(tablet)
 endif()
