--- a/CMakeLists.txt      2026-01-07 17:43:48.515232213 +0530
+++ b/CMakeLists.txt    2026-01-07 18:46:45.463232438 +0530
@@ -212,7 +212,7 @@
     PURPOSE "Support audible bell in kaccess"
     TYPE REQUIRED)

-find_package(UDev)
+#find_package(UDev)
 set_package_properties(UDev PROPERTIES
     DESCRIPTION "API for enumerating and introspecting local devices (part of systemd)"
     URL "https://www.freedesktop.org/wiki/Software/systemd/"

--- a/kcms/CMakeLists.txt 2026-01-07 17:43:48.483232213 +0530
+++ b/kcms/CMakeLists.txt       2026-01-07 19:10:09.835232522 +0530
@@ -16,7 +16,7 @@
 add_subdirectory( desktoppaths )

 if (TARGET SDL2::SDL2)
-    add_subdirectory( gamecontroller )
+#    add_subdirectory( gamecontroller )
 endif()

 add_subdirectory(activities)
 
 
