diff -uNr dash-0.5.10.2/src/var.c dash-0.5.10.2.mod/src/var.c
--- dash-0.5.10.2/src/var.c	2018-05-17 14:10:58.000000000 +0300
+++ dash-0.5.10.2.mod/src/var.c	2018-06-21 10:46:49.260059837 +0300
@@ -74,7 +74,7 @@
 MKINIT struct localvar_list *localvar_stack;
 
 const char defpathvar[] =
-	"PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin";
+	"PATH=@TERMUX_PREFIX@/bin:@TERMUX_PREFIX@/bin/applets";
 #ifdef IFS_BROKEN
 const char defifsvar[] = "IFS= \t\n";
 #else
