diff -u -r ../byobu-5.122/usr/bin/byobu.in ./usr/bin/byobu.in
--- ../byobu-5.122/usr/bin/byobu.in	2017-08-09 23:53:48.000000000 +0200
+++ ./usr/bin/byobu.in	2017-09-20 01:31:55.576848354 +0200
@@ -46,7 +46,7 @@
 		. "$HOME/.byoburc"
 	fi
 fi
-export BYOBU_CHARMAP=$(locale charmap)
+export BYOBU_CHARMAP=UTF-8
 [ -r "$HOME/.byoburc" ] && . "$HOME/.byoburc"
 [ -z "${BYOBU_PREFIX}" ] && export BYOBU_PREFIX="@prefix@" || export BYOBU_PREFIX
 . "${BYOBU_PREFIX}/lib/${PKG}/include/common"
