diff --git a/crates/tinymist-std/Cargo.toml b/crates/tinymist-std/Cargo.toml
index c20b92d..b723b34 100644
--- a/crates/tinymist-std/Cargo.toml
+++ b/crates/tinymist-std/Cargo.toml
@@ -79,7 +79,7 @@ rkyv-validation = ["rkyv/validation"]
 
 __web = ["wasm-bindgen", "js-sys", "web-time"]
 web = ["__web"]
-system = ["tempfile", "same-file", "battery"]
+system = ["tempfile", "same-file"]
 bi-hash = []
 
 [lints]
