diff -uNr dar-2.7.0/configure dar-2.7.0.mod/configure
--- dar-2.7.0/configure	2021-04-24 12:18:42.000000000 +0000
+++ dar-2.7.0.mod/configure	2021-05-03 18:46:56.060794046 +0000
@@ -24003,7 +24003,7 @@
 
                  { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libgcrypt usability" >&5
 $as_echo_n "checking for libgcrypt usability... " >&6; }
-                 if test "$cross_compiling" = yes; then :
+                 if test "$cross_compiling" = fakeyes; then :
 
                                   # if cross compiling
 
@@ -24071,13 +24071,13 @@
 }
 
 _ACEOF
-if ac_fn_cxx_try_run "$LINENO"; then :
+if true; then :
 
 $as_echo "#define CRYPTO_AVAILABLE 1" >>confdefs.h
 
                                   { $as_echo "$as_me:${as_lineno-$LINENO}: result: ok" >&5
 $as_echo "ok" >&6; }
-                                  if test "$cross_compiling" = yes; then :
+                                  if test "$cross_compiling" = "fakeyes"; then :
   { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
 $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
 as_fn_error $? "cannot run test program while cross compiling
@@ -24106,7 +24106,7 @@
 }
 
 _ACEOF
-if ac_fn_cxx_try_run "$LINENO"; then :
+if true; then :
 
 else
    libgcrypt_hash_bug="yes"
@@ -26596,7 +26596,7 @@
 }
 
 _ACEOF
-if ac_fn_cxx_try_run "$LINENO"; then :
+if false; then :
 
                                  local_threadar=yes
 
