diff -uNr dar-2.6.12/configure dar-2.6.12.mod/configure
--- dar-2.6.12/configure	2020-09-11 14:26:19.000000000 +0000
+++ dar-2.6.12.mod/configure	2020-11-05 21:36:52.997365365 +0000
@@ -23331,7 +23331,7 @@
 
                  { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libgcrypt usability" >&5
 $as_echo_n "checking for libgcrypt usability... " >&6; }
-                 if test "$cross_compiling" = yes; then :
+                 if test "$cross_compiling" = "fakeyes"; then :
   { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
 $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
 as_fn_error $? "cannot run test program while cross compiling
@@ -23365,13 +23365,13 @@
 }
 
 _ACEOF
-if ac_fn_cxx_try_run "$LINENO"; then :
+if true; then :
 
 $as_echo "#define CRYPTO_AVAILABLE 1" >>confdefs.h
 
                                   { $as_echo "$as_me:${as_lineno-$LINENO}: result: ok" >&5
 $as_echo "ok" >&6; }
-                                  if test "$cross_compiling" = yes; then :
+                                  if test "$cross_compiling" = "fakeyes"; then :
   { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
 $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
 as_fn_error $? "cannot run test program while cross compiling
@@ -23400,7 +23400,7 @@
 }
 
 _ACEOF
-if ac_fn_cxx_try_run "$LINENO"; then :
+if true; then :
 
 else
    libgcrypt_hash_bug="yes"
@@ -25756,7 +25756,7 @@
 
                { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libthreadar() operationability" >&5
 $as_echo_n "checking for libthreadar() operationability... " >&6; }
-               if test "$cross_compiling" = yes; then :
+               if test "$cross_compiling" = "fakeyes"; then :
   { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
 $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
 as_fn_error $? "cannot run test program while cross compiling
@@ -25799,7 +25799,7 @@
 }
 
 _ACEOF
-if ac_fn_cxx_try_run "$LINENO"; then :
+if false; then :
 
                                  local_threadar=yes
 
