diff --git a/configure b/configure
--- a/configure
+++ b/configure
@@ -26165,7 +26165,7 @@ printf "%s\n" "#define MIN_VERSION_GCRYPT_HASH_BUG \"$min_version_gcrypt_hash_bu
 
                  { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for libgcrypt usability" >&5
 printf %s "checking for libgcrypt usability... " >&6; }
-                 if test "$cross_compiling" = yes
+                 if test "$cross_compiling" = fakeyes;
 then :
 
                                   # if cross compiling
@@ -26235,14 +26235,14 @@ main (void)
 }
 
 _ACEOF
-if ac_fn_cxx_try_run "$LINENO"
+if true;
 then :
 
 printf "%s\n" "#define CRYPTO_AVAILABLE 1" >>confdefs.h
 
                                   { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: ok" >&5
 printf "%s\n" "ok" >&6; }
-                                  if test "$cross_compiling" = yes
+                                  if test "$cross_compiling" = "fakeyes"
 then :
   { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
 printf "%s\n" "$as_me: error: in \`$ac_pwd':" >&2;}
@@ -26272,7 +26272,7 @@ main (void)
 }
 
 _ACEOF
-if ac_fn_cxx_try_run "$LINENO"
+if true;
 then :
 
 else $as_nop
@@ -28990,7 +28990,7 @@ main (void)
 }
 
 _ACEOF
-if ac_fn_cxx_try_run "$LINENO"
+if false;
 then :
 
                                  local_threadar=yes
