--- ./src/third_party/blake3/CMakeLists.txt.orig	2021-09-01 18:47:43.285956872 +0530
+++ ./src/third_party/blake3/CMakeLists.txt	2021-09-01 18:49:01.195956842 +0530
@@ -27,7 +27,7 @@
 
   # First check if it's possible to use the assembler variant for the feature.
   string(TOUPPER "have_asm_${feature}" have_feature)
-  if(NOT DEFINED "${have_feature}" AND CMAKE_SIZEOF_VOID_P EQUAL 8)
+  if(NOT DEFINED "${have_feature}" AND CMAKE_SIZEOF_VOID_P EQUAL 8 AND 0)
     if(NOT CMAKE_REQUIRED_QUIET)
       message(STATUS "Performing Test ${have_feature}")
     endif()
