--- ./gio/gdbusprivate.c.orig	2021-12-10 17:28:44.551721009 +0530
+++ ./gio/gdbusprivate.c	2021-12-10 19:24:17.521718364 +0530
@@ -2477,8 +2477,8 @@
 
   /* Copy what dbus.git does: allow the /var/lib path to be configurable at
    * build time, but hard-code the system-wide machine ID path in /etc. */
-  const gchar *var_lib_path = LOCALSTATEDIR "/lib/dbus/machine-id";
-  const gchar *etc_path = "/etc/machine-id";
+  const gchar *var_lib_path = LOCALSTATEDIR "@TERMUX_PREFIX@/lib/dbus/machine-id";
+  const gchar *etc_path = "@TERMUX_PREFIX@/etc/machine-id";
 
   if (!g_file_get_contents (var_lib_path,
                             &ret,
