--- a/girepository/introspection/meson.build
+++ b/girepository/introspection/meson.build
@@ -22,9 +22,11 @@
 # Use currently built libraries to run g-ir-scanner and the various tools
 # this may not happen if we don't set the library paths.
 # FIXME: https://github.com/mesonbuild/meson/issues/16127
-gi_gen_env_variables.prepend(glib_exec_var_library_path,
-  fs.parent(libglib.full_path()), fs.parent(libgobject.full_path()),
-  fs.parent(libgmodule.full_path()), fs.parent(libgio.full_path()))
+if meson.can_run_host_binaries()
+  gi_gen_env_variables.prepend(glib_exec_var_library_path,
+    fs.parent(libglib.full_path()), fs.parent(libgobject.full_path()),
+    fs.parent(libgmodule.full_path()), fs.parent(libgio.full_path()))
+endif
 
 if 'address' in glib_sanitizers
   gi_gen_env_variables.append('ASAN_OPTIONS',
