https://gitlab.gnome.org/GNOME/glib/-/issues/2820

--- a/glib/ghash.h
+++ b/glib/ghash.h
@@ -162,7 +162,7 @@
                          gconstpointer  v2);
 
 /* Macro for optimization in the case it is not used as callback function */
-#define g_str_equal(v1, v2) (strcmp ((gconstpointer) (v1), (gconstpointer) (v2)) == 0)
+#define g_str_equal(v1, v2) (strcmp ((const char *) (gconstpointer) (v1), (const char *) (gconstpointer) (v2)) == 0)
 
 GLIB_AVAILABLE_IN_ALL
 guint    g_str_hash     (gconstpointer  v);
