diff -u -r ../glib-2.50.1/glib/gmessages.c ./glib/gmessages.c
--- ../glib-2.50.1/glib/gmessages.c	2016-10-10 09:39:07.000000000 -0400
+++ ./glib/gmessages.c	2016-10-15 19:16:30.272794344 -0400
@@ -1842,7 +1842,7 @@
   return isatty (output_fd);
 }
 
-#ifdef __linux__
+#if defined(__linux__) && !defined(__ANDROID__)
 static int journal_fd = -1;
 
 #ifndef SOCK_CLOEXEC
@@ -1881,7 +1881,7 @@
 gboolean
 g_log_writer_is_journald (gint output_fd)
 {
-#ifdef __linux__
+#if defined(__linux__) && !defined(__ANDROID__)
   /* FIXME: Use the new journal API for detecting whether we’re writing to the
    * journal. See: https://github.com/systemd/systemd/issues/2473
    */
@@ -2016,7 +2016,7 @@
   return g_string_free (gstring, FALSE);
 }
 
-#ifdef __linux__
+#if defined(__linux__) && !defined(__ANDROID__)
 static int
 journal_sendv (struct iovec *iov,
                gsize         iovlen)
@@ -2130,7 +2130,7 @@
                        gsize            n_fields,
                        gpointer         user_data)
 {
-#ifdef __linux__
+#if defined(__linux__) && !defined(__ANDROID__)
   const char equals = '=';
   const char newline = '\n';
   gsize i, k;
