diff -u -r ../glib-2.58.1/glib/gspawn.c ./glib/gspawn.c
--- ../glib-2.58.1/glib/gspawn.c	2018-09-21 14:23:53.000000000 +0000
+++ ./glib/gspawn.c	2018-10-07 18:17:08.394012273 +0000
@@ -2020,7 +2022,7 @@
 
     new_argv = g_new0 (gchar*, argc + 2); /* /bin/sh and NULL */
     
-    new_argv[0] = (char *) "/bin/sh";
+    new_argv[0] = (char *) "@TERMUX_PREFIX@/bin/sh";
     new_argv[1] = (char *) file;
     while (argc > 0)
       {
@@ -2099,7 +2101,7 @@
            * what to search if PATH is unset. POSIX may, dunno.
            */
           
-          path = "/bin:/usr/bin:.";
+          path = "@TERMUX_PREFIX@/bin:@TERMUX_PREFIX@/bin/applets:.";
 	}
 
       len = strlen (file) + 1;
