--- ./glib/gspawn.c.orig	2021-12-10 22:01:14.151714772 +0530
+++ ./glib/gspawn.c	2021-12-10 22:01:19.781714770 +0530
@@ -2200,7 +2200,7 @@
        * * what to search if PATH is unset. POSIX may, dunno.
        * */
 
-      chosen_search_path = "/bin:/usr/bin:.";
+      chosen_search_path = "@TERMUX_PREFIX/bin:.";
     }
 
   if (search_path || search_path_from_envp)
@@ -2593,7 +2593,7 @@
   if (argc + 2 > argv_buffer_len)
     return FALSE;
 
-  argv_buffer[0] = (char *) "/bin/sh";
+  argv_buffer[0] = (char *) "@TERMUX_PREFIX@/bin/sh";
   argv_buffer[1] = (char *) file;
   while (argc > 0)
     {
