--- ./glib/gspawn.c.orig	2020-07-20 22:15:45.985010959 +0200
+++ ./glib/gspawn.c	2020-07-20 22:18:29.707654650 +0200
@@ -1825,7 +1825,7 @@
        * * what to search if PATH is unset. POSIX may, dunno.
        * */
 
-      chosen_search_path = "/bin:/usr/bin:.";
+      chosen_search_path = "@TERMUX_PREFIX@/bin:.";
     }
 
   /* Allocate a buffer which the fork()ed child can use to assemble potential
@@ -2246,7 +2246,7 @@
   if (argc + 2 > argv_buffer_len)
     return FALSE;
 
-  argv_buffer[0] = (char *) "/bin/sh";
+  argv_buffer[0] = (char *) "@TERMUX_PREFIX@/bin/sh";
   argv_buffer[1] = (char *) file;
   while (argc > 0)
     {
