--- ./meson.build.orig	2021-12-10 06:56:41.609999906 +0530
+++ ./meson.build	2021-12-11 17:23:07.992103796 +0530
@@ -443,7 +443,6 @@
     # A zero-length format string shouldn't be considered an issue.
     '-Wno-format-zero-length',
     '-Werror=declaration-after-statement',
-    '-Werror=format=2',
     '-Werror=implicit-function-declaration',
     '-Werror=init-self',
     '-Werror=missing-include-dirs',
@@ -2047,9 +2046,8 @@
 # FIXME: glib-gettext.m4 has much more checks to detect broken/uncompatible
 # implementations. This could be extended if issues are found in some platforms.
 libintl_deps = []
-if cc.has_function('ngettext', args : osx_ldflags)
-  have_bind_textdomain_codeset = cc.has_function('bind_textdomain_codeset')
-else
+have_bind_textdomain_codeset = cc.has_function('bind_textdomain_codeset')
+if host_system != 'android'
   # First just find the bare library.
   libintl = cc.find_library('intl', required : false)
   # The bare library probably won't link without help if it's static.
