--- a/meson.build
+++ b/meson.build
@@ -464,7 +464,6 @@
     '-Wno-format-zero-length',
     # We explicitly require variadic macros
     '-Wno-variadic-macros',
-    '-Werror=format=2',
     '-Werror=init-self',
     '-Werror=missing-include-dirs',
     '-Werror=pointer-arith',
@@ -545,7 +544,6 @@
   'getresuid',
   'getvfsstat',
   'gmtime_r',
-  'hasmntopt',
   'inotify_init1',
   'issetugid',
   'kevent',
@@ -900,7 +898,7 @@
                int main (int argc, char ** argv) {
                  syscall (__NR_futex_time64, NULL, FUTEX_WAKE, FUTEX_WAIT);
                  return 0;
-               }''', name : 'futex(2) system call')
+               }''', name : 'futex(2) system call') and host_system != 'android'
   glib_conf.set('HAVE_FUTEX_TIME64', 1)
 endif
 
@@ -907,7 +905,7 @@
                  syscall (SYS_pidfd_open, 0, 0);
                  waitid (P_PIDFD, 0, &child_info, WEXITED | WNOHANG);
                  return 0;
-               }''', name : 'pidfd_open(2) system call')
+               }''', name : 'pidfd_open(2) system call') and host_system != 'android'
   glib_conf.set('HAVE_PIDFD', 1)
 endif
 
@@ -2092,7 +2090,7 @@
 # implementations. This could be extended if issues are found in some platforms.
 libintl_deps = []
 libintl = dependency('intl', required: false)
-if libintl.found()
+if false
   # libintl supports different threading APIs, which may not
   # require additional flags, but it defaults to using pthreads if
   # found. Meson's "threads" dependency does not allow you to
@@ -2114,7 +2112,7 @@
   endif
 endif
 
-if libintl.found()
+if true
   have_bind_textdomain_codeset = cc.has_function('bind_textdomain_codeset', dependencies: libintl_deps)
 else
   libintl = subproject('proxy-libintl').get_variable('intl_dep')
