diff -u -r ../glib-2.60.2/meson.build ./meson.build
--- ../glib-2.60.2/meson.build	2019-05-03 13:43:28.000000000 +0000
+++ ./meson.build	2019-05-11 20:10:26.054269305 +0000
@@ -360,7 +360,6 @@
     # https://wiki.gnome.org/Projects/GLib/CompilerRequirements#Function_pointer_conversions.
     '-Wno-pedantic',
     '-Werror=declaration-after-statement',
-    '-Werror=format=2',
     '-Werror=implicit-function-declaration',
     '-Werror=init-self',
     '-Werror=missing-include-dirs',
