--- a/meson.build
+++ b/meson.build
@@ -464,7 +464,6 @@
     '-Wno-format-zero-length',
     # We explicitly require variadic macros
     '-Wno-variadic-macros',
-    '-Werror=format=2',
     '-Werror=init-self',
     '-Werror=missing-include-dirs',
     '-Werror=pointer-arith',
@@ -545,7 +544,6 @@
   'getresuid',
   'getvfsstat',
   'gmtime_r',
-  'hasmntopt',
   'inotify_init1',
   'issetugid',
   'kevent',
@@ -2088,9 +2086,8 @@
 # FIXME: glib-gettext.m4 has much more checks to detect broken/uncompatible
 # implementations. This could be extended if issues are found in some platforms.
 libintl_deps = []
-if cc.has_function('ngettext')
-  have_bind_textdomain_codeset = cc.has_function('bind_textdomain_codeset')
-else
+have_bind_textdomain_codeset = cc.has_function('bind_textdomain_codeset')
+if host_system != 'android'
   # First just find the bare library.
   libintl = cc.find_library('intl', required : false)
   # The bare library probably won't link without help if it's static.
