# TODO: Remove this after new Ubuntu provides autoconf 2.72

--- a/configure.ac
+++ b/configure.ac
@@ -1,5 +1,5 @@
 dnl Process this file with autoconf to produce a configure script.
-AC_PREREQ([2.72])
+AC_PREREQ([2.71])
 AC_INIT([alsa-utils], [1.2.16])
 AC_CONFIG_SRCDIR([aplay/aplay.c])
 AC_PREFIX_DEFAULT(/usr)
@@ -8,7 +8,7 @@
 AM_MAINTAINER_MODE([enable])

 AM_GNU_GETTEXT([external])
-AM_GNU_GETTEXT_VERSION([0.25.1])
+AM_GNU_GETTEXT_VERSION([0.21])

 dnl Checks for programs.

