diff -uNr gl4es/src/CMakeLists.txt gl4es.mod/src/CMakeLists.txt
--- gl4es/src/CMakeLists.txt	2021-06-21 07:53:58.734942738 +0000
+++ gl4es.mod/src/CMakeLists.txt	2021-06-21 08:02:39.158267430 +0000
@@ -224,10 +224,10 @@ else()
     endif()
     install(TARGETS GL
       LIBRARY
-      DESTINATION "/usr/lib/gl4es/"
+      DESTINATION "@TERMUX_PREFIX@/lib/gl4es/"
     )
     install(FILES "../include/gl4esinit.h" "../include/gl4eshint.h"
-      DESTINATION "/usr/include/gl4es/"
+      DESTINATION "@TERMUX_PREFIX@/include/gl4es/"
     )
 endif()

