--- ./common/autoconf/generated-configure.sh	2017-05-12 23:47:56.000000000 +0530
+++ ../generated-configure.sh	2017-05-13 10:06:09.764530254 +0530
@@ -5195,13 +5195,7 @@
 #CUSTOM_AUTOCONF_INCLUDE
 
 # Do not change or remove the following line, it is needed for consistency checks:
-<<<<<<< variant A
-DATE_WHEN_GENERATED=1493303684
->>>>>>> variant B
 DATE_WHEN_GENERATED=1493884285
-####### Ancestor
-DATE_WHEN_GENERATED=1492975963
-======= end
 
 ###############################################################################
 #
