--- ./jdk/src/java.base/unix/native/libnio/fs/UnixNativeDispatcher.c	2017-03-02 21:55:47.000000000 +0530
+++ ../UnixNativeDispatcher.c	2017-03-03 21:37:02.370375158 +0530
@@ -168,40 +168,6 @@
 #endif
 static fdopendir_func* my_fdopendir_func = NULL;
 
-#ifdef __ANDROID__
-/*
- * TODO: Android lacks support for the methods listed below.  In it's place are
- * alternatives that use existing Android functionality, but lack reentrant
- * support.  Determine if the following are the most suitable alternatives.
- *
- */
-int getgrgid_r(gid_t gid, struct group* grp, char* buf, size_t buflen, struct group** result)
-{
-  *result = NULL;
-  errno = 0;
-  grp = getgrgid(gid);
-  if (grp == NULL) {
-        return errno;
-  }
-  // buf not used by caller (see below)
-  *result = grp;
-  return 0;
-}
-
-int getgrnam_r(const char *name, struct group* grp, char* buf, size_t buflen, struct group** result)
-{
-  *result = NULL;
-  errno = 0;
-  grp = getgrnam(name);
-  if (grp == NULL) {
-        return errno;
-  }
-  // buf not used by caller (see below)
-  *result = grp;
-  return 0;
-}
-#endif
-
 /**
  * fstatat missing from glibc on Linux. Temporary workaround
  * for x86/x64.
