shoutout to @its-pointless (live_the_dream)
--- ./hotspot/src/share/vm/runtime/os.cpp	2017-03-30 20:45:45.000000000 +0530
+++ ../os.cpp	2017-03-31 11:04:17.097618021 +0530
@@ -74,7 +74,7 @@
 int               os::_processor_count    = 0;
 int               os::_initial_active_processor_count = 0;
 size_t            os::_page_sizes[os::page_sizes_max];
-const char *      os::_java_home = NULL;
+const char *      os::_java_home = "@TERMUX_PREFIX@/lib/jvm/openjdk-9";
 
 #ifndef PRODUCT
 julong os::num_mallocs = 0;         // # of calls to malloc/realloc
