thanks @fornwall for finding this
--- ./jdk/src/java.base/unix/classes/java/lang/ProcessImpl.java	2017-03-30 20:45:48.000000000 +0530
+++ ../ProcessImpl.java	2017-04-01 23:04:39.565293249 +0530
@@ -157,7 +157,7 @@
         static Platform get() {
             String osName = GetPropertyAction.privilegedGetProperty("os.name");
 
-            if (osName.equals("Linux")) { return LINUX; }
+            if (osName.equals("Linux") || osName.equals("Android")) { return LINUX; }
             if (osName.contains("OS X")) { return BSD; }
             if (osName.equals("SunOS")) { return SOLARIS; }
             if (osName.equals("AIX")) { return AIX; }
