diff --git a/components/enterprise/client_certificates/core/private_key_factory.cc b/components/enterprise/client_certif
icates/core/private_key_factory.cc
index f2365aaf5a..b4282d8cb3 100644
--- a/components/enterprise/client_certificates/core/private_key_factory.cc
+++ b/components/enterprise/client_certificates/core/private_key_factory.cc
@@ -122,9 +122,12 @@ void PrivateKeyFactoryImpl::OnPrivateKeyCreated(
     PrivateKeyCallback callback,
     scoped_refptr<PrivateKey> private_key) {
   if (!private_key && source != PrivateKeySource::kSoftwareKey) {
-    for (auto fallback_source =
-             ++std::find(std::begin(kKeySourcesOrderedBySecurity),
-                         std::end(kKeySourcesOrderedBySecurity), source);
+    auto it = std::find(std::begin(kKeySourcesOrderedBySecurity),
+                        std::end(kKeySourcesOrderedBySecurity), source);
+    if (it == std::end(kKeySourcesOrderedBySecurity)) {
+      return;
+    }
+    for (auto fallback_source = std::next(it);
          fallback_source != std::end(kKeySourcesOrderedBySecurity);
          fallback_source++) {
       auto it = sub_factories_.find(*fallback_source);
