diff -uNr qemu-4.2.0/audio/ossaudio.c qemu-4.2.0.mod/audio/ossaudio.c
--- qemu-4.2.0/audio/ossaudio.c	2019-12-12 20:20:47.000000000 +0200
+++ qemu-4.2.0.mod/audio/ossaudio.c	2019-12-15 00:04:31.280813450 +0200
@@ -24,7 +24,7 @@
 
 #include "qemu/osdep.h"
 #include <sys/ioctl.h>
-#include <sys/soundcard.h>
+#include <linux/soundcard.h>
 #include "qemu/main-loop.h"
 #include "qemu/module.h"
 #include "qemu/host-utils.h"
