diff -uNr qemu-5.2.0/hw/9pfs/9p-local.c qemu-5.2.0.mod/hw/9pfs/9p-local.c
--- qemu-5.2.0/hw/9pfs/9p-local.c	2020-12-08 18:59:44.000000000 +0200
+++ qemu-5.2.0.mod/hw/9pfs/9p-local.c	2021-08-07 17:01:43.567841976 +0300
@@ -299,8 +299,6 @@
 
     map_fd = fileno(fp);
     assert(map_fd != -1);
-    ret = fchmod(map_fd, 0600);
-    assert(ret == 0);
 
     if (credp->fc_uid != -1) {
         uid = credp->fc_uid;
