diff -uNr qemu-6.1.0/meson.build qemu-6.1.0.mod/meson.build
--- qemu-6.1.0/meson.build	2021-08-25 21:20:39.873631512 +0300
+++ qemu-6.1.0.mod/meson.build	2021-08-25 21:21:35.135670419 +0300
@@ -1411,11 +1411,6 @@
   #include <sys/mman.h>
   #include <stddef.h>
   int main(void) { return posix_madvise(NULL, 0, POSIX_MADV_DONTNEED); }'''))
-config_host_data.set('CONFIG_SIGNALFD', cc.links(gnu_source_prefix + '''
-  #include <unistd.h>
-  #include <sys/syscall.h>
-  #include <signal.h>
-  int main(void) { return syscall(SYS_signalfd, -1, NULL, _NSIG / 8); }'''))
 config_host_data.set('CONFIG_SPLICE', cc.links(gnu_source_prefix + '''
   #include <unistd.h>
   #include <fcntl.h>
