diff --git a/daemon/main/Options.cpp b/daemon/main/Options.cpp
index f41bd824..50fa8dd9 100644
--- a/daemon/main/Options.cpp
+++ b/daemon/main/Options.cpp
@@ -182,10 +182,7 @@ const int BoolCount = 12;
 const char* PossibleConfigLocations[] =
 	{
 		"~/.nzbget",
-		"/etc/nzbget.conf",
-		"/usr/etc/nzbget.conf",
-		"/usr/local/etc/nzbget.conf",
-		"/opt/etc/nzbget.conf",
+		"@TERMUX_PREFIX@/etc/nzbget.conf",
 		nullptr
 	};
 #endif
