diff -uNr syncthing/lib/osutil/lowprio_linux.go syncthing.mod/lib/osutil/lowprio_linux.go
--- syncthing/lib/osutil/lowprio_linux.go	2018-03-06 08:19:07.000000000 +0200
+++ syncthing.mod/lib/osutil/lowprio_linux.go	2018-05-02 13:03:12.463566932 +0300
@@ -30,13 +30,8 @@
 )
 
 func ioprioSet(class ioprioClass, value int) error {
-	res, _, err := syscall.Syscall(syscall.SYS_IOPRIO_SET,
-		uintptr(ioprioWhoProcess), 0,
-		uintptr(class)<<ioprioClassShift|uintptr(value))
-	if res == 0 {
-		return nil
-	}
-	return err
+    // Android 8 does not allow syscall SYS_IOPRIO_SET
+    return nil
 }
 
 // SetLowPriority lowers the process CPU scheduling priority, and possibly
