diff -ur orig/src/lincity/MapPoint.hpp mod/src/lincity/MapPoint.hpp
--- orig/src/lincity/MapPoint.hpp	2025-09-23 03:23:26.000000000 +0200
+++ mod/src/lincity/MapPoint.hpp	2025-12-16 16:36:32.010211288 +0100
@@ -24,8 +24,7 @@
 
 #include <cstddef>      // for size_t
 #include <ostream>      // for ostream
-
-namespace std { template<class Key> struct hash; }  // IWYU pragma: keep
+#include <functional>
 
 class MapPoint {
 public:
