diff -ur orig/src/lincity-ng/Config.cpp mod/src/lincity-ng/Config.cpp
--- orig/src/lincity-ng/Config.cpp	2025-09-23 03:23:26.000000000 +0200
+++ mod/src/lincity-ng/Config.cpp	2025-12-16 16:57:49.183708805 +0100
@@ -436,11 +436,6 @@
 }
 
 
-template class Config::Option<int>;
-template class Config::Option<bool>;
-template class Config::Option<std::string>;
-template class Config::Option<std::filesystem::path>;
-
 template<typename T>
 Config::Option<T>::Option() :
   default_(std::nullopt)
@@ -595,4 +590,9 @@
       (xmlStr)"default");
 }
 
+template class Config::Option<int>;
+template class Config::Option<bool>;
+template class Config::Option<std::string>;
+template class Config::Option<std::filesystem::path>;
+
 /** @file lincity-ng/Config.cpp */
