diff --git a/CMakeLists.txt b/CMakeLists.txt
index 036890e..a7970b1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -197,7 +197,6 @@ ecm_qt_install_logging_categories(
     DESTINATION ${KDE_INSTALL_LOGGINGCATEGORIESDIR}
 )
 
-update_xdg_mimetypes(${KDE_INSTALL_MIMEDIR})
 ki18n_install(po)
 if(KF6DocTools_FOUND)
     kdoctools_install(po)
diff --git a/src/data/CMakeLists.txt b/src/data/CMakeLists.txt
index c76f3eb..cbe4965 100644
--- a/src/data/CMakeLists.txt
+++ b/src/data/CMakeLists.txt
@@ -4,5 +4,4 @@ if(UNIX AND NOT APPLE)
     install(PROGRAMS org.kde.drawy.desktop DESTINATION ${KDE_INSTALL_APPDIR})
     install(FILES org.kde.drawy.metainfo.xml DESTINATION ${KDE_INSTALL_METAINFODIR})
     install(FILES application-x-drawy.xml DESTINATION ${KDE_INSTALL_MIMEDIR})
-    update_xdg_mimetypes(${KDE_INSTALL_MIMEDIR})
 endif()
