--- a/src/cfte.cpp
+++ b/src/cfte.cpp
@@ -1887,7 +1887,7 @@
             //fprintf(stderr, "Looking for %s\n", Cfg);
             if (!FileExists(Cfg)) {
                 // 3. /usr/share/fte
-                sprintf(tmp, "/usr/share/fte/%s", CfgName);
+                sprintf(tmp, "@TERMUX_PREFIX@/share/fte/config/%s", CfgName);
                 ExpandPath(tmp, Cfg, sizeof(Cfg));
                 if (!FileExists(Cfg)) {
                     sprintf(tmp, "%sconfig/%s", StartDir, CfgName);
