diff -uNr libpciaccess-0.16/src/linux_sysfs.c libpciaccess-0.16.mod/src/linux_sysfs.c
--- libpciaccess-0.16/src/linux_sysfs.c	2019-07-17 19:25:27.000000000 +0300
+++ libpciaccess-0.16.mod/src/linux_sysfs.c	2019-07-23 00:31:23.703286583 +0300
@@ -462,7 +462,7 @@


     while ( temp_size > 0 ) {
-	const ssize_t bytes = pread64( fd, data_bytes, temp_size, offset );
+	const ssize_t bytes = pread( fd, data_bytes, temp_size, offset );

 	/* If zero bytes were read, then we assume it's the end of the
 	 * config file.
@@ -522,7 +522,7 @@


     while ( temp_size > 0 ) {
-	const ssize_t bytes = pwrite64( fd, data_bytes, temp_size, offset );
+	const ssize_t bytes = pwrite( fd, data_bytes, temp_size, offset );

 	/* If zero bytes were written, then we assume it's the end of the
 	 * config file.
