diff --git a/CMakeLists.txt b/CMakeLists.txt
index f6de091a0c..d21e04a24d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -301,7 +301,7 @@ if(UNIX)
   find_package(Threads)
   set(_gRPC_ALLTARGETS_LIBRARIES ${CMAKE_DL_LIBS} m Threads::Threads)
   if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_POSIX)
-    set(_gRPC_ALLTARGETS_LIBRARIES ${_gRPC_ALLTARGETS_LIBRARIES} rt)
+    set(_gRPC_ALLTARGETS_LIBRARIES ${_gRPC_ALLTARGETS_LIBRARIES})
   endif()
 endif()
 
@@ -1432,12 +1432,6 @@ target_link_libraries(gpr
   absl::strings
   absl::optional
 )
-if(_gRPC_PLATFORM_ANDROID)
-  target_link_libraries(gpr
-    android
-    log
-  )
-endif()
 
 foreach(_hdr
   include/grpc/support/alloc.h
