diff -u -r ../cache/checkout-1.9.1/src/core/lib/gpr/log_android.cc ./src/core/lib/gpr/log_android.cc
--- ../cache/checkout-1.9.1/src/core/lib/gpr/log_android.cc	2018-02-12 23:15:11.039292462 +0000
+++ ./src/core/lib/gpr/log_android.cc	2018-02-12 23:28:01.274274589 +0000
@@ -63,7 +63,7 @@
 
   asprintf(&output, "%s:%d] %s", display_file, args->line, args->message);
 
-  __android_log_write(severity_to_log_priority(args->severity), "GRPC", output);
+  // __android_log_write(severity_to_log_priority(args->severity), "GRPC", output);
 
   /* allocated by asprintf => use free, not gpr_free */
   free(output);
