diff -u -r ../cache/checkout-1.12.0/src/core/lib/gpr/wrap_memcpy.cc ./src/core/lib/gpr/wrap_memcpy.cc
--- ../cache/checkout-1.12.0/src/core/lib/gpr/wrap_memcpy.cc	2018-05-22 13:07:29.307092222 +0000
+++ ./src/core/lib/gpr/wrap_memcpy.cc	2018-05-23 22:21:40.602239126 +0000
@@ -28,7 +28,7 @@
 
 extern "C" {
 #ifdef __linux__
-#if defined(__x86_64__) && !defined(GPR_MUSL_LIBC_COMPAT)
+#if defined(__x86_64__) && !defined(GPR_MUSL_LIBC_COMPAT) && !defined(__ANDROID__)
 __asm__(".symver memcpy,memcpy@GLIBC_2.2.5");
 void* __wrap_memcpy(void* destination, const void* source, size_t num) {
   return memcpy(destination, source, num);
