diff --git a/meson.build b/meson.build
index c1b9ce4..dc8454c 100644
--- a/meson.build
+++ b/meson.build
@@ -19,7 +19,7 @@ datadir = join_paths(prefix, get_option('datadir'))
 privexecdir = join_paths(prefix, get_option('libexecdir'), meson.project_name())
 c_comp = meson.get_compiler('c')
 
-meson.add_install_script('postinstall.py')
+#meson.add_install_script('postinstall.py')
 
 # Dependencies
 
@@ -119,7 +119,7 @@ c_args += c_comp.get_supported_arguments(test_args)
 # Subdirectories
 
 subdir('data')
-subdir('help')
+#subdir('help')
 if build_nautilus_actions
   subdir('nautilus')
 endif
