diff --git a/Cargo.toml b/Cargo.toml
index 83da4c1..01f8708 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -29,8 +29,7 @@ The minimal, blazing-fast, and infinitely customizable prompt for any shell! ☄
 """
 
 [features]
-default = ["battery", "notify", "gix-max-perf"]
-battery = ["starship-battery"]
+default = ["notify", "gix-max-perf"]
 config-schema = ["schemars"]
 notify = ["notify-rust"]
 
@@ -73,9 +72,6 @@ serde = { version = "1.0.210", features = ["derive"] }
 serde_json = "1.0.129"
 sha1 = "0.10.6"
 shadow-rs = { version = "0.35.1", default-features = false }
-# battery is optional (on by default) because the crate doesn't currently build for Termux
-# see: https://github.com/svartalf/rust-battery/issues/33
-starship-battery = { version = "0.10.0", optional = true }
 strsim = "0.11.1"
 systemstat = "=0.2.3"
 terminal_size = "0.4.0"
