diff --git a/Cargo.toml b/Cargo.toml
index 4728f41d..429c3ed8 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -29,8 +29,7 @@ The minimal, blazing-fast, and infinitely customizable prompt for any shell! ☄
 """
 
 [features]
-default = ["battery", "notify", "gix-max-perf"]
-battery = ["starship-battery"]
+default = ["notify", "gix-max-perf"]
 config-schema = ["schemars"]
 notify = ["notify-rust"]
 
@@ -74,9 +73,6 @@ serde = { version = "1.0.204", features = ["derive"] }
 serde_json = "1.0.120"
 sha1 = "0.10.6"
 shadow-rs = { version = "0.30.0", default-features = false }
-# battery is optional (on by default) because the crate doesn't currently build for Termux
-# see: https://github.com/svartalf/rust-battery/issues/33
-starship-battery = { version = "0.9.1", optional = true }
 strsim = "0.11.1"
 systemstat = "=0.2.3"
 terminal_size = "0.3.0"
