--- a/Cargo.toml
+++ b/Cargo.toml
@@ -72,6 +72,9 @@
 serde_json = "1.0.108"
 sha1 = "0.10.6"
 shadow-rs = { version = "0.25.0", default-features = false }
+# battery is optional (on by default) because the crate doesn't currently build for Termux
+# see: https://github.com/svartalf/rust-battery/issues/33
+starship-battery = { version = "0.8.2", optional = true }
 strsim = "0.10.0"
 systemstat = "=0.2.3"
 terminal_size = "0.3.0"
