--- a/src/lib.rs
+++ b/src/lib.rs
@@ -440,7 +440,7 @@
 
         // Some decisions later on are made if CMAKE_TOOLCHAIN_FILE is defined,
         // so we need to read it from the environment variables from the beginning.
-        if !self.defined("CMAKE_TOOLCHAIN_FILE") {
+        if false {
             if let Some(s) = self.getenv_target_os("CMAKE_TOOLCHAIN_FILE") {
                 self.define("CMAKE_TOOLCHAIN_FILE", s);
             } else {
