diff -u -r ../starship-0.48.0/Cargo.toml ./Cargo.toml
--- ../starship-0.48.0/Cargo.toml	2021-01-06 15:03:16.000000000 +0800
+++ ./Cargo.toml	2021-01-06 15:04:13.000000000 +0800
@@ -70,6 +70,9 @@
 native-tls = { version = "0.2.7", optional = true }
 shell-words = "1.0.0"
 
+[patch.crates-io]
+sys-info = { git = 'https://github.com/FillZpp/sys-info-rs' }
+
 [target.'cfg(windows)'.dependencies]
 winapi = { version = "0.3.9", features = [
   "winuser",
