diff --git a/vendor/systemstat/src/platform/mod.rs b/vendor/systemstat/src/platform/mod.rs
index 1ebc9fec..8bebe135 100644
--- a/vendor/systemstat/src/platform/mod.rs
+++ b/vendor/systemstat/src/platform/mod.rs
@@ -10,7 +10,7 @@ pub use self::windows::PlatformImpl;
 #[cfg(unix)]
 pub mod unix;
 
-#[cfg(any(target_os = "linux", target_os = "hurd"))]
+#[cfg(any(target_os = "android", target_os = "linux", target_os = "hurd"))]
 mod procfs;
 
 #[cfg(any(
