diff -uNr gopass-1.8.4/pkg/clipboard/clipboard.go gopass-1.8.4.mod/pkg/clipboard/clipboard.go
--- gopass-1.8.4/pkg/clipboard/clipboard.go	2018-12-26 19:16:38.000000000 +0200
+++ gopass-1.8.4.mod/pkg/clipboard/clipboard.go	2019-01-30 14:46:13.327735051 +0200
@@ -14,7 +14,7 @@
 
 var (
 	// ErrNotSupported is returned when the clipboard is not accessible
-	ErrNotSupported = fmt.Errorf("WARNING: No clipboard available. Install xsel or xclip or use -p to print to console")
+	ErrNotSupported = fmt.Errorf("WARNING: No clipboard available. Install termux-api or use -p to print to console")
 )
 
 // CopyTo copies the given data to the clipboard and enqueues automatic
