diff -uNr haproxy-1.9.5/include/common/config.h haproxy-1.9.5.mod/include/common/config.h
--- haproxy-1.9.5/include/common/config.h	2019-03-19 15:40:44.000000000 +0200
+++ haproxy-1.9.5.mod/include/common/config.h	2019-03-27 14:36:33.223563480 +0200
@@ -44,8 +44,6 @@
 #if defined(__x86_64__) || defined (__aarch64__)
 #define HA_HAVE_CAS_DW
 #define HA_CAS_IS_8B
-#elif defined(__arm__) && (defined(__ARM_ARCH_7__) || defined(__ARM_ARCH_7A__))
-#define HA_HAVE_CAS_DW
 #endif
 
