diff -uNr srelay-0.4.8p1/configure srelay-0.4.8p1.mod/configure
--- srelay-0.4.8p1/configure	2017-12-25 05:10:37.000000000 +0200
+++ srelay-0.4.8p1.mod/configure	2018-06-26 00:55:42.529288733 +0300
@@ -3998,41 +3998,6 @@
 	elif test "$OS" = "SOLARIS"; then
 	    LDFLAGS="$LDFLAGS -lpthread"
 	fi
-	if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
-else
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-#include <pthread.h>
-void
-init_routine()
-{
-    return;
-}
-main()
-{
-    pthread_once_t once_control = PTHREAD_ONCE_INIT;
-    pthread_once(&once_control, &init_routine);
-    return(0);
-}
-_ACEOF
-if ac_fn_c_try_run "$LINENO"; then :
-  $as_echo "#define USE_THREAD 1" >>confdefs.h
- { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-$as_echo "yes" >&6; }
-else
-  CFLAGS="$save_CFLAGS"; LDFLAGS="$save_LDFLAGS"; { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-
-fi
-rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
-  conftest.$ac_objext conftest.beam conftest.$ac_ext
-fi
-
     fi
 fi
 
