diff -uNr srelay-0.4.8p1/srelay.h srelay-0.4.8p1.mod/srelay.h
--- srelay-0.4.8p1/srelay.h	2017-12-25 05:10:37.000000000 +0200
+++ srelay-0.4.8p1.mod/srelay.h	2018-06-26 00:55:42.545955400 +0300
@@ -106,13 +106,13 @@
 #define version  "srelay 0.4.8p1 2017/12/25 (Tomo.M)"
 
 #ifndef SYSCONFDIR
-# define SYSCONFDIR "/usr/local/etc"
+# define SYSCONFDIR "@TERMUX_PREFIX@/etc"
 #endif
 #define CONFIG    SYSCONFDIR "/srelay.conf"
 #define PWDFILE   SYSCONFDIR "/srelay.passwd"
-#define PIDFILE   "/var/run/srelay.pid"
-#define WORKDIR0  "/var/run"
-#define WORKDIR1  "/var/tmp"
+#define PIDFILE   "@TERMUX_PREFIX@/var/run/srelay.pid"
+#define WORKDIR0  "@TERMUX_PREFIX@/var/run"
+#define WORKDIR1  "@TERMUX_PREFIX@/var/tmp"
 
 #define S4DEFUSR  "user"
 
