--- a/utils/moontracker/moontracker.c
+++ b/utils/moontracker/moontracker.c
@@ -23,11 +23,12 @@
 #include <stdio.h>
 #include <math.h>
 #include <string.h>
+#include <strings.h>
 #include <fcntl.h>
 #include <termios.h>
 #include <signal.h>
 
-char	callsign[20], output[20];
+char	callsign[20], output[21];
 int	qthalt, pid;
 double	qthlat, qthlong, moon_az, moon_el, moonrise,
 	PI=3.141592653589793, TWOPI=6.28318530717958623,
