diff -uNr zlib-1.3/configure zlib-1.3.mod/configure
--- zlib-1.3/configure	2023-08-18 11:45:36.000000000 +0300
+++ zlib-1.3.mod/configure	2023-11-29 01:51:17.924387123 +0200
@@ -438,7 +438,7 @@
 extern int getchar();
 int hello() {return getchar();}
 EOF
-if test $shared -eq 1; then
+if false; then
   echo Checking for shared library support... | tee -a configure.log
   # we must test in two steps (cc then ld), required at least on SunOS 4.x
   if try $CC -w -c $SFLAGS $test.c &&
