diff -uNr xorg-server-1.20.1/hw/xfree86/parser/write.c xorg-server-1.20.1.mod/hw/xfree86/parser/write.c
--- xorg-server-1.20.1/hw/xfree86/parser/write.c	2018-08-07 19:31:03.000000000 +0300
+++ xorg-server-1.20.1.mod/hw/xfree86/parser/write.c	2018-10-08 21:07:16.817587230 +0300
@@ -129,7 +129,7 @@
 int
 xf86writeConfigFile(const char *filename, XF86ConfigPtr cptr)
 {
-#ifndef HAS_NO_UIDS
+#if !defined(HAS_NO_UIDS) && !defined(__ANDROID__)
     int ret;
 
     if (getuid() != geteuid()) {
