diff -uNr xorg-server-1.20.1/hw/xfree86/os-support/linux/lnx_acpi.c xorg-server-1.20.1.mod/hw/xfree86/os-support/linux/lnx_acpi.c
--- xorg-server-1.20.1/hw/xfree86/os-support/linux/lnx_acpi.c	2018-08-07 19:31:03.000000000 +0300
+++ xorg-server-1.20.1.mod/hw/xfree86/os-support/linux/lnx_acpi.c	2018-10-08 21:07:16.807582583 +0300
@@ -15,7 +15,7 @@
 #include <fcntl.h>
 #include <errno.h>
 
-#define ACPI_SOCKET  "/var/run/acpid.socket"
+#define ACPI_SOCKET  "/data/data/com.termux/files/usr/var/run/acpid.socket"
 
 #define ACPI_VIDEO_NOTIFY_SWITCH	0x80
 #define ACPI_VIDEO_NOTIFY_PROBE		0x81
