diff -uNr xorg-server-1.20.1/os/backtrace.c xorg-server-1.20.1.mod/os/backtrace.c
--- xorg-server-1.20.1/os/backtrace.c	2018-08-07 19:31:05.000000000 +0300
+++ xorg-server-1.20.1.mod/os/backtrace.c	2018-10-08 21:07:16.834261640 +0300
@@ -244,14 +244,13 @@
         /* CHILD */
         char parent[16];
 
-        seteuid(0);
         close(STDIN_FILENO);
         close(STDOUT_FILENO);
         dup2(pipefd[1], STDOUT_FILENO);
         closefrom(STDERR_FILENO);
 
         snprintf(parent, sizeof(parent), "%d", getppid());
-        execle("/usr/bin/pstack", "pstack", parent, NULL);
+        execle("/data/data/com.termux/files/usr/bin/pstack", "pstack", parent, NULL);
         exit(1);
     }
     else {
