diff -uNr xorg-server-1.20.1/hw/xfree86/parser/scan.c xorg-server-1.20.1.mod/hw/xfree86/parser/scan.c
--- xorg-server-1.20.1/hw/xfree86/parser/scan.c	2018-08-07 19:31:03.000000000 +0300
+++ xorg-server-1.20.1.mod/hw/xfree86/parser/scan.c	2018-10-08 21:07:16.814252348 +0300
@@ -873,13 +873,13 @@
  */
 
 #ifndef DEFAULT_CONF_PATH
-#define DEFAULT_CONF_PATH	"/etc/X11/%S," \
+#define DEFAULT_CONF_PATH	"@TERMUX_PREFIX@/etc/X11/%S," \
 							"%P/etc/X11/%S," \
-							"/etc/X11/%G," \
+							"@TERMUX_PREFIX@/etc/X11/%G," \
 							"%P/etc/X11/%G," \
-							"/etc/X11/%X-%M," \
-							"/etc/X11/%X," \
-							"/etc/%X," \
+							"@TERMUX_PREFIX@/etc/X11/%X-%M," \
+							"@TERMUX_PREFIX@/etc/X11/%X," \
+							"@TERMUX_PREFIX@/etc/%X," \
 							"%P/etc/X11/%X.%H," \
 							"%P/etc/X11/%X-%M," \
 							"%P/etc/X11/%X," \
