diff -uNr xorg-server-1.20.1/hw/xquartz/mach-startup/bundle-main.c xorg-server-1.20.1.mod/hw/xquartz/mach-startup/bundle-main.c
--- xorg-server-1.20.1/hw/xquartz/mach-startup/bundle-main.c	2018-08-07 19:31:04.000000000 +0300
+++ xorg-server-1.20.1.mod/hw/xquartz/mach-startup/bundle-main.c	2018-10-08 21:07:16.820922112 +0300
@@ -79,7 +79,7 @@
 
 #define DEFAULT_CLIENT X11BINDIR "/xterm"
 #define DEFAULT_STARTX X11BINDIR "/startx -- " X11BINDIR "/Xquartz"
-#define DEFAULT_SHELL  "/bin/sh"
+#define DEFAULT_SHELL  "@TERMUX_PREFIX@/bin/sh"
 
 #ifndef BUILD_DATE
 #define BUILD_DATE ""
@@ -534,7 +534,7 @@
     temp = getenv("PATH");
     if (temp == NULL || temp[0] == 0) {
         snprintf(buf, sizeof(buf),
-                 "/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:%s",
+                 "@TERMUX_PREFIX@/bin:@TERMUX_PREFIX@/bin/applets:%s",
                  dir);
         setenv("PATH", buf, TRUE);
     }
