diff -uNr xorg-server-1.20.1/os/connection.c xorg-server-1.20.1.mod/os/connection.c
--- xorg-server-1.20.1/os/connection.c	2018-08-07 19:31:05.000000000 +0300
+++ xorg-server-1.20.1.mod/os/connection.c	2018-10-08 21:07:16.837596523 +0300
@@ -996,7 +996,7 @@
     XtransConnInfo ciptr;
     const char *display_env = getenv("DISPLAY");
 
-    if (display_env && (strncmp(display_env, "/tmp/launch", 11) == 0)) {
+    if (display_env && (strncmp(display_env, "/data/data/com.termux/files/usr/tmp/launch", 11) == 0)) {
         /* Make the path the launchd socket if our DISPLAY is set right */
         strcpy(port, display_env);
     }
