diff -uNr xorg-server-1.20.1/include/site.h xorg-server-1.20.1.mod/include/site.h
--- xorg-server-1.20.1/include/site.h	2018-08-07 19:31:05.000000000 +0300
+++ xorg-server-1.20.1.mod/include/site.h	2018-10-08 21:07:16.824256995 +0300
@@ -72,7 +72,7 @@
  * DO NOT CHANGE THESE VALUES OR THE DIX IMAKEFILE!
  */
 #ifndef COMPILEDDEFAULTFONTPATH
-#define COMPILEDDEFAULTFONTPATH	"/usr/share/fonts/X11/misc/"
+#define COMPILEDDEFAULTFONTPATH	"/data/data/com.termux/files/usr/share/fonts/X11/misc/"
 #endif
 
 /*
