--- a/menus.c
+++ b/menus.c
@@ -1252,8 +1252,8 @@
 static void
 AutoInstallProtocol (char *settingsFile, char *protocol)
 {   // install new engines for given protocol (both from package and source)
-    InstallFromDir("/usr/local/share/games/plugins", protocol, settingsFile);
-    InstallFromDir("/usr/share/games/plugins", protocol, settingsFile);
+    InstallFromDir("@TERMUX_PREFIX@/local/share/games/plugins", protocol, settingsFile);
+    InstallFromDir("@TERMUX_PREFIX@/share/games/plugins", protocol, settingsFile);
 }
 
 void
