--- a/electron/shell/browser/api/electron_api_service_worker_main.cc
+++ b/electron/shell/browser/api/electron_api_service_worker_main.cc
@@ -347,7 +347,7 @@

 }  // namespace electron::api
 
-namespace {
+namespace a_electron_api_service_worker_main {
 
 using electron::api::ServiceWorkerMain;
 
@@ -364,4 +364,6 @@
 }  // namespace
 
+#define _module _module_electron_api_service_worker_main
 NODE_LINKED_BINDING_CONTEXT_AWARE(electron_browser_service_worker_main,
-                                  Initialize)
+                                  a_electron_api_service_worker_main::Initialize)
+#undef _module
--- a/electron/shell/browser/api/electron_api_msix_updater.cc
+++ b/electron/shell/browser/api/electron_api_msix_updater.cc
@@ -41,7 +41,7 @@
 
 }  // namespace electron
 
-namespace {
+namespace a_electron_api_msix_updater {
 
 #if BUILDFLAG(IS_WIN)
 
@@ -906,4 +906,6 @@
 
 }  // namespace
 
+#define _module _module_electron_browser_msix_updater
-NODE_LINKED_BINDING_CONTEXT_AWARE(electron_browser_msix_updater, Initialize)
+NODE_LINKED_BINDING_CONTEXT_AWARE(electron_browser_msix_updater, a_electron_api_msix_updater::Initialize)
+#undef _module
