--- a/third_party/pdfium/BUILD.gn
+++ b/third_party/pdfium/BUILD.gn
@@ -219,7 +219,23 @@
   ]
 }
 
-component("pdfium") {
+group("pdfium") {
+  public_deps = [ ":pdfium_public_headers" ]
+  public_configs = [
+    ":pdfium_public_config",
+    ":pdfium_prebuilt",
+  ]
+}
+
+config("pdfium_prebuilt") {
+  lib_dirs = [ ]
+  libs = [ "@TERMUX_PREFIX@/opt/electron-host-tools-for-code-oss/obj/third_party/pdfium/libpdfium.a" ]
+
+  configs = [ ":pdfium_public_config" ]
+  visibility = [ ":pdfium" ]
+}
+
+component("pdfium_1") {
   output_name = "pdfium"
   libs = []
   configs += [ ":pdfium_strict_config" ]
