--- a/electron/js2c_toolchain.gni
+++ b/electron/js2c_toolchain.gni
@@ -4,6 +4,8 @@
 
 declare_args() {
   electron_js2c_toolchain = ""
+
+  prebuilt_js2c_binary = ""
 }
 
 if (electron_js2c_toolchain == "") {
--- a/electron/BUILD.gn
+++ b/electron/BUILD.gn
@@ -231,8 +231,13 @@
     ":electron_sandboxed_renderer_bundle",
     ":electron_utility_bundle",
     ":electron_worker_bundle",
+  ]
+
+if (prebuilt_js2c_binary == "") {
+  deps += [
     "//third_party/electron_node:node_js2c($electron_js2c_toolchain)",
   ]
+}
 
   sources = [
     "$target_gen_dir/js2c/browser_init.js",
@@ -248,10 +253,14 @@
   outputs = [ "$root_gen_dir/electron_natives.cc" ]
 
   script = "build/js2c.py"
+  node_js2c_binary = prebuilt_js2c_binary
+if (node_js2c_binary == "") {
   out_dir =
       get_label_info(":anything($electron_js2c_toolchain)", "root_out_dir")
+  node_js2c_binary = rebase_path("$out_dir/node_js2c")
+}
   args = [
-           rebase_path("$out_dir/node_js2c"),
+           node_js2c_binary,
            rebase_path("$root_gen_dir"),
          ] + rebase_path(outputs, root_gen_dir) +
          rebase_path(sources, root_gen_dir)
--- a/third_party/electron_node/BUILD.gn
+++ b/third_party/electron_node/BUILD.gn
@@ -101,8 +101,12 @@
     "//electron:generate_config_gypi",
     ":node_js2c_original_fs",
     ":node_js2c_inputs",
+  ]
+if (prebuilt_js2c_binary == "") {
+  deps += [
     ":node_js2c($electron_js2c_toolchain)"
   ]
+}
   config_gypi = [ "$root_gen_dir/config.gypi" ]
   inputs = library_files + get_target_outputs(":node_js2c_original_fs") + config_gypi
   outputs = [
@@ -110,8 +114,12 @@
   ]
 
   script = "//electron/build/run-in-dir.py"
+  node_js2c_binary = prebuilt_js2c_binary
+if (prebuilt_js2c_binary == "") {
   out_dir = get_label_info(":anything($electron_js2c_toolchain)", "root_out_dir")
-  args = [ rebase_path("$target_gen_dir/js2c_inputs"), rebase_path("$out_dir/node_js2c") ] +
+  node_js2c_binary = rebase_path("$out_dir/node_js2c")
+}
+  args = [ rebase_path("$target_gen_dir/js2c_inputs"), node_js2c_binary ] +
          rebase_path(outputs) + library_files + fs_files + original_fs_files + rebase_path(config_gypi)
 }
 
@@ -226,7 +234,7 @@
   }
 }
 
-if (current_toolchain == electron_js2c_toolchain) {
+if (current_toolchain == electron_js2c_toolchain && prebuilt_js2c_binary == "") {
   executable("node_js2c") {
     defines = []
     sources = [
