diff -ur src-o/configure src/configure
--- src-o/configure	2017-07-31 04:44:24.215397674 +0200
+++ src/configure	2017-07-31 04:45:19.667222873 +0200
@@ -12719,7 +12719,7 @@
 fi
 
 
-THREAD_LIBS=-lpthread
+THREAD_LIBS=
 
 
 
diff -ur src-o/configure.ac src/configure.ac
--- src-o/configure.ac	2017-07-31 04:44:24.215397674 +0200
+++ src/configure.ac	2017-07-31 04:44:40.903346135 +0200
@@ -113,7 +113,7 @@
         ,
         [#include <unistd.h>])
 
-THREAD_LIBS=-lpthread
+THREAD_LIBS=
 AC_SUBST(THREAD_LIBS)
 
