diff -uNr libcap-2.26/libcap/Makefile libcap-2.26.mod/libcap/Makefile
--- libcap-2.26/libcap/Makefile	2018-09-11 06:02:10.000000000 +0300
+++ libcap-2.26.mod/libcap/Makefile	2019-03-18 21:53:27.413691421 +0200
@@ -35,7 +35,7 @@
 		$< >$@
 
 _makenames: _makenames.c cap_names.list.h
-	$(BUILD_CC) $(BUILD_CFLAGS) $< -o $@
+	gcc $(BUILD_CFLAGS) $< -o $@
 
 cap_names.h: _makenames
 	./_makenames > cap_names.h
