diff -u -r ../libcap-2.28/Make.Rules ./Make.Rules
--- ../libcap-2.28/Make.Rules	2019-12-07 20:10:11.000000000 +0000
+++ ./Make.Rules	2019-12-11 23:41:49.847418000 +0000
@@ -62,7 +62,7 @@
 LD=$(CC) -Wl,-x -shared
 LDFLAGS := #-g
 LIBCAPLIB := -L$(topdir)/libcap -lcap
-LIBPSXLIB := -L$(topdir)/libcap -lpsx -lpthread
+LIBPSXLIB := -L$(topdir)/libcap -lpsx
 
 BUILD_GPERF := $(shell which gperf >/dev/null 2>/dev/null && echo yes)
 
