diff -u -r ../libcap-2.34/progs/capsh.c ./progs/capsh.c
--- ../libcap-2.34/progs/capsh.c	2020-05-02 22:47:28.000000000 +0000
+++ ./progs/capsh.c	2020-05-11 08:14:15.943279000 +0000
@@ -26,7 +26,7 @@
 #include <unistd.h>
 
 #ifndef SHELL
-#define SHELL "/bin/bash"
+#define SHELL "@TERMUX_PREFIX@/bin/bash"
 #endif /* ndef SHELL */
 
 #define MAX_GROUPS       100   /* max number of supplementary groups for user */
