diff -u -r ../cvs-1.11.23/lib/getline.h ./lib/getline.h
--- ../cvs-1.11.23/lib/getline.h	2005-04-04 16:46:05.000000000 -0400
+++ ./lib/getline.h	2016-03-29 17:31:15.020232824 -0400
@@ -12,8 +12,6 @@
 #define GETLINE_NO_LIMIT -1
 
 int
-  getline __PROTO ((char **_lineptr, size_t *_n, FILE *_stream));
-int
   getline_safe __PROTO ((char **_lineptr, size_t *_n, FILE *_stream,
                          int limit));
 int
