diff -u -r ../cvs-1.11.23/lib/getline.c ./lib/getline.c
--- ../cvs-1.11.23/lib/getline.c	2005-04-04 16:46:05.000000000 -0400
+++ ./lib/getline.c	2016-03-29 17:34:23.536876927 -0400
@@ -155,15 +155,6 @@
 }
 
 int
-getline (lineptr, n, stream)
-     char **lineptr;
-     size_t *n;
-     FILE *stream;
-{
-  return getstr (lineptr, n, stream, '\n', 0, GETLINE_NO_LIMIT);
-}
-
-int
 getline_safe (lineptr, n, stream, limit)
      char **lineptr;
      size_t *n;
