diff -uNr moreutils-0.64/is_utf8/Makefile moreutils-0.64.mod/is_utf8/Makefile
--- moreutils-0.64/is_utf8/Makefile	2020-08-29 20:59:47.000000000 +0300
+++ moreutils-0.64.mod/is_utf8/Makefile	2020-11-30 02:59:43.379944170 +0200
@@ -37,7 +37,7 @@
 SONAME = $(LINKERNAME).$(VERSION)
 REALNAME = $(SONAME).$(MINOR).$(RELEASE)
 
-CC = gcc
+CC ?= gcc
 CFLAGS ?= -O3 -Wextra -Wall -ansi -Wstrict-prototypes
 
 $(NAME): $(OBJ)
