diff -uNr moreutils-0.63/Makefile moreutils-0.63.mod/Makefile
--- moreutils-0.63/Makefile	2019-01-09 17:14:34.000000000 +0200
+++ moreutils-0.63.mod/Makefile	2020-01-03 20:48:12.342120852 +0200
@@ -1,8 +1,8 @@
-BINS=isutf8 ifdata ifne pee sponge mispipe lckdo parallel errno
+BINS=isutf8 ifdata ifne pee sponge mispipe lckdo errno
 PERLSCRIPTS=vidir vipe ts combine zrun chronic
 MANS=sponge.1 vidir.1 vipe.1 isutf8.1 ts.1 combine.1 ifdata.1 ifne.1 pee.1 zrun.1 chronic.1 mispipe.1 lckdo.1 parallel.1 errno.1
 CFLAGS?=-O2 -g -Wall
-INSTALL_BIN?=install -s
+INSTALL_BIN?=install
 PREFIX?=/usr
 
 ifneq (,$(findstring CYGWIN,$(shell uname)))
