diff --git a/Makefile b/Makefile
index 8541b8c..78432f2 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
-LUA ?= lua
-LUA_VERSION ?= $(shell $(LUA) -e 'v=_VERSION:gsub("^Lua *","");print(v)')
+LUA_VERSION ?= 5.3
+LUA ?= lua$(LUA_VERSION)
 DESTDIR ?=
 PREFIX ?= /usr/local
 BIN_DIR ?= $(PREFIX)/bin
@@ -18,7 +18,7 @@ SRC=$(LIB_SRC) src/launcher.fnl src/fennel/binary.fnl
 
 MAN_PANDOC := pandoc -f gfm -t man -s --lua-filter=build/manfilter.lua \
 	     --metadata author="Fennel Maintainers" \
-	     --variable footer="fennel $(shell ./fennel -e '(. (require :fennel) :version)')"
+	     --variable footer="fennel"
 
 unexport NO_COLOR # this causes test failures
 unexport FENNEL_PATH FENNEL_MACRO_PATH # ensure isolation
@@ -74,7 +74,7 @@ check:
 
 ## Binaries
 
-BIN_LUA_VERSION ?= 5.4.6
+BIN_LUA_VERSION ?= 5.3.6
 BIN_LUAJIT_VERSION ?= 2.0.5
 BIN_LUA_DIR ?= $(PWD)/lua-$(BIN_LUA_VERSION)
 BIN_LUAJIT_DIR ?= $(PWD)/LuaJIT-$(BIN_LUAJIT_VERSION)
