diff -uNr amber-0.5.9/Cargo.toml amber-0.5.9.mod/Cargo.toml
--- amber-0.5.9/Cargo.toml	2021-04-06 09:31:20.000000000 +0800
+++ amber-0.5.9.mod/Cargo.toml	2022-07-18 13:40:51.902186988 +0800
@@ -32,7 +32,7 @@
 ctrlc             = "3"
 dirs              = "3"
 filetime          = "0.2"
-getch             = "0.2"
+getch             = "0.3.0"
 glob              = "0.3"
 lazy_static       = "1"
 memmap            = "0.7"
@@ -52,7 +52,7 @@
 unicode-width     = "0.1"
 
 [target.'cfg(not(target_os = "windows"))'.dependencies]
-termios           = "0.3"
+termios           = "0.3.3"
 
 [profile.dev]
 opt-level        = 0
