diff -u -r ../alpine-2.21/configure ./configure
--- ../alpine-2.21/configure	2017-02-06 00:06:22.491217965 +0000
+++ ./configure	2017-06-14 23:28:28.132302332 +0000
@@ -17646,6 +17646,14 @@
 
     fi
     ;;
+  *-*-*-android*)
+    alpine_SSLDIR="@TERMUX_PREFIX@/etc/tls"
+    alpine_SSLCERTS="$alpine_SSLDIR/certs"
+    alpine_SSLKEYS="$alpine_SSLDIR/private"
+    alpine_SSLINCLUDE="@TERMUX_PREFIX@/include/openssl"
+    alpine_SSLLIB="@TERMUX_PREFIX@/lib"
+    alpine_path_delim="/"
+	 ;;
   *-apple-darwin*)
     systype="OSX"
     alpine_path_delim="/"
