--- ./configure.orig	2021-09-18 20:41:42.000000000 +0530
+++ ./configure	2021-10-05 14:01:19.899930440 +0530
@@ -981,7 +981,7 @@
 sharedstatedir='${prefix}/com'
 localstatedir='${prefix}/var'
 includedir='${prefix}/include'
-oldincludedir='/usr/include'
+oldincludedir='${prefix}/usr/include'
 docdir='${datarootdir}/doc/${PACKAGE_TARNAME}'
 infodir='${datarootdir}/info'
 htmldir='${docdir}'
@@ -17745,6 +17745,14 @@
       fi
     fi
     ;;
+  *-*-*-android*)
+    alpine_SSLDIR="/data/data/com.termux/files/usr/etc/tls"
+    alpine_SSLCERTS="$alpine_SSLDIR/certs"
+    alpine_SSLKEYS="$alpine_SSLDIR/private"
+    alpine_SSLINCLUDE="/data/data/com.termux/files/usr/include/openssl"
+    alpine_SSLLIB="/data/data/com.termux/files/usr/lib"
+    alpine_path_delim="/"
+	 ;;
   *-apple-darwin*)
     systype="OSX"
     alpine_path_delim="/"
