diff --git a/imap/src/osdep/unix/env_unix.c b/imap/src/osdep/unix/env_unix.c
index 89e5092..f5a734e 100644
--- a/imap/src/osdep/unix/env_unix.c
+++ b/imap/src/osdep/unix/env_unix.c
@@ -123,7 +123,7 @@ static char *lockpgm = LOCKPGM;
  * systems.
  */
 
-static const char *tmpdir = "/tmp";
+static const char *tmpdir = "@TERMUX_PREFIX@/tmp";
 
 /* Do not change shlock_mode.  Doing so can cause mailbox corruption and
  * denial of service.  It also defeats the entire purpose of the shared
