diff -Naur ./alpine.orig/imap/src/osdep/unix/os_lnx.h ./alpine.patch/imap/src/osdep/unix/os_lnx.h
--- ./imap/src/osdep/unix/os_lnx.h	2021-05-16 06:44:05.008507235 +0000
+++ ./imap/src/osdep/unix/os_lnx.h	2021-05-16 06:44:57.139105846 +0000
@@ -58,6 +58,13 @@
 
 #define direct dirent
 
+#ifndef L_SET
+#define L_SET SEEK_SET
+#endif
+#ifndef FNDELAY
+#define FNDELAY O_NDELAY
+#endif
+
 int safe_flock (int, int);
 #define flock safe_flock
 
