diff -uNr alpine-2.21/imap/src/osdep/unix/os_lnx.c alpine-2.21.mod/imap/src/osdep/unix/os_lnx.c
--- alpine-2.21/imap/src/osdep/unix/os_lnx.c	2017-02-06 02:06:22.499218141 +0200
+++ alpine-2.21.mod/imap/src/osdep/unix/os_lnx.c	2018-07-03 21:08:21.624860068 +0300
@@ -35,7 +35,6 @@
 #include <netdb.h>
 #include <ctype.h>
 #include <errno.h>
-extern int errno;		/* just in case */
 #include <pwd.h>
 #include "misc.h"
 
@@ -50,3 +49,7 @@
 #include "tz_sv4.c"
 #include "flocklnx.c"
 #include "utime.c"
+long gethostid (void)
+{
+return 0xdeadface;
+}
