--- ../cache/alpine-2.20/pith/Makefile.am	2015-01-12 05:12:25.585178823 +0000
+++ ./pith/Makefile.am	2016-12-11 05:35:52.697766827 +0000
@@ -29,19 +29,13 @@
 	state.c status.c store.c stream.c string.c strlst.c takeaddr.c tempfile.c text.c \
 	thread.c adjtime.c url.c util.c helptext.c smkeys.c smime.c
 
-help_c_gen$(EXEEXT): $(help_c_gen_OBJECTS) $(help_c_gen_DEPENDENCIES) 
-	@rm -f help_c_gen$(EXEEXT)
-	$(LINK) $(help_c_gen_OBJECTS) $(help_c_gen_LDADD)
-help_h_gen$(EXEEXT): $(help_h_gen_OBJECTS) $(help_h_gen_DEPENDENCIES) 
-	@rm -f help_h_gen$(EXEEXT)
-	$(LINK) $(help_h_gen_OBJECTS) $(help_h_gen_LDADD)
 
-helptext.c: help_c_gen pine.hlp
+helptext.c:  pine.hlp
 	./help_c_gen < pine.hlp > $@
 
-helptext.h: help_h_gen pine.hlp
+helptext.h:  pine.hlp
 	./help_h_gen < pine.hlp > $@
 
 AM_CPPFLAGS = -I@top_builddir@/include -I@top_srcdir@/include
 
-CLEANFILES = helptext.c helptext.h help_h_gen help_c_gen
+CLEANFILES = helptext.c helptext.h 
