diff -uNr goaccess-1.3/src/tcbtdb.h goaccess-1.3.mod/src/tcbtdb.h
--- goaccess-1.3/src/tcbtdb.h	2018-11-23 04:00:08.000000000 +0200
+++ goaccess-1.3.mod/src/tcbtdb.h	2019-02-01 01:30:19.191246767 +0200
@@ -46,7 +46,7 @@
 #define TC_LMEMB 128
 #define TC_NMEMB 256
 #define TC_BNUM  32749
-#define TC_DBPATH "/tmp/goaccess"
+#define TC_DBPATH "@TERMUX_PREFIX@/tmp/goaccess"
 #define TC_DBPMODE 0755
 #define TC_ZLIB 1
 #define TC_BZ2  2
