diff -uNr eja-13.11.20/eja.lua eja-13.11.20.mod/eja.lua
--- eja-13.11.20/eja.lua	2020-11-20 12:34:27.000000000 +0200
+++ eja-13.11.20.mod/eja.lua	2020-11-25 03:53:10.134610701 +0200
@@ -33,9 +33,9 @@
  end
 
  eja.path=_eja_path or '/'
- eja.pathBin=eja.pathBin or eja.path..'/usr/bin/'
+ eja.pathBin=eja.pathBin or eja.path..'/bin/'
  eja.pathEtc=eja.pathEtc or eja.path..'/etc/eja/'
- eja.pathLib=eja.pathLib or eja.path..'/usr/lib/eja/'
+ eja.pathLib=eja.pathLib or eja.path..'/lib/eja/'
  eja.pathVar=eja.pathVar or eja.path..'/var/eja/'
  eja.pathTmp=eja.pathTmp or eja.path..'/tmp/'
  eja.pathLock=eja.pathLock or eja.path..'/var/lock/'
