--- ../src.orig/runit-2.1.2/src/conf-cc	2020-06-10 13:15:17.927086588 +0200
+++ ./runit-2.1.2/src/conf-cc	2020-06-10 13:27:28.210306192 +0200
@@ -1,5 +1,5 @@
-gcc -O2 -Wall
+${CC} ${CPPFLAGS} ${CFLAGS}
 
-gcc -O2 -Wimplicit -Wunused -Wcomment -Wchar-subscripts -Wuninitialized -Wshadow -Wcast-qual -Wcast-align -Wwrite-strings
+${CC} -Wimplicit -Wunused -Wcomment -Wchar-subscripts -Wuninitialized -Wshadow -Wcast-qual -Wcast-align -Wwrite-strings ${CPPFLAGS} ${CFLAGS}
 
 This will be used to compile .c files.
