diff --git a/projects/make/wren_test.make b/projects/make/wren_test.make
index 6d8c6383..8309bf5b 100644
--- a/projects/make/wren_test.make
+++ b/projects/make/wren_test.make
@@ -40,7 +40,7 @@ ALL_CFLAGS += $(CFLAGS) $(ALL_CPPFLAGS) -m64 -O2 -std=c99
 ALL_CXXFLAGS += $(CXXFLAGS) $(ALL_CPPFLAGS) -m64 -O2
 LIBS += ../../lib/libwren.a -lm
 LDDEPS += ../../lib/libwren.a
-ALL_LDFLAGS += $(LDFLAGS) -L/usr/lib64 -m64 -s
+ALL_LDFLAGS += $(LDFLAGS) -m64 -s
 
 else ifeq ($(config),release_32bit)
 TARGETDIR = ../../bin
@@ -51,7 +51,7 @@ ALL_CFLAGS += $(CFLAGS) $(ALL_CPPFLAGS) -m32 -O2 -std=c99
 ALL_CXXFLAGS += $(CXXFLAGS) $(ALL_CPPFLAGS) -m32 -O2
 LIBS += ../../lib/libwren.a -lm
 LDDEPS += ../../lib/libwren.a
-ALL_LDFLAGS += $(LDFLAGS) -L/usr/lib32 -m32 -s
+ALL_LDFLAGS += $(LDFLAGS) -m32 -s
 
 else ifeq ($(config),release_64bit-no-nan-tagging)
 TARGETDIR = ../../bin
--- ../wren.make.orig	2021-09-22 16:33:01.100347594 +0200
+++ ./projects/make/wren.make	2021-09-22 16:33:36.664323142 +0200
@@ -40,7 +40,7 @@
 DEFINES += -DNDEBUG
 ALL_CFLAGS += $(CFLAGS) $(ALL_CPPFLAGS) -m64 -O2 -std=c99
 ALL_CXXFLAGS += $(CXXFLAGS) $(ALL_CPPFLAGS) -m64 -O2
-ALL_LDFLAGS += $(LDFLAGS) -L/usr/lib64 -m64 -s
+ALL_LDFLAGS += $(LDFLAGS) -m64 -s
 
 else ifeq ($(config),release_32bit)
 TARGETDIR = ../../lib
@@ -49,7 +49,7 @@
 DEFINES += -DNDEBUG
 ALL_CFLAGS += $(CFLAGS) $(ALL_CPPFLAGS) -m32 -O2 -std=c99
 ALL_CXXFLAGS += $(CXXFLAGS) $(ALL_CPPFLAGS) -m32 -O2
-ALL_LDFLAGS += $(LDFLAGS) -L/usr/lib32 -m32 -s
+ALL_LDFLAGS += $(LDFLAGS) -m32 -s
 
 else ifeq ($(config),release_64bit-no-nan-tagging)
 TARGETDIR = ../../lib
--- ../wren_shared.make.orig	2021-09-22 16:34:05.898184395 +0200
+++ ./projects/make/wren_shared.make	2021-09-22 16:34:22.975159452 +0200
@@ -40,7 +40,7 @@
 DEFINES += -DNDEBUG
 ALL_CFLAGS += $(CFLAGS) $(ALL_CPPFLAGS) -m64 -O2 -fPIC -std=c99
 ALL_CXXFLAGS += $(CXXFLAGS) $(ALL_CPPFLAGS) -m64 -O2 -fPIC
-ALL_LDFLAGS += $(LDFLAGS) -L/usr/lib64 -m64 -shared -Wl,-soname=libwren.so -s
+ALL_LDFLAGS += $(LDFLAGS) -m64 -shared -Wl,-soname=libwren.so -s
 
 else ifeq ($(config),release_32bit)
 TARGETDIR = ../../lib
@@ -49,7 +49,7 @@
 DEFINES += -DNDEBUG
 ALL_CFLAGS += $(CFLAGS) $(ALL_CPPFLAGS) -m32 -O2 -fPIC -std=c99
 ALL_CXXFLAGS += $(CXXFLAGS) $(ALL_CPPFLAGS) -m32 -O2 -fPIC
-ALL_LDFLAGS += $(LDFLAGS) -L/usr/lib32 -m32 -shared -Wl,-soname=libwren.so -s
+ALL_LDFLAGS += $(LDFLAGS) -m32 -shared -Wl,-soname=libwren.so -s
 
 else ifeq ($(config),release_64bit-no-nan-tagging)
 TARGETDIR = ../../lib
