From b4a44dff88fdb8fd2526c70d197037ec5190a7f7 Mon Sep 17 00:00:00 2001
From: xingguangcuican6666 <xingguangcuican666@foxmail.com>
Date: Sun, 26 Oct 2025 12:31:34 +0800
Subject: [PATCH 02/13] remove_setuid_using

---
 include/xconfig.h.in | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/include/xconfig.h.in b/include/xconfig.h.in
index c130600..52a49c2 100644
--- a/include/xconfig.h.in
+++ b/include/xconfig.h.in
@@ -132,12 +132,12 @@
 #cmakedefine HAVE_MLOCKALL
 	/* working mlockall() is present in libc */
 #define HAVE_FLOCK 1		/* *BSD flock() is present in libc */
-#define HAVE_SETREUID 1		/* setreuid() is present in libc */
-#define HAVE_SETEUID 1		/* seteuid() is present in libc */
-#define HAVE_SETUID 1		/* setuid() is present in libc */
-#define HAVE_SETREGID 1		/* setregid() is present in libc */
-#define HAVE_SETEGID 1		/* setegid() is present in libc */
-#define HAVE_SETGID 1		/* setgid() is present in libc */
+//#define HAVE_SETREUID 1		/* setreuid() is present in libc */
+//#define HAVE_SETEUID 1		/* seteuid() is present in libc */
+//#define HAVE_SETUID 1		/* setuid() is present in libc */
+//#define HAVE_SETREGID 1		/* setregid() is present in libc */
+//#define HAVE_SETEGID 1		/* setegid() is present in libc */
+//#define HAVE_SETGID 1		/* setgid() is present in libc */
 #define HAVE_TCGETATTR 1		/* tcgetattr() is present in libc */
 #define HAVE_TCSETATTR 1		/* tcsetattr() is present in libc */
 #define HAVE_GETHOSTID 1		/* gethostid() is present in libc */
-- 
2.51.1

