From 0737a9c7227bc9e2a799211451ae8796394222d2 Mon Sep 17 00:00:00 2001
From: xingguangcuican6666 <xingguangcuican666@foxmail.com>
Date: Thu, 23 Oct 2025 19:10:10 +0800
Subject: [PATCH 05/12] fix_checksum

---
 genisoimage/genisoimage.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/genisoimage/genisoimage.c b/genisoimage/genisoimage.c
index 46f0cb7..46998fc 100644
--- a/genisoimage/genisoimage.c
+++ b/genisoimage/genisoimage.c
@@ -85,7 +85,7 @@ extern char     *optarg;
 #include <locale.h>
 #include <langinfo.h>
 #endif
-
+#include "checksum.h"
 struct directory *root = NULL;
 int		path_ind;
 
-- 
2.51.1

