From ca18d5fc5b6f8343c5e2d8cf720aafe08d943346 Mon Sep 17 00:00:00 2001
From: xingguangcuican6666 <xingguangcuican666@foxmail.com>
Date: Sun, 26 Oct 2025 12:31:34 +0800
Subject: [PATCH 05/13] fix_checksum

---
 genisoimage/genisoimage.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/genisoimage/genisoimage.c b/genisoimage/genisoimage.c
index 46f0cb7..46998fc 100644
--- a/genisoimage/genisoimage.c
+++ b/genisoimage/genisoimage.c
@@ -85,7 +85,7 @@ extern char     *optarg;
 #include <locale.h>
 #include <langinfo.h>
 #endif
-
+#include "checksum.h"
 struct directory *root = NULL;
 int		path_ind;
 
-- 
2.51.1

