From 316ca652382be64081ea596bde985d7987a24ac1 Mon Sep 17 00:00:00 2001
From: xingguangcuican6666 <xingguangcuican666@foxmail.com>
Date: Thu, 23 Oct 2025 19:20:27 +0800
Subject: [PATCH 07/12] remove_pthreads_for_termux

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c22b720..7b21bde 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,6 @@
 cmake_minimum_required(VERSION 3.10)
 PROJECT (cdrkit C)
 add_definitions("-D__THROW=")
+add_definitions("-UTHREADED_CHECKSUMS")
 SUBDIRS(include genisoimage wodim libedc libhfs_iso libparanoia icedax libusal librols libunls readom netscsid 3rd-party/dirsplit)
 
-- 
2.51.1

