From e5db68441a0e222a263d4b33505621136bc4e4e6 Mon Sep 17 00:00:00 2001
From: xingguangcuican6666 <xingguangcuican666@foxmail.com>
Date: Sun, 26 Oct 2025 12:31:34 +0800
Subject: [PATCH 07/13] remove_pthreads_for_termux

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c22b720..7b21bde 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,6 @@
 cmake_minimum_required(VERSION 3.10)
 PROJECT (cdrkit C)
 add_definitions("-D__THROW=")
+add_definitions("-UTHREADED_CHECKSUMS")
 SUBDIRS(include genisoimage wodim libedc libhfs_iso libparanoia icedax libusal librols libunls readom netscsid 3rd-party/dirsplit)
 
-- 
2.51.1

