From 8636fa79d8c5c01a064b4a1dba897acf082d218f Mon Sep 17 00:00:00 2001
From: xingguangcuican6666 <xingguangcuican666@foxmail.com>
Date: Thu, 23 Oct 2025 19:23:25 +0800
Subject: [PATCH 08/12] fix_outfile_duplicate

---
 genisoimage/genisoimage.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/genisoimage/genisoimage.h b/genisoimage/genisoimage.h
index bbedfb0..6f0f7d1 100644
--- a/genisoimage/genisoimage.h
+++ b/genisoimage/genisoimage.h
@@ -376,7 +376,7 @@ extern int	use_fileversion;
 extern int	split_SL_component;
 extern int	split_SL_field;
 extern char	*trans_tbl;
-char		*outfile;
+extern char		*outfile;
 
 #define	JMAX		64	/* maximum Joliet file name length (spec) */
 #define	JLONGMAX	103	/* out of spec Joliet file name length */
-- 
2.51.1

