--- a/bazel/build_defs.bzl
+++ b/bazel/build_defs.bzl
@@ -36,6 +36,7 @@
     # "-Wshorten-64-to-32",  # not in GCC (and my Kokoro images doesn't have Clang)
     "-Werror",
     "-Wno-long-long",
+    "-Wno-gnu-offsetof-extensions",
 ])
 _DEFAULT_COPTS.extend([
     "-std=c99",
@@ -43,6 +44,7 @@
     "-Werror=pedantic",
     "-Wall",
     "-Wstrict-prototypes",
+    "-Wno-gnu-offsetof-extensions",
     # GCC (at least) emits spurious warnings for this that cannot be fixed
     # without introducing redundant initialization (with runtime cost):
     #   https://gcc.gnu.org/bugzilla/show_bug.cgi?id=80635
