--- a/adapter/CMakeLists.txt
+++ b/adapter/CMakeLists.txt
@@ -52,15 +52,13 @@ externalproject_add(cargo_build
     CONFIGURE_COMMAND ""
     SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}"
     BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}"
-    BUILD_COMMAND cargo rustc --package=codelldb --bin=codelldb ${CargoFlags} --features=weaklink -- ${RustcFlags}
+    BUILD_COMMAND cargo rustc --package=codelldb --bin=codelldb ${CargoFlags} -- ${RustcFlags}
     BUILD_BYPRODUCTS ${CargoOutDir}/${CodelldbExe}
     BUILD_ALWAYS ON
     USES_TERMINAL_BUILD ON
     INSTALL_COMMAND ""
 )
 
-add_dependencies(cargo_build lldb)
-
 # Exe
 add_custom_target(codelldb_exe
     COMMAND ${UpdateFile} ${CargoOutDir}/${CodelldbExe} ${CMAKE_CURRENT_BINARY_DIR}
