--- lua-language-server/3rd/bee.lua/project/common.lua	2021-10-22 15:08:44.191055637 +0530
+++ lua-language-server-patch/3rd/bee.lua/project/common.lua	2021-10-22 15:13:31.211055527 +0530
@@ -10,7 +10,11 @@
     },
     linux = {
         flags = "-fPIC"
+    },
+    android = {
+        flags = "-fPIC"
     }
+
 }
 
 lm:source_set "source_bee" {
@@ -45,6 +49,7 @@
         }
     },
     android = {
+        flags ="-fPIC",
         sources = {
             "!bee/**_win.cpp",
             "!bee/**_osx.cpp",
