--- lua-language-server/3rd/bee.lua/project/common.lua	2021-10-22 15:08:44.191055637 +0530
+++ lua-language-server-patch/3rd/bee.lua/project/common.lua	2021-10-25 22:52:35.474777452 +0530
@@ -45,6 +45,8 @@
         }
     },
     android = {
+        flags ="-fPIC",
+        ldflags ="-L@TERMUX_PREFIX@/lib -landroid-spawn",
         sources = {
             "!bee/**_win.cpp",
             "!bee/**_osx.cpp",
