--- lua-language-server/make/install.lua	2021-08-09 14:36:08.073522963 +0530
+++ lua-language-server-patch/make/install.lua	2021-08-10 10:55:34.845906577 +0530
@@ -3,7 +3,7 @@
 local pf = require 'bee.platform'
 
 local CWD = fs.current_path()
-local output = CWD / 'bin' / pf.OS
+local output = CWD / 'bin' / 'Android'
 local bindir = CWD / builddir / 'bin'
 local exe = pf.OS == 'Windows' and ".exe" or ""
 local dll = pf.OS == 'Windows' and ".dll" or ".so"
