--- lua-language-server/3rd/bee.lua/3rd/lua54/luaconf.h
+++ lua-language-server/3rd/bee.lua/3rd/lua54/luaconf.h
@@ -223,7 +223,7 @@
 
 #else			/* }{ */
 
-#define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT	"@TERMUX_PREFIX@/"
 #define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR "/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR "/"
 
--- lua-language-server/3rd/bee.lua/3rd/lua55/luaconf.h
+++ lua-language-server/3rd/bee.lua/3rd/lua55/luaconf.h
@@ -239,7 +239,7 @@
 
 #else			/* }{ */
 
-#define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT	"@TERMUX_PREFIX@/"
 #define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR "/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR "/"
 
