--- lua-language-server/3rd/bee.lua/3rd/lua/luaconf.h	2021-08-09 14:37:07.223522941 +0530
+++ lua-language-server-patch/3rd/bee.lua/3rd/lua/luaconf.h	2021-08-10 00:18:08.283997562 +0530
@@ -227,7 +227,7 @@
 
 #else			/* }{ */
 
-#define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT	"@TERMUX_PREFIX@/"
 #define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR "/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR "/"
 
