--- lua-language-server/3rd/luamake/scripts/compiler/gcc.lua	2021-08-10 10:20:07.635907389 +0530
+++ lua-language-server-patch/3rd/luamake/scripts/compiler/gcc.lua	2021-08-10 14:26:20.565901753 +0530
@@ -64,7 +64,8 @@
 function gcc.update_ldflags(ldflags, attribute)
     if globals.os == "android" then
         --TODO android不支持static crt
-        ldflags[#ldflags+1] = "-lstdc++"
+        ldflags[#ldflags+1] = "-lc++_static"
+        ldflags[#ldflags+1] = "-lc++abi"
     elseif attribute.crt == "dynamic" then
         ldflags[#ldflags+1] = "-lstdc++"
     else
