--- lua-language-server/make.lua	2021-10-04 00:13:48.079999941 +0530
+++ lua-language-server-patch/make.lua	2021-10-04 00:18:30.059999834 +0530
@@ -21,16 +21,3 @@
         'bee',
     }
 }
-
-local fs = require 'bee.filesystem'
-local pf = require 'bee.platform'
-local exe = pf.OS == 'Windows' and ".exe" or ""
-lm:build 'unittest' {
-    fs.path 'bin' / pf.OS / ('lua-language-server' .. exe), 'test.lua', '-E',
-    pool = "console",
-    deps = {
-        'install',
-    }
-}
-
-lm:default 'unittest'
