--- lua-language-server/make.lua	2021-10-16 18:04:03.750643184 +0530
+++ lua-language-server-patch/make.lua	2021-10-16 18:05:34.250643150 +0530
@@ -40,25 +40,6 @@
     output = lm.bindir.."/main.lua",
 }
 
-lm:build "bee-test" {
-    lm.bindir.."/lua-language-server"..exe, "3rd/bee.lua/test/test.lua",
-    pool = "console",
-    deps = {
-        "lua-language-server",
-        "copy_bootstrap"
-    },
-}
-
-lm:build 'unit-test' {
-    lm.bindir.."/lua-language-server"..exe, 'test.lua',
-    pool = "console",
-    deps = {
-        "bee-test",
-    }
-}
-
 lm:default {
-    "bee-test",
-    "unit-test",
     "install",
 }
