diff -uNr clamav-0.100.2/libclamav/others_common.c clamav-0.100.2.mod/libclamav/others_common.c
--- clamav-0.100.2/libclamav/others_common.c	2018-09-19 22:29:07.000000000 +0300
+++ clamav-0.100.2.mod/libclamav/others_common.c	2018-10-19 13:03:39.917694079 +0300
@@ -434,7 +434,7 @@
 #ifdef _WIN32
 #define P_tmpdir "C:\\"
 #else
-#define P_tmpdir "/tmp"
+#define P_tmpdir "@TERMUX_PREFIX@/tmp"
 #endif /* _WIN32 */
 #endif /* P_tmpdir */
 
