diff -uNr clamav-0.100.2/configure clamav-0.100.2.mod/configure
--- clamav-0.100.2/configure	2018-09-19 22:29:07.000000000 +0300
+++ clamav-0.100.2.mod/configure	2018-10-19 13:09:14.015897561 +0300
@@ -18809,10 +18809,8 @@
 fi
 
 
-if test ! -f "$ZLIB_HOME/include/zlib.h"
+if true
 then
-    as_fn_error $? "Please install zlib and zlib-devel packages" "$LINENO" 5
-else
 
     vuln=`grep "ZLIB_VERSION \"1.2.0\"" $ZLIB_HOME/include/zlib.h`
     if test -z "$vuln"; then
@@ -18873,11 +18871,7 @@
 fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_z_inflateEnd" >&5
 $as_echo "$ac_cv_lib_z_inflateEnd" >&6; }
-if test "x$ac_cv_lib_z_inflateEnd" = xyes; then :
-  LIBCLAMAV_LIBS="$LIBCLAMAV_LIBS -L$ZLIB_HOME/lib -lz"; FRESHCLAM_LIBS="$FRESHCLAM_LIBS -L$ZLIB_HOME/lib -lz"
-else
-  as_fn_error $? "Please install zlib and zlib-devel packages" "$LINENO" 5
-fi
+LIBCLAMAV_LIBS="$LIBCLAMAV_LIBS -L$ZLIB_HOME/lib -lz"; FRESHCLAM_LIBS="$FRESHCLAM_LIBS -L$ZLIB_HOME/lib -lz"
 
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for gzopen in -lz" >&5
 $as_echo_n "checking for gzopen in -lz... " >&6; }
@@ -20873,7 +20867,7 @@
 
     have_fanotify="no"
     if test "$have_pthreads" = "yes"; then
-	THREAD_LIBS="-lpthread"
+	THREAD_LIBS=""
 	TH_SAFE="-thread-safe"
 	if test "$want_fanotify" = "yes"; then
 	    ac_fn_c_check_header_mongrel "$LINENO" "sys/fanotify.h" "ac_cv_header_sys_fanotify_h" "$ac_includes_default"
@@ -21097,7 +21091,7 @@
 if test "$have_milter" = "yes"; then
 
         save_LIBS="$LIBS"
-    CLAMAV_MILTER_LIBS="$CLAMAV_MILTER_LIBS -lpthread"
+    CLAMAV_MILTER_LIBS="$CLAMAV_MILTER_LIBS"
     if test -d /usr/lib/libmilter ; then
 	CLAMAV_MILTER_LIBS="$CLAMAV_MILTER_LIBS -L/usr/lib/libmilter"
     fi
