diff -uNr clamav-0.102.0/configure clamav-0.102.0.mod/configure
--- clamav-0.102.0/configure	2019-10-01 20:24:08.000000000 +0300
+++ clamav-0.102.0.mod/configure	2019-11-02 23:32:29.043631184 +0200
@@ -26711,7 +26711,7 @@
 
     have_fanotify="no"
     if test "$have_pthreads" = "yes"; then
-	THREAD_LIBS="-lpthread"
+	THREAD_LIBS=""
 	TH_SAFE="-thread-safe"
 	if test "$want_fanotify" = "yes"; then
 	    ac_fn_c_check_header_mongrel "$LINENO" "sys/fanotify.h" "ac_cv_header_sys_fanotify_h" "$ac_includes_default"
