Prevents:
error: use of undeclared identifier 'LLONG_MAX'

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,7 +7,7 @@ else()
     cmake_minimum_required( VERSION 3.14 )
 endif()
 
-set(CMAKE_C_STANDARD 90)
+set(CMAKE_C_STANDARD 99)
 set(CMAKE_C_STANDARD_REQUIRED ON)
 # CMAKE_CXX_STANDARD not set because we may need to match the compiler used to build LLVM.
 # There's probably a good chance this issue will go away if using modern LLVM versions.
