diff -uNr luv-1.36.0-0/CMakeLists.txt luv-1.36.0-0.mod/CMakeLists.txt
--- luv-1.36.0-0/CMakeLists.txt	2020-04-19 08:44:19.000000000 +0000
+++ luv-1.36.0-0.mod/CMakeLists.txt	2021-02-05 22:07:47.397165825 +0000
@@ -108,7 +108,6 @@
       include_directories(${LUAJIT_INCLUDE_DIR})
       link_directories(${LUAJIT_LIBRARIES})
     else (USE_LUAJIT)
-      find_package(Lua REQUIRED)
       include_directories(${LUA_INCLUDE_DIR})
     endif (USE_LUAJIT)
 
@@ -187,7 +186,7 @@
   elseif("${CMAKE_SYSTEM_NAME}" MATCHES "Linux")
     target_link_libraries(${TARGET_NAME} ${LIBUV_LIBRARIES} rt)
   else()
-    target_link_libraries(${TARGET_NAME} ${LIBUV_LIBRARIES})
+    target_link_libraries(${TARGET_NAME} ${LIBUV_LIBRARIES} lua5.3)
   endif()
 endforeach()
 
