diff -uNr aterm-1.0.1/src/feature.h aterm-1.0.1.mod/src/feature.h
--- aterm-1.0.1/src/feature.h	2005-12-06 19:08:44.000000000 +0200
+++ aterm-1.0.1.mod/src/feature.h	2018-11-28 19:48:54.161918612 +0200
@@ -342,10 +342,10 @@
 			"taipei16"
 #  define NFONT_LIST	"8x16", "10x20", "12x24", "10x20", "8x16"
 # else				/* no Kanji or Big5 support */
-#  define NFONTS	5
-#  define FONT0_IDX	2
+#  define NFONTS	4
+#  define FONT0_IDX	0
 #  undef  MFONT_LIST
-#  define NFONT_LIST	"7x14", "6x10", "6x13", "8x13", "9x15"
+#  define NFONT_LIST	"6x13", "7x14", "8x13", "9x15"
 # endif
 #endif
 
