diff -uNr aterm-1.0.1/autoconf/Make.common.in aterm-1.0.1.mod/autoconf/Make.common.in
--- aterm-1.0.1/autoconf/Make.common.in	2005-05-26 19:44:49.000000000 +0300
+++ aterm-1.0.1.mod/autoconf/Make.common.in	2018-11-28 19:42:34.675482946 +0200
@@ -41,7 +41,7 @@
 CP = @CP@
 SED = @SED@
 INSTALL = @INSTALL@
-INSTALL_PROGRAM = @INSTALL@ -s -m 755
+INSTALL_PROGRAM = @INSTALL@ -m 755
 INSTALL_DATA = @INSTALL@ -m 644
 
 # Flags & libs
