--- a/front-end/src/ui/event.rs
+++ b/front-end/src/ui/event.rs
@@ -467,7 +467,7 @@
         let mut state = state_original.lock().unwrap();
 
         // TODO: Ask for conformation before downloading
-        let mut command = tokio::process::Command::new("youtube-dl");
+        let mut command = tokio::process::Command::new("yt-dlp");
         let download_url;
         if let Some(focused_index) = state.musicbar.1.selected() {
             let music_id = &state.musicbar.0[focused_index].id;
