--- a/CMakeLists.txt	2025-11-12 23:26:18.092349694 +0530
+++ b/CMakeLists.txt	2025-11-12 23:26:30.042038277 +0530
@@ -391,14 +391,6 @@
     endif()
   endif()
 
-  # Add example subdirectories and an "examples" target.
-  if(MSVC)
-    add_subdirectory(examples)
-  else()
-    add_subdirectory(examples EXCLUDE_FROM_ALL)
-    get_property(examples GLOBAL PROPERTY DART_EXAMPLES)
-    add_custom_target(examples DEPENDS ${examples})
-  endif()
 
   if(DART_VERBOSE)
     message(STATUS "")
