diff -uNr netpbm-10.73.31/buildtools/installnetpbm.pl netpbm-10.73.31.mod/buildtools/installnetpbm.pl
--- netpbm-10.73.31/buildtools/installnetpbm.pl	2019-06-29 02:07:55.000000000 +0300
+++ netpbm-10.73.31.mod/buildtools/installnetpbm.pl	2020-05-12 00:06:51.166788332 +0300
@@ -31,17 +31,7 @@
 sub prompt($$) {
 
     my ($prompt, $default) = @_;
-
-    print("$prompt ($default) ==> ");
-
-    my $response = <STDIN>;
-
-    chomp($response);
-    if ($response eq "") {
-        $response = $default;
-    }
-
-    return $response;
+    return $default;
 }
 
 
@@ -123,7 +113,7 @@
         $default = "/usr/local/netpbm";
     }
 
-    my $response = prompt("install prefix", $default);
+    my $response = "@TERMUX_PREFIX@";
 
     my $prefix;
 
@@ -511,7 +501,7 @@
         $done = $FALSE;
 
         while (!$done) {
-            my $response = prompt("Y(es) or N(o)", "Y");
+            my $response = "N";
 
             if (uc($response) eq "Y") {
                 execLdconfig();
@@ -766,7 +756,7 @@
     my $manDir;
 
     while (!$manDir) {
-        my $default = "$prefix/man";
+        my $default = "$prefix/share/man";
 
         my $response = prompt("man page directory", $default);
 
@@ -954,10 +944,10 @@
 
     print("Installing man pages...\n");
 
-    my $rc = system("$cpCommand $pkgdir/man/* $manDir/");
+    my $rc = system("$cpCommand $pkgdir/share/man/* $manDir/");
 
     if ($rc != 0) {
-        print("copy of man pages from $pkgdir/man to $manDir failed.\n");
+        print("copy of man pages from $pkgdir/share/man to $manDir failed.\n");
         print("cp exit code is $rc\n");
     } else {
         print("done.\n");
@@ -967,8 +957,6 @@
 
     removeObsoleteManPage($manDir);
 
-    makeInManwebPath("$manDir/web");
-    
     $$mandirR = $manDir;
 }
 
