diff -u -r ../netpbm-advanced-r3084/buildtools/installnetpbm.pl ./buildtools/installnetpbm.pl
--- ../netpbm-advanced-r3084/buildtools/installnetpbm.pl	2017-10-06 07:49:15.000000000 +0200
+++ ./buildtools/installnetpbm.pl	2017-10-06 12:18:07.809552955 +0200
@@ -32,16 +32,7 @@
 
     my ($prompt, $default) = @_;
 
-    print("$prompt ($default) ==> ");
-
-    my $response = <STDIN>;
-
-    chomp($response);
-    if ($response eq "") {
-        $response = $default;
-    }
-
-    return $response;
+    return $default;
 }
 
 
@@ -185,7 +176,7 @@
         $default = "/usr/local/netpbm";
     }
 
-    my $response = fsObjPrompt("install prefix", $default);
+    my $response = "@TERMUX_PREFIX@";
 
     my $prefix;
 
@@ -573,7 +564,7 @@
         $done = $FALSE;
 
         while (!$done) {
-            my $response = prompt("Y(es) or N(o)", "Y");
+            my $response = "N";
 
             if (uc($response) eq "Y") {
                 execLdconfig();
