diff -u -r ../netpbm-10.35.95/lib/Makefile ./lib/Makefile
--- ../netpbm-10.35.95/lib/Makefile	2009-09-18 15:04:37.000000000 -0400
+++ ./lib/Makefile	2014-12-29 01:36:13.772104906 -0500
@@ -117,7 +117,7 @@
 	rm -f $@
 	$(SYMLINK) $< $@
 libnetpbm.$(NETPBMLIBSUFFIX).$(MAJ).$(MIN): $(LIBOBJECTS) $(LIBOBJECTS_X)
-	$(LD) $(LDSHLIB) -o $@ $(LIBOBJECTS) $(LIBOBJECTS_X) \
+	$(LD) $(LDFLAGS) $(LDSHLIB) -o $@ $(LIBOBJECTS) $(LIBOBJECTS_X) \
           $(SHLIB_CLIB) -lm $(LADD)
 endif
 
