diff -uNr iverilog-10_2/Makefile.in iverilog-10_2.mod/Makefile.in
--- iverilog-10_2/Makefile.in	2017-08-24 18:22:49.000000000 +0300
+++ iverilog-10_2.mod/Makefile.in	2019-02-25 20:18:55.327140615 +0200
@@ -67,8 +67,8 @@
 dllib=@DLLIB@
 
 # For a cross compile these defines will need to be set accordingly.
-HOSTCC = @CC@
-HOSTCFLAGS = @WARNING_FLAGS@ @WARNING_FLAGS_CC@ @CFLAGS@
+HOSTCC = gcc
+HOSTCFLAGS = @WARNING_FLAGS@ @WARNING_FLAGS_CC@
 
 CC = @CC@
 CXX = @CXX@
