diff -uNr iverilog-10_2/vvp/Makefile.in iverilog-10_2.mod/vvp/Makefile.in
--- iverilog-10_2/vvp/Makefile.in	2017-08-24 18:22:49.000000000 +0300
+++ iverilog-10_2.mod/vvp/Makefile.in	2019-02-25 20:18:31.607079821 +0200
@@ -32,8 +32,8 @@
 includedir = @includedir@
 
 # For a cross compile these defines will need to be set accordingly.
-HOSTCC = @CC@
-HOSTCFLAGS = @WARNING_FLAGS@ @WARNING_FLAGS_CC@ @CFLAGS@
+HOSTCC = gcc
+HOSTCFLAGS = @WARNING_FLAGS@ @WARNING_FLAGS_CC@
 
 CC = @CC@
 CXX = @CXX@
