diff -uNr transmission-2.94/libtransmission/platform.c transmission-2.94.mod/libtransmission/platform.c
--- transmission-2.94/libtransmission/platform.c	2018-05-01 20:18:29.000000000 +0300
+++ transmission-2.94.mod/libtransmission/platform.c	2018-10-06 01:44:35.722770450 +0300
@@ -543,7 +543,7 @@
           {
             const char * pkg = PACKAGE_DATA_DIR;
             char * xdg = tr_env_get_string ("XDG_DATA_DIRS", NULL);
-            const char * fallback = "/usr/local/share:/usr/share";
+            const char * fallback = "@TERMUX_PREFIX@/share";
             char * buf = tr_strdup_printf ("%s:%s:%s", (pkg?pkg:""), (xdg?xdg:""), fallback);
             tr_free (xdg);
             tmp = buf;
