diff -uNr transmission-2.94/third-party/miniupnp/miniupnpc.c transmission-2.94.mod/third-party/miniupnp/miniupnpc.c
--- transmission-2.94/third-party/miniupnp/miniupnpc.c	2018-05-01 20:18:29.000000000 +0300
+++ transmission-2.94.mod/third-party/miniupnp/miniupnpc.c	2018-10-06 01:43:49.465939838 +0300
@@ -385,7 +385,7 @@
 #if !defined(_WIN32) && !defined(__amigaos__) && !defined(__amigaos4__)
 	/* first try to get infos from minissdpd ! */
 	if(!minissdpdsock)
-		minissdpdsock = "/var/run/minissdpd.sock";
+		minissdpdsock = "@TERMUX_PREFIX@/var/run/minissdpd.sock";
 	while(!devlist && deviceList[deviceIndex]) {
 		devlist = getDevicesFromMiniSSDPD(deviceList[deviceIndex],
 		                                  minissdpdsock);
