diff -uNr transmission-2.94/third-party/libnatpmp/natpmp.c transmission-2.94.mod/third-party/libnatpmp/natpmp.c
--- transmission-2.94/third-party/libnatpmp/natpmp.c	2018-05-01 20:18:29.000000000 +0300
+++ transmission-2.94.mod/third-party/libnatpmp/natpmp.c	2018-10-06 01:38:32.814708049 +0300
@@ -53,6 +53,8 @@
 #include "natpmp.h"
 #include "getgateway.h"
 
+#include <arpa/inet.h>
+
 LIBSPEC int initnatpmp(natpmp_t * p, int forcegw, in_addr_t forcedgw)
 {
 #ifdef WIN32
