diff -uNr transmission-2.94/third-party/miniupnp/miniwget.c transmission-2.94.mod/third-party/miniupnp/miniwget.c
--- transmission-2.94/third-party/miniupnp/miniwget.c	2018-05-01 20:18:29.000000000 +0300
+++ transmission-2.94.mod/third-party/miniupnp/miniwget.c	2018-10-06 01:38:32.821374744 +0300
@@ -40,7 +40,7 @@
 #include <netdb.h>
 #define closesocket close
 #endif /* #else _WIN32 */
-#if defined(__sun) || defined(sun)
+#if defined(__sun) || defined(sun) || defined __ANDROID__
 #define MIN(x,y) (((x)<(y))?(x):(y))
 #endif
 
