diff -uNr transmission-2.94/third-party/libnatpmp/getgateway.h transmission-2.94.mod/third-party/libnatpmp/getgateway.h
--- transmission-2.94/third-party/libnatpmp/getgateway.h	2018-05-01 20:18:29.000000000 +0300
+++ transmission-2.94.mod/third-party/libnatpmp/getgateway.h	2018-10-06 01:38:32.811374701 +0300
@@ -40,6 +40,8 @@
 #endif
 #include "declspec.h"
 
+#include <arpa/inet.h>
+
 /* getdefaultgateway() :
  * return value :
  *    0 : success
