diff -uNr transmission-2.94/libtransmission/utils.c transmission-2.94.mod/libtransmission/utils.c
--- transmission-2.94/libtransmission/utils.c	2018-05-01 20:18:29.000000000 +0300
+++ transmission-2.94.mod/libtransmission/utils.c	2018-10-06 01:38:32.831374787 +0300
@@ -1416,7 +1416,7 @@
   char buf[128];
   const int max_precision = (int) log10 (1.0 / DBL_EPSILON) - 1;
   tr_snprintf (buf, sizeof (buf), "%.*f", max_precision, x);
-  if ((pt = strstr (buf, localeconv ()->decimal_point)))
+  if ((pt = strstr (buf, ".")))
     pt[precision ? precision+1 : 0] = '\0';
   return atof (buf);
 }
