--- a/src/ffaudio/ffaudio-core.cc
+++ b/src/ffaudio/ffaudio-core.cc
@@ -288,7 +288,7 @@
         AVProbeData d = {name, buf, filled};
         score = target;
 
-        f = av_probe_input_format2 (& d, true, & score);
+        f = const_cast<AVInputFormat *> (av_probe_input_format2 (& d, true, & score));
         if (f)
             break;
 
@@ -361,7 +361,7 @@
 #endif
         if (stream && stream->codecpar && stream->codecpar->codec_type == AVMEDIA_TYPE_AUDIO)
         {
-            AVCodec * codec = avcodec_find_decoder (stream->codecpar->codec_id);
+            AVCodec * codec = const_cast<AVCodec *> (avcodec_find_decoder (stream->codecpar->codec_id));
 
             if (codec)
             {
