--- ./configure	2004-06-04 06:39:10.000000000 +0000
+++ ../configure	2018-03-11 08:11:27.716604849 +0000
@@ -13,16 +13,13 @@
 	exit 0
 fi
 
-CC=${CC:=cc}
-
-echo build byteorder.c...
-$CC byteorder.c -o byteorder || exit 1
+#CC = $CC
 
 INSTALL_MANPATH=`echo $MANPATH|cut -f1 -d:`
 if [ "$INSTALL_MANPATH" = "" ]; then
-	INSTALL_MANPATH="/usr/local/man"
+	INSTALL_MANPATH="@TERMUX_PREFIX@/share/man"
 fi
-BYTEORDER=`./byteorder -m`
+BYTEORDER=__LITTLE_ENDIAN_BITFIELD
 
 echo create byteorder.h...
 cat > byteorder.h <<EOF
@@ -61,46 +58,13 @@
 #
 # TCL detection
 #
-for TCLPATH_TRY in "/usr/bin/" "/usr/local/bin/" "/bin/"
-do
-	for TCLVER_TRY in "8.4" "8.3" "8.2" "8.1" "8.0"
-	do
-		if [ -z $TCLSH ]
-		then
-			TCLSH_TRY=${TCLPATH_TRY}tclsh${TCLVER_TRY}
-			if [ -f $TCLSH_TRY ]
-			then
-				TCLSH=$TCLSH_TRY
-				echo "===> Found Tclsh in: $TCLSH"
-			fi
-		fi
-	done
-done
-if [ -f $TCLSH ]
-then
-	TCL_VER=`echo puts \\$tcl_version | $TCLSH -`
-	USE_TCL='-DUSE_TCL'
-	TCL_LIB="-ltcl${TCL_VER}"
-	if [ -e /usr/include/tcl${TCL_VER} ]
-	then
-		TCL_INC="-I/usr/include/tcl${TCL_VER}"
-	elif [ -e /usr/include/tcl.h ]
-	then
-		TCL_INC=""
-	elif [ -e /usr/local/include/tcl${TCL_VER} ]
-	then
-		TCL_INC="-I/usr/local/include/tcl${TCL_VER}"
-	else
-		USE_TCL=""
-		TCL_LIB=""
-		echo "==> WARNING: no Tcl header files found!"
-	fi
-fi
+USE_TCL='-DUSE_TCL'
 if [ -n $USE_TCL ]
 then
-	LIBPOSTFIX=`ls -1 /usr/local/lib/ /usr/lib | grep 'libtcl[0-9]' | grep so | sed -e 's/\.so.*//g' -e 's/libtcl//g' | sort -r | head -1`
+	LIBPOSTFIX=`ls -1 @TERMUX_PREFIX@/lib | grep 'libtcl[0-9]' | grep so | sed -e 's/\.so.*//g' -e 's/libtcl//g' | sort -r | head -1`
 	TCL_LIB="-ltcl${LIBPOSTFIX} -lm -lpthread"
 fi
+TCL_INC="-I@TERMUX_PREFIX@/include"
 
 #
 # configurable stuff
@@ -161,6 +125,6 @@
 EOF
 
 echo creating dependences...
-$CC -MM *.c > .depend
+$CC -MM $TCL_INC *.c > .depend
 
 echo now you can try \`make\'
