diff --git a/src/recipe.rs b/src/recipe.rs
index bfa8b4b5..0d631cdd 100644
--- a/src/recipe.rs
+++ b/src/recipe.rs
@@ -489,6 +489,10 @@ impl<'src> Recipe<'src> {
           .unwrap_or_else(|| Interpreter::default_script_interpreter().clone()),
       )
     } else if self.body.first().is_some_and(Line::is_shebang) {
+      evaluated_lines[0] = format!(
+        "#!@TERMUX_PREFIX@{}",
+        evaluated_lines[0][2..].to_string()
+      ).replace("/usr/usr", "/usr");
       let shebang = &evaluated_lines[0];
       Executor::Shebang(Shebang::new(shebang).ok_or_else(|| Error::InvalidShebang {
         recipe: self.name,
