diff --git a/internal/frontend/cli/frontend.go b/internal/frontend/cli/frontend.go
index 82c0dbc..c34d73a 100644
--- a/internal/frontend/cli/frontend.go
+++ b/internal/frontend/cli/frontend.go
@@ -241,21 +241,16 @@ func (f *frontendCLI) Loop(credentialsError error) error {
 	}
 
 	f.Print(`
-            Welcome to ProtonMail Bridge interactive shell
-                              ___....___
-    ^^                __..-:'':__:..:__:'':-..__
-                  _.-:__:.-:'':  :  :  :'':-.:__:-._
-                .':.-:  :  :  :  :  :  :  :  :  :._:'.
-             _ :.':  :  :  :  :  :  :  :  :  :  :  :'.: _
-            [ ]:  :  :  :  :  :  :  :  :  :  :  :  :  :[ ]
-            [ ]:  :  :  :  :  :  :  :  :  :  :  :  :  :[ ]
-   :::::::::[ ]:__:__:__:__:__:__:__:__:__:__:__:__:__:[ ]:::::::::::
-   !!!!!!!!![ ]!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!![ ]!!!!!!!!!!!
-   ^^^^^^^^^[ ]^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^[ ]^^^^^^^^^^^
-            [ ]                                        [ ]
-            [ ]                                        [ ]
-      jgs   [ ]                                        [ ]
-    ~~^_~^~/   \~^-~^~ _~^-~_^~-^~_^~~-^~_~^~-~_~-^~_^/   \~^ ~~_ ^
+  Welcome to  ProtonMail Bridge 
+  interactive shell for Android
+
+  ^^
+    _                       _
+___[ ]_____________________[ ]___
+!!![ ]!!!!!!!!!!!!!!!!!!!!![ ]!!!
+^^^[ ]^^^^^^^^^^^^^^^^^^^^^[ ]^^^
+jgs[ ]                     [ ]
+^~/   \^~~~~~~~~~~~~~~~~~^/   \~^
 `)
 	f.Run()
 	return nil
