diff --git a/go.mod b/go.mod
index c0e8754..08c6caf 100644
--- a/go.mod
+++ b/go.mod
@@ -77,4 +77,5 @@ replace (
 	github.com/emersion/go-imap => github.com/ProtonMail/go-imap v0.0.0-20201102134601-418cd74e9474
 	github.com/jameskeane/bcrypt => github.com/ProtonMail/bcrypt v0.0.0-20170924085257-7509ea014998
 	golang.org/x/crypto => github.com/ProtonMail/crypto v0.0.0-20200818122824-ed5d25e28db8
+	github.com/emersion/go-imap-specialuse => github.com/exander77/go-imap-specialuse noannouncecapability
 )
