--- ./internal/updater/version.go.orig	2021-10-30 14:03:57.695224491 +0000
+++ ./internal/updater/version.go	2021-10-30 14:04:11.848561016 +0000
@@ -86,5 +86,5 @@
 //  - https://protonmail.com/download/bridge/version_linux.json
 //  - https://protonmail.com/download/ie/version_linux.json
 func (u *Updater) getVersionFileURL() string {
-	return fmt.Sprintf("%v/%v/version_%v.json", Host, u.updateURLName, u.platform)
+	return fmt.Sprintf("%v/%v/version_linux.json", Host, u.updateURLName)
 }
