diff -uNr libXfont2-2.0.1/src/fc/fslibos.h libXfont2-2.0.1.mod/src/fc/fslibos.h
--- libXfont2-2.0.1/src/fc/fslibos.h	2015-09-19 06:18:54.000000000 +0300
+++ libXfont2-2.0.1.mod/src/fc/fslibos.h	2017-11-19 20:23:16.615579918 +0200
@@ -68,11 +68,7 @@
 #     ifdef __OSF1__
 #      define OPEN_MAX 256
 #     else
-#      ifdef NOFILE
-#       define OPEN_MAX NOFILE
-#      else
-#       define OPEN_MAX NOFILES_MAX
-#      endif
+#      define OPEN_MAX 512
 #     endif
 #    endif
 #   endif
