--- a/m4/vapigen.m4
+++ b/m4/vapigen.m4
@@ -82,7 +82,7 @@
 
   AS_CASE([$enable_vala],
     [yes], [
-      VAPIGEN=$PKG_CONFIG_SYSROOT_DIR`$PKG_CONFIG --variable=vapigen $vapigen_pkg_name`
+      VAPIGEN=vapigen
       VAPIGEN_MAKEFILE=$PKG_CONFIG_SYSROOT_DIR`$PKG_CONFIG --variable=datadir $vapigen_pkg_name`/vala/Makefile.vapigen
       AS_IF([test "x$2" = "x"], [
           VAPIGEN_VAPIDIR=$PKG_CONFIG_SYSROOT_DIR`$PKG_CONFIG --variable=vapidir $vapigen_pkg_name`
