diff -uNr lftp-4.8.3/src/lftp_ssl.cc lftp-4.8.3.mod/src/lftp_ssl.cc
--- lftp-4.8.3/src/lftp_ssl.cc	2017-08-10 12:57:37.000000000 +0300
+++ lftp-4.8.3.mod/src/lftp_ssl.cc	2017-12-19 17:06:58.526580533 +0200
@@ -214,6 +214,8 @@
       "/usr/local/ssl/certs/ca-bundle.crt",
       "/etc/apache/ssl.crt/ca-bundle.crt",
       "/usr/share/curl/curl-ca-bundle.crt",
+      "@TERMUX_PREFIX@/etc/tls/cert.pem",
+      "@TERMUX_PREFIX@/etc/tls/ca-bundle.crt",
       0};
    for(int i=0; ca_file_location[i]; i++)
    {
