diff -u -r ../lftp-4.7.1/m4/ax_check_zlib.m4 ./m4/ax_check_zlib.m4
--- ../lftp-4.7.1/m4/ax_check_zlib.m4	2016-02-20 08:57:52.000000000 -0500
+++ ./m4/ax_check_zlib.m4	2016-05-03 13:06:09.166707065 -0400
@@ -70,7 +70,7 @@
 # Handle user hints
 #
 [AC_MSG_CHECKING(if zlib is wanted)
-zlib_places="/usr/local /usr /opt/local /sw"
+zlib_places=""
 AC_ARG_WITH([zlib],
 [  --with-zlib=DIR         root directory path of zlib installation @<:@defaults to
                           /usr/local or /usr if not found in /usr/local@:>@
