--- a/nala/__init__.py
+++ b/nala/__init__.py
@@ -29,7 +29,7 @@
 # Before we import anything else we need to make sure that this is in our path
 # Else apt_pkg will give an import error. Conda is an example of this breaking
 # pylint: disable=wrong-import-position
-PY_PATH = "/usr/lib/python3/dist-packages"
+PY_PATH = "@TERMUX_PREFIX@/lib/python3/dist-packages"
 if PY_PATH not in sys.path:
 	sys.path.append(PY_PATH)
 
