--- a/nala/error.py
+++ b/nala/error.py
@@ -58,7 +58,7 @@
 
 INSTALL_FAILED = _(
 	"{error} Installation has failed.\n"
-	"If you'd like to file a bug report please include '/var/log/nala/dpkg-debug.log'"
+	"If you'd like to file a bug report please include '@TERMUX_PREFIX@/var/log/nala/dpkg-debug.log'"
 )
 NO_PROPER_ERR = _(
 	"{error} python-apt gave us {apt_err} This isn't a proper error as it's empty"
