--- a/nala/constants.py
+++ b/nala/constants.py
@@ -37,13 +37,13 @@
 from nala import _, color
 
 # File Constants
-LICENSE = Path("/usr/share/common-licenses/GPL-3")
+LICENSE = Path("@TERMUX_PREFIX@/share/LICENSES/GPL-3.0.txt")
 """/usr/share/common-licenses/GPL-3"""
-NALA_SOURCES = Path("/etc/apt/sources.list.d/nala-sources.list")
+NALA_SOURCES = Path("@TERMUX_PREFIX@/etc/apt/sources.list.d/nala-sources.list")
 """/etc/apt/sources.list.d/nala-sources.list"""
-NALA_DIR = Path("/var/lib/nala")
+NALA_DIR = Path("@TERMUX_PREFIX@/var/lib/nala")
 """/var/lib/nala"""
-NALA_LOGDIR = Path("/var/log/nala")
+NALA_LOGDIR = Path("@TERMUX_PREFIX@/var/log/nala")
 """/var/log/nala"""
 NALA_DEBUGLOG = NALA_LOGDIR / "nala-debug.log"
 """/var/log/nala/nala.debug.log"""
@@ -51,17 +51,17 @@
 """/var/log/nala/dpkg-debug.log"""
 DPKG_STATUS_LOG = NALA_LOGDIR / "dpkg-status.log"
 """/var/log/nala/dpkg-status.log"""
-NALA_TERM_LOG = Path("/var/log/nala/term.log")
+NALA_TERM_LOG = Path("@TERMUX_PREFIX@/var/log/nala/term.log")
 """/var/log/nala/term.log"""
-NALA_HISTORY = Path("/var/lib/nala/history.json")
+NALA_HISTORY = Path("@TERMUX_PREFIX@/var/lib/nala/history.json")
 """/var/lib/nala/history.json"""
-PACSTALL_METADATA = Path("/var/log/pacstall/metadata")
+PACSTALL_METADATA = Path("@TERMUX_PREFIX@/var/log/pacstall/metadata")
 """/var/log/pacstall/metadata"""
-REBOOT_REQUIRED = Path("/var/run/reboot-required")
+REBOOT_REQUIRED = Path("@TERMUX_PREFIX@/var/run/reboot-required")
 """/var/run/reboot-required"""
-REBOOT_PKGS = Path("/var/run/reboot-required.pkgs")
+REBOOT_PKGS = Path("@TERMUX_PREFIX@/var/run/reboot-required.pkgs")
 """'/var/run/reboot-required.pkgs'"""
-NEED_RESTART = Path("/var/run/needrestart")
+NEED_RESTART = Path("@TERMUX_PREFIX@/var/run/needrestart")
 """/var/run/needrestart"""
 
 # Apt Directories
