--- a/setup.py
+++ b/setup.py
@@ -3,8 +3,6 @@
 from pathlib import Path
 
 import tomli
-from DistUtilsExtra.auto import clean_build_tree
-from DistUtilsExtra.command import build_extra, build_i18n
 from setuptools import setup
 
 PYPROJECT = Path("./pyproject.toml")
@@ -16,11 +14,6 @@
 	version=POETRY["version"],
 	author=POETRY["authors"][0].split()[0],
 	author_email=POETRY["authors"][0].split()[1],
-	cmdclass={
-		"build": build_extra.build_extra,
-		"build_i18n": build_i18n.build_i18n,
-		"clean": clean_build_tree,
-	},
 	license=POETRY["license"],
 	packages=["nala"],
 	entry_points={
