Revert https://github.com/atuinsh/atuin/pull/1249

--- a/atuin/Cargo.toml
+++ b/atuin/Cargo.toml
@@ -33,11 +33,10 @@
 atuin = { path = "/usr/bin/atuin" }
 
 [features]
-default = ["client", "sync", "server", "clipboard", "check-update"]
+default = ["client", "sync", "server", "check-update"]
 client = ["atuin-client"]
 sync = ["atuin-client/sync"]
 server = ["atuin-server", "atuin-server-postgres", "tracing-subscriber"]
-clipboard = ["cli-clipboard"]
 check-update = ["atuin-client/check-update"]
 
 [dependencies]
@@ -75,7 +74,6 @@
 colored = "2.0.4"
 ratatui = "0.25"
 tracing = "0.1"
-cli-clipboard = { version = "0.4.0", optional = true }
 uuid = { workspace = true }
