diff -uNr profanity-0.10.0/src/command/cmd_funcs.c profanity-0.10.0.mod/src/command/cmd_funcs.c
--- profanity-0.10.0/src/command/cmd_funcs.c	2021-01-08 15:35:03.000000000 +0000
+++ profanity-0.10.0.mod/src/command/cmd_funcs.c	2021-02-07 20:14:53.389297396 +0000
@@ -5151,7 +5151,7 @@
 gboolean
 cmd_charset(ProfWin* window, const char* const command, gchar** args)
 {
-    char* codeset = nl_langinfo(CODESET);
+    char* codeset = "UTF-8";
     char* lang = getenv("LANG");
 
     cons_show("Charset information:");
