diff --git a/configure b/configure
index 57c0ba1..24309bc 100755
--- a/configure
+++ b/configure
@@ -11933,7 +11933,7 @@ $as_echo "$as_me: Symlinking Python.framework to $PYTHON_FRAMEWORK" >&6;}
         ln -s $PYTHON_FRAMEWORK Python.framework
 fi
     # Extract the first word of "python-config", so it can be a program name with args.
-set dummy python-config; ac_word=$2
+set dummy python3-config; ac_word=$2
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
 $as_echo_n "checking for $ac_word... " >&6; }
 if ${ac_cv_prog_PYTHON_CONFIG_EXISTS+:} false; then :
