--- a/utils/libcecutil/cec-parse.cpp
+++ b/utils/libcecutil/cec-parse.cpp
@@ -9,6 +9,8 @@
 
 #include <unistd.h>
 
+#include "getsubopt.h"
+
 #include <linux/cec-funcs.h>
 #include "cec-htng-funcs.h"
 #include "cec-log.h"
--- a/utils/rds-ctl/rds-ctl.cpp
+++ b/utils/rds-ctl/rds-ctl.cpp
@@ -24,6 +24,8 @@
 #include <sys/ioctl.h>
 #include <unistd.h>
 
+#include "getsubopt.h"
+
 #include <linux/videodev2.h>
 #include <libv4l2rds.h>
 
--- a/utils/v4l2-compliance/v4l2-compliance.cpp
+++ b/utils/v4l2-compliance/v4l2-compliance.cpp
@@ -30,6 +30,8 @@
 #include <sys/types.h>
 #include <unistd.h>
 
+#include "getsubopt.h"
+
 #include "v4l2-compliance.h"
 #include <media-info.h>
 
--- a/utils/v4l2-ctl/v4l2-ctl.h
+++ b/utils/v4l2-ctl/v4l2-ctl.h
@@ -7,6 +7,8 @@
 #include <config.h>
 #endif
 
+#include "getsubopt.h"
+
 #include <linux/videodev2.h>
 #include <linux/v4l2-subdev.h>
 
--- a/utils/v4l2-dbg/v4l2-dbg.cpp
+++ b/utils/v4l2-dbg/v4l2-dbg.cpp
@@ -37,6 +37,8 @@
 #include <sys/klog.h>
 #endif
 
+#include "getsubopt.h"
+
 #include <linux/videodev2.h>
 
 #include "v4l2-dbg-bttv.h"
