--- a/utils/cec-ctl/cec-ctl.cpp
+++ b/utils/cec-ctl/cec-ctl.cpp
@@ -25,7 +25,7 @@
 #include "cec-log.h"
 #include "cec-parse.h"
 
-#ifdef __ANDROID__
+#ifdef ANDROID
 #include <android-config.h>
 #else
 #include <config.h>
--- a/utils/cec-ctl/cec-pin.cpp
+++ b/utils/cec-ctl/cec-pin.cpp
@@ -8,7 +8,7 @@
 #include <linux/cec.h>
 #include "cec-htng.h"
 
-#ifdef __ANDROID__
+#ifdef ANDROID
 #include <android-config.h>
 #else
 #include <config.h>
