diff -u -r ../inotify-tools-3.20.1/src/Makefile.am ./src/Makefile.am
--- ../inotify-tools-3.20.1/src/Makefile.am	2018-01-06 12:10:51.000000000 +0100
+++ ./src/Makefile.am	2018-05-19 00:38:08.962826874 +0200
@@ -2,7 +2,7 @@
 inotifywait_SOURCES = inotifywait.c common.c common.h
 inotifywatch_SOURCES = inotifywatch.c common.c common.h
 
-AM_CFLAGS = -Wall -Werror -Wpointer-arith -std=c99 -I../libinotifytools/src -L../libinotifytools/src
+AM_CFLAGS = -Wall -Wpointer-arith -std=c99 -I../libinotifytools/src -L../libinotifytools/src
 AM_CPPFLAGS = -I$(top_srcdir)/libinotifytools/src
 LDADD = ../libinotifytools/src/libinotifytools.la
 
