diff -uNr inotify-tools-3.20.2/src/Makefile.am inotify-tools-3.20.2.mod/src/Makefile.am
--- inotify-tools-3.20.2/src/Makefile.am	2020-01-30 02:04:48.000000000 +0200
+++ inotify-tools-3.20.2.mod/src/Makefile.am	2020-01-30 21:18:17.734398158 +0200
@@ -2,7 +2,7 @@
 inotifywait_SOURCES = inotifywait.c common.c common.h
 inotifywatch_SOURCES = inotifywatch.c common.c common.h
 
-AM_CFLAGS = -Wall -Wextra -Wpointer-arith -Werror -std=c99 -I../libinotifytools/src
+AM_CFLAGS = -Wall -Wextra -Wpointer-arith -std=c99 -I../libinotifytools/src
 AM_CPPFLAGS = -I$(top_srcdir)/libinotifytools/src
 LDADD = ../libinotifytools/src/libinotifytools.la
 
