From 30474bd5928950d2c88f1b6cc4efac271e900635 Mon Sep 17 00:00:00 2001
From: Princess Grace <grace@princess.software>
Date: Mon, 30 Mar 2026 11:55:12 -0700
Subject: [PATCH] sbindir contains the full path.

---
 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index 97378e5..12fee7c 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -1,4 +1,4 @@
-targetdir = $(DESTDIR)@prefix@@sbindir@
+targetdir = $(DESTDIR)@sbindir@
 man8dir = $(DESTDIR)@prefix@@mandir@/man8
 datarootdir =
 
-- 
2.47.3

