diff -u -r ../pure-ftpd-1.0.45/src/ftpd_p.h ./src/ftpd_p.h
--- ../pure-ftpd-1.0.45/src/ftpd_p.h	2017-01-09 20:51:00.000000000 +0100
+++ ./src/ftpd_p.h	2017-02-24 23:16:25.721476594 +0100
@@ -350,7 +350,7 @@
 # define getpwnam(A) fakegetpwnam(A)
 # define getpwuid(A) fakegetpwnam(NULL)
 #endif
-#define NON_ROOT_ANON_DIR "/ftp"
+#define NON_ROOT_ANON_DIR "@TERMUX_HOME@"
 
 #ifdef PROBE_RANDOM_AT_RUNTIME
 static const char *random_device;
