diff -u -r ../pure-ftpd-1.0.47/src/ftpd.h ./src/ftpd.h
--- ../pure-ftpd-1.0.47/src/ftpd.h	2017-08-19 01:47:43.000000000 +0200
+++ ./src/ftpd.h	2017-11-13 00:02:42.291967297 +0100
@@ -416,7 +416,7 @@
 
 #ifdef WITH_TLS
 # ifndef TLS_CONFDIR
-#  define TLS_CONFDIR "/etc/ssl/private"
+#  define TLS_CONFDIR CONFDIR "/ssl/private"
 # endif
 # ifndef TLS_CERTIFICATE_FILE
 #  define TLS_CERTIFICATE_FILE TLS_CONFDIR "/pure-ftpd.pem"
