diff -uNr xarchiver-0.5.4.13/src/xdgmime/xdgmime.c xarchiver-0.5.4.13.mod/src/xdgmime/xdgmime.c
--- xarchiver-0.5.4.13/src/xdgmime/xdgmime.c	2018-03-13 12:22:11.000000000 +0200
+++ xarchiver-0.5.4.13.mod/src/xdgmime/xdgmime.c	2018-06-23 22:34:22.195187606 +0300
@@ -259,7 +259,7 @@
 
   xdg_data_dirs = getenv ("XDG_DATA_DIRS");
   if (xdg_data_dirs == NULL)
-    xdg_data_dirs = "/usr/local/share/:/usr/share/";
+    xdg_data_dirs = "@TERMUX_PREFIX@/share/:@TERMUX_PREFIX@/local/share/";
 
   ptr = xdg_data_dirs;
 
