diff -uNr qemu-2.12.0/contrib/ivshmem-server/main.c qemu-2.12.0.mod/contrib/ivshmem-server/main.c
--- qemu-2.12.0/contrib/ivshmem-server/main.c	2018-04-24 19:30:46.000000000 +0300
+++ qemu-2.12.0.mod/contrib/ivshmem-server/main.c	2018-06-25 14:12:36.683968721 +0300
@@ -14,8 +14,8 @@
 
 #define IVSHMEM_SERVER_DEFAULT_VERBOSE        0
 #define IVSHMEM_SERVER_DEFAULT_FOREGROUND     0
-#define IVSHMEM_SERVER_DEFAULT_PID_FILE       "/var/run/ivshmem-server.pid"
-#define IVSHMEM_SERVER_DEFAULT_UNIX_SOCK_PATH "/tmp/ivshmem_socket"
+#define IVSHMEM_SERVER_DEFAULT_PID_FILE       "@TERMUX_PREFIX@/var/run/ivshmem-server.pid"
+#define IVSHMEM_SERVER_DEFAULT_UNIX_SOCK_PATH "@TERMUX_PREFIX@/tmp/ivshmem_socket"
 #define IVSHMEM_SERVER_DEFAULT_SHM_PATH       "ivshmem"
 #define IVSHMEM_SERVER_DEFAULT_SHM_SIZE       (4*1024*1024)
 #define IVSHMEM_SERVER_DEFAULT_N_VECTORS      1
