diff -uNr qemu-2.12.0/contrib/ivshmem-client/main.c qemu-2.12.0.mod/contrib/ivshmem-client/main.c
--- qemu-2.12.0/contrib/ivshmem-client/main.c	2018-04-24 19:30:46.000000000 +0300
+++ qemu-2.12.0.mod/contrib/ivshmem-client/main.c	2018-06-25 14:12:24.330634752 +0300
@@ -12,7 +12,7 @@
 #include "ivshmem-client.h"
 
 #define IVSHMEM_CLIENT_DEFAULT_VERBOSE        0
-#define IVSHMEM_CLIENT_DEFAULT_UNIX_SOCK_PATH "/tmp/ivshmem_socket"
+#define IVSHMEM_CLIENT_DEFAULT_UNIX_SOCK_PATH "@TERMUX_PREFIX@/tmp/ivshmem_socket"
 
 typedef struct IvshmemClientArgs {
     bool verbose;
