diff -uNr qemu-2.12.0/util/qemu-sockets.c qemu-2.12.0.mod/util/qemu-sockets.c
--- qemu-2.12.0/util/qemu-sockets.c	2018-04-24 19:30:47.000000000 +0300
+++ qemu-2.12.0.mod/util/qemu-sockets.c	2018-06-25 14:11:33.290632124 +0300
@@ -841,7 +841,7 @@
         path = saddr->path;
     } else {
         const char *tmpdir = getenv("TMPDIR");
-        tmpdir = tmpdir ? tmpdir : "/tmp";
+        tmpdir = tmpdir ? tmpdir : "@TERMUX_PREFIX@/tmp";
         path = pathbuf = g_strdup_printf("%s/qemu-socket-XXXXXX", tmpdir);
     }
 
