diff -uNr qemu-2.12.0/hw/usb/ccid-card-emulated.c qemu-2.12.0.mod/hw/usb/ccid-card-emulated.c
--- qemu-2.12.0/hw/usb/ccid-card-emulated.c	2018-04-24 19:30:47.000000000 +0300
+++ qemu-2.12.0.mod/hw/usb/ccid-card-emulated.c	2018-06-25 14:11:33.227298788 +0300
@@ -412,7 +412,7 @@
     return 0;
 }
 
-#define CERTIFICATES_DEFAULT_DB "/etc/pki/nssdb"
+#define CERTIFICATES_DEFAULT_DB "@TERMUX_PREFIX@/etc/pki/nssdb"
 #define CERTIFICATES_ARGS_TEMPLATE\
     "db=\"%s\" use_hw=no soft=(,Virtual Reader,CAC,,%s,%s,%s)"
 
