diff -uNr qemu-2.12.0/linux-user/strace.c qemu-2.12.0.mod/linux-user/strace.c
--- qemu-2.12.0/linux-user/strace.c	2018-04-24 19:30:47.000000000 +0300
+++ qemu-2.12.0.mod/linux-user/strace.c	2018-06-25 14:11:33.243965455 +0300
@@ -1,7 +1,7 @@
 #include "qemu/osdep.h"
 #include <sys/ipc.h>
 #include <sys/msg.h>
-#include <sys/sem.h>
+#include <linux/sem.h>
 #include <sys/shm.h>
 #include <sys/select.h>
 #include <sys/mount.h>
