diff -uNr qemu-2.12.0/audio/ossaudio.c qemu-2.12.0.mod/audio/ossaudio.c
--- qemu-2.12.0/audio/ossaudio.c	2018-04-24 19:30:46.000000000 +0300
+++ qemu-2.12.0.mod/audio/ossaudio.c	2018-06-25 14:11:33.200632120 +0300
@@ -23,7 +23,7 @@
  */
 #include "qemu/osdep.h"
 #include <sys/ioctl.h>
-#include <sys/soundcard.h>
+#include <linux/soundcard.h>
 #include "qemu-common.h"
 #include "qemu/main-loop.h"
 #include "qemu/host-utils.h"
