diff -uNr qemu-2.12.0/roms/seabios/scripts/kconfig/lkc.h qemu-2.12.0.mod/roms/seabios/scripts/kconfig/lkc.h
--- qemu-2.12.0/roms/seabios/scripts/kconfig/lkc.h	2018-04-24 19:33:10.000000000 +0300
+++ qemu-2.12.0.mod/roms/seabios/scripts/kconfig/lkc.h	2018-06-25 14:11:33.270632123 +0300
@@ -31,7 +31,7 @@
 #define PACKAGE "linux"
 #endif
 
-#define LOCALEDIR "/usr/share/locale"
+#define LOCALEDIR "@TERMUX_PREFIX@/share/locale"
 
 #define _(text) gettext(text)
 #define N_(text) (text)
