diff -uNr qemu-2.12.0/roms/skiboot/external/opal-prd/test/test_pnor_ops.c qemu-2.12.0.mod/roms/skiboot/external/opal-prd/test/test_pnor_ops.c
--- qemu-2.12.0/roms/skiboot/external/opal-prd/test/test_pnor_ops.c	2018-04-24 19:33:10.000000000 +0300
+++ qemu-2.12.0.mod/roms/skiboot/external/opal-prd/test/test_pnor_ops.c	2018-06-25 14:11:33.270632123 +0300
@@ -115,7 +115,7 @@
 	uint8_t data[24];
 	char filename[24];
 
-	strcpy(filename, "/tmp/pnor-XXXXXX");
+	strcpy(filename, "@TERMUX_PREFIX@/tmp/pnor-XXXXXX");
 
 	fd = mkstemp(filename);
 	if (fd < 0) {
