--- a/src/sundials/sundials_hashmap.h
+++ b/src/sundials/sundials_hashmap.h
@@ -20,11 +20,12 @@
 #ifndef _SUNDIALS_HASHMAP_H
 #define _SUNDIALS_HASHMAP_H
 
+#include <stdint.h>
 #include <stdlib.h>
 #include <string.h>
 
-static const unsigned long HASH_PRIME        = 14695981039346656037U;
-static const unsigned long HASH_OFFSET_BASIS = 1099511628211U;
+static const uint64_t HASH_PRIME        = 14695981039346656037U;
+static const uint64_t HASH_OFFSET_BASIS = 1099511628211U;
 
 /*
   For a nice discussion on popular hashing algorithms see:
@@ -33,9 +34,9 @@
   This is a 64-bit implementation of the 'a' modification of the
   Fowler–Noll–Vo hash (i.e., FNV1-a).
  */
-static unsigned long fnv1a_hash(const char* str)
+static uint64_t fnv1a_hash(const char* str)
 {
-  unsigned long hash = HASH_OFFSET_BASIS;
+  uint64_t hash = HASH_OFFSET_BASIS;
   char c;
   while ((c = *str++))
   {
