--- ../Makefile.orig	2019-11-01 17:25:45.000000000 +0100
+++ ./Makefile	2020-02-23 21:55:43.646881105 +0100
@@ -11,11 +11,9 @@
 
 # User-defined variables
 
-Q		:= @
-PREFIX		:= /usr/local
 bindir		:= ${PREFIX}/bin
 mandir		:= ${PREFIX}/man
-STRIP		:= -s
+STRIP		:=
 BINMODE		:= 555
 MANMODE		:= 444
 CHECKPATCH	:= ../linux/scripts/checkpatch.pl
@@ -31,13 +29,9 @@
 
 WARNFLAGS	:= -Wall
 
-# Overridable CFLAGS
-CFLAGS		:= -g
 # Non-overridable CFLAGS
 REALCFLAGS	:= ${CFLAGS} ${WARNFLAGS} -std=c99 -D_POSIX_C_SOURCE=200809L \
 		   -Iinclude -DBUILD_VERSION_STRING=\"${VERSION_STRING}\"
-# Overridable LDFLAGS
-LDFLAGS		:=
 # Non-overridable LDFLAGS
 REALLDFLAGS	:= ${LDFLAGS} ${WARNFLAGS}
 
