diff -uNr rgbds/Makefile rgbds.mod/Makefile
--- rgbds/Makefile	2020-04-04 12:01:55.000000000 +0300
+++ rgbds.mod/Makefile	2020-04-14 23:07:08.846835851 +0300
@@ -11,11 +11,9 @@
 
 # User-defined variables
 
-Q		:= @
-PREFIX		:= /usr/local
 bindir		:= ${PREFIX}/bin
 mandir		:= ${PREFIX}/man
-STRIP		:= -s
+STRIP		:=
 BINMODE		:= 555
 MANMODE		:= 444
 CHECKPATCH	:= ../linux/scripts/checkpatch.pl
@@ -31,13 +29,9 @@
 
 WARNFLAGS	:= -Wall
 
-# Overridable CFLAGS
-CFLAGS		:= -O3
 # Non-overridable CFLAGS
 REALCFLAGS	:= ${CFLAGS} ${WARNFLAGS} -std=c11 -D_POSIX_C_SOURCE=200809L \
 		   -D_DEFAULT_SOURCE -Iinclude
-# Overridable LDFLAGS
-LDFLAGS		:=
 # Non-overridable LDFLAGS
 REALLDFLAGS	:= ${LDFLAGS} ${WARNFLAGS} \
 		   -DBUILD_VERSION_STRING=\"${VERSION_STRING}\"
