diff -uNr rgbds/src/extern/getopt.c rgbds.mod/src/extern/getopt.c
--- rgbds/src/extern/getopt.c	2021-03-20 00:41:50.000000000 +0000
+++ rgbds.mod/src/extern/getopt.c	2021-05-05 14:58:53.925498951 +0000
@@ -208,7 +208,7 @@
 			}
 			cnt++;
 		}
-		if (cnt == 1 && longonly && arg - start == mblen(start, MB_LEN_MAX)) {
+		if (cnt == 1 && longonly && arg - start == mbtowc(0, start, MB_LEN_MAX)) {
 			int l = arg - start;
 
 			for (i = 0; optstring[i]; i++) {
