diff -uNr gtk-3.24.8/autogen.sh gtk-3.24.8.mod/autogen.sh
--- gtk-3.24.8/autogen.sh	2019-04-11 00:12:29.000000000 +0300
+++ gtk-3.24.8.mod/autogen.sh	2019-05-20 01:01:58.516541976 +0300
@@ -23,12 +23,6 @@
         exit 1
 fi
 
-pkg-config --print-errors gobject-introspection-1.0
-if [ $? != 0 ]; then
-	echo "You probably need to install 'libgirepository1.0-dev'"
-	exit 1
-fi
-
 # README and INSTALL are required by automake, but may be deleted by clean
 # up rules. to get automake to work, simply touch these here, they will be
 # regenerated from their corresponding *.in files by ./configure anyway.
