diff -uNr gtk-3.24.8/autogen.sh gtk-3.24.8.mod/autogen.sh
--- gtk-3.24.8/autogen.sh	2019-04-11 00:12:29.000000000 +0300
+++ gtk-3.24.8.mod/autogen.sh	2019-05-20 01:01:58.516541976 +0300
@@ -23,12 +23,6 @@
         exit 1
 fi
 
-pkg-config --print-errors gobject-introspection-1.0
-if [ $? != 0 ]; then
-	echo "You probably need to install 'libgirepository1.0-dev'"
-	exit 1
-fi
-
 # README and INSTALL are required by automake, but may be deleted by clean
 # up rules. to get automake to work, simply touch these here, they will be
 # regenerated from their corresponding *.in files by ./configure anyway.
diff -uNr gtk-3.24.8/build/win32/Makefile.am gtk-3.24.8.mod/build/win32/Makefile.am
--- gtk-3.24.8/build/win32/Makefile.am	2019-04-11 00:12:29.000000000 +0300
+++ gtk-3.24.8.mod/build/win32/Makefile.am	2019-05-20 01:01:58.516541976 +0300
@@ -1,30 +1,7 @@
 include $(top_srcdir)/Makefile.decl
 
-if HAVE_INTROSPECTION
-GENERATED_ITEMS = \
-	introspection.body.mak	\
-	Gdk_3_0_gir_list		\
-	GdkWin32_3_0_gir_list	\
-	Gtk_3_0_gir_list
-
-MSVC_INTROSPECTION_INTERMEDIATE_FILES = Gdk-3.0.gir.msvc.introspect GdkWin32-3.0.gir.msvc.introspect Gtk-3.0.gir.msvc.introspect
-
-introspection.body.mak: $(MSVC_INTROSPECTION_INTERMEDIATE_FILES)
-	-$(RM) introspection.body.mak
-	for F in `ls *.msvc.introspect`; do \
-		case $$F in \
-			*)	cat $(top_builddir)/build/win32/$$F >>introspection.body.mak \
-			;; \
-		esac; \
-	done
-	$(RM) $(MSVC_INTROSPECTION_INTERMEDIATE_FILES)
-
-DISTCLEANFILES = $(GENERATED_ITEMS)
-
-else
 GENERATED_ITEMS =
 DISTCLEANFILES =
-endif
 
 SUBDIRS =	\
 	vs9 	\
diff -uNr gtk-3.24.8/configure.ac gtk-3.24.8.mod/configure.ac
--- gtk-3.24.8/configure.ac	2019-04-11 00:12:29.000000000 +0300
+++ gtk-3.24.8.mod/configure.ac	2019-05-20 01:01:58.517541982 +0300
@@ -1705,12 +1705,6 @@
 GLIB_GSETTINGS
 
 ##################################################
-# GObject introspection
-##################################################
-
-GOBJECT_INTROSPECTION_CHECK(introspection_required_version)
-
-##################################################
 # colord module
 #################################################
 
diff -uNr gtk-3.24.8/gdk/Makefile.am gtk-3.24.8.mod/gdk/Makefile.am
--- gtk-3.24.8/gdk/Makefile.am	2019-04-11 00:12:29.000000000 +0300
+++ gtk-3.24.8.mod/gdk/Makefile.am	2019-05-20 01:03:36.915120535 +0300
@@ -239,145 +239,6 @@
 libgdk_3_la_LIBADD += mir/libgdk-mir.la
 endif
 
-if HAVE_INTROSPECTION
-
-introspection_files = 		\
-	$(filter-out gdkkeysyms-compat.h, $(gdk_h_sources))	\
-	$(gdk_c_sources)	\
-	$(gdk_built_sources)
-
-Gdk-3.0.gir: libgdk-3.la Makefile
-Gdk_3_0_gir_SCANNERFLAGS = 	\
-	--c-include="gdk/gdk.h"
-Gdk_3_0_gir_INCLUDES = Gio-2.0 GdkPixbuf-2.0 Pango-1.0 cairo-1.0
-Gdk_3_0_gir_LIBS = libgdk-3.la
-Gdk_3_0_gir_FILES = $(introspection_files)
-Gdk_3_0_gir_CFLAGS = $(AM_CPPFLAGS)
-Gdk_3_0_gir_EXPORT_PACKAGES = gdk-3.0
-INTROSPECTION_GIRS += Gdk-3.0.gir
-if OS_WIN32
-Gdk_3_0_gir_CFLAGS += -I$(srcdir)/win32 -I$(srcdir)
-endif
-if USE_X11
-x11_introspection_files = 		\
-	x11/gdkapplaunchcontext-x11.c	\
-	x11/gdkasync.c			\
-	x11/gdkcursor-x11.c		\
-	x11/gdkdevice-core-x11.c	\
-	x11/gdkdevicemanager-core-x11.c	\
-	x11/gdkdevicemanager-x11.c	\
-	x11/gdkdevicemanager-xi2.c	\
-	x11/gdkdevice-xi2.c		\
-	x11/gdkdisplay-x11.c		\
-	x11/gdkdisplaymanager-x11.c	\
-	x11/gdkdnd-x11.c		\
-	x11/gdkeventsource.c		\
-	x11/gdkeventtranslator.c	\
-	x11/gdkgeometry-x11.c		\
-	x11/gdkglcontext-x11.c		\
-	x11/gdkkeys-x11.c		\
-	x11/gdkmain-x11.c		\
-	x11/gdkmonitor-x11.c		\
-	x11/gdkproperty-x11.c		\
-	x11/gdkscreen-x11.c		\
-	x11/gdkselection-x11.c		\
-	x11/gdksettings.c		\
-	x11/gdktestutils-x11.c		\
-	x11/gdkvisual-x11.c		\
-	x11/gdkwindow-x11.c		\
-	x11/gdkxftdefaults.c		\
-	x11/gdkxid.c			\
-	x11/xsettings-client.c		\
-	x11/gdkx.h			\
-	x11/gdkx11applaunchcontext.h	\
-	x11/gdkx11cursor.h		\
-	x11/gdkx11device.h		\
-	x11/gdkx11device-core.h		\
-	x11/gdkx11device-xi2.h		\
-	x11/gdkx11devicemanager.h	\
-	x11/gdkx11devicemanager-core.h	\
-	x11/gdkx11devicemanager-xi2.h	\
-	x11/gdkx11display.h		\
-	x11/gdkx11displaymanager.h	\
-	x11/gdkx11dnd.h			\
-	x11/gdkx11glcontext.h		\
-	x11/gdkx11keys.h		\
-	x11/gdkx11monitor.h		\
-	x11/gdkx11property.h		\
-	x11/gdkx11screen.h		\
-	x11/gdkx11selection.h		\
-	x11/gdkx11utils.h		\
-	x11/gdkx11visual.h		\
-	x11/gdkx11window.h
-
-GdkX11-3.0.gir: libgdk-3.la Gdk-3.0.gir Makefile
-GdkX11_3_0_gir_SCANNERFLAGS = 		\
-	--identifier-prefix=Gdk		\
-	--c-include="gdk/gdkx.h"	\
-	--include-uninstalled=$(top_builddir)/gdk/Gdk-3.0.gir
-GdkX11_3_0_gir_INCLUDES = Gio-2.0 GdkPixbuf-2.0 Pango-1.0 xlib-2.0
-GdkX11_3_0_gir_LIBS = libgdk-3.la
-GdkX11_3_0_gir_FILES = $(x11_introspection_files)
-GdkX11_3_0_gir_CFLAGS = $(AM_CPPFLAGS) -L$(top_builddir)/gdk
-GdkX11_3_0_gir_EXPORT_PACKAGES = gdk-x11-3.0
-INTROSPECTION_GIRS += GdkX11-3.0.gir
-
-endif # USE_X11
-
-w32_introspection_files =		\
-	win32/gdkcursor-win32.c		\
-	win32/gdkdevicemanager-win32.c	\
-	win32/gdkdevice-virtual.c	\
-	win32/gdkdevice-win32.c		\
-	win32/gdkdevice-wintab.c	\
-	win32/gdkdisplaymanager-win32.c	\
-	win32/gdkdisplay-win32.c	\
-	win32/gdkdnd-win32.c		\
-	win32/gdkevents-win32.c		\
-	win32/gdkgeometry-win32.c	\
-	win32/gdkglobals-win32.c	\
-	win32/gdkkeys-win32.c		\
-	win32/gdkmain-win32.c		\
-	win32/gdkproperty-win32.c	\
-	win32/gdkscreen-win32.c		\
-	win32/gdkselection-win32.c	\
-	win32/gdktestutils-win32.c	\
-	win32/gdkwin32.h		\
-	win32/gdkwin32cursor.h		\
-	win32/gdkwin32display.h		\
-	win32/gdkwin32displaymanager.h	\
-	win32/gdkwin32dnd.h		\
-	win32/gdkwin32id.c		\
-	win32/gdkwin32keys.h		\
-	win32/gdkwin32misc.h		\
-	win32/gdkwin32screen.h		\
-	win32/gdkwin32window.h		\
-	win32/gdkwindow-win32.c
-
-if OS_WIN32
-GdkWin32-3.0.gir: libgdk-3.la Gdk-3.0.gir Makefile
-GdkWin32_3_0_gir_SCANNERFLAGS = \
-	--identifier-prefix=Gdk		\
-	--c-include="gdk/gdkwin32.h"	\
-	--include-uninstalled=$(top_builddir)/gdk/Gdk-3.0.gir
-GdkWin32_3_0_gir_INCLUDES = Gio-2.0 GdkPixbuf-2.0 Pango-1.0
-GdkWin32_3_0_gir_LIBS = libgdk-3.la
-GdkWin32_3_0_gir_FILES = $(w32_introspection_files)
-GdkWin32_3_0_gir_CFLAGS = $(AM_CPPFLAGS) -L$(top_builddir)/gdk -I$(srcdir)/win32 -I$(srcdir)
-INTROSPECTION_GIRS += GdkWin32-3.0.gir
-
-endif # OS_WIN32
-
-girdir = $(datadir)/gir-1.0
-gir_DATA = $(INTROSPECTION_GIRS)
-
-typelibsdir = $(libdir)/girepository-1.0
-typelibs_DATA = $(INTROSPECTION_GIRS:.gir=.typelib)
-
-CLEANFILES += $(gir_DATA) $(typelibs_DATA)
-
-endif # HAVE_INTROSPECTION
-
 if OS_WIN32
 gdk.def: libgdk-3.la
 	echo "LIBRARY libgdk-$(GTK_MAJOR_VERSION)-@LT_CURRENT_MINUS_AGE@" >$@
@@ -510,41 +371,7 @@
 
 include $(top_srcdir)/build/Makefile.msvcproj
 
-if HAVE_INTROSPECTION
-# Introspection Items for MSVC
-MSVC_INTROSPECT_GIRS = Gdk-3.0.gir GdkWin32-3.0.gir
-
-BASE_MSVC_GIR_CFLAGS =			\
-	$(GDK_CFLAGS_DEFINES)		\
-	-I../.. -I../../gdk -I.../../gdk/win32
-
-INTROSPECTION_INTERMEDIATE_ITEMS = \
-	$(top_builddir)/build/win32/Gdk-3.0.gir.msvc.introspect	\
-	$(top_builddir)/build/win32/Gdk_3_0_gir_list			\
-	$(top_builddir)/build/win32/GdkWin32-3.0.gir.msvc.introspect	\
-	$(top_builddir)/build/win32/GdkWin32_3_0_gir_list
-
-Gdk_3_0_gir_MSVC_FILES = $(introspection_files)
-Gdk_3_0_gir_MSVC_EXPORT_PACKAGES = $(Gdk_3_0_gir_EXPORT_PACKAGES)
-Gdk_3_0_gir_MSVC_INCLUDE_GIRS = $(Gdk_3_0_gir_INCLUDES)
-Gdk_3_0_gir_MSVC_LIBS = gdk-3.0
-Gdk_3_0_gir_MSVC_CFLAGS = $(BASE_MSVC_GIR_CFLAGS)
-Gdk_3_0_gir_MSVC_SCANNERFLAGS = $(Gdk_3_0_gir_SCANNERFLAGS)
-
-GdkWin32_3_0_gir_MSVC_FILES = $(w32_introspection_files)
-GdkWin32_3_0_gir_MSVC_INCLUDE_GIRS = $(GdkWin32_3_0_gir_INCLUDES)
-GdkWin32_3_0_gir_MSVC_LIBS = $(Gdk_3_0_gir_MSVC_LIBS)
-GdkWin32_3_0_gir_MSVC_CFLAGS = $(BASE_MSVC_GIR_CFLAGS)
-GdkWin32_3_0_gir_MSVC_SCANNERFLAGS =	\
-	--identifier-prefix=Gdk		\
-	--c-include="gdk/gdkwin32.h"	\
-	--include-uninstalled='./vs$$$$(VSVER)/$$$$(CFG)/$$$$(PLAT)/bin/Gdk-3.0.gir'
-
-include $(top_srcdir)/build/Makefile.msvc-introspection
-
-else
 INTROSPECTION_INTERMEDIATE_ITEMS =
-endif
 
 dist-hook: \
 	$(top_builddir)/build/win32/vs9/gdk-3.vcproj	\
diff -uNr gtk-3.24.8/gtk/Makefile.am gtk-3.24.8.mod/gtk/Makefile.am
--- gtk-3.24.8/gtk/Makefile.am	2019-04-11 00:12:29.000000000 +0300
+++ gtk-3.24.8.mod/gtk/Makefile.am	2019-05-20 01:02:02.665566760 +0300
@@ -1523,56 +1523,8 @@
 
 include $(top_srcdir)/build/Makefile.msvcproj
 
-if HAVE_INTROSPECTION
-# Introspection Items for MSVC
-MSVC_INTROSPECT_GIRS = Gtk-3.0.gir
-
-GTK_MSVC_GIR_CFLAGS =						\
-	$(GTK_BASE_CFLAGS_DEFINES)				\
-	-DGTK_LIBDIR=\\\"/dummy/lib\\\"				\
-	-DGTK_DATADIR=\\\"/dummy/share\\\"			\
-	-DGTK_DATA_PREFIX=\\\"/dummy\\\"			\
-	-DGTK_SYSCONFDIR=\\\"/dummy/etc\\\"			\
-	-DGTK_HOST=\\\"'$$$$(AT_PLAT)'-pc-vs'$$$$(VSVER)'\\\"	\
-	-DGTK_PRINT_BACKENDS=\\\"file\\\"			\
-	-DINCLUDE_IM_am_et			\
-	-DINCLUDE_IM_cedilla			\
-	-DINCLUDE_IM_cyrillic_translit		\
-	-DINCLUDE_IM_ime			\
-	-DINCLUDE_IM_inuktitu			\
-	-DINCLUDE_IM_ipa			\
-	-DINCLUDE_IM_multipress			\
-	-DINCLUDE_IM_thai			\
-	-DINCLUDE_IM_ti_er			\
-	-DINCLUDE_IM_ti_et			\
-	-DINCLUDE_IM_viqr			\
-	-DGTK_TEXT_USE_INTERNAL_UNSUPPORTED_API	\
-	-I../.. -I../../gtk -I../../gdk
-
-INTROSPECTION_INTERMEDIATE_ITEMS = \
-	$(top_builddir)/build/win32/Gtk-3.0.gir.msvc.introspect		\
-	$(top_builddir)/build/win32/Gtk_3_0_gir_list
-
-Gtk_3_0_gir_MSVC_FILES =		\
-	$(introspected_pub_headers)	\
-	$(gtk_base_c_sources)		\
-	$(gtk_os_win32_c_sources)	\
-	gtktypebuiltins.h		\
-	gtktypebuiltins.c
-
-Gtk_3_0_gir_MSVC_EXPORT_PACKAGES = $(Gtk_3_0_gir_EXPORT_PACKAGES)
-Gtk_3_0_gir_MSVC_INCLUDE_GIRS = Atk-1.0
-Gtk_3_0_gir_MSVC_LIBS = gtk-3.0 gdk-3.0
-Gtk_3_0_gir_MSVC_CFLAGS = $(GTK_MSVC_GIR_CFLAGS)
-Gtk_3_0_gir_MSVC_SCANNERFLAGS = --warn-all --include-uninstalled='./vs$$$$(VSVER)/$$$$(CFG)/$$$$(PLAT)/bin/Gdk-3.0.gir'
-
-include $(top_srcdir)/build/Makefile.msvc-introspection
-
-else
 INTROSPECTION_INTERMEDIATE_ITEMS =
 
-endif
-
 dist-hook: \
 	$(top_builddir)/build/win32/vs9/gtk-3.vcproj	\
 	$(top_builddir)/build/win32/vs9/gtk-3.headers	\
@@ -1596,58 +1548,6 @@
 	  rm -f $(MAINTAINERCLEANFILES); \
 	fi
 
-if HAVE_INTROSPECTION
-introspected_pub_headers = $(filter-out %private.h gtktextdisplay.h gtktextlayout.h gtkx.h, $(gtkinclude_HEADERS) $(a11yinclude_HEADERS) $(deprecatedinclude_HEADERS))
-
-introspection_files = \
-    $(introspected_pub_headers) \
-    $(filter-out %win32.c, $(gtk_base_c_sources)) \
-    gtkprintoperation-unix.c \
-    gtktypebuiltins.h \
-    gtktypebuiltins.c
-
-if USE_X11
-introspection_files += \
-    gtksocket.c \
-    gtkplug.c
-endif
-
-include $(INTROSPECTION_MAKEFILE)
-INTROSPECTION_SCANNER_ENV = CC="$(CC)"
-INTROSPECTION_COMPILER_ARGS = \
-	--includedir=$(srcdir) \
-	--includedir=. \
-	--includedir=../gdk
-
-Gtk-3.0.gir: $(INTROSPECTION_SCANNER) libgtk-3.la $(top_builddir)/gdk/Gdk-3.0.gir Makefile
-Gtk_3_0_gir_SCANNERFLAGS = 			\
-	--warn-all				\
-	--add-include-path=$(top_builddir)/gdk	\
-	--include-uninstalled=$(top_builddir)/gdk/Gdk-3.0.gir
-Gtk_3_0_gir_INCLUDES = Atk-1.0
-Gtk_3_0_gir_C_INCLUDES = gtk/gtk.h gtk/gtk-a11y.h
-if USE_X11
-Gtk_3_0_gir_SCANNERFLAGS += --add-include-path=$(top_builddir)/gdk/x11
-Gtk_3_0_gir_INCLUDES += xlib-2.0
-Gtk_3_0_gir_C_INCLUDES += gtk/gtkx.h
-endif
-Gtk_3_0_gir_CFLAGS = \
-		$(AM_CPPFLAGS) \
-		-DGTK_TEXT_USE_INTERNAL_UNSUPPORTED_API
-Gtk_3_0_gir_FILES = $(introspection_files)
-Gtk_3_0_gir_LIBS = libgtk-3.la $(top_builddir)/gdk/libgdk-3.la
-Gtk_3_0_gir_EXPORT_PACKAGES = gtk+-3.0
-INTROSPECTION_GIRS = Gtk-3.0.gir
-
-girdir = $(datadir)/gir-1.0
-gir_DATA = $(INTROSPECTION_GIRS)
-
-typelibsdir = $(libdir)/girepository-1.0
-typelibs_DATA = $(INTROSPECTION_GIRS:.gir=.typelib)
-
-CLEANFILES += $(gir_DATA) $(typelibs_DATA)
-endif
-
 #
 # Installed tools
 #
