diff -uNr gtk-3.22.30/demos/gtk-demo/Makefile.am gtk-3.22.30.mod/demos/gtk-demo/Makefile.am
--- gtk-3.22.30/demos/gtk-demo/Makefile.am	2018-04-16 21:06:02.000000000 +0300
+++ gtk-3.22.30.mod/demos/gtk-demo/Makefile.am	2018-06-22 20:55:46.621292924 +0300
@@ -176,7 +176,7 @@
 dist_appsicon48_DATA = data/48x48/gtk3-demo.png data/48x48/gtk3-demo-symbolic.symbolic.png
 dist_appsicon256_DATA = data/256x256/gtk3-demo.png data/256x256/gtk3-demo-symbolic.symbolic.png
 
-update_icon_cache = $(top_builddir)/gtk/gtk-update-icon-cache$(EXEEXT) --ignore-theme-index --force
+update_icon_cache = true
 
 install-data-hook: install-update-icon-cache
 uninstall-hook: uninstall-update-icon-cache
diff -uNr gtk-3.22.30/demos/widget-factory/Makefile.am gtk-3.22.30.mod/demos/widget-factory/Makefile.am
--- gtk-3.22.30/demos/widget-factory/Makefile.am	2018-04-16 21:06:02.000000000 +0300
+++ gtk-3.22.30.mod/demos/widget-factory/Makefile.am	2018-06-22 20:55:46.624626257 +0300
@@ -47,7 +47,7 @@
 dist_appsicon48_DATA = data/48x48/gtk3-widget-factory.png data/48x48/gtk3-widget-factory-symbolic.symbolic.png
 dist_appsicon256_DATA = data/256x256/gtk3-widget-factory.png data/256x256/gtk3-widget-factory-symbolic.symbolic.png
 
-update_icon_cache = $(top_builddir)/gtk/gtk-update-icon-cache$(EXEEXT) --ignore-theme-index --force
+update_icon_cache = true
 
 install-data-hook: install-update-icon-cache
 uninstall-hook: uninstall-update-icon-cache
