--- a/modules/input/meson.build
+++ b/modules/input/meson.build
@@ -14,7 +14,7 @@
     shared_module('im-@0@'.format(name),
                   immod_sources,
                   c_args: common_cflags + cflags,
-                  dependencies: [ libgtk_dep ],
+                  dependencies: [ libgtk_dep, libgdk_dep ],
                   install_dir: immodules_install_dir,
                   include_directories: [confinc, gtkinc],
                   install : true,
