diff -uNr gtk-3.24.10/gdk/Makefile.am gtk-3.24.10.mod/gdk/Makefile.am
--- gtk-3.24.10/gdk/Makefile.am	2019-07-03 23:25:10.000000000 +0300
+++ gtk-3.24.10.mod/gdk/Makefile.am	2019-07-23 00:25:53.122144498 +0300
@@ -434,13 +434,13 @@
 gdkenumtypes.h: stamp-gdkenumtypes.h
 	@true
 stamp-gdkenumtypes.h: $(gdk_h_sources) gdkenumtypes.h.template
-	$(AM_V_GEN) ( cd $(srcdir) && $(GLIB_MKENUMS) --template gdkenumtypes.h.template \
+	$(AM_V_GEN) ( cd $(srcdir) && glib-mkenums --template gdkenumtypes.h.template \
 		$(gdk_h_sources) ) >> xgen-geth \
 	&& (cmp -s xgen-geth gdkenumtypes.h || cp xgen-geth gdkenumtypes.h ) \
 	&& rm -f xgen-geth \
 	&& echo timestamp > $(@F)
 gdkenumtypes.c: $(gdk_h_sources) gdkenumtypes.c.template
-	$(AM_V_GEN) ( cd $(srcdir) && $(GLIB_MKENUMS) --template gdkenumtypes.c.template \
+	$(AM_V_GEN) ( cd $(srcdir) && glib-mkenums --template gdkenumtypes.c.template \
 		$(gdk_h_sources) ) > xgen-getc \
 	&& cp xgen-getc gdkenumtypes.c  \
 	&& rm -f xgen-getc
@@ -449,7 +449,7 @@
 # Marshaller generation
 #
 gdkmarshalers.h: gdkmarshalers.list
-	$(AM_V_GEN) $(GLIB_GENMARSHAL) \
+	$(AM_V_GEN) glib-genmarshal \
 		--prefix=_gdk_marshal \
 		--valist-marshallers \
 		--header \
@@ -457,7 +457,7 @@
 	&& mv gdkmarshalers-h.tmp gdkmarshalers.h \
 	|| ( rm -f gdkmarshalers-h.tmp && exit 1)
 gdkmarshalers.c: gdkmarshalers.list
-	$(AM_V_GEN) $(GLIB_GENMARSHAL) \
+	$(AM_V_GEN) glib-genmarshal \
 		--prefix=_gdk_marshal \
 		--valist-marshallers \
 		--body \
diff -uNr gtk-3.24.10/gtk/Makefile.am gtk-3.24.10.mod/gtk/Makefile.am
--- gtk-3.24.10/gtk/Makefile.am	2019-07-03 23:25:10.000000000 +0300
+++ gtk-3.24.10.mod/gtk/Makefile.am	2019-07-23 00:24:47.805264300 +0300
@@ -1252,7 +1252,7 @@
 gtkmarshalers.h: stamp-gtkmarshalers.h
 	@true
 stamp-gtkmarshalers.h: gtkmarshalers.list
-	$(AM_V_GEN) $(GLIB_GENMARSHAL) \
+	$(AM_V_GEN) glib-genmarshal \
 		--prefix=_gtk_marshal \
 		--header \
 		--valist-marshallers \
@@ -1262,7 +1262,7 @@
 	&& echo timestamp > $(@F)
 gtkmarshalers.c: gtkmarshalers.list
 	$(AM_V_GEN) (echo "#undef G_ENABLE_DEBUG"; \
-	 $(GLIB_GENMARSHAL) \
+	 glib-genmarshal \
 	 	--prefix=_gtk_marshal \
 		--body \
 		--valist-marshallers $(srcdir)/gtkmarshalers.list) >> xgen-gmlc \
@@ -1272,13 +1272,13 @@
 gtktypebuiltins.h: stamp-gtktypebuiltins.h
 	@true
 stamp-gtktypebuiltins.h: $(gtk_public_h_sources) $(a11y_h_sources) $(gtk_deprecated_h_sources) gtktypebuiltins.h.template
-	$(AM_V_GEN) ( cd $(srcdir) && $(GLIB_MKENUMS) --template gtktypebuiltins.h.template \
+	$(AM_V_GEN) ( cd $(srcdir) && glib-mkenums --template gtktypebuiltins.h.template \
 		$(gtk_public_h_sources) $(a11y_h_sources) $(gtk_deprecated_h_sources) ) > xgen-gtbh \
 	&& (cmp -s xgen-gtbh gtktypebuiltins.h || cp xgen-gtbh gtktypebuiltins.h ) \
 	&& rm -f xgen-gtbh \
 	&& echo timestamp > $(@F)
 gtktypebuiltins.c: $(gtk_public_h_sources) $(a11y_h_sources) $(gtk_deprecated_h_sources) gtktypebuiltins.c.template
-	$(AM_V_GEN) ( cd $(srcdir) && $(GLIB_MKENUMS) --template gtktypebuiltins.c.template \
+	$(AM_V_GEN) ( cd $(srcdir) && glib-mkenums --template gtktypebuiltins.c.template \
 		$(gtk_public_h_sources) $(a11y_h_sources) $(gtk_deprecated_h_sources) ) > xgen-gtbc \
 	&& cp xgen-gtbc gtktypebuiltins.c  \
 	&& rm -f xgen-gtbc
@@ -1430,13 +1430,13 @@
 gtkprivatetypebuiltins.h: stamp-gtkprivatetypebuiltins.h
 	@true
 stamp-gtkprivatetypebuiltins.h: $(gtk_private_type_h_sources) gtkprivatetypebuiltins.h.template
-	$(AM_V_GEN) ( cd $(srcdir) && $(GLIB_MKENUMS) --template gtkprivatetypebuiltins.h.template \
+	$(AM_V_GEN) ( cd $(srcdir) && glib-mkenums --template gtkprivatetypebuiltins.h.template \
 		$(gtk_private_type_h_sources) ) > xgen-gptbh \
 	&& (cmp -s xgen-gptbh gtkprivatetypebuiltins.h || cp xgen-gptbh gtkprivatetypebuiltins.h ) \
 	&& rm -f xgen-gptbh \
 	&& echo timestamp > $(@F)
 gtkprivatetypebuiltins.c: $(gtk_private_type_h_sources)  gtkprivatetypebuiltins.c.template
-	$(AM_V_GEN) ( cd $(srcdir) && $(GLIB_MKENUMS) --template gtkprivatetypebuiltins.c.template \
+	$(AM_V_GEN) ( cd $(srcdir) && glib-mkenums --template gtkprivatetypebuiltins.c.template \
 		$(gtk_private_type_h_sources) ) > xgen-gptbc \
 	&& cp xgen-gptbc gtkprivatetypebuiltins.c  \
 	&& rm -f xgen-gptbc
