diff -uNr gtk-3.22.30/examples/sunny.c gtk-3.22.30.mod/examples/sunny.c
--- gtk-3.22.30/examples/sunny.c	2018-04-16 21:06:02.000000000 +0300
+++ gtk-3.22.30.mod/examples/sunny.c	2018-06-22 20:55:46.621292924 +0300
@@ -56,7 +56,7 @@
 }
 
 static void
-open (GApplication  *application,
+opened (GApplication  *application,
       GFile        **files,
       gint           n_files,
       const gchar   *hint)
@@ -173,7 +173,7 @@
 
   application_class->startup = startup;
   application_class->activate = activate;
-  application_class->open = open;
+  application_class->open = opened;
 }
 
 MenuButton *
