diff -uNr gtk-3.22.30/gtk/a11y/gtkaccessibility.c gtk-3.22.30.mod/gtk/a11y/gtkaccessibility.c
--- gtk-3.22.30/gtk/a11y/gtkaccessibility.c	2018-04-16 21:06:02.000000000 +0300
+++ gtk-3.22.30.mod/gtk/a11y/gtkaccessibility.c	2018-06-22 20:55:46.611292924 +0300
@@ -37,10 +37,6 @@
 #include <gtk/gtktogglebutton.h>
 #include <gtk/gtkaccessible.h>
 
-#ifdef GDK_WINDOWING_X11
-#include <atk-bridge.h>
-#endif
-
 static gboolean gail_focus_watcher      (GSignalInvocationHint *ihint,
                                          guint                  n_param_values,
                                          const GValue          *param_values,
@@ -988,9 +984,5 @@
   _gtk_accessibility_override_atk_util ();
   do_window_event_initialization ();
 
-#ifdef GDK_WINDOWING_X11
-  atk_bridge_adaptor_init (NULL, NULL);
-#endif
-
   atk_misc_instance = g_object_new (GTK_TYPE_MISC_IMPL, NULL);
 }
