diff -uNr gtk-3.22.30/m4/xsize.m4 gtk-3.22.30.mod/m4/xsize.m4
--- gtk-3.22.30/m4/xsize.m4	1970-01-01 03:00:00.000000000 +0300
+++ gtk-3.22.30.mod/m4/xsize.m4	2018-06-22 20:55:46.581292922 +0300
@@ -0,0 +1,12 @@
+# xsize.m4 serial 5
+dnl Copyright (C) 2003-2004, 2008-2015 Free Software Foundation, Inc.
+dnl This file is free software; the Free Software Foundation
+dnl gives unlimited permission to copy and/or distribute it,
+dnl with or without modifications, as long as this notice is preserved.
+
+AC_DEFUN([gl_XSIZE],
+[
+  dnl Prerequisites of lib/xsize.h.
+  AC_REQUIRE([gl_SIZE_MAX])
+  AC_CHECK_HEADERS([stdint.h])
+])
