diff -uNr gtk-3.22.30/configure.ac gtk-3.22.30.mod/configure.ac
--- gtk-3.22.30/configure.ac	2018-04-16 21:06:02.000000000 +0300
+++ gtk-3.22.30.mod/configure.ac	2018-06-22 20:55:46.617959590 +0300
@@ -1393,11 +1393,7 @@
 # Check for Accessibility Toolkit flags
 ########################################
 
-if test x$enable_x11_backend = xyes; then
-   ATK_PACKAGES="atk atk-bridge-2.0"
-else
-   ATK_PACKAGES="atk"
-fi
+ATK_PACKAGES="atk"
 
 PKG_CHECK_MODULES(ATK, $ATK_PACKAGES)
 
