diff --git a/inode2prog.cpp b/inode2prog.cpp
index cfac566..4c1c4c5 100644
--- a/inode2prog.cpp
+++ b/inode2prog.cpp
@@ -117,7 +117,7 @@ std::string getprogname (pid_t pid) {
 	const int maxfilenamelen = 14 + MAX_PID_LENGTH + 1;
 	char filename[maxfilenamelen];

-	std::snprintf(filename, maxfilenamelen, "/proc/%d/cmdline", pid);
+	snprintf(filename, maxfilenamelen, "/proc/%d/cmdline", pid);
 	return read_file(filename);
 }
