diff -uNr db-6.2.32/src/os/os_tmpdir.c db-6.2.32.mod/src/os/os_tmpdir.c
--- db-6.2.32/src/os/os_tmpdir.c	2017-04-13 17:06:21.000000000 +0300
+++ db-6.2.32.mod/src/os/os_tmpdir.c	2019-03-02 02:06:03.557665103 +0200
@@ -125,9 +125,9 @@
 	DB_TEMP_DIRECTORY("C:/temp");
 	DB_TEMP_DIRECTORY("C:/tmp");
 #else
-	DB_TEMP_DIRECTORY("/var/tmp");
-	DB_TEMP_DIRECTORY("/usr/tmp");
-	DB_TEMP_DIRECTORY("/tmp");
+	DB_TEMP_DIRECTORY("@TERMUX_PREFIX@/var/tmp");
+	DB_TEMP_DIRECTORY("@TERMUX_PREFIX@/usr/tmp");
+	DB_TEMP_DIRECTORY("@TERMUX_PREFIX@/tmp");
 #if defined(ANDROID) || defined(DB_ANDROID)
 	DB_TEMP_DIRECTORY("/cache");
 #endif
