diff -u -r ../db-18.1.32/lang/sql/sqlite/src/os_unix.c ./lang/sql/sqlite/src/os_unix.c
--- ../db-18.1.32/lang/sql/sqlite/src/os_unix.c	2019-02-19 21:51:18.000000000 +0000
+++ ./lang/sql/sqlite/src/os_unix.c	2019-04-13 07:37:00.461881967 +0000
@@ -5421,9 +5421,7 @@
   static const char *azDirs[] = {
      0,
      0,
-     "/var/tmp",
-     "/usr/tmp",
-     "/tmp",
+     "@TERMUX_PREFIX@/tmp",
      "."
   };
   unsigned int i = 0;
