diff -uNr db-18.1.40/dist/Makefile.in db-18.1.40.mod/dist/Makefile.in
--- db-18.1.40/dist/Makefile.in	2020-05-29 20:58:22.000000000 +0300
+++ db-18.1.40.mod/dist/Makefile.in	2020-06-26 02:52:37.740818006 +0300
@@ -1215,7 +1215,7 @@
 # Library and standard utilities install.
 ##################################################
 library_install: install_setup
-library_install: install_include install_lib install_utilities @ADDITIONAL_INSTALL@ install_docs
+library_install: install_include install_lib install_utilities @ADDITIONAL_INSTALL@
 
 uninstall: uninstall_include uninstall_lib uninstall_utilities @ADDITIONAL_UNINSTALL@ uninstall_docs
 
