From 496eed950f6d0e7fd92619d47e3cec8f06e96ace Mon Sep 17 00:00:00 2001
From: Jochen Sprickerhof <jspricke@debian.org>
Date: Mon, 8 Aug 2022 21:10:13 +0200
Subject: [PATCH] Add missing header for LIBAVCODEC_VERSION_INT

---
 modules/videoio/src/ffmpeg_codecs.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/videoio/src/ffmpeg_codecs.hpp b/modules/videoio/src/ffmpeg_codecs.hpp
index 61788e0345a6..faad2596edc5 100644
--- a/modules/videoio/src/ffmpeg_codecs.hpp
+++ b/modules/videoio/src/ffmpeg_codecs.hpp
@@ -60,6 +60,7 @@ extern "C" {
 #include <errno.h>
 #endif
 
+#include <libavcodec/version.h>
 #include <libavformat/avformat.h>
 
 #ifdef __cplusplus
