--- a/doc/info/Makefile.am
+++ b/doc/info/Makefile.am
@@ -87,8 +87,8 @@
 # maxima to verify the index.
 maxima-index-html.lisp : index.html build-html-index.lisp
 	$(MAKE) -C $(top_builddir)/src
-	../../maxima-local --no-init --preload=$(top_srcdir)/doc/info/build-html-index.lisp --batch-string='build_and_dump_html_index("./*.html");'
-	../../maxima-local --no-init --verify-html-index --batch-string="quit();"
+	-../../maxima-local --no-init --preload=$(top_srcdir)/doc/info/build-html-index.lisp --batch-string='build_and_dump_html_index("./*.html");'
+	-../../maxima-local --no-init --verify-html-index --batch-string="quit();"
 
 DOLLAR_SIGN = $$
 
--- a/doc/info/Makefile.in
+++ b/doc/info/Makefile.in
@@ -1064,8 +1064,8 @@
 # maxima to verify the index.
 maxima-index-html.lisp : index.html build-html-index.lisp
 	$(MAKE) -C $(top_builddir)/src
-	../../maxima-local --no-init --preload=$(top_srcdir)/doc/info/build-html-index.lisp --batch-string='build_and_dump_html_index("./*.html");'
-	../../maxima-local --no-init --verify-html-index --batch-string="quit();"
+	-../../maxima-local --no-init --preload=$(top_srcdir)/doc/info/build-html-index.lisp --batch-string='build_and_dump_html_index("./*.html");'
+	-../../maxima-local --no-init --verify-html-index --batch-string="quit();"
 
 maxima_singlepage.html index.html: maxima.texi $(maxima_TEXINFOS) $(figurefiles) manual.css texi2html.init
 	./build_html.sh
