--- a/doc/info/Makefile.am
+++ b/doc/info/Makefile.am
@@ -689,7 +689,7 @@
 # This includes the indices, maxima_singlepage.html and any other html
 # file that doesn't start with "maxima".
 maxima-index-html.lisp : maxima_toc.html build-html-index.lisp
-	../../maxima-local --batch-lisp=build-html-index.lisp
+	-../../maxima-local --batch-lisp=build-html-index.lisp
 
 DOLLAR_SIGN = $$
 
--- a/doc/info/Makefile.in
+++ b/doc/info/Makefile.in
@@ -1650,7 +1650,7 @@
 # This includes the indices, maxima_singlepage.html and any other html
 # file that doesn't start with "maxima".
 maxima-index-html.lisp : maxima_toc.html build-html-index.lisp
-	../../maxima-local --batch-lisp=build-html-index.lisp
+	-../../maxima-local --batch-lisp=build-html-index.lisp
 
 maxima_singlepage.html maxima_toc.html: maxima.texi $(maxima_TEXINFOS) $(figurefiles) manual.css texi2html.init
 	./build_html.sh
