--- a/src/mpfr.h
+++ b/src/mpfr.h
@@ -1026,15 +1026,14 @@
 
 #if __GNUC__ > 2 || __GNUC_MINOR__ >= 95
 #define mpfr_custom_get_kind(x)                                         \
-  __extension__ ({                                                      \
-    mpfr_ptr _x = (x);                                                  \
-    _x->_mpfr_exp >  __MPFR_EXP_INF ?                                   \
-      (mpfr_int) MPFR_REGULAR_KIND * MPFR_SIGN (_x)                     \
-      : _x->_mpfr_exp == __MPFR_EXP_INF ?                               \
-      (mpfr_int) MPFR_INF_KIND * MPFR_SIGN (_x)                         \
-      : _x->_mpfr_exp == __MPFR_EXP_NAN ? (mpfr_int) MPFR_NAN_KIND      \
-      : (mpfr_int) MPFR_ZERO_KIND * MPFR_SIGN (_x);                     \
-  })
+  (                                                                     \
+    (x)->_mpfr_exp >  __MPFR_EXP_INF ?                                  \
+      (mpfr_int) MPFR_REGULAR_KIND * MPFR_SIGN (x)                      \
+      : (x)->_mpfr_exp == __MPFR_EXP_INF ?                              \
+      (mpfr_int) MPFR_INF_KIND * MPFR_SIGN (x)                          \
+      : (x)->_mpfr_exp == __MPFR_EXP_NAN ? (mpfr_int) MPFR_NAN_KIND     \
+      : (mpfr_int) MPFR_ZERO_KIND * MPFR_SIGN (x)                       \
+  )
 #else
 #define mpfr_custom_get_kind(x) ((mpfr_custom_get_kind)(x))
 #endif
