--- sc-7.16/Makefile
+++ Makefile
@@ -20,13 +20,13 @@
 # packaging this with an O/S, for example, you'll probably want to change
 # this to /usr.  Otherwise, /usr/local is probably more appropriate, unless
 # you're replacing the vendor-supplied version.
-prefix=/usr
+prefix=/data/data/com.termux/files/usr
 
 # This is where the install step puts it.
 EXDIR=${prefix}/bin
 
 # This is where the man page goes.
-MANDIR=${prefix}/man/man1
+MANDIR=${prefix}/share/man/man1
 MANEXT=1
 MANMODE=644
 
@@ -175,9 +175,9 @@
 YTAB=y.tab
 
 # Command to use to make temporary copies of some source files.
-#LN=ln -s
+LN=ln -s
 #LN=cp
-LN=ln
+#LN=ln
 
 #### SYSTEM DEFINES ####
 
