diff -u -r ../transmission-2.82/third-party/libnatpmp/getgateway.h ./third-party/libnatpmp/getgateway.h
--- ../transmission-2.82/third-party/libnatpmp/getgateway.h	2013-08-09 04:47:55.000000000 +0200
+++ ./third-party/libnatpmp/getgateway.h	2014-01-28 23:57:43.000000000 +0100
@@ -40,6 +40,8 @@
 #endif
 #include "declspec.h"
 
+#include <arpa/inet.h>
+
 /* getdefaultgateway() :
  * return value :
  *    0 : success
--- ../natpmp.c.orig	2020-05-14 20:17:03.002657737 +0200
+++ ./third-party/libnatpmp/natpmp.c	2020-05-14 20:17:34.527090133 +0200
@@ -54,6 +54,7 @@
 #include "natpmp.h"
 #include "getgateway.h"
 #include <stdio.h>
+#include <arpa/inet.h>
 
 NATPMP_LIBSPEC int initnatpmp(natpmp_t * p, int forcegw, in_addr_t forcedgw)
 {
diff -u -r ../transmission-2.82/third-party/libnatpmp/natpmp.h ./third-party/libnatpmp/natpmp.h
--- ../transmission-2.82/third-party/libnatpmp/natpmp.h	2013-08-09 04:47:55.000000000 +0200
+++ ./third-party/libnatpmp/natpmp.h	2014-01-28 23:58:05.000000000 +0100
@@ -51,6 +51,8 @@
 #include <netinet/in.h>
 #endif
 
+#include <arpa/inet.h>
+
 typedef struct {
 	int s;	/* socket */
 	in_addr_t gateway;	/* default gateway (IPv4) */
