diff -u -r ../transmission-2.92/libtransmission/fdlimit.c ./libtransmission/fdlimit.c
--- ../transmission-2.92/libtransmission/fdlimit.c	2015-10-23 21:35:22.638234970 -0400
+++ ./libtransmission/fdlimit.c	2016-03-07 20:32:58.075869242 -0500
@@ -11,6 +11,7 @@
 #include <errno.h>
 #include <inttypes.h>
 #include <string.h>
+#include <sys/select.h>
 
 #ifndef _WIN32
  #include <sys/time.h> /* getrlimit */
