diff -u -r ../transmission-2.82/libtransmission/utils.c ./libtransmission/utils.c
--- ../transmission-2.82/libtransmission/utils.c	2013-08-09 04:45:40.000000000 +0200
+++ ./libtransmission/utils.c	2014-01-29 00:08:31.000000000 +0100
@@ -1412,7 +1412,7 @@
   char buf[128];
   const int max_precision = (int) log10 (1.0 / DBL_EPSILON) - 1;
   tr_snprintf (buf, sizeof (buf), "%.*f", max_precision, x);
-  if ((pt = strstr (buf, localeconv ()->decimal_point)))
+  if ((pt = strstr (buf, ".")))
     pt[precision ? precision+1 : 0] = '\0';
   return atof (buf);
 }
