--- ../Makefile.orig	2021-10-30 15:39:30.272534156 +0000
+++ ./Makefile	2021-10-30 15:40:46.089228652 +0000
@@ -55,7 +55,7 @@
 	rm -f *.o ts
 
 install: ts
-	$(INSTALL) -d $(PREFIX)/bin
-	$(INSTALL) ts $(PREFIX)/bin
-	$(INSTALL) -d $(PREFIX)/share/man/man1
-	$(INSTALL) -m 644 ts.1 $(PREFIX)/share/man/man1
+	$(INSTALL) -d $(DESTDIR)$(PREFIX)/bin
+	$(INSTALL) ts $(DESTDIR)$(PREFIX)/bin
+	$(INSTALL) -d $(DESTDIR)$(PREFIX)/share/man/man1
+	$(INSTALL) -m 644 ts.1 $(DESTDIR)$(PREFIX)/share/man/man1
