diff -u -r /home/builder/lib/android-ndk-r28b/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/sys/time.h ./usr/include/sys/time.h
--- /home/builder/lib/android-ndk-r28b/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/sys/time.h	2024-11-14 00:52:25.000000000 +0000
+++ ./usr/include/sys/time.h	2025-05-09 18:58:25.785705103 +0000
@@ -48,10 +48,8 @@
 
 #if defined(__USE_BSD)
 
-#if __BIONIC_AVAILABILITY_GUARD(26)
-int futimes(int __fd, const struct timeval __times[_Nullable 2]) __INTRODUCED_IN(26);
-int lutimes(const char* _Nonnull __path, const struct timeval __times[_Nullable 2]) __INTRODUCED_IN(26);
-#endif /* __BIONIC_AVAILABILITY_GUARD(26) */
+int futimes(int __fd, const struct timeval __times[_Nullable 2]);
+int lutimes(const char* _Nonnull __path, const struct timeval __times[_Nullable 2]);
 
 #endif
 
@@ -70,9 +68,7 @@
  * Available since API level 26.
  */
 
-#if __BIONIC_AVAILABILITY_GUARD(26)
-int futimesat(int __dir_fd, const char* __BIONIC_COMPLICATED_NULLNESS __path, const struct timeval __times[_Nullable 2]) __INTRODUCED_IN(26);
-#endif /* __BIONIC_AVAILABILITY_GUARD(26) */
+int futimesat(int __dir_fd, const char* __BIONIC_COMPLICATED_NULLNESS __path, const struct timeval __times[_Nullable 2]);
 
 #endif
 
