--- a/Makefile
+++ b/Makefile
@@ -10,7 +10,7 @@ CONFIG := clang
 
 # features (the more the better)
 ENABLE_TCL := 1
-ENABLE_ABC := 1
+ENABLE_ABC := 0
 ENABLE_GLOB := 1
 ENABLE_PLUGINS := 1
 ENABLE_READLINE := 1
@@ -217,7 +217,7 @@ endif
 
 ifeq ($(CONFIG),clang)
 CXX = clang++
-CXXFLAGS += -std=$(CXXSTD) -Os
+CXXFLAGS += -std=$(CXXSTD)
 ABCMKARGS += ARCHFLAGS="-DABC_USE_STDINT_H $(ABC_ARCHFLAGS)"
 
 ifneq ($(SANITIZER),)
