--- a/Makefile
+++ b/Makefile
@@ -11,7 +11,7 @@
 
 # features (the more the better)
 ENABLE_TCL := 1
-ENABLE_ABC := 1
+ENABLE_ABC := 0
 ENABLE_GLOB := 1
 ENABLE_PLUGINS := 1
 ENABLE_READLINE := 1
@@ -138,7 +138,6 @@
 else
 LDFLAGS += -rdynamic
 ifneq ($(OS), OpenBSD)
-LDLIBS += -lrt
 endif
 endif
 
@@ -216,9 +215,8 @@
 endif
 
 ifeq ($(CONFIG),clang)
-CXX = clang
-LD = clang++
-CXXFLAGS += -std=$(CXXSTD) -Os
+LD = $(CXX)
+CXXFLAGS += -std=$(CXXSTD)
 ABCMKARGS += ARCHFLAGS="-DABC_USE_STDINT_H -Wno-c++11-narrowing $(ABC_ARCHFLAGS)"
 
 ifneq ($(SANITIZER),)
