--- a/Makefile
+++ b/Makefile
@@ -222,7 +222,7 @@
 
 ifeq ($(CONFIG),clang)
 CXX = clang++
-CXXFLAGS += -std=$(CXXSTD) $(OPT_LEVEL)
+CXXFLAGS += -std=$(CXXSTD)
 ifeq ($(ENABLE_LTO),1)
 LINKFLAGS += -fuse-ld=lld
 endif
--- a/abc/Makefile
+++ b/abc/Makefile
@@ -57,7 +57,7 @@ ARCHFLAGS := $(ARCHFLAGS)
 
 OPTFLAGS  ?= -g -O
 
-CFLAGS    += -Wall -Wno-unused-function -Wno-write-strings -Wno-sign-compare $(ARCHFLAGS)
+CFLAGS    += -Wall -Wno-unused-function -Wno-write-strings -Wno-sign-compare -Wno-c++11-narrowing $(ARCHFLAGS)
 ifneq ($(findstring arm,$(shell uname -m)),)
 	CFLAGS += -DABC_MEMALIGN=4
 endif
@@ -76,8 +76,8 @@ ifndef ABC_USE_NO_CUDD
   $(info $(MSG_PREFIX)Compiling with CUDD)
 endif
 
-ABC_READLINE_INCLUDES ?=
-ABC_READLINE_LIBRARIES ?= -lreadline
+ABC_READLINE_INCLUDES ?= -I@TERMUX_PREFIX@/include -DNCURSES_WIDECHAR
+ABC_READLINE_LIBRARIES ?= -L@TERMUX_PREFIX@/lib -lreadline
 
 # whether to use libreadline
 ifndef ABC_USE_NO_READLINE
