--- a/passes/sat/qbfsat.cc
+++ b/passes/sat/qbfsat.cc
@@ -251,7 +251,7 @@
 
 QbfSolutionType qbf_solve(RTLIL::Module *mod, const QbfSolveOptions &opt) {
 	QbfSolutionType ret, best_soln;
-	const std::string tempdir_name = make_temp_dir("/tmp/yosys-qbfsat-XXXXXX");
+	const std::string tempdir_name = make_temp_dir("@TERMUX_PREFIX@/tmp/yosys-qbfsat-XXXXXX");
 	RTLIL::Module *module = mod;
 	RTLIL::Design *design = module->design;
 	std::string module_name = module->name.str();
