--- a/techlibs/easic/synth_easic.cc
+++ b/techlibs/easic/synth_easic.cc
@@ -45,7 +45,7 @@
 		log("        an output file is omitted if this parameter is not specified.\n");
 		log("\n");
 		log("    -etools <path>\n");
-		log("        set path to the eTools installation. (default=/opt/eTools)\n");
+		log("        set path to the eTools installation. (default=@TERMUX_PREFIX@/opt/eTools)\n");
 		log("\n");
 		log("    -run <from_label>:<to_label>\n");
 		log("        only run the commands between the labels (see below). an empty\n");
@@ -71,7 +71,7 @@
 	{
 		top_opt = "-auto-top";
 		vlog_file = "";
-		etools_path = "/opt/eTools";
+		etools_path = "@TERMUX_PREFIX@/opt/eTools";
 		flatten = true;
 		retime = false;
 	}
