--- a/src/lib_ccx/params.c
+++ b/src/lib_ccx/params.c
@@ -1,6 +1,7 @@
 #include "png.h"
-#include "protobuf-c.h"
+#include "protobuf-c/protobuf-c.h"
 #include "zlib.h"
+#include "gpac/setup.h"
 #include "gpac/version.h"
 #include "lib_ccx.h"
 #include "ccx_common_option.h"
@@ -10,10 +11,10 @@
 #include "ccx_common_common.h"
 #include "ccx_decoders_708.h"
 #include "compile_info.h"
-#include "../lib_hash/sha2.h"
+#include "sha256.h"
 #include <string.h>
 #include <stdio.h>
-#include <utf8proc/utf8proc.h>
+#include <utf8proc.h>
 
 #ifdef ENABLE_OCR
 #include <tesseract/capi.h>
