--- a/src/ccextractor.c
+++ b/src/ccextractor.c
@@ -222,25 +222,30 @@
 					ret = tmp;
 				break;
 			case CCX_SM_MP4:
+			{
+				int mp4_file_index = ctx->current_file;
+				if (mp4_file_index >= ctx->num_input_files)
+					mp4_file_index = ctx->num_input_files - 1;
 				mprint("\rAnalyzing data with GPAC (MP4 library)\n");
 				close_input_file(ctx);	     // No need to have it open. GPAC will do it for us
-				if (ctx->current_file == -1) // We don't have a file to open, must be stdin, and GPAC is incompatible with stdin
+				if (mp4_file_index < 0) // We don't have a file to open, must be stdin, and GPAC is incompatible with stdin
 				{
 					fatal(EXIT_INCOMPATIBLE_PARAMETERS, "MP4 requires an actual file, it's not possible to read from a stream, including stdin.\n");
 				}
 				if (ccx_options.extract_chapters)
 				{
-					tmp = dumpchapters(ctx, &ctx->mp4_cfg, ctx->inputfile[ctx->current_file]);
+					tmp = dumpchapters(ctx, &ctx->mp4_cfg, ctx->inputfile[mp4_file_index]);
 				}
 				else
 				{
-					tmp = processmp4(ctx, &ctx->mp4_cfg, ctx->inputfile[ctx->current_file]);
+					tmp = processmp4(ctx, &ctx->mp4_cfg, ctx->inputfile[mp4_file_index]);
 				}
 				if (ccx_options.print_file_reports)
 					print_file_report(ctx);
 				if (!ret)
 					ret = tmp;
 				break;
+			}
 			case CCX_SM_MKV:
 				mprint("\rAnalyzing data in Matroska mode\n");
 				tmp = matroska_loop(ctx);
