--- a/src/rust/src/decoder/tv_screen.rs
+++ b/src/rust/src/decoder/tv_screen.rs
@@ -259,7 +259,7 @@
             &mut buf,
         );
         // Tags can still be crossed e.g <f><i>text</f></i>, but testing HTML code has shown that they still are handled correctly.
-        if writer.writer_ctx.cd != (-1_isize) as iconv_t {
+        if writer.writer_ctx.cd != (-1_isize) as _ {
             if writer.writer_ctx.charset.is_null() {
                 debug!("Charset: null");
             } else {
