diff --git a/Source/WTF/wtf/PlatformGTK.cmake b/Source/WTF/wtf/PlatformGTK.cmake
index fa6958fbd3..c6d4a51fec 100644
--- a/Source/WTF/wtf/PlatformGTK.cmake
+++ b/Source/WTF/wtf/PlatformGTK.cmake
@@ -13,7 +13,7 @@ list(APPEND WTF_PUBLIC_HEADERS
     glib/WTFGType.h
 )

-if (CMAKE_SYSTEM_NAME MATCHES "Linux")
+if (CMAKE_SYSTEM_NAME MATCHES "Android")
     list(APPEND WTF_PUBLIC_HEADERS
         linux/ProcessMemoryFootprint.h
         linux/CurrentProcessMemoryStatus.h
@@ -47,7 +47,7 @@ list(APPEND WTF_SOURCES
     unix/UniStdExtrasUnix.cpp
 )

-if (CMAKE_SYSTEM_NAME MATCHES "Linux")
+if (CMAKE_SYSTEM_NAME MATCHES "Android")
     list(APPEND WTF_SOURCES
         linux/CurrentProcessMemoryStatus.cpp
         linux/MemoryFootprintLinux.cpp
