diff -u -r ../binutils-2.24/bfd/archive.c ./bfd/archive.c
--- ../binutils-2.24/bfd/archive.c	2013-11-04 16:33:37.000000000 +0100
+++ ./bfd/archive.c	2014-01-01 14:44:49.000000000 +0100
@@ -1880,7 +1880,8 @@
     {
       /* Assume we just "made" the member, and fake it.  */
       struct bfd_in_memory *bim = (struct bfd_in_memory *) member->iostream;
-      time (&status.st_mtime);
+      /* termux diff: explicit cast */
+      time ((time_t*) &status.st_mtime);
       status.st_uid = getuid ();
       status.st_gid = getgid ();
       status.st_mode = 0644;
