diff -uNr antiword-0.37/antiword.h antiword-0.37.mod/antiword.h
--- antiword-0.37/antiword.h	2005-07-06 20:46:22.000000000 +0300
+++ antiword-0.37.mod/antiword.h	2019-03-05 19:51:12.173846939 +0200
@@ -204,7 +204,7 @@
 #define ANTIWORD_DIR		"antiword"
 #define FONTNAMES_FILE		"fontname.txt"
 #elif defined(__vms)
-#define GLOBAL_ANTIWORD_DIR	"/usr/share/antiword"
+#define GLOBAL_ANTIWORD_DIR	"@TERMUX_PREFIX@/share/antiword"
 #define ANTIWORD_DIR		"antiword"
 #define FONTNAMES_FILE		"fontnames"
 #elif defined(__BEOS__)
@@ -224,7 +224,7 @@
 #define ANTIWORD_DIR		".antiword"
 #define FONTNAMES_FILE		"fontnames"
 #else	/* All others */
-#define GLOBAL_ANTIWORD_DIR	"/usr/share/antiword"
+#define GLOBAL_ANTIWORD_DIR	"@TERMUX_PREFIX@/share/antiword"
 #define ANTIWORD_DIR		".antiword"
 #define FONTNAMES_FILE		"fontnames"
 #endif /* __dos */
