From 5a0697aa6d71d1c3c11ec623f02bb0192c85c914 Mon Sep 17 00:00:00 2001
From: cnjhb <hikariz@qq.com>
Date: Sat, 3 Jan 2026 22:32:47 +0800
Subject: [PATCH] add python path for ibus-setup

---
 setup/ibus-setup.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/setup/ibus-setup.in b/setup/ibus-setup.in
index 474ce8a8..dce67ea4 100644
--- a/setup/ibus-setup.in
+++ b/setup/ibus-setup.in
@@ -27,5 +27,6 @@ export IBUS_PREFIX=@prefix@
 export IBUS_DATAROOTDIR=@datarootdir@
 export IBUS_LOCALEDIR=@localedir@
 export IBUS_LIBEXECDIR=${libexecdir}
+export PYTHON=@TERMUX_PREFIX@/bin/python3
 exec ${PYTHON:-@PYTHON@} @prefix@/share/ibus/setup/main.py "$@"
 
-- 
2.52.0

