--- a/platform/unix/deps.m4
+++ b/platform/unix/deps.m4
@@ -38,7 +38,7 @@
 AC_DEFUN([ACLOVE_DEP_GME], [
 	AC_SEARCH_LIBS([gme_open_data], [gme], [], [LOVE_MSG_ERROR([gme])])
 	AC_DEFINE([LOVE_SUPPORT_GME], [], [Enable gme])
-	AC_CHECK_HEADER([gme/gme.h], [includes="$includes -I/usr/include/gme"], [])])
+	AC_CHECK_HEADER([gme/gme.h], [includes="$includes -I@TERMUX_PREFIX@/include/gme"], [])])
 
 # For enet
 AC_DEFUN([ACLOVE_SOCKLEN_T], [
