diff --git a/sfdo-basedir/basedir.c b/sfdo-basedir/basedir.c
index 19a4aff..cc81e30 100644
--- a/sfdo-basedir/basedir.c
+++ b/sfdo-basedir/basedir.c
@@ -14,8 +14,8 @@
 #define STATE_HOME_FALLBACK "/.local/state/"
 #define CACHE_HOME_FALLBACK "/.cache/"
 
-#define DATA_DIRS_FALLBACK "/usr/local/share/:/usr/share/"
-#define CONFIG_DIRS_FALLBACK "/etc/xdg/"
+#define DATA_DIRS_FALLBACK "@TERMUX_PREFIX@/local/share/:@TERMUX_PREFIX@/share/"
+#define CONFIG_DIRS_FALLBACK "@TERMUX_PREFIX@/etc/xdg/"
 
 // Lists in ctx include home directories
 
