--- a/src/display_buffer.hh
+++ b/src/display_buffer.hh
@@ -137,6 +137,21 @@
     template<typename It>
     iterator insert(iterator pos, It beg, It end)
     {
+#ifdef __ANDROID__
+        auto first = beg;
+        for (; first != end; ++first)
+            if (first->has_buffer_range())
+                break;
+        if (first != end) {
+            auto last = end - 1;
+            for (; last >= first; --last)
+                if (last->has_buffer_range())
+                    break;
+            m_range.begin = std::min(m_range.begin, first->begin());
+            m_range.end = std::max(m_range.end, last->end());
+        }
+        return m_atoms.insert(pos, beg, end);
+#else
         auto has_buffer_range = std::mem_fn(&DisplayAtom::has_buffer_range);
         if (auto first = std::find_if(beg, end, has_buffer_range); first != end)
         {
@@ -145,6 +160,7 @@
             m_range.end = std::max(m_range.end, last->end());
         }
         return m_atoms.insert(pos, beg, end);
+#endif
     }
 
     DisplayLine extract(iterator beg, iterator end);
