diff -uNr mobile-ec285598849a27f681ea6269342cf03cf382eb56/src/java.base/share/native/libjli/java.c mobile-ec285598849a27f681ea6269342cf03cf382eb56.mod/src/java.base/share/native/libjli/java.c
--- mobile-ec285598849a27f681ea6269342cf03cf382eb56/src/java.base/share/native/libjli/java.c	2021-07-06 14:29:24.000000000 +0300
+++ mobile-ec285598849a27f681ea6269342cf03cf382eb56.mod/src/java.base/share/native/libjli/java.c	2021-07-30 13:28:40.900486911 +0300
@@ -50,7 +50,7 @@
  * the CreateExecutionEnviroment will remove the -d<n> flags.
  */
 
-
+#include <stdbool.h>
 #include "java.h"
 #include "jni.h"
 
@@ -219,6 +219,10 @@
 #define STACK_SIZE_MINIMUM (64 * KB)
 #endif
 
+#ifdef __TERMUX__
+extern bool android_mallopt(int opcode, void* arg, size_t arg_size);
+#endif
+
 /*
  * Entry point.
  */
@@ -252,6 +256,11 @@
     _is_java_args = javaargs;
     _wc_enabled = cpwildcard;
 
+#ifdef __TERMUX__
+    int android_malloc_tag_level = 0;
+    android_mallopt(8, &android_malloc_tag_level, sizeof(android_malloc_tag_level)));
+#endif
+
     InitLauncher(javaw);
     DumpState();
     if (JLI_IsTraceLauncher()) {
