diff -uNr mobile-ec285598849a27f681ea6269342cf03cf382eb56/make/modules/java.instrument/Lib.gmk mobile-ec285598849a27f681ea6269342cf03cf382eb56.mod/make/modules/java.instrument/Lib.gmk
--- mobile-ec285598849a27f681ea6269342cf03cf382eb56/make/modules/java.instrument/Lib.gmk	2021-07-06 14:29:24.000000000 +0300
+++ mobile-ec285598849a27f681ea6269342cf03cf382eb56.mod/make/modules/java.instrument/Lib.gmk	2021-07-29 19:38:19.486712301 +0300
@@ -36,7 +36,6 @@
 $(eval $(call SetupJdkLibrary, BUILD_LIBINSTRUMENT, \
     NAME := instrument, \
     OPTIMIZATION := LOW, \
-    EXTRA_SRC := java.base:libtinyiconv, \
     CFLAGS := $(CFLAGS_JDKLIB) $(LIBINSTRUMENT_CFLAGS), \
     CFLAGS_debug := -DJPLIS_LOGGING, \
     CFLAGS_release := -DNO_JPLIS_LOGGING, \
@@ -51,7 +50,7 @@
     LDFLAGS_aix := -L$(SUPPORT_OUTPUTDIR)/native/java.base, \
     LIBS := $(JDKLIB_LIBS), \
     LIBS_unix := -ljava -ljvm $(LIBZ_LIBS), \
-    LIBS_linux := -ljli $(LIBDL), \
+    LIBS_linux := -ljli -liconv $(LIBDL), \
     LIBS_aix := -liconv -ljli_static $(LIBDL), \
     LIBS_macosx := -ljli -liconv -framework Cocoa -framework Security \
         -framework ApplicationServices, \
diff -uNr mobile-ec285598849a27f681ea6269342cf03cf382eb56/make/modules/jdk.jdwp.agent/Lib.gmk mobile-ec285598849a27f681ea6269342cf03cf382eb56.mod/make/modules/jdk.jdwp.agent/Lib.gmk
--- mobile-ec285598849a27f681ea6269342cf03cf382eb56/make/modules/jdk.jdwp.agent/Lib.gmk	2021-07-06 14:29:24.000000000 +0300
+++ mobile-ec285598849a27f681ea6269342cf03cf382eb56.mod/make/modules/jdk.jdwp.agent/Lib.gmk	2021-07-29 19:38:46.415058534 +0300
@@ -59,11 +59,10 @@
     EXTRA_HEADER_DIRS := \
       include \
       libjdwp/export, \
-    EXTRA_SRC := java.base:libtinyiconv, \
     LDFLAGS := $(LDFLAGS_JDKLIB) \
         $(call SET_SHARED_LIBRARY_ORIGIN), \
     LIBS := $(JDKLIB_LIBS), \
-    LIBS_linux := $(LIBDL), \
+    LIBS_linux := -liconv $(LIBDL), \
     LIBS_macosx := -liconv, \
     LIBS_aix := -liconv, \
 ))
