diff -uNr tesseract-5.2.0/CMakeLists.txt tesseract-5.2.0.mod/CMakeLists.txt
--- tesseract-5.2.0/CMakeLists.txt	2022-07-07 05:15:49.000000000 +0900
+++ tesseract-5.2.0.mod/CMakeLists.txt	2022-09-30 13:12:18.222593892 +0900
@@ -859,7 +859,7 @@
   target_link_libraries(tesseract ${TIFF_LIBRARIES})
 endif()
 
-if(OPENMP_BUILD AND UNIX)
+if(OPENMP_BUILD AND UNIX AND NOT ANDROID)
   target_link_libraries(tesseract pthread)
 endif()
