diff -u -r ../tesseract-4.0.0/src/api/baseapi.cpp ./src/api/baseapi.cpp
--- ../tesseract-4.0.0/src/api/baseapi.cpp	2018-10-29 08:53:12.000000000 +0000
+++ ./src/api/baseapi.cpp	2019-02-03 22:55:00.470748529 +0000
@@ -206,11 +206,8 @@
       image_height_(0) {
   const char *locale;
   locale = std::setlocale(LC_ALL, nullptr);
-  ASSERT_HOST(!strcmp(locale, "C"));
   locale = std::setlocale(LC_CTYPE, nullptr);
-  ASSERT_HOST(!strcmp(locale, "C"));
   locale = std::setlocale(LC_NUMERIC, nullptr);
-  ASSERT_HOST(!strcmp(locale, "C"));
 }
 
 TessBaseAPI::~TessBaseAPI() {
