diff -uNr wireshark-2.6.1/epan/maxmind_db.c wireshark-2.6.1.mod/epan/maxmind_db.c
--- wireshark-2.6.1/epan/maxmind_db.c	2018-05-22 21:53:09.000000000 +0300
+++ wireshark-2.6.1.mod/epan/maxmind_db.c	2018-06-26 18:49:28.834971786 +0300
@@ -69,8 +69,8 @@
     { "C:\\ProgramData\\GeoIP" },
     { "C:\\GeoIP" },
 #else
-    { "/usr/share/GeoIP" },
-    { "/var/lib/GeoIP" },
+    { "@TERMUX_PREFIX@/share/GeoIP" },
+    { "@TERMUX_PREFIX@/var/lib/GeoIP" },
 #endif
     { NULL }
 };
