diff -uNr wireshark-2.6.1/plugins/epan/wimax/Makefile.in wireshark-2.6.1.mod/plugins/epan/wimax/Makefile.in
--- wireshark-2.6.1/plugins/epan/wimax/Makefile.in	2018-05-22 21:53:30.000000000 +0300
+++ wireshark-2.6.1.mod/plugins/epan/wimax/Makefile.in	2018-06-26 19:41:20.178461345 +0300
@@ -925,7 +925,7 @@
 	}
 
 wimax.la: $(wimax_la_OBJECTS) $(wimax_la_DEPENDENCIES) $(EXTRA_wimax_la_DEPENDENCIES) 
-	$(AM_V_CCLD)$(wimax_la_LINK) -rpath $(epan_plugindir) $(wimax_la_OBJECTS) $(wimax_la_LIBADD) $(LIBS)
+	$(AM_V_CCLD)$(wimax_la_LINK) -rpath $(epan_plugindir) $(wimax_la_OBJECTS) $(wimax_la_LIBADD) -L$(top_srcdir)/epan/.libs -L$(top_srcdir)/wsutil/.libs -L$(top_srcdir)/wiretap/.libs -lwireshark -lglib-2.0 $(LIBS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
