--- ../../build/libjemalloc/cache/jemalloc-4.5.0/Makefile.in	2017-03-01 04:10:36.000000000 +0100
+++ ./Makefile.in	2017-03-02 21:17:46.972199302 +0100
@@ -383,7 +383,7 @@
 
 install_doc: install_doc_html install_doc_man
 
-install: install_bin install_include install_lib install_doc
+install: install_bin install_include install_lib
 
 tests_unit: $(TESTS_UNIT:$(srcroot)%.c=$(objroot)%$(EXE))
 tests_integration: $(TESTS_INTEGRATION:$(srcroot)%.c=$(objroot)%$(EXE))
