diff --git a/src/video-downloader.in b/src/video-downloader.in
index 434f73f..f05116d 100644
--- a/src/video-downloader.in
+++ b/src/video-downloader.in
@@ -30,8 +30,6 @@ localedir = @localedir@
 
 sys.path.insert(1, pkgdatadir)
 signal.signal(signal.SIGINT, signal.SIG_DFL)
-locale.bindtextdomain('video-downloader', localedir)
-locale.textdomain('video-downloader')
 gettext.bindtextdomain('video-downloader', localedir)
 gettext.textdomain('video-downloader')
 with contextlib.suppress(locale.Error):
