diff -uNr sdcv-0.5.3/src/sdcv.cpp sdcv-0.5.3.mod/src/sdcv.cpp
--- sdcv-0.5.3/src/sdcv.cpp     2020-08-14 18:06:51.000000000 +0800
+++ sdcv-0.5.3.mod/src/sdcv.cpp 2021-07-19 00:10:19.833094383 +0800
@@ -136,7 +136,7 @@
             if (stardict_data_dir)
                 data_dir = stardict_data_dir;
             else
-                data_dir = "/usr/share/stardict/dic";
+                data_dir = "@TERMUX_PREFIX@/share/stardict/dic";
         }
     } else {
         data_dir = get_impl(opt_data_dir);
