diff -uNr sdcv-0.5.3/src/sdcv.cpp sdcv-0.5.3.mod/src/sdcv.cpp
--- sdcv-0.5.3/src/sdcv.cpp     2020-08-14 18:06:51.000000000 +0800
+++ sdcv-0.5.3.mod/src/sdcv.cpp 2021-07-18 23:30:05.063094239 +0800
@@ -136,7 +136,13 @@
             if (stardict_data_dir)
                 data_dir = stardict_data_dir;
             else
-                data_dir = "/usr/share/stardict/dic";
+                data_dir =
+#ifdef __TERMUX__
+                  __TERMUX_PREFIX__
+#else
+                  "/usr"
+#endif
+                  "/share/stardict/dic";
         }
     } else {
         data_dir = get_impl(opt_data_dir);
