diff -u -r -N ../tmux-1.8/compat/imsg-buffer.c ./compat/imsg-buffer.c
--- ../tmux-1.8/compat/imsg-buffer.c	2013-02-10 17:20:15.000000000 +0100
+++ ./compat/imsg-buffer.c	2014-01-08 12:33:53.721206934 +0100
@@ -28,6 +28,10 @@
 
 #include "tmux.h"
 
+#ifndef IOV_MAX
+# define IOV_MAX 1024
+#endif
+
 int	ibuf_realloc(struct ibuf *, size_t);
 void	ibuf_enqueue(struct msgbuf *, struct ibuf *);
 void	ibuf_dequeue(struct msgbuf *, struct ibuf *);
