diff -u -r ../tmux-1.8/session.c ./session.c
--- ../tmux-1.8/session.c	2013-03-26 20:22:31.000000000 +0100
+++ ./session.c	2014-02-13 18:55:45.000000000 +0100
@@ -23,6 +23,7 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <time.h>
+#include <sys/stat.h>
 
 #include "tmux.h"
 
@@ -245,8 +246,10 @@
 	server_fill_environ(s, &env);
 
 	shell = options_get_string(&s->options, "default-shell");
-	if (*shell == '\0' || areshell(shell))
-		shell = _PATH_BSHELL;
+	if (*shell == '\0' || areshell(shell)) {
+		struct stat st;
+		shell = (stat("@TERMUX_PREFIX@/home/.aterm/shell", &st) == 0) ? "@TERMUX_PREFIX@/home/.aterm/shell" : _PATH_BSHELL;
+	}
 
 	hlimit = options_get_number(&s->options, "history-limit");
 	w = window_create(
