diff -u -r ../tmux-2.2/session.c ./session.c
--- ../tmux-2.2/session.c	2016-01-20 19:00:28.000000000 -0500
+++ ./session.c	2016-04-11 05:28:53.669311770 -0400
@@ -22,6 +22,7 @@
 #include <string.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <pwd.h>
 #include <time.h>
 
 #include "tmux.h"
@@ -340,7 +341,7 @@
 
 	shell = options_get_string(s->options, "default-shell");
 	if (*shell == '\0' || areshell(shell))
-		shell = _PATH_BSHELL;
+		shell = getpwuid(getuid())->pw_shell;
 
 	hlimit = options_get_number(s->options, "history-limit");
 	w = window_create(name, argc, argv, path, shell, cwd, env, s->tio,
