diff -u -r ../tmux-2.1/tmux.h ./tmux.h
--- ../tmux-2.1/tmux.h	2015-09-27 05:51:49.000000000 -0400
+++ ./tmux.h	2015-10-19 17:33:45.737891900 -0400
@@ -40,7 +40,7 @@
 extern char   **environ;
 
 /* Default global configuration file. */
-#define TMUX_CONF "/etc/tmux.conf"
+#define TMUX_CONF "@TERMUX_PREFIX@/etc/tmux.conf"
 
 /*
  * Minimum layout cell size, NOT including separator line. The scroll region
@@ -72,7 +72,7 @@
 #define fatalx(msg) log_fatalx("%s: %s", __func__, msg);
 
 /* Definition to shut gcc up about unused arguments. */
-#define unused __attribute__ ((unused))
+#define unused
 
 /* Attribute to make gcc check printf-like arguments. */
 #define printflike(a, b) __attribute__ ((format (printf, a, b)))
