diff -uNr libvips-8.14.2/cplusplus/include/vips/vips8 libvips-8.14.2.mod/cplusplus/include/vips/vips8
--- libvips-8.14.2/cplusplus/include/vips/vips8	2023-03-21 20:42:23.000000000 +0800
+++ libvips-8.14.2.mod/cplusplus/include/vips/vips8	2023-05-03 17:10:48.209181929 +0800
@@ -32,7 +32,7 @@
 
 #include <vips/version.h>
 
-#include <glib-object.h>
+#include <glib-2.0/glib-object.h>
 
 /* Define VIPS_CPLUSPLUS_EXPORTS to build a DLL using MSVC.
  */
