--- a/data/CMakeLists.txt
+++ b/data/CMakeLists.txt
@@ -13,13 +13,13 @@
 add_custom_command(
   OUTPUT "${LM_OUTPUT}"
   DEPENDS "${LM_SRC}" LibIME::slm_build_binary
-  COMMAND LibIME::slm_build_binary -s -a 22 -q 8 trie "${LM_SRC}" "${LM_OUTPUT}")
+  COMMAND echo LibIME::slm_build_binary -s -a 22 -q 8 trie "${LM_SRC}" "${LM_OUTPUT}")
 add_custom_target(lm ALL DEPENDS "${LM_OUTPUT}")

 add_custom_command(
   OUTPUT "${LM_PREDICT_OUTPUT}"
   DEPENDS "${LM_SRC}" lm LibIME::prediction
-  COMMAND LibIME::prediction "${LM_OUTPUT}" "${LM_SRC}" "${LM_PREDICT_OUTPUT}")
+  COMMAND echo LibIME::prediction "${LM_OUTPUT}" "${LM_SRC}" "${LM_PREDICT_OUTPUT}")
 add_custom_target(lm-predict ALL DEPENDS "${LM_PREDICT_OUTPUT}")

 install(FILES "${LM_OUTPUT}" RENAME zh_CN.lm DESTINATION "${LIBIME_INSTALL_LIBDATADIR}")
@@ -38,19 +38,20 @@
 add_custom_command(
   OUTPUT "${DICT_OUTPUT}"
   DEPENDS "${DICT_SRC}" LibIME::pinyindict
-  COMMAND LibIME::pinyindict "${DICT_SRC}" "${DICT_OUTPUT}")
+  COMMAND echo LibIME::pinyindict "${DICT_SRC}" "${DICT_OUTPUT}")

 set(DICT_EXTB_SRC "${CMAKE_CURRENT_BINARY_DIR}/dict_extb.txt")
 set(DICT_EXTB_OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/extb.dict")
 add_custom_command(
   OUTPUT "${DICT_EXTB_OUTPUT}"
   DEPENDS "${DICT_EXTB_SRC}" LibIME::pinyindict
-  COMMAND LibIME::pinyindict "${DICT_EXTB_SRC}" "${DICT_EXTB_OUTPUT}")
+  COMMAND echo LibIME::pinyindict "${DICT_EXTB_SRC}" "${DICT_EXTB_OUTPUT}")
 add_custom_target(dict ALL DEPENDS "${DICT_OUTPUT}" "${DICT_EXTB_OUTPUT}")
 install(FILES "${DICT_OUTPUT}" "${DICT_EXTB_OUTPUT}" DESTINATION "${LIBIME_INSTALL_PKGDATADIR}")

 set(TABLE_DICT_TAR "table-20240108.tar.zst")
 set(TABLE_DICT_URL "https://download.fcitx-im.org/data/${TABLE_DICT_TAR}")
+# if this list changes, rebase this patch and modify the matching list in build.sh
 set(TABLE_TXT_FILES db.txt erbi.txt qxm.txt wanfeng.txt
                       wbpy.txt wbx.txt zrm.txt cj.txt)
 fcitx5_download(table-dict-download ${TABLE_DICT_URL} ${TABLE_DICT_TAR}
@@ -63,7 +64,7 @@
   string(REPLACE .txt .main.dict TABLE_DICT_FILE ${TABLE_TXT_FILE})
   add_custom_command(OUTPUT ${TABLE_DICT_FILE}
                      DEPENDS ${TABLE_TXT_FILE} LibIME::tabledict
-                     COMMAND LibIME::tabledict ${TABLE_TXT_FILE} ${CMAKE_CURRENT_BINARY_DIR}/${TABLE_DICT_FILE})
+                     COMMAND echo LibIME::tabledict ${TABLE_TXT_FILE} ${CMAKE_CURRENT_BINARY_DIR}/${TABLE_DICT_FILE})
   list(APPEND TABLE_DICT_FILES ${CMAKE_CURRENT_BINARY_DIR}/${TABLE_DICT_FILE})
 endforeach()

