diff --git a/src/vendor/Soup/build_common.php b/src/vendor/Soup/build_common.php
index 3f0a7f4..8845a4b 100644
--- a/src/vendor/Soup/build_common.php
+++ b/src/vendor/Soup/build_common.php
@@ -1,6 +1,6 @@
 <?php
 // Config
-$clang = "clang -std=c++17 -fno-rtti -DSOUP_USE_INTRIN -O3";
+$clang = getenv("CXX")." -std=c++17 -fno-rtti -DSOUP_USE_INTRIN -O3";
 if (defined("PHP_WINDOWS_VERSION_MAJOR"))
 {
 	$clang .= " -D_CRT_SECURE_NO_WARNINGS";
