diff --git a/src/loslib.cpp b/src/loslib.cpp
index e496940..3e4cb3d 100644
--- a/src/loslib.cpp
+++ b/src/loslib.cpp
@@ -111,10 +111,10 @@
 
 #include <unistd.h>
 
-#define LUA_TMPNAMBUFSIZE	32
+#define LUA_TMPNAMBUFSIZE	256
 
 #if !defined(LUA_TMPNAMTEMPLATE)
-#define LUA_TMPNAMTEMPLATE	"/tmp/lua_XXXXXX"
+#define LUA_TMPNAMTEMPLATE	"@TERMUX_PREFIX@/tmp/lua_XXXXXX"
 #endif
 
 #define lua_tmpnam(b,e) { \
