+++ ./src/vendor/Soup/build_lib.php
@@ -28,13 +28,11 @@
 }
 if(is_dir(__DIR__."/Intrin"))
 {
-	if (php_uname("m") == "aarch64"
-		|| php_uname("m") == "arm64" // For people who think different
-		)
+	if (getenv("TARGET_ARCH") == "aarch64")
 	{
 		$clang .= " -march=armv8+crypto+crc";
 	}
-	else
+	elseif (getenv("TARGET_ARCH") != "arm")
 	{
 		$clang .= " -maes -mpclmul -mrdrnd -mrdseed -msha -msse4.1";
 	}
