diff -uNr debootstrap/scripts/aequorea debootstrap.mod/scripts/aequorea
--- debootstrap/scripts/aequorea	2020-10-15 15:29:51.356809238 +0300
+++ debootstrap.mod/scripts/aequorea	2020-10-15 15:31:36.534273799 +0300
@@ -54,7 +54,7 @@
 	setup_etc
 	if [ ! -e "$TARGET/etc/fstab" ]; then
 		echo '# UNCONFIGURED FSTAB FOR BASE SYSTEM' > "$TARGET/etc/fstab"
-		chown 0:0 "$TARGET/etc/fstab"; chmod 644 "$TARGET/etc/fstab"
+		chmod 644 "$TARGET/etc/fstab"
 	fi
 
 	setup_devices
diff -uNr debootstrap/scripts/amber debootstrap.mod/scripts/amber
--- debootstrap/scripts/amber	2020-03-13 03:58:18.000000000 +0200
+++ debootstrap.mod/scripts/amber	2020-10-15 15:31:36.530273898 +0300
@@ -63,7 +63,7 @@
 	setup_etc
 	if [ ! -e "$TARGET/etc/fstab" ]; then
 		echo '# UNCONFIGURED FSTAB FOR BASE SYSTEM' > "$TARGET/etc/fstab"
-		chown 0:0 "$TARGET/etc/fstab"; chmod 644 "$TARGET/etc/fstab"
+		chmod 644 "$TARGET/etc/fstab"
 	fi
 
 	setup_devices
diff -uNr debootstrap/scripts/breezy debootstrap.mod/scripts/breezy
--- debootstrap/scripts/breezy	2018-04-01 19:11:04.000000000 +0300
+++ debootstrap.mod/scripts/breezy	2020-10-15 15:31:36.538273700 +0300
@@ -42,7 +42,7 @@
     setup_etc
     if [ ! -e "$TARGET/etc/fstab" ]; then
         echo '# UNCONFIGURED FSTAB FOR BASE SYSTEM' > "$TARGET/etc/fstab"
-        chown 0:0 "$TARGET/etc/fstab"; chmod 644 "$TARGET/etc/fstab"
+        chmod 644 "$TARGET/etc/fstab"
     fi
 
     setup_devices
diff -uNr debootstrap/scripts/dapper debootstrap.mod/scripts/dapper
--- debootstrap/scripts/dapper	2018-04-01 19:11:04.000000000 +0300
+++ debootstrap.mod/scripts/dapper	2020-10-15 15:31:36.534273799 +0300
@@ -49,7 +49,7 @@
     setup_etc
     if [ ! -e "$TARGET/etc/fstab" ]; then
         echo '# UNCONFIGURED FSTAB FOR BASE SYSTEM' > "$TARGET/etc/fstab"
-        chown 0:0 "$TARGET/etc/fstab"; chmod 644 "$TARGET/etc/fstab"
+        chmod 644 "$TARGET/etc/fstab"
     fi
 
     setup_devices
diff -uNr debootstrap/scripts/edgy debootstrap.mod/scripts/edgy
--- debootstrap/scripts/edgy	2018-08-01 10:02:13.000000000 +0300
+++ debootstrap.mod/scripts/edgy	2020-10-15 15:31:36.522274097 +0300
@@ -55,7 +55,7 @@
     setup_etc
     if [ ! -e "$TARGET/etc/fstab" ]; then
         echo '# UNCONFIGURED FSTAB FOR BASE SYSTEM' > "$TARGET/etc/fstab"
-        chown 0:0 "$TARGET/etc/fstab"; chmod 644 "$TARGET/etc/fstab"
+        chmod 644 "$TARGET/etc/fstab"
     fi
 
     setup_devices
diff -uNr debootstrap/scripts/feisty debootstrap.mod/scripts/feisty
--- debootstrap/scripts/feisty	2018-08-01 10:02:13.000000000 +0300
+++ debootstrap.mod/scripts/feisty	2020-10-15 15:31:36.526273998 +0300
@@ -54,7 +54,7 @@
     setup_etc
     if [ ! -e "$TARGET/etc/fstab" ]; then
         echo '# UNCONFIGURED FSTAB FOR BASE SYSTEM' > "$TARGET/etc/fstab"
-        chown 0:0 "$TARGET/etc/fstab"; chmod 644 "$TARGET/etc/fstab"
+        chmod 644 "$TARGET/etc/fstab"
     fi
 
     setup_devices
diff -uNr debootstrap/scripts/gutsy debootstrap.mod/scripts/gutsy
--- debootstrap/scripts/gutsy	2020-10-15 15:29:51.356809238 +0300
+++ debootstrap.mod/scripts/gutsy	2020-10-15 15:31:36.550273401 +0300
@@ -94,7 +94,7 @@
 	setup_etc
 	if [ ! -e "$TARGET/etc/fstab" ]; then
 		echo '# UNCONFIGURED FSTAB FOR BASE SYSTEM' > "$TARGET/etc/fstab"
-		chown 0:0 "$TARGET/etc/fstab"; chmod 644 "$TARGET/etc/fstab"
+		chmod 644 "$TARGET/etc/fstab"
 	fi
 
 	setup_devices
diff -uNr debootstrap/scripts/hoary debootstrap.mod/scripts/hoary
--- debootstrap/scripts/hoary	2017-09-04 13:44:05.000000000 +0300
+++ debootstrap.mod/scripts/hoary	2020-10-15 15:31:36.526273998 +0300
@@ -62,7 +62,7 @@
     setup_etc
     if [ ! -e "$TARGET/etc/fstab" ]; then
         echo '# UNCONFIGURED FSTAB FOR BASE SYSTEM' > "$TARGET/etc/fstab"
-        chown 0:0 "$TARGET/etc/fstab"; chmod 644 "$TARGET/etc/fstab"
+        chmod 644 "$TARGET/etc/fstab"
     fi
 
     setup_devices
diff -uNr debootstrap/scripts/hoary.buildd debootstrap.mod/scripts/hoary.buildd
--- debootstrap/scripts/hoary.buildd	2017-09-04 13:44:05.000000000 +0300
+++ debootstrap.mod/scripts/hoary.buildd	2020-10-15 15:31:36.530273898 +0300
@@ -58,7 +58,7 @@
     setup_etc
     if [ ! -e "$TARGET/etc/fstab" ]; then
         echo '# UNCONFIGURED FSTAB FOR BASE SYSTEM' > "$TARGET/etc/fstab"
-        chown 0:0 "$TARGET/etc/fstab"; chmod 644 "$TARGET/etc/fstab"
+        chmod 644 "$TARGET/etc/fstab"
     fi
 
     setup_devices
diff -uNr debootstrap/scripts/potato debootstrap.mod/scripts/potato
--- debootstrap/scripts/potato	2020-10-15 15:29:51.356809238 +0300
+++ debootstrap.mod/scripts/potato	2020-10-15 15:31:36.546273501 +0300
@@ -39,7 +39,7 @@
 
     setup_etc
     echo '# UNCONFIGURED FSTAB FOR BASE SYSTEM' > "$TARGET/etc/fstab"
-    chown 0:0 "$TARGET/etc/fstab"; chmod 644 "$TARGET/etc/fstab"
+    chmod 644 "$TARGET/etc/fstab"
 
     x_feign_install () {
         local pkg=$1
diff -uNr debootstrap/scripts/sarge debootstrap.mod/scripts/sarge
--- debootstrap/scripts/sarge	2020-10-15 15:29:51.356809238 +0300
+++ debootstrap.mod/scripts/sarge	2020-10-15 15:31:36.550273401 +0300
@@ -106,7 +106,7 @@
     setup_etc
     if [ ! -e "$TARGET/etc/fstab" ]; then
         echo '# UNCONFIGURED FSTAB FOR BASE SYSTEM' > "$TARGET/etc/fstab"
-        chown 0:0 "$TARGET/etc/fstab"; chmod 644 "$TARGET/etc/fstab"
+        chmod 644 "$TARGET/etc/fstab"
     fi
 
     setup_devices
diff -uNr debootstrap/scripts/sarge.buildd debootstrap.mod/scripts/sarge.buildd
--- debootstrap/scripts/sarge.buildd	2017-09-04 13:44:05.000000000 +0300
+++ debootstrap.mod/scripts/sarge.buildd	2020-10-15 15:31:36.550273401 +0300
@@ -65,7 +65,7 @@
     setup_etc
     if [ ! -e "$TARGET/etc/fstab" ]; then
         echo '# UNCONFIGURED FSTAB FOR BASE SYSTEM' > "$TARGET/etc/fstab"
-        chown 0:0 "$TARGET/etc/fstab"; chmod 644 "$TARGET/etc/fstab"
+        chmod 644 "$TARGET/etc/fstab"
     fi
 
     setup_devices
diff -uNr debootstrap/scripts/sarge.fakechroot debootstrap.mod/scripts/sarge.fakechroot
--- debootstrap/scripts/sarge.fakechroot	2018-08-01 10:02:13.000000000 +0300
+++ debootstrap.mod/scripts/sarge.fakechroot	2020-10-15 15:31:36.546273501 +0300
@@ -70,7 +70,7 @@
     setup_etc
     if [ ! -e "$TARGET/etc/fstab" ]; then
         echo '# UNCONFIGURED FSTAB FOR BASE SYSTEM' > "$TARGET/etc/fstab"
-        chown 0:0 "$TARGET/etc/fstab"; chmod 644 "$TARGET/etc/fstab"
+        chmod 644 "$TARGET/etc/fstab"
     fi
 
     x_feign_install () {
diff -uNr debootstrap/scripts/warty debootstrap.mod/scripts/warty
--- debootstrap/scripts/warty	2017-09-04 13:44:05.000000000 +0300
+++ debootstrap.mod/scripts/warty	2020-10-15 15:31:36.526273998 +0300
@@ -50,7 +50,7 @@
     setup_etc
     if [ ! -e "$TARGET/etc/fstab" ]; then
         echo '# UNCONFIGURED FSTAB FOR BASE SYSTEM' > "$TARGET/etc/fstab"
-        chown 0:0 "$TARGET/etc/fstab"; chmod 644 "$TARGET/etc/fstab"
+        chmod 644 "$TARGET/etc/fstab"
     fi
 
     setup_devices
diff -uNr debootstrap/scripts/warty.buildd debootstrap.mod/scripts/warty.buildd
--- debootstrap/scripts/warty.buildd	2017-09-04 13:44:05.000000000 +0300
+++ debootstrap.mod/scripts/warty.buildd	2020-10-15 15:31:36.534273799 +0300
@@ -58,7 +58,7 @@
     setup_etc
     if [ ! -e "$TARGET/etc/fstab" ]; then
         echo '# UNCONFIGURED FSTAB FOR BASE SYSTEM' > "$TARGET/etc/fstab"
-        chown 0:0 "$TARGET/etc/fstab"; chmod 644 "$TARGET/etc/fstab"
+        chmod 644 "$TARGET/etc/fstab"
     fi
 
     setup_devices
diff -uNr debootstrap/scripts/woody debootstrap.mod/scripts/woody
--- debootstrap/scripts/woody	2020-10-15 15:29:51.360809145 +0300
+++ debootstrap.mod/scripts/woody	2020-10-15 15:31:36.554273302 +0300
@@ -92,7 +92,7 @@
     setup_etc
     if [ ! -e "$TARGET/etc/fstab" ]; then
         echo '# UNCONFIGURED FSTAB FOR BASE SYSTEM' > "$TARGET/etc/fstab"
-        chown 0:0 "$TARGET/etc/fstab"; chmod 644 "$TARGET/etc/fstab"
+        chmod 644 "$TARGET/etc/fstab"
     fi
 
     setup_devices
diff -uNr debootstrap/scripts/woody.buildd debootstrap.mod/scripts/woody.buildd
--- debootstrap/scripts/woody.buildd	2020-10-15 15:29:51.360809145 +0300
+++ debootstrap.mod/scripts/woody.buildd	2020-10-15 15:31:36.562273103 +0300
@@ -67,7 +67,7 @@
     setup_etc
     if [ ! -e "$TARGET/etc/fstab" ]; then
         echo '# UNCONFIGURED FSTAB FOR BASE SYSTEM' > "$TARGET/etc/fstab"
-        chown 0:0 "$TARGET/etc/fstab"; chmod 644 "$TARGET/etc/fstab"
+        chmod 644 "$TARGET/etc/fstab"
     fi
 
     setup_devices
