diff -uNr debootstrap-1.0.106/scripts/debian-common debootstrap-1.0.106.mod/scripts/debian-common
--- debootstrap-1.0.106/scripts/debian-common	2018-07-05 11:48:22.000000000 +0300
+++ debootstrap-1.0.106.mod/scripts/debian-common	2018-07-10 15:33:41.289906934 +0300
@@ -60,7 +60,7 @@
 	setup_etc
 	if [ ! -e "$TARGET/etc/fstab" ]; then
 		echo '# UNCONFIGURED FSTAB FOR BASE SYSTEM' > "$TARGET/etc/fstab"
-		chown 0:0 "$TARGET/etc/fstab"; chmod 644 "$TARGET/etc/fstab"
+		chmod 644 "$TARGET/etc/fstab"
 	fi
 
 	setup_devices
@@ -102,7 +102,6 @@
 	}
 
 	if ! doing_variant fakechroot; then
-		setup_proc
 		in_target /sbin/ldconfig
 	fi
 
@@ -172,6 +171,20 @@
 echo \"Warning: Fake start-stop-daemon called, doing nothing\"" > "$TARGET/sbin/start-stop-daemon"
 	chmod 755 "$TARGET/sbin/start-stop-daemon"
 
+	# Replace problematic binaries with a stub
+	echo "" > "$TARGET/bin/chown"
+	echo "" > "$TARGET/usr/sbin/groupadd"
+	echo "" > "$TARGET/usr/sbin/groupdel"
+	echo "" > "$TARGET/usr/sbin/groupmod"
+	echo "" > "$TARGET/usr/sbin/useradd"
+	echo "" > "$TARGET/usr/sbin/userdel"
+	echo "" > "$TARGET/usr/sbin/usermod"
+	echo "" > "$TARGET/usr/bin/chage"
+	echo "" > "$TARGET/usr/bin/chfn"
+	echo "" > "$TARGET/usr/bin/dpkg-statoverride"
+
+	echo "nameserver 8.8.8.8\nnameserver 8.8.4.4" > "$TARGET/etc/resolv.conf"
+
 	setup_dselect_method apt
 
 	smallyes '' |
