--- a/src/basedir.c
+++ b/src/basedir.c
@@ -91,9 +91,9 @@
 static const char
 	DefaultRelativeDataHome[] = DIR_SEPARATOR_STR ".local" DIR_SEPARATOR_STR "share",
 	DefaultRelativeConfigHome[] = DIR_SEPARATOR_STR ".config",
-	DefaultDataDirectories1[] = DIR_SEPARATOR_STR "usr" DIR_SEPARATOR_STR "local" DIR_SEPARATOR_STR "share",
-	DefaultDataDirectories2[] = DIR_SEPARATOR_STR "usr" DIR_SEPARATOR_STR "share",
-	DefaultConfigDirectories[] = DIR_SEPARATOR_STR "etc" DIR_SEPARATOR_STR "xdg",
+	DefaultDataDirectories1[] = "@TERMUX_PREFIX@" DIR_SEPARATOR_STR "local" DIR_SEPARATOR_STR "share",
+	DefaultDataDirectories2[] = "@TERMUX_PREFIX@" DIR_SEPARATOR_STR "share",
+	DefaultConfigDirectories[] = "@TERMUX_PREFIX@" DIR_SEPARATOR_STR "etc" DIR_SEPARATOR_STR "xdg",
 	DefaultRelativeCacheHome[] = DIR_SEPARATOR_STR ".cache";
 
 static const char
