diff -u -r ../libuuid-1.0.3/gen_uuid.c ./gen_uuid.c
--- ../libuuid-1.0.3/gen_uuid.c	2014-08-12 10:07:18.000000000 +0200
+++ ./gen_uuid.c	2017-07-07 01:46:59.528994512 +0200
@@ -38,6 +38,8 @@
  */
 #define _SVID_SOURCE
 
+#include "c.h"
+
 #ifdef _WIN32
 #define _WIN32_WINNT 0x0500
 #include <windows.h>
@@ -91,7 +97,6 @@
 #include "uuidP.h"
 #include "uuidd.h"
 #include "randutils.h"
-#include "c.h"
 
 #ifdef HAVE_TLS
 #define THREAD_LOCAL static __thread
