diff -uNr libuuid-1.0.3/uuidP.h libuuid-1.0.3.mod/uuidP.h
--- libuuid-1.0.3/uuidP.h	2014-08-12 11:07:18.000000000 +0300
+++ libuuid-1.0.3.mod/uuidP.h	2018-06-19 16:34:52.864262517 +0300
@@ -37,7 +37,7 @@
 
 #include "uuid.h"
 
-#define LIBUUID_CLOCK_FILE	"/var/lib/libuuid/clock.txt"
+#define LIBUUID_CLOCK_FILE	"@TERMUX_PREFIX@/var/lib/libuuid/clock.txt"
 
 /*
  * Offset between 15-Oct-1582 and 1-Jan-70
