diff -uNr kirigami2-5.99.0/src/libkirigami/kirigamipluginfactory.cpp kirigami2-5.99.0.mod/src/libkirigami/kirigamipluginfactory.cpp
--- kirigami2-5.99.0/src/libkirigami/kirigamipluginfactory.cpp	2022-10-01 22:20:15.000000000 +0800
+++ kirigami2-5.99.0.mod/src/libkirigami/kirigamipluginfactory.cpp	2022-11-12 16:57:50.498159341 +0800
@@ -7,7 +7,7 @@
 #include "kirigamipluginfactory.h"
 
 #include <QDebug>
-
+#include <QCoreApplication>
 #include <QDir>
 #include <QQuickStyle>
 #include <QPluginLoader>
