--- src/CMakeLists.txt	2020-06-07 19:29:04.000000000 +0100
+++ src.mod/CMakeLists.txt	2021-05-19 13:56:28.854533182 +0100
@@ -98,23 +98,16 @@
 )
 
 #use dbus on linux, bsd etc, but not andoid and apple stuff
-if (UNIX AND NOT ANDROID AND NOT(APPLE) AND NOT(DISABLE_DBUS))
+if (UNIX AND NOT(APPLE) AND NOT(DISABLE_DBUS))
     find_package(Qt5DBus)
     add_definitions(-DKIRIGAMI_ENABLE_DBUS)
 endif()
 
-if(BUILD_EXAMPLES AND ANDROID)
-    find_package(Qt5AndroidExtras ${REQUIRED_QT_VERSION} REQUIRED)
-endif()
-
-
 ecm_find_qmlmodule(QtGraphicalEffects 1.0)
 
 add_definitions(-DQT_DISABLE_DEPRECATED_BEFORE=0x050d00)
 add_subdirectory(src)
-if (NOT ANDROID)
-    add_subdirectory(templates)
-endif()
+add_subdirectory(templates)
 
 if (BUILD_EXAMPLES AND BUILD_SHARED_LIBS)
     add_subdirectory(examples)
--- src/src/libkirigami/CMakeLists.txt	2020-06-07 19:29:04.000000000 +0100
+++ src.mod/src/libkirigami/CMakeLists.txt	2021-05-19 13:59:42.909309847 +0100
@@ -8,7 +8,7 @@
 )
 
 #use dbus on linux, bsd etc, but not andoid and apple stuff
-if (UNIX AND NOT ANDROID AND NOT(APPLE) AND NOT(DISABLE_DBUS))
+if (UNIX AND NOT(APPLE) AND NOT(DISABLE_DBUS))
     qt5_add_dbus_interface(libkirigami_SRCS org.kde.KWin.TabletModeManager.xml tabletmodemanager_interface)
     set(LIBKIRIGAMKI_EXTRA_LIBS Qt5::DBus)
 endif()
