diff -uNr lzlib-1.11/configure lzlib-1.11.mod/configure
--- lzlib-1.11/configure	2019-01-02 13:51:19.000000000 +0200
+++ lzlib-1.11.mod/configure	2019-09-23 19:34:31.423708870 +0300
@@ -28,10 +28,10 @@
 infodir='$(datarootdir)/info'
 libdir='$(exec_prefix)/lib'
 mandir='$(datarootdir)/man'
-CC=gcc
-CPPFLAGS=
-CFLAGS='-Wall -W -O2'
-LDFLAGS=
+CC?=gcc
+CPPFLAGS?=
+CFLAGS?='-Wall -W -O2'
+LDFLAGS?=
 
 # checking whether we are using GNU C.
 /bin/sh -c "${CC} --version" > /dev/null 2>&1 ||
