diff -u -r ../gnushogi-1.4.2/configure ./configure
--- ../gnushogi-1.4.2/configure	2014-02-17 21:30:14.000000000 +0100
+++ ./configure	2014-02-25 23:37:49.000000000 +0100
@@ -3315,7 +3315,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_curses_clrtoeol" >&5
 $as_echo "$ac_cv_lib_curses_clrtoeol" >&6; }
 if test "x$ac_cv_lib_curses_clrtoeol" = xyes; then :
-  LIBCURSES=-lcurses
+  LIBCURSES=-lncurses
 
      CURSESDSP=cursesdsp.o
 
