--- mtr/ui/asn.c        2017-06-02 08:54:58.000000000 +0000
+++ ./ui/asn.c          2017-10-21 22:21:43.076825733 +0000
@@ -41,7 +41,7 @@
 #include <resolv.h>
 #include <string.h>
 #include <sys/socket.h>
-#include <search.h>
+#include "portability/search.h"
 
 #include "mtr.h"
 #include "asn.h"
--- mtr/Makefile.am     2017-06-02 08:54:58.000000000 +0000
+++ ./Makefile.am       2017-10-21 22:39:26.628914946 +0000
@@ -59,6 +59,12 @@
               img/mtr_icon.xpm \
               ui/mtr-gtk.h
 
+mtr_SOURCES += \
+	portability/hcreate.c \
+	portability/hcreate_r.c \
+	portability/hsearch_r.c \
+	portability/hdestroy_r.c
+
 if WITH_ERROR
 mtr_SOURCES += \
 	portability/error.h \

