diff -uNr abuild-3.2.0/abuild-fetch.c abuild-3.2.0.mod/abuild-fetch.c
--- abuild-3.2.0/abuild-fetch.c	2018-06-22 10:24:10.000000000 +0300
+++ abuild-3.2.0.mod/abuild-fetch.c	2018-11-18 13:46:36.361440934 +0200
@@ -163,7 +163,7 @@
 int main(int argc, char *argv[])
 {
 	int opt, r=0, i;
-	char *destdir = "/var/cache/distfiles";
+	char *destdir = "@TERMUX_PREFIX@/var/cache/distfiles";
 
 	program = argv[0];
 	while ((opt = getopt(argc, argv, "hd:")) != -1) {
