--- abuild-3.2.0/abuild-sign.in	2018-06-22 10:24:10.000000000 +0300
+++ abuild-3.2.0.mod/abuild-sign.in	2018-11-18 20:28:48.283970248 +0200
@@ -51,7 +51,7 @@
 		  -e, --installed    Check only of there exist a private key for signing
 		  -k, --private KEY  The private key to use for signing
 		  -p, --public KEY   The name of public key. apk add will look for
-		                     /etc/apk/keys/KEY
+		                     \$PREFIX/etc/apk/keys/KEY
 		  -q, --quiet
 		  -h, --help         Show this help
 
