diff -uNr openssh-portable-V_9_5_P1/auth2-passwd.c openssh-portable-V_9_5_P1.mod/auth2-passwd.c
--- openssh-portable-V_9_5_P1/auth2-passwd.c	2023-10-04 07:34:10.000000000 +0300
+++ openssh-portable-V_9_5_P1.mod/auth2-passwd.c	2023-11-23 16:29:34.257875879 +0200
@@ -32,6 +32,8 @@
 #include <stdarg.h>
 #include <stdio.h>
 
+#include <termux-auth.h>
+
 #include "packet.h"
 #include "ssherr.h"
 #include "log.h"
@@ -66,7 +68,7 @@
 
 	if (change)
 		logit("password change not supported");
-	else if (PRIVSEP(auth_password(ssh, password)) == 1)
+	else if (termux_auth(((Authctxt *)ssh->authctxt)->user, password))
 		authenticated = 1;
 	freezero(password, len);
 	return authenticated;
