diff -u -r ../openssh-6.4p1/config.h.in ./config.h.in
--- ../openssh-6.4p1/config.h.in	2013-11-08 02:41:08.000000000 +0100
+++ ./config.h.in	2014-02-03 04:32:13.000000000 +0100
@@ -336,7 +336,7 @@
 #undef HAVE_DES_CRYPT
 
 /* Define if you have /dev/ptmx */
-#undef HAVE_DEV_PTMX
+#define HAVE_DEV_PTMX 1
 
 /* Define if you have /dev/ptc */
 #undef HAVE_DEV_PTS_AND_PTC
diff -u -r ../openssh-6.4p1/openbsd-compat/bsd-openpty.c ./openbsd-compat/bsd-openpty.c
--- ../openssh-6.4p1/openbsd-compat/bsd-openpty.c	2006-08-24 11:52:30.000000000 +0200
+++ ./openbsd-compat/bsd-openpty.c	2014-02-03 04:33:42.000000000 +0100
@@ -121,15 +121,17 @@
 		return (-1);
 	}
 
+# ifndef __ANDROID__
 	/*
 	 * Try to push the appropriate streams modules, as described 
 	 * in Solaris pts(7).
 	 */
 	ioctl(*aslave, I_PUSH, "ptem");
 	ioctl(*aslave, I_PUSH, "ldterm");
-# ifndef __hpux
+#  ifndef __hpux
 	ioctl(*aslave, I_PUSH, "ttcompat");
-# endif /* __hpux */
+#  endif /* __hpux */
+# endif /* __ANDROID__ */
 
 	return (0);
 
