diff -uNr openssh-portable-V_9_5_P1/servconf.c openssh-portable-V_9_5_P1.mod/servconf.c
--- openssh-portable-V_9_5_P1/servconf.c	2023-10-04 07:34:10.000000000 +0300
+++ openssh-portable-V_9_5_P1.mod/servconf.c	2023-11-23 16:29:34.267876057 +0200
@@ -299,7 +299,7 @@
 	}
 	/* No certificates by default */
 	if (options->num_ports == 0)
-		options->ports[options->num_ports++] = SSH_DEFAULT_PORT;
+		options->ports[options->num_ports++] = 8022 /*SSH_DEFAULT_PORT*/;
 	if (options->address_family == -1)
 		options->address_family = AF_UNSPEC;
 	if (options->listen_addrs == NULL)
@@ -450,9 +450,13 @@
 
 	assemble_algorithms(options);
 
-	/* Turn privilege separation and sandboxing on by default */
 	if (use_privsep == -1)
+#ifdef __ANDROID__
+		use_privsep = PRIVSEP_OFF;
+#else
+		/* Turn privilege separation and sandboxing on by default */
 		use_privsep = PRIVSEP_ON;
+#endif
 
 #define CLEAR_ON_NONE(v) \
 	do { \
@@ -887,7 +891,7 @@
 	struct queued_listenaddr *qla;
 
 	if (options->num_ports == 0)
-		options->ports[options->num_ports++] = SSH_DEFAULT_PORT;
+		options->ports[options->num_ports++] = 8022 /*SSH_DEFAULT_PORT*/;
 	if (options->address_family == -1)
 		options->address_family = AF_UNSPEC;
 
