diff -uNr openssh-portable-V_9_5_P1/serverloop.c openssh-portable-V_9_5_P1.mod/serverloop.c
--- openssh-portable-V_9_5_P1/serverloop.c	2023-10-04 07:34:10.000000000 +0300
+++ openssh-portable-V_9_5_P1.mod/serverloop.c	2023-11-23 16:29:34.271209449 +0200
@@ -827,8 +827,7 @@
 		/* check permissions */
 		if ((options.allow_streamlocal_forwarding & FORWARD_REMOTE) == 0
 		    || !auth_opts->permit_port_forwarding_flag ||
-		    options.disable_forwarding ||
-		    (pw->pw_uid != 0 && !use_privsep)) {
+		    options.disable_forwarding) {
 			success = 0;
 			ssh_packet_send_debug(ssh, "Server has disabled "
 			    "streamlocal forwarding.");
