diff -u -r ../openssh-8.2p1/ssh-agent.c ./ssh-agent.c
--- ../openssh-8.2p1/ssh-agent.c	2020-02-14 00:40:54.000000000 +0000
+++ ./ssh-agent.c	2020-02-15 22:01:29.025352000 +0000
@@ -93,7 +93,7 @@
 #include "sk-api.h"
 
 #ifndef DEFAULT_PROVIDER_WHITELIST
-# define DEFAULT_PROVIDER_WHITELIST "/usr/lib*/*,/usr/local/lib*/*"
+# define DEFAULT_PROVIDER_WHITELIST "/data/data/com.termux/files/usr/lib*/*"
 #endif
 
 /* Maximum accepted message length */
@@ -1199,9 +1199,11 @@
 	/* Ensure that fds 0, 1 and 2 are open or directed to /dev/null */
 	sanitise_stdfd();
 
+#ifndef __ANDROID__
 	/* drop */
 	setegid(getgid());
 	setgid(getgid());
+#endif
 
 	platform_disable_tracing(0);	/* strict=no */
 
