--- a/sshd.0
+++ b/sshd.0
@@ -661,8 +661,7 @@ FILES
      /var/empty
              chroot(2) directory used by sshd during privilege separation in
              the pre-authentication phase.  The directory should not contain
-             any files and must be owned by root and not group or world-
-             writable.
+             any files.
 
      /var/run/sshd.pid
              Contains the process ID of the sshd listening for connections (if