diff -u -r ../openssh-6.5p1/sftp-server.c ./sftp-server.c
--- ../openssh-6.5p1/sftp-server.c	2014-01-19 05:25:35.000000000 +0100
+++ ./sftp-server.c	2014-02-11 11:06:33.000000000 +0100
@@ -1259,7 +1259,7 @@
 	debug3("request %u: statvfs", id);
 	logit("statvfs \"%s\"", path);
 
-	if (statvfs(path, &st) != 0)
+	if (statfs(path, &st) != 0)
 		send_status(id, errno_to_portable(errno));
 	else
 		send_statvfs(id, &st);
@@ -1279,7 +1279,7 @@
 		send_status(id, SSH2_FX_FAILURE);
 		return;
 	}
-	if (fstatvfs(fd, &st) != 0)
+	if (fstat(fd, &st) != 0)
 		send_status(id, errno_to_portable(errno));
 	else
 		send_statvfs(id, &st);
