diff -uNr faust-2.70.3/compiler/tlib/compatibility.cpp faust-2.70.3.mod/compiler/tlib/compatibility.cpp
--- faust-2.70.3/compiler/tlib/compatibility.cpp	2024-01-05 16:34:56.000000000 +0800
+++ faust-2.70.3.mod/compiler/tlib/compatibility.cpp	2024-01-29 10:13:44.602047935 +0800
@@ -275,7 +275,7 @@
             strncpy(str, path, size);
         } else {
             // prevent the case of _ undefined
-            strncpy(str, "/usr/local/bin/faust", size);
+            strncpy(str, "@TERMUX_PREFIX@/bin/faust", size);
         }
     }
 }
