diff --git a/include/czmq_prelude.h b/include/czmq_prelude.h
index 8a1d7ea7..785ae052 100644
--- a/include/czmq_prelude.h
+++ b/include/czmq_prelude.h
@@ -286,9 +286,8 @@
 #   include <sys/un.h>
 #   include <sys/uio.h>             //  Let CZMQ build with libzmq/3.x
 #   include <netinet/in.h>          //  Must come before arpa/inet.h
-#   if (!defined (__UTYPE_ANDROID)) && (!defined (__UTYPE_IBMAIX)) \
-    && (!defined (__UTYPE_HPUX))
-#       include <ifaddrs.h>
+#   if (!defined (__UTYPE_IBMAIX)) && (!defined (__UTYPE_HPUX))
+#       include <ifaddrs.h>
 #   endif
 #   if defined (__UTYPE_SUNSOLARIS) || defined (__UTYPE_SUNOS)
 #       include <sys/sockio.h>
