# error: call to undeclared function 'explicit_bzero'

--- a/kitty/disk-cache.c
+++ b/kitty/disk-cache.c
@@ -49,7 +49,7 @@
 #ifdef __APPLE__
     memset_s(cv->encryption_key, sizeof(cv->encryption_key), 0, sizeof(cv->encryption_key));
 #else
-    explicit_bzero(cv->encryption_key, sizeof(cv->encryption_key));
+    memset(cv->encryption_key, 0, sizeof(cv->encryption_key));
 #endif
     free(cv->data); cv->data = NULL;
     free(cv);
