diff -u -r ../kitty-0.36.4/setup.py ./setup.py
--- ../kitty-0.36.4/setup.py	2024-09-27 04:50:49.000000000 +0000
+++ ./setup.py	2024-09-27 22:13:43.049667431 +0000
@@ -963,7 +963,7 @@
 
 
 def compile_glfw(compilation_database: CompilationDatabase, build_dsym: bool = False) -> None:
-    modules = 'cocoa' if is_macos else 'x11 wayland'
+    modules = 'cocoa' if is_macos else 'x11'
     for module in modules.split():
         try:
             genv = glfw.init_env(env, pkg_config, pkg_version, at_least_version, test_compile, module)
