diff -u -r ../zstd-1.3.6/programs/fileio.c ./programs/fileio.c
--- ../zstd-1.3.6/programs/fileio.c	2018-10-05 16:34:52.000000000 +0000
+++ ./programs/fileio.c	2018-10-08 00:53:00.087407054 +0000
@@ -21,7 +21,8 @@
 #endif
 
 #if !defined(BACKTRACES_ENABLE) && \
-   (defined(__linux__) || (defined(__APPLE__) && defined(__MACH__)) )
+   (defined(__linux__) || (defined(__APPLE__) && defined(__MACH__)) ) \
+   && !defined(__ANDROID__)
 #  define BACKTRACES_ENABLE 1
 #endif
 
