diff --git a/main.go b/main.go
index 553961a..31ef2f8 100644
--- a/main.go
+++ b/main.go
@@ -342,8 +342,8 @@ All long form (--) flags can be toggled with the dig-standard +[no]flag notation
 			opts.Server[0] = os.Getenv(defaultServerVar)
 			log.Debugf("Using %s from %s environment variable", opts.Server, defaultServerVar)
 		} else {
-			log.Debugf("No server specified or %s set, using /etc/resolv.conf", defaultServerVar)
-			conf, err := dns.ClientConfigFromFile("/etc/resolv.conf")
+			log.Debugf("No server specified or %s set, using @TERMUX_PREFIX@/etc/resolv.conf", defaultServerVar)
+			conf, err := dns.ClientConfigFromFile("@TERMUX_PREFIX@/etc/resolv.conf")
 			if err != nil {
 				opts.Server[0] = "https://cloudflare-dns.com/dns-query"
 				log.Debugf("no server set, using %s", opts.Server)
@@ -353,7 +353,7 @@ All long form (--) flags can be toggled with the dig-standard +[no]flag notation
 					log.Debugf("no server set, using %s", opts.Server)
 				} else {
 					opts.Server[0] = conf.Servers[0]
-					log.Debugf("found server %s from /etc/resolv.conf", opts.Server)
+					log.Debugf("found server %s from @TERMUX_PREFIX@/etc/resolv.conf", opts.Server)
 				}
 			}
 		}
