--- ../cpl_port.h.orig	2019-05-24 13:50:47.754028778 +0200
+++ ./port/cpl_port.h	2019-05-24 14:28:25.219164021 +0200
@@ -140,7 +140,7 @@
 #include <stdarg.h>
 #include <string.h>
 #include <ctype.h>
-#include <limits.h>
+#include <linux/limits.h>
 
 #include <time.h>
 
@@ -1118,7 +1118,7 @@
 #include <set>
 #include <string>
 #include <cstddef>
-#include <limits>
+#include <linux/limits>
 #include <sstream>
 #include <fstream>
 #include <algorithm>
