Submitted upstream at https://github.com/vim/vim/pull/1314

diff -u -r ../vim-8.0.0104/src/auto/configure ./src/auto/configure
--- ../vim-8.0.0104/src/auto/configure	2016-11-26 11:45:53.000000000 -0500
+++ ./src/auto/configure	2016-12-07 21:36:01.937616265 -0500
@@ -7411,7 +7411,7 @@
 	/* Check bitfields */
 	struct nbbuf {
 	unsigned int  initDone:1;
-	ushort signmaplen;
+	unsigned short signmaplen;
 	};
 
 int
