--- ../dcgettext.c.orig	2021-02-04 19:26:04.839747458 +0000
+++ ./intl/dcgettext.c	2021-02-04 19:26:39.223307963 +0000
@@ -86,12 +86,14 @@
 
 /* @@ end of prolog @@ */
 
-#ifdef _LIBC
+#if defined(_LIBC) || defined(__BIONIC__)
 /* Rename the non ANSI C functions.  This is required by the standard
    because some ANSI C functions will require linking with this object
    file and the name space must not be polluted.  */
 # define getcwd __getcwd
 # define stpcpy __stpcpy
+char *getcwd(char *, size_t);
+char *stpcpy(char *, const char *);
 #else
 # if !defined HAVE_GETCWD
 char *getwd ();
