diff -u -r ../oleo-1.99.16/lib/getopt.c ./lib/getopt.c
--- ../oleo-1.99.16/lib/getopt.c	1998-04-19 04:14:12.000000000 +0000
+++ ./lib/getopt.c	2024-05-15 09:46:20.098744039 +0000
@@ -42,6 +42,7 @@
 #endif
 
 #include <stdio.h>
+#include <string.h>
 
 /* Comment out all this code if we are using the GNU C Library, and are not
    actually compiling the library itself.  This code is part of the GNU C
