--- plotter.c	2018-02-22 08:59:06.651246905 +0000
+++ src/src/plotter.c	2018-02-22 09:00:15.471539332 +0000
@@ -2639,8 +2639,6 @@
 	case SP_PLOT_PIE:
 		sp_pie_plot_point(mg, point);
 		break;
-	default:
-		/* ?? */
 	}
 }
 
@@ -2770,8 +2768,6 @@
 	case SP_PLOT_PIE:
 		sp_pie_begin_graph(mg, scale, trans_x, trans_y);
 		break;
-	default:
-		/* ??? */
 	}
 
 	sp_first_dataset(mg);
@@ -2790,8 +2786,6 @@
 	case SP_PLOT_PIE:
 		sp_pie_end_graph(mg);
 		break;
-	default:
-		/* ??? */
 	}
 }
 
