--- a/src/qtxdg/xdgdirs.cpp
+++ b/src/qtxdg/xdgdirs.cpp
@@ -125,7 +125,7 @@
     const QString home = QFile::decodeName(qgetenv("HOME"));
 
     if (home.isEmpty())
-        return "/tmp"_L1;
+        return "@TERMUX_PREFIX@/tmp"_L1;
     else if (dir == XdgDirs::Desktop)
         fallback = "%1/%2"_L1.arg(home, "Desktop"_L1);
     else
@@ -284,8 +284,8 @@
     QStringList dirs = d.split(u':', Qt::SkipEmptyParts);
 
     if (dirs.isEmpty()) {
-        dirs.append("/usr/local/share"_L1);
-        dirs.append("/usr/share"_L1);
+        dirs.append("@TERMUX_PREFIX@/local/share"_L1);
+        dirs.append("@TERMUX_PREFIX@/share"_L1);
     } else {
         QMutableListIterator<QString> it(dirs);
         while (it.hasNext()) {
@@ -307,7 +307,7 @@
     QStringList dirs;
     const QString env = QFile::decodeName(qgetenv("XDG_CONFIG_DIRS"));
     if (env.isEmpty())
-        dirs.append("/etc/xdg"_L1);
+        dirs.append("@TERMUX_PREFIX@/etc/xdg"_L1);
     else
         dirs = env.split(u':', Qt::SkipEmptyParts);
 
