diff -uNr jbig2dec-0.19/autogen.sh jbig2dec-0.19.mod/autogen.sh
--- jbig2dec-0.19/autogen.sh	2020-03-12 14:26:37.000000000 +0200
+++ jbig2dec-0.19.mod/autogen.sh	2020-04-05 21:53:20.165790568 +0300
@@ -169,12 +169,3 @@
 
 echo "  autoconf"
 autoconf
-
-if test -z "$*"; then
-        echo "I am going to run ./configure with no arguments - if you wish "
-        echo "to pass any to it, please specify them on the $0 command line."
-else
-	echo "running ./configure $@"
-fi
-
-$srcdir/configure "$@" && echo
