--- ./Makefile	2024-04-16 20:58:39.980928203 +0800
+++ ./Makefile	2024-04-16 21:01:18.432928142 +0800
@@ -40,7 +40,6 @@
 FAST_CFLAGS=-O3 -flto=auto -fno-plt -fno-stack-protector
 ARCH_CFLAGS=-march=native
 ifeq ($(SYS), Linux)
-	CC=gcc
 	DEFS+=-std=gnu99 -DLUA_USE_LINUX
 	LIBS+=-lm -ldl -Wl,-E
 	ifeq ($(CC), musl-gcc)
@@ -226,7 +225,7 @@
 # Install
 
 # Install paths
-PREFIX=/usr/local
+PREFIX=@TERMUX_PREFIX@
 DPREFIX=$(DESTDIR)$(PREFIX)
 PREFIX_BIN=$(DPREFIX)/bin
 PREFIX_LIB=$(DPREFIX)/lib
@@ -349,7 +348,7 @@
 
 CACHE_DIR=.cache
 ifdef HOME
-	CACHE_DIR=$(HOME)/.cache/nelua
+	CACHE_DIR=@TERMUX_HOME@/.cache/nelua
 else ifdef USERPROFILE
 	CACHE_DIR=$(USERPROFILE)\\.cache\\nelua
 endif
