--- ./src/lua/luaconf.h	2024-04-16 21:05:14.688928052 +0800
+++ ./src/lua/luaconf.h	2024-04-16 21:06:20.924928027 +0800
@@ -223,8 +223,8 @@
 
 #else			/* }{ */
 
-#define LUA_ROOT	"/usr/local/"
-#define LUA_ROOT2	"/usr/"
+#define LUA_ROOT	"@TERMUX_PREFIX@/local/"
+#define LUA_ROOT2	"@TERMUX_PREFIX@/"
 #define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR "/"
 #define LUA_LDIR2	LUA_ROOT2 "share/lua/" LUA_VDIR "/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR "/"
