diff --git a/parallel b/parallel
index 873f294..4fa24cb 100644
--- a/src/parallel
+++ b/src/parallel
@@ -2131,7 +2131,7 @@ sub parse_options(@) {
     if($opt::bug) { ::die_bug("test-bug"); }
     $Global::debug = $opt::D;
     $Global::shell = $ENV{'PARALLEL_SHELL'} || parent_shell($$)
-	|| $ENV{'SHELL'} || "/bin/sh";
+	|| $ENV{'SHELL'} || "@TERMUX_PREFIX@/bin/sh";
     if(not -x $Global::shell and not which($Global::shell)) {
 	::error("Shell '$Global::shell' not found.");
 	wait_and_exit(255);
@@ -2869,14 +2869,14 @@ sub init_globals() {
     $Global::trim = 'n';
     $Global::max_jobs_running = 0;
     $Global::job_already_run = '';
-    $ENV{'TMPDIR'} ||= "/tmp";
-    $ENV{'PARALLEL_REMOTE_TMPDIR'} ||= "/tmp";
+    $ENV{'TMPDIR'} ||= "@TERMUX_PREFIX@/tmp";
+    $ENV{'PARALLEL_REMOTE_TMPDIR'} ||= "@TERMUX_PREFIX@/tmp";
     # bug #55398: set $OLDPWD when using --wd
     $ENV{'OLDPWD'} = $ENV{'PWD'};
     if(not $ENV{HOME}) {
 	# $ENV{HOME} is sometimes not set if called from PHP
-	::warning("\$HOME not set. Using /tmp.");
-	$ENV{HOME} = "/tmp";
+	::warning("\$HOME not set. Using @TERMUX_PREFIX@/tmp.");
+	$ENV{HOME} = "@TERMUX_PREFIX@/tmp";
     }
     # no warnings to allow for undefined $XDG_*
     no warnings 'uninitialized';
@@ -4688,7 +4688,7 @@ sub _run_gnu_parallel() {
     unlink $stderrname;
 
     my $pid = ::open3($stdin_fh,$stdout_fh,$stderr_fh,
-		      $0,qw(--plain --shell /bin/sh --will-cite), @args);
+		      $0,qw(--plain --shell @TERMUX_PREFIX@/bin/sh --will-cite), @args);
     if(my $writerpid = fork()) {
 	close $stdin_fh;
 	@stdout = <$stdout_fh>;
