diff -uNr strace-4.24/ipc_msgctl.c strace-4.24.mod/ipc_msgctl.c
--- strace-4.24/ipc_msgctl.c	2017-05-22 20:33:51.000000000 +0300
+++ strace-4.24.mod/ipc_msgctl.c	2018-10-03 01:14:05.344818748 +0300
@@ -75,7 +75,9 @@
 			break;
 		}
 
+#ifndef __ANDROID__
 		tprintf(", key=%u", (unsigned) msqid_ds.msg_perm.__key);
+#endif
 		printuid(", cuid=", msqid_ds.msg_perm.cuid);
 		printuid(", cgid=", msqid_ds.msg_perm.cgid);
 		tprints("}");
