--- a/src/defs.h
+++ b/src/defs.h
@@ -849,8 +849,6 @@
  * part.
  */
 # define QUOTE_HEXSTR_MASK			(0x3 << QUOTE_HEXSTR_SHIFT)
-static_assert((NUM_HEXSTR_OPTS - 1) <= (QUOTE_HEXSTR_MASK >> QUOTE_HEXSTR_SHIFT),
-	      "xflag options do not fit into QUOTE_HEXSTR_MASK");
 
 # define QUOTE_HEXSTR_NONE			(HEXSTR_NONE << QUOTE_HEXSTR_SHIFT)
 # define QUOTE_HEXSTR_ALL			(HEXSTR_ALL << QUOTE_HEXSTR_SHIFT)
