diff -u -r ../strace-4.8/strace.c ./strace.c
--- ../strace-4.8/strace.c	2013-05-28 23:49:16.000000000 +0200
+++ ./strace.c	2013-12-30 16:05:39.000000000 +0100
@@ -540,7 +540,7 @@
 tprints(const char *str)
 {
 	if (current_tcp) {
-		int n = fputs_unlocked(str, current_tcp->outf);
+		int n = fputs(str, current_tcp->outf);
 		if (n >= 0) {
 			current_tcp->curcol += strlen(str);
 			return;
@@ -917,7 +917,7 @@
 					struct tcb *cur_tcp;
 					int tid;
 
-					if (de->d_fileno == 0)
+					if (de->d_ino == 0)
 						continue;
 					/* we trust /proc filesystem */
 					tid = atoi(de->d_name);
