diff -u -r ../strace-4.12/ubi.c ./ubi.c
--- ../strace-4.12/ubi.c	2016-05-28 06:29:30.000000000 -0400
+++ ./ubi.c	2016-06-09 09:56:41.495280471 -0400
@@ -30,7 +30,7 @@
 
 /* The UBI api changes, so we have to keep a local copy */
 #include <linux/version.h>
-#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 7, 0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 7, 0) || defined __ANDROID__
 # include "ubi-user.h"
 #else
 # include <mtd/ubi-user.h>
