diff -u -r ../bash-completion-2.8/completions/man ./completions/man
--- ../bash-completion-2.8/completions/man	2018-03-17 01:02:23.000000000 +0100
+++ ./completions/man	2018-05-13 01:31:56.406265511 +0200
@@ -51,8 +51,7 @@
         return
     fi
 
-    local manpath=$( manpath 2>/dev/null || command man -w 2>/dev/null )
-    [[ -z $manpath ]] && manpath="/usr/share/man:/usr/local/share/man"
+    local manpath="@TERMUX_PREFIX@/share/man"
 
     # determine manual section to search
     local sect
