--- ./fpcsrc/compiler/systems/t_android.pas.orig	2022-06-02 16:53:16.091736014 +0000
+++ ./fpcsrc/compiler/systems/t_android.pas	2022-06-02 18:05:28.159582322 +0000
@@ -354,7 +354,6 @@
       add('    KEEP (*(.fpc .fpc.n_version .fpc.n_links))');
       add('  }');
       add('}');
-      add('INSERT AFTER .data1');

       // Define different aliases for normal and JNI libraries
       if FJNIOnLoadName <> '' then
