--- ./libpolyml/polyffi.cpp.old	2020-07-13 18:36:05.435290513 +0800
+++ ./libpolyml/polyffi.cpp	2020-07-13 18:39:49.757178225 +0800
@@ -110,6 +110,8 @@
     {"ms_cdecl", FFI_MS_CDECL},
 #elif defined(X86_WIN64)
     {"win64", FFI_WIN64},
+#elif defined(X86_64) || (defined (__x86_64__) && defined (X86_DARWIN)) || defined (__ANDROID__)
+    {"unix64", FFI_UNIX64},
 #elif defined(X86_ANY)
     {"sysv", FFI_SYSV},
     {"unix64", FFI_UNIX64},
