diff -u -r ../apt-1.6~alpha3/apt-pkg/contrib/fileutl.cc ./apt-pkg/contrib/fileutl.cc
--- ../apt-1.6~alpha3/apt-pkg/contrib/fileutl.cc	2017-10-28 17:57:05.000000000 +0200
+++ ./apt-pkg/contrib/fileutl.cc	2017-11-02 17:38:43.633482989 +0100
@@ -37,7 +37,9 @@
 #include <dirent.h>
 #include <errno.h>
 #include <fcntl.h>
+#ifndef __ANDROID__
 #include <glob.h>
+#endif
 #include <grp.h>
 #include <pwd.h>
 #include <signal.h>
@@ -2790,6 +2792,7 @@
 }
 									/*}}}*/
 
+#ifndef __ANDROID__
 // Glob - wrapper around "glob()"					/*{{{*/
 std::vector<std::string> Glob(std::string const &pattern, int flags)
 {
@@ -2815,6 +2818,7 @@
    globfree(&globbuf);
    return result;
 }
+#endif
 									/*}}}*/
 static std::string APT_NONNULL(1) GetTempDirEnv(char const * const env)	/*{{{*/
 {
diff -u -r ../apt-1.6~alpha3/apt-pkg/contrib/fileutl.h ./apt-pkg/contrib/fileutl.h
--- ../apt-1.6~alpha3/apt-pkg/contrib/fileutl.h	2017-10-28 17:57:05.000000000 +0200
+++ ./apt-pkg/contrib/fileutl.h	2017-11-02 17:37:38.518249720 +0100
@@ -262,7 +262,9 @@
 APT_HIDDEN std::string flNormalize(std::string file);
 
 // simple c++ glob
+#ifndef __ANDROID__
 std::vector<std::string> Glob(std::string const &pattern, int flags=0);
+#endif
 
 /** \brief Popen() implementation that execv() instead of using a shell
  *
