diff -u -r ../apt-1.2.12/apt-private/private-download.cc ./apt-private/private-download.cc
--- ../apt-1.2.12/apt-private/private-download.cc	2016-05-11 10:57:51.000000000 +0200
+++ ./apt-private/private-download.cc	2017-07-01 00:31:32.131776100 +0200
@@ -121,7 +121,6 @@
 									/*}}}*/
 bool CheckFreeSpaceBeforeDownload(std::string const &Dir, unsigned long long FetchBytes)/*{{{*/
 {
-   uint32_t const RAMFS_MAGIC = 0x858458f6;
    /* Check for enough free space, but only if we are actually going to
       download */
    if (_config->FindB("APT::Get::Print-URIs", false) == true ||
