diff -u -r ../apt-1.6~alpha3/CMakeLists.txt ./CMakeLists.txt
--- ../apt-1.6~alpha3/CMakeLists.txt	2017-10-28 17:57:05.000000000 +0200
+++ ./CMakeLists.txt	2017-11-02 17:42:28.162838378 +0100
@@ -33,7 +33,6 @@
 include(GNUInstallDirs)
 include(TestBigEndian)
 find_package(Threads REQUIRED)
-find_package(LFS REQUIRED)
 find_package(Iconv REQUIRED)
 
 find_package(Perl REQUIRED)
@@ -44,11 +43,6 @@
   include_directories(${Intl_INCLUDE_DIRS})
 endif()
 
-# Add large file support
-add_compile_options(${LFS_COMPILE_OPTIONS})
-add_definitions(${LFS_DEFINITIONS})
-link_libraries(${LFS_LIBRARIES})
-
 # Set compiler flags
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
@@ -69,12 +63,6 @@
 add_optional_compile_options(Wsign-promo)
 add_optional_compile_options(Wundef)
 
-# apt-ftparchive dependencies
-find_package(BerkeleyDB REQUIRED)
-if (BERKELEY_DB_FOUND)
-  set(HAVE_BDB 1)
-endif()
-
 find_package(GnuTLS REQUIRED)
 if (GNUTLS_FOUND)
   set(HAVE_GNUTLS 1)
@@ -87,22 +75,12 @@
 endif()
 
 
-find_package(BZip2 REQUIRED)
-if (BZIP2_FOUND)
-  set(HAVE_BZ2 1)
-endif()
-
 find_package(LZMA REQUIRED)
 if (LZMA_FOUND)
   set(HAVE_LZMA 1)
 endif()
 
 
-find_package(LZ4 REQUIRED)
-if (LZ4_FOUND)
-  set(HAVE_LZ4 1)
-endif()
-
 find_package(Udev)
 if (UDEV_FOUND)
   set(HAVE_UDEV 1)
@@ -184,10 +162,10 @@
   message(STATUS "Found dpkg data dir: ${DPKG_DATADIR_CMD}")
   set(DPKG_DATADIR "${DPKG_DATADIR_CMD}" CACHE PATH "dpkg data directory")
 endif()
-if (NOT DEFINED COMMON_ARCH)
-  execute_process(COMMAND dpkg-architecture -qDEB_HOST_ARCH
-                  OUTPUT_VARIABLE COMMON_ARCH OUTPUT_STRIP_TRAILING_WHITESPACE)
-endif()
+# if (NOT DEFINED COMMON_ARCH)
+#  execute_process(COMMAND dpkg-architecture -qDEB_HOST_ARCH
+                  #OUTPUT_VARIABLE COMMON_ARCH OUTPUT_STRIP_TRAILING_WHITESPACE)
+# endif()
 if (NOT DEFINED ROOT_GROUP)
   execute_process(COMMAND id -gn root
                   OUTPUT_VARIABLE ROOT_GROUP OUTPUT_STRIP_TRAILING_WHITESPACE)
@@ -215,11 +193,8 @@
 add_subdirectory(apt-inst)
 add_subdirectory(cmdline)
 add_subdirectory(completions)
-add_subdirectory(doc)
 add_subdirectory(dselect)
-add_subdirectory(ftparchive)
 add_subdirectory(methods)
-add_subdirectory(test)
 
 if (USE_NLS)
 add_subdirectory(po)
