diff -u -r ../apt-1.6~alpha3/methods/connect.cc ./methods/connect.cc
--- ../apt-1.6~alpha3/methods/connect.cc	2017-10-28 17:57:05.000000000 +0200
+++ ./methods/connect.cc	2017-11-02 17:47:18.803413361 +0100
@@ -164,7 +164,7 @@
 
    // Check the socket for an error condition
    unsigned int Err;
-   unsigned int Len = sizeof(Err);
+   socklen_t Len = sizeof(Err);
    if (getsockopt(Fd->Fd(), SOL_SOCKET, SO_ERROR, &Err, &Len) != 0)
       return _error->Errno("getsockopt",_("Failed"));
    
@@ -335,7 +335,10 @@
    if(LastHost != Host || LastPort != Port)
    {
       SrvRecords.clear();
-      if (_config->FindB("Acquire::EnableSrvRecords", true) == true)
+      /* Disable by default in Termux due to it breaking on (some) HTC
+         devices (https://github.com/termux/termux-packages/issues/99). */
+	
+	 if (_config->FindB("Acquire::EnableSrvRecords", false) == true)
       {
          GetSrvRecords(Host, DefPort, SrvRecords);
 	 // RFC2782 defines that a lonely '.' target is an abort reason
