diff -u -r ../apt-1.6~alpha3/methods/CMakeLists.txt ./methods/CMakeLists.txt
--- ../apt-1.6~alpha3/methods/CMakeLists.txt	2017-10-28 17:57:05.000000000 +0200
+++ ./methods/CMakeLists.txt	2017-11-02 17:45:55.436395981 +0100
@@ -9,11 +9,8 @@
 add_executable(copy copy.cc)
 add_executable(store store.cc)
 add_executable(gpgv gpgv.cc)
-add_executable(cdrom cdrom.cc)
 add_executable(http http_main.cc $<TARGET_OBJECTS:httplib> $<TARGET_OBJECTS:connectlib>)
 add_executable(mirror mirror.cc $<TARGET_OBJECTS:httplib> $<TARGET_OBJECTS:connectlib>)
-add_executable(ftp ftp.cc $<TARGET_OBJECTS:connectlib>)
-add_executable(rred rred.cc)
 add_executable(rsh rsh.cc)
 
 target_compile_definitions(connectlib PRIVATE ${GNUTLS_DEFINITIONS})
@@ -22,10 +19,9 @@
 # Additional libraries to link against for networked stuff
 target_link_libraries(http ${GNUTLS_LIBRARIES})
 target_link_libraries(mirror ${RESOLV_LIBRARIES} ${GNUTLS_LIBRARIES})
-target_link_libraries(ftp ${GNUTLS_LIBRARIES})
 
 # Install the library
-install(TARGETS file copy store gpgv cdrom http ftp rred rsh mirror
+install(TARGETS file copy store gpgv http rsh mirror
         RUNTIME DESTINATION ${CMAKE_INSTALL_LIBEXECDIR}/apt/methods)
 
 add_slaves(${CMAKE_INSTALL_LIBEXECDIR}/apt/methods store)
