diff -uNr wren-0.1.0/src/cli/vm.h wren-0.1.0.mod/src/cli/vm.h
--- wren-0.1.0/src/cli/vm.h	2016-05-21 20:53:05.000000000 +0300
+++ wren-0.1.0.mod/src/cli/vm.h	2019-07-01 21:53:28.116581394 +0300
@@ -1,7 +1,7 @@
 #ifndef vm_h
 #define vm_h
 
-#include "uv.h"
+#include <uv.h>
 #include "wren.h"
 
 // Executes the Wren script at [path] in a new VM.
diff -uNr wren-0.1.0/src/module/io.c wren-0.1.0.mod/src/module/io.c
--- wren-0.1.0/src/module/io.c	2016-05-21 20:53:05.000000000 +0300
+++ wren-0.1.0.mod/src/module/io.c	2019-07-01 21:53:23.563209454 +0300
@@ -1,7 +1,7 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include "uv.h"
+#include <uv.h>
 
 #include "scheduler.h"
 #include "vm.h"
diff -uNr wren-0.1.0/src/module/scheduler.c wren-0.1.0.mod/src/module/scheduler.c
--- wren-0.1.0/src/module/scheduler.c	2016-05-21 20:53:05.000000000 +0300
+++ wren-0.1.0.mod/src/module/scheduler.c	2019-07-01 21:53:03.086368797 +0300
@@ -1,7 +1,7 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include "uv.h"
+#include <uv.h>
 
 #include "scheduler.h"
 #include "wren.h"
diff -uNr wren-0.1.0/src/module/timer.c wren-0.1.0.mod/src/module/timer.c
--- wren-0.1.0/src/module/timer.c	2016-05-21 20:53:05.000000000 +0300
+++ wren-0.1.0.mod/src/module/timer.c	2019-07-01 21:53:32.426617909 +0300
@@ -1,7 +1,7 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include "uv.h"
+#include <uv.h>
 
 #include "scheduler.h"
 #include "vm.h"
