diff -ruN tauri-plugin-src-orig/src/build/mobile.rs tauri-plugin-src/src/build/mobile.rs
--- tauri-plugin-src-orig/src/build/mobile.rs	2026-09-27 15:20:06.000000000 +0000
+++ tauri-plugin-src/src/build/mobile.rs	2026-09-27 15:20:21.000000000 +0000
@@ -57,7 +57,7 @@
   #[allow(unused_variables)] ios_path: Option<PathBuf>,
 ) -> Result<()> {
   let target_os = build_var("CARGO_CFG_TARGET_OS")?;
-  let mobile = target_os == "android" || target_os == "ios";
+  let mobile = target_os == "ios";
   cfg_alias("mobile", mobile);
   cfg_alias("desktop", !mobile);
 
