diff -ruN tauri-plugin-clipboard-manager-src-orig/Cargo.toml tauri-plugin-clipboard-manager-src/Cargo.toml
--- tauri-plugin-clipboard-manager-src-orig/Cargo.toml	2026-09-21 14:13:20.000000000 +0000
+++ tauri-plugin-clipboard-manager-src/Cargo.toml	2026-09-21 14:13:35.000000000 +0000
@@ -85,7 +85,7 @@
 version = "2.5"
 features = ["build"]
 
-[target.'cfg(any(target_os = "macos", windows, target_os = "linux", target_os = "dragonfly", target_os = "freebsd", target_os = "openbsd", target_os = "netbsd"))'.dependencies.arboard]
+[target.'cfg(any(target_os = "macos", windows, target_os = "linux", target_os = "dragonfly", target_os = "freebsd", target_os = "openbsd", target_os = "netbsd", target_os = "android"))'.dependencies.arboard]
 version = "3"
 features = ["wayland-data-control"]
 
