--- ./Makefile.in.orig	2020-07-20 20:22:23.089300679 +0000
+++ ./Makefile.in	2020-07-20 20:25:25.129583303 +0000
@@ -352,7 +352,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = foreign
-SUBDIRS = config m4 term src docs man demo share
+SUBDIRS = config m4 term src man share
 EXTRA_DIST = BUGS Copyright FAQ.pdf INSTALL INSTALL.gnu \
 PATCHLEVEL PGPKEYS README RELEASE_NOTES \
 configure.vms win
