For WEXITSTATUS macro.

diff -u -r ../gnuplot-5.2.0/src/command.c ./src/command.c
--- ../gnuplot-5.2.0/src/command.c	2017-08-30 18:48:26.000000000 +0000
+++ ./src/command.c	2017-10-14 20:56:13.905856980 +0000
@@ -138,6 +138,7 @@
 int vms_ktid;			/* key table id, for translating keystrokes */
 #endif /* VMS */
 
+#include <sys/wait.h>
 
 /* static prototypes */
 static void command __PROTO((void));
