diff --git a/configure.ac b/configure.ac
index c72148d06..1cecd2faa 100644
--- a/configure.ac
+++ b/configure.ac
@@ -318,7 +318,7 @@ dnl Add /usr/local/share/zsh/site-functions if not yet present
 dnl owing to $sitefndir, whether or not explicitly given.
 dnl If not explicitly given, it hasn't been expanded yet.
 if test X$sitefndir = X/usr/local/share/zsh/site-functions || \
-   test X$sitefndir = Xno
+   test X$sitefndir = Xno || true
 then fixed_sitefndir=''
 elif test X$prefix != X/usr/local; then
   if test X$prefix = XNONE && test X$ac_default_prefix = X/usr/local; then
