--- a/configure.ac
+++ b/configure.ac
@@ -3212,7 +3212,7 @@ AH_TOP([/***** begin user configuration section *****/
 #define DEFAULT_FCEDIT "vi"
 
 /* The default prefix for temporary files */
-#define DEFAULT_TMPPREFIX "/tmp/zsh"
+#define DEFAULT_TMPPREFIX "@TERMUX_PREFIX@/tmp/zsh"
 
 /***** end of user configuration section            *****/
 /***** shouldn't have to change anything below here *****/
