diff -u -r ../zsh-5.0.5/config.h.in ./config.h.in
--- ../zsh-5.0.5/config.h.in	2013-11-30 00:28:56.000000000 +0100
+++ ./config.h.in	2014-02-03 02:21:35.000000000 +0100
@@ -25,7 +25,7 @@
 #define DEFAULT_FCEDIT "vi"
 
 /* The default prefix for temporary files */
-#define DEFAULT_TMPPREFIX "/tmp/zsh"
+#define DEFAULT_TMPPREFIX "@TERMUX_PREFIX@/tmp/zsh"
 
 /***** end of user configuration section            *****/
 /***** shouldn't have to change anything below here *****/
