--- a/build/moz.configure/rust.configure
+++ b/build/moz.configure/rust.configure
@@ -524,17 +524,27 @@ def rust_host_triple(rustc, host, rustc_host, rust_supported_targets, arm_target
     assert_rust_compile(host, rustc_target, rustc)
     return rustc_target
 
+option("--custom-rust-target-triple", nargs=1, help="Customize Rust Target Triple")
 
-@depends(rustc, target, rust_supported_targets, arm_target, when=rust_compiler)
+@depends("--custom-rust-target-triple")
+def get_custom_rust_target_triple(custom_rust_target_triple):
+    if custom_rust_target_triple:
+        return custom_rust_target_triple[0]
+    return None
+
+@depends(get_custom_rust_target_triple, rustc, target, rust_supported_targets, arm_target, when=rust_compiler)
 @checking("for rust target triplet")
-def rust_target_triple(rustc, target, rust_supported_targets, arm_target):
-    rustc_target = detect_rustc_target(target, arm_target, rust_supported_targets)
+def rust_target_triple(custom_rust_target_triple, rustc, target, rust_supported_targets, arm_target):
+    if custom_rust_target_triple:
+        rustc_target = custom_rust_target_triple
+    else:
+        rustc_target = detect_rustc_target(target, arm_target, rust_supported_targets)
     assert_rust_compile(target, rustc_target, rustc)
     return rustc_target
 
 
 set_config("RUST_TARGET", rust_target_triple)
 set_config("RUST_HOST_TARGET", rust_host_triple)
 
 
 # This is used for putting source info into symbol files.
@@ -618,8 +618,6 @@ set_define("MOZ_DEBUG_RUST", debug_rust)
 
 # ==============================================================
 
-option(env="RUSTFLAGS", nargs=1, help="Rust compiler flags")
-set_config("RUSTFLAGS", depends("RUSTFLAGS")(lambda flags: flags))
 
 
 # Rust compiler flags
