--- a/dom/media/webaudio/FFTBlock.h
+++ b/dom/media/webaudio/FFTBlock.h
@@ -7,6 +7,10 @@
 #ifndef FFTBlock_h_
 #define FFTBlock_h_
 
+#if defined(BUILD_ARM_NEON) && defined(__TERMUX__)
+#undef BUILD_ARM_NEON
+#endif
+
 #ifdef BUILD_ARM_NEON
 #  include <cmath>
 #  include "mozilla/arm.h"
--- a/gfx/cairo/libpixman/src/pixman-arm.c
+++ b/gfx/cairo/libpixman/src/pixman-arm.c
@@ -34,6 +34,12 @@
     ARM_IWMMXT		= (1 << 4)
 } arm_cpu_features_t;
 
+#ifdef __TERMUX__
+#undef USE_ARM_SIMD
+#undef USE_ARM_NEON
+#undef USE_ARM_IWMMXT
+#endif
+
 #if defined(USE_ARM_SIMD) || defined(USE_ARM_NEON) || defined(USE_ARM_IWMMXT)
 
 #if defined(_MSC_VER)
--- a/media/libtheora/moz.build
+++ b/media/libtheora/moz.build
@@ -80,7 +80,7 @@
         ]
 
 if CONFIG['GNU_AS']:
-    if CONFIG['CPU_ARCH'] == 'arm':
+    if False:
         SOURCES += [
             'lib/arm/armcpu.c',
             'lib/arm/armstate.c',
--- a/media/openmax_dl/dl/moz.build
+++ b/media/openmax_dl/dl/moz.build
@@ -4,7 +4,7 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-if CONFIG['CPU_ARCH'] == 'arm' and CONFIG['BUILD_ARM_NEON']:
+if False:
     Library('openmax_dl')
 
     EXPORTS.dl.api += [
--- a/third_party/aom/aom_ports/arm_cpudetect.c
+++ b/third_party/aom/aom_ports/arm_cpudetect.c
@@ -87,7 +87,7 @@
   return flags & mask;
 }
 
-#elif defined(__ANDROID__) /* end _MSC_VER */
+#elif defined(__ANDROID__) && !defined(__TERMUX__) /* end _MSC_VER */
 #include <cpu-features.h>
 
 int aom_arm_cpu_caps(void) {
