--- a/build/moz.configure/rust.configure
+++ b/build/moz.configure/rust.configure
@@ -414,7 +414,7 @@
     rustc_target = find_candidate(candidates)
 
     if rustc_target is None:
-        die("Don't know how to translate {} for rustc".format(host_or_target.alias))
+        rustc_target = "@CARGO_TARGET_NAME@"
 
     return rustc_target
 
@@ -599,8 +599,6 @@
 
 # ==============================================================
 
-option(env="RUSTFLAGS", nargs=1, help="Rust compiler flags")
-set_config("RUSTFLAGS", depends("RUSTFLAGS")(lambda flags: flags))
 
 
 # Rust compiler flags
