--- ./coregrind/vg_preloaded.c.orig	2019-05-08 23:37:08.483459377 +0000
+++ ./coregrind/vg_preloaded.c	2019-05-08 23:37:43.340001515 +0000
@@ -61,7 +61,8 @@
       && !defined(VGPV_arm_linux_android) \
       && !defined(VGPV_x86_linux_android) \
       && !defined(VGPV_mips32_linux_android) \
-      && !defined(VGPV_arm64_linux_android)
+      && !defined(VGPV_arm64_linux_android) \
+      && !defined(VGPV_amd64_linux_android)
 
    /* g++ mangled __gnu_cxx::__freeres yields -> _ZN9__gnu_cxx9__freeresEv */
    extern void _ZN9__gnu_cxx9__freeresEv(void) __attribute__((weak));
