diff -u -r ../orig-src/tools/clang/tools/CMakeLists.txt ./tools/clang/tools/CMakeLists.txt
--- ../orig-src/tools/clang/tools/CMakeLists.txt       2014-03-06 11:07:50.000000000 +0100
+++ ./tools/clang/tools/CMakeLists.txt 2014-03-06 11:06:10.000000000 +0100
@@ -3,7 +3,6 @@
 add_subdirectory(clang-format)
 add_subdirectory(clang-format-vs)
 
-add_subdirectory(c-index-test)
 add_subdirectory(libclang)
 
 if(CLANG_ENABLE_ARCMT)
