diff -u -r ../orig-src/tools/clang/tools/CMakeLists.txt ./tools/clang/tools/CMakeLists.txt
--- ../orig-src/tools/clang/tools/CMakeLists.txt	2016-01-12 21:03:50.000000000 -0500
+++ ./tools/clang/tools/CMakeLists.txt	2016-03-10 16:32:04.320055824 -0500
@@ -6,7 +6,6 @@
 add_clang_subdirectory(clang-format-vs)
 add_clang_subdirectory(clang-fuzzer)
 
-add_clang_subdirectory(c-index-test)
 add_clang_subdirectory(libclang)
 
 if(CLANG_ENABLE_ARCMT)
