diff -u -r ../llvm-3.8.0.src/Makefile ./Makefile
--- ../llvm-3.8.0.src/Makefile	2014-03-25 17:45:41.000000000 -0400
+++ ./Makefile	2016-03-10 16:28:00.142389801 -0500
@@ -69,7 +69,6 @@
 ifeq ($(MAKECMDGOALS),install-clang)
   DIRS := tools/clang/tools/driver tools/clang/lib/Headers \
           tools/clang/tools/libclang \
-          tools/clang/tools/c-index-test \
           tools/clang/include/clang-c \
           tools/clang/runtime tools/clang/docs \
           tools/lto
