diff --color -uNr zpaq-7.15/Makefile zpaq-7.15.mod/Makefile
--- zpaq-7.15/Makefile	2016-08-17 18:49:18.000000000 -0300
+++ zpaq-7.15.mod/Makefile	2022-06-07 18:22:22.816761025 -0300
@@ -1,8 +1,6 @@
-CXX=g++
-CPPFLAGS+=-Dunix
-# CPPFLAGS+=NOJIT
-CXXFLAGS=-O3 -march=native
-PREFIX=/usr/local
+# CPPFLAGS+=-Dunix
+# CPPFLAGS+=-DNOJIT
+PREFIX=@TERMUX_PREFIX@
 BINDIR=$(PREFIX)/bin
 MANDIR=$(PREFIX)/share/man
 
