diff --git a/src/vendor/Soup/Intrin/Makefile b/src/vendor/Soup/Intrin/Makefile
index 5accbf7..7fe2bc5 100644
--- a/src/vendor/Soup/Intrin/Makefile
+++ b/src/vendor/Soup/Intrin/Makefile
@@ -1,10 +1,10 @@
 CXX=g++ -std=c++17 -O3 -flto -fvisibility=hidden -fPIC
 CFLAGS=-c -Wall
 
-ifeq ($(shell uname -m),x86_64)
+ifneq (,$(filter $(TERMUX_ARCH),x86_64 i686))
 	CFLAGS+= -maes -mpclmul -mrdrnd -mrdseed -msha -msse4.1
 endif
-ifeq ($(shell uname -m),aarch64)
+ifeq ($(TERMUX_ARCH),aarch64)
 	CFLAGS+= -march=armv8-a+crypto+crc
 endif
 
