diff --git a/src/vendor/Soup/soup/CpuInfo.cpp b/src/vendor/Soup/soup/CpuInfo.cpp
index 4a8d447..55a91de 100644
--- a/src/vendor/Soup/soup/CpuInfo.cpp
+++ b/src/vendor/Soup/soup/CpuInfo.cpp
@@ -5,6 +5,7 @@
 
 	#if defined(_MSC_VER) && !defined(__clang__)
 		#include <intrin.h>
+        #include <immintrin.h>
 	#else
 		#include <cpuid.h>
 	#endif
@@ -71,7 +72,7 @@ namespace soup
 	#undef EBX
 	#undef EDX
 	#undef ECX
-#elif SOUP_ARM
+#elif SOUP_ARM && SOUP_BITS == 64
 	#if SOUP_WINDOWS
 		armv8_aes = IsProcessorFeaturePresent(PF_ARM_V8_CRYPTO_INSTRUCTIONS_AVAILABLE);
 		armv8_sha1 = IsProcessorFeaturePresent(PF_ARM_V8_CRYPTO_INSTRUCTIONS_AVAILABLE);
@@ -127,7 +128,7 @@ namespace soup
 		}
 
 		return str;
-#elif SOUP_ARM
+#elif SOUP_ARM && SOUP_BITS == 64
 		std::string str = "ARMv8 AES: ";
 		str.append(armv8_aes ? "true" : "false");
 		str.append("\nARMv8 SHA1: ").append(armv8_sha1 ? "true" : "false");
