diff --git a/Makefile b/Makefile
index 0c60fd2..4766ad8 100644
--- a/Makefile
+++ b/Makefile
@@ -38,28 +38,27 @@ RM= rm -f
 # Convenience platforms targets.
 PLATS= mingw linux macos
 
+# Lua version and release.
+V= 5.4
+R= $V.4
+
 # What to install.
 TO_BIN= pluto plutoc
 TO_INC= lua.h luaconf.h lualib.h lauxlib.h lua.hpp
-TO_LIB= libpluto.a
+TO_LIB= libpluto.a libpluto.so.$(R) libpluto.so.$(V) libpluto.so
 TO_MAN= lua.1 luac.1
 
-# Lua version and release.
-V= 5.4
-R= $V.4
-
 # Targets start here.
 all:	$(PLAT)
 
 $(PLATS) help test clean:
-	@cd src && $(MAKE) $@
+	@cd src && $(MAKE) $@ V=$(V) R=$(R)
 
 install: dummy
 	cd src && $(MKDIR) $(INSTALL_BIN) $(INSTALL_INC) $(INSTALL_LIB) $(INSTALL_MAN) $(INSTALL_LMOD) $(INSTALL_CMOD)
 	cd src && $(INSTALL_EXEC) $(TO_BIN) $(INSTALL_BIN)
 	cd src && $(INSTALL_DATA) $(TO_INC) $(INSTALL_INC)
 	cd src && $(INSTALL_DATA) $(TO_LIB) $(INSTALL_LIB)
-	cd doc && $(INSTALL_DATA) $(TO_MAN) $(INSTALL_MAN)
 
 uninstall:
 	cd src && cd $(INSTALL_BIN) && $(RM) $(TO_BIN)
