diff --git a/src/vendor/Soup/soup/aes.cpp b/src/vendor/Soup/soup/aes.cpp
index 0f75724..8da0e6e 100644
--- a/src/vendor/Soup/soup/aes.cpp
+++ b/src/vendor/Soup/soup/aes.cpp
@@ -4,7 +4,7 @@
 
 #include "base.hpp"
 
-#if defined(SOUP_USE_INTRIN) && SOUP_BITS == 64 && (SOUP_X86 || SOUP_ARM)
+#if defined(SOUP_USE_INTRIN) && SOUP_BITS == 64 && SOUP_X86
 #define AES_USE_INTRIN true
 #else
 #define AES_USE_INTRIN false
