--- ../osdep-Makefile.inc.orig	2021-01-05 11:01:19.954675451 +0000
+++ ./lib/osdep/Makefile.inc	2021-01-05 11:01:30.914916657 +0000
@@ -96,7 +96,7 @@
 endif
 
 lib_LTLIBRARIES += libaircrack-osdep.la
-libaircrack_osdep_la_LDFLAGS = -release $(LT_VER) -no-undefined
+libaircrack_osdep_la_LDFLAGS = -no-undefined
 libaircrack_osdep_la_CPPFLAGS = -I$(top_srcdir)/include/aircrack-ng/osdep $(AM_CPPFLAGS)
 
 EXTRA_DIST +=     %D%/openbsd.c \
