--- a/build/moz.configure/init.configure
+++ b/build/moz.configure/init.configure
@@ -499,10 +499,7 @@
     raw_os = os = os.replace("/", "_")
     abi = None
     sub_configure_alias = triplet
-    if "android" in os:
-        canonical_os = "Android"
-        canonical_kernel = "Linux"
-    elif os.startswith("linux"):
+    if os.startswith("linux"):
         canonical_os = "GNU"
         canonical_kernel = "Linux"
     elif os.startswith("kfreebsd") and os.endswith("-gnu"):
