--- a/comm/chat/modules/CLib.sys.mjs
+++ b/comm/chat/modules/CLib.sys.mjs
@@ -39,7 +39,7 @@
     break;
   case "linux":
     libcAbi = ctypes.default_abi;
-    libcPath = "libc.so.6";
+    libcPath = "libc.so";
     break;
   default:
     throw new Error("Unknown OS");
--- a/comm/mail/modules/DNS.worker.mjs
+++ b/comm/mail/modules/DNS.worker.mjs
@@ -41,6 +41,7 @@
           { name: "resolv.9", suffix: "" },
           { name: "resolv", suffix: ".2" },
           { name: "resolv", suffix: "" },
+          { name: "c", suffix: "" },
         ];
       }
       const tried = [];
