diff -uNr at-spi2-atk/subprojects/at-spi2-core.wrap at-spi2-atk.mod/subprojects/at-spi2-core.wrap
--- at-spi2-atk/subprojects/at-spi2-core.wrap	2022-07-22 20:50:31.163946073 +0800
+++ at-spi2-atk.mod/subprojects/at-spi2-core.wrap	2022-07-22 20:50:51.743946065 +0800
@@ -2,4 +2,4 @@
 directory=at-spi2-core
 url=https://gitlab.gnome.org/GNOME/at-spi2-core.git
 push-url=git@gitlab.gnome.org:GNOME/at-spi2-core.git
-revision=master
+revision=main
