diff -u -r ../babl-0.1.70/babl/meson.build ./babl/meson.build
--- ../babl-0.1.70/babl/meson.build	2019-08-02 16:16:52.000000000 +0000
+++ ./babl/meson.build	2019-08-08 09:55:50.648892290 +0000
@@ -128,7 +128,7 @@
   c_args: babl_c_args,
   link_whole: [ babl_base, ],
   link_args: [ babl_link_args, ],
-  dependencies: [ math, thread, dl, lcms, ],
+  dependencies: [ math, thread, dl, lcms, log, ],
   link_depends: [ version_script_target, ],
   version: so_version,
   install: true,
