diff -uNr babl-0.1.86/babl/base/meson.build babl-0.1.86.mod/babl/base/meson.build
--- babl-0.1.86/babl/base/meson.build	2021-02-26 12:07:12.687609700 +0000
+++ babl-0.1.86.mod/babl/base/meson.build	2021-05-03 18:24:40.532776557 +0000
@@ -19,5 +19,5 @@
 babl_base = static_library('babl_base',
   babl_base_sources,
   include_directories: [rootInclude, bablInclude],
-  dependencies: [math, lcms],
+  dependencies: [math, lcms, log],
 )
