diff -u -r ../babl-BABL_0_1_34/tools/Makefile.am ./tools/Makefile.am
--- ../babl-BABL_0_1_34/tools/Makefile.am	2017-10-06 15:15:57.000000000 +0200
+++ ./tools/Makefile.am	2017-10-14 20:30:33.268973098 +0200
@@ -2,7 +2,7 @@
 AM_LDFLAGS = -no-install
 
 LDADD = $(top_builddir)/babl/libbabl-@BABL_API_VERSION@.la \
-	$(MATH_LIB) $(THREAD_LIB)
+	$(MATH_LIB) $(THREAD_LIB) $(LOG_LIB)
 
 noinst_PROGRAMS =		\
 	babl-verify		\
