diff -u -r ../babl-0.1.70/tools/meson.build ./tools/meson.build
--- ../babl-0.1.70/tools/meson.build	2019-08-02 16:16:52.000000000 +0000
+++ ./tools/meson.build	2019-08-08 09:59:07.362648769 +0000
@@ -17,7 +17,7 @@
     tool_name + '.c',
     include_directories: [rootInclude, bablInclude],
     link_with: babl,
-    dependencies: [math, thread],
+    dependencies: [math, thread, log],
     install: false,
   )
 
