diff -uNr gtk-3.22.30/po-properties/boldquot.sed gtk-3.22.30.mod/po-properties/boldquot.sed
--- gtk-3.22.30/po-properties/boldquot.sed	1970-01-01 03:00:00.000000000 +0300
+++ gtk-3.22.30.mod/po-properties/boldquot.sed	2018-06-22 20:55:46.584626256 +0300
@@ -0,0 +1,10 @@
+s/"\([^"]*\)"/“\1”/g
+s/`\([^`']*\)'/‘\1’/g
+s/ '\([^`']*\)' / ‘\1’ /g
+s/ '\([^`']*\)'$/ ‘\1’/g
+s/^'\([^`']*\)' /‘\1’ /g
+s/“”/""/g
+s/“/“[1m/g
+s/”/[0m”/g
+s/‘/‘[1m/g
+s/’/[0m’/g
