We don't need a warning every time.

diff -u -r ../mpv-0.19.0/player/main.c ./player/main.c
--- ../mpv-0.19.0/player/main.c	2016-08-15 10:10:59.000000000 -0400
+++ ./player/main.c	2016-08-26 18:40:46.624646426 -0400
@@ -455,10 +455,12 @@
     }
 #endif
 
+#ifndef __ANDROID__
 #if !HAVE_LIBASS
     MP_WARN(mpctx, "Compiled without libass.\n");
     MP_WARN(mpctx, "There will be no OSD and no text subtitles.\n");
 #endif
+#endif
 
     mpctx->osd = osd_create(mpctx->global);
 
