diff --git a/osdep/strnlen.h b/osdep/strnlen.h
index 0a971d0ab0..9a8b2a104d 100644
--- a/osdep/strnlen.h
+++ b/osdep/strnlen.h
@@ -19,11 +19,4 @@
 
 #ifndef MP_OSDEP_STRNLEN
 #define MP_OSDEP_STRNLEN
-
-#ifdef __ANDROID__
-// strnlen is broken on current android ndk, see https://code.google.com/p/android/issues/detail?id=74741
-#include "osdep/android/strnlen.h"
-#define strnlen freebsd_strnlen
-#endif
-
 #endif
