diff -uNr mpv-0.30.0/wscript_build.py mpv-0.30.0.mod/wscript_build.py
--- mpv-0.30.0/wscript_build.py	2019-10-25 16:08:15.000000000 +0300
+++ mpv-0.30.0.mod/wscript_build.py	2019-10-28 16:11:51.777815918 +0200
@@ -413,7 +413,7 @@
         ( "video/img_format.c" ),
         ( "video/mp_image.c" ),
         ( "video/mp_image_pool.c" ),
-        ( "video/out/android_common.c",          "android" ),
+        ( "video/out/android_common.c",          "egl-android" ),
         ( "video/out/aspect.c" ),
         ( "video/out/bitmap_packer.c" ),
         ( "video/out/cocoa/events_view.m",       "cocoa" ),
