diff -u -r ../hunspell-1.3.3/src/tools/hunspell.cxx ./src/tools/hunspell.cxx
--- ../hunspell-1.3.3/src/tools/hunspell.cxx	2014-06-02 13:35:50.000000000 +0200
+++ ./src/tools/hunspell.cxx	2014-07-02 13:14:52.000000000 +0200
@@ -68,7 +68,7 @@
 #include "odfparser.hxx"
 
 #define LIBDIR \
-    "/usr/share/hunspell:" \
+    "@TERMUX_PREFIX@/share/hunspell:" \
     "/usr/share/myspell:" \
     "/usr/share/myspell/dicts:" \
     "/Library/Spelling"
@@ -92,7 +92,7 @@
     "/usr/lib/openoffice.org2.0/share/dict/ooo"
 #define HOME getenv("HOME")
 #define DICBASENAME ".hunspell_"
-#define LOGFILE "/tmp/hunspell.log"
+#define LOGFILE "@TERMUX_PREFIX@/tmp/hunspell.log"
 #define DIRSEPCH '/'
 #define DIRSEP "/"
 #define PATHSEP ":"
