--- a/src/tools/hunspell.cxx
+++ b/src/tools/hunspell.cxx
@@ -119,7 +119,7 @@
   DATADIR "/hunspell:"        \
   DATADIR "/myspell:"         \
   DATADIR "/myspell/dicts:"   \
-  "/usr/share/hunspell:"      \
+  "@TERMUX_PREFIX@/share/hunspell:"      \
   "/usr/share/myspell:"       \
   "/usr/share/myspell/dicts:" \
   "/Library/Spelling"
@@ -143,7 +143,7 @@
   "/usr/lib/openoffice.org2.0/share/dict/ooo"
 #define HOME getenv("HOME")
 #define DICBASENAME ".hunspell_"
-#define LOGFILE "/tmp/hunspell.log"
+#define LOGFILE "@TERMUX_PREFIX@/tmp/hunspell.log"
 #define DIRSEPCH '/'
 #define DIRSEP "/"
 #define PATHSEP ":"
