diff --git a/utils/data-generators/create-tree-from-real-data b/utils/data-generators/create-tree-from-real-data
index 6fc5a37..28d3aa9 100755
--- a/utils/data-generators/create-tree-from-real-data
+++ b/utils/data-generators/create-tree-from-real-data
@@ -37,7 +37,7 @@ def argument_parser():
     parser = argparse.ArgumentParser(
         description="Assemble test data for the Tracker extractors.")
     parser.add_argument(
-        '--rules-dir', default='/usr/share/tracker/extract-rules',
+        '--rules-dir', default='@TERMUX_PREFIX@/share/tracker/extract-rules',
         help="location to find Tracker extract rules (default: %(default)s)")
     parser.add_argument(
         '--output-dir', default=None,
