diff -uNr lesspipe-2.06/configure lesspipe-2.06.mod/configure
--- lesspipe-2.06/configure	2022-08-17 17:56:28.000000000 +0800
+++ lesspipe-2.06.mod/configure	2022-08-18 08:56:15.742341358 +0800
@@ -30,16 +30,16 @@
 	exit ! $opt_help ? 1 : 0;
 }
 my $bash_complete_dir = `pkg-config --variable=completionsdir bash-completion`;
-$bash_complete_dir ||= '/usr/share/bash-completion/completions';
+$bash_complete_dir ||= '@TERMUX_PREFIX@/share/bash-completion/completions';
 # dynamic loading not working, use the following dir instead:
-$bash_complete_dir = '/etc/bashcompletion.d';
+$bash_complete_dir = '@TERMUX_PREFIX@/etc/bashcompletion.d';
 if ($opt_prefix and $opt_prefix !~ /^\/usr/) {
-	$bash_complete_dir = "$ENV{HOME}/.local/share/bash-completion";
+	$bash_complete_dir = "@TERMUX_HOME@/.local/share/bash-completion";
 	# dynamic loading not working, use the following dir instead:
-	$bash_complete_dir = "$opt_prefix/share/bash-completion";
+	$bash_complete_dir = "@TERMUX_PREFIX@/share/bash-completion";
 	print "installing bash completion in $bash_complete_dir\nIn bash, please preload the completion, dynamic invocation does not work\n. $bash_complete_dir/less_completion\n";
 } else {
-	$opt_prefix = '/usr/local';
+	$opt_prefix = '@TERMUX_PREFIX@';
 }
 # remove trailing slash and trailing bin directory
 print "removed trailing /bin dir from prefix\n" if $opt_prefix =~ m|/bin/?$|;
