Install both bash and zsh shell completions.

--- a/configure
+++ b/configure
@@ -44,8 +44,8 @@
 # gererate Makefile
 my @bad = ();
 my $shell = check_shell_vers();
-my $no_bash = (grep {/bash/} @bad);
-my $no_zsh = (grep {/zsh/} @bad);
+my $no_bash = 0;
+my $no_zsh = 0;
 my $bash_complete_dir = "$prefix/share/bash-completion";
 # override prefix and use the system defined directory if known
 if (`pkg-config --version 2>/dev/null`) {
