diff -uNr telepathy-glib-0.24.1/configure telepathy-glib-0.24.1.mod/configure
--- telepathy-glib-0.24.1/configure	2014-08-25 15:30:45.000000000 +0300
+++ telepathy-glib-0.24.1.mod/configure	2019-07-13 18:01:59.342405992 +0300
@@ -13298,10 +13298,10 @@
 
 
 
-GLIB_GENMARSHAL=`$PKG_CONFIG --variable=glib_genmarshal glib-2.0`
+GLIB_GENMARSHAL=glib-genmarshal
 
 
-GLIB_MKENUMS=`$PKG_CONFIG --variable=glib_mkenums glib-2.0`
+GLIB_MKENUMS=glib-mkenums
 
 
 
