--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1878,13 +1878,6 @@
         # TODO(https://crbug.com/1490607): Fix and re-enable.
         "-Wno-thread-safety-reference-return",
       ]
-
-      if (!is_nacl) {
-        cflags_cc += [
-          # TODO(https://crbug.com/1513724): Fix and re-enable.
-          "-Wno-c++11-narrowing-const-reference",
-        ]
-      }
     }
 
     # Some builders, such as Cronet, use a different version of Clang than
