This reverts https://github.com/chromium/chromium/commit/97c864b32a7f4d64f541e2b7bf9006d2f582082f

--- a/build/install-build-deps.py
+++ b/build/install-build-deps.py
@@ -374,8 +374,6 @@ def lib_list():
       "libxrender1",
       "libxtst6",
       "x11-utils",
-      "xserver-xorg-core",  # TODO(crbug.com/1417069): Experimental.
-      "xserver-xorg-video-dummy",  # TODO(crbug.com/1417069): Experimental.
       "xvfb",
       "zlib1g",
   ]
