--- a/v8/src/wasm/wasm-objects.cc
+++ b/v8/src/wasm/wasm-objects.cc
@@ -11,6 +11,11 @@
 #undef MAP_TYPE
 #endif  // V8_TARGET_OS_LINUX
 
+#if defined __ANDROID__ && __ANDROID_API__ < 30
+#include <sys/syscall.h>
+#define memfd_create(name,flags) syscall(SYS_memfd_create,name,flags)
+#endif
+
 #include "src/wasm/wasm-objects.h"
 
 #include <optional>
diff --git a/v8/src/wasm/wasm-memory-map-descriptor.cc b/v8/src/wasm/wasm-memory-map-descriptor.cc
index 53c4286730..5235bb5339 100644
--- a/v8/src/wasm/wasm-memory-map-descriptor.cc
+++ b/v8/src/wasm/wasm-memory-map-descriptor.cc
@@ -14,6 +14,11 @@
 #undef MAP_TYPE
 #endif  // V8_TARGET_OS_LINUX
 
+#if defined __ANDROID__ && __ANDROID_API__ < 30
+#include <sys/syscall.h>
+#define memfd_create(name,flags) syscall(SYS_memfd_create,name,flags)
+#endif
+
 #include "include/cppgc/allocation.h"
 #include "include/v8-cppgc.h"
 #include "include/v8-template.h"
