--- a/third_party/libsync/src/include/ndk/sync.h
+++ b/third_party/libsync/src/include/ndk/sync.h
@@ -64,7 +64,7 @@
  *
  * Available since API level 26.
  */
-int32_t sync_merge(const char* name, int32_t fd1, int32_t fd2) __INTRODUCED_IN(26);
+int32_t sync_merge(const char* name, int32_t fd1, int32_t fd2);
 
 /**
  * Retrieve detailed information about a sync file and its fences.
@@ -73,7 +73,7 @@
  *
  * Available since API level 26.
  */
-struct sync_file_info* sync_file_info(int32_t fd) __INTRODUCED_IN(26);
+struct sync_file_info* sync_file_info(int32_t fd);
 
 /**
  * Get the array of fence infos from the sync file's info.
@@ -97,7 +97,7 @@
  *
  * Available since API level 26.
  */
-void sync_file_info_free(struct sync_file_info* info) __INTRODUCED_IN(26);
+void sync_file_info_free(struct sync_file_info* info);
 
 __END_DECLS
 
