--- a/v8/src/regexp/regexp-bytecodes-inl.h
+++ b/v8/src/regexp/regexp-bytecodes-inl.h
@@ -118,14 +118,14 @@
   using Traits = RegExpBytecodeOperandsTraits<OpTypes...>;
   static constexpr int kCount = Traits::kOperandCount;
   static constexpr int kTotalSize = Traits::kSize;
-  static consteval int Index(Operand op) { return static_cast<uint8_t>(op); }
-  static consteval int Size(Operand op) {
+  static constexpr int Index(Operand op) { return static_cast<uint8_t>(op); }
+  static constexpr int Size(Operand op) {
     return Traits::kOperandSizes[Index(op)];
   }
-  static consteval int Offset(Operand op) {
+  static constexpr int Offset(Operand op) {
     return Traits::kOperandOffsets[Index(op)];
   }
-  static consteval RegExpBytecodeOperandType Type(Operand op) {
+  static constexpr RegExpBytecodeOperandType Type(Operand op) {
     return Traits::kOperandTypes[Index(op)];
   }
 
