--- a/base/containers/span.h
+++ b/base/containers/span.h
@@ -785,7 +785,7 @@
   template <typename OtherElementType,
             size_t OtherExtent,
             typename OtherInternalPtrType>
-    requires((OtherExtent == dynamic_extent || extent == OtherExtent) &&
+    requires((OtherExtent == dynamic_extent || Extent == OtherExtent) &&
              std::equality_comparable_with<const element_type,
                                            const OtherElementType>)
   friend constexpr bool operator==(
@@ -820,7 +820,7 @@
   template <typename OtherElementType,
             size_t OtherExtent,
             typename OtherInternalPtrType>
-    requires((OtherExtent == dynamic_extent || extent == OtherExtent) &&
+    requires((OtherExtent == dynamic_extent || Extent == OtherExtent) &&
              std::three_way_comparable_with<const element_type,
                                             const OtherElementType>)
   friend constexpr auto operator<=>(
