--- a/Cargo.toml
+++ b/Cargo.toml
@@ -28,7 +28,7 @@ rand = "0.9.2"
 postgres = "0.19.10"
 rayon = "1.6.1"
 config = { version = "0.15.13", features = ["yaml", "json", "toml", "ini"] }
-headless_chrome = { git = "https://github.com/atroche/rust-headless-chrome", rev = "61ce783806e5d75a03f731330edae6156bb0a2e0" }
+headless_chrome = "1.0.17"
 addr = "0.15.6"
 serde_json = "1.0.141"
 rusolver = { git = "https://github.com/Edu4rdSHL/rusolver", tag = "0.10.1" }
