--- a/Cargo.toml
+++ b/Cargo.toml
@@ -36,7 +36,8 @@
 
 # https://github.com/bluejekyll/trust-dns/pull/1632
 [patch.crates-io]
-trust-dns-resolver = { git = "https://github.com/Findomain/trust-dns", package = "trust-dns-resolver", branch = "custombranch" }
+traitobject = { path = "./vendor/traitobject" }
+trust-dns-resolver = { path = "./vendor/trust-dns/crates/resolver" }
 
 [target.arm-unknown-linux-gnueabihf.dependencies]
 openssl = { version = "0.10.60", features = ["vendored"] }
