diff -uNr dmenu-4.9/config.mk dmenu-4.9.mod/config.mk
--- dmenu-4.9/config.mk	2019-02-02 14:55:02.000000000 +0200
+++ dmenu-4.9.mod/config.mk	2020-01-04 23:50:39.788733241 +0200
@@ -2,11 +2,11 @@
 VERSION = 4.9
 
 # paths
-PREFIX = /usr/local
+PREFIX ?= /usr/local
 MANPREFIX = $(PREFIX)/share/man
 
-X11INC = /usr/X11R6/include
-X11LIB = /usr/X11R6/lib
+X11INC = $(PREFIX)/include
+X11LIB = $(PREFIX)/lib
 
 # Xinerama, comment if you don't want it
 XINERAMALIBS  = -lXinerama
@@ -14,7 +14,7 @@
 
 # freetype
 FREETYPELIBS = -lfontconfig -lXft
-FREETYPEINC = /usr/include/freetype2
+FREETYPEINC = $(PREFIX)/include/freetype2
 # OpenBSD (uncomment)
 #FREETYPEINC = $(X11INC)/freetype2
 
@@ -25,7 +25,7 @@
 # flags
 CPPFLAGS = -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_XOPEN_SOURCE=700 -D_POSIX_C_SOURCE=200809L -DVERSION=\"$(VERSION)\" $(XINERAMAFLAGS)
 CFLAGS   = -std=c99 -pedantic -Wall -Os $(INCS) $(CPPFLAGS)
-LDFLAGS  = $(LIBS)
+LDFLAGS  = -Wl,-rpath=@TERMUX_PREFIX@/lib,--enable-new-dtags $(LIBS)
 
 # compiler and linker
-CC = cc
+CC ?= cc
