--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,7 +50,6 @@
     # handle cases where the libdir should be lib/, lib64/, or lib/<arch>/
     # depending on the target system and cmake options.
     include(GNUInstallDirs)
-    string(REGEX REPLACE "lib(.*)$" "\\1" LLVM_LIBDIR_SUFFIX "${CMAKE_INSTALL_LIBDIR}")
     set(LLVM_LIBDIR_SUFFIX "${LLVM_LIBDIR_SUFFIX}" CACHE STRING "Define suffix of library directory name (32/64)")
 endif()
 
