diff --git a/configure b/configure
index 50f3cad..31256c2 100755
--- a/configure
+++ b/configure
@@ -21,10 +21,10 @@ datarootdir='$(prefix)/share'
 infodir='$(datarootdir)/info'
 mandir='$(datarootdir)/man'
 program_prefix=
-CC=gcc
-CPPFLAGS=
-CFLAGS='-Wall -W -O2'
-LDFLAGS=
+CC?=gcc
+CPPFLAGS?=
+CFLAGS?='-Wall -W -O2'
+LDFLAGS?=
 
 # checking whether we are using GNU C.
 ${CC} --version > /dev/null 2>&1
