diff --git a/README b/README
new file mode 100644
index 00000000..cd578cb7
--- /dev/null
+++ b/README
@@ -0,0 +1,18 @@
+This is LDC, the LLVM-based D compiler.  It will natively
+compile D on Android devices.
+
+The compiler configuration file is etc/ldc2.conf, and by
+default only include/d is on the module search path.
+
+To develop for Android, you will find the D headers and
+sample apps at this github repository useful:
+
+https://github.com/joakim-noah/android
+
+You can find instructions on building Android apps at the
+D wiki:
+
+http://wiki.dlang.org/Build_D_for_Android
+
+For further information, including how to report bugs,
+please refer to the LDC wiki: http://wiki.dlang.org/LDC.
