--- ./Makefile.am~	2023-09-05 14:53:22.000000000 +0000
+++ ./Makefile.am	2023-09-29 20:28:32.058027371 +0000
@@ -9,11 +9,8 @@
 tempfile_SOURCES = tempfile.c
 ischroot_SOURCES = ischroot.c
 
-dist_bin_SCRIPTS = which.debianutils savelog
+dist_bin_SCRIPTS = savelog
 
-dist_sbin_SCRIPTS = installkernel add-shell remove-shell update-shells
+dist_sbin_SCRIPTS =
 
-dist_man_MANS = run-parts.8 \
-	   installkernel.8 savelog.8 \
-	   tempfile.1 which.debianutils.1 add-shell.8 \
-	   remove-shell.8 update-shells.8 ischroot.1
+dist_man_MANS = savelog.8 tempfile.1 ischroot.1
