diff --git a/arch-desktop-directories/Arch-Accessibility.directory b/termux-desktop-directories/Termux-Accessibility.directory
similarity index 100%
rename from arch-desktop-directories/Arch-Accessibility.directory
rename to termux-desktop-directories/Termux-Accessibility.directory
diff --git a/arch-desktop-directories/Arch-Accessories.directory b/termux-desktop-directories/Termux-Accessories.directory
similarity index 100%
rename from arch-desktop-directories/Arch-Accessories.directory
rename to termux-desktop-directories/Termux-Accessories.directory
diff --git a/arch-desktop-directories/Arch-Applications.directory b/termux-desktop-directories/Termux-Applications.directory
similarity index 100%
rename from arch-desktop-directories/Arch-Applications.directory
rename to termux-desktop-directories/Termux-Applications.directory
diff --git a/arch-desktop-directories/Arch-Development.directory b/termux-desktop-directories/Termux-Development.directory
similarity index 100%
rename from arch-desktop-directories/Arch-Development.directory
rename to termux-desktop-directories/Termux-Development.directory
diff --git a/arch-desktop-directories/Arch-Education.directory b/termux-desktop-directories/Termux-Education.directory
similarity index 100%
rename from arch-desktop-directories/Arch-Education.directory
rename to termux-desktop-directories/Termux-Education.directory
diff --git a/arch-desktop-directories/Arch-Games.directory b/termux-desktop-directories/Termux-Games.directory
similarity index 100%
rename from arch-desktop-directories/Arch-Games.directory
rename to termux-desktop-directories/Termux-Games.directory
diff --git a/arch-desktop-directories/Arch-Graphics.directory b/termux-desktop-directories/Termux-Graphics.directory
similarity index 100%
rename from arch-desktop-directories/Arch-Graphics.directory
rename to termux-desktop-directories/Termux-Graphics.directory
diff --git a/arch-desktop-directories/Arch-Internet.directory b/termux-desktop-directories/Termux-Internet.directory
similarity index 100%
rename from arch-desktop-directories/Arch-Internet.directory
rename to termux-desktop-directories/Termux-Internet.directory
diff --git a/arch-desktop-directories/Arch-Multimedia.directory b/termux-desktop-directories/Termux-Multimedia.directory
similarity index 100%
rename from arch-desktop-directories/Arch-Multimedia.directory
rename to termux-desktop-directories/Termux-Multimedia.directory
diff --git a/arch-desktop-directories/Arch-Office.directory b/termux-desktop-directories/Termux-Office.directory
similarity index 100%
rename from arch-desktop-directories/Arch-Office.directory
rename to termux-desktop-directories/Termux-Office.directory
diff --git a/arch-desktop-directories/Arch-Other.directory b/termux-desktop-directories/Termux-Other.directory
similarity index 100%
rename from arch-desktop-directories/Arch-Other.directory
rename to termux-desktop-directories/Termux-Other.directory
diff --git a/arch-desktop-directories/Arch-Science.directory b/termux-desktop-directories/Termux-Science.directory
similarity index 100%
rename from arch-desktop-directories/Arch-Science.directory
rename to termux-desktop-directories/Termux-Science.directory
diff --git a/arch-desktop-directories/Arch-System-Tools.directory b/termux-desktop-directories/Termux-System-Tools.directory
similarity index 100%
rename from arch-desktop-directories/Arch-System-Tools.directory
rename to termux-desktop-directories/Termux-System-Tools.directory
diff --git a/arch-xdg-menu/arch-applications.menu b/termux-xdg-menu/termux-applications.menu
similarity index 76%
rename from arch-xdg-menu/arch-applications.menu
rename to termux-xdg-menu/termux-applications.menu
index 41b9c64..e44f6eb 100644
--- a/arch-xdg-menu/arch-applications.menu
+++ b/termux-xdg-menu/termux-applications.menu
@@ -3,22 +3,22 @@
 <Menu>
 
   <Name>Applications</Name>
-  <Directory>Arch-Applications.directory</Directory>
+  <Directory>Termux-Applications.directory</Directory>
   <DefaultAppDirs/>
   <DefaultDirectoryDirs/>
   <DefaultMergeDirs/>
 
   <Menu>
-    <Name>Archlinux</Name>
-    <Directory>Archlinux.directory</Directory>
+    <Name>Termux</Name>
+    <Directory>Termux.directory</Directory>
     <Include>
-      <Category>Archlinux</Category>
+      <Category>Termux</Category>
     </Include>
   </Menu>
 
   <Menu>
     <Name>Accessories</Name>
-    <Directory>Arch-Accessories.directory</Directory>
+    <Directory>Termux-Accessories.directory</Directory>
     <Include>
       <And>
         <Category>Utility</Category>
@@ -31,7 +31,7 @@
 
   <Menu>
     <Name>Accessibility</Name>
-    <Directory>Arch-Accessibility.directory</Directory>
+    <Directory>Termux-Accessibility.directory</Directory>
     <Include>
       <And>
         <Category>Accessibility</Category>
@@ -44,7 +44,7 @@
 
   <Menu>
     <Name>Development</Name>
-    <Directory>Arch-Development.directory</Directory>
+    <Directory>Termux-Development.directory</Directory>
     <Include>
       <And>
         <Category>Development</Category>
@@ -55,7 +55,7 @@
 
   <Menu>
     <Name>Education</Name>
-    <Directory>Arch-Education.directory</Directory>
+    <Directory>Termux-Education.directory</Directory>
     <Include>
       <And>
         <Category>Education</Category>
@@ -65,7 +65,7 @@
 
   <Menu>
     <Name>Games</Name>
-    <Directory>Arch-Games.directory</Directory>
+    <Directory>Termux-Games.directory</Directory>
     <Include>
       <And>
         <Category>Game</Category>
@@ -75,7 +75,7 @@
 
   <Menu>
     <Name>Graphics</Name>
-    <Directory>Arch-Graphics.directory</Directory>
+    <Directory>Termux-Graphics.directory</Directory>
     <Include>
       <And>
         <Category>Graphics</Category>
@@ -85,7 +85,7 @@
 
   <Menu>
     <Name>Internet</Name>
-    <Directory>Arch-Internet.directory</Directory>
+    <Directory>Termux-Internet.directory</Directory>
     <Include>
       <And>
         <Category>Network</Category>
@@ -95,7 +95,7 @@
 
   <Menu>
     <Name>Multimedia</Name>
-    <Directory>Arch-Multimedia.directory</Directory>
+    <Directory>Termux-Multimedia.directory</Directory>
     <Include>
       <And>
         <Category>AudioVideo</Category>
@@ -105,7 +105,7 @@
 
   <Menu>
     <Name>Office</Name>
-    <Directory>Arch-Office.directory</Directory>
+    <Directory>Termux-Office.directory</Directory>
     <Include>
       <And>
         <Category>Office</Category>
@@ -115,7 +115,7 @@
 
     <Menu>
 	<Name>Science</Name>
-	<Directory>Arch-Science.directory</Directory>
+	<Directory>Termux-Science.directory</Directory>
 	<Include>
 		<And>
 			<Or>
@@ -135,7 +135,7 @@
 
   <Menu>
     <Name>System</Name>
-    <Directory>Arch-System-Tools.directory</Directory>
+    <Directory>Termux-System-Tools.directory</Directory>
     <Include>
       <And>
         <Category>System</Category>
@@ -146,7 +146,7 @@
 
   <Menu>
     <Name>Other</Name>
-    <Directory>Arch-Other.directory</Directory>
+    <Directory>Termux-Other.directory</Directory>
     <OnlyUnallocated/>
     <Include>
       <And>
diff --git a/update-menus b/update-menus
index 68233f7..0053640 100755
--- a/update-menus
+++ b/update-menus
@@ -1,10 +1,10 @@
-#!/usr/bin/perl -w
+#!@TERMUX_PREFIX@/bin/perl -w
 
 # SPDX-License-Identifier: GPL2+
 
-$XDG_CFG="/etc/xdg/menus/arch-applications.menu";
-$CACHE_DIR="/var/cache/xdg-menu";
-$LOG_FILE="/var/log/update-menus.log";
+$XDG_CFG="@TERMUX_PREFIX@/etc/xdg/menus/termux-applications.menu";
+$CACHE_DIR="@TERMUX_PREFIX@/var/cache/xdg-menu";
+$LOG_FILE="@TERMUX_PREFIX@/var/log/update-menus.log";
 
 sub get_cache_dir
 {
@@ -25,7 +25,7 @@ my $cachedir = get_cache_dir;
 system("rm -rf $cachedir");
 (system("rm -rf $CACHE_DIR/*") == 0) || die("Cannot purge cache!");
 
-open(FH, "</etc/update-menus.conf") || die("No config file!");
+open(FH, "<@TERMUX_PREFIX@/etc/update-menus.conf") || die("No config file!");
 
 while(<FH>)
 {
diff --git a/xdg_menu b/xdg_menu
index e45ca4e..1bb5368 100755
--- a/xdg_menu
+++ b/xdg_menu
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -w
+#!@TERMUX_PREFIX@/bin/perl -w
 
 #
 # xdg-menu for archlinux. based on suse xdg-menu written by <nadvornik@suse.cz>
@@ -74,7 +74,7 @@ sub findicon
         my $iconname = shift;
 
         my $home = $ENV{"HOME"};
-        my $xdg_data_dirs = $ENV{"XDG_DATA_DIRS"} || "/usr/local/share:/usr/share:$home/.local/share";
+        my $xdg_data_dirs = $ENV{"XDG_DATA_DIRS"} || "@TERMUX_PREFIX@/local/share:@TERMUX_PREFIX@/share:$home/.local/share";
 
         my (@xdg_data_dirs);
         @xdg_data_dirs = split(":",$xdg_data_dirs);
@@ -121,7 +121,7 @@ sub findicon
                 }
         }
 
-        push @icon_search_path, "/usr/share/pixmaps";
+        push @icon_search_path, "@TERMUX_PREFIX@/share/pixmaps";
 
         my $filename;
 
@@ -1579,9 +1579,9 @@ sub output_blackbox_menu ($;$;$)
     $output .= '[config] (Configuration)
                 [workspaces] (Workspace)
                 [submenu] (System Styles) {Choose a style...}
-                        [stylesdir] (/usr/share/blackbox/styles)
-			[stylesdir] (/usr/share/fluxbox/styles)
-			[stylesdir] (/usr/share/openbox/styles)
+                        [stylesdir] (@TERMUX_PREFIX@/share/blackbox/styles)
+			[stylesdir] (@TERMUX_PREFIX@/share/fluxbox/styles)
+			[stylesdir] (@TERMUX_PREFIX@/share/openbox/styles)
                 [end]
                 [submenu] (User Styles) {Choose a style...}
                         [stylesdir] (~/.blackbox/styles)
@@ -1986,7 +1986,7 @@ sub output_openbox3_menu ($;$)
 <openbox_menu xmlns="http://openbox.org/"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://openbox.org/
-                file:///usr/share/openbox/menu.xsd">';
+                file://@TERMUX_PREFIX@/share/openbox/menu.xsd">';
     $output .= "<menu id=\"xdg\" label=\"xdg\">\n";
     $output .= output_openbox3_inner_menu ($menu, $indent);
     $output .= "</menu>\n";
@@ -2200,10 +2200,10 @@ sub output_jwm_menu ($;$)
 
 sub get_root_menu
 {
-    foreach my $dir (split(/:/, $ENV{XDG_CONFIG_DIRS}), "/etc/xdg")
+    foreach my $dir (split(/:/, $ENV{XDG_CONFIG_DIRS}), "@TERMUX_PREFIX@/etc/xdg")
     {
-        check_file("$dir/menus/arch-applications.menu");
-        return "$dir/menus/arch-applications.menu" if -f "$dir/menus/arch-applications.menu";
+        check_file("$dir/menus/termux-applications.menu");
+        return "$dir/menus/termux-applications.menu" if -f "$dir/menus/termux-applications.menu";
     }
     return "";
 }
@@ -2211,13 +2211,13 @@ sub get_root_menu
 sub get_kde_config
 {
     my $ret = 'true';
-    if(-x '/usr/bin/kde4-config')
+    if(-x '@TERMUX_PREFIX@/bin/kde4-config')
     {
-        $ret = '/usr/bin/kde4-config';
+        $ret = '@TERMUX_PREFIX@/bin/kde4-config';
     }
-    elsif(-x '/opt/kde/bin/kde-config')
+    elsif(-x '@TERMUX_PREFIX@/opt/kde/bin/kde-config')
     {
-        $ret = '/opt/kde/bin/kde-config';
+        $ret = '@TERMUX_PREFIX@/opt/kde/bin/kde-config';
     }
     return $ret;
 }
@@ -2236,7 +2236,7 @@ sub get_app_dirs
     };
 
     my $home = $ENV{"HOME"};
-    foreach my $d (split(/:/, $ENV{XDG_DATA_DIRS}), @kde_xdgdata, "/usr/share", "$home/.local/share", "/opt/gnome/share")
+    foreach my $d (split(/:/, $ENV{XDG_DATA_DIRS}), @kde_xdgdata, "@TERMUX_PREFIX@/share", "$home/.local/share", "@TERMUX_PREFIX@/opt/gnome/share")
     {
         my $dir = $d;
         $dir =~ s/\/*$//;
@@ -2254,7 +2254,7 @@ sub get_desktop_dirs
     my %used;
     my $ret = '';
     my $home = $ENV{"HOME"};
-    foreach my $dir (split(/:/, $ENV{XDG_DATA_DIRS}), "/usr/share", "$home/.local/share", "/opt/kde3/share", "/opt/gnome/share")
+    foreach my $dir (split(/:/, $ENV{XDG_DATA_DIRS}), "@TERMUX_PREFIX@/share", "$home/.local/share", "@TERMUX_PREFIX@/opt/kde3/share", "@TERMUX_PREFIX@/opt/gnome/share")
     {
         next if defined $used{$dir};
         next if check_file("$dir/desktop-directories") ne 'D';
@@ -2269,7 +2269,7 @@ sub get_KDE_legacy_dirs
 {
     my %used;
     my @ret;
-    foreach my $d ("/etc/opt/kde3/share/applnk", "/opt/kde3/share/applnk", reverse(split(/:/, `$kdeconfig --path apps`)))
+    foreach my $d ("@TERMUX_PREFIX@/etc/opt/kde3/share/applnk", "@TERMUX_PREFIX@/opt/kde3/share/applnk", reverse(split(/:/, `$kdeconfig --path apps`)))
     {
         my $dir = $d;
         chomp $dir;
@@ -2496,9 +2496,6 @@ my $root_menu = get_root_menu();
 $charset = langinfo(CODESET);
 $language = setlocale(LC_MESSAGES);
 
-$root_cmd = "/usr/bin/gnomesu" if -x '/usr/bin/gnomesu';
-$root_cmd = "/usr/bin/kdesu" if -x '/usr/bin/kdesu';
-
 my $help;
 
 GetOptions("format=s" => \$format,
diff --git a/xdg_menu_su b/xdg_menu_su
index db5f23b..e6303ba 100755
--- a/xdg_menu_su
+++ b/xdg_menu_su
@@ -4,5 +4,5 @@
 
 xterm -bg LightGoldenRod -fn 9x15 -geometry 34x4 \
       -vb +sb -T "root permission" \
-      -e sh -c "echo \"Run command with root privileges\"; echo \"Command: $*\"; sux -c \"/usr/bin/env LANG=${LANG} $*\""
+      -e sh -c "echo \"Run command with root privileges\"; echo \"Command: $*\"; su -c \"@TERMUX_PREFIX@/bin/env LANG=${LANG} $*\""
 
