diff --git a/m4/dpkg-arch.m4 b/m4/dpkg-arch.m4
index 8859289b5..8087de293 100644
--- a/m4/dpkg-arch.m4
+++ b/m4/dpkg-arch.m4
@@ -20,7 +20,7 @@ AC_DEFUN([DPKG_CPU_TYPE], [
   AC_MSG_CHECKING([dpkg cpu type])
   _DPKG_ARCHITECTURE([DEB_HOST_ARCH_CPU], [cpu_type])
   AS_IF([test "x$cpu_type" = "x"], [
-    cpu_type=$host_cpu
+    cpu_type=TERMUX_ARCH
     AC_MSG_RESULT([$cpu_type])
     AC_MSG_WARN([$host_cpu not found in cputable])
   ], [
