diff -uNr dpkg-1.19.4/src/help.c dpkg-1.19.4.mod/src/help.c
--- dpkg-1.19.4/src/help.c	2019-01-23 14:04:28.000000000 +0200
+++ dpkg-1.19.4.mod/src/help.c	2019-05-12 22:08:27.729956232 +0300
@@ -114,8 +114,7 @@
 void checkpath(void) {
   static const char *const prog_list[] = {
     DEFAULTSHELL,
-    RM,
-    TAR,
+    "busybox",
     DIFF,
     BACKEND,
     /* Mac OS X uses dyld (Mach-O) instead of ld.so (ELF), and does not have
