diff -u -r ../dpkg-1.18.2/lib/dpkg/triglib.c ./lib/dpkg/triglib.c
--- ../dpkg-1.18.2/lib/dpkg/triglib.c	2015-07-12 22:38:47.000000000 -0400
+++ ./lib/dpkg/triglib.c	2015-08-25 17:54:10.350853590 -0400
@@ -787,10 +787,13 @@
 			if (errno != EEXIST)
 				ohshite(_("unable to create triggers state"
 				          " directory '%.250s'"), triggersdir);
-		} else if (chown(triggersdir, 0, 0)) {
+		}
+#ifndef __ANDROID__
+		else if (chown(triggersdir, 0, 0)) {
 			ohshite(_("unable to set ownership of triggers state"
 			          " directory '%.250s'"), triggersdir);
 		}
+#endif
 		ur = trigdef_update_start(tduf);
 	}
 	switch (ur) {
