diff --git a/lib/dpkg/path-remove.c b/lib/dpkg/path-remove.c
index de595cd9e..54fbd407e 100644
--- a/lib/dpkg/path-remove.c
+++ b/lib/dpkg/path-remove.c
@@ -131,7 +131,7 @@ path_remove_tree(const char *pathname)
 	if (!rmdir(pathname))
 		/* Deleted it OK, it was a directory. */
 		return;
-	if (errno == ENOENT || errno == ELOOP)
+	if (errno == ENOENT || errno == ELOOP || errno == EROFS)
 		return;
 	if (errno == ENOTDIR) {
 		/* Either it's a file, or one of the path components is. If
