diff --git a/scripts/dpkg-scanpackages.pl b/scripts/dpkg-scanpackages.pl
index b5d0c8498..62e05dc03 100755
--- a/scripts/dpkg-scanpackages.pl
+++ b/scripts/dpkg-scanpackages.pl
@@ -17,6 +17,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
+use lib '@TERMUX_PREFIX@/share/perl5';
 use v5.36;
 
 use Getopt::Long qw(:config posix_default bundling_values no_ignorecase);
