diff -u -r ./src/statoverride/main.c ./main.c
--- ./src/statoverride/main.c	2022-05-01 09:51:32.298164682 -0300
+++ ./main.c	2022-05-01 09:56:12.247431464 -0300
@@ -185,8 +185,10 @@
 static void
 statdb_node_apply(const char *filename, struct file_stat *filestat)
 {
+#ifndef __ANDROID__
 	if (chown(filename, filestat->uid, filestat->gid) < 0)
 		ohshite(_("error setting ownership of '%.255s'"), filename);
+#endif
 	if (chmod(filename, filestat->mode & ~S_IFMT))
 		ohshite(_("error setting permissions of '%.255s'"), filename);
 
