diff --color -uNr ../dpkg-1.21.8/src/statoverride/main.c ./src/statoverride/main.c
--- ../dpkg-1.21.8/src/statoverride/main.c	2022-05-24 14:36:59.000000000 -0300
+++ ./src/statoverride/main.c	2022-05-31 07:39:44.829552700 -0300
@@ -188,9 +188,11 @@
 {
 	int rc;
 
+#ifndef __ANDROID__
 	rc = chown(filename, filestat->uid, filestat->gid);
 	if (forcible_nonroot_error(rc) < 0)
 		ohshite(_("error setting ownership of '%.255s'"), filename);
+#endif
 	rc = chmod(filename, filestat->mode & ~S_IFMT);
 	if (forcible_nonroot_error(rc) < 0)
 		ohshite(_("error setting permissions of '%.255s'"), filename);
