diff --git a/src/main/configure.c b/src/main/configure.c
index 9a7516335..12facaeb8 100644
--- a/src/main/configure.c
+++ b/src/main/configure.c
@@ -504,8 +504,10 @@ deferred_configure_conffile(struct pkginfo *pkg, struct conffile *conff)
 			        pkg_name(pkg, pnaw_nonambig), cdr_old.buf,
 			        strerror(errno));
 		if (!(what & CFOF_USER_DEL))
-			if (link(cdr.buf, cdr_old.buf))
-				warning(_("%s: failed to link '%s' to '%s': %s"),
+			/** Termux modification: Use rename(2) instead of link(2), to avoid hard
+			links which do not work on Android 6.0 or later. */
+			if (rename(cdr.buf, cdr_old.buf))
+				warning(_("%s: failed to rename '%s' to '%s': %s"),
 				        pkg_name(pkg, pnaw_nonambig),
 				        cdr.buf,
 				        cdr_old.buf, strerror(errno));
