diff --git a/src/main/configure.c b/src/main/configure.c
index 4cdd67f0e..3306e00b4 100644
--- a/src/main/configure.c
+++ b/src/main/configure.c
@@ -499,8 +499,10 @@ deferred_configure_conffile(struct pkginfo *pkg, struct conffile *conff)
 			        pkg_name(pkg, pnaw_nonambig), cdr_old.buf,
 			        strerror(errno));
 		if (!(what & CFOF_USER_DEL))
-			if (link(cdr.buf, cdr_old.buf))
-				warning(_("%s: failed to link '%.250s' to '%.250s': %s"),
+		/** Termux modification: Use rename(2) instead of link(2), to avoid hard
+			links which do not work on Android 6.0 or later. */
+		if (rename(cdr.buf, cdr_old.buf))
+			warning(_("%s: failed to rename '%.250s' to '%.250s': %s"),
 				        pkg_name(pkg, pnaw_nonambig), cdr.buf,
 				        cdr_old.buf, strerror(errno));
 		/* Fall through. */
