diff -u -r ../dpkg-1.18.4/src/configure.c ./src/configure.c
--- ../dpkg-1.18.4/src/configure.c	2015-11-26 18:53:41.000000000 -0500
+++ ./src/configure.c	2016-03-03 17:41:42.494272593 -0500
@@ -496,8 +496,10 @@
 			        pkg_name(pkg, pnaw_nonambig), cdr2.buf,
 			        strerror(errno));
 		if (!(what & CFOF_USER_DEL))
-			if (link(cdr.buf, cdr2.buf))
-				warning(_("%s: failed to link '%.250s' to '%.250s': %s"),
+			/** Termux modification: Use rename(2) instead of link(2), to avoid hard
+			    links which does not work on Android 6.0 or later. */
+			if (rename(cdr.buf, cdr2.buf))
+				warning(_("%s: failed to rename '%.250s' to '%.250s': %s"),
 				        pkg_name(pkg, pnaw_nonambig), cdr.buf,
 				        cdr2.buf, strerror(errno));
 		/* Fall through. */
