--- ext/openjpeg/gstopenjpeg.h.orig	2017-08-17 05:39:18.000000000 -0500
+++ ./ext/openjpeg/gstopenjpeg.h	2017-10-05 20:36:20.000000000 -0500
@@ -38,13 +38,7 @@
 #define OPJ_CPRL CPRL
 #else
 #include <stdio.h>
-# if defined(HAVE_OPENJPEG_2_2)
-#  include <openjpeg-2.2/openjpeg.h>
-# elif defined(HAVE_OPENJPEG_2_1)
-#  include <openjpeg-2.1/openjpeg.h>
-# else
-#  include <openjpeg-2.0/openjpeg.h>
-# endif
+#include <openjpeg-2.3/openjpeg.h>
 #endif
 
 #endif /* __GST_OPENJPEG_H__ */
