diff -uNr libtorrent-0.13.8/scripts/ax_check_zlib.m4 libtorrent-0.13.8.mod/scripts/ax_check_zlib.m4
--- libtorrent-0.13.8/scripts/ax_check_zlib.m4	2019-07-19 14:38:12.000000000 +0300
+++ libtorrent-0.13.8.mod/scripts/ax_check_zlib.m4	2020-05-07 20:33:39.695941588 +0300
@@ -70,7 +70,7 @@
 # Handle user hints
 #
 [AC_MSG_CHECKING(if zlib is wanted)
-zlib_places="/usr/local /usr /opt/local /sw"
+zlib_places="@TERMUX_PREFIX@"
 AC_ARG_WITH([zlib],
 [  --with-zlib=DIR         root directory path of zlib installation @<:@defaults to
                           /usr/local or /usr if not found in /usr/local@:>@
