diff -ru fcgi-2.4.0-orig/configure.in fcgi-2.4.0/configure.in
--- fcgi-2.4.0-orig/configure.in	2003-01-19 18:19:41.000000000 +0100
+++ fcgi-2.4.0/configure.in	2007-03-05 20:48:14.000000000 +0100
@@ -4,11 +4,13 @@
 dnl     generate the file "configure", which is run during the build
 dnl     to configure the system for the local environment.
 
-AC_INIT
-AM_INIT_AUTOMAKE(fcgi, 2.4.0)
+AC_INIT(fcgi, 2.4.0)
+AM_INIT_AUTOMAKE(1.9 foreign)
 
 AM_CONFIG_HEADER(fcgi_config.h)
 
+AM_MAINTAINER_MODE
+
 AC_PROG_CC
 AC_PROG_CPP 
 AC_PROG_INSTALL 
