diff -u -r ../emacs-25.0.92/lib-src/emacsclient.c ./lib-src/emacsclient.c
--- ../emacs-25.0.92/lib-src/emacsclient.c	2016-03-02 05:21:42.000000000 -0500
+++ ./lib-src/emacsclient.c	2016-04-09 01:28:10.881760020 -0400
@@ -1210,7 +1210,7 @@
               }
             else
 #endif
-              tmpdir = "/tmp";
+              tmpdir = "@TERMUX_PREFIX@/tmp";
           }
 	socket_name_storage =
 	  xmalloc (strlen (tmpdir) + strlen (server_name) + EXTRA_SPACE);
