diff -u -r ../emacs-24.3/src/editfns.c ./src/editfns.c
--- ../emacs-24.3/src/editfns.c	2013-01-01 21:37:17.000000000 +0100
+++ ./src/editfns.c	2014-02-19 03:21:55.000000000 +0100
@@ -1304,7 +1304,7 @@
   if (!pw)
     return Qnil;
 
-  p = USER_FULL_NAME;
+  p = "unknown";
   /* Chop off everything after the first comma. */
   q = strchr (p, ',');
   full = make_string (p, q ? q - p : strlen (p));
