diff -ruN ./src/conf.c /home/vishal/AndroidDev/alsa/src/conf.c
--- ./src/conf.c	2016-08-02 23:18:38.000000000 +0530
+++ /home/vishal/AndroidDev/alsa/src/conf.c	2016-12-02 23:42:04.689785910 +0530
@@ -3558,7 +3558,7 @@
 			int n;
 
 #ifndef DOC_HIDDEN
-#if defined(_GNU_SOURCE) && !defined(__NetBSD__) && !defined(__FreeBSD__) && !defined(__sun)
+#if defined(_GNU_SOURCE) && !defined(__NetBSD__) && !defined(__FreeBSD__) && !defined(__sun) && !defined(__ANDROID__) && !defined(ANDROID)
 #define SORTFUNC	versionsort
 #else
 #define SORTFUNC	alphasort
diff -ruN ./src/ucm/parser.c /home/vishal/AndroidDev/alsa/src/ucm/parser.c
--- ./src/ucm/parser.c	2016-08-02 23:18:38.000000000 +0530
+++ /home/vishal/AndroidDev/alsa/src/ucm/parser.c	2016-12-02 23:42:17.406370931 +0530
@@ -1274,7 +1274,7 @@
 		"%s", env ? env : ALSA_USE_CASE_DIR);
 	filename[MAX_FILE-1] = '\0';
 
-#if defined(_GNU_SOURCE) && !defined(__NetBSD__) && !defined(__FreeBSD__) && !defined(__sun)
+#if defined(_GNU_SOURCE) && !defined(__NetBSD__) && !defined(__FreeBSD__) && !defined(__sun) && !defined(__ANDROID__) && !defined(ANDROID)
 #define SORTFUNC	versionsort
 #else
 #define SORTFUNC	alphasort
