diff -uNr xorgproto-2018.4/Makefile.in xorgproto-2018.4.mod/Makefile.in
--- xorgproto-2018.4/Makefile.in	2018-02-28 18:45:10.000000000 +0200
+++ xorgproto-2018.4.mod/Makefile.in	2018-09-28 13:17:51.642119182 +0300
@@ -475,21 +475,21 @@
         include/X11/extensions/applewmconst.h \
 	include/X11/extensions/applewmproto.h
 
-applewmprotopkgconfigdir = $(datadir)/pkgconfig
+applewmprotopkgconfigdir = $(libdir)/pkgconfig
 applewmprotopkgconfig_DATA = applewmproto.pc
 bigreqsdir = $(includedir)/X11/extensions
 bigreqs_HEADERS = \
 	include/X11/extensions/bigreqsproto.h \
 	include/X11/extensions/bigreqstr.h
 
-bigreqsprotopkgconfigdir = $(datadir)/pkgconfig
+bigreqsprotopkgconfigdir = $(libdir)/pkgconfig
 bigreqsprotopkgconfig_DATA = bigreqsproto.pc
 @LEGACY_TRUE@XCalibrateincludedir = $(includedir)/X11/extensions
 @LEGACY_TRUE@XCalibrateinclude_HEADERS = \
 @LEGACY_TRUE@	include/X11/extensions/xcalibrateproto.h \
 @LEGACY_TRUE@	include/X11/extensions/xcalibratewire.h
 
-@LEGACY_TRUE@calibrateprotopkgconfigdir = $(datadir)/pkgconfig
+@LEGACY_TRUE@calibrateprotopkgconfigdir = $(libdir)/pkgconfig
 @LEGACY_TRUE@calibrateprotopkgconfig_DATA = xcalibrateproto.pc
 @LEGACY_TRUE@calibrateproto_EXTRA_DIST = autogen.sh xcalibrateproto.pc.in
 compositedir = $(includedir)/X11/extensions
@@ -497,41 +497,41 @@
 	include/X11/extensions/composite.h \
 	include/X11/extensions/compositeproto.h
 
-compositeprotopkgconfigdir = $(datadir)/pkgconfig
+compositeprotopkgconfigdir = $(libdir)/pkgconfig
 compositeprotopkgconfig_DATA = compositeproto.pc
 damagedir = $(includedir)/X11/extensions
 damage_HEADERS = \
 	include/X11/extensions/damageproto.h \
 	include/X11/extensions/damagewire.h
 
-damageprotopkgconfigdir = $(datadir)/pkgconfig
+damageprotopkgconfigdir = $(libdir)/pkgconfig
 damageprotopkgconfig_DATA = damageproto.pc
 dmxdir = $(includedir)/X11/extensions
 dmx_HEADERS = \
 	include/X11/extensions/dmx.h \
 	include/X11/extensions/dmxproto.h
 
-dmxprotopkgconfigdir = $(datadir)/pkgconfig
+dmxprotopkgconfigdir = $(libdir)/pkgconfig
 dmxprotopkgconfig_DATA = dmxproto.pc
 dri2dir = $(includedir)/X11/extensions
 dri2_HEADERS = \
 	include/X11/extensions/dri2proto.h \
 	include/X11/extensions/dri2tokens.h
 
-dri2protopkgconfigdir = $(datadir)/pkgconfig
+dri2protopkgconfigdir = $(libdir)/pkgconfig
 dri2protopkgconfig_DATA = dri2proto.pc
 dri3dir = $(includedir)/X11/extensions
 dri3_HEADERS = \
 	include/X11/extensions/dri3proto.h
 
-dri3protopkgconfigdir = $(datadir)/pkgconfig
+dri3protopkgconfigdir = $(libdir)/pkgconfig
 dri3protopkgconfig_DATA = dri3proto.pc
 @LEGACY_TRUE@eviedir = $(includedir)/X11/extensions
 @LEGACY_TRUE@evie_HEADERS = \
 @LEGACY_TRUE@	include/X11/extensions/evieproto.h \
 @LEGACY_TRUE@	include/X11/extensions/Xeviestr.h
 
-@LEGACY_TRUE@evieprotopkgconfigdir = $(datadir)/pkgconfig
+@LEGACY_TRUE@evieprotopkgconfigdir = $(libdir)/pkgconfig
 @LEGACY_TRUE@evieprotopkgconfig_DATA = evieproto.pc
 @LEGACY_TRUE@evieproto_EXTRA_DIST = evieproto.pc.in
 fixesdir = $(includedir)/X11/extensions
@@ -539,7 +539,7 @@
 	include/X11/extensions/xfixesproto.h \
 	include/X11/extensions/xfixeswire.h
 
-fixesprotopkgconfigdir = $(datadir)/pkgconfig
+fixesprotopkgconfigdir = $(libdir)/pkgconfig
 fixesprotopkgconfig_DATA = fixesproto.pc
 @LEGACY_TRUE@fontcachedir = $(includedir)/X11/extensions
 @LEGACY_TRUE@fontcache_HEADERS = \
@@ -547,7 +547,7 @@
 @LEGACY_TRUE@	include/X11/extensions/fontcacheP.h \
 @LEGACY_TRUE@	include/X11/extensions/fontcachstr.h
 
-@LEGACY_TRUE@fontcacheprotopkgconfigdir = $(datadir)/pkgconfig
+@LEGACY_TRUE@fontcacheprotopkgconfigdir = $(libdir)/pkgconfig
 @LEGACY_TRUE@fontcacheprotopkgconfig_DATA = fontcacheproto.pc
 @LEGACY_TRUE@fontcacheproto_EXTRA_DIST = fontcacheproto.pc.in
 fontsdir = $(includedir)/X11/fonts
@@ -559,7 +559,7 @@
 	include/X11/fonts/fsmasks.h \
 	include/X11/fonts/FSproto.h
 
-fontsprotopkgconfigdir = $(datadir)/pkgconfig
+fontsprotopkgconfigdir = $(libdir)/pkgconfig
 fontsprotopkgconfig_DATA = fontsproto.pc
 gldir = $(includedir)/GL
 gl_HEADERS = \
@@ -572,7 +572,7 @@
 glinternal_HEADERS = \
 	include/GL/internal/glcore.h
 
-glprotopkgconfigdir = $(datadir)/pkgconfig
+glprotopkgconfigdir = $(libdir)/pkgconfig
 glprotopkgconfig_DATA = glproto.pc
 glproto_EXTRA_DIST = meson.build
 inputdir = $(includedir)/X11/extensions
@@ -582,7 +582,7 @@
 	include/X11/extensions/XI2.h \
 	include/X11/extensions/XI2proto.h
 
-inputprotopkgconfigdir = $(datadir)/pkgconfig
+inputprotopkgconfigdir = $(libdir)/pkgconfig
 inputprotopkgconfig_DATA = inputproto.pc
 kbdir = $(includedir)/X11/extensions
 kb_HEADERS = \
@@ -592,13 +592,13 @@
 	include/X11/extensions/XKBsrv.h \
 	include/X11/extensions/XKBstr.h
 
-kbprotopkgconfigdir = $(datadir)/pkgconfig
+kbprotopkgconfigdir = $(libdir)/pkgconfig
 kbprotopkgconfig_DATA = kbproto.pc
 @LEGACY_TRUE@lg3ddir = $(includedir)/X11/extensions
 @LEGACY_TRUE@lg3d_HEADERS = \
 @LEGACY_TRUE@	include/X11/extensions/lgewire.h
 
-@LEGACY_TRUE@lg3dprotopkgconfigdir = $(datadir)/pkgconfig
+@LEGACY_TRUE@lg3dprotopkgconfigdir = $(libdir)/pkgconfig
 @LEGACY_TRUE@lg3dprotopkgconfig_DATA = lg3dproto.pc
 @LEGACY_TRUE@lg3dproto_EXTRA_DIST = autogen.sh lg3dproto.pc.in
 xproxymngdir = $(includedir)/X11/PM
@@ -606,7 +606,7 @@
 	include/X11/PM/PM.h \
 	include/X11/PM/PMproto.h
 
-pmprotopkgconfigdir = $(datadir)/pkgconfig
+pmprotopkgconfigdir = $(libdir)/pkgconfig
 pmprotopkgconfig_DATA = xproxymngproto.pc
 pmproto_EXTRA_DIST = xproxymngproto.pc.in
 presentdir = $(includedir)/X11/extensions
@@ -614,14 +614,14 @@
 	include/X11/extensions/presentproto.h \
 	include/X11/extensions/presenttokens.h
 
-presentprotopkgconfigdir = $(datadir)/pkgconfig
+presentprotopkgconfigdir = $(libdir)/pkgconfig
 presentprotopkgconfig_DATA = presentproto.pc
 @LEGACY_TRUE@printdir = $(includedir)/X11/extensions
 @LEGACY_TRUE@print_HEADERS = \
 @LEGACY_TRUE@	include/X11/extensions/Print.h \
 @LEGACY_TRUE@	include/X11/extensions/Printstr.h
 
-@LEGACY_TRUE@printprotopkgconfigdir = $(datadir)/pkgconfig
+@LEGACY_TRUE@printprotopkgconfigdir = $(libdir)/pkgconfig
 @LEGACY_TRUE@printprotopkgconfig_DATA = printproto.pc
 @LEGACY_TRUE@miscmandir = $(MISC_MAN_DIR)
 @LEGACY_TRUE@miscman_DATA = man/Xprint.$(MISC_MAN_SUFFIX)
@@ -644,7 +644,7 @@
 	include/X11/extensions/randr.h \
 	include/X11/extensions/randrproto.h
 
-randrprotopkgconfigdir = $(datadir)/pkgconfig
+randrprotopkgconfigdir = $(libdir)/pkgconfig
 randrprotopkgconfig_DATA = randrproto.pc
 recorddir = $(includedir)/X11/extensions
 record_HEADERS = \
@@ -652,27 +652,27 @@
 	include/X11/extensions/recordproto.h \
 	include/X11/extensions/recordstr.h
 
-recordprotopkgconfigdir = $(datadir)/pkgconfig
+recordprotopkgconfigdir = $(libdir)/pkgconfig
 recordprotopkgconfig_DATA = recordproto.pc
 renderdir = $(includedir)/X11/extensions
 render_HEADERS = \
 	include/X11/extensions/render.h \
 	include/X11/extensions/renderproto.h
 
-renderprotopkgconfigdir = $(datadir)/pkgconfig
+renderprotopkgconfigdir = $(libdir)/pkgconfig
 renderprotopkgconfig_DATA = renderproto.pc
 resourcedir = $(includedir)/X11/extensions
 resource_HEADERS = \
 	include/X11/extensions/XResproto.h
 
-resourceprotopkgconfigdir = $(datadir)/pkgconfig
+resourceprotopkgconfigdir = $(libdir)/pkgconfig
 resourceprotopkgconfig_DATA = resourceproto.pc
 scrnsaverdir = $(includedir)/X11/extensions
 scrnsaver_HEADERS = \
 	include/X11/extensions/saver.h \
 	include/X11/extensions/saverproto.h
 
-scrnsaverprotopkgconfigdir = $(datadir)/pkgconfig
+scrnsaverprotopkgconfigdir = $(libdir)/pkgconfig
 scrnsaverprotopkgconfig_DATA = scrnsaverproto.pc
 trapdir = $(includedir)/X11/extensions
 trap_HEADERS = \
@@ -684,7 +684,7 @@
 	include/X11/extensions/xtraplibp.h \
 	include/X11/extensions/xtrapproto.h
 
-trapprotopkgconfigdir = $(datadir)/pkgconfig
+trapprotopkgconfigdir = $(libdir)/pkgconfig
 trapprotopkgconfig_DATA = trapproto.pc
 trapproto_EXTRA_DIST = trapproto.pc.in
 videodir = $(includedir)/X11/extensions
@@ -695,14 +695,14 @@
 	include/X11/extensions/XvMCproto.h \
 	include/X11/extensions/Xvproto.h
 
-videoprotopkgconfigdir = $(datadir)/pkgconfig
+videoprotopkgconfigdir = $(libdir)/pkgconfig
 videoprotopkgconfig_DATA = videoproto.pc
 windowswmdir = $(includedir)/X11/extensions
 windowswm_HEADERS = \
         include/X11/extensions/windowswm.h \
 	include/X11/extensions/windowswmstr.h
 
-windowswmprotopkgconfigdir = $(datadir)/pkgconfig
+windowswmprotopkgconfigdir = $(libdir)/pkgconfig
 windowswmprotopkgconfig_DATA = windowswmproto.pc
 xprotodir = $(includedir)/X11
 xproto_HEADERS = \
@@ -735,14 +735,14 @@
 nodist_xproto_HEADERS = \
 	include/X11/Xpoll.h
 
-x11protopkgconfigdir = $(datadir)/pkgconfig
+x11protopkgconfigdir = $(libdir)/pkgconfig
 x11protopkgconfig_DATA = xproto.pc
 xcmiscdir = $(includedir)/X11/extensions
 xcmisc_HEADERS = \
 	include/X11/extensions/xcmiscproto.h \
 	include/X11/extensions/xcmiscstr.h
 
-xcmiscprotopkgconfigdir = $(datadir)/pkgconfig
+xcmiscprotopkgconfigdir = $(libdir)/pkgconfig
 xcmiscprotopkgconfig_DATA = xcmiscproto.pc
 xextdir = $(includedir)/X11/extensions
 xext_HEADERS = \
@@ -783,7 +783,7 @@
 	include/X11/extensions/shmstr.h \
 	include/X11/extensions/syncstr.h
 
-xextprotopkgconfigdir = $(datadir)/pkgconfig
+xextprotopkgconfigdir = $(libdir)/pkgconfig
 xextprotopkgconfig_DATA = xextproto.pc
 xf86bigfontdir = $(includedir)/X11/extensions
 xf86bigfont_HEADERS = \
@@ -791,7 +791,7 @@
 	include/X11/extensions/xf86bigfproto.h \
 	include/X11/extensions/xf86bigfstr.h
 
-xf86bigfontprotopkgconfigdir = $(datadir)/pkgconfig
+xf86bigfontprotopkgconfigdir = $(libdir)/pkgconfig
 xf86bigfontprotopkgconfig_DATA = xf86bigfontproto.pc
 xf86dgadir = $(includedir)/X11/extensions
 xf86dga_HEADERS = \
@@ -803,7 +803,7 @@
 	include/X11/extensions/xf86dgaproto.h \
 	include/X11/extensions/xf86dgastr.h
 
-xf86dgaprotopkgconfigdir = $(datadir)/pkgconfig
+xf86dgaprotopkgconfigdir = $(libdir)/pkgconfig
 xf86dgaprotopkgconfig_DATA = xf86dgaproto.pc
 xf86dridir = $(includedir)/X11/dri
 xf86dri_HEADERS = \
@@ -811,14 +811,14 @@
 	include/X11/dri/xf86dristr.h \
 	include/X11/dri/xf86driproto.h
 
-xf86driprotopkgconfigdir = $(datadir)/pkgconfig
+xf86driprotopkgconfigdir = $(libdir)/pkgconfig
 xf86driprotopkgconfig_DATA = xf86driproto.pc
 xf86miscdir = $(includedir)/X11/extensions
 xf86misc_HEADERS = \
 	include/X11/extensions/xf86misc.h \
 	include/X11/extensions/xf86mscstr.h
 
-xf86miscprotopkgconfigdir = $(datadir)/pkgconfig
+xf86miscprotopkgconfigdir = $(libdir)/pkgconfig
 xf86miscprotopkgconfig_DATA = xf86miscproto.pc
 xf86miscproto_EXTRA_DIST = xf86miscproto.pc.in
 @LEGACY_TRUE@xf86rushdir = $(includedir)/X11/extensions
@@ -826,7 +826,7 @@
 @LEGACY_TRUE@	include/X11/extensions/xf86rush.h \
 @LEGACY_TRUE@	include/X11/extensions/xf86rushstr.h
 
-@LEGACY_TRUE@xf86rushprotopkgconfigdir = $(datadir)/pkgconfig
+@LEGACY_TRUE@xf86rushprotopkgconfigdir = $(libdir)/pkgconfig
 @LEGACY_TRUE@xf86rushprotopkgconfig_DATA = xf86rushproto.pc
 @LEGACY_TRUE@xf86rushproto_EXTRA_DIST = xf86rushproto.pc.in
 xf86vidmodedir = $(includedir)/X11/extensions
@@ -835,13 +835,13 @@
 	include/X11/extensions/xf86vmproto.h \
 	include/X11/extensions/xf86vmstr.h
 
-xf86vidmodeprotopkgconfigdir = $(datadir)/pkgconfig
+xf86vidmodeprotopkgconfigdir = $(libdir)/pkgconfig
 xf86vidmodeprotopkgconfig_DATA = xf86vidmodeproto.pc
 xineramadir = $(includedir)/X11/extensions
 xinerama_HEADERS = \
 	include/X11/extensions/panoramiXproto.h
 
-xineramaprotopkgconfigdir = $(datadir)/pkgconfig
+xineramaprotopkgconfigdir = $(libdir)/pkgconfig
 xineramaprotopkgconfig_DATA = xineramaproto.pc
 
 # SUBDIRS=specs
