diff -uNr xorgproto-2018.4/Makefile.am xorgproto-2018.4.mod/Makefile.am
--- xorgproto-2018.4/Makefile.am	2018-02-28 18:45:07.000000000 +0200
+++ xorgproto-2018.4.mod/Makefile.am	2018-04-09 19:14:29.228322693 +0300
@@ -3,7 +3,7 @@
         include/X11/extensions/applewmconst.h \
 	include/X11/extensions/applewmproto.h
 
-applewmprotopkgconfigdir = $(datadir)/pkgconfig
+applewmprotopkgconfigdir = $(libdir)/pkgconfig
 applewmprotopkgconfig_DATA = applewmproto.pc
 
 bigreqsdir = $(includedir)/X11/extensions
@@ -11,7 +11,7 @@
 	include/X11/extensions/bigreqsproto.h \
 	include/X11/extensions/bigreqstr.h
 
-bigreqsprotopkgconfigdir = $(datadir)/pkgconfig
+bigreqsprotopkgconfigdir = $(libdir)/pkgconfig
 bigreqsprotopkgconfig_DATA = bigreqsproto.pc
 
 if LEGACY
@@ -20,7 +20,7 @@
 	include/X11/extensions/xcalibrateproto.h \
 	include/X11/extensions/xcalibratewire.h
 
-calibrateprotopkgconfigdir = $(datadir)/pkgconfig
+calibrateprotopkgconfigdir = $(libdir)/pkgconfig
 calibrateprotopkgconfig_DATA = xcalibrateproto.pc
 
 calibrateproto_EXTRA_DIST = autogen.sh xcalibrateproto.pc.in
@@ -31,7 +31,7 @@
 	include/X11/extensions/composite.h \
 	include/X11/extensions/compositeproto.h
 
-compositeprotopkgconfigdir = $(datadir)/pkgconfig
+compositeprotopkgconfigdir = $(libdir)/pkgconfig
 compositeprotopkgconfig_DATA = compositeproto.pc
 
 damagedir = $(includedir)/X11/extensions
@@ -39,7 +39,7 @@
 	include/X11/extensions/damageproto.h \
 	include/X11/extensions/damagewire.h
 
-damageprotopkgconfigdir = $(datadir)/pkgconfig
+damageprotopkgconfigdir = $(libdir)/pkgconfig
 damageprotopkgconfig_DATA = damageproto.pc
 
 dmxdir = $(includedir)/X11/extensions
@@ -47,7 +47,7 @@
 	include/X11/extensions/dmx.h \
 	include/X11/extensions/dmxproto.h
 
-dmxprotopkgconfigdir = $(datadir)/pkgconfig
+dmxprotopkgconfigdir = $(libdir)/pkgconfig
 dmxprotopkgconfig_DATA = dmxproto.pc
 
 dri2dir = $(includedir)/X11/extensions
@@ -55,14 +55,14 @@
 	include/X11/extensions/dri2proto.h \
 	include/X11/extensions/dri2tokens.h
 
-dri2protopkgconfigdir = $(datadir)/pkgconfig
+dri2protopkgconfigdir = $(libdir)/pkgconfig
 dri2protopkgconfig_DATA = dri2proto.pc
 
 dri3dir = $(includedir)/X11/extensions
 dri3_HEADERS = \
 	include/X11/extensions/dri3proto.h
 
-dri3protopkgconfigdir = $(datadir)/pkgconfig
+dri3protopkgconfigdir = $(libdir)/pkgconfig
 dri3protopkgconfig_DATA = dri3proto.pc
 
 if LEGACY
@@ -71,7 +71,7 @@
 	include/X11/extensions/evieproto.h \
 	include/X11/extensions/Xeviestr.h
 
-evieprotopkgconfigdir = $(datadir)/pkgconfig
+evieprotopkgconfigdir = $(libdir)/pkgconfig
 evieprotopkgconfig_DATA = evieproto.pc
 
 evieproto_EXTRA_DIST = evieproto.pc.in
@@ -82,7 +82,7 @@
 	include/X11/extensions/xfixesproto.h \
 	include/X11/extensions/xfixeswire.h
 
-fixesprotopkgconfigdir = $(datadir)/pkgconfig
+fixesprotopkgconfigdir = $(libdir)/pkgconfig
 fixesprotopkgconfig_DATA = fixesproto.pc
 
 if LEGACY
@@ -92,7 +92,7 @@
 	include/X11/extensions/fontcacheP.h \
 	include/X11/extensions/fontcachstr.h
 
-fontcacheprotopkgconfigdir = $(datadir)/pkgconfig
+fontcacheprotopkgconfigdir = $(libdir)/pkgconfig
 fontcacheprotopkgconfig_DATA = fontcacheproto.pc
 
 fontcacheproto_EXTRA_DIST = fontcacheproto.pc.in
@@ -107,7 +107,7 @@
 	include/X11/fonts/fsmasks.h \
 	include/X11/fonts/FSproto.h
 
-fontsprotopkgconfigdir = $(datadir)/pkgconfig
+fontsprotopkgconfigdir = $(libdir)/pkgconfig
 fontsprotopkgconfig_DATA = fontsproto.pc
 
 gldir = $(includedir)/GL
@@ -121,7 +121,7 @@
 glinternal_HEADERS = \
 	include/GL/internal/glcore.h
 
-glprotopkgconfigdir = $(datadir)/pkgconfig
+glprotopkgconfigdir = $(libdir)/pkgconfig
 glprotopkgconfig_DATA = glproto.pc
 
 glproto_EXTRA_DIST = meson.build
@@ -133,7 +133,7 @@
 	include/X11/extensions/XI2.h \
 	include/X11/extensions/XI2proto.h
 
-inputprotopkgconfigdir = $(datadir)/pkgconfig
+inputprotopkgconfigdir = $(libdir)/pkgconfig
 inputprotopkgconfig_DATA = inputproto.pc
 
 kbdir = $(includedir)/X11/extensions
@@ -144,7 +144,7 @@
 	include/X11/extensions/XKBsrv.h \
 	include/X11/extensions/XKBstr.h
 
-kbprotopkgconfigdir = $(datadir)/pkgconfig
+kbprotopkgconfigdir = $(libdir)/pkgconfig
 kbprotopkgconfig_DATA = kbproto.pc
 
 if LEGACY
@@ -152,7 +152,7 @@
 lg3d_HEADERS = \
 	include/X11/extensions/lgewire.h
 
-lg3dprotopkgconfigdir = $(datadir)/pkgconfig
+lg3dprotopkgconfigdir = $(libdir)/pkgconfig
 lg3dprotopkgconfig_DATA = lg3dproto.pc
 
 lg3dproto_EXTRA_DIST = autogen.sh lg3dproto.pc.in
@@ -163,7 +163,7 @@
 	include/X11/PM/PM.h \
 	include/X11/PM/PMproto.h
 
-pmprotopkgconfigdir = $(datadir)/pkgconfig
+pmprotopkgconfigdir = $(libdir)/pkgconfig
 pmprotopkgconfig_DATA = xproxymngproto.pc
 
 pmproto_EXTRA_DIST = xproxymngproto.pc.in
@@ -173,7 +173,7 @@
 	include/X11/extensions/presentproto.h \
 	include/X11/extensions/presenttokens.h
 
-presentprotopkgconfigdir = $(datadir)/pkgconfig
+presentprotopkgconfigdir = $(libdir)/pkgconfig
 presentprotopkgconfig_DATA = presentproto.pc
 
 if LEGACY
@@ -182,7 +182,7 @@
 	include/X11/extensions/Print.h \
 	include/X11/extensions/Printstr.h
 
-printprotopkgconfigdir = $(datadir)/pkgconfig
+printprotopkgconfigdir = $(libdir)/pkgconfig
 printprotopkgconfig_DATA = printproto.pc
 
 miscmandir = $(MISC_MAN_DIR)
@@ -226,7 +226,7 @@
 	include/X11/extensions/randr.h \
 	include/X11/extensions/randrproto.h
 
-randrprotopkgconfigdir = $(datadir)/pkgconfig
+randrprotopkgconfigdir = $(libdir)/pkgconfig
 randrprotopkgconfig_DATA = randrproto.pc
 
 recorddir = $(includedir)/X11/extensions
@@ -235,7 +235,7 @@
 	include/X11/extensions/recordproto.h \
 	include/X11/extensions/recordstr.h
 
-recordprotopkgconfigdir = $(datadir)/pkgconfig
+recordprotopkgconfigdir = $(libdir)/pkgconfig
 recordprotopkgconfig_DATA = recordproto.pc
 
 renderdir = $(includedir)/X11/extensions
@@ -243,14 +243,14 @@
 	include/X11/extensions/render.h \
 	include/X11/extensions/renderproto.h
 
-renderprotopkgconfigdir = $(datadir)/pkgconfig
+renderprotopkgconfigdir = $(libdir)/pkgconfig
 renderprotopkgconfig_DATA = renderproto.pc
 
 resourcedir = $(includedir)/X11/extensions
 resource_HEADERS = \
 	include/X11/extensions/XResproto.h
 
-resourceprotopkgconfigdir = $(datadir)/pkgconfig
+resourceprotopkgconfigdir = $(libdir)/pkgconfig
 resourceprotopkgconfig_DATA = resourceproto.pc
 
 scrnsaverdir = $(includedir)/X11/extensions
@@ -258,7 +258,7 @@
 	include/X11/extensions/saver.h \
 	include/X11/extensions/saverproto.h
 
-scrnsaverprotopkgconfigdir = $(datadir)/pkgconfig
+scrnsaverprotopkgconfigdir = $(libdir)/pkgconfig
 scrnsaverprotopkgconfig_DATA = scrnsaverproto.pc
 
 trapdir = $(includedir)/X11/extensions
@@ -271,7 +271,7 @@
 	include/X11/extensions/xtraplibp.h \
 	include/X11/extensions/xtrapproto.h
 
-trapprotopkgconfigdir = $(datadir)/pkgconfig
+trapprotopkgconfigdir = $(libdir)/pkgconfig
 trapprotopkgconfig_DATA = trapproto.pc
 
 trapproto_EXTRA_DIST = trapproto.pc.in
@@ -284,7 +284,7 @@
 	include/X11/extensions/XvMCproto.h \
 	include/X11/extensions/Xvproto.h
 
-videoprotopkgconfigdir = $(datadir)/pkgconfig
+videoprotopkgconfigdir = $(libdir)/pkgconfig
 videoprotopkgconfig_DATA = videoproto.pc
 
 windowswmdir = $(includedir)/X11/extensions
@@ -292,7 +292,7 @@
         include/X11/extensions/windowswm.h \
 	include/X11/extensions/windowswmstr.h
 
-windowswmprotopkgconfigdir = $(datadir)/pkgconfig
+windowswmprotopkgconfigdir = $(libdir)/pkgconfig
 windowswmprotopkgconfig_DATA = windowswmproto.pc
 
 xprotodir = $(includedir)/X11
@@ -326,7 +326,7 @@
 nodist_xproto_HEADERS = \
 	include/X11/Xpoll.h
 
-x11protopkgconfigdir = $(datadir)/pkgconfig
+x11protopkgconfigdir = $(libdir)/pkgconfig
 x11protopkgconfig_DATA = xproto.pc
 
 xcmiscdir = $(includedir)/X11/extensions
@@ -334,7 +334,7 @@
 	include/X11/extensions/xcmiscproto.h \
 	include/X11/extensions/xcmiscstr.h
 
-xcmiscprotopkgconfigdir = $(datadir)/pkgconfig
+xcmiscprotopkgconfigdir = $(libdir)/pkgconfig
 xcmiscprotopkgconfig_DATA = xcmiscproto.pc
 
 xextdir = $(includedir)/X11/extensions
@@ -376,7 +376,7 @@
 	include/X11/extensions/shmstr.h \
 	include/X11/extensions/syncstr.h
 
-xextprotopkgconfigdir = $(datadir)/pkgconfig
+xextprotopkgconfigdir = $(libdir)/pkgconfig
 xextprotopkgconfig_DATA = xextproto.pc
 
 xf86bigfontdir = $(includedir)/X11/extensions
@@ -385,7 +385,7 @@
 	include/X11/extensions/xf86bigfproto.h \
 	include/X11/extensions/xf86bigfstr.h
 
-xf86bigfontprotopkgconfigdir = $(datadir)/pkgconfig
+xf86bigfontprotopkgconfigdir = $(libdir)/pkgconfig
 xf86bigfontprotopkgconfig_DATA = xf86bigfontproto.pc
 
 xf86dgadir = $(includedir)/X11/extensions
@@ -398,7 +398,7 @@
 	include/X11/extensions/xf86dgaproto.h \
 	include/X11/extensions/xf86dgastr.h
 
-xf86dgaprotopkgconfigdir = $(datadir)/pkgconfig
+xf86dgaprotopkgconfigdir = $(libdir)/pkgconfig
 xf86dgaprotopkgconfig_DATA = xf86dgaproto.pc
 
 xf86dridir = $(includedir)/X11/dri
@@ -407,7 +407,7 @@
 	include/X11/dri/xf86dristr.h \
 	include/X11/dri/xf86driproto.h
 
-xf86driprotopkgconfigdir = $(datadir)/pkgconfig
+xf86driprotopkgconfigdir = $(libdir)/pkgconfig
 xf86driprotopkgconfig_DATA = xf86driproto.pc
 
 xf86miscdir = $(includedir)/X11/extensions
@@ -415,7 +415,7 @@
 	include/X11/extensions/xf86misc.h \
 	include/X11/extensions/xf86mscstr.h
 
-xf86miscprotopkgconfigdir = $(datadir)/pkgconfig
+xf86miscprotopkgconfigdir = $(libdir)/pkgconfig
 xf86miscprotopkgconfig_DATA = xf86miscproto.pc
 
 xf86miscproto_EXTRA_DIST = xf86miscproto.pc.in
@@ -426,7 +426,7 @@
 	include/X11/extensions/xf86rush.h \
 	include/X11/extensions/xf86rushstr.h
 
-xf86rushprotopkgconfigdir = $(datadir)/pkgconfig
+xf86rushprotopkgconfigdir = $(libdir)/pkgconfig
 xf86rushprotopkgconfig_DATA = xf86rushproto.pc
 
 xf86rushproto_EXTRA_DIST = xf86rushproto.pc.in
@@ -438,14 +438,14 @@
 	include/X11/extensions/xf86vmproto.h \
 	include/X11/extensions/xf86vmstr.h
 
-xf86vidmodeprotopkgconfigdir = $(datadir)/pkgconfig
+xf86vidmodeprotopkgconfigdir = $(libdir)/pkgconfig
 xf86vidmodeprotopkgconfig_DATA = xf86vidmodeproto.pc
 
 xineramadir = $(includedir)/X11/extensions
 xinerama_HEADERS = \
 	include/X11/extensions/panoramiXproto.h
 
-xineramaprotopkgconfigdir = $(datadir)/pkgconfig
+xineramaprotopkgconfigdir = $(libdir)/pkgconfig
 xineramaprotopkgconfig_DATA = xineramaproto.pc
 
 # SUBDIRS=specs
