diff -u -r ../autoconf-2.69/lib/Autom4te/General.pm ./lib/Autom4te/General.pm
--- ../autoconf-2.69/lib/Autom4te/General.pm	2012-04-24 16:44:15.000000000 -0400
+++ ./lib/Autom4te/General.pm	2015-05-17 16:11:45.492445577 -0400
@@ -300,7 +300,7 @@
 sub mktmpdir ($)
 {
   my ($signature) = @_;
-  my $TMPDIR = $ENV{'TMPDIR'} || '/tmp';
+  my $TMPDIR = $ENV{'TMPDIR'} || '@TERMUX_PREFIX@/tmp';
   my $quoted_tmpdir = shell_quote ($TMPDIR);
 
   # If mktemp supports dirs, use it.
