--- ./Cargo.toml.bak	2022-02-19 02:03:29.085335654 +0530
+++ ./Cargo.toml	2022-02-19 02:04:45.461335659 +0530
@@ -20,7 +20,7 @@
 
 [dependencies]
 ansi_term = "0.12.0"
-app_dirs = { version = "2", package = "app_dirs2" }
+app_dirs = { version = "1.2.1" }
 atty = "0.2"
 clap = { version = "=3.0.0-rc.11", features = ["std", "derive", "suggestions", "color"], default-features = false }
 env_logger = { version = "0.9", optional = true }
