--- src/lib/egif_lib.c.orig	2017-12-09 15:46:41.761228962 +0000
+++ src/lib/egif_lib.c	2017-12-09 15:47:16.631668749 +0000
@@ -25,6 +25,14 @@
 #include "gif_lib.h"
 #include "gif_lib_private.h"
 
+#ifndef S_IREAD
+#define S_IREAD  00400
+#endif
+
+#ifndef S_IWRITE
+#define S_IWRITE 00200
+#endif
+
 /* Masks given codes to BitsPerPixel, to make sure all codes are in range: */
 /*@+charint@*/
 static const GifPixelType CodeMask[] = {
