--- a/libxtables/Makefile.in
+++ b/libxtables/Makefile.in
@@ -148,7 +148,7 @@
 am__v_lt_1 = 
 libxtables_la_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(libxtables_la_CFLAGS) \
-	$(CFLAGS) $(libxtables_la_LDFLAGS) $(LDFLAGS) -o $@
+	$(CFLAGS) $(libxtables_la_LDFLAGS) $(LDFLAGS) -landroid-spawn -o $@
 AM_V_P = $(am__v_P_@AM_V@)
 am__v_P_ = $(am__v_P_@AM_DEFAULT_V@)
 am__v_P_0 = false
