--- a/source/common/primitives.cpp
+++ b/source/common/primitives.cpp
@@ -295,7 +295,7 @@
 void PFX(cpu_cpuid)(uint32_t, uint32_t *eax, uint32_t *, uint32_t *, uint32_t *) { *eax = 0; }
 void PFX(cpu_xgetbv)(uint32_t, uint32_t *, uint32_t *) {}
 
-#if X265_ARCH_ARM == 0
+#if X265_ARCH_ARM == 0 || !ENABLE_ASSEMBLY
 void PFX(cpu_neon_test)(void) {}
 int PFX(cpu_fast_neon_mrc_test)(void) { return 0; }
 #endif // X265_ARCH_ARM
