diff -u -r ../postgresql-11.0/src/bin/initdb/initdb.c ./src/bin/initdb/initdb.c
--- ../postgresql-11.0/src/bin/initdb/initdb.c	2018-10-15 21:12:02.000000000 +0000
+++ ./src/bin/initdb/initdb.c	2018-10-18 17:54:16.149615230 +0000
@@ -120,8 +120,8 @@
 
 /* values to be obtained from arguments */
 static char *pg_data = NULL;
-static char *encoding = NULL;
-static char *locale = NULL;
+static char *encoding = "UTF-8";
+static char *locale = "en_US.UTF-8";
 static char *lc_collate = NULL;
 static char *lc_ctype = NULL;
 static char *lc_monetary = NULL;
