--- ./src/include/pg_config_manual.h.orig	2020-09-21 22:47:36.000000000 +0200
+++ ./src/include/pg_config_manual.h	2020-10-18 10:17:39.656801301 +0200
@@ -13,6 +13,8 @@
  *------------------------------------------------------------------------
  */
 
+#include <paths.h>
+
 /*
  * This is the default value for wal_segment_size to be used when initdb is run
  * without the --wal-segsize option.  It must be a valid segment size.
@@ -203,8 +205,12 @@
 #ifndef WIN32
 #define DEFAULT_PGSOCKET_DIR  "/tmp"
 #else
+#ifdef _PATH_TMP
+#define DEFAULT_PGSOCKET_DIR _PATH_TMP
+#else
 #define DEFAULT_PGSOCKET_DIR ""
 #endif
+#endif
 
 /*
  * This is the default event source for Windows event log.
