diff -u -r ../postgresql-10.0/src/bin/initdb/initdb.c ./src/bin/initdb/initdb.c
--- ../postgresql-10.0/src/bin/initdb/initdb.c	2017-10-02 23:09:15.000000000 +0200
+++ ./src/bin/initdb/initdb.c	2017-10-06 21:45:24.088346650 +0200
@@ -119,8 +119,8 @@
 
 /* values to be obtained from arguments */
 static char *pg_data = "";
-static char *encoding = "";
-static char *locale = "";
+static char *encoding = "UTF-8";
+static char *locale = "en_US.UTF-8";
 static char *lc_collate = "";
 static char *lc_ctype = "";
 static char *lc_monetary = "";
