diff -uNr microsocks/Makefile microsocks-mod/Makefile
--- microsocks/Makefile 2020-10-02 23:56:02.119088328 +0600
+++ microsocks-mod/Makefile     2020-10-03 00:58:49.228942473 +0600
@@ -1,7 +1,7 @@
 # if you want to change/override some variables, do so in a file called
 # config.mak, which is gets included automatically if it exists.

-prefix = /usr/local
+prefix = @TERMUX_PREFIX@
 bindir = $(prefix)/bin

 PROG = microsocks
