diff --git a/yt_dlp/utils/_utils.py b/yt_dlp/utils/_utils.py
index c6ae21f..f7111da 100644
--- a/yt_dlp/utils/_utils.py
+++ b/yt_dlp/utils/_utils.py
@@ -4727,7 +4727,7 @@ def get_user_config_dirs(package_name):
 
 def get_system_config_dirs(package_name):
     # /etc/package_name
-    yield os.path.join('/etc', package_name)
+    yield os.path.join('@TERMUX_PREFIX@/etc', package_name)
 
 
 def time_seconds(**kwargs):
