--- a/sdk/src/posix/net.cpp
+++ b/sdk/src/posix/net.cpp
@@ -23,7 +23,7 @@
 #include "mega/posix/meganet.h"
 #include "mega/logging.h"
 
-#if defined(__ANDROID__) && ARES_VERSION >= 0x010F00
+#if defined(__ANDROID__) && !defined(__TERMUX__) && ARES_VERSION >= 0x010F00
 #include <jni.h>
 extern JavaVM *MEGAjvm;
 #endif
@@ -298,7 +298,7 @@
             LOG_debug << "c-ares version: " << aresversion;
         }
 
-#if (defined(ANDROID) || defined(__ANDROID__)) && ARES_VERSION >= 0x010F00
+#if (defined(ANDROID) || defined(__ANDROID__)) && !defined(__TERMUX__) && ARES_VERSION >= 0x010F00
         initialize_android();
 #endif
 #endif
@@ -2941,7 +2941,7 @@
 }
 
 #ifdef MEGA_USE_C_ARES
-#if (defined(ANDROID) || defined(__ANDROID__)) && ARES_VERSION >= 0x010F00
+#if (defined(ANDROID) || defined(__ANDROID__)) && !defined(__TERMUX__) && ARES_VERSION >= 0x010F00
 
 void CurlHttpIO::initialize_android()
 {
