--- ../enscript.cfg.in.orig	2022-02-22 20:02:37.744671349 +0000
+++ ./lib/enscript.cfg.in	2022-02-22 20:03:09.737858022 +0000
@@ -76,7 +76,7 @@
 # HighlightBars: 2
 
 # Library path.  Let's use built-in default.
-# LibraryPath:
+LibraryPath: @TERMUX_PREFIX@/share/enscript
 
 # How wrapped lines are marked in printout.  Possible values are: `none',
 # `plus', `box' and `arrow'.  As a default, we do not mark them.
