--- a/src/Makefile
+++ b/src/Makefile
@@ -710,7 +710,7 @@ endif
         # enable UTF8 filename translation
         # For Linux (GLibc), iconv is built into the C library so no LOPT needed.
         COPT += -DENABLE_FILENAMES_ICONV
-        ifneq ($(filter-out NetBSD Linux,$(PLATFORM)),)
+        ifneq ($(filter-out NetBSD,$(PLATFORM)),)
           ifneq ($(DEB_HOST_ARCH_OS), kfreebsd)
             LOPT += -liconv
           endif

