diff -uNr thunar-thunar-4.15.0/icons/Makefile.am thunar-thunar-4.15.0.mod/icons/Makefile.am
--- thunar-thunar-4.15.0/icons/Makefile.am	2020-06-19 01:50:01.000000000 +0300
+++ thunar-thunar-4.15.0.mod/icons/Makefile.am	2020-06-22 15:54:54.826868756 +0300
@@ -5,7 +5,7 @@
 	128x128								\
 	scalable
 
-gtk_update_icon_cache = gtk-update-icon-cache -f -t $(datadir)/icons/hicolor
+gtk_update_icon_cache = true
 
 install-data-hook:
 	@-if test -z "$(DESTDIR)"; then					\
