diff -uNr thunar-thunar-4.15.0/plugins/thunar-uca/Makefile.am thunar-thunar-4.15.0.mod/plugins/thunar-uca/Makefile.am
--- thunar-thunar-4.15.0/plugins/thunar-uca/Makefile.am	2020-06-19 01:50:01.000000000 +0300
+++ thunar-thunar-4.15.0.mod/plugins/thunar-uca/Makefile.am	2020-06-22 16:18:21.197997945 +0300
@@ -57,7 +57,7 @@
 thunar_uca_templates =							\
 	thunar-uca-ui.xml
 
-GLIB_COMPILE_RESOURCES = $(shell $(PKG_CONFIG) --variable=glib_compile_resources gio-2.0)
+GLIB_COMPILE_RESOURCES = glib-compile-resources
 
 resource_files = $(shell $(GLIB_COMPILE_RESOURCES) --sourcedir=$(srcdir) --generate-dependencies $(srcdir)/thunar-uca.gresource.xml)
 
