https://github.com/termux/termux-packages/issues/12865

--- a/thunar/thunar-preferences-dialog.c
+++ b/thunar/thunar-preferences-dialog.c
@@ -1333,6 +1333,7 @@ thunar_preferences_dialog_init (ThunarPr
       gtk_container_add (GTK_CONTAINER (frame), infobar);
     }
 
+#if 0
   /*
    Shortcuts
   */
@@ -1354,6 +1355,7 @@ thunar_preferences_dialog_init (ThunarPr
                                     );
   gtk_container_add (GTK_CONTAINER (frame), grid);
   gtk_widget_show (grid);
+#endif
 }
 
 
