Fixes https://github.com/termux/termux-packages/issues/12865.

--- a/thunar/thunar-preferences-dialog.c
+++ b/thunar/thunar-preferences-dialog.c
@@ -1347,10 +1347,10 @@ thunar_preferences_dialog_init (ThunarPr
   gtk_widget_show (frame);
 
   grid = xfce_shortcuts_editor_new (13,
-                                    _("Window"), thunar_window_get_action_entries (), THUNAR_WINDOW_N_ACTIONS,
-                                    _("View"), thunar_standard_view_get_action_entries (), THUNAR_STANDARD_VIEW_N_ACTIONS,
-                                    _("Launcher"), thunar_action_manager_get_action_entries (), THUNAR_ACTION_MANAGER_N_ACTIONS,
-                                    _("Status Bar"), thunar_statusbar_get_action_entries (), THUNAR_STATUS_BAR_N_ACTIONS
+                                    _("Window"), thunar_window_get_action_entries (), (size_t)THUNAR_WINDOW_N_ACTIONS,
+                                    _("View"), thunar_standard_view_get_action_entries (), (size_t)THUNAR_STANDARD_VIEW_N_ACTIONS,
+                                    _("Launcher"), thunar_action_manager_get_action_entries (), (size_t)THUNAR_ACTION_MANAGER_N_ACTIONS,
+                                    _("Status Bar"), thunar_statusbar_get_action_entries (), (size_t)THUNAR_STATUS_BAR_N_ACTIONS
                                     );
   gtk_container_add (GTK_CONTAINER (frame), grid);
   gtk_widget_show (grid);
