--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -226,6 +226,7 @@
    message( STATUS "Building for 32-bit target" )
 endif()
 
+if( FALSE )
 # Check for compiler flags
 if( CMAKE_CXX_COMPILER_ID MATCHES "AppleClang|Clang|GNU" )
    check_cxx_compiler_flag( "-mmmx" HAVE_MMX )
@@ -255,6 +256,7 @@
       set( SSE2_FLAG "/arch:SSE2" )
    endif()
 endif()
+endif( FALSE )
 
 check_include_files( "float.h;stdarg.h;stdlib.h;string.h" STDC_HEADERS )
 
