--- a/cmake-proxies/portmidi/CMakeLists.txt
+++ b/cmake-proxies/portmidi/CMakeLists.txt
@@ -3,7 +3,7 @@
 
 def_vars()
 
-if( CMAKE_SYSTEM_NAME MATCHES "Linux|FreeBSD" )
+if( CMAKE_SYSTEM_NAME MATCHES "Android|Linux|FreeBSD" )
    if( ${_OPT}use_pa_alsa )
       find_package( ALSA )
    endif()
@@ -29,7 +29,7 @@
          ${TARGET_ROOT}/porttime/ptmacosx_mach.c
       >
 
-      $<$<PLATFORM_ID:Linux,FreeBSD,CYGWIN>:
+      $<$<PLATFORM_ID:Android,Linux,FreeBSD,CYGWIN>:
          ${TARGET_ROOT}/pm_linux/finddefault.c
          ${TARGET_ROOT}/pm_linux/pmlinux.c
          ${TARGET_ROOT}/porttime/ptlinux.c
@@ -53,7 +53,7 @@
          ${TARGET_ROOT}/pm_mac
       >
 
-      $<$<PLATFORM_ID:Linux,FreeBSD>:
+      $<$<PLATFORM_ID:Android,Linux,FreeBSD>:
          ${TARGET_ROOT}/pm_linux
       >
 )
