--- a/cmake-proxies/portaudio-v19/CMakeLists.txt
+++ b/cmake-proxies/portaudio-v19/CMakeLists.txt
@@ -56,7 +56,7 @@
          "Use the portaudio CoreAudio interface if available"
          YES
       )
-   elseif( CMAKE_SYSTEM_NAME MATCHES "Linux|FreeBSD" )
+   elseif( CMAKE_SYSTEM_NAME MATCHES "Android|Linux|FreeBSD" )
       cmd_option(
          ${_OPT}use_pa_alsa
          "Use the portaudio ALSA interface if available"
@@ -118,7 +118,7 @@
          ${TARGET_ROOT}/src/hostapi/coreaudio/pa_mac_core_utilities.c
       >
 
-      $<$<PLATFORM_ID:Darwin,Linux,FreeBSD,CYGWIN>:
+      $<$<PLATFORM_ID:Darwin,Android,Linux,FreeBSD,CYGWIN>:
          ${TARGET_ROOT}/src/os/unix/pa_unix_hostapis.c
          ${TARGET_ROOT}/src/os/unix/pa_unix_util.c
       >
@@ -158,7 +158,7 @@
          ${TARGET_ROOT}/src/os/win
       >
 
-      $<$<PLATFORM_ID:Darwin,Linux,FreeBSD,CYGWIN>:
+      $<$<PLATFORM_ID:Darwin,Android,Linux,FreeBSD,CYGWIN>:
          ${TARGET_ROOT}/src/os/unix
       >
 
