+++ ./libraries/lib-wx-wrappers/CMakeLists.txt
@@ -25,7 +25,7 @@
       FileDialog/mac/FileDialogPrivate.mm
       FileDialog/mac/FileDialogPrivate.h
    >
-   $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD,CYGWIN>:
+   $<$<PLATFORM_ID:Android,Linux,FreeBSD,OpenBSD,CYGWIN>:
       FileDialog/gtk/FileDialogPrivate.cpp
       FileDialog/gtk/FileDialogPrivate.h
    >
@@ -42,5 +42,5 @@
 
 audacity_library( lib-wx-wrappers "${SOURCES}" "${LIBRARIES}"
    ""
-   "$<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD,NetBSD,CYGWIN>:PkgConfig::GTK>"
+   "$<$<PLATFORM_ID:Android,Linux,FreeBSD,OpenBSD,NetBSD,CYGWIN>:PkgConfig::GTK>"
 )
