+++ ./src/AudacityApp.cpp
@@ -1148,7 +1148,7 @@
 #define WL(lang,sublang)
 #endif
 
-#if wxCHECK_VERSION(3, 0, 1) && !wxCHECK_VERSION(3, 1, 6)
+#if wxCHECK_VERSION(3, 0, 1) && !wxCHECK_VERSION(3, 1, 6) && !defined __ANDROID__
 wxLanguageInfo userLangs[] =
 {
    // Included upstream in version 3.1.6
@@ -1361,7 +1361,7 @@
    //
    // TODO:  The whole Language initialization really need to be reworked.
    //        It's all over the place.
-#if wxCHECK_VERSION(3, 0, 1) && !wxCHECK_VERSION(3, 1, 6)
+#if wxCHECK_VERSION(3, 0, 1) && !wxCHECK_VERSION(3, 1, 6) && !defined __ANDROID__
    for (size_t i = 0, cnt = WXSIZEOF(userLangs); i < cnt; i++)
    {
       wxLocale::AddLanguage(userLangs[i]);
