error: no matching function for call to ‘wxTimerEvent::wxTimerEvent()’

--- a/src/TimerRecordDialog.cpp
+++ b/src/TimerRecordDialog.cpp
@@ -254,7 +254,8 @@ void TimerRecordDialog::OnDatePicker_Start(wxDateEvent& WXUNUSED(event))
 
    // User might have had the dialog up for a while, or
    // had a future day, set hour of day less than now's, then changed day to today.
-   wxTimerEvent dummyTimerEvent;
+   wxTimer dummyTimer;
+   wxTimerEvent dummyTimerEvent(dummyTimer);
    this->OnTimer(dummyTimerEvent);
 
    // Always update End for changed Start, keeping Duration constant.
@@ -572,7 +573,8 @@ int TimerRecordDialog::RunWaitDialog()
 
          // Make sure that start and end time are updated, so we always get the full
          // duration, even if there's some delay getting here.
-         wxTimerEvent dummyTimerEvent;
+         wxTimer dummyTimer;
+         wxTimerEvent dummyTimerEvent(dummyTimer);
          this->OnTimer(dummyTimerEvent);
 
          // Loop for progress display during recording.
