--- ../cache/GEGL_0_3_14/configure.ac	2017-02-27 14:43:22.000000000 +0000
+++ ./configure.ac	2017-05-06 03:37:24.376507193 +0000
@@ -979,7 +979,7 @@
  
 AC_ARG_WITH(graphviz, [  --without-graphviz      build without graphviz support])
 
-have_graphviz="no"
+have_graphviz="yes"
 if test "x$with_graphviz" != "xno"; then
   AC_PATH_PROG(GRAPHVIZ, dot, no)
 
@@ -990,7 +990,7 @@
     have_graphviz="yes"
   fi
 fi
-
+have_graphviz="yes"
 AM_CONDITIONAL(HAVE_GRAPHVIZ, test "$have_graphviz" = "yes")
 
 
