--- ../cache/GEGL_0_3_14/libs/npd/Makefile.am	2017-02-27 14:43:22.000000000 +0000
+++ ./libs/npd/Makefile.am	2017-05-11 04:11:43.452797750 +0000
@@ -31,7 +31,7 @@
 	$(GEGL_NPD_SOURCES)
 
 libgegl_npd_@GEGL_API_VERSION@_la_LDFLAGS = \
-	-avoid-version -export-dynamic $(no_undefined)
+	-lm -avoid-version -export-dynamic $(no_undefined)
 
 libgegl_npd_@GEGL_API_VERSION@_la_CFLAGS = \
   $(AM_CFLAGS)
