--- ../cache/GEGL_0_3_14/operations/external/Makefile.am	2017-02-27 14:43:22.000000000 +0000
+++ ./operations/external/Makefile.am	2017-05-11 03:58:57.688240785 +0000
@@ -107,11 +107,11 @@
 if HAVE_AVFORMAT
 ops += ff-load.la ff-save.la
 ff_load_la_SOURCES = ff-load.c
-ff_load_la_LIBADD = $(op_libs) $(AVFORMAT_LIBS)
+ff_load_la_LIBADD = $(op_libs) $(AVFORMAT_LIBS) -lavutil
 ff_load_la_CFLAGS = $(AM_CFLAGS) $(AVFORMAT_CFLAGS) 
 
 ff_save_la_SOURCES = ff-save.c
-ff_save_la_LIBADD = $(op_libs) $(AVFORMAT_LIBS)
+ff_save_la_LIBADD = $(op_libs) $(AVFORMAT_LIBS) -lavutil
 ff_save_la_CFLAGS = $(AM_CFLAGS) $(AVFORMAT_CFLAGS) -Wno-deprecated-declarations
 endif
 
