diff -uNr delve-1.20.0/service/debugger/debugger_unix_test.go delve-1.20.0.mod/service/debugger/debugger_unix_test.go
--- delve-1.20.0/service/debugger/debugger_unix_test.go	2022-12-09 02:34:57.000000000 +0800
+++ delve-1.20.0.mod/service/debugger/debugger_unix_test.go	2022-12-09 22:38:00.158612578 +0800
@@ -32,6 +32,7 @@
 		"windows": "linux",
 		"freebsd": "windows",
 		"linux":   "windows",
+		"android": "windows",
 	}
 	if runtime.GOARCH == "arm64" && runtime.GOOS == "linux" {
 		os.Setenv("GOARCH", "amd64")
