diff -uNr delve-1.9.0/service/debugger/debugger_unix.go delve-1.9.0.mod/service/debugger/debugger_unix.go
--- delve-1.9.0/service/debugger/debugger_unix.go	2022-07-07 01:18:27.000000000 +0800
+++ delve-1.9.0.mod/service/debugger/debugger_unix.go	2022-07-14 16:58:54.736488762 +0800
@@ -32,7 +32,7 @@
 	switch runtime.GOOS {
 	case "darwin":
 		exe, err = macho.NewFile(f)
-	case "linux", "freebsd":
+	case "linux", "freebsd", "android":
 		exe, err = elf.NewFile(f)
 	default:
 		panic("attempting to open file Delve cannot parse")
