--- ./pkg/proc/gdbserial/gdbserver.go	2020-12-09 10:38:21.000000000 +0000
+++ ./pkg/proc/gdbserial/gdbserver.go_patch	2020-12-16 08:22:33.439803407 +0000
@@ -1400,7 +1400,7 @@
 	case "windows", "darwin", "freebsd":
 		// mov rcx, QWORD PTR gs:{uint32(off)}
 		op = []byte{0x65, 0x48, 0x8b, 0x0c, 0x25}
-	case "linux":
+	case "linux", "android":
 		// mov rcx,QWORD PTR fs:{uint32(off)}
 		op = []byte{0x64, 0x48, 0x8B, 0x0C, 0x25}
 	default:
