diff -uNr delve-1.9.0/pkg/proc/gdbserial/gdbserver_conn.go delve-1.9.0.mod/pkg/proc/gdbserial/gdbserver_conn.go
--- delve-1.9.0/pkg/proc/gdbserial/gdbserver_conn.go	2022-07-07 01:18:27.000000000 +0800
+++ delve-1.9.0.mod/pkg/proc/gdbserial/gdbserver_conn.go	2022-07-14 17:05:29.976488611 +0800
@@ -619,7 +619,7 @@
 
 	var _SIGBUS uint8
 	switch conn.goos {
-	case "linux":
+	case "linux", "android":
 		_SIGBUS = 0x7
 	case "darwin":
 		_SIGBUS = 0xa
