diff -uNr libvgm/audio/AudDrv_OSS.c libvgm.mod/audio/AudDrv_OSS.c
--- libvgm/audio/AudDrv_OSS.c	2022-12-16 17:19:46.378761717 +0800
+++ libvgm.mod/audio/AudDrv_OSS.c	2022-12-16 17:21:12.758761684 +0800
@@ -7,7 +7,7 @@
 
 #include <fcntl.h>	// for open() constants
 #include <unistd.h>	// for I/O calls
-#include <sys/soundcard.h>	// OSS sound stuff
+#include <linux/soundcard.h>	// OSS sound stuff
 #include <sys/ioctl.h>
 
 #ifdef ENABLE_OSS_THREAD
