diff -uNr bzip2-1.0.7/Makefile-libbz2_so bzip2-1.0.7.mod/Makefile-libbz2_so
--- bzip2-1.0.7/Makefile-libbz2_so	2019-06-27 21:15:39.000000000 +0300
+++ bzip2-1.0.7.mod/Makefile-libbz2_so	2019-07-07 14:27:21.338209706 +0300
@@ -22,9 +22,7 @@
 
 
 SHELL=/bin/sh
-CC=gcc
 BIGFILES=-D_FILE_OFFSET_BITS=64
-CFLAGS=-fpic -fPIC -Wall -Winline -O2 -g $(BIGFILES)
 
 OBJS= blocksort.o  \
       huffman.o    \
@@ -35,8 +33,8 @@
       bzlib.o
 
 all: $(OBJS)
-	$(CC) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.7 $(OBJS)
-	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.7
+	$(CC) $(LDFLAGS) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.7 $(OBJS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.7
 	rm -f libbz2.so.1.0
 	ln -s libbz2.so.1.0.7 libbz2.so.1.0
 
