diff -u -r ../bzip2-1.0.6/Makefile-libbz2_so ./Makefile-libbz2_so
--- ../bzip2-1.0.6/Makefile-libbz2_so	2010-09-11 01:07:52.000000000 +0200
+++ ./Makefile-libbz2_so	2014-02-24 22:00:02.000000000 +0100
@@ -22,9 +22,7 @@
 
 
 SHELL=/bin/sh
-CC=gcc
 BIGFILES=-D_FILE_OFFSET_BITS=64
-CFLAGS=-fpic -fPIC -Wall -Winline -O2 -g $(BIGFILES)
 
 OBJS= blocksort.o  \
       huffman.o    \
@@ -35,8 +33,8 @@
       bzlib.o
 
 all: $(OBJS)
-	$(CC) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.6 $(OBJS)
-	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.6
+	$(CC) $(LDFLAGS) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.6 $(OBJS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.6
 	rm -f libbz2.so.1.0
 	ln -s libbz2.so.1.0.6 libbz2.so.1.0
 
