--- a/liblxqt-config-cursor/crtheme.cpp
+++ b/liblxqt-config-cursor/crtheme.cpp
@@ -309,7 +309,7 @@
         QString name, inherits;
         QString path = QDir::home().absolutePath() + QStringLiteral("/.icons/default/index.theme");
         if(! QFile::exists(path))
-            path = QStringLiteral("/usr/share/icons/default/index.theme");
+            path = QStringLiteral("@TERMUX_PREFIX@/share/icons/default/index.theme");
         if(! QFile::exists(path))
             return QString();
         QSettings cursorTheme(path, QSettings::IniFormat);
