--- a/lxqt-config-input/keyboardlayoutconfig.h
+++ b/lxqt-config-input/keyboardlayoutconfig.h
@@ -23,13 +23,13 @@
 
 #include <QtCore/QtGlobal>
 #ifdef Q_OS_LINUX
-#define XKBD_BASELIST_PATH "/usr/share/X11/xkb/rules/base.lst"
+#define XKBD_BASELIST_PATH "@TERMUX_PREFIX@/share/X11/xkb/rules/base.lst"
 #elif defined(Q_OS_FREEBSD)
-#define XKBD_BASELIST_PATH "/usr/local/share/X11/xkb/rules/base.lst"
+#define XKBD_BASELIST_PATH "@TERMUX_PREFIX@/local/share/X11/xkb/rules/base.lst"
 #elif defined(Q_OS_OPENBSD)
-#define XKBD_BASELIST_PATH "/usr/X11R6/share/X11/xkb/rules/base.lst"
+#define XKBD_BASELIST_PATH "@TERMUX_PREFIX@/X11R6/share/X11/xkb/rules/base.lst"
 #else
-#define XKBD_BASELIST_PATH "/usr/local/share/X11/xkb/rules/base.lst"
+#define XKBD_BASELIST_PATH "@TERMUX_PREFIX@/local/share/X11/xkb/rules/base.lst"
 #endif
 
 #include <QWidget>
