diff -uNr fd-8.1.1/Cargo.toml fd-8.1.1.mod/Cargo.toml
--- fd-8.1.1/Cargo.toml	2020-05-25 16:38:30.000000000 +0300
+++ fd-8.1.1.mod/Cargo.toml	2020-05-25 22:43:17.350694509 +0300
@@ -61,8 +61,8 @@
 # FIXME: Re-enable jemalloc on macOS
 # jemalloc is currently disabled on macOS due to a bug in jemalloc in combination with macOS
 # Catalina. See https://github.com/sharkdp/fd/issues/498 for details.
-[target.'cfg(all(not(windows), not(target_os = "macos"), not(target_env = "musl")))'.dependencies]
-jemallocator = "0.3.0"
+#[target.'cfg(all(not(windows), not(target_os = "macos"), not(target_env = "musl")))'.dependencies]
+#jemallocator = "0.3.0"
 
 [dev-dependencies]
 diff = "0.1"
