diff -u -r ../fd-7.4.0/Cargo.toml ./Cargo.toml
--- ../fd-7.4.0/Cargo.toml	2019-09-15 17:29:15.000000000 +0000
+++ ./Cargo.toml	2019-09-17 19:37:35.384516258 +0000
@@ -52,9 +52,6 @@
 [target.'cfg(all(unix, not(target_os = "redox")))'.dependencies]
 libc = "0.2"
 
-[target.'cfg(all(not(windows), not(target_env = "musl")))'.dependencies]
-jemallocator = "0.3.0"
-
 [dev-dependencies]
 diff = "0.1"
 tempdir = "0.3"
