diff -uNr fd-8.1.1/src/main.rs fd-8.1.1.mod/src/main.rs
--- fd-8.1.1/src/main.rs	2020-05-25 16:38:30.000000000 +0300
+++ fd-8.1.1.mod/src/main.rs	2020-05-26 00:15:13.833867073 +0300
@@ -34,7 +34,7 @@
 
 // We use jemalloc for performance reasons, see https://github.com/sharkdp/fd/pull/481
 // FIXME: re-enable jemalloc on macOS, see comment in Cargo.toml file for more infos
-#[cfg(all(not(windows), not(target_os = "macos"), not(target_env = "musl")))]
+#[cfg(all(not(windows), not(target_os = "macos"), not(target_env = "musl"), not(target_os = "android")))]
 #[global_allocator]
 static ALLOC: jemallocator::Jemalloc = jemallocator::Jemalloc;
 
