diff -u -r ../fd-8.0.0/src/main.rs ./src/main.rs
--- ../fd-8.0.0/src/main.rs	2020-04-16 08:11:05.000000000 +0000
+++ ./src/main.rs	2020-04-18 22:36:31.485685000 +0000
@@ -30,7 +30,7 @@
 use crate::regex_helper::pattern_has_uppercase_char;
 
 // We use jemalloc for performance reasons, see https://github.com/sharkdp/fd/pull/481
-#[cfg(all(not(windows), not(target_env = "musl")))]
+#[cfg(all(not(windows), not(target_env = "musl"), not(target_os = "android")))]
 #[global_allocator]
 static ALLOC: jemallocator::Jemalloc = jemallocator::Jemalloc;
 
