diff -u -r ../fd-7.4.0/src/main.rs ./src/main.rs
--- ../fd-7.4.0/src/main.rs	2019-09-15 17:29:15.000000000 +0000
+++ ./src/main.rs	2019-09-17 19:37:47.620364849 +0000
@@ -35,11 +35,6 @@
     pattern_has_uppercase_char, transform_args_with_exec, FileTypes,
 };
 
-// We use jemalloc for performance reasons, see https://github.com/sharkdp/fd/pull/481
-#[cfg(all(not(windows), not(target_env = "musl")))]
-#[global_allocator]
-static ALLOC: jemallocator::Jemalloc = jemallocator::Jemalloc;
-
 fn main() {
     let checked_args = transform_args_with_exec(env::args_os());
     let matches = app::build_app().get_matches_from(checked_args);
