--- a/src/lib/ares_android.c
+++ b/src/lib/ares_android.c
@@ -12,7 +12,7 @@
  * this software for any purpose.  It is provided "as is"
  * without express or implied warranty.
  */
-#if defined(ANDROID) || defined(__ANDROID__)
+#if 0
 
 #include <jni.h>
 
--- a/src/lib/ares_init.c
+++ b/src/lib/ares_init.c
@@ -47,7 +47,9 @@
 
 #if defined(ANDROID) || defined(__ANDROID__)
 #  include <sys/system_properties.h>
+#if 0
 #  include "ares_android.h"
+#endif
 /* From the Bionic sources */
 #  define DNS_PROP_NAME_PREFIX "net.dns"
 #  define MAX_DNS_PROPERTIES   8
@@ -1321,6 +1323,7 @@
   char  *domains;
   size_t num_servers;
 
+#if 0
   /* Use the Android connectivity manager to get a list
    * of DNS servers. As of Android 8 (Oreo) net.dns#
    * system properties are no longer available. Google claims this
@@ -1346,6 +1349,7 @@
     set_search(channel, domains);
     ares_free(domains);
   }
+#endif
 
 #  ifdef HAVE___SYSTEM_PROPERTY_GET
   /* Old way using the system property still in place as
--- a/src/lib/ares_library_init.c
+++ b/src/lib/ares_library_init.c
@@ -32,7 +32,7 @@
 
 /* library-private global and unique instance vars */
 
-#if defined(ANDROID) || defined(__ANDROID__)
+#if 0
 #  include "ares_android.h"
 #endif
 
@@ -141,7 +141,7 @@
 
   /* NOTE: ARES_LIB_INIT_WIN32 flag no longer used */
 
-#if defined(ANDROID) || defined(__ANDROID__)
+#if 0
   ares_library_cleanup_android();
 #endif
 
