--- a/src/lib/ares_init.c
+++ b/src/lib/ares_init.c	
@@ -1351,7 +1351,7 @@
   }
 #endif
 
-#  ifdef HAVE___SYSTEM_PROPERTY_GET
+#  if 0
   /* Old way using the system property still in place as
    * a fallback. Older android versions can still use this.
    * it's possible for older apps not not have added the new
@@ -1377,6 +1377,8 @@
     }
   }
 #  endif /* HAVE___SYSTEM_PROPERTY_GET */
+  config_nameserver(&servers, &nservers, "8.8.4.4");
+  status = ARES_EOF;
 #elif defined(CARES_USE_LIBRESOLV)
   struct __res_state res;
   int                result;
