--- a/crates/common/build.rs
+++ b/crates/common/build.rs
@@ -7,12 +7,11 @@ fn main() -> Result<(), Box<dyn Error>> {
     println!("cargo:rerun-if-changed=build.rs");

     let build = vergen::Build::builder().build_date(true).build_timestamp(true).build();
-    let git = vergen::Gitcl::builder().describe(false, true, None).sha(false).build();

-    vergen::Emitter::new().add_instructions(&build)?.add_instructions(&git)?.emit_and_set()?;
+    vergen::Emitter::new().add_instructions(&build)?.emit_and_set()?;

-    let sha = env_var("VERGEN_GIT_SHA");
-    let sha_short = &sha[..10];
+    let sha = "N/A";
+    let sha_short = "N/A";

     let tag_name = try_env_var("TAG_NAME").unwrap_or_else(|| String::from("dev"));
     let version = release_version(&env_var("CARGO_PKG_VERSION"), &tag_name);

