diff --git a/crates/svm-builds/build.rs b/crates/svm-builds/build.rs
index 1e4b255..c8642b8 100644
--- a/build.rs
+++ b/build.rs
@@ -16,6 +16,9 @@ use svm::Releases;
 /// - "macosx-aarch64"
 /// - "windows-amd64"
 /// - "android-aarch64"
+/// - "android-arm"
+/// - "android-x86"
+/// - "android-x86_64"
 pub const SVM_TARGET_PLATFORM: &str = "SVM_TARGET_PLATFORM";
 
 /// The path to the releases JSON file, that was pre-fetched manually.
