diff -uNr fftw-3.3.6-pl1/api/import-system-wisdom.c fftw-3.3.6-pl1.mod/api/import-system-wisdom.c
--- fftw-3.3.6-pl1/api/import-system-wisdom.c	2017-01-15 14:03:24.000000000 +0200
+++ fftw-3.3.6-pl1.mod/api/import-system-wisdom.c	2017-09-12 11:20:08.432992283 +0300
@@ -32,7 +32,7 @@
 #if defined(__DJGPP__)
 #  define WISDOM_DIR "/dev/env/DJDIR/etc/fftw/"
 #else
-#  define WISDOM_DIR "/etc/fftw/"
+#  define WISDOM_DIR "@TERMUX_PREFIX@/etc/fftw/"
 #endif
 
 int X(import_system_wisdom)(void)
