diff -u -r ../gnuchess-6.2.1/src/main.cc ./src/main.cc
--- ../gnuchess-6.2.1/src/main.cc	2015-01-04 05:10:41.000000000 -0500
+++ ./src/main.cc	2015-01-05 20:40:16.816570715 -0500
@@ -237,8 +237,8 @@
   char opt_addbook[MAXSTR+1] = "";
   char *endptr;
 
-  /*disable graphic output by default */
-  graphicmodeoutput = 0;
+  /*enable graphic output by default */
+  graphicmodeoutput = 1;
 
 
   progname = argv[0]; /* Save in global for cmd_usage */
