diff --git a/rhubarb/lib/whereami/src/whereami.c b/rhubarb/lib/whereami/src/whereami.c
index 19d3781..1e8ff6d 100644
--- a/rhubarb/lib/whereami/src/whereami.c
+++ b/rhubarb/lib/whereami/src/whereami.c
@@ -167,6 +167,7 @@ int WAI_PREFIX(getModulePath)(char* out, int capacity, int* dirname_length)
 #define __STDC_FORMAT_MACROS
 #endif
 #include <inttypes.h>
+#include <unistd.h>
 
 #if !defined(WAI_PROC_SELF_EXE)
 #define WAI_PROC_SELF_EXE "/proc/self/exe"
