diff -u -r ../gnupg-2.1.6/agent/agent.h ./agent/agent.h
--- ../gnupg-2.1.6/agent/agent.h	2015-06-30 16:26:08.000000000 -0400
+++ ./agent/agent.h	2015-07-06 08:39:31.806515184 -0400
@@ -310,7 +310,7 @@
   agent_Lunderscore (ctrl_t ctrl, const char *string)           \
   {                                                             \
     return ctrl? i18n_localegettext (ctrl->lc_messages, string) \
-      /*     */: gettext (string);                              \
+      /*     */: string;                              \
   }
 
 
