diff -u -r ../gnupg-2.1.14/configure ./configure
--- ../gnupg-2.1.14/configure	2016-07-14 10:03:06.000000000 -0400
+++ ./configure	2016-07-14 19:20:04.539115863 -0400
@@ -14729,13 +14729,6 @@
 done
 
 
-if test "$have_android_system" = yes; then
-   # On Android ttyname is a stub but prints an error message.
-
-$as_echo "#define HAVE_BROKEN_TTYNAME 1" >>confdefs.h
-
-fi
-
 ac_fn_c_check_type "$LINENO" "struct sigaction" "ac_cv_type_struct_sigaction" "#include <signal.h>
 "
 if test "x$ac_cv_type_struct_sigaction" = xyes; then :
