--- gnupg-1.4.20/mpi/Makefile.in	2015-12-20 08:53:15.000000000 +0100
+++ src/mpi/Makefile.in	2016-04-19 19:33:55.167315560 +0200
@@ -713,7 +713,7 @@
 # cancel the default rules used by libtool which do not really
 # work and add one to cpp .S files
 .S.o:
-	 $(CPP) $(INCLUDES) $(DEFS) $< | grep -v '^#' > _$*.s
+	 $(CPP) $(INCLUDES) $(DEFS) -DPIC $< | grep -v '^#' > _$*.s
 	 $(COMPILE) $(AM_CCASFLAGS) -c _$*.s
 	 mv -f _$*.o $*.o
 
