diff -uNr slang-2.3.2/slsh/lib/sldbsock.sl slang-2.3.2.mod/slsh/lib/sldbsock.sl
--- slang-2.3.2/slsh/lib/sldbsock.sl	2018-03-05 01:16:36.000000000 +0200
+++ slang-2.3.2.mod/slsh/lib/sldbsock.sl	2020-01-01 20:12:25.652335085 +0200
@@ -23,7 +23,7 @@
 
 % Simple line-based protocol:
 
-private variable SLDB_SOCKET_PREFIX	= "/tmp/.sldb_";
+private variable SLDB_SOCKET_PREFIX	= "@TERMUX_PREFIX@/tmp/.sldb_";
 
 private variable OK_CMD_RECEIVED	= 200;
 private variable OK_CONT_XFER		= 201;
