diff -uNr slang-2.3.2/src/sltermin.c slang-2.3.2.mod/src/sltermin.c
--- slang-2.3.2/src/sltermin.c	2018-03-05 01:16:36.000000000 +0200
+++ slang-2.3.2.mod/src/sltermin.c	2020-01-01 20:18:03.859706757 +0200
@@ -276,14 +276,9 @@
 #ifdef MISC_TERMINFO_DIRS
    MISC_TERMINFO_DIRS,
 #endif
-   "/usr/local/etc/terminfo",
-   "/usr/local/share/terminfo",
-   "/usr/local/lib/terminfo",
-   "/etc/terminfo",
-   "/usr/share/terminfo",
-   "/usr/lib/terminfo",
-   "/usr/share/lib/terminfo",
-   "/lib/terminfo",
+   "@TERMUX_PREFIX@/etc/terminfo",
+   "@TERMUX_PREFIX@/lib/terminfo",
+   "@TERMUX_PREFIX@/share/terminfo",
    NULL,
 };
 
