diff -u -r ../php-5.6.15/ext/posix/posix.c ./ext/posix/posix.c
--- ../php-5.6.15/ext/posix/posix.c	2015-10-29 05:55:01.000000000 -0400
+++ ./ext/posix/posix.c	2015-11-10 15:38:41.542443608 -0500
@@ -1098,7 +1098,9 @@
 	add_assoc_string(return_value, "passwd",    pw->pw_passwd, 1);
 	add_assoc_long  (return_value, "uid",       pw->pw_uid);
 	add_assoc_long  (return_value, "gid",		pw->pw_gid);
+#ifdef __LP64__
 	add_assoc_string(return_value, "gecos",     pw->pw_gecos, 1);
+#endif
 	add_assoc_string(return_value, "dir",       pw->pw_dir, 1);
 	add_assoc_string(return_value, "shell",     pw->pw_shell, 1);
 	return 1;
