diff -u -r ../php-7.3.1/ext/opcache/config.m4 ./ext/opcache/config.m4
--- ../php-7.3.1/ext/opcache/config.m4	2019-01-08 13:55:52.000000000 +0000
+++ ./ext/opcache/config.m4	2019-01-10 23:53:48.558906864 +0000
@@ -145,7 +145,9 @@
 }
 ]])],[dnl
     AC_DEFINE(HAVE_SHM_MMAP_ANON, 1, [Define if you have mmap(MAP_ANON) SHM support])
-    msg=yes],[msg=no],[msg=no])
+    msg=yes],[msg=no],
+    AC_DEFINE(HAVE_SHM_MMAP_ANON, 1, [Define if you have mmap(MAP_ANON) SHM support])
+    msg=yes)
   AC_MSG_RESULT([$msg])
 
   AC_MSG_CHECKING(for mmap() using /dev/zero shared memory support)
@@ -356,7 +358,12 @@
 	flock_type=linux
     AC_DEFINE([HAVE_FLOCK_LINUX], [], [Struct flock is Linux-type])
     AC_MSG_RESULT("yes")
-], [AC_MSG_RESULT("no")], [AC_MSG_RESULT([no])])
+], [AC_MSG_RESULT("no")], [
+    dnl cross-compiling; assume Linux
+    flock_type=linux
+    AC_DEFINE([HAVE_FLOCK_LINUX], [], [Struct flock is Linux-type])
+    AC_MSG_RESULT("yes")
+])
 
 AC_MSG_CHECKING("whether flock struct is BSD ordered")
 AC_RUN_IFELSE([AC_LANG_SOURCE([[
@@ -416,6 +423,9 @@
 	Optimizer/zend_dump.c,
 	shared,,-DZEND_ENABLE_STATIC_TSRMLS_CACHE=1,,yes)
 
+  OPCACHE_SHARED_LIBADD=-lpcre
+  PHP_SUBST(OPCACHE_SHARED_LIBADD)
+
   PHP_ADD_BUILD_DIR([$ext_builddir/Optimizer], 1)
   PHP_ADD_EXTENSION_DEP(opcache, pcre)
 fi
