--- ./ext/opcache/jit/Makefile.frag.orig	2024-11-27 23:45:28.237709258 +0700
+++ ./ext/opcache/jit/Makefile.frag	2024-11-27 23:47:10.000000000 +0700
@@ -2,7 +2,7 @@
 	$(BUILD_CC) $(srcdir)/jit/ir/dynasm/minilua.c -lm -o $@
 
 $(builddir)/jit/ir/ir_emit_$(DASM_ARCH).h: $(srcdir)/jit/ir/ir_$(DASM_ARCH).dasc $(srcdir)/jit/ir/dynasm/*.lua $(builddir)/jit/ir/minilua
-	$(builddir)/jit/ir/minilua $(srcdir)/jit/ir/dynasm/dynasm.lua  $(DASM_FLAGS) -o $@ $(srcdir)/jit/ir/ir_$(DASM_ARCH).dasc
+	$(NATIVE_MINILUA_EXECUTABLE) $(srcdir)/jit/ir/dynasm/dynasm.lua  $(DASM_FLAGS) -o $@ $(srcdir)/jit/ir/ir_$(DASM_ARCH).dasc
 
 $(builddir)/jit/ir/ir_emit.lo: \
 	$(srcdir)/jit/ir/ir_emit.c $(builddir)/jit/ir/ir_emit_$(DASM_ARCH).h
