--- ./ext/mysqli/config.m4	2017-10-25 12:34:36.000000000 +0530
+++ ./ext/mysqli/config.m4	2017-11-07 21:59:15.529313719 +0530
@@ -120,6 +120,7 @@
                   mysqli_fe.c mysqli_report.c mysqli_driver.c mysqli_warning.c \
                   mysqli_exception.c mysqli_result_iterator.c $mysqli_extra_sources"
   PHP_NEW_EXTENSION(mysqli, $mysqli_sources, $ext_shared,, -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1)
+  PHP_ADD_LIBRARY(mariadb,, MYSQLI_SHARED_LIBADD)
   PHP_SUBST(MYSQLI_SHARED_LIBADD)
   PHP_INSTALL_HEADERS([ext/mysqli/php_mysqli_structs.h])
 
