diff -u -r ../php-7.1.0RC3/ext/phar/Makefile.frag ./ext/phar/Makefile.frag
--- ../php-7.1.0RC3/ext/phar/Makefile.frag	2016-09-28 22:15:43.000000000 -0400
+++ ./ext/phar/Makefile.frag	2016-10-03 05:47:53.106218451 -0400
@@ -4,20 +4,7 @@
 pharcmd: $(builddir)/phar.php $(builddir)/phar.phar
 
 PHP_PHARCMD_SETTINGS = -n -d 'open_basedir=' -d 'output_buffering=0' -d 'memory_limit=-1' -d phar.readonly=0 -d 'safe_mode=0'
-PHP_PHARCMD_EXECUTABLE = ` \
-	if test -x "$(top_builddir)/$(SAPI_CLI_PATH)"; then \
-		$(top_srcdir)/build/shtool echo -n -- "$(top_builddir)/$(SAPI_CLI_PATH) -n"; \
-		if test "x$(PHP_MODULES)" != "x"; then \
-		$(top_srcdir)/build/shtool echo -n -- " -d extension_dir=$(top_builddir)/modules"; \
-		for i in bz2 zlib phar; do \
-			if test -f "$(top_builddir)/modules/$$i.la"; then \
-				. $(top_builddir)/modules/$$i.la; $(top_srcdir)/build/shtool echo -n -- " -d extension=$$dlname"; \
-			fi; \
-		done; \
-		fi; \
-	else \
-		$(top_srcdir)/build/shtool echo -n -- "$(PHP_EXECUTABLE)"; \
-	fi;`
+PHP_PHARCMD_EXECUTABLE = $(NATIVE_PHP_EXECUTABLE)
 PHP_PHARCMD_BANG = `$(top_srcdir)/build/shtool echo -n -- "$(INSTALL_ROOT)$(bindir)/$(program_prefix)php$(program_suffix)$(EXEEXT)";`
 
 $(builddir)/phar/phar.inc: $(srcdir)/phar/phar.inc
