diff -u -r ../php-5.6.15/ext/standard/basic_functions.c ./ext/standard/basic_functions.c
--- ../php-5.6.15/ext/standard/basic_functions.c	2015-10-29 05:55:01.000000000 -0400
+++ ./ext/standard/basic_functions.c	2015-11-10 16:01:20.858601990 -0500
@@ -1834,6 +1834,7 @@
 ZEND_END_ARG_INFO()
 /* }}} */
 /* {{{ password.c */
+#if HAVE_CRYPT
 ZEND_BEGIN_ARG_INFO_EX(arginfo_password_hash, 0, 0, 2)
 	ZEND_ARG_INFO(0, password)
 	ZEND_ARG_INFO(0, algo)
@@ -1851,6 +1852,7 @@
 	ZEND_ARG_INFO(0, password)
 	ZEND_ARG_INFO(0, hash)
 ZEND_END_ARG_INFO()
+#endif
 /* }}} */
 /* {{{ proc_open.c */
 #ifdef PHP_CAN_SUPPORT_PROC_OPEN
@@ -2860,12 +2862,14 @@
 	PHP_FE(base64_decode,													arginfo_base64_decode)
 	PHP_FE(base64_encode,													arginfo_base64_encode)
 
+#if HAVE_CRYPT
 	PHP_FE(password_hash,													arginfo_password_hash)
 	PHP_FE(password_get_info,												arginfo_password_get_info)
 	PHP_FE(password_needs_rehash,											arginfo_password_needs_rehash)
 	PHP_FE(password_verify,													arginfo_password_verify)
 	PHP_FE(convert_uuencode,												arginfo_convert_uuencode)
 	PHP_FE(convert_uudecode,												arginfo_convert_uudecode)
+#endif
 
 	PHP_FE(abs,																arginfo_abs)
 	PHP_FE(ceil,															arginfo_ceil)
@@ -3620,7 +3624,9 @@
 	BASIC_MINIT_SUBMODULE(browscap)
 	BASIC_MINIT_SUBMODULE(standard_filters)
 	BASIC_MINIT_SUBMODULE(user_filters)
+#if HAVE_CRYPT
 	BASIC_MINIT_SUBMODULE(password)
+#endif
 
 #if defined(HAVE_LOCALECONV) && defined(ZTS)
 	BASIC_MINIT_SUBMODULE(localeconv)
