--- sapi/cgi/cgi_main.c.orig	2020-01-19 21:52:00.467616255 +0000
+++ ./sapi/cgi/cgi_main.c	2020-01-19 21:52:20.367984090 +0000
@@ -84,7 +84,7 @@
 
 #include "php_getopt.h"
 
-#include "fastcgi.h"
+#include "../../main/fastcgi.h"
 
 #if defined(PHP_WIN32) && defined(HAVE_OPENSSL)
 # include "openssl/applink.c"
--- ./sapi/fpm/fpm/fpm_log.c.orig	2020-01-19 21:52:34.868252066 +0000
+++ ./sapi/fpm/fpm/fpm_log.c	2020-01-19 21:53:28.639245436 +0000
@@ -16,7 +16,7 @@
 #include "fpm_process_ctl.h"
 #include "fpm_signals.h"
 #include "fpm_scoreboard.h"
-#include "fastcgi.h"
+#include "../../../main/fastcgi.h"
 #include "zlog.h"
 
 #ifdef MAX_LINE_LENGTH
--- ./sapi/fpm/fpm/fpm_main.c.orig	2020-01-19 21:54:35.180473976 +0000
+++ ./sapi/fpm/fpm/fpm_main.c	2020-01-19 21:54:47.060693226 +0000
@@ -80,7 +80,7 @@
 
 #include "http_status_codes.h"
 
-#include "fastcgi.h"
+#include "../../../main/fastcgi.h"
 
 #include <php_config.h>
 #include "fpm.h"
