diff -uNr php-8.0.0.orig/sapi/cgi/cgi_main.c php-8.0.0/sapi/cgi/cgi_main.c
--- php-8.0.0.orig/sapi/cgi/cgi_main.c	2020-11-24 22:34:03.000000000 +0530
+++ php-8.0.0/sapi/cgi/cgi_main.c	2020-12-06 21:46:09.648909912 +0530
@@ -83,7 +83,7 @@
 
 #include "php_getopt.h"
 
-#include "fastcgi.h"
+#include "../../main/fastcgi.h"
 #include "cgi_main_arginfo.h"
 
 #if defined(PHP_WIN32) && defined(HAVE_OPENSSL)
diff -uNr php-8.0.0.orig/sapi/fpm/fpm/fpm_log.c php-8.0.0/sapi/fpm/fpm/fpm_log.c
--- php-8.0.0.orig/sapi/fpm/fpm/fpm_log.c	2020-11-24 22:34:03.000000000 +0530
+++ php-8.0.0/sapi/fpm/fpm/fpm_log.c	2020-12-06 21:49:08.433333740 +0530
@@ -16,7 +16,7 @@
 #include "fpm_process_ctl.h"
 #include "fpm_signals.h"
 #include "fpm_scoreboard.h"
-#include "fastcgi.h"
+#include "../../../main/fastcgi.h"
 #include "zlog.h"
 
 #ifdef MAX_LINE_LENGTH
diff -uNr php-8.0.0.orig/sapi/fpm/fpm/fpm_main.c php-8.0.0/sapi/fpm/fpm/fpm_main.c
--- php-8.0.0.orig/sapi/fpm/fpm/fpm_main.c	2020-11-24 22:34:03.000000000 +0530
+++ php-8.0.0/sapi/fpm/fpm/fpm_main.c	2020-12-06 21:47:59.656234130 +0530
@@ -78,7 +78,7 @@
 
 #include "http_status_codes.h"
 
-#include "fastcgi.h"
+#include "../../../main/fastcgi.h"
 
 #include <php_config.h>
 #include "fpm.h"
