diff -uNr OpenTTD-1.9.1/config.lib OpenTTD-1.9.1.mod/config.lib
--- OpenTTD-1.9.1/config.lib	2019-04-08 22:37:59.000000000 +0300
+++ OpenTTD-1.9.1.mod/config.lib	2019-06-30 00:21:04.831588247 +0300
@@ -1616,10 +1616,6 @@
 		fi
 	fi
 
-	if [ "$os" != "CYGWIN" ] && [ "$os" != "HAIKU" ] && [ "$os" != "OPENBSD" ] && [ "$os" != "MINGW" ] && [ "$os" != "MORPHOS" ] && [ "$os" != "OSX" ] && [ "$os" != "DOS" ] && [ "$os" != "OS2" ]; then
-		LIBS="$LIBS -lpthread"
-	fi
-
 	if [ "$os" != "CYGWIN" ] && [ "$os" != "HAIKU" ] && [ "$os" != "MINGW" ] && [ "$os" != "DOS" ]; then
 		LIBS="$LIBS -lc"
 	fi
@@ -2651,7 +2647,7 @@
 	log 2 "detecting $2"
 
 	if [ "$1" = "1" ] || [ "$1" = "" ] || [ "$1" = "2" ]; then
-		eval "$2=`ls -1 /usr/include/$4*.h 2>/dev/null | egrep \"\/$5\$\"`"
+		eval "$2=`ls -1 @TERMUX_PREFIX@/include/$4*.h 2>/dev/null | egrep \"\/$5\$\"`"
 		eval "res=\$$2"
 		if [ -z "$res" ]; then
 			log 2 "  trying /usr/include/$4$5... no"
