--- a/savers/slideshow.c
+++ b/savers/slideshow.c
@@ -119,7 +119,7 @@
 
     if (location == NULL) {
         location = xdg_user_dir_lookup ("PICTURES");
-        if (location == NULL || strcmp (location, "/tmp") == 0 || strcmp (location, g_get_home_dir ()) == 0) {
+        if (location == NULL || strcmp (location, "@TERMUX_PREFIX@/tmp") == 0 || strcmp (location, g_get_home_dir ()) == 0) {
             g_free (location);
             location = g_build_filename (g_get_home_dir (), "Pictures", NULL);
         }
--- a/savers/xdg-user-dir-lookup.c
+++ b/savers/xdg-user-dir-lookup.c
@@ -179,7 +179,7 @@
     home_dir = getenv ("HOME");
 
     if (home_dir == NULL)
-        return g_strdup ("/tmp");
+        return g_strdup ("@TERMUX_PREFIX@/tmp");
 
     /* Special case desktop for historical compatibility */
     if (strcmp (type, "DESKTOP") == 0) {
