diff -u -r ../Python-2.7.10/configure ./configure
--- ../Python-2.7.10/configure	2015-05-23 12:09:25.000000000 -0400
+++ ./configure	2015-10-20 09:54:55.481316335 -0400
@@ -10310,7 +10310,7 @@
 _ACEOF
 if ac_fn_c_try_compile "$LINENO"; then :
 
-$as_echo "#define HAVE_LINK 1" >>confdefs.h
+# $as_echo "#define HAVE_LINK 1" >>confdefs.h
 
    { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
