diff -u -r ../Python-2.7.10/Python/bltinmodule.c ./Python/bltinmodule.c
--- ../Python-2.7.10/Python/bltinmodule.c	2015-05-23 12:09:24.000000000 -0400
+++ ./Python/bltinmodule.c	2015-09-16 21:01:23.750245871 -0400
@@ -22,7 +22,7 @@
 #elif defined(__APPLE__)
 const char *Py_FileSystemDefaultEncoding = "utf-8";
 #else
-const char *Py_FileSystemDefaultEncoding = NULL; /* use default */
+const char *Py_FileSystemDefaultEncoding = "utf-8"; /* use default */
 #endif
 
 /* Forward */
