diff -u -r ../Python-2.7.10/setup.py ./setup.py
--- ../Python-2.7.10/setup.py	2015-05-23 12:09:25.000000000 -0400
+++ ./setup.py	2015-09-16 20:36:19.281639713 -0400
@@ -294,6 +294,9 @@
                           (ext.name, sys.exc_info()[1]))
             self.failed.append(ext.name)
             return
+
+	return # Skip import check which does not work when cross compiling
+
         # Workaround for Mac OS X: The Carbon-based modules cannot be
         # reliably imported into a command-line Python
         if 'Carbon' in ext.extra_link_args:
