--- src/lxqtprogramfinder.cpp	2021-04-10 07:24:13.000000000 +0000
+++ src.mod/lxqtprogramfinder.cpp	2021-05-20 12:09:53.024301000 +0000
@@ -23,7 +23,6 @@
  * END_COMMON_COPYRIGHT_HEADER */
 
 #include "lxqtprogramfinder.h"
-#include <wordexp.h>
 #include <QDir>
 #include <QFileInfo>
 
@@ -62,11 +61,7 @@
     return availPrograms;
 }
 
-LXQT_API QString ProgramFinder::programName(const QString& command)
+LXQT_API QString ProgramFinder::programName(const QString& )
 {
-    wordexp_t we;
-    if (wordexp(command.toLocal8Bit().constData(), &we, WRDE_NOCMD) == 0)
-        if (we.we_wordc > 0)
-            return QString::fromLocal8Bit(we.we_wordv[0]);
     return QString();
 }
