diff --git a/misc/fts.c b/misc/fts.c
index dc5480c50..6836a9939 100644
--- a/misc/fts.c
+++ b/misc/fts.c
@@ -80,7 +80,7 @@ static char sccsid[] = "@(#)fts.c	8.6 (Berkeley) 8/14/94";
 #include <dirent.h>
 #include <errno.h>
 #include "misc/rpmfts.h"
-#   define __set_errno(val) (*__errno_location ()) = (val)
+#   define __set_errno(val) (*__errno ()) = (val)
 #   define __open	open
 #   define __close	close
 #   define __fchdir	fchdir
