diff --git a/lib/fsm.c b/lib/fsm.c
index b2e5fe525..e0db49316 100644
--- a/lib/fsm.c
+++ b/lib/fsm.c
@@ -1014,7 +1014,7 @@ int rpmPackageFilesInstall(rpmts ts, rpmte te, rpmfiles files,
                     rc = RPMERR_UNKNOWN_FILETYPE;
             }
 
-setmeta:
+setmeta:;
 	    /* Special files require path-based ops */
 	    int mayopen = S_ISREG(fp->sb.st_mode) || S_ISDIR(fp->sb.st_mode);
 	    if (!rc && fd == -1 && mayopen) {
