diff -u -r ../racket-6.6/collects/racket/system.rkt ./collects/racket/system.rkt
--- ../racket-6.6/collects/racket/system.rkt	2016-04-15 16:33:56.000000000 -0400
+++ ./collects/racket/system.rkt	2016-11-22 16:45:09.576844179 -0500
@@ -17,7 +17,7 @@
 
 (define (shell-path/args who argstr)
   (case (system-type)
-    [(unix macosx) (append '("/bin/sh" "-c") (list argstr))]
+    [(unix macosx) (append '("@TERMUX_PREFIX@/bin/sh" "-c") (list argstr))]
     [(windows) (let ([cmd
                       (let ([d (find-system-path 'sys-dir)])
                         (let ([cmd (build-path d "cmd.exe")])
