--- ./bin/pil.orig	2022-02-20 20:18:10.391495466 +0530
+++ ./bin/pil	2022-02-20 20:18:26.511495460 +0530
@@ -1,4 +1,4 @@
-#!/usr/bin/picolisp /usr/lib/picolisp/lib.l
+#!/usr/bin/picolisp @TERMUX_PREFIX@/lib/picolisp/lib.l
 (load "@lib/net.l" "@lib/misc.l" "@lib/btree.l" "@lib/db.l" "@lib/pilog.l")
 
 `*Dbg
--- ./bin/vip.orig	2022-02-20 20:17:42.441495477 +0530
+++ ./bin/vip	2022-02-20 20:18:01.041495470 +0530
@@ -1,4 +1,4 @@
-#!/usr/bin/picolisp /usr/lib/picolisp/lib.l
+#!/usr/bin/picolisp @TERMUX_PREFIX@/lib/picolisp/lib.l
 # 19feb22abu
 
 (unless *Dbg
