diff -u -r ../picoLisp/src/Makefile ./src/Makefile
--- ../picoLisp/src/Makefile	2016-02-02 10:52:49.000000000 -0500
+++ ./src/Makefile	2016-04-12 05:55:14.050621923 -0400
@@ -6,119 +6,14 @@
 
 picoFiles = main.c gc.c apply.c flow.c sym.c subr.c big.c io.c net.c tab.c
 
-CC = gcc
 # CCLD is the cc (compiler frontend) to use for the link step.
-CCLD = gcc
-
-CFLAGS = -c -O2 -pipe \
-	-falign-functions=32 -fomit-frame-pointer -fno-strict-aliasing \
-	-W -Wimplicit -Wreturn-type -Wunused -Wformat \
-	-Wuninitialized -Wstrict-prototypes \
-	-D_GNU_SOURCE  -D_FILE_OFFSET_BITS=64
-
-
-ifeq ($(shell uname), Linux)
-	OS = Linux
-	CFLAGS += -m32
-	PICOLISP-FLAGS = -m32 -rdynamic
-	LIB-FLAGS = -lm -ldl
-	DYNAMIC-LIB-FLAGS = -m32 -shared -export-dynamic
-	LCRYPT = -lcrypt
-	STRIP = strip
-else
-ifeq ($(shell uname), OpenBSD)
-	OS = OpenBSD
-	CFLAGS += -m32
-	PICOLISP-FLAGS = -m32 -rdynamic -Wl,-E
-	LIB-FLAGS = -lm
-	DYNAMIC-LIB-FLAGS = -Wl,-E -Wl,-shared
-	LCRYPT = -lcrypto
-	STRIP = strip
-else
-ifeq ($(shell uname), FreeBSD)
-	OS = FreeBSD
-	CC = gcc
-	CCLD = clang
-	CFLAGS += -m32
-	PICOLISP-FLAGS = -m32 -rdynamic
-	LIB-FLAGS = -lm
-	DYNAMIC-LIB-FLAGS = -m32 -shared -export-dynamic
-	LCRYPT = -lcrypt
-	STRIP = strip
-else
-ifeq ($(shell uname), NetBSD)
-	OS = NetBSD
-	CFLAGS += -m32
-	PICOLISP-FLAGS = -m32 -rdynamic
-	LIB-FLAGS = -lm
-	DYNAMIC-LIB-FLAGS = -m32 -shared -export-dynamic
-	LCRYPT = -lcrypto
-	STRIP = strip
-else
-ifeq ($(shell uname), Darwin)
-	OS = Darwin
-	CFLAGS += -m32
-	PICOLISP-FLAGS = -m32
-	LIB-FLAGS = -lc -lm -ldl
-	DYNAMIC-LIB-FLAGS = -m32 -dynamiclib -undefined dynamic_lookup
-	LCRYPT = -lcrypto
-	STRIP = strip -x
-else
-ifeq ($(shell uname), SunOS)
-	OS = SunOS
-	CFLAGS += -m32
-	PICOLISP-FLAGS =
-	LIB-FLAGS = -lm -lnsl -lsocket
-	DYNAMIC-LIB-FLAGS = -G
-	LCRYPT = -lcrypt
-	STRIP = strip
-else
-ifeq ($(shell uname), AIX)
-	OS = AIX
-	CFLAGS += -maix32
-	PICOLISP-FLAGS = -maix32 -Wl,-brtl -Wl,-bexpall
-	LIB-FLAGS = -lm
-	DYNAMIC-LIB-FLAGS = -maix32 -Wl,-G -shared -lm
-	LCRYPT = -lcrypt
-	STRIP = strip
-else
-ifeq ($(shell uname), HP-UX)
-	OS = HP-UX
-	CFLAGS +=
-	PICOLISP-FLAGS = -m32 -rdynamic
-	LIB-FLAGS = -lm -ldl
-	DYNAMIC-LIB-FLAGS = -m32 -shared
-	LCRYPT =
-	STRIP = strip
-else
-ifeq ($(shell uname), IRIX64)
-    OS = IRIX64
-    CFLAGS += -std=gnu99
-    DYNAMIC-LIB-FLAGS = -shared
-    STRIP = strip -f
-else
-ifeq ($(shell uname -o), Cygwin)
-	OS = Cygwin
-	CFLAGS += -m32
-	DYNAMIC-LIB-FLAGS = -shared
-	PICOLISP-FLAGS =
-	DLL-DEFS = $(bin)/picolisp.dll
-	LCRYPT = -lcrypt
-	STRIP = strip
-	exe = .exe
-	dll = .dll
-endif
-endif
-endif
-endif
-endif
-endif
-endif
-endif
-endif
-endif
-
+CCLD = $(CC)
 
+LCRYPT = -lcrypt
+OS = Android
+PICOLISP-FLAGS = -rdynamic -lm $(LDFLAGS)
+DYNAMIC-LIB-FLAGS = -shared -export-dynamic $(LDFLAGS) -lm
+ 
 picolisp: $(bin)/picolisp $(lib)/ext$(dll) $(lib)/ht$(dll)
 tools: $(bin)/lat1 $(bin)/utf2 $(bin)/balance
 gate: $(bin)/ssl $(bin)/httpGate
