diff --git a/src/Makefile.am b/src/Makefile.am
index 8a9b682..303f17f 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -10,7 +10,7 @@ if ENABLE_STRUCTS
 ctypes_la_LIBADD     += libstruct.la
 noinst_LTLIBRARIES   += libstruct.la
 noinst_HEADERS       += struct/dutil.h struct/dwarves.h struct/elf_symtab.h struct/gobuffer.h struct/hash.h struct/list.h struct/rbtree.h struct/strings.h
-libstruct_la_SOURCES  = struct/dutil.c struct/dwarves.c struct/gobuffer.c struct/struct.c struct/strings.c struct/dwarf_loader.c struct/dwarves_fprintf.c struct/elf_symtab.c struct/rbtree.c
+libstruct_la_SOURCES  = ../lib/obstack.c struct/dutil.c struct/dwarves.c struct/gobuffer.c struct/struct.c struct/strings.c struct/dwarf_loader.c struct/dwarves_fprintf.c struct/elf_symtab.c struct/rbtree.c
 libstruct_la_CFLAGS   = -std=gnu99 -D_GNU_SOURCE
 libstruct_la_CPPFLAGS = -I../include -I../lib
 endif
