--- ./src/Makefile.am.orig	2020-07-20 19:30:45.535909736 +0000
+++ ./src/Makefile.am	2020-07-20 19:32:50.544300161 +0000
@@ -10,7 +10,7 @@
 ctypes_la_LIBADD     += libstruct.la
 noinst_LTLIBRARIES   += libstruct.la
 noinst_HEADERS       += struct/dutil.h struct/dwarves.h struct/elf_symtab.h struct/gobuffer.h struct/hash.h struct/list.h struct/rbtree.h struct/strings.h
-libstruct_la_SOURCES  = struct/dutil.c struct/dwarves.c struct/gobuffer.c struct/struct.c struct/strings.c struct/dwarf_loader.c struct/dwarves_fprintf.c struct/elf_symtab.c struct/rbtree.c
+libstruct_la_SOURCES  = ../lib/obstack.c struct/dutil.c struct/dwarves.c struct/gobuffer.c struct/struct.c struct/strings.c struct/dwarf_loader.c struct/dwarves_fprintf.c struct/elf_symtab.c struct/rbtree.c
 libstruct_la_CFLAGS   = -std=gnu99 -D_GNU_SOURCE $(FFI_CFLAGS)
 libstruct_la_CPPFLAGS = -I../include -I../lib
 endif
