--- a/libgnome-desktop/meson.build
+++ b/libgnome-desktop/meson.build
@@ -108,17 +108,6 @@
   link_with: libgnome_desktop
 )
 
-gnome.generate_gir(libgnome_desktop,
-  sources: introspection_sources + libgnome_desktop_headers,
-  export_packages: 'gnome-desktop-3.0',
-  namespace: 'GnomeDesktop',
-  nsversion: '3.0',
-  includes: ['GObject-2.0', 'Gtk-3.0', 'GDesktopEnums-3.0'],
-  extra_args: '-DGNOME_DESKTOP_USE_UNSTABLE_API',
-  identifier_prefix: 'Gnome',
-  symbol_prefix: 'gnome',
-  install: true
-)
 
 desktop_pc = pkg.generate(
   libraries: libgnome_desktop,
