--- a/third_party/electron_node/deps/cares/config/linux/ares_config.h
+++ b/third_party/electron_node/deps/cares/config/linux/ares_config.h
@@ -50,10 +50,10 @@
 #define HAVE_ARPA_INET_H 1
 
 /* Define to 1 if you have the <arpa/nameser_compat.h> header file. */
-#define HAVE_ARPA_NAMESER_COMPAT_H 1
+// #define HAVE_ARPA_NAMESER_COMPAT_H 1
 
 /* Define to 1 if you have the <arpa/nameser.h> header file. */
-#define HAVE_ARPA_NAMESER_H 1
+// #define HAVE_ARPA_NAMESER_H 1
 
 /* Define to 1 if you have the <assert.h> header file. */
 #define HAVE_ASSERT_H 1
@@ -116,10 +116,10 @@
 #define HAVE_GETNAMEINFO 1

 /* Define to 1 if you have `getrandom` */
-#define HAVE_GETRANDOM 1
+// #define HAVE_GETRANDOM 1

 /* Define to 1 if you have `getservbyport_r` */
-#define HAVE_GETSERVBYPORT_R 1
+// #define HAVE_GETSERVBYPORT_R 1

 /* Define to 1 if you have `gettimeofday` */
 #define HAVE_GETTIMEOFDAY 1
@@ -146,7 +146,7 @@
 #define HAVE_INTTYPES_H 1
 
 /* Define to 1 if you have `ioctl` */
-#define HAVE_IOCTL 1
+// #define HAVE_IOCTL 1
 
 /* Define to 1 if you have `ioctlsocket` */
 /* #undef HAVE_IOCTLSOCKET */
@@ -158,7 +158,7 @@
 /* #undef HAVE_IOCTLSOCKET_FIONBIO */
 
 /* ioctl() with FIONBIO support */
-#define HAVE_IOCTL_FIONBIO 1 
+// #define HAVE_IOCTL_FIONBIO 1 
 
 /* Define to 1 if you have the <iphlpapi.h> header file. */
 /* #undef HAVE_IPHLPAPI_H */
@@ -212,7 +212,7 @@
 #define HAVE_PIPE 1
 
 /* Define to 1 if you have `pipe2` */
-#define HAVE_PIPE2 1
+// #define HAVE_PIPE2 1
 
 /* Define to 1 if you have `poll` */
 #define HAVE_POLL 1
