diff --git a/src/steps/generic.rs b/src/steps/generic.rs
index 0caeff1..5beafcc 100644
--- a/src/steps/generic.rs
+++ b/src/steps/generic.rs
@@ -285,9 +285,9 @@ pub fn run_stack_update(run_type: RunType) -> Result<()> {
 
 pub fn run_tlmgr_update(ctx: &ExecutionContext) -> Result<()> {
     cfg_if::cfg_if! {
-        if #[cfg(target_os = "linux")] {
+        if #[cfg(any(target_os = "linux", target_os = "android"))] {
             if !ctx.config().enable_tlmgr_linux() {
-                return Err(SkipStep(String::from("tlmgr must be explicity enabled in the configuration to run in Linux")).into());
+                return Err(SkipStep(String::from("tlmgr must be explicity enabled in the configuration to run in Android/Linux")).into());
             }
         }
     }
