--- ./pacman.c~	2019-08-11 11:40:25.424773750 +0200
+++ ./pacman.c	2019-08-11 11:46:32.051044212 +0200
@@ -33,6 +33,7 @@
 void IntroScreen();                                     //Show introduction screen and menu
 int  CheckCollision();                                  //See if Pacman and Ghosts collided
 void CheckScreenSize();                                 //Make sure resolution is at least 32x29
+int  GameOverScreen();
 void CreateWindows(int y, int x, int y0, int x0);       //Make ncurses windows
 void Delay();                                           //Slow down game for better control
 void DrawWindow();                                      //Refresh display
