diff -uNr tigervnc-1.8.0/unix/tx/TXDialog.h tigervnc-1.8.0.mod/unix/tx/TXDialog.h
--- tigervnc-1.8.0/unix/tx/TXDialog.h	2017-05-16 16:53:28.000000000 +0300
+++ tigervnc-1.8.0.mod/unix/tx/TXDialog.h	2018-06-15 23:44:13.101940995 +0300
@@ -36,6 +36,10 @@
 #include <sys/proto.h>
 #endif
 
+#ifdef __ANDROID__
+#include <sys/select.h>
+#endif
+
 class TXDialog : public TXWindow, public TXDeleteWindowCallback {
 public:
   TXDialog(Display* dpy, int width, int height, const char* name,
