diff -uNr tigervnc-1.9.0/unix/vncpasswd/vncpasswd.cxx tigervnc-1.9.0.mod/unix/vncpasswd/vncpasswd.cxx
--- tigervnc-1.9.0/unix/vncpasswd/vncpasswd.cxx	2018-07-16 17:08:55.000000000 +0300
+++ tigervnc-1.9.0.mod/unix/vncpasswd/vncpasswd.cxx	2018-09-12 11:15:13.158490703 +0300
@@ -50,7 +50,7 @@
   else
     attrs.c_lflag &= ~ECHO;
   attrs.c_lflag |= ECHONL;
-  tcsetattr(fileno(stdin), TCSAFLUSH, &attrs);
+  tcsetattr(fileno(stdin), TCSANOW, &attrs);
 }
 
 static char* getpassword(const char* prompt) {
