diff -uNr tigervnc-1.9.0/common/os/CMakeLists.txt tigervnc-1.9.0.mod/common/os/CMakeLists.txt
--- tigervnc-1.9.0/common/os/CMakeLists.txt	2018-07-16 17:08:55.000000000 +0300
+++ tigervnc-1.9.0.mod/common/os/CMakeLists.txt	2018-09-12 11:14:21.776323226 +0300
@@ -6,7 +6,7 @@
   w32tiger.c
   os.cxx)
 
-if(UNIX)
+if(UNIX AND NOT ${CMAKE_SYSTEM_NAME} MATCHES "Android")
   target_link_libraries(os pthread)
 endif()
 
