diff -u -r ../iperf-3.1/src/iperf_api.c ./src/iperf_api.c
--- ../iperf-3.1/src/iperf_api.c	2015-10-16 13:01:09.000000000 -0400
+++ ./src/iperf_api.c	2015-11-08 01:20:52.345913816 -0500
@@ -2594,7 +2594,7 @@
     if (test->template) {
         snprintf(template, sizeof(template) / sizeof(char), "%s", test->template);
     } else {
-        char buf[] = "/tmp/iperf3.XXXXXX";
+        char buf[] = "@TERMUX_PREFIX@/tmp/iperf3.XXXXXX";
         snprintf(template, sizeof(template) / sizeof(char), "%s", buf);
     }
 
