diff -u -r ../iperf-3.0.6/src/Makefile.in ./src/Makefile.in
--- ../iperf-3.0.6/src/Makefile.in	2014-07-28 18:33:11.000000000 +0200
+++ ./src/Makefile.in	2014-08-27 10:04:22.000000000 +0200
@@ -593,9 +593,9 @@
 iperf3_profile_SOURCES = main.c \
 						  $(libiperf_la_SOURCES)
 
-iperf3_profile_CFLAGS = -pg -g
+iperf3_profile_CFLAGS = -g
 iperf3_profile_LDADD = libiperf.la
-iperf3_profile_LDFLAGS = -pg -g
+iperf3_profile_LDFLAGS = -g
 
 # Specify the sources and various flags for the test cases
 t_timer_SOURCES = t_timer.c
