diff -uNr a/macho/main.cc b/macho/main.cc
--- a/macho/main.cc	2022-09-04 12:59:58.000000000 +0800
+++ b/macho/main.cc	2022-09-21 12:05:34.587244400 +0800
@@ -450,9 +450,17 @@
   });
 
   static Counter counter("num_merged_strings");
+#ifndef __ANDROID__
   counter += std::erase_if(osec.members, [](Subsection<E> *subsec) {
     return subsec->is_coalesced;
   });
+#else
+  auto old_size = osec.members.size();
+  std::erase_if(osec.members, [](Subsection<E> *subsec) {
+    return subsec->is_coalesced;
+  });
+  counter += (old_size - osec.members.size());
+#endif
 }
 
 template <typename E>
