diff -u -r ../mold-2.34.0/lib/jobs-unix.cc ./lib/jobs-unix.cc
--- ../mold-2.34.0/lib/jobs-unix.cc	2024-09-25 01:01:39.000000000 +0000
+++ ./lib/jobs-unix.cc	2024-09-25 17:00:39.457328015 +0000
@@ -31,7 +31,7 @@
   if (char *dir = getenv("XDG_RUNTIME_DIR"))
     path = dir + "/mold-lock"s;
   else
-    path = "/tmp/mold-lock-"s + getpwuid(getuid())->pw_name;
+    path = "@TERMUX_PREFIX@/tmp/mold-lock-"s + getpwuid(getuid())->pw_name;
 
   int fd = open(path.c_str(), O_WRONLY | O_CREAT | O_CLOEXEC, 0600);
   if (fd == -1)
