https://github.com/termux/termux-packages/issues/25061

--- a/src/cmdline.cc
+++ b/src/cmdline.cc
@@ -1318,6 +1318,7 @@
     } else if (read_flag("O0")) {
     } else if (read_flag("O1")) {
     } else if (read_flag("O2")) {
+    } else if (read_flag("use-android-relr-tags")) {
     } else if (read_flag("verbose")) {
     } else if (read_flag("color-diagnostics")) {
     } else if (read_flag("eh-frame-hdr")) {
