diff --git a/src/ARMJIT_Memory.h b/src/ARMJIT_Memory.h
index 0fbce17c..4bf660e1 100644
--- a/src/ARMJIT_Memory.h
+++ b/src/ARMJIT_Memory.h
@@ -190,7 +190,7 @@ private:
     static void SigsegvHandler(int sig, siginfo_t* info, void* rawContext);
     int MemoryFile = -1;
 #endif
-#ifdef ANDROID
+#if defined(__ANDROID__)
     Platform::DynamicLibrary* Libandroid = nullptr;
 #endif
     u8 MappingStatus9[1 << (32-12)] {};
