--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,9 +6,6 @@
 set(CMAKE_AUTOMOC ON)
 set(CMAKE_AUTORCC ON)
 
-# C++11
-set(CMAKE_CXX_STANDARD 11)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 # Optional D-Bus support
 option(USE_DBUS "Use D-Bus integration" ON)
@@ -76,7 +76,7 @@
 endif ()
 
 # Create main executable
-if (ANDROID)
+if (FALSE)
 	add_library(spotify-qt SHARED res.qrc)
 elseif (MSVC)
 	add_executable(spotify-qt WIN32 res.qrc
