--- a/meson.build
+++ b/meson.build
@@ -666,6 +666,8 @@
                output: 'xkeyboard-config-test',
                configuration: xkct_config)
 
+if false
+
 # Tests
 test_env = environment()
 test_env.set('XKB_LOG_LEVEL', 'debug')
@@ -1107,6 +1109,7 @@
   )
 endif
 
+endif
 
 # Documentation.
 if get_option('enable-docs')
@@ -1227,7 +1230,4 @@
       'rules': get_option('default-rules'),
       'variant': get_option('default-variant'),
     }, section: 'Defaults')
-    summary({
-      'merge-modes': has_merge_modes_tests,
-    }, section: 'Tests')
 endif
