diff -uNr fvwm-2.6.9/configure fvwm-2.6.9.mod/configure
--- fvwm-2.6.9/configure	2019-09-05 20:55:22.000000000 +0000
+++ fvwm-2.6.9.mod/configure	2019-10-13 18:54:47.276104871 +0000
@@ -10796,74 +10796,12 @@
   fribidi_in_path=yes
   ac_save_LIBS="$LIBS"
   LIBS="$LIBS $Bidi_LIBS"
-  if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
-else
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-#include <stdlib.h>
-#include <fribidi/fribidi.h>
-int main()
-{
-	FriBidiChar *logical_unicode_str =
-		(FriBidiChar *)malloc((4 + 1) * sizeof(FriBidiChar));
-	fribidi_charset_to_unicode(
-		fribidi_parse_charset("iso8859-8"), "test", 4,
-		logical_unicode_str);
-	return 0;
-}
 
-_ACEOF
-if ac_fn_c_try_run "$LINENO"; then :
-  :
-else
-  fribidi_in_path=no
-fi
-rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
-  conftest.$ac_objext conftest.beam conftest.$ac_ext
-fi
-
-  LIBS="$ac_save_LIBS"
   if test ! x"$fribidi_in_path" = xyes; then
     ac_save_CFLAGS="$CFLAGS"
     ac_save_LIBS="$LIBS"
     CFLAGS="$CFLAGS $Bidi_CFLAGS"
     LIBS="$LIBS $Bidi_LIBS"
-  if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
-else
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-#include <stdlib.h>
-#include <fribidi/fribidi.h>
-int main()
-{
-	FriBidiChar *logical_unicode_str =
-		(FriBidiChar *)malloc((4 + 1) * sizeof(FriBidiChar));
-	fribidi_charset_to_unicode(
-		fribidi_parse_charset("iso8859-8"), "test", 4,
-		logical_unicode_str);
-	return 0;
-}
-
-_ACEOF
-if ac_fn_c_try_run "$LINENO"; then :
-  :
-else
-  with_bidi=no; problem_bidi=": Bad fribidi version, see config.log"
-fi
-rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
-  conftest.$ac_objext conftest.beam conftest.$ac_ext
-fi
-
-    CFLAGS="$ac_save_CFLAGS"
-    LIBS="$ac_save_LIBS"
   fi
   if test ! x"$fribidi_in_path" = xno; then
     Bidi_CFLAGS=
@@ -10878,29 +10816,6 @@
   Bidi_LIBS=
 else
   $as_echo "#define HAVE_BIDI 1" >>confdefs.h
-
-  if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
-else
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-#include <fribidi/fribidi.h>
-    int main(int c, char **v) {
-      return FRIBIDI_CHARSET_NOT_FOUND * 0;
-    }
-
-_ACEOF
-if ac_fn_c_try_run "$LINENO"; then :
-  $as_echo "#define FRIBIDI_CHARSET_SPELLING 1" >>confdefs.h
-
-fi
-rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
-  conftest.$ac_objext conftest.beam conftest.$ac_ext
-fi
-
 fi
 
 
@@ -13015,41 +12930,6 @@
 CPPFLAGS="$CPPFLAGS $X_CFLAGS"
 
 
-if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
-else
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-#include <stdio.h>
-int main(int c, char **v) {
-#include <X11/keysym.h>
-#if defined XK_Page_Up && defined XK_Page_Down
-  return 0;
-#else
-  return 1;
-#endif
-}
-
-_ACEOF
-if ac_fn_c_try_run "$LINENO"; then :
-  :
-else
-  $as_echo "#define COMPAT_OLD_KEYSYMDEF 1" >>confdefs.h
-
-fi
-rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
-  conftest.$ac_objext conftest.beam conftest.$ac_ext
-fi
-
-
-CPPFLAGS="$mg_save_CPPFLAGS"
-
-
-
 if test x"$with_stroke" = xyes; then
 
 mg_save_CPPFLAGS="$CPPFLAGS"
