--- ./Makefile.orig	2021-05-15 19:20:40.634439555 +0000
+++ ./Makefile	2021-05-15 19:21:29.281077019 +0000
@@ -109,17 +109,16 @@
 	rm -f *~
 
 install: tree
-	install -d $(BINDIR)
-	install -d $(MANDIR)
+	install -d $(DESTDIR)$(BINDIR)
+	install -d $(DESTDIR)$(MANDIR)
 	if [ -e $(TREE_DEST) ]; then \
-		install $(TREE_DEST) $(BINDIR)/$(TREE_DEST); \
+		install $(TREE_DEST) $(DESTDIR)$(BINDIR)/$(TREE_DEST); \
 	fi
-	install doc/$(MAN) $(MANDIR)/$(MAN)
+	install doc/$(MAN) $(DESTDIR)$(MANDIR)/$(MAN)
 
 distclean:
 	if [ -f tree.o ]; then rm *.o; fi
 	rm -f *~
-	
 
 dist:	distclean
 	tar zcf ../tree-$(VERSION).tgz -C .. `cat .tarball`
