--- ./strverscmp.c.orig	2022-04-17 17:30:13.079263409 +0000
+++ ./strverscmp.c	2022-04-17 17:37:27.495463877 +0000
@@ -21,7 +21,7 @@
 /* #include "libiberty.h" */
 /* #include "safe-ctype.h" */
 #include <ctype.h>
-#ifndef __linux__
+#if defined(__ANDROID__) || !defined(__linux__) 
 
 /* 
 @deftypefun int strverscmp (const char *@var{s1}, const char *@var{s2})
@@ -157,4 +157,4 @@
       return state;
     }
 }
-#endif
\ No newline at end of file
+#endif
