--- a/cmake-scripts/boomerang-flags.cmake
+++ b/cmake-scripts/boomerang-flags.cmake
@@ -60,7 +60,7 @@
     set(DEBUG_LIB dbghelp.lib)
 
 else () # GCC / Clang
-    BOOMERANG_ADD_COMPILE_FLAGS(-Wall -Wextra -Werror -Werror=pedantic)
+    BOOMERANG_ADD_COMPILE_FLAGS(-Wall -Wextra)
     BOOMERANG_ADD_COMPILE_FLAGS(-Wshadow)
     BOOMERANG_ADD_COMPILE_FLAGS(-Wformat=2)
     BOOMERANG_ADD_COMPILE_FLAGS(-Wmissing-include-dirs)
