diff -uNr mtpaint-3.40/src/spawn.c mtpaint-3.40.mod/src/spawn.c
--- mtpaint-3.40/src/spawn.c	2011-10-30 23:30:53.000000000 +0200
+++ mtpaint-3.40.mod/src/spawn.c	2018-06-24 23:31:22.228892213 +0300
@@ -45,7 +45,7 @@
 #ifdef WIN32
 		env = "\\";
 #else
-		env = "/tmp";
+		env = "@TERMUX_PREFIX@/tmp";
 #endif
 	return (env);
 }
@@ -758,8 +758,8 @@
 #else /* Linux */
 
 #define HANDBOOK_BROWSER "firefox"
-#define HANDBOOK_LOCATION "/usr/doc/mtpaint/index.html"
-#define HANDBOOK_LOCATION2 "/usr/share/doc/mtpaint/index.html"
+#define HANDBOOK_LOCATION "@TERMUX_PREFIX@/doc/mtpaint/index.html"
+#define HANDBOOK_LOCATION2 "@TERMUX_PREFIX@/share/doc/mtpaint/index.html"
 
 #endif
 
