diff -uNr mtpaint-3.40/src/font.c mtpaint-3.40.mod/src/font.c
--- mtpaint-3.40/src/font.c	2011-12-29 20:03:08.000000000 +0200
+++ mtpaint-3.40.mod/src/font.c	2018-06-24 23:31:22.212225544 +0300
@@ -1510,8 +1510,8 @@
 		FILE *fp;
 		char buf[4096], buf2[128], *s;
 
-		if (!(fp = fopen("/etc/X11/xorg.conf", "r")))
-			fp = fopen("/etc/X11/XF86Config", "r");
+		if (!(fp = fopen("@TERMUX_PREFIX@/etc/X11/xorg.conf", "r")))
+			fp = fopen("@TERMUX_PREFIX@/etc/X11/XF86Config", "r");
 
 		// If these files are not found the user will have to manually enter directories
 
@@ -1537,7 +1537,7 @@
 			fclose(fp);
 		}
 
-		if (!new_dirs && (fp = fopen("/etc/fonts/fonts.conf", "r")))
+		if (!new_dirs && (fp = fopen("@TERMUX_PREFIX@/etc/fonts/fonts.conf", "r")))
 		{
 			char *s1, *s2;
 
