diff -uNr mtpaint-3.40/src/main.c mtpaint-3.40.mod/src/main.c
--- mtpaint-3.40/src/main.c	2011-10-30 23:29:17.000000000 +0200
+++ mtpaint-3.40.mod/src/main.c	2018-06-24 23:31:22.215558878 +0300
@@ -287,7 +287,7 @@
 	/* Enable threading for GLib, but NOT for GTK+ (at least, not yet) */
 	g_thread_init(NULL);
 #endif
-	inifile_init("/etc/mtpaint/mtpaintrc", "~/.mtpaint");
+	inifile_init("@TERMUX_PREFIX@/etc/mtpaint/mtpaintrc", "~/.mtpaint");
 
 #ifdef U_NLS
 #if GTK_MAJOR_VERSION == 1
