--- ../cache/ncmpcpp-8134e6e23b2787322fa10e65e44d286da82eea91/configure.ac	2017-04-23 02:24:57.000000000 +0000
+++ ./configure.ac	2017-05-06 00:32:15.468042009 +0000
@@ -147,9 +147,9 @@
 fi
 
 # pthread
-AC_CHECK_LIB(pthread, pthread_create, LIBS="$LIBS -lpthread",
-	AC_MSG_ERROR([pthread library is required])
-)
+#AC_CHECK_LIB(pthread, pthread_create, LIBS="$LIBS -lboost_thread",
+#	AC_MSG_ERROR([pthread library is required])
+#)
 
 # ncursesw
 PKG_CHECK_MODULES([ncursesw], [ncursesw], [
