--- ../cache/ncmpcpp-8134e6e23b2787322fa10e65e44d286da82eea91/doc/config	2017-04-23 02:24:57.000000000 +0000
+++ ./doc/config	2017-05-13 23:29:50.055629354 +0000
@@ -49,7 +49,13 @@
 ## }
 ##
 #
-#visualizer_fifo_path = /tmp/mpd.fifo
+visualizer_fifo_path = "@TERMUX_PREFIX@/tmp/mpd.fifo"
+visualizer_output_name = "my_fifo"
+visualizer_sync_interval = "30"
+visualizer_in_stereo = "yes"
+visualizer_type = "spectrum"
+visualizer_look = "+|"
+
 #
 ##
 ## Note: Below parameter is needed for ncmpcpp to determine which output
