--- ../cache/ncmpcpp-8134e6e23b2787322fa10e65e44d286da82eea91/src/configuration.cpp	2017-04-23 02:24:57.000000000 +0000
+++ ./src/configuration.cpp	2017-05-14 00:48:22.456559395 +0000
@@ -84,8 +84,8 @@
 
 	po::options_description options("Options");
 	options.add_options()
-		("host,h", po::value<std::string>()->value_name("HOST")->default_value("localhost"), "connect to server at host")
-		("port,p", po::value<int>()->value_name("PORT")->default_value(6600), "connect to server at port")
+		("host,h", po::value<std::string>()->value_name("HOST")->default_value("~/.mpd/socket"), "connect to server at host")
+		("port,p", po::value<int>()->value_name("PORT")->default_value(6600), "connect to server at port(for termux local its 8600)")
 		("current-song", po::value<std::string>()->value_name("FORMAT")->implicit_value("{{{(%l) }{{%a - }%t}}|{%f}}"), "print current song using given format and exit")
 		("config,c", po::value<std::vector<std::string>>(&config_paths)->value_name("PATH")->default_value(default_config_paths, join<std::string>(default_config_paths, " AND ")), "specify configuration file(s)")
 		("ignore-config-errors", "ignore unknown and invalid options in configuration files")
