diff -u -r ../apt-1.1.3/apt-pkg/deb/debrecords.cc ./apt-pkg/deb/debrecords.cc
--- ../apt-1.1.3/apt-pkg/deb/debrecords.cc	2015-11-30 03:08:24.000000000 -0500
+++ ./apt-pkg/deb/debrecords.cc	2015-12-03 16:30:51.506746167 -0500
@@ -25,7 +25,9 @@
 #include <sstream>
 #include <string>
 #include <vector>
+#ifndef __ANDROID__
 #include <langinfo.h>
+#endif
 
 #include <apti18n.h>
 									/*}}}*/
@@ -143,12 +145,14 @@
 	 orig = Section.FindS("Description");
    }
 
+#ifndef __ANDROID__
    char const * const codeset = nl_langinfo(CODESET);
    if (strcmp(codeset,"UTF-8") != 0) {
       string dest;
       UTF8ToCodeset(codeset, orig, &dest);
       return dest;
    }
+#endif
 
    return orig;
 }
