--- src/src/tools/clangpchmanagerbackend/source/generatepchactionfactory.h	2020-07-07 06:44:20.000000000 +0000
+++ src.mod//src/tools/clangpchmanagerbackend/source/generatepchactionfactory.h	2021-05-24 12:46:08.430407000 +0000
@@ -44,7 +44,7 @@
 #if LLVM_VERSION_MAJOR >= 12
         compilerInstance.getPreprocessorOpts().DisablePCHOrModuleValidation = clang::DisableValidationForModuleKind::PCH;
 #else
-        compilerInstance.getPreprocessorOpts().DisablePCHValidation = true;
+        // compilerInstance.getPreprocessorOpts().DisablePCHValidation = true;
 #endif
         compilerInstance.getPreprocessorOpts().AllowPCHWithCompilerErrors = true;
         compilerInstance.getDiagnosticOpts().ErrorLimit = 0;
