--- src/src/plugins/plugins.pro	2021-05-24 18:00:34.384117000 +0000
+++ src.mod/src/plugins/plugins.pro	2021-05-24 18:01:03.292691000 +0000
@@ -16,7 +16,6 @@
     bookmarks \
     projectexplorer \
     vcsbase \
-    perforce \
     subversion \
     git \
     cvs \
@@ -35,9 +34,6 @@
     qmlprojectmanager \
     glsleditor \
     python \
-    nim \
-    mercurial \
-    bazaar \
     classview \
     tasklist \
     qmljstools \
@@ -46,13 +42,8 @@
     android \
     valgrind \
     todo \
-    qnx \
-    clearcase \
-    baremetal \
-    ios \
     beautifier \
     modeleditor \
-    winrt \
     updateinfo \
     scxmleditor \
     welcome \
@@ -61,9 +52,6 @@
     cppcheck \
     compilationdatabaseprojectmanager \
     qmlpreview \
-    studiowelcome \
-    webassembly \
-    mcusupport \
     marketplace \
     incredibuild \
     conan
