--- src/src/plugins/plugins.pro	2020-07-07 06:44:20.000000000 +0000
+++ src.mod/src/plugins/plugins.pro	2021-05-24 11:08:50.489277117 +0000
@@ -72,12 +72,6 @@
     warning("SerialTerminal plugin has been disabled since the Qt SerialPort module is not available.")
 }
 
-qtHaveModule(quick) {
-    SUBDIRS += qmlprofiler perfprofiler ctfvisualizer
-} else {
-    warning("QmlProfiler, PerfProfiler and CTF Visualizer plugins have been disabled since the Qt Quick module is not available.")
-}
-
 qtHaveModule(help) {
     SUBDIRS += help
 } else {
--- src/src/libs/libs.pro	2020-07-07 06:44:20.000000000 +0000
+++ src.mod/src/libs/libs.pro	2021-05-24 10:03:26.192640937 +0000
@@ -18,11 +18,6 @@
     languageserverprotocol \
     sqlite
 
-qtHaveModule(quick) {
-    SUBDIRS += \
-        tracing
-}
-
 QTC_DO_NOT_BUILD_QMLDESIGNER = $$(QTC_DO_NOT_BUILD_QMLDESIGNER)
 isEmpty(QTC_DO_NOT_BUILD_QMLDESIGNER):qtHaveModule(quick-private) {
     exists($$[QT_INSTALL_QML]/QtQuick/Controls/qmldir) {
