diff -u -r ../node-v5.3.0/node.gyp ./node.gyp
--- ../node-v5.3.0/node.gyp	2015-12-16 13:49:23.000000000 -0500
+++ ./node.gyp	2015-12-25 23:03:42.313063892 -0500
@@ -670,31 +670,6 @@
           ],
         } ],
       ]
-    },
-    {
-      'target_name': 'cctest',
-      'type': 'executable',
-      'dependencies': [
-        'deps/gtest/gtest.gyp:gtest',
-        'deps/v8/tools/gyp/v8.gyp:v8',
-        'deps/v8/tools/gyp/v8.gyp:v8_libplatform'
-      ],
-      'include_dirs': [
-        'src',
-        'deps/v8/include'
-      ],
-      'defines': [
-        # gtest's ASSERT macros conflict with our own.
-        'GTEST_DONT_DEFINE_ASSERT_EQ=1',
-        'GTEST_DONT_DEFINE_ASSERT_GE=1',
-        'GTEST_DONT_DEFINE_ASSERT_GT=1',
-        'GTEST_DONT_DEFINE_ASSERT_LE=1',
-        'GTEST_DONT_DEFINE_ASSERT_LT=1',
-        'GTEST_DONT_DEFINE_ASSERT_NE=1',
-      ],
-      'sources': [
-        'test/cctest/util.cc',
-      ],
     }
   ], # end targets
 
