diff -uNr quick-lint-js/src/quick-lint-js/port/have.h quick-lint-js.mod/src/quick-lint-js/port/have.h
--- quick-lint-js/src/quick-lint-js/port/have.h	2022-07-26 10:45:47.584150790 +0800
+++ quick-lint-js.mod/src/quick-lint-js/port/have.h	2022-07-26 10:47:14.734150757 +0800
@@ -285,7 +285,7 @@
 #endif
 
 #if !defined(QLJS_HAVE_ARM_NEON)
-#if defined(__ARM_NEON)
+#if defined(__ARM_NEON) && defined(__aarch64__)
 #define QLJS_HAVE_ARM_NEON 1
 #else
 #define QLJS_HAVE_ARM_NEON 0
