diff --git a/moby-20.10.1/cmd/dockerd/daemon.go b/moby/cmd/dockerd/daemon.go
index 7fe8a6c..684f297 100644
--- a/moby-20.10.1/cmd/dockerd/daemon.go
+++ b/moby/cmd/dockerd/daemon.go
@@ -554,6 +554,11 @@ func (cli *DaemonCli) getContainerdDaemonOpts() ([]supervisor.DaemonOpt, error)
 		opts = append(opts, supervisor.WithPlugin("cri", nil))
 	}
 
+	type Config struct {Path string `toml:"path"`}
+	opts = append(opts, supervisor.WithPlugin("opt", &Config{
+		Path: "/data/docker/opt",
+	}))
+
 	return opts, nil
 }
