--- ./cli/scripts/docs/generate-man.sh.orig	2021-12-05 15:26:07.599604265 +0100
+++ ./cli/scripts/docs/generate-man.sh	2021-12-05 15:28:20.809486509 +0100
@@ -6,12 +6,12 @@
 
 if ! command -v go-md2man &> /dev/null; then
 	# yay, go install creates a binary named "v2" ¯\_(ツ)_/¯
-	go build -o "/go/bin/go-md2man" ./vendor/github.com/cpuguy83/go-md2man/v2
+	go build -o "build/go-md2man" ./vendor/github.com/cpuguy83/go-md2man/v2
 fi
 
 # Generate man pages from cobra commands
-go build -o /tmp/gen-manpages github.com/docker/cli/man
-/tmp/gen-manpages --root "$(pwd)" --target "$(pwd)/man/man1"
+go build -o build/gen-manpages github.com/docker/cli/man
+build/gen-manpages --root "$(pwd)" --target "$(pwd)/man/man1"
 
 # Generate legacy pages from markdown
 ./man/md2man-all.sh -q
