diff -uNr ./moby-24.0.6.orig/cmd/dockerd/daemon_unix.go ./moby/cmd/dockerd/daemon_unix.go
--- ./moby-24.0.6.orig/cmd/dockerd/daemon_unix.go	2023-08-29 22:14:17.000000000 +0300
+++ ./moby/cmd/dockerd/daemon_unix.go	2024-02-01 19:50:22.423211230 +0200
@@ -24,7 +24,7 @@
 
 func getDefaultDaemonConfigDir() (string, error) {
 	if !honorXDG {
-		return "/etc/docker", nil
+		return "@TERMUX_PREFIX@/etc/docker", nil
 	}
 	// NOTE: CLI uses ~/.docker while the daemon uses ~/.config/docker, because
 	// ~/.docker was not designed to store daemon configurations.
