From d57808bd313d9fc9fb6849dabde31bd0ae1b43f4 Mon Sep 17 00:00:00 2001
From: Marius Kintel <marius@kintel.net>
Date: Mon, 18 Aug 2025 20:15:44 -0400
Subject: [PATCH] Boost system is not longer in use

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -80,7 +80,7 @@ if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
   endif()
 endif()
 
-set(BOOST_DIRECTLY_REQUIRED_LIBRARIES filesystem system regex program_options)
+set(BOOST_DIRECTLY_REQUIRED_LIBRARIES filesystem regex program_options)
 
 find_package(PkgConfig)
 
