--- ../CMakeLists.txt.orig	2020-02-14 20:41:56.576884809 +0100
+++ ./CMakeLists.txt	2020-02-14 20:44:41.792539280 +0100
@@ -201,9 +201,6 @@
   # Check if boost dependency libraries have been found too
   list(LENGTH BOOST_DIRECTLY_REQUIRED_LIBRARIES BOOST_DIRECTLY_REQUIRED_LIBRARIES_LENGTH)
   list(LENGTH Boost_LIBRARIES Boost_LIBRARIES_LENGTH)
-  if(Boost_LIBRARIES_LENGTH EQUAL BOOST_DIRECTLY_REQUIRED_LIBRARIES_LENGTH)
-    message(FATAL_ERROR "No dependent Boost libraries found. Your CMake (${CMAKE_VERSION}) version might be too old for the version of boost you are using (${Boost_MAJOR_VERSION}.${Boost_MINOR_VERSION}.${Boost_SUBMINOR_VERSION}). In that case, you should have received warnings above of the type: 'Imported targets not available for Boost version ${Boost_VERSION}'")
-  endif()
 
   find_package(HarfBuzz 0.9.19 REQUIRED QUIET)
   message(STATUS "Harfbuzz: ${HARFBUZZ_VERSION}")
@@ -688,6 +685,8 @@
   target_link_libraries(OpenSCAD spnav)
 endif()
 
+list(APPEND COMMON_LIBRARIES boost_thread boost_program_options boost_filesystem boost_system boost_regex)
+
 target_link_libraries(OpenSCAD PRIVATE ${COMMON_LIBRARIES} ${PLATFORM_LIBS})
 if(NOT HEADLESS)
   target_link_libraries(OpenSCAD PRIVATE Qt5::Core Qt5::Widgets Qt5::Multimedia Qt5::OpenGL Qt5::Concurrent Qt5::Network ${QT5QSCINTILLA_LIBRARY} ${Qt5DBus_LIBRARIES} ${Qt5Gamepad_LIBRARIES})
