--- ./src/FileModule.cc.orig	2024-12-22 12:44:51.462463542 +0000
+++ ./src/FileModule.cc	2024-12-22 12:45:36.708971549 +0000
@@ -65,7 +65,7 @@
 	auto ext = fs::path(path).extension().generic_string();
 
 	if (boost::iequals(ext, ".otf") || boost::iequals(ext, ".ttf")) {
-		if (fs::is_regular(path)) {
+		if (fs::is_regular_file(path)) {
 			FontCache::instance()->register_font_file(path);
 		} else {
 			LOG(message_group::Error,Location::NONE,"","Can't read font with path '%1$s'",path);
