+++ a/src/NULLGL.cc
@@ -23,10 +23,11 @@
 	assert(false && "not implemented");
 	return BoundingBox();
 }
-void ThrownTogetherRenderer::renderCSGProducts(const CSGProducts &products, bool highlight_mode, bool background_mode, bool showedges, 
+void ThrownTogetherRenderer::renderCSGProducts(const CSGProducts &products, const GLView::shaderinfo_t *, bool highlight_mode, bool background_mode, bool showedges, 
                         bool fberror) const {}
-void ThrownTogetherRenderer::renderChainObject(const class CSGChainObject &csgobj, bool highlight_mode,
+void ThrownTogetherRenderer::renderChainObject(const class CSGChainObject &csgobj, const GLView::shaderinfo_t *, bool highlight_mode,
                         bool background_mode, bool showedges, bool fberror, OpenSCADOperator type) const {}
+void ThrownTogetherRenderer::draw_with_shader(const GLView::shaderinfo_t *shaderinfo, bool showedges) const {}
 
 #include "CGALRenderer.h"
 
