diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e591492..c719caa 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -53,7 +53,6 @@ install( TARGETS cantor ${KDE_INSTALL_TARGETS_DEFAULT_ARGS} )
 install( FILES org.kde.cantor.desktop  DESTINATION ${KDE_INSTALL_APPDIR} )
 install( FILES cantor.knsrc cantor-documentation.knsrc DESTINATION ${KDE_INSTALL_KNSRCDIR} )
 install( FILES cantor.xml DESTINATION ${KDE_INSTALL_MIMEDIR} )
-update_xdg_mimetypes(${KDE_INSTALL_MIMEDIR})
 
 #########################################################################
 # KPART SECTION
