diff -u -r ../emacs-24.4/lib-src/Makefile.in ./lib-src/Makefile.in
--- ../emacs-24.4/lib-src/Makefile.in	2014-04-29 16:52:57.000000000 +0200
+++ ./lib-src/Makefile.in	2014-11-01 12:29:44.881760880 +0100
@@ -128,7 +128,7 @@
 UTILITIES = profile${EXEEXT} movemail${EXEEXT} hexl${EXEEXT} \
             update-game-score${EXEEXT}
 
-DONT_INSTALL= test-distrib${EXEEXT} make-docfile${EXEEXT}
+DONT_INSTALL= make-docfile${EXEEXT}
 
 # Like UTILITIES, but they're not system-dependent, and should not be
 #  deleted by the distclean target.
@@ -303,14 +303,6 @@
 TAGS: etags${EXEEXT}
 	etags *.[ch]
 
-## This verifies that the non-ASCII characters in the file \`testfile\'
-## have not been clobbered by whatever means were used to copy and
-## distribute Emacs.  If they were clobbered, all the .elc files were
-## clobbered too.
-test-distrib${EXEEXT}: ${srcdir}/test-distrib.c
-	$(CC) ${ALL_CFLAGS} -o test-distrib${EXEEXT} ${srcdir}/test-distrib.c
-	./test-distrib ${srcdir}/testfile
-
 ../lib/libgnu.a: $(config_h)
 	cd ../lib && $(MAKE) libgnu.a
 
