diff --git a/src/btop_config.cpp b/src/btop_config.cpp
index 3faafc3..a203530 100644
--- a/src/btop_config.cpp
+++ b/src/btop_config.cpp
@@ -306,7 +306,8 @@ namespace Config {
 		{"swap_disk", true},
 		{"show_disks", true},
 		{"only_physical", true},
-		{"use_fstab", true},
+		// There is no /etc/fstab on Android.
+		{"use_fstab", false},
 		{"zfs_hide_datasets", false},
 		{"show_io_stat", true},
 		{"io_mode", false},
