diff -uNr tk8.6.11/unix/installManPage tk8.6.11.mod/unix/installManPage
--- tk8.6.11/unix/installManPage	2021-01-13 20:14:00.000000000 +0000
+++ tk8.6.11.mod/unix/installManPage	2021-05-05 11:59:01.889458448 +0000
@@ -117,7 +117,7 @@
 	chmod 644 "$Dir/$First"
 	$Gzip "$Dir/$First"
     else
-	ln $SymOrLoc"$First$Gz" "$Dir/$Target$Gz"
+	ln -sfr $SymOrLoc"$First$Gz" "$Dir/$Target$Gz"
     fi
 done
 
