diff --git a/lib/libpaper.c.in.in b/lib/libpaper.c.in.in
index 02af84a..1aa982c 100644
--- a/lib/libpaper.c.in.in
+++ b/lib/libpaper.c.in.in
@@ -31,7 +31,7 @@
 #include <strings.h>
 #include <ctype.h>
 #include <locale.h>
-#if defined _GNU_SOURCE && defined LC_PAPER && defined HAVE_NL_PAPER_CONSTANTS
+#if defined _GNU_SOURCE && defined LC_PAPER && defined HAVE_NL_PAPER_CONSTANTS && !defined(__ANDROID__)
 #define USE_NL_PAPER
 #include <langinfo.h>
 #endif
