--- a/config/paricfg.h.SH
+++ b/config/paricfg.h.SH
@@ -58,19 +58,19 @@
   echo '#define ASMINLINE' >> $file
 fi
 
-if test -n "$gzip"; then
+if test -n "@TERMUX_PREFIX@/bin/gzip"; then
   cat >> $file << EOT
 
 /*  Location of GNU gzip program (enables reading of .Z and .gz files). */
 #define GNUZCAT
-#define ZCAT "$gzip -dc"
+#define ZCAT "@TERMUX_PREFIX@/bin/gzip -dc"
 
 EOT
-else if test -n "$zcat"; then
+else if test -n "@TERMUX_PREFIX@/bin/zcat"; then
   cat >> $file << EOT
 
 /*  Location of zcat program (enables reading of .Z files). */
-#define ZCAT "$zcat"
+#define ZCAT "@TERMUX_PREFIX@/bin/zcat"
 
 EOT
   fi
