--- a/config/get_double_format
+++ b/config/get_double_format
@@ -1,7 +1,7 @@
 exe=$osname-$arch-endian$$$exe_suff
 cmd="$CC $CFLAGS $extraflag endian.c -o $exe"; . log_cmd
 if test -r $exe; then
-  doubleformat=`$RUNTEST $exe`;
+  doubleformat=$PARI_DOUBLE_FORMAT;
 else
   echo "***************************************************************"
   echo "Cannot compile endian.c. Aborting. PLEASE REPORT!"
