diff -uNr libmpeg2-0.5.1/configure.ac libmpeg2-0.5.1.mod/configure.ac
--- libmpeg2-0.5.1/configure.ac	2008-07-18 22:30:17.000000000 +0800
+++ libmpeg2-0.5.1.mod/configure.ac	2023-10-29 08:54:41.536435236 +0800
@@ -101,7 +101,7 @@
 	AC_TRY_CFLAGS([$TRY_CFLAGS $CFLAGS],[OPT_CFLAGS="$TRY_CFLAGS"]);;
     alpha*)
 	AC_DEFINE([ARCH_ALPHA],,[alpha architecture]);;
-    arm*)
+    arm*-gnueabi*)
 	arm_conditional=:
 	AC_DEFINE([ARCH_ARM],,[ARM architecture]);;
     esac
