--- bvi-1.4.0/bm_unix.c	2013-08-25 11:59:33.000000000 -0700
+++ src/bm_unix.c	2017-02-19 00:34:01.618869104 -0800
@@ -260,7 +260,6 @@
 vgetc()
 {
     char cha;
-    extern int errno;
 
     errno = 0;
     if (read(2, &cha, 1) <= 0) {
--- bvi-1.4.0/comm.c	2014-01-29 06:54:53.000000000 -0800
+++ src/comm.c	2017-02-19 00:34:01.622869129 -0800
@@ -84,7 +84,6 @@
 extern	char	*name;			/* actual filename */
 extern	char	**files;		/* used for "next" and "rewind" */
 extern	int		numfiles, curfile;
-extern	int		errno;
 
 static	char	oldbuf[CMDSZ];		/** for :!! command **/
 
