--- src/bm_unix.c~	2013-08-25 11:59:33.000000000 -0700
+++ src/bm_unix.c	2017-01-03 23:37:36.697875250 -0800
@@ -260,7 +260,6 @@
 vgetc()
 {
     char cha;
-    extern int errno;
 
     errno = 0;
     if (read(2, &cha, 1) <= 0) {
--- src/bvi.c~	2014-10-07 13:09:45.000000000 -0700
+++ src/bvi.c	2017-01-03 23:37:09.469794620 -0800
@@ -729,7 +729,6 @@
 	char	*arg;
 {
 	off_t	val;
-	extern int errno;
 	char	*poi;
 
 	errno = 0;
--- src/comm.c~	2014-01-29 06:54:53.000000000 -0800
+++ src/comm.c	2017-01-03 23:37:18.305820838 -0800
@@ -84,7 +84,6 @@
 extern	char	*name;			/* actual filename */
 extern	char	**files;		/* used for "next" and "rewind" */
 extern	int		numfiles, curfile;
-extern	int		errno;
 
 static	char	oldbuf[CMDSZ];		/** for :!! command **/
 
