diff -u -r ../openssl-1.1.1-pre8/Configurations/15-android.conf ./Configurations/15-android.conf
--- ../openssl-1.1.1-pre8/Configurations/15-android.conf	2018-06-20 14:48:09.000000000 +0000
+++ ./Configurations/15-android.conf	2018-08-17 21:34:02.996980008 +0000
@@ -127,10 +127,10 @@
         # systems are perfectly capable of executing binaries targeting
         # Froyo. Keep in mind that in the nutshell Android builds are
         # about JNI, i.e. shared libraries, not applications.
-        cflags           => add(sub { android_ndk()->{cflags} }),
-        cppflags         => add(sub { android_ndk()->{cppflags} }),
-        cxxflags         => add(sub { android_ndk()->{cflags} }),
-        bn_ops           => sub { android_ndk()->{bn_ops} },
+        #cflags           => add(sub { android_ndk()->{cflags} }),
+        #cppflags         => add(sub { android_ndk()->{cppflags} }),
+        #cxxflags         => add(sub { android_ndk()->{cflags} }),
+        #bn_ops           => sub { android_ndk()->{bn_ops} },
         bin_cflags       => "-pie",
         enable           => [ ],
     },
