--- src/rc/Makefile.in
+++ src/rc/Makefile.in
@@ -503,7 +503,7 @@
 jpicorc : jpicorc.in
 rjoerc : rjoerc.in
 .in:
-	$(AM_V_SED) LC_ALL=C $(SED) 's,@JOERC\@,$(sysconf_joedir),; s,@JOEDOC\@,$(data_docdir),; s,@JOEDATA\@,$(data_joedir),; s,@SPELL\@,@SPELL@,' $(srcdir)/$< > $@
+	$(AM_V_SED) LC_ALL=C $(SED) 's,@JOERC\@,$(sysconf_joedir),; s,@JOEDOC\@,$(data_docdir),; s,@JOEDATA\@,$(data_joedir),; s,@SPELL\@,@SPELL@,' $< > $@

 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
