diff --git a/src/mainwindow.cpp b/src/mainwindow.cpp
index 0c602f84..c5bbe91b 100644
--- a/src/mainwindow.cpp
+++ b/src/mainwindow.cpp
@@ -148,7 +148,7 @@ MainWindow::MainWindow()
     } else {
         libJack.unload();
     }
-    QLibrary libSDL("libSDL2-2.0.so.0");
+    QLibrary libSDL("libSDL2-2.0.so");
     if (!libSDL.load()) {
         QMessageBox::critical(this, qApp->applicationName(),
                               tr("Error: This program requires the SDL 2 library.\n\nPlease install it using your package manager. It may be named libsdl2-2.0-0, SDL2, or similar."));
