--- a/hwloc/utils/hwloc/hwloc-gather-topology.in 
+++ b/hwloc/utils/hwloc/hwloc-gather-topology.in 
@@ -54,7 +54,7 @@
    echo "  --no-cpuid  Do not gather x86 CPUID using hwloc-gather-cpuid"
    echo "  --keep      Keep the temporary copy of dumped files"
    echo "Example:"
-   echo "  $0 /tmp/\$(uname -n)"
+   echo "  $0 /data/data/com.termux/files/usr/tmp/\$(uname -n)"
 }
 
 while [ x`echo "$1" | cut -c1` = x- ] ; do

