--- a/clutter/Makefile.in
+++ b/clutter/Makefile.in
@@ -3745,13 +3745,8 @@
 	    rm -f $$lib.0.$(CLUTTER_LT_CURRENT).$(CLUTTER_LT_REVISION); \
 	  ) ; \
 	  (cd $(DESTDIR)$(libdir) && \
-	    { ln -s -f libclutter-$(CLUTTER_API_VERSION).so.0.$(CLUTTER_LT_CURRENT).$(CLUTTER_LT_REVISION) $$lib.0 || \
-	      { rm -f $$lib.0 && ln -s libclutter-1.0.so.0.$(CLUTTER_LT_CURRENT).$(CLUTTER_LT_REVISION) $$lib.0; }; \
-	    } \
-	  ) ; \
-	  (cd $(DESTDIR)$(libdir) && \
-	    { ln -s -f libclutter-$(CLUTTER_API_VERSION).so.0.$(CLUTTER_LT_CURRENT).$(CLUTTER_LT_REVISION) $$lib || \
-	      { rm -f $$lib && ln -s libclutter-1.0.so.0.$(CLUTTER_LT_CURRENT).$(CLUTTER_LT_REVISION) $$lib; }; \
+	    { ln -s -f libclutter-$(CLUTTER_API_VERSION).so $$lib || \
+	      { rm -f $$lib && ln -s libclutter-1.0.so $$lib; }; \
 	    } \
 	  ) ; \
 	done
