--- a/cmake/modules/FindInotify.cmake
+++ b/cmake/modules/FindInotify.cmake
@@ -34,7 +34,7 @@ find_path(Inotify_INCLUDE_DIRS sys/inotify.h)
 if(Inotify_INCLUDE_DIRS)
 # On Linux there is no library to link against, on the BSDs there is.
 # On the BSD's, inotify is implemented through a library, libinotify.
-    if( CMAKE_SYSTEM_NAME MATCHES "Linux")
+    if( CMAKE_SYSTEM_NAME MATCHES "Android")
         set(Inotify_FOUND TRUE)
         set(Inotify_LIBRARIES "")
         set(Inotify_INCLUDE_DIRS "")
