diff -uNr desktop-4.0.1/CMakeLists.txt desktop-4.0.1.mod/CMakeLists.txt
--- desktop-4.0.1/CMakeLists.txt	2025-10-27 21:28:23.000000000 +0530
+++ desktop-4.0.1.mod/CMakeLists.txt	2025-11-09 14:05:19.447245986 +0530
@@ -248,9 +248,9 @@
         find_package(Sparkle)
     endif()
 
-    if(UNIX AND NOT APPLE)
-        find_package(Inotify REQUIRED)
-    endif()
+#   if(UNIX AND NOT APPLE)
+#       find_package(Inotify REQUIRED)
+#   endif()
     find_package(OpenSSL 1.1 REQUIRED )
     find_package(PkgConfig REQUIRED)
     pkg_check_modules(OPENSC-LIBP11 libp11 REQUIRED IMPORTED_TARGET)
