--- a/src/libsync/ocsprofileconnector.cpp
+++ b/src/libsync/ocsprofileconnector.cpp
@@ -55,9 +55,9 @@ OCC::Optional<QPixmap> createPixmapFromSvgData(const QByteArray &iconData)
         return {};
     }
     QSize imageSize{16, 16};
-    if (OCC::Theme::isHidpi()) {
-        imageSize = QSize{32, 32};
-    }
+    // if (OCC::Theme::isHidpi()) {
+    //     imageSize = QSize{32, 32};
+    // }
     QImage scaledSvg(imageSize, QImage::Format_ARGB32);
     scaledSvg.fill("transparent");
     QPainter svgPainter{&scaledSvg};
