--- a/units.c
+++ b/units.c
@@ -130,7 +130,7 @@
 #ifdef _WIN32
 #  define DEFAULTPAGER "more"             /* Default pager for Windows */
 #else
-#  define DEFAULTPAGER "/usr/bin/pager"   /* Default pager for Unix */
+#  define DEFAULTPAGER "@TERMUX_PREFIX@/bin/less"   /* Default pager for Unix */
 #endif
 #define DEFAULTLOCALE "en_US"   /* Default locale */
 #define MAXINCLUDE 5            /* Max depth of include files */
