--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -60,7 +60,6 @@
     qml/pages/PlasmaUpdate.qml
     qml/pages/Live.qml
     qml/pages/Welcome.qml
-    qml/pages/Network.qml
     qml/pages/network/PlasmaNM.qml
     qml/pages/network/PlasmaNMLoader.qml
     qml/pages/SimpleByDefault.qml

--- a/src/qml/Main.qml
+++ b/src/qml/Main.qml
@@ -133,7 +133,6 @@
             case Private.App.Welcome:
                 _pushPage(_createPage("Welcome.qml"));

-                _pushPage(_createPage("Network.qml"));

                 _pushPage(_createPage("SimpleByDefault.qml"));
                 _pushPage(_createPage("PowerfulWhenNeeded.qml"));
