--- ../cache/vifm-0.9/src/modes/file_info.c	2017-06-18 14:31:25.000000000 +0000
+++ ./src/modes/file_info.c	2017-10-03 23:51:20.005137148 +0000
@@ -31,7 +31,7 @@
 #include <stdlib.h> /* free() */
 #include <string.h> /* strlen() */
 #include <time.h> /* tm localtime() strftime() */
-
+#include <sys/sysmacros.h>
 #include "../cfg/config.h"
 #include "../compat/fs_limits.h"
 #include "../compat/os.h"
