diff -uNr vifm-0.10.1/src/Makefile.am vifm-0.10.1.mod/src/Makefile.am
--- vifm-0.10.1/src/Makefile.am	2019-07-29 00:19:39.000000000 +0300
+++ vifm-0.10.1.mod/src/Makefile.am	2019-08-03 15:03:06.481245003 +0300
@@ -44,7 +44,6 @@
 							 ../data/vifmrc-osx
 dist_pkgdata_SCRIPTS = ../data/vifm-media \
 							 ../data/vifm-media-osx
-nodist_pkgdata_DATA = ../data/vifm-help.txt
 
 bin_PROGRAMS = vifm
 
