--- a/src/platforms/wayland/shm.cpp
+++ b/src/platforms/wayland/shm.cpp
@@ -78,8 +78,9 @@
     const int stride = image.bytesPerLine();
     const int32_t byteCount = image.size().height() * stride;
 
+    int fd = -1;
 #if defined HAVE_MEMFD
-    int fd = memfd_create("kwayland-shared", MFD_CLOEXEC | MFD_ALLOW_SEALING);
+    fd = memfd_create("kwayland-shared", MFD_CLOEXEC | MFD_ALLOW_SEALING);
     if (fd >= 0) {
         fcntl(fd, F_ADD_SEALS, F_SEAL_SHRINK | F_SEAL_SEAL);
     } else
