diff --git a/data/meson.build b/data/meson.build
index fd570b3..4e2e2e8 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -1,7 +1,6 @@
 gnome = import('gnome')
 
 desktop_file = i18n.merge_file(
-  'desktop',
   type: 'desktop',
   input: 'org.gnome.Screenshot.desktop.in',
   output: 'org.gnome.Screenshot.desktop',
@@ -23,7 +22,6 @@ if desktop_file_validate.found()
 endif
 
 metainfo_file = i18n.merge_file(
-  'appdata',
   input: 'org.gnome.Screenshot.metainfo.xml.in',
   output: 'org.gnome.Screenshot.metainfo.xml',
   po_dir: join_paths(meson.current_source_dir(), '../po'),
