https://github.com/termux/termux-packages/issues/15852

--- a/configure
+++ b/configure
@@ -1280,6 +1280,7 @@
 #line 1281 "configure"
 #include "confdefs.h"
 
+int
 main(){return(0);}
 EOF
 if { (eval echo configure:1286: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
@@ -3026,6 +3027,7 @@
 #line 3027 "configure"
 #include "confdefs.h"
 
+void exit();
 int main() {
 extern void foo (); (void) exit (0);
 ; return 0; }
@@ -3154,6 +3156,7 @@
 #line 3155 "configure"
 #include "confdefs.h"
 
+void *memset();
 int main() {
  char *i; int j, k; memset(i, j, k); 
 ; return 0; }
@@ -3186,6 +3189,7 @@
 #line 3187 "configure"
 #include "confdefs.h"
 
+int memcmp();
 int main() {
  char *i, *j; int k; memcmp(i, j, k); 
 ; return 0; }
@@ -3218,6 +3222,7 @@
 #line 3219 "configure"
 #include "confdefs.h"
 
+void *memcpy();
 int main() {
  char *i, *j; int k; memcpy(i, j, k); 
 ; return 0; }
