--- openbox-3.5.0/openbox/client.c    2011-08-01 22:14:58.000000000 +0200
+++ openbox-3.5.0/openbox/client.c    2011-10-22 14:59:56.000000000 +0200
@@ -883,7 +883,7 @@
                  !g_pattern_match(app->role,
                                   strlen(self->role), self->role, NULL))
             match = FALSE;
-        else if (app->title &&
+        else if (app->title && self->title &&
                  !g_pattern_match(app->title,
                                   strlen(self->title), self->title, NULL))
             match = FALSE;
