diff -uNr openbox-3.6.1/tools/gdm-control/gdm-control.c openbox-3.6.1.mod/tools/gdm-control/gdm-control.c
--- openbox-3.6.1/tools/gdm-control/gdm-control.c	2014-11-05 17:19:42.000000000 +0200
+++ openbox-3.6.1.mod/tools/gdm-control/gdm-control.c	2017-11-27 11:43:38.691355227 +0200
@@ -21,8 +21,8 @@
     SWITCHUSER
 } Action;
 
-#define GDM_PROTOCOL_SOCKET_PATH1 "/var/run/gdm_socket"
-#define GDM_PROTOCOL_SOCKET_PATH2 "/tmp/.gdm_socket"
+#define GDM_PROTOCOL_SOCKET_PATH1 "@TERMUX_PREFIX@/var/run/gdm_socket"
+#define GDM_PROTOCOL_SOCKET_PATH2 "@TERMUX_PREFIX@/tmp/.gdm_socket"
 
 #define GDM_PROTOCOL_MSG_CLOSE         "CLOSE"
 #define GDM_PROTOCOL_MSG_VERSION       "VERSION"
