diff -u -r ../cscope-15.8b/src/main.c ./src/main.c
--- ../cscope-15.8b/src/main.c	2014-11-20 22:12:54.000000000 +0100
+++ ./src/main.c	2017-05-22 14:30:00.303282400 +0200
@@ -62,9 +62,9 @@
 #define HOME	"/"	/* no $HOME --> use root directory */
 #define	SHELL	"sh"
 #define LINEFLAG "+%s"	/* default: used by vi and emacs */
-#define TMPDIR	"/tmp"
+#define TMPDIR	"@TERMUX_PREFIX@/tmp"
 #ifndef DFLT_INCDIR
-#define DFLT_INCDIR "/usr/include"
+#define DFLT_INCDIR "@TERMUX_PREFIX@/include"
 #endif
 
 static char const rcsid[] = "$Id: main.c,v 1.57 2014/11/20 21:12:54 broeker Exp $";
