diff -u -r ../rlwrap-0.42/configure ./configure
--- ../rlwrap-0.42/configure	2014-11-14 05:16:59.000000000 -0500
+++ ./configure	2015-05-24 18:45:34.387435144 -0400
@@ -6610,41 +6610,9 @@
 # but they wil fail unless /no is a procfs mountpoint ;-)
 
 echo "Will rlwrap find command's working directory under ${opt_proc_mountpoint}/<commands pid>/cwd? let's see..."
-as_ac_File=`$as_echo "ac_cv_file_$opt_proc_mountpoint/$$/cwd/configure.ac" | $as_tr_sh`
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for $opt_proc_mountpoint/$$/cwd/configure.ac" >&5
-$as_echo_n "checking for $opt_proc_mountpoint/$$/cwd/configure.ac... " >&6; }
-if eval \${$as_ac_File+:} false; then :
-  $as_echo_n "(cached) " >&6
-else
-  test "$cross_compiling" = yes &&
-  as_fn_error $? "cannot check for file existence when cross compiling" "$LINENO" 5
-if test -r "$opt_proc_mountpoint/$$/cwd/configure.ac"; then
-  eval "$as_ac_File=yes"
-else
-  eval "$as_ac_File=no"
-fi
-fi
-eval ac_res=\$$as_ac_File
-	       { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_res" >&5
-$as_echo "$ac_res" >&6; }
-if eval test \"x\$"$as_ac_File"\" = x"yes"; then :
-
-cat >>confdefs.h <<_ACEOF
-#define `$as_echo "HAVE_$opt_proc_mountpoint/$$/cwd/configure.ac" | $as_tr_cpp` 1
-_ACEOF
-
-
-$as_echo "#define HAVE_PROC_PID_CWD 1 " >>confdefs.h
-
                  HAVE_PROC_PID_CWD=yes
 
 
-else
-  HAVE_PROC_PID_CWD=no
-
-fi
-
-
 
 
 cat >>confdefs.h <<_ACEOF
