https://github.com/termux/termux-packages/issues/12282

--- a/src/readline.c
+++ b/src/readline.c
@@ -103,7 +103,7 @@
 				   in .inputrc */
 
   /* put the next variable binding(s) *after* rl_initialize(), so they cannot be overridden */
-  /* rl_variable_bind("enable-bracketed-paste","off"); */     /* enable-bracketed-paste changes cursor positioning after printing the prompt ...
+  rl_variable_bind("enable-bracketed-paste","off");     /* enable-bracketed-paste changes cursor positioning after printing the prompt ...
                                                            ... causing rlwrap to overwrite it after accepting input                        */
  
   using_history();
