diff -uNr sqlcipher-4.5.2/src/crypto.h sqlcipher-4.5.2.mod/src/crypto.h
--- sqlcipher-4.5.2/src/crypto.h	2022-07-28 03:23:29.000000000 +0800
+++ sqlcipher-4.5.2.mod/src/crypto.h	2022-08-04 09:04:30.948978453 +0800
@@ -40,10 +40,6 @@
 #include "pager.h"
 #include "vdbeInt.h"
 
-#ifdef __ANDROID__
-#include <android/log.h>
-#endif
-
 #include <time.h>
 
 #if defined(_WIN32) || defined(SQLITE_OS_WINRT)
diff -uNr sqlcipher-4.5.2/src/crypto_impl.c sqlcipher-4.5.2.mod/src/crypto_impl.c
--- sqlcipher-4.5.2/src/crypto_impl.c	2022-07-28 03:23:29.000000000 +0800
+++ sqlcipher-4.5.2.mod/src/crypto_impl.c	2022-08-04 09:06:21.678978411 +0800
@@ -1648,11 +1648,6 @@
   FILE *f = (FILE*) file;
   char *fmt = "Elapsed time:%.3f ms - %s\n";
   double elapsed = (*((sqlite3_uint64*)run_time))/1000000.0;
-#ifdef __ANDROID__
-  if(f == NULL) {
-    __android_log_print(ANDROID_LOG_DEBUG, "sqlcipher", fmt, elapsed, sqlite3_sql((sqlite3_stmt*)stmt));
-  }
-#endif
   if(f) fprintf(f, fmt, elapsed, sqlite3_sql((sqlite3_stmt*)stmt));
   return SQLITE_OK;
 }
@@ -1702,13 +1697,9 @@
   va_start(params, message);
 
 #ifdef CODEC_DEBUG
-#ifdef __ANDROID__
-    __android_log_vprint(ANDROID_LOG_DEBUG, "sqlcipher", message, params);
-#else
     vfprintf(stderr, message, params);
     fprintf(stderr, "\n");
 #endif
-#endif
 
   if(level > sqlcipher_log_level || (sqlcipher_log_logcat == 0 && sqlcipher_log_file == NULL)) {
     /* no log target or tag not in included filters */
@@ -1740,11 +1731,7 @@
       fprintf((FILE*)sqlcipher_log_file, "\n");
     }
   }
-#ifdef __ANDROID__
-  if(sqlcipher_log_logcat) {
-    __android_log_vprint(ANDROID_LOG_DEBUG, "sqlcipher", message, params);
-  }
-#endif
+
 end:
   va_end(params);
 }
