--- src/src/plugins/sensors/dummy/dummy.pro	2020-10-15 06:53:19.000000000 +0000
+++ src.mod/src/plugins/sensors/dummy/dummy.pro	2021-05-14 14:22:51.194482639 +0000
@@ -12,8 +12,6 @@
 
 OTHER_FILES = plugin.json
 
-unix:!darwin:!qnx:!android:!openbsd: LIBS += -lrt
-
 PLUGIN_TYPE = sensors
 PLUGIN_CLASS_NAME = dummySensorPlugin
 load(qt_plugin)
--- src/src/plugins/sensors/linux/linux.pro	2020-10-15 06:53:19.000000000 +0000
+++ src.mod/src/plugins/sensors/linux/linux.pro	2021-05-14 14:22:09.604788205 +0000
@@ -3,7 +3,6 @@
 
 OTHER_FILES = plugin.json
 
-!android:LIBS += -lrt
 HEADERS += linuxsysaccelerometer.h
 SOURCES += linuxsysaccelerometer.cpp \
 main.cpp
--- src/src/plugins/sensors/iio-sensor-proxy/iio-sensor-proxy.pro	2020-10-15 06:53:19.000000000 +0000
+++ src.mod/src/plugins/sensors/iio-sensor-proxy/iio-sensor-proxy.pro	2021-05-14 14:28:09.397847029 +0000
@@ -5,7 +5,6 @@
 PLUGIN_CLASS_NAME = IIOSensorProxySensorPlugin
 load(qt_plugin)
 
-!android:LIBS += -lrt
 HEADERS += iiosensorproxysensorbase.h \
     iiosensorproxylightsensor.h \
     iiosensorproxyorientationsensor.h \
