diff -u -r ../elfutils-0.168/config.h.in ./config.h.in
--- ../elfutils-0.168/config.h.in	2016-12-28 10:33:20.000000000 +0100
+++ ./config.h.in	2017-07-09 23:32:21.305290105 +0200
@@ -1,4 +1,7 @@
 /* config.h.in.  Generated from configure.ac by autoheader.  */
+#include <libgen.h> /* for basename(3) */
+#include <stdio_ext.h> /* for fputc_unlocked(3) */
+#define DL_CALL_FCT(f, args) ((*(f)) args)
 
 /* Building with -fsanitize=undefined or not */
 #undef CHECK_UNDEFINED
@@ -117,4 +120,14 @@
 /* Define for large files, on AIX-style hosts. */
 #undef _LARGE_FILES
 
+#ifdef __ANDROID__
+# ifndef ANDROID_MEMPCPY
+# define ANDROID_MEMPCPY 1
+static __inline__ void* mempcpy(void* dest, void const* src, size_t n)
+{
+      return memcpy(dest, src, n) + n;
+}
+# endif
+#endif
+
 #include <eu-config.h>
