--- ./libelf/Makefile.am.orig	2019-05-29 05:19:01.983292465 +0000
+++ ./libelf/Makefile.am	2019-05-29 05:19:19.759573256 +0000
@@ -62,7 +62,7 @@
 		   gelf_update_ehdr.c \
 		   elf32_getphdr.c elf64_getphdr.c gelf_getphdr.c \
 		   elf32_newphdr.c elf64_newphdr.c gelf_newphdr.c \
-		   gelf_update_phdr.c \
+		   gelf_update_phdr.c aligned_alloc.c \
 		   elf_getarhdr.c elf_getarsym.c \
 		   elf_rawfile.c elf_readall.c elf_cntl.c \
 		   elf_getscn.c elf_nextscn.c elf_ndxscn.c elf_newscn.c \
