diff -u -r ../elfutils-0.166/src/strings.c ./src/strings.c
--- ../elfutils-0.166/src/strings.c	2015-11-27 08:36:29.000000000 -0500
+++ ./src/strings.c	2016-05-04 14:19:55.593602311 -0400
@@ -494,7 +494,6 @@
       if (mem != MAP_FAILED)
 	{
 	  /* We will go through the mapping sequentially.  */
-	  (void) posix_madvise (mem, map_size, POSIX_MADV_SEQUENTIAL);
 	  break;
 	}
       if (errno != EINVAL && errno != ENOMEM)
