--- ./texk/web2c/pdftexdir/pdftex.web.orig	2021-06-25 11:05:46.198310283 +0000
+++ ./texk/web2c/pdftexdir/pdftex.web	2021-06-25 11:06:42.138353401 +0000
@@ -10932,7 +10932,7 @@
 @!save_scanner_status:small_number; {|scanner_status| upon entry}
 @!save_def_ref: pointer; {|def_ref| upon entry, important if inside `\.{\\message}'}
 @!save_warning_index: pointer;
-@!bool: boolean; {temp boolean}
+@!mybool: boolean; {temp boolean}
 @!i: integer; {first temp integer}
 @!j: integer; {second temp integer}
 @!b:pool_pointer; {base of temporary string}
@@ -11132,7 +11132,7 @@
     save_warning_index := warning_index;
     save_def_ref := def_ref;
     save_cur_string;
-    bool := scan_keyword("file");
+    mybool := scan_keyword("file");
     scan_pdf_ext_toks;
     s := tokens_to_string(def_ref);
     delete_token_ref(def_ref);
@@ -11140,7 +11140,7 @@
     warning_index := save_warning_index;
     scanner_status := save_scanner_status;
     b := pool_ptr;
-    getmd5sum(s, bool);
+    getmd5sum(s, mybool);
     link(garbage) := str_toks(b);
     flush_str(s);
     ins_list(link(temp_head));
@@ -11203,7 +11203,7 @@
     save_def_ref := def_ref;
     save_cur_string;
     {scan for icase}
-    bool := scan_keyword("icase");
+    mybool := scan_keyword("icase");
     {scan for subcount}
     i := -1; {default for subcount}
     if scan_keyword("subcount") then begin
@@ -11220,7 +11220,7 @@
     warning_index := save_warning_index;
     scanner_status := save_scanner_status;
     b := pool_ptr;
-    matchstrings(s, t, i, bool);
+    matchstrings(s, t, i, mybool);
     link(garbage) := str_toks(b);
     flush_str(t);
     flush_str(s);
@@ -11259,7 +11259,7 @@
   end;
 pdf_colorstack_init_code:
   begin
-    bool := scan_keyword("page");
+    mybool := scan_keyword("page");
     if scan_keyword("direct") then
         cur_val := direct_always
     else
@@ -11277,7 +11277,7 @@
     def_ref := save_def_ref;
     warning_index := save_warning_index;
     scanner_status := save_scanner_status;
-    cur_val := newcolorstack(s, cur_val, bool);
+    cur_val := newcolorstack(s, cur_val, mybool);
     flush_str(s);
     cur_val_level := int_val;
     if cur_val < 0 then begin
