--- ./texk/web2c/luatexdir/luaffi/ffi.h	2017-11-17 14:32:47.517562104 +0000
+++ ../ffi.h	2017-11-17 14:32:26.333562953 +0000
@@ -42,7 +42,7 @@
 #include <sys/mman.h>
 #endif
 
-#if ( defined( _WIN32) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__bsdi__) || defined(__DragonFly__))
+#if ( defined( _WIN32) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__bsdi__) || defined(__DragonFly__) || defined(__ANDROID__))
 /* We should include something equivalent to */
 /* complex.h                                 */
 #else
