diff -u -r ../texlive-source-texlive-2018.2.orig/texk/web2c/xetexdir/pdfimage.cpp ./texk/web2c/xetexdir/pdfimage.cpp
--- ../texlive-source-texlive-2018.2.orig/texk/web2c/xetexdir/pdfimage.cpp	2018-12-23 10:51:16.249652209 +0100
+++ ./texk/web2c/xetexdir/pdfimage.cpp	2018-12-23 12:00:04.419369459 +0100
@@ -82,19 +82,19 @@
 	switch (pdf_box) {
 		default:
 		case pdfbox_crop:
-			r = page->getCropBox();
+			r = (PDFRectangle *)page->getCropBox();
 			break;
 		case pdfbox_media:
-			r = page->getMediaBox();
+			r = (PDFRectangle *)page->getMediaBox();
 			break;
 		case pdfbox_bleed:
-			r = page->getBleedBox();
+			r = (PDFRectangle *)page->getBleedBox();
 			break;
 		case pdfbox_trim:
-			r = page->getTrimBox();
+			r = (PDFRectangle *)page->getTrimBox();
 			break;
 		case pdfbox_art:
-			r = page->getArtBox();
+			r = (PDFRectangle *)page->getArtBox();
 			break;
 	}
 
