From 75547149552c9027478ef5cc8b27adf729a1e1fc Mon Sep 17 00:00:00 2001
From: Henrik Grimler <grimler@termux.dev>
Date: Mon, 13 Feb 2023 13:49:47 +0100
Subject: [PATCH 32/37] make: cflags: handle arm like other arches

---
 make/autoconf/flags-cflags.m4 | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/make/autoconf/flags-cflags.m4 b/make/autoconf/flags-cflags.m4
index f8fbe14cc..4cfc77586 100644
--- a/make/autoconf/flags-cflags.m4
+++ b/make/autoconf/flags-cflags.m4
@@ -66,13 +66,7 @@ AC_DEFUN([FLAGS_SETUP_SHARED_LIBS],
       SET_SHARED_LIBRARY_NAME='-Wl,-soname=[$]1'
       SET_SHARED_LIBRARY_MAPFILE='-Wl,-version-script=[$]1'
 
-      # arm specific settings
-      if test "x$OPENJDK_TARGET_CPU" = "xarm"; then
-        # '-Wl,-z,origin' isn't used on arm.
-        SET_SHARED_LIBRARY_ORIGIN='-Wl,-rpath,\$$$$ORIGIN[$]1'
-      else
-        SET_SHARED_LIBRARY_ORIGIN="-Wl,-z,origin $SET_EXECUTABLE_ORIGIN"
-      fi
+      SET_SHARED_LIBRARY_ORIGIN="-Wl,-z,origin $SET_EXECUTABLE_ORIGIN"
     fi
 
   elif test "x$TOOLCHAIN_TYPE" = xxlc; then
-- 
2.45.2

