diff -u -r ../src/make/autoconf/flags-cflags.m4 ./make/autoconf/flags-cflags.m4
--- ../src/make/autoconf/flags-cflags.m4	2024-09-01 20:09:35.611261408 +0000
+++ ./make/autoconf/flags-cflags.m4	2024-09-01 20:10:49.387795908 +0000
@@ -66,9 +66,6 @@
       # Default works for linux, might work on other platforms as well.
       SHARED_LIBRARY_FLAGS='-shared'
       SET_EXECUTABLE_ORIGIN='-Wl,-rpath,\$$ORIGIN[$]1'
-      if test "x$OPENJDK_TARGET_OS" = xlinux; then
-        SET_EXECUTABLE_ORIGIN="$SET_EXECUTABLE_ORIGIN -Wl,--disable-new-dtags"
-      fi
       SET_SHARED_LIBRARY_NAME='-Wl,-soname=[$]1'
       SET_SHARED_LIBRARY_MAPFILE='-Wl,-version-script=[$]1'
 
