# TODO: Remove this after new Ubuntu provides autoconf 2.72

--- a/configure.ac
+++ b/configure.ac
@@ -1,5 +1,5 @@
 dnl Process this file with autoconf to produce a configure script.
-AC_PREREQ([2.72])
+AC_PREREQ([2.71])
 AC_INIT([alsa-lib], [1.2.16])
 
 AC_CONFIG_SRCDIR([src/control/control.c])
@@ -31,7 +31,7 @@ AC_PROG_CC
 AC_PROG_CPP
 AC_USE_SYSTEM_EXTENSIONS
 AC_PROG_INSTALL
-AC_PROG_LN_S 
+AC_PROG_LN_S
 AC_DISABLE_STATIC
 _LT_SET_OPTION([LT_INIT],[dlopen])
 LT_INIT
