--- ../frida-python-src-meson.build.orig	2020-03-25 07:54:47.948054199 +0100
+++ ./frida-python/src/meson.build	2020-03-25 08:44:34.189059770 +0100
@@ -9,8 +9,8 @@
   name_prefix: '',
   name_suffix: 'so',
   c_args: frida_component_cflags,
-  include_directories: include_directories(python_incdir),
-  link_args: extra_link_args,
+  include_directories: include_directories('@TERMUX_PREFIX@/include', python_incdir),
+  link_args: extra_link_args + ['../frida-core/lib/selinux/libfrida-selinux.a'],
   dependencies: [frida_core_dep],
   install: true,
   install_dir: python_site_packages,
