--- ./releng/env_android.py	2025-07-11 09:11:03.507422814 +0000
+++ ./releng/env_android.py.mod	2025-07-11 09:11:59.628425590 +0000
@@ -41,7 +41,7 @@
 
     android_build_os = "darwin" if build_machine.os == "macos" else build_machine.os
     android_build_arch = "x86_64" if build_machine.os in {"macos", "linux"} else build_machine.arch
-    android_api = 19 if machine.arch in {"x86", "arm"} else 21
+    android_api = @TERMUX_PKG_API_LEVEL@
 
     llvm_bindir = ndk_root / "toolchains" / "llvm" / "prebuilt" / f"{android_build_os}-{android_build_arch}" / "bin"
 
@@ -111,7 +111,7 @@
     pass
 
 
-NDK_REQUIRED = 25
+NDK_REQUIRED = @TERMUX_NDK_VERSION_NUM@
 
 NDK_BINARIES = [
     ("c",       "clang"),
