--- ./releng/setup-env.sh.orig	2021-12-05 13:14:56.373457925 +0100
+++ ./releng/setup-env.sh	2021-12-05 13:16:27.716458982 +0100
@@ -123,7 +123,7 @@
 fi
 
 if [ "$host_os" == "android" ]; then
-  ndk_required=24
+  ndk_required=23
   if [ -n "$ANDROID_NDK_ROOT" ]; then
     if [ -f "$ANDROID_NDK_ROOT/source.properties" ]; then
       ndk_installed_version=$(grep Pkg.Revision "$ANDROID_NDK_ROOT/source.properties" | awk '{ split($NF, v, "."); print v[1]; }')
