--- ./releng/env_generic.py.orig	2025-07-11 09:03:53.705401553 +0000
+++ ./releng/env_generic.py	2025-07-11 09:04:19.281402818 +0000
@@ -266,7 +266,7 @@
         return "gnu-ld"
     if "GNU gold " in linker_version:
         return "gnu-gold"
-    if linker_version.startswith("LLD "):
+    if linker_version.startswith("LLD ") or "compatible with GNU linkers" in linker_version:
         return "lld"
     if linker_version.startswith("ld: "):
         return "apple"
