diff -uNr wordgrinder-0.7.2/Makefile wordgrinder-0.7.2.mod/Makefile
--- wordgrinder-0.7.2/Makefile	2018-11-29 21:12:27.000000000 +0200
+++ wordgrinder-0.7.2.mod/Makefile	2019-01-28 20:03:20.720358428 +0200
@@ -86,7 +86,7 @@
 
 # Do you want your binaries stripped on installation?
 
-WANT_STRIPPED_BINARIES ?= yes
+WANT_STRIPPED_BINARIES ?= no
 
 # ===========================================================================
 #                       END OF CONFIGURATION OPTIONS
