diff -uNr rc-1.7.4/Makefile.in rc-1.7.4.mod/Makefile.in
--- rc-1.7.4/Makefile.in	2015-05-14 00:26:18.000000000 +0300
+++ rc-1.7.4.mod/Makefile.in	2020-01-01 19:24:41.726859279 +0200
@@ -438,13 +438,13 @@
 	@rm -f history$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(history_OBJECTS) $(history_LDADD) $(LIBS)
 
-mksignal$(EXEEXT): $(mksignal_OBJECTS) $(mksignal_DEPENDENCIES) $(EXTRA_mksignal_DEPENDENCIES) 
+mksignal$(EXEEXT): $(mksignal_SOURCES) $(mksignal_DEPENDENCIES) $(EXTRA_mksignal_DEPENDENCIES) 
 	@rm -f mksignal$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(mksignal_OBJECTS) $(mksignal_LDADD) $(LIBS)
+	$(AM_V_CCLD)gcc $(mksignal_SOURCES) $(mksignal_LDADD) $(LIBS) -o mksignal$(EXEEXT)
 
 mkstatval$(EXEEXT): $(mkstatval_OBJECTS) $(mkstatval_DEPENDENCIES) $(EXTRA_mkstatval_DEPENDENCIES) 
 	@rm -f mkstatval$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(mkstatval_OBJECTS) $(mkstatval_LDADD) $(LIBS)
+	$(AM_V_CCLD)gcc $(mkstatval_SOURCES) $(mkstatval_LDADD) $(LIBS) -o mkstatval$(EXEEXT)
 
 rc$(EXEEXT): $(rc_OBJECTS) $(rc_DEPENDENCIES) $(EXTRA_rc_DEPENDENCIES) 
 	@rm -f rc$(EXEEXT)
