diff -uNr rc-1.7.4/history.c rc-1.7.4.mod/history.c
--- rc-1.7.4/history.c	2015-04-03 11:07:24.000000000 +0300
+++ rc-1.7.4.mod/history.c	2020-01-01 19:16:57.983948748 +0200
@@ -338,7 +338,7 @@
 		if (!editit)
 			fprintf(stderr, "%s\n", s);
 		if (shell == NULL)
-			shell = "/bin/sh";
+			shell = "@TERMUX_PREFIX@/bin/sh";
 		execl(shell, rc_basename(shell), "-c", s, NULL);
 		perror(shell);
 		exit(1);
