diff -u -r ../PRoot-5.1.0/src/ptrace/wait.h ./src/ptrace/wait.h
--- ../PRoot-5.1.0/src/ptrace/wait.h	2014-12-15 09:18:11.000000000 -0500
+++ ./src/ptrace/wait.h	2016-01-08 09:07:20.342650218 -0500
@@ -25,6 +25,8 @@
 
 #include "tracee/tracee.h"
 
+#include <sys/wait.h> /* for __WALL */
+
 extern int translate_wait_enter(Tracee *ptracer);
 extern int translate_wait_exit(Tracee *ptracer);
 extern bool handle_ptracee_event(Tracee *ptracee, int wait_status);
