diff -u -r ../PRoot-5.1.0/src/tracee/tracee.c ./src/tracee/tracee.c
--- ../PRoot-5.1.0/src/tracee/tracee.c	2014-12-15 09:18:11.000000000 -0500
+++ ./src/tracee/tracee.c	2016-01-08 19:32:43.493157624 -0500
@@ -538,6 +538,9 @@
 			/* Sanity check.  */
 			assert(!child->as_ptracee.tracing_started);
 
+#ifndef __W_STOPCODE
+#define __W_STOPCODE(sig) ((sig) << 8 | 0x7f)
+#endif
 			keep_stopped = handle_ptracee_event(child, __W_STOPCODE(SIGSTOP));
 
 			/* Note that this event was already handled by
