From https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=788113

diff -u -r ../PRoot-5.1.0/src/arch.h ./src/arch.h
--- ../PRoot-5.1.0/src/arch.h	2014-12-15 09:18:11.000000000 -0500
+++ ./src/arch.h	2016-01-08 19:53:23.789378336 -0500
@@ -125,6 +125,9 @@
     #define OFFSETOF_STAT_UID_32 0
     #define OFFSETOF_STAT_GID_32 0
 
+    #define EXEC_PIC_ADDRESS   0x500000000000
+    #define INTERP_PIC_ADDRESS 0x6f0000000000
+
 #elif defined(ARCH_X86)
 
     #define SYSNUMS_HEADER1 "syscall/sysnums-i386.h"
