--- readline-8.0/rlconf.h	2017-02-21 20:55:30.000000000 +0530
+++ readline-8.0-mod/rlconf.h	2019-02-20 20:45:45.583654990 +0530
@@ -43,7 +43,7 @@
 #define DEFAULT_INPUTRC "~/.inputrc"
 
 /* The ultimate last-ditch filename for an init file -- system-wide. */
-#define SYS_INPUTRC "/etc/inputrc"
+#define SYS_INPUTRC "@TERMUX_PREFIX@/etc/inputrc"
 
 /* If defined, expand tabs to spaces. */
 #define DISPLAY_TABS
