--- ../readline-6.2/rlconf.h	2009-01-04 20:32:33.000000000 +0100
+++ ./rlconf.h	2014-02-04 18:30:16.000000000 +0100
@@ -39,7 +39,7 @@
 #define DEFAULT_INPUTRC "~/.inputrc"
 
 /* The ultimate last-ditch filenname for an init file -- system-wide. */
-#define SYS_INPUTRC "/etc/inputrc"
+#define SYS_INPUTRC "@TERMUX_PREFIX@/etc/inputrc"
 
 /* If defined, expand tabs to spaces. */
 #define DISPLAY_TABS
