diff --git a/cmake/Helper.cmake b/cmake/Helper.cmake
index 992e469..4075621 100644
--- a/cmake/Helper.cmake
+++ b/cmake/Helper.cmake
@@ -37,13 +37,6 @@ else()
     -Wextra
   )
 
-  if(NOT WASMEDGE_PLUGIN_WASI_NN_GGML_LLAMA_CUBLAS)
-    list(APPEND WASMEDGE_CFLAGS
-      -Werror
-      -Wno-error=pedantic
-    )
-  endif()
-
   if(WASMEDGE_ENABLE_UB_SANITIZER)
     list(APPEND WASMEDGE_CFLAGS -fsanitize=undefined)
   endif()
