diff -uNr WasmEdge/lib/host/wasi/linux.h WasmEdge.mod/lib/host/wasi/linux.h
--- WasmEdge/lib/host/wasi/linux.h	2022-07-28 18:56:12.000000000 +0800
+++ WasmEdge.mod/lib/host/wasi/linux.h	2022-08-15 01:48:53.044855896 +0800
@@ -319,7 +319,11 @@
   }
 }
 
+#if !defined __ANDROID__ || defined __aarch64__ || defined __x86_64__
 inline constexpr __wasi_filetype_t fromFileType(mode_t Mode) noexcept {
+#else
+inline constexpr __wasi_filetype_t fromFileType(unsigned int Mode) noexcept {
+#endif
   switch (Mode & S_IFMT) {
   case S_IFBLK:
     return __WASI_FILETYPE_BLOCK_DEVICE;
