diff -u -r ../ocrad-0.23/configure ./configure
--- ../ocrad-0.23/configure	2014-03-10 21:19:33.000000000 +0100
+++ ./configure	2014-08-03 03:00:32.678020824 +0200
@@ -16,7 +16,7 @@
 LC_ALL=C
 export LC_ALL
 srcdir=
-prefix=/usr/local
+prefix=@TERMUX_PREFIX@
 exec_prefix='$(prefix)'
 bindir='$(exec_prefix)/bin'
 datarootdir='$(prefix)/share'
@@ -24,10 +24,6 @@
 infodir='$(datarootdir)/info'
 libdir='${exec_prefix}/lib'
 mandir='$(datarootdir)/man'
-CXX=g++
-CPPFLAGS=
-CXXFLAGS='-Wall -W -O2'
-LDFLAGS=
 
 # checking whether we are using GNU C++.
 ${CXX} --version > /dev/null 2>&1
