--- a/debian/move-subset.py
+++ b/debian/move-subset.py
@@ -40,7 +40,7 @@
         for ext in EXTS:
             wanted[name + ext] = name
 
-    for dirpath, dirnames, filenames in os.walk(dest / 'usr/share/icons'):
+    for dirpath, dirnames, filenames in os.walk(dest / 'share/icons'):
         rel = Path(dirpath).relative_to(dest)
 
         if 'cursors' in dirnames:
@@ -51,7 +51,7 @@
             # remove -rtl or -symbolic
             already_have[Path(Path(name).stem).stem] = rel / name
 
-    for dirpath, dirnames, filenames in os.walk(source / 'usr/share/icons'):
+    for dirpath, dirnames, filenames in os.walk(source / 'share/icons'):
         rel = Path(dirpath).relative_to(source)
 
         for name in filenames:
