diff -ur orig/Source/Audio/openal_wrapper.cpp mod/Source/Audio/openal_wrapper.cpp
--- orig/Source/Audio/openal_wrapper.cpp	2017-02-11 20:09:37.000000000 +0100
+++ mod/Source/Audio/openal_wrapper.cpp	2025-07-06 10:30:30.359535461 +0200
@@ -316,7 +316,7 @@
     // !!! FIXME: if it's not Ogg, we don't have a decoder. I'm lazy.  :/
     char* fname = (char*)alloca(strlen(_fname) + 16);
     strcpy(fname, _fname);
-    char* ptr = strchr(fname, '.');
+    char* ptr = strrchr(fname, '.');
     if (ptr) {
         *ptr = '\0';
     }
