diff --git a/include/llvm/Config/config.h b/include/llvm/Config/config.h
index 9cf1c89..9af2945 100644
--- a/include/llvm/Config/config.h
+++ b/include/llvm/Config/config.h
@@ -14,7 +14,9 @@
 #define ENABLE_CRASH_OVERRIDES 1
 
 /* Define to 1 if you have the `backtrace' function. */
+#if !defined(__ANDROID__)
 #define HAVE_BACKTRACE TRUE
+#endif
 
 #define BACKTRACE_HEADER <execinfo.h>
 
@@ -74,7 +76,11 @@
 /* #undef HAVE_FFI_H */
 
 /* Define to 1 if you have the `futimens' function. */
+#if defined(__ANDROID__)
+#define HAVE_FUTIMENS 1
+#else
 /* #undef HAVE_FUTIMENS */
+#endif
 
 /* Define to 1 if you have the `futimes' function. */
 #define HAVE_FUTIMES 1
@@ -104,7 +110,9 @@
 #define HAVE_LIBPTHREAD 1
 
 /* Define to 1 if you have the `pthread_getname_np' function. */
+#if !defined(__ANDROID__)
 #define HAVE_PTHREAD_GETNAME_NP 1
+#endif
 
 /* Define to 1 if you have the `pthread_setname_np' function. */
 #define HAVE_PTHREAD_SETNAME_NP 1
