--- ./lib/vasnprintf.c~	2019-12-12 20:53:07.000000000 +0100
+++ ./lib/vasnprintf.c	2022-01-06 08:58:41.897056563 +0100
@@ -4872,7 +4872,8 @@
 # if ! (((__GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 3))        \
          && !defined __UCLIBC__)                                            \
         || (defined __APPLE__ && defined __MACH__)                          \
-        || (defined _WIN32 && ! defined __CYGWIN__))
+        || (defined _WIN32 && ! defined __CYGWIN__)                         \
+        || defined __ANDROID__)
                 fbp[1] = '%';
                 fbp[2] = 'n';
                 fbp[3] = '\0';
