--- a/src/meson.build
+++ b/src/meson.build
@@ -1,8 +1,6 @@
 
 waypipe_source_files = ['dmabuf.c', 'handlers.c', 'kernel.c', 'mainloop.c', 'parsing.c', 'platform.c', 'shadow.c', 'interval.c', 'util.c', 'video.c']
 waypipe_deps = [
-	pthreads,        # To run expensive computations in parallel
-	rt,              # For shared memory
 ]
 if config_data.has('HAS_DMABUF')
 	# General GPU buffer creation, aligned with dmabuf proto
