--- a/src/meson.build
+++ b/src/meson.build
@@ -1,8 +1,6 @@
 
 waypipe_source_files = ['dmabuf.c', 'handlers.c', 'kernel.c', 'mainloop.c', 'parsing.c', 'platform.c', 'shadow.c', 'interval.c', 'util.c', 'video.c']
 waypipe_deps = [
-	pthreads,        # To run expensive computations in parallel
-	rt,              # For shared memory
 ]
 if config_data.has('HAS_DMABUF')
 	# General GPU buffer creation, aligned with dmabuf proto
@@ -53,7 +51,7 @@
 	config_data.set('HAVE_NEON', 1, description: 'Compiler supports NEON')
 endif
 if config_data.has('HAS_SECURITY_CONTEXT')
-	wayland_scanner_prog = find_program(wayland_scanner.get_variable(pkgconfig: 'wayland_scanner'))
+	wayland_scanner_prog = find_program('wayland-scanner')
 	wl_security_context_xml = '../protocols/security-context-v1.xml'
 	protocols_src += custom_target(
 		'security-context-v1-protocol.c',
