diff -u -r ../libqalculate-2.0.0/libqalculate/Calculator.cc ./libqalculate/Calculator.cc
--- ../libqalculate-2.0.0/libqalculate/Calculator.cc	2017-08-27 21:38:22.000000000 +0200
+++ ./libqalculate/Calculator.cc	2017-08-31 16:09:58.987236004 +0200
@@ -9360,7 +9360,7 @@
 	char filename[MAX_PATH]; 
 	return SearchPath(NULL, "gnuplot", ".exe", MAX_PATH, filename, &lpFilePart);
 #else
-	FILE *pipe = popen("gnuplot -", "w");
+	FILE *pipe = popen("gnuplot - 2> /dev/null", "w");
 	if(!pipe) return false;
 	return pclose(pipe) == 0;
 #endif
