--- ../../build/qalc/cache/libqalculate-0.9.9/libqalculate/Number.cc	2016-06-02 18:40:18.000000000 +0200
+++ ./libqalculate/Number.cc	2016-08-15 00:35:08.556225106 +0200
@@ -2708,7 +2708,7 @@
 				} catch(runtime_exception &e) {
 					CALCULATOR->error(true, _("CLN Exception: %s"), e.what());
 				}
-				pthread_testcancel();
+				//pthread_testcancel();
 				if(b_zero) {
 					try {
 						inr = cln::exquo(inr, last_prime);
