diff -u -r ../nginx-1.14.0/auto/threads ./auto/threads
--- ../nginx-1.14.0/auto/threads	2018-04-17 15:22:36.000000000 +0000
+++ ./auto/threads	2018-05-19 01:20:25.703633445 +0000
@@ -16,6 +16,5 @@
     have=NGX_THREADS . auto/have
     CORE_DEPS="$CORE_DEPS $THREAD_POOL_DEPS"
     CORE_SRCS="$CORE_SRCS $THREAD_POOL_SRCS"
-    CORE_LIBS="$CORE_LIBS -lpthread"
-    NGX_LIBPTHREAD="-lpthread"
+    NGX_LIBPTHREAD=""
 fi
