diff -uNr pypy3.6-v7.3.2-src/pypy/tool/release/package.py pypy3.6-v7.3.2-src.mod/pypy/tool/release/package.py
--- pypy3.6-v7.3.2-src/pypy/tool/release/package.py     2020-09-23 15:02:22.000000000 +0800
+++ pypy3.6-v7.3.2-src.mod/pypy/tool/release/package.py 2021-12-30 23:25:59.068175100 +0800
@@ -35,7 +35,7 @@
 ARCH = get_arch()


-USE_ZIPFILE_MODULE = ARCH == 'win32'
+USE_ZIPFILE_MODULE = True

 STDLIB_VER = "3"
