diff -uNr pypy2.7-v7.3.6-src/lib-python/2.7/uuid.py pypy2.7-v7.3.6-src.mod/lib-python/2.7/uuid.py
--- pypy2.7-v7.3.6-src/lib-python/2.7/uuid.py           2021-10-03 14:36:11.000000000 +0800
+++ pypy2.7-v7.3.6-src.mod/lib-python/2.7/uuid.py       2022-01-18 00:06:31.330416000 +0800
@@ -314,7 +314,7 @@
 def _popen(command, args):
     import os
     path = os.environ.get("PATH", os.defpath).split(os.pathsep)
-    path.extend(('/sbin', '/usr/sbin'))
+    path.extend(('@TERMUX_PREFIX@/bin',))
     for dir in path:
         executable = os.path.join(dir, command)
         if (os.path.exists(executable) and
