diff -uNr pypy2.7-v7.3.6-src/lib-python/2.7/mailcap.py pypy2.7-v7.3.6-src.mod/lib-python/2.7/mailcap.py
--- pypy2.7-v7.3.6-src/lib-python/2.7/mailcap.py        2021-10-03 14:36:11.000000000 +0800
+++ pypy2.7-v7.3.6-src.mod/lib-python/2.7/mailcap.py    2022-01-17 23:57:37.136118200 +0800
@@ -44,7 +44,8 @@
             # Don't bother with getpwuid()
             home = '.' # Last resort
         mailcaps = [home + '/.mailcap', '/etc/mailcap',
-                '/usr/etc/mailcap', '/usr/local/etc/mailcap']
+                '/usr/etc/mailcap', '/usr/local/etc/mailcap',
+                '@TERMUX_PREFIX@/etc/mailcap']
     return mailcaps


