diff -uNr pypy2.7-v7.3.6-src/lib-python/2.7/subprocess.py pypy2.7-v7.3.6-src.mod/lib-python/2.7/subprocess.py
--- pypy2.7-v7.3.6-src/lib-python/2.7/subprocess.py     2021-10-03 14:36:11.000000000 +0800
+++ pypy2.7-v7.3.6-src.mod/lib-python/2.7/subprocess.py 2022-01-17 22:21:06.651605000 +0800
@@ -941,7 +941,7 @@
                 args = list(args)

             if shell:
-                args = ["/bin/sh", "-c"] + args
+                args = ["@TERMUX_PREFIX@/bin/sh", "-c"] + args
                 if executable:
                     args[0] = executable
