diff -uNr pypy3.6-v7.3.2-src/lib_pypy/_pwdgrp_build.py pypy3.6-v7.3.2-src.mod/lib_pypy/_pwdgrp_build.py
--- pypy3.6-v7.3.2-src/lib_pypy/_pwdgrp_build.py        2020-09-23 15:02:22.000000000 +0800
+++ pypy3.6-v7.3.2-src.mod/lib_pypy/_pwdgrp_build.py    2021-11-09 18:32:36.933096400 +0800
@@ -35,9 +35,9 @@
 struct passwd *getpwuid(uid_t uid);
 struct passwd *getpwnam(const char *name);

-struct passwd *getpwent(void);
-void setpwent(void);
-void endpwent(void);
+// struct passwd *getpwent(void);
+// void setpwent(void);
+// void endpwent(void);

 struct group *getgrgid(gid_t gid);
 struct group *getgrnam(const char *name);
