diff -uNr pypy3.6-v7.3.2-src/pypy/module/pwd/moduledef.py pypy3.6-v7.3.2-src.mod/pypy/module/pwd/moduledef.py
--- pypy3.6-v7.3.2-src/pypy/module/pwd/moduledef.py     2020-09-23 15:02:22.000000000 +0800
+++ pypy3.6-v7.3.2-src.mod/pypy/module/pwd/moduledef.py 2021-11-09 18:31:05.132729100 +0800
@@ -15,11 +15,11 @@
     interpleveldefs = {
         'getpwuid': 'interp_pwd.getpwuid',
         'getpwnam': 'interp_pwd.getpwnam',
-        'getpwall': 'interp_pwd.getpwall',
+        # 'getpwall': 'interp_pwd.getpwall',
     }

     appleveldefs = {
         'struct_passwd': 'app_pwd.struct_passwd',
-        'struct_pwent': 'app_pwd.struct_passwd',
+        # 'struct_pwent': 'app_pwd.struct_passwd',
     }
