diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 30d97cb9..341e782e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -335,7 +335,7 @@ if (USE_PYTHON)
       $<TARGET_FILE:libledger> "${CMAKE_BINARY_DIR}/${_ledger_python_module_name}")
     install(
       FILES "${CMAKE_BINARY_DIR}/${_ledger_python_module_name}"
-      DESTINATION ${Python_SITEARCH})
+      DESTINATION @TERMUX_PREFIX@/lib/python3.10/site-packages/)
   else()
     message(WARNING "Python_SITEARCH not set. Will not install python module.")
   endif()
