From dab0039680cde39bd50404d1466f3ad50e012b68 Mon Sep 17 00:00:00 2001
From: Ivy Foster <iff@escondida.tk>
Date: Thu, 26 Oct 2017 17:10:48 -0500
Subject: [PATCH] src/global.cc: do not parse user-specified init-file twice

---
 src/global.cc | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/global.cc b/src/global.cc
index b1c45d45f..37765a9b3 100644
--- a/src/global.cc
+++ b/src/global.cc
@@ -136,9 +136,7 @@ void global_scope_t::read_init()
   path init_file;
   if (HANDLED(init_file_)) {
     init_file=HANDLER(init_file_).str();
-    if (exists(init_file)) {
-      parse_init(init_file);
-    } else {
+    if (!exists(init_file)) {
       throw_(parse_error, _f("Could not find specified init file %1%") % init_file);
     }
   } else {
