--- ./src/system.hh.in.orig	2024-12-22 11:06:03.977157646 +0000
+++ ./src/system.hh.in	2024-12-22 11:06:09.980469189 +0000
@@ -144,7 +144,6 @@
 #include <boost/date_time/posix_time/posix_time_io.hpp>
 #include <boost/date_time/gregorian/gregorian_io.hpp>
 
-#include <boost/filesystem/convenience.hpp>
 #include <boost/filesystem/exception.hpp>
 #include <boost/filesystem/fstream.hpp>
 #include <boost/filesystem/operations.hpp>
