--- ../cache/vlc-3.0.4/modules/video_output/caca.c	2017-11-24 15:29:18.000000000 +0000
+++ ./modules/video_output/caca.c	2018-09-21 03:55:20.348684964 +0000
@@ -34,7 +34,7 @@
 #include <vlc_plugin.h>
 #include <vlc_vout_display.h>
 #include <vlc_picture_pool.h>
-#if !defined(_WIN32) && !defined(__APPLE__)
+#if !defined(_WIN32) && !defined(__APPLE__) && !defined(__ANDROID__)
 # ifdef X_DISPLAY_MISSING
 #  error Xlib required due to XInitThreads
 # endif
