--- ../cache/vlc-3.0.4/modules/keystore/file_crypt_android.c	2017-11-24 15:29:18.000000000 +0000
+++ ./modules/keystore/file_crypt_android.c	2018-09-21 02:35:48.106486086 +0000
@@ -17,7 +17,7 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston MA 02110-1301, USA.
  *****************************************************************************/
-
+#ifndef __termux__
 #ifdef HAVE_CONFIG_H
 # include "config.h"
 #endif
@@ -618,3 +618,4 @@
 
     return VLC_SUCCESS;
 }
+#endif
