--- ../cache/vlc-3.0.4/modules/codec/omxil/utils.c	2017-11-24 15:29:17.000000000 +0000
+++ ./modules/codec/omxil/utils.c	2018-09-21 02:34:28.645111295 +0000
@@ -211,11 +211,13 @@
     uint8_t *p_dst;
     int i_dst_stride;
     int i_plane, i_width, i_line;
+#ifndef __termux__
     if( i_color_format == QOMX_COLOR_FormatYUV420PackedSemiPlanar64x32Tile2m8ka )
     {
         qcom_convert(p_src, p_pic);
         return;
     }
+#endif
 #ifdef CAN_COMPILE_SSE2
     if( i_color_format == OMX_COLOR_FormatYUV420SemiPlanar
         && vlc_CPU_SSE2() && p_architecture_specific && p_architecture_specific->data )
