diff -uNr i3status-2.12/i3status.conf i3status-2.12.mod/i3status.conf
--- i3status-2.12/i3status.conf	2018-05-11 12:10:46.000000000 +0300
+++ i3status-2.12.mod/i3status.conf	2018-06-21 19:34:01.046914197 +0300
@@ -7,41 +7,18 @@
 # If the above line is not correctly displayed, fix your editor first!
 
 general {
-        colors = true
-        interval = 5
+    colors = true
+    interval = 5
 }
 
-order += "ipv6"
-order += "disk /"
-order += "wireless _first_"
-order += "ethernet _first_"
-order += "battery all"
-order += "load"
+order += "disk /data"
 order += "tztime local"
 
-wireless _first_ {
-        format_up = "W: (%quality at %essid) %ip"
-        format_down = "W: down"
-}
-
-ethernet _first_ {
-        # if you use %speed, i3status requires root privileges
-        format_up = "E: %ip (%speed)"
-        format_down = "E: down"
-}
-
-battery all {
-        format = "%status %percentage %remaining"
+disk "/data" {
+    format = "/data (%avail avail)"
 }
 
 tztime local {
-        format = "%Y-%m-%d %H:%M:%S"
+    format = "%Y-%m-%d %H:%M:%S"
 }
 
-load {
-        format = "%1min"
-}
-
-disk "/" {
-        format = "%avail"
-}
