diff -uNr i3status-2.13/configure.ac i3status-2.13.mod/configure.ac
--- i3status-2.13/configure.ac	2019-06-30 20:53:24.000000000 +0300
+++ i3status-2.13.mod/configure.ac	2019-07-19 02:18:04.275231603 +0300
@@ -73,7 +73,7 @@
 
 AC_SEARCH_LIBS([floor], [m], , [AC_MSG_FAILURE([cannot find the required floor() function despite trying to link with -lm])])
 
-AC_SEARCH_LIBS([shm_open], [rt])
+AC_SEARCH_LIBS([glob], [android-glob])
 
 AC_CANONICAL_HOST
 
@@ -83,7 +83,6 @@
 case $host_os in
 	linux*)
 	PKG_CHECK_MODULES([NLGENL], [libnl-genl-3.0])
-	PKG_CHECK_MODULES([ALSA], [alsa])
 	PKG_CHECK_MODULES([PULSE], [libpulse])
 	;;
 	netbsd*)
