diff -uNr i3status-2.12/Makefile i3status-2.12.mod/Makefile
--- i3status-2.12/Makefile	2018-05-11 12:10:46.000000000 +0300
+++ i3status-2.12.mod/Makefile	2018-06-21 19:33:34.403580664 +0300
@@ -1,21 +1,17 @@
 TOPDIR=$(shell pwd)
 
 ifndef PREFIX
-  PREFIX=/usr
+  PREFIX=@TERMUX_PREFIX@
 endif
 ifndef MANPREFIX
   MANPREFIX=$(PREFIX)
 endif
 ifndef SYSCONFDIR
-  ifeq ($(PREFIX),/usr)
-    SYSCONFDIR=/etc
-  else
-    SYSCONFDIR=$(PREFIX)/etc
-  endif
+  SYSCONFDIR=@TERMUX_PREFIX@/etc
 endif
 
 PKG_CONFIG ?= pkg-config
-CFLAGS+=-Wall -Wshadow -Wpointer-arith -Wcast-qual -Wsign-compare
+CFLAGS+=-Wall -Wshadow -Wpointer-arith -Wcast-qual -Wsign-compare -DLINUX
 CFLAGS+=-g
 CFLAGS+=-std=gnu99
 CFLAGS+=-pedantic
@@ -26,7 +22,7 @@
 LIBS+=-lyajl
 LIBS+=-lpulse
 LIBS+=-lm
-LIBS+=-lpthread
+LIBS+=-landroid-glob
 
 ifeq ($(wildcard .git),)
   # not in git repository
@@ -43,7 +39,6 @@
 CPPFLAGS+=-D_GNU_SOURCE
 CFLAGS += $(shell $(PKG_CONFIG) --cflags libnl-genl-3.0)
 LIBS += $(shell $(PKG_CONFIG) --libs libnl-genl-3.0)
-LIBS+=-lasound
 endif
 
 ifeq ($(OS),GNU/kFreeBSD)
