diff -uNr i3status-2.13/Makefile.am i3status-2.13.mod/Makefile.am
--- i3status-2.13/Makefile.am	2019-02-21 16:35:12.000000000 +0200
+++ i3status-2.13.mod/Makefile.am	2019-07-19 02:20:40.232423055 +0300
@@ -29,7 +29,6 @@
 	$(YAJL_CFLAGS) \
 	$(PULSE_CFLAGS) \
 	$(NLGENL_CFLAGS) \
-	$(ALSA_CFLAGS) \
 	$(PTHREAD_CFLAGS)
 
 i3status_CPPFLAGS = \
@@ -41,7 +40,6 @@
 	$(YAJL_LIBS) \
 	$(PULSE_LIBS) \
 	$(NLGENL_LIBS) \
-	$(ALSA_LIBS) \
 	$(PTHREAD_LIBS)
 
 i3status_SOURCES = \
@@ -52,20 +50,16 @@
 	src/first_network_device.c \
 	src/general.c \
 	src/output.c \
-	src/print_battery_info.c \
 	src/print_cpu_temperature.c \
 	src/print_cpu_usage.c \
 	src/print_ddate.c \
 	src/print_disk_info.c \
-	src/print_eth_info.c \
 	src/print_ip_addr.c \
 	src/print_ipv6_addr.c \
-	src/print_load.c \
 	src/print_mem.c \
 	src/print_path_exists.c \
 	src/print_run_watch.c \
 	src/print_time.c \
-	src/print_volume.c \
 	src/print_wireless_info.c \
 	src/print_file_contents.c \
 	src/process_runs.c \
