--- a/include/i3status.h
+++ b/include/i3status.h
@@ -278,7 +278,6 @@
     bool hide_seconds;
 } battery_info_ctx_t;
 
-void print_battery_info(battery_info_ctx_t *ctx);
 
 typedef struct {
     yajl_gen json_gen;
@@ -382,7 +381,6 @@
     const char *format_down;
 } eth_info_ctx_t;
 
-void print_eth_info(eth_info_ctx_t *ctx);
 
 typedef struct {
     yajl_gen json_gen;
@@ -393,7 +391,6 @@
     const float max_threshold;
 } load_ctx_t;
 
-void print_load(load_ctx_t *ctx);
 
 typedef struct {
     yajl_gen json_gen;
@@ -421,7 +418,6 @@
     int mixer_idx;
 } volume_ctx_t;
 
-void print_volume(volume_ctx_t *ctx);
 
 bool process_runs(const char *path);
 int volume_pulseaudio(uint32_t sink_idx, const char *sink_name);
