--- ../cache/racket-6.12/src/racket/sconfig.h	2017-10-12 18:23:27.000000000 +0000
+++ ./src/racket/sconfig.h	2018-02-07 01:18:13.911569478 +0000
@@ -124,10 +124,12 @@
 # endif
 # if defined(__x86_64__)
 #  define SCHEME_PLATFORM_LIBRARY_SUBPATH "x86_64-"SPLS_LINUX
 #  define REGISTER_POOR_MACHINE
 #  define ASM_DBLPREC_CONTROL_87
+#ifndef __ANDROID__
 #  define MZ_TRY_EXTFLONUMS
 # endif
+# endif
 # ifndef SCHEME_PLATFORM_LIBRARY_SUBPATH
 #  define SCHEME_PLATFORM_LIBRARY_SUBPATH "unknown-"SPLS_LINUX
 # endif
@@ -147,12 +149,16 @@
 #if defined(__i386__)
 # define MZ_USE_JIT_I386
 # define MZ_JIT_USE_MPROTECT
+#ifndef __ANDROID__
 # define MZ_USE_DWARF_LIBUNWIND
 #endif
+#endif
 #if defined(__x86_64__)
 # define MZ_USE_JIT_X86_64
 # define MZ_JIT_USE_MPROTECT
+#ifndef __ANDROID__
 # define MZ_USE_DWARF_LIBUNWIND
+#endif 
 #endif
 #if defined(__powerpc__) && !defined(__powerpc64__)
 # define MZ_USE_JIT_PPC
