diff -uNr iodine-0.7.0/src/android_dns.h iodine-0.7.0.mod/src/android_dns.h
--- iodine-0.7.0/src/android_dns.h	2014-06-16 23:28:43.000000000 +0300
+++ iodine-0.7.0.mod/src/android_dns.h	2019-01-20 16:35:42.911541961 +0200
@@ -2,7 +2,7 @@
 #ifndef __FIX_ANDROID_H__
 #define __FIX_ANDROID_H__
 
-typedef struct {
+/*typedef struct {
 	unsigned id :16;
 	unsigned rd :1;
 	unsigned tc :1;
@@ -18,7 +18,7 @@
 	unsigned ancount :16;
 	unsigned nscount :16;
 	unsigned arcount :16;
-} HEADER;
+} HEADER;*/
 
 #define NOERROR		0
 #define FORMERR		1
diff -uNr iodine-0.7.0/src/iodined.c iodine-0.7.0.mod/src/iodined.c
--- iodine-0.7.0/src/iodined.c	2014-06-16 23:28:43.000000000 +0300
+++ iodine-0.7.0.mod/src/iodined.c	2019-01-20 16:37:23.255524447 +0200
@@ -92,7 +92,7 @@
 static int bind_port;
 static int debug;
 
-#if !defined(BSD) && !defined(__GLIBC__)
+#if !defined(BSD) && !defined(__GLIBC__) && !defined(__ANDROID__)
 static char *__progname;
 #else
 extern char *__progname;
