--- src/Source/JavaScriptCore/inspector/JSGlobalObjectInspectorController.cpp	2021-05-14 19:39:34.587524906 +0000
+++ src.mod/Source/JavaScriptCore/inspector/JSGlobalObjectInspectorController.cpp	2021-05-14 19:38:52.431421074 +0000
@@ -49,7 +49,7 @@
 #include <wtf/Stopwatch.h>
 
 #include <cxxabi.h>
-#if OS(DARWIN) || (OS(LINUX) && !PLATFORM(GTK))
+#if OS(DARWIN) || (OS(LINUX) && defined(__GLIBC__) && !defined(__UCLIBC__))
 #include <dlfcn.h>
 #include <execinfo.h>
 #endif
@@ -187,7 +187,7 @@
 
 void JSGlobalObjectInspectorController::appendAPIBacktrace(ScriptCallStack* callStack)
 {
-#if OS(DARWIN) || (OS(LINUX) && !PLATFORM(GTK))
+#if OS(DARWIN) || (OS(LINUX) && defined(__GLIBC__) && !defined(__UCLIBC__))
     static const int framesToShow = 31;
     static const int framesToSkip = 3; // WTFGetBacktrace, appendAPIBacktrace, reportAPIException.
 
