--- ../configure.in.orig	2020-05-10 14:28:06.173300836 +0200
+++ ./configure.in	2020-05-10 14:36:06.592909972 +0200
@@ -44,7 +44,7 @@
 AC_HEADER_TIME
 
 dnl Checks for library functions.
-AC_CHECK_FUNCS(getcwd gettimeofday getwd putenv select strdup strerror,,
+AC_CHECK_FUNCS(getcwd gettimeofday putenv select strdup strerror,,
 	AC_MSG_ERROR([Missing required function]))
 AC_FUNC_ALLOCA
 AC_FUNC_MMAP
