diff --color -ur tree-sitter-old/Makefile tree-sitter-new/Makefile
--- tree-sitter-old/Makefile	2021-06-30 06:18:39.000000000 +0600
+++ tree-sitter-new/Makefile	2021-07-04 21:29:33.621254194 +0600
@@ -1,7 +1,7 @@
 VERSION := 0.6.3
 
 # install directory layout
-PREFIX ?= /usr/local
+PREFIX ?= ${TERMUX_PREFIX}/usr/local
 INCLUDEDIR ?= $(PREFIX)/include
 LIBDIR ?= $(PREFIX)/lib
 PCLIBDIR ?= $(LIBDIR)/pkgconfig
