diff -uNr st-0.8.2/config.mk st-0.8.2.mod/config.mk
--- st-0.8.2/config.mk	2019-02-09 13:50:41.000000000 +0200
+++ st-0.8.2.mod/config.mk	2019-07-09 19:55:40.777392265 +0300
@@ -4,11 +4,11 @@
 # Customize below to fit your system
 
 # paths
-PREFIX = /usr/local
+PREFIX = @TERMUX_PREFIX@
 MANPREFIX = $(PREFIX)/share/man
 
-X11INC = /usr/X11R6/include
-X11LIB = /usr/X11R6/lib
+X11INC = @TERMUX_PREFIX@/include/X11
+X11LIB = @TERMUX_PREFIX@/lib
 
 PKG_CONFIG = pkg-config
 
