diff --git a/internal/dashboard/sse.go b/internal/dashboard/sse.go
index e8287ad..0e3ae99 100644
--- a/internal/dashboard/sse.go
+++ b/internal/dashboard/sse.go
@@ -80,7 +80,7 @@ func (emitter *eventEmitter) onEvent(name string, data any) {
 	var retry []byte
 
 	if name == stopEvent {
-		retry = []byte(strconv.Itoa(maxSafeInteger))
+		retry = []byte(strconv.FormatInt(int64(maxSafeInteger), 10))
 	}
 
 	id := strconv.FormatInt(emitter.id.Add(1), 10)
diff --git a/internal/output/prometheusrw/remote/client.go b/internal/output/prometheusrw/remote/client.go
index 7f1db14..a4efb78 100644
--- a/internal/output/prometheusrw/remote/client.go
+++ b/internal/output/prometheusrw/remote/client.go
@@ -128,7 +128,7 @@ func newWriteRequestBody(series []*prompb.TimeSeries) ([]byte, error) {
 	}
 	if snappy.MaxEncodedLen(len(b)) < 0 {
 		return nil, fmt.Errorf("the protobuf message is too large to be handled by Snappy encoder; "+
-			"size: %d, limit: %d", len(b), math.MaxUint32)
+			"size: %d, limit: %d", len(b), uint64(math.MaxUint32))
 	}
 	return snappy.Encode(nil, b), nil
 }
diff --git a/output/cloud/expv2/metrics_client.go b/output/cloud/expv2/metrics_client.go
index f0131b2..34203f5 100644
--- a/output/cloud/expv2/metrics_client.go
+++ b/output/cloud/expv2/metrics_client.go
@@ -101,7 +101,7 @@ func newRequestBody(data *pbcloud.MetricSet) ([]byte, error) {
 	// https://pkg.go.dev/github.com/klauspost/compress/snappy#NewBufferedWriter
 	if snappy.MaxEncodedLen(len(b)) < 0 {
 		return nil, fmt.Errorf("the Protobuf message is too large to be handled by Snappy encoder; "+
-			"size: %d, limit: %d", len(b), 0xffffffff)
+			"size: %d, limit: %d", len(b), uint64(0xffffffff))
 	}
 	return snappy.Encode(nil, b), nil
 }
