--- ./configure.orig	2019-03-10 15:10:39.226464224 +0100
+++ ./configure	2019-03-10 15:13:22.879796367 +0100
@@ -377,37 +377,6 @@
    }
 done
 
-if [ "$lua_interp_found" != "yes" ]
-then
-   if [ "$LUA_VERSION_SET" ]
-   then
-      interp="Lua $LUA_VERSION"
-   else
-      interp="Lua"
-   fi
-   if [ "$LUA_DIR_SET" ] || [ "$LUA_BINDIR_SET" ]
-   then
-      where="$LUA_BINDIR"
-   else
-      where="\$PATH"
-   fi
-   echo "$(RED $interp interpreter not found) in $where"
-   echo "You may want to use the flags $(BOLD --with-lua), $(BOLD --with-lua-bin) and/or $(BOLD --lua-version)"
-   die "Run $(BOLD ./configure --help) for details."
-fi
-
-if [ "$LUA_VERSION_SET" = "yes" ]
-then
-   echo_n "Checking if $LUA_BINDIR/$LUA_INTERPRETER is Lua version $LUA_VERSION... "
-   if detect_lua_version "$LUA_BINDIR/$LUA_INTERPRETER"
-   then
-      echo "yes"
-   else
-      echo "no"
-      die "You may want to use the flags --with-lua, --with-lua-bin and/or --lua-version. See --help."
-   fi
-fi
-
 # ----------------------------------------
 # Additional checks
 # ----------------------------------------
