--- ./GNUmakefile.orig	2022-01-30 15:09:08.447445934 +0100
+++ ./GNUmakefile	2022-01-30 15:09:48.444214490 +0100
@@ -54,7 +54,7 @@
 	echo "unset LUA_PATH LUA_PATH_5_2 LUA_PATH_5_3 LUA_PATH_5_4 LUA_CPATH LUA_CPATH_5_2 LUA_CPATH_5_3 LUA_CPATH_5_4" >> luarocks
 	echo 'LUAROCKS_SYSCONFDIR="$(luarocksconfdir)" LUA_PATH="$(CURDIR)/src/?.lua;;" exec "$(LUA)" "$(CURDIR)/src/bin/luarocks" --project-tree="$(CURDIR)/lua_modules" "$$@"' >> luarocks
 	chmod +rx ./luarocks
-	./luarocks init
+	./luarocks init --lua-dir=$(PREFIX)
 
 luarocks-admin: config.unix
 	rm -f src/luarocks/core/hardcoded.lua
