--- ./src/luarocks/fs/lua.lua.orig	2019-03-10 15:14:28.333129223 +0100
+++ ./src/luarocks/fs/lua.lua	2019-03-10 15:32:29.349788026 +0100
@@ -961,7 +961,7 @@
    assert(type(name_pattern) == "string")
    name_pattern = dir.normalize(name_pattern)
 
-   return posix.mkdtemp((os.getenv("TMPDIR") or "/tmp") .. "/luarocks_" .. name_pattern:gsub("/", "_") .. "-XXXXXX")
+   return posix.mkdtemp((os.getenv("TMPDIR") or "@TERMUX_PREFIX@/tmp") .. "/luarocks_" .. name_pattern:gsub("/", "_") .. "-XXXXXX")
 end
 
 end -- if posix.mkdtemp
