--- ./src/luarocks/fs/lua.lua.orig	2019-06-07 22:33:12.441995154 +0000
+++ ./src/luarocks/fs/lua.lua	2019-06-07 22:33:22.938066139 +0000
@@ -259,7 +259,7 @@
 end
 
 function fs_lua.system_temp_dir()
-   return os.getenv("TMPDIR") or os.getenv("TEMP") or "/tmp"
+   return os.getenv("TMPDIR") or os.getenv("TEMP") or "@TERMUX_PREFIX@/tmp"
 end
 
 ---------------------------------------------------------------------
