--- ./src/luarocks/core/cfg.lua~	2018-10-30 18:31:40.000000000 +0100
+++ ./src/luarocks/core/cfg.lua	2019-03-10 16:33:24.676428404 +0100
@@ -358,7 +358,7 @@
       defaults.static_lib_extension = "a"
       defaults.external_lib_extension = "so"
       defaults.obj_extension = "o"
-      defaults.external_deps_dirs = { "/usr/local", "/usr", "/" }
+      defaults.external_deps_dirs = { "@TERMUX_PREFIX@/local", "@TERMUX_PREFIX@" }
       defaults.variables.CFLAGS = "-O2"
       defaults.cmake_generator = "Unix Makefiles"
       defaults.variables.CC = "gcc"
