--- ../cfg.lua.orig	2018-05-25 06:56:17.183526289 +0000
+++ ./src/luarocks/cfg.lua	2018-05-25 06:57:25.695490218 +0000
@@ -517,10 +517,10 @@
    defaults.lib_extension = "so"
    defaults.external_lib_extension = "so"
    defaults.obj_extension = "o"
-   defaults.external_deps_dirs = { "/usr/local", "/usr" }
-   defaults.variables.LUA_BINDIR = site_config.LUA_BINDIR or "/usr/local/bin"
-   defaults.variables.LUA_INCDIR = site_config.LUA_INCDIR or "/usr/local/include"
-   defaults.variables.LUA_LIBDIR = site_config.LUA_LIBDIR or "/usr/local/lib"
+   defaults.external_deps_dirs = { "@TERMUX_PREFIX@/local", "@TERMUX_PREFIX@" }
+   defaults.variables.LUA_BINDIR = site_config.LUA_BINDIR or "@TERMUX_PREFIX@/local/bin"
+   defaults.variables.LUA_INCDIR = site_config.LUA_INCDIR or "@TERMUX_PREFIX@/local/include"
+   defaults.variables.LUA_LIBDIR = site_config.LUA_LIBDIR or "@TERMUX_PREFIX@/local/lib"
    defaults.variables.CFLAGS = "-O2"
    defaults.cmake_generator = "Unix Makefiles"
    defaults.variables.CC = "gcc"
