diff -uNr dosbox-0.74/src/ints/int10_vesa.cpp dosbox-0.74.mod/src/ints/int10_vesa.cpp
--- dosbox-0.74/src/ints/int10_vesa.cpp	2010-05-10 20:43:54.000000000 +0300
+++ dosbox-0.74.mod/src/ints/int10_vesa.cpp	2018-09-22 19:17:25.564727346 +0300
@@ -39,7 +39,7 @@
 static char string_oem[]="S3 Incorporated. Trio64";
 static char string_vendorname[]="DOSBox Development Team";
 static char string_productname[]="DOSBox - The DOS Emulator";
-static char string_productrev[]="DOSBox "VERSION;
+static char string_productrev[]="DOSBox " VERSION;
 
 #ifdef _MSC_VER
 #pragma pack (1)
