--- a/src/lib/util/kosrelease.cpp
+++ b/src/lib/util/kosrelease.cpp
@@ -78,6 +78,11 @@ public:
         , id(QStringLiteral("linux"))
         , prettyName(QStringLiteral("Linux"))
     {
+        homeUrl = QStringLiteral("https://termux.dev/");
+        documentationUrl = QStringLiteral("https://termux.dev/en/docs/index.html");
+        supportUrl = QStringLiteral("https://wiki.termux.dev/wiki/Community");
+        bugReportUrl = QStringLiteral("https://github.com/termux/termux-packages/issues");
+        privacyPolicyUrl = QStringLiteral("https://termux.dev/en/privacy-policy");
         // Default values for non-optional fields set above ^.
 
         QHash<QString, QString *> stringHash = {{QStringLiteral("NAME"), &name},
