diff --git a/tools/build/src/tools/python.jam b/tools/build/src/tools/python.jam
index 273b28a..3de7f40 100644
--- a/tools/build/src/tools/python.jam
+++ b/tools/build/src/tools/python.jam
@@ -651,7 +651,7 @@ local rule system-library-dependencies ( target-os )
 
         case aix : return  <library>pthread <library>dl ;
 
-        case * : return  <library>pthread <library>dl
+        case * : return  <library>dl
             <toolset>gcc:<library>util <toolset-intel:platform>linux:<library>util ;
     }
 }
