https://github.com/bfgroup/b2/commit/78fd284a42caabe8815cb0870b46e5567872e75b

--- a/tools/build/src/tools/stage.jam
+++ b/tools/build/src/tools/stage.jam
@@ -478,6 +478,10 @@ class install-target-class : basic-target
         return [ sequence.unique $(result2) ] ;
     }
 
+    rule skip-from-usage-requirements ( )
+    {
+    }
+
     # Returns true iff 'type' is subtype of some element of 'types-to-include'.
     #
     local rule include-type ( type : types-to-include * )
