diff -uNr codecrypt-1.8/autogen.sh codecrypt-1.8.mod/autogen.sh
--- codecrypt-1.8/autogen.sh	2017-10-31 22:42:06.000000000 +0200
+++ codecrypt-1.8.mod/autogen.sh	2019-04-25 01:43:13.982529470 +0300
@@ -3,10 +3,10 @@
 # simple autogen script that generates basic layout for autotools.
 
 NAME="ccr"
-COMMON_CPPFLAGS="-I/usr/local/include"
+COMMON_CPPFLAGS=""
 COMMON_CFLAGS="-Wall"
 COMMON_CXXFLAGS="${COMMON_CFLAGS} -std=c++11"
-COMMON_LDFLAGS="-L/usr/local/lib"
+COMMON_LDFLAGS=""
 COMMON_LDADD=""
 
 OUT=Makefile.am
