diff -u -r ../lua-5.3.0/src/luaconf.h ./src/luaconf.h
--- ../lua-5.3.0/src/luaconf.h	2014-12-29 08:27:55.000000000 -0500
+++ ./src/luaconf.h	2015-05-24 18:07:15.413550950 -0400
@@ -175,7 +175,7 @@
 
 #else			/* }{ */
 
-#define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT	"@TERMUX_PREFIX@/"
 #define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR "/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR "/"
 #define LUA_PATH_DEFAULT  \
