diff -u -r ../lua-5.3.0/src/Makefile ./src/Makefile
--- ../lua-5.3.0/src/Makefile	2015-01-05 11:04:52.000000000 -0500
+++ ./src/Makefile	2015-05-24 18:06:15.958404755 -0400
@@ -6,13 +6,7 @@
 # Your platform. See PLATS for possible values.
 PLAT= none
 
-CC= gcc -std=gnu99
-CFLAGS= -O2 -Wall -Wextra -DLUA_COMPAT_5_2 $(SYSCFLAGS) $(MYCFLAGS)
-LDFLAGS= $(SYSLDFLAGS) $(MYLDFLAGS)
 LIBS= -lm $(SYSLIBS) $(MYLIBS)
-
-AR= ar rcu
-RANLIB= ranlib
 RM= rm -f
 
 SYSCFLAGS=
