--- ../Configure.orig	2021-11-08 19:52:16.878340986 +0000
+++ ./Configure	2021-11-08 19:53:37.598405439 +0000
@@ -2976,27 +2976,6 @@
       fi	#}
     fi	# }
 
-    # Test for SELinux support.
-
-    LSOF_TMP1=0
-    if test "X$LINUX_HASSELINUX" = "X"	# {
-    then
-      if test -r ${LSOF_INCLUDE}/selinux/selinux.h	# {
-      then
-	LSOF_TMP1=1
-      fi	# }
-    else
-      if test "X$LINUX_HASSELINUX" = "XY" -o "X$LINUX_HASSELINUX" = "Xy" # {
-      then
-	LSOF_TMP1=1
-      fi	# }
-    fi	# }
-    if test $LSOF_TMP1 -eq 1	# {
-    then
-      LSOF_CFGF="$LSOF_CFGF -DHASSELINUX"
-      LSOF_CFGL="$LSOF_CFGL -lselinux"
-    fi	# }
-
   # Test for UNIX socket endpoint support.
 
     if test -r ${LSOF_INCLUDE}/linux/sock_diag.h -a -r ${LSOF_INCLUDE}/linux/unix_diag.h  # {
@@ -5625,11 +5604,11 @@
   LSOF_CFGF="$LSOF_CFGF -DLSOF_VSTR=\\\"$LSOF_TMP\\\""
 fi	# }
 echo "" >> $LSOF_MKFC
-echo "CFGF=	$LSOF_CFGF" >> $LSOF_MKFC
+echo "CFGF=	$LSOF_CFGF -I@TERMUX_PREFIX@/include/tirpc" >> $LSOF_MKFC
 if test "X$LSOF_LIB_NO" = "X"	# {
 then
   echo "" >> $LSOF_MKFC
-  echo "CFGL=	$LSOF_FCFGL -L./$LSOF_LIB -llsof $LSOF_CFGL" >> $LSOF_MKFC
+  echo "CFGL=	$LSOF_FCFGL -Wl,-rpath=@TERMUX_PREFIX@/lib,--enable-new-dtags -L./$LSOF_LIB -llsof -L@TERMUX_PREFIX@/lib -ltirpc -landroid-support $LSOF_CFGL" >> $LSOF_MKFC
 fi	# }
 echo "" >> $LSOF_MKFC
 if test "X$LSOF_DEBUG" = "X"	# {
