diff -r -u PRoot-next/src/arch.h src/src/arch.h
--- PRoot-next/src/arch.h	2015-07-23 21:50:10.000000000 +0200
+++ src/src/arch.h	2016-08-17 16:15:24.197684187 +0200
@@ -133,6 +133,10 @@
     #define EXEC_PIC_ADDRESS   0x3000000000
     #define INTERP_PIC_ADDRESS 0x3f00000000
 
+    /* Syscall -2 appears to cause some odd side effects, use -1 */
+    #undef SYSCALL_AVOIDER
+    #define SYSCALL_AVOIDER ((word_t) -1)
+
 #elif defined(ARCH_X86)
 
     #define SYSNUMS_HEADER1 "syscall/sysnums-i386.h"
