diff -u -r ../PRoot-5.1.0/src/ptrace/ptrace.c ./src/ptrace/ptrace.c
--- ../PRoot-5.1.0/src/ptrace/ptrace.c	2014-12-15 09:18:11.000000000 -0500
+++ ./src/ptrace/ptrace.c	2016-01-08 15:12:27.089144325 -0500
@@ -27,6 +27,7 @@
 #include <signal.h>     /* siginfo_t, */
 #include <sys/uio.h>    /* struct iovec, */
 #include <sys/param.h>  /* MIN(), MAX(), */
+#include <sys/wait.h>   /* __WALL */
 #include <string.h>     /* memcpy(3), */
 
 #include "ptrace/ptrace.h"
@@ -58,7 +59,13 @@
 #define user_fpregs_struct user_fpsimd_struct
 #endif
 
-static const char *stringify_ptrace(enum __ptrace_request request)
+static const char *stringify_ptrace(
+#ifdef __ANDROID__
+		int
+#else
+		enum __ptrace_request
+#endif
+		request)
 {
 #define CASE_STR(a) case a: return #a; break;
 	switch ((int) request) {
