diff -u -r ../PRoot-5.1.0/src/tracee/tracee.h ./src/tracee/tracee.h
--- ../PRoot-5.1.0/src/tracee/tracee.h	2014-12-15 09:18:11.000000000 -0500
+++ ./src/tracee/tracee.h	2016-01-08 09:04:52.144913929 -0500
@@ -140,7 +140,12 @@
 				     && get_sysnum((tracee), ORIGINAL) == sysnum)
 
 	/* How this tracee is restarted.  */
-	enum __ptrace_request restart_how;
+#ifdef __ANDROID__
+	int
+#else
+	enum __ptrace_request
+#endif
+	restart_how;
 
 	/* Value of the tracee's general purpose registers.  */
 	struct user_regs_struct _regs[NB_REG_VERSION];
