diff -uNr abuild-3.2.0/abuild.conf abuild-3.2.0.mod/abuild.conf
--- abuild-3.2.0/abuild.conf	2018-06-22 10:24:10.000000000 +0300
+++ abuild-3.2.0.mod/abuild.conf	2018-11-18 16:42:45.897130881 +0200
@@ -1,4 +1,4 @@
-export CFLAGS="-Os -fomit-frame-pointer"
+export CFLAGS="-Os"
 export CXXFLAGS="$CFLAGS"
 export CPPFLAGS="$CFLAGS"
 export LDFLAGS="-Wl,--as-needed"
@@ -9,7 +9,7 @@
 # remove line below to disable colors
 USE_COLORS=1
 
-SRCDEST=/var/cache/distfiles
+SRCDEST=@TERMUX_PREFIX@/var/cache/distfiles
 
 # uncomment line below to store built packages in other location
 # The package will be stored as $REPODEST/$repo/$pkgname-$pkgver-r$pkgrel.apk
