diff -uNr abuild-3.4.0/abuild-fetch.c abuild-3.4.0.mod/abuild-fetch.c
--- abuild-3.4.0/abuild-fetch.c	2019-06-14 15:13:23.000000000 +0300
+++ abuild-3.4.0.mod/abuild-fetch.c	2019-07-21 16:45:34.475241394 +0300
@@ -189,7 +189,7 @@
 int main(int argc, char *argv[])
 {
 	int opt;
-	char *destdir = "/var/cache/distfiles";
+	char *destdir = "@TERMUX_PREFIX@/var/cache/distfiles";
 
 	program = argv[0];
 	while ((opt = getopt(argc, argv, "hd:")) != -1) {
