--- abuild-3.2.0/sample.APKBUILD	2018-06-22 10:24:10.000000000 +0300
+++ abuild-3.2.0.mod/sample.APKBUILD	2018-11-19 01:28:10.783277331 +0200
@@ -27,10 +27,10 @@
 
 build() {
 	cd "$builddir"
-	./configure --prefix=/usr \
-		--sysconfdir=/etc \
-		--mandir=/usr/share/man \
-		--infodir=/usr/share/info
+	./configure --prefix=$PREFIX \
+		--sysconfdir=$PREFIX/etc \
+		--mandir=$PREFIX/share/man \
+		--infodir=$PREFIX/share/info
 	make
 }
 
