diff -uNr imlib2-1.5.1/src/modules/loaders/loader_zlib.c imlib2-1.5.1.mod/src/modules/loaders/loader_zlib.c
--- imlib2-1.5.1/src/modules/loaders/loader_zlib.c	2017-12-16 15:03:10.000000000 +0200
+++ imlib2-1.5.1.mod/src/modules/loaders/loader_zlib.c	2018-11-14 19:06:27.008030924 +0200
@@ -44,7 +44,7 @@
 {
    ImlibLoader        *loader;
    int                 src, dest, res;
-   char               *file, *p, *q, tmp[] = "/tmp/imlib2_loader_zlib-XXXXXX";
+   char               *file, *p, *q, tmp[] = "@TERMUX_PREFIX@/tmp/imlib2_loader_zlib-XXXXXX";
    char               *real_ext;
    struct stat         st;
 
