diff -uNr imlib2-1.4.10/src/modules/loaders/loader_zlib.c imlib2-1.4.10.mod/src/modules/loaders/loader_zlib.c
--- imlib2-1.4.10/src/modules/loaders/loader_zlib.c	2016-02-09 18:52:49.000000000 +0200
+++ imlib2-1.4.10.mod/src/modules/loaders/loader_zlib.c	2017-11-27 14:13:27.571355069 +0200
@@ -44,7 +44,7 @@
 {
    ImlibLoader        *loader;
    int                 src, dest, res;
-   char               *file, *p, *q, tmp[] = "/tmp/imlib2_loader_zlib-XXXXXX";
+   char               *file, *p, *q, tmp[] = "/data/data/com.termux/files/usr/tmp/imlib2_loader_zlib-XXXXXX";
    char               *real_ext;
    struct stat         st;
 
