diff -uNr imlib2-1.4.10/src/modules/loaders/loader_id3.c imlib2-1.4.10.mod/src/modules/loaders/loader_id3.c
--- imlib2-1.4.10/src/modules/loaders/loader_id3.c	2017-03-25 08:48:49.000000000 +0200
+++ imlib2-1.4.10.mod/src/modules/loaders/loader_id3.c	2017-11-27 14:13:30.241355067 +0200
@@ -507,7 +507,7 @@
 
    if (loader)
      {
-        char               *ofile, tmp[] = "/tmp/imlib2_loader_id3-XXXXXX";
+        char               *ofile, tmp[] = "/data/data/com.termux/files/usr/tmp/imlib2_loader_id3-XXXXXX";
         int                 dest;
 
         if ((dest = mkstemp(tmp)) < 0)
