diff -uNr imlib2-1.4.10/src/modules/loaders/loader_bz2.c imlib2-1.4.10.mod/src/modules/loaders/loader_bz2.c
--- imlib2-1.4.10/src/modules/loaders/loader_bz2.c	2016-02-09 18:53:49.000000000 +0200
+++ imlib2-1.4.10.mod/src/modules/loaders/loader_bz2.c	2017-11-27 14:13:24.631355071 +0200
@@ -52,7 +52,7 @@
    ImlibLoader        *loader;
    FILE               *fp;
    int                 dest, res;
-   char               *file, *p, *q, tmp[] = "/tmp/imlib2_loader_bz2-XXXXXX";
+   char               *file, *p, *q, tmp[] = "/data/data/com.termux/files/usr/tmp/imlib2_loader_bz2-XXXXXX";
    char               *real_ext;
 
    assert(im);
