diff --git a/cinnamon-session/cinnamon-session.in b/cinnamon-session/cinnamon-session.in
index d9d7cb2..1c52c90 100755
--- a/cinnamon-session/cinnamon-session.in
+++ b/cinnamon-session/cinnamon-session.in
@@ -3,7 +3,6 @@
 if [ "x$XDG_SESSION_TYPE" = "xwayland" ] &&
    [ "x$XDG_SESSION_CLASS" != "xgreeter" ] &&
    [  -n "$SHELL" ] &&
-   grep -q "$SHELL" /etc/shells &&
    ! (echo "$SHELL" | grep -q "false") &&
    ! (echo "$SHELL" | grep -q "nologin"); then
   if [ "$1" != '-l' ]; then
