--- a/term_tag.c
+++ b/term_tag.c
@@ -87,7 +87,7 @@
 
 	if (outfilename == NULL) {
 		(void)snprintf(tag_files.ofn, sizeof(tag_files.ofn),
-		    "/tmp/man.XXXXXXXXXX%s", suffix);
+		    "@TERMUX_PREFIX@/tmp/man.XXXXXXXXXX%s", suffix);
 		if ((ofd = mkstemps(tag_files.ofn, strlen(suffix))) == -1) {
 			mandoc_msg(MANDOCERR_MKSTEMP, 0, 0,
 			    "%s: %s", tag_files.ofn, strerror(errno));
@@ -105,7 +105,7 @@
 		}
 	}
 	if (tagfilename == NULL) {
-		(void)strlcpy(tag_files.tfn, "/tmp/man.XXXXXXXXXX",
+		(void)strlcpy(tag_files.tfn, "@TERMUX_PREFIX@/tmp/man.XXXXXXXXXX",
 		    sizeof(tag_files.tfn));
 		if ((tfd = mkstemp(tag_files.tfn)) == -1) {
 			mandoc_msg(MANDOCERR_MKSTEMP, 0, 0,
@@ -201,11 +201,11 @@
 		    getpgid(tc_pgid) == -1)
 			(void)tcsetpgrp(STDOUT_FILENO, tag_files.tcpgid);
 	}
-	if (strncmp(tag_files.ofn, "/tmp/man.", 9) == 0) {
+	if (strncmp(tag_files.ofn, "@TERMUX_PREFIX@/tmp/man.", 9) == 0) {
 		unlink(tag_files.ofn);
 		*tag_files.ofn = '\0';
 	}
-	if (strncmp(tag_files.tfn, "/tmp/man.", 9) == 0) {
+	if (strncmp(tag_files.tfn, "@TERMUX_PREFIX@/tmp/man.", 9) == 0) {
 		unlink(tag_files.tfn);
 		*tag_files.tfn = '\0';
 	}
