Fixes link-time conflict with graphviz

--- a/subprojects/gvc/meson.build
+++ b/subprojects/gvc/meson.build
@@ -90,7 +90,7 @@ else
     error('Installing shared library, but pkglibdir is unset!')
   endif
 
-  libgvc_shared = shared_library('gvc',
+  libgvc_shared = shared_library('gnomevolumecontrol',
     sources: libgvc_gir_sources + libgvc_no_gir_sources + libgvc_enums,
     dependencies: libgvc_deps,
     c_args: c_args,
