--- a/subprojects/extensions-app/meson.build
+++ b/subprojects/extensions-app/meson.build
@@ -47,7 +47,7 @@ metainfodir = join_paths(datadir, 'metainfo')
 schemadir = join_paths(datadir, 'glib-2.0', 'schemas')
 servicedir = join_paths(datadir, 'dbus-1', 'services')
 
-gjs = find_program('gjs')
+gjs = find_program('@TERMUX_PREFIX@/bin/gjs', native : false)
 appstreamcli = find_program('appstreamcli', required: false)
 desktop_file_validate = find_program('desktop-file-validate', required: false)
 
