--- a/src/Operation.h
+++ b/src/Operation.h
@@ -78,7 +78,7 @@
   // Define a partial order on Operations:
   //  - Create < Update < Delete < UndoPoint
   //  - Given two updates, sort by timestamp
-  bool operator<(Operation &other) const;
+  bool operator<(const Operation &other) const;
 
  private:
   const tc::Operation *op;
--- a/src/Operation.cpp
+++ b/src/Operation.cpp
@@ -47,7 +47,7 @@
 }
 
 ////////////////////////////////////////////////////////////////////////////////
-bool Operation::operator<(Operation& other) const {
+bool Operation::operator<(const Operation& other) const {
   if (is_create()) {
     return !other.is_create();
   } else if (is_update()) {
