diff -u -r ../task-2.5.1/CMakeLists.txt ./CMakeLists.txt
--- ../task-2.5.1/CMakeLists.txt	2016-02-24 23:18:11.000000000 +0100
+++ ./CMakeLists.txt	2017-01-15 23:14:01.677031016 +0100
@@ -43,10 +43,6 @@
  message (FATAL_ERROR "C++11 support missing. Try upgrading your C++ compiler. If you have a good reason for using an outdated compiler, please let us know at support@taskwarrior.org.")
 endif (_HAS_CXX11)
 
-if (${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang")
-  set (_CXX11_FLAGS "${_CXX11_FLAGS} -stdlib=libc++")
-endif (${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang")
-
 if (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
   set (LINUX true)
 elseif (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
