diff -uNr net-tools/lib/pathnames.h net-tools.mod/lib/pathnames.h
--- net-tools/lib/pathnames.h	2017-02-21 20:24:32.000000000 +0200
+++ net-tools.mod/lib/pathnames.h	2019-03-01 19:05:45.935478227 +0200
@@ -47,7 +47,7 @@
 #define _PATH_PROCNET_IFINET6		"/proc/net/if_inet6"
 #define _PATH_PROCNET_DEV		"/proc/net/dev"
 #define _PATH_PROCNET_RARP		"/proc/net/rarp"
-#define _PATH_ETHERS			"/etc/ethers"
+#define _PATH_ETHERS			"@TERMUX_PREFIX@/etc/ethers"
 #define _PATH_PROCNET_ROSE		"/proc/net/rose"
 #define _PATH_PROCNET_ROSE_NEIGH	"/proc/net/rose_neigh"
 #define _PATH_PROCNET_ROSE_NODES	"/proc/net/rose_nodes"
