diff -uNr net-tools/nameif.c net-tools.mod/nameif.c
--- net-tools/nameif.c	2017-02-21 20:24:32.000000000 +0200
+++ net-tools.mod/nameif.c	2019-03-01 19:02:26.447822359 +0200
@@ -10,7 +10,7 @@
 #endif
 #include <stdio.h>
 #include <getopt.h>
-#include <sys/syslog.h>
+#include <syslog.h>
 #include <errno.h>
 #include <stdlib.h>
 #include <ctype.h>
@@ -25,7 +25,7 @@
 #include "net-support.h"
 #include "util.h"
 
-const char default_conf[] = "/etc/mactab";
+const char default_conf[] = "@TERMUX_PREFIX@/etc/mactab";
 const char *fname = default_conf;
 int use_syslog;
 int ctl_sk = -1;
