diff -uNr net-tools/slattach.c net-tools.mod/slattach.c
--- net-tools/slattach.c	2017-02-21 20:24:32.000000000 +0200
+++ net-tools.mod/slattach.c	2019-03-01 21:54:06.558251056 +0200
@@ -55,7 +55,7 @@
 #include "util.h"
 
 #ifndef _PATH_LOCKD
-#define _PATH_LOCKD	"/var/lock"		/* lock files   */
+#define _PATH_LOCKD	"@TERMUX_PREFIX@/var/lock"		/* lock files   */
 #endif
 #ifndef _UID_UUCP
 #define _UID_UUCP	"uucp"			/* owns locks   */
