commit b4a22f6cf4512b07feef38354b76a97db1f92f5c
Author: Brian McKenna <brian@brianmckenna.org>
Date:   Thu Aug 3 08:10:55 2017 +1000

    Directly call AutoCloseFD::get

diff --git a/src/libstore/local-store.cc b/src/libstore/local-store.cc
index 95b05f8..23bd165 100644
--- a/src/libstore/local-store.cc
+++ b/src/libstore/local-store.cc
@@ -1313,7 +1313,7 @@ static void makeMutable(const Path & path)
        mutable bit on the target of a symlink (which would be a
        security hole). */
     AutoCloseFD fd = open(path.c_str(), O_RDONLY | O_NOFOLLOW | O_CLOEXEC);
-    if (fd == -1) {
+    if (fd.get() == -1) {
         if (errno == ELOOP) return; // it's a symlink
         throw SysError(format("opening file '%1%'") % path);
     }
@@ -1322,11 +1322,11 @@ static void makeMutable(const Path & path)
 
     /* Silently ignore errors getting/setting the immutable flag so
        that we work correctly on filesystems that don't support it. */
-    if (ioctl(fd, FS_IOC_GETFLAGS, &flags)) return;
+    if (ioctl(fd.get(), FS_IOC_GETFLAGS, &flags)) return;
     old = flags;
     flags &= ~FS_IMMUTABLE_FL;
     if (old == flags) return;
-    if (ioctl(fd, FS_IOC_SETFLAGS, &flags)) return;
+    if (ioctl(fd.get(), FS_IOC_SETFLAGS, &flags)) return;
 }
 
 /* Upgrade from schema 6 (Nix 0.15) to schema 7 (Nix >= 1.3). */
