--- /dev/null	2018-05-15 17:54:13.876121969 +0000
+++ ./lib/config.h	2018-05-15 20:51:48.262434344 +0000
@@ -0,0 +1,15 @@
+#define PCI_CONFIG_H
+#define PCI_ARCH_@TERMUX_ARCH@
+#define PCI_OS_LINUX
+#define PCI_HAVE_PM_LINUX_SYSFS
+#define PCI_HAVE_PM_LINUX_PROC
+#define PCI_HAVE_LINUX_BYTEORDER_H
+#define PCI_PATH_PROC_BUS_PCI "/proc/bus/pci"
+#define PCI_PATH_SYS_BUS_PCI "/sys/bus/pci"
+#define PCI_HAVE_64BIT_ADDRESS
+#define PCI_HAVE_PM_DUMP
+#define PCI_COMPRESSED_IDS
+#define PCI_IDS "pci.ids.gz"
+#define PCI_PATH_IDS_DIR "@TERMUX_PREFIX@/share"
+#define PCI_SHARED_LIB
+#define PCILIB_VERSION "3.5.6"
