diff -u -r ../graphite-1.3.9/src/CMakeLists.txt ./src/CMakeLists.txt
--- ../graphite-1.3.9/src/CMakeLists.txt	2016-11-11 10:53:21.000000000 +0100
+++ ./src/CMakeLists.txt	2017-01-19 05:12:08.903024071 +0100
@@ -152,4 +152,4 @@
 
 
 install(TARGETS graphite2 EXPORT graphite2 LIBRARY DESTINATION lib${LIB_SUFFIX} ARCHIVE DESTINATION lib${LIB_SUFFIX} PUBLIC_HEADER DESTINATION include/graphite2 RUNTIME DESTINATION bin)
-install(EXPORT graphite2 DESTINATION share/graphite2 NAMESPACE gr2_)
+install(EXPORT graphite2 DESTINATION lib/cmake/graphite2 NAMESPACE gr2_)
