diff -uNr onnxruntime-1.23.0/cmake/external/abseil-cpp.cmake onnxruntime-1.23.0.mod/cmake/external/abseil-cpp.cmake
--- onnxruntime-1.23.0/cmake/external/abseil-cpp.cmake	2025-09-25 07:04:17.000000000 +0530
+++ onnxruntime-1.23.0.mod/cmake/external/abseil-cpp.cmake	2025-10-06 18:34:44.412636167 +0530
@@ -36,7 +36,7 @@
     URL_HASH SHA1=${DEP_SHA1_abseil_cpp}
     EXCLUDE_FROM_ALL
     PATCH_COMMAND ${ABSL_PATCH_COMMAND}
-    FIND_PACKAGE_ARGS 20250512 NAMES absl
+    FIND_PACKAGE_ARGS NAMES absl
 )
 
 onnxruntime_fetchcontent_makeavailable(abseil_cpp)
@@ -65,9 +65,6 @@
 # TODO: since multiple ORT's dependencies depend on Abseil, the list below would vary from version to version.
 # We'd better to not manually manage the list.
 set(ABSEIL_LIBS
-absl::absl_log
-absl::log_internal_log_impl
-absl::log_internal_strip
 absl::log_internal_message
 absl::log_internal_format
 absl::synchronization
@@ -122,7 +119,6 @@
 absl::hash_function_defaults
 absl::function_ref
 absl::city
-absl::low_level_hash
 absl::fixed_array
 absl::variant
 absl::meta
