diff -uNr onnxruntime-1.23.0/cmake/CMakeLists.txt onnxruntime-1.23.0.mod/cmake/CMakeLists.txt
--- onnxruntime-1.23.0/cmake/CMakeLists.txt	2025-09-25 07:04:17.000000000 +0530
+++ onnxruntime-1.23.0.mod/cmake/CMakeLists.txt	2025-10-06 17:55:12.700446767 +0530
@@ -495,6 +495,7 @@
     if (onnxruntime_REQUIRE_PYTHON_EMBED_LIB)
       find_package(Python 3.10 COMPONENTS Interpreter Development NumPy)
     else()
+      find_package(Python 3.10 COMPONENTS Development)
       find_package(Python 3.10 COMPONENTS Interpreter Development.Module NumPy)
     endif()
     message("Numpy version: ${Python_NumPy_VERSION}")
