--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -593,6 +593,7 @@
     if (onnxruntime_REQUIRE_PYTHON_EMBED_LIB)
       find_package(Python 3.8 COMPONENTS Interpreter Development NumPy)
     else()
+      find_package(Python 3.8 COMPONENTS Development)
       find_package(Python 3.8 COMPONENTS Interpreter Development.Module NumPy)
     endif()
     message("Numpy version: ${Python_NumPy_VERSION}")
