diff -uNr onnxruntime-1.23.0/cmake/CMakeLists.txt onnxruntime-1.23.0.mod/cmake/CMakeLists.txt
--- onnxruntime-1.23.0/cmake/CMakeLists.txt	2025-09-25 07:04:17.000000000 +0530
+++ onnxruntime-1.23.0.mod/cmake/CMakeLists.txt	2025-10-06 17:57:49.290194740 +0530
@@ -1800,4 +1800,9 @@
 
     include("${CMAKE_CURRENT_SOURCE_DIR}/arm64x.cmake")
   endif()
+
+target_compile_options(onnx PRIVATE "-Wno-unused-variable")
+target_compile_options(onnxruntime_providers PRIVATE "-Wno-unused-variable")
+target_compile_options(onnxruntime_providers_xnnpack PRIVATE -Wno-unused-variable)
+target_compile_options(onnxruntime_providers_nnapi PRIVATE "-Wno-unused-variable")
 endif()
