diff -u -r ../neovim-b57d9a4ff08fc1ef8db79d422b441b792493ff4e/runtime/autoload/provider/clipboard.vim ./runtime/autoload/provider/clipboard.vim
--- ../neovim-b57d9a4ff08fc1ef8db79d422b441b792493ff4e/runtime/autoload/provider/clipboard.vim	2017-11-25 22:22:46.000000000 +0000
+++ ./runtime/autoload/provider/clipboard.vim	2017-11-25 22:51:34.076263411 +0000
@@ -103,6 +103,12 @@
     let s:copy['*'] = s:copy['+']
     let s:paste['*'] = s:paste['+']
     return 'win32yank'
+  elseif executable('termux-clipboard-set')
+    let s:copy['+'] = 'termux-clipboard-set'
+    let s:paste['+'] = 'termux-clipboard-get'
+    let s:copy['*'] = s:copy['+']
+    let s:paste['*'] = s:paste['+']
+    return 'termux-clipboard'
   elseif exists('$TMUX') && executable('tmux')
     let s:copy['+'] = 'tmux load-buffer -'
     let s:paste['+'] = 'tmux save-buffer -'
