diff -u -r ../neovim-master/runtime/CMakeLists.txt ./runtime/CMakeLists.txt
--- ../neovim-master/runtime/CMakeLists.txt	2015-08-03 15:36:33.000000000 -0400
+++ ./runtime/CMakeLists.txt	2015-08-03 17:38:07.657949138 -0400
@@ -35,7 +35,7 @@
 add_custom_command(OUTPUT ${GENERATED_HELP_TAGS}
   COMMAND ${CMAKE_COMMAND} -E copy_directory
     ${PROJECT_SOURCE_DIR}/runtime/doc ${GENERATED_RUNTIME_DIR}/doc
-  COMMAND "${PROJECT_BINARY_DIR}/bin/nvim"
+  COMMAND "${PROJECT_BINARY_DIR}/../host-build/bin/nvim"
     -u NONE
     -i NONE
     -esX
