diff -u -r ../neovim-master/runtime/autoload/provider/clipboard.vim ./runtime/autoload/provider/clipboard.vim
--- ../neovim-master/runtime/autoload/provider/clipboard.vim	2016-04-23 15:32:43.000000000 -0400
+++ ./runtime/autoload/provider/clipboard.vim	2016-04-23 17:44:02.062815241 -0400
@@ -37,6 +37,11 @@
   let s:copy['*'] = s:copy['+']
   let s:paste['*'] = s:paste['+']
   let s:cache_enabled = 0
+elseif executable('termux-clipboard-set')
+  let s:copy['+'] = 'termux-clipboard-set'
+  let s:paste['+'] = 'termux-clipboard-get'
+  let s:copy['*'] = 'termux-clipboard-set'
+  let s:paste['*'] = 'termux-clipboard-get'
 elseif exists('$DISPLAY') && executable('xsel')
   let s:copy['+'] = 'xsel --nodetach -i -b'
   let s:paste['+'] = 'xsel -o -b'
