diff -u -r ../neovim-a062cd4ce58ba9aca6fdce443b014c9c0949ecde/runtime/autoload/provider/clipboard.vim ./runtime/autoload/provider/clipboard.vim
--- ../neovim-a062cd4ce58ba9aca6fdce443b014c9c0949ecde/runtime/autoload/provider/clipboard.vim	2017-01-16 23:18:19.000000000 +0100
+++ ./runtime/autoload/provider/clipboard.vim	2017-01-17 10:14:57.997439109 +0100
@@ -78,6 +78,12 @@
     let s:copy['*'] = s:copy['+']
     let s:paste['*'] = s:paste['+']
     return 'doitclient'
+  elseif executable('termux-clipboard-set')
+    let s:copy['+'] = 'termux-clipboard-set'
+    let s:paste['+'] = 'termux-clipboard-get'
+    let s:copy['*'] = s:copy['+']
+    let s:paste['*'] = s:paste['+']
+    return 'termux-clipboard'
   endif

   let s:err = 'clipboard: No clipboard tool available. See :help clipboard'
