diff -u -r ../neovim-76ea97c809e50fccc5ca6615943ac6da1db1e030/runtime/autoload/provider/clipboard.vim ./runtime/autoload/provider/clipboard.vim
--- ../neovim-76ea97c809e50fccc5ca6615943ac6da1db1e030/runtime/autoload/provider/clipboard.vim	2017-06-19 00:01:29.000000000 +0000
+++ ./runtime/autoload/provider/clipboard.vim	2017-06-19 20:10:55.376978820 +0000
@@ -90,6 +90,12 @@
     let s:copy['*'] = s:copy['+']
     let s:paste['*'] = s:paste['+']
     return 'tmux'
+  elseif executable('termux-clipboard-set')
+    let s:copy['+'] = 'termux-clipboard-set'
+    let s:paste['+'] = 'termux-clipboard-get'
+    let s:copy['*'] = s:copy['+']
+    let s:paste['*'] = s:paste['+']
+    return 'termux-clipboard'
   endif
 
   let s:err = 'clipboard: No clipboard tool available. :help clipboard'
