diff -u -r ../neovim-master/runtime/CMakeLists.txt ./runtime/CMakeLists.txt
--- ../neovim-master/runtime/CMakeLists.txt	2016-02-05 10:02:38.000000000 -0500
+++ ./runtime/CMakeLists.txt	2016-02-05 17:56:54.645492669 -0500
@@ -32,7 +32,7 @@
 )
 
 add_custom_command(OUTPUT ${GENERATED_HELP_TAGS}
-  COMMAND "${PROJECT_BINARY_DIR}/bin/nvim"
+  COMMAND "${PROJECT_BINARY_DIR}/../host-build/bin/nvim"
     -u NONE
     -i NONE
     -es
