--- ./src/nvim/os/stdpaths.c.orig	2022-07-02 15:46:11.997439729 +0530
+++ ./src/nvim/os/stdpaths.c	2022-07-02 15:47:38.817439696 +0530
@@ -51,8 +51,8 @@
   [kXDGCacheHome] = "~/.cache",
   [kXDGStateHome] = "~/.local/state",
   [kXDGRuntimeDir] = NULL,  // Decided by vim_mktempdir().
-  [kXDGConfigDirs] = "/etc/xdg/",
-  [kXDGDataDirs] = "/usr/local/share/:/usr/share/",
+  [kXDGConfigDirs] = "@TERMUX_PREFIX@/etc/xdg/",
+  [kXDGDataDirs] = "@TERMUX_PREFIX@/local/share/:@TERMUX_PREFIX@/share/",
 #endif
 };
