--- a/src/quicktemplates2/accessible/qaccessiblequickpage_p.h
+++ b/src/quicktemplates2/accessible/qaccessiblequickpage_p.h
@@ -52,6 +52,8 @@
 
 QT_BEGIN_NAMESPACE
 
+#if QT_CONFIG(accessibility)
+
 class QQuickPage;
 
 class QAccessibleQuickPage : public QAccessibleQuickItem
@@ -65,6 +67,8 @@
     QList<QQuickItem *> orderedChildItems() const;
 };
 
+#endif // accessibility
+
 QT_END_NAMESPACE
 
 #endif // QACCESSIBLEQUICKPAGE_H
