--- a/src/quicktemplates2/accessible/qaccessiblequickpage.cpp
+++ b/src/quicktemplates2/accessible/qaccessiblequickpage.cpp
@@ -39,6 +39,8 @@
 
 QT_BEGIN_NAMESPACE
 
+#if QT_CONFIG(accessibility)
+
 QAccessibleQuickPage::QAccessibleQuickPage(QQuickPage *page)
     : QAccessibleQuickItem(page)
 {
@@ -77,5 +79,7 @@
     return static_cast<QQuickPage*>(object());
 }
 
+#endif // accessibility
+
 QT_END_NAMESPACE
 
