--- a/libdjvu/GBitmap.h
+++ b/libdjvu/GBitmap.h
@@ -620,7 +620,7 @@
 inline int
 GBitmap::read_run(unsigned char *&data)
 {
-  register int z=*data++;
+  const int z=*data++;
   return (z>=RUNOVERFLOWVALUE)?
     ((z&~RUNOVERFLOWVALUE)<<8)|(*data++):z;
 }
@@ -628,7 +628,7 @@
 inline int
 GBitmap::read_run(const unsigned char *&data)
 {
-  register int z=*data++;
+  const int z=*data++;
   return (z>=RUNOVERFLOWVALUE)?
     ((z&~RUNOVERFLOWVALUE)<<8)|(*data++):z;
 }
