--- a/vendor/golang.design/x/clipboard/clipboard_android.c
+++ b/vendor/golang.design/x/clipboard/clipboard_android.c
@@ -4,7 +4,7 @@
 //
 // Written by Changkun Ou <changkun.de>
 
-//go:build android
+//go:build disabling_this_because_it_is_for_building_an_apk
 
 #include <android/log.h>
 #include <jni.h>
--- a/vendor/golang.design/x/clipboard/clipboard_android.go
+++ b/vendor/golang.design/x/clipboard/clipboard_android.go
@@ -4,7 +4,7 @@
 //
 // Written by Changkun Ou <changkun.de>
 
-//go:build android
+//go:build disabling_this_because_it_is_for_building_an_apk
 
 package clipboard
 
--- a/vendor/golang.design/x/clipboard/clipboard_linux.go
+++ b/vendor/golang.design/x/clipboard/clipboard_linux.go
@@ -4,7 +4,7 @@
 //
 // Written by Changkun Ou <changkun.de>
 
-//go:build linux && !android
+//go:build linux
 
 package clipboard
 
--- a/vendor/golang.design/x/clipboard/clipboard_wayland.go
+++ b/vendor/golang.design/x/clipboard/clipboard_wayland.go
@@ -4,7 +4,7 @@
 //
 // Written by Changkun Ou <changkun.de>
 
-//go:build (linux || freebsd || openbsd || netbsd) && !android
+//go:build linux || freebsd || openbsd || netbsd
 
 package clipboard
 
--- a/vendor/golang.design/x/clipboard/clipboard_x11.go
+++ b/vendor/golang.design/x/clipboard/clipboard_x11.go
@@ -4,7 +4,7 @@
 //
 // Written by Changkun Ou <changkun.de>
 
-//go:build (linux || freebsd || openbsd || netbsd) && !android
+//go:build linux || freebsd || openbsd || netbsd
 
 package clipboard
 
