--- a/src/app/CMakeLists.txt
+++ b/src/app/CMakeLists.txt
@@ -344,12 +344,6 @@ install(TARGETS MuseScoreStudio
     RUNTIME DESTINATION bin
 )
 
-qt_generate_deploy_qml_app_script(
-    TARGET MuseScoreStudio
-    OUTPUT_SCRIPT deploy_script
-    MACOS_BUNDLE_POST_BUILD
-)
-
 if(OS_IS_WIN)
     # Doing this only on Windows for now. On other platforms, it is not necessary
     # for development builds because rpath exists, and it takes very long. However,
