--- old/Project/GNU/Library/libmediainfo.pc.in	2019-09-10 21:27:35.000000000 +0200
+++ Project/GNU/Library/libmediainfo.pc.in	2019-12-05 13:05:47.469137700 +0100
@@ -3,7 +3,7 @@
 libdir=@libdir@
 includedir=@includedir@
 Unicode=@MediaInfoLib_Unicode@
-Libs_Static=${libdir}/lib@MediaInfoLib_LibName@.a ${libdir}/libzen.a -lpthread -lz@Curl_Lib@
+Libs_Static=${libdir}/lib@MediaInfoLib_LibName@.a ${libdir}/libzen.a -lz@Curl_Lib@
 la_name=lib@MediaInfoLib_LibName@.la
 
 Name: libmediainfo
