--- ../kona-Win.3.36-64/src/getline.h	2015-07-30 11:18:02.000000000 -0400
+++ ./src/getline.h	2015-12-06 03:21:00.000000000 -0500
@@ -1,16 +1,16 @@
 K _p();
 S strdupn(S s,I k);
 I strlenn(S s,I k);
-I getdelim_(S *s,size_t * __restrict__ n,I d,FILE *f);
-I getline_(S *s,size_t * __restrict__ n,FILE *f);
+I new_getdelim_(S *s,size_t * __restrict__ n,I d,FILE *f);
+I new_getline_(S *s,size_t * __restrict__ n,FILE *f);
 I appender(S *s,I *n,S t,I k);
 I expander(S *s,I n);
 
 #if defined(__OpenBSD__) || defined(__NetBSD__) || \
    (defined(__MACH__) && __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ < 1070) || \
    defined(__ANDROID__)
-I getline(S *s,size_t * __restrict__ n,FILE *f);
-I getdelim(S *s,size_t * __restrict__ n,I d,FILE *f);
+I new_getline(S *s,size_t * __restrict__ n,FILE *f);
+I new_getdelim(S *s,size_t * __restrict__ n,I d,FILE *f);
 #endif
 
 #if WIN32
