--- ../kona-Win.3.36-64/Makefile	2015-07-30 11:18:02.000000000 -0400
+++ ./Makefile	2015-12-06 04:14:40.000000000 -0500
@@ -1,10 +1,10 @@
-PREFIX = /usr/local
-CFLAGS=-g
+PREFIX = @TERMUX_PREFIX@
+CFLAGS=-O3
 PRODFLAGS = -O3 #-pg -g3
 LIB=libkona.a
 DEVFLAGS = -O0 -g3 -DDEBUG -Wunused -Wreturn-type -Wimplicit-int #-Wall
 
-OS := $(shell uname -s | tr "[:upper:]" "[:lower:]")
+OS = android
 
 # Win-64
 ifeq (mingw32_nt-6.2,$(OS))
@@ -34,7 +34,7 @@
       src/ks.o src/v.o src/va.o src/vc.o src/vd.o src/vf.o src/vg.o src/vq.o
 LDFLAGS = -Wl,--gc-sections -Wl,-z,nocopyreloc -lgcc -no-canonical-prefixes \
           -Wl,--no-undefined -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now -mthumb \
-          -lc -lm -ldl
+          -fPIE -pie -lc -lm -ldl
 CFLAGS += -fPIE -fpic -ffunction-sections -funwind-tables -fstack-protector \
           -no-canonical-prefixes -mtune=xscale -msoft-float -mthumb \
           -fomit-frame-pointer -fno-strict-aliasing
@@ -109,7 +109,7 @@
 test: k_test
 
 install:
-	install k $(PREFIX)/bin/k
+	install k ${PREFIX}/bin/k
 
 clean:
 	$(RM) -r k k_test *.exe k.dSYM k_test.dSYM src/*.o src/win/*.o
