diff -uNr kona-Win64-20190226/src/v.h kona-Win64-20190226.mod/src/v.h
--- kona-Win64-20190226/src/v.h	2019-02-26 22:13:07.000000000 +0200
+++ kona-Win64-20190226.mod/src/v.h	2019-05-21 14:40:16.130994527 +0300
@@ -6,3 +6,6 @@
 extern V offsetSSR,offsetWhat,offsetAt,offsetDot,offsetColon;  //k.c
 extern K KTREE;                      //k.c
 K wd_(S s,int n,K*dict,K func);        //p.c
+
+K lookupEntryOrCreate(K *p, S k);
+K* EAP(K e);
