diff -uNr kona-Win64-20190226/src/getline.h kona-Win64-20190226.mod/src/getline.h
--- kona-Win64-20190226/src/getline.h	2019-02-26 22:13:07.000000000 +0200
+++ kona-Win64-20190226.mod/src/getline.h	2019-05-21 14:38:35.428447096 +0300
@@ -6,8 +6,7 @@
 I appender(S *s,I *n,S t,I k);
 I expander(S *s,I n);
 
-#if defined(__MACH__) && __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ < 1070 || \
-   defined(__ANDROID__)
+#if defined(__MACH__) && __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ < 1070
 I getline(S *s,size_t * __restrict__ n,FILE *f);
 I getdelim(S *s,size_t * __restrict__ n,I d,FILE *f);
 #endif
