diff -u -r ../kona-Win.3.39-69/src/getline.h ./src/getline.h
--- ../kona-Win.3.39-69/src/getline.h	2016-04-17 14:10:22.000000000 -0400
+++ ./src/getline.h	2016-05-03 14:15:16.755266190 -0400
@@ -7,8 +7,7 @@
 I expander(S *s,I n);
 
 #if defined(__OpenBSD__) || defined(__NetBSD__) || \
-   (defined(__MACH__) && __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ < 1070) || \
-   defined(__ANDROID__)
+   (defined(__MACH__) && __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ < 1070)
 I getline(S *s,size_t * __restrict__ n,FILE *f);
 I getdelim(S *s,size_t * __restrict__ n,I d,FILE *f);
 #endif
