diff -uNr fetchmail-6.4.1/fetchmail.h fetchmail-6.4.1.mod/fetchmail.h
--- fetchmail-6.4.1/fetchmail.h	2019-09-28 11:52:03.000000000 +0300
+++ fetchmail-6.4.1.mod/fetchmail.h	2019-10-31 15:03:00.008488320 +0200
@@ -114,7 +114,7 @@
 #define		MSGBUFSIZE	8192
 
 #define		NAMELEN		64	/* max username length */
-#define		PASSWORDLEN	256	/* max password length */
+#define		FETCHMAIL_PASSWORDLEN	256	/* max password length */
 #define		DIGESTLEN	33	/* length of MD5 digest */
 
 /* exit code values */
@@ -463,7 +463,7 @@
 extern flag configdump;		/* dump control blocks as Python dictionary */
 extern const char *fetchmailhost; /* either "localhost" or an FQDN */
 extern int suppress_tags;	/* suppress tags in tagged protocols? */
-extern char shroud[PASSWORDLEN*2+3];	/* string to shroud in debug output */
+extern char shroud[FETCHMAIL_PASSWORDLEN*2+3];	/* string to shroud in debug output */
 #ifdef SDPS_ENABLE
 extern char *sdps_envfrom;
 extern char *sdps_envto;
diff -uNr fetchmail-6.4.1/getpass.c fetchmail-6.4.1.mod/getpass.c
--- fetchmail-6.4.1/getpass.c	2019-08-24 11:49:43.000000000 +0300
+++ fetchmail-6.4.1.mod/getpass.c	2019-10-31 15:03:10.211855090 +0200
@@ -24,7 +24,7 @@
 #include "fetchmail.h"
 #include "i18n.h"
 
-#define INPUT_BUF_SIZE	PASSWORDLEN
+#define INPUT_BUF_SIZE	FETCHMAIL_PASSWORDLEN
 
 #if defined(HAVE_TERMIOS_H) && defined(HAVE_TCSETATTR)
 #  include <termios.h>
diff -uNr fetchmail-6.4.1/transact.c fetchmail-6.4.1.mod/transact.c
--- fetchmail-6.4.1/transact.c	2019-08-28 01:30:05.000000000 +0300
+++ fetchmail-6.4.1.mod/transact.c	2019-10-31 15:03:15.288538378 +0200
@@ -56,7 +56,7 @@
 /** Macro to generate the tag and store it in #tag. */
 #define GENSYM	(sprintf(tag, "A%04u", ++tagnum % TAGMOD), tag)
 static const struct method *protocol; /**< description of the protocol used for the current poll */
-char shroud[PASSWORDLEN*2+3];	/**< string to shroud in debug output */
+char shroud[FETCHMAIL_PASSWORDLEN*2+3];	/**< string to shroud in debug output */
 
 /* session variables initialized in do_session() */
 int mytimeout;		/**< value of nonreponse timeout */
