--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1251,16 +1251,16 @@
 if (UNIX AND NOT APPLE)
     # Install desktop entry
     install(FILES ${LENMUS_ROOT_DIR}/res/desktop/org.lenmus.lenmus.desktop
-            DESTINATION "/usr/share/applications"
+            DESTINATION "@TERMUX_PREFIX@/share/applications"
             COMPONENT main )
     # Install icon
     install(FILES ${LENMUS_ROOT_DIR}/res/icons/lenmus.png
-            DESTINATION "/usr/share/pixmaps"
+            DESTINATION "@TERMUX_PREFIX@/share/pixmaps"
             COMPONENT main )
 
     # Install AppStream upstream metadata
     install(FILES ${LENMUS_ROOT_DIR}/res/other/org.lenmus.lenmus.appdata.xml
-            DESTINATION "/usr/share/metainfo"
+            DESTINATION "@TERMUX_PREFIX@/share/metainfo"
             COMPONENT main )
 endif(UNIX AND NOT APPLE)
 
