--- a/include/lenmus_content_box_ctrol.h
+++ b/include/lenmus_content_box_ctrol.h
@@ -129,7 +129,7 @@
     wxWindow* GetHTMLWindow();
     wxColour GetHTMLBackgroundColour() const;
     void SetHTMLBackgroundColour(const wxColour& WXUNUSED(clr));
-    void SetHTMLBackgroundImage(const wxBitmap& WXUNUSED(bmpBg));
+    void SetHTMLBackgroundImage(const wxBitmapBundle& WXUNUSED(bmpBg));
     void SetHTMLStatusText(const wxString& WXUNUSED(text));
     wxCursor GetHTMLCursor(HTMLCursor type) const;
     wxPoint GetRootCellCoords(size_t n) const;
--- a/src/app/lenmus_content_box_ctrol.cpp
+++ b/src/app/lenmus_content_box_ctrol.cpp
@@ -989,7 +989,7 @@
     // nothing to do
 }
 
-void ContentBoxCtrol::SetHTMLBackgroundImage(const wxBitmap& WXUNUSED(bmpBg))
+void ContentBoxCtrol::SetHTMLBackgroundImage(const wxBitmapBundle& WXUNUSED(bmpBg))
 {
     // nothing to do
 }
