--- a/runtime/lua/man.lua
+++ b/runtime/lua/man.lua
@@ -271,9 +271,9 @@
   -- inconsistently supported. Instead, call -w with a section and a name.
   local cmd
   if sect == '' then
-    cmd = { 'man', FIND_ARG, name }
+    cmd = { 'mandoc', FIND_ARG, name }
   else
-    cmd = { 'man', FIND_ARG, sect, name }
+    cmd = { 'mandoc', FIND_ARG, sect, name }
   end
 
   local lines = system(cmd, silent)
