--- a/configure
+++ b/configure
@@ -5911,7 +5911,7 @@
 if eval \${$as_ac_File+:} false; then :
   $as_echo_n "(cached) " >&6
 else
-  test "$cross_compiling" = yes &&
+  false &&
   as_fn_error $? "cannot check for file existence when cross compiling" "$LINENO" 5
 if test -r "$GCFILE.tar.gz"; then
   eval "$as_ac_File=yes"
@@ -5947,7 +5947,7 @@
    case _$ac_cv_use_gc in
    _|_system|_*[\\/]*)
       if test "x$ac_cv_use_gc" = "xsystem" ; then
-         INCL="-I. -I$prefix/include/gc -I/usr/include/gc"
+         INCL="-I. -I$prefix/include/gc"
          LIBS=$LIBS"-L$prefix/lib "
       else
          INCL="-I$ac_cv_use_gc/include/gc"
@@ -10239,7 +10239,7 @@
 fi
 
 
-if test "x$GCDIR" != "x" ; then
+if false; then
    as_ac_File=`$as_echo "ac_cv_file_$GCDIR.tar.gz" | $as_tr_sh`
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $GCDIR.tar.gz" >&5
 $as_echo_n "checking for $GCDIR.tar.gz... " >&6; }
