--- asymptote-2.95/configure	2024-11-25 14:49:19.000000000 +0530
+++ asymptote-2.95.mod/configure	2025-01-22 20:05:44.133806620 +0530
@@ -4591,7 +4591,7 @@
 then :
   printf %s "(cached) " >&6
 else $as_nop
-  test "$cross_compiling" = yes &&
+  false &&
   as_fn_error $? "cannot check for file existence when cross compiling" "$LINENO" 5
 if test -r "$VCPKG_ROOT/vcpkg"; then
   eval "$as_ac_File=yes"
@@ -6940,7 +6940,7 @@
 then :
   printf %s "(cached) " >&6
 else $as_nop
-  test "$cross_compiling" = yes &&
+  false &&
   as_fn_error $? "cannot check for file existence when cross compiling" "$LINENO" 5
 if test -r "$GCFILE.tar.gz"; then
   eval "$as_ac_File=yes"
@@ -7025,7 +7025,7 @@
       case _$ac_cv_use_gc in
       _|_system|_*[\\/]*)
          if test "x$ac_cv_use_gc" = "xsystem" ; then
-            INCL=$INCL" -I$prefix/include/gc -I/usr/include/gc "
+            INCL=$INCL" -I$prefix/include/gc "
             LIBS=$LIBS"-L$prefix/lib "
          else
             INCL=$INCL" -I$ac_cv_use_gc/include/gc "
