--- a/configure
+++ b/configure
@@ -15473,7 +15473,7 @@
 			 -Werror=missing-prototypes -Werror=nested-externs \
 			 -Werror=implicit-function-declaration \
 			 -Wmissing-declarations -Wno-pointer-sign \
-			 -Werror=format-security -Wbitwise -Wcast-to-as \
+			 -Wno-error=format-security -Wbitwise -Wcast-to-as \
 			 -Wdefault-bitfield-sign -Wdo-while -Wparen-string \
 			 -Wptr-subtraction-blows -Wreturn-void -Wtypesign \
 			 -Wstrict-prototypes -Wno-error=format-nonliteral "
