--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -119,7 +119,7 @@
 
 # ALSA
 find_package(ALSA)
-if(ALSA_FOUND OR ALSA)
+if(RTMIDI_API_ALSA)
   set(NEED_PTHREAD ON)
   list(APPEND INCDIRS ${ALSA_INCLUDE_DIR})
   list(APPEND LINKLIBS ALSA::ALSA)
@@ -150,7 +150,7 @@
 endif()
 
 # Android AMIDI
-if(ANDROID)
+if(RTMIDI_API_AMIDI)
   set(NEED_PTHREAD ON)
   set(JAVA_INCLUDE_PATH2 NotNeeded)
   set(JAVA_AWT_INCLUDE_PATH NotNeeded)
