diff -uNr tinyemu-2018-09-23/host_config.h tinyemu-2018-09-23.mod/host_config.h
--- tinyemu-2018-09-23/host_config.h	1970-01-01 03:00:00.000000000 +0300
+++ tinyemu-2018-09-23.mod/host_config.h	2018-10-04 18:07:24.374021311 +0300
@@ -0,0 +1,24 @@
+#ifndef HOST_CONFIG_H
+#define HOST_CONFIG_H
+
+#ifndef PREFIX
+# define PREFIX "/usr/local"
+#endif
+
+#ifndef ETC_DIR
+# define ETC_DIR PREFIX "/etc"
+#endif
+
+#ifndef TMP_DIR
+# define TMP_DIR PREFIX "/tmp"
+#endif
+
+#ifndef TUN_DEVICE
+# ifdef __ANDROID__
+#  define TUN_DEVICE "/dev/tun"
+# else
+#  define TUN_DEVICE "/dev/net/tun"
+# endif
+#endif
+
+#endif
