diff -uNr tinyemu-2018-09-23/riscv_cpu.c tinyemu-2018-09-23.mod/riscv_cpu.c
--- tinyemu-2018-09-23/riscv_cpu.c	2018-09-23 15:17:19.000000000 +0300
+++ tinyemu-2018-09-23.mod/riscv_cpu.c	2018-10-04 18:07:24.377354692 +0300
@@ -29,6 +29,8 @@
 #include <assert.h>
 #include <fcntl.h>
 
+#include "host_config.h"
+
 #include "cutils.h"
 #include "iomem.h"
 #include "riscv_cpu.h"
@@ -69,7 +71,7 @@
 static void log_vprintf(const char *fmt, va_list ap)
 {
     if (!log_file)
-        log_file = fopen("/tmp/riscemu.log", "wb");
+        log_file = fopen(TMP_DIR "/riscemu.log", "wb");
     vfprintf(log_file, fmt, ap);
 }
 #else
