diff -uNr tinyemu-2018-09-23/Makefile tinyemu-2018-09-23.mod/Makefile
--- tinyemu-2018-09-23/Makefile	2018-09-23 15:17:20.000000000 +0300
+++ tinyemu-2018-09-23.mod/Makefile	2018-10-04 18:07:55.647799869 +0300
@@ -26,10 +26,10 @@
 # (openssl) must be installed.
 CONFIG_FS_NET=y
 # SDL support (optional)
-CONFIG_SDL=y
+#CONFIG_SDL=y
 # if set, compile the 128 bit emulator. Note: the 128 bit target does
 # not compile if gcc does not support the int128 type (32 bit hosts).
-CONFIG_INT128=y
+#CONFIG_INT128=y
 # build x86 emulator
 CONFIG_X86EMU=y
 # win32 build (not usable yet)
@@ -44,11 +44,11 @@
 CROSS_PREFIX=
 EXE=
 endif
-CC=$(CROSS_PREFIX)gcc
+CC=$(CROSS_PREFIX)clang
 STRIP=$(CROSS_PREFIX)strip
-CFLAGS=-O2 -Wall -g -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -MMD
+CFLAGS=$(TERMUX_CFLAGS) -Wall -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -MMD -I.
 CFLAGS+=-D_GNU_SOURCE -DCONFIG_VERSION=\"$(shell cat VERSION)\"
-LDFLAGS=
+LDFLAGS=$(TERMUX_LDFLAGS)
 
 bindir=/usr/local/bin
 INSTALL=install
@@ -72,7 +72,6 @@
 
 ifndef CONFIG_WIN32
 EMU_OBJS+=fs_disk.o
-EMU_LIBS=-lrt
 endif
 ifdef CONFIG_FS_NET
 CFLAGS+=-DCONFIG_FS_NET
