diff -uNr tinyemu-2018-09-23/fs_net.c tinyemu-2018-09-23.mod/fs_net.c
--- tinyemu-2018-09-23/fs_net.c	2018-09-23 15:17:19.000000000 +0300
+++ tinyemu-2018-09-23.mod/fs_net.c	2018-10-04 18:07:24.367354547 +0300
@@ -30,6 +30,8 @@
 #include <sys/time.h>
 #include <ctype.h>
 
+#include "host_config.h"
+
 #include "cutils.h"
 #include "list.h"
 #include "fs.h"
@@ -2093,7 +2095,7 @@
 static FSDevice *fs_import_fs;
 #endif
 
-#define DEFAULT_IMPORT_FILE_PATH "/tmp"
+#define DEFAULT_IMPORT_FILE_PATH TMP_DIR
 
 FSDevice *fs_net_init(const char *url, void (*start_cb)(void *opaque),
                       void *start_opaque)
