diff -uNr poac-0.9.3/src/Git2/Exception.cc poac-0.9.3.mod/src/Git2/Exception.cc
--- poac-0.9.3/src/Git2/Exception.cc    2024-02-02 23:01:37.000000000 +0800
+++ poac-0.9.3.mod/src/Git2/Exception.cc        2024-03-24 13:22:34.353993706 +0800
@@ -3,6 +3,7 @@
 #include "../Rustify.hpp"
 
 #include <git2/errors.h>
+#include <git2/deprecated.h>
 #include <git2/version.h>
 
 namespace git2 {
@@ -25,7 +26,7 @@
   if (const git_error* error = git_error_last(); error != nullptr) {
     this->msg += error->message;
     this->cat = static_cast<git_error_t>(error->klass);
-    git_error_clear();
+    giterr_clear();
   }
 }
 
