diff -uNr dbus-glib-0.110/configure dbus-glib-0.110.mod/configure
--- dbus-glib-0.110/configure	2018-01-29 14:42:36.000000000 +0200
+++ dbus-glib-0.110.mod/configure	2019-07-13 17:05:40.542291530 +0300
@@ -12922,7 +12922,7 @@

 fi

-GLIB_GENMARSHAL=`$PKG_CONFIG --variable=glib_genmarshal glib-2.0`
+GLIB_GENMARSHAL=glib-genmarshal



diff -uNr dbus-glib-0.110/configure.ac dbus-glib-0.110.mod/configure.ac
--- dbus-glib-0.110/configure.ac	2018-01-29 14:42:18.000000000 +0200
+++ dbus-glib-0.110.mod/configure.ac	2019-07-13 17:04:09.085145079 +0300
@@ -265,7 +265,7 @@
   [Warn on use of APIs deprecated before GLib 2.32])
 PKG_CHECK_MODULES([GLIB], [glib-2.0 >= 2.40, gobject-2.0 >= 2.40, gio-2.0 >= 2.40])

-GLIB_GENMARSHAL=`$PKG_CONFIG --variable=glib_genmarshal glib-2.0`
+GLIB_GENMARSHAL=glib-genmarshal
 AC_SUBST(GLIB_GENMARSHAL)

 dnl GLib flags
