diff --git a/audio/out/ao_opensles.c b/audio/out/ao_opensles.c
index ea48de892e..2e637adf82 100644
--- a/audio/out/ao_opensles.c
+++ b/audio/out/ao_opensles.c
@@ -129,7 +129,7 @@ static int init(struct ao *ao)
     CHK((*p->output_mix)->Realize(p->output_mix, SL_BOOLEAN_FALSE));
 
     locator_buffer_queue.locatorType = SL_DATALOCATOR_BUFFERQUEUE;
-    locator_buffer_queue.numBuffers = 1;
+    locator_buffer_queue.numBuffers = 8;
 
     pcm.formatType = SL_DATAFORMAT_PCM;
     pcm.numChannels = 2;
