diff -u -r ../mpv-0.24.0/waftools/checks/custom.py ./waftools/checks/custom.py
--- ../mpv-0.24.0/waftools/checks/custom.py	2017-02-12 02:31:16.000000000 +0100
+++ ./waftools/checks/custom.py	2017-02-28 16:53:22.687772818 +0100
@@ -61,7 +61,7 @@
         ( '51obsd', 'lua51 >= 5.1.0'), # OpenBSD
         ( '51deb',  'lua5.1 >= 5.1.0'), # debian
         ( '51fbsd', 'lua-5.1 >= 5.1.0'), # FreeBSD
-        ( '52',     'lua >= 5.2.0 lua < 5.3.0' ),
+        ( '52',     'lua >= 5.2.0' ),
         ( '52arch', 'lua52 >= 5.2.0'), # Arch
         ( '52deb',  'lua5.2 >= 5.2.0'), # debian
         ( '52fbsd', 'lua-5.2 >= 5.2.0'), # FreeBSD
