diff --git a/wscript b/wscript
index 830201a6bd..543a30feea 100644
--- a/wscript
+++ b/wscript
@@ -154,7 +154,7 @@ main_dependencies = [
     }, {
         'name': '--android',
         'desc': 'Android environment',
-        'func': check_statement('android/api-level.h', '(void)__ANDROID__'),  # arbitrary android-specific header
+        'func': check_cc(lib=['android']),
     }, {
         'name': '--tvos',
         'desc': 'tvOS environment',
@@ -167,7 +167,7 @@ main_dependencies = [
         'desc': 'Android EGL support',
         'deps': 'android',
         'groups': [ 'gl' ],
-        'func': check_cc(lib=['android', 'EGL']),
+        'func': check_cc(lib=['EGL']),
     }, {
         'name': 'posix-or-mingw',
         'desc': 'development environment',
