diff -u -r ../mpv-0.28.2/wscript_build.py ./wscript_build.py
--- ../mpv-0.28.2/wscript_build.py	2018-02-13 02:07:05.000000000 +0000
+++ ./wscript_build.py	2018-06-24 00:28:49.967625363 +0000
@@ -409,7 +409,7 @@
         ( "video/out/opengl/context_drm_egl.c",  "egl-drm" ),
         ( "video/out/opengl/context_dxinterop.c","gl-dxinterop" ),
         ( "video/out/opengl/context_mali_fbdev.c","mali-fbdev" ),
-        ( "video/out/opengl/context_android.c",  "android" ),
+        ( "video/out/opengl/context_android.c",  "android && gl" ),
         ( "video/out/opengl/context_rpi.c",      "rpi" ),
         ( "video/out/opengl/context_vdpau.c",    "vdpau-gl-x11" ),
         ( "video/out/opengl/context_wayland.c",  "gl-wayland" ),
