diff -u -r ../mpv-0.28.2/wscript ./wscript
--- ../mpv-0.28.2/wscript	2018-02-13 03:07:05.000000000 +0100
+++ ./wscript	2018-06-25 14:48:26.288156610 +0200
@@ -154,9 +154,7 @@
         'name': '--android',
         'desc': 'Android environment',
         'func': compose_checks(
-            check_statement('android/api-level.h', '(void)__ANDROID__'),  # arbitrary android-specific header
-            check_cc(lib="android"),
-            check_cc(lib="EGL"),
+            check_statement('android/api-level.h', '(void)__ANDROID__')  # arbitrary android-specific header
         )
     }, {
         'name': 'posix-or-mingw',
@@ -813,7 +811,7 @@
         'name': 'egl-helpers',
         'desc': 'EGL helper functions',
         'deps': 'egl-x11 || mali-fbdev || rpi || gl-wayland || egl-drm || ' +
-                'egl-angle-win32 || android',
+                'egl-angle-win32',
         'func': check_true
     }
 ]
