diff -uNr wavemon-0.9.0/configure wavemon-0.9.0.mod/configure
--- wavemon-0.9.0/configure	2018-12-30 03:08:25.000000000 +0200
+++ wavemon-0.9.0.mod/configure	2019-07-03 16:41:58.114071620 +0300
@@ -3986,7 +3986,7 @@
 fi
 
 # Testing library functions
-for ac_func in gettimeofday ether_ntohost
+for ac_func in gettimeofday
 do :
   as_ac_var=`$as_echo "ac_cv_func_$ac_func" | $as_tr_sh`
 ac_fn_c_check_func "$LINENO" "$ac_func" "$as_ac_var"
