diff -uNr wavemon-0.9.4/utils.c wavemon-0.9.4.mod/utils.c
--- wavemon-0.9.4/utils.c	2021-09-18 02:25:37.000000000 +0200
+++ wavemon-0.9.4.mod/utils.c	2022-12-02 12:50:45.290264283 +0100
@@ -69,10 +69,6 @@
 /* Print mac-address translation from /etc/ethers if available */
 char *ether_lookup(const struct ether_addr *ea)
 {
-	static char hostname[BUFSIZ];
-
-	if (ether_ntohost(hostname, ea) == 0)
-		return hostname;
 	return ether_addr(ea);
 }
 
