diff -uNr wavemon-0.9.0/utils.c wavemon-0.9.0.mod/utils.c
--- wavemon-0.9.0/utils.c	2018-12-30 03:08:25.000000000 +0200
+++ wavemon-0.9.0.mod/utils.c	2019-07-03 16:40:46.503509776 +0300
@@ -39,10 +39,6 @@
 /* Print mac-address translation from /etc/ethers if available */
 char *ether_lookup(const struct ether_addr *ea)
 {
-	static char hostname[BUFSIZ];
-
-	if (ether_ntohost(hostname, ea) == 0)
-		return hostname;
 	return ether_addr(ea);
 }
 
