diff -u -r ../getmail-5.1/getmailcore/utilities.py ./getmailcore/utilities.py
--- ../getmail-5.1/getmailcore/utilities.py	2017-07-16 05:26:01.000000000 +0200
+++ ./getmailcore/utilities.py	2017-07-26 13:36:30.196152312 +0200
@@ -47,6 +47,7 @@
 import sys
 import tempfile
 import errno
+import shutil
 try:
     import subprocess
 except ImportError, o:
@@ -340,7 +341,7 @@
 
     # Move message file from Maildir/tmp to Maildir/new
     try:
-        os.link(fname_tmp, fname_new)
+        shutil.copy2(fname_tmp, fname_new)
         os.unlink(fname_tmp)
 
     except OSError:
