diff --git a/configure b/configure
index 8e304282a..354bfda29 100755
--- a/configure
+++ b/configure
@@ -612,20 +612,26 @@ if test_eq "${os_arch}" "i686"; then
     os_arch="i386"
 fi
 
-# set the default target platform and architecture
+# set the default target platform
 _target_plat_default=${os_host}
 if is_host "msys"; then
     _target_plat_default="mingw"
 elif is_host "freebsd"; then
     _target_plat_default="bsd"
 fi
+
+# set the default target architecture
 _target_arch_default=${os_arch}
 if is_host "msys" && test_nz "${MSYSTEM_CARCH}"; then
     _target_arch_default="${MSYSTEM_CARCH}"
+elif test_nz "${TERMUX_ARCH}"; then
+    _target_arch_default="${TERMUX_ARCH}"
 fi
 if test_eq "${_target_arch_default}" "i686"; then
     _target_arch_default="i386"
 fi
+
+# set the default target mode
 _target_mode_default="release"
 
 # set the default project generator and build program
diff --git a/core/src/demo/xmake.sh b/core/src/demo/xmake.sh
index 0a04b8bfb..2f091b6ef 100755
--- a/core/src/demo/xmake.sh
+++ b/core/src/demo/xmake.sh
@@ -34,6 +34,8 @@ target "demo"
         add_syslinks "ws2_32" "pthread" "m"
     elif is_plat "bsd"; then
         add_syslinks "pthread" "m"
+    elif test_nz "${TERMUX_ARCH}"; then
+        add_syslinks "m" "dl"
     else
         add_syslinks "pthread" "dl" "m" "c"
     fi
