--- bthread.h
+++ ../bthread.h
@@ -1,3 +1,5 @@
+#include <pthread.h>
+
 # define PTHREAD_CANCEL_ENABLE    	 0x00000010
 # define PTHREAD_CANCEL_DISABLE   	 0x00000000
 
@@ -9,4 +11,5 @@
 int pthread_setcancelstate (int , int *);
 int pthread_setcanceltype (int , int *);
 void pthread_testcancel (void);
-int pthread_cancel (pthread_t t);
\ No newline at end of file
+int pthread_cancel (pthread_t t);
+
