--- a/configure.ac
+++ b/configure.ac
@@ -510,11 +510,13 @@
 ]], [[]])],
 [
 	AC_MSG_RESULT(yes)
+	AS_CASE(["$target_os"], [*android*], [], [
 	target_os=${target_os}-android
+	])
 	AS_IF([test "x$cross_compiling" = xno], [
                 AC_MSG_CHECKING([for Android API version])
                 # hacky workaround: https://github.com/termux/termux-packages/issues/6176
-                rb_android_api=`getprop ro.build.version.sdk`
+                rb_android_api=24
                 AC_MSG_RESULT($rb_android_api)
                 RUBY_APPEND_OPTIONS(CPPFLAGS, -D__ANDROID_API__=$rb_android_api -Wno-macro-redefined)
 	])
