--- /home/vagrant/ruby-2.5.0/configure	2017-12-25 07:00:28.000000000 +0000
+++ src/configure	2018-02-14 19:06:28.742701948 +0000
@@ -7434,7 +7434,7 @@
 
     if test $gcc_major -ge 4; then :
 
-	extra_warning=-Werror=extra-tokens
+	extra_warning=
 
 else
 
@@ -7464,9 +7464,7 @@
 		 -Werror=pointer-arith \
 		 -Werror=write-strings \
 		 -Werror=declaration-after-statement \
-		 -Werror=shorten-64-to-32 \
 		 -Werror=implicit-function-declaration \
-		 -Werror=division-by-zero \
 		 -Werror=deprecated-declarations \
 		 -Werror=misleading-indentation \
 		 -Wno-packed-bitfield-compat \
