--- /home/vagrant/ruby-2.5.0/configure.ac	2017-12-15 09:43:40.000000000 +0000
+++ src/configure.ac	2018-02-14 19:07:11.272257521 +0000
@@ -742,7 +742,7 @@
 rb_cv_warnflags="$warnflags"
 AS_IF([test "$GCC:${warnflags+set}:no" = yes::no], [
     AS_IF([test $gcc_major -ge 4], [
-	extra_warning=-Werror=extra-tokens
+	extra_warning=
     ], [
 	extra_warning=
     ])
@@ -765,9 +765,7 @@
 		 -Werror=pointer-arith \
 		 -Werror=write-strings \
 		 -Werror=declaration-after-statement \
-		 -Werror=shorten-64-to-32 \
 		 -Werror=implicit-function-declaration \
-		 -Werror=division-by-zero \
 		 -Werror=deprecated-declarations \
 		 -Werror=misleading-indentation \
 		 -Wno-packed-bitfield-compat \
