diff -u -r ../ruby-2.1.2/dln.c ./dln.c
--- ../ruby-2.1.2/dln.c	2012-12-01 16:25:28.000000000 +0100
+++ ./dln.c	2014-06-27 10:10:16.539935164 +0200
@@ -1308,14 +1308,17 @@
 	void *handle;
 	void (*init_fct)();
 
-#ifndef RTLD_LAZY
-# define RTLD_LAZY 1
-#endif
-#ifdef __INTERIX
-# undef RTLD_GLOBAL
-#endif
-#ifndef RTLD_GLOBAL
-# define RTLD_GLOBAL 0
+#ifndef __ANDROID__
+  /* Android has RTLD_* as enum values and not defines */
+# ifndef RTLD_LAZY
+#  define RTLD_LAZY 1
+# endif
+# ifdef __INTERIX
+#  undef RTLD_GLOBAL
+# endif
+# ifndef RTLD_GLOBAL
+#  define RTLD_GLOBAL 0
+# endif
 #endif
 
 #ifdef __native_client__
