diff -u -r ../ruby-2.4.0/lib/rubygems/defaults.rb ./lib/rubygems/defaults.rb
--- ../ruby-2.4.0/lib/rubygems/defaults.rb	2016-06-24 06:13:11.000000000 +0200
+++ ./lib/rubygems/defaults.rb	2017-01-31 22:53:59.125074751 +0100
@@ -191,6 +191,9 @@
   # end
 
   def self.platform_defaults
-    {}
+     {
+         'install' => '--no-rdoc --no-ri --env-shebang',
+         'update' => '--no-rdoc --no-ri --env-shebang'
+     }
   end
 end
