--- /home/vagrant/ruby-2.5.0/lib/fileutils.rb	2017-12-22 08:00:10.000000000 +0000
+++ ./lib/fileutils.rb	2018-02-17 23:19:55.344655246 +0000
@@ -744,8 +744,8 @@
   #
   def compare_stream(a, b)
     bsize = fu_stream_blksize(a, b)
-    sa = String.new(capacity: bsize)
-    sb = String.new(capacity: bsize)
+    sa = String.new
+    sb = String.new
     begin
       a.read(bsize, sa)
       b.read(bsize, sb)
