diff -u -r ../ruby-2.4.0/lib/rubygems/install_update_options.rb ./lib/rubygems/install_update_options.rb
--- ../ruby-2.4.0/lib/rubygems/install_update_options.rb	2016-04-06 08:01:14.000000000 +0200
+++ ./lib/rubygems/install_update_options.rb	2017-01-31 22:48:03.877466602 +0100
@@ -232,7 +232,7 @@
   # Default options for the gem install command.
 
   def install_update_defaults_str
-    '--document=rdoc,ri --wrappers'
+    '--document=rdoc,ri --wrappers,--env-shebang'
   end
 
 end
