diff '--color=auto' -uNr a/process.c b/process.c
--- a/process.c	2024-12-25 07:43:20.000000000 +0000
+++ b/process.c	2025-02-04 23:00:22.000000000 +0000
@@ -4120,6 +4120,20 @@
     return result;
 }
 
+#ifdef __ANDROID__
+void __rb_atfork_prepare() {
+    rb_thread_release_fork_lock();
+}
+
+void __rb_atfork_parent() {
+    // NOOP
+}
+
+void __rb_atfork_child() {
+    // NOOP
+}
+#endif
+
 rb_pid_t
 rb_fork_ruby(int *status)
 {
@@ -4128,6 +4142,12 @@
     rb_pid_t pid;
     int try_gc = 1, err = 0;
     struct child_handler_disabler_state old;
+
+#ifdef __ANDROID__
+    if ((err = pthread_atfork(__rb_atfork_prepare, __rb_atfork_parent, __rb_atfork_child))) {
+        rb_bug_errno("pthread_atfork", err);
+    }
+#endif
 
     do {
         prefork();
@@ -4141,6 +4161,8 @@
         if (
 #if defined(__FreeBSD__)
             pid != 0 &&
+#elif defined(__ANDROID__)
+            false && /* the lock was released in the prepare hook registered at pthread_atfork */
 #endif
             true) {
             rb_thread_release_fork_lock();
