diff -u -r ../ruby-3.4.1/yjit/src/yjit.rs ./yjit/src/yjit.rs
--- ../ruby-3.4.1/yjit/src/yjit.rs	2024-12-25 07:43:20.000000000 +0000
+++ ./yjit/src/yjit.rs	2025-02-04 23:00:22.788803376 +0000
@@ -76,7 +76,7 @@
 
     // Make sure --yjit-perf doesn't append symbols to an old file
     if get_option!(perf_map).is_some() {
-        let perf_map = format!("/tmp/perf-{}.map", std::process::id());
+        let perf_map = format!("@TERMUX_PREFIX@/tmp/perf-{}.map", std::process::id());
         let _ = std::fs::remove_file(&perf_map);
         println!("YJIT perf map: {perf_map}");
     }
