diff -u -r ../ruby-3.3.2/yjit/src/codegen.rs ./yjit/src/codegen.rs
--- ../ruby-3.3.2/yjit/src/codegen.rs	2024-05-30 00:23:11.000000000 +0000
+++ ./yjit/src/codegen.rs	2024-06-10 02:06:30.788015891 +0000
@@ -261,7 +261,7 @@
 
     /// Flush addresses and symbols to /tmp/perf-{pid}.map
     fn flush_perf_symbols(&self, cb: &CodeBlock) {
-        let path = format!("/tmp/perf-{}.map", std::process::id());
+        let path = format!("@TERMUX_PREFIX@/tmp/perf-{}.map", std::process::id());
         let mut f = std::fs::File::options().create(true).append(true).open(path).unwrap();
         for sym in self.perf_map.borrow().iter() {
             if let (start, Some(end), name) = sym {
