--- a/cmake-proxies/cmake-modules/TenacityFunctions.cmake	2025-01-01 07:28:49.000000000 -0500
+++ b/cmake-proxies/cmake-modules/TenacityFunctions.cmake	2026-02-26 00:30:41.633177701 -0500
@@ -427,7 +427,7 @@
    target_link_options( ${TARGET} PRIVATE ${LOPTS} )
    target_link_libraries( ${TARGET} PUBLIC ${LIBRARIES} )
 
-   if( NOT CMAKE_SYSTEM_NAME MATCHES "Windows" )
+   if( NOT CMAKE_SYSTEM_NAME MATCHES "Windows" AND NOT ANDROID)
       add_custom_command(
          TARGET "${TARGET}"
          POST_BUILD
