--- a/libraries/lib-ffmpeg-support/wrappers/AVFrameWrapper.h	2025-01-01 07:28:49.000000000 -0500
+++ b/libraries/lib-ffmpeg-support/wrappers/AVFrameWrapper.h	2026-02-24 23:22:28.355009519 -0500
@@ -81,7 +81,6 @@
    virtual AVDictionaryWrapper GetMetadata() const noexcept = 0;
    virtual int GetDecodeErrorFlags() const noexcept = 0;
    virtual int GetChannels() const noexcept = 0;
-   virtual int GetPacketSize() const noexcept = 0;
 protected:
    const FFmpegFunctions& mFFmpeg;
    AVFrame* mAVFrame { nullptr };
