--- a/libraries/lib-ffmpeg-support/CMakeLists.txt	2026-02-11 00:08:39.000000000 -0500
+++ b/libraries/lib-ffmpeg-support/CMakeLists.txt	2026-02-22 22:42:06.619469852 -0500
@@ -79,6 +79,13 @@
       impl/avformat/61/AVFormatImpl.cpp
       impl/avutil/59/AVUtilImpl.cpp
 
+      impl/ffmpeg-8.0.0-single-header.h
+      impl/avutil/60/avconfig.h
+      impl/avcodec/62/AVCodecIDLookup.cpp
+      impl/avcodec/62/AVCodecImpl.cpp
+      impl/avformat/62/AVFormatImpl.cpp
+      impl/avutil/60/AVUtilImpl.cpp
+
       # Loaders
       impl/avcodec/AVCodecFunctionsLoader.cpp
       impl/avcodec/AVCodecFunctionsLoader.h

