--- a/src/main.c	2023-07-08 14:55:42.000000000 +0200
+++ b/src/main.c	2023-08-14 18:50:52.863338600 +0200
@@ -25,12 +25,9 @@
 #include "cmusfm.h"
 
 #include <errno.h>
-#include <spawn.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <sys/types.h>
-#include <sys/wait.h>
 #include <unistd.h>
 
 #include "cache.h"
@@ -119,21 +116,6 @@
 	printf("Grant access to your Last.fm account using the link below,"
 			" then press ENTER:\n  %s\n", url);
 
-#ifdef __APPLE__
-	char *cmd = "open";
-	char * const argv[] = { cmd, (char *)url, NULL };
-#else
-	char *cmd = "x-www-browser";
-	char * const argv[] = { cmd, (char *)url, NULL };
-#endif
-
-	pid_t pid;
-	int status;
-
-	/* spawn "open" command and forget */
-	if (posix_spawnp(&pid, cmd, NULL, NULL, argv, environ) == 0)
-		waitpid(pid, &status, WNOHANG);
-
 	getchar();
 
 	return 0;
