diff -u -r ../gnupg-2.1.11/g10/gpg.c ./g10/gpg.c
--- ../gnupg-2.1.11/g10/gpg.c	2016-01-26 07:05:55.000000000 -0500
+++ ./g10/gpg.c	2016-04-14 21:42:15.168898294 -0400
@@ -4669,6 +4669,9 @@
   if (opt.debug)
     gcry_control (GCRYCTL_DUMP_SECMEM_STATS );
 
+  /* XXX: Workaround for libgpg-error atexit()-callback not working. */
+  es_fflush(NULL);
+
   emergency_cleanup ();
 
   rc = rc? rc : log_get_errorcount(0)? 2 : g10_errors_seen? 1 : 0;
