diff -u -r ../gnupg-2.1.11/tools/gpgconf.c ./tools/gpgconf.c
--- ../gnupg-2.1.11/tools/gpgconf.c	2016-01-18 02:14:56.000000000 -0500
+++ ./tools/gpgconf.c	2016-04-14 20:06:56.058590579 -0400
@@ -393,5 +395,8 @@
     if (es_fclose (outfp))
       gc_error (1, errno, "error closing '%s'", opt.outfile);
 
+  /* XXX: Workaround for libgpg-error atexit()-callback not working. */
+  es_fflush(NULL);
+
   return 0;
 }
