--- helix/helix-syntax/build.rs	2021-08-19 21:14:05.097999871 +0530
+++ helix-patch/helix-syntax/build.rs	2021-08-19 23:43:08.417999841 +0530
@@ -67,7 +67,7 @@
     let mut config = cc::Build::new();
     config.cpp(true).opt_level(2).cargo_metadata(false);
     let compiler = config.get_compiler();
-    let mut command = Command::new(compiler.path());
+    let mut command = Command::new("@TERMUX_CC@");
     command.current_dir(src_path);
     for (key, value) in compiler.env() {
         command.env(key, value);
@@ -89,6 +89,11 @@
     } else {
         command
             .arg("-shared")
+            .arg("-L@TERMUX_PREFIX/lib")
+            .arg("-Wl,-rpath=@TERMUX_PREFIX@/lib")
+            .arg("-lc++")
+            .arg("-lm")
+            .arg("-lc++_shared")
             .arg("-fPIC")
             .arg("-fno-exceptions")
             .arg("-g")
