diff -uNr dropbear-2018.76/Makefile.in dropbear-2018.76.mod/Makefile.in
--- dropbear-2018.76/Makefile.in	2018-02-27 16:25:10.000000000 +0200
+++ dropbear-2018.76.mod/Makefile.in	2018-04-21 14:05:03.923792282 +0300
@@ -9,7 +9,7 @@
 # dbclient functionality, and includes the progress-bar functionality in scp.
 
 ifndef PROGRAMS
-	PROGRAMS=dropbear dbclient dropbearkey dropbearconvert
+	PROGRAMS=dropbear dbclient dropbearkey dropbearconvert scp
 endif
 
 STATIC_LTC=libtomcrypt/libtomcrypt.a
