+++ ./src/sshpty.c
@@ -21,6 +21,7 @@
 #include "dbutil.h"
 #include "errno.h"
 #include "sshpty.h"
+# define USE_DEV_PTMX 1
 
 /* Pty allocated with _getpty gets broken if we do I_PUSH:es to it. */
 #if defined(HAVE__GETPTY) || defined(HAVE_OPENPTY)
@@ -247,6 +248,7 @@
 void
 pty_release(const char *tty_name)
 {
+#if 0
 	if (chown(tty_name, (uid_t) 0, (gid_t) 0) < 0
 			&& (errno != ENOENT)) {
 		dropbear_log(LOG_ERR,
@@ -257,6 +259,7 @@
 		dropbear_log(LOG_ERR,
 			"chmod %.100s 0666 failed: %.100s", tty_name, strerror(errno));
 	}
+#endif
 }
 
 /* Makes the tty the processes controlling tty and sets it to sane modes. */
@@ -355,6 +358,7 @@
 void
 pty_setowner(struct passwd *pw, const char *tty_name)
 {
+#if 0
 	struct group *grp;
 	gid_t gid;
 	mode_t mode;
@@ -411,4 +415,5 @@
 			}
 		}
 	}
+#endif 0
 }
