diff -uNr dropbear-2018.76/svr-agentfwd.c dropbear-2018.76.mod/svr-agentfwd.c
--- dropbear-2018.76/svr-agentfwd.c	2018-02-27 16:25:12.000000000 +0200
+++ dropbear-2018.76.mod/svr-agentfwd.c	2018-04-21 13:46:57.660403020 +0300
@@ -41,7 +41,7 @@
 #include "listener.h"
 #include "auth.h"
 
-#define AGENTDIRPREFIX "/tmp/dropbear-"
+#define AGENTDIRPREFIX "@TERMUX_PREFIX@/tmp/dropbear-"
 
 static int send_msg_channel_open_agent(int fd);
 static int bindagent(int fd, struct ChanSess * chansess);
