--- ../cache/gc-7.6.2/include/private/gcconfig.h	2017-12-23 08:46:29.000000000 +0000
+++ ./include/private/gcconfig.h	2017-12-30 07:35:01.202778219 +0000
@@ -2716,7 +2716,7 @@
 # define GETPAGESIZE() (unsigned)getpagesize()
 #endif
 
-#if defined(PLATFORM_ANDROID) && !(__ANDROID_API__ >= 23) \
+#if defined(PLATFORM_ANDROID) && !(__ANDROID_API__ >= 21) \
     && ((defined(MIPS) && (CPP_WORDSZ == 32)) \
         || defined(ARM32) || defined(I386) /* but not x32 */)
   /* tkill() exists only on arm32/mips(32)/x86. */
