diff -u -r ../gc-8.0.2/include/private/gcconfig.h ./include/private/gcconfig.h
--- ../gc-8.0.2/include/private/gcconfig.h	2018-12-23 21:28:41.000000000 +0000
+++ ./include/private/gcconfig.h	2019-01-20 21:31:50.050945027 +0000
@@ -2283,6 +2283,7 @@
 #     define HBLKSIZE 4096
 #   endif
 #   ifdef LINUX
+#     define SEARCH_FOR_DATA_START
 #     define OS_TYPE "LINUX"
 #     define LINUX_STACKBOTTOM
 #     define DYNAMIC_LOADING
