--- ../cache/gc-7.6.0/os_dep.c	2016-08-02 19:36:14.000000000 +0000
+++ ./os_dep.c	2017-10-19 03:43:03.800275032 +0000
@@ -426,7 +426,11 @@
 
 #   if (defined(LINUX) || defined(HURD)) && !defined(IGNORE_PROG_DATA_START)
       /* Try the easy approaches first: */
-#     ifdef PLATFORM_ANDROID
+	/* this workaround for gold linker when only aarch64 is using it in 
+	  termux breaks x86_64 and x86 builds and also arm w3m apply it to
+	 aarch64 only  */
+
+#   if defined(PLATFORM_ANDROID) && defined(__aarch64__) 
         /* Workaround for "gold" (default) linker (as of Android NDK r10e). */
         if ((word)__data_start < (word)_etext
             && (word)_etext < (word)__dso_handle) {
