diff -u -r ../gc-8.0.2/pthread_stop_world.c ./pthread_stop_world.c
--- ../gc-8.0.2/pthread_stop_world.c	2018-12-23 21:28:41.000000000 +0000
+++ ./pthread_stop_world.c	2019-01-20 19:48:08.502759151 +0000
@@ -502,7 +502,8 @@
 
 # ifdef USE_TKILL_ON_ANDROID
     EXTERN_C_BEGIN
-    extern int tkill(pid_t tid, int sig); /* from sys/linux-unistd.h */
+    static int tkill(pid_t tid, int sig) { return tgkill(-1, tid, sig); }
+
     EXTERN_C_END
 
     static int android_thread_kill(pid_t tid, int sig)
