diff -u -r ../gc-7.6.0/pthread_stop_world.c ./pthread_stop_world.c
--- ../gc-7.6.0/pthread_stop_world.c	2016-08-02 15:36:14.000000000 -0400
+++ ./pthread_stop_world.c	2016-08-14 16:52:41.334806302 -0400
@@ -365,7 +365,7 @@
 }
 
 # ifdef USE_TKILL_ON_ANDROID
-    extern int tkill(pid_t tid, int sig); /* from sys/linux-unistd.h */
+    static int tkill(pid_t tid, int sig) { return tgkill(-1, tid, sig); }
 
     static int android_thread_kill(pid_t tid, int sig)
     {
