diff -uNr nim-1.4.0/config/nim.cfg nim-1.4.0.mod/config/nim.cfg
--- nim-1.4.0/config/nim.cfg	2020-10-30 15:23:34.955027000 +0100
+++ nim-1.4.0.mod/config/nim.cfg	2020-10-30 15:22:59.586260700 +0100
@@ -136,13 +136,13 @@
 
 @if android:
   cc = clang
-  @if termux:
+#  @if termux:
     gcc.options.linker = "-landroid-glob"
     gcc.cpp.options.linker = "-landroid-glob"
     clang.options.linker = "-landroid-glob"
     clang.cpp.options.linker = "-landroid-glob"
     tcc.options.linker = "-landroid-glob"
-  @end
+#  @end
 @end
 
 @if nintendoswitch:
