--- ./compiler/installer.a.ini	2018-11-20 22:15:41.349488361 +0000
+++ ./compiler/installer.ini	2018-11-20 22:14:38.431003964 +0000
@@ -5,17 +5,8 @@
 Name: "Nim"
 Version: "$version"
 Platforms: """
-  windows: i386;amd64
-  linux: i386;amd64;powerpc64;arm;sparc;mips;mipsel;mips64;mips64el;powerpc;powerpc64el;arm64;riscv64
-  macosx: i386;amd64;powerpc64
-  solaris: i386;amd64;sparc;sparc64
-  freebsd: i386;amd64
-  netbsd: i386;amd64
-  openbsd: i386;amd64
-  dragonfly: i386;amd64
-  haiku: i386;amd64
-  android: i386;arm;arm64
-  nintendoswitch: arm64
+  linux: i386;amd64;arm;arm64
+  android: i386;amd64;arm;arm64
 """
 
 Authors: "Andreas Rumpf"
