--- ../asd/nim-0.17.2/compiler/installer.ini	2017-09-07 18:45:45.000000000 +0000
+++ ./compiler/installer.ini	2017-11-24 21:57:57.121844345 +0000
@@ -5,16 +5,8 @@
 Name: "Nim"
 Version: "$version"
 Platforms: """
-  windows: i386;amd64
-  linux: i386;amd64;powerpc64;arm;sparc;mips;mipsel;mips64;mips64el;powerpc;powerpc64el;arm64
-  macosx: i386;amd64;powerpc64
-  solaris: i386;amd64;sparc;sparc64
-  freebsd: i386;amd64
-  netbsd: i386;amd64
-  openbsd: i386;amd64
-  dragonfly: i386;amd64
-  haiku: i386;amd64
-  android: i386;arm;arm64
+  linux: i386;amd64;arm;arm64
+  android: i386;arm;arm64;amd64
 """
 
 Authors: "Andreas Rumpf"
