diff -uNr nim-0.20.0/compiler/installer.ini nim-0.20.0.mod/compiler/installer.ini
--- nim-0.20.0/compiler/installer.ini	2019-06-06 22:33:02.000000000 +0300
+++ nim-0.20.0.mod/compiler/installer.ini	2019-06-08 02:16:49.950336896 +0300
@@ -5,17 +5,8 @@
 Name: "Nim"
 Version: "$version"
 Platforms: """
-  windows: i386;amd64
-  linux: i386;alpha;amd64;powerpc64;arm;sparc;sparc64;m68k;mips;mipsel;mips64;mips64el;powerpc;powerpc64el;arm64;riscv64
-  macosx: i386;amd64;powerpc64
-  solaris: i386;amd64;sparc;sparc64
-  freebsd: i386;amd64
-  netbsd: i386;amd64
-  openbsd: i386;amd64
-  dragonfly: i386;amd64
-  haiku: i386;amd64
-  android: i386;arm;arm64
-  nintendoswitch: arm64
+  linux: i386;amd64;arm;arm64
+  android: i386;amd64;arm;arm64
 """
 
 Authors: "Andreas Rumpf"
