diff --git a/makefile b/makefile
index b5ef490..c19c558 100644
--- a/makefile
+++ b/makefile
@@ -2,10 +2,10 @@
 # Template is in tools/niminst/makefile.tmpl
 # To regenerate run ``niminst csource`` or ``koch csource``

-CC = gcc
-LINKER = gcc
+CC ?= gcc
+LD ?= gcc
 COMP_FLAGS = $(CPPFLAGS) $(CFLAGS) -w -O3 -fno-strict-aliasing
-LINK_FLAGS = $(LDFLAGS)
+LDFLAGS = $(LDFLAGS)
 binDir = bin

 koch := $(shell sh -c 'test -s ../koch.nim && echo "yes"')
@@ -19,46 +19,46 @@ uosname := $(shell sh -c 'uname -o | tr "[:upper:]" "[:lower:]"')

 ifeq ($(uos),linux)
   myos = linux
-  LINK_FLAGS += -ldl -lm
+  LDFLAGS += -ldl -lm
 endif
 ifeq ($(uos),dragonfly)
   myos = freebsd
-  LINK_FLAGS += -lm
+  LDFLAGS += -lm
 endif
 ifeq ($(uos),freebsd)
   myos= freebsd
   CC = clang
-  LINKER = clang
-  LINK_FLAGS += -lm
+  LD = clang
+  LDFLAGS += -lm
 endif
 ifeq ($(uos),openbsd)
   myos = openbsd
-  LINK_FLAGS += -lm
+  LDFLAGS += -lm
 endif
 ifeq ($(uos),netbsd)
   myos = netbsd
-  LINK_FLAGS += -lm
+  LDFLAGS += -lm
 endif
 ifeq ($(uos),darwin)
   myos = macosx
   CC = clang
-  LINKER = clang
-  LINK_FLAGS += -ldl -lm
+  LD = clang
+  LDFLAGS += -ldl -lm
   ifeq ($(HOSTTYPE),x86_64)
     ucpu = amd64
   endif
 endif
 ifeq ($(uos),aix)
   myos = aix
-  LINK_FLAGS += -dl -lm
+  LDFLAGS += -dl -lm
 endif
 ifeq ($(uos),solaris)
   myos = solaris
-  LINK_FLAGS += -ldl -lm -lsocket -lnsl
+  LDFLAGS += -ldl -lm -lsocket -lnsl
 endif
 ifeq ($(uos),sun)
   myos = solaris
-  LINK_FLAGS += -ldl -lm -lsocket -lnsl
+  LDFLAGS += -ldl -lm -lsocket -lnsl
 endif
 ifeq ($(uos),haiku)
   myos = haiku
@@ -69,7 +69,7 @@ endif

 ifeq ($(uosname),android)
   myos = android
-  LINK_FLAGS += -landroid-glob
+  LDFLAGS += -landroid-glob
 endif

 ifeq ($(ucpu),i386)
@@ -109,7 +109,7 @@ ifeq ($(ucpu),ppc64)
   mycpu = powerpc64
   ifeq ($(myos),linux)
     COMP_FLAGS += -m64
-    LINK_FLAGS += -m64
+    LDFLAGS += -m64
   endif
 endif
 ifeq ($(ucpu),powerpc)
@@ -560,7 +560,7 @@ endif

 $(binDir)/nim: $(oFiles)
 	@mkdir -p $(binDir)
-	$(LINKER) -o $@ $^ $(LINK_FLAGS)
+	$(LD) -o $@ $^ $(LDFLAGS)
 	@echo "SUCCESS"

 .PHONY: clean
