--- a/src/load_pat.cpp
+++ b/src/load_pat.cpp
@@ -52,8 +52,8 @@
 #define PATHFORPAT	"C:\\TIMIDITY\\INSTRUMENTS"
 #else
 #define DIRDELIM		'/'
-#define TIMIDITYCFG	"/etc/timidity.cfg" /*"/usr/share/timidity/timidity.cfg"*/
-#define PATHFORPAT	"/usr/share/timidity/instruments"
+#define TIMIDITYCFG	"@TERMUX_PREFIX@/share/timidity/timidity.cfg"
+#define PATHFORPAT	"@TERMUX_PREFIX@/share/timidity/instruments"
 #endif
 
 #define PAT_ENV_PATH2CFG			"MMPAT_PATH_TO_CFG"
