--- a/src/build_mozc.py
+++ b/src/build_mozc.py
@@ -162,7 +162,7 @@ def GetGypFileNames(options):
     gyp_file_names.extend(glob.glob('%s/win32/*/*.gyp' % OSS_SRC_DIR))
   elif options.target_platform == 'Linux':
     gyp_file_names.extend(glob.glob('%s/unix/emacs/*.gyp' % OSS_SRC_DIR))
-    gyp_file_names.extend(glob.glob('%s/unix/fcitx/*.gyp' % OSS_SRC_DIR))
+    #gyp_file_names.extend(glob.glob('%s/unix/fcitx/*.gyp' % OSS_SRC_DIR))
     gyp_file_names.extend(glob.glob('%s/unix/fcitx5/*.gyp' % OSS_SRC_DIR))
   gyp_file_names.sort()
   return gyp_file_names
