--- a/src/gyp/common.gypi
+++ b/src/gyp/common.gypi
@@ -221,6 +221,19 @@
           # <unordered_map> and <unordered_set>.
           '-Wno-deprecated',
         ],
+        'conditions': [
+          ['target_platform=="Linux"', {
+            'target_conditions' : [
+              ['_toolset=="host"', {
+                'include_dirs': [],
+              }],
+              ['_toolset=="target"', {
+                'include_dirs': [ '@TERMUX_PREFIX@/include' ],
+                'library_dirs': [ '@TERMUX_PREFIX@/lib' ],
+              }],
+            ],
+	  }],
+	],
       }],
       ['OS=="mac"', {
         'make_global_settings': [
@@ -273,4 +286,20 @@
       }],
     ],
   },
+  'conditions': [
+    ['target_platform=="Linux"', {
+      'make_global_settings': [
+        ['AR', '<!(which ar)'],
+        ['LD', '<!(which ld)'],
+        ['NM', '<!(which nm)'],
+        ['READELF', '<!(which readelf)'],
+        ['AR.host', '/usr/bin/ar'],
+        ['CC.host', '/usr/bin/gcc'],
+        ['CXX.host', '/usr/bin/g++'],
+        ['LD.host', '/usr/bin/ld'],
+        ['NM.host', '/usr/bin/nm'],
+        ['READELF.host', '/usr/bin/readelf'],
+      ],
+    }],
+  ],
 }
