diff -uNr RHash-1.3.8/parse_cmdline.c RHash-1.3.8.mod/parse_cmdline.c
--- RHash-1.3.8/parse_cmdline.c	2019-02-04 06:05:28.000000000 +0200
+++ RHash-1.3.8.mod/parse_cmdline.c	2019-03-01 20:33:22.425704712 +0200
@@ -547,7 +547,7 @@
 static const char* find_conf_file(void)
 {
 #ifndef SYSCONFDIR
-# define SYSCONFDIR "/etc"
+# define SYSCONFDIR "@TERMUX_PREFIX@/etc"
 #endif
 #define CONFIG_FILENAME "rhashrc"

