--- ./configure	2017-06-06 19:59:58.000000000 +0530
+++ ../configure	2017-06-09 22:41:43.475719606 +0530
@@ -4464,8 +4464,8 @@
 fi
 
 
-want_passwd=yes
-want_passwd_file=yes
+want_passwd=no
+want_passwd_file=no
 want_checkpassword=yes
 want_prefetch_userdb=yes
 
--- ./configure	2017-06-10 10:57:04.390830105 +0530
+++ ../configure	2017-06-10 11:14:06.734990847 +0530
@@ -26351,12 +26351,6 @@
 
 
 have_rquota=no
-if test -f /usr/include/rpcsvc/rquota.x && test -n "$RPCGEN"; then
-
-$as_echo "#define HAVE_RQUOTA /**/" >>confdefs.h
-
-  have_rquota=yes
-fi
  if test "$have_rquota" = "yes"; then
   HAVE_RQUOTA_TRUE=
   HAVE_RQUOTA_FALSE='#'
