diff -u -r ../ctags-ff448e2e2a08325c95f2e162ae50fb899392d99f/configure.ac ./configure.ac
--- ../ctags-ff448e2e2a08325c95f2e162ae50fb899392d99f/configure.ac	2016-11-28 06:04:11.000000000 -0500
+++ ./configure.ac	2016-11-28 17:57:01.052817678 -0500
@@ -7,7 +7,14 @@
 #	Process this file with autoconf to produce a configure script.
 
 AC_PREREQ([2.59])
-AC_INIT([universal-ctags],[0.0.0])
+# Avoid identifying as
+#   Universal Ctags 0.0.0, Copyright (C) 2015 Universal Ctags Team
+#   Universal Ctags is derived from Exuberant Ctags.
+#   Exuberant Ctags 5.8, Copyright (C) 1996-2009 Darren Hiebert
+# since that causes at least vim-easytags to identify the ctags
+# version as 0.0.0, see
+# https://github.com/xolox/vim-easytags/blob/master/autoload/xolox/easytags.vim
+AC_INIT([universal-ctags],[5.8])
 AM_INIT_AUTOMAKE([foreign subdir-objects])
 AM_SILENT_RULES([yes])
 AC_CONFIG_HEADERS([config.h])
@@ -593,7 +600,7 @@
 		       [have_libxml=yes
 		       AC_DEFINE(HAVE_LIBXML)],
 		       [have_libxml=no])
-AM_CONDITIONAL(HAVE_LIBXML, test "x$have_libxml" = xyes)
+AM_CONDITIONAL(HAVE_LIBXML, test "no" = xyes)
 
 AH_TEMPLATE([HAVE_JANSSON],
 	[Define this value if jansson is available.])
