diff -u -r ../ctags-61cc66cfc796e707cfb13c5fed493af280378c75/configure.ac ./configure.ac
--- ../ctags-61cc66cfc796e707cfb13c5fed493af280378c75/configure.ac	2019-04-12 19:56:22.000000000 +0000
+++ ./configure.ac	2019-04-13 07:08:50.478780864 +0000
@@ -7,7 +7,14 @@
 #	Process this file with autoconf to produce a configure script.
 
 AC_PREREQ([2.59])
-AC_INIT([universal-ctags],[0.0.0])
+# Avoid identifying as
+#   Universal Ctags 0.0.0, Copyright (C) 2015 Universal Ctags Team
+#   Universal Ctags is derived from Exuberant Ctags.
+#   Exuberant Ctags 5.8, Copyright (C) 1996-2009 Darren Hiebert
+# since that causes at least vim-easytags to identify the ctags
+# version as 0.0.0, see
+# https://github.com/xolox/vim-easytags/blob/master/autoload/xolox/easytags.vim
+AC_INIT([universal-ctags],[5.8])
 AM_INIT_AUTOMAKE([foreign subdir-objects tar-ustar])
 AM_SILENT_RULES([yes])
 AC_CONFIG_HEADERS([config.h])
@@ -409,7 +416,7 @@
 	sprintf(buffer, "%zu", i);
 	return !(buffer[0] == '9' && buffer[1] == '9' && buffer[2] == '9');
 }
-]])],[have_zu_formatter=yes],[have_zu_formatter=no])
+]])],[have_zu_formatter=yes],[have_zu_formatter=no], [have_zu_formatter=yes])
 AC_MSG_RESULT($have_zu_formatter)
 
 if test x$have_zu_formatter != xyes; then
@@ -594,16 +601,7 @@
 
 AH_TEMPLATE([HAVE_LIBXML],
 	[Define this value if libxml is available.])
-dnl About the condition of version
-dnl see https://mail.gnome.org/archives/xml/2010-February/msg00008.html
-AS_IF([test "x$enable_xml" != "xno"], [
-	PKG_CHECK_MODULES(LIBXML, [libxml-2.0 >= 2.7.7],
-			       [have_libxml=yes
-			       AC_DEFINE(HAVE_LIBXML)],
-			       [AS_IF([test "x$enable_xml" = "xyes"], [
-			           AC_MSG_ERROR([libxml2 not found])])])
-])
-AM_CONDITIONAL(HAVE_LIBXML, test "x$have_libxml" = xyes)
+AM_CONDITIONAL(HAVE_LIBXML, [test true = false])
 
 AC_ARG_ENABLE([json],
 	[AS_HELP_STRING([--disable-json],
