--- a/electron/shell/app/electron_main_delegate.cc
+++ b/electron/shell/app/electron_main_delegate.cc
@@ -19,7 +19,7 @@
 #include "base/logging.h"
 #include "base/path_service.h"
 #include "base/strings/cstring_view.h"
-#include "base/strings/string_number_conversions.cc"
+#include "base/strings/string_number_conversions.h"
 #include "base/strings/string_util_internal.h"
 #include "chrome/common/chrome_paths.h"
 #include "chrome/common/chrome_switches.h"
