--- ./crates/io_base/Cargo.toml.orig	2022-09-12 18:10:19.409234320 +0200
+++ ./crates/io_base/Cargo.toml	2022-09-12 18:10:47.816387800 +0200
@@ -16,7 +16,7 @@
 edition = "2018"
 
 [dependencies]
-app_dirs2 = "^2.3"
+app_dirs2 = { package = "app_dirs", version = "1.2.1" }
 flate2 = { version = "^1.0.19", default-features = false, features = ["zlib"] }
 libc = "^0.2" # for EISDIR :-(
 sha2 = "^0.9" # for digest computations
