--- a/electrum/daemon.py
+++ b/electrum/daemon.py
@@ -606,7 +606,7 @@
         threading.current_thread().name = 'GUI'
         gui_name = self.config.GUI_NAME
         if gui_name in ['lite', 'classic']:
-            gui_name = 'qt'
+            gui_name = 'text'
         self._plugins = Plugins(self.config, gui_name)  # init plugins
         self.logger.info(f'launching GUI: {gui_name}')
         try:
--- a/electrum/simple_config.py
+++ b/electrum/simple_config.py
@@ -1061,7 +1061,7 @@
     RPC_SOCKET_TYPE = ConfigVar('rpcsock', default='auto', type_=str)
     RPC_SOCKET_FILEPATH = ConfigVar('rpcsockpath', default=None, type_=str)
 
-    GUI_NAME = ConfigVar('gui', default='qt', type_=str)
+    GUI_NAME = ConfigVar('gui', default='text', type_=str)
     GUI_LAST_WALLET = ConfigVar('gui_last_wallet', default=None, type_=str)
 
     GUI_QT_COLOR_THEME = ConfigVar(
