--- a/electrum/daemon.py
+++ b/electrum/daemon.py
@@ -646,7 +646,7 @@
 
     def run_gui(self, config, plugins):
         threading.current_thread().name = 'GUI'
-        gui_name = config.get('gui', 'qt')
+        gui_name = config.get('gui', 'text')
         if gui_name in ['lite', 'classic']:
             gui_name = 'qt'
         self.logger.info(f'launching GUI: {gui_name}')
--- a/electrum/electrum
+++ b/electrum/electrum
@@ -426,7 +426,7 @@
         configure_logging(config)
         fd = daemon.get_file_descriptor(config)
         if fd is not None:
-            plugins = init_plugins(config, config.get('gui', 'qt'))
+            plugins = init_plugins(config, config.get('gui', 'text'))
             d = daemon.Daemon(config, fd)
             try:
                 d.run_gui(config, plugins)
