diff --git a/CMakeLists.txt b/CMakeLists.txt
index 553e0a7..e0bbff2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -111,7 +111,6 @@ set_package_properties(Qt6Quick3D PROPERTIES
     TYPE RUNTIME
 )
 
-find_package(Corrosion REQUIRED)
 
 add_definitions(
     -DQT_DEPRECATED_WARNINGS
diff --git a/kdeds/kameleon/CMakeLists.txt b/kdeds/kameleon/CMakeLists.txt
index 0af95d3..a3ec5c3 100644
--- a/kdeds/kameleon/CMakeLists.txt
+++ b/kdeds/kameleon/CMakeLists.txt
@@ -51,7 +51,3 @@ ecm_qt_declare_logging_category(kameleonhelper
                       EXPORT KDEPLASMAADDONS
                   )
 
-# FreeBSD is missing dependencies to CI.
-if(NOT CMAKE_SYSTEM_NAME MATCHES FreeBSD)
-    add_subdirectory(qmk/kameleon-qmk-helper)
-endif()
