--- ./src/main.c	2021-07-02 18:16:01.000000000 +0530
+++ ./src/main.c.mod	2021-08-03 07:01:36.763978266 +0530
@@ -351,11 +351,11 @@
   */
   db_unsave_encryption_key();
 #endif
-#if defined(_WIN32) || (defined(__BIONIC__) && !defined(FOSSIL_HAVE_GETPASS))
+#if defined(__ANDROID__)
   /*
   ** Free the secure getpass() buffer now.
   */
-  freepass();
+  fossil_freepass();
 #endif
 #if defined(_WIN32) && !defined(_WIN64) && defined(FOSSIL_ENABLE_TCL) && \
     defined(USE_TCL_STUBS)
