diff -uNr rinetd-0973bd2d1c187da9706fd83c8aa68fabc1775a31/Makefile rinetd-0973bd2d1c187da9706fd83c8aa68fabc1775a31.mod/Makefile
--- rinetd-0973bd2d1c187da9706fd83c8aa68fabc1775a31/Makefile	2019-10-22 14:45:57.000000000 +0300
+++ rinetd-0973bd2d1c187da9706fd83c8aa68fabc1775a31.mod/Makefile	2020-02-29 14:02:04.947085665 +0200
@@ -1,9 +1,10 @@
-CFLAGS=-DLINUX -g
+CC?=gcc
+CFLAGS+=-DLINUX -g
+PREFIX?=/usr/local
 
 rinetd: rinetd.o match.o
-	gcc rinetd.o match.o -o rinetd
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) rinetd.o match.o -o rinetd
 
 install: rinetd
-	install -m 700 rinetd /usr/sbin
-	install -m 644 rinetd.8 /usr/man/man8
-
+	install -Dm 755 rinetd $(PREFIX)/bin/rinetd
+	install -Dm 644 rinetd.8 $(PREFIX)/share/man/man8/rinetd.8
