--- rinetd-0.73/src/rinetd.h
+++ rinetd-0.73.mod/src/rinetd.h
@@ -17,8 +17,8 @@
 static int const RINETD_LISTEN_BACKLOG = 128;
 static int const RINETD_DEFAULT_UDP_TIMEOUT = 72;
 
-#define RINETD_CONFIG_FILE "/etc/rinetd.conf"
-#define RINETD_PID_FILE "/var/run/rinetd.pid"
+#define RINETD_CONFIG_FILE "@TERMUX_PREFIX@/etc/rinetd.conf"
+#define RINETD_PID_FILE "@TERMUX_PREFIX@/var/run/rinetd.pid"
 
 /* Global configuration */
 
