diff --git a/src/rinetd.h b/src/rinetd.h
index 79be926..95aff52 100644
--- a/src/rinetd.h
+++ b/src/rinetd.h
@@ -16,8 +16,8 @@ static int const RINETD_BUFFER_SIZE = 16384;
 static int const RINETD_LISTEN_BACKLOG = 128;
 static int const RINETD_DEFAULT_UDP_TIMEOUT = 72;
 
-#define RINETD_CONFIG_FILE "/etc/rinetd.conf"
-#define RINETD_PID_FILE "/var/run/rinetd.pid"
+#define RINETD_CONFIG_FILE "@TERMUX_PREFIX@/etc/rinetd.conf"
+#define RINETD_PID_FILE "@TERMUX_PREFIX@/var/run/rinetd.pid"
 
 /* Global configuration */
 
