From d9b94967df710d607b6f31ff096f1e8627420413 Mon Sep 17 00:00:00 2001
From: yangfl <yangfl@users.noreply.github.com>
Date: Mon, 7 Jan 2019 16:49:10 +0800
Subject: [PATCH 04/10] willuslib/CMakeLists.txt: typo

---
 willuslib/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/willuslib/CMakeLists.txt b/willuslib/CMakeLists.txt
index 2bbe248..641626b 100644
--- a/willuslib/CMakeLists.txt
+++ b/willuslib/CMakeLists.txt
@@ -29,6 +29,6 @@ message("-- Willuslib sources: ${WILLUSLIB_SRC}")
 
 add_library(willuslib ${WILLUSLIB_SRC})
 
-# HAVE_GOCR_LIB: ocrjocr.c
+# HAVE_GOCR_LIB: ocrgocr.c
 # HAVE_TESSERACT_LIB: ocrtess.c 
 
-- 
2.28.0

