From 4bde754a3fab11b4c7d0e966dca2ba7a3cee7666 Mon Sep 17 00:00:00 2001
From: yangfl <yangfl@users.noreply.github.com>
Date: Mon, 7 Jan 2019 16:49:10 +0800
Subject: [PATCH 4/7] willuslib/CMakeLists.txt: typo

---
 willuslib/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/willuslib/CMakeLists.txt b/willuslib/CMakeLists.txt
index 463bbc9..500c926 100644
--- a/willuslib/CMakeLists.txt
+++ b/willuslib/CMakeLists.txt
@@ -6,7 +6,7 @@ include_directories(..)
 set(WILLUSLIB_SRC
     ansi.c array.c bmp.c bmpdjvu.c bmpmupdf.c dtcompress.c filelist.c
     fontdata.c fontrender.c gslpolyfit.c linux.c math.c mem.c ocr.c
-    ocrjocr.c ocrtess.c pdfwrite.c point2d.c render.c strbuf.c string.c
+    ocrgocr.c ocrtess.c pdfwrite.c point2d.c render.c strbuf.c string.c
     token.c wfile.c wgs.c wgui.c willusversion.c win.c winbmp.c
     wincomdlg.c winmbox.c winshell.c wmupdf.c wmupdfinfo.c wpdf.c wsys.c
     wzfile.c wleptonica.c
@@ -26,6 +26,6 @@ message("-- Willuslib sources: ${WILLUSLIB_SRC}")
 
 add_library(willuslib ${WILLUSLIB_SRC})
 
-# HAVE_GOCR_LIB: ocrjocr.c
+# HAVE_GOCR_LIB: ocrgocr.c
 # HAVE_TESSERACT_LIB: ocrtess.c 
 
-- 
2.20.1

