From 81b02c399d9acbb347f6822f8f5107cac6b62e9d Mon Sep 17 00:00:00 2001
From: yangfl <yangfl@users.noreply.github.com>
Date: Mon, 7 Jan 2019 16:52:13 +0800
Subject: [PATCH 06/10] k2pdfopt.h: dst_ocr_visibility_flags workaround

---
 k2pdfoptlib/k2pdfopt.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/k2pdfoptlib/k2pdfopt.h b/k2pdfoptlib/k2pdfopt.h
index 0aae31e..0183d37 100644
--- a/k2pdfoptlib/k2pdfopt.h
+++ b/k2pdfoptlib/k2pdfopt.h
@@ -267,7 +267,7 @@ typedef struct
     int text_only; /* Do not send figures to output file */
 
     /* OCR */
-#ifdef HAVE_OCR_LIB
+#if 1
     char ocrout[128];
     int dst_ocr;
     int ocrvbb;             /* New in v2.53 -ocrvbb option */
-- 
2.28.0

