diff -uNr k2pdfopt_v2.53/CMakeLists.txt k2pdfopt_v2.53.mod/CMakeLists.txt
--- k2pdfopt_v2.53/CMakeLists.txt	2020-06-01 01:34:34.000000000 +0300
+++ k2pdfopt_v2.53.mod/CMakeLists.txt	2022-07-02 18:19:48.605603955 +0300
@@ -128,7 +128,7 @@
 
 # ms
 add_executable(k2pdfopt k2pdfopt.c)
-target_link_libraries (k2pdfopt k2pdfoptlib willuslib ${K2PDFOPT_LIB} pthread)
+target_link_libraries (k2pdfopt k2pdfoptlib willuslib ${K2PDFOPT_LIB})
 
 
 message("")
