diff -uNr k2pdfopt_v2.53/willuslib/linux.c k2pdfopt_v2.53.mod/willuslib/linux.c
--- k2pdfopt_v2.53/willuslib/linux.c	2020-06-12 19:55:01.000000000 +0300
+++ k2pdfopt_v2.53.mod/willuslib/linux.c	2022-06-28 12:49:39.692922903 +0300
@@ -33,7 +33,7 @@
 #include <unistd.h>
 /* 12-6-18:  Apparently sys/termios.h no longer needed to compile in some cases. */
 // #if (defined(__MACH__) || !defined(__GNUC__) || __GNUC__ < 7)
-#include <sys/termios.h>
+#include <termios.h>
 // #endif
 #include <sys/wait.h>
 #include <sys/types.h>
