diff -uNr html2text-2.0.0/Makefile.in html2text-2.0.0.mod/Makefile.in
--- html2text-2.0.0/Makefile.in	2020-06-18 09:11:22.000000000 +0300
+++ html2text-2.0.0.mod/Makefile.in	2020-06-21 17:55:30.409483227 +0300
@@ -32,7 +32,7 @@
 YFLAGS =
 
 INSTALLER = install
-PREFIX    = /usr/local
+PREFIX    ?= /usr/local
 BINDIR    = $(PREFIX)/bin
 MANDIR    = $(PREFIX)/man
 DOCDIR    = $(PREFIX)/share/doc/html2text
