From a54bac8f55365be24a0804cbb1294bbe744fb791 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fr=C3=A9d=C3=A9ric=20Mangano?= <fmang@mg0.fr>
Date: Fri, 1 Nov 2024 10:30:12 +0900
Subject: [PATCH] Fix the warning on comparison of size_t and long

---
 src/system.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/system.cc b/src/system.cc
index b04cb9e..40dad53 100644
--- a/src/system.cc
+++ b/src/system.cc
@@ -127,7 +127,7 @@ ot::byte_string ot::slurp_binary_file(const char* filename)
 
 	byte_string content;
 	long file_size = get_file_size(f.get());
-	if (file_size == -1) {
+	if (file_size < 0) {
 		// Read the input stream block by block and resize the output byte string as needed.
 		uint8_t buffer[4096];
 		while (!feof(f.get())) {
@@ -140,7 +140,7 @@ ot::byte_string ot::slurp_binary_file(const char* filename)
 	} else {
 		// Lucky! We know the file size, so let’s slurp it at once.
 		content.resize(file_size);
-		if (fread(content.data(), 1, file_size, f.get()) < file_size)
+		if (fread(content.data(), 1, file_size, f.get()) < size_t(file_size))
 			throw status { st::standard_error,
 				       "Could not read '"s + filename + "': " + strerror(errno) + "." };
 	}
