--- ../CMakeLists.txt.orig	2022-02-07 21:37:34.253637300 +0000
+++ ./CMakeLists.txt	2022-02-07 21:37:47.633618078 +0000
@@ -36,7 +36,7 @@
 
 set(SOURCES src/util/StringUtils.cpp)
 
-add_library(p8-platform ${SOURCES} ${PLAT_SOURCES})
+add_library(p8-platform SHARED ${SOURCES} ${PLAT_SOURCES})
 target_link_libraries(p8-platform ${p8-platform_LIBRARIES})
 set_target_properties(p8-platform PROPERTIES VERSION ${p8-platform_VERSION_MAJOR}.${p8-platform_VERSION_MINOR}.${p8-platform_VERSION_PATCH}
                                               SOVERSION ${p8-platform_VERSION_MAJOR})
