--- ../inlines.cpp.orig	2019-12-07 11:13:58.242315736 +0000
+++ ./src/inlines.cpp	2019-12-07 11:14:56.612224731 +0000
@@ -33,7 +33,7 @@
 // If using cygwin then we suppress the "multiple definition" errors by
 // ignoring this section completely; the cygwin linker seems to handle
 // the stubs correctly at link time by itself
-#if !defined(__CYGWIN__)
+#if !defined(__CYGWIN__) && !defined(__ANDROID__)
 
 // Undefine GEOS_INLINE so that .inl files
 // will be ready for an implementation file
