diff -u -r ../lighttpd-1.4.35/src/network.c ./src/network.c
--- ../lighttpd-1.4.35/src/network.c	2014-03-06 15:08:00.000000000 +0100
+++ ./src/network.c	2014-03-13 01:22:10.235965757 +0100
@@ -356,7 +356,6 @@
 				goto error_free_socket;
 			}
 			memcpy(srv_socket->addr.un.sun_path, host, hostlen);
-		}
 		srv_socket->addr.un.sun_family = AF_UNIX;
 
 #ifdef SUN_LEN
@@ -365,6 +364,7 @@
 		/* stevens says: */
 		addr_len = hostlen + sizeof(srv_socket->addr.un.sun_family);
 #endif
+		}
 
 		/* check if the socket exists and try to connect to it. */
 		if (-1 != (fd = connect(srv_socket->fd, (struct sockaddr *) &(srv_socket->addr), addr_len))) {
