diff -uNr cronie-1.7.0/anacron/global.h cronie-1.7.0.mod/anacron/global.h
--- cronie-1.7.0/anacron/global.h	2023-07-26 04:41:47.000000000 +0800
+++ cronie-1.7.0.mod/anacron/global.h	2023-10-14 11:21:23.674892041 +0800
@@ -33,7 +33,7 @@
 #define DEBUG_LEVEL LOG_DEBUG     /* only used when DEBUG is defined */
 
 /* Mail interface.  (All MTAs should supply this command) */
-#define SENDMAIL "/usr/sbin/sendmail"
+#define SENDMAIL "@TERMUX_PREFIX@/bin/sendmail"
 
 /* End of user-configurable section */
 
diff -uNr cronie-1.7.0/configure cronie-1.7.0.mod/configure
--- cronie-1.7.0/configure	2023-10-13 17:00:09.000000000 +0800
+++ cronie-1.7.0.mod/configure	2023-10-14 11:17:58.102892119 +0800
@@ -5961,7 +5961,7 @@
 
 
 
-printf "%s\n" "#define MAILARG \"/usr/sbin/sendmail\"" >>confdefs.h
+printf "%s\n" "#define MAILARG \"@TERMUX_PREFIX@/bin/sendmail\"" >>confdefs.h
 
 
 
diff -uNr cronie-1.7.0/src/cron.c cronie-1.7.0.mod/src/cron.c
--- cronie-1.7.0/src/cron.c	2023-07-26 04:41:47.000000000 +0800
+++ cronie-1.7.0.mod/src/cron.c	2023-10-14 11:19:16.262892089 +0800
@@ -302,7 +302,7 @@
 
 	log_it("CRON", getpid(), "STARTUP", PACKAGE_VERSION, 0);
 
-	if (!SyslogOutput && MailCmd[0] == '\0' && access("/usr/sbin/sendmail", X_OK) != 0) {
+	if (!SyslogOutput && MailCmd[0] == '\0' && access("@TERMUX_PREFIX@/bin/sendmail", X_OK) != 0) {
 		SyslogOutput=1;
 		log_it("CRON", pid, "INFO","Syslog will be used instead of sendmail.", 0);
 	}
diff -uNr cronie-1.7.0/src/crontab.c cronie-1.7.0.mod/src/crontab.c
--- cronie-1.7.0/src/crontab.c	2023-03-13 16:20:25.000000000 +0800
+++ cronie-1.7.0.mod/src/crontab.c	2023-10-14 11:19:43.314892079 +0800
@@ -613,7 +613,7 @@
 	if ((getuid() == geteuid()) && (getgid() == getegid())) {
 		tmpdir = getenv("TMPDIR");
 	}
-	return tmpdir ? tmpdir : "/tmp";
+	return tmpdir ? tmpdir : "@TERMUX_PREFIX@/tmp";
 }
 
 static char *host_specific_filename(const char *prefix, const char *suffix) {
diff -uNr cronie-1.7.0/src/pathnames.h cronie-1.7.0.mod/src/pathnames.h
--- cronie-1.7.0/src/pathnames.h	2023-03-13 16:20:25.000000000 +0800
+++ cronie-1.7.0.mod/src/pathnames.h	2023-10-14 11:20:45.206892055 +0800
@@ -50,15 +50,15 @@
 #define REBOOT_LOCK     PIDDIR "cron.reboot"
 
 #ifndef _PATH_BSHELL
-# define _PATH_BSHELL "/bin/sh"
+# define _PATH_BSHELL "@TERMUX_PREFIX@/bin/sh"
 #endif
 
 #ifndef _PATH_STDPATH
-# define _PATH_STDPATH "/usr/bin:/bin:/usr/sbin:/sbin"
+# define _PATH_STDPATH "@TERMUX_PREFIX@/bin"
 #endif
 
 #ifndef _PATH_TMP
-# define _PATH_TMP "/tmp"
+# define _PATH_TMP "@TERMUX_PREFIX@/tmp"
 #endif
 
 #ifndef _PATH_DEVNULL
