diff -uNr cronie-1.5.5/src/env.c cronie-1.5.5.mod/src/env.c
--- cronie-1.5.5/src/env.c	2019-04-15 11:41:50.000000000 +0300
+++ cronie-1.5.5.mod/src/env.c	2020-07-11 01:00:33.448944127 +0300
@@ -123,21 +123,20 @@
 int env_set_from_environ(char ***envpp) {
 	static const char *names[] = {
 		"LANG",
-		"LC_CTYPE",
-		"LC_NUMERIC",
-		"LC_TIME",
-		"LC_COLLATE",
-		"LC_MONETARY",
-		"LC_MESSAGES",
-		"LC_PAPER",
-		"LC_NAME",
-		"LC_ADDRESS",
-		"LC_TELEPHONE",
-		"LC_MEASUREMENT",
-		"LC_IDENTIFICATION",
-		"LC_ALL",
-		"LANGUAGE",
 		"RANDOM_DELAY",
+		"ANDROID_ART_ROOT",
+		"ANDROID_DATA",
+		"ANDROID_I18N_ROOT",
+		"ANDROID_ROOT",
+		"ANDROID_RUNTIME_ROOT",
+		"ANDROID_TZDATA_ROOT",
+		"BOOTCLASSPATH",
+		"DEX2OATBOOTCLASSPATH",
+		"EXTERNAL_STORAGE",
+		"LD_PRELOAD",
+		"PREFIX",
+		"TMPDIR",
+		"TZ",
 		NULL
 	};
 	const char **name;
