diff --git a/build.nims b/build.nims
index 9c3f812ef..0bdaee0e1 100755
--- a/build.nims
+++ b/build.nims
@@ -53,7 +53,7 @@ let
     # configuration options
     OPTIONS = {
         "arm"               : "--cpu:arm",
-        "arm64"             : "--cpu:arm64 --gcc.path:/usr/bin --gcc.exe:aarch64-linux-gnu-gcc --gcc.linkerexe:aarch64-linux-gnu-gcc",
+        "arm64"             : "--cpu:arm64",
         "debug"             : "-d:DEBUG --debugger:on --debuginfo --linedir:on",
         "dev"               : "--embedsrc:on -d:DEV --listCmd",
         "docgen"            : "-d:DOCGEN",
@@ -100,11 +100,13 @@ var
     IS_DEV              = false 
     MODE                = ""       
 
-    FLAGS*              = "--verbosity:1 --hints:on --hint:ProcessingStmt:off --hint:XCannotRaiseY:off --warning:GcUnsafe:off --warning:ProveInit:off --warning:ProveField:off --warning:Uninit:off " & 
+    FLAGS*              = "--cc:clang --clang.path:@TERMUX_STANDALONE_TOOLCHAIN@/bin --clang.exe:@TERMUX_HOST_PLATFORM@clang --clang.linkerexe:@TERMUX_HOST_PLATFORM@clang --passC:'@CFLAGS@' --passL:'@LDFLAGS@' " &
+                          "--verbosity:1 --hints:on --hint:ProcessingStmt:off --hint:XCannotRaiseY:off --warning:GcUnsafe:off --warning:ProveInit:off --warning:ProveField:off --warning:Uninit:off " & 
                           "--skipUserCfg:on --colors:off -d:danger " &
                           "--panics:off --mm:orc -d:useMalloc --checks:off " &
                           "-d:ssl --cincludes:extras --opt:speed --nimcache:.cache --passL:'-pthread' " & 
-                          "--path:src "
+                          "--path:src " &
+                          "@NIMFLAGS@ "
     CONFIG              ="@full"
 
     ARGS: seq[string]   = @[] 
@@ -549,4 +551,4 @@ except:
 #=======================================
 # This is the end,
 # my only friend, the end...
-#=======================================
\ No newline at end of file
+#=======================================
