diff --git a/src/library/Numbers.nim b/src/library/Numbers.nim
index 403bef8a7..5174991bb 100644
--- a/src/library/Numbers.nim
+++ b/src/library/Numbers.nim
@@ -949,7 +949,7 @@ proc defineSymbols*() =
             elif x.kind==Floating:
                 push(newLogical(x.f < 0.0))
             elif x.kind==Rational:
-                push(newLogical(x.rat.num < 0)):
+                push(newLogical(x.rat.num < 0))
             elif x.kind==Complex:
                 push(newLogical(x.z.re < 0.0 or (x.z.re == 0.0 and x.z.im < 0.0)))
 
@@ -1033,7 +1033,7 @@ proc defineSymbols*() =
             elif x.kind==Floating:
                 push(newLogical(x.f > 0.0))
             elif x.kind==Rational:
-                push(newLogical(x.rat.num > 0)):
+                push(newLogical(x.rat.num > 0))
             elif x.kind==Complex:
                 push(newLogical(x.z.re > 0.0 or (x.z.re == 0.0 and x.z.im > 0.0)))
     
@@ -1402,4 +1402,4 @@ proc defineSymbols*() =
 # Add Library
 #=======================================
 
-Libraries.add(defineSymbols)
\ No newline at end of file
+Libraries.add(defineSymbols)
