diff --git a/erts/emulator/asmjit/core/virtmem.cpp b/erts/emulator/asmjit/core/virtmem.cpp
index a1b0f95..d1eb1ad 100644
--- a/erts/emulator/asmjit/core/virtmem.cpp
+++ b/erts/emulator/asmjit/core/virtmem.cpp
@@ -518,7 +518,7 @@ enum class AnonymousMemoryStrategy : uint32_t {
 #if !defined(SHM_ANON)
 static const char* get_tmp_dir() noexcept {
   const char* tmp_dir = getenv("TMPDIR");
-  return tmp_dir ? tmp_dir : "/tmp";
+  return tmp_dir ? tmp_dir : "@TERMUX_PREFIX@/tmp";
 }
 #endif
 
