--- ./erts/emulator/nifs/common/socket_util.c	2021-06-24 17:00:43.000000000 +0530
+++ ./erts/emulator/nifs/common/socket_util.c.mod	2021-06-30 09:38:32.380499249 +0530
@@ -1392,18 +1392,6 @@
         break;
 #endif
 
-#if defined(PACKET_USER)
-    case PACKET_USER:
-        *ePktType = esock_atom_user;
-        break;
-#endif
-
-#if defined(PACKET_KERNEL)
-    case PACKET_KERNEL:
-        *ePktType = esock_atom_kernel;
-        break;
-#endif
-
         /* Unused? Not user space?
          * Also, has the same value as PACKET_USER,
          * so may result in a compiler error (at least
