diff -u -r ../otp-OTP-22.2.1/erts/emulator/nifs/common/socket_util.c ./erts/emulator/nifs/common/socket_util.c
--- ../otp-OTP-22.2.1/erts/emulator/nifs/common/socket_util.c	2019-12-18 16:48:36.000000000 +0000
+++ ./erts/emulator/nifs/common/socket_util.c	2020-01-02 21:53:42.278551000 +0000
@@ -2325,7 +2325,7 @@
         *ePktType = esock_atom_loopback;
         break;
 #endif
-
+#ifndef __ANDROID__
 #if defined(PACKET_USER)
     case PACKET_USER:
         *ePktType = esock_atom_user;
@@ -2337,7 +2337,7 @@
         *ePktType = esock_atom_kernel;
         break;
 #endif
-
+#endif
         /* Unused? Not user space?
          * Also, has the same value as PACKET_USER,
          * so may result in a compiler error (at least
