diff -u -r ../otp-OTP-21.0.5/lib/crypto/c_src/crypto.c ./lib/crypto/c_src/crypto.c
--- ../otp-OTP-21.0.5/lib/crypto/c_src/crypto.c	2018-08-10 17:02:51.000000000 +0000
+++ ./lib/crypto/c_src/crypto.c	2018-08-23 22:08:35.654582842 +0000
@@ -193,7 +193,8 @@
 #endif
 
 #if OPENSSL_VERSION_NUMBER >= PACKED_OPENSSL_VERSION(0,9,8,'h') \
-    && defined(HAVE_EC)
+    && defined(HAVE_EC) \
+    && !defined(OPENSSL_NO_ENGINE)
 /* If OPENSSL_NO_EC is set, there will be an error in ec.h included from engine.h
    So if EC is disabled, you can't use Engine either....
 */
