--- ./lib/kernel/src/os.erl	2021-06-24 17:00:43.000000000 +0530
+++ ./lib/kernel/src/os.erl.mod	2021-06-30 09:41:45.550499175 +0530
@@ -313,7 +313,7 @@
     %% We use an absolute path here because we do not want the path to be
     %% searched in case a stale NFS handle is somewhere in the path before
     %% the sh command.
-    {"/bin/sh -s unix:cmd", [out],
+    {"@TERMUX_PREFIX@/bin/sh -s unix:cmd", [out],
      %% We insert a new line after the command, in case the command
      %% contains a comment character.
      %%
