--- a/packages/next-swc/package.json
+++ b/packages/next-swc/package.json
@@ -32,7 +32,9 @@
         "x86_64-unknown-linux-musl",
         "aarch64-unknown-linux-musl",
         "aarch64-pc-windows-msvc",
-        "wasm32-wasip1-threads"
+        "wasm32-wasip1-threads",
+        "aarch64-linux-android",
+        "x86_64-linux-android"
       ]
     }
   },
--- a/packages/next/src/build/swc/index.ts
+++ b/packages/next/src/build/swc/index.ts
@@ -149,7 +149,6 @@ function checkVersionMismatch(pkgData: any) {
 // we'll not include native bindings for these platform at all.
 const knownDefaultWasmFallbackTriples = [
   'x86_64-unknown-freebsd',
-  'aarch64-linux-android',
   'arm-linux-androideabi',
   'armv7-unknown-linux-gnueabihf',
   'i686-pc-windows-msvc',
