diff -uNr bochs-2.6.9/bx_debug/Makefile.in bochs-2.6.9.mod/bx_debug/Makefile.in
--- bochs-2.6.9/bx_debug/Makefile.in    2016-08-13 01:06:14.000000000 +0800
+++ bochs-2.6.9.mod/bx_debug/Makefile.in        2019-08-26 21:47:49.744442900 +0800
@@ -58,7 +58,7 @@

 BX_INCLUDES = debug.h

-BX_INCDIRS = -I.. -I$(srcdir)/.. -I../@INSTRUMENT_DIR@ -I$(srcdir)/../@INSTRUMENT_DIR@ -I. -I$(srcdir)/.
+BX_INCDIRS = -I.. -I@TERMUX_PREFIX@/include -I$(srcdir)/.. -I../@INSTRUMENT_DIR@ -I$(srcdir)/../@INSTRUMENT_DIR@ -I. -I$(srcdir)/.

 all: libdebug.a
