--- bochs-2.8/configure	2024-03-10 12:29:37.000000000 +0530
+++ bochs-2.8.mod/configure	2025-01-24 17:41:16.822615995 +0530
@@ -23706,70 +23706,13 @@
 
     { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking if readline works without -lcurses" >&5
 printf %s "checking if readline works without -lcurses... " >&6; }
-    OLD_LIBS=$LIBS
-    LIBS="$LIBS -lreadline"
-    if test "$cross_compiling" = yes
-then :
-  { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-printf "%s\n" "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
-else $as_nop
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-      #include <stdio.h>
-      #include <readline/readline.h>
-      int main() { rl_initialize(); return 0; }
-
-_ACEOF
-if ac_fn_c_try_run "$LINENO"
-then :
+   
    { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 printf "%s\n" "yes" >&6; }
         rl_without_curses_ok=yes
-else $as_nop
-   { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: no" >&5
-printf "%s\n" "no" >&6; }
-
-fi
-rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
-  conftest.$ac_objext conftest.beam conftest.$ac_ext
-fi
-
-    { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking if readline works with -lcurses" >&5
-printf %s "checking if readline works with -lcurses... " >&6; }
-    LIBS="$LIBS -lcurses"
-    if test "$cross_compiling" = yes
-then :
-  { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-printf "%s\n" "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
-else $as_nop
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-      #include <stdio.h>
-      #include <readline/readline.h>
-      int main() { rl_initialize(); return 0; }
-
-_ACEOF
-if ac_fn_c_try_run "$LINENO"
-then :
-  { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-printf "%s\n" "yes" >&6; }
-       rl_with_curses_ok=yes
-else $as_nop
-   { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: no" >&5
-printf "%s\n" "no" >&6; }
 
-fi
-rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
-  conftest.$ac_objext conftest.beam conftest.$ac_ext
-fi
 
-    LIBS=$OLD_LIBS
+  
 
     if test "$rl_without_curses_ok" = yes; then
       { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: yes" >&5
