diff -uNr bochs-2.6.9/iodev/network/slirp/misc.cc bochs-2.6.9.mod/iodev/network/slirp/misc.cc
--- bochs-2.6.9/iodev/network/slirp/misc.cc     2016-12-14 04:22:28.000000000 +0800
+++ bochs-2.6.9.mod/iodev/network/slirp/misc.cc 2019-08-26 20:05:37.060533000 +0800
@@ -16,7 +16,7 @@
 #include "libslirp.h"

 #if BX_NETWORKING && BX_NETMOD_SLIRP
-
+#define ANDROID
 #ifdef DEBUG
 int slirp_debug = DBG_CALL|DBG_MISC|DBG_ERROR;
 #endif
