--- a/cmake/inputs/tiledb.pc.in
+++ b/cmake/inputs/tiledb.pc.in
@@ -1,6 +1,6 @@
 prefix="@CMAKE_INSTALL_PREFIX@"
 exec_prefix="${prefix}"
-libdir="${prefix}/@CMAKE_INSTALL_LIBDIR@"
+libdir="@CMAKE_INSTALL_LIBDIR@"
 includedir="${prefix}/include"
 
 Name: TileDB
@@ -11,4 +11,4 @@
 Requires.private: @PKGCONF_REQ_PRIV@
 Cflags: -I"${includedir}"
 Libs: -L"${libdir}" -ltiledb
-Libs.private: -L"${libdir}" -ltiledb @PKGCONF_LIBS_PRIV@
\ No newline at end of file
+Libs.private: -L"${libdir}" -ltiledb @PKGCONF_LIBS_PRIV@
