--- a/tiledb/CMakeLists.txt
+++ b/tiledb/CMakeLists.txt
@@ -533,7 +533,7 @@
 # On Linux, must explicitly link -lpthread -ldl in order for static linking
 # to libzstd.
 if (NOT WIN32)
-  target_link_libraries(TILEDB_CORE_OBJECTS_ILIB INTERFACE pthread dl)
+  target_link_libraries(TILEDB_CORE_OBJECTS_ILIB INTERFACE dl)
 endif()
 
 # Copy over dependency info (e.g. include directories) to the core objects.
