diff -u -r ../cadaver-0.24/configure ./configure
--- ../cadaver-0.24/configure	2022-10-30 10:51:38.000000000 +0000
+++ ./configure	2024-05-15 22:27:42.263548376 +0000
@@ -6954,7 +6954,7 @@
 printf "%s\n" "$ne_cv_lib_neon" >&6; }
     if test "$ne_cv_lib_neon" = "yes"; then
        ne_cv_lib_neonver=no
-       for v in 27 28 29 30 31 32; do
+       for v in 27 28 29 30 31 32 33; do
           case $ne_libver in
           0.$v.*) ne_cv_lib_neonver=yes ;;
           esac
