--- a/src/modules/spell/CMakeLists.txt
+++ b/src/modules/spell/CMakeLists.txt
@@ -27,6 +27,7 @@
 install(TARGETS comp-spell-dict
   DESTINATION "${FCITX_INSTALL_ADDONDIR}/libexec")
 
+if(FALSE)
 set(SPELL_EN_DICT_VER 20121020)
 set(SPELL_EN_DICT "${CMAKE_CURRENT_BINARY_DIR}/en_dict.fscd")
 set(SPELL_EN_DICT_SRC "${CMAKE_CURRENT_BINARY_DIR}/en_dict.txt")
@@ -46,3 +47,4 @@
   "${SPELL_EN_DICT_SRC}" "${SPELL_EN_DICT}")
 add_custom_target(spell_en_dict ALL DEPENDS "${SPELL_EN_DICT}")
 install(FILES "${SPELL_EN_DICT}" DESTINATION "${FCITX_INSTALL_PKGDATADIR}/spell")
+endif(FALSE)
