diff -u -r ../OpenFOAM-v1906/applications/solvers/heatTransfer/chtMultiRegionFoam/chtMultiRegionTwoPhaseEulerFoam/Make/options ./applications/solvers/heatTransfer/chtMultiRegionFoam/chtMultiRegionTwoPhaseEulerFoam/Make/options
--- ../OpenFOAM-v1906/applications/solvers/heatTransfer/chtMultiRegionFoam/chtMultiRegionTwoPhaseEulerFoam/Make/options	2019-06-26 16:16:55.000000000 +0200
+++ ./applications/solvers/heatTransfer/chtMultiRegionFoam/chtMultiRegionTwoPhaseEulerFoam/Make/options	2019-09-02 21:43:00.546582802 +0200
@@ -38,4 +38,4 @@
     -lregionModels \
     -lsampling \
     -lreactingTwoPhaseSystem \
-
+    -lreactingPhaseSystem
