--- ./etc/config.sh/CGAL.orig	2022-02-08 17:17:26.353636901 +0100
+++ ./etc/config.sh/CGAL	2022-02-08 17:18:02.813601658 +0100
@@ -43,8 +43,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-boost_version=boost_1_74_0
-cgal_version=CGAL-4.14.3
+boost_version=boost-system
+cgal_version=CGAL-system
 
 export BOOST_ARCH_PATH="$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$boost_version"
 export CGAL_ARCH_PATH="$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$cgal_version"
