--- ./etc/config.sh/CGAL	2017-06-30 08:50:14.000000000 +0000
+++ ../CGAL	2017-12-01 20:25:07.418203062 +0000
@@ -50,7 +50,7 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-boost_version=boost_1_64_0
+boost_version=boost-system
 cgal_version=CGAL-4.9.1
 
 export BOOST_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$boost_version
