--- ./etc/config.sh/unset.orig	2020-07-04 12:31:53.433426922 +0200
+++ ./etc/config.sh/unset	2020-07-04 12:32:27.270319808 +0200
@@ -25,7 +25,7 @@
 
 # The old dirs to be cleaned from the environment variables
 foamOldDirs="$WM_PROJECT_DIR $WM_THIRD_PARTY_DIR \
-    $HOME/$WM_PROJECT/$USER $FOAM_USER_APPBIN $FOAM_USER_LIBBIN \
+    $HOME/$WM_PROJECT/termux $FOAM_USER_APPBIN $FOAM_USER_LIBBIN \
     $WM_PROJECT_SITE $FOAM_SITE_APPBIN $FOAM_SITE_LIBBIN \
     $ParaView_DIR $ParaView_MESA_DIR"
 
