--- ../POSIX.C.orig	2018-09-15 11:32:55.355398128 +0200
+++ ./src/OSspecific/POSIX/POSIX.C	2018-09-15 11:43:08.586328601 +0200
@@ -1107,7 +1127,7 @@
     // Fill sockaddr_in structure with dest address and port
     memset(reinterpret_cast<char *>(&destAddr), '\0', sizeof(destAddr));
     destAddr.sin_family = AF_INET;
-    destAddr.sin_port = htons(ushort(destPort));
+    destAddr.sin_port = htons(u_short(destPort));
     destAddr.sin_addr.s_addr = addr;
 
 
