--- ../OpenFOAM-v1906/wmake/makefiles/general	2019-06-26 16:16:55.000000000 +0200
+++ ./wmake/makefiles/general	2019-09-01 18:21:57.531631905 +0200
@@ -52,8 +52,8 @@
 LIB_PLATFORMS   = $(WM_PROJECT_DIR)/platforms/$(WM_OPTIONS)/lib
 OBJECTS_DIR     = $(MAKE_DIR)/$(WM_OPTIONS)
 
-SYS_INC         =
-SYS_LIBS        =
+SYS_INC         = -I@TERMUX_PREFIX@/include
+SYS_LIBS        = -L@TERMUX_PREFIX@/lib
 
 PROJECT_INC     = \
     -I$(LIB_SRC)/$(WM_PROJECT)/lnInclude \
@@ -61,8 +61,8 @@
 
 PROJECT_LIBS    = -l$(WM_PROJECT)
 
-EXE_INC         =
-EXE_LIBS        =
+EXE_INC         = -I@TERMUX_PREFIX@/include
+EXE_LIBS        = -L@TERMUX_PREFIX@/lib
 LIB_LIBS        =
 
 
