diff -u -r ../OpenFOAM-v1906/wmake/scripts/have_adios2 ./wmake/scripts/have_adios2
--- ../OpenFOAM-v1906/wmake/scripts/have_adios2	2019-06-26 16:16:55.000000000 +0200
+++ ./wmake/scripts/have_adios2	2019-09-04 20:56:13.310624168 +0200
@@ -82,7 +82,7 @@
         library="$(findExtLib $libName)"
     elif isSystem "$prefix"
     then
-        header=$(findFirstFile "/usr/local/include/$incName" "/usr/include/$incName")
+        header=$(findFirstFile @TERMUX_PREFIX@/local/include/$incName" @TERMUX_PREFIX@/include/$incName")
         prefix=$(sysPrefix "$header")
     else
         unset prefix
diff -u -r ../OpenFOAM-v1906/wmake/scripts/have_boost ./wmake/scripts/have_boost
--- ../OpenFOAM-v1906/wmake/scripts/have_boost	2019-06-26 16:16:55.000000000 +0200
+++ ./wmake/scripts/have_boost	2019-09-04 20:56:13.313959165 +0200
@@ -75,7 +75,7 @@
         library="$(findExtLib $libName)"
     elif isSystem "$prefix"
     then
-        header=$(findFirstFile "/usr/local/include/$incName" "/usr/include/$incName")
+        header=$(findFirstFile @TERMUX_PREFIX@/local/include/$incName" @TERMUX_PREFIX@/include/$incName")
         prefix=$(sysPrefix "$header")
     else
         unset prefix
diff -u -r ../OpenFOAM-v1906/wmake/scripts/have_cgal ./wmake/scripts/have_cgal
--- ../OpenFOAM-v1906/wmake/scripts/have_cgal	2019-06-26 16:16:55.000000000 +0200
+++ ./wmake/scripts/have_cgal	2019-09-04 20:56:13.317294161 +0200
@@ -75,7 +75,7 @@
         library="$(findExtLib $libName)"
     elif isSystem "$prefix"
     then
-        header=$(findFirstFile "/usr/local/include/$incName" "/usr/include/$incName")
+        header=$(findFirstFile @TERMUX_PREFIX@/local/include/$incName" @TERMUX_PREFIX@/include/$incName")
         prefix=$(sysPrefix "$header")
     else
         unset prefix
diff -u -r ../OpenFOAM-v1906/wmake/scripts/have_fftw ./wmake/scripts/have_fftw
--- ../OpenFOAM-v1906/wmake/scripts/have_fftw	2019-06-26 16:16:55.000000000 +0200
+++ ./wmake/scripts/have_fftw	2019-09-04 20:56:13.320629158 +0200
@@ -75,7 +75,7 @@
         library="$(findExtLib $libName)"
     elif isSystem "$prefix"
     then
-        header=$(findFirstFile "/usr/local/include/$incName" "/usr/include/$incName")
+        header=$(findFirstFile @TERMUX_PREFIX@/local/include/$incName" @TERMUX_PREFIX@/include/$incName")
         prefix=$(sysPrefix "$header")
     else
         unset prefix
diff -u -r ../OpenFOAM-v1906/wmake/scripts/have_hypre ./wmake/scripts/have_hypre
--- ../OpenFOAM-v1906/wmake/scripts/have_hypre	2019-06-26 16:16:55.000000000 +0200
+++ ./wmake/scripts/have_hypre	2019-09-04 20:56:13.323964153 +0200
@@ -85,7 +85,7 @@
         library="$(findExtLib $libName)"
     elif isSystem "$prefix"
     then
-        header=$(findFirstFile "/usr/local/include/$incName" "/usr/include/$incName")
+        header=$(findFirstFile @TERMUX_PREFIX@/local/include/$incName" @TERMUX_PREFIX@/include/$incName")
         prefix=$(sysPrefix "$header")
     else
         unset prefix
diff -u -r ../OpenFOAM-v1906/wmake/scripts/have_kahip ./wmake/scripts/have_kahip
--- ../OpenFOAM-v1906/wmake/scripts/have_kahip	2019-06-26 16:16:55.000000000 +0200
+++ ./wmake/scripts/have_kahip	2019-09-04 20:56:13.323964153 +0200
@@ -83,7 +83,7 @@
         library="$(findExtLib $libName)"
     elif isSystem "$prefix"
     then
-        header=$(findFirstFile "/usr/local/include/$incName" "/usr/include/$incName")
+        header=$(findFirstFile @TERMUX_PREFIX@/local/include/$incName" @TERMUX_PREFIX@/include/$incName")
         prefix=$(sysPrefix "$header")
     else
         unset prefix
diff -u -r ../OpenFOAM-v1906/wmake/scripts/have_metis ./wmake/scripts/have_metis
--- ../OpenFOAM-v1906/wmake/scripts/have_metis	2019-06-26 16:16:55.000000000 +0200
+++ ./wmake/scripts/have_metis	2019-09-04 20:56:13.327299150 +0200
@@ -83,7 +83,7 @@
         library="$(findExtLib $libName)"
     elif isSystem "$prefix"
     then
-        header=$(findFirstFile "/usr/local/include/$incName" "/usr/include/$incName")
+        header=$(findFirstFile @TERMUX_PREFIX@/local/include/$incName" @TERMUX_PREFIX@/include/$incName")
         prefix=$(sysPrefix "$header")
     else
         unset prefix
diff -u -r ../OpenFOAM-v1906/wmake/scripts/have_mgridgen ./wmake/scripts/have_mgridgen
--- ../OpenFOAM-v1906/wmake/scripts/have_mgridgen	2019-06-26 16:16:55.000000000 +0200
+++ ./wmake/scripts/have_mgridgen	2019-09-04 20:56:13.330634146 +0200
@@ -84,7 +84,7 @@
         library="$(findExtLib $libName $libName2)"
     elif isSystem "$prefix"
     then
-        header=$(findFirstFile "/usr/local/include/$incName" "/usr/include/$incName")
+        header=$(findFirstFile @TERMUX_PREFIX@/local/include/$incName" @TERMUX_PREFIX@/include/$incName")
         prefix=$(sysPrefix "$header")
     else
         unset prefix
diff -u -r ../OpenFOAM-v1906/wmake/scripts/have_petsc ./wmake/scripts/have_petsc
--- ../OpenFOAM-v1906/wmake/scripts/have_petsc	2019-06-26 16:16:55.000000000 +0200
+++ ./wmake/scripts/have_petsc	2019-09-04 20:56:13.333969141 +0200
@@ -85,7 +85,7 @@
         library="$(findExtLib $libName)"
     elif isSystem "$prefix"
     then
-        header=$(findFirstFile "/usr/local/include/$incName" "/usr/include/$incName")
+        header=$(findFirstFile @TERMUX_PREFIX@/local/include/$incName" @TERMUX_PREFIX@/include/$incName")
         prefix=$(sysPrefix "$header")
     else
         unset prefix
diff -u -r ../OpenFOAM-v1906/wmake/scripts/have_readline ./wmake/scripts/have_readline
--- ../OpenFOAM-v1906/wmake/scripts/have_readline	2019-06-26 16:16:55.000000000 +0200
+++ ./wmake/scripts/have_readline	2019-09-04 20:56:13.337304138 +0200
@@ -71,7 +71,7 @@
         library="$(findExtLib $libName)"
     elif isSystem "$prefix"
     then
-        header=$(findFirstFile "/usr/local/include/$incName" "/usr/include/$incName")
+        header=$(findFirstFile @TERMUX_PREFIX@/local/include/$incName" @TERMUX_PREFIX@/include/$incName")
         prefix=$(sysPrefix "$header")
     else
         unset prefix
diff -u -r ../OpenFOAM-v1906/wmake/scripts/have_scotch ./wmake/scripts/have_scotch
--- ../OpenFOAM-v1906/wmake/scripts/have_scotch	2019-06-26 16:16:55.000000000 +0200
+++ ./wmake/scripts/have_scotch	2019-09-04 20:56:13.337304138 +0200
@@ -90,10 +90,10 @@
     elif isSystem "$prefix"
     then
         header=$(findFirstFile \
-            "/usr/local/include/scotch/$incName" \
-            "/usr/local/include/$incName" \
-            "/usr/include/scotch/$incName" \
-            "/usr/include/$incName" \
+            @TERMUX_PREFIX@/local/include/scotch/$incName" \
+            @TERMUX_PREFIX@/local/include/$incName" \
+            @TERMUX_PREFIX@/include/scotch/$incName" \
+            @TERMUX_PREFIX@/include/$incName" \
         )
         prefix=$(sysPrefix "$header")
     else
@@ -208,12 +208,12 @@
     then
         prefix=/usr
         header=$(findFirstFile \
-            "/usr/local/include/ptscotch/$incName" \
-            "/usr/local/include/scotch/$incName" \
-            "/usr/local/include/$incName" \
-            "/usr/include/ptscotch/$incName" \
-            "/usr/include/scotch/$incName" \
-            "/usr/include/$incName" \
+            @TERMUX_PREFIX@/local/include/ptscotch/$incName" \
+            @TERMUX_PREFIX@/local/include/scotch/$incName" \
+            @TERMUX_PREFIX@/local/include/$incName" \
+            @TERMUX_PREFIX@/include/ptscotch/$incName" \
+            @TERMUX_PREFIX@/include/scotch/$incName" \
+            @TERMUX_PREFIX@/include/$incName" \
         )
         case "$header" in (/usr/local/*) prefix=/usr/local ;; esac
     else
diff -u -r ../OpenFOAM-v1906/wmake/scripts/have_zoltan ./wmake/scripts/have_zoltan
--- ../OpenFOAM-v1906/wmake/scripts/have_zoltan	2019-06-26 16:16:55.000000000 +0200
+++ ./wmake/scripts/have_zoltan	2019-09-04 20:56:13.340639134 +0200
@@ -82,7 +82,7 @@
         library="$(findExtLib $libName)"
     elif isSystem "$prefix"
     then
-        header=$(findFirstFile "/usr/local/include/$incName" "/usr/include/$incName")
+        header=$(findFirstFile @TERMUX_PREFIX@/local/include/$incName" @TERMUX_PREFIX@/include/$incName")
         prefix=$(sysPrefix "$header")
     else
         unset prefix
diff -u -r ../OpenFOAM-v1906/wmake/scripts/paraviewFunctions ./wmake/scripts/paraviewFunctions
--- ../OpenFOAM-v1906/wmake/scripts/paraviewFunctions	2019-06-26 16:16:55.000000000 +0200
+++ ./wmake/scripts/paraviewFunctions	2019-09-04 20:58:15.698308442 +0200
@@ -166,10 +166,10 @@
             "$(paraview-config --include 2>/dev/null |sed -ne 's/^ *-I//p')/$header"\
             "${includeDir:+$includeDir/paraview-$pv_api/$header}" \
             "${includeDir:+$includeDir/paraview/$header}" \
-            /usr/local/include/"paraview-$pv_api/$header" \
-            /usr/local/include/paraview/"$header" \
-            /usr/include/"paraview-$pv_api/$header" \
-            /usr/include/paraview/"$header" \
+            @TERMUX_PREFIX@/local/include/"paraview-$pv_api/$header" \
+            @TERMUX_PREFIX@/local/include/paraview/"$header" \
+            @TERMUX_PREFIX@/include/"paraview-$pv_api/$header" \
+            @TERMUX_PREFIX@/include/paraview/"$header" \
             )
     fi
 
diff -u -r ../OpenFOAM-v1906/wmake/scripts/sysFunctions ./wmake/scripts/sysFunctions
--- ../OpenFOAM-v1906/wmake/scripts/sysFunctions	2019-06-26 16:16:55.000000000 +0200
+++ ./wmake/scripts/sysFunctions	2019-09-04 20:56:13.363984106 +0200
@@ -91,7 +91,7 @@
 
     # Return system prefix (/usr, /usr/local, ...) based on hint provided
     # Eg,
-    #    sysPrefix "/usr/local/include/fftw3.h"  -> "/usr/local"
+    #    sysPrefix @TERMUX_PREFIX@/local/include/fftw3.h"  -> "/usr/local"
     #
     # Without a hint, echoes "/usr"
     sysPrefix()
