From bc89c504276a00e655d7ba4d0ba3aa19e3f8d70d Mon Sep 17 00:00:00 2001
From: OldWorldOrdr <joey.t.reinhart@gmail.com>
Date: Fri, 30 Sep 2022 06:34:19 -0400
Subject: [PATCH] Update neofetch
Origin: https://github.com/hykilpikonna/hyfetch/commit/e72fe4d934e3fb0c606cd5afaccca89995412613

---
 neofetch | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/neofetch b/neofetch
index 7c7a2de3e..3e3f81467 100755
--- a/neofetch
+++ b/neofetch
@@ -4636,8 +4636,8 @@ get_cols() {
         printf -v block_spaces "%${block_height}s"
 
         # Convert the spaces into rows of blocks.
-        [[ "$blocks"  ]] && cols+="${block_spaces// /${blocks}[mnl}"
-        [[ "$blocks2" ]] && cols+="${block_spaces// /${blocks2}[mnl}"
+        [[ "$blocks"  ]] && cols+="${block_spaces// /${blocks}\[mnl}"
+        [[ "$blocks2" ]] && cols+="${block_spaces// /${blocks2}\[mnl}"
 
         # Determine the horizontal offset of the blocks.
         case $col_offset in
