diff -u -r ../neofetch-3.0/neofetch ./neofetch
--- ../neofetch-3.0/neofetch	2017-01-23 04:40:31.000000000 +0100
+++ ./neofetch	2017-01-23 13:12:25.682702897 +0100
@@ -3285,8 +3285,8 @@
     if [[ -f "/etc/neofetch/config" ]]; then
         default_config="/etc/neofetch/config"
 
-    elif [[ -f "/data/data/com.termux/files/etc/neofetch/config" ]]; then
-        default_config="/data/data/com.termux/files/etc/neofetch/config"
+    elif [[ -f "/data/data/com.termux/files/usr/etc/neofetch/config" ]]; then
+        default_config="/data/data/com.termux/files/usr/etc/neofetch/config"
 
     else
         [[ -z "$script_dir" ]] && script_dir="$(get_full_path "$0")"
