--- ./depends/packages/boost.mk.orig	2021-05-07 07:56:23.903846226 +0000
+++ ./depends/packages/boost.mk	2021-05-07 07:56:47.650501607 +0000
@@ -1,6 +1,6 @@
 package=boost
 $(package)_version=1_70_0
-$(package)_download_path=https://dl.bintray.com/boostorg/release/1.70.0/source/
+$(package)_download_path=https://boostorg.jfrog.io/artifactory/main/release/1.70.0/source/
 $(package)_file_name=boost_$($(package)_version).tar.bz2
 $(package)_sha256_hash=430ae8354789de4fd19ee52f3b1f739e1fba576f0aded0897c3c2bc00fb38778
 $(package)_patches=unused_var_in_process.patch
