diff -u -r ../MPD-0.20.21/configure.ac ./configure.ac
--- ../MPD-0.20.21/configure.ac	2018-08-17 17:50:59.000000000 +0000
+++ ./configure.ac	2018-08-20 20:45:16.603943001 +0000
@@ -97,7 +97,7 @@
 
 case "$host_os" in
 linux-android*)
-	host_is_android=yes
+	host_is_android=no
 	host_is_linux=yes
 	linux_auto=auto
 	AM_CPPFLAGS="$AM_CPPFLAGS -DANDROID"
@@ -202,7 +202,7 @@
 		[android_abi="armeabi-v7a"])
 fi
 
-AC_SUBST(ANDROID_SDK, [$with_android_sdk])
+#AC_SUBST(ANDROID_SDK, [$with_android_sdk])
 AC_SUBST(ANDROID_ABI, [$android_abi])
 
 dnl ---------------------------------------------------------------------------
