--- ../cache/MPD-0.20.6/configure.ac	2017-03-10 15:57:59.000000000 +0000
+++ ./configure.ac	2017-05-13 01:18:13.566696356 +0000
@@ -97,7 +97,7 @@
 
 case "$host_os" in
 linux-android*)
-	host_is_android=yes
+	host_is_android=no
 	host_is_linux=yes
 	linux_auto=auto
 	AM_CPPFLAGS="$AM_CPPFLAGS -DANDROID"
@@ -196,7 +196,7 @@
 	fi
 fi
 
-AC_SUBST(ANDROID_SDK, [$with_android_sdk])
+#AC_SUBST(ANDROID_SDK, [$with_android_sdk])
 
 dnl ---------------------------------------------------------------------------
 dnl Language Checks
