diff -u -r ../MPD-0.21.7/src/decoder/plugins/HybridDsdDecoderPlugin.cxx ./src/decoder/plugins/HybridDsdDecoderPlugin.cxx
--- ../MPD-0.21.7/src/decoder/plugins/HybridDsdDecoderPlugin.cxx	2019-04-03 10:18:29.000000000 +0000
+++ ./src/decoder/plugins/HybridDsdDecoderPlugin.cxx	2019-04-11 23:28:00.503100747 +0000
@@ -39,8 +39,6 @@
 	   without a DSD DAC, the PCM (=ALAC) part of the file is
 	   better */
 	if (block.GetBlockParam("enabled") == nullptr) {
-		LogDebug(hybrid_dsd_domain,
-			 "The Hybrid DSD decoder is disabled because it was not explicitly enabled");
 		return false;
 	}
 
