--- ../HybridDsdDecoderPlugin.cxx.orig	2019-01-07 20:58:59.223328152 +0100
+++ ./src/decoder/plugins/HybridDsdDecoderPlugin.cxx	2019-01-07 21:00:02.953327686 +0100
@@ -40,8 +40,6 @@
 	   without a DSD DAC, the PCM (=ALAC) part of the file is
 	   better */
 	if (block.GetBlockParam("enabled") == nullptr) {
-		LogInfo(hybrid_dsd_domain,
-			"The Hybrid DSD decoder is disabled because it was not explicitly enabled");
 		return false;
 	}
 
