--- ../cache/MPD-0.20.6/Makefile.am	2017-03-10 15:57:59.000000000 +0000
+++ ./Makefile.am	2017-05-13 03:28:47.725309586 +0000
@@ -1481,7 +1481,14 @@
 if HAVE_OPENAL
 liboutput_plugins_a_SOURCES += \
 	src/output/plugins/OpenALOutputPlugin.cxx \
-	src/output/plugins/OpenALOutputPlugin.hxx
+	src/output/plugins/OpenALOutputPlugin.hxx \
+	src/output/plugins/sles/Object.hxx \
+	src/output/plugins/sles/Engine.hxx \
+	src/output/plugins/sles/Play.hxx \
+	src/output/plugins/sles/AndroidSimpleBufferQueue.hxx \
+	src/output/plugins/sles/SlesOutputPlugin.cxx \
+	src/output/plugins/sles/SlesOutputPlugin.hxx
+OUTPUT_LIBS += -lOpenSLES
 endif
 
 if HAVE_OSX
