diff -u -r ../MPD-0.20.8/src/Main.cxx ./src/Main.cxx
--- ../MPD-0.20.8/src/Main.cxx	2017-05-19 20:10:22.000000000 +0200
+++ ./src/Main.cxx	2017-05-23 01:19:38.629090778 +0200
@@ -123,7 +123,7 @@
 static constexpr size_t DEFAULT_BUFFER_SIZE = 4 * MEGABYTE;
 
 static
-#if GCC_OLDER_THAN(5,0)
+#if GCC_OLDER_THAN(5,0) || defined(__ANDROID__)
 /* gcc 4.x has no "constexpr" for std::max() */
 const
 #else
