diff -u -r ../MPD-0.21.3/src/lib/gcrypt/meson.build ./src/lib/gcrypt/meson.build
--- ../MPD-0.21.3/src/lib/gcrypt/meson.build	2018-11-16 12:27:58.000000000 +0000
+++ ./src/lib/gcrypt/meson.build	2019-01-06 23:35:51.108609898 +0000
@@ -1,4 +1,4 @@
-gcrypt_dep = c_compiler.find_library('gcrypt', required: get_option('qobuz'))
+gcrypt_dep = c_compiler.find_library('gcrypt_disabled', required: get_option('qobuz'))
 if not gcrypt_dep.found()
   subdir_done()
 endif
