diff --git a/src/thread/Util.cxx~ b/src/thread/Util.cxx
index 0dbf818..671eb6a 100644
--- a/src/thread/Util.cxx~
+++ b/src/thread/Util.cxx
@@ -40,7 +40,7 @@
 
 #ifdef __linux__
 
-#ifndef ANDROID
+#if !defined(ANDROID) && !defined(TERMUX)
 
 static int
 linux_ioprio_set(int which, int who, int ioprio)
@@ -60,7 +60,7 @@ ioprio_set_idle()
 	linux_ioprio_set(_IOPRIO_WHO_PROCESS, 0, _IOPRIO_IDLE);
 }
 
-#endif /* !ANDROID */
+#endif /* !ANDROID && !TERMUX */
 
 /**
  * Wrapper for the "sched_setscheduler" system call.  We don't use the
@@ -84,7 +84,7 @@ SetThreadIdlePriority()
 	linux_sched_setscheduler(0, SCHED_IDLE, &sched_param);
 #endif
 
-#ifndef ANDROID
+#if !defined(ANDROID) && !defined(TERMUX)
 	/* this system call is forbidden via seccomp on Android 8 and
 	   leads to crash (SIGSYS) */
 	ioprio_set_idle();
