diff -uNr lxc-3.1.0/src/lxc/conf.c lxc-3.1.0.mod/src/lxc/conf.c
--- lxc-3.1.0/src/lxc/conf.c	2018-12-14 01:13:51.000000000 +0200
+++ lxc-3.1.0.mod/src/lxc/conf.c	2019-07-13 19:02:58.727793440 +0300
@@ -4104,6 +4104,7 @@
 	free(conf->init_cmd);
 	free(conf->init_cwd);
 	free(conf->unexpanded_config);
+#undef syslog
 	free(conf->syslog);
 	lxc_free_networks(&conf->network);
 	free(conf->lsm_aa_profile);
