--- a/src/bootstrap/src/core/build_steps/compile.rs
+++ b/src/bootstrap/src/core/build_steps/compile.rs
@@ -443,7 +443,6 @@
         // should be deleted and the native objects should be used.
         let srcdir = if target == "wasm32-wasip3" {
-            assert!(!srcdir.exists(), "wasip3 support is in wasi-libc, this should be updated now");
             builder.wasi_libdir(TargetSelection::from_user("wasm32-wasip2")).unwrap()
         } else {
             srcdir
         };
