--- a/compiler/rustc_driver_impl/src/lib.rs
+++ b/compiler/rustc_driver_impl/src/lib.rs
@@ -1296,7 +1296,7 @@
     }
 }
 
-#[cfg(all(unix, any(target_env = "gnu", target_os = "macos")))]
+#[cfg(all(unix, any(target_env = "gnu", target_os = "macos"), not(target_os = "android")))]
 mod signal_handler {
     extern "C" {
         fn backtrace_symbols_fd(
@@ -1340,7 +1340,7 @@
     }
 }
 
-#[cfg(not(all(unix, any(target_env = "gnu", target_os = "macos"))))]
+#[cfg(not(all(unix, any(target_env = "gnu", target_os = "macos"), not(target_os = "android"))))]
 mod signal_handler {
     pub(super) fn install() {}
 }
