--- ../rustc-1.44.0-src/vendor/libc/src/unix/mod.rs	2020-06-01 12:45:24.000000000 -0500
+++ ./vendor/libc/src/unix/mod.rs	2020-06-11 23:46:08.059193124 -0500
@@ -1235,7 +1235,7 @@ extern "C" {
             target_os = "dragonfly",
             target_os = "haiku"
         ),
-        link_name = "__res_init"
+        link_name = "res_init"
     )]
     #[cfg_attr(
         any(target_os = "macos", target_os = "ios"),

