--- ../rustc-1.29.2-src/src/etc/rust-lldb	2018-10-05 15:19:57.000000000 -0700
+++ ./src/etc/rust-lldb	2018-10-22 20:18:13.686641765 -0700
@@ -24,7 +24,7 @@
 fi
 
 # Create a tempfile containing the LLDB script we want to execute on startup
-TMPFILE=`mktemp /tmp/rust-lldb-commands.XXXXXX`
+TMPFILE=`mktemp @TERMUX_PREFIX@/tmp/rust-lldb-commands.XXXXXX`
 
 # Make sure to delete the tempfile no matter what
 trap "rm -f $TMPFILE; exit" INT TERM EXIT
