--- a/src/tools/rustc-workspace-hack/Cargo.toml
+++ b/src/tools/rustc-workspace-hack/Cargo.toml
@@ -85,6 +85,7 @@
 regex = { version = "1.5.6" }
 serde_json = { version = "1.0.31", features = ["raw_value", "unbounded_depth"] }
 syn = { version = "1", features = ['full', 'visit', 'visit-mut'] } # `visit-mut` required by Cargo via `gix`
+syn_2 = { version = "2", features = ['full'], package = "syn" }
 url = { version = "2.0", features = ['serde'] }
 # Ensure default features of rand, which are disabled in some scenarios.
 rand = { version = "0.8.5" }
