diff -uNr gw-0.7.0/src/main.cpp gw-0.7.0.mod/src/main.cpp
--- gw-0.7.0/src/main.cpp	2023-03-20 12:59:19.000000000 +0200
+++ gw-0.7.0.mod/src/main.cpp	2023-03-22 11:45:32.090313596 +0200
@@ -35,6 +35,7 @@
 #include "include/core/SkDocument.h"
 #include "include/docs/SkPDFDocument.h"
 
+namespace fs = std::filesystem;
 
 // skia context has to be managed from global space to work
 GrDirectContext *sContext = nullptr;
diff -uNr gw-0.7.0/src/plot_controls.cpp gw-0.7.0.mod/src/plot_controls.cpp
--- gw-0.7.0/src/plot_controls.cpp	2023-03-20 12:59:19.000000000 +0200
+++ gw-0.7.0.mod/src/plot_controls.cpp	2023-03-22 11:45:36.670355876 +0200
@@ -20,6 +20,7 @@
 #include "term_out.h"
 #include "themes.h"
 
+namespace fs = std::filesystem;
 
 namespace Manager {
 
diff -uNr gw-0.7.0/src/plot_manager.cpp gw-0.7.0.mod/src/plot_manager.cpp
--- gw-0.7.0/src/plot_manager.cpp	2023-03-20 12:59:19.000000000 +0200
+++ gw-0.7.0.mod/src/plot_manager.cpp	2023-03-22 11:45:43.594419791 +0200
@@ -31,6 +31,7 @@
 #include "../include/termcolor.h"
 #include "themes.h"
 
+namespace fs = std::filesystem;
 
 using namespace std::literals;
 
diff -uNr gw-0.7.0/src/plot_manager.h gw-0.7.0.mod/src/plot_manager.h
--- gw-0.7.0/src/plot_manager.h	2023-03-20 12:59:19.000000000 +0200
+++ gw-0.7.0.mod/src/plot_manager.h	2023-03-22 11:45:19.706199285 +0200
@@ -41,6 +41,7 @@
 #include "include/core/SkColorSpace.h"
 #include "include/core/SkSurface.h"
 
+namespace fs = std::filesystem;
 
 namespace Manager {
 
@@ -246,4 +247,4 @@
         long stop;
     };
 
-}
\ В конце файла нет новой строки
+}
