--- ../Makefile.orig	2019-12-21 16:18:41.380353060 +0100
+++ ./Makefile	2019-12-21 16:42:45.894677981 +0100
@@ -297,7 +300,7 @@
 		$(NAME)src/doc/*.6 $(NAME)src/doc/frotz.conf* \
 		$(NAME)src/doc/Xresources  > /dev/null
 else
-        @echo "Not in a git repository or git command not found.  Cannot make a tarball."
+	@echo "Not in a git repository or git command not found.  Cannot make a tarball."
 endif
 
 all: $(FROTZ_BIN) $(DFROTZ_BIN) $(SFROTZ_BIN)
