diff -u -r ../frotz-2.43d/Makefile ./Makefile
--- ../frotz-2.43d/Makefile	2012-01-03 09:32:58.000000000 +0100
+++ ./Makefile	2014-03-06 22:50:06.580642826 +0100
@@ -1,22 +1,22 @@
 # Define your C compiler.  I recommend gcc if you have it.
 # MacOS users should use "cc" even though it's really "gcc".
 #
-CC = gcc
+#CC = gcc
 #CC = cc
 
 # Define your optimization flags.  Most compilers understand -O and -O2,
 # Standard (note: Solaris on UltraSparc using gcc 2.8.x might not like this.)
 #
-OPTS = -O2
+OPTS = -Os
 
 # Pentium with gcc 2.7.0 or better
 #OPTS = -O2 -fomit-frame-pointer -malign-functions=2 -malign-loops=2 \
 #-malign-jumps=2
 
 # Define where you want Frotz installed.  Usually this is /usr/local
-PREFIX = /usr/local
+#PREFIX = /usr/local
 
-MAN_PREFIX = $(PREFIX)
+MAN_PREFIX = $(PREFIX)/share
 #MAN_PREFIX = /usr/local/share
 
 CONFIG_DIR = $(PREFIX)/etc
@@ -24,7 +24,7 @@
 
 # Define where you want Frotz to look for frotz.conf.
 #
-CONFIG_DIR = /usr/local/etc
+#CONFIG_DIR = /usr/local/etc
 #CONFIG_DIR = /etc
 #CONFIG_DIR = /usr/pkg/etc
 #CONFIG_DIR =
@@ -73,8 +73,8 @@
 # curses library won't work, comment out the first option and uncomment
 # the second.
 #
-CURSES = -lcurses
-#CURSES = -lncurses
+#CURSES = -lcurses
+CURSES = -lncurses
 
 # Uncomment this if your need to use ncurses instead of the
 # vendor-supplied curses library.  This just tells the compile process
@@ -187,24 +187,24 @@
 $(COMMON_TARGET): $(COMMON_OBJECT)
 	@echo
 	@echo "Archiving common code..."
-	ar rc $(COMMON_TARGET) $(COMMON_OBJECT)
-	ranlib $(COMMON_TARGET)
+	$(AR) rc $(COMMON_TARGET) $(COMMON_OBJECT)
+	$(RANLIB) $(COMMON_TARGET)
 	@echo
 
 curses_lib:	config_curses $(CURSES_TARGET)
 $(CURSES_TARGET): $(CURSES_OBJECT)
 	@echo
 	@echo "Archiving curses interface code..."
-	ar rc $(CURSES_TARGET) $(CURSES_OBJECT)
-	ranlib $(CURSES_TARGET)
+	$(AR) rc $(CURSES_TARGET) $(CURSES_OBJECT)
+	$(RANLIB) $(CURSES_TARGET)
 	@echo
 
 dumb_lib:	$(DUMB_TARGET)
 $(DUMB_TARGET): $(DUMB_OBJECT)
 	@echo
 	@echo "Archiving dumb interface code..."
-	ar rc $(DUMB_TARGET) $(DUMB_OBJECT)
-	ranlib $(DUMB_TARGET)
+	$(AR) rc $(DUMB_TARGET) $(DUMB_OBJECT)
+	$(RANLIB) $(DUMB_TARGET)
 	@echo
 
 soundcard.h:
@@ -213,7 +213,7 @@
 	fi
 
 install: $(NAME)
-	strip $(BINNAME)$(EXTENSION)
+	$(STRIP) $(BINNAME)$(EXTENSION)
 	install -d $(PREFIX)/bin
 	install -d $(MAN_PREFIX)/man/man6
 	install -c -m 755 $(BINNAME)$(EXTENSION) $(PREFIX)/bin
@@ -226,7 +226,7 @@
 deinstall: uninstall
 
 install_dumb: d$(NAME)
-	strip d$(BINNAME)$(EXTENSION)
+	$(STRIP) d$(BINNAME)$(EXTENSION)
 	install -d $(PREFIX)/bin
 	install -d $(MAN_PREFIX)/man/man6
 	install -c -m 755 d$(BINNAME)$(EXTENSION) $(PREFIX)/bin
