--- ../ux_frotz.h.orig	2019-12-21 18:00:15.498402287 +0100
+++ ./src/curses/ux_frotz.h	2019-12-21 18:00:32.478537779 +0100
@@ -40,7 +40,7 @@
 #define ERR_REPORT_DEF		ERR_REPORT_ONCE
 #define	QUETZAL_DEF		1
 #define SAVEDIR_DEF		"if-saves"
-#define ZCODEPATH_DEF		"/usr/games/zcode:/usr/local/games/zcode"
+#define ZCODEPATH_DEF		"@TERMUX_PREFIX@/games/zcode:@TERMUX_PREFIX@/local/games/zcode"
 
 
 #define LINELEN		256	/* for getconfig()	*/
