diff -uNr pytorch/torch/csrc/autograd/python_function.cpp pytorch.mod/torch/csrc/autograd/python_function.cpp
--- pytorch/torch/csrc/autograd/python_function.cpp	2022-10-03 21:59:02.604422931 +0900
+++ pytorch.mod/torch/csrc/autograd/python_function.cpp	2022-10-03 23:39:51.207871384 +0900
@@ -510,7 +510,7 @@
       throw torch::TypeError(
           "save_for_backward can only save variables, but argument %ld is of "
           "type %s",
-          i,
+          (long)(i),
           Py_TYPE(obj)->tp_name);
     }
   }
diff -uNr pytorch/torch/csrc/utils/python_arg_parser.h pytorch.mod/torch/csrc/utils/python_arg_parser.h
--- pytorch/torch/csrc/utils/python_arg_parser.h	2022-10-03 19:00:05.000000000 +0900
+++ pytorch.mod/torch/csrc/utils/python_arg_parser.h	2022-10-03 23:39:46.263853740 +0900
@@ -511,7 +511,7 @@
       args->signature.params[i].name.c_str(),
       args->signature.params[i].type_name().c_str(),
       Py_TYPE(obj)->tp_name,
-      idx + 1);
+      (long)(idx + 1));
 }
 
 inline std::vector<c10::SymInt> PythonArgs::symintlist(int i) {
@@ -671,7 +671,7 @@
           signature.params[i].name.c_str(),
           signature.params[i].type_name().c_str(),
           Py_TYPE(obj)->tp_name,
-          idx + 1);
+          (long)(idx + 1));
     }
   }
   return res;
