--- a/caffe2/CMakeLists.txt
+++ b/caffe2/CMakeLists.txt
@@ -852,7 +852,6 @@
 
 if("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang" AND NOT USE_IOS AND NOT USE_COREML_DELEGATE)
   target_compile_options_if_supported(torch_cpu "-Wmissing-prototypes")
-  target_compile_options_if_supported(torch_cpu "-Werror=missing-prototypes")
   if(TARGET torch_cuda)
     target_compile_options_if_supported(torch_cuda "-Wmissing-prototypes")
     target_compile_options_if_supported(torch_cuda "-Werror=missing-prototypes")
