diff -u -r ../cache/tmp-checkout/torch/csrc/jit/python/script_init.cpp ./torch/csrc/jit/python/script_init.cpp
--- ../cache/tmp-checkout/torch/csrc/jit/python/script_init.cpp	2024-10-02 13:05:35.577599122 +0000
+++ ./torch/csrc/jit/python/script_init.cpp	2024-10-02 13:16:36.541910940 +0000
@@ -722,7 +722,7 @@
   std::shared_ptr<char> bytes_copy(
       static_cast<char*>(_aligned_malloc(size, kFlatbufferDataAlignmentBytes)),
       _aligned_free);
-#elif defined(__APPLE__)
+#elif defined(__APPLE__) || defined(__ANDROID__)
   void* p;
   ::posix_memalign(&p, kFlatbufferDataAlignmentBytes, size);
   TORCH_INTERNAL_ASSERT(p, "Could not allocate memory for flatbuffer");
