diff -uNr a/cmake/MiscCheck.cmake b/cmake/MiscCheck.cmake
--- a/cmake/MiscCheck.cmake	2022-11-03 20:21:14.485923000 +0900
+++ b/cmake/MiscCheck.cmake	2022-11-03 20:23:46.858831635 +0900
@@ -114,7 +114,7 @@
 cmake_pop_check_state()
 
 # ---[ Check if the compiler has AVX/AVX2 support. We only check AVX2.
-if(NOT INTERN_BUILD_MOBILE)
+if(False)
   find_package(AVX) # checks AVX and AVX2
   if(CXX_AVX2_FOUND)
     message(STATUS "Current compiler supports avx2 extension. Will build perfkernels.")
