diff -uNr lxqt-runner-1.2.0/dialog.cpp lxqt-runner-1.2.0.mod/dialog.cpp
--- lxqt-runner-1.2.0/dialog.cpp	2022-11-05 16:33:14.000000000 +0800
+++ lxqt-runner-1.2.0.mod/dialog.cpp	2022-11-06 17:22:18.897744844 +0800
@@ -534,11 +534,7 @@
     if (res)
     {
         hide();
-        if (mClearOnRunning
-            && !qobject_cast<const MathItem*>(command)) // don't clear math results
-        {
-            ui->commandEd->clear();
-        }
+        ui->commandEd->clear();
     }
 
 }
