diff -uNr putty-0.70/unix/uxpgnt.c putty-0.70.mod/unix/uxpgnt.c
--- putty-0.70/unix/uxpgnt.c	2017-07-04 22:30:56.000000000 +0300
+++ putty-0.70.mod/unix/uxpgnt.c	2018-10-05 20:13:03.107470567 +0300
@@ -153,7 +153,7 @@
     /* Nothing needs doing in Unix Pageant */
 }
 
-#define PAGEANT_DIR_PREFIX "/tmp/pageant"
+#define PAGEANT_DIR_PREFIX "@TERMUX_PREFIX@/tmp/pageant"
 
 const char *const appname = "Pageant";
 
