diff -uNr putty-0.70/unix/uxproxy.c putty-0.70.mod/unix/uxproxy.c
--- putty-0.70/unix/uxproxy.c	2017-07-04 22:30:56.000000000 +0300
+++ putty-0.70.mod/unix/uxproxy.c	2018-10-05 20:17:20.194819370 +0300
@@ -355,7 +355,7 @@
             }
 	    noncloexec(0);
 	    noncloexec(1);
-	    execl("/bin/sh", "sh", "-c", cmd, (void *)NULL);
+	    execl("@TERMUX_PREFIX@/bin/sh", "sh", "-c", cmd, (void *)NULL);
 	    _exit(255);
 	}
 
