diff -uNr putty-0.70/unix/uxnoise.c putty-0.70.mod/unix/uxnoise.c
--- putty-0.70/unix/uxnoise.c	2017-07-04 22:30:56.000000000 +0300
+++ putty-0.70.mod/unix/uxnoise.c	2018-10-05 20:12:53.697446775 +0300
@@ -70,7 +70,7 @@
 	exit(1);
     }
 
-    fp = popen("ls -al /tmp 2>/dev/null", "r");
+    fp = popen("ls -al @TERMUX_PREFIX@/tmp 2>/dev/null", "r");
     if (fp) {
 	while ( (ret = fread(buf, 1, sizeof(buf), fp)) > 0)
 	    func(buf, ret);
