diff -uNr putty-0.70/unix/uxmisc.c putty-0.70.mod/unix/uxmisc.c
--- putty-0.70/unix/uxmisc.c	2017-07-04 22:30:56.000000000 +0300
+++ putty-0.70.mod/unix/uxmisc.c	2018-10-05 20:12:00.693981290 +0300
@@ -131,7 +131,6 @@
      * coping correctly with people who have su'ed.
      */
     user = getlogin();
-    setpwent();
     if (user)
 	p = getpwnam(user);
     else
@@ -153,7 +152,6 @@
 	    return NULL;
 	ret = p->pw_name;
     }
-    endpwent();
 
     return dupstr(ret);
 }
