--- ../screen.c.orig	2022-02-04 21:18:58.796769500 +0000
+++ ./src/screen.c	2022-02-04 21:21:13.233194482 +0000
@@ -451,7 +451,7 @@
 
     keypad(tmpwin, TRUE);
 
-    mvwprintw(tmpwin,2,3, msg);				/* output mesg        */
+    mvwprintw(tmpwin,2,3, "%s", msg);				/* output mesg        */
     wmove(tmpwin,2,len+4);
     wrefresh(tmpwin);
 
@@ -492,7 +492,7 @@
 
     tmpwin = drawbox(y, x, 5, len + 6);			/* create window      */
 
-    mvwprintw(tmpwin,2,3, msg);
+    mvwprintw(tmpwin,2,3, "%s", msg);
     wmove(tmpwin,2,len+4);
     wrefresh(tmpwin);
 
