diff --git a/vendor/core/fastboot/fastboot.cpp b/vendor/core/fastboot/fastboot.cpp
index 0062e7e..5f2894d 100644
--- a/vendor/core/fastboot/fastboot.cpp
+++ b/vendor/core/fastboot/fastboot.cpp
@@ -1343,6 +1343,12 @@ static bool is_userspace_fastboot() {
 
 static void reboot_to_userspace_fastboot() {
     fb->RebootTo("fastboot");
+#ifdef __ANDROID__
+    if (fb->transport()->NotReusable()) {
+        fb->WaitForDisconnect();
+        return;
+    }
+#endif
 
     auto* old_transport = fb->set_transport(nullptr);
     delete old_transport;
