diff --git a/vendor/core/fastboot/usb.h b/vendor/core/fastboot/usb.h
index e5f56e2..eea0075 100644
--- a/vendor/core/fastboot/usb.h
+++ b/vendor/core/fastboot/usb.h
@@ -28,6 +28,10 @@
 
 #pragma once
 
+#ifdef ANDROID
+#include <libusb-1.0/libusb.h>
+#endif
+
 #include "transport.h"
 
 struct usb_ifc_info {
@@ -52,6 +56,10 @@ struct usb_ifc_info {
     char device_path[256];
 
     char interface[256];
+
+#ifdef ANDROID
+    bool termux_usb;
+#endif
 };
 
 class UsbTransport : public Transport {
@@ -64,4 +72,8 @@ class UsbTransport : public Transport {
 typedef int (*ifc_match_func)(usb_ifc_info *ifc);
 
 // 0 is non blocking
+#ifdef ANDROID
+UsbTransport* usb_open(ifc_match_func callback, libusb_context *context, uint32_t timeout_ms = 0);
+#else
 UsbTransport* usb_open(ifc_match_func callback, uint32_t timeout_ms = 0);
+#endif
