diff --git a/vendor/incremental_delivery/incfs/util/include/util/map_ptr.h b/vendor/incremental_delivery/incfs/util/include/util/map_ptr.h
index 6fd38ee..bb62fd4 100644
--- a/vendor/incremental_delivery/incfs/util/include/util/map_ptr.h
+++ b/vendor/incremental_delivery/incfs/util/include/util/map_ptr.h
@@ -23,6 +23,7 @@
 #include <android-base/logging.h>
 #include <android-base/off64_t.h>
 
+#undef __ANDROID__
 #ifdef __ANDROID__
 #include <linux/incrementalfs.h>
 #endif
