diff --git a/vendor/core/fastboot/fastboot.h b/vendor/core/fastboot/fastboot.h
index c23793a..b7c07f0 100644
--- a/vendor/core/fastboot/fastboot.h
+++ b/vendor/core/fastboot/fastboot.h
@@ -26,11 +26,18 @@
  * SUCH DAMAGE.
  */
 
+#ifdef ANDROID
+#include <libusb-1.0/libusb.h>
+#endif
+
 #include <bootimg.h>
 
 class FastBootTool {
   public:
     int Main(int argc, char* argv[]);
+#ifdef ANDROID
+    int RealMain(int argc, char* argv[], libusb_context *context);
+#endif
 
     void ParseOsPatchLevel(boot_img_hdr_v1*, const char*);
     void ParseOsVersion(boot_img_hdr_v1*, const char*);
