--- a/vendor/core/fastboot/fastboot_driver_interface.h	2026-08-01 08:20:11.000000000 +0300
+++ b/vendor/core/fastboot/fastboot_driver_interface.h	2026-09-18 12:14:55.990251721 +0300
@@ -125,6 +125,7 @@
     virtual RetCode SetActive(const std::string& slot, std::string* response = nullptr,
                               std::vector<std::string>* info = nullptr) = 0;
     virtual void set_transport(std::unique_ptr<Transport> transport) = 0;
+    virtual Transport* transport() = 0;
     virtual RetCode SnapshotUpdateCommand(const std::string& command,
                                           std::string* response = nullptr,
                                           std::vector<std::string>* info = nullptr) = 0;
