--- a/crawl-ref/source/util/gen_ver.pl
+++ b/crawl-ref/source/util/gen_ver.pl
@@ -20,7 +20,7 @@
 #
 # Source tarbells distributed as part of a release include this file already
 # generated with the release version.
-$_ = `git describe --tags $mergebase 2> /dev/null`
+$_ = "@VERSION@"
     || (open(IN, "<", "$scriptpath/release_ver") ? <IN>
         : die "Can't get version information: `git describe --tags` failed (no git, no repository, or shallow clone), and $scriptpath/release_ver doesn't exist.\n")
     or die "couldn't get the version information\n";
