--- a/source/util/gen_ver.pl
+++ b/source/util/gen_ver.pl
@@ -21,7 +21,7 @@
 #
 # Source tarbells distributed as part of a release include this file already
 # generated with the release version.
-$_ = `git describe $mergebase 2> $nullfile`
+$_ = "@VERSION@"
     || (open(IN, "<", "$scriptpath/release_ver") ? <IN>
         : die "Error: Can't get version information: `git describe` failed (no git, no repository, or shallow clone), and $scriptpath/release_ver doesn't exist.\n")
     or die "Error: couldn't get the version information\n";
