diff -uNr signify-31/signify.c signify-31.mod/signify.c
--- signify-31/signify.c	2022-03-13 21:29:52.000000000 +0000
+++ signify-31.mod/signify.c	2023-06-10 15:53:46.000000000 +0000
@@ -505,7 +505,7 @@
 readpubkey(const char *pubkeyfile, struct pubkey *pubkey,
     const char *sigcomment, const char *keytype)
 {
-	const char *safepath = "/etc/signify";
+	const char *safepath = "@TERMUX_PREFIX@/etc/signify";
 	char keypath[PATH_MAX];
 
 	if (!pubkeyfile) {
