diff -uNr libssh-0.9.0/src/dh-gex.c libssh-0.9.0.mod/src/dh-gex.c
--- libssh-0.9.0/src/dh-gex.c	2019-06-27 11:37:37.000000000 +0300
+++ libssh-0.9.0.mod/src/dh-gex.c	2019-06-29 17:28:16.549587113 +0300
@@ -294,7 +294,7 @@
 
 #ifdef WITH_SERVER
 
-#define MODULI_FILE "/etc/ssh/moduli"
+#define MODULI_FILE "@TERMUX_PREFIX@/etc/ssh/moduli"
 /* 2     "Safe" prime; (p-1)/2 is also prime. */
 #define SAFE_PRIME 2
 /* 0x04  Probabilistic Miller-Rabin primality tests. */
diff -uNr libssh-0.9.0/src/options.c libssh-0.9.0.mod/src/options.c
--- libssh-0.9.0/src/options.c	2019-06-27 11:34:16.000000000 +0300
+++ libssh-0.9.0.mod/src/options.c	2019-06-29 17:28:52.049933003 +0300
@@ -666,7 +666,7 @@
             SAFE_FREE(session->opts.global_knownhosts);
             if (v == NULL) {
                 session->opts.global_knownhosts =
-                    strdup("/etc/ssh/ssh_known_hosts");
+                    strdup("@TERMUX_PREFIX@/etc/ssh/ssh_known_hosts");
                 if (session->opts.global_knownhosts == NULL) {
                     ssh_set_error_oom(session);
                     return -1;
@@ -1449,7 +1449,7 @@
     session->opts.knownhosts = tmp;
 
     if (session->opts.global_knownhosts == NULL) {
-        tmp = strdup("/etc/ssh/ssh_known_hosts");
+        tmp = strdup("@TERMUX_PREFIX@/etc/ssh/ssh_known_hosts");
     } else {
         tmp = ssh_path_expand_escape(session, session->opts.global_knownhosts);
     }
diff -uNr libssh-0.9.0/src/socket.c libssh-0.9.0.mod/src/socket.c
--- libssh-0.9.0/src/socket.c	2019-06-27 11:34:16.000000000 +0300
+++ libssh-0.9.0.mod/src/socket.c	2019-06-29 17:26:21.278467080 +0300
@@ -908,7 +908,7 @@
     if (shell == NULL || shell[0] == '\0') {
         /* Fall back to bash. There are issues with dash or
          * whatever people tend to link to /bin/sh */
-        shell = "/bin/bash";
+        shell = "@TERMUX_PREFIX@/bin/bash";
     }
     args[0] = shell;
 
