diff -uNr libssh-0.9.0/src/dh-gex.c libssh-0.9.0.mod/src/dh-gex.c
--- libssh-0.9.0/src/dh-gex.c	2019-06-27 11:37:37.000000000 +0300
+++ libssh-0.9.0.mod/src/dh-gex.c	2019-06-29 17:28:16.549587113 +0300
@@ -294,7 +294,7 @@
 
 #ifdef WITH_SERVER
 
-#define MODULI_FILE "/etc/ssh/moduli"
+#define MODULI_FILE "@TERMUX_PREFIX@/etc/ssh/moduli"
 /* 2     "Safe" prime; (p-1)/2 is also prime. */
 #define SAFE_PRIME 2
 /* 0x04  Probabilistic Miller-Rabin primality tests. */
diff -uNr libssh-0.9.0/src/socket.c libssh-0.9.0.mod/src/socket.c
--- libssh-0.9.0/src/socket.c	2019-06-27 11:34:16.000000000 +0300
+++ libssh-0.9.0.mod/src/socket.c	2019-06-29 17:26:21.278467080 +0300
@@ -1149,11 +1149,11 @@
     if (shell == NULL || shell[0] == '\0') {
         /* Fall back to the /bin/sh only if the bash is not available. But there are
          * issues with dash or whatever people tend to link to /bin/sh */
-        rc = access("/bin/bash", 0);
+        rc = access("@TERMUX_PREFIX@/bin/bash", 0);
         if (rc != 0) {
-            shell = "/bin/sh";
+            shell = "@TERMUX_PREFIX@/bin/sh";
         } else {
-            shell = "/bin/bash";
+            shell = "@TERMUX_PREFIX@/bin/bash";
         }
     }
     args[0] = shell;
--- a/DefineOptions.cmake
+++ b/DefineOptions.cmake
@@ -71,7 +71,7 @@
     set(WITH_SYMBOL_VERSIONING ON)
 endif (WITH_ABI_BREAK)
 
-set(GLOBAL_CONF_DIR "/etc/ssh")
+set(GLOBAL_CONF_DIR "@TERMUX_PREFIX@/etc/ssh")
 if (WIN32)
     # Use PROGRAMDATA on Windows
     if (DEFINED ENV{PROGRAMDATA})
