--- ../configure.orig	2022-02-18 20:35:05.112179802 +0000
+++ ./configure	2022-02-18 20:35:16.316087977 +0000
@@ -7598,7 +7598,7 @@
         sys/param.h sys/time.h sys/times.h \
         sys/types.h sys/stat.h fcntl.h termios.h sys/ioctl.h \
         sys/resource.h sys/stropts.h windows.h dirent.h getopt.h malloc.h \
-        semaphore.h pthread.h time.h spawn.h fenv.h sys/mman.h sys/sem.h \
+        semaphore.h pthread.h time.h fenv.h sys/mman.h sys/sem.h \
         sched.h utmpx.h dispatch/dispatch.h stdint.h sys/select.h ; do
 	mercury_cv_header_define="MR_HAVE_`echo $mercury_cv_header | \
 		tr abcdefghijklmnopqrstuvwxyz./ ABCDEFGHIJKLMNOPQRSTUVWXYZ__`"
