diff -uNr ddrescue-1.22/configure ddrescue-1.22.mod/configure
--- ddrescue-1.22/configure	2017-02-03 14:22:59.000000000 +0200
+++ ddrescue-1.22.mod/configure	2017-09-09 22:03:38.904467078 +0300
@@ -15,23 +15,15 @@
 LC_ALL=C
 export LC_ALL
 srcdir=
-prefix=/usr/local
+prefix=@TERMUX_PREFIX@
 exec_prefix='$(prefix)'
 bindir='$(exec_prefix)/bin'
 datarootdir='$(prefix)/share'
 infodir='$(datarootdir)/info'
 mandir='$(datarootdir)/man'
-CXX=g++
 CPPFLAGS=
 CXXFLAGS='-Wall -W -O2'
-LDFLAGS=
-
-# checking whether we are using GNU C++.
-/bin/sh -c "${CXX} --version" > /dev/null 2>&1 ||
-	{
-	CXX=c++
-	CXXFLAGS='-W -O2'
-	}
+LDFLAGS=-L/data/data/com.termux/files/usr/lib
 
 # Loop over all args
 args=
