--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -74,7 +74,6 @@
 # cmake modules
 include(CheckCCompilerFlag)
 include(CheckCSourceCompiles)
-include(CheckIPOSupported)
 include(GNUInstallDirs)
 
 # custom cmake modules
@@ -132,7 +131,6 @@
 # required dependencies
 find_package(Threads REQUIRED)
 find_package(PCRE2 REQUIRED)
-find_library(MATH_LIB m REQUIRED)
 find_package(OpenSSL REQUIRED)
 
 # check dependencies versions
@@ -169,6 +167,8 @@
   endif()
   message("Searching for lua")
   find_package(Lua)
+  set(LUA_LIBRARIES "lua5.4")
+  set(LUA_FOUND TRUE)
   if(LUA_FOUND)
     if(NOT LUA_VERSION_STRING VERSION_GREATER_EQUAL "5.3.0")
       message("Lua is disabled because a version lower than 5.3.0 was found")
@@ -238,7 +238,7 @@
 configure_file(contrib/initscripts/mympd.sysVinit.in contrib/initscripts/mympd.sysVinit @ONLY)
 configure_file(contrib/initscripts/mympd.openrc.in contrib/initscripts/mympd.openrc @ONLY)
 
-if(CMAKE_BUILD_TYPE MATCHES "(Release|Debug)")
+if(FALSE)
   # set strict global compile flags
   add_compile_options(
     "-fdata-sections"
@@ -329,8 +329,7 @@
     add_compile_options("-g")
   endif()
   # IPO/LTO support
-  check_ipo_supported()
-  set(CMAKE_INTERPROCEDURAL_OPTIMIZATION TRUE)
+  set(CMAKE_INTERPROCEDURAL_OPTIMIZATION FALSE)
 elseif(CMAKE_BUILD_TYPE STREQUAL "Debug")
   add_compile_options(
     "-ggdb"
