--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -71,7 +71,6 @@
 # cmake modules
 include(CheckCCompilerFlag)
 include(CheckCSourceCompiles)
-include(CheckIPOSupported)
 include(GNUInstallDirs)
 
 # custom cmake modules
@@ -123,7 +122,6 @@
 # required dependencies
 find_package(Threads REQUIRED)
 find_package(PCRE2 REQUIRED)
-find_library(MATH_LIB m REQUIRED)
 
 # optional dependencies
 if(MYMPD_ENABLE_SSL)
@@ -227,7 +225,7 @@
 configure_file(contrib/initscripts/mympd.sysVinit.in contrib/initscripts/mympd.sysVinit @ONLY)
 configure_file(contrib/initscripts/mympd.openrc.in contrib/initscripts/mympd.openrc @ONLY)
 
-if(CMAKE_BUILD_TYPE MATCHES "(Release|Debug)")
+if(FALSE)
   # set strict global compile flags
   add_compile_options(
     "-fdata-sections"
@@ -317,8 +315,7 @@
     add_compile_options("-g")
   endif()
   # IPO/LTO support
-  check_ipo_supported()
-  set(CMAKE_INTERPROCEDURAL_OPTIMIZATION TRUE)
+  set(CMAKE_INTERPROCEDURAL_OPTIMIZATION FALSE)
 elseif(CMAKE_BUILD_TYPE STREQUAL "Debug")
   add_compile_options(
     "-ggdb"
