--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -125,7 +125,6 @@
 # required dependencies
 find_package(Threads REQUIRED)
 find_package(PCRE2 REQUIRED)
-find_library(MATH_LIB m REQUIRED)
 
 # optional dependencies
 if(MYMPD_ENABLE_SSL)
@@ -229,7 +228,7 @@
 configure_file(contrib/initscripts/mympd.sysVinit.in contrib/initscripts/mympd.sysVinit @ONLY)
 configure_file(contrib/initscripts/mympd.openrc.in contrib/initscripts/mympd.openrc @ONLY)
 
-if(CMAKE_BUILD_TYPE MATCHES "(Release|Debug)")
+if(FALSE)
   # set strict global compile flags
   add_compile_options(
     "-fdata-sections"
