--- a/cmake/CopyConfig.cmake.in
+++ b/cmake/CopyConfig.cmake.in
@@ -2,6 +2,7 @@
 # myMPD (c) 2018-2023 Juergen Mang <mail@jcgames.de>
 # https://github.com/jcorporation/mympd
 
+if(FALSE)
 if("@MYMPD_STARTUP_SCRIPT@" STREQUAL "ON")
   # Install startup script
   if(EXISTS "/lib/systemd/system" OR EXISTS "/usr/lib/systemd/system")
@@ -27,6 +28,7 @@
     message("No supported init system found, no startup script was installed")
   endif()
 endif()
+endif(FALSE)
 
 # Install and compress manpages
 if("@MYMPD_MANPAGES@" STREQUAL "ON")
