--- a/src/lib/config/cacertstore.c
+++ b/src/lib/config/cacertstore.c
@@ -20,12 +20,7 @@
  * Paths to check for the ca cert store
  */
 const char *check_ca_cert_paths[] = {
-    "/etc/ssl/ca-bundle.pem",  // openSUSE
-    "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem", // Fedora
-    "/etc/pki/tls/certs/ca-bundle.crt",  // Fedora
-    "/etc/ssl/certs/ca-bundle.crt",
-    "/etc/ssl/certs/ca-certificates.crt",  // Debian
-    "/var/lib/ca-certificates/ca-bundle.pem",  // openSUSE
+    "@TERMUX_PREFIX@/etc/tls/cert.pem",
     NULL
 };
 
