--- a/alc/alconfig.cpp
+++ b/alc/alconfig.cpp
@@ -384,13 +384,13 @@
 void ReadALConfig()
 {
     namespace fs = std::filesystem;
-    fs::path path{"/etc/openal/alsoft.conf"};
+    fs::path path{"@TERMUX_PREFIX@/etc/openal/alsoft.conf"};
 
     TRACE("Loading config %s...\n", reinterpret_cast<const char*>(path.u8string().c_str()));
     if(std::ifstream f{path}; f.is_open())
         LoadConfigFromFile(f);
 
-    std::string confpaths{al::getenv("XDG_CONFIG_DIRS").value_or("/etc/xdg")};
+    std::string confpaths{al::getenv("XDG_CONFIG_DIRS").value_or("@TERMUX_PREFIX@/etc/xdg")};
     /* Go through the list in reverse, since "the order of base directories
      * denotes their importance; the first directory listed is the most
      * important". Ergo, we need to load the settings from the later dirs
