From 85c195b9705162566e092ad6683a4d616d62967d Mon Sep 17 00:00:00 2001
From: Chris Robinson <chris.kcat@gmail.com>
Date: Wed, 31 Dec 2025 04:43:58 -0800
Subject: [PATCH] Return the correct size for the rinbuffer read/write spans

---
 common/ringbuffer.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/common/ringbuffer.h b/common/ringbuffer.h
index 55b4ba62a2..8ab352a6bb 100644
--- a/common/ringbuffer.h
+++ b/common/ringbuffer.h
@@ -143,9 +143,9 @@ class RingBuffer {
              * ptr, plus some from the start of the buffer.
              */
             return DataPair{std::span{mBuffer}.subspan(read_idx*mElemSize),
-                std::span{mBuffer}.first(rdend&mSizeMask)};
+                std::span{mBuffer}.first((rdend&mSizeMask)*mElemSize)};
         }
-        return DataPair{std::span{mBuffer}.subspan(read_idx*mElemSize, readable), {}};
+        return DataPair{std::span{mBuffer}.subspan(read_idx*mElemSize, readable*mElemSize), {}};
     }
 
     /** Advance the read pointer `count' places. */
@@ -210,9 +210,9 @@ class RingBuffer {
              * ptr, plus some from the start of the buffer.
              */
             return DataPair{std::span{mBuffer}.subspan(write_idx*mElemSize),
-                std::span{mBuffer}.first(wrend&mSizeMask)};
+                std::span{mBuffer}.first((wrend&mSizeMask)*mElemSize)};
         }
-        return DataPair{std::span{mBuffer}.subspan(write_idx*mElemSize, writable), {}};
+        return DataPair{std::span{mBuffer}.subspan(write_idx*mElemSize, writable*mElemSize), {}};
     }
 
     /** Advance the write pointer `count' places. */
