diff -uNr openal-soft-openal-soft-1.19.1/native-tools/CMakeLists.txt openal-soft-openal-soft-1.19.1.mod/native-tools/CMakeLists.txt
--- openal-soft-openal-soft-1.19.1/native-tools/CMakeLists.txt	2018-10-12 01:05:31.000000000 +0300
+++ openal-soft-openal-soft-1.19.1.mod/native-tools/CMakeLists.txt	2019-07-24 15:09:41.667923002 +0300
@@ -4,6 +4,11 @@
 
 include(CheckLibraryExists)
 
+set(CMAKE_C_COMPILER gcc)
+set(CMAKE_C_FLAGS "")
+set(CMAKE_LINKER "")
+set(CMAKE_EXE_LINKER_FLAGS "")
+
 set(CPP_DEFS )
 if(WIN32)
     set(CPP_DEFS ${CPP_DEFS} _WIN32)
