--- a/greed.rs	2026-05-20 17:22:59.464795567 +0530
+++ b/greed.rs	2026-05-20 17:24:04.885148321 +0530
@@ -17,12 +17,12 @@
 
 const ME: char = '@';
 const MAXSCORES: usize = 10;
-const LOCKPATH: &str = "/tmp/Greed.lock";
+const LOCKPATH: &str = "@TERMUX_PREFIX@/tmp/Greed.lock";
 const LOCALSCOREFILE: &str = ".greedscores";
 const USERNAMELEN: usize = 32;
 const USER_FIELD_LEN: usize = USERNAMELEN + 1;
 const DEFAULT_RELEASE: &str = " (dev-build)";
-const DEFAULT_SCOREFILE: &str = "/usr/games/lib/greed.hs";
+const DEFAULT_SCOREFILE: &str = "@TERMUX_PREFIX@/var/greed.hs";
 
 type TimeT = i64;
 
