diff -u -r ../greed-3.9/greed.c ./greed.c
--- ../greed-3.9/greed.c	2013-10-23 18:40:41.000000000 +0200
+++ ./greed.c	2014-02-22 11:00:49.000000000 +0100
@@ -69,7 +69,7 @@
 /* rnd() returns a random number between 1 and x */
 #define rnd(x) (int) ((lrand48() % (x))+1)
 
-#define LOCKPATH "/tmp/Greed.lock"	/* lock path for high score file */
+#define LOCKPATH "@TERMUX_PREFIX@/tmp/Greed.lock"	/* lock path for high score file */
 
 /* 
  * changing stuff in this struct
