diff -u -r ../greed-3.9/Makefile ./Makefile
--- ../greed-3.9/Makefile	2013-10-23 18:57:16.000000000 +0200
+++ ./Makefile	2014-02-12 14:33:14.000000000 +0100
@@ -2,12 +2,12 @@
 
 VERS=3.9
 
-SFILE=/usr/games/lib/greed.hs
+SFILE=@TERMUX_PREFIX@/var/greed.hs
 # Location of game executable
-BIN=/usr/games
+BIN=@TERMUX_PREFIX@/bin
 
 greed: greed.c
-	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -DSCOREFILE=\"$(SFILE)\" -DRELEASE=\"$(VERS)\" -o greed greed.c -O3 -lcurses
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -DSCOREFILE=\"$(SFILE)\" -DRELEASE=\"$(VERS)\" -o greed greed.c -lncurses
 
 greed.6: greed.xml
 	xmlto man greed.xml
@@ -17,7 +17,8 @@
 
 install: greed.6 uninstall
 	cp greed $(BIN)
-	cp greed.6 /usr/share/man/man6/greed.6
+	mkdir -p @TERMUX_PREFIX@/share/man/man6
+	cp greed.6 @TERMUX_PREFIX@/share/man/man6/greed.6
 
 uninstall:
 	rm -f $(BIN)/install /usr/share/man/man6/greed.6
