--- a/python/recoll/setup.py.in
+++ b/python/recoll/setup.py.in
@@ -16,8 +16,8 @@
 # Ensure that we can build with an uninstalled lib from a build tree
 library_dirs = [os.path.realpath(os.path.join(localtop, '.libs'))]

-if "CYGWIN" in os.environ:
-        libraries =  ['recoll', 'xapian', 'iconv', 'z']
+if True:
+        libraries = ['recoll', 'python@PYTHON_VERSION@', 'xslt', 'xml2', 'xapian', 'iconv', 'z']
 else:
         libraries = ['recoll']

