diff -uNr libuv-v1.41.0/Makefile.am libuv-v1.41.0.mod/Makefile.am
--- libuv-v1.41.0/Makefile.am	2021-02-13 17:57:24.000000000 +0000
+++ libuv-v1.41.0.mod/Makefile.am	2021-05-04 18:36:54.447549290 +0000
@@ -386,9 +386,7 @@
 uvinclude_HEADERS += include/uv/android-ifaddrs.h
 libuv_la_CFLAGS += -D_GNU_SOURCE
 libuv_la_SOURCES += src/unix/android-ifaddrs.c \
-                    src/unix/pthread-fixes.c \
-                    src/unix/random-getrandom.c \
-                    src/unix/random-sysctl-linux.c
+                    src/unix/pthread-fixes.c
 endif
 
 if CYGWIN
