--- ./Makefile.am	2021-07-20 08:14:04.000000000 +0530
+++ ./Makefile.am.mod	2021-09-19 10:24:59.478382337 +0530
@@ -390,10 +390,7 @@
 uvinclude_HEADERS += include/uv/android-ifaddrs.h
 libuv_la_CFLAGS += -D_GNU_SOURCE
 libuv_la_SOURCES += src/unix/android-ifaddrs.c \
-                    src/unix/pthread-fixes.c \
-                    src/unix/random-getrandom.c \
-                    src/unix/random-sysctl-linux.c \
-                    src/unix/epoll.c
+                    src/unix/pthread-fixes.c
 endif
 
 if CYGWIN
