diff -uNr libuv-v1.38.1/Makefile.am libuv-v1.38.1.mod/Makefile.am
--- libuv-v1.38.1/Makefile.am	2020-07-03 19:51:27.000000000 +0300
+++ libuv-v1.38.1.mod/Makefile.am	2020-07-22 19:44:17.806814523 +0300
@@ -382,13 +382,7 @@
 uvinclude_HEADERS += include/uv/android-ifaddrs.h
 libuv_la_CFLAGS += -D_GNU_SOURCE
 libuv_la_SOURCES += src/unix/android-ifaddrs.c \
-                    src/unix/linux-core.c \
-                    src/unix/linux-inotify.c \
-                    src/unix/linux-syscalls.c \
-                    src/unix/procfs-exepath.c \
-                    src/unix/pthread-fixes.c \
-                    src/unix/random-getrandom.c \
-                    src/unix/random-sysctl-linux.c
+                    src/unix/pthread-fixes.c
 endif
 
 if CYGWIN
