diff -u -r ../libuv-v1.34.0/Makefile.am ./Makefile.am
--- ../libuv-v1.34.0/Makefile.am	2019-12-04 03:11:28.000000000 +0000
+++ ./Makefile.am	2020-01-02 19:24:23.274396000 +0000
@@ -387,14 +387,7 @@
 if ANDROID
 uvinclude_HEADERS += include/uv/android-ifaddrs.h
 libuv_la_SOURCES += src/unix/android-ifaddrs.c \
-                    src/unix/linux-core.c \
-                    src/unix/linux-inotify.c \
-                    src/unix/linux-syscalls.c \
-                    src/unix/procfs-exepath.c \
-                    src/unix/pthread-fixes.c \
-                    src/unix/random-getrandom.c \
-                    src/unix/random-sysctl-linux.c \
-                    src/unix/sysinfo-loadavg.c
+                    src/unix/pthread-fixes.c
 endif
 
 if CYGWIN
