--- gst-plugins-good-1.20.0/ext/libcaca/gstcacasink.c	2021-11-26 09:39:43.543715367 +0000
+++ gst-plugins-good-1.20.0.mod/ext/libcaca/gstcacasink.c	2021-11-26 09:42:13.306342649 +0000
@@ -88,7 +88,7 @@
 
 #define gst_cacasink_parent_class parent_class
 G_DEFINE_TYPE (GstCACASink, gst_cacasink, GST_TYPE_BASE_SINK);
-GST_ELEMENT_REGISTER_DEFINE (cacasink, "cacasink", GST_RANK_NONE,
+GST_ELEMENT_REGISTER_DEFINE (cacasink, "cacasink", GST_RANK_PRIMARY,
     GST_TYPE_CACASINK);
 
 #define GST_TYPE_CACADITHER (gst_cacasink_dither_get_type())
