--- src/ext/libcaca/gstcacasink.c.orig	2017-10-05 05:57:01.990156123 +0000
+++ src/ext/libcaca/gstcacasink.c	2017-10-05 06:17:58.320938634 +0000
@@ -407,7 +407,7 @@
   if (!gst_element_register (plugin, "cacatv", GST_RANK_NONE, GST_TYPE_CACATV))
     return FALSE;
 
-  if (!gst_element_register (plugin, "cacasink", GST_RANK_NONE,
+  if (!gst_element_register (plugin, "cacasink", GST_RANK_PRIMARY,
           GST_TYPE_CACASINK))
     return FALSE;
 
