diff -uNr android-trusty-4.4/tools/build/fixdep.c android-trusty-4.4.mod/tools/build/fixdep.c
--- android-trusty-4.4/tools/build/fixdep.c	2024-04-11 13:24:41.206000000 +0800
+++ android-trusty-4.4.mod/tools/build/fixdep.c	2024-04-12 21:58:38.375997235 +0800
@@ -49,7 +49,7 @@
 	char *end = m + len;
 	char *p;
 	char s[PATH_MAX];
-	int is_target;
+	int is_target, has_target = 0;
 	int saw_any_target = 0;
 	int is_first_dep = 0;
 
@@ -67,7 +67,8 @@
 		if (is_target) {
 			/* The /next/ file is the first dependency */
 			is_first_dep = 1;
-		} else {
+			has_target = 1;
+		} else if (has_target) {
 			/* Save this token/filename */
 			memcpy(s, m, p-m);
 			s[p - m] = 0;
