diff -uNr android-trusty-4.4/tools/lib/traceevent/event-parse.c android-trusty-4.4.mod/tools/lib/traceevent/event-parse.c
--- android-trusty-4.4/tools/lib/traceevent/event-parse.c	2024-04-11 13:24:41.255000000 +0800
+++ android-trusty-4.4.mod/tools/lib/traceevent/event-parse.c	2024-04-12 18:31:07.427997595 +0800
@@ -31,8 +31,11 @@
 #include <errno.h>
 #include <stdint.h>
 #include <limits.h>
+#include <inttypes.h>
 
 #include <netinet/ip6.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
 #include "event-parse.h"
 #include "event-utils.h"
 
