diff -uNr android-trusty-4.4/tools/perf/util/cpumap.h android-trusty-4.4.mod/tools/perf/util/cpumap.h
--- android-trusty-4.4/tools/perf/util/cpumap.h	2024-04-11 13:24:41.516000000 +0800
+++ android-trusty-4.4.mod/tools/perf/util/cpumap.h	2024-04-12 22:05:04.719996959 +0800
@@ -56,9 +56,9 @@
 	return map ? map->map[0] == -1 : true;
 }
 
-int max_cpu_num;
-int max_node_num;
-int *cpunode_map;
+extern int max_cpu_num;
+extern int max_node_num;
+extern int *cpunode_map;
 
 int cpu__setup_cpunode_map(void);
 
