diff -uNr android-trusty-4.4/tools/lib/api/Makefile android-trusty-4.4.mod/tools/lib/api/Makefile
--- android-trusty-4.4/tools/lib/api/Makefile	2024-04-11 13:24:41.230000000 +0800
+++ android-trusty-4.4.mod/tools/lib/api/Makefile	2024-04-12 15:28:35.424995725 +0800
@@ -8,8 +8,8 @@
 #$(info Determined 'srctree' to be $(srctree))
 endif
 
-CC = $(CROSS_COMPILE)gcc
-AR = $(CROSS_COMPILE)ar
+#CC = $(CROSS_COMPILE)gcc
+#AR = $(CROSS_COMPILE)ar
 
 MAKEFLAGS += --no-print-directory
 
