diff -uNr android-trusty-4.4/tools/lib/bpf/bpf.c android-trusty-4.4.mod/tools/lib/bpf/bpf.c
--- android-trusty-4.4/tools/lib/bpf/bpf.c	2024-04-11 13:24:41.232000000 +0800
+++ android-trusty-4.4.mod/tools/lib/bpf/bpf.c	2024-04-12 20:01:21.227993722 +0800
@@ -9,6 +9,7 @@
 #include <stdlib.h>
 #include <memory.h>
 #include <unistd.h>
+#include <strings.h>
 #include <asm/unistd.h>
 #include <linux/bpf.h>
 #include "bpf.h"
