diff -uNr android-trusty-4.4/tools/build/feature/Makefile android-trusty-4.4.mod/tools/build/feature/Makefile
--- android-trusty-4.4/tools/build/feature/Makefile	2024-04-11 13:24:41.199000000 +0800
+++ android-trusty-4.4.mod/tools/build/feature/Makefile	2024-04-14 01:24:30.873985820 +0800
@@ -44,7 +44,7 @@
 all: $(FILES)
 
 __BUILD = $(CC) $(CFLAGS) -Wall -Werror -o $(OUTPUT)$@ $(patsubst %.bin,%.c,$@) $(LDFLAGS)
-  BUILD = $(__BUILD) > $(OUTPUT)$(@:.bin=.make.output) 2>&1
+  BUILD = $(__BUILD)
 
 ###############################
 
