diff -u -r ../jsoncpp-1.6.2/src/lib_json/json_writer.cpp ./src/lib_json/json_writer.cpp
--- ../jsoncpp-1.6.2/src/lib_json/json_writer.cpp	2015-04-11 15:45:33.000000000 -0400
+++ ./src/lib_json/json_writer.cpp	2015-05-20 18:05:22.382255829 -0400
@@ -29,7 +29,7 @@
 
 #if defined(_MSC_VER) && _MSC_VER < 1500 // VC++ 8.0 and below
 #define snprintf _snprintf
-#elif __cplusplus >= 201103L
+#elif (__cplusplus >= 201103L) && !defined(__ANDROID__)
 #define snprintf std::snprintf
 #endif
 
