diff --git a/CMakeLists.txt b/CMakeLists.txt
index 19e61cd14..a4b9a8aa4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -193,14 +193,6 @@ if (NOT HAVE_SIGHANDLER_T)
   endif()
 endif()
 
-# Handle resolving
-check_function_exists(res_ninit HAVE_LIBC_RESOLV)
-if(HAVE_LIBC_RESOLV)
-    set(RESOLV_LIBRARIES)
-else()
-    set(RESOLV_LIBRARIES -lresolv)
-endif()
-
 # Configure some variables like package, version and architecture.
 set(PACKAGE ${PROJECT_NAME})
 set(PACKAGE_MAIL "APT Development Team <deity@lists.debian.org>")
@@ -255,7 +247,6 @@ add_subdirectory(doc)
 add_subdirectory(dselect)
 add_subdirectory(ftparchive)
 add_subdirectory(methods)
-add_subdirectory(test)
 
 if (USE_NLS)
 add_subdirectory(po)
diff --git a/apt-pkg/CMakeLists.txt b/apt-pkg/CMakeLists.txt
index 7aaacd9d9..b577e4f12 100644
--- a/apt-pkg/CMakeLists.txt
+++ b/apt-pkg/CMakeLists.txt
@@ -61,7 +61,7 @@ target_include_directories(apt-pkg
 )
 
 target_link_libraries(apt-pkg
-                      PRIVATE -lutil ${CMAKE_DL_LIBS} ${RESOLV_LIBRARIES}
+                      PRIVATE ${CMAKE_DL_LIBS} ${RESOLV_LIBRARIES}
                              ${CMAKE_THREAD_LIBS_INIT}
                              ${ZLIB_LIBRARIES}
                              ${BZIP2_LIBRARIES}
diff --git a/doc/CMakeLists.txt b/doc/CMakeLists.txt
index 2df0ce721..25441d936 100644
--- a/doc/CMakeLists.txt
+++ b/doc/CMakeLists.txt
@@ -3,15 +3,6 @@ include(Documentation)
 endif()
 
 set(LINGUAS
-    de
-    es
-    fr
-    it
-    ja
-    nl
-    pl
-    pt_BR
-    pt
 )
 
 set(TRANSLATED_ENTITIES
