diff -u -r apt-2.9.2/apt-pkg/CMakeLists.txt src/apt-pkg/CMakeLists.txt
--- apt-2.9.2/apt-pkg/CMakeLists.txt	2024-04-22 17:36:30.000000000 +0000
+++ src/apt-pkg/CMakeLists.txt	2024-05-05 21:30:19.498701160 +0000
@@ -57,7 +57,7 @@
 )
 
 target_link_libraries(apt-pkg
-                      PRIVATE -lutil ${CMAKE_DL_LIBS} ${RESOLV_LIBRARIES}
+                      PRIVATE ${CMAKE_DL_LIBS} ${RESOLV_LIBRARIES}
                              ${CMAKE_THREAD_LIBS_INIT}
                              ${ZLIB_LIBRARIES}
                              ${BZIP2_LIBRARIES}
diff -u -r apt-2.9.2/CMakeLists.txt src/CMakeLists.txt
--- apt-2.9.2/CMakeLists.txt	2024-04-22 17:36:30.000000000 +0000
+++ src/CMakeLists.txt	2024-05-05 21:30:19.498701160 +0000
@@ -195,14 +195,6 @@
   endif()
 endif()
 
-# Handle resolving
-check_function_exists(res_ninit HAVE_LIBC_RESOLV)
-if(HAVE_LIBC_RESOLV)
-    set(RESOLV_LIBRARIES)
-else()
-    set(RESOLV_LIBRARIES -lresolv)
-endif()
-
 # Configure some variables like package, version and architecture.
 set(PACKAGE ${PROJECT_NAME})
 set(PACKAGE_MAIL "APT Development Team <deity@lists.debian.org>")
@@ -215,10 +207,6 @@
   message(STATUS "Found dpkg data dir: ${DPKG_DATADIR_CMD}")
   set(DPKG_DATADIR "${DPKG_DATADIR_CMD}" CACHE PATH "dpkg data directory")
 endif()
-if (NOT DEFINED COMMON_ARCH)
-  execute_process(COMMAND dpkg-architecture -qDEB_HOST_ARCH
-                  OUTPUT_VARIABLE COMMON_ARCH OUTPUT_STRIP_TRAILING_WHITESPACE)
-endif()
 if (NOT DEFINED ROOT_GROUP)
   execute_process(COMMAND id -gn root
                   OUTPUT_VARIABLE ROOT_GROUP OUTPUT_STRIP_TRAILING_WHITESPACE)
@@ -251,7 +239,6 @@
 add_subdirectory(dselect)
 add_subdirectory(ftparchive)
 add_subdirectory(methods)
-add_subdirectory(test)
 
 if (USE_NLS)
 add_subdirectory(po)
diff -u -r apt-2.9.2/methods/CMakeLists.txt src/methods/CMakeLists.txt
--- apt-2.9.2/methods/CMakeLists.txt	2024-04-22 17:36:30.000000000 +0000
+++ src/methods/CMakeLists.txt	2024-05-05 21:31:42.650799537 +0000
@@ -11,7 +11,6 @@
 add_executable(cdrom cdrom.cc)
 add_executable(http http.cc basehttp.cc $<TARGET_OBJECTS:connectlib>)
 add_executable(mirror mirror.cc)
-add_executable(ftp ftp.cc $<TARGET_OBJECTS:connectlib>)
 add_executable(rred rred.cc)
 add_executable(rsh rsh.cc)
 
@@ -23,12 +22,11 @@
 
 # Additional libraries to link against for networked stuff
 target_link_libraries(http $<$<BOOL:${GNUTLS_FOUND}>:${GNUTLS_LIBRARIES}> $<$<BOOL:${SYSTEMD_FOUND}>:${SYSTEMD_LIBRARIES}>)
-target_link_libraries(ftp $<$<BOOL:${GNUTLS_FOUND}>:${GNUTLS_LIBRARIES}>)
 
 target_link_libraries(rred apt-private)
 
 # Install the library
-install(TARGETS file copy store gpgv cdrom http ftp rred rsh mirror
+install(TARGETS file copy store gpgv cdrom http rred rsh mirror
         RUNTIME DESTINATION ${CMAKE_INSTALL_LIBEXECDIR}/apt/methods)
 
 add_links(${CMAKE_INSTALL_LIBEXECDIR}/apt/methods mirror mirror+ftp mirror+http mirror+https mirror+file mirror+copy)
diff -u -r apt-2.9.2/doc/CMakeLists.txt src/doc/CMakeLists.txt
--- apt-2.9.2/doc/CMakeLists.txt	2024-04-22 17:36:30.000000000 +0000
+++ src/doc/CMakeLists.txt	2024-05-05 21:30:19.499701137 +0000
@@ -3,15 +3,6 @@
 endif()
 
 set(LINGUAS
-    de
-    es
-    fr
-    it
-    ja
-    nl
-    pl
-    pt_BR
-    pt
 )
 
 set(TRANSLATED_ENTITIES
