diff -uNr apt-2.1.11/apt-private/private-download.cc apt-2.1.11.mod/apt-private/private-download.cc
--- apt-2.1.11/apt-private/private-download.cc	2020-10-21 12:53:18.000000000 +0300
+++ apt-2.1.11.mod/apt-private/private-download.cc	2020-11-05 14:22:21.865355149 +0200
@@ -126,7 +126,6 @@
 									/*}}}*/
 bool CheckFreeSpaceBeforeDownload(std::string const &Dir, unsigned long long FetchBytes)/*{{{*/
 {
-   uint32_t const RAMFS_MAGIC = 0x858458f6;
    /* Check for enough free space, but only if we are actually going to
       download */
    if (_config->FindB("APT::Get::Print-URIs", false) == true ||
