diff --git a/apt-private/private-download.cc b/apt-private/private-download.cc
index 60cdfc9ec..7602b18d2 100644
--- a/apt-private/private-download.cc
+++ b/apt-private/private-download.cc
@@ -126,8 +126,6 @@ bool AcquireRun(pkgAcquire &Fetcher, int const PulseInterval, bool * const Failu
 									/*}}}*/
 bool CheckFreeSpaceBeforeDownload(std::string const &Dir, unsigned long long FetchBytes)/*{{{*/
 {
-   uint32_t const RAMFS_MAGIC = 0x858458f6;
-   uint32_t const TMPFS_MAGIC = 0x01021994;
    /* Check for enough free space, but only if we are actually going to
       download */
    if (_config->FindB("APT::Get::Print-URIs", false) == true ||
