diff -uNr apt-1.4.9/apt-pkg/contrib/fileutl.cc apt-1.4.9.mod/apt-pkg/contrib/fileutl.cc
--- apt-1.4.9/apt-pkg/contrib/fileutl.cc	2019-11-12 22:12:25.625803839 +0200
+++ apt-1.4.9.mod/apt-pkg/contrib/fileutl.cc	2019-11-12 22:36:46.021615823 +0200
@@ -46,7 +46,9 @@
 #include <dirent.h>
 #include <signal.h>
 #include <errno.h>
+#ifndef __ANDROID__
 #include <glob.h>
+#endif
 #include <pwd.h>
 #include <grp.h>
 
@@ -2752,6 +2754,7 @@
 }
 									/*}}}*/
 
+#ifndef __ANDROID__
 // Glob - wrapper around "glob()"					/*{{{*/
 std::vector<std::string> Glob(std::string const &pattern, int flags)
 {
@@ -2777,6 +2780,7 @@
    globfree(&globbuf);
    return result;
 }
+#endif
 									/*}}}*/
 static std::string APT_NONNULL(1) GetTempDirEnv(char const * const env)	/*{{{*/
 {
diff -uNr apt-1.4.9/apt-pkg/contrib/fileutl.h apt-1.4.9.mod/apt-pkg/contrib/fileutl.h
--- apt-1.4.9/apt-pkg/contrib/fileutl.h	2019-01-18 12:42:07.000000000 +0200
+++ apt-1.4.9.mod/apt-pkg/contrib/fileutl.h	2019-11-12 22:36:46.021615823 +0200
@@ -239,7 +239,9 @@
 APT_HIDDEN std::string flNormalize(std::string file);
 
 // simple c++ glob
+#ifndef __ANDROID__
 std::vector<std::string> Glob(std::string const &pattern, int flags=0);
+#endif
 
 /** \brief Popen() implementation that execv() instead of using a shell
  *
