diff --git a/apt-pkg/contrib/srvrec.cc b/apt-pkg/contrib/srvrec.cc
index 3ddb95abb..3275b6da2 100644
--- a/apt-pkg/contrib/srvrec.cc
+++ b/apt-pkg/contrib/srvrec.cc
@@ -6,6 +6,7 @@
 
    ##################################################################### */
 									/*}}}*/
+#ifndef __ANDROID__
 #include <config.h>
 
 #include <netdb.h>
@@ -210,3 +211,4 @@ SrvRec PopFromSrvRecs(std::vector<SrvRec> &Recs)
 
    return selected;
 }
+#endif
diff --git a/apt-pkg/contrib/srvrec.h b/apt-pkg/contrib/srvrec.h
index 1e981d3d7..f49c91737 100644
--- a/apt-pkg/contrib/srvrec.h
+++ b/apt-pkg/contrib/srvrec.h
@@ -8,6 +8,7 @@
 									/*}}}*/
 #ifndef SRVREC_H
 #define SRVREC_H
+#ifndef __ANDROID__
 
 #include <string>
 #include <vector>
@@ -55,3 +56,4 @@ APT_PUBLIC bool GetSrvRecords(std::string host, int port, std::vector<SrvRec> &R
 APT_PUBLIC SrvRec PopFromSrvRecs(std::vector<SrvRec> &Recs);
 
 #endif
+#endif
diff --git a/cmdline/apt-helper.cc b/cmdline/apt-helper.cc
index 4404259d8..518c98179 100644
--- a/cmdline/apt-helper.cc
+++ b/cmdline/apt-helper.cc
@@ -106,6 +106,7 @@ static bool DoDownloadFile(CommandLine &CmdL)				/*{{{*/
 
    return true;
 }
+#ifndef __ANDROID__
 									/*}}}*/
 static bool DoSrvLookup(CommandLine &CmdL)				/*{{{*/
 {
@@ -134,6 +135,7 @@ static bool DoSrvLookup(CommandLine &CmdL)				/*{{{*/
    return true;
 }
 									/*}}}*/
+#endif
 static const APT::Configuration::Compressor *FindCompressor(std::vector<APT::Configuration::Compressor> const &compressors, std::string const &name) /*{{{*/
 {
    APT::Configuration::Compressor const * compressor = NULL;
@@ -341,7 +343,9 @@ static std::vector<aptDispatchWithHelp> GetCommands()			/*{{{*/
 {
    return {
       {"download-file", &DoDownloadFile, _("download the given uri to the target-path")},
+#ifndef __ANDROID__
       {"srv-lookup", &DoSrvLookup, _("lookup a SRV record (e.g. _http._tcp.ftp.debian.org)")},
+#endif
       {"cat-file", &DoCatFile, _("concatenate files, with automatic decompression")},
       {"hash-file", &DoHashFile, _("hash file")},
       {"auto-detect-proxy", &DoAutoDetectProxy, _("detect proxy using apt.conf")},
diff --git a/methods/connect.cc b/methods/connect.cc
index 96c169e79..27d7c727b 100644
--- a/methods/connect.cc
+++ b/methods/connect.cc
@@ -52,7 +52,9 @@ static std::string LastService;
 static struct addrinfo *LastHostAddr = 0;
 static struct addrinfo *LastUsed = 0;
 
+#ifndef __ANDROID__
 static std::vector<SrvRec> SrvRecords;
+#endif
 
 // Set of IP/hostnames that we timed out before or couldn't resolve
 static std::set<std::string> bad_addr;
@@ -489,6 +491,10 @@ ResultState Connect(std::string Host, int Port, const char *Service,
    // Used by getaddrinfo(); prefer port if given, else fallback to service
    std::string ServiceNameOrPort = Port != 0 ? std::to_string(Port) : Service;
 
+   size_t stackSize = 0;
+
+#ifndef __ANDROID__
+
    if(LastHost != Host || LastService != ServiceNameOrPort)
    {
       SrvRecords.clear();
@@ -506,7 +512,6 @@ ResultState Connect(std::string Host, int Port, const char *Service,
       }
    }
 
-   size_t stackSize = 0;
    // try to connect in the priority order of the srv records
    std::string initialHost{std::move(Host)};
    auto const initialPort = Port;
@@ -528,6 +533,7 @@ ResultState Connect(std::string Host, int Port, const char *Service,
    }
    Host = std::move(initialHost);
    Port = initialPort;
+#endif
 
    // we have no (good) SrvRecords for this host, connect right away
    _error->PushToStack();
