diff -uNr apt-1.4.9/cmdline/apt-key.in apt-1.4.9.mod/cmdline/apt-key.in
--- apt-1.4.9/cmdline/apt-key.in	2019-11-12 22:27:00.015963736 +0200
+++ apt-1.4.9.mod/cmdline/apt-key.in	2019-11-12 22:46:00.284024293 +0200
@@ -16,10 +16,7 @@
 aptkey_echo() { echo "$@"; }
 
 requires_root() {
-	if [ "$(id -u)" -ne 0 ]; then
-		apt_error "This command can only be used by root."
-		exit 1
-	fi
+	:
 }
 
 command_available() {
