diff -uNr apt-1.4.9/apt-pkg/init.cc apt-1.4.9.mod/apt-pkg/init.cc
--- apt-1.4.9/apt-pkg/init.cc	2019-01-18 12:42:07.000000000 +0200
+++ apt-1.4.9.mod/apt-pkg/init.cc	2019-11-12 22:55:03.409615793 +0200
@@ -51,7 +51,7 @@
    return vec;
 }
 
-
+#ifndef __ANDROID__
 // pkgInitArchTupleMap - Initialize the architecture tuple map				/*{{{*/
 // ---------------------------------------------------------------------
 /* This initializes */
@@ -116,7 +116,7 @@
 
    return true;
 }
-									/*}}}*/
+#endif									/*}}}*/
 
     
 // pkgInitConfig - Initialize the configuration class			/*{{{*/
@@ -133,6 +133,7 @@
    Cnf.CndSet("APT::Install-Recommends", true);
    Cnf.CndSet("APT::Install-Suggests", false);
    Cnf.CndSet("Dir","/");
+   Cnf.CndSet("Acquire::Languages", "none");
    
    // State
    Cnf.CndSet("Dir::State", STATE_DIR + 1);
@@ -158,9 +159,9 @@
    Cnf.CndSet("Dir::Etc::trusted", "trusted.gpg");
    Cnf.CndSet("Dir::Etc::trustedparts","trusted.gpg.d");
    Cnf.CndSet("Dir::Bin::methods", LIBEXEC_DIR "/methods");
-   Cnf.CndSet("Dir::Bin::solvers::",LIBEXEC_DIR  "/solvers");
-   Cnf.CndSet("Dir::Bin::planners::",LIBEXEC_DIR  "/planners");
-   Cnf.CndSet("Dir::Media::MountPath","/media/apt");
+   Cnf.CndSet("Dir::Bin::solvers::",LIBEXEC_DIR "/solvers");
+   Cnf.CndSet("Dir::Bin::planners::", LIBEXEC_DIR "/planners");
+   Cnf.CndSet("Dir::Media::MountPath","media/apt");
 
    // State
    Cnf.CndSet("Dir::Log", LOG_DIR + 1);
@@ -283,8 +284,6 @@
 	 return _error->Error(_("Unable to determine a suitable packaging system type"));
    }
 
-   if (pkgInitArchTupleMap() == false)
-      return false;
    
    return Sys->Initialize(Cnf);
 }
