diff -uNr apt-2.3.14/apt-pkg/acquire-item.cc apt-2.3.14.mod/apt-pkg/acquire-item.cc
--- apt-2.3.14/apt-pkg/acquire-item.cc	2022-01-16 10:27:00.000000000 +0200
+++ apt-2.3.14.mod/apt-pkg/acquire-item.cc	2022-01-16 10:28:00.000000000 +0200
@@ -170,6 +170,7 @@
 {
    std::string m;
    strprintf(m, msg, repo.c_str());
+#ifndef __ANDROID__
    if (isError)
    {
       _error->Error("%s", m.c_str());
@@ -181,6 +182,13 @@
       _error->Notice("%s", _("Data from such a repository can't be authenticated and is therefore potentially dangerous to use."));
    }
    _error->Notice("%s", _("See apt-secure(8) manpage for repository creation and user configuration details."));
+#else
+   _error->Error("%s", m.c_str());
+   _error->Notice("%s", _("Metadata integrity can't be verified, repository is disabled now."));
+   _error->Notice("%s", _("Possible causes: unstable or tampered Internet connection, wrong sources.list, outdated keyring or host is down currently."));
+   _error->Notice("%s", _("Please note that all hosting problems or other serious issues we announce on our social media pages."));
+   _error->Notice("%s", _("Use termux-change-repo for switching to a mirror."));
+#endif
    return false;
 }
 									/*}}}*/
