diff -uNr apt-1.4.9/apt-private/private-download.cc apt-1.4.9.mod/apt-private/private-download.cc
--- apt-1.4.9/apt-private/private-download.cc	2019-01-18 12:42:07.000000000 +0200
+++ apt-1.4.9.mod/apt-private/private-download.cc	2019-11-13 14:22:40.305931068 +0200
@@ -127,7 +127,6 @@
 									/*}}}*/
 bool CheckFreeSpaceBeforeDownload(std::string const &Dir, unsigned long long FetchBytes)/*{{{*/
 {
-   uint32_t const RAMFS_MAGIC = 0x858458f6;
    /* Check for enough free space, but only if we are actually going to
       download */
    if (_config->FindB("APT::Get::Print-URIs", false) == true ||
