diff --git a/apt-private/private-cmndline.cc b/apt-private/private-cmndline.cc
index c26a3f7f4..bfd42d84f 100644
--- a/apt-private/private-cmndline.cc
+++ b/apt-private/private-cmndline.cc
@@ -512,7 +512,7 @@ static void BinarySpecificConfiguration(char const * const Binary)	/*{{{*/
    _config->CndSet("Version::1.1::APT::Get::Upgrade-Allow-New", true);
    _config->CndSet("Version::1.1::APT::Cmd::Show-Update-Stats", true);
    _config->CndSet("Version::1.2::DPkg::Progress-Fancy", true);
-   _config->CndSet("Version::1.2::APT::Keep-Downloaded-Packages", false);
+   _config->CndSet("Version::1.2::APT::Keep-Downloaded-Packages", true);
    _config->CndSet("Version::1.5::APT::Get::Update::InteractiveReleaseInfoChanges", true);
    _config->CndSet("Version::2.0::APT::Cmd::Pattern-Only", true);
    _config->CndSet("Version::3.0::Pager", true);
