diff -uNr apt-1.4.9/apt-pkg/acquire-item.cc apt-1.4.9.mod/apt-pkg/acquire-item.cc
--- apt-1.4.9/apt-pkg/acquire-item.cc	2019-01-18 12:42:07.000000000 +0200
+++ apt-1.4.9.mod/apt-pkg/acquire-item.cc	2020-01-16 01:46:26.562224655 +0200
@@ -180,17 +180,11 @@
 {
    std::string m;
    strprintf(m, msg, repo.c_str());
-   if (isError)
-   {
-      _error->Error("%s", m.c_str());
-      _error->Notice("%s", _("Updating from such a repository can't be done securely, and is therefore disabled by default."));
-   }
-   else
-   {
-      _error->Warning("%s", m.c_str());
-      _error->Notice("%s", _("Data from such a repository can't be authenticated and is therefore potentially dangerous to use."));
-   }
-   _error->Notice("%s", _("See apt-secure(8) manpage for repository creation and user configuration details."));
+
+   _error->Error("%s", m.c_str());
+   _error->Notice("%s", _("Metadata integrity can't be verified, repository is disabled now."));
+   _error->Notice("%s", _("Possible cause: repository is under maintenance or down (wrong sources.list URL?)."));
+
    return false;
 }
 									/*}}}*/
