diff -uNr apt-2.1.11/apt-private/private-cmndline.cc apt-2.1.11.mod/apt-private/private-cmndline.cc
--- apt-2.1.11/apt-private/private-cmndline.cc	2020-10-21 12:53:18.000000000 +0300
+++ apt-2.1.11.mod/apt-private/private-cmndline.cc	2020-11-05 17:01:40.119381470 +0200
@@ -473,7 +473,7 @@
       _config->CndSet("Binary::apt::APT::Get::Upgrade-Allow-New", true);
       _config->CndSet("Binary::apt::APT::Cmd::Show-Update-Stats", true);
       _config->CndSet("Binary::apt::DPkg::Progress-Fancy", true);
-      _config->CndSet("Binary::apt::APT::Keep-Downloaded-Packages", false);
+      _config->CndSet("Binary::apt::APT::Keep-Downloaded-Packages", true);
       _config->CndSet("Binary::apt::APT::Get::Update::InteractiveReleaseInfoChanges", true);
       _config->CndSet("Binary::apt::APT::Cmd::Pattern-Only", true);
 
