diff --git a/apt-private/private-cmndline.cc b/apt-private/private-cmndline.cc
index de3992a00..96b2c1251 100644
--- a/apt-private/private-cmndline.cc
+++ b/apt-private/private-cmndline.cc
@@ -461,7 +461,7 @@ static void BinarySpecificConfiguration(char const * const Binary)	/*{{{*/
       _config->CndSet("Binary::apt::APT::Get::Upgrade-Allow-New", true);
       _config->CndSet("Binary::apt::APT::Cmd::Show-Update-Stats", true);
       _config->CndSet("Binary::apt::DPkg::Progress-Fancy", true);
-      _config->CndSet("Binary::apt::APT::Keep-Downloaded-Packages", false);
+      _config->CndSet("Binary::apt::APT::Keep-Downloaded-Packages", true);
    }
    if (binary == "apt-config")
       _config->CndSet("Binary::apt-get::Acquire::AllowInsecureRepositories", true);
