diff -u -r ../apt-1.2.3/apt-pkg/deb/debindexfile.cc ./apt-pkg/deb/debindexfile.cc
--- ../apt-1.2.3/apt-pkg/deb/debindexfile.cc	2016-02-10 12:02:45.000000000 -0500
+++ ./apt-pkg/deb/debindexfile.cc	2016-02-18 02:44:56.707323056 -0500
@@ -26,8 +26,10 @@
 
 #include <stdio.h>
 #include <iostream>
+#include <stdlib.h>
 #include <string>
 #include <sstream>
+#include <unistd.h>
 
 #include <sys/stat.h>
 									/*}}}*/
