diff -u -r ../apt-1.1.3/apt-pkg/makefile ./apt-pkg/makefile
--- ../apt-1.1.3/apt-pkg/makefile	2015-11-30 03:08:24.000000000 -0500
+++ ./apt-pkg/makefile	2015-12-03 16:37:32.041569392 -0500
@@ -15,7 +15,7 @@
 LIBRARY=apt-pkg
 MAJOR=$(LIBAPTPKG_MAJOR)
 MINOR=$(LIBAPTPKG_RELEASE)
-SLIBS=$(PTHREADLIB) $(INTLLIBS) -lutil -ldl -lresolv
+SLIBS=$(PTHREADLIB) $(INTLLIBS) -ldl
 ifeq ($(HAVE_ZLIB),yes)
 SLIBS+= -lz
 endif
