diff -u -r ../apt-1.4.8/apt-pkg/CMakeLists.txt ./apt-pkg/CMakeLists.txt
--- ../apt-1.4.8/apt-pkg/CMakeLists.txt	2017-09-13 18:47:33.000000000 +0200
+++ ./apt-pkg/CMakeLists.txt	2017-12-29 01:09:50.231708153 +0100
@@ -48,13 +48,12 @@
 )
 
 target_link_libraries(apt-pkg
-                      PRIVATE -lutil ${CMAKE_DL_LIBS} ${RESOLV_LIBRARIES}
+                      PRIVATE ${CMAKE_DL_LIBS} ${RESOLV_LIBRARIES}
                              ${CMAKE_THREAD_LIBS_INIT}
                              ${ZLIB_LIBRARIES}
                              ${BZIP2_LIBRARIES}
                              ${LZMA_LIBRARIES}
                              ${LZ4_LIBRARIES}
-                             ${ICONV_LIBRARIES}
 )
 set_target_properties(apt-pkg PROPERTIES VERSION ${MAJOR}.${MINOR})
 set_target_properties(apt-pkg PROPERTIES SOVERSION ${MAJOR})
