diff -u -r ../apt-1.1.3/cmdline/makefile ./cmdline/makefile
--- ../apt-1.1.3/cmdline/makefile	2015-11-30 03:08:24.000000000 -0500
+++ ./cmdline/makefile	2015-12-03 16:50:46.792294882 -0500
@@ -49,7 +49,7 @@
 
 # The apt-helper
 PROGRAM=apt-helper
-SLIBS = -lapt-pkg -lapt-private $(INTLLIBS) -lresolv
+SLIBS = -lapt-pkg -lapt-private $(INTLLIBS)
 LIB_MAKES = apt-pkg/makefile apt-private/makefile
 SOURCE = apt-helper.cc
 include $(PROGRAM_H)
