diff -u -r ../upstream.git/configure.ac ./configure.ac
--- ../upstream.git/configure.ac	2014-06-18 14:12:32.000000000 +0200
+++ ./configure.ac	2014-06-18 23:18:51.219899072 +0200
@@ -89,11 +89,6 @@
      AC_MSG_ERROR([failed: I need CURL due https support]),
 )
 
-AC_LANG_PUSH([C++])
-AC_CHECK_HEADER(gtest/gtest.h,,
-     AC_MSG_ERROR([failed: I need gtest to build tests]),
-)
-AC_LANG_POP([C++])
 
 
 AC_SUBST(BDBLIB)
