diff -u -r ../apt-1.2.12/configure ./configure
--- ../apt-1.2.12/configure	2016-05-11 10:58:04.000000000 +0200
+++ ./configure	2017-03-03 13:31:05.373948035 +0100
@@ -569,7 +569,7 @@
 ac_clean_files=
 ac_config_libobj_dir=.
 LIBOBJS=
-cross_compiling=no
+cross_compiling=yes
 subdirs=
 MFLAGS=
 MAKEFLAGS=
@@ -4611,7 +4611,7 @@
   return 0;
 }
 _ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
+if false; then :
   $as_echo "#define HAVE_BDB 1" >>confdefs.h
 
        BDBLIB="-ldb"
@@ -5056,7 +5056,7 @@
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking debian architecture" >&5
 $as_echo_n "checking debian architecture... " >&6; }
-archset="`dpkg-architecture -qDEB_HOST_ARCH`"
+archset=TERMUX_ARCH
 if test "x$archset" = "x"; then
    as_fn_error $? "failed: use --host= or output from dpkg-architecture" "$LINENO" 5
 fi
