diff -u -r ../apt-0.9.16.1/apt-pkg/deb/debsrcrecords.cc ./apt-pkg/deb/debsrcrecords.cc
--- ../apt-0.9.16.1/apt-pkg/deb/debsrcrecords.cc	2014-03-15 17:23:45.000000000 +0100
+++ ./apt-pkg/deb/debsrcrecords.cc	2014-03-25 01:23:30.652689687 +0100
@@ -26,6 +26,14 @@
 #include <string>
 #include <vector>
 									/*}}}*/
+#if defined(__ANDROID__)
+static char* strchrnul(char const* s, int c)
+{
+	char const* result = strchr(s, c);
+	return const_cast<char*>((result == NULL) ? (s + strlen(s)) : result);
+}
+#endif
+
 
 using std::max;
 using std::string;
