diff -u -r ../apt-1.0.9.4/apt-pkg/deb/dpkgpm.cc ./apt-pkg/deb/dpkgpm.cc
--- ../apt-1.0.9.4/apt-pkg/deb/dpkgpm.cc	2014-12-03 10:06:58.000000000 -0500
+++ ./apt-pkg/deb/dpkgpm.cc	2014-12-07 07:17:09.210804305 -0500
@@ -27,7 +27,9 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <grp.h>
-#include <pty.h>
+#ifndef __ANDROID__
+# include <pty.h>
+#endif
 #include <pwd.h>
 #include <signal.h>
 #include <stddef.h>
@@ -1075,7 +1077,11 @@
 
    _error->PushToStack();
 
+#ifdef __ANDROID__
+   d->master = open("/dev/ptmx", O_RDWR | O_NOCTTY);
+# else
    d->master = posix_openpt(O_RDWR | O_NOCTTY);
+#endif
    if (d->master == -1)
       _error->Errno("posix_openpt", _("Can not write log (%s)"), _("Is /dev/pts mounted?"));
    else if (unlockpt(d->master) == -1)
