--- ../cache/apt-1.4.7/methods/connect.cc	2017-07-14 07:45:39.000000000 +1000
+++ ./methods/connect.cc	2017-07-23 10:50:27.755510218 +1000
@@ -129,7 +129,7 @@
 
    // Check the socket for an error condition
    unsigned int Err;
-   unsigned int Len = sizeof(Err);
+   socklen_t Len = sizeof(Err);
    if (getsockopt(Fd,SOL_SOCKET,SO_ERROR,&Err,&Len) != 0)
       return _error->Errno("getsockopt",_("Failed"));
    
@@ -301,7 +301,10 @@
    if(LastHost != Host || LastPort != Port)
    {
       SrvRecords.clear();
-      if (_config->FindB("Acquire::EnableSrvRecords", true) == true)
+      /* Disable by default in Termux due to it breaking on (some) HTC
+         devices (https://github.com/termux/termux-packages/issues/99). */
+	
+	 if (_config->FindB("Acquire::EnableSrvRecords", false) == true)
       {
          GetSrvRecords(Host, DefPort, SrvRecords);
 	 // RFC2782 defines that a lonely '.' target is an abort reason
