diff -u -r ../apt-0.9.16.1/test/interactive-helper/makefile ./test/interactive-helper/makefile
--- ../apt-0.9.16.1/test/interactive-helper/makefile	2014-01-05 20:06:21.000000000 +0100
+++ ./test/interactive-helper/makefile	2014-03-25 01:33:54.600672630 +0100
@@ -41,7 +41,7 @@
 
 # Program for testing udevcdrom
 PROGRAM=aptwebserver
-SLIBS = -lapt-pkg -lpthread
+SLIBS = -lapt-pkg
 LIB_MAKES = apt-pkg/makefile
 SOURCE = aptwebserver.cc
 include $(PROGRAM_H)
