--- a/modules/libpref/init/all.js
+++ b/modules/libpref/init/all.js
@@ -3632,7 +3632,7 @@
 pref("webextensions.storage.session.enforceQuota", true);
 
 // Should we use the old kinto-based implementation of storage.sync? To be removed in bug 1637465.
-pref("webextensions.storage.sync.kinto", false);
+pref("webextensions.storage.sync.kinto", true);
 // Server used by the old kinto-based implementation of storage.sync.
 pref("webextensions.storage.sync.serverURL", "https://webextensions.settings.services.mozilla.com/v1");
 
