--- a/gfx/wr/webrender/src/texture_cache.rs
+++ b/gfx/wr/webrender/src/texture_cache.rs
@@ -275,7 +275,7 @@ enum BudgetType {
 impl BudgetType {
     const COUNT: usize = 7;
 
-    const VALUES: [BudgetType; BudgetType::COUNT] = [
+    const VALUES: [BudgetType; 7] = [
         BudgetType::SharedColor8Linear,
         BudgetType::SharedColor8Nearest,
         BudgetType::SharedColor8Glyphs,
@@ -285,7 +285,7 @@ impl BudgetType {
         BudgetType::Standalone,
     ];
 
-    const PRESSURE_COUNTERS: [usize; BudgetType::COUNT] = [
+    const PRESSURE_COUNTERS: [usize; 7] = [
         profiler::ATLAS_COLOR8_LINEAR_PRESSURE,
         profiler::ATLAS_COLOR8_NEAREST_PRESSURE,
         profiler::ATLAS_COLOR8_GLYPHS_PRESSURE,
