--- a/configure.ac
+++ b/configure.ac
@@ -2157,7 +2157,7 @@
 if test "$enable_tcl" = yes; then
 	AC_MSG_CHECKING([for tclConfig.sh])
 	TCLCONFIG=no
-	TCLCONFIGDIRS="/usr/lib \
+	TCLCONFIGDIRS="@TERMUX_PREFIX@/lib /usr/lib \
 			/usr/lib64 \
 			/usr/lib/tcl8.7 \
 			/usr/lib/tcl8.6 \
@@ -2234,7 +2234,7 @@
 if test "$enable_tcl" = yes -a "$enable_tk" = yes; then
 	AC_MSG_CHECKING([for tkConfig.sh])
 	TKCONFIG=no
-	TKCONFIGDIRS="/usr/lib \
+	TKCONFIGDIRS="@TERMUX_PREFIX@/lib /usr/lib \
 			/usr/lib64 \
 			/usr/lib/tk8.7 \
 			/usr/lib/tk8.6 \
