--- a/src/libtsduck/base/system/tsSysUtils.h
+++ b/src/libtsduck/base/system/tsSysUtils.h
@@ -72,6 +72,8 @@ namespace ts {
     #elif defined(TS_WINDOWS)
         // Second parameter of ::DeviceIoControl().
         using ioctl_request_t = ::DWORD;
+    #elif defined(TS_ANDROID)
+        using ioctl_request_t = int;
     #else
         // Extract the type of the second parameter of ::ioctl().
         // It is "unsigned long" on most Linux systems but "int" on Alpine Linux.
