--- a/src/libtsduck/config/Makefile
+++ b/src/libtsduck/config/Makefile
@@ -42,7 +42,7 @@
 
 $(TABLES_DEST): $(TABLES_SRC) $(TABLES_SUBS) $(BINDIR)/tsxml
 	@echo '  [GEN] $(notdir $@)'; \
-	$(TSXML) --merge $(TABLES_SRC) $(TABLES_SUBS) --sort _tables --sort _descriptors --uncomment -o $@
+	tsxml --merge $(TABLES_SRC) $(TABLES_SUBS) --sort _tables --sort _descriptors --uncomment -o $@
 
 $(NAMES_DEST): $(NAMES_SRC)
 	@echo '  [GEN] $(notdir $@)'; \
