--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -134,8 +134,8 @@
   if(LIBSAMPLERATE_INSTALL_PKGCONFIG_MODULE)
     set(prefix ${CMAKE_INSTALL_PREFIX})
     set(exec_prefix "\${prefix}")
-    set(includedir "\${prefix}/${CMAKE_INSTALL_INCLUDEDIR}")
-    set(libdir "\${exec_prefix}/${CMAKE_INSTALL_LIBDIR}")
+    set(includedir "${CMAKE_INSTALL_INCLUDEDIR}")
+    set(libdir "${CMAKE_INSTALL_LIBDIR}")
     set(VERSION "${PROJECT_VERSION}")
     if(LIBM_REQUIRED)
       set(LIBS "-lm")
