--- ./mutt/file.c.orig	2021-12-15 17:56:42.465005012 +0530
+++ ./mutt/file.c	2021-12-15 18:16:25.832322927 +0530
@@ -354,6 +354,10 @@
  */
 int mutt_file_safe_rename(const char *src, const char *target)
 {
+/* Since Android 6.0, hardlinks are unsupported */
+#if defined(__ANDROID__)
+  return rename(src, target);
+#endif
   struct stat st_src = { 0 };
   struct stat st_target = { 0 };
   int link_errno;
