diff --git a/test/file/mutt_file_mkstemp_full.c b/test/file/mutt_file_mkstemp_full.c
index 49d863d66..b87cf2037 100644
--- a/test/file/mutt_file_mkstemp_full.c
+++ b/test/file/mutt_file_mkstemp_full.c
@@ -29,7 +29,7 @@ void test_mutt_file_mkstemp_full(void)
 {
   // FILE *mutt_file_mkstemp_full(const char *file, int line, const char *func);
 
-  C_Tmpdir = "/tmp";
+  C_Tmpdir = "@TERMUX_PREFIX@/tmp";
 
   {
     FILE *fp = NULL;
