diff --git a/mutt_config.c b/mutt_config.c
index 2ae78194a..e41cd8787 100644
--- a/mutt_config.c
+++ b/mutt_config.c
@@ -702,7 +702,7 @@ struct ConfigDef MainVars[] = {
   { "timeout", DT_NUMBER|DT_NOT_NEGATIVE, &C_Timeout, 600, 0, NULL,
     "Time to wait for user input in menus"
   },
-  { "tmpdir", DT_PATH|DT_PATH_DIR|DT_NOT_EMPTY, &C_Tmpdir, IP "/tmp", 0, NULL,
+  { "tmpdir", DT_PATH|DT_PATH_DIR|DT_NOT_EMPTY, &C_Tmpdir, IP "@TERMUX_PREFIX@/tmp", 0, NULL,
     "Directory for temporary files"
   },
   { "toggle_quoted_show_levels", DT_NUMBER|DT_NOT_NEGATIVE, &C_ToggleQuotedShowLevels, 0, 0, NULL,
