diff -uNr proxmark3-011f973e7ff145ee0f6ccf58c850765513c43d04/client/deps/hardnested/Makefile proxmark3-011f973e7ff145ee0f6ccf58c850765513c43d04.mod/client/deps/hardnested/Makefile
--- proxmark3-011f973e7ff145ee0f6ccf58c850765513c43d04/client/deps/hardnested/Makefile	2020-04-24 19:21:38.000000000 +0300
+++ proxmark3-011f973e7ff145ee0f6ccf58c850765513c43d04.mod/client/deps/hardnested/Makefile	2020-04-25 02:07:55.045517847 +0300
@@ -4,7 +4,7 @@
 MYDEFS =
 MYSRCS = hardnested_bruteforce.c
 
-cpu_arch = $(shell uname -m)
+cpu_arch = generic
 ifneq ($(findstring 86, $(cpu_arch)), )
     MULTIARCHSRCS = hardnested_bf_core.c hardnested_bitarray_core.c
 endif
@@ -24,7 +24,7 @@
             $(MULTIARCHSRCS:%.c=$(OBJDIR)/%_AVX.o) \
             $(MULTIARCHSRCS:%.c=$(OBJDIR)/%_AVX2.o)
 
-SUPPORTS_AVX512 :=  $(shell echo | gcc -E -mavx512f - > /dev/null 2>&1 && echo "True" )
+SUPPORTS_AVX512 := False
 
 HARD_SWITCH_NOSIMD = -mno-mmx -mno-sse2 -mno-avx -mno-avx2
 HARD_SWITCH_MMX = -mmmx -mno-sse2 -mno-avx -mno-avx2
diff -uNr proxmark3-011f973e7ff145ee0f6ccf58c850765513c43d04/client/deps/reveng/Makefile proxmark3-011f973e7ff145ee0f6ccf58c850765513c43d04.mod/client/deps/reveng/Makefile
--- proxmark3-011f973e7ff145ee0f6ccf58c850765513c43d04/client/deps/reveng/Makefile	2020-04-24 19:21:38.000000000 +0300
+++ proxmark3-011f973e7ff145ee0f6ccf58c850765513c43d04.mod/client/deps/reveng/Makefile	2020-04-25 02:08:19.501640877 +0300
@@ -30,6 +30,5 @@
 	$(info [-] CC $<)
 	$(Q)$(CC) $(CFLAGS) -DBMPTST -o $@ $<
 	$(info [=] TEST $@)
-	$(Q)( ./$@ && $(TOUCH) $@ ) || ( $(RM) $@ && $(FALSE) )
 
 .PHONY: all clean
diff -uNr proxmark3-011f973e7ff145ee0f6ccf58c850765513c43d04/client/Makefile proxmark3-011f973e7ff145ee0f6ccf58c850765513c43d04.mod/client/Makefile
--- proxmark3-011f973e7ff145ee0f6ccf58c850765513c43d04/client/Makefile	2020-04-24 19:21:38.000000000 +0300
+++ proxmark3-011f973e7ff145ee0f6ccf58c850765513c43d04.mod/client/Makefile	2020-04-25 02:09:05.293871331 +0300
@@ -22,7 +22,7 @@
 OBJDIR = obj
 
 LDLIBS ?= -L/usr/local/lib
-LDLIBS += -lreadline -lpthread -lm
+LDLIBS += -lreadline -lm
 
 # RPi Zero gcc requires -latomic
 # but MacOSX /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
@@ -54,7 +54,7 @@
 ZLIB = $(OBJDIR)/libz.a
 
 LIBS = -I$(LUALIBPATH) -I$(MBEDTLSLIBPATH) -I$(JANSSONLIBPATH) -I$(CBORLIBPATH) -I$(ZLIBPATH) -I$(REVENGPATH) -I$(AMIIBOLIBPATH) -I$(HARDNESTEDPATH) -I$(CLIPARSERPATH)
-INCLUDES_CLIENT = -I./src -I../include -I../common -I../common_fpga $(LIBS)
+INCLUDES_CLIENT = -I./src -I../include -I../common -I../common_fpga -I/data/data/com.termux/files/usr/include $(LIBS)
 CFLAGS ?= -Wall -Werror -O3
 # We cannot just use CFLAGS+=... because it has impact on sub-makes if CFLAGS is defined in env:
 PM3CFLAGS = $(CFLAGS) $(INCLUDES_CLIENT)
@@ -286,7 +286,7 @@
 proxmark3: LDLIBS+=$(LUALIB) $(JANSSONLIB) $(MBEDTLSLIB) $(CBORLIB) $(ZLIB) $(REVENGLIB) $(AMIIBOLIB) $(HARDNESTEDLIB) $(CLIPARSERLIB) $(QTLDLIBS)
 proxmark3: $(OBJDIR)/proxmark3.o $(COREOBJS) $(CMDOBJS) $(OBJCOBJS) $(QTGUIOBJS) $(LUALIB) $(JANSSONLIB) $(CBORLIB) $(REVENGLIB) $(MBEDTLSLIB) $(ZLIB) $(AMIIBOLIB) $(HARDNESTEDLIB) $(CLIPARSERLIB) lualibs/pm3_cmd.lua lualibs/mfc_default_keys.lua
 	$(info [=] LD $@)
-	$(Q)$(LD) $(LDFLAGS) $(OBJDIR)/proxmark3.o $(COREOBJS) $(CMDOBJS) $(OBJCOBJS) $(QTGUIOBJS) $(LDLIBS)  -o $@
+	$(Q)$(LD) $(LDFLAGS) $(OBJDIR)/proxmark3.o $(COREOBJS) $(CMDOBJS) $(OBJCOBJS) $(QTGUIOBJS) $(LDLIBS)  -o $@ deps/hardnested/libhardnested.a deps/cliparser/libcliparser.a deps/tinycbor/tinycbor.a deps/reveng/libreveng.a deps/jansson/libjansson.a deps/liblua/liblua.a obj/libmbedtls.a obj/libz.a
 
 src/proxgui.cpp: src/ui/ui_overlays.h
 
diff -uNr proxmark3-011f973e7ff145ee0f6ccf58c850765513c43d04/Makefile.defs proxmark3-011f973e7ff145ee0f6ccf58c850765513c43d04.mod/Makefile.defs
--- proxmark3-011f973e7ff145ee0f6ccf58c850765513c43d04/Makefile.defs	2020-04-24 19:21:38.000000000 +0300
+++ proxmark3-011f973e7ff145ee0f6ccf58c850765513c43d04.mod/Makefile.defs	2020-04-25 02:07:55.046517852 +0300
@@ -21,9 +21,6 @@
 TARFLAGS ?= -v --ignore-failed-read -r
 TARFLAGS += -C .. -f
 CROSS  ?= arm-none-eabi-
-CC = gcc
-CXX = g++
-LD = g++
 
 PATHSEP=/
 PREFIX ?=              /usr/local
