diff -uNr dnote-server-v2.0.1/scripts/cli/build.sh dnote-server-v2.0.1.mod/scripts/cli/build.sh
--- dnote-server-v2.0.1/scripts/cli/build.sh	2022-05-09 19:37:02.000000000 +0800
+++ dnote-server-v2.0.1.mod/scripts/cli/build.sh	2022-10-29 18:03:03.282780947 +0800
@@ -76,37 +76,9 @@
     if [ "$platform" == "windows" ]; then
       flags+=("-buildmode=exe")
     fi
-
-    xgo \
-      -go "$goVersion" \
-      -targets="$platform/$arch" \
-      -ldflags "$ldflags" \
-      -dest="$destDir" \
-      -out="cli" \
-      "${flags[@]}" \
-      -tags "$tags" \
-      -pkg pkg/cli \
-      .
   fi
-
-  popd
-
   binaryName=$(get_binary_name "$platform")
   mv "$destDir/cli-${platform}-"* "$destDir/$binaryName"
-
-  # build tarball
-  tarballName="dnote_${version}_${platform}_${arch}.tar.gz"
-  tarballPath="$outputDir/$tarballName"
-
-  cp "$projectDir/licenses/GPLv3.txt" "$destDir"
-  cp "$basedir/README.md" "$destDir"
-  tar -C "$destDir" -zcvf "$tarballPath" "."
-  rm -rf "$destDir"
-
-  # calculate checksum
-  pushd "$outputDir"
-  shasum -a 256 "$tarballName" >> "$outputDir/dnote_${version}_checksums.txt"
-  popd
 }
 
 if [ -z "$GOOS" ] && [ -z "$GOARCH" ]; then
