diff -uNr dnote-server-v2.0.1/scripts/server/build.sh dnote-server-v2.0.1.mod/scripts/server/build.sh
--- dnote-server-v2.0.1/scripts/server/build.sh	2022-05-09 19:37:02.000000000 +0800
+++ dnote-server-v2.0.1.mod/scripts/server/build.sh	2022-10-29 17:44:33.332781370 +0800
@@ -26,8 +26,8 @@
 fi
 
 build() {
-  platform=$1
-  arch=$2
+  platform=android
+  arch=$GOARCH
 
   pushd "$basedir"
 
@@ -46,20 +46,8 @@
 
   popd
 
-  # build tarball
-  tarballName="dnote_server_${version}_${platform}_${arch}.tar.gz"
-  tarballPath="$outputDir/$tarballName"
-
   cp "$projectDir/licenses/AGPLv3.txt" "$destDir"
   cp "$basedir/README.md" "$destDir"
-  tar -C "$destDir" -zcvf "$tarballPath" "."
-  rm -rf "$destDir"
-
-  # calculate checksum
-  pushd "$outputDir"
-  shasum -a 256 "$tarballName" >> "$outputDir/dnote_${version}_checksums.txt"
-  popd
-
 }
 
 build linux amd64
