--- ./src/sat/smt/euf_solver.h	2021-07-26 09:42:33.123349945 +0530
+++ ./src/sat/smt/euf_solver.h.mod	2021-07-26 09:42:55.080018799 +0530
@@ -101,7 +101,7 @@
         sat::sat_internalizer* m_to_si;
         scoped_ptr<euf::ackerman>    m_ackerman;
         scoped_ptr<sat::dual_solver> m_dual_solver;
-        user::solver*          m_user_propagator{ nullptr };
+        z3_user::solver*          m_user_propagator{ nullptr };
         th_solver*             m_qsolver { nullptr };
         unsigned               m_generation { 0 };
         mutable ptr_vector<expr> m_todo;
