diff -u -r ../node-v22.2.0/common.gypi ./common.gypi
--- ../node-v22.2.0/common.gypi	2024-05-15 12:34:59.000000000 +0000
+++ ./common.gypi	2024-05-30 20:14:10.892850620 +0000
@@ -239,10 +239,6 @@
               },],
             ],
           },],
-          ['OS == "android"', {
-            'cflags': [ '-fPIC', '-I<(android_ndk_path)/sources/android/cpufeatures' ],
-            'ldflags': [ '-fPIC' ]
-          }],
           ['clang==1', {
             'msbuild_toolset': 'ClangCL',
           }],
