--- ./common.gypi.orig	2024-02-16 19:07:34.899138955 +0530
+++ ./common.gypi	2024-02-16 19:08:47.540634360 +0530
@@ -237,10 +237,6 @@
               },],
             ],
           },],
-          ['OS == "android"', {
-            'cflags': [ '-fPIC', '-I<(android_ndk_path)/sources/android/cpufeatures' ],
-            'ldflags': [ '-fPIC' ]
-          }],
         ],
         'msvs_settings': {
           'VCCLCompilerTool': {
