diff -uNr ./deps/uv/src/unix/core.c ./deps/uv/src/unix/core.c.mod
--- ./deps/uv/src/unix/core.c	2021-06-03 07:15:30.000000000 +0530
+++ ./deps/uv/src/unix/core.c.mod	2021-06-18 20:10:13.705642672 +0530
@@ -1116,7 +1116,7 @@
 
   /* No temp environment variables defined */
   #if defined(__ANDROID__)
-    buf = "/data/local/tmp";
+    buf = "@TERMUX_PREFIX@/tmp";
   #else
     buf = "/tmp";
   #endif
