--- ./deps/v8/BUILD.gn	2021-06-03 07:15:30.000000000 +0530
+++ ./deps/v8/BUILD.gn.mod	2021-07-17 17:32:54.134091233 +0530
@@ -17,6 +17,8 @@
 import("gni/snapshot_toolchain.gni")
 import("gni/v8.gni")
 
+v8_target_cpu = getenv("DEST_CPU")
+
 # Specifies if the target build is a simulator build. Comparing target cpu
 # with v8 target cpu to not affect simulator builds for making cross-compile
 # snapshots.
