--- ./deps/npm/lib/npm.js.orig	2024-02-21 20:21:11.251716444 +0530
+++ ./deps/npm/lib/npm.js	2024-02-21 20:25:09.207937428 +0530
@@ -113,12 +113,6 @@
 
     const timeEnd = this.time(`command:${cmd}`)
 
-    // this is async but we dont await it, since its ok if it doesnt
-    // finish before the command finishes running. it uses command and argv
-    // so it must be initiated here, after the command name is set
-    // eslint-disable-next-line promise/catch-or-return
-    updateNotifier(this).then((msg) => (this.updateNotification = msg))
-
     // Options are prefixed by a hyphen-minus (-, \u2d).
     // Other dash-type chars look similar but are invalid.
     if (!this.#warnedNonDashArg) {
