--- ./deps/v8/src/logging/log.cc	2021-06-03 07:15:31.000000000 +0530
+++ ./deps/v8/src/logging/log.cc.mod	2021-06-18 20:27:34.605642275 +0530
@@ -291,7 +291,7 @@
   FILE* perf_output_handle_;
 };
 
-const char PerfBasicLogger::kFilenameFormatString[] = "/tmp/perf-%d.map";
+const char PerfBasicLogger::kFilenameFormatString[] = "@TERMUX_PREFIX@/tmp/perf-%d.map";
 // Extra space for the PID in the filename
 const int PerfBasicLogger::kFilenameBufferPadding = 16;
 
