--- ../node-v6.0.0/deps/v8/src/flag-definitions.h	2016-04-26 15:50:11.000000000 -0400
+++ ./deps/v8/src/flag-definitions.h	2016-05-02 16:46:31.289030161 -0400
@@ -172,7 +172,10 @@
 //
 #define FLAG FLAG_FULL
 
-DEFINE_BOOL(warn_template_set, true,
+/* Termux modification: Do not enable this warning per default
+   as it will only confuse users needlessly. The root causes
+   of this warning will get fixed anyway soon in npm packages. */
+DEFINE_BOOL(warn_template_set, false,
             "warn on deprecated v8::Template::Set() use")
 
 DEFINE_BOOL(experimental_extras, false,
