Disable update check for npm
--- ./deps/npm/lib/cli/entry.js.orig	2025-11-13 23:01:06.006769023 +0530
+++ ./deps/npm/lib/cli/entry.js	2025-11-13 23:01:09.000312729 +0530
@@ -57,13 +57,6 @@
 
     const execPromise = npm.exec(command, args)
 
-    // this is async but we don't await it, since its ok if it doesnt
-    // finish before the command finishes running. it uses command and argv
-    // so it must be initiated here, after the command name is set
-    const updateNotifier = require('./update-notifier.js')
-    // eslint-disable-next-line promise/catch-or-return
-    updateNotifier(npm).then((msg) => (npm.updateNotification = msg))
-
     await execPromise
     return exitHandler.exit()
   } catch (err) {
