diff -u -r ../node-v10.9.0/deps/v8/gypfiles/v8.gyp ./deps/v8/gypfiles/v8.gyp
--- ../node-v10.9.0/deps/v8/gypfiles/v8.gyp	2018-08-15 13:53:24.000000000 +0000
+++ ./deps/v8/gypfiles/v8.gyp	2018-08-23 21:43:30.588264328 +0000
@@ -2053,8 +2053,7 @@
                   # library order and break (see crbug.com/469973).
                   # These libraries do not exist on Mac hosted builds.
                   'libraries': [
-                    '-ldl',
-                    '-lrt'
+                    '-ldl'
                   ]
                 }]
               ]
