--- ./deps/uv/uv.gyp.orig	2022-11-14 07:56:05.000000000 +0530
+++ ./deps/uv/uv.gyp	2022-11-25 18:51:25.022527139 +0530
@@ -40,7 +40,7 @@
     {
       'target_name': 'libuv',
       'type': '<(uv_library)',
-      'include_dirs': [
+      'include_dirs+': [
         'include',
         'src/',
       ],
@@ -55,7 +55,7 @@
           '<@(shared_unix_defines)',
           '<@(shared_zos_defines)',
         ],
-        'include_dirs': [ 'include' ],
+        'include_dirs+': [ 'include' ],
         'conditions': [
           ['OS == "linux"', {
             'defines': [ '_POSIX_C_SOURCE=200112' ],
@@ -267,6 +267,7 @@
             'src/unix/random-sysctl-linux.c',
             'src/unix/epoll.c',
           ],
+          'defines': [ '__USE_GNU' ],
           'link_settings': {
             'libraries': [ '-ldl' ],
           },
