diff -uNr node-v14.0.0/deps/v8/BUILD.gn node-v14.0.0.mod/deps/v8/BUILD.gn
--- node-v14.0.0/deps/v8/BUILD.gn	2020-04-21 14:38:50.000000000 +0300
+++ node-v14.0.0.mod/deps/v8/BUILD.gn	2020-05-06 13:58:58.867691869 +0300
@@ -28,7 +28,7 @@
 
 declare_args() {
   # Print to stdout on Android.
-  v8_android_log_stdout = false
+  v8_android_log_stdout = true
 
   # Dynamically set an additional dependency from v8/custom_deps.
   v8_custom_deps = ""
