diff -u -r ../node-v8.1.4/deps/uv/uv.gyp ./deps/uv/uv.gyp
--- ../node-v8.1.4/deps/uv/uv.gyp	2017-07-11 05:48:01.000000000 +0200
+++ ./deps/uv/uv.gyp	2017-07-19 01:25:10.988614742 +0200
@@ -45,12 +45,12 @@
     {
       'target_name': 'libuv',
       'type': '<(uv_library)',
-      'include_dirs': [
+      'include_dirs+': [
         'include',
         'src/',
       ],
       'direct_dependent_settings': {
-        'include_dirs': [ 'include' ],
+        'include_dirs+': [ 'include' ],
         'conditions': [
           ['OS != "win"', {
             'defines': [
