--- ./deps/v8/src/logging/log.cc.orig	2022-11-23 18:22:22.362329609 +0530
+++ ./deps/v8/src/logging/log.cc	2022-11-23 18:26:15.492329520 +0530
@@ -357,7 +357,7 @@
   static uint64_t reference_count_;
 };
 
-const char LinuxPerfBasicLogger::kFilenameFormatString[] = "/tmp/perf-%d.map";
+const char LinuxPerfBasicLogger::kFilenameFormatString[] = "@TERMUX_PREFIX@/perf-%d.map";
 // Extra space for the PID in the filename
 const int LinuxPerfBasicLogger::kFilenameBufferPadding = 16;
 
