--- ../linux-syscalls.c.orig	2019-04-01 12:59:57.770124746 +0000
+++ ./deps/uv/src/unix/linux-syscalls.c	2019-04-01 13:00:38.703457159 +0000
@@ -187,7 +187,7 @@
 # endif
 #endif /* __NR_pwritev */
 
-#ifndef __NR_statx
+#if !defined(__NR_statx) && !defined(__ANDROID__)
 # if defined(__x86_64__)
 #  define __NR_statx 332
 # elif defined(__i386__)
@@ -358,7 +358,7 @@
               int flags,
               unsigned int mask,
               struct uv__statx* statxbuf) {
-#if defined(__NR_statx)
+#if defined(__NR_statx) && !defined(__ANDROID__)
   return syscall(__NR_statx, dirfd, path, flags, mask, statxbuf);
 #else
   return errno = ENOSYS, -1;
