--- ./lib/internal/test_runner/test.js.orig	2022-05-21 11:46:57.293199020 +0530
+++ ./lib/internal/test_runner/test.js	2022-05-21 13:15:42.882384210 +0530
@@ -19,7 +19,7 @@
 const { createDeferredPromise } = require('internal/util');
 const { isPromise } = require('internal/util/types');
 const { isUint32 } = require('internal/validators');
-const { cpus } = require('os');
+const { numcpus } = require('os');
 const { bigint: hrtime } = process.hrtime;
 const kCallbackAndPromisePresent = 'callbackAndPromisePresent';
 const kCancelledByParent = 'cancelledByParent';
@@ -32,7 +32,7 @@
 const isTestRunner = getOptionValue('--test');
 const testOnlyFlag = !isTestRunner && getOptionValue('--test-only');
 // TODO(cjihrig): Use uv_available_parallelism() once it lands.
-const rootConcurrency = isTestRunner ? cpus().length : 1;
+const rootConcurrency = isTestRunner ? numcpus() : 1;
 
 class TestContext {
   #test;
