See https://github.com/nodejs/node-v0.x-archive/issues/8540

diff -u -r ../node-v4.0.0/lib/net.js ./lib/net.js
--- ../node-v4.0.0/lib/net.js	2015-09-08 11:30:45.000000000 -0400
+++ ./lib/net.js	2015-09-08 17:37:04.293684663 -0400
@@ -951,7 +951,7 @@
     // systems. See
     // http://lists.freebsd.org/pipermail/freebsd-bugs/2008-February/028260.html
     // for more information on the lack of support for FreeBSD.
-    if (process.platform !== 'freebsd')
+    if (process.platform !== 'freebsd' && process.platform !== 'android')
       dnsopts.hints |= dns.V4MAPPED;
   }
 
