diff -u -r ../node-v6.3.0/node.gyp ./node.gyp
--- ../node-v6.3.0/node.gyp	2016-07-06 13:14:10.000000000 -0400
+++ ./node.gyp	2016-07-15 09:30:10.150224361 -0400
@@ -805,54 +805,7 @@
           ],
         } ],
       ]
-    },
-    {
-      'target_name': 'cctest',
-      'type': 'executable',
-      'dependencies': [ 'deps/gtest/gtest.gyp:gtest' ],
-      'include_dirs': [
-        'src',
-        'deps/v8/include'
-      ],
-      'defines': [
-        # gtest's ASSERT macros conflict with our own.
-        'GTEST_DONT_DEFINE_ASSERT_EQ=1',
-        'GTEST_DONT_DEFINE_ASSERT_GE=1',
-        'GTEST_DONT_DEFINE_ASSERT_GT=1',
-        'GTEST_DONT_DEFINE_ASSERT_LE=1',
-        'GTEST_DONT_DEFINE_ASSERT_LT=1',
-        'GTEST_DONT_DEFINE_ASSERT_NE=1',
-        'NODE_WANT_INTERNALS=1',
-      ],
-      'sources': [
-        'test/cctest/util.cc',
-      ],
-
-      'conditions': [
-        ['v8_inspector=="true"', {
-          'dependencies': [
-            'deps/openssl/openssl.gyp:openssl',
-            'deps/http_parser/http_parser.gyp:http_parser',
-            'deps/uv/uv.gyp:libuv'
-          ],
-          'sources': [
-            'src/inspector_socket.cc',
-            'test/cctest/test_inspector_socket.cc'
-          ]
-        }],
-        [ 'node_use_v8_platform=="true"', {
-          'dependencies': [
-            'deps/v8/tools/gyp/v8.gyp:v8_libplatform',
-          ],
-        }],
-        [ 'node_use_bundled_v8=="true"', {
-          'dependencies': [
-            'deps/v8/tools/gyp/v8.gyp:v8',
-            'deps/v8/tools/gyp/v8.gyp:v8_libplatform'
-          ],
-        }],
-      ]
     }
   ], # end targets
 
   'conditions': [
