Backport of https://github.com/nodejs/node/pull/6820

diff -u -r ../node-v6.2.0/src/node_config.cc ./src/node_config.cc
--- ../node-v6.2.0/src/node_config.cc	2016-05-17 15:53:07.000000000 -0400
+++ ./src/node_config.cc	2016-05-18 16:45:34.588991777 -0400
@@ -29,8 +29,8 @@
 void InitConfig(Local<Object> target,
                 Local<Value> unused,
                 Local<Context> context) {
-#ifdef NODE_HAVE_I18N_SUPPORT
   Environment* env = Environment::GetCurrent(context);
+#ifdef NODE_HAVE_I18N_SUPPORT
 
   READONLY_BOOLEAN_PROPERTY("hasIntl");
 
