--- ./test/parallel/test-blob-buffer-too-large.js.orig	2024-02-16 18:12:39.119954543 +0530
+++ ./test/parallel/test-blob-buffer-too-large.js	2024-02-16 18:47:24.110742373 +0530
@@ -4,10 +4,14 @@
 const common = require('../common');
 const assert = require('assert');
 const { Blob, kMaxLength } = require('buffer');
+const { platform } = require('os');
 
 if (common.isFreeBSD)
   common.skip('Oversized buffer make the FreeBSD CI runner crash');
 
+if(platform() === 'android')
+  common.skip('Android will kill heavy memory using processes sometimes crashing Termux');
+
 try {
   new Blob([new Uint8Array(kMaxLength), [1]]);
 } catch (e) {
