--- ./test/parallel/test-child-process-spawnsync-shell.js.orig	2022-06-05 08:34:30.587601172 +0530
+++ ./test/parallel/test-child-process-spawnsync-shell.js	2022-06-05 08:34:48.497601165 +0530
@@ -82,7 +82,7 @@
   test('darwin', '/bin/csh', '/bin/csh');
 
   // Test Android platforms.
-  test('android', true, '/system/bin/sh');
+  test('android', true, '@TERMUX_PREFIX@/bin/sh');
 
   // Test Windows platforms with a user specified shell.
   test('win32', 'powershell.exe', 'powershell.exe');
