--- ./test/sequential/test-gc-http-client.js.orig	2022-06-05 10:17:44.127598809 +0530
+++ ./test/sequential/test-gc-http-client.js	2022-06-05 10:17:53.947598805 +0530
@@ -5,7 +5,7 @@
 const common = require('../common');
 const onGC = require('../common/ongc');
 
-const cpus = require('os').cpus().length;
+const cpus = require('os').numcpus();
 
 function serverHandler(req, res) {
   res.writeHead(200, { 'Content-Type': 'text/plain' });
