diff -u -r ../node-v12.9.0/tools/v8_gypfiles/v8.gyp ./tools/v8_gypfiles/v8.gyp
--- ../node-v12.9.0/tools/v8_gypfiles/v8.gyp	2019-08-20 17:14:53.000000000 +0000
+++ ./tools/v8_gypfiles/v8.gyp	2019-08-24 22:02:57.943558855 +0000
@@ -1010,6 +1010,7 @@
             '<(V8_ROOT)/src/base/platform/platform-posix.h',
             '<(V8_ROOT)/src/base/platform/platform-posix-time.cc',
             '<(V8_ROOT)/src/base/platform/platform-posix-time.h',
+            '<(V8_ROOT)/src/base/platform/platform-linux.cc',
           ],
           'conditions': [
             ['_toolset=="host"', {
