diff -u -r ../mu-1.0.orig/lib/mu-util.c ./lib/mu-util.c
--- ../mu-1.0.orig/lib/mu-util.c	2018-02-03 21:02:45.000000000 +0900
+++ ./lib/mu-util.c	2018-02-07 13:19:46.836536134 +0900
@@ -44,8 +44,36 @@
 #include <glib/gstdio.h>
 #include <errno.h>
 
-#include <langinfo.h>
+#include <libandroid-support/langinfo.h>
+#include <termios.h>
+#include <readline/readline.h>
 
+#ifdef __ANDROID__
+static char* getpass(const char *prompt) {
+  struct termios term_old, term_new;
+  int nread;
+
+  /* Turn echoing off and fail if we can't. */
+  if (tcgetattr (0, &term_old) != 0) {
+    return NULL;
+  }
+  
+  term_new = term_old;
+  term_new.c_lflag &= ~ECHO;
+  
+  if (tcsetattr (0, TCSAFLUSH, &term_new) != 0) {
+    return NULL;
+  }
+
+  /* Read the password. */
+  char *password = readline(prompt);
+
+  /* Restore terminal. */
+  (void) tcsetattr (0, TCSAFLUSH, &term_old);
+
+  return password;
+}
+#endif
 
 static char*
 do_wordexp (const char *path)
diff -u -r ../mu-1.0.orig/lib/tests/test-mu-common.c ./lib/tests/test-mu-common.c
--- ../mu-1.0.orig/lib/tests/test-mu-common.c	2018-02-03 21:02:45.000000000 +0900
+++ ./lib/tests/test-mu-common.c	2018-02-07 13:21:03.726536104 +0900
@@ -28,7 +28,7 @@
 #include <unistd.h>
 #include <string.h>
 
-#include <langinfo.h>
+#include <libandroid-support/langinfo.h>
 #include <locale.h>
 
 #include "test-mu-common.h"
diff -u -r ../mu-1.0.orig/lib/tests/test-mu-util.c ./lib/tests/test-mu-util.c
--- ../mu-1.0.orig/lib/tests/test-mu-util.c	2018-02-03 21:02:45.000000000 +0900
+++ ./lib/tests/test-mu-util.c	2018-02-07 13:31:38.396535862 +0900
@@ -82,13 +82,13 @@
 	const char *expected;
 
 	/* skip the test if there's no /tmp */
-	if (access ("/tmp", F_OK))
+	if (access ("/data/data/com.termux/files/usr/tmp", F_OK))
 		return;
 
-	g_setenv ("MAILDIR", "/tmp", TRUE);
+	g_setenv ("MAILDIR", "/data/data/com.termux/files/usr/tmp", TRUE);
 
 	got = mu_util_guess_maildir ();
-	expected = "/tmp";
+	expected = "/data/data/com.termux/files/usr/tmp";
 
 	g_assert_cmpstr (got,==,expected);
 	g_free (got);
@@ -119,11 +119,11 @@
 static void
 test_mu_util_check_dir_01 (void)
 {
-	if (g_access ("/usr/bin", F_OK) == 0) {
+	if (g_access ("/data/data/com.termux/files/usr/bin", F_OK) == 0) {
 		g_assert_cmpuint (
-			mu_util_check_dir ("/usr/bin", TRUE, FALSE) == TRUE,
+			mu_util_check_dir ("/data/data/com.termux/files/usr/bin", TRUE, FALSE) == TRUE,
 			==,
-			g_access ("/usr/bin", R_OK) == 0);
+			g_access ("/data/data/com.termux/files/usr/bin", R_OK) == 0);
 	}
 }
 
@@ -131,11 +131,11 @@
 static void
 test_mu_util_check_dir_02 (void)
 {
-	if (g_access ("/tmp", F_OK) == 0) {
+	if (g_access ("/data/data/com.termux/files/usr/tmp", F_OK) == 0) {
 		g_assert_cmpuint (
-			mu_util_check_dir ("/tmp", FALSE, TRUE) == TRUE,
+			mu_util_check_dir ("/data/data/com.termux/files/usr/tmp", FALSE, TRUE) == TRUE,
 			==,
-			g_access ("/tmp", W_OK) == 0);
+			g_access ("/data/data/com.termux/files/usr/tmp", W_OK) == 0);
 	}
 }
 
