diff -uNr rsync-3.1.3/rsync.h rsync-3.1.3.mod/rsync.h
--- rsync-3.1.3/rsync.h	2018-01-15 20:58:31.000000000 +0200
+++ rsync-3.1.3.mod/rsync.h	2019-03-01 22:40:29.926940976 +0200
@@ -29,7 +29,7 @@
 /* RSYNCD_SYSCONF is now set in config.h */
 #define RSYNCD_USERCONF "rsyncd.conf"
 
-#define DEFAULT_LOCK_FILE "/var/run/rsyncd.lock"
+#define DEFAULT_LOCK_FILE "@TERMUX_PREFIX@/var/run/rsyncd.lock"
 #define URL_PREFIX "rsync://"
 
 #define SYMLINK_PREFIX "/rsyncd-munged/"  /* This MUST have a trailing slash! */
