diff --git a/apt-file b/apt-file
index 57c76b489..0cbd041d5 100644
--- a/apt-file
+++ b/apt-file
@@ -199,7 +199,7 @@ sub open_data_pipeline {
         my @cat_cmd = (
             'xargs',
             '-0r',
-            '/usr/lib/apt/apt-helper',
+            '@TERMUX_PREFIX@/lib/apt/apt-helper',
             @apt_options,
             'cat-file'
         );
@@ -924,7 +924,7 @@ eval {
     exit(0);
 };
 if (my $err = $@) {
-    stop_with_message($err, 255);
+    stop_with_msg($err, 255);
 }
 
 __END__
