--- a/plugin-fancymenu/lxqtfancymenuconfiguration.cpp
+++ b/plugin-fancymenu/lxqtfancymenuconfiguration.cpp
@@ -191,7 +191,7 @@
 {
     QFileDialog *d = new QFileDialog(this,
                                      tr("Choose menu file"),
-                                     QLatin1String("/etc/xdg/menus"),
+                                     QLatin1String("@TERMUX_PREFIX@/etc/xdg/menus"),
                                      tr("Menu files (*.menu)"));
     d->setWindowModality(Qt::WindowModal);
     d->setAttribute(Qt::WA_DeleteOnClose);
