--- fcitx5-configtool-5.0.17/layout/keyboardlayoutwidget.cpp.orig	2023-03-11 13:01:54
+++ fcitx5-configtool-5.0.17/layout/keyboardlayoutwidget.cpp	2023-04-22 02:28:28
@@ -37,7 +37,7 @@
 constexpr unsigned int INVALID_KEYCODE = (unsigned int)(-1);
 
 #ifndef XKB_RULES_XML_FILE
-#define XKB_RULES_XML_FILE "/usr/share/X11/xkb/rules/evdev.xml"
+#define XKB_RULES_XML_FILE "@TERMUX_PREFIX@/share/X11/xkb/rules/evdev.xml"
 #endif
 
 namespace fcitx {
