diff -uNr global-6.6.1.orig/globash/globash.in global-6.6.1/globash/globash.in
--- global-6.6.1.orig/globash/globash.in	2017-12-16 07:19:15.000000000 +0200
+++ global-6.6.1/globash/globash.in	2017-12-19 18:37:51.034558917 +0200
@@ -36,7 +36,7 @@
 # Use /bin/echo, because in some systems, built-in echo of /bin/sh
 # does not accept -n option.
 #
-ECHO='/bin/echo'
+ECHO='@TERMUX_PREFIX@/bin/echo'
 #
 # global command name
 #
