--- a/v8/src/regexp/experimental/experimental-compiler.cc
+++ b/v8/src/regexp/experimental/experimental-compiler.cc
@@ -1218,7 +1218,7 @@
 
 ZoneList<Instruction> ExperimentalCompiler::Compile(Tree* tree, Flags flags,
                                                     Zone* zone) {
-  return CompileVisitor::Compile(tree, flags, zone);
+  return experimental::CompileVisitor::Compile(tree, flags, zone);
 }
 
 }  // namespace regexp
