--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -113,7 +113,7 @@
     DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/coturn"
         COMPONENT Development)
 
-install(DIRECTORY man DESTINATION .
+install(DIRECTORY man DESTINATION ./share
     COMPONENT Runtime)
 install(DIRECTORY turndb/
     DESTINATION share/turnserver
@@ -119,7 +119,7 @@
     DESTINATION share/turnserver
         COMPONENT Runtime)
 install(DIRECTORY turndb/
-    DESTINATION doc/turnserver
+    DESTINATION share/doc/turnserver
         COMPONENT Runtime)
 install(FILES
             LICENSE
@@ -128,7 +128,7 @@
             README.turnutils
             INSTALL
             postinstall.txt
-        DESTINATION doc/turnserver
+        DESTINATION share/doc/turnserver
             COMPONENT Runtime)
 install(FILES examples/etc/turnserver.conf
     DESTINATION ${CMAKE_INSTALL_SYSCONFDIR}
