diff --git a/runtime/autoload/tar.vim b/runtime/autoload/tar.vim
index 9c4f16d7f..837c8cd1c 100644
--- a/runtime/autoload/tar.vim
+++ b/runtime/autoload/tar.vim
@@ -416,9 +416,6 @@ fun! tar#Write(fname)
   let curdir= getcwd()
   let tmpdir= tempname()
 "  call Decho("orig tempname<".tmpdir.">")
-  if tmpdir =~ '\.'
-   let tmpdir= substitute(tmpdir,'\.[^.]*$','','e')
-  endif
 "  call Decho("tmpdir<".tmpdir.">")
   call mkdir(tmpdir,"p")
 
diff --git a/runtime/autoload/zip.vim b/runtime/autoload/zip.vim
index 4a53fc5f2..e2bd35064 100644
--- a/runtime/autoload/zip.vim
+++ b/runtime/autoload/zip.vim
@@ -249,9 +249,6 @@ fun! zip#Write(fname)
 
   let curdir= getcwd()
   let tmpdir= tempname()
-  if tmpdir =~ '\.'
-   let tmpdir= substitute(tmpdir,'\.[^.]*$','','e')
-  endif
   call mkdir(tmpdir,"p")
 
   " attempt to change to the indicated directory
