diff -uNr vim-8.1.0050/src/if_cscope.c vim-8.1.0050.mod/src/if_cscope.c
--- vim-8.1.0050/src/if_cscope.c	2018-06-12 21:25:52.000000000 +0300
+++ vim-8.1.0050.mod/src/if_cscope.c	2018-06-19 20:32:38.661663690 +0300
@@ -962,7 +962,7 @@
 	    PERROR(_("cs_create_connection setpgid failed"));
 #  endif
 # endif
-	if (execl("/bin/sh", "sh", "-c", cmd, (char *)NULL) == -1)
+	if (execl("@TERMUX_PREFIX@/bin/sh", "sh", "-c", cmd, (char *)NULL) == -1)
 	    PERROR(_("cs_create_connection exec failed"));
 
 	exit(127);
