--- a/src/ssdb/ssdb_impl.cpp
+++ b/src/ssdb/ssdb_impl.cpp
@@ -44,7 +44,7 @@
 	ssdb->options.block_cache = leveldb::NewLRUCache(opt.cache_size * 1048576);
 	ssdb->options.block_size = opt.block_size * 1024;
 	ssdb->options.write_buffer_size = opt.write_buffer_size * 1024 * 1024;
-	ssdb->options.compaction_speed = opt.compaction_speed;
+	//ssdb->options.compaction_speed = opt.compaction_speed;
 	if(opt.compression == "yes"){
 		ssdb->options.compression = leveldb::kSnappyCompression;
 	}else{
