--- a/src/ssdb-server.cpp
+++ b/src/ssdb-server.cpp
@@ -55,7 +55,7 @@
 	log_info("block_size       : %d KB", option.block_size);
 	log_info("write_buffer     : %d MB", option.write_buffer_size);
 	log_info("max_open_files   : %d", option.max_open_files);
-	log_info("compaction_speed : %d MB/s", option.compaction_speed);
+	//log_info("compaction_speed : %d MB/s", option.compaction_speed);
 	log_info("compression      : %s", option.compression.c_str());
 	log_info("binlog           : %s", option.binlog? "yes" : "no");
 	log_info("binlog_capacity  : %d", option.binlog_capacity);
