--- ../ssdb.conf.orig	2022-09-29 08:54:57.859034312 +0200
+++ ./ssdb.conf	2022-09-29 08:55:34.403114520 +0200
@@ -2,8 +2,8 @@
 # MUST indent by TAB!
 
 # absolute path, or relative to path of this file, directory must exists
-work_dir = ./var
-pidfile = ./var/ssdb.pid
+work_dir = @TERMUX_PREFIX@/var
+pidfile = @TERMUX_PREFIX@/var/run/ssdb.pid
 
 server:
 	# specify an ipv6 address to enable ipv6 support
