--- ./go/pkg/mod/github.com/containerd/containerd@v1.7.16/platforms/cpuinfo_other.go.orig
+++ ./go/pkg/mod/github.com/containerd/containerd@v1.7.16/platforms/cpuinfo_other.go
@@ -29,7 +29,7 @@
 
 	var variant string
 
-	if runtime.GOOS == "windows" || runtime.GOOS == "darwin" {
+	if runtime.GOOS == "windows" || runtime.GOOS == "darwin" || runtime.GOOS == "android" {
 		// Windows/Darwin only supports v7 for ARM32 and v8 for ARM64 and so we can use
 		// runtime.GOARCH to determine the variants
 		switch runtime.GOARCH {
