--- blender-3.4.1/build_files/cmake/Modules/FindEmbree.cmake	2023-02-21 14:42:54.000000000 +0530
+++ blender-3.4.1.mod/build_files/cmake/Modules/FindEmbree.cmake	2023-03-11 19:51:37.406029673 +0530
@@ -18,12 +18,12 @@
 
 SET(_embree_SEARCH_DIRS
   ${EMBREE_ROOT_DIR}
-  /opt/lib/embree
+  @TERMUX_PREFIX@/lib/
 )
 
 FIND_PATH(EMBREE_INCLUDE_DIR
   NAMES
-    embree3/rtcore.h
+    embree4/rtcore.h
   HINTS
     ${_embree_SEARCH_DIRS}
   PATH_SUFFIXES
@@ -39,7 +39,7 @@
 ENDIF()
 
 SET(_embree_FIND_COMPONENTS
-  embree3
+  embree4
   ${_embree_SIMD_COMPONENTS}
   lexers
   math
@@ -61,10 +61,10 @@
       lib64 lib
     )
   IF(NOT EMBREE_${UPPERCOMPONENT}_LIBRARY)
-    IF(EMBREE_EMBREE3_LIBRARY)
+    IF(EMBREE_EMBREE4_LIBRARY)
       # If we can't find all the static libraries, try to fall back to the shared library if found.
       # This allows building with a shared embree library
-      SET(_embree_LIBRARIES ${EMBREE_EMBREE3_LIBRARY})
+      SET(_embree_LIBRARIES ${EMBREE_EMBREE4_LIBRARY})
       BREAK()
     ENDIF()
   ENDIF()
