diff -u -r ../abook-0.6.0pre2/database.h ./database.h
--- ../abook-0.6.0pre2/database.h	2006-09-04 21:24:18.000000000 +0200
+++ ./database.h	2017-01-15 21:47:00.441932233 +0100
@@ -60,7 +60,7 @@
 /*
  * Field operations
  */
-inline int field_id(int i);
+int field_id(int i);
 abook_field *find_standard_field(char *key, int do_declare);
 abook_field *real_find_field(char *key, abook_field_list *list, int *nb);
 #define find_field(key, list)		real_find_field(key, list, NULL)
