https://github.com/termux/termux-packages/issues/15852

--- a/cnf/configure_func.sh
+++ b/cnf/configure_func.sh
@@ -33,7 +33,7 @@
 checkfunc d_alarm 'alarm' "0" 'unistd.h'
 checkfunc d_asctime64 'asctime64'
 checkfunc d_atolf 'atolf'
-checkfunc d_atoll 'atoll'
+checkfunc d_atoll 'atoll' "NULL" 'stdlib.h'
 checkfunc d_backtrace 'backtrace' 'NULL, 0' 'execinfo.h'
 checkfunc d_bcmp 'bcmp' "NULL,NULL,0" 'string.h'
 checkfunc d_bcopy 'bcopy' "NULL,NULL,0" 'string.h'
--- a/cnf/configure_func_dbl.sh
+++ b/cnf/configure_func_dbl.sh
@@ -40,7 +40,7 @@
 checkfunc d_ldexpl 'ldexpl' '0.0, 0'
 checkfunc d_lgamma 'lgamma' '0.0'
 checkfunc d_lgamma_r 'lgamma_r' '0.0, NULL'
-checkfunc d_llabs 'llabs' '0LL'
+checkfunc d_llabs 'llabs' '0LL' 'stdlib.h'
 checkfunc d_llrint 'llrint' '0.0'
 checkfunc d_llrintl 'llrintl' '0.0'
 checkfunc d_llround 'llround' '0.0'
