--- a/cpan/CPAN/lib/CPAN/FirstTime.pm
+++ b/cpan/CPAN/lib/CPAN/FirstTime.pm
@@ -904,7 +904,6 @@ sub init {
     unless ( $matcher
         || _can_write_to_libdirs() || _using_installbase() || _using_sudo()
     ) {
-        local $auto_config = 0; # We *must* ask, even under autoconfig
         local *_real_prompt;    # We *must* show prompt
         my_prompt_loop(install_help => 'local::lib', $matcher,
                    'local::lib|sudo|manual');
