diff -u -r ../perl-5.22.1/cpan/ExtUtils-MakeMaker/lib/ExtUtils/Liblist/Kid.pm ./cpan/ExtUtils-MakeMaker/lib/ExtUtils/Liblist/Kid.pm
--- ../perl-5.22.1/cpan/ExtUtils-MakeMaker/lib/ExtUtils/Liblist/Kid.pm	2015-10-17 08:38:37.000000000 -0400
+++ ./cpan/ExtUtils-MakeMaker/lib/ExtUtils/Liblist/Kid.pm	2016-04-22 18:38:59.367520265 -0400
@@ -19,9 +19,9 @@
 use File::Spec;
 
 sub ext {
-    if    ( $^O eq 'VMS' )     { return &_vms_ext; }
-    elsif ( $^O eq 'MSWin32' ) { return &_win32_ext; }
-    else                       { return &_unix_os2_ext; }
+    # Termux modification to work with android linker.
+    my $res = $Config{perllibs} . " -L" . $ENV{'TERMUX_PKG_SRCDIR'} . " -lperl";
+    return ( $res, '', $res, '' );
 }
 
 sub _unix_os2_ext {
