--- a/ext/text/gettext.scm
+++ b/ext/text/gettext.scm
@@ -88,7 +88,7 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;; Customize this to the appropriate value for your system:
 
-(define message-path '("/usr/share/locale"))
+(define message-path '("@TERMUX_PREFIX@/share/locale"))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;; store meta info for gettext files
