--- a/sources/polyphone.pro
+++ b/sources/polyphone.pro
@@ -89,9 +89,9 @@
         lib/_option_rtaudio/rtaudio/include/iasiothiscallresolver.cpp
 }
 unix:!macx {
-    DEFINES += __LINUX_ALSASEQ__ __UNIX_JACK__ __LINUX_ALSA__ #__LINUX_PULSE__
+    DEFINES += __LINUX_PULSE__
     CONFIG += link_pkgconfig
-    PKGCONFIG += alsa jack zlib ogg vorbis vorbisfile vorbisenc sndfile
+    PKGCONFIG += libpulse libpulse-simple zlib ogg vorbis vorbisfile vorbisenc sndfile
     contains(DEFINES, USE_WOLFSSL) {
         PKGCONFIG += wolfssl
     } else {
