--- a/core/thread.cpp
+++ b/core/thread.cpp
@@ -16,6 +16,20 @@
 #include "thread.h"
 #include <fstream>
 
+#if defined(__BIONIC__)
+static inline int pthread_setaffinity_np(pthread_t thread, size_t cpusetsize,
+                                         cpu_set_t *cpuset) {
+  assert(pthread_equal(pthread_self(), thread));
+  return sched_setaffinity(0, cpusetsize, cpuset);
+}
+
+static inline int pthread_getaffinity_np(pthread_t thread, size_t cpusetsize,
+                                         cpu_set_t *cpuset) {
+  assert(pthread_equal(pthread_self(), thread));
+  return sched_getaffinity(0, cpusetsize, cpuset);
+}
+#endif
+
 OIDN_NAMESPACE_BEGIN
 
 #if defined(_WIN32)
