--- a/mkl-dnn/src/common/primitive_attr.cpp
+++ b/mkl-dnn/src/common/primitive_attr.cpp
@@ -21,6 +21,17 @@
 #include "type_helpers.hpp"
 #include "utils.hpp"
 
+// /home/builder/.termux-build/oidn/src/mkl-dnn/src/common/primitive_attr.cpp:302:54: error: reference to 'user' is ambiguous
+//     const bool ok = one_of(scratchpad_mode, library, user);
+//                                                      ^
+// /home/builder/.termux-build/_cache/android-r25-api-24-v1/sysroot/usr/include/sys/user.h:153:8: note: candidate found by name lookup is 'user'
+// struct user {
+//        ^
+// /home/builder/.termux-build/oidn/src/mkl-dnn/src/common/c_types_map.hpp:158:25: note: candidate found by name lookup is 'dnnl::impl::scratchpad_mode::user'
+// const scratchpad_mode_t user = dnnl_scratchpad_mode_user;
+#undef user
+#define user dnnl::impl::scratchpad_mode::user
+
 using namespace dnnl::impl;
 using namespace dnnl::impl::status;
 using namespace dnnl::impl::utils;
