diff -uNr tinyxml2-master/Makefile tinyxml2-master.mod/Makefile
--- tinyxml2-master/Makefile	2022-03-05 21:31:46.312207486 +0800
+++ tinyxml2-master.mod/Makefile	2022-03-05 22:07:28.092206669 +0800
@@ -16,7 +16,7 @@
 INSTALL_PROGRAM = $(INSTALL)
 INSTALL_DATA = $(INSTALL) -m 644
 
-prefix = /usr/local
+prefix = "@TERMUX_PREFIX@/local"
 bindir = $(prefix)/bin
 libdir = $(prefix)/lib
 includedir = $(prefix)/include
