diff -uNr borgbackup-1.1.11/src/borg/repository.py borgbackup-1.1.11.mod/src/borg/repository.py
--- borgbackup-1.1.11/src/borg/repository.py	2020-03-08 01:32:06.000000000 +0200
+++ borgbackup-1.1.11.mod/src/borg/repository.py	2020-03-08 23:13:43.555530400 +0200
@@ -294,20 +294,6 @@
             logger.warning("Old config file not securely erased on previous config update")
             secure_erase(old_config_path)
 
-        if os.path.isfile(config_path):
-            link_error_msg = ("Failed to securely erase old repository config file (hardlinks not supported>). "
-                              "Old repokey data, if any, might persist on physical storage.")
-            try:
-                os.link(config_path, old_config_path)
-            except OSError as e:
-                if e.errno in (errno.EMLINK, errno.ENOSYS, errno.EPERM, errno.EACCES, errno.ENOTSUP):
-                    logger.warning(link_error_msg)
-                else:
-                    raise
-            except AttributeError:
-                # some python ports have no os.link, see #4901
-                logger.warning(link_error_msg)
-
         with SaveFile(config_path) as fd:
             config.write(fd)
 
