diff -uNr borgbackup-1.1.10/src/borg/xattr.py borgbackup-1.1.10.mod/src/borg/xattr.py
--- borgbackup-1.1.10/src/borg/xattr.py	2019-05-16 05:58:15.000000000 +0300
+++ borgbackup-1.1.10.mod/src/borg/xattr.py	2019-12-12 00:27:59.831938631 +0200
@@ -68,7 +68,7 @@
     # tools used by find_library, like ldconfig, gcc/cc, objdump.
     # so we can only try some "usual" names for the C library:
     if sys.platform.startswith('linux'):
-        libc_name = 'libc.so.6'
+        libc_name = 'libc.so'
     elif sys.platform.startswith(('freebsd', 'netbsd')):
         libc_name = 'libc.so'
     elif sys.platform == 'darwin':
