--- a/tools/scan6.c
+++ b/tools/scan6.c
@@ -1604,7 +1604,7 @@
 
 
 	if(!configfile_f){
-		strncpy(configfile, "/etc/ipv6toolkit.conf", MAX_FILENAME_SIZE);
+		strncpy(configfile, "@TERMUX_PREFIX@/etc/ipv6toolkit.conf", MAX_FILENAME_SIZE);
 	}
 
 	if(tgt_vendor_f || portscan_f){
@@ -5854,13 +5854,13 @@
 	fclose(fp);
 
 	if(!fname_f)
-		strncpy(fname, "/usr/local/share/ipv6toolkit/oui.txt", MAX_FILENAME_SIZE-1);
+		strncpy(fname, "@TERMUX_PREFIX@/share/ipv6toolkit/oui.txt", MAX_FILENAME_SIZE-1);
 
 	if(!portsfname_f)
-		strncpy(portsfname, "/usr/local/share/ipv6toolkit/service-names-port-numbers.csv", MAX_FILENAME_SIZE-1);
+		strncpy(portsfname, "@TERMUX_PREFIX@/share/ipv6toolkit/service-names-port-numbers.csv", MAX_FILENAME_SIZE-1);
 
 	if(!topportsfname_f)
-		strncpy(topportsfname, "/usr/local/share/ipv6toolkit/top-port-numbers.csv", MAX_FILENAME_SIZE-1);
+		strncpy(topportsfname, "@TERMUX_PREFIX@/share/ipv6toolkit/top-port-numbers.csv", MAX_FILENAME_SIZE-1);
 
 	return(1);
 }
