--- libglade-2.0.1/glade/glade-gtk.c.nowarning	2003-08-29 14:50:10.000000000 -0400
+++ libglade-2.0.1/glade/glade-gtk.c	2003-08-29 14:58:41.000000000 -0400
@@ -639,9 +639,8 @@
 		if (!strcmp (childinfo->properties[j].name, "label")) {
 		    label = childinfo->properties[j].value;
 		    break;
-		} else {
-		    g_warning ("Unknown CList child property: %s", childinfo->properties[j].name);
 		}
+		/* Ignore all other properties */
 	    }
 
 	    if (label) {
@@ -683,6 +682,7 @@
 	    char *icon = NULL;
 	    gboolean use_stock = FALSE, active = FALSE, new_group = FALSE;
 	    gboolean use_underline = FALSE;
+	    gboolean sensitive = TRUE;
 	    GtkWidget *iconw = NULL;
 	    int j;
 
@@ -708,6 +708,8 @@
 		    group_name = value;
 		} else if (!strcmp (name, "new_group")) {
 		    new_group = BOOL (value);
+		} else if (!strcmp (name, "sensitive")) {
+		    sensitive = BOOL (value);
 		} else if (!strcmp (name, "visible")) {
 		    /* ignore for now */
 		} else if (!strcmp (name, "tooltip")) {
@@ -785,6 +787,8 @@
 		gtk_label_set_use_underline (GTK_LABEL (toolbar_child->label),
 					     TRUE);
 	    }
+	    
+	    gtk_widget_set_sensitive (child, sensitive);
 
 	    glade_xml_set_common_params (xml, child, childinfo->child);
 	} else {
