--- ./cli/threadexecutor.cpp.orig	2021-10-26 21:30:46.557469590 +0530
+++ ./cli/threadexecutor.cpp	2021-10-26 21:32:35.957469548 +0530
@@ -149,7 +149,7 @@
 
 bool ThreadExecutor::checkLoadAverage(size_t nchildren)
 {
-#if defined(__CYGWIN__) || defined(__QNX__) || defined(__HAIKU__)  // getloadavg() is unsupported on Cygwin, Qnx, Haiku.
+#if defined(__CYGWIN__) || defined(__QNX__) || defined(__HAIKU__) || defined(__ANDROID__) // getloadavg() is unsupported on Cygwin, Qnx, Haiku, Android.
     return true;
 #else
     if (!nchildren || !mSettings.loadAverage) {
