diff --git a/third_party/blink/renderer/platform/geometry/layout_unit.cc b/third_party/blink/renderer/platform/geometry/layout_unit.cc
index 9d3b36ef25..ffbf29ecb8 100644
--- a/third_party/blink/renderer/platform/geometry/layout_unit.cc
+++ b/third_party/blink/renderer/platform/geometry/layout_unit.cc
@@ -57,3 +57,5 @@ INSTANTIATE(16, int32_t);
 INSTANTIATE(16, int64_t);
 
 }  // namespace blink
+
+#undef INSTANTIATE
diff --git a/third_party/blink/renderer/platform/geometry/physical_offset.cc b/third_party/blink/renderer/platform/geometry/physical_offset.cc
index f3101af87e..2e8455a011 100644
--- a/third_party/blink/renderer/platform/geometry/physical_offset.cc
+++ b/third_party/blink/renderer/platform/geometry/physical_offset.cc
@@ -29,3 +29,5 @@ std::ostream& operator<<(std::ostream& os,
 INSTANTIATE(LayoutUnit);
 
 }  // namespace blink
+
+#undef INSTANTIATE
