diff --git a/ui/accessibility/platform/inspect/ax_inspect_utils_auralinux.cc b/ui/accessibility/platform/inspect/ax_inspect_utils_auralinux.cc
index 00c2c5dc9e..08ac87ba7c 100644
--- a/ui/accessibility/platform/inspect/ax_inspect_utils_auralinux.cc
+++ b/ui/accessibility/platform/inspect/ax_inspect_utils_auralinux.cc
@@ -47,11 +47,13 @@ const char* GetNameForPlatformConstant(
 #define QUOTE(X) \
   { X, #X }
 
+#ifndef ATSPI_CHECK_VERSION
 #define ATSPI_CHECK_VERSION(major, minor, micro)                           \
   (major) < ATSPI_MAJOR_VERSION ||                                         \
       ((major) == ATSPI_MAJOR_VERSION && (minor) < ATSPI_MINOR_VERSION) || \
       ((major) == ATSPI_MAJOR_VERSION && (minor) == ATSPI_MINOR_VERSION && \
        (micro) <= ATSPI_MICRO_VERSION)
+#endif
 
 #define CHECK_ATSPI_ERROR_NULLPTR(error) \
   if (error) {                           \
