--- a/content/browser/renderer_host/render_frame_host_manager.cc
+++ b/content/browser/renderer_host/render_frame_host_manager.cc
@@ -1340,7 +1340,7 @@
   // state on RFH, because RFH exists for a long time after navigation, and
   // because it can own multiple pending-commit NavigationRequests at the same
   // time, which might clobber each other.
-  if (features::ShouldAckCOREarlyForViewTransition() &&
+  if (::features::ShouldAckCOREarlyForViewTransition() &&
       !old_render_frame_host->GetParentOrOuterDocument() &&
       render_frame_host_ &&
       render_frame_host_->did_last_navigation_have_view_transition()) {
