diff -uNr upx-4.0.1-src/src/util/snprintf.h upx-4.0.1-src.mod/src/util/snprintf.h
--- upx-4.0.1-src/src/util/snprintf.h	2022-11-17 03:45:38.000000000 +0530
+++ upx-4.0.1-src.mod/src/util/snprintf.h	2022-11-17 10:52:28.160885021 +0530
@@ -54,7 +54,6 @@
 #undef sprintf
 #undef vsnprintf
 #define snprintf  upx_safe_snprintf
-#define sprintf   ERROR_sprintf_IS_DANGEROUS_USE_snprintf
 #define vsnprintf upx_safe_vsnprintf
 
 /*************************************************************************
