--- a/src/sys/statfs.rs
+++ b/src/sys/statfs.rs
@@ -94,6 +94,8 @@
 pub const CGROUP_SUPER_MAGIC: FsType = FsType(libc::CGROUP_SUPER_MAGIC as fs_type_t);
 #[cfg(all(target_os = "linux", not(target_env = "musl")))]
 pub const CGROUP2_SUPER_MAGIC: FsType = FsType(libc::CGROUP2_SUPER_MAGIC as fs_type_t);
+#[cfg(target_os = "android")]
+pub const CGROUP2_SUPER_MAGIC: FsType = FsType(0x63677270 as fs_type_t);
 
 
 impl Statfs {
