diff -u -r ../dialog-1.2-20150528/configure ./configure
--- ../dialog-1.2-20150528/configure	2015-05-28 20:57:16.000000000 -0400
+++ ./configure	2015-06-06 18:35:01.620952945 -0400
@@ -12803,7 +12803,7 @@
 fi
 
 cf_ncuconfig_root=$cf_cv_screen
-cf_have_ncuconfig=no
+cf_have_ncuconfig=yes
 
 if test "x$PKG_CONFIG" != xnone; then
 	echo "$as_me:12809: checking pkg-config for $cf_ncuconfig_root" >&5
@@ -12814,7 +12814,7 @@
 
 		echo "$as_me:12815: checking if the $cf_ncuconfig_root package files work" >&5
 echo $ECHO_N "checking if the $cf_ncuconfig_root package files work... $ECHO_C" >&6
-		cf_have_ncuconfig=unknown
+		cf_have_ncuconfig=yes
 
 		cf_save_CPPFLAGS="$CPPFLAGS"
 		cf_save_LIBS="$LIBS"
@@ -12862,7 +12862,7 @@
   echo "$as_me:12862: \$? = $ac_status" >&5
   (exit $ac_status); }; }; then
   if test "$cross_compiling" = yes; then
-  cf_have_ncuconfig=unknown
+  cf_have_ncuconfig=yes
 else
   cat >conftest.$ac_ext <<_ACEOF
 #line 12868 "configure"
@@ -12887,14 +12887,14 @@
   echo "$as_me: program exited with status $ac_status" >&5
 echo "$as_me: failed program was:" >&5
 cat conftest.$ac_ext >&5
-cf_have_ncuconfig=no
+cf_have_ncuconfig=yes
 fi
 rm -f core core.* *.core conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 else
   echo "$as_me: failed program was:" >&5
 cat conftest.$ac_ext >&5
-cf_have_ncuconfig=no
+cf_have_ncuconfig=yes
 fi
 rm -f conftest.$ac_objext conftest$ac_exeext conftest.$ac_ext
 
