--- a/libvbisam/Makefile.am
+++ b/libvbisam/Makefile.am
@@ -1,8 +1,8 @@
 lib_LTLIBRARIES = libvbisam.la
 libvbisam_la_SOURCES = isdecimal.c ishelper.c isaudit.c isbuild.c isdelete.c \
 	isopen.c isread.c isrecover.c isrewrite.c istrans.c iswrite.c ischeck.c \
-	vbdataio.c vbindexio.c vbkeysio.c vblocking.c \
-	vblowlevel.c vbmemio.c vbnodememio.c isinternal.h byteswap.h
+	vbdataio.c vbindexio.c vbkeysio.c vblocking.c efgcvt_r-template.c \
+	vblowlevel.c vbmemio.c vbnodememio.c isinternal.h byteswap.h efgcvt-dbl-macros.h
 
 libvbisam_la_CFLAGS = -I$(top_srcdir)
 libvbisam_la_LDFLAGS = -no-undefined -version-info 1:0:0
