From 3e35ee0ee9a2701198e4439373421173b0116ac0 Mon Sep 17 00:00:00 2001
From: alexytomi <60690056+alexytomi@users.noreply.github.com>
Date: Thu, 1 Jan 2026 23:53:31 +0800
Subject: [PATCH] [plist-d] Add filename for termux libplist-2.0.so

---
 source/plist/c.d | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/plist/c.d b/source/plist/c.d
index ca31040..1182a35 100644
--- a/source/plist/c.d
+++ b/source/plist/c.d
@@ -35,7 +35,7 @@ version (Windows) {
 } else version (OSX) {
     enum libplist = LibImport("libplist-2.0.3.dylib", "libplist-2.0.4.dylib");
 } else {
-    enum libplist = LibImport("libplist-2.0.so.3", "libplist.so.3", "libplist-2.0.so.4", "libplist.so.4");
+    enum libplist = LibImport("libplist-2.0.so", "libplist-2.0.so.3", "libplist.so.3", "libplist-2.0.so.4", "libplist.so.4");
 }
 
 mixin makeBindings;
-- 
2.51.2

