diff -uNr qtbase/src/corelib/thread/qthread_unix.cpp qtbase.mod/src/corelib/thread/qthread_unix.cpp
--- qtbase/src/corelib/thread/qthread_unix.cpp	2018-09-13 07:25:10.000000000 +0300
+++ qtbase.mod/src/corelib/thread/qthread_unix.cpp	2018-09-21 16:25:01.058662816 +0300
@@ -92,7 +92,7 @@
 # define SCHED_IDLE    5
 #endif
 
-#if defined(Q_OS_DARWIN) || !defined(Q_OS_ANDROID) && !defined(Q_OS_OPENBSD) && defined(_POSIX_THREAD_PRIORITY_SCHEDULING) && (_POSIX_THREAD_PRIORITY_SCHEDULING-0 >= 0)
+#if !defined(__ANDROID__) && (defined(Q_OS_DARWIN) || !defined(Q_OS_OPENBSD) && defined(_POSIX_THREAD_PRIORITY_SCHEDULING) && (_POSIX_THREAD_PRIORITY_SCHEDULING-0 >= 0))
 #define QT_HAS_THREAD_PRIORITY_SCHEDULING
 #endif
 
@@ -312,7 +312,7 @@
 
 void *QThreadPrivate::start(void *arg)
 {
-#if !defined(Q_OS_ANDROID)
+#if !defined(__ANDROID__)
     pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, NULL);
 #endif
     pthread_cleanup_push(QThreadPrivate::finish, arg);
@@ -360,7 +360,7 @@
 #endif
 
         emit thr->started(QThread::QPrivateSignal());
-#if !defined(Q_OS_ANDROID)
+#if !defined(__ANDROID__)
         pthread_setcancelstate(PTHREAD_CANCEL_ENABLE, NULL);
         pthread_testcancel();
 #endif
@@ -747,7 +747,7 @@
 
 void QThread::terminate()
 {
-#if !defined(Q_OS_ANDROID)
+#if !defined(__ANDROID__)
     Q_D(QThread);
     QMutexLocker locker(&d->mutex);
 
@@ -789,7 +789,7 @@
                "Current thread was not started with QThread.");
 
     Q_UNUSED(thr)
-#if defined(Q_OS_ANDROID)
+#if defined(__ANDROID__)
     Q_UNUSED(enabled);
 #else
     pthread_setcancelstate(enabled ? PTHREAD_CANCEL_ENABLE : PTHREAD_CANCEL_DISABLE, NULL);
