--- src/src/corelib/global/qsystemdetection.h.orig	2021-05-06 14:36:25.054047886 +0000
+++ src.mod/src/corelib/global/qsystemdetection.h	2021-05-06 14:43:45.917806447 +0000
@@ -109,7 +109,7 @@
 #    error "Qt has not been ported to this Apple platform - see http://www.qt.io/developers"
 #  endif
 #elif defined(__ANDROID__) || defined(ANDROID)
-#  define Q_OS_ANDROID
+#  define Q_OS_ANDROID_EMBEDDED
 #  define Q_OS_LINUX
 #elif defined(__CYGWIN__)
 #  define Q_OS_CYGWIN
