diff -uNr qtbase/mkspecs/termux-cross/qplatformdefs.h qtbase.mod/mkspecs/termux-cross/qplatformdefs.h
--- qtbase/mkspecs/termux-cross/qplatformdefs.h	1970-01-01 03:00:00.000000000 +0300
+++ qtbase.mod/mkspecs/termux-cross/qplatformdefs.h	2018-09-21 16:25:01.055329471 +0300
@@ -0,0 +1,4 @@
+#include "../android-clang/qplatformdefs.h"
+#define fseeko64 fseeko
+#define ftello64 ftello
+#define fopen64 fopen
