--- src/src/corelib/io/qtemporaryfile_p.h	2021-05-09 10:40:38.972403873 +0100
+++ src.mod/src/corelib/io/qtemporaryfile_p.h	2021-05-09 10:40:32.520423082 +0100
@@ -58,7 +58,7 @@
 #include "private/qfile_p.h"
 #include "qtemporaryfile.h"
 
-#if defined(Q_OS_LINUX) && QT_CONFIG(linkat)
+#if defined(Q_OS_LINUX) && !defined(__ANDROID__) && QT_CONFIG(linkat)
 #  include <fcntl.h>
 #  ifdef O_TMPFILE
 // some early libc support had the wrong values for O_TMPFILE
