diff --git a/Makefile.in b/Makefile.in
index 6986591..d8f1de0 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -491,7 +491,7 @@ top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 include_HEADERS = libcob.h
-SUBDIRS = . lib libcob bin cobc config copy po extras doc tests
+SUBDIRS = . lib libcob bin cobc config copy po extras 
 BUILT_SOURCES = tarstamp.h
 ACLOCAL_AMFLAGS = -I m4 --install
 # CLEANFILES = $(bin_SCRIPTS)
