--- src/config.tests/gstreamer/gstreamer.pro	2020-10-15 06:53:20.000000000 +0000
+++ src.mod/config.tests/gstreamer/gstreamer.pro	2021-05-16 13:59:14.419703082 +0000
@@ -1,2 +1,5 @@
 SOURCES += main.cpp
 
+INCLUDEPATH += "@TERMUX_PREFIX@/include/gstreamer-1.0/"
+INCLUDEPATH += "@TERMUX_PREFIX@/include/glib-2.0/"
+INCLUDEPATH += "@TERMUX_PREFIX@/lib/glib-2.0/include"
--- src/config.tests/gstreamer_photography/gstreamer_photography.pro	2020-10-15 06:53:20.000000000 +0000
+++ src.mod/config.tests/gstreamer_photography/gstreamer_photography.pro	2021-05-16 13:59:57.407079643 +0000
@@ -1 +1,4 @@
 SOURCES += main.cpp
+INCLUDEPATH += "@TERMUX_PREFIX@/include/gstreamer-1.0/"
+INCLUDEPATH += "@TERMUX_PREFIX@/include/glib-2.0/"
+INCLUDEPATH += "@TERMUX_PREFIX@/lib/glib-2.0/include"
--- src/config.tests/gstreamer_appsrc/gstreamer_appsrc.pro	2020-10-15 06:53:20.000000000 +0000
+++ src.mod/config.tests/gstreamer_appsrc/gstreamer_appsrc.pro	2021-05-16 14:03:35.948676966 +0000
@@ -1 +1,4 @@
 SOURCES += main.cpp
+INCLUDEPATH += "@TERMUX_PREFIX@/include/gstreamer-1.0/"
+INCLUDEPATH += "@TERMUX_PREFIX@/include/glib-2.0/"
+INCLUDEPATH += "@TERMUX_PREFIX@/lib/glib-2.0/include"
--- src/config.tests/gstreamer_encodingprofiles/gstreamer_encodingprofiles.pro	2020-10-15 06:53:20.000000000 +0000
+++ src.mod/config.tests/gstreamer_encodingprofiles/gstreamer_encodingprofiles.pro	2021-05-16 14:02:47.817110561 +0000
@@ -1 +1,4 @@
 SOURCES += main.cpp
+INCLUDEPATH += "@TERMUX_PREFIX@/include/gstreamer-1.0/"
+INCLUDEPATH += "@TERMUX_PREFIX@/include/glib-2.0/"
+INCLUDEPATH += "@TERMUX_PREFIX@/lib/glib-2.0/include"
