diff -uNr binaryen/CMakeLists.txt binaryen.mod/CMakeLists.txt
--- binaryen/CMakeLists.txt	2022-02-26 23:46:13.824326901 +0800
+++ binaryen.mod/CMakeLists.txt	2022-02-27 00:20:59.532106415 +0800
@@ -317,7 +317,10 @@
 add_subdirectory(test/lit)
 
 # Configure GTest unit tests
-add_subdirectory(test/gtest)
+option(ENABLE_GTEST "Enable GTest unit tests" ON)
+if(ENABLE_GTEST)
+  add_subdirectory(test/gtest)
+endif()
 
 # Object files
 set(binaryen_objs
