diff -uNr binaryen/third_party/CMakeLists.txt binaryen.mod/third_party/CMakeLists.txt
--- binaryen/third_party/CMakeLists.txt	2022-02-26 23:46:14.268336047 +0800
+++ binaryen.mod/third_party/CMakeLists.txt	2022-02-27 00:30:42.657030709 +0800
@@ -2,15 +2,14 @@
   add_subdirectory(llvm-project)
 endif()
 
-include_directories(
-  googletest/googletest
-  googletest/googletest/include
-)
+if(ENABLE_GTEST)
+  include_directories(
+    googletest/googletest
+    googletest/googletest/include)
 
-add_library(gtest STATIC
-  googletest/googletest/src/gtest-all.cc
-)
+  add_library(gtest STATIC
+    googletest/googletest/src/gtest-all.cc)
 
-add_library(gtest_main STATIC
-  googletest/googletest/src/gtest_main.cc
-)
+  add_library(gtest_main STATIC
+    googletest/googletest/src/gtest_main.cc)
+endif()
