diff -uNr llvm-project/llvm/lib/ExecutionEngine/Orc/MemoryMapper.cpp llvm-project.mod/llvm/lib/ExecutionEngine/Orc/MemoryMapper.cpp
--- llvm-project/llvm/lib/ExecutionEngine/Orc/MemoryMapper.cpp	2022-07-22 22:56:15.000000000 +0800
+++ llvm-project.mod/llvm/lib/ExecutionEngine/Orc/MemoryMapper.cpp	2022-07-23 16:39:13.276357002 +0800
@@ -186,7 +186,7 @@
 
 void SharedMemoryMapper::reserve(size_t NumBytes,
                                  OnReservedFunction OnReserved) {
-#if defined(LLVM_ON_UNIX) || defined(_WIN32)
+#if (defined(LLVM_ON_UNIX) || defined(_WIN32)) && !defined(__ANDROID__)
 
   EPC.callSPSWrapperAsync<
       rt::SPSExecutorSharedMemoryMapperServiceReserveSignature>(
