diff -uNr llvm-project/clang/lib/Driver/ToolChain.cpp llvm-project.mod/clang/lib/Driver/ToolChain.cpp
--- llvm-project/clang/lib/Driver/ToolChain.cpp	2021-11-04 18:24:28.000000000 +0800
+++ llvm-project.mod/clang/lib/Driver/ToolChain.cpp	2021-11-14 21:15:21.092854100 +0800
@@ -406,7 +406,7 @@
   case llvm::Triple::AIX:
     return "aix";
   default:
-    return getOS();
+    return Triple.isAndroid() ? "android":getOS();
   }
 }
 
@@ -944,7 +944,7 @@
 
   switch (Type) {
   case ToolChain::CST_Libcxx:
-    CmdArgs.push_back("-lc++");
+    CmdArgs.push_back("-lc++_shared");
     break;
 
   case ToolChain::CST_Libstdcxx:
