diff -uNr llvm-project/clang/lib/Driver/ToolChain.cpp llvm-project.mod/clang/lib/Driver/ToolChain.cpp
--- llvm-project/clang/lib/Driver/ToolChain.cpp	2022-06-20 05:49:00.000000000 +0800
+++ llvm-project.mod/clang/lib/Driver/ToolChain.cpp	2022-06-28 23:33:05.425005209 +0800
@@ -1012,7 +1012,7 @@
 
   switch (Type) {
   case ToolChain::CST_Libcxx:
-    CmdArgs.push_back("-lc++");
+    CmdArgs.push_back("-lc++_shared");
     break;
 
   case ToolChain::CST_Libstdcxx:
