diff -uNr llvm-project/clang/lib/Driver/ToolChain.cpp llvm-project.mod/clang/lib/Driver/ToolChain.cpp
--- llvm-project/clang/lib/Driver/ToolChain.cpp	2022-05-27 00:44:58.000000000 +0800
+++ llvm-project.mod/clang/lib/Driver/ToolChain.cpp	2022-06-06 00:15:23.523958529 +0800
@@ -1004,7 +1004,7 @@
 
   switch (Type) {
   case ToolChain::CST_Libcxx:
-    CmdArgs.push_back("-lc++");
+    CmdArgs.push_back("-lc++_shared");
     break;
 
   case ToolChain::CST_Libstdcxx:
