diff -uNr llvm-project/compiler-rt/lib/builtins/CMakeLists.txt llvm-project.mod/compiler-rt/lib/builtins/CMakeLists.txt
--- llvm-project/compiler-rt/lib/builtins/CMakeLists.txt	2022-02-24 01:38:33.000000000 +0800
+++ llvm-project.mod/compiler-rt/lib/builtins/CMakeLists.txt	2022-02-26 20:33:34.373468611 +0800
@@ -738,6 +738,8 @@
       # Release mode already sets it via -O2/3, Debug mode doesn't.
       if (${arch} STREQUAL "armhf")
         list(APPEND BUILTIN_CFLAGS_${arch} -fomit-frame-pointer -DCOMPILER_RT_ARMHF_TARGET)
+      elseif(${arch} STREQUAL "arm" AND ANDROID)
+        list(APPEND BUILTIN_CFLAGS_${arch} -fomit-frame-pointer)
       endif()
 
       # For RISCV32, we must force enable int128 for compiling long
