diff -uNr llvm-project/clang/lib/Driver/ToolChain.cpp llvm-project.mod/clang/lib/Driver/ToolChain.cpp
--- llvm-project/clang/lib/Driver/ToolChain.cpp	2022-05-03 21:10:32.000000000 +0800
+++ llvm-project.mod/clang/lib/Driver/ToolChain.cpp	2022-05-08 23:47:42.482356300 +0800
@@ -967,7 +967,7 @@
 
   switch (Type) {
   case ToolChain::CST_Libcxx:
-    CmdArgs.push_back("-lc++");
+    CmdArgs.push_back("-lc++_shared");
     break;
 
   case ToolChain::CST_Libstdcxx:
