--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -280,17 +280,16 @@ option(EXTENSION_STATIC_BUILD
 set(JEMALLOC_DEFAULT_BUILD FALSE)
 if(NOT CLANG_TIDY AND OS_NAME STREQUAL "linux")
   # build jemalloc by default for linux
-  set(JEMALLOC_DEFAULT_BUILD TRUE)
 endif()
 
-option(BUILD_ICU_EXTENSION "Build the ICU extension." FALSE)
-option(BUILD_PARQUET_EXTENSION "Build the Parquet extension." FALSE)
+option(BUILD_ICU_EXTENSION "Build the ICU extension." TRUE)
+option(BUILD_PARQUET_EXTENSION "Build the Parquet extension." TRUE)
 option(BUILD_TPCH_EXTENSION "Build the TPC-H extension." FALSE)
 option(BUILD_TPCDS_EXTENSION "Build the TPC-DS extension." FALSE)
 option(BUILD_FTS_EXTENSION "Build the FTS extension." FALSE)
-option(BUILD_HTTPFS_EXTENSION "Build the HTTP File System extension." FALSE)
+option(BUILD_HTTPFS_EXTENSION "Build the HTTP File System extension." TRUE)
 option(BUILD_VISUALIZER_EXTENSION "Build the profiler-output visualizer extension." FALSE)
-option(BUILD_JSON_EXTENSION "Build the JSON extension." FALSE)
+option(BUILD_JSON_EXTENSION "Build the JSON extension." TRUE)
 option(BUILD_JEMALLOC_EXTENSION "Build the JEMalloc extension." ${JEMALLOC_DEFAULT_BUILD})
 option(BUILD_EXCEL_EXTENSION "Build the excel extension." FALSE)
 option(BUILD_INET_EXTENSION "Build the inet extension." FALSE)
