--- qrsspig-v0.8.0/src/QGPluginLoader.h	2022-08-04 01:22:10.000000000 +0530
+++ qrsspig-v0.8.0.mod/src/QGPluginLoader.h	2025-01-28 02:19:15.574174901 +0530
@@ -109,7 +109,7 @@
 	struct path_leaf_string
     {
 	    std::string operator()(const boost::filesystem::directory_entry& entry) const {
-		    return entry.path().leaf().string();
+		    return entry.path().filename().string();
 	    }
     };
 
