--- a/src/modules/espeak.c	2022-04-15 04:49:54.448394598 +0000
+++ b/src/modules/espeak.c	2022-04-15 14:59:04.648322417 +0000
@@ -35,7 +35,7 @@
 #include <ctype.h>
 #include <glib.h>
 #include <fcntl.h>
-
+#include <sys/stat.h>
 /* espeak header file */
 #ifdef ESPEAK_NG_INCLUDE
 #include <espeak-ng/espeak_ng.h>
@@ -890,7 +890,7 @@
 
 			if (!strncmp(identifier, "mb/mb-", 6)) {
 				voicename = g_strdup(identifier + 6);
-				dash = index(voicename, '-');
+				dash = strchr(voicename, '-');
 				if (dash)
 					/* Ignore "-en" language specification */
 					*dash = 0;
