--- ./src/common/i18n.c.orig	2021-04-05 19:46:45.166906583 +0000
+++ ./src/common/i18n.c	2021-04-05 19:53:41.936476655 +0000
@@ -29,18 +29,5 @@
 
 void i18n_init(void)
 {
-	if (setlocale(LC_ALL, "") == NULL) {
-		perror("setlocale");
-		exit(1);
-	}
-
-	if (bindtextdomain(GETTEXT_PACKAGE, LOCALEDIR) == NULL) {
-		perror("bindtextdomain");
-		exit(1);
-	}
-
-	if (textdomain(GETTEXT_PACKAGE) == NULL) {
-		perror("textdomain");
-		exit(1);
-	}
+  
 }
