--- ../speechd-0.10.2/src/clients/say/say.c	2020-11-24 23:38:03.000000000 +0000
+++ ./src/clients/say/say.c	2021-04-05 18:31:24.825310937 +0000
@@ -128,13 +128,13 @@
 			printf("Invalid language!\n");
 	} else {
 		char *locale = strdup(setlocale(LC_MESSAGES, NULL));
-		char *dot = index(locale, '.');
+		char *dot = strchr(locale, '.');
 		if (dot)
 			*dot = 0;
-		char *at = index(locale, '@');
+		char *at = strchr(locale, '@');
 		if (at)
 			*at = 0;
-		char *underscore = index(locale, '_');
+		char *underscore = strchr(locale, '_');
 		if (underscore)
 			*underscore = '-';
 		if (spd_set_language(conn, locale))
