diff -uNr speechd-0.11.2/src/clients/say/Makefile.am speechd-0.11.2.mod/src/clients/say/Makefile.am
--- speechd-0.11.2/src/clients/say/Makefile.am	2022-08-26 06:13:57.000000000 +0800
+++ speechd-0.11.2.mod/src/clients/say/Makefile.am	2022-08-26 10:07:12.045684832 +0800
@@ -23,7 +23,7 @@
 bin_PROGRAMS = spd-say
 spd_say_CPPFLAGS = $(inc_local) $(GLIB_CFLAGS)
 spd_say_SOURCES = say.c options.c options.h
-spd_say_LDADD = $(c_api)/libspeechd.la $(EXTRA_SOCKET_LIBS) $(top_builddir)/src/common/libcommon.la -lpthread
+spd_say_LDADD = $(c_api)/libspeechd.la $(EXTRA_SOCKET_LIBS) $(top_builddir)/src/common/libcommon.la
 
 if HAVE_HELP2MAN
 spd-say.1: spd-say$(EXEEXT)
diff -uNr speechd-0.11.2/src/clients/spdsend/Makefile.am speechd-0.11.2.mod/src/clients/spdsend/Makefile.am
--- speechd-0.11.2/src/clients/spdsend/Makefile.am	2022-08-26 06:13:57.000000000 +0800
+++ speechd-0.11.2.mod/src/clients/spdsend/Makefile.am	2022-08-26 10:07:29.335684826 +0800
@@ -19,7 +19,7 @@
 
 bin_PROGRAMS = spdsend
 spdsend_SOURCES = spdsend.h spdsend.c server.c client.c common.c 
-spdsend_LDADD = $(EXTRA_SOCKET_LIBS) -lpthread
+spdsend_LDADD = $(EXTRA_SOCKET_LIBS)
 
 
 -include $(top_srcdir)/git.mk
diff -uNr speechd-0.11.2/src/modules/Makefile.am speechd-0.11.2.mod/src/modules/Makefile.am
--- speechd-0.11.2/src/modules/Makefile.am	2022-08-26 06:13:57.000000000 +0800
+++ speechd-0.11.2.mod/src/modules/Makefile.am	2022-08-26 10:07:48.025684819 +0800
@@ -28,7 +28,7 @@
 
 inc_local = -I$(top_srcdir)/include -I$(top_srcdir)/src/common
 common_SOURCES = module_main.c module_readline.c module_process.c module_config.c module_utils.c module_utils.h
-common_LDADD = $(DOTCONF_LIBS) $(GLIB_LIBS) $(audio_dlopen) -lpthread
+common_LDADD = $(DOTCONF_LIBS) $(GLIB_LIBS) $(audio_dlopen)
 
 AM_CFLAGS = $(ERROR_CFLAGS)
 AM_CXXFLAGS = $(ERROR_CFLAGS)
diff -uNr speechd-0.11.2/src/tests/Makefile.am speechd-0.11.2.mod/src/tests/Makefile.am
--- speechd-0.11.2/src/tests/Makefile.am	2022-08-26 06:13:57.000000000 +0800
+++ speechd-0.11.2.mod/src/tests/Makefile.am	2022-08-26 10:08:07.925684811 +0800
@@ -61,7 +61,7 @@
 clibrary2_LDADD = $(c_api)/libspeechd.la $(EXTRA_SOCKET_LIBS)
 
 clibrary3_SOURCES = clibrary2.c
-clibrary3_LDADD = $(c_api)/libspeechd.la $(EXTRA_SOCKET_LIBS) -lpthread
+clibrary3_LDADD = $(c_api)/libspeechd.la $(EXTRA_SOCKET_LIBS)
 clibrary3_CPPFLAGS = $(AM_CPPFLAGS) -DTHOROUGH
 
 connection_recovery_SOURCES = connection-recovery.c
