# Mitigates https://github.com/termux/termux-packages/issues/15425

--- a/config/speechd.conf
+++ b/config/speechd.conf
@@ -45,8 +45,9 @@
 # By default, Speech Dispatcher is configured to shut itself down after a period of
 # time if no clients are connected. The timeout value is in seconds, and is started when
 # the last client disconnects. A value of 0 disables the timeout.
+# Note for Termux: timeout is disabled by default, for a reason.
 
-# Timeout 5
+# Timeout 0
 
 # -----LOGGING CONFIGURATION-----
 
--- ./src/api/c/libspeechd.c.orig	2025-05-07 00:38:06.000000000 +0530
+++ ./src/api/c/libspeechd.c	2025-11-17 13:45:27.210794201 +0530
@@ -654,7 +654,9 @@
 
 	/* close the socket */
 	if (connection->socket >= 0) {
+#ifndef __ANDROID__
 		close(connection->socket);
+#endif
 		connection->socket = -1;
 		connection->stream = NULL;
 	}
--- ./src/server/configuration.c.orig	2025-05-07 00:38:06.000000000 +0530
+++ ./src/server/configuration.c	2025-11-17 13:46:01.070332858 +0530
@@ -517,7 +517,7 @@
 	if (!SpeechdOptions.localhost_access_only_set)
 		SpeechdOptions.localhost_access_only = 1;
 	if (!SpeechdOptions.server_timeout_set)
-		SpeechdOptions.server_timeout = 5;
+		SpeechdOptions.server_timeout = 0;
 	
 
 	logfile = stderr;
