diff -u -r ../mutt-1.14.0/doc/Muttrc.head ./doc/Muttrc.head
--- ../mutt-1.14.0/doc/Muttrc.head	2020-05-02 21:08:56.000000000 +0000
+++ ./doc/Muttrc.head	2020-05-11 09:29:03.417162000 +0000
@@ -23,7 +23,7 @@
 "call urlview to extract URLs out of a message"
 
 # Show documentation when pressing F1
-macro generic,pager <F1> "<shell-escape> less @docdir@/manual.txt<Enter>" "show Mutt documentation"
+macro generic,pager <F1> "<shell-escape> less @TERMUX_PREFIX@/share/doc/mutt/manual.txt<Enter>" "show Mutt documentation"
 
 # show the incoming mailboxes list (just like "mutt -y") and back when pressing "y"
 # note: these macros have been subsumed by the <browse-mailboxes> function.
