diff -u -r ../mutt-1.5.23/Makefile.in ./Makefile.in
--- ../mutt-1.5.23/Makefile.in	2014-03-12 17:26:44.000000000 +0100
+++ ./Makefile.in	2014-07-02 04:15:30.884707410 +0200
@@ -451,7 +451,7 @@
 AUTOMAKE_OPTIONS = 1.6 foreign
 @BUILD_IMAP_TRUE@IMAP_SUBDIR = imap
 @BUILD_IMAP_TRUE@IMAP_INCLUDES = -I$(top_srcdir)/imap
-SUBDIRS = m4 po intl doc contrib $(IMAP_SUBDIR)
+SUBDIRS = m4 po intl contrib $(IMAP_SUBDIR)
 bin_SCRIPTS = muttbug flea @SMIMEAUX_TARGET@
 @BUILD_HCACHE_TRUE@HCVERSION = hcversion.h
 BUILT_SOURCES = keymap_defs.h patchlist.c reldate.h conststrings.c $(HCVERSION)
