diff -u -r ../mutt-1.5.24/smime_keys.pl ./smime_keys.pl
--- ../mutt-1.5.24/smime_keys.pl	2015-08-30 13:06:38.000000000 -0400
+++ ./smime_keys.pl	2015-11-08 01:41:12.179119879 -0500
@@ -76,7 +76,7 @@
 
 
 my $mutt = $ENV{MUTT_CMDLINE} || 'mutt';
-my $opensslbin = "/usr/bin/openssl";
+my $opensslbin = "@TERMUX_PREFIX@/bin/openssl";
 my $tmpdir;
 
 # Get the directories mutt uses for certificate/key storage.
