diff -u -r ../nushell-0.18.1/Cargo.toml ./Cargo.toml
--- ../nushell-0.18.1/Cargo.toml	2020-08-12 03:59:28.000000000 +0000
+++ ./Cargo.toml	2020-08-13 08:02:58.145134000 +0000
@@ -64,8 +64,6 @@ toml = "0.5.6"
 
 [features]
 default = [
-  "sys",
-  "ps",
   "textview",
   "inc",
   "git-support",
@@ -79,7 +77,6 @@ default = [
   "match",
   "post",
   "fetch",
-  "rich-benchmark",
 ]
 extra = ["default", "binaryview", "tree", "clipboard-cli", "trash-support", "start", "bson", "sqlite", "s3"]
 stable = ["default"]
