diff -uNr nushell-0.14.0/Cargo.toml nushell-0.14.0.mod/Cargo.toml
--- nushell-0.14.0/Cargo.toml	2020-05-12 19:32:51.000000000 +0300
+++ nushell-0.14.0.mod/Cargo.toml	2020-05-13 12:22:00.470388369 +0300
@@ -62,7 +62,7 @@
 # Test executables
 test-bins = []
 
-default = ["sys", "ps", "textview", "inc", "str"]
+default = ["textview", "inc", "str"]
 stable = ["default", "starship-prompt", "binaryview", "match", "tree", "average", "post", "fetch", "clipboard-cli", "trash-support", "start"]
 
 # Default
