diff -uNr nushell-0.15.0/Cargo.toml nushell-0.15.0.mod/Cargo.toml
--- nushell-0.15.0/Cargo.toml	2020-06-09 23:39:15.000000000 +0300
+++ nushell-0.15.0.mod/Cargo.toml	2020-06-21 19:36:42.914555630 +0300
@@ -57,7 +57,7 @@
 nu-build = { version = "0.15.0", path = "./crates/nu-build" }
 
 [features]
-default = ["sys", "ps", "textview", "inc"]
+default = ["textview", "inc"]
 stable = ["default", "starship-prompt", "binaryview", "match", "tree", "post", "fetch", "clipboard-cli", "trash-support", "start"]
 
 # Default
