diff -uNr nushell-0.25.1/Cargo.toml nushell-0.25.1.mod/Cargo.toml
--- nushell-0.25.1/Cargo.toml	2021-01-06 04:58:04.000000000 +0200
+++ nushell-0.25.1.mod/Cargo.toml	2021-01-07 16:24:22.942615177 +0200
@@ -73,8 +73,6 @@
 which-support = ["nu-cli/ichwh", "nu-cli/which"]
 
 default = [
-  "sys",
-  "ps",
   "textview",
   "inc",
   "git-support",
@@ -88,7 +86,6 @@
   "match",
   "post",
   "fetch",
-  "rich-benchmark",
   "zip-support",
 ]
 extra = ["default", "binaryview", "tree", "clipboard-cli", "trash-support", "start", "bson", "sqlite", "s3", "chart", "xpath", "selector"]
