diff -u -r ../nushell-0.18.1/Cargo.toml ./Cargo.toml
--- ../nushell-0.18.1/Cargo.toml	2020-08-12 03:59:28.000000000 +0000
+++ ./Cargo.toml	2020-08-13 08:02:58.145134000 +0000
@@ -53,7 +53,6 @@
 log = "0.4.8"
 pretty_env_logger = "0.4.0"
 quick-xml = "0.18.1"
-starship = "0.43.0"
 
 [dev-dependencies]
 nu-test-support = {version = "0.18.1", path = "./crates/nu-test-support"}
@@ -64,8 +63,6 @@
 
 [features]
 default = [
-  "sys",
-  "ps",
   "textview",
   "inc",
   "git-support",
