diff --git a/crates/nu-cli/Cargo.toml b/crates/nu-cli/Cargo.toml
index 6368b7b8..720bd6b6 100644
--- a/crates/nu-cli/Cargo.toml
+++ b/crates/nu-cli/Cargo.toml
@@ -24,7 +24,7 @@ app_dirs = "1.2.1"
 async-stream = "0.2"
 base64 = "0.12.0"
 bigdecimal = { version = "0.1.0", features = ["serde"] }
-bson = { version = "0.14.1", features = ["decimal128"] }
+bson = "0.14.1"
 byte-unit = "3.0.3"
 bytes = "0.5.4"
 calamine = "0.16"
