diff -uNr xfdesktop-xfdesktop-4.15.0/pixmaps/Makefile.am xfdesktop-xfdesktop-4.15.0.mod/pixmaps/Makefile.am
--- xfdesktop-xfdesktop-4.15.0/pixmaps/Makefile.am	2020-06-19 02:20:25.000000000 +0300
+++ xfdesktop-xfdesktop-4.15.0.mod/pixmaps/Makefile.am	2020-06-22 16:03:49.597436313 +0300
@@ -40,7 +40,7 @@
 	$(pixmaps_DATA) \
 	$(pixmapsx_DATA)
 
-gtk_update_icon_cache = gtk-update-icon-cache -f -t $(datadir)/icons/hicolor
+gtk_update_icon_cache = true
 
 install-data-hook:
 	@-if test -z "$(DESTDIR)"; then \
--- a/pixmaps/Makefile.in
+++ b/pixmaps/Makefile.in
@@ -416,7 +416,7 @@
 	$(pixmaps_DATA) \
 	$(pixmapsx_DATA)
 
-gtk_update_icon_cache = gtk-update-icon-cache -f -t $(datadir)/icons/hicolor
+gtk_update_icon_cache = true
 all: all-am
 
 .SUFFIXES:
