From 20c30f6d189eb78bf6c0eae4b5f4b9d5e1baf0ba Mon Sep 17 00:00:00 2001
From: Rui Chen <rui@chenrui.dev>
Date: Sun, 28 Apr 2024 12:44:20 -0400
Subject: [PATCH] build: fix variable scope issue in undo functionality

Signed-off-by: Rui Chen <rui@chenrui.dev>
---
 src/main.c | 23 ++++++++++++-----------
 1 file changed, 12 insertions(+), 11 deletions(-)

diff --git a/src/main.c b/src/main.c
index c2e84bb..af124dd 100644
--- a/src/main.c
+++ b/src/main.c
@@ -708,18 +708,20 @@ int main(int argc, char *argv[])
 				}
 				break;
 			case 'u': // Undo
-				move_t old_move;
-				if (undo_stack_pop(&old_move))
-				{	// Stack empty
+				{
+					move_t old_move;
+					if (undo_stack_pop(&old_move))
+					{	// Stack empty
+						break;
+					}
+					x = old_move.x;
+					y = old_move.y;
+					posy = (y-GRID_NUMBER_START_Y)/GRID_COL_DELTA;
+					posx = (x-GRID_NUMBER_START_X)/GRID_LINE_DELTA;
+					user_board[posy*9+posx] = old_move.prev_val;
+					fill_grid(user_board, plain_board, x, y);
 					break;
 				}
-				x = old_move.x;
-				y = old_move.y;
-				posy = (y-GRID_NUMBER_START_Y)/GRID_COL_DELTA;
-				posx = (x-GRID_NUMBER_START_X)/GRID_LINE_DELTA;
-				user_board[posy*9+posx] = old_move.prev_val;
-				fill_grid(user_board, plain_board, x, y);
-				break;
 
 			default:
 				break;
@@ -755,4 +757,3 @@ int main(int argc, char *argv[])
 	endwin();
 	return EXIT_SUCCESS;
 }
-
