--- ./src/Makefile
+++ ../src/Makefile
@@ -3,8 +3,8 @@
 ## License.....: MIT
 ##                                                                                                                                                       -SHARED                  := 0                                                -DEBUG                   := 0
+SHARED                  := 1                                                +DEBUG                   := 1
 PRODUCTION              := 1
                                                                              ##
@@ -44,7 +44,7 @@
 ##
 
 DESTDIR                 ?=
-PREFIX                  ?= /usr/local
+PREFIX                  ?= @TERMUX_PREFIX@
 
 INSTALL_FOLDER          ?= $(PREFIX)/bin
 LIBRARY_FOLDER          ?= $(PREFIX)/lib
@@ -126,8 +126,8 @@
 # (which is automatically run for every github release)
 # the value will be something like this: "tag: vX.Y.Z, refs/pull/Y/head"
 
-VERSION_EXPORT          := tag: v4.0.0
-VERSION_TAG             := $(shell test -d .git && git describe --tags --dirty=+ || echo "$(VERSION_EXPORT)" | $(SED) 's/.*: v\([\.0-9]*\),.*/v\1/')
+VERSION_EXPORT          := HEAD -> master, tag: v4.0.0
+VERSION_TAG             := $(shell test -d .git && git describe --tags --dirty=+ || echo "$(VERSION_EXPORT)" | $(SED) 's/.*: v\([\.0-9]*\).*/v\1/')
 
 ##
 ## General compiler and linker options
