https://github.com/termux/termux-packages/issues/15852

--- a/configure
+++ b/configure
@@ -18946,6 +18946,7 @@
 #line 18946 "configure"
 #include "confdefs.h"
 #include <${cf_header}>
+char *tgoto();
 int
 main (void)
 {
@@ -19223,6 +19224,7 @@
 #line 19223 "configure"
 #include "confdefs.h"
 #include <${cf_cv_ncurses_header:-curses.h}>
+char *tgoto();
 int
 main (void)
 {
@@ -20093,6 +20095,7 @@
 #include "confdefs.h"
 
 #include <stdlib.h>
+#include <wchar.h>
 int
 main (void)
 {
@@ -20669,6 +20672,7 @@
 #line 20669 "configure"
 #include "confdefs.h"
 #include <${cf_cv_ncurses_header:-curses.h}>
+char *tgoto();
 int
 main (void)
 {
@@ -20963,6 +20967,7 @@
 #line 20963 "configure"
 #include "confdefs.h"
 #include <${cf_header}>
+char *tgoto();
 int
 main (void)
 {
@@ -22213,6 +22218,7 @@
 #line 22213 "configure"
 #include "confdefs.h"
 #include <${cf_cv_ncurses_header:-curses.h}>
+char *tgoto();
 int
 main (void)
 {
@@ -22299,6 +22305,7 @@
 #line 22299 "configure"
 #include "confdefs.h"
 #include <${cf_cv_ncurses_header:-curses.h}>
+char *tgoto();
 int
 main (void)
 {
@@ -22593,6 +22600,7 @@
 #line 22593 "configure"
 #include "confdefs.h"
 #include <${cf_header}>
+char *tgoto();
 int
 main (void)
 {
@@ -27618,6 +27626,7 @@
 #line 27618 "configure"
 #include "confdefs.h"
 
+#include <stdio.h>
 #if HAVE_TERMIO_H
 #include <termio.h>
 #endif
