https://github.com/termux/termux-packages/issues/15852

--- a/configure
+++ b/configure
@@ -21109,6 +21109,7 @@
 #line 21109 "configure"
 #include "confdefs.h"
 #include <${cf_header}>
+char *tgoto();
 int
 main (void)
 {
@@ -21388,6 +21389,7 @@
 #line 21388 "configure"
 #include "confdefs.h"
 #include <${cf_cv_ncurses_header:-curses.h}>
+char *tgoto();
 int
 main (void)
 {
@@ -23002,6 +23004,7 @@
 #line 23002 "configure"
 #include "confdefs.h"
 #include <${cf_cv_ncurses_header:-curses.h}>
+char *tgoto();
 int
 main (void)
 {
@@ -23636,6 +23639,7 @@
 #line 23636 "configure"
 #include "confdefs.h"
 #include <${cf_header}>
+char *tgoto();
 int
 main (void)
 {
@@ -24897,6 +24901,7 @@
 #line 24897 "configure"
 #include "confdefs.h"
 #include <${cf_cv_ncurses_header:-curses.h}>
+char *tgoto();
 int
 main (void)
 {
@@ -25090,6 +25095,7 @@
 #line 25090 "configure"
 #include "confdefs.h"
 #include <${cf_cv_ncurses_header:-curses.h}>
+char *tgoto();
 int
 main (void)
 {
@@ -25724,6 +25730,7 @@
 #line 25724 "configure"
 #include "confdefs.h"
 #include <${cf_header}>
+char *tgoto();
 int
 main (void)
 {
