diff -u -r ../lynx2-8-8/src/LYMain.c ./src/LYMain.c
--- ../lynx2-8-8/src/LYMain.c	2014-02-14 14:00:16.000000000 +0100
+++ ./src/LYMain.c	2014-07-02 08:39:17.064274768 +0200
@@ -398,7 +398,7 @@
 BOOLEAN LYSetCookies = SET_COOKIES;	/* Process Set-Cookie headers? */
 BOOLEAN LYUseDefSelPop = TRUE;	/* Command line -popup toggle */
 BOOLEAN LYUseDefaultRawMode = TRUE;
-BOOLEAN LYUseMouse = FALSE;
+BOOLEAN LYUseMouse = TRUE;
 BOOLEAN LYisConfiguredForX = FALSE;
 BOOLEAN UCForce8bitTOUPPER = FALSE;	/* override locale for case-conversion? */
 BOOLEAN UCSaveBookmarksInUnicode = FALSE;
