diff -uNr lynx2.8.9dev.19/userdefs.h lynx2.8.9dev.19.mod/userdefs.h
--- lynx2.8.9dev.19/userdefs.h	2018-05-16 00:12:20.000000000 +0300
+++ lynx2.8.9dev.19.mod/userdefs.h	2018-06-19 19:40:07.061501423 +0300
@@ -366,7 +366,7 @@
  * the "TMPDIR" (unix), or "TEMP" or "TMP" (Windows,DOS,OS/2)
  * variable.
  */
-#define TEMP_SPACE "/tmp/"
+#define TEMP_SPACE "@TERMUX_PREFIX@/tmp/"
 
 /********************************
  * Comment this line out to disable code that implements command logging
@@ -741,7 +741,7 @@
  *     RFC 1345 Mnemonic                  mnemonic
  *     Transparent                        x-transparent
  */
-#define CHARACTER_SET "iso-8859-1"
+#define CHARACTER_SET "utf-8"
 
 /*****************************
  * PREFERRED_LANGUAGE is the language in MIME notation (e.g., "en",
@@ -770,7 +770,7 @@
  * an error response with the 406 (not acceptable) status code, though
  * the sending of an unacceptable response is also allowed. (RFC2068)
  */
-#define PREFERRED_CHARSET ""
+#define PREFERRED_CHARSET "utf-8"
 
 /*****************************
 * If MULTI_BOOKMARK_SUPPORT is set to MBM_STANDARD or MBM_ADVANCED, and
@@ -924,7 +924,7 @@
  * The default defined here can be changed in lynx.cfg, and .lynxrc, or
  * toggled via the -accept_all_cookies command line switch.
  */
-#define ACCEPT_ALL_COOKIES FALSE
+#define ACCEPT_ALL_COOKIES TRUE
 
 /****************************************************************
  *   Section 2.   Things that you probably want to change or review
@@ -975,7 +975,7 @@
  * and configuration defaults, and the default always can be toggled
  * via the -show_cursor command line switch.
  */
-#define SHOW_CURSOR FALSE
+#define SHOW_CURSOR TRUE
 
 /******************************
 * UNDERLINE_LINKS controls whether links are underlined by default, or shown
