diff -uNr lynx2.8.9dev.19/src/LYMain.c lynx2.8.9dev.19.mod/src/LYMain.c
--- lynx2.8.9dev.19/src/LYMain.c	2018-05-12 02:11:18.000000000 +0300
+++ lynx2.8.9dev.19.mod/src/LYMain.c	2018-06-19 19:40:07.051501423 +0300
@@ -394,7 +394,7 @@
 BOOLEAN LYSetCookies = SET_COOKIES;	/* Process Set-Cookie headers? */
 BOOLEAN LYUseDefSelPop = TRUE;	/* Command line -popup toggle */
 BOOLEAN LYUseDefaultRawMode = TRUE;
-BOOLEAN LYUseMouse = FALSE;
+BOOLEAN LYUseMouse = TRUE;
 BOOLEAN LYisConfiguredForX = FALSE;
 BOOLEAN UCForce8bitTOUPPER = FALSE;	/* override locale for case-conversion? */
 BOOLEAN UCSaveBookmarksInUnicode = FALSE;
