diff -uNr lynx2.8.9dev.19/src/LYUtils.c lynx2.8.9dev.19.mod/src/LYUtils.c
--- lynx2.8.9dev.19/src/LYUtils.c	2018-05-16 00:20:52.000000000 +0300
+++ lynx2.8.9dev.19.mod/src/LYUtils.c	2018-06-19 19:40:07.058168089 +0300
@@ -5185,7 +5185,7 @@
 		/*
 		 * Use /tmp; it should be writable.
 		 */
-		StrAllocCopy(HomeDir, "/tmp");
+		StrAllocCopy(HomeDir, "@TERMUX_PREFIX@/tmp");
 	    }
 #endif
 #endif /* VMS */
