diff -u -r ../lynx2-8-8/WWW/Library/Implementation/www_tcp.h ./WWW/Library/Implementation/www_tcp.h
--- ../lynx2-8-8/WWW/Library/Implementation/www_tcp.h	2013-07-29 23:38:35.000000000 +0200
+++ ./WWW/Library/Implementation/www_tcp.h	2016-06-20 00:19:17.000000000 +0200
@@ -797,7 +797,7 @@
 
 #if defined(VMS)
 #define socklen_t unsigned
-#else
+#elif !defined(__APPLE__)
 #define socklen_t int		/* used for default LY_SOCKLEN definition */
 #endif
 
