From 256c911eee50ac9deb8e50015debe896d0d61edb Mon Sep 17 00:00:00 2001
From: dalance <dalance@gmail.com>
Date: Tue, 18 Jan 2022 18:53:02 +0900
Subject: [PATCH 1/4] Try android support

---
 src/columns.rs | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/columns.rs b/src/columns.rs
index 3088319..e724a4f 100644
--- a/src/columns.rs
+++ b/src/columns.rs
@@ -1,5 +1,7 @@
 #[cfg(target_os = "linux")]
 include!("./columns/os_linux.rs");
+#[cfg(target_os = "android")]
+include!("./columns/os_linux.rs");
 #[cfg(target_os = "macos")]
 include!("./columns/os_macos.rs");
 #[cfg(target_os = "windows")]
-- 
2.33.0

