--- a/CMakeLists.txt	2025-10-16 14:36:48.000000000 +0530
+++ b/CMakeLists.txt	2025-10-21 20:18:50.876173014 +0530
@@ -317,7 +317,7 @@
     set(KWIN_BUILD_X11_BACKEND OFF CACHE BOOL "Enable building kwin_x11" FORCE)
 endif()
 
-find_package(UDev)
+#find_package(UDev)
 set_package_properties(UDev PROPERTIES
     URL "https://www.freedesktop.org/wiki/Software/systemd/"
     DESCRIPTION "Linux device library."

--- a/src/CMakeLists.txt	2025-10-16 14:36:48.000000000 +0530
+++ b/src/CMakeLists.txt	2025-10-21 20:19:10.616173015 +0530
@@ -259,7 +259,7 @@
         KDecoration3::KDecoration
         KDecoration3::KDecoration3Private
 
-        UDev::UDev
+#       UDev::UDev
         XKB::XKB
         EGL::EGL
         epoxy::epoxy

--- a/src/utils/CMakeLists.txt	2025-10-16 14:36:49.000000000 +0530
+++ b/src/utils/CMakeLists.txt	2025-10-26 22:01:08.779716176 +0530
@@ -9,7 +9,7 @@
     realtime.cpp
     softwarevsyncmonitor.cpp
     subsurfacemonitor.cpp
-    udev.cpp
+#   udev.cpp
     vsyncmonitor.cpp
 )
 if (KWIN_BUILD_X11)
 
