diff -uNr --color src.mod/src/resultdom.h src/src/resultdom.h
--- src.mod/src/resultdom.h	2020-03-26 16:48:48.851000843 +0800
+++ src/src/resultdom.h	2020-03-26 16:52:05.101000768 +0800
@@ -22,6 +22,7 @@
 #include <vector>
 #include <list>
 #include <stdlib.h>
+#include <locale.h>
 
 
 class Item //элемент дерева данных (например <disk_usage>1129827635.000000</disk_usage>)
@@ -53,4 +54,4 @@
     Item*		parent;
 };
 
-#endif //RESULTDOM_H
\ No newline at end of file
+#endif //RESULTDOM_H
