diff -uNr boinctui-2.5.0/configure.in boinctui-2.5.0.mod/configure.in
--- boinctui-2.5.0/configure.in	2016-09-10 20:01:45.000000000 +0300
+++ boinctui-2.5.0.mod/configure.in	2020-03-26 15:21:40.598627902 +0200
@@ -39,7 +39,6 @@
 )
 
 AC_CHECK_LIB(expat,    main,  , [AC_MSG_ERROR([Couldn't find expat library])] )
-AC_CHECK_LIB(pthread,    main,  , [AC_MSG_ERROR([Couldn't find pthread library])] )
 
 AC_ARG_WITH(
     [gnutls],
