--- a/cpp/cmake_modules/FindPython3Alt.cmake
+++ b/cpp/cmake_modules/FindPython3Alt.cmake
@@ -58,7 +58,7 @@
                                                                      Development NumPy)
 else()
   find_package(Python3 ${Python3Alt_FIND_PACKAGE_OPTIONS}
-               COMPONENTS Interpreter Development.Module NumPy)
+               COMPONENTS Interpreter Development.Module)
 endif()
 
 if(NOT Python3_FOUND)
@@ -70,7 +70,9 @@
 set(PYTHON_LIBRARIES ${Python3_LIBRARIES})
 set(PYTHON_OTHER_LIBS)
 
+if(FALSE)
 get_target_property(NUMPY_INCLUDE_DIRS Python3::NumPy INTERFACE_INCLUDE_DIRECTORIES)
+endif(FALSE)
 
 # CMake's python3_add_library() doesn't apply the required extension suffix,
 # detect it ourselves.
@@ -94,5 +96,6 @@
   set_target_properties(${name} PROPERTIES SUFFIX ${_EXT_SUFFIX})
 endfunction()
 
+include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(
   Python3Alt REQUIRED_VARS PYTHON_EXECUTABLE PYTHON_INCLUDE_DIRS NUMPY_INCLUDE_DIRS)
