diff -uNr streamripper-1.64.6/configure streamripper-1.64.6.mod/configure
--- streamripper-1.64.6/configure	2009-03-31 04:51:13.000000000 +0300
+++ streamripper-1.64.6.mod/configure	2020-01-02 01:33:25.603677178 +0200
@@ -5374,7 +5374,7 @@
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lpthread  $LIBS"
+LIBS="$LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -5434,7 +5434,7 @@
 { $as_echo "$as_me:$LINENO: result: $ac_cv_lib_pthread_pthread_create" >&5
 $as_echo "$ac_cv_lib_pthread_pthread_create" >&6; }
 if test "x$ac_cv_lib_pthread_pthread_create" = x""yes; then
-  THREADLIBS="-lpthread"
+  THREADLIBS=""
 fi
 
 if test "$THREADLIBS" = no; then
@@ -5586,7 +5586,7 @@
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lpthreads  $LIBS"
+LIBS="$LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -5646,7 +5646,7 @@
 { $as_echo "$as_me:$LINENO: result: $ac_cv_lib_pthreads_pthread_create" >&5
 $as_echo "$ac_cv_lib_pthreads_pthread_create" >&6; }
 if test "x$ac_cv_lib_pthreads_pthread_create" = x""yes; then
-  THREADLIBS="-lpthreads"
+  THREADLIBS=""
 fi
 
 fi
