--- ../cache/soxr-0.1.2-Source/go	2015-08-27 07:53:52.000000000 +0000
+++ ./go	2018-01-18 23:39:51.044488639 +0000
@@ -12,6 +12,6 @@
 mkdir -p $build
 cd $build
 
-cmake -DCMAKE_BUILD_TYPE=$build -Wno-dev .. &&
-  make $j &&
-    (ctest $j || echo "FAILURE details in $build/Testing/Temporary/LastTest.log")
+cmake -DCMAKE_BUILD_TYPE=$build -DCMAKE_INSTALL_PREFIX=@TERMUX_PREFIX@  -DBUILD_TESTS=off -Wno-dev .. &&
+  make $j
+make install
