diff --git a/cava.c b/cava.c
index de8f98a..e88154e 100644
--- a/cava.c
+++ b/cava.c
@@ -268,7 +268,7 @@ FILE *fp;
 	}
 	if (strcmp(inputMethod, "fifo") == 0) {
 		im = 2;
-		audio.source = (char *)iniparser_getstring(ini, "input:source", "/tmp/mpd.fifo");
+		audio.source = (char *)iniparser_getstring(ini, "input:source", "@TERMUX_PREFIX@/tmp/mpd.fifo");
 	}
 	if (strcmp(inputMethod, "pulse") == 0) {
 		im = 3;
