--- a/configure.ac
+++ b/configure.ac
@@ -436,7 +436,7 @@
         CPPFLAGS="$CPPFLAGS -I/usr/local/include/iniparser/"
         CPPFLAGS="$CPPFLAGS -I/opt/homebrew/include/iniparser/"
       else
-        CPPFLAGS="$CPPFLAGS -I/usr/include/iniparser"
+        CPPFLAGS="$CPPFLAGS -I@TERMUX_PREFIX@/include/iniparser"
       fi
       AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <iniparser.h>]],
         [[dictionary* ini;
