--- a/configure.ac
+++ b/configure.ac
@@ -260,7 +260,7 @@
 AC_CHECK_LIB(iniparser,iniparser_load, have_iniparser=yes, have_iniparser=no)
     if [[ $have_iniparser = "yes" ]] ; then
     LIBS="$LIBS -liniparser"
-      CPPFLAGS="$CPPFLAGS -I/usr/include/iniparser"
+      CPPFLAGS="$CPPFLAGS -I@TERMUX_PREFIX@/include/iniparser"
     AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <iniparser.h>]],
       [[dictionary* ini;
       const char *keys[3];
