--- a/configure.ac
+++ b/configure.ac
@@ -63,9 +63,9 @@
 
 AC_CHECK_HEADERS([pthread.h],
 	AC_CHECK_LIB(pthread, pthread_create, LIBS="$LIBS -lpthread",
-		AC_MSG_ERROR([pthread.h found but there is no pthread library to make use of])
+		AC_MSG_NOTICE([pthread.h found but there is no pthread library to make use of])
 	),
-	AC_MSG_ERROR([no pthread.h header header file found])
+	AC_MSG_NOTICE([no pthread.h header header file found])
 )
 
 dnl ######################
@@ -260,7 +260,7 @@
 AC_CHECK_LIB(iniparser,iniparser_load, have_iniparser=yes, have_iniparser=no)
     if [[ $have_iniparser = "yes" ]] ; then
     LIBS="$LIBS -liniparser"
-      CPPFLAGS="$CPPFLAGS -I/usr/include/iniparser"
+      CPPFLAGS="$CPPFLAGS -I@TERMUX_PREFIX@/include/iniparser"
     AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <iniparser.h>]],
       [[dictionary* ini;
       const char *keys[3];
