--- a/Makefile.am
+++ b/Makefile.am
@@ -8,13 +8,13 @@
 cava_CPPFLAGS = -DPACKAGE=\"$(PACKAGE)\" -DVERSION=\"$(VERSION)\" \
            -D_POSIX_SOURCE -D _POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE_EXTENDED \
 	   -DFONTDIR=\"@FONT_DIR@\"
-cava_CFLAGS = -std=c99 -Wall -Werror -Wextra -Wno-unused-result -Wno-unknown-warning-option -Wno-maybe-uninitialized -Wno-vla-parameter
+cava_CFLAGS = -std=c99 -Wall -Werror -Wextra -Wno-unused-result -Wno-unknown-warning-option -Wno-maybe-uninitialized -Wno-vla-parameter -Wno-unused-variable
 
 if OSX
     cava_CFLAGS += -DNORT
     cava_LDADD =
 else
-    cava_LDADD = -lrt
+    cava_LDADD =
     cava_font_dir = @FONT_DIR@
     cava_font__DATA = cava.psf
 endif
