diff -uNr cava-0.8.3/Makefile.am cava-0.8.3.mod/Makefile.am
--- cava-0.8.3/Makefile.am	2022-11-04 16:01:05.000000000 +0800
+++ cava-0.8.3.mod/Makefile.am	2022-11-05 16:58:02.059784566 +0800
@@ -8,7 +8,7 @@
 cava_CPPFLAGS = -DPACKAGE=\"$(PACKAGE)\" -DVERSION=\"$(VERSION)\" \
            -D_POSIX_SOURCE -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE_EXTENDED \
 	   -DFONTDIR=\"@FONT_DIR@\"
-cava_CFLAGS = -std=c99 -Wall -Wextra -Wno-unused-result -Wno-unknown-warning-option -Wno-maybe-uninitialized -Wno-vla-parameter
+cava_CFLAGS = -std=c99 -Wall -Wextra -Wno-unused-result -Wno-unknown-warning-option -Wno-maybe-uninitialized -Wno-vla-parameter -Wno-unused-variable
 
 if OSX
     cava_CFLAGS += -DNORT
