diff -uNr cava-0.6.1/Makefile.am cava-0.6.1.mod/Makefile.am
--- cava-0.6.1/Makefile.am	2018-05-03 19:07:33.000000000 +0300
+++ cava-0.6.1.mod/Makefile.am	2019-08-16 13:48:27.945938729 +0300
@@ -11,7 +11,7 @@
 bin_PROGRAMS = cava
 cava_SOURCES = cava.c
 cava_LDADD = -liniparser
-cava_LDFLAGS = -L/usr/local/lib -Wl,-rpath /usr/local/lib
+cava_LDFLAGS =
 cava_CPPFLAGS = -DPACKAGE=\"$(PACKAGE)\" -DVERSION=\"$(VERSION)\" \
            -D_POSIX_SOURCE -D _POSIX_C_SOURCE=200809L
 cava_CFLAGS = -std=c99 -Wall -Wextra -Wno-unused-result -Wno-maybe-uninitialized 
