--- a/lib_pypy/pypy_tools/build_cffi_imports.py
+++ b/lib_pypy/pypy_tools/build_cffi_imports.py
@@ -39,6 +39,11 @@
     ("xx", None),    # for testing: 'None' should be completely ignored
     ]
 
+def fake_cpu_count():
+    return 1
+
+multiprocessing.cpu_count = fake_cpu_count
+
 # for distribution, we may want to fetch dependencies not provided by
 # the OS, such as a recent openssl/libressl.
 curdir = os.path.abspath(os.path.dirname(__file__))
