--- a/rpython/memory/gc/env.py
+++ b/rpython/memory/gc/env.py
@@ -140,7 +140,7 @@
         return get_L2cache_linux2_cpuinfo(label='L2 cache')
     #if arch == 's390x':    untested
     #    return get_L2cache_linux2_cpuinfo_s390x()
-    if arch in ('ia64', 'aarch64'):
+    if arch in ('ia64'):
         return get_L2cache_linux2_system_cpu_index()
     if arch in ('parisc', 'parisc64'):
         return get_L2cache_linux2_cpuinfo(label='D-cache')
