--- ../saltyim/Makefile	2022-05-13 15:01:00.000000000 +0000
+++ ./Makefile	2023-07-17 14:04:19.499079755 +0000
@@ -8,8 +8,6 @@
 GOCMD=go
 GOVER=$(shell go version | grep -o -E 'go1\.17\.[0-9]+')
 
-DESTDIR=/usr/local/bin
-
 ifeq ($(BRANCH), master)
 IMAGE := prologic/saltyim
 TAG := latest
@@ -75,8 +73,8 @@
 pwa: internal/web/app.wasm
 
 install: build ## Install salty-chat (cli) and saltyd (server) to $DESTDIR
-	@install -D -m 755 salty-chat $(DESTDIR)/salty-chat
-	@install -D -m 755 saltyd $(DESTDIR)/saltyd
+	@install -D -m 755 salty-chat $(DESTDIR)$(PREFIX)/bin/salty-chat
+	@install -D -m 755 saltyd $(DESTDIR)$(PREFIX)/bin/saltyd
 
 ifeq ($(PUBLISH), 1)
 image: generate ## Build the Docker image
