diff -u -r ../leveldb-1.20/port/port_posix.h ./port/port_posix.h
--- ../leveldb-1.20/port/port_posix.h	2017-03-02 01:08:02.000000000 +0100
+++ ./port/port_posix.h	2017-09-22 22:18:21.814123643 +0200
@@ -52,7 +52,7 @@
 
 #if defined(OS_MACOSX) || defined(OS_SOLARIS) || defined(OS_FREEBSD) ||\
     defined(OS_NETBSD) || defined(OS_OPENBSD) || defined(OS_DRAGONFLYBSD) ||\
-    defined(OS_ANDROID) || defined(OS_HPUX) || defined(CYGWIN)
+    defined(OS_ANDROID) || defined(OS_HPUX) || defined(CYGWIN) || defined(__ANDROID__)
 // Use fread/fwrite/fflush on platforms without _unlocked variants
 #define fread_unlocked fread
 #define fwrite_unlocked fwrite
