From bf5650f0dd41b500102a129d6867cb568f4edee4 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sat, 9 Nov 2024 13:50:05 -0500
Subject: [PATCH] remove new operator

---
 Magick++/lib/Magick++/Include.h | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/Magick++/lib/Magick++/Include.h b/Magick++/lib/Magick++/Include.h
index 7a4d3d6..cddacf2 100644
--- a/Magick++/lib/Magick++/Include.h
+++ b/Magick++/lib/Magick++/Include.h
@@ -32,10 +32,6 @@
 #include <errno.h>
 #include <stdint.h>
 
-#if __cplusplus >= 201703L
-void* operator new(size_t, void*) throw();
-#endif
-
 //
 // Include ImageMagick headers into namespace "MagickCore". If
 // MAGICKCORE_IMPLEMENTATION is defined, include ImageMagick development
-- 
2.47.0

