diff -uNr azpainter-2.1.5/configure azpainter-2.1.5.mod/configure
--- azpainter-2.1.5/configure	2019-12-02 18:06:30.000000000 +0200
+++ azpainter-2.1.5.mod/configure	2019-12-30 17:50:21.049470937 +0200
@@ -4780,29 +4780,6 @@
 
 
 
-as_ac_File=`$as_echo "ac_cv_file_"$freetype_dir/ft2build.h"" | $as_tr_sh`
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for \"$freetype_dir/ft2build.h\"" >&5
-$as_echo_n "checking for \"$freetype_dir/ft2build.h\"... " >&6; }
-if eval \${$as_ac_File+:} false; then :
-  $as_echo_n "(cached) " >&6
-else
-  test "$cross_compiling" = yes &&
-  as_fn_error $? "cannot check for file existence when cross compiling" "$LINENO" 5
-if test -r ""$freetype_dir/ft2build.h""; then
-  eval "$as_ac_File=yes"
-else
-  eval "$as_ac_File=no"
-fi
-fi
-eval ac_res=\$$as_ac_File
-	       { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_res" >&5
-$as_echo "$ac_res" >&6; }
-if eval test \"x\$"$as_ac_File"\" = x"yes"; then :
-
-else
-  as_fn_error $? "(freetype) ft2build.h not found" "$LINENO" 5
-fi
-
 ac_fn_c_check_header_mongrel "$LINENO" "fontconfig/fontconfig.h" "ac_cv_header_fontconfig_fontconfig_h" "$ac_includes_default"
 if test "x$ac_cv_header_fontconfig_fontconfig_h" = xyes; then :
 
