--- azpainter-v3.0.5/configure.orig	2022-03-21 11:06:20.000000000 +0900
+++ azpainter-v3.0.5/configure	2022-04-21 02:00:04.000000000 +0900
@@ -344,7 +344,6 @@
 
 fc_check_sharedlib "iconv"
 
-fc_check_funclink rt "clock_gettime"
 fc_check_lib libpng "libpng" "" "-lpng -lz" "png.h" "" "libpng-dev or libpng-devel or libpng or png" ""
 fc_check_lib zlib "zlib" "" "" "zlib.h" "" "zlib1g-dev or zlib-devel or zlib" ""
 fc_check_lib libjpeg "libjpeg" "" "-ljpeg" "stdio.h jpeglib.h" "" "libjpeg-dev or libjpeg-devel or libjpeg-turbo or jpeg-turbo" ""
