diff -uNr azpainter-2.1.5/src/Makefile.in azpainter-2.1.5.mod/src/Makefile.in
--- azpainter-2.1.5/src/Makefile.in	2019-12-02 18:06:30.000000000 +0200
+++ azpainter-2.1.5.mod/src/Makefile.in	2019-12-30 17:55:24.808888627 +0200
@@ -478,7 +478,7 @@
 AM_CPPFLAGS = -I../mlib/include -Iinclude -I@freetype_dir@ -DPACKAGE_DATA_DIR=\""$(datadir)/azpainter"\"
 AM_CFLAGS = -pthread
 azpainter_LDADD = ../mlib/libmlib.a \
-  -lX11 -lXext -lfontconfig -lfreetype -lpng -ljpeg -lz -lrt -lm
+  -lX11 -lXext -lfontconfig -lfreetype -lpng -ljpeg -lz -lm
 
 azpainter_SOURCES = \
 AppCursor.c \
