diff -uNr azpainter-2.1.5/src/Makefile.am azpainter-2.1.5.mod/src/Makefile.am
--- azpainter-2.1.5/src/Makefile.am	2019-12-02 18:06:30.000000000 +0200
+++ azpainter-2.1.5.mod/src/Makefile.am	2019-12-30 17:55:21.320849401 +0200
@@ -3,7 +3,7 @@
 
 bin_PROGRAMS=azpainter
 azpainter_LDADD=../mlib/libmlib.a \
-  -lX11 -lXext -lfontconfig -lfreetype -lpng -ljpeg -lz -lrt -lm
+  -lX11 -lXext -lfontconfig -lfreetype -lpng -ljpeg -lz -lm
 
 azpainter_SOURCES= \
 AppCursor.c \
