diff -u -r ../vim-7.4.1049/src/os_unix.h ./src/os_unix.h
--- ../vim-7.4.1049/src/os_unix.h	2016-01-04 06:45:29.000000000 -0500
+++ ./src/os_unix.h	2016-01-04 11:23:21.698790120 -0500
@@ -368,7 +368,7 @@
 # ifdef VMS
 #  define DFLT_DIR     "./,sys$login:,tmp:"
 # else
-#  define DFLT_DIR     ".,~/tmp,/var/tmp,/tmp" /* default for 'directory' */
+#  define DFLT_DIR     ".,~/tmp,@TERMUX_PREFIX@/var/tmp,@TERMUX_PREFIX@/tmp" /* default for 'directory' */
 # endif
 #endif
 
