diff --git a/src/sdk/scripting/bindings/sc_wxtypes.cpp b/src/sdk/scripting/bindings/sc_wxtypes.cpp
index 33cd09850..9a162bf9a 100644
--- a/src/sdk/scripting/bindings/sc_wxtypes.cpp
+++ b/src/sdk/scripting/bindings/sc_wxtypes.cpp
@@ -389,7 +389,6 @@ namespace ScriptBindings
                 func(&wxFileName::IsDir, "IsDir").
                 func(&wxFileName::MakeAbsolute, "MakeAbsolute").
                 func(&wxFileName::MakeRelativeTo, "MakeRelativeTo").
-                func(&wxFileName::Normalize, "Normalize").
                 func(&wxFileName::PrependDir, "PrependDir").
                 func(&wxFileName::RemoveDir, "RemoveDir").
                 func(&wxFileName::RemoveLastDir, "RemoveLastDir").

