--- a/src/mime/Makefile.am
+++ b/src/mime/Makefile.am
@@ -20,7 +20,7 @@
 ## Update mime only when system wide installation takes place
 
 install-data-hook: install-xdgmimeDATA
-	if [ -f $(DESTDIR)$(datadir)/mime/packages/freedesktop.org.xml ] ; then \
+	if false && [ -f $(DESTDIR)$(datadir)/mime/packages/freedesktop.org.xml ] ; then \
 	  if which update-mime-database>/dev/null 2>&1; then \
 	    update-mime-database $(DESTDIR)$(datadir)/mime; \
 	  fi; \
--- a/src/mime/Makefile.in
+++ b/src/mime/Makefile.in
@@ -713,7 +713,7 @@
 
 
 install-data-hook: install-xdgmimeDATA
-	if [ -f $(DESTDIR)$(datadir)/mime/packages/freedesktop.org.xml ] ; then \
+	if false && [ -f $(DESTDIR)$(datadir)/mime/packages/freedesktop.org.xml ] ; then \
 	  if which update-mime-database>/dev/null 2>&1; then \
 	    update-mime-database $(DESTDIR)$(datadir)/mime; \
 	  fi; \
