--- ./opal/mca/memory/patcher/memory_patcher_component.c	2017-05-10 17:41:36.000000000 +0200
+++ ../memory_patcher_component.c	2017-08-21 17:18:25.778728017 +0200
@@ -254,9 +254,9 @@
 
     if (advice == MADV_DONTNEED ||
 #ifdef MADV_REMOVE
-        advice == MADV_REMOVE ||
+        advice == MADV_REMOVE
 #endif
-        advice == POSIX_MADV_DONTNEED)
+        )
     {
         opal_mem_hooks_release_hook (start, length, false);
     }
@@ -342,7 +342,7 @@
 
 #include <stdio.h>
 #include <fcntl.h>
-#include <sys/shm.h>
+#include <linux/shm.h>
 
 static size_t memory_patcher_get_shm_seg_size (const void *shmaddr)
 {
