--- ./oshmem/mca/memheap/ptmalloc/malloc.c	2017-05-10 17:40:49.000000000 +0200
+++ ../malloc.c	2017-08-21 19:00:22.757971004 +0200
@@ -648,10 +648,10 @@
   system version and #define HAVE_USR_INCLUDE_MALLOC_H.
 */
 
-/* #define HAVE_USR_INCLUDE_MALLOC_H */
+#define HAVE_USR_INCLUDE_MALLOC_H
 
 #ifdef HAVE_USR_INCLUDE_MALLOC_H
-#include "/usr/include/malloc.h"
+#include <malloc.h>
 #else /* HAVE_USR_INCLUDE_MALLOC_H */
 
 struct mallinfo {
