diff -uNr NetHack-NetHack-3.6.4_Released/sys/unix/hints/linux NetHack-NetHack-3.6.4_Released.mod/sys/unix/hints/linux
--- NetHack-NetHack-3.6.4_Released/sys/unix/hints/linux	2019-12-18 22:29:40.000000000 +0200
+++ NetHack-NetHack-3.6.4_Released.mod/sys/unix/hints/linux	2020-01-13 15:33:51.932991035 +0200
@@ -10,7 +10,7 @@
 
 
 #PREFIX=/usr
-PREFIX=$(wildcard ~)/nh/install
+PREFIX=@TERMUX_PREFIX@
 HACKDIR=$(PREFIX)/games/lib/$(GAME)dir
 SHELLDIR = $(PREFIX)/games
 INSTDIR=$(HACKDIR)
@@ -20,9 +20,9 @@
 
 POSTINSTALL=cp -n sys/unix/sysconf $(INSTDIR)/sysconf; $(CHOWN) $(GAMEUID) $(INSTDIR)/sysconf; $(CHGRP) $(GAMEGRP) $(INSTDIR)/sysconf; chmod $(VARFILEPERM) $(INSTDIR)/sysconf;
 
-CFLAGS=-g -O -I../include -DNOTPARMDECL
+CFLAGS+=-g -I../include -DNOTPARMDECL
 CFLAGS+=-DDLB
-CFLAGS+=-DCOMPRESS=\"/bin/gzip\" -DCOMPRESS_EXTENSION=\".gz\"
+CFLAGS+=-DCOMPRESS=\"@TERMUX_PREFIX@/bin/gzip\" -DCOMPRESS_EXTENSION=\".gz\"
 CFLAGS+=-DSYSCF -DSYSCF_FILE=\"$(HACKDIR)/sysconf\" -DSECURE
 CFLAGS+=-DTIMED_DELAY
 CFLAGS+=-DHACKDIR=\"$(HACKDIR)\"
