--- ../Makefile.utl.orig	2019-08-11 17:36:35.716609211 +0200
+++ ./sys/unix/Makefile.utl	2019-08-11 17:37:38.569874628 +0200
@@ -239,11 +239,11 @@
 	perl mdgrep.pl
 
 ../include/onames.h: makedefs
-	./makedefs -o
+	./makedefs.host -o
 ../include/pm.h: makedefs
-	./makedefs -p
+	./makedefs.host -p
 ../include/vis_tab.h: makedefs
-	./makedefs -z
+	./makedefs.host -z
 # makedefs -z makes both vis_tab.h and vis_tab.c, but writes the .h first
 ../src/vis_tab.c: ../include/vis_tab.h
 
