--- wafw00f-2.1.0/Makefile	2020-01-29 18:24:09.000000000 +0600
+++ wafw00f-2.1.0-mod/Makefile	2020-02-25 10:56:31.903174066 +0600
@@ -1,15 +1,11 @@
 SRC_DIR = wafw00f
 DOC_DIR = docs
 MAKE = make
-
 all:
 	make install
-	make test
-	make html
-	make clean
 
 install:
-	pip install -q -e .[dev,docs]
+	export PYTHONPATH=@TERMUX_PREFIX@/lib/python3.8/site-packages && python3.8 setup.py install --prefix=@TERMUX_PREFIX@
 
 lint:
 	prospector $(SRC_DIR) --strictness veryhigh
