https://github.com/wlynxg/anet/pull/9
--- ./go/pkg/mod/github.com/wlynxg/anet@v0.0.3/interface_android.go
+++ ./go/pkg/mod/github.com/wlynxg/anet@v0.0.3/interface_android.go
@@ -141,8 +141,10 @@
 	toName       map[int]string // interface index to its name
 }
 
-//go:linkname zoneCache net.zoneCache
-var zoneCache ipv6ZoneCache
+var zoneCache = ipv6ZoneCache{
+	toIndex: make(map[string]int),
+	toName:  make(map[int]string),
+}
 
 // update refreshes the network interface information if the cache was last
 // updated more than 1 minute ago, or if force is set. It reports whether the
