diff --git a/CMakeLists.txt b/CMakeLists.txt
index b99e161..5c8d0a3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -554,7 +554,7 @@ if (UNIX AND NOT APPLE)
         install(PROGRAMS ${CMAKE_BINARY_DIR}/cantata-remote DESTINATION ${SHARE_INSTALL_PREFIX}/${CMAKE_PROJECT_NAME}/scripts)
         install(FILES cantata.desktop DESTINATION ${XDG_APPS_INSTALL_DIR})
     endif ()
-    target_link_libraries(cantata -lpthread)
+    target_link_libraries(cantata)
 endif ()
 
 configure_file(config.h.cmake ${CMAKE_BINARY_DIR}/config.h)
diff --git a/replaygain/CMakeLists.txt b/replaygain/CMakeLists.txt
index ae456ea..864066c 100644
--- a/replaygain/CMakeLists.txt
+++ b/replaygain/CMakeLists.txt
@@ -45,7 +45,7 @@ if (FFMPEG_FOUND OR MPG123_FOUND)
     endif ()
     target_link_libraries(cantata-replaygain ${QTCORELIBS})
     if (UNIX AND NOT APPLE)
-        target_link_libraries(cantata-replaygain -lpthread)
+        target_link_libraries(cantata-replaygain)
     endif ()
 endif ()
 
