--- resolv-0.1.2.0/cbits/hs_resolv.h	2001-09-09 07:16:40.000000000 +0530
+++ resolv-0.1.2.0-patch/cbits/hs_resolv.h	2022-02-03 20:54:49.602546255 +0530
@@ -84,12 +84,8 @@
 {
   assert(!s);
 
-  if (!(_res.options & RES_INIT)) {
     int rc = res_init();
     if (rc) return rc;
-  }
-
-  _res.options |= RES_USE_DNSSEC | RES_USE_EDNS0;
 
   return 0;
 }
@@ -112,7 +108,7 @@
 {
   assert(!s);
 
-  return res_send(msg, msglen, answer, anslen);
+  return -1;
 }
 
 inline static int
