diff -uNr samba-4.12.5/lib/util/util_paths.c samba-4.12.5.mod/lib/util/util_paths.c
--- samba-4.12.5/lib/util/util_paths.c	2020-07-02 14:12:21.000000000 +0300
+++ samba-4.12.5.mod/lib/util/util_paths.c	2020-07-19 14:38:37.598039736 +0300
@@ -64,6 +64,10 @@
 	return get_dyn_SHLIBEXT();
 }
 
+#ifndef NSS_BUFLEN_PASSWD
+#define NSS_BUFLEN_PASSWD 1024
+#endif
+
 static char *get_user_home_dir(TALLOC_CTX *mem_ctx)
 {
 	struct passwd pwd = {0};
