diff --git a/lib/util/debug.c b/lib/util/debug.c
index f79b881..cebab1a 100644
--- a/lib/util/debug.c
+++ b/lib/util/debug.c
@@ -1505,6 +1505,7 @@ void check_log_size( void )
 {
 	off_t maxlog;
 
+#ifndef __ANDROID__
 	if (geteuid() != 0) {
 		/*
 		 * We need to be root to change the log file (tests use a fake
@@ -1514,6 +1515,7 @@ void check_log_size( void )
 		 */
 		return;
 	}
+#endif
 
 	if(log_overflow || (!state.schedule_reopen_logs && !need_to_check_log_size())) {
 		return;
