diff --color -uNr samba-4.16.2/third_party/heimdal_build/et_compile_wrapper.sh samba-4.16.2.mod/third_party/heimdal_build/et_compile_wrapper.sh
--- samba-4.16.2/third_party/heimdal_build/et_compile_wrapper.sh	2022-01-24 07:27:00.621315500 -0300
+++ samba-4.16.2.mod/third_party/heimdal_build/et_compile_wrapper.sh	2022-06-23 16:45:46.258872083 -0300
@@ -43,6 +43,7 @@
 	mv "${SOURCE}" "${TMP}" && {
 		echo "#include \"config.h\"" > "${SOURCE}" && {
 			cat "${TMP}" >> "${SOURCE}"
+			sed -i 's/link/link_et/g' "${SOURCE}" || :
 		}
 	}
 	rm -f "${TMP}"
