diff --git a/source4/heimdal_build/et_compile_wrapper.sh b/source4/heimdal_build/et_compile_wrapper.sh
index 939b434..84f8615 100755
--- a/source4/heimdal_build/et_compile_wrapper.sh
+++ b/source4/heimdal_build/et_compile_wrapper.sh
@@ -43,6 +43,7 @@ cd "${DESTDIR}" && {
 	mv "${SOURCE}" "${TMP}" && {
 		echo "#include \"config.h\"" > "${SOURCE}" && {
 			cat "${TMP}" >> "${SOURCE}"
+			sed -i 's/link/link_et/g' "${SOURCE}" || :
 		}
 	}
 	rm -f "${TMP}"
