diff -u -r ../hub-2.8.3/script/build ./script/build
--- ../hub-2.8.3/script/build	2019-01-26 17:47:05.000000000 +0000
+++ ./script/build	2019-01-27 08:10:53.521236847 +0000
@@ -14,7 +14,7 @@
 build_hub() {
   mkdir -p "$(dirname "$1")"
   go build \
-	  -ldflags "-X github.com/github/hub/version.Version=`./script/version` $LDFLAGS" \
+	  -ldflags "-X github.com/github/hub/version.Version=`./script/version`" \
 	  -gcflags "$GCFLAGS" \
 	  -asmflags "$ASMFLAGS" \
 	  -o "$1"
