diff -uNr hub-2.12.1/script/build hub-2.12.1.mod/script/build
--- hub-2.12.1/script/build	2019-06-28 23:19:41.000000000 +0300
+++ hub-2.12.1.mod/script/build	2019-06-29 17:10:47.759789557 +0300
@@ -15,7 +15,7 @@
   mkdir -p "$(dirname "$1")"
   go build \
 	  $MOD_VENDOR_ARG \
-	  -ldflags "-X github.com/github/hub/version.Version=`./script/version` $LDFLAGS" \
+	  -ldflags "-X github.com/github/hub/version.Version=`./script/version`" \
 	  -gcflags "$GCFLAGS" \
 	  -asmflags "$ASMFLAGS" \
 	  -o "$1"
