diff -uNr dbus-1.12.16/dbus/dbus-sysdeps-unix.c dbus-1.12.16.mod/dbus/dbus-sysdeps-unix.c
--- dbus-1.12.16/dbus/dbus-sysdeps-unix.c	2019-05-13 12:33:56.000000000 +0300
+++ dbus-1.12.16.mod/dbus/dbus-sysdeps-unix.c	2019-08-15 02:15:37.825982953 +0300
@@ -3602,7 +3602,7 @@
 
       /* And this is the sane fallback. */
       if (tmpdir == NULL)
-        tmpdir = "/tmp";
+        tmpdir = "@TERMUX_PREFIX@/tmp";
     }
 
   _DBUS_UNLOCK (sysdeps);
