https://github.com/termux/termux-packages/issues/24739#issuecomment-2888679949

--- a/src/vs/workbench/contrib/extensions/browser/extensions.contribution.ts
+++ b/src/vs/workbench/contrib/extensions/browser/extensions.contribution.ts
@@ -265,7 +265,7 @@
 			[VerifyExtensionSignatureConfigKey]: {
 				type: 'boolean',
 				description: localize('extensions.verifySignature', "When enabled, extensions are verified to be signed before getting installed."),
-				default: true,
+				default: false,
 				scope: ConfigurationScope.APPLICATION,
 				included: isNative
 			},
