https://gitlab.freedesktop.org/wayland/wayland/-/merge_requests/438

--- a/src/wayland-util.h
+++ b/src/wayland-util.h
@@ -48,7 +48,8 @@
 #endif
 
 /** Deprecated attribute */
-#if __STDC_VERSION__ >= 202311L
+#if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 202311L \
+	|| defined(__cplusplus) && __cplusplus >= 201402L
 #define WL_DEPRECATED [[deprecated]]
 #elif defined(__GNUC__) && __GNUC__ >= 4
 #define WL_DEPRECATED __attribute__ ((deprecated))
@@ -70,7 +71,7 @@
 #define WL_PRINTF(x, y)
 #endif
 
-#if __STDC_VERSION__ >= 202311L
+#if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 202311L
 #define WL_TYPEOF(expr) typeof(expr)
 #else
 #define WL_TYPEOF(expr) __typeof__(expr)
