diff -u -r ../tmate-2.2.1/tmux.h ./tmux.h
--- ../tmate-2.2.1/tmux.h	2016-03-28 23:30:07.000000000 -0400
+++ ./tmux.h	2016-08-26 16:57:50.901952193 -0400
@@ -53,7 +53,7 @@
 struct tmuxproc;
 
 /* Default global configuration file. */
-#define TMUX_CONF "/etc/tmux.conf"
+#define TMUX_CONF "@TERMUX_PREFIX@/etc/tmux.conf"
 
 /*
  * Minimum layout cell size, NOT including separator line. The scroll region
