diff --git a/Xfce-Theme-Manager/src/globals.h b/Xfce-Theme-Manager/src/globals.h
index d09a124..931271b 100644
--- a/Xfce-Theme-Manager/src/globals.h
+++ b/Xfce-Theme-Manager/src/globals.h
@@ -42,9 +42,9 @@
 
 #define DELETETHEME 100
 
-#define GLOBALICONS "/usr/share/icons"
-#define GLOBALTHEMES "/usr/share/themes"
-#define GLOBALWALLPAPERS "/usr/share/backgrounds/xfce"
+#define GLOBALICONS "@TERMUX_PREFIX@/share/icons"
+#define GLOBALTHEMES "@TERMUX_PREFIX@/share/themes"
+#define GLOBALWALLPAPERS "@TERMUX_PREFIX@/share/backgrounds/xfce"
 
 //new props defines
 #define XSETTINGS "xsettings"
