diff -u -r ../espeak-ng-1.49.1/src/include/compat/stdint.h ./src/include/compat/stdint.h
--- ../espeak-ng-1.49.1/src/include/compat/stdint.h	2017-01-21 21:57:13.000000000 +0100
+++ ./src/include/compat/stdint.h	2017-02-01 00:28:46.687277703 +0100
@@ -19,6 +19,8 @@
 #ifndef STDINT_H_COMPAT_SHIM
 #define STDINT_H_COMPAT_SHIM
 
+#include "config.h"
+
 #ifdef _MSC_VER
 
 #if _MSC_VER >= 1600 // Visual C++ 10 (Visual Studio 2010) and above...
