diff -u -r ../espeak-1.48.01-source/src/Makefile ./src/Makefile
--- ../espeak-1.48.01-source/src/Makefile	2014-02-02 11:58:11.000000000 +0100
+++ ./src/Makefile	2014-02-11 22:18:52.000000000 +0100
@@ -1,8 +1,8 @@
 # remove -fno-exceptions  if it causes problems with stdio library on some platforms (ARM)
 
-DATADIR=/usr/share/espeak-data
 
-PREFIX=/usr
+PREFIX=@TERMUX_PREFIX@
+DATADIR=$(PREFIX)/share/espeak
 BINDIR=$(PREFIX)/bin
 INCDIR=$(PREFIX)/include/espeak
 LIBDIR=$(PREFIX)/lib
@@ -27,7 +27,7 @@
 
 # 'runtime' uses pulseaudio if it is running, else uses portaudio
 #AUDIO = runtime
-AUDIO = portaudio
+#AUDIO = portaudio
 #AUDIO = portaudio0
 #AUDIO = portaudio2
 #AUDIO = pulseaudio
@@ -83,11 +83,11 @@
 
 SRCS1=$(speak_SOURCES)
 OBJS1=$(patsubst %.cpp,%.o,$(SRCS1))
-LIBS1=-lstdc++ $(LIB_AUDIO) -lpthread $(EXTRA_LIBS)
+LIBS1=-lstdc++ $(LIB_AUDIO) $(EXTRA_LIBS)
 
 SRCS2=$(libespeak_SOURCES)
 OBJS2=$(patsubst %.cpp,x_%.o,$(SRCS2))
-LIBS2=-lstdc++ $(LIB_AUDIO) -lpthread
+LIBS2=-lstdc++ $(LIB_AUDIO)
 
 SRCS3 = espeak.cpp
 OBJS3=$(patsubst %.cpp,%.o,$(SRCS3))
