diff -u -r ../espeak-ng-1.49.1/Makefile.am ./Makefile.am
--- ../espeak-ng-1.49.1/Makefile.am	2017-01-21 21:57:13.000000000 +0100
+++ ./Makefile.am	2017-02-01 00:30:07.810286033 +0100
@@ -121,7 +121,7 @@
 
 lib_LTLIBRARIES += src/libespeak-ng.la
 
-src_libespeak_ng_la_LDFLAGS = -version-info $(SHARED_VERSION) -lpthread -lm \
+src_libespeak_ng_la_LDFLAGS = -version-info $(SHARED_VERSION) -lm \
 	${PCAUDIOLIB_LIBS}
 
 src_libespeak_ng_la_CFLAGS = -Isrc/include -Isrc/include/compat \
diff -u -r ../espeak-ng-1.49.1/Makefile.in ./Makefile.in
--- ../espeak-ng-1.49.1/Makefile.in	2017-01-21 21:57:17.000000000 +0100
+++ ./Makefile.in	2017-02-01 00:30:24.202085646 +0100
@@ -472,7 +472,7 @@
 	src/include/espeak-ng/espeak_ng.h \
 	src/include/espeak-ng/speak_lib.h
 
-src_libespeak_ng_la_LDFLAGS = -version-info $(SHARED_VERSION) -lpthread -lm \
+src_libespeak_ng_la_LDFLAGS = -version-info $(SHARED_VERSION) -lm \
 	${PCAUDIOLIB_LIBS}
 
 src_libespeak_ng_la_CFLAGS = -Isrc/include -Isrc/include/compat -fPIC \
