diff -u -r ../espeak-1.48.01-source/src/speech.h ./src/speech.h
--- ../espeak-1.48.01-source/src/speech.h	2014-02-02 11:52:30.000000000 +0100
+++ ./src/speech.h	2014-02-11 22:16:17.000000000 +0100
@@ -47,12 +47,12 @@
 //#define ESPEAK_API  extern "C"
 
 #ifdef LIBRARY
-#define USE_ASYNC
+//#define USE_ASYNC
 #endif
 
 #ifdef _ESPEAKEDIT
 #define USE_PORTAUDIO
-#define USE_ASYNC
+//#define USE_ASYNC
 #define LOG_FRAMES      // write keyframe info to log-espeakedit
 #endif
 
