diff --git a/CLI/Repl.cpp b/CLI/Repl.cpp
index 814d7c8..e04298f 100644
--- a/CLI/Repl.cpp
+++ b/CLI/Repl.cpp
@@ -781,9 +781,9 @@ int replMain(int argc, char** argv)
 
     if (codegenPerf)
     {
-#if __linux__
+#if __ANDROID__
         char path[128];
-        snprintf(path, sizeof(path), "/tmp/perf-%d.map", getpid());
+        snprintf(path, sizeof(path), "@TERMUX_PREFIX@/tmp/perf-%d.map", getpid());
 
         // note, there's no need to close the log explicitly as it will be closed when the process exits
         FILE* codegenPerfLog = fopen(path, "w");
