diff -uNr fluxbox-1.3.7/util/fluxbox-generate_menu.in fluxbox-1.3.7.mod/util/fluxbox-generate_menu.in
--- fluxbox-1.3.7/util/fluxbox-generate_menu.in	2015-02-08 12:44:45.377187009 +0200
+++ fluxbox-1.3.7.mod/util/fluxbox-generate_menu.in	2018-10-17 17:15:44.175731942 +0300
@@ -1509,7 +1509,7 @@
 your PATH or specify your favourite terminal-emulator with the -t option
 
 EOF
-    DEFAULT_TERM=xterm
+    DEFAULT_TERM=aterm
 fi
 
 DEFAULT_TERMNAME=`echo $DEFAULT_TERM|awk '{print $1}'`
@@ -1533,7 +1533,7 @@
 if [ "x$DEFAULT_BROWSERNAME" != "x" ]; then
 	DEFAULT_BROWSERNAME=`basename $DEFAULT_BROWSERNAME`
 else
-	DEFAULT_BROWSERNAME="firefox"
+	DEFAULT_BROWSERNAME="lynx"
 fi
 
 if [ -z "$LAUNCHER" ]; then
