diff -uNr libgit2-1.5.0/src/libgit2/sysdir.c libgit2-1.5.0.mod/src/libgit2/sysdir.c
--- libgit2-1.5.0/src/libgit2/sysdir.c	2022-07-14 08:06:59.000000000 +0800
+++ libgit2-1.5.0.mod/src/libgit2/sysdir.c	2022-07-14 09:11:02.940851706 +0800
@@ -33,7 +33,7 @@
 #ifdef GIT_WIN32
 	return git_win32__find_system_dirs(out, "etc");
 #else
-	return git_str_sets(out, "/etc");
+	return git_str_sets(out, "@TERMUX_PREFIX@/etc");
 #endif
 }
 
