diff -uNr libgit2-1.5.0/src/util/rand.c libgit2-1.5.0.mod/src/util/rand.c
--- libgit2-1.5.0/src/util/rand.c	2022-07-14 08:06:59.000000000 +0800
+++ libgit2-1.5.0.mod/src/util/rand.c	2022-07-14 09:27:21.190851333 +0800
@@ -23,6 +23,9 @@
 } bits;
 
 #if defined(GIT_WIN32)
+#if defined __ANDROID__ && __ANDROID_API__ < 29
+int getloadavg(double[], int);
+#endif
 GIT_INLINE(int) getseed(uint64_t *seed)
 {
 	HCRYPTPROV provider;
