--- a/winpr/libwinpr/utils/trio/trio.c
+++ b/winpr/libwinpr/utils/trio/trio.c
@@ -127,7 +127,7 @@
 
 #if defined(__STDC_ISO_10646__) || defined(MB_LEN_MAX) || defined(USE_MULTIBYTE) || \
     TRIO_FEATURE_WIDECHAR
-#if (!defined(TRIO_PLATFORM_WINCE) && !defined(ANDROID))
+#if (!defined(TRIO_PLATFORM_WINCE) && !defined(__ANDROID__))
 #define TRIO_COMPILER_SUPPORTS_MULTIBYTE
 #if !defined(MB_LEN_MAX)
 #define MB_LEN_MAX 6
