diff -u -r ../leptonica-1.74.2/src/recogbasic.c ./src/recogbasic.c
--- ../leptonica-1.74.2/src/recogbasic.c	2017-05-20 03:30:27.000000000 +0200
+++ ./src/recogbasic.c	2017-06-07 01:17:15.994340419 +0200
@@ -1065,7 +1065,6 @@
         return ERROR_INT("stream not opened", procName, 1);
     ret = recogWriteStream(fp, recog);
 #else
-    L_INFO("work-around: writing to a temp file\n", procName);
   #ifdef _WIN32
     if ((fp = fopenWriteWinTempfile()) == NULL)
         return ERROR_INT("tmpfile stream not opened", procName, 1);
