# TODO: Remove this after new Ubuntu 24.04 provides autoconf 2.72
--- html-xml-utils-8.8/configure.ac	2026-06-09 23:32:08.000000000 +0800
+++ html-xml-utils-8.8.mod/configure.ac	2026-06-10 20:40:37.842962804 +0800
@@ -1,6 +1,6 @@
 #                                               -*- Autoconf -*-
 # Process this file with autoconf to produce a configure script.
-AC_PREREQ([2.72])
+AC_PREREQ([2.71])
 AC_INIT([html-xml-utils],[8.8])
 dnl print all automake warnings with -Wall
 dnl http://sources.redhat.com/automake/automake.html#Options
