diff -uNr html-xml-utils-7.9/hxnum.c html-xml-utils-7.9.mod/hxnum.c
--- html-xml-utils-7.9/hxnum.c	2018-08-02 01:21:37.000000000 +0000
+++ html-xml-utils-7.9.mod/hxnum.c	2020-11-05 23:15:40.166574007 +0000
@@ -38,6 +38,8 @@
 #include "openurl.e"
 #include "errexit.e"
 
+#include <getopt.h>
+
 #define SECNO "secno"				/* class attribute */
 #define NO_NUM "no-num"				/* class-attribute */
 #define OPTS ":l:h:n:1:2:3:4:5:6:?"		/* Command line options */
