diff -uNr html-xml-utils-7.9/Makefile.am html-xml-utils-7.9.mod/Makefile.am
--- html-xml-utils-7.9/Makefile.am	2020-08-04 20:34:12.000000000 +0000
+++ html-xml-utils-7.9.mod/Makefile.am	2020-11-05 23:34:14.389537863 +0000
@@ -51,7 +51,7 @@
 			fopencookie.h\
 			class.c hash.c headers.c dict.c fopencookie.c
 cexport_SOURCES =	cexport.c
-hxcite_SOURCES =	heap.c errexit.c hxcite.c
+hxcite_SOURCES =	heap.c errexit.c hxcite.c hash.c
 hxcount_SOURCES =	hxcount.c html.y scan.l types.c errexit.c heap.c\
 			openurl.c url.c connectsock.c headers.c dict.c\
 			fopencookie.h fopencookie.c
@@ -132,6 +132,7 @@
 SUFFIXES =		.1 .e .html
 
 $(EXPORTS): $(cexport_SOURCES)
+	gcc -I. $(cexport_SOURCES) -o cexport
 
 .c.e:
 	$(MAKE) cexport
