diff -urN olsrd-0.6.8+termux/lib/pud/src/netTools.h olsrd-0.6.8+in_port_t/lib/pud/src/netTools.h
--- olsrd-0.6.8+termux/lib/pud/src/netTools.h	2014-12-29 11:17:09.000000000 +0000
+++ olsrd-0.6.8+in_port_t/lib/pud/src/netTools.h	2016-09-26 01:47:47.717744923 +0000
@@ -15,7 +15,7 @@
 
 
 #ifdef __ANDROID__
-typedef __in_port_t in_port_t;
+typedef unsigned short int in_port_t;
 #endif /* __ANDROID_API__ */
 
 /**
