# Although it just builds, still disable it.
--- userspace-rcu-0.15.0/Makefile.in	2024-12-19 19:49:26.000000000 +0530
+++ userspace-rcu-0.15.0.mod/Makefile.in	2025-01-23 16:33:27.627942132 +0530
@@ -400,7 +400,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = include src doc tests extras
+SUBDIRS = include src doc extras
 dist_doc_DATA = \
 	LICENSE.md \
 	README.md
