--- a/frontend/libpicofe/menu.c	2023-01-14 18:47:37.198748509 -0300
+++ b/frontend/libpicofe/menu.c	2023-01-14 21:01:15.266742774 -0300
@@ -313,7 +313,7 @@
 	}
 
 	// load custom font and selector (stored as 1st symbol in font table)
-	pos = plat_get_skin_dir(buff, sizeof(buff));
+	pos = plat_get_skin_dir(buff);
 	strcpy(buff + pos, "font.png");
 	readpng(menu_font_data, buff, READPNG_FONT,
 		MENU_X2 ? 256 : 128, MENU_X2 ? 320 : 160);
