--- a/frontend/libpicofe/plat.h	2023-01-14 16:15:27.861589338 -0300
+++ b/frontend/libpicofe/plat.h	2023-01-15 03:29:10.852522982 -0300
@@ -101,11 +101,8 @@
 void plat_video_flip(void);
 void plat_video_wait_vsync(void);
 
-/* return the dir/ where configs, saves, bios, etc. are found */
-int  plat_get_root_dir(char *dst, int len);
-
 /* return the dir/ where skin files are found */
-int  plat_get_skin_dir(char *dst, int len);
+int  plat_get_skin_dir(char *dst);
 
 /* return the top level dir for image files */
 int  plat_get_data_dir(char *dst, int len);
