--- a/frontend/main.h	2023-01-14 16:14:07.969589395 -0300
+++ b/frontend/main.h	2023-01-14 16:51:46.017587780 -0300
@@ -21,16 +21,17 @@
 
 #include "config.h"
 
-#define DEFAULT_MEM_CARD_1 "/.pcsx/memcards/card1.mcd"
-#define DEFAULT_MEM_CARD_2 "/.pcsx/memcards/card2.mcd"
-#define MEMCARD_DIR "/.pcsx/memcards/"
-#define PLUGINS_DIR "/.pcsx/plugins/"
-#define PLUGINS_CFG_DIR "/.pcsx/plugins/cfg/"
-#define PCSX_DOT_DIR "/.pcsx/"
-#define STATES_DIR "/.pcsx/sstates/"
-#define CHEATS_DIR "/.pcsx/cheats/"
-#define PATCHES_DIR "/.pcsx/patches/"
-#define BIOS_DIR "/bios/"
+#define DEFAULT_MEM_CARD_1 "@TERMUX_PREFIX@/etc/pcsx/memcards/card1.mcd"
+#define DEFAULT_MEM_CARD_2 "@TERMUX_PREFIX@/etc/pcsx/memcards/card2.mcd"
+#define MEMCARD_DIR "@TERMUX_PREFIX@/etc/pcsx/memcards/"
+#define PLUGINS_DIR "@TERMUX_PREFIX@/etc/pcsx/plugins/"
+#define PLUGINS_CFG_DIR "@TERMUX_PREFIX@/etc/pcsx/plugins/cfg/"
+#define PCSX_DOT_DIR "@TERMUX_PREFIX@/etc/pcsx/"
+#define STATES_DIR "@TERMUX_PREFIX@/etc/pcsx/sstates/"
+#define CHEATS_DIR "@TERKUX_PREFIX@/etc/pcsx/cheats/"
+#define PATCHES_DIR "@TERMUX_PREFIX@/etc/pcsx/patches/"
+#define BIOS_DIR "@TERMUX_PREFIX@/etc/pcsx/bios/"
+#define SCREENSHOTS_DIR "@TERMUX_PREFIX@/etc/pcsx/screenshots/"
 
 extern char cfgfile_basename[MAXPATHLEN];
 
