diff -u -r texmf-dist/scripts/crossrefware.orig/bbl2bib.pl texmf-dist/scripts/crossrefware/bbl2bib.pl
--- texmf-dist/scripts/crossrefware.orig/bbl2bib.pl	2022-05-01 22:05:42.101394014 +0200
+++ texmf-dist/scripts/crossrefware/bbl2bib.pl	2022-05-01 22:08:14.421393905 +0200
@@ -98,7 +98,7 @@
 use strict;
 BEGIN {
     # find files relative to our installed location within TeX Live
-    chomp(my $TLMaster = `kpsewhich -var-value=SELFAUTOPARENT`); # TL root
+    chomp(my $TLMaster = `kpsewhich -var-value=TEXMFROOT`); # TL root
     if (length($TLMaster)) {
 	unshift @INC, "$TLMaster/texmf-dist/scripts/bibtexperllibs";
     }
diff -u -r texmf-dist/scripts/crossrefware.orig/bibdoiadd.pl texmf-dist/scripts/crossrefware/bibdoiadd.pl
--- texmf-dist/scripts/crossrefware.orig/bibdoiadd.pl	2022-05-01 22:05:42.101394014 +0200
+++ texmf-dist/scripts/crossrefware/bibdoiadd.pl	2022-05-01 22:08:14.421393905 +0200
@@ -97,7 +97,7 @@
 use strict;
 BEGIN {
     # find files relative to our installed location within TeX Live
-    chomp(my $TLMaster = `kpsewhich -var-value=SELFAUTOPARENT`); # TL root
+    chomp(my $TLMaster = `kpsewhich -var-value=TEXMFROOT`); # TL root
     if (length($TLMaster)) {
 	unshift @INC, "$TLMaster/texmf-dist/scripts/bibtexperllibs";
     }
diff -u -r texmf-dist/scripts/crossrefware.orig/bibmradd.pl texmf-dist/scripts/crossrefware/bibmradd.pl
--- texmf-dist/scripts/crossrefware.orig/bibmradd.pl	2022-05-01 22:05:42.105394014 +0200
+++ texmf-dist/scripts/crossrefware/bibmradd.pl	2022-05-01 22:08:14.421393905 +0200
@@ -64,7 +64,7 @@
 use strict;
 BEGIN {
     # find files relative to our installed location within TeX Live
-    chomp(my $TLMaster = `kpsewhich -var-value=SELFAUTOPARENT`); # TL root
+    chomp(my $TLMaster = `kpsewhich -var-value=TEXMFROOT`); # TL root
     if (length($TLMaster)) {
 	unshift @INC, "$TLMaster/texmf-dist/scripts/bibtexperllibs";
     }
diff -u -r texmf-dist/scripts/crossrefware.orig/biburl2doi.pl texmf-dist/scripts/crossrefware/biburl2doi.pl
--- texmf-dist/scripts/crossrefware.orig/biburl2doi.pl	2022-05-01 22:05:42.105394014 +0200
+++ texmf-dist/scripts/crossrefware/biburl2doi.pl	2022-05-01 22:08:14.425393905 +0200
@@ -51,7 +51,7 @@
 use strict;
 BEGIN {
     # find files relative to our installed location within TeX Live
-    chomp(my $TLMaster = `kpsewhich -var-value=SELFAUTOPARENT`); # TL root
+    chomp(my $TLMaster = `kpsewhich -var-value=TEXMFROOT`); # TL root
     if (length($TLMaster)) {
 	unshift @INC, "$TLMaster/texmf-dist/scripts/bibtexperllibs";
     }
diff -u -r texmf-dist/scripts/crossrefware.orig/bibzbladd.pl texmf-dist/scripts/crossrefware/bibzbladd.pl
--- texmf-dist/scripts/crossrefware.orig/bibzbladd.pl	2022-05-01 22:05:42.109394014 +0200
+++ texmf-dist/scripts/crossrefware/bibzbladd.pl	2022-05-01 22:08:14.425393905 +0200
@@ -64,7 +64,7 @@
 use strict;
 BEGIN {
     # find files relative to our installed location within TeX Live
-    chomp(my $TLMaster = `kpsewhich -var-value=SELFAUTOPARENT`); # TL root
+    chomp(my $TLMaster = `kpsewhich -var-value=TEXMFROOT`); # TL root
     if (length($TLMaster)) {
 	unshift @INC, "$TLMaster/texmf-dist/scripts/bibtexperllibs";
     }
diff -u -r texmf-dist/scripts/crossrefware.orig/ltx2crossrefxml.pl texmf-dist/scripts/crossrefware/ltx2crossrefxml.pl
--- texmf-dist/scripts/crossrefware.orig/ltx2crossrefxml.pl	2022-05-01 22:05:42.109394014 +0200
+++ texmf-dist/scripts/crossrefware/ltx2crossrefxml.pl	2022-05-01 22:08:14.425393905 +0200
@@ -265,7 +265,7 @@
 
  BEGIN {
      # find files relative to our installed location within TeX Live
-     chomp(my $TLMaster = `kpsewhich -var-value=SELFAUTOPARENT`); # TL root
+     chomp(my $TLMaster = `kpsewhich -var-value=TEXMFROOT`); # TL root
      if (length($TLMaster)) {
 	 unshift @INC, "$TLMaster/texmf-dist/scripts/bibtexperllibs";
      }
