--- ./texmf-dist/scripts/pdfjam/pdfjam.orig	2022-05-01 22:13:53.209393663 +0200
+++ ./texmf-dist/scripts/pdfjam/pdfjam	2022-05-01 22:14:19.829393644 +0200
@@ -21,7 +21,7 @@
 ##  The path searched for site-wide configuration files can be set     ##
 ##  by editing the following variable:                                 ##
 ##                                                                     ##
-    configpath='/etc:/usr/share/etc:/usr/local/share:/usr/local/etc'   ##
+    configpath='@TERMUX_PREFIX@/etc:@TERMUX_PREFIX@/share/etc:/usr/local/share:/usr/local/etc'   ##
 ##                                                                     ##
 ##  Nothing else in this file should need to be changed.               ##
 ##                                                                     ##
@@ -361,7 +361,7 @@
 ##
 ##  Next a permitted location for temporary files on your system:
 ##
-tempfileDir='/var/tmp'   ##  /var/tmp is standard on most unix systems
+tempfileDir='@TERMUX_PREFIX@/tmp'   ##  /var/tmp is standard on most unix systems
 ##
 ##
 ##  Default for the output file location:
