--- ../pedigree.pl.orig	2020-05-01 17:26:30.863222262 +0200
+++ ./texmf-dist/scripts/pedigree-perl/pedigree.pl	2020-05-01 17:28:24.936404899 +0200
@@ -110,7 +110,7 @@
 
 BEGIN {
     # find files relative to our installed location within TeX Live
-    chomp(my $TLMaster = `kpsewhich -var-value=SELFAUTOPARENT`); # TL root
+    chomp(my $TLMaster = `kpsewhich -var-value=TEXMFROOT`); # TL root
     if (length($TLMaster)) {
 	unshift @INC, "$TLMaster/texmf-dist/scripts/pedigree-perl";
 	$TLCONF = "$TLMaster/texmf-config/pedigree/pedigree.cfg";
