--- ../texconfig.sh.orig	2020-04-04 09:33:30.373390496 +0200
+++ ./texmf-dist/scripts/texlive-extra/texconfig.sh	2020-04-04 09:34:14.749963681 +0200
@@ -53,8 +53,8 @@
   case $FMT in
     "") 
       FMT=fmt
-      test ! -x /bin/fmt && test ! -f /usr/bin/fmt &&
-        { test -x /bin/adjust || test -x /usr/bin/adjust; } && FMT=adjust
+      test ! -x /bin/fmt && test ! -f @TERMUX_PREFIX@/bin/fmt &&
+        { test -x /bin/adjust || test -x @TERMUX_PREFIX@/bin/adjust; } && FMT=adjust
       ;;
     *)
       return
