--- ./texmf-dist/scripts/texdef/texdef.pl.orig	2020-09-28 23:01:50.000000000 +0200
+++ ./texmf-dist/scripts/texdef/texdef.pl	2022-05-01 22:22:11.349393307 +0200
@@ -284,14 +284,14 @@
             $EDITOR = 'texworks "%f"';
         }
         else {
-            for my $ed (qw(/usr/bin/vim /usr/bin/emacs /usr/bin/nano)) {
+            for my $ed (qw(@TERMUX_PREFIX@/bin/vim @TERMUX_PREFIX@/bin/emacs @TERMUX_PREFIX@/bin/nano)) {
                 if (-x $ed) {
                     $EDITOR = $ed;
                     last;
                 }
             }
             if (!$EDITOR) {
-                for my $ed (qw(/usr/bin/editor /usr/bin/open /bin/open)) {
+                for my $ed (qw(@TERMUX_PREFIX@/bin/editor @TERMUX_PREFIX@/bin/open @TERMUX_PREFIX@/bin/open)) {
                     if (-x $ed) {
                         $EDITOR = "$ed \"%f\"";
                         last;
