--- ./texmf-dist/scripts/texlive/tlmgr.pl.orig	2022-05-01 22:25:19.677393172 +0200
+++ ./texmf-dist/scripts/texlive/tlmgr.pl	2022-05-01 22:25:52.269393149 +0200
@@ -65,7 +65,7 @@
   }
   if (-r "$bindir/$kpsewhichname") {
     # if not in bootstrapping mode => kpsewhich exists, so use it to get $Master
-    chomp($Master = `kpsewhich -var-value=SELFAUTOPARENT`);
+    chomp($Master = `kpsewhich -var-value=TEXMFROOT`);
   }
 
   # if we have no directory in which to find our modules,
