--- ./tlpkg/TeXLive/TLUtils.pm.orig	2022-05-01 20:24:48.919325797 +0200
+++ ./tlpkg/TeXLive/TLUtils.pm	2022-05-01 20:30:27.652807200 +0200
@@ -1871,7 +1871,7 @@
   }
   my $cmd = $keyval{'cmd'};
 
-  my $texdir = `kpsewhich -var-value=TEXMFROOT`;
+  my $texdir = "@TERMUX_PREFIX@/share/texlive";
   chomp($texdir);
   my $texdir_bsl = conv_to_w32_path($texdir);
   $cmd =~ s!^("?)TEXDIR/!$1$texdir/!g;
@@ -1946,7 +1946,7 @@
   if (win32() && defined($keyval{'filew32'})) {
     $file = $keyval{'filew32'};
   }
-  my $texdir = `kpsewhich -var-value=TEXMFROOT`;
+  my $texdir = "@TERMUX_PREFIX@/share/texlive";
   chomp($texdir);
   my @syscmd;
   if ($file =~ m/\.pl$/i) {
@@ -2021,7 +2021,7 @@
 
   &log("postaction $how shortcut for " . $tlpobj->name . "\n");
   if ($how eq "install") {
-    my $texdir = `kpsewhich -var-value=TEXMFROOT`;
+    my $texdir = "@TERMUX_PREFIX@/share/texlive";
     chomp($texdir);
     my $texdir_bsl = conv_to_w32_path($texdir);
     $icon =~ s!^TEXDIR/!$texdir/!;
