# Build demos as programs like in Linux

--- a/demo/meson.build
+++ b/demo/meson.build
@@ -69,7 +69,7 @@
 # This results in a warning, that android_exe_type is only available
 # starting with meson >= 1.8.0. As there are some jobs running with
 # fatal warnings, put this behind a host_machine is android guard too.
-if meson.version().version_compare('>= 1.8.0') and host_machine.system() == 'android'
+if false and meson.version().version_compare('>= 1.8.0')
   demo_kwargs += { 'android_exe_type': 'application' }
 endif
 
