diff -uNr hwinfo-21.60/Makefile hwinfo-21.60.mod/Makefile
--- hwinfo-21.60/Makefile	2018-10-17 16:23:47.000000000 +0300
+++ hwinfo-21.60.mod/Makefile	2019-06-29 01:57:27.596632042 +0300
@@ -22,11 +22,7 @@
 
 include Makefile.common
 
-ifeq "$(ARCH)" "x86_64"
-LIBDIR		?= /usr/lib64
-else
-LIBDIR		?= /usr/lib
-endif
+LIBDIR		= /lib
 ULIBDIR		= $(LIBDIR)
 
 # ia64
@@ -102,26 +98,26 @@
 	@cd doc ; doxygen libhd.doxy
 
 install:
-	install -d -m 755 $(DESTDIR)/sbin $(DESTDIR)/usr/sbin $(DESTDIR)$(ULIBDIR) \
-		$(DESTDIR)$(ULIBDIR)/pkgconfig $(DESTDIR)/usr/include
-	install -m 755 hwinfo $(DESTDIR)/usr/sbin
-	install -m 755 src/ids/check_hd $(DESTDIR)/usr/sbin
-	install -m 755 src/ids/convert_hd $(DESTDIR)/usr/sbin
+	install -d -m 755 $(DESTDIR)/bin $(DESTDIR)$(ULIBDIR) \
+		$(DESTDIR)$(ULIBDIR)/pkgconfig $(DESTDIR)/include
+	install -m 755 hwinfo $(DESTDIR)/bin/
+	install -m 755 src/ids/check_hd $(DESTDIR)/bin/
+	install -m 755 src/ids/convert_hd $(DESTDIR)/bin/
 	if [ -f $(LIBHD_SO) ] ; then \
-		install $(LIBHD_SO) $(DESTDIR)$(ULIBDIR) ; \
+		install $(LIBHD_SO) $(DESTDIR)$(ULIBDIR)/ ; \
 		ln -snf $(LIBHD_NAME) $(DESTDIR)$(ULIBDIR)/$(LIBHD_SONAME) ; \
 		ln -snf $(LIBHD_SONAME) $(DESTDIR)$(ULIBDIR)/$(LIBHD_BASE).so ; \
 	else \
 		install -m 644 $(LIBHD) $(DESTDIR)$(ULIBDIR) ; \
 	fi
-	install -m 644 hwinfo.pc $(DESTDIR)$(ULIBDIR)/pkgconfig
-	install -m 644 src/hd/hd.h $(DESTDIR)/usr/include
-	install -m 755 getsysinfo $(DESTDIR)/usr/sbin
-	install -m 755 src/isdn/cdb/mk_isdnhwdb $(DESTDIR)/usr/sbin
-	install -d -m 755 $(DESTDIR)/usr/share/hwinfo
+	install -m 644 hwinfo.pc $(DESTDIR)$(ULIBDIR)/pkgconfig/
+	install -m 644 src/hd/hd.h $(DESTDIR)/include/
+	install -m 755 getsysinfo $(DESTDIR)/bin/
+	install -m 755 src/isdn/cdb/mk_isdnhwdb $(DESTDIR)/bin/
+	install -d -m 755 $(DESTDIR)/share/hwinfo
 	install -d -m 755 $(DESTDIR)/var/lib/hardware/udi
-	install -m 644 src/isdn/cdb/ISDN.CDB.txt $(DESTDIR)/usr/share/hwinfo
-	install -m 644 src/isdn/cdb/ISDN.CDB.hwdb $(DESTDIR)/usr/share/hwinfo
+	install -m 644 src/isdn/cdb/ISDN.CDB.txt $(DESTDIR)/share/hwinfo
+	install -m 644 src/isdn/cdb/ISDN.CDB.hwdb $(DESTDIR)/share/hwinfo
 
 archive: changelog
 	@if [ ! -d .git ] ; then echo no git repo ; false ; fi
