diff -uNr hwinfo-21.60/src/isdn/cdb/Makefile hwinfo-21.60.mod/src/isdn/cdb/Makefile
--- hwinfo-21.60/src/isdn/cdb/Makefile	2018-10-17 16:23:47.000000000 +0300
+++ hwinfo-21.60.mod/src/isdn/cdb/Makefile	2019-06-25 22:54:25.404041321 +0300
@@ -1,6 +1,6 @@
 TOPDIR		= ../../..
-TARGETS		= mk_isdnhwdb isdn_cdb.h ISDN.CDB.hwdb
-CLEANFILES	= isdn_cdb isdn_cdb.h lex.yy.c ISDN.CDB.txt mk_isdnhwdb ISDN.CDB.hwdb
+TARGETS		= mk_isdnhwdb mk_isdnhwdb_host isdn_cdb.h ISDN.CDB.hwdb
+CLEANFILES	= isdn_cdb isdn_cdb_host isdn_cdb.h lex.yy.c ISDN.CDB.txt mk_isdnhwdb ISDN.CDB.hwdb
 
 include $(TOPDIR)/Makefile.common
 
@@ -11,23 +11,40 @@
 		ln -s ISDN.ppc.txt ISDN.CDB.txt; \
 	fi
 
-isdn_cdb.h: isdn_cdb ISDN.CDB.txt
+isdn_cdb.h: isdn_cdb_host ISDN.CDB.txt
 	rm -f isdn_cdb.h
-	./isdn_cdb ISDN.CDB.txt isdn_cdb.h
+	./isdn_cdb_host ISDN.CDB.txt isdn_cdb.h
 
 lex.yy.c: isdn_cdb.lex
 	lex isdn_cdb.lex
 
 isdn_cdb: isdn_cdb.o cdb_read.o
 
+isdn_cdb_host: isdn_cdb_host.o cdb_read_host.o
+	gcc isdn_cdb_host.o cdb_read_host.o -o isdn_cdb_host
+
 mk_isdnhwdb: mk_isdnhwdb.o cdb_read.o
 
+mk_isdnhwdb_host: mk_isdnhwdb_host.o cdb_read_host.o
+	gcc mk_isdnhwdb_host.o cdb_read_host.o -o mk_isdnhwdb_host
+
 cdb_read.o: cdb_read.c cdb_read.h
 
+cdb_read_host.o: cdb_read.c cdb_read.h
+	gcc -O2 -I../../hd -c cdb_read.c -o cdb_read_host.o
+
 isdn_cdb.o: isdn_cdb.c cdb_read.h lex.yy.c
 
+isdn_cdb_host.o: isdn_cdb.c cdb_read.h lex.yy.c
+	gcc -O2 -I../../hd -c isdn_cdb.c -o isdn_cdb_host.o
+	gcc -O2 -I../../hd -c lex.yy.c -o lex.yy.host.o
+
 mk_isdnhwdb.o: mk_isdnhwdb.c cdb_read.h lex.yy.c
 
-ISDN.CDB.hwdb: mk_isdnhwdb ISDN.CDB.txt
+mk_isdnhwdb_host.o: mk_isdnhwdb.c cdb_read.h lex.yy.c
+	gcc -O2 -I../../hd -c mk_isdnhwdb.c -o mk_isdnhwdb_host.o
+	gcc -O2 -I../../hd -c lex.yy.c -o lex.yy.host.o
+
+ISDN.CDB.hwdb: mk_isdnhwdb_host ISDN.CDB.txt
 	rm -f  ISDN.CDB.hwdb
-	./mk_isdnhwdb ISDN.CDB.txt ISDN.CDB.hwdb
+	./mk_isdnhwdb_host ISDN.CDB.txt ISDN.CDB.hwdb
