diff -uNr xtrans-1.3.5/Xtranssock.c xtrans-1.3.5.mod/Xtranssock.c
--- xtrans-1.3.5/Xtranssock.c	2014-09-12 21:32:47.000000000 +0300
+++ xtrans-1.3.5.mod/Xtranssock.c	2018-06-15 19:11:49.445727641 +0300
@@ -208,28 +208,28 @@
 
 
 #if defined(X11_t)
-#define UNIX_PATH "/tmp/.X11-unix/X"
-#define UNIX_DIR "/tmp/.X11-unix"
+#define UNIX_PATH "/data/data/com.termux/files/usr/tmp/.X11-unix/X"
+#define UNIX_DIR "/data/data/com.termux/files/usr/tmp/.X11-unix"
 #endif /* X11_t */
 #if defined(XIM_t)
-#define UNIX_PATH "/tmp/.XIM-unix/XIM"
-#define UNIX_DIR "/tmp/.XIM-unix"
+#define UNIX_PATH "/data/data/com.termux/files/usr/tmp/.XIM-unix/XIM"
+#define UNIX_DIR "/data/data/com.termux/files/usr/tmp/.XIM-unix"
 #endif /* XIM_t */
 #if defined(FS_t) || defined(FONT_t)
-#define UNIX_PATH "/tmp/.font-unix/fs"
-#define UNIX_DIR "/tmp/.font-unix"
+#define UNIX_PATH "/data/data/com.termux/files/usr/tmp/.font-unix/fs"
+#define UNIX_DIR "/data/data/com.termux/files/usr/tmp/.font-unix"
 #endif /* FS_t || FONT_t */
 #if defined(ICE_t)
-#define UNIX_PATH "/tmp/.ICE-unix/"
-#define UNIX_DIR "/tmp/.ICE-unix"
+#define UNIX_PATH "/data/data/com.termux/files/usr/tmp/.ICE-unix/"
+#define UNIX_DIR "/data/data/com.termux/files/usr/tmp/.ICE-unix"
 #endif /* ICE_t */
 #if defined(TEST_t)
-#define UNIX_PATH "/tmp/.Test-unix/test"
-#define UNIX_DIR "/tmp/.Test-unix"
+#define UNIX_PATH "/data/data/com.termux/files/usr/tmp/.Test-unix/test"
+#define UNIX_DIR "/data/data/com.termux/files/usr/tmp/.Test-unix"
 #endif
 #if defined(LBXPROXY_t)
-#define UNIX_PATH "/tmp/.X11-unix/X"
-#define UNIX_DIR  "/tmp/.X11-unix"
+#define UNIX_PATH "/data/data/com.termux/files/usr/tmp/.X11-unix/X"
+#define UNIX_DIR  "/data/data/com.termux/files/usr/tmp/.X11-unix"
 #endif
 
 
