diff -uNr xtrans-1.4.0/Xtranssock.c xtrans-1.4.0.mod/Xtranssock.c
--- xtrans-1.4.0/Xtranssock.c	2019-03-16 23:30:31.000000000 +0200
+++ xtrans-1.4.0.mod/Xtranssock.c	2019-05-19 00:52:00.404623112 +0300
@@ -200,20 +200,20 @@
 
 
 #if defined(X11_t)
-#define UNIX_PATH "/tmp/.X11-unix/X"
-#define UNIX_DIR "/tmp/.X11-unix"
+#define UNIX_PATH "@TERMUX_PREFIX@/tmp/.X11-unix/X"
+#define UNIX_DIR "@TERMUX_PREFIX@/tmp/.X11-unix"
 #endif /* X11_t */
 #if defined(XIM_t)
-#define UNIX_PATH "/tmp/.XIM-unix/XIM"
-#define UNIX_DIR "/tmp/.XIM-unix"
+#define UNIX_PATH "@TERMUX_PREFIX@/tmp/.XIM-unix/XIM"
+#define UNIX_DIR "@TERMUX_PREFIX@/tmp/.XIM-unix"
 #endif /* XIM_t */
 #if defined(FS_t) || defined(FONT_t)
-#define UNIX_PATH "/tmp/.font-unix/fs"
-#define UNIX_DIR "/tmp/.font-unix"
+#define UNIX_PATH "@TERMUX_PREFIX@/tmp/.font-unix/fs"
+#define UNIX_DIR "@TERMUX_PREFIX@/tmp/.font-unix"
 #endif /* FS_t || FONT_t */
 #if defined(ICE_t)
-#define UNIX_PATH "/tmp/.ICE-unix/"
-#define UNIX_DIR "/tmp/.ICE-unix"
+#define UNIX_PATH "@TERMUX_PREFIX@/tmp/.ICE-unix/"
+#define UNIX_DIR "@TERMUX_PREFIX@/tmp/.ICE-unix"
 #endif /* ICE_t */
 
 
