diff --git a/Makefile b/Makefile
index f110b150..b7307c5f 100644
--- a/Makefile
+++ b/Makefile
@@ -3,7 +3,7 @@
 ###############################################################################
 
 LUA_EXE   = lua5.3
-CEU_EXE   = /usr/local/bin/ceu
+CEU_EXE   = @TERMUX_PREFIX@/bin/ceu
 CEU_ARGS_ = --ceu-features-trace=true --ceu-err-unused=pass $(CEU_ARGS)
 CC_ARGS_  = -llua5.3 -lpthread $(CC_ARGS)
 
