--- ../ssl_buckets.c.orig	2022-02-28 10:07:45.179525952 +0100
+++ ./buckets/ssl_buckets.c	2022-02-28 10:21:21.360065552 +0100
@@ -1325,8 +1325,7 @@
                 return 0;
             }
             else {
-                printf("OpenSSL cert error: %d %d %d\n", ERR_GET_LIB(err),
-                       ERR_GET_FUNC(err),
+                printf("OpenSSL cert error: %d %d\n", ERR_GET_LIB(err),
                        ERR_GET_REASON(err));
                 PKCS12_free(p12);
                 bio_meth_free(biom);
