diff -u -r ../imlib2-1.6.0/src/modules/loaders/loader_zlib.c ./src/modules/loaders/loader_zlib.c
--- ../imlib2-1.6.0/src/modules/loaders/loader_zlib.c	2019-11-23 05:36:44.000000000 +0000
+++ ./src/modules/loaders/loader_zlib.c	2019-12-12 20:15:28.361709201 +0000
@@ -44,7 +44,7 @@
    ImlibLoader        *loader;
    int                 src;
    int                 dest, res;
-   char               *file, *p, *q, tmp[] = "/tmp/imlib2_loader_zlib-XXXXXX";
+   char               *file, *p, *q, tmp[] = "@TERMUX_PREFIX@/tmp/imlib2_loader_zlib-XXXXXX";
    char               *real_ext;
 
    /* make sure this file ends in ".gz" and that there's another ext
