diff -uNr libtirpc-1.3.1/libtirpc.pc.in libtirpc-1.3.1.mod/libtirpc.pc.in
--- libtirpc-1.3.1/libtirpc.pc.in	2020-12-01 15:06:57.000000000 +0000
+++ libtirpc-1.3.1.mod/libtirpc.pc.in	2020-12-07 21:49:59.657182464 +0000
@@ -8,5 +8,5 @@
 Requires:
 Version: @PACKAGE_VERSION@
 Libs: -L${libdir} -ltirpc
-Libs.private: -lpthread
+Libs.private:
 Cflags: -I${includedir}/tirpc
diff -uNr libtirpc-1.3.1/src/Makefile.am libtirpc-1.3.1.mod/src/Makefile.am
--- libtirpc-1.3.1/src/Makefile.am	2020-12-01 15:06:57.000000000 +0000
+++ libtirpc-1.3.1.mod/src/Makefile.am	2020-12-07 21:50:06.577875138 +0000
@@ -12,7 +12,7 @@
 
 lib_LTLIBRARIES = libtirpc.la
 
-libtirpc_la_LDFLAGS = @LDFLAG_NOUNDEFINED@ -no-undefined -lpthread
+libtirpc_la_LDFLAGS = @LDFLAG_NOUNDEFINED@ -no-undefined
 libtirpc_la_LDFLAGS += -version-info @LT_VERSION_INFO@
 
 libtirpc_la_SOURCES = auth_none.c auth_unix.c authunix_prot.c \
