--- ../clnt_bcast.c.orig	2021-09-22 16:24:18.434244238 +0200
+++ ./src/clnt_bcast.c	2021-09-22 16:24:38.977948416 +0200
@@ -69,10 +69,6 @@
 #define	INITTIME 4000	/* Time to wait initially */
 #define	WAITTIME 8000	/* Maximum time to wait */
 
-# define POLLRDNORM     0x040           /* Normal data may be read.  */
-# define POLLRDBAND     0x080           /* Priority data may be read.  */
-
-
 
 /*
  * If nettype is NULL, it broadcasts on all the available
