--- ../main.c.orig	2020-03-10 21:37:55.340743885 +0100
+++ ./wpa_supplicant/main.c	2020-03-10 21:38:08.970690217 +0100
@@ -105,7 +105,7 @@
 	       "  -W = wait for a control interface monitor before starting\n");
 
 	printf("example:\n"
-	       "  wpa_supplicant -D%s -iwlan0 -c/etc/wpa_supplicant.conf\n",
+	       "  wpa_supplicant -D%s -iwlan0 -c@TERMUX_PREFIX@/etc/wpa_supplicant.conf\n",
 	       wpa_drivers[0] ? wpa_drivers[0]->name : "nl80211");
 #endif /* CONFIG_NO_STDOUT_DEBUG */
 }
