diff -uNr xorg-server-1.19.1/os/backtrace.c xorg-server-1.19.1.mod/os/backtrace.c
--- xorg-server-1.19.1/os/backtrace.c	2017-01-04 23:32:10.000000000 +0200
+++ xorg-server-1.19.1.mod/os/backtrace.c	2017-11-24 18:44:44.672683897 +0200
@@ -250,7 +250,7 @@
         closefrom(STDERR_FILENO);
 
         snprintf(parent, sizeof(parent), "%d", getppid());
-        execle("/usr/bin/pstack", "pstack", parent, NULL);
+        execle("/data/data/com.termux/files/usr/bin/pstack", "pstack", parent, NULL);
         exit(1);
     }
     else {
