diff -uNr xorg-server-1.19.1/include/site.h xorg-server-1.19.1.mod/include/site.h
--- xorg-server-1.19.1/include/site.h	2017-01-04 23:28:03.000000000 +0200
+++ xorg-server-1.19.1.mod/include/site.h	2017-11-24 18:44:44.672683897 +0200
@@ -72,7 +72,7 @@
  * DO NOT CHANGE THESE VALUES OR THE DIX IMAKEFILE!
  */
 #ifndef COMPILEDDEFAULTFONTPATH
-#define COMPILEDDEFAULTFONTPATH	"/usr/share/fonts/X11/misc/"
+#define COMPILEDDEFAULTFONTPATH	"/data/data/com.termux/files/usr/share/fonts/X11/misc/"
 #endif
 
 /*
