diff -uNr xorg-server-1.19.1/os/connection.c xorg-server-1.19.1.mod/os/connection.c
--- xorg-server-1.19.1/os/connection.c	2017-01-11 22:04:12.000000000 +0200
+++ xorg-server-1.19.1.mod/os/connection.c	2017-11-24 18:44:44.672683897 +0200
@@ -1120,7 +1120,7 @@
     XtransConnInfo ciptr;
     const char *display_env = getenv("DISPLAY");
 
-    if (display_env && (strncmp(display_env, "/tmp/launch", 11) == 0)) {
+    if (display_env && (strncmp(display_env, "/data/data/com.termux/files/usr/tmp/launch", 11) == 0)) {
         /* Make the path the launchd socket if our DISPLAY is set right */
         strcpy(port, display_env);
     }
