diff -uNr xorg-server-1.19.1/hw/xfree86/os-support/linux/lnx_acpi.c xorg-server-1.19.1.mod/hw/xfree86/os-support/linux/lnx_acpi.c
--- xorg-server-1.19.1/hw/xfree86/os-support/linux/lnx_acpi.c	2017-01-04 23:27:52.000000000 +0200
+++ xorg-server-1.19.1.mod/hw/xfree86/os-support/linux/lnx_acpi.c	2017-11-24 18:44:44.662683772 +0200
@@ -15,7 +15,7 @@
 #include <fcntl.h>
 #include <errno.h>
 
-#define ACPI_SOCKET  "/var/run/acpid.socket"
+#define ACPI_SOCKET  "/data/data/com.termux/files/usr/var/run/acpid.socket"
 
 #define ACPI_VIDEO_NOTIFY_SWITCH	0x80
 #define ACPI_VIDEO_NOTIFY_PROBE		0x81
