diff -uNr xorg-server-1.19.1/os/access.c xorg-server-1.19.1.mod/os/access.c
--- xorg-server-1.19.1/os/access.c	2017-01-04 23:32:10.000000000 +0200
+++ xorg-server-1.19.1.mod/os/access.c	2017-11-24 18:44:44.672683897 +0200
@@ -961,7 +961,7 @@
 #if defined WIN32 && defined __MINGW32__
 #define ETC_HOST_PREFIX "X"
 #else
-#define ETC_HOST_PREFIX "/etc/X"
+#define ETC_HOST_PREFIX "/data/data/com.termux/files/usr/etc/X"
 #endif
 #define ETC_HOST_SUFFIX ".hosts"
     fnamelen = strlen(ETC_HOST_PREFIX) + strlen(ETC_HOST_SUFFIX) +
