+++ ./configure.ac
@@ -26,7 +26,7 @@
 PKG_CHECK_MODULES(XCB_RENDERUTIL, xcb-renderutil)
 PKG_CHECK_MODULES(XCB_IMAGE, xcb-image)
 
-DEF_CURSORPATH="~/.icons:/usr/share/icons:/usr/share/pixmaps:/usr/X11R6/lib/X11/icons"
+DEF_CURSORPATH="~/.icons:@TERMUX_PREFIX@/share/icons:@TERMUX_PREFIX@/share/pixmaps"
 AC_ARG_WITH(cursorpath,
 	AS_HELP_STRING([--with-cursorpath=<paths>],
 		[Set default search path for cursors]),
