--- a/filters/words/wiki/CMakeLists.txt
+++ b/filters/words/wiki/CMakeLists.txt
@@ -3,7 +3,6 @@
 if(SHOULD_BUILD_FILTER_ODT_TO_WIKI)
 if (SharedMimeInfo_FOUND)
     install(FILES wiki-format.xml DESTINATION ${KDE_INSTALL_MIMEDIR})
-    update_xdg_mimetypes(${KDE_INSTALL_MIMEDIR})
 endif()
 
 add_subdirectory(export)
--- a/plugins/vectorshape/CMakeLists.txt
+++ b/plugins/vectorshape/CMakeLists.txt
@@ -36,7 +36,6 @@ install(TARGETS calligra_shape_vector DESTINATION ${KDE_INSTALL_PLUGINDIR}/calli
 
 if (SharedMimeInfo_FOUND)
     install(FILES calligra_svm.xml DESTINATION ${KDE_INSTALL_MIMEDIR})
-    update_xdg_mimetypes(${KDE_INSTALL_MIMEDIR})
 endif ()
 
 ########### subdirectories ###############
