diff --git a/src/daemon/caps.c~ b/src/daemon/caps.c
index fd135c0..135d842 100644
--- a/src/daemon/caps.c~
+++ b/src/daemon/caps.c
@@ -92,8 +92,5 @@ void pa_drop_caps(void) {
 #error "Don't know how to do capabilities on your system.  Please send a patch."
 #endif /* __linux__ */
 #else /* HAVE_SYS_CAPABILITY_H */
-    pa_log_warn("Normally all extra capabilities would be dropped now, but "
-                "that's impossible because PulseAudio was built without "
-                "capabilities support.");
 #endif
 }
