--- ../cache/pulseaudio-11.1/src/Makefile.am	2017-09-18 10:41:02.000000000 +0000
+++ ./src/Makefile.am	2017-12-22 22:51:30.628573929 +0000
@@ -1594,6 +1595,11 @@
 module_simple_protocol_unix_la_LIBADD = $(MODULE_LIBADD) libprotocol-simple.la
 
 # CLI protocol
+modlibexec_LTLIBRARIES += module-sles-sink.la
+module_sles_sink_la_SOURCES = modules/sles/module-sles-sink.c
+module_sles_sink_la_LDFLAGS = $(MODULE_LDFLAGS) -lOpenSLES
+module_sles_sink_la_LIBADD = $(MODULE_LIBADD)
+module_sles_sink_la_CFLAGS = $(AM_CFLAGS) -DPA_MODULE_NAME=module_sles_sink
 
 module_cli_la_SOURCES = modules/module-cli.c
 module_cli_la_LDFLAGS = $(MODULE_LDFLAGS)
