diff --git a/src/Makefile.am~ b/src/Makefile.am
index f4464d2..a2c201d 100644
--- a/src/Makefile.am~
+++ b/src/Makefile.am
@@ -1495,6 +1495,12 @@ bin_SCRIPTS += utils/qpaeq
 endif
 endif
 
+modlibexec_LTLIBRARIES += module-sles-sink.la
+module_sles_sink_la_SOURCES = modules/sles/module-sles-sink.c
+module_sles_sink_la_LDFLAGS = $(MODULE_LDFLAGS) -lOpenSLES
+module_sles_sink_la_LIBADD = $(MODULE_LIBADD)
+module_sles_sink_la_CFLAGS = $(AM_CFLAGS) -DPA_MODULE_NAME=module_sles_sink
+
 # Simple protocol
 
 module_simple_protocol_tcp_la_SOURCES = modules/module-protocol-stub.c
