diff -uNr xfce4-session-4.16.0/scripts/startxfce4.in xfce4-session-4.16.0.mod/scripts/startxfce4.in
--- xfce4-session-4.16.0/scripts/startxfce4.in	2016-05-02 21:02:14.000000000 +0000
+++ xfce4-session-4.16.0.mod/scripts/startxfce4.in	2022-04-12 08:25:51.362449242 +0000
@@ -88,14 +88,9 @@
 fi
 export XDG_CONFIG_DIRS
 
-if test "x$DISPLAY" = "x"
-then
-  echo "$0: Starting X server"
-  prog=xinit
-
-  if test ! "x$XDG_VTNR" = "x"; then
-    SERVERRC="$SERVERRC vt$XDG_VTNR"
-  fi
+if test "x$DISPLAY" = "x"; then
+  echo "$0: Failed to connect to display"
+  exit 2
 else
   echo "$0: X server already running on display $DISPLAY"
   prog=/bin/sh
