--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -218,7 +218,7 @@
 set(EXE_FILE "${OUTPUT_DIR}/blade${CMAKE_EXECUTABLE_SUFFIX}")
 
 add_custom_command(TARGET blade POST_BUILD
-		COMMAND	"${EXE_FILE}" "${PROJECT_SOURCE_DIR}/scripts/make_blade.b" "${PROJECT_SOURCE_DIR}"
+		COMMAND	blade "${PROJECT_SOURCE_DIR}/scripts/make_blade.b" "${PROJECT_SOURCE_DIR}"
 		COMMENT "Generating Blade header and folders..."
 )
 
