diff --git a/pyproject.toml b/pyproject.toml
index b66ffd7..a0f51f1 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -37,7 +37,6 @@ dependencies = [
     "PyMySQL ~= 1.1.2",
     "sqlparse >= 0.3.0,<0.6.0",
     "sqlglot ~= 30.17.0",
-    "sqlglotc ~= 30.17.0",
     "configobj ~= 5.0.9",
     "cli_helpers[styles] ~= 2.15.1",
     "wcwidth ~= 0.8.2",                     # can upgrade after 4 Sep 2026
