--- ./CMakeLists.txt.orig	2022-10-17 18:45:12.854205329 +0000
+++ ./CMakeLists.txt	2022-10-17 18:46:08.305988462 +0000
@@ -343,7 +343,7 @@
     configure_package_config_file(
         "cmake/MbedTLSConfig.cmake.in"
         "cmake/MbedTLSConfig.cmake"
-            INSTALL_DESTINATION "cmake")
+            INSTALL_DESTINATION "lib/cmake")
 
     write_basic_package_version_file(
         "cmake/MbedTLSConfigVersion.cmake"
