https://github.com/termux/termux-packages/issues/15852

--- a/src/bftextview2_autocomp.c
+++ b/src/bftextview2_autocomp.c
@@ -429,7 +429,7 @@ acwin_create(BluefishTextView * btv)
 	/*gtk_widget_set_size_request(acw->reflabel,150,-1); */
 	gtk_widget_show_all(acw->scroll);
 	gtk_widget_show(hbox);
-	g_signal_connect(acw->reflabel, "activate-link", acw_label_active_link_lcb, acw);
+	g_signal_connect(acw->reflabel, "activate-link", G_CALLBACK(acw_label_active_link_lcb), acw);
 	/*gtk_widget_set_size_request(GTK_WIDGET(acw->tree),100,200); */
 	/*gtk_widget_set_size_request(acw->win, 150, 200); */
 	/*g_signal_connect(G_OBJECT(acw->win),"key-release-event",G_CALLBACK(acwin_key_release_lcb),acw); */
--- a/src/external_commands.c
+++ b/src/external_commands.c
@@ -483,7 +483,7 @@ create_commandstring(Texternalp * ep, co
 		gtk_dialog_set_default_response(GTK_DIALOG(dialog),GTK_RESPONSE_ACCEPT);
 		tmp = g_strdup_printf(_("Supply arguments to define %%a in '%s'"), formatstring);
 		entry = dialog_entry_labeled(NULL, tmp, gtk_dialog_get_content_area(GTK_DIALOG(dialog)), 6);
-		g_signal_connect(G_OBJECT(entry), "activate", command_dialog_entry_activated_lcb, dialog);
+		g_signal_connect(G_OBJECT(entry), "activate", G_CALLBACK(command_dialog_entry_activated_lcb), dialog);
 		g_free(tmp);
 		gtk_widget_show_all(dialog);
 		result = gtk_dialog_run(GTK_DIALOG(dialog));
