--- sysroot.orig/usr/include/bits/ioctl.h	2019-07-30 14:11:49.196931147 -0400
+++ sysroot/usr/include/bits/ioctl.h	2019-07-30 14:12:44.446936650 -0400
@@ -55,7 +55,7 @@
  * - defining BIONIC_IOCTL_NO_SIGNEDNESS_OVERLOAD, which will make the
  *   overloading go away.
  */
-#if !defined(BIONIC_IOCTL_NO_SIGNEDNESS_OVERLOAD)
+#if !defined(BIONIC_IOCTL_NO_SIGNEDNESS_OVERLOAD) && defined(__clang__)
 /* enable_if(1) just exists to break overloading ties. */
 int ioctl(int __fd, unsigned __request, ...) __overloadable __enable_if(1, "") __RENAME(ioctl);
 #endif
