diff -u -r /home/fornwall/lib/android-ndk/sysroot/usr/include/limits.h ./usr/include/limits.h
--- /home/fornwall/lib/android-ndk/sysroot/usr/include/limits.h	2017-06-07 01:07:52.000000000 +0200
+++ ./usr/include/limits.h	2017-06-18 02:05:17.404056395 +0200
@@ -139,7 +139,7 @@
 #define HOST_NAME_MAX _POSIX_HOST_NAME_MAX
 
 #define  _POSIX_VERSION             200809L   /* Posix C language bindings version */
-#define  _POSIX2_VERSION            -1        /* we don't support Posix command-line tools */
+#define  _POSIX2_VERSION            200809    /* Termux change to avoid coreutils obsolete breakage */
 #define  _XOPEN_VERSION             700       /* by Posix definition */
 
 /* >= _POSIX_THREAD_DESTRUCTOR_ITERATIONS */
