diff -u -r /home/fornwall/lib/android-ndk/sysroot/usr/include/linux/fcntl.h ./usr/include/linux/fcntl.h
--- /home/fornwall/lib/android-ndk/sysroot/usr/include/linux/fcntl.h	2017-11-09 09:57:12.000000000 +0100
+++ ./usr/include/linux/fcntl.h	2017-11-15 11:49:10.709710452 +0100
@@ -40,6 +40,12 @@
 #define DN_ATTRIB 0x00000020
 #define DN_MULTISHOT 0x80000000
 #define AT_FDCWD - 100
+/* Termux patch: This flag is not supported on Android 6.0+, and should not
+   be needed since there are not setuid binaries. Define AT_EACCESS to 0 to
+   avoid packages defining their own AT_EACCESS (which e.g. coreutil does)
+   since that will only result in errors. See:
+   https://android.googlesource.com/platform/bionic/+/3577825%5E!/      */
+#define AT_EACCESS 0
 #define AT_SYMLINK_NOFOLLOW 0x100
 #define AT_REMOVEDIR 0x200
 #define AT_SYMLINK_FOLLOW 0x400
