--- ./usr/include/mntent.h.orig	2021-12-10 16:17:07.041722648 +0530
+++ ./usr/include/mntent.h	2021-12-10 16:17:41.171722635 +0530
@@ -70,9 +70,9 @@
 #endif /* __ANDROID_API__ >= 21 */
 
 
-#if __ANDROID_API__ >= 26
-char* hasmntopt(const struct mntent* __entry, const char* __option) __INTRODUCED_IN(26);
-#endif /* __ANDROID_API__ >= 26 */
+#if __ANDROID_API__ >= 21
+char* hasmntopt(const struct mntent* __entry, const char* __option) __INTRODUCED_IN(21);
+#endif /* __ANDROID_API__ >= 21 */
 
 
 __END_DECLS
