diff -u -r /home/fornwall/lib/android-ndk/sysroot/usr/include/paths.h ./usr/include/paths.h
--- /home/fornwall/lib/android-ndk/sysroot/usr/include/paths.h	2017-11-09 09:57:12.000000000 +0100
+++ ./usr/include/paths.h	2017-11-15 11:51:24.456129565 +0100
@@ -34,15 +34,19 @@
 
 #include <sys/cdefs.h>
 
-#ifndef _PATH_BSHELL
-#define	_PATH_BSHELL	"/system/bin/sh"
-#endif
+#define	_PATH_BSHELL	"@TERMUX_PREFIX@/bin/sh"
 #define	_PATH_CONSOLE	"/dev/console"
-#define	_PATH_DEFPATH	"/sbin:/system/sbin:/system/bin:/system/xbin:/odm/bin:/vendor/bin:/vendor/xbin"
+#define	_PATH_DEFPATH	"@TERMUX_PREFIX@/bin:@TERMUX_PREFIX@/bin/applets"
 #define	_PATH_DEV	"/dev/"
 #define	_PATH_DEVNULL	"/dev/null"
 #define	_PATH_KLOG	"/proc/kmsg"
 #define	_PATH_MOUNTED	"/proc/mounts"
 #define	_PATH_TTY	"/dev/tty"
 
+#define	_PATH_STDPATH	_PATH_DEFPATH
+#define	_PATH_TMP	"@TERMUX_PREFIX@/tmp/"
+#define	_PATH_VARDB	"@TERMUX_PREFIX@/var/db/"
+#define	_PATH_VARRUN	"@TERMUX_PREFIX@/var/run/"
+#define	_PATH_VARTMP	"@TERMUX_PREFIX@/var/tmp/"
+
 #endif /* !_PATHS_H_ */
