--- ./usr/include/paths.h.orig	2020-07-10 21:50:03.778299300 +0000
+++ ./usr/include/paths.h	2020-07-10 21:51:03.127602196 +0000
@@ -40,14 +40,14 @@
 
 #ifndef _PATH_BSHELL
 /** Path to the default system shell. Historically the 'B' was to specify the Bourne shell. */
-#define _PATH_BSHELL "/system/bin/sh"
+#define _PATH_BSHELL "@TERMUX_PREFIX@/bin/sh"
 #endif
 
 /** Path to the system console. */
 #define _PATH_CONSOLE "/dev/console"
 
 /** Default shell search path. */
-#define _PATH_DEFPATH "/product/bin:/apex/com.android.runtime/bin:/apex/com.android.art/bin:/system/bin:/system/xbin:/odm/bin:/vendor/bin:/vendor/xbin"
+#define _PATH_DEFPATH "@TERMUX_PREFIX@/bin"
 
 /** Path to the directory containing device files. */
 #define _PATH_DEV "/dev/"
@@ -63,3 +63,9 @@
 
 /** Path to the calling process' tty. */
 #define _PATH_TTY "/dev/tty"
+
+#define        _PATH_STDPATH   _PATH_DEFPATH
+#define        _PATH_TMP       "@TERMUX_PREFIX@/tmp/"
+#define        _PATH_VARDB     "@TERMUX_PREFIX@/var/db/"
+#define        _PATH_VARRUN    "@TERMUX_PREFIX@/var/run/"
+#define        _PATH_VARTMP    "@TERMUX_PREFIX@/var/tmp/"
