diff -u -r /home/fornwall/lib/android-ndk/sysroot/usr/include/spawn.h ./usr/include/spawn.h
--- /home/fornwall/lib/android-ndk/sysroot/usr/include/spawn.h	2017-11-09 09:57:12.000000000 +0100
+++ ./usr/include/spawn.h	2017-11-15 12:08:34.911786251 +0100
@@ -36,6 +36,8 @@
 
 __BEGIN_DECLS
 
+#if __ANDROID_API__ >= __ANDROID_API_FUTURE__
+
 #define POSIX_SPAWN_RESETIDS 1
 #define POSIX_SPAWN_SETPGROUP 2
 #define POSIX_SPAWN_SETSIGDEF 4
@@ -51,7 +53,6 @@
 typedef struct __posix_spawn_file_actions* posix_spawn_file_actions_t;
 
 
-#if __ANDROID_API__ >= __ANDROID_API_FUTURE__
 int posix_spawn(pid_t* __pid, const char* __path, const posix_spawn_file_actions_t* __actions, const posix_spawnattr_t* __attr, char* const __argv[], char* const __env[]) __INTRODUCED_IN_FUTURE;
 int posix_spawnp(pid_t* __pid, const char* __file, const posix_spawn_file_actions_t* __actions, const posix_spawnattr_t* __attr, char* const __argv[], char* const __env[]) __INTRODUCED_IN_FUTURE;
 
