--- ./usr/include/stdlib.h.orig	2020-12-22 09:37:22.122377110 +0100
+++ ./usr/include/stdlib.h	2020-12-22 09:38:22.159360733 +0100
@@ -33,6 +33,7 @@
 #include <bits/wait.h>
 #include <malloc.h>
 #include <stddef.h>
+#include <stdint.h>
 #include <sys/cdefs.h>
 #include <xlocale.h>
 
@@ -224,8 +225,7 @@
 size_t wcstombs(char* __dst, const wchar_t* __src, size_t __n);
 
 #if __ANDROID_API__ >= 21
-size_t __ctype_get_mb_cur_max(void) __INTRODUCED_IN(21);
-#define MB_CUR_MAX __ctype_get_mb_cur_max()
+#define MB_CUR_MAX 4
 #else
 /*
  * Pre-L we didn't have any locale support and so we were always the POSIX
