diff -u -r /home/fornwall/lib/android-ndk/sysroot/usr/include/stdlib.h ./usr/include/stdlib.h
--- /home/fornwall/lib/android-ndk/sysroot/usr/include/stdlib.h	2017-11-09 09:57:12.000000000 +0100
+++ ./usr/include/stdlib.h	2017-11-15 12:49:51.790799140 +0100
@@ -35,6 +35,7 @@
 #include <alloca.h>
 #include <malloc.h>
 #include <stddef.h>
+#include <stdint.h>
 #include <sys/cdefs.h>
 #include <xlocale.h>
 
@@ -205,8 +206,7 @@
 size_t wcstombs(char* __dst, const wchar_t* __src, size_t __n);
 
 #if __ANDROID_API__ >= __ANDROID_API_L__
-size_t __ctype_get_mb_cur_max(void) __INTRODUCED_IN(21);
-#define MB_CUR_MAX __ctype_get_mb_cur_max()
+#define MB_CUR_MAX 4
 #else
 /*
  * Pre-L we didn't have any locale support and so we were always the POSIX
