--- sysroot.orig/usr/include/string.h	2019-07-30 14:11:49.196931147 -0400
+++ sysroot/usr/include/string.h	2019-07-30 14:12:44.446936650 -0400
@@ -188,7 +188,7 @@
 #endif
 
 /* Const-correct overloads. Placed after FORTIFY so we call those functions, if possible. */
-#if defined(__cplusplus)
+#if defined(__cplusplus) && defined(__clang__)
 /*
  * Use two enable_ifs so these overloads don't conflict with + are preferred over libcxx's. This can
  * be reduced to 1 after libcxx recognizes that we have const-correct overloads.
