diff -u -r /home/fornwall/lib/android-ndk/sysroot/usr/include/sys/cdefs.h ./usr/include/sys/cdefs.h
--- /home/fornwall/lib/android-ndk/sysroot/usr/include/sys/cdefs.h	2017-06-20 17:41:56.000000000 +0200
+++ ./usr/include/sys/cdefs.h	2017-07-07 03:27:59.327850834 +0200
@@ -234,13 +234,6 @@
 # define __USE_BSD 1
 #endif
 
-/* _FILE_OFFSET_BITS 64 support. */
-#if !defined(__LP64__) && defined(_FILE_OFFSET_BITS)
-#if _FILE_OFFSET_BITS == 64
-#define __USE_FILE_OFFSET64 1
-#endif
-#endif
-
 #define  __BIONIC__   1
 #include <android/api-level.h>
 
