diff -u -r /home/fornwall/lib/android-ndk/sysroot/usr/include/sys/cdefs.h ./usr/include/sys/cdefs.h
--- /home/fornwall/lib/android-ndk/sysroot/usr/include/sys/cdefs.h	2017-06-20 17:41:56.000000000 +0200
+++ ./usr/include/sys/cdefs.h	2017-07-09 23:37:26.453911992 +0200
@@ -237,9 +237,15 @@
 /* _FILE_OFFSET_BITS 64 support. */
 #if !defined(__LP64__) && defined(_FILE_OFFSET_BITS)
 #if _FILE_OFFSET_BITS == 64
+#ifdef TERMUX_EXPOSE_FILE_OFFSET64
+/* Using _FILE_OFFSET_BITS=64 does not work very well on Android
+ * and can cause problems when mixing libraries, which is why
+ * Termux hides away this unless TERMUX_EXPOSE_FILE_OFFSET64
+ * is defined. */
 #define __USE_FILE_OFFSET64 1
 #endif
 #endif
+#endif
 
 #define  __BIONIC__   1
 #include <android/api-level.h>
