diff -u -r /home/builder/lib/android-ndk/sysroot/usr/include/sys/cdefs.h ./usr/include/sys/cdefs.h
--- /home/builder/lib/android-ndk/sysroot/usr/include/sys/cdefs.h	2019-05-22 12:49:28.000000000 +0000
+++ ./usr/include/sys/cdefs.h	2019-09-19 19:18:03.978640526 +0000
@@ -176,7 +176,11 @@
  * _FILE_OFFSET_BITS 64 support.
  * See https://android.googlesource.com/platform/bionic/+/master/docs/32-bit-abi.md
  */
-#if !defined(__LP64__) && defined(_FILE_OFFSET_BITS) && _FILE_OFFSET_BITS == 64
+/* Using _FILE_OFFSET_BITS=64 does not work well on Android
+ * and can cause problems when mixing libraries, which is why
+ * Termux hides away this unless TERMUX_EXPOSE_FILE_OFFSET64
+ * is defined. */
+#if !defined(__LP64__) && defined(_FILE_OFFSET_BITS) && _FILE_OFFSET_BITS == 64 && defined(TERMUX_EXPOSE_FILE_OFFSET64)
 #  define __USE_FILE_OFFSET64 1
 /*
  * Note that __RENAME_IF_FILE_OFFSET64 is only valid if the off_t and off64_t
@@ -336,3 +340,6 @@
 
 #include <android/versioning.h>
 #include <android/api-level.h>
+
+#define __TERMUX__ 1
+#define __TERMUX_PREFIX__ "@TERMUX_PREFIX@"
