diff -u -r /home/fornwall/lib/android-ndk/platforms/android-21/arch-arm/usr/include/termios.h ./usr/include/termios.h
--- /home/fornwall/lib/android-ndk/platforms/android-21/arch-arm/usr/include/termios.h	2014-10-14 22:53:49.000000000 -0400
+++ ./usr/include/termios.h	2015-05-15 18:23:49.264331535 -0400
@@ -49,6 +49,8 @@
 int tcsendbreak(int, int);
 int tcsetattr(int, int, const struct termios*);

+#define _POSIX_VDISABLE 0
+
 __END_DECLS

 #endif /* _TERMIOS_H_ */
