--- sysroot/usr/include/utmp.h	2021-05-07 05:45:33.518467959 +0000
+++ usr/include/utmp.h	2021-05-07 05:46:25.195022662 +0000
@@ -32,9 +32,9 @@
 #include <sys/types.h>
 #include <time.h>
 
-#define _PATH_UTMP      "/var/run/utmp"
-#define _PATH_WTMP      "/var/log/wtmp"
-#define _PATH_LASTLOG   "/var/log/lastlog"
+#define _PATH_UTMP      "@TERMUX_PREFIX@/var/run/utmp"
+#define _PATH_WTMP      "@TERMUX_PREFIX@/var/log/wtmp"
+#define _PATH_LASTLOG   "@TERMUX_PREFIX@/var/log/lastlog"
 
 #ifdef __LP64__
 #define UT_NAMESIZE 32
