diff -u -r /home/fornwall/lib/android-ndk/sysroot/usr/include/utmp.h ./usr/include/utmp.h
--- /home/fornwall/lib/android-ndk/sysroot/usr/include/utmp.h	2017-11-09 09:57:12.000000000 +0100
+++ ./usr/include/utmp.h	2017-11-15 12:01:37.108811692 +0100
@@ -101,11 +101,8 @@
 struct utmp* getutent(void);
 void endutent(void);
 
-
-#if __ANDROID_API__ >= 23
-int login_tty(int __fd) __INTRODUCED_IN(23);
-#endif /* __ANDROID_API__ >= 23 */
-
+/* In Termux login_tty() is implemented in the libutil package to support android-21. */
+int login_tty(int __fd);
 
 __END_DECLS
 
