diff -u -r ../gmime-2.6.23/Makefile.in ./Makefile.in
--- ../gmime-2.6.23/Makefile.in	2017-01-21 23:27:33.000000000 +0100
+++ ./Makefile.in	2017-04-06 18:58:31.616613454 +0200
@@ -194,7 +194,7 @@
 ETAGS = etags
 CTAGS = ctags
 CSCOPE = cscope
-DIST_SUBDIRS = m4 build util gmime mono tests docs examples tools .
+DIST_SUBDIRS = m4 build util gmime mono docs examples tools .
 am__DIST_COMMON = $(srcdir)/Makefile.in $(srcdir)/config.h.in \
 	$(srcdir)/gmime.pc.in $(srcdir)/gmime.spec.in AUTHORS COPYING \
 	ChangeLog NEWS README TODO compile config.guess config.rpath \
@@ -433,7 +433,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = m4 build util gmime $(am__append_1) tests docs \
+SUBDIRS = m4 build util gmime $(am__append_1) docs \
 	$(am__append_2) tools .
 
 # build documentation when doing distcheck
