diff -u -r ../subversion-1.8.5/configure ./configure
--- ../subversion-1.8.5/configure	2013-11-15 02:07:12.000000000 +0100
+++ ./configure	2014-02-12 08:43:44.000000000 +0100
@@ -18377,10 +18377,7 @@
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for Mach-O dynamic module iteration functions" >&5
 $as_echo_n "checking for Mach-O dynamic module iteration functions... " >&6; }
   if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
+  echo no
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
