diff --git a/cfg/vnstat.conf b/cfg/vnstat.conf
index f68cad4..306aefc 100644
--- a/cfg/vnstat.conf
+++ b/cfg/vnstat.conf
@@ -8,7 +8,7 @@
 ;Interface ""
 
 # location of the database directory
-;DatabaseDir "/var/lib/vnstat"
+;DatabaseDir "@TERMUX_PREFIX@/var/lib/vnstat"
 
 # locale (LC_ALL) ("-" = use system locale)
 ;Locale "-"
@@ -168,10 +168,10 @@
 ;UpdateFileOwner 1
 
 # file used for logging if UseLogging is set to 1
-;LogFile "/var/log/vnstat/vnstat.log"
+;LogFile "@TERMUX_PREFIX@/var/log/vnstat/vnstat.log"
 
 # file used as daemon pid / lock file
-;PidFile "/var/run/vnstat/vnstat.pid"
+;PidFile "@TERMUX_PREFIX@/var/run/vnstat/vnstat.pid"
 
 # 1 = 64-bit, 0 = 32-bit, -1 = old style logic, -2 = automatic detection
 ;64bitInterfaceCounters -2
