diff --git a/src/common.h b/src/common.h
index c528a5d..8e782c0 100644
--- a/src/common.h
+++ b/src/common.h
@@ -64,7 +64,7 @@ and most can be changed later from the config file.
 #if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__APPLE__)
 #define DATABASEDIR "/var/db/vnstat"
 #else
-#define DATABASEDIR "/var/lib/vnstat"
+#define DATABASEDIR "@TERMUX_PREFIX@/var/lib/vnstat"
 #endif
 #endif
 
@@ -259,8 +259,8 @@ and most can be changed later from the config file.
 #define USELOGGING 2
 #define CREATEDIRS 1
 #define UPDATEFILEOWNER 1
-#define LOGFILE "/var/log/vnstat/vnstat.log"
-#define PIDFILE "/var/run/vnstat/vnstat.pid"
+#define LOGFILE "@TERMUX_PREFIX@/var/log/vnstat/vnstat.log"
+#define PIDFILE "@TERMUX_PREFIX@/var/run/vnstat/vnstat.pid"
 #define IS64BIT -2
 #define WALDB 0
 #define WALDBCHECKPOINTINTERVALMINS 240
