--- a/src/ReadConfig.cxx	2018-11-23 13:05:51.749928637 +0100
+++ b/src/ReadConfig.cxx	2018-11-23 13:07:24.143261299 +0100
@@ -38,7 +38,7 @@
 
 #ifndef _WIN32
 
-#define FILE_CACHE "/var/cache/mpdscribble/mpdscribble.cache"
+#define FILE_CACHE "@TERMUX_PREFIX@/var/cache/mpdscribble/mpdscribble.cache"
 #define FILE_HOME_CONF "~/.mpdscribble/mpdscribble.conf"
 #define FILE_HOME_CACHE "~/.mpdscribble/mpdscribble.cache"
 
