diff -uNr recordmydesktop-0.3.8.1/configure recordmydesktop-0.3.8.1.mod/configure
--- recordmydesktop-0.3.8.1/configure	2008-12-13 19:40:03.000000000 +0200
+++ recordmydesktop-0.3.8.1.mod/configure	2020-02-11 14:34:18.712282563 +0200
@@ -8580,7 +8580,7 @@
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lpthread  $LIBS"
+LIBS=" $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -8644,7 +8644,7 @@
 #define HAVE_LIBPTHREAD 1
 _ACEOF
 
-  LIBS="-lpthread $LIBS"
+  LIBS=" $LIBS"
 
 else
   { { $as_echo "$as_me:$LINENO: error: Can't find libpthread" >&5
