diff -uNr recordmydesktop-0.3.8.1/src/rmd_initialize_data.c recordmydesktop-0.3.8.1.mod/src/rmd_initialize_data.c
--- recordmydesktop-0.3.8.1/src/rmd_initialize_data.c	2008-12-13 17:49:09.000000000 +0200
+++ recordmydesktop-0.3.8.1.mod/src/rmd_initialize_data.c	2020-02-11 15:00:47.318593830 +0200
@@ -240,7 +240,7 @@
     strcpy(args->device, DEFAULT_AUDIO_DEVICE);
 
     args->workdir = (char *) malloc(5);
-    strcpy(args->workdir, "/tmp");
+    strcpy(args->workdir, "@TERMUX_PREFIX@/tmp");
 
     args->pause_shortcut = (char *) malloc(15);
     strcpy(args->pause_shortcut, "Control+Mod1+p");
