Sound is unsupported.

diff -uNr recordmydesktop-0.3.8.1/src/rmd_initialize_data.c recordmydesktop-0.3.8.1.mod/src/rmd_initialize_data.c
--- recordmydesktop-0.3.8.1/src/rmd_initialize_data.c	2008-12-13 17:49:09.000000000 +0200
+++ recordmydesktop-0.3.8.1.mod/src/rmd_initialize_data.c	2020-02-11 14:59:43.758378031 +0200
@@ -202,7 +202,7 @@
     args->width                = 0;
     args->height               = 0;
     args->rescue_path          = NULL;
-    args->nosound              = 0;
+    args->nosound              = 1;
     args->full_shots           = 0;
     args->follow_mouse         = 0;
     args->encOnTheFly          = 0;
