diff -u -r ../daemonize-release-1.7.7/daemonize.c ./daemonize.c
--- ../daemonize-release-1.7.7/daemonize.c	2015-12-11 17:54:43.000000000 -0500
+++ ./daemonize.c	2016-12-23 03:40:30.055592421 -0500
@@ -38,7 +38,10 @@
 static bool         be_verbose = FALSE;
 static const char  *user       = NULL;
 static char       **cmd        = NULL;
-static const char  *cwd        = "/";
+/* Termux patch: Use a readable directory for e.g. runsvdir to work,
+   as / is not readable starting from Android 7:
+   https://github.com/termux/termux-packages/issues/614 */
+static const char  *cwd        = "@TERMUX_PREFIX@";
 static int          null_fd    = -1;
 static int          out_fd     = -1;
 static int          err_fd     = -1;
