diff -uNr boinc-client_release-7.18-7.18.1/configure.ac boinc-client_release-7.18-7.18.1.mod/configure.ac
--- boinc-client_release-7.18-7.18.1/configure.ac	2021-08-03 23:52:19.000000000 +0800
+++ boinc-client_release-7.18-7.18.1.mod/configure.ac	2021-12-27 20:13:32.143967208 +0800
@@ -1071,7 +1071,7 @@
 AM_CONDITIONAL(OS_WIN32_MINGW, [echo $host_os | grep '^mingw' > /dev/null])
 dnl or OS2
 AM_CONDITIONAL(OS_OS2, [echo $host_os | grep '^os2' > /dev/null])
-AM_CONDITIONAL(OS_ARM_LINUX, [echo $host_alias | grep '^arm-linux' > /dev/null])
+AM_CONDITIONAL(OS_ARM_LINUX, [echo $host_alias | grep '^arm-linux-skip' > /dev/null])
 AM_CONDITIONAL(OS_ARMV6_LINUX, [echo $host_alias | grep '^armv6-linux' > /dev/null])
 AM_CONDITIONAL(ANDROID, [test x"${ANDROID}" = xyes])
 AM_CONDITIONAL(BUILD_WITH_VCPKG, [test "${enable_vcpkg}" = yes])
