diff -uNr boinc-client_release-7.16-7.16.11/client/Makefile.am boinc-client_release-7.16-7.16.11.mod/client/Makefile.am
--- boinc-client_release-7.16-7.16.11/client/Makefile.am	2020-08-30 16:04:59.000000000 +0800
+++ boinc-client_release-7.16-7.16.11.mod/client/Makefile.am	2020-09-11 11:31:44.280000000 +0800
@@ -25,7 +23,7 @@
 endif
 
 bin_PROGRAMS = boinc_client boinccmd
-if !OS_WIN32
+if OS_DARWIN
 bin_PROGRAMS += switcher
 endif
 
