diff -uNr boinc-client_release-7.16-7.16.11/client/hostinfo_unix.cpp boinc-client_release-7.16-7.16.11.mod/client/hostinfo_unix.cpp
--- boinc-client_release-7.16-7.16.11/client/hostinfo_unix.cpp	2020-08-30 16:04:59.000000000 +0800
+++ boinc-client_release-7.16-7.16.11.mod/client/hostinfo_unix.cpp	2020-09-11 11:47:27.990000000 +0800
@@ -1414,11 +1414,11 @@
 #if HAVE_SYS_UTSNAME_H
     struct utsname u;
     uname(&u);
-#ifdef ANDROID
+#ifdef __ANDROID__
     safe_strcpy(os_name, "Android");
 #else
     safe_strcpy(os_name, u.sysname);
-#endif //ANDROID
+#endif // __ANDROID__
 #if defined(__EMX__) // OS2: version is in u.version
     safe_strcpy(os_version, u.version);
 #elif defined(__HAIKU__)
