diff -uNr boinc-client_release-7.16-7.16.6/lib/synch.cpp boinc-client_release-7.16-7.16.6.mod/lib/synch.cpp
--- boinc-client_release-7.16-7.16.6/lib/synch.cpp	2020-03-23 07:13:30.000000000 +0200
+++ boinc-client_release-7.16-7.16.6.mod/lib/synch.cpp	2020-04-16 14:37:09.972107225 +0300
@@ -18,7 +18,7 @@
 
 // interfaces for accessing sempahores
 
-#ifndef ANDROID
+#ifndef __ANDROID__
 
 #include "config.h"
 #include <cstdio>
