diff -uNr boinc/version.h.in boinc.mod/version.h.in
--- boinc/version.h.in	2022-07-18 18:32:21.000000000 +0800
+++ boinc.mod/version.h.in	2022-09-11 21:07:05.475844109 +0800
@@ -22,7 +22,7 @@
 #define BOINC_VERSION_STRING "@BOINC_VERSION_STRING@"
 
 /* Package is a pre-release (Alpha/Beta) package */
-#define BOINC_PRERELEASE 1
+//#define BOINC_PRERELEASE 1
 
 #if (defined(_WIN32) || defined(__APPLE__))
 /* Name of package */
