diff -uNr boinc-client_release-7.18-7.18.1/version.h.in boinc-client_release-7.18-7.18.1.mod/version.h.in
--- boinc-client_release-7.18-7.18.1/version.h.in	2021-08-03 23:52:19.000000000 +0800
+++ boinc-client_release-7.18-7.18.1.mod/version.h.in	2021-12-29 21:29:50.585455988 +0800
@@ -22,7 +22,7 @@
 #define BOINC_VERSION_STRING "@BOINC_VERSION_STRING@"
 
 /* Package is a pre-release (Alpha/Beta) package */
-#define BOINC_PRERELEASE 1
+//#define BOINC_PRERELEASE 1
 
 #if (defined(_WIN32) || defined(__APPLE__))
 /* Name of package */
