diff -u -r ../socat-1.7.2.3/configure ./configure
--- ../socat-1.7.2.3/configure	2013-03-25 17:46:09.000000000 +0100
+++ ./configure	2014-02-19 01:01:02.000000000 +0100
@@ -3884,7 +3884,7 @@
 done
 
 
-for ac_header in sys/types.h netinet/in.h arpa/nameser.h netdb.h resolv.h
+for ac_header in sys/types.h netinet/in.h arpa/nameser.h netdb.h
 do :
   as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
 ac_fn_c_check_header_compile "$LINENO" "$ac_header" "$as_ac_Header" "#ifdef HAVE_SYS_TYPES_H
