--- a/xio-udp.c
+++ b/xio-udp.c
@@ -291,6 +291,7 @@
       switch (xioparms.default_ip) {
       case '4': pf = PF_INET; break;
       case '6': pf = PF_INET6; break;
+      case '0': pf = PF_INET; break;
       default: break;		/* includes \0 */
       }
 #elif WITH_IP6
@@ -544,6 +545,7 @@
       switch (xioparms.default_ip) {
       case '4': pf = PF_INET; break;
       case '6': pf = PF_INET6; break;
+      case '0': pf = PF_INET; break;
       default: break;		/* includes \0 */
       }
 #elif WITH_IP6
@@ -632,6 +634,7 @@
       switch (xioparms.default_ip) {
       case '4': pf = PF_INET; break;
       case '6': pf = PF_INET6; break;
+      case '0': pf = PF_INET; break;
       default: break;		/* includes \0 */
       }
 #elif WITH_IP6
