diff -u -r ../socat-1.7.3.0/sslcls.c ./sslcls.c
--- ../socat-1.7.3.0/sslcls.c	2015-01-24 05:15:22.000000000 -0500
+++ ./sslcls.c	2015-03-06 15:30:30.168143538 -0500
@@ -375,7 +375,7 @@
 }
 #endif /* WITH_FIPS */
 
-#if OPENSSL_VERSION_NUMBER >= 0x00908000L
+#if OPENSSL_VERSION_NUMBER >= 0x00908000L && !defined(OPENSSL_NO_COMP)
 const COMP_METHOD *sycSSL_get_current_compression(SSL *ssl) {
    const COMP_METHOD *result;
    Debug1("SSL_get_current_compression(%p)", ssl);
