diff -u -r ../socat-1.7.3.0/sslcls.h ./sslcls.h
--- ../socat-1.7.3.0/sslcls.h	2015-01-24 05:15:22.000000000 -0500
+++ ./sslcls.h	2015-03-06 15:30:17.128143529 -0500
@@ -55,7 +55,7 @@
 
 int sycFIPS_mode_set(int onoff);
 
-#if OPENSSL_VERSION_NUMBER >= 0x00908000L
+#if OPENSSL_VERSION_NUMBER >= 0x00908000L && !defined(OPENSSL_NO_COMP)
 const COMP_METHOD *sycSSL_get_current_compression(SSL *ssl);
 const COMP_METHOD *sycSSL_get_current_expansion(SSL *ssl);
 const char *sycSSL_COMP_get_name(const COMP_METHOD *comp);
