--- spice-0.14.91/Makefile.in	2020-10-26 12:20:20.000000000 +0000
+++ spice-0.14.91-fix/Makefile.in	2021-07-04 16:19:48.925982437 +0000
@@ -425,7 +425,7 @@
 valgrind_tools = @valgrind_tools@
 NULL = 
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = subprojects/spice-common server docs tools
+SUBDIRS = subprojects/spice-common server docs
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = spice-server.pc
 DISTCHECK_CONFIGURE_FLAGS = \
--- spice-0.14.91/Makefile.am   2020-10-26 12:18:53.000000000 +0000
+++ spice-0.14.91-fix/Makefile.am       2021-07-04 16:20:05.886952189 +0000
@@ -1,7 +1,7 @@
 NULL =
 ACLOCAL_AMFLAGS = -I m4

-SUBDIRS = subprojects/spice-common server docs tools
+SUBDIRS = subprojects/spice-common server docs

 if ENABLE_TESTS
 check-valgrind:
