--- ../configure.orig	2019-10-24 14:56:07.273121338 +0200
+++ ./configure	2019-10-24 14:58:24.726415961 +0200
@@ -21930,11 +21930,8 @@
           LIBS="$ac_save_LIBS"
        fi
      fi
-     GLIB_CFLAGS=""
-     GLIB_LIBS=""
-     { { echo "$as_me:$LINENO: error: Cannot find glib - aborting" >&5
-echo "$as_me: error: Cannot find glib - aborting" >&2;}
-   { (exit 1); exit 1; }; }
+     GLIB_CFLAGS="-I@TERMUX_PREFIX@/include/glib-2.0 -I@TERMUX_PREFIX@/lib/glib-2.0/include"
+     GLIB_LIBS="-lglib-2.0"
   fi
 
 
