diff -uNr dopewars-1.5.12/src/serverside.c dopewars-1.5.12.mod/src/serverside.c
--- dopewars-1.5.12/src/serverside.c	2005-12-30 23:26:13.000000000 +0200
+++ dopewars-1.5.12.mod/src/serverside.c	2020-04-28 03:01:36.285480545 +0300
@@ -800,7 +800,7 @@
     dopelog(1, LF_SERVER, _("Maintaining pid file %s"), PidFile);
     fprintf(fp, "%ld\n", (long)getpid());
     fclose(fp);
-    chmod(PidFile, S_IREAD | S_IWRITE);
+    chmod(PidFile, S_IRUSR | S_IWUSR);
   } else {
     gchar *OpenError = ErrStrFromErrno(errno);
     g_warning(_("Cannot create pid file %s: %s"), PidFile, OpenError);
@@ -1089,7 +1089,7 @@
 }
 
 #ifndef CYGWIN
-static gchar sockpref[] = "/tmp/.dopewars";
+static gchar sockpref[] = "@TERMUX_PREFIX@/tmp/.dopewars";
 
 static gchar *GetLocalSockDir(void)
 {
@@ -1192,7 +1192,7 @@
   if (localsock == -1) {
     dopelog(0, LF_SERVER,
             _("Could not set up Unix domain socket for admin "
-              "connections - check permissions on /tmp!"));
+              "connections - check permissions on @TERMUX_PREFIX@/tmp!"));
   }
 #endif
 
