diff -uNr proxychains-ng-4.14/src/main.c proxychains-ng-4.14.mod/src/main.c
--- proxychains-ng-4.14/src/main.c	2019-03-17 01:04:10.000000000 +0200
+++ proxychains-ng-4.14.mod/src/main.c	2019-12-13 20:38:27.768235332 +0200
@@ -65,8 +65,8 @@
 
 int main(int argc, char *argv[]) {
 	char *path = NULL;
-	char buf[256];
-	char pbuf[256];
+	char buf[PATH_MAX];
+	char pbuf[PATH_MAX];
 	int start_argv = 1;
 	int quiet = 0;
 	size_t i;
