diff -uNr proxychains-ng-4.13/Makefile proxychains-ng-4.13.mod/Makefile
--- proxychains-ng-4.13/Makefile	2018-06-25 02:26:05.000000000 +0300
+++ proxychains-ng-4.13.mod/Makefile	2019-03-05 19:35:28.924332422 +0200
@@ -8,7 +8,7 @@
 exec_prefix = /usr/local
 bindir = $(exec_prefix)/bin
 
-prefix = /usr/local/
+prefix = @TERMUX_PREFIX@
 includedir = $(prefix)/include
 libdir = $(prefix)/lib
 sysconfdir=$(prefix)/etc
@@ -25,7 +25,7 @@
 CFLAGS  += -Wall -O0 -g -std=c99 -D_GNU_SOURCE -pipe
 NO_AS_NEEDED = -Wl,--no-as-needed
 LIBDL   = -ldl
-LDFLAGS = -fPIC $(NO_AS_NEEDED) $(LIBDL) -lpthread
+LDFLAGS = -fPIC $(NO_AS_NEEDED) $(LIBDL)
 INC     = 
 PIC     = -fPIC
 AR      = $(CROSS_COMPILE)ar
