--- ../cache/cgdb-0.7.0/cgdb/cgdb.cpp	2017-03-10 16:44:18.000000000 +0000
+++ ./cgdb/cgdb.cpp	2018-10-10 05:05:23.164961985 +0000
@@ -1523,10 +1523,10 @@
 
     /* Shut down debugger */
     tgdb_shutdown(tgdb);
-
+#if !defined(__ANDROID__)
     if (tty_set_attributes(STDIN_FILENO, &term_attributes) == -1)
         clog_error(CLOG_CGDB, "tty_reset error");
-
+#endif
     /* Close our logfiles */
     tgdb_close_logfiles();
 
@@ -1768,12 +1768,12 @@
         clog_error(CLOG_CGDB, "Unable to init readline");
         cgdb_cleanup_and_exit(-1);
     }
-
+#if !defined(__ANDROID__)
     if (tty_cbreak(STDIN_FILENO, &term_attributes) == -1) {
         clog_error(CLOG_CGDB, "tty_cbreak error");
         cgdb_cleanup_and_exit(-1);
     }
-
+#endif
     if (init_kui() == -1) {
         clog_error(CLOG_CGDB, "init_kui error");
         cgdb_cleanup_and_exit(-1);
