diff -u -r ../cgdb-0.7.0/configure ./configure
--- ../cgdb-0.7.0/configure	2017-03-21 14:30:21.000000000 +0100
+++ ./configure	2017-05-23 00:03:31.352677383 +0200
@@ -6146,41 +6146,8 @@
 	curses_lib_name="curses"
 fi
 
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking Checking attribute printf support" >&5
-$as_echo_n "checking Checking attribute printf support... " >&6; }
-    if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
-else
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-#include <stdlib.h>
-
-static void test(const char *fmt, ...) __attribute__((format(__printf__, 1, 2)));
-static void test(const char *fmt, ...) { (void)fmt; }
-
-int main() {
-    test("foobar");
-    return 0;
-}
-_ACEOF
-if ac_fn_c_try_run "$LINENO"; then :
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-$as_echo "yes" >&6; }
-
 $as_echo "#define HAVE_ATTRIBUTE_PRINTF 1" >>confdefs.h
 
-else
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-fi
-rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
-  conftest.$ac_objext conftest.beam conftest.$ac_ext
-fi
-
 
 
 if test "X$bash_cv_termcap_lib" = "X"; then
