--- ./CMakeLists.txt	2021-07-04 13:00:34.583855568 +0530
+++ ./CMakeLists.txt.mod	2021-07-04 13:01:48.993855539 +0530
@@ -30,11 +30,6 @@
     set(CMAKE_BUILD_TYPE "${DEFAULT_BUILD_TYPE}")
 endif()
 
-# Error out when linking statically, it doesn't work.
-if (CMAKE_EXE_LINKER_FLAGS MATCHES ".*-static.*")
-    message(FATAL_ERROR "Fish does not support static linking")
-endif()
-
 # Force colored warnings in Ninja's output, if the compiler has -fdiagnostics-color support.
 # Rationale in https://github.com/ninja-build/ninja/issues/814
 if (CMAKE_GENERATOR STREQUAL "Ninja" AND
@@ -185,7 +180,7 @@
 target_sources(fishlib PRIVATE ${FISH_HEADERS})
 target_link_libraries(fishlib
   ${CURSES_LIBRARY} ${CURSES_EXTRA_LIBRARY} Threads::Threads ${CMAKE_DL_LIBS}
-  ${PCRE2_LIB} ${Intl_LIBRARIES} ${ATOMIC_LIBRARY})
+  ${PCRE2_LIB} ${Intl_LIBRARIES} ${ATOMIC_LIBRARY} -landroid-spawn)
 target_include_directories(fishlib PRIVATE
   ${CURSES_INCLUDE_DIRS})
 
