diff -u -r ../fish-shell-master/src/common.cpp ./src/common.cpp
--- ../fish-shell-master/src/common.cpp	2015-12-16 10:35:17.000000000 -0500
+++ ./src/common.cpp	2015-12-17 05:09:51.542032656 -0500
@@ -90,8 +90,12 @@
     int trace_size = 0;
 
     trace_size = backtrace(trace, 32);
+#ifdef __ANDROID__
+    debug(0, L"Backtrace: Not available on Android");
+#else
     debug(0, L"Backtrace:");
     backtrace_symbols_fd(trace, trace_size, STDERR_FILENO);
+#endif
 }
 
 int fgetws2(wcstring *s, FILE *f)
