diff -u -r ../fish-3.2.0/share/tools/create_manpage_completions.py ./share/tools/create_manpage_completions.py
--- ../fish-3.2.0/share/tools/create_manpage_completions.py.o	2021-03-18 00:14:41.517297567 +0530
+++ ./share/tools/create_manpage_completions.py	2021-03-18 00:44:55.699988162 +0530
@@ -971,7 +971,7 @@
 def get_paths_from_man_locations():
     # Return all the paths to man(1) and man(8) files in the manpath
 
-    parent_paths = []
+    parent_paths = ["@TERMUX_PREFIX@/share/man"]
 
     # Most (GNU, macOS, Haiku) modern implementations of man support being called with `--path`.
     # Traditional implementations require a second `manpath` program: examples include FreeBSD and Solaris.
