diff -u -r ../fish-3.1.0/share/tools/create_manpage_completions.py ./share/tools/create_manpage_completions.py
--- ../fish-3.1.0/share/tools/create_manpage_completions.py	2020-02-12 14:04:07.000000000 +0000
+++ ./share/tools/create_manpage_completions.py	2020-02-15 21:40:21.963831000 +0000
@@ -953,17 +953,7 @@
     if os.getenv("MANPATH"):
         parent_paths = os.getenv("MANPATH").strip().split(":")
     else:
-        # Some systems have manpath, others have `man --path` (like Haiku).
-        for prog in [["manpath"], ["man", "--path"]]:
-            try:
-                proc = subprocess.Popen(
-                    prog, stdout=subprocess.PIPE, stderr=subprocess.PIPE
-                )
-            except OSError:  # Command does not exist, keep trying
-                continue
-            break  # Command exists, use it.
-        manpath, err_data = proc.communicate()
-        parent_paths = manpath.decode().strip().split(":")
+        parent_paths = ["@TERMUX_PREFIX@/share/man"]
     if (not parent_paths) or (proc and proc.returncode > 0):
         # HACK: Use some fallbacks in case we can't get anything else.
         # `mandoc` does not provide `manpath` or `man --path` and $MANPATH might not be set.
