diff -u -r ../fish-2.4.0/share/tools/create_manpage_completions.py ./share/tools/create_manpage_completions.py
--- ../fish-2.4.0/share/tools/create_manpage_completions.py	2016-11-07 22:20:54.000000000 -0500
+++ ./share/tools/create_manpage_completions.py	2016-12-23 15:09:00.703224799 -0500
@@ -857,12 +857,7 @@
 def get_paths_from_manpath():
     # Return all the paths to man(1) and man(8) files in the manpath
     import subprocess, os
-    proc = subprocess.Popen(['manpath'], stdout=subprocess.PIPE)
-    manpath, err_data = proc.communicate()
-    parent_paths = manpath.decode().strip().split(':')
-    if not parent_paths:
-        sys.stderr.write("Unable to get the manpath (tried manpath)\n")
-        sys.exit(-1)
+    parent_paths = ['@TERMUX_PREFIX@/share/man']
     result = []
     for parent_path in parent_paths:
         for section in ['man1', 'man6', 'man8']:
