diff -u -r ../fish-2.2.0/env_universal_common.h ./env_universal_common.h
--- ../fish-2.2.0/env_universal_common.h	2015-07-03 15:46:59.000000000 -0400
+++ ./src/env_universal_common.h	2015-07-25 17:52:47.555388223 -0400
@@ -123,8 +123,10 @@
         // Default meta-strategy to use the 'best' notifier for the system
         strategy_default,
         
+#ifndef __ANDROID__
         // Use a value in shared memory. Simple, but requires polling and therefore semi-frequent wakeups.
         strategy_shmem_polling,
+#endif
         
         // Strategy that uses a named pipe. Somewhat complex, but portable and doesn't require polling most of the time.
         strategy_named_pipe,
