diff -u -r ../fish-2.4b1/share/functions/__fish_print_help.fish ./share/functions/__fish_print_help.fish
--- ../fish-2.4b1/share/functions/__fish_print_help.fish	2016-10-18 10:17:06.000000000 -0400
+++ ./share/functions/__fish_print_help.fish	2016-10-19 17:37:08.948931562 -0400
@@ -40,7 +40,7 @@
 		set rLL -rLL=$cols[1]n
 	end
     if test -e "$__fish_datadir/man/man1/$item.1"
-	    set help (nroff -man -c -t $rLL "$__fish_datadir/man/man1/$item.1" ^/dev/null)
+	    set help (mandoc "$__fish_datadir/man/man1/$item.1" ^/dev/null)
     else if test -e "$__fish_datadir/man/man1/$item.1.gz"
 	    set help (gunzip -c "$__fish_datadir/man/man1/$item.1.gz" ^/dev/null | nroff -man -c -t $rLL ^/dev/null)
     end
@@ -98,6 +98,6 @@
 				# skip it
 			end
 		end
-	end | ul # post-process with `ul`, to interpret the old-style grotty escapes
+	end # post-process with `ul`, to interpret the old-style grotty escapes
 	echo # print a trailing blank line
 end
