diff -u -r ../fish-shell-master/share/functions/help.fish ./share/functions/help.fish
--- ../fish-shell-master/share/functions/help.fish	2016-03-12 15:26:01.000000000 -0500
+++ ./share/functions/help.fish	2016-03-14 12:08:13.804718800 -0400
@@ -125,7 +125,7 @@
 		end
 	else
 		# Go to the web. Only include one dot in the version string
-		set -l version_string (echo $FISH_VERSION| cut -d . -f 1,2)
+		set -l version_string current # Termux: We're building from git master for now.
 		set page_url http://fishshell.com/docs/$version_string/$fish_help_page
 	end
 	
