--- a/share/functions/__fish_print_apt_packages.fish
+++ b/share/functions/__fish_print_apt_packages.fish
@@ -9,7 +9,7 @@
 
     set -l search_term (commandline -ct | string replace -ar '[\'"\\\\]' '' | string lower)
 
-    if ! test -f /var/lib/dpkg/status
+    if ! test -f @TERMUX_PREFIX@/var/lib/dpkg/status
         return 1
     end
 
@@ -59,6 +59,6 @@
     print pkg "\t" desc
     installed=0 # Prevent multiple description translations from being printed
   }
-}' </var/lib/dpkg/status
+}' <@TERMUX_PREFIX@/var/lib/dpkg/status
     end
 end
