https://reviews.freebsd.org/D14058

diff -u -r ../fish-2.7.1/src/common.h ./src/common.h
--- ../fish-2.7.1/src/common.h	2017-12-22 16:16:29.000000000 +0000
+++ ./src/common.h	2018-09-21 22:05:33.472645301 +0000
@@ -658,7 +658,7 @@
 void append_format(wcstring &str, const wchar_t *format, ...);
 void append_formatv(wcstring &str, const wchar_t *format, va_list ap);
 
-#ifdef __cpp_lib_make_unique
+#if __cplusplus >= 201402L
 using std::make_unique;
 #else
 /// make_unique implementation
