diff -u -r ../fish-2.7.0/src/builtin_argparse.cpp ./src/builtin_argparse.cpp
--- ../fish-2.7.0/src/builtin_argparse.cpp	2017-11-23 06:45:49.000000000 +0100
+++ ./src/builtin_argparse.cpp	2017-11-30 00:33:41.800475273 +0100
@@ -444,7 +444,7 @@
 
         if (!opt_spec->long_flag.empty()) {
             long_options.get()[i++] = {opt_spec->long_flag.c_str(), arg_type, NULL,
-                                       opt_spec->short_flag};
+                                       (int) opt_spec->short_flag};
         }
     }
     long_options.get()[i] = {NULL, 0, NULL, 0};
