diff -u -r ../fish-2.4b1/src/env_universal_common.cpp ./src/env_universal_common.cpp
--- ../fish-2.4b1/src/env_universal_common.cpp	2016-10-18 10:17:06.000000000 -0400
+++ ./src/env_universal_common.cpp	2016-10-19 17:42:06.900520935 -0400
@@ -137,7 +137,7 @@
         }
 
         // /tmp/fish.user
-        std::string tmpdir = "/tmp/fish.";
+        std::string tmpdir = "@TERMUX_PREFIX@/tmp/fish.";
         tmpdir.append(uname);
         if (check_runtime_path(tmpdir.c_str()) != 0) {
             debug(0,
