https://github.com/termux/termux-packages/issues/15677

--- a/src/proc.h
+++ b/src/proc.h
@@ -117,7 +117,7 @@
     }
 
     /// \return if we are stopped (as in SIGSTOP).
-    bool stopped() const { return WIFSTOPPED(status_); }
+    bool stopped() const { return WIFSTOPPED(status_) && !WIFCONTINUED(status_); }
 
     /// \return if we are continued (as in SIGCONT).
     bool continued() const { return WIFCONTINUED(status_); }
