diff -u -r ../fish-2.1.1/wutil.cpp ./wutil.cpp
--- ../fish-2.1.1/wutil.cpp	2014-09-24 05:51:07.000000000 -0400
+++ ./wutil.cpp	2015-02-05 17:40:19.462415848 -0500
@@ -324,11 +324,13 @@
     // See #808
     return strerror(err);
 #else
+# ifndef __ANDROID__
     if (err >= 0 && err < sys_nerr && sys_errlist[err] != NULL)
     {
         return sys_errlist[err];
     }
     else
+# endif
     {
         int saved_err = errno;
 
