diff --git a/src/prompt/main.rs b/src/prompt/main.rs
index f27e9c5..e3bf353 100644
--- a/src/prompt/main.rs
+++ b/src/prompt/main.rs
@@ -4,9 +4,7 @@ use crate::execute;
 use crate::libs;
 use crate::shell;
 
-const DEFAULT_PROMPT: &str = "${COLOR_STATUS}$USER${RESET}\
-    @${COLOR_STATUS}$HOSTNAME${RESET}: \
-    ${COLOR_STATUS}$CWD${RESET}$ ";
+const DEFAULT_PROMPT: &str = "${COLOR_STATUS}>> $CWD${RESET}$ ";
 use super::preset::apply_preset_item;
 use super::preset::apply_pyenv;
 
