--- a/completions/bun.bash
+++ b/completions/bun.bash
@@ -57,7 +57,7 @@
     local re_prev_script="(^| )${prev}($| )";
     [[
         ( "${COMPREPLY[*]}" =~ ${re_prev_script} && -n "${COMP_WORDS[2]}" ) || \
-            ( "${COMPREPLY[*]}" =~ ${re_comp_word_script} )
+            ( -n "${re_comp_word_script}" && "${COMPREPLY[*]}" =~ ${re_comp_word_script} )
     ]] && {
         local filtered_reply=();
         local reply_word script_name keep;
