https://github.com/termux/termux-packages/issues/29240

--- a/Makefile
+++ b/Makefile
@@ -122,7 +122,7 @@ else ifeq ($(DEBUG), 2)
           -fno-optimize-sibling-calls
     LDFLAGS += -fsanitize=address,undefined
 else
-    COMMONFLAGS += $(SYMBOL_FLAGS) -O2 -fstack-protector-strong -Wformat -ffunction-sections -fdata-sections -Werror=format-security -D_FORTIFY_SOURCE=2
+    COMMONFLAGS += $(SYMBOL_FLAGS) -O2 -fstack-protector-strong -Wformat -ffunction-sections -fdata-sections -D_FORTIFY_SOURCE=2
 endif
 
 ifneq ($(strip $(KEW_VERSION)),)
