--- ./qtbase/mkspecs/linux-termux-clang/qmake.conf	1970-01-01 05:30:00.000000000 +0530
+++ ./qtbase/mkspecs/linux-termux-clang/qmake.conf	2017-07-14 23:06:45.553367565 +0530
@@ -0,0 +1,32 @@
+MAKEFILE_GENERATOR      = UNIX
+CONFIG                 += incremental unversioned_libname unversioned_soname plugin_with_soname qt
+QMAKE_INCREMENTAL_STYLE = sublib
+
+include(../common/linux.conf)
+include(../common/gcc-base-unix.conf)
+include(../common/clang.conf)
+
+QMAKE_CC                = @TERMUX_HOST_PLATFORM@-clang
+QMAKE_CXX               = @TERMUX_HOST_PLATFORM@-clang++
+QMAKE_LINK              = @TERMUX_HOST_PLATFORM@-clang++
+QMAKE_LINK_SHLIB        = @TERMUX_HOST_PLATFORM@-clang++
+
+QMAKE_AR                = @TERMUX_HOST_PLATFORM@-ar cqs
+QMAKE_OBJCOPY           = @TERMUX_HOST_PLATFORM@-objcopy
+QMAKE_NM                = @TERMUX_HOST_PLATFORM@-nm -P
+QMAKE_STRIP             = @TERMUX_HOST_PLATFORM@-strip
+
+QMAKE_CFLAGS            = @CFLAGS@
+QMAKE_CXXFLAGS          = @CXXFLAGS@
+QMAKE_LFLAGS            = @LDFLAGS@
+
+QMAKE_LFLAGS_APP        = -Wl,--no-undefined -Wl,-z,noexecstack -shared
+QMAKE_LFLAGS_SHLIB      = -Wl,--no-undefined -Wl,-z,noexecstack -shared
+QMAKE_LFLAGS_PLUGIN     = $$QMAKE_LFLAGS_SHLIB
+QMAKE_LFLAGS_NOUNDEF    = -Wl,--no-undefined
+
+QMAKE_LIBS_THREAD       =
+QMAKE_LIBS_X11          =
+QMAKE_LIBS_OPENGL       =
+QMAKE_LIBS_OPENGL_ES2   = -lGLESv3
+load(qt_config)
--- ./qtbase/mkspecs/linux-termux-clang/qplatformdefs.h	1970-01-01 05:30:00.000000000 +0530
+++ ./qtbase/mkspecs/linux-termux-clang/qplatformdefs.h	2017-07-14 23:08:02.031153237 +0530
@@ -0,0 +1,43 @@
+/****************************************************************************
+**
+** Copyright (C) 2016 The Qt Company Ltd.
+** Contact: https://www.qt.io/licensing/
+**
+** This file is part of the qmake spec of the Qt Toolkit.
+**
+** $QT_BEGIN_LICENSE:LGPL$
+** Commercial License Usage
+** Licensees holding valid commercial Qt licenses may use this file in
+** accordance with the commercial license agreement provided with the
+** Software or, alternatively, in accordance with the terms contained in
+** a written agreement between you and The Qt Company. For licensing terms
+** and conditions see https://www.qt.io/terms-conditions. For further
+** information use the contact form at https://www.qt.io/contact-us.
+**
+** GNU Lesser General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU Lesser
+** General Public License version 3 as published by the Free Software
+** Foundation and appearing in the file LICENSE.LGPL3 included in the
+** packaging of this file. Please review the following information to
+** ensure the GNU Lesser General Public License version 3 requirements
+** will be met: https://www.gnu.org/licenses/lgpl-3.0.html.
+**
+** GNU General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU
+** General Public License version 2.0 or (at your option) the GNU General
+** Public license version 3 or any later version approved by the KDE Free
+** Qt Foundation. The licenses are as published by the Free Software
+** Foundation and appearing in the file LICENSE.GPL2 and LICENSE.GPL3
+** included in the packaging of this file. Please review the following
+** information to ensure the GNU General Public License requirements will
+** be met: https://www.gnu.org/licenses/gpl-2.0.html and
+** https://www.gnu.org/licenses/gpl-3.0.html.
+**
+** $QT_END_LICENSE$
+**
+****************************************************************************/
+
+#include "../android-clang/qplatformdefs.h"
+#define fseeko64 fseeko
+#define ftello64 ftello
+#define fopen64 fopen
