--- ./qtserialport/src/serialport/qserialport_unix.cpp	2017-06-20 12:29:20.000000000 +0530
+++ ../qserialport_unix.cpp	2017-10-01 10:21:47.899850693 +0530
@@ -66,7 +66,7 @@
 #  define __ANDROID_API__ 21
 # endif
 
-# if !defined(Q_OS_ANDROID) || (!defined(Q_PROCESSOR_X86) && __ANDROID_API__ < 21)
+# if 0
 struct termios2 {
     tcflag_t c_iflag;       /* input mode flags */
     tcflag_t c_oflag;       /* output mode flags */
