diff -uNr ne-3.3.4.orig/doc/default.keys ne-3.3.4/doc/default.keys
--- ne-3.3.4.orig/doc/default.keys	2026-09-21 14:13:17.071926639 +0530
+++ ne-3.3.4/doc/default.keys	2026-09-21 14:13:17.119926639 +0530
@@ -302,7 +302,7 @@
 
 # Delete
 
-KEY	7F	DeleteChar
+KEY	7F	BackSpace
 
 KEY	1cf	OpenClip
 KEY	1d0	Paragraph
diff -uNr ne-3.3.4.orig/src/inputclass.c ne-3.3.4/src/inputclass.c
--- ne-3.3.4.orig/src/inputclass.c	2026-09-21 14:13:17.075926639 +0530
+++ ne-3.3.4/src/inputclass.c	2026-09-21 14:13:17.111926639 +0530
@@ -95,7 +95,7 @@
 	NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL,  /* P Q R S T U V W X Y Z [ \ ] ^ _  */
 	NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL,  /* ` a b c d e f g h i j k l m n o  */
 	NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL,        /* p q r s t u v w x y z { | } ~    */
-	DELETECHAR_ABBREV,                         /*  127  7f */
+	BACKSPACE_ABBREV,                          /*  127  7f */
 
 	/* Control-meta-letter bindings (Ctrl-Alt-X)  128..159 80..9f */
 
