diff -u -r ../ne-3.1.1/makefile ./makefile
--- ../ne-3.1.1/makefile	2017-06-04 22:43:12.000000000 +0200
+++ ./makefile	2017-06-06 09:45:51.221176799 +0200
@@ -8,10 +8,12 @@
 # and installed under the $(PREFIX) hierarchy. You can even use "make install PREFIX=$HOME/<dir>"
 # to install ne locally into the directory <dir>.
 
-PREFIX=/usr/local
+PREFIX?=/usr/local
 
 PROGRAM       = ne
 
+STRIP?=strip
+
 ifeq ($(OS), Windows_NT)
 	OS := Windows
 else
@@ -20,7 +22,7 @@
 
 
 build: docs
-	( cd src; make clean; make NE_GLOBAL_DIR=$(PREFIX)/share/ne; strip ne )
+	( cd src; make clean; make NE_GLOBAL_DIR=$(PREFIX)/share/ne; $(STRIP) ne )
 
 docs:
 	( cd doc; make )
