diff -u -r ../ne-3.1.1/src/makefile ./src/makefile
--- ../ne-3.1.1/src/makefile	2017-04-04 12:35:59.000000000 +0200
+++ ./src/makefile	2017-06-06 10:07:56.594364223 +0200
@@ -28,7 +28,7 @@
 # where ne tries to read system-wide information (configuration
 # files, etc.).
 
-NE_GLOBAL_DIR = /usr/local/share/ne
+NE_GLOBAL_DIR ?= /usr/local/share/ne
 
 PROGRAM       = ne
 
@@ -81,7 +81,7 @@
 NE_DEBUG=
 NE_TEST=
 
-CC=c99
+CC?=c99
 
 ifeq ($(CC),gcc)
 GCCFLAGS=-std=c99 -Wall -Wno-parentheses
@@ -96,7 +96,7 @@
 	$(if $(NE_ANSI),    -DTERMCAP -DANSI,)
 
 
-LIBS=$(if $(NE_TERMCAP)$(NE_ANSI),,-lcurses)
+LIBS=$(if $(NE_TERMCAP)$(NE_ANSI),,-lncursesw)
 
 ne:	$(OBJS) $(if $(NE_TERMCAP)$(NE_ANSI),$(TERMCAPOBJS),)
 	$(CC) $(OPTS) $(LDFLAGS) $(if $(NE_TEST), -coverage -lefence,) $^ -lm $(LIBS) -o $(PROGRAM)
