diff -uNr tsocks-1.8/Makefile.in tsocks-1.8.mod/Makefile.in
--- tsocks-1.8/Makefile.in	2002-03-16 12:12:40.000000000 +0200
+++ tsocks-1.8.mod/Makefile.in	2019-08-16 03:39:24.865614750 +0300
@@ -47,7 +47,7 @@
 	${SHCC} ${CFLAGS} ${INCLUDES} -o ${INSPECT} ${INSPECT}.c ${COMMON}.o ${LIBS} 
 
 ${SAVE}: ${SAVE}.c
-	${SHCC} ${CFLAGS} ${INCLUDES} -static -o ${SAVE} ${SAVE}.c
+	${SHCC} ${CFLAGS} ${INCLUDES} -o ${SAVE} ${SAVE}.c
 
 ${SHLIB}: ${OBJS} ${COMMON}.o ${PARSER}.o
 	${SHCC} ${CFLAGS} ${INCLUDES} -nostdlib -shared -o ${SHLIB} ${OBJS} ${COMMON}.o ${PARSER}.o ${DYNLIB_FLAGS} ${SPECIALLIBS} ${LIBS}
