diff -u -r ../ExtUtils-LibBuilder-0.08.orig/t/01-simple.t ./t/01-simple.t
--- ../ExtUtils-LibBuilder-0.08.orig/t/01-simple.t	2020-03-19 18:35:24.423661248 +0100
+++ ./t/01-simple.t	2020-03-19 18:37:50.234669799 +0100
@@ -59,7 +59,7 @@
 } elsif ($^O =~ /darwin/i) {
     $ENV{DYLD_LIBRARY_PATH} = ".";
 }
-elsif ($^O =~ /(?:linux|bsd|sun|sol|dragonfly|hpux|irix)/i) {
+elsif ($^O =~ /(?:linux|bsd|sun|sol|dragonfly|hpux|irix|android)/i) {
     $ENV{LD_LIBRARY_PATH} = ".";
     if ($^O =~ /irix/) {
         # trying my luck
