--- ./Makefile.PL.orig	2022-03-10 09:53:22.009385239 +0100
+++ ./Makefile.PL	2022-03-10 09:54:33.541385188 +0100
@@ -357,6 +357,7 @@
 	'/ssl111$exe/openssl.exe'        => '/ssl111$root',# VMS, VSI install
 	'/ssl1$exe/openssl.exe'          => '/ssl1$root',# VMS, VSI or HPE install
 	'/ssl$exe/openssl.exe'           => '/ssl$root', # VMS, HP install
+	'@TERMUX_PREFIX@/bin/openssl' => '@TERMUX_PREFIX@', # Termux
     );
 
     while (my $k = shift @guesses
