--- gmic-3.5.0/src/CImg.h	2024-12-31 19:11:35.000000000 +0800
+++ gmic-3.5.0.mod/src/CImg.h	2025-01-04 03:25:46.407412118 +0800
@@ -3274,7 +3274,11 @@
 
       ~X11_attr() {
         if (events_thread) {
+#ifndef __ANDROID__
           pthread_cancel(*events_thread);
+#else
+          pthread_kill(*events_thread, SIGUSR1);
+#endif
           delete events_thread;
         }
         pthread_cond_destroy(&wait_event);
@@ -9897,8 +9901,10 @@
     static void* _events_thread(void *arg) { // Thread to manage events for all opened display windows
       Display *const dpy = cimg::X11_attr::ref().display;
       XEvent event;
+#ifndef __ANDROID__
       pthread_setcanceltype(PTHREAD_CANCEL_DEFERRED,0);
       pthread_setcancelstate(PTHREAD_CANCEL_ENABLE,0);
+#endif
       if (!arg) for ( ; ; ) {
         cimg_lock_display();
         bool event_flag = XCheckTypedEvent(dpy,ClientMessage,&event);
@@ -9911,7 +9917,9 @@
             if (!cimg::X11_attr::ref().wins[i]->_is_closed && event.xany.window==cimg::X11_attr::ref().wins[i]->_window)
               cimg::X11_attr::ref().wins[i]->_handle_events(&event);
         cimg_unlock_display();
+#ifndef __ANDROID__
         pthread_testcancel();
+#endif
         cimg::sleep(8);
       }
       return 0;
@@ -68269,8 +68277,8 @@
         _cimg_test_temporary_path("D:\\Temp");
         _cimg_test_temporary_path("D:");
 #else
-        _cimg_test_temporary_path("/tmp");
-        _cimg_test_temporary_path("/var/tmp");
+        _cimg_test_temporary_path("@TERMUX_PREFIX@/tmp");
+        _cimg_test_temporary_path("@TERMUX_PREFIX@/var/tmp");
 #endif
         if (!path_found) {
           *s_path = 0;
