diff -uNr gmic-2.6.7/src/CImg.h gmic-2.6.7.mod/src/CImg.h
--- gmic-2.6.7/src/CImg.h	2019-06-28 09:56:12.000000000 +0300
+++ gmic-2.6.7.mod/src/CImg.h	2019-06-29 18:45:02.595895578 +0300
@@ -60774,8 +60774,8 @@
         _cimg_test_temporary_path("D:\\Temp");
         _cimg_test_temporary_path("D:");
 #else
-        _cimg_test_temporary_path("/tmp");
-        _cimg_test_temporary_path("/var/tmp");
+        _cimg_test_temporary_path("@TERMUX_PREFIX@/tmp");
+        _cimg_test_temporary_path("@TERMUX_PREFIX@/var/tmp");
 #endif
         if (!path_found) {
           *s_path = 0;
