--- ../cache/gmic-1.7.9/src/gmic_stdlib.gmic	2016-12-02 11:29:58.000000000 +0000
+++ ./src/gmic_stdlib.gmic	2017-05-04 00:49:42.898379733 +0000
@@ -17282,7 +17282,7 @@
     -if {narg({'${TMP}'})} _path_tmp=${TMP}
     -elif {narg({'${TEMP}'})} _path_tmp=${TEMP}
     -elif {narg({'${TMPDIR}'})} _path_tmp=${TMPDIR}
-    -elif {narg({'${HOME}'})} _path_tmp="/tmp"
+    -elif {narg({'${HOME}'})} _path_tmp="/data/data/com.termux/files/usr/tmp"
     -endif
     -if ${-is_windows} _path_tmp=$_path_tmp{`92`} -else _path_tmp=$_path_tmp/ -endif
   -endif
