diff -u -r ../gmic-2.2.3/src/Makefile ./src/Makefile
--- ../gmic-2.2.3/src/Makefile	2018-05-16 13:18:08.000000000 +0000
+++ ./src/Makefile	2018-05-18 22:44:33.038378433 +0000
@@ -61,7 +61,7 @@
 
 # Try to auto-detect target OS.
 OS = $(shell uname)
-USR = /usr
+USR = $(PREFIX)
 LIB = lib
 BIN = bin
 INCLUDE = include
@@ -136,9 +136,7 @@
 endif
 
 ifdef NOSTRIP
-STRIP = echo skip strip
-else
-STRIP = strip
+STRIP = echo skip stripe
 endif
 
 ifndef QMAKE
@@ -190,7 +188,7 @@
 # Enable multi-threading support.
 PARALLEL_CFLAGS = -Dgmic_is_parallel
 ifneq ($(OS),Windows)
-PARALLEL_LIBS = -lpthread
+PARALLEL_LIBS = #-lpthread
 endif
 
 # Enable parallelization in CImg, using OpenMP.
@@ -326,8 +324,8 @@
 GMIC_CLI_CFLAGS = $(MANDATORY_CFLAGS) $(ABORT_CFLAGS) $(PARALLEL_CFLAGS) $(FFTW_CFLAGS) $(CURL_CFLAGS) $(PNG_CFLAGS) $(JPEG_CFLAGS) $(TIFF_CFLAGS)
 GMIC_CLI_LIBS = $(MANDATORY_LIBS) $(PARALLEL_LIBS) $(FFTW_LIBS) $(CURL_LIBS) $(PNG_LIBS) $(JPEG_LIBS) $(TIFF_LIBS)
 ifeq ($(OS),Unix) # Unix.
-GMIC_CLI_CFLAGS += $(OPENMP_CFLAGS) $(X11_CFLAGS) $(OPENEXR_CFLAGS) $(OPENCV_CFLAGS) # $(XSHM_CFLAGS) $(MAGICK_CFLAGS)
-GMIC_CLI_LIBS += $(OPENMP_LIBS) $(X11_LIBS) $(OPENEXR_LIBS) $(OPENCV_LIBS) # $(XSHM_LIBS) # $(MAGICK_LIBS)
+GMIC_CLI_CFLAGS += $(MAGICK_CFLAGS) # $(OPENMP_CFLAGS) $(X11_CFLAGS) $(OPENEXR_CFLAGS) $(OPENCV_CFLAGS) # $(XSHM_CFLAGS) $(MAGICK_CFLAGS)
+GMIC_CLI_LIBS += $(MAGICK_LIBS) # $(OPENMP_LIBS) $(X11_LIBS) $(OPENEXR_LIBS) $(OPENCV_LIBS) # $(XSHM_LIBS) # $(MAGICK_LIBS)
 else
 ifeq ($(OS),Darwin) # MacOSX.
 GMIC_CLI_CFLAGS += $(X11_CFLAGS) $(OPENEXR_CFLAGS) $(OPENCV_CFLAGS)
@@ -339,8 +337,7 @@
 endif
 
 cli:
-	$(MAKE) "CFLAGS+=$(GMIC_CLI_CFLAGS) $(OPT_CFLAGS)" "LIBS+=$(GMIC_CLI_LIBS)" _cli
-	$(STRIP) gmic$(EXE)
+	$(MAKE) "CFLAGS+= $(CXXFLAGS) $(LDFLAGS) $(GMIC_CLI_CFLAGS) $(OPT_CFLAGS)" "LIBS+=$(GMIC_CLI_LIBS)" _cli
 
 debug:
 	$(MAKE) "CFLAGS+=$(GMIC_CLI_CFLAGS) $(DEBUG_CFLAGS)" "LIBS+=$(GMIC_CLI_LIBS)" _cli
