--- ../cache/gmic-2.3.1/src/Makefile	2018-06-25 12:57:35.000000000 +0000
+++ ./src/Makefile	2018-06-27 05:33:59.637954036 +0000
@@ -60,7 +60,7 @@
 
 # Try to auto-detect target OS.
 OS = $(shell uname)
-USR = /usr
+USR = $(PREFIX)
 LIB = lib
 BIN = bin
 INCLUDE = include
@@ -136,8 +136,6 @@
 
 ifdef NOSTRIP
 STRIP = echo skip strip
-else
-STRIP = strip
 endif
 
 ifndef QMAKE
@@ -154,7 +152,7 @@
 
 # Minimal set of flags mandatory to compile G'MIC.
 MANDATORY_CFLAGS = -Dgmic_build -Dcimg_use_zlib `pkg-config --cflags zlib || echo -I$(USR)/$(INCLUDE)` $(PRERELEASE_CFLAGS) $(EXTRA_CFLAGS)
-MANDATORY_LIBS = `pkg-config --libs zlib || echo -lz` $(EXTRA_LIBS)
+# MANDATORY_LIBS = `pkg-config --libs zlib || echo -lz` $(EXTRA_LIBS)
 ifndef NO_SRIPDLIB
 MANDATORY_CFLAGS += -std=c++11 -pedantic
 endif
@@ -189,7 +187,7 @@
 # Enable multi-threading support.
 PARALLEL_CFLAGS = -Dgmic_is_parallel
 ifneq ($(OS),Windows)
-PARALLEL_LIBS = -lpthread
+PARALLEL_LIBS = #-lpthread
 endif
 
 # Enable parallelization in CImg, using OpenMP.
@@ -238,7 +236,7 @@
 # (http://www.libpng.org/pub/png/libpng.html)
 ifneq ($(OS),Darwin)
 PNG_CFLAGS = -Dcimg_use_png `pkg-config --cflags libpng`
-PNG_LIBS = `pkg-config --libs libpng || echo -lpng -lz`
+PNG_LIBS = `echo -lpng`
 else
 ifeq (,$(wildcard /tmp/skl))
 PNG_CFLAGS = -Dcimg_use_png `pkg-config --cflags libpng`
@@ -277,7 +275,7 @@
 # Enable support of most classical image file formats, using the GraphicsMagick++ library.
 # (http://www.graphicsmagick.org/Magick++/)
 MAGICK_CFLAGS = -Dcimg_use_magick `pkg-config --cflags GraphicsMagick++ || echo -I$(USR)/$(INCLUDE)/GraphicsMagick`
-MAGICK_LIBS = `pkg-config --libs GraphicsMagick++ || echo -lGraphicsMagick++`
+MAGICK_LIBS = `echo -L/data/data/com.termux/files/usr/lib -lGraphicsMagick++ -lGraphicsMagick -lz`
 
 # Enable native support of EXR image files, using the OpenEXR library.
 # (http://www.openexr.com/)
@@ -325,8 +323,8 @@
 GMIC_CLI_CFLAGS = $(MANDATORY_CFLAGS) $(ABORT_CFLAGS) $(PARALLEL_CFLAGS) $(FFTW_CFLAGS) $(CURL_CFLAGS) $(PNG_CFLAGS) $(JPEG_CFLAGS) $(TIFF_CFLAGS)
 GMIC_CLI_LIBS = $(MANDATORY_LIBS) $(PARALLEL_LIBS) $(FFTW_LIBS) $(CURL_LIBS) $(PNG_LIBS) $(JPEG_LIBS) $(TIFF_LIBS)
 ifeq ($(OS),Unix) # Unix.
-GMIC_CLI_CFLAGS += $(OPENMP_CFLAGS) $(X11_CFLAGS) $(OPENEXR_CFLAGS) $(OPENCV_CFLAGS) # $(XSHM_CFLAGS) $(MAGICK_CFLAGS)
-GMIC_CLI_LIBS += $(OPENMP_LIBS) $(X11_LIBS) $(OPENEXR_LIBS) $(OPENCV_LIBS) # $(XSHM_LIBS) # $(MAGICK_LIBS)
+GMIC_CLI_CFLAGS += $(MAGICK_CFLAGS) # $(OPENMP_CFLAGS) $(X11_CFLAGS) $(OPENEXR_CFLAGS) $(OPENCV_CFLAGS) # $(XSHM_CFLAGS) $(MAGICK_CFLAGS)
+GMIC_CLI_LIBS += $(MAGICK_LIBS) # $(OPENMP_LIBS) $(X11_LIBS) $(OPENEXR_LIBS) $(OPENCV_LIBS) # $(XSHM_LIBS) # $(MAGICK_LIBS)
 else
 ifeq ($(OS),Darwin) # MacOSX.
 GMIC_CLI_CFLAGS += $(X11_CFLAGS) $(OPENEXR_CFLAGS) $(OPENCV_CFLAGS)
@@ -338,8 +336,7 @@
 endif
 
 cli:
-	$(MAKE) "CFLAGS+=$(GMIC_CLI_CFLAGS) $(OPT_CFLAGS)" "LIBS+=$(GMIC_CLI_LIBS)" _cli
-	$(STRIP) gmic$(EXE)
+	$(MAKE) "CFLAGS+= $(CXXFLAGS) $(GMIC_CLI_CFLAGS) $(OPT_CFLAGS)" "LIBS+=$(GMIC_CLI_LIBS)" _cli
 
 debug:
 	$(MAKE) "CFLAGS+=$(GMIC_CLI_CFLAGS) $(DEBUG_CFLAGS)" "LIBS+=$(GMIC_CLI_LIBS)" _cli
