--- a/src/CImg.h
+++ b/src/CImg.h
@@ -68470,8 +68485,8 @@
         _cimg_test_temporary_path("D:\\Temp");
         _cimg_test_temporary_path("D:");
 #else
-        _cimg_test_temporary_path("/tmp");
-        _cimg_test_temporary_path("/var/tmp");
+        _cimg_test_temporary_path("@TERMUX_PREFIX@/tmp");
+        _cimg_test_temporary_path("@TERMUX_PREFIX@/var/tmp");
 #endif
         if (!path_found) {
           *s_path = 0;
