diff -u -r ../nano-4.6/src/text.c ./src/text.c
--- ../nano-4.6/src/text.c	2019-11-29 10:12:07.000000000 +0000
+++ ./src/text.c	2019-12-01 16:32:08.956643000 +0000
@@ -962,7 +962,7 @@
 		const char *theshell = getenv("SHELL");
 
 		if (theshell == NULL)
-			theshell = (char *)"/bin/sh";
+			theshell = (char *)"@TERMUX_PREFIX@/bin/sh";
 
 		/* Child: close the unused read end of the output pipe. */
 		close(from_fd[0]);
