diff -u -r ../nano-4.3/src/text.c ./src/text.c
--- ../nano-4.3/src/text.c	2019-06-11 11:55:31.000000000 +0000
+++ ./src/text.c	2019-06-23 21:36:28.191047000 +0000
@@ -956,7 +956,7 @@
 	/* Check which shell to use.  If none is specified, use /bin/sh. */
 	shellenv = getenv("SHELL");
 	if (shellenv == NULL)
-		shellenv = (char *)"/bin/sh";
+		shellenv = (char *)"@TERMUX_PREFIX@/bin/sh";
 
 	/* Fork a child process to run the command in. */
 	if ((pid_of_command = fork()) == 0) {
