diff -u -r ../nano-2.8.6/src/files.c ./src/files.c
--- ../nano-2.8.6/src/files.c	2017-07-21 09:44:46.000000000 +0200
+++ ./src/files.c	2017-07-22 16:01:40.806964191 +0200
@@ -2337,7 +2337,7 @@
 	    get_homedir();
 	    tilde_dir = mallocstrcpy(NULL, homedir);
 	} else {
-#ifdef HAVE_PWD_H
+#if defined(HAVE_PWD_H) && !defined(__ANDROID__)
 	    const struct passwd *userdata;
 
 	    tilde_dir = mallocstrncpy(NULL, buf, i + 1);
@@ -2451,7 +2451,7 @@
 
     *num_matches = 0;
 
-#ifdef HAVE_PWD_H
+#if defined(HAVE_PWD_H) && !defined(__ANDROID__)
     while ((userdata = getpwent()) != NULL) {
 	if (strncmp(userdata->pw_name, buf + 1, buf_len - 1) == 0) {
 	    /* Cool, found a match.  Add it to the list.  This makes a
