--- a/main/main.cpp
+++ b/main/main.cpp
@@ -1125,7 +1125,7 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 	bool delta_smoothing_override = false;
 	bool load_shell_env = false;
 
-	String rendering_driver = "";
+	String rendering_driver = "opengl3"; // There are problems with the render method on termux
 	String rendering_method = "";
 	OS::RenderingSource rendering_driver_source = OS::RenderingSource::RENDERING_SOURCE_DEFAULT;
 	OS::RenderingSource rendering_method_source = OS::RenderingSource::RENDERING_SOURCE_DEFAULT;
