--- a/modules/openxr/extensions/platform/openxr_opengl_extension.cpp
+++ b/modules/openxr/extensions/platform/openxr_opengl_extension.cpp
@@ -175,7 +175,7 @@ void *OpenXROpenGLExtension::set_session_create_and_get_next_pointer(void *p_nex
 		graphics_binding_egl.type = XR_TYPE_GRAPHICS_BINDING_EGL_MNDX;
 		graphics_binding_egl.next = p_next_pointer;
 
-		graphics_binding_egl.getProcAddress = eglGetProcAddress;
+		graphics_binding_egl.getProcAddress = (__typeof__(graphics_binding_egl.getProcAddress)) eglGetProcAddress;
 		graphics_binding_egl.display = (void *)display_server->window_get_native_handle(DisplayServerEnums::EGL_DISPLAY);
 		graphics_binding_egl.config = (void *)display_server->window_get_native_handle(DisplayServerEnums::EGL_CONFIG);
 		graphics_binding_egl.context = (void *)display_server->window_get_native_handle(DisplayServerEnums::OPENGL_CONTEXT);
