diff -ur ../nSnake-3.0.1/doc/man/nsnake.6 ./doc/man/nsnake.6
--- ../nSnake-3.0.1/doc/man/nsnake.6	2014-07-31 07:02:32.000000000 +0200
+++ ./doc/man/nsnake.6	2022-02-07 22:47:25.386119328 +0100
@@ -60,7 +60,5 @@
 This manual page and nsnake were both written by Alexandre Dantas <eu@alexdantas.net>.
 \"----------------------------------- WWW -------------------------------------
 .SH WWW
-* http://nsnake.alexdantas.net/
-
 * https://github.com/alexdantas/nsnake/
 
diff -ur ../nSnake-3.0.1/levels/lenny.nsnake ./levels/lenny.nsnake
--- ../nSnake-3.0.1/levels/lenny.nsnake	2014-07-31 07:02:32.000000000 +0200
+++ ./levels/lenny.nsnake	2022-02-07 22:50:12.829211049 +0100
@@ -1,7 +1,7 @@
 name=Le Lenny Faec
 author=Alexandre Dantas
 date=Jul 23, 2014
-comment=http://alexdantas.net/lenny
+comment=
 
 start
 ##############################################################################
diff -ur ../nSnake-3.0.1/src/Config/Arguments.cpp ./src/Config/Arguments.cpp
--- ../nSnake-3.0.1/src/Config/Arguments.cpp	2014-07-31 07:02:32.000000000 +0200
+++ ./src/Config/Arguments.cpp	2022-02-07 22:48:15.056047681 +0100
@@ -24,7 +24,6 @@
 		"       :o_o:\n"
 		"         -\n"
 		"\n"
-		"Homepage:      http://nsnake.alexdantas.net/\n"
 		"Source Code:   https://github.com/alexdantas/nsnake/\n"
 		"Contact:       Alexandre Dantas <eu@alexdantas.net>\n";
 
diff -ur ../nSnake-3.0.1/src/Flow/GameStateMainMenu.cpp ./src/Flow/GameStateMainMenu.cpp
--- ../nSnake-3.0.1/src/Flow/GameStateMainMenu.cpp	2014-07-31 07:02:32.000000000 +0200
+++ ./src/Flow/GameStateMainMenu.cpp	2022-02-07 22:48:48.159333247 +0100
@@ -274,10 +274,7 @@
 					             "Please copy the default level files from\n"
 					             "`"  SYSTEM_LEVEL_DIR "/`\n"
 					             "to\n"
-					             "`" + BoardParser::directory + "`\n" +
-					             "\n"
-					             "You can also download more levels from the website:\n"
-					             "http://nsnake.alexdantas.net/", true);
+					             "`" + BoardParser::directory + "`\n", true);
 				}
 				else
 				{
diff -ur ../nSnake-3.0.1/src/Interface/WindowGameHelp.cpp ./src/Interface/WindowGameHelp.cpp
--- ../nSnake-3.0.1/src/Interface/WindowGameHelp.cpp	2014-07-31 07:02:32.000000000 +0200
+++ ./src/Interface/WindowGameHelp.cpp	2022-02-07 22:48:03.032731685 +0100
@@ -131,8 +131,6 @@
 			                                             "contact him at <eu@alexdantas.net>\n"
 			                                             "Thanks for playing this game :)\n"
 			                                             "\n"
-			                                             "Homepage:\n"
-			                                             " http://nsnake.alexdantas.net/\n"
 			                                             "Source Code:\n"
 			                                             " https://github.com/alexdantas/nsnake/", '\n'),
 			                        0, 5, Globals::Theme::text);
