diff -uNr qemu-6.1.0/meson.build qemu-6.1.0.mod/meson.build
--- qemu-6.1.0/meson.build	2021-08-24 20:35:41.000000000 +0300
+++ qemu-6.1.0.mod/meson.build	2021-08-25 16:17:56.185294702 +0300
@@ -1508,7 +1508,7 @@
   }
 endif
 
-have_ivshmem = config_host_data.get('CONFIG_EVENTFD')
+have_ivshmem = false
 host_kconfig = \
   ('CONFIG_TPM' in config_host ? ['CONFIG_TPM=y'] : []) + \
   ('CONFIG_SPICE' in config_host ? ['CONFIG_SPICE=y'] : []) + \
