diff -uNr qemu-5.2.0/ui/curses.c qemu-5.2.0.mod/ui/curses.c
--- qemu-5.2.0/ui/curses.c	2020-12-08 18:59:44.000000000 +0200
+++ qemu-5.2.0.mod/ui/curses.c	2021-01-30 01:29:40.987065827 +0200
@@ -566,14 +566,14 @@
       0x25bc
     };
 
-    ucs2_to_nativecharset = iconv_open(local_codeset, "UCS-2");
+    ucs2_to_nativecharset = iconv_open(local_codeset, "UCS-2LE");
     if (ucs2_to_nativecharset == (iconv_t) -1) {
         fprintf(stderr, "Could not convert font glyphs from UCS-2: '%s'\n",
                         strerror(errno));
         exit(1);
     }
 
-    nativecharset_to_ucs2 = iconv_open("UCS-2", local_codeset);
+    nativecharset_to_ucs2 = iconv_open("UCS-2LE", local_codeset);
     if (nativecharset_to_ucs2 == (iconv_t) -1) {
         iconv_close(ucs2_to_nativecharset);
         fprintf(stderr, "Could not convert font glyphs to UCS-2: '%s'\n",
