diff -uNr qemu-8.2.6/block/nfs.c qemu-8.2.6.mod/block/nfs.c
--- qemu-8.2.6/block/nfs.c	2024-07-16 23:23:46.000000000 +0300
+++ qemu-8.2.6.mod/block/nfs.c	2024-12-22 04:04:31.484038274 +0200
@@ -272,9 +272,11 @@
     nfs_co_init_task(bs, &task);
     task.iov = iov;
 
+    char *buf = g_try_malloc(bytes);
+    if (!buf) return -ENOMEM;
+
     WITH_QEMU_LOCK_GUARD(&client->mutex) {
-        if (nfs_pread_async(client->context, client->fh,
-                            offset, bytes, nfs_co_generic_cb, &task) != 0) {
+        if (nfs_pread_async(client->context, client->fh, buf, bytes, offset, nfs_co_generic_cb, &task) != 0) {
             return -ENOMEM;
         }
 
@@ -320,7 +322,7 @@
 
     WITH_QEMU_LOCK_GUARD(&client->mutex) {
         if (nfs_pwrite_async(client->context, client->fh,
-                             offset, bytes, buf,
+                             buf, bytes, offset,
                              nfs_co_generic_cb, &task) != 0) {
             if (my_buffer) {
                 g_free(buf);
