--- a/src/discovery.c
+++ b/src/discovery.c
@@ -262,9 +262,9 @@
         };
         const char *global_dirs[] = {
                 globalconf_marker,
-                "/etc/xdg/boxes",
-                "/usr/local/share/boxes",
-                "/usr/share/boxes"
+                "@TERMUX_PREFIX@/etc/xdg/boxes",
+                "@TERMUX_PREFIX@/local/share/boxes",
+                "@TERMUX_PREFIX@/share/boxes"
         };
         const char *dirs[global_only ? 4 : 7];
         if (global_only) {
