--- a/src/Makefile
+++ b/src/Makefile
@@ -46,7 +46,7 @@
 	mkdir $(OUT_DIR)
 
 build: flags_$(BOXES_PLATFORM) | $(OUT_DIR)
-	$(MAKE) -C $(OUT_DIR) -f $(SRC_DIR)/Makefile BOXES_PLATFORM=$(BOXES_PLATFORM) ALL_OBJ="$(ALL_OBJ)" STRIP=true \
+	$(MAKE) -C $(OUT_DIR) -f $(SRC_DIR)/Makefile BOXES_PLATFORM=$(BOXES_PLATFORM) ALL_OBJ="$(ALL_OBJ)" STRIP=false \
 	    CFLAGS_ADDTL="-O $(CFLAGS_ADDTL)" flags_$(BOXES_PLATFORM) $(BOXES_EXECUTABLE_NAME)
 
 cov: flags_$(BOXES_PLATFORM) | $(OUT_DIR)
