--- tsu.sh	2026-02-13 14:37:35.895000000 +0800
+++ ./shell/tsu.sh	2026-02-13 14:22:05.220999797 +0800
@@ -64,7 +84,7 @@
 TERMUX_PATH="$TERMUX_PREFIX/bin:$TERMUX_PREFIX/bin/applets"
 ROOT_HOME="$TERMUX_FS/home/.suroot"
 ANDROID_SYSPATHS="/system/bin:/system/xbin"
-EXTRA_SYSPATHS="/sbin:/sbin/bin"
+EXTRA_SYSPATHS="/sbin:/sbin/bin:/debug_ramdisk"
 #ANDROID_ASROOT_SYSPATHS="/bin:/xbin"
 
 # Some constants that may change in future.
@@ -267,6 +287,10 @@
 	SKIP_SBIN=1
 fi
 
+if [[ -x "/debug_ramdisk" ]]; then
+        SU_BINARY_SEARCH+=("/debug_ramdisk/su")
+fi
+
 # Unset all Termux LD_* enviroment variables to prevent symbols missing , dlopen()ing of wrong libs.
 if [[ -n "$ENVIRONMENT_PRESERVE" ]]; then
 	EXP_ENV[LD_PRELOAD]="$LD_PRELOAD"
@@ -288,42 +312,24 @@
 
 ### TODO: Implement this cleanly.
 
-### ----- MAGISKSU
-# shellcheck disable=SC2117
-if [[ -z "$SKIP_SBIN" && "$(/sbin/su -v)" == *"MAGISKSU" ]]; then
-	# We are on Magisk su
-	su_args=("/sbin/su")
-	[[ -z "$SWITCH_USER" ]] || su_args+=("$SWITCH_USER")
-
-	if [[ -n "$ENVIRONMENT_PRESERVE" ]]; then
-		su_args+=("--preserve-environment")
-		su_cmdline="PATH=$BB_MAGISK:$PATH $ENV_BUILT $STARTUP_SCRIPT"
-	else
-		su_cmdline="PATH=$BB_MAGISK env -i $ENV_BUILT $STARTUP_SCRIPT"
-	fi
-	su_args+=("-c")
-	exec "${su_args[@]}" "${su_cmdline}"
-	##### ----- END MAGISKSU
-else
-	##### ----- OTHERS SU
-	for SU_BINARY in "${SU_BINARY_SEARCH[@]}"; do
-		if [[ -x "$SU_BINARY" ]]; then
-
-			su_args=("$SU_BINARY")
-			[[ -z "$SWITCH_USER" ]] || su_args+=("$SWITCH_USER")
-
-			# Let's use the system toybox/toolbox for now
-			if [[ -n "$ENVIRONMENT_PRESERVE" ]]; then
-				su_args+=("--preserve-environment")
-				su_cmdline="PATH=$ANDROID_SYSPATHS:$PATH $ENV_BUILT $STARTUP_SCRIPT "
-			else
-				su_cmdline="PATH=$ANDROID_SYSPATHS env -i $ENV_BUILT $STARTUP_SCRIPT"
-			fi
-			su_args+=("-c")
-			exec "${su_args[@]}" "${su_cmdline}"
+##### ----- OTHERS SU
+for SU_BINARY in "${SU_BINARY_SEARCH[@]}"; do
+	if [[ -x "$SU_BINARY" ]]; then
+
+		su_args=("$SU_BINARY")
+		[[ -z "$SWITCH_USER" ]] || su_args+=("$SWITCH_USER")
+
+		# Let's use the system toybox/toolbox for now
+		if [[ -n "$ENVIRONMENT_PRESERVE" ]]; then
+			su_args+=("--preserve-environment")
+			su_cmdline="PATH=$ANDROID_SYSPATHS:$PATH $ENV_BUILT $STARTUP_SCRIPT "
+		else
+			su_cmdline="PATH=$ANDROID_SYSPATHS env -i $ENV_BUILT $STARTUP_SCRIPT"
 		fi
-	done
-fi
+		su_args+=("-c")
+		exec "${su_args[@]}" "${su_cmdline}"
+	fi
+done
 ##### ----- END OTHERS SU
 
 # We didnt find any su binary