--- ../Makefile.orig	2019-07-22 17:41:28.077788655 +0200
+++ ./Makefile	2019-07-22 17:43:01.536030090 +0200
@@ -1,7 +1,16 @@
 .PHONY: all lint prettify
 
+PREFIX ?= /data/data/com.termux/files/usr/local
+
+all:
+
 lint:
 	./lint.sh
 
 prettify:
-	./prettify.sh
\ No newline at end of file
+	./prettify.sh
+
+install:
+	mkdir -p $(PREFIX)/bin/
+	install -m700 tsu $(PREFIX)/bin/
+	install -m700 tsudo $(PREFIX)/bin/
