--- ../configure.orig	2021-12-05 10:26:50.984021147 +0000
+++ ./configure	2021-12-05 10:49:16.032713588 +0000
@@ -14497,7 +14497,7 @@
 esac
 
 CPPFLAGS_SAVE="$CPPFLAGS"
-CPPFLAGS="$XML2_CPPFLAGS"
+CPPFLAGS="$XML2_CPPFLAGS -I@TERMUX_PREFIX@/include"
 for ac_header in libxml/tree.h libxml/parser.h libxml/xpath.h libxml/xpathInternals.h
 do :
   as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
