diff -uNr qemu-3.0.0/audio/ossaudio.c qemu-3.0.0.mod/audio/ossaudio.c
--- qemu-3.0.0/audio/ossaudio.c	2018-08-14 22:10:34.000000000 +0300
+++ qemu-3.0.0.mod/audio/ossaudio.c	2018-10-26 16:49:45.240264962 +0300
@@ -23,7 +23,7 @@
  */
 #include "qemu/osdep.h"
 #include <sys/ioctl.h>
-#include <sys/soundcard.h>
+#include <linux/soundcard.h>
 #include "qemu-common.h"
 #include "qemu/main-loop.h"
 #include "qemu/host-utils.h"
