diff -uNr qemu-3.0.0/linux-user/strace.c qemu-3.0.0.mod/linux-user/strace.c
--- qemu-3.0.0/linux-user/strace.c	2018-08-14 22:10:34.000000000 +0300
+++ qemu-3.0.0.mod/linux-user/strace.c	2018-10-26 16:49:45.273598481 +0300
@@ -1,7 +1,7 @@
 #include "qemu/osdep.h"
 #include <sys/ipc.h>
 #include <sys/msg.h>
-#include <sys/sem.h>
+#include <linux/sem.h>
 #include <sys/shm.h>
 #include <sys/select.h>
 #include <sys/mount.h>
