diff -uNr qemu-4.0.0/configure qemu-4.0.0.mod/configure
--- qemu-4.0.0/configure	2019-04-23 21:14:45.000000000 +0300
+++ qemu-4.0.0.mod/configure	2019-04-27 04:00:44.774004972 +0300
@@ -2301,23 +2301,10 @@
 fi
 
 ##########################################
-# lzo check
+# disable lzo check, force enable instead
 
-if test "$lzo" != "no" ; then
-    cat > $TMPC << EOF
-#include <lzo/lzo1x.h>
-int main(void) { lzo_version(); return 0; }
-EOF
-    if compile_prog "" "-llzo2" ; then
-        libs_softmmu="$libs_softmmu -llzo2"
-        lzo="yes"
-    else
-        if test "$lzo" = "yes"; then
-            feature_not_found "liblzo2" "Install liblzo2 devel"
-        fi
-        lzo="no"
-    fi
-fi
+libs_softmmu="$libs_softmmu -llzo2"
+lzo="yes"
 
 ##########################################
 # snappy check
@@ -2339,22 +2326,9 @@
 fi
 
 ##########################################
-# bzip2 check
+# disable bzip2 check, force enable bz2
 
-if test "$bzip2" != "no" ; then
-    cat > $TMPC << EOF
-#include <bzlib.h>
-int main(void) { BZ2_bzlibVersion(); return 0; }
-EOF
-    if compile_prog "" "-lbz2" ; then
-        bzip2="yes"
-    else
-        if test "$bzip2" = "yes"; then
-            feature_not_found "libbzip2" "Install libbzip2 devel"
-        fi
-        bzip2="no"
-    fi
-fi
+bzip2="yes"
 
 ##########################################
 # lzfse check
@@ -3183,6 +3157,7 @@
 # VNC SASL detection
 if test "$vnc" = "yes" && test "$vnc_sasl" != "no" ; then
   cat > $TMPC <<EOF
+#include <sys/types.h>
 #include <sasl/sasl.h>
 #include <stdio.h>
 int main(void) { sasl_server_init(NULL, "qemu"); return 0; }
@@ -4381,16 +4356,6 @@
 ##########################################
 # signalfd probe
 signalfd="no"
-cat > $TMPC << EOF
-#include <unistd.h>
-#include <sys/syscall.h>
-#include <signal.h>
-int main(void) { return syscall(SYS_signalfd, -1, NULL, _NSIG / 8); }
-EOF
-
-if compile_prog "" "" ; then
-  signalfd=yes
-fi
 
 # check if optreset global is declared by <getopt.h>
 optreset="no"
@@ -6029,7 +5994,7 @@
 
 # We can only support ivshmem if we have eventfd
 if [ "$eventfd" = "yes" ]; then
-  ivshmem=yes
+  ivshmem=no
 fi
 
 tools=""
@@ -6699,9 +6664,6 @@
 if test "$posix_fallocate" = "yes" ; then
   echo "CONFIG_POSIX_FALLOCATE=y" >> $config_host_mak
 fi
-if test "$sync_file_range" = "yes" ; then
-  echo "CONFIG_SYNC_FILE_RANGE=y" >> $config_host_mak
-fi
 if test "$fiemap" = "yes" ; then
   echo "CONFIG_FIEMAP=y" >> $config_host_mak
 fi
