diff -uNr qemu-3.1.0/tests/Makefile.include qemu-3.1.0.mod/tests/Makefile.include
--- qemu-3.1.0/tests/Makefile.include	2018-12-11 19:44:35.000000000 +0200
+++ qemu-3.1.0.mod/tests/Makefile.include	2019-05-27 15:42:32.554434009 +0300
@@ -777,10 +777,6 @@
 	rm $(INITRD_WORK_DIR)/init
 	rmdir $(INITRD_WORK_DIR)
 
-ifeq ($(CONFIG_POSIX),y)
-LIBS += -lutil
-endif
-
 # QTest rules
 
 TARGETS=$(patsubst %-softmmu,%, $(filter %-softmmu,$(TARGET_DIRS)))
