diff -u -r ../exa-57e4c08411f59e7db91fa7d8127fbd412667ea32/Cargo.toml ./Cargo.toml
--- ../exa-57e4c08411f59e7db91fa7d8127fbd412667ea32/Cargo.toml	2018-10-14 15:21:13.000000000 +0000
+++ ./Cargo.toml	2018-10-18 03:04:56.228613011 +0000
@@ -41,7 +41,7 @@
 term_grid = "0.1.6"
 term_size = "0.3.0"
 unicode-width = "0.1.4"
-users = "0.7"
+users = "0.8.1"
 zoneinfo_compiled = "0.4.7"
 
 [build-dependencies]
