diff -u -r ../mosh-1.2.4/src/util/pty_compat.cc ./src/util/pty_compat.cc
--- ../mosh-1.2.4/src/util/pty_compat.cc        2013-03-10 21:45:37.000000000 +0100
+++ ./src/util/pty_compat.cc    2014-01-23 03:18:24.000000000 +0100
@@ -39,7 +39,6 @@
 #include <unistd.h>
 #include <fcntl.h>
 #include <sys/ioctl.h>
-#include <sys/stropts.h>
 #include <termios.h>

 #include "pty_compat.h"
@@ -92,7 +91,7 @@
     return -1;
   }

-#ifndef _AIX
+#ifndef __ANDROID__
   if ( ioctl(slave, I_PUSH, "ptem") < 0 ||
        ioctl(slave, I_PUSH, "ldterm") < 0 ) {
     perror( "ioctl(I_PUSH)" );
