--- a/ggml/src/ggml-vulkan/ggml-vulkan.cpp	2026-03-26 11:58:35.579786395 +0000
+++ b/ggml/src/ggml-vulkan/ggml-vulkan.cpp	2026-03-26 11:58:35.591786516 +0000
@@ -5107,12 +5107,12 @@
 #endif
         device->name = GGML_VK_NAME + std::to_string(idx);
 
-        device_create_info = {
+        device_create_info = vk::DeviceCreateInfo(
             vk::DeviceCreateFlags(),
             device_queue_create_infos,
             {},
             device_extensions
-        };
+        );
         device_create_info.setPNext(&device_features2);
         device->device = device->physical_device.createDevice(device_create_info);
 
