+++ a/app/src/cli.c
@@ -2155,6 +2155,7 @@
                 }
                 break;
             case OPT_ORIENTATION:
+            {
                 enum sc_orientation orientation;
                 if (!parse_orientation(optarg, &orientation)) {
                     return false;
@@ -2162,6 +2163,7 @@
                 opts->display_orientation = orientation;
                 opts->record_orientation = orientation;
                 break;
+			}
             case OPT_RENDER_DRIVER:
                 opts->render_driver = optarg;
                 break;
