From 69074d38c40bff266e32671affcc6030b9705908 Mon Sep 17 00:00:00 2001
From: Md Arif <111168803+sabamdarif@users.noreply.github.com>
Date: Thu, 10 Jul 2025 19:34:45 +0530
Subject: [PATCH] Werror

---
 meson.build | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/meson.build b/meson.build
index d12a545..0ebfdff 100644
--- a/meson.build
+++ b/meson.build
@@ -70,6 +70,8 @@ add_project_arguments(
     '-Wextra',
     '-Wno-cast-function-type',
     '-Wno-unused-parameter',
+    '-Wno-format-nonliteral',
+    '-Wno-format-security',
     # Context: https://gitlab.com/freedesktop-sdk/freedesktop-sdk/commit/ce54a2527555e51e4ebf4cce9cbb6259cafa89a4
     # -O2 and -fexceptions unmask some cases of -Wmaybe-uninitialized, which
     # become errors for some reason (I’m guessing because of the
-- 
2.48.1

