--- a/src/fastfetch.c
+++ b/src/fastfetch.c
@@ -395,7 +395,7 @@
     ffStrbufAppendS(&folder, "/.local/share/fastfetch/presets/");
     listAvailablePresetsFromFolder(&folder, 0, NULL);
 
-    ffStrbufSetS(&folder, "/usr/share/fastfetch/presets/");
+    ffStrbufSetS(&folder, "@TERMUX_PREFIX@/share/fastfetch/presets/");
     listAvailablePresetsFromFolder(&folder, 0, NULL);
 
     ffStrbufDestroy(&folder);
@@ -490,7 +490,7 @@
         return;
     }
 
-    ffStrbufSetS(&filename, "/usr/share/fastfetch/presets/");
+    ffStrbufSetS(&filename, "@TERMUX_PREFIX@/share/fastfetch/presets/");
     ffStrbufAppendS(&filename, value);
 
     file = fopen(filename.chars, "r");
