--- a/src/modules/cursor.c
+++ b/src/modules/cursor.c
@@ -147,7 +147,7 @@
     if(user)
         ffParsePropFileHome(instance, ".icons/default/index.theme", "Inherits =", &theme);
     else
-        ffParsePropFile("/usr/share/icons/default/index.theme", "Inherits =", &theme);
+        ffParsePropFile("@TERMUX_PREFIX@/share/icons/default/index.theme", "Inherits =", &theme);
 
     if(theme.length == 0)
     {
