--- ../Makefile.am.orig	2018-12-10 09:34:44.539940037 +0100
+++ ./Makefile.am	2018-12-10 09:35:23.876606447 +0100
@@ -10,7 +10,7 @@
 #
 dist_man_MANS = arp-scan.1 get-oui.1 get-iab.1 arp-fingerprint.1 mac-vendor.5
 #
-arp_scan_SOURCES = arp-scan.c arp-scan.h error.c wrappers.c utils.c mt19937ar.c mt19937ar.h
+arp_scan_SOURCES = arp-scan.c arp-scan.h error.c wrappers.c utils.c mt19937ar.c mt19937ar.h hcreate.c hcreate_r.c hsearch_r.c hdestroy_r.c
 arp_scan_LDADD = $(LIBOBJS)
 #
 dist_pkgdata_DATA = ieee-oui.txt ieee-iab.txt mac-vendor.txt
