diff --git a/backends/build_system/awscli_venv.py b/backends/build_system/awscli_venv.py
index be995bf81..e50deda12 100644
--- a/backends/build_system/awscli_venv.py
+++ b/backends/build_system/awscli_venv.py
@@ -77,7 +77,7 @@ class AwsCliVenv:

     def _install_requirements(self, requirements_file, cwd=None):
         self._pip_install(
-            ["--no-build-isolation", "-r", requirements_file],
+            ["-r", requirements_file],
             cwd=cwd,
         )

