--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -350,9 +350,8 @@
 #
 install(TARGETS ${PROJECT_NAME} RUNTIME                     DESTINATION bin)
 install(FILES   ${CMAKE_CURRENT_BINARY_DIR}/man/biboumi.1   DESTINATION share/man/man1         OPTIONAL COMPONENT documentation)
-install(FILES   ${CMAKE_CURRENT_BINARY_DIR}/biboumi.service DESTINATION lib/systemd/system     COMPONENT init)
 file(GLOB policy_files conf/*policy.txt)
-install(FILES   ${policy_files}                             DESTINATION /etc/biboumi           COMPONENT configuration)
+install(FILES   ${policy_files}                             DESTINATION etc/biboumi           COMPONENT configuration)
 
 #
 ## Dist target
