diff --color -uNr at-3.2.5/getloadavg.c at-3.2.5.mod/getloadavg.c
--- at-3.2.5/getloadavg.c	2022-02-05 07:00:57.000000000 -0300
+++ at-3.2.5.mod/getloadavg.c	2022-06-25 15:48:17.911936826 -0300
@@ -69,8 +69,6 @@
 #include "config.h"
 #endif
 
-#include <fcntl.h> /* for encapsulated open, close, read, write */
-
 #ifndef HAVE_GETLOADAVG
 
 /* The existing Emacs configuration files define a macro called
