diff --color -uNr at-3.2.5/Makefile.in at-3.2.5.mod/Makefile.in
--- at-3.2.5/Makefile.in	2022-02-05 07:00:57.000000000 -0300
+++ at-3.2.5.mod/Makefile.in	2022-06-25 16:05:18.213639977 -0300
@@ -102,43 +102,41 @@
 	$(CC) -c $(CFLAGS) $(DEFS) $*.c
 
 install: all
-	$(INSTALL) -g root -o root -m 755 -d $(DESTDIR)$(etcdir)
-	$(INSTALL) -g root -o root -m 755 -d $(DESTDIR)$(bindir)
-	$(INSTALL) -g root -o root -m 755 -d $(DESTDIR)$(sbindir)
-	$(INSTALL) -g root -o root -m 755 -d $(DESTDIR)$(atdatadir)
-	$(INSTALL) -g root -o root -m 755 -d $(DESTDIR)$(docdir)
-	$(INSTALL) -g root -o root -m 755 -d $(DESTDIR)$(atdocdir)
-	$(INSTALL) -g $(DAEMON_GROUPNAME) -o $(DAEMON_USERNAME) -m 755 -d $(DESTDIR)$(ATSPOOL_DIR) $(DESTDIR)$(ATJOB_DIR)
-	chmod 1770 $(DESTDIR)$(ATSPOOL_DIR) $(DESTDIR)$(ATJOB_DIR)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(etcdir)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(bindir)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(sbindir)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(atdatadir)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(docdir)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(atdocdir)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(ATSPOOL_DIR) $(DESTDIR)$(ATJOB_DIR)
+	chmod 0700 $(DESTDIR)$(ATSPOOL_DIR) $(DESTDIR)$(ATJOB_DIR)
 	touch $(DESTDIR)$(LFILE)
 	chmod 600 $(DESTDIR)$(LFILE)
-	chown $(DAEMON_USERNAME):$(DAEMON_GROUPNAME) $(DESTDIR)$(LFILE)
-	test -f $(DESTDIR)$(etcdir)/at.allow || test -f $(DESTDIR)$(etcdir)/at.deny || $(INSTALL) -o root -g $(DAEMON_GROUPNAME) -m 640 at.deny $(DESTDIR)$(etcdir)/
-	$(INSTALL) -g $(DAEMON_GROUPNAME) -o $(DAEMON_USERNAME) -m 6755 at $(DESTDIR)$(bindir)
+	$(INSTALL) -m 755 at $(DESTDIR)$(bindir)
 	$(LN_S) -f at $(DESTDIR)$(bindir)/atq
 	$(LN_S) -f at $(DESTDIR)$(bindir)/atrm
-	$(INSTALL) -g root -o root -m 755 batch $(DESTDIR)$(bindir)
-	$(INSTALL) -g root -o root -m 755 batch-job $(DESTDIR)$(atdatadir)
-	$(INSTALL) -d -o root -g root -m 755 $(DESTDIR)$(man1dir)
-	$(INSTALL) -d -o root -g root -m 755 $(DESTDIR)$(man5dir)
-	$(INSTALL) -d -o root -g root -m 755 $(DESTDIR)$(man8dir)
-	$(INSTALL) -g root -o root -m 755 atd $(DESTDIR)$(sbindir)
-	$(INSTALL) -g root -o root -m 755 atrun $(DESTDIR)$(sbindir)
-	$(INSTALL) -g root -o root -m 644 at.1 $(DESTDIR)$(man1dir)/
+	$(INSTALL) -m 755 batch $(DESTDIR)$(bindir)
+	$(INSTALL) -m 755 batch-job $(DESTDIR)$(atdatadir)
+	$(INSTALL) -d -m 755 $(DESTDIR)$(man1dir)
+	$(INSTALL) -d -m 755 $(DESTDIR)$(man5dir)
+	$(INSTALL) -d -m 755 $(DESTDIR)$(man8dir)
+	$(INSTALL) -m 755 atd $(DESTDIR)$(sbindir)
+	$(INSTALL) -m 755 atrun $(DESTDIR)$(sbindir)
+	$(INSTALL) -m 644 at.1 $(DESTDIR)$(man1dir)/
 	cd $(DESTDIR)$(man1dir) && $(LN_S) -f at.1 atq.1 && $(LN_S) -f at.1 batch.1 && $(LN_S) -f at.1 atrm.1
-	$(INSTALL) -g root -o root -m 644 atd.8 $(DESTDIR)$(man8dir)/
+	$(INSTALL) -m 644 atd.8 $(DESTDIR)$(man8dir)/
 	sed "s,\$${exec_prefix},$(exec_prefix),g" <atrun.8>tmpman
-	$(INSTALL) -g root -o root -m 644 tmpman $(DESTDIR)$(man8dir)/atrun.8
+	$(INSTALL) -m 644 atd.8 $(DESTDIR)$(man8dir)/
 	rm -f tmpman
-	$(INSTALL) -g root -o root -m 644 at.allow.5 $(DESTDIR)$(man5dir)/
+	$(INSTALL) -m 644 at.allow.5 $(DESTDIR)$(man5dir)/
 	cd $(DESTDIR)$(man5dir) && $(LN_S) -f at.allow.5 at.deny.5
-	$(INSTALL) -g root -o root -m 644 $(DOCS) $(DESTDIR)$(atdocdir)
+	$(INSTALL) -m 644 $(DOCS) $(DESTDIR)$(atdocdir)
 	rm -f $(DESTDIR)$(mandir)/cat1/at.1* $(DESTDIR)$(mandir)/cat1/batch.1* \
 		$(DESTDIR)$(mandir)/cat1/atq.1*
 	rm -f $(DESTDIR)$(mandir)/cat1/atd.8*
 	if test x"$(systemdsystemunitdir)" != xno; then \
-		$(INSTALL) -o root -g root -m 755 -d $(DESTDIR)$(systemdsystemunitdir); \
-		$(INSTALL) -o root -g root -m 644 atd.service $(DESTDIR)$(systemdsystemunitdir); \
+		$(INSTALL) -m 755 -d $(DESTDIR)$(systemdsystemunitdir); \
+		$(INSTALL) -m 644 atd.service $(DESTDIR)$(systemdsystemunitdir); \
 	fi
 
 dist: checkin $(DIST) $(LIST) Filelist.asc
