diff -uNr at-3.2.1/getloadavg.c at-3.2.1.mod/getloadavg.c
--- at-3.2.1/getloadavg.c	2019-08-04 15:16:16.000000000 +0300
+++ at-3.2.1.mod/getloadavg.c	2020-01-02 20:33:06.650907870 +0200
@@ -69,9 +69,6 @@
 #include <config.h>
 #endif
 
-#include "lisp.h"
-#include "sysfile.h" /* for encapsulated open, close, read, write */
-
 #ifndef HAVE_GETLOADAVG
 
 /* The existing Emacs configuration files define a macro called
