diff -u -r ../unzip60/unzpriv.h ./unzpriv.h
--- ../unzip60/unzpriv.h	2009-04-19 23:59:26.000000000 +0000
+++ ./unzpriv.h	2017-07-07 00:38:24.057603217 +0000
@@ -1284,9 +1284,9 @@
 
 #ifdef ZMEM
 #  undef ZMEM
-#  define memcmp(b1,b2,len)      bcmp(b2,b1,len)
-#  define memcpy(dest,src,len)   bcopy(src,dest,len)
-#  define memzero                bzero
+#  define memcmp(b1,b2,len)      memcmp(b2,b1,len)
+#  define memcpy(b1,b2,len)      (void)(__builtin_memmove((b2), (b1), (len)))
+#  define memzero(b,len)         (void)(__builtin_memset((b), '\0', (len)))
 #else
 #  define memzero(dest,len)      memset(dest,0,len)
 #endif
