--- ./tools/lldb/source/Host/CMakeLists.txt	2015-10-30 08:24:52.000000000 +0530
+++ ./tools/lldb/source/Host/CMakeLists.txt	2017-01-10 21:14:16.132877692 +0530
@@ -115,7 +115,7 @@
       macosx/cfcpp/CFCString.cpp
       )
 
-  elseif (CMAKE_SYSTEM_NAME MATCHES "Linux")
+  elseif (CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Android")
     if (__ANDROID_NDK__)
       add_host_subdirectory(android
         android/HostInfoAndroid.cpp
--- ./tools/lldb/scripts/Python/modules/CMakeLists.txt	2015-02-13 15:52:00.000000000 +0530
+++ ./tools/lldb/scripts/Python/modules/CMakeLists.txt	2017-01-10 22:03:15.069857246 +0530
@@ -5,7 +5,6 @@
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-macro-redefined")
 endif ()
 
-# build the Python readline suppression module only on Linux
-if (CMAKE_SYSTEM_NAME MATCHES "Linux" AND NOT __ANDROID_NDK__)
+if (CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Android")
    add_subdirectory(readline)
 endif()
--- ./tools/lldb/tools/lldb-server/CMakeLists.txt	2016-01-29 17:29:57.000000000 +0530
+++ ./tools/lldb/tools/lldb-server/CMakeLists.txt	2017-01-10 22:06:03.981423698 +0530
@@ -1,4 +1,4 @@
-if ( CMAKE_SYSTEM_NAME MATCHES "Linux" )
+if ( CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Android" )
 include_directories(
   ../../../../llvm/include
   ../../source/Plugins/Process/Linux
--- ./tools/lldb/source/CMakeLists.txt	2015-10-24 05:57:04.000000000 +0530
+++ ./tools/lldb/source/CMakeLists.txt	2017-01-10 22:10:44.805026352 +0530
@@ -1,6 +1,6 @@
 include_directories(.)
 
-if ( CMAKE_SYSTEM_NAME MATCHES "Linux" )
+if ( CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Android" )
 include_directories(
   Plugins/Process/Linux
   Plugins/Process/POSIX
--- ./tools/lldb/source/Plugins/Platform/CMakeLists.txt	2015-10-24 05:57:04.000000000 +0530
+++ ./tools/lldb/source/Plugins/Platform/CMakeLists.txt	2017-01-10 22:15:36.255075635 +0530
@@ -1,4 +1,4 @@
-#if (CMAKE_SYSTEM_NAME MATCHES "Linux")
+#if (CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Android")
   add_subdirectory(Linux)
 #elseif (CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
   add_subdirectory(FreeBSD)
--- ./tools/lldb/source/Plugins/Process/CMakeLists.txt	2015-10-28 23:51:45.000000000 +0530
+++ ./tools/lldb/source/Plugins/Process/CMakeLists.txt	2017-01-10 22:18:20.019868513 +0530
@@ -1,4 +1,4 @@
-if (CMAKE_SYSTEM_NAME MATCHES "Linux")
+if (CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Android")
   add_subdirectory(Linux)
   add_subdirectory(POSIX)
 elseif (CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
--- ./tools/llvm-shlib/CMakeLists.txt	2016-05-26 10:05:35.000000000 +0530
+++ ./tools/llvm-shlib/CMakeLists.txt	2017-01-12 16:43:02.136137108 +0530
@@ -39,7 +39,7 @@
 add_llvm_library(LLVM SHARED DISABLE_LLVM_LINK_LLVM_DYLIB SONAME ${SOURCES})
 
 list(REMOVE_DUPLICATES LIB_NAMES)
-if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux") # FIXME: It should be "GNU ld for elf"
+if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux" OR "${CMAKE_SYSTEM_NAME}" STREQUAL "Android") # FIXME: It should be "GNU ld for elf"
   # GNU ld doesn't resolve symbols in the version script.
   set(LIB_NAMES -Wl,--whole-archive ${LIB_NAMES} -Wl,--no-whole-archive)
 elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
