diff -uNr plzip-1.8/configure plzip-1.8.mod/configure
--- plzip-1.8/configure	2019-01-04 20:33:44.000000000 +0200
+++ plzip-1.8.mod/configure	2019-09-23 19:37:59.464411919 +0300
@@ -21,10 +21,10 @@
 datarootdir='$(prefix)/share'
 infodir='$(datarootdir)/info'
 mandir='$(datarootdir)/man'
-CXX=g++
-CPPFLAGS=
-CXXFLAGS='-Wall -W -O2'
-LDFLAGS=
+CXX?=g++
+CPPFLAGS?=
+CXXFLAGS?='-Wall -W -O2'
+LDFLAGS?=
 
 # checking whether we are using GNU C++.
 /bin/sh -c "${CXX} --version" > /dev/null 2>&1 ||
