diff -uNr plzip-1.8/Makefile.in plzip-1.8.mod/Makefile.in
--- plzip-1.8/Makefile.in	2018-10-12 00:56:58.000000000 +0300
+++ plzip-1.8.mod/Makefile.in	2019-09-23 19:39:16.088047518 +0300
@@ -4,7 +4,7 @@
 INSTALL_PROGRAM = $(INSTALL) -m 755
 INSTALL_DATA = $(INSTALL) -m 644
 INSTALL_DIR = $(INSTALL) -d -m 755
-LIBS = -llz -lpthread
+LIBS = -llz
 SHELL = /bin/sh
 CAN_RUN_INSTALLINFO = $(SHELL) -c "install-info --version" > /dev/null 2>&1
 
