diff -uNr unrar/makefile unrar.mod/makefile
--- unrar/makefile	2017-06-13 14:04:49.000000000 +0300
+++ unrar.mod/makefile	2017-09-14 20:40:07.511902817 +0300
@@ -2,14 +2,15 @@
 # Makefile for UNIX - unrar
 
 # Linux using GCC
-CXX=c++
-CXXFLAGS=-O2
-LIBFLAGS=-fPIC
+CXX?=c++
+CXXFLAGS?=-O2
+LIBFLAGS=$(LDFLAGS)
 DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -DRAR_SMP
-STRIP=strip
-AR=ar
-LDFLAGS=-pthread
-DESTDIR=/usr
+STRIP?=strip
+AR?=ar
+LDFLAGS?=-pthread
+DESTDIR=$(PREFIX)
+LIBS=-lreadline
 
 # Linux using LCC
 #CXX=lcc
