diff -u -r ../unrar/makefile ./makefile
--- ../unrar/makefile	2016-05-10 11:51:35.000000000 +0200
+++ ./makefile	2017-01-31 12:48:13.868329984 +0100
@@ -2,13 +2,13 @@
 # Makefile for UNIX - unrar
 
 # Linux using GCC
-CXX=c++
-CXXFLAGS=-O2
-LIBFLAGS=-fPIC
+CXX?=c++
+CXXFLAGS?=-O2
+LIBFLAGS=$(LDFLAGS)
 DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -DRAR_SMP
-STRIP=strip
-LDFLAGS=-pthread
-DESTDIR=/usr
+STRIP?=strip
+LDFLAGS?=-pthread
+DESTDIR=@TERMUX_PREFIX@
 
 # Linux using LCC
