diff -uNr unrar/makefile unrar.mod/makefile
--- unrar/makefile	2017-11-22 08:53:39.000000000 +0200
+++ unrar.mod/makefile	2018-07-03 15:18:13.359399811 +0300
@@ -2,14 +2,15 @@
 # Makefile for UNIX - unrar
 
 # Linux using GCC
-CXX=c++
-CXXFLAGS=-O2 -Wno-logical-op-parentheses -Wno-switch -Wno-dangling-else
-LIBFLAGS=-fPIC
+CXX?=c++
+CXXFLAGS?=-O2 -Wno-logical-op-parentheses -Wno-switch -Wno-dangling-else
+LIBFLAGS=$(LDFLAGS)
 DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -DRAR_SMP
-STRIP=strip
-AR=ar
-LDFLAGS=-pthread
-DESTDIR=/usr
+STRIP?=strip
+AR?=ar
+LDFLAGS?=-pthread
+DESTDIR=$(PREFIX)
+LIBS=-lgetpass
 
 # Linux using LCC
 #CXX=lcc
