diff -u -r ../unrar/makefile ./makefile
--- ../unrar/makefile	2017-11-22 06:53:39.000000000 +0000
+++ ./makefile	2018-03-12 10:30:16.141557520 +0000
@@ -2,14 +2,14 @@
 # Makefile for UNIX - unrar
 
 # Linux using GCC
-CXX=c++
-CXXFLAGS=-O2 -Wno-logical-op-parentheses -Wno-switch -Wno-dangling-else
-LIBFLAGS=-fPIC
+CXX?=c++
+CXXFLAGS?=-O2 -Wno-logical-op-parentheses -Wno-switch -Wno-dangling-else
+LIBFLAGS=$(LDFLAGS)
 DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -DRAR_SMP
-STRIP=strip
-AR=ar
-LDFLAGS=-pthread
-DESTDIR=/usr
+STRIP?=strip
+AR?=ar
+LDFLAGS?=-pthread
+DESTDIR=$(PREFIX)
 
 # Linux using LCC
 #CXX=lcc
