Patch submitted to dev@rarlab.com on 2015-08-25.

diff -u -r ../unrar/os.hpp ./os.hpp
--- ../unrar/os.hpp	2015-08-04 06:22:40.000000000 -0400
+++ ./os.hpp	2015-08-25 16:58:59.525380417 -0400
@@ -160,7 +160,7 @@
 #define SAVE_LINKS
 #endif
 
-#if defined(__linux) && !defined (_ANDROID) || defined(__FreeBSD__)
+#if defined(__linux) && !defined(__ANDROID__) || defined(__FreeBSD__)
 #include <sys/time.h>
 #define USE_LUTIMES
 #endif
