diff -u -r ../unrar/strfn.cpp ./strfn.cpp
--- ../unrar/strfn.cpp	2016-08-14 21:17:47.000000000 +0200
+++ ./strfn.cpp	2017-01-31 12:50:41.922525083 +0100
@@ -21,10 +21,6 @@
     DestSize=SrcLength;
   OemToCharBuffA(Src,Dest,(DWORD)DestSize);
   Dest[DestSize-1]=0;
-#elif defined(_ANDROID)
-  wchar DestW[NM];
-  JniCharToWide(Src,DestW,ASIZE(DestW),true);
-  WideToChar(DestW,Dest,DestSize);
 #else
   if (Dest!=Src)
     strncpyz(Dest,Src,DestSize);
