--- a/src/storage/src/db_checkpoint.cc
+++ b/src/storage/src/db_checkpoint.cc
@@ -18,11 +18,24 @@
 #  include <cinttypes>
 
 #include <glog/logging.h>
-#  include "file/file_util.h"
+// #  include "file/file_util.h"
 #  include "rocksdb/db.h"
 // #include "file/filename.h"
 
 namespace rocksdb {
+class IOTracer;
+
+extern Status CopyFile(FileSystem* env, const std::string& source,
+                       const std::string& destination, uint64_t size,
+                       bool use_fsync,
+                       const std::shared_ptr<IOTracer>& io_tracer = nullptr,
+                       const Temperature temperature = Temperature::kUnknown);
+
+extern Status CreateFile(FileSystem* env, const std::string& destination,
+                         const std::string& contents, bool);
+
+extern bool ParseFileName(const std::string& filename, uint64_t* number,
+                          FileType* type, WalFileType* log_type = nullptr);
 
 class DBCheckpointImpl : public DBCheckpoint {
  public:
