diff -uNr fltk-1.3.4/fluid/ExternalCodeEditor_UNIX.cxx fltk-1.3.4.mod/fluid/ExternalCodeEditor_UNIX.cxx
--- fltk-1.3.4/fluid/ExternalCodeEditor_UNIX.cxx	2016-08-16 23:42:22.000000000 +0300
+++ fltk-1.3.4.mod/fluid/ExternalCodeEditor_UNIX.cxx	2017-12-10 15:15:58.522896625 +0200
@@ -214,8 +214,8 @@
 //     Returns pointer to static memory.
 //
 const char* ExternalCodeEditor::tmpdir_name() {
-  static char dirname[100];
-  snprintf(dirname, sizeof(dirname), "/tmp/.fluid-%ld", (long)getpid());
+  static char dirname[100] = {0};
+  snprintf(dirname, sizeof(dirname), "/data/data/com.termux/files/usr/tmp/.fluid-%ld", (long)getpid());
   return dirname;
 }
 
