diff -uNr fltk-1.3.4/src/fl_open_uri.cxx fltk-1.3.4.mod/src/fl_open_uri.cxx
--- fltk-1.3.4/src/fl_open_uri.cxx	2013-10-04 19:48:08.000000000 +0300
+++ fltk-1.3.4.mod/src/fl_open_uri.cxx	2017-12-10 15:30:04.032588233 +0200
@@ -275,7 +275,7 @@
 		*end;			// End of filename buffer
 
 
-  if ((path = getenv("PATH")) == NULL) path = "/bin:/usr/bin";
+  if ((path = getenv("PATH")) == NULL) path = "/data/data/com.termux/files/usr/bin:/data/data/com.termux/files/usr/bin/applets:/system/xbin:/system/bin";
 
   for (ptr = filename, end = filename + filesize - 1; *path; path ++) {
     if (*path == ':') {
