diff -uNr loksh-6.4/c_sh.c loksh-6.4.mod/c_sh.c
--- loksh-6.4/c_sh.c	2018-10-18 19:25:07.000000000 +0300
+++ loksh-6.4.mod/c_sh.c	2019-02-20 21:08:11.282582629 +0200
@@ -8,6 +8,7 @@
 #include <sys/resource.h>
 #include <sys/stat.h>
 #include <sys/time.h>
+#include <linux/time.h>
 
 #include <ctype.h>
 #include <errno.h>
diff -uNr loksh-6.4/mail.c loksh-6.4.mod/mail.c
--- loksh-6.4/mail.c	2018-10-18 19:25:07.000000000 +0300
+++ loksh-6.4.mod/mail.c	2019-02-20 21:08:16.325940697 +0200
@@ -7,6 +7,7 @@
 
 #include <sys/stat.h>
 #include <sys/time.h>
+#include <linux/time.h>
 
 #include <string.h>
 #include <time.h>
diff -uNr loksh-6.4/var.c loksh-6.4.mod/var.c
--- loksh-6.4/var.c	2018-10-18 19:25:07.000000000 +0300
+++ loksh-6.4.mod/var.c	2019-02-20 21:08:21.149297670 +0200
@@ -2,6 +2,7 @@
 
 #include <sys/stat.h>
 #include <sys/time.h>
+#include <linux/time.h>
 
 #include <ctype.h>
 #include <errno.h>
