diff -uNr loksh-6.6/Makefile loksh-6.6.mod/Makefile
--- loksh-6.6/Makefile	2019-10-17 17:47:17.000000000 +0300
+++ loksh-6.6.mod/Makefile	2019-11-03 03:27:27.575357097 +0200
@@ -33,7 +33,6 @@
 install: all
 	install -v -D -m 755 ksh $(DESTDIR)/$(BIN_DIR)/$(BIN_NAME)
 	install -v -D -m 644 ksh.1 $(DESTDIR)/$(MAN_DIR)/man1/$(BIN_NAME).1
-	install -v -m 644 sh.1 $(DESTDIR)/$(MAN_DIR)/man1/sh.1
 	install -v -D -m 644 README.md $(DESTDIR)/$(DOC_DIR)/README.md
 	install -v -m 644 README $(DESTDIR)/$(DOC_DIR)/README
 	install -v -m 644 CONTRIBUTORS $(DESTDIR)/$(DOC_DIR)/CONTRIBUTORS
