--- ../loksh/Makefile	2018-04-14 11:35:14.000000000 +0200
+++ ./Makefile	2018-10-04 00:20:52.213906936 +0200
@@ -30,7 +30,6 @@
 install: all
 	install -v -D -m 755 ksh $(DESTDIR)/$(BIN_DIR)/$(BIN_NAME)
 	install -v -D -m 644 ksh.1 $(DESTDIR)/$(MAN_DIR)/man1/$(BIN_NAME).1
-	install -v -m 644 sh.1 $(DESTDIR)/$(MAN_DIR)/man1/sh.1
 	install -v -D -m 644 README $(DESTDIR)/$(DOC_DIR)/README
 	install -v -m 644 README.upstream $(DESTDIR)/$(DOC_DIR)/README.upstream
 	install -v -m 644 CONTRIBUTORS $(DESTDIR)/$(DOC_DIR)/CONTRIBUTORS
