diff -uNr lua-5.5.0/src/luaconf.h lua-5.5.0.mod/src/luaconf.h
--- lua-5.5.0/src/luaconf.h	2025-12-15 16:14:40.000000000 +0530
+++ lua-5.5.0.mod/src/luaconf.h	2026-01-09 10:41:00.285336305 +0530
@@ -239,7 +239,7 @@
 
 #else			/* }{ */
 
-#define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT	"@TERMUX_PREFIX@/"
 #define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR "/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR "/"
 
