diff -uNr The-Powder-Toy-94.1/src/gui/game/GameView.cpp The-Powder-Toy-94.1.mod/src/gui/game/GameView.cpp
--- The-Powder-Toy-94.1/src/gui/game/GameView.cpp	2019-02-21 07:23:45.000000000 +0200
+++ The-Powder-Toy-94.1.mod/src/gui/game/GameView.cpp	2019-05-19 20:37:48.967690493 +0300
@@ -1628,9 +1628,7 @@
 			c->AdjustBrushSize(-1, !alt, shiftBehaviour, ctrlBehaviour);
 		break;
 	case SDL_SCANCODE_I:
-		if(ctrl)
-			c->Install();
-		else
+		if(!ctrl)
 			c->InvertAirSim();
 		break;
 	case SDL_SCANCODE_SEMICOLON:
