--- a/meson.build
+++ b/meson.build
@@ -46,7 +46,7 @@
 elif host_platform in [ 'darwin' ]
 	host_libc = 'macos'
 elif host_platform in [ 'android' ]
-	host_platform = 'android'
+	host_platform = 'linux'
 	host_libc = 'bionic'
 else
 	host_libc = 'gnu'
@@ -315,6 +315,7 @@
 else
 	ident_platform = 'UNKNOWN'
 endif
+ident_platform = 'TERMUX'
 
 app_exe = get_option('app_exe')
 app_id = get_option('app_id')
