diff -uNr The-Powder-Toy-93.3/src/gui/game/GameModel.cpp The-Powder-Toy-93.3.mod/src/gui/game/GameModel.cpp
--- The-Powder-Toy-93.3/src/gui/game/GameModel.cpp	2018-04-02 00:19:03.000000000 +0300
+++ The-Powder-Toy-93.3.mod/src/gui/game/GameModel.cpp	2018-06-15 13:55:36.209587222 +0300
@@ -105,6 +105,7 @@
 	brushList.push_back(new TriangleBrush(ui::Point(4, 4)));
 
 	//Load more from brushes folder
+    Client::Ref().MakeDirectory("/data/data/com.termux/files/home/.powdertoy");
 	std::vector<string> brushFiles = Client::Ref().DirectorySearch(BRUSH_DIR, "", ".ptb");
 	for (size_t i = 0; i < brushFiles.size(); i++)
 	{
