diff -uNr The-Powder-Toy-93.3/src/gui/options/OptionsView.cpp The-Powder-Toy-93.3.mod/src/gui/options/OptionsView.cpp
--- The-Powder-Toy-93.3/src/gui/options/OptionsView.cpp	2018-04-02 00:19:03.000000000 +0300
+++ The-Powder-Toy-93.3.mod/src/gui/options/OptionsView.cpp	2018-06-15 13:55:36.216253888 +0300
@@ -260,7 +260,7 @@
 			const char* openCommand = "xdg-open ";
 #endif
 			char* workingDirectory = new char[FILENAME_MAX+strlen(openCommand)];
-			sprintf(workingDirectory, "%s\"%s\"", openCommand, getcwd(NULL, 0));
+			sprintf(workingDirectory, "%s\"%s\"", openCommand, "/data/data/com.termux/files/home/.powdertoy");
 			system(workingDirectory);
 			delete[] workingDirectory;
 		}
