diff -uNr The-Powder-Toy-93.3/src/gui/game/GameView.cpp The-Powder-Toy-93.3.mod/src/gui/game/GameView.cpp
--- The-Powder-Toy-93.3/src/gui/game/GameView.cpp	2018-04-02 00:19:03.000000000 +0300
+++ The-Powder-Toy-93.3.mod/src/gui/game/GameView.cpp	2018-06-15 13:55:36.212920556 +0300
@@ -1636,9 +1636,7 @@
 			c->AdjustBrushSize(-1, !alt, shiftBehaviour, ctrlBehaviour);
 		break;
 	case 'i':
-		if(ctrl)
-			c->Install();
-		else
+		if(!ctrl)
 			c->InvertAirSim();
 		break;
 	case ';':
