diff -uNr The-Powder-Toy-93.3/SConstruct The-Powder-Toy-93.3.mod/SConstruct
--- The-Powder-Toy-93.3/SConstruct	2018-04-02 00:19:03.000000000 +0300
+++ The-Powder-Toy-93.3.mod/SConstruct	2018-06-15 13:55:36.199587222 +0300
@@ -2,7 +2,7 @@
 #run generator.py
 if not GetOption('clean'):
 	if sys.version_info[0] < 3:
-		execfile("generator.py")
+		exec(compile(open("generator.py").read(), "generator.py", 'exec'))
 	else:
 		with open("generator.py") as f:
 			code = compile(f.read(), "generator.py", 'exec')
