diff -uNr The-Powder-Toy-93.3/src/Config.h The-Powder-Toy-93.3.mod/src/Config.h
--- The-Powder-Toy-93.3/src/Config.h	2018-04-02 00:19:03.000000000 +0300
+++ The-Powder-Toy-93.3.mod/src/Config.h	2018-06-15 13:55:36.206253889 +0300
@@ -61,11 +61,7 @@
 #define IDENT_PLATFORM "WIN32"
 #endif
 #elif defined(LIN)
-#if defined(_64BIT)
-#define IDENT_PLATFORM "LIN64"
-#else
-#define IDENT_PLATFORM "LIN32"
-#endif
+#define IDENT_PLATFORM "TERMUX"
 #elif defined(MACOSX)
 #define IDENT_PLATFORM "MACOSX"
 #else
@@ -88,11 +84,11 @@
 #define SERVER "powdertoy.co.uk"
 #define STATICSERVER "static.powdertoy.co.uk"
 
-#define LOCAL_SAVE_DIR "Saves"
+#define LOCAL_SAVE_DIR "/data/data/com.termux/files/home/.powdertoy/saves"
 
-#define STAMPS_DIR "stamps"
+#define STAMPS_DIR "/data/data/com.termux/files/home/.powdertoy/stamps"
 
-#define BRUSH_DIR "Brushes"
+#define BRUSH_DIR "/data/data/com.termux/files/home/.powdertoy/brushes"
 
 //Number of unique thumbnails to have in cache at one time
 #define THUMB_CACHE_SIZE 256
