diff -uNr The-Powder-Toy-93.3/src/gui/save/LocalSaveActivity.cpp The-Powder-Toy-93.3.mod/src/gui/save/LocalSaveActivity.cpp
--- The-Powder-Toy-93.3/src/gui/save/LocalSaveActivity.cpp	2018-04-02 00:19:03.000000000 +0300
+++ The-Powder-Toy-93.3.mod/src/gui/save/LocalSaveActivity.cpp	2018-06-15 13:55:36.216253888 +0300
@@ -109,6 +109,7 @@
 
 void LocalSaveActivity::saveWrite(std::string finalFilename)
 {
+	Client::Ref().MakeDirectory("/data/data/com.termux/files/home/.powdertoy");
 	Client::Ref().MakeDirectory(LOCAL_SAVE_DIR);
 	GameSave *gameSave = save.GetGameSave();
 	Json::Value localSaveInfo;
