diff -uNr The-Powder-Toy-93.3/data/IntroText.h The-Powder-Toy-93.3.mod/data/IntroText.h
--- The-Powder-Toy-93.3/data/IntroText.h	2018-04-02 00:19:03.000000000 +0300
+++ The-Powder-Toy-93.3.mod/data/IntroText.h	2018-06-15 13:55:36.196253889 +0300
@@ -28,6 +28,8 @@
 	"\brThis is a BETA, you cannot save things publicly. If you are planning on publishing any saves, use the release version\n"
 #endif
 	"\n"
+    "\brWarning: this is an experimental build and it may contain bugs.\n"
+    "\n"
 	"\bt" MTOS(SAVE_VERSION) "." MTOS(MINOR_VERSION) "." MTOS(BUILD_NUM) " " IDENT_PLATFORM " "
 #ifdef SNAPSHOT
 	"SNAPSHOT " MTOS(SNAPSHOT_ID) " "
