diff -uNr doxygen-Release_1_8_16/addon/doxyparse/doxyparse.cpp doxygen-Release_1_8_16.mod/addon/doxyparse/doxyparse.cpp
--- doxygen-Release_1_8_16/addon/doxyparse/doxyparse.cpp	2019-08-08 16:36:52.000000000 +0300
+++ doxygen-Release_1_8_16.mod/addon/doxyparse/doxyparse.cpp	2019-08-11 20:18:30.675734491 +0300
@@ -284,7 +284,7 @@
       Argument * argument = iterator.toFirst();
       if(argument != NULL) {
         temp = argumentData(argument);
-// TODO: This is a workaround; better not include "void" in argList, in the first place. 
+// TODO: This is a workaround; better not include "void" in argList, in the first place.
           if(temp != "void") {
               printNumberOfArguments(argList->count());
           }
@@ -452,7 +452,7 @@
 #else
   unsigned int pid = (uint)GetCurrentProcessId();
 #endif
-  tmpdir << "/tmp/doxyparse-" << pid;
+  tmpdir << "@TERMUX_PREFIX@/tmp/doxyparse-" << pid;
   Config_getString(OUTPUT_DIRECTORY)= tmpdir.str().c_str();
   // enable HTML (fake) output to omit warning about missing output format
   Config_getBool(GENERATE_HTML)=TRUE;
