diff -uNr doxygen-Release_1_8_17/addon/doxyparse/doxyparse.cpp doxygen-Release_1_8_17.mod/addon/doxyparse/doxyparse.cpp
--- doxygen-Release_1_8_17/addon/doxyparse/doxyparse.cpp	2019-12-27 19:52:36.000000000 +0200
+++ doxygen-Release_1_8_17.mod/addon/doxyparse/doxyparse.cpp	2019-12-31 14:59:53.592555381 +0200
@@ -442,7 +442,7 @@
 #else
   unsigned int pid = (uint)GetCurrentProcessId();
 #endif
-  tmpdir << "/tmp/doxyparse-" << pid;
+  tmpdir << "@TERMUX_PREFIX@/tmp/doxyparse-" << pid;
   Config_getString(OUTPUT_DIRECTORY)= tmpdir.str().c_str();
   // enable HTML (fake) output to omit warning about missing output format
   Config_getBool(GENERATE_HTML)=TRUE;
