diff -u -r ../mdocml-1.13.4/configure ./configure
--- ../mdocml-1.13.4/configure	2016-07-14 07:13:40.000000000 -0400
+++ ./configure	2016-07-16 17:59:13.906589870 -0400
@@ -31,7 +31,7 @@
 # Initialize all variables here,
 # such that nothing can leak in from the environment.
 
-MANPATH_DEFAULT="/usr/share/man:/usr/X11R6/man:/usr/local/man"
+MANPATH_DEFAULT="@TERMUX_PREFIX@/share/man"
 OSNAME=
 
 CC=`printf "all:\\n\\t@echo \\\$(CC)\\n" | make -f -`
@@ -300,7 +300,7 @@
 [ ${HAVE_GETLINE} -eq 0 ] && echo "#include <stdio.h>"
 
 echo
-echo "#define MAN_CONF_FILE \"/etc/${MANM_MANCONF}\""
+echo "#define MAN_CONF_FILE \"@TERMUX_PREFIX@/etc/${MANM_MANCONF}\""
 echo "#define MANPATH_DEFAULT \"${MANPATH_DEFAULT}\""
 [ -n "${OSNAME}" ] && echo "#define OSNAME \"${OSNAME}\""
 [ -n "${HOMEBREWDIR}" ] && echo "#define HOMEBREWDIR \"${HOMEBREWDIR}\""
