diff -u -r ../mdocml-1.13.1/mandocdb.c ./mandocdb.c
--- ../mdocml-1.13.1/mandocdb.c	2014-08-10 19:26:46.000000000 +0200
+++ ./mandocdb.c	2014-08-18 08:22:05.529753054 +0200
@@ -1440,7 +1440,7 @@
 	if (TYPE_arch == type)
 		for (cp = value; *cp; cp++)
 			if (isupper((unsigned char)*cp))
-				*cp = _tolower((unsigned char)*cp);
+				*cp = tolower((unsigned char)*cp);
 	putkeys(mpage, value, strlen(value), type);
 }
 
