diff --git a/qt6/CMakeLists.txt b/qt6/CMakeLists.txt
index bff3a03..28ee62e 100644
--- a/qt6/CMakeLists.txt
+++ b/qt6/CMakeLists.txt
@@ -2,9 +2,18 @@
 find_package(Qt6 ${REQUIRED_QT6_VERSION} CONFIG REQUIRED Core DBus Widgets)
 find_package(Qt6Gui ${REQUIRED_QT6_VERSION} REQUIRED Private)
 if (ENABLE_QT6_WAYLAND_WORKAROUND)
+if(Qt6_VERSION VERSION_GREATER_EQUAL 6.10)
+  find_package(Qt6 COMPONENTS WaylandClientPrivate REQUIRED)
+else()
   find_package(Qt6WaylandClient ${REQUIRED_QT6_VERSION} REQUIRED Private)
   find_package(Qt6WaylandGlobalPrivate ${REQUIRED_QT6_VERSION} REQUIRED)
 endif()
+endif()
+if(Qt6_VERSION VERSION_GREATER_EQUAL 6.10)
+  find_package(Qt6 COMPONENTS GuiPrivate REQUIRED)
+else()
+  find_package(Qt6Gui ${REQUIRED_QT6_VERSION} REQUIRED Private)
+endif()
 
 add_subdirectory(dbusaddons)
 add_subdirectory(platforminputcontext)
diff --git a/qt6/immodule-probing/CMakeLists.txt b/qt6/immodule-probing/CMakeLists.txt
index efa5e4b..a1bfce4 100644
--- a/qt6/immodule-probing/CMakeLists.txt
+++ b/qt6/immodule-probing/CMakeLists.txt
@@ -1,5 +1,5 @@
 add_executable(fcitx5-qt6-immodule-probing main.cpp)
 target_include_directories(fcitx5-qt6-immodule-probing PRIVATE ${Qt6Gui_PRIVATE_INCLUDE_DIRS})
-target_link_libraries(fcitx5-qt6-immodule-probing Qt6::Gui)
+target_link_libraries(fcitx5-qt6-immodule-probing Qt6::Gui Qt6::GuiPrivate)
 
-install(TARGETS fcitx5-qt6-immodule-probing DESTINATION "${CMAKE_INSTALL_BINDIR}")
\ No newline at end of file
+install(TARGETS fcitx5-qt6-immodule-probing DESTINATION "${CMAKE_INSTALL_BINDIR}")
