Tests doesn't work when cross compiling.

diff -u -r ../libconfig-1.5/Makefile.in ./Makefile.in
--- ../libconfig-1.5/Makefile.in	2014-05-28 01:54:57.000000000 -0400
+++ ./Makefile.in	2016-01-19 17:58:57.829691367 -0500
@@ -337,7 +337,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = lib doc tinytest tests $(am__append_1)
+SUBDIRS = lib doc tinytest $(am__append_1)
 EXTRA_DIST = \
 	m4/*.m4 \
 	*.sln \
