--- a/tools/ipv6calc-db-update.sh.in
+++ b/tools/ipv6calc-db-update.sh.in
@@ -12,10 +12,10 @@
 [ -z "$IPV6CALC_DAT_DIR" -a -n "$IPV6CALC_DB_EXTERNAL_DIR" ] && IPV6CALC_DAT_DIR="$IPV6CALC_DB_EXTERNAL_DIR"
 [ -z "$IPV6CALC_DAT_DIR" ] && IPV6CALC_DAT_DIR="$IPV6CALC_DAT_DIR_DEFAULT"
 
-IPV6CALC_UPDATE_LIST="./ipv6calc-update-registries.sh /usr/share/ipv6calc/tools/ipv6calc-update-registries.sh"
-IPV6CALC_CHECK_LIST="./ipv6calc-registries-check-run-create.sh /usr/share/ipv6calc/tools/ipv6calc-registries-check-run-create.sh"
-IPV6CALC_CREATE_IPV4_LIST="./ipv6calc-create-registry-list-ipv4.pl /usr/share/ipv6calc/tools/ipv6calc-create-registry-list-ipv4.pl"
-IPV6CALC_CREATE_IPV6_LIST="./ipv6calc-create-registry-list-ipv6.pl /usr/share/ipv6calc/tools/ipv6calc-create-registry-list-ipv6.pl"
+IPV6CALC_UPDATE_LIST="./ipv6calc-update-registries.sh @TERMUX_PREFIX@/share/ipv6calc/tools/ipv6calc-update-registries.sh"
+IPV6CALC_CHECK_LIST="./ipv6calc-registries-check-run-create.sh @TERMUX_PREFIX@/share/ipv6calc/tools/ipv6calc-registries-check-run-create.sh"
+IPV6CALC_CREATE_IPV4_LIST="./ipv6calc-create-registry-list-ipv4.pl @TERMUX_PREFIX@/share/ipv6calc/tools/ipv6calc-create-registry-list-ipv4.pl"
+IPV6CALC_CREATE_IPV6_LIST="./ipv6calc-create-registry-list-ipv6.pl @TERMUX_PREFIX@/share/ipv6calc/tools/ipv6calc-create-registry-list-ipv6.pl"
 
 
 help() {
