diff -uNr restic/internal/fs/file_unix.go restic.mod/internal/fs/file_unix.go
--- restic/internal/fs/file_unix.go	2019-07-29 10:22:22.000000000 +0300
+++ restic.mod/internal/fs/file_unix.go	2020-01-01 17:54:08.797875856 +0200
@@ -47,12 +47,6 @@
 
 // Chmod changes the mode of the named file to mode.
 func Chmod(name string, mode os.FileMode) error {
-	err := os.Chmod(fixpath(name), mode)
-
-	// ignore the error if the FS does not support setting this mode (e.g. CIFS with gvfs on Linux)
-	if err != nil && isNotSupported(err) {
-		return nil
-	}
-
-	return err
+	os.Chmod(fixpath(name), mode)
+	return nil
 }
