diff -u -r ../colordiff-1.0.18/Makefile ./Makefile
--- ../colordiff-1.0.18/Makefile	2016-11-12 13:12:14.000000000 +0100
+++ ./Makefile	2017-06-06 22:31:52.645026869 +0200
@@ -37,8 +37,6 @@
 	  install -d ${DESTDIR}${ETC_DIR}; \
 	fi
 	cp colordiffrc ${DESTDIR}${ETC_DIR}/colordiffrc
-	-chown root.root ${DESTDIR}${ETC_DIR}/colordiffrc
-	chmod 644 ${DESTDIR}${ETC_DIR}/colordiffrc
 
 .PHONY: uninstall
 uninstall:
