diff -u -r ../tor-0.3.4.8/src/common/crypto_rand.c ./src/common/crypto_rand.c
--- ../tor-0.3.4.8/src/common/crypto_rand.c	2018-06-15 21:20:53.000000000 +0000
+++ ./src/common/crypto_rand.c	2018-09-12 04:25:23.741656806 +0000
@@ -33,6 +33,7 @@
 
 DISABLE_GCC_WARNING(redundant-decls)
 #include <openssl/rand.h>
+#include <openssl/sha.h>
 ENABLE_GCC_WARNING(redundant-decls)
 
 #if __GNUC__ && GCC_VERSION >= 402
