diff -u -r ../tor-0.3.4.8/src/common/compat_openssl.h ./src/common/compat_openssl.h
--- ../tor-0.3.4.8/src/common/compat_openssl.h	2018-06-15 21:20:53.000000000 +0000
+++ ./src/common/compat_openssl.h	2018-09-12 04:22:24.387770893 +0000
@@ -28,10 +28,6 @@
 #define OPENSSL_1_1_API
 #endif /* OPENSSL_VERSION_NUMBER >= OPENSSL_V_SERIES(1,1,0) && ... */
 
-#ifndef OPENSSL_VERSION
-#define OPENSSL_VERSION SSLEAY_VERSION
-#endif
-
 #ifndef OPENSSL_1_1_API
 #define OpenSSL_version(v) SSLeay_version(v)
 #define OpenSSL_version_num() SSLeay()
