diff --git a/CMakeLists.txt b/CMakeLists.txt
index 44cd9ac6e..b061830b5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -170,6 +170,9 @@ else (MACOSX) # --> Linux
     endif (IS_64_BIT)
   endif (DEBIAN OR ARCHLINUX)
 endif (MACOSX)
+set (CMAKE_MOUNT_INSTALL_BINDIR "${CMAKE_INSTALL_PREFIX}/sbin")
+set (CVMFS_LIBEXEC_DIR "lib/cvmfs")
+set (CMAKE_INSTALL_LIBDIR "lib")
 message ("Installing shared libraries to: ${CMAKE_INSTALL_LIBDIR}")
 
 #
diff --git a/cvmfs/CMakeLists.txt b/cvmfs/CMakeLists.txt
index d7409cc67..e49f1434f 100644
--- a/cvmfs/CMakeLists.txt
+++ b/cvmfs/CMakeLists.txt
@@ -372,11 +372,11 @@ if (BUILD_CVMFS)
   endif ()
   if (NOT MACOSX)
   # Systemd service unit for reloads after package upgrades
-  install (
-    FILES pkg/cvmfs-reload.service
-    DESTINATION /usr/lib/systemd/system
-    PERMISSIONS   OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ
-  )
+  # install (
+  #   FILES pkg/cvmfs-reload.service
+  #   DESTINATION /usr/lib/systemd/system
+  #   PERMISSIONS   OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ
+  # )
 
   endif()
 
