diff -u -r ../orig-src/lib/getline.cc ./lib/getline.cc
--- ../orig-src/lib/getline.cc	2023-10-02 11:41:12.201611420 +0000
+++ ./lib/getline.cc	2023-10-02 11:43:12.677004096 +0000
@@ -55,7 +55,7 @@
 
   for (;;)
     {
-      register int c = getc (stream);
+      int c = getc (stream);
 
       /* We always want at least one char left in the buffer, since we
          always (unless we get an error while reading the first char)
