--- ./ext/iconv/config.m4.orig	2019-03-23 10:05:07.289878202 +0000
+++ ./ext/iconv/config.m4	2019-03-23 10:06:27.207978972 +0000
@@ -12,22 +12,6 @@
   ])
 
   if test "$iconv_avail" != "no"; then
-    if test -z "$ICONV_DIR"; then
-      for i in /usr/local /usr; do
-        if test -f "$i/include/iconv.h" || test -f "$i/include/giconv.h"; then
-          PHP_ICONV_PREFIX="$i"
-          break
-        fi
-      done
-      if test -z "$PHP_ICONV_PREFIX"; then
-        PHP_ICONV_PREFIX="/usr"
-      fi
-    else
-      PHP_ICONV_PREFIX="$ICONV_DIR"
-    fi
-
-    CFLAGS="-I$PHP_ICONV_PREFIX/include $CFLAGS"
-    LDFLAGS="-L$PHP_ICONV_PREFIX/$PHP_LIBDIR $LDFLAGS"
 
     if test -r "$PHP_ICONV_PREFIX/include/giconv.h"; then
       PHP_ICONV_H_PATH="$PHP_ICONV_PREFIX/include/giconv.h"
