diff -uNr csol-1.5.0/CMakeLists.txt csol-1.5.0.mod/CMakeLists.txt
--- csol-1.5.0/CMakeLists.txt	2022-03-17 03:21:53.000000000 +0800
+++ csol-1.5.0.mod/CMakeLists.txt	2022-07-11 20:49:58.806999088 +0800
@@ -3,7 +3,7 @@
 cmake_minimum_required(VERSION 2.8.12)
 
 set(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
-set(CMAKE_C_FLAGS "--std=c89 -Wall -pedantic -DUSE_GETOPT")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} --std=c89 -Wall -pedantic -DUSE_GETOPT")
 
 configure_file(csolrc csolrc COPYONLY)
 file(COPY games DESTINATION .)
@@ -22,7 +22,7 @@
 target_link_libraries(csol ${CURSES_LIBRARIES})
 
 install(TARGETS csol DESTINATION bin COMPONENT binaries)
-install(FILES "${CMAKE_BINARY_DIR}/csolrc" DESTINATION /etc/xdg/csol COMPONENT config)
-install(DIRECTORY "${CMAKE_BINARY_DIR}/themes" DESTINATION /etc/xdg/csol COMPONENT config)
-install(DIRECTORY "${CMAKE_BINARY_DIR}/games" DESTINATION /etc/xdg/csol COMPONENT config)
+install(FILES "${CMAKE_BINARY_DIR}/csolrc" DESTINATION @TERMUX_PREFIX@/etc/xdg/csol COMPONENT config)
+install(DIRECTORY "${CMAKE_BINARY_DIR}/themes" DESTINATION @TERMUX_PREFIX@/etc/xdg/csol COMPONENT config)
+install(DIRECTORY "${CMAKE_BINARY_DIR}/games" DESTINATION @TERMUX_PREFIX@/etc/xdg/csol COMPONENT config)
 install(FILES "${PROJECT_SOURCE_DIR}/doc/csol.6" DESTINATION ${CMAKE_INSTALL_PREFIX}/share/man/man6)
