diff --git a/src/glsl/meson.build b/src/glsl/meson.build
index c9a85429..1c13fc44 100644
--- a/src/glsl/meson.build
+++ b/src/glsl/meson.build
@@ -39,7 +39,7 @@ elif not glslang_req.disabled()
   # glslang = dependency('glslang', method: 'cmake', modules: ['glslang::SPIRV'])
 
   vulkan_sdk = get_option('vulkan-sdk')
-  vulkan_lib_dirs = []
+  vulkan_lib_dirs = ['@TERMUX_PREFIX@/lib']
   if vulkan_sdk != ''
     vulkan_lib_dirs += [vulkan_sdk / 'lib']
   endif
