# gtk4 is not installed in host builder

--- a/meson-post-install.py
+++ b/meson-post-install.py
@@ -12,7 +12,7 @@
 	subprocess.call(['glib-compile-schemas', os.path.join(datadir, 'glib-2.0', 'schemas')])
 
 	print('Updating icon cache...')
-	subprocess.call(['gtk4-update-icon-cache', '-qtf', os.path.join(datadir, 'icons', 'hicolor')])
+	subprocess.call(['gtk-update-icon-cache', '-qtf', os.path.join(datadir, 'icons', 'hicolor')])
 
 	print('Updating desktop database...')
 	subprocess.call(['update-desktop-database', '-q', os.path.join(datadir, 'applications')])
