--- a/cups/dest.c
+++ b/cups/dest.c
@@ -125,6 +125,8 @@
 {
   int		num_dests;		/* Number of destinations */
   cups_dest_t	*dests;			/* Destinations */
+  int		num_local;		/* Number of local cupsd queues */
+  cups_dest_t	*local_dests;		/* Local cupsd queues */
   char		def_name[1024],		/* Default printer name, if any */
 		*def_instance;		/* Default printer instance, if any */
 } _cups_getdata_t;
