--- a/meson.build
+++ b/meson.build
@@ -36,7 +36,7 @@
 gdk_pixbuf = dependency('gdk-pixbuf-2.0', required: get_option('gdk-pixbuf'))
 
 git = find_program('git', required: false, native: true)
-scdoc = find_program('scdoc', required: get_option('man-pages'), native: true)
+scdoc = find_program('scdoc', required: get_option('man-pages'))
 
 version = '"@0@"'.format(meson.project_version())
 if git.found()
