https://github.com/termux/termux-packages/issues/15852

--- a/configure
+++ b/configure
@@ -1001,6 +1001,7 @@
 #line 1004 "configure"
 #include "confdefs.h"
 
+int
 main(){return(0);}
 EOF
 if { (eval echo configure:1009: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
@@ -2054,7 +2055,7 @@
 #include "confdefs.h"
 
 int main() {
-} $ac_kw foo() {
+} $ac_kw int foo() {
 ; return 0; }
 EOF
 if { (eval echo configure:2063: \"$ac_compile\") 1>&5; (eval $ac_compile) 2>&5; }; then
@@ -2321,6 +2322,7 @@
 #include "confdefs.h"
 #include <stdio.h>
 #include <sys/types.h>
+int
 main()
 {
   FILE *f=fopen("conftestval", "w");
@@ -2361,6 +2363,7 @@
 #include "confdefs.h"
 #include <stdio.h>
 #include <sys/types.h>
+int
 main()
 {
   FILE *f=fopen("conftestval", "w");
@@ -2401,6 +2404,7 @@
 #include "confdefs.h"
 #include <stdio.h>
 #include <sys/types.h>
+int
 main()
 {
   FILE *f=fopen("conftestval", "w");
@@ -2442,6 +2446,7 @@
 #include "confdefs.h"
 #include <stdio.h>
 #include <sys/types.h>
+int
 main()
 {
   FILE *f=fopen("conftestval", "w");
@@ -8943,6 +8948,7 @@
 cat > conftest.$ac_ext <<EOF
 #line 8943 "configure"
 #include "confdefs.h"
+char *${x_direct_test_function}();
 
 int main() {
 ${x_direct_test_function}()
