--- ../CMakeLists.txt.orig	2019-01-26 00:17:38.419818085 +0000
+++ ./src/CMakeLists.txt	2019-01-26 00:25:12.349814559 +0000
@@ -874,13 +874,6 @@
       INTERFACE $<$<PLATFORM_ID:Emscripten>:"-O0">
   )
 
-  add_custom_command (
-      OUTPUT ${HDF5_GENERATED_SOURCE_DIR}/H5Tinit.c
-      COMMAND ${CMAKE_CROSSCOMPILING_EMULATOR}$<TARGET_FILE:H5detect>
-      ARGS ${HDF5_GENERATED_SOURCE_DIR}/H5Tinit.c
-      DEPENDS H5detect
-      WORKING_DIRECTORY ${HDF5_GENERATED_SOURCE_DIR}
-  )
   if (BUILD_SHARED_LIBS)
     if (NOT EXISTS ${HDF5_GENERATED_SOURCE_DIR}/shared/H5Tinit.c)
       add_custom_command (
@@ -916,23 +909,6 @@
     INTERFACE $<$<PLATFORM_ID:Emscripten>:"-O0">
 )
 
-add_custom_command (
-    OUTPUT ${HDF5_BINARY_DIR}/H5lib_settings.c
-    COMMAND ${CMAKE_CROSSCOMPILING_EMULATOR}$<TARGET_FILE:H5make_libsettings>
-    ARGS ${HDF5_BINARY_DIR}/H5lib_settings.c
-    DEPENDS H5make_libsettings
-    WORKING_DIRECTORY ${HDF5_BINARY_DIR}
-)
-if (BUILD_SHARED_LIBS)
-  add_custom_command (
-      OUTPUT ${HDF5_BINARY_DIR}/shared/H5lib_settings.c
-      COMMAND ${CMAKE_CROSSCOMPILING_EMULATOR}$<TARGET_FILE:H5make_libsettings>
-      ARGS ${HDF5_BINARY_DIR}/shared/H5lib_settings.c
-      DEPENDS H5make_libsettings
-      WORKING_DIRECTORY ${HDF5_BINARY_DIR}
-  )
-endif ()
-
 ## all_packages="AC,B,B2,D,F,FA,FL,FS,HL,I,O,S,ST,T,Z"
 #all_packages="AC,B2,CX,D,F,HL,I,O,S,ST,T,Z"
 option (HDF5_ENABLE_DEBUG_APIS "Turn on extra debug output in all packages" OFF)
