--- ./usr/include/stdlib.h.orig	2021-08-16 11:50:47.432239182 +0200
+++ ./usr/include/stdlib.h	2021-08-16 11:51:31.102222894 +0200
@@ -33,6 +33,7 @@
 #include <bits/wait.h>
 #include <malloc.h>
 #include <stddef.h>
+#include <stdint.h>
 #include <sys/cdefs.h>
 #include <xlocale.h>
 
@@ -223,12 +224,8 @@
 
 size_t wcstombs(char* _Nullable __dst, const wchar_t* _Nullable __src, size_t __n) __INTRODUCED_IN_NO_GUARD_FOR_NDK(21);
 
 
-#if __ANDROID_API__ >= 21
-size_t __ctype_get_mb_cur_max(void) __INTRODUCED_IN(21);
-#endif /* __ANDROID_API__ >= 21 */
-
-#define MB_CUR_MAX __ctype_get_mb_cur_max()
+#define MB_CUR_MAX 4
 
 #if defined(__BIONIC_INCLUDE_FORTIFY_HEADERS)
 #include <bits/fortify/stdlib.h>
