--- ./usr/include/sys/time.h.orig	2024-03-21 21:20:39.291006515 +0000
+++ ./usr/include/sys/time.h	2024-03-23 15:15:08.754434710 +0000
@@ -48,10 +48,8 @@
 
 #if defined(__USE_BSD)
 
-#if __ANDROID_API__ >= 26
-int futimes(int __fd, const struct timeval __times[_Nullable 2]) __INTRODUCED_IN(26);
-int lutimes(const char* _Nonnull __path, const struct timeval __times[_Nullable 2]) __INTRODUCED_IN(26);
-#endif /* __ANDROID_API__ >= 26 */
+int futimes(int __fd, const struct timeval __times[_Nullable 2]);
+int lutimes(const char* _Nonnull __path, const struct timeval __times[_Nullable 2]);
 
 #endif
 
@@ -70,9 +68,7 @@
  * Available since API level 26.
  */
 
-#if __ANDROID_API__ >= 26
-int futimesat(int __dir_fd, const char* __BIONIC_COMPLICATED_NULLNESS __path, const struct timeval __times[_Nullable 2]) __INTRODUCED_IN(26);
-#endif /* __ANDROID_API__ >= 26 */
+int futimesat(int __dir_fd, const char* __BIONIC_COMPLICATED_NULLNESS __path, const struct timeval __times[_Nullable 2]);
 
 #endif
 
