--- a/uwsgi.h
+++ b/uwsgi.h
@@ -4967,6 +4967,15 @@
 int uwsgi_wait_for_mountpoint(char *);
 int uwsgi_wait_for_socket(char *);
 
+#ifdef __ANDROID__
+#define pivot_root(new_root, put_old) (-1)
+#define pthread_setcancelstate(state, oldstate) (-1)
+#define pthread_setcanceltype(type, oldtype) (-1)
+#if __ANDROID_API__ < 26
+int futimes(int, const struct timeval[2]);
+#endif
+#endif
+
 #ifdef __cplusplus
 }
 #endif
