diff --git a/libnemo-private/nemo-file-utilities.c b/libnemo-private/nemo-file-utilities.c
index 8a45214..2e03ac8 100644
--- a/libnemo-private/nemo-file-utilities.c
+++ b/libnemo-private/nemo-file-utilities.c
@@ -862,7 +862,7 @@ nemo_ensure_unique_file_name (const char *directory_uri,
 char *
 nemo_unique_temporary_file_name (void)
 {
-	const char *prefix = "/tmp/nemo-temp-file";
+	const char *prefix = "@TERMUX_PREFIX@/tmp/nemo-temp-file";
 	char *file_name;
 	int fd;
 
