diff -u -r ../rasqal-0.9.33/src/Makefile.in ./src/Makefile.in
--- ../rasqal-0.9.33/src/Makefile.in	2014-11-23 22:11:38.000000000 +0000
+++ ./src/Makefile.in	2024-05-21 21:19:23.003617513 +0000
@@ -654,7 +654,7 @@
 	$(am__append_5) $(am__append_6) $(am__append_7)
 nodist_librasqal_la_SOURCES = $(am__append_8)
 librasqal_la_LDFLAGS = -version-info @RASQAL_LIBTOOL_VERSION@
-librasqal_la_LIBADD = @LTLIBOBJS@ @RASQAL_INTERNAL_LIBS@ \
+librasqal_la_LIBADD = -lm @LTLIBOBJS@ @RASQAL_INTERNAL_LIBS@ \
 	@RASQAL_EXTERNAL_LIBS@ $(MEM_LIBS) \
 	$(top_builddir)/libsv/libsv.la $(am__append_10)
 librasqal_la_DEPENDENCIES = @LTLIBOBJS@ @RASQAL_INTERNAL_LIBS@ \
