--- ./CMakeLists.txt	2017-11-12 18:14:32.000000000 +0000
+++ ../CMakeLists.txt	2017-11-18 19:48:53.494691943 +0000
@@ -60,6 +60,7 @@
 option(USE_FIXEDPOINT "Use fixed point arithmetic in the Splash backend" OFF)
 option(USE_FLOAT "Use single precision arithmetic in the Splash backend" OFF)
 option(BUILD_SHARED_LIBS "Build poppler as a shared library" ON)
+option(ENABLE_MULTITHREAD "Build with multithread support" ON)
 if(WIN32)
   option(ENABLE_RELOCATABLE "Do not hardcode the poppler library location (on Windows)." ON)
 else()
@@ -105,7 +106,7 @@
   message(FATAL_ERROR "Invalid font configuration setting: ${FONT_CONFIGURATION}")
 endif()
 
-if(CMAKE_USE_PTHREADS_INIT OR WIN32)
+if((CMAKE_USE_PTHREADS_INIT OR WIN32) AND ENABLE_MULTITHREAD)
   set(MULTITHREADED ON)
 endif()
 
