diff -uNr poppler-0.65.0/goo/gfile.cc poppler-0.65.0.mod/goo/gfile.cc
--- poppler-0.65.0/goo/gfile.cc	2018-05-19 18:05:25.000000000 +0300
+++ poppler-0.65.0.mod/goo/gfile.cc	2018-06-19 17:01:32.437678208 +0300
@@ -439,7 +439,7 @@
   if ((s = getenv("TMPDIR"))) {
     *name = new GooString(s);
   } else {
-    *name = new GooString("/tmp");
+    *name = new GooString("@TERMUX_PREFIX@/tmp");
   }
   (*name)->append("/XXXXXX");
   fd = mkstemp((*name)->getCString());
