diff -uNr poppler-0.65.0/poppler/GlobalParams.cc poppler-0.65.0.mod/poppler/GlobalParams.cc
--- poppler-0.65.0/poppler/GlobalParams.cc	2018-05-19 18:05:25.000000000 +0300
+++ poppler-0.65.0.mod/poppler/GlobalParams.cc	2018-06-19 16:04:51.997503128 +0300
@@ -1239,11 +1239,11 @@
 };
 
 static const char *displayFontDirs[] = {
-  "/usr/share/ghostscript/fonts",
-  "/usr/local/share/ghostscript/fonts",
-  "/usr/share/fonts/default/Type1",
-  "/usr/share/fonts/default/ghostscript",
-  "/usr/share/fonts/type1/gsfonts",
+  "@TERMUX_PREFIX@/share/ghostscript/fonts",
+  "@TERMUX_PREFIX@/local/share/ghostscript/fonts",
+  "@TERMUX_PREFIX@/share/fonts/default/Type1",
+  "@TERMUX_PREFIX@/share/fonts/default/ghostscript",
+  "@TERMUX_PREFIX@/share/fonts/type1/gsfonts",
   NULL
 };
 
