--- a/configure
+++ b/configure
@@ -10490,7 +10490,7 @@
 do
   LIBS=$SAVED_LIBS
   LIBS="$LIBS $extralib"
-  if test "$cross_compiling" = yes; then :
+  if false; then :
 
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: cross-compiling" >&5
 $as_echo "cross-compiling" >&6; }  # FIXME
@@ -10515,7 +10515,7 @@
 }
 
 _ACEOF
-if ac_fn_cxx_try_run "$LINENO"; then :
+if ac_fn_cxx_try_link "$LINENO"; then :
 
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
@@ -10777,7 +10777,7 @@
           # check library version, update LIBS
           { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libz version >= $required_libz_version" >&5
 $as_echo_n "checking for libz version >= $required_libz_version... " >&6; }
-          if test "$cross_compiling" = yes; then :
+          if false; then :
   { $as_echo "$as_me:${as_lineno-$LINENO}: result: cross-compiling, forced" >&5
 $as_echo "cross-compiling, forced" >&6; }
 
@@ -10797,9 +10797,9 @@
 
 
 _ACEOF
-if ac_fn_cxx_try_run "$LINENO"; then :
+if ac_fn_cxx_try_link "$LINENO"; then :
 
-              ax_check_lib_version=`eval $ac_try 2>&1`
+              ax_check_lib_version=
 
 
 
@@ -10831,7 +10831,7 @@
 
 
 
-    if test "$ax_compare_version" = "true" ; then
+    if true ; then
     ax_check_lib="ok"
     else ax_check_lib="bad"
   fi
@@ -10963,7 +10963,7 @@
           # check library version, update LIBS
           { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libpng version >= $required_libpng_version" >&5
 $as_echo_n "checking for libpng version >= $required_libpng_version... " >&6; }
-          if test "$cross_compiling" = yes; then :
+          if false; then :
   { $as_echo "$as_me:${as_lineno-$LINENO}: result: cross-compiling, forced" >&5
 $as_echo "cross-compiling, forced" >&6; }
 
@@ -10983,9 +10983,9 @@
 
 
 _ACEOF
-if ac_fn_cxx_try_run "$LINENO"; then :
+if ac_fn_cxx_try_link "$LINENO"; then :
 
-              ax_check_lib_version=`eval $ac_try 2>&1`
+              ax_check_lib_version=
 
 
 
@@ -11017,7 +11017,7 @@
 
 
 
-    if test "$ax_compare_version" = "true" ; then
+    if true ; then
     ax_check_lib="ok"
     else ax_check_lib="bad"
   fi
@@ -11164,7 +11164,7 @@
             { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libjpeg version >= $required_libjpeg_version" >&5
 $as_echo_n "checking for libjpeg version >= $required_libjpeg_version... " >&6; }
           fi
-          if test "$cross_compiling" = yes; then :
+          if false; then :
   { $as_echo "$as_me:${as_lineno-$LINENO}: result: cross-compiling, forced" >&5
 $as_echo "cross-compiling, forced" >&6; }
 
@@ -11182,9 +11182,9 @@
 
 
 _ACEOF
-if ac_fn_cxx_try_run "$LINENO"; then :
+if ac_fn_cxx_try_link "$LINENO"; then :
 
-              ax_check_libjpeg_version=`eval $ac_try 2>&1`
+              ax_check_libjpeg_version=
 
 
 
@@ -11216,7 +11216,7 @@
 
 
 
-    if test "$ax_compare_version" = "true" ; then
+    if true ; then
     ax_check_libjpeg="ok"
     else ax_check_libjpeg="bad"
   fi
@@ -11348,7 +11348,7 @@
           # check library version, update LIBS
           { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libtiff version >= $required_libtiff_version" >&5
 $as_echo_n "checking for libtiff version >= $required_libtiff_version... " >&6; }
-          if test "$cross_compiling" = yes; then :
+          if false; then :
   { $as_echo "$as_me:${as_lineno-$LINENO}: result: cross-compiling, forced" >&5
 $as_echo "cross-compiling, forced" >&6; }
 
@@ -11377,9 +11377,9 @@
 
 
 _ACEOF
-if ac_fn_cxx_try_run "$LINENO"; then :
+if ac_fn_cxx_try_link "$LINENO"; then :
 
-              ax_check_libtiff_version=`eval $ac_try 2>&1`
+              ax_check_libtiff_version=
 
 
 
@@ -11411,7 +11411,7 @@
 
 
 
-    if test "$ax_compare_version" = "true" ; then
+    if true ; then
     ax_check_libtiff="ok"
     else ax_check_libtiff="bad"
   fi
