diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9db2b6b6e3..dd768e5ebf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -576,7 +576,7 @@ set (Boost_FIND_QUIETLY ON)
 if (NOT DEFINED ${BOOST_ROOT})
   set(BOOST_ROOT $ENV{BOOST_ROOT})
 endif()
-find_package (Boost 1.67.0 COMPONENTS date_time filesystem locale program_options regex system)
+find_package (Boost 1.67.0 COMPONENTS date_time filesystem locale program_options regex)
 
 if (Boost_FOUND)
   include_directories(${Boost_INCLUDE_DIRS})
