--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -108,7 +108,7 @@ set(LIBDIR ${CMAKE_INSTALL_LIBDIR} CACHE STRING "object code libraries")
 set(LOCALEDIR ${DATAROOTDIR}/locale CACHE STRING "locale-dependent data")
 set(GNC_HELPDIR ${DATADIR} CACHE STRING "where to store help files")
 set(DATADIRNAME share)
-set(GNC_SYSTEM_XDG_DATA_DIRS /usr/local/share /usr/share)
+set(GNC_SYSTEM_XDG_DATA_DIRS @TERMUX_PREFIX@/share)
 if (NOT DEFINED GNC_DBD_DIR)
   set(GNC_DBD_DIR $ENV{GNC_DBD_DIR} CACHE PATH "Hint for location of libdbi-drivers.")
 endif()
