--- a/configure.ac
+++ b/configure.ac
@@ -76,7 +76,7 @@
   *)
     AC_MSG_RESULT([${host_os}])
     AX_PTHREAD([], [AC_MSG_ERROR([pthread is required to build $PACKAGE_NAME])])
-    AC_CHECK_LIB(pthread, [pthread_once], [], [AC_MSG_ERROR([pthread with pthread_once required to build $PACKAGE_NAME])])
+    AC_CHECK_LIB(pthread, [pthread_once], [], [AC_MSG_NOTICE([pthread with pthread_once required to build $PACKAGE_NAME])])
     ;;
 esac
 AM_CONDITIONAL(WIN32, test x$win32 = xtrue)

