--- a/server/server_settings.hpp
+++ b/server/server_settings.hpp
@@ -40,7 +40,7 @@
         /// Number of worker threads
         int threads{-1};
         /// PID file, if running as daemon
-        std::string pid_file{"/var/run/snapserver/pid"};
+        std::string pid_file{"@TERMUX_PREFIX@/var/run/snapserver/pid"};
         /// User when running as deaemon
         std::string user{"snapserver"};
         /// Group when running as deaemon
@@ -213,9 +213,9 @@
     struct Stream
     {
         /// Directory for stream plugins
-        std::filesystem::path plugin_dir{"/usr/share/snapserver/plug-ins"};
+        std::filesystem::path plugin_dir{"@TERMUX_PREFIX@/share/snapserver/plug-ins"};
         /// Directory for executable process stream sources
-        std::filesystem::path sandbox_dir{"/usr/share/snapserver/sandbox"};
+        std::filesystem::path sandbox_dir{"@TERMUX_PREFIX@/share/snapserver/sandbox"};
         /// Stream sources
         std::vector<std::string> sources;
         /// Default codec
