--- a/server/CMakeLists.txt
+++ b/server/CMakeLists.txt
@@ -41,7 +41,7 @@
     ${CMAKE_SOURCE_DIR}/server
     ${CMAKE_SOURCE_DIR}/common)
 
-if (ANDROID)
+if (FALSE)
     find_package(vorbis REQUIRED CONFIG)
     list(APPEND SERVER_LIBRARIES boost::boost)
     list(APPEND SERVER_LIBRARIES flac::flac)
@@ -90,12 +90,6 @@
         list(APPEND SERVER_INCLUDE ${OPUS_INCLUDE_DIRS})
     endif (OPUS_FOUND)
 
-    if (ALSA_FOUND)
-        list(APPEND SERVER_SOURCES streamreader/alsa_stream.cpp)
-        list(APPEND SERVER_LIBRARIES ${ALSA_LIBRARIES})
-        list(APPEND SERVER_INCLUDE ${ALSA_INCLUDE_DIRS})
-    endif (ALSA_FOUND)
-
     if (EXPAT_FOUND)
         list(APPEND SERVER_LIBRARIES ${EXPAT_LIBRARIES})
         list(APPEND SERVER_INCLUDE ${EXPAT_INCLUDE_DIRS})
@@ -105,7 +99,7 @@
 #list(APPEND SERVER_LIBRARIES Boost::boost)
 
 include_directories(${SERVER_INCLUDE})
-if (ANDROID)
+if (FALSE)
     add_executable(libsnapserver.so ${SERVER_SOURCES})
     target_link_libraries(libsnapserver.so ${SERVER_LIBRARIES} log)
 else()
@@ -117,6 +111,6 @@
     install(FILES etc/snapserver.conf COMPONENT server DESTINATION ${CMAKE_INSTALL_SYSCONFDIR})
     install(FILES etc/index.html COMPONENT server DESTINATION ${CMAKE_INSTALL_DATADIR}/snapserver)
     install(DIRECTORY etc/snapweb/ DESTINATION ${CMAKE_INSTALL_DATADIR}/snapserver/snapweb)
-    install(FILES etc/plug-ins/meta_mpd.py PERMISSIONS OWNER_EXECUTE GROUP_EXECUTE WORLD_EXECUTE DESTINATION ${CMAKE_INSTALL_DATADIR}/snapserver/plug-ins/)
+    install(FILES etc/plug-ins/meta_mpd.py PERMISSIONS OWNER_READ OWNER_EXECUTE GROUP_EXECUTE WORLD_EXECUTE DESTINATION ${CMAKE_INSTALL_DATADIR}/snapserver/plug-ins/)
     #install(FILES ../debian/snapserver.service DESTINATION ${SYSTEMD_SERVICES_INSTALL_DIR})
 endif()
