diff --git a/src/log.h b/src/log.h
index e40fe3c..720b58d 100644
--- a/src/log.h
+++ b/src/log.h
@@ -4,9 +4,16 @@
 #pragma once
 #include <assert.h>
 #include <stdio.h>
+#include <time.h>
 
 #include "compiler.h"
 
+#if __ANDROID_API__ < 29
+#define __TIMESPEC__GET_WRAPPER__
+int timespec_get_wrapper(struct timespec *ts, int base);
+#define timespec_get timespec_get_wrapper
+#endif
+
 enum log_level {
        LOG_LEVEL_INVALID = -1,
        LOG_LEVEL_TRACE = 0,
