--- a/io.c
+++ b/io.c
@@ -2080,7 +2080,7 @@
 
 		set_sigpipe_to_default();
 
-		execl("/bin/sh", "sh", "-c", command, NULL);
+		execl("@TERMUX_PREFIX@/bin/sh", "sh", "-c", command, NULL);
 		_exit(errno == ENOENT ? 127 : 126);
 
 	case -1:
@@ -2132,7 +2132,7 @@
 
 		signal(SIGPIPE, SIG_DFL);
 
-		execl("/bin/sh", "sh", "-c", command, NULL);
+		execl("@TERMUX_PREFIX@/bin/sh", "sh", "-c", command, NULL);
 		_exit(errno == ENOENT ? 127 : 126);
 
 	case -1:
@@ -2471,7 +2471,7 @@
 			fatal(_("close of pipe failed: %s"), strerror(errno));
 		/* stderr does NOT get dup'ed onto child's stdout */
 		set_sigpipe_to_default();
-		execl("/bin/sh", "sh", "-c", str, NULL);
+		execl("@TERMUX_PREFIX@/bin/sh", "sh", "-c", str, NULL);
 		_exit(errno == ENOENT ? 127 : 126);
 	}
 #endif /* NOT __MINGW32__ */
@@ -2719,7 +2719,7 @@
 		if (close(p[0]) == -1 || close(p[1]) == -1)
 			fatal(_("close of pipe failed: %s"), strerror(errno));
 		set_sigpipe_to_default();
-		execl("/bin/sh", "sh", "-c", cmd, NULL);
+		execl("@TERMUX_PREFIX@/bin/sh", "sh", "-c", cmd, NULL);
 		_exit(errno == ENOENT ? 127 : 126);
 	}
 #endif /* NOT __MINGW32__ */
