--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -98,7 +98,6 @@ find_package(assimp CONFIG REQUIRED)
 find_package(cpptrace CONFIG REQUIRED)
 find_package(freeimage CONFIG REQUIRED)
 find_package(freetype CONFIG REQUIRED)
-find_package(Catch2 CONFIG REQUIRED)
 find_package(fmt CONFIG REQUIRED)
 find_package(miniz CONFIG REQUIRED)
 find_package(tinyxml2 CONFIG REQUIRED)
--- a/lib/KdLib/CMakeLists.txt
+++ b/lib/KdLib/CMakeLists.txt
@@ -39,4 +39,3 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
   )
 endif()
 
-add_subdirectory(test)
--- a/lib/TbBaseLib/CMakeLists.txt
+++ b/lib/TbBaseLib/CMakeLists.txt
@@ -32,5 +32,3 @@ target_link_libraries(TbBaseLib
     VmLib
 )
 
-add_subdirectory(test)
-add_subdirectory(test-utils)
--- a/lib/TbBaseLib/test-utils/CMakeLists.txt
+++ b/lib/TbBaseLib/test-utils/CMakeLists.txt
@@ -21,4 +21,3 @@ target_link_libraries(TbBaseTestUtilsLib
     TbBaseLib
 )
 
-add_subdirectory(test)
--- a/lib/TbElLib/CMakeLists.txt
+++ b/lib/TbElLib/CMakeLists.txt
@@ -26,5 +26,3 @@ target_link_libraries(TbElLib
     fmt::fmt-header-only
 )
 
-add_subdirectory(test)
-add_subdirectory(test-utils)
--- a/lib/TbFsLib/CMakeLists.txt
+++ b/lib/TbFsLib/CMakeLists.txt
@@ -33,5 +33,3 @@ target_link_libraries(TbFsLib
     fmt::fmt-header-only
 )
 
-add_subdirectory(test)
-add_subdirectory(test-utils)
--- a/lib/TbFsLib/test-utils/CMakeLists.txt
+++ b/lib/TbFsLib/test-utils/CMakeLists.txt
@@ -20,4 +20,3 @@ target_link_libraries(TbFsTestUtilsLib
     VmLib
 )
 
-add_subdirectory(test)
--- a/lib/TbGlLib/CMakeLists.txt
+++ b/lib/TbGlLib/CMakeLists.txt
@@ -62,5 +62,3 @@ target_link_libraries(TbGlLib
     VmLib
 )
 
-add_subdirectory(test)
-add_subdirectory(test-utils)
--- a/lib/TbMdlLib/CMakeLists.txt
+++ b/lib/TbMdlLib/CMakeLists.txt
@@ -221,5 +221,3 @@ target_link_libraries(TbMdlLib
     TbGlLib
   )
 
-add_subdirectory(test)
-add_subdirectory(test-utils)
--- a/lib/TbMdlLib/test-utils/CMakeLists.txt
+++ b/lib/TbMdlLib/test-utils/CMakeLists.txt
@@ -24,4 +24,3 @@ target_link_libraries(TbMdlTestUtilsLib
     TbMdlLib
 )
 
-add_subdirectory(test)
--- a/lib/TbRenderLib/CMakeLists.txt
+++ b/lib/TbRenderLib/CMakeLists.txt
@@ -56,4 +56,3 @@ target_link_libraries(TbRenderLib
     VmLib
 )
 
-add_subdirectory(test)
--- a/lib/TbUiLib/CMakeLists.txt
+++ b/lib/TbUiLib/CMakeLists.txt
@@ -503,5 +503,3 @@ if(APPLE)
     target_compile_definitions(TbUiLib PRIVATE GL_SILENCE_DEPRECATION)
 endif()
 
-add_subdirectory(test)
-add_subdirectory(test-utils)
--- a/lib/UpdateLib/CMakeLists.txt
+++ b/lib/UpdateLib/CMakeLists.txt
@@ -53,4 +53,3 @@ elseif (APPLE)
   set(LIB_UPDATE_SCRIPT "${CMAKE_CURRENT_SOURCE_DIR}/scripts/install_update.sh" CACHE INTERNAL "Path to the update script")
 endif()
 
-add_subdirectory(test)
--- a/lib/VmLib/CMakeLists.txt
+++ b/lib/VmLib/CMakeLists.txt
@@ -19,4 +19,3 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
   )
 endif()
 
-add_subdirectory(test)
