diff -uNr cgal-releases-CGAL-4.14.1/Installation/cmake/modules/CGAL_SetupBoost.cmake cgal-releases-CGAL-4.14.1.mod/Installation/cmake/modules/CGAL_SetupBoost.cmake
--- cgal-releases-CGAL-4.14.1/Installation/cmake/modules/CGAL_SetupBoost.cmake	2019-09-30 10:18:05.000000000 +0300
+++ cgal-releases-CGAL-4.14.1.mod/Installation/cmake/modules/CGAL_SetupBoost.cmake	2019-10-05 15:13:45.119729964 +0300
@@ -27,16 +27,6 @@
     set ( CGAL_requires_Boost_libs TRUE )
     if ( DEFINED  MSVC_VERSION AND "${MSVC_VERSION}" GREATER 1800)
       set ( CGAL_requires_Boost_libs FALSE )
-    else()
-      try_run( CGAL_test_cpp_version_RUN_RES CGAL_test_cpp_version_COMPILE_RES
-        "${CMAKE_BINARY_DIR}"
-        "${CGAL_MODULES_DIR}/config/support/CGAL_test_cpp_version.cpp"
-        RUN_OUTPUT_VARIABLE CGAL_cplusplus)
-      message(STATUS "__cplusplus is ${CGAL_cplusplus}")
-      if(NOT CGAL_test_cpp_version_RUN_RES)
-        set ( CGAL_requires_Boost_libs FALSE )
-        message(STATUS "  --> Do not link with Boost.Thread")
-      endif()
     endif()
   endif()
   set_property(GLOBAL PROPERTY CGAL_requires_Boost_Thread ${CGAL_requires_Boost_libs})
