diff -u -r ../findutils-4.10.0/configure ./configure
--- ../findutils-4.10.0/configure	2024-06-01 16:18:04.000000000 +0000
+++ ./configure	2024-09-18 22:09:54.499067347 +0000
@@ -24336,7 +24336,7 @@
           else
             if case "$gl_cv_have_weak" in *yes) true;; *) false;; esac; then
 
-printf "%s\n" "#define USE_POSIX_THREADS_WEAK 1" >>confdefs.h
+# printf "%s\n" "#define USE_POSIX_THREADS_WEAK 1" >>confdefs.h
 
               LIBTHREAD= LTLIBTHREAD=
             else
