diff -u -r ../findutils-4.4.2/configure ./configure
--- ../findutils-4.4.2/configure	2009-05-16 17:28:18.000000000 +0200
+++ ./configure	2014-02-12 18:47:49.000000000 +0100
@@ -21410,7 +21410,7 @@
 
 
 # Determine how to get the list of mounted file systems.
-ac_list_mounted_fs=
+ac_list_mounted_fs=found
 
 # If the getmntent function is available but not in the standard library,
 # make sure LIBS contains the appropriate -l option.
