diff -uNr findutils-4.6.0/configure findutils-4.6.0.mod/configure
--- findutils-4.6.0/configure	2015-12-28 23:31:23.000000000 +0200
+++ findutils-4.6.0.mod/configure	2019-08-16 17:21:04.358640627 +0300
@@ -15945,7 +15945,7 @@
           if test -n "$LIBMULTITHREAD" || test -n "$LTLIBMULTITHREAD"; then
             if case "$gl_cv_have_weak" in *yes) true;; *) false;; esac; then
 
-$as_echo "#define USE_POSIX_THREADS_WEAK 1" >>confdefs.h
+#$as_echo "#define USE_POSIX_THREADS_WEAK 1" >>confdefs.h
 
               LIBTHREAD=
               LTLIBTHREAD=
@@ -31716,7 +31716,7 @@
 
 
 # Determine how to get the list of mounted file systems.
-ac_list_mounted_fs=
+ac_list_mounted_fs=found
 
 # If the getmntent function is available but not in the standard library,
 # make sure LIBS contains the appropriate -l option.
