diff -u -r ../findutils-4.7.0/find/parser.c ./find/parser.c
--- ../findutils-4.7.0/find/parser.c	2019-08-29 19:10:57.000000000 +0000
+++ ./find/parser.c	2019-09-08 08:27:46.805564375 +0000
@@ -69,10 +69,10 @@
 
 
 #ifndef HAVE_ENDGRENT
-# define endgrent ()
+# define endgrent()
 #endif
 #ifndef HAVE_ENDPWENT
-# define endpwent ()
+# define endpwent()
 #endif
 
 static bool parse_accesscheck   (const struct parser_table*, char *argv[], int *arg_ptr);
