diff -u -r ../findutils-4.5.14/find/parser.c ./find/parser.c
--- ../findutils-4.5.14/find/parser.c	2014-07-19 12:12:59.000000000 +0200
+++ ./find/parser.c	2014-08-11 22:58:04.839140727 +0200
@@ -75,10 +75,10 @@
 #endif
 
 #ifndef HAVE_ENDGRENT
-#define endgrent ()
+#define endgrent()
 #endif
 #ifndef HAVE_ENDPWENT
-#define endpwent ()
+#define endpwent()
 #endif
 
 static bool parse_accesscheck   (const struct parser_table*, char *argv[], int *arg_ptr);
