--- a/libs/tk/ytk/config.h
+++ b/libs/tk/ytk/config.h
@@ -28,7 +28,7 @@
 #define HAVE_GETRESUID 1
 #endif
 
-#if !defined(__APPLE__) && !defined(__NetBSD__)
+#if !defined(__APPLE__) && !defined(__NetBSD__) && !defined(__ANDROID__)
 /* Have GNU ftw */
 #define HAVE_GNU_FTW 1
 #endif
diff --git a/libs/tk/ydk/config.h b/libs/tk/ydk/config.h
index 484758b..8b8bc15 100644
--- a/libs/tk/ydk/config.h
+++ b/libs/tk/ydk/config.h
@@ -27,7 +27,7 @@
 #define HAVE_GETRESUID 1
 #endif
 
-#if !defined(__APPLE__) && !defined(__NetBSD__)
+#if !defined(__APPLE__) && !defined(__NetBSD__) && !defined(__ANDROID___)
 /* Have GNU ftw */
 #define HAVE_GNU_FTW 1
 #endif

