--- a/filter/filter_compare.c
+++ b/filter/filter_compare.c
@@ -182,7 +182,7 @@
 
 			if (strlen(results_name) == 0) {
 				// Ponemos el nombre del fichero al original con extension dat
-				strlcpy(results_name, "/tmp/compare.dat", sizeof(results_name));
+				strlcpy(results_name, "@TERMUX_PREFIX@/tmp/compare.dat", sizeof(results_name));
 
 			}
 			if (!(compare[instance]->results = fopen(results_name, "w")))
