--- a/filter/parse_csv.awk.in
+++ b/filter/parse_csv.awk.in
@@ -3,7 +3,7 @@
 #  "strength", "Blending factor.", "%f", "0.9", "0.0", "2.0"
 #  "range", "Apply the filter only to a range", "%u-%u", "0-0", "0", "-1", "0", "-1"
 #  "pre", "Run as a PRE filter", "", "0"
-#  "log", "Logfile for verbose info", "%s", "/tmp/logfile"
+#  "log", "Logfile for verbose info", "%s", "@TERMUX_PREFIX@/tmp/logfile"
 
 
 # "filter_logoaway.so", "remove an image from the video", "v0.1.0 (2002-12-04)", "Thomas Wehrspann", "VRY", "1"
