--- ./Makefile.in	1995-08-17 00:54:29.000000000 +0000
+++ ./Makefile.in.fixed	2018-02-09 02:16:31.746398040 +0000
@@ -46,7 +46,7 @@
 # so compilers besides gcc can find it by default.
 # If it is empty or not defined, termcap.h will only be installed in
 # includedir. 
-oldincludedir = /usr/include
+#oldincludedir = /usr/include
 
 # Directory in which to install the documentation info files.
 infodir = $(prefix)/info
