diff -u -r ../weechat-2.4/cmake/FindRuby.cmake ./cmake/FindRuby.cmake
--- ../weechat-2.4/cmake/FindRuby.cmake	2019-02-17 07:20:07.000000000 +0000
+++ ./cmake/FindRuby.cmake	2019-02-24 18:49:43.397998264 +0000
@@ -33,7 +33,7 @@
 
 find_package(PkgConfig)
 if(PKG_CONFIG_FOUND)
-  pkg_search_module(RUBY ruby-2.5 ruby-2.4 ruby-2.3 ruby-2.2 ruby-2.1 ruby-2.0 ruby-1.9 ruby-1.8)
+  pkg_search_module(RUBY ruby-2.6 ruby-2.5 ruby-2.4 ruby-2.3 ruby-2.2 ruby-2.1 ruby-2.0 ruby-1.9 ruby-1.8)
 endif()
 
 if(RUBY_FOUND)
