diff -u -r ../weechat-1.3/cmake/FindRuby.cmake ./cmake/FindRuby.cmake
--- ../weechat-1.3/cmake/FindRuby.cmake	2015-08-16 02:27:07.000000000 -0400
+++ ./cmake/FindRuby.cmake	2016-01-03 07:50:12.392704575 -0500
@@ -33,7 +33,7 @@
 
 find_package(PkgConfig)
 if(PKG_CONFIG_FOUND)
-  pkg_search_module(RUBY ruby-2.2 ruby-2.1 ruby-2.0 ruby-1.9 ruby-1.8)
+  pkg_search_module(RUBY ruby-2.3 ruby-2.2 ruby-2.1 ruby-2.0 ruby-1.9 ruby-1.8)
 endif()
 
 if(RUBY_FOUND)
