diff -u -r ../weechat-1.7/cmake/FindRuby.cmake ./cmake/FindRuby.cmake
--- ../weechat-1.7/cmake/FindRuby.cmake	2017-01-15 07:41:25.000000000 +0100
+++ ./cmake/FindRuby.cmake	2017-01-17 22:41:59.681388711 +0100
@@ -33,7 +33,7 @@
 
 find_package(PkgConfig)
 if(PKG_CONFIG_FOUND)
-  pkg_search_module(RUBY ruby-2.3 ruby-2.2 ruby-2.1 ruby-2.0 ruby-1.9 ruby-1.8)
+  pkg_search_module(RUBY ruby-2.4 ruby-2.3 ruby-2.2 ruby-2.1 ruby-2.0 ruby-1.9 ruby-1.8)
 endif()
 
 if(RUBY_FOUND)
