diff -u -r ../weechat-1.1.1/src/gui/curses/CMakeLists.txt ./src/gui/curses/CMakeLists.txt
--- ../weechat-1.1.1/src/gui/curses/CMakeLists.txt	2015-01-25 02:40:23.000000000 -0500
+++ ./src/gui/curses/CMakeLists.txt	2015-05-05 14:18:28.743723739 -0400
@@ -59,8 +59,6 @@
   list(APPEND EXTRA_LIBS "socket" "nsl")
 endif()
 
-list(APPEND EXTRA_LIBS "pthread")
-
 if(ICONV_LIBRARY)
   list(APPEND EXTRA_LIBS ${ICONV_LIBRARY})
 endif()
