Enable mouse support by default.

--- a/src/core/core-config.c
+++ b/src/core/core-config.c
@@ -2993,7 +2993,7 @@
             weechat_config_file, weechat_config_section_look,
             "mouse", "boolean",
             N_("enable mouse support"),
-            NULL, 0, 0, "off", NULL, 0,
+            NULL, 0, 0, "on", NULL, 0,
             NULL, NULL, NULL,
             &config_change_mouse, NULL, NULL,
             NULL, NULL, NULL);
