diff -u -r ../weechat-2.7/src/gui/curses/CMakeLists.txt ./src/gui/curses/CMakeLists.txt
--- ../weechat-2.7/src/gui/curses/CMakeLists.txt	2019-12-08 08:48:53.000000000 +0000
+++ ./src/gui/curses/CMakeLists.txt	2019-12-11 23:30:56.807370000 +0000
@@ -31,7 +31,7 @@
   list(APPEND EXTRA_LIBS "network")
 endif()
 
-if(NOT ${CMAKE_SYSTEM_NAME} STREQUAL "Haiku")
+if(NOT ${CMAKE_SYSTEM_NAME} STREQUAL "Haiku" AND NOT ${CMAKE_SYSTEM_NAME} STREQUAL "Android")
   list(APPEND EXTRA_LIBS "pthread")
 endif()
 
