diff -u -r ../TermuxAm-0.1/am-libexec-packaged ./am-libexec-packaged
--- ../TermuxAm-0.1/am-libexec-packaged	2018-02-03 16:43:05.000000000 +0100
+++ ./am-libexec-packaged	2018-02-05 01:11:01.744982848 +0100
@@ -1,4 +1,4 @@
 #!/data/data/com.termux/files/usr/bin/bash
 export CLASSPATH=/data/data/com.termux/files/usr/libexec/termux-am/am.apk
-unset LD_LIBRARY_PATH
+unset LD_LIBRARY_PATH LD_PRELOAD
 exec /system/bin/app_process / com.example.termuxam.Am "$@"
