--- a/makepanda/installpanda.py
+++ b/makepanda/installpanda.py
@@ -175,0 +176,3 @@
+        python_lib = prefix+"/lib/python"+python_version["version"]+"/site-packages"
+        python_version["purelib"] = python_lib
+        python_version["platlib"] = python_lib
@@ -179,6 +181,0 @@
-    if (sys.platform.startswith("freebsd")):
-        MakeDirectory(destdir+prefix+"/etc", mode=0o755)
-        MakeDirectory(destdir+"/usr/local/libdata/ldconfig", mode=0o755, recursive=True)
-    else:
-        MakeDirectory(destdir+"/etc/ld.so.conf.d", mode=0o755, recursive=True)
-
@@ -185,0 +183 @@
+    MakeDirectory(destdir+prefix+"/etc", mode=0o755, recursive=True)
@@ -188 +186 @@
-    if (sys.platform.startswith("freebsd")):
+    if True:
@@ -191,3 +188,0 @@
-    else:
-        WriteFile(destdir+"/etc/Config.prc", Configrc)
-        oscmd("cp "+outputdir+"/etc/Confauto.prc "+destdir+"/etc/Confauto.prc")
@@ -225,5 +219,0 @@
-
-    if (sys.platform.startswith("freebsd")):
-        oscmd("echo '"+libdir+"/panda3d'>    "+destdir+"/usr/local/libdata/ldconfig/panda3d")
-    else:
-        oscmd("echo '"+libdir+"/panda3d'>    "+destdir+"/etc/ld.so.conf.d/panda3d.conf")
