diff --git a/makepanda/makepanda.py b/makepanda/makepanda.py
index ad6230f0ed..031842b9e0 100755
--- a/makepanda/makepanda.py
+++ b/makepanda/makepanda.py
@@ -5546,7 +5546,7 @@ if (not RTDIST and not RUNTIME and PkgSkip("PVIEW")==0):
 # DIRECTORY: panda/src/android/
 #
 
-if (not RUNTIME and GetTarget() == 'android'):
+if (not RUNTIME and GetTarget() == 'android' and False):
   OPTS=['DIR:panda/src/android']
   TargetAdd('org/panda3d/android/NativeIStream.class', opts=OPTS, input='NativeIStream.java')
   TargetAdd('org/panda3d/android/NativeOStream.class', opts=OPTS, input='NativeOStream.java')
