--- a/makepanda/makepandacore.py
+++ b/makepanda/makepandacore.py
@@ -497,27 +497,36 @@ def GetCXX():
     else:
         return os.environ.get('CXX', TOOLCHAIN_PREFIX + 'g++')
 
-def GetStrip():
+def _GetStrip():
     # Hack
     if TARGET == 'android':
         return TOOLCHAIN_PREFIX + 'strip'
     else:
         return 'strip'
 
-def GetAR():
+def GetStrip():
+    return os.environ.get('STRIP', _GetStrip())
+
+def _GetAR():
     # Hack
     if TARGET == 'android':
         return TOOLCHAIN_PREFIX + 'ar'
     else:
         return 'ar'
 
-def GetRanlib():
+def GetAR():
+    return os.environ.get('AR', _GetAR())
+
+def _GetRanlib():
     # Hack
     if TARGET == 'android':
         return TOOLCHAIN_PREFIX + 'ranlib'
     else:
         return 'ranlib'
 
+def GetRanlib():
+    return os.environ.get('RANLIB', _GetRanlib())
+
 BISON = None
 def GetBison():
     global BISON
@@ -3086,12 +3098,6 @@ def SetupBuildEnvironment(compiler):
 
         # Now extract the preprocessor's include directories.
         cmd = GetCXX() + " -x c++ -v -E " + os.devnull
-        if "ANDROID_NDK" in SDK:
-            ndk_dir = SDK["ANDROID_NDK"].replace('\\', '/')
-            cmd += ' -isystem %s/sysroot/usr/include' % (ndk_dir)
-            cmd += ' -isystem %s/sysroot/usr/include/%s' % (ndk_dir, SDK["ANDROID_TRIPLE"])
-        else:
-            cmd += sysroot_flag
 
         null = open(os.devnull, 'w')
         handle = subprocess.Popen(cmd, stdout=null, stderr=subprocess.PIPE, shell=True)
