diff -uNr redis-6.0.1/deps/lua/src/luaconf.h redis-6.0.1.mod/deps/lua/src/luaconf.h
--- redis-6.0.1/deps/lua/src/luaconf.h	2020-05-02 01:10:20.000000000 +0300
+++ redis-6.0.1.mod/deps/lua/src/luaconf.h	2020-05-13 12:58:07.276076703 +0300
@@ -94,7 +94,7 @@
 	".\\?.dll;"  LUA_CDIR"?.dll;" LUA_CDIR"loadall.dll"
 
 #else
-#define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT	"@TERMUX_PREFIX@"
 #define LUA_LDIR	LUA_ROOT "share/lua/5.1/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/5.1/"
 #define LUA_PATH_DEFAULT  \
@@ -645,9 +645,9 @@
 
 #if defined(LUA_USE_MKSTEMP)
 #include <unistd.h>
-#define LUA_TMPNAMBUFSIZE	32
+#define LUA_TMPNAMBUFSIZE	256
 #define lua_tmpnam(b,e)	{ \
-	strcpy(b, "/tmp/lua_XXXXXX"); \
+	strcpy(b, "@TERMUX_PREFIX@/tmp/lua_XXXXXX"); \
 	e = mkstemp(b); \
 	if (e != -1) close(e); \
 	e = (e == -1); }
