--- ./src/setproctitle.c.orig	2021-05-01 14:35:04.385944179 +0000
+++ ./src/setproctitle.c	2021-05-01 14:36:32.979189534 +0000
@@ -190,7 +190,7 @@ static int spt_copyargs(int argc, char *argv[]) {
  */
 void spt_init(int argc, char *argv[]) {
         char **envp = environ;
-	char *base, *end, *nul, *tmp;
+	char *base, *end, *nul;
 	int i, error, envc;
 
 	if (!(base = argv[0]))
@@ -232,7 +232,7 @@ void spt_init(int argc, char *argv[]) {
 	if (!(SPT.arg0 = strdup(argv[0])))
 		goto syerr;
 
-#if defined __linux__
+#if defined __linux__ && !defined __ANDROID__
 	if (!(tmp = strdup(program_invocation_name)))
 		goto syerr;
 
