--- ../fuse_common.h.orig	2018-10-13 19:54:21.093146271 +0200
+++ ./include/fuse_common.h	2018-10-13 19:57:05.539651897 +0200
@@ -759,12 +760,12 @@
  *
  * On 32bit systems please add -D_FILE_OFFSET_BITS=64 to your compile flags!
  */
-
+/* This probably introduces problem for arm and i686
 #if defined(__GNUC__) && (__GNUC__ > 4 || __GNUC__ == 4 && __GNUC_MINOR__ >= 6) && !defined __cplusplus
 _Static_assert(sizeof(off_t) == 8, "fuse: off_t must be 64bit");
 #else
 struct _fuse_off_t_must_be_64bit_dummy_struct \
 	{ unsigned _fuse_off_t_must_be_64bit:((sizeof(off_t) == 8) ? 1 : -1); };
 #endif
-
+*/
 #endif /* FUSE_COMMON_H_ */
