--- ../fuse_loop_mt.c_2.9.8_orig	2018-10-13 22:18:23.067614196 +0200
+++ ./lib/fuse_loop_mt.c	2018-10-13 22:16:17.256721029 +0200
@@ -77,9 +77,7 @@
 		};
 		int res;
 
-		pthread_setcancelstate(PTHREAD_CANCEL_ENABLE, NULL);
 		res = fuse_session_receive_buf(mt->se, &fbuf, &ch);
-		pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, NULL);
 		if (res == -EINTR)
 			continue;
 		if (res <= 0) {
@@ -243,7 +241,7 @@
 
 		pthread_mutex_lock(&mt.lock);
 		for (w = mt.main.next; w != &mt.main; w = w->next)
-			pthread_cancel(w->thread_id);
+			pthread_kill(w->thread_id, 0);
 		mt.exit = 1;
 		pthread_mutex_unlock(&mt.lock);
 
