--- ../lib-meson.build.orig	2018-10-13 18:50:35.614173402 +0200
+++ ./lib/meson.build	2018-10-13 19:31:58.316854967 +0200
@@ -3,11 +3,8 @@
                    'fuse_signals.c', 'buffer.c', 'cuse_lowlevel.c',
 	           'helper.c', 'modules/subdir.c' ]
 
-if host_machine.system().startswith('linux')
-   libfuse_sources += [ 'mount.c', 'mount_util.c' ]
-else
-   libfuse_sources += [ 'mount_bsd.c' ]
-endif
+libfuse_sources += [ 'mount.c', 'mount_util.c' ]
+
 
 if cfg.get('HAVE_ICONV')
    libfuse_sources += [ 'modules/iconv.c' ]
@@ -22,9 +19,6 @@
 if host_machine.system().startswith('netbsd')
    deps += [ cc.find_library('perfuse'),
              cc.find_library('puffs') ]
-else
-   # Required for clock_gettime before glibc 2.17
-   deps += cc.find_library('rt')
 endif
 
 fusermount_path = join_paths(get_option('prefix'), get_option('bindir'))
