--- a/capplets/system-info/mate-system-info.c
+++ b/capplets/system-info/mate-system-info.c
@@ -547,7 +547,7 @@ get_mate_desktop_version ()
     int   status;
     int   i = 0;
     char *version = NULL;
-    char *argv[] = {"/usr/bin/mate-about", "-v", NULL};
+    char *argv[] = {"@TERMUX_PREFIX@/bin/mate-about", "-v", NULL};
     g_autoptr(GError) error = NULL;
 
     g_debug ("About to launch '%s'", argv[0]);
