diff -u -r ../alpine-2.20/imap/src/osdep/unix/os_slx.c ./imap/src/osdep/unix/os_slx.c
--- ../alpine-2.20/imap/src/osdep/unix/os_slx.c	2015-01-12 00:12:25.504178437 -0500
+++ ./imap/src/osdep/unix/os_slx.c	2016-04-14 17:13:21.985871059 -0400
@@ -35,9 +35,7 @@
 #include <netdb.h>
 #include <ctype.h>
 #include <errno.h>
-extern int errno;		/* just in case */
 #include <pwd.h>
-#include <shadow.h>
 #include "misc.h"
 
 
