diff -u -r ../alpine-2.20/imap/src/osdep/unix/Makefile ./imap/src/osdep/unix/Makefile
--- ../alpine-2.20/imap/src/osdep/unix/Makefile	2015-01-12 00:12:25.505178442 -0500
+++ ./imap/src/osdep/unix/Makefile	2016-04-14 17:06:00.090973009 -0400
@@ -132,10 +132,7 @@
 
 # Commands possibly overriden by the individual port
 
-ARRC=ar rc
-CC=cc
 LN=ln -s
-RANLIB=ranlib
 
 
 # Standard distribution build parameters
