--- ./src/CpuAffinitySet.cc	2016-10-10 01:28:01.000000000 +0530
+++ ../CpuAffinitySet.cc	2016-12-07 22:53:47.745146503 +0530
@@ -37,7 +37,7 @@
     } else {
         cpu_set_t cpuSet;
         memcpy(&cpuSet, &theCpuSet, sizeof(cpuSet));
-        (void) CPU_AND(&cpuSet, &cpuSet, &theOrigCpuSet);
+        CPU_AND(&cpuSet, &cpuSet, &theOrigCpuSet);
         if (CPU_COUNT(&cpuSet) <= 0) {
             debugs(54, DBG_IMPORTANT, "ERROR: invalid CPU affinity for process "
                    "PID " << getpid() << ", may be caused by an invalid core in "
