commit 55151c545a8e0bd2cb69036da5794c9cb21018b2
Author: Amos Jeffries <yadij@users.noreply.github.com>
Date:   2022-08-17 23:32:43 +0000

    Fix typo in manager ACL (#1113)

diff --git a/src/cf.data.pre b/src/cf.data.pre
index a0bdb2f83..118256437 100644
--- a/src/cf.data.pre
+++ b/src/cf.data.pre
@@ -1036,7 +1036,7 @@ DEFAULT: ssl::certUntrusted ssl_error X509_V_ERR_INVALID_CA X509_V_ERR_SELF_SIGN
 DEFAULT: ssl::certSelfSigned ssl_error X509_V_ERR_DEPTH_ZERO_SELF_SIGNED_CERT
 ENDIF
 DEFAULT: all src all
-DEFAULT: manager url_regex -i ^cache_object:// +i ^https?://[^/]+/squid-internal-mgr/
+DEFAULT: manager url_regex -i ^cache_object:// +i ^[^:]+://[^/]+/squid-internal-mgr/
 DEFAULT: localhost src 127.0.0.1/32 ::1
 DEFAULT: to_localhost dst 127.0.0.0/8 0.0.0.0/32 ::1/128 ::/128
 DEFAULT: CONNECT method CONNECT
