diff -u -r ../squid-4.2/src/ssl/support.cc ./src/ssl/support.cc
--- ../squid-4.2/src/ssl/support.cc	2018-08-05 14:13:41.000000000 +0000
+++ ./src/ssl/support.cc	2018-08-25 07:56:09.290337943 +0000
@@ -472,7 +472,7 @@
     SSL_load_error_strings();
     SSLeay_add_ssl_algorithms();
 
-#if HAVE_OPENSSL_ENGINE_H
+#if HAVE_OPENSSL_ENGINE_H && !defined(OPENSSL_NO_ENGINE)
     if (::Config.SSL.ssl_engine) {
         ENGINE_load_builtin_engines();
         ENGINE *e;
