--- ./argp-fmtstream.h.orig	2003-12-11 09:37:05.000000000 +0100
+++ ./argp-fmtstream.h	2011-08-12 11:56:43.000000000 +0200
@@ -192,7 +192,7 @@
 extern int _argp_fmtstream_ensure (argp_fmtstream_t __fs, size_t __amount);
 extern int __argp_fmtstream_ensure (argp_fmtstream_t __fs, size_t __amount);
 
-#ifdef __OPTIMIZE__
+#if defined(__OPTIMIZE__) && !defined(__clang__)
 /* Inline versions of above routines.  */
 
 #if !_LIBC
