--- a/meson.build       2025-05-20 22:37:14.000000000 +0530
+++ b/meson.build     2026-01-17 20:18:54.340352597 +0530
@@ -110,7 +110,7 @@
 gio_dep = dependency('gio-2.0')
 gio_unix_dep = dependency('gio-unix-2.0')
 json_glib_dep = dependency('json-glib-1.0')
-fuse3_dep = dependency('fuse3', version: '>= 3.10.0')
+fuse3_dep = dependency('fuse3', version: '>= 3.10.0', required: false)
 gdk_pixbuf_dep = dependency('gdk-pixbuf-2.0')
 gst_pbutils_dep = dependency('gstreamer-pbutils-1.0')
 geoclue_dep = dependency(
@@ -215,7 +215,6 @@

 subdir('data')
 subdir('src')
-subdir('document-portal')
 subdir('po')
 subdir('doc')
