diff -uNr redis-6.0.9/src/server.c redis-6.0.9.mod/src/server.c
--- redis-6.0.9/src/server.c	2020-10-27 07:12:01.000000000 +0000
+++ redis-6.0.9.mod/src/server.c	2020-11-22 17:03:52.033053867 +0000
@@ -2813,8 +2813,10 @@
  * can work reliably (default cancelability type is PTHREAD_CANCEL_DEFERRED).
  * Needed for pthread_cancel used by the fast memory test used by the crash report. */
 void makeThreadKillable(void) {
+#ifndef __ANDROID__
     pthread_setcancelstate(PTHREAD_CANCEL_ENABLE, NULL);
     pthread_setcanceltype(PTHREAD_CANCEL_ASYNCHRONOUS, NULL);
+#endif
 }
 
 void initServer(void) {
