--- ./src/Makefile.orig	2021-05-14 11:06:16.748969206 +0000
+++ ./src/Makefile	2021-05-14 11:07:41.892446036 +0000
@@ -45,7 +45,7 @@
 endif
 
 PREFIX?=/usr/local
-INSTALL_BIN=$(PREFIX)/bin
+INSTALL_BIN=$(DESTDIR)$(PREFIX)/bin
 INSTALL=install
 PKG_CONFIG?=pkg-config
 
@@ -177,7 +177,7 @@
 else
 	# All the other OSes (notably Linux)
 	FINAL_LDFLAGS+= -rdynamic
-	FINAL_LIBS+=-ldl -pthread -lrt
+	FINAL_LIBS+=-ldl
 endif
 endif
 endif
