--- a/src/common/error.cpp
+++ b/src/common/error.cpp
@@ -30,7 +30,7 @@
     return ret;
 #else
     char err_str[255];
-#if defined(ANDROID) ||                                                                            \
+#if defined(ANDROID) || defined(__ANDROID__) ||                                                   \
     (defined(__GLIBC__) && (_GNU_SOURCE || (_POSIX_C_SOURCE < 200112L && _XOPEN_SOURCE < 600)))
     // Thread safe (GNU-specific)
     const char* str = strerror_r(e, err_str, sizeof(err_str));
