--- ../cache/node-v8.2.1/deps/uv/uv.gyp	2017-07-21 07:28:21.000000000 +1000
+++ ./deps/uv/uv.gyp	2017-08-01 09:44:01.147497665 +1000
@@ -45,12 +45,12 @@
     {
       'target_name': 'libuv',
       'type': '<(uv_library)',
-      'include_dirs': [
+      'include_dirs+': [
         'include',
         'src/',
       ],
       'direct_dependent_settings': {
-        'include_dirs': [ 'include' ],
+        'include_dirs+': [ 'include' ],
         'conditions': [
           ['OS != "win"', {
             'defines': [
