diff -u -r ../node-v5.10.0/lib/os.js ./lib/os.js
--- ../node-v5.10.0/lib/os.js	2016-03-31 21:52:17.000000000 -0400
+++ ./lib/os.js	2016-04-04 04:46:05.148105544 -0400
@@ -36,7 +36,7 @@
     path = process.env.TMPDIR ||
            process.env.TMP ||
            process.env.TEMP ||
-           '/tmp';
+           '@TERMUX_PREFIX@/tmp';
     if (path.length > 1 && path.endsWith('/'))
       path = path.slice(0, -1);
   }
