diff -u -r ../node-v7.6.0/node.gyp ./node.gyp
--- ../node-v7.6.0/node.gyp	2017-02-21 23:53:43.000000000 +0100
+++ ./node.gyp	2017-02-24 22:34:31.607945917 +0100
@@ -874,82 +874,6 @@
           ],
         } ],
       ]
-    },
-    {
-      'target_name': 'cctest',
-      'type': 'executable',
-      'dependencies': [ 'deps/gtest/gtest.gyp:gtest' ],
-      'include_dirs': [
-        'src',
-        'deps/v8/include',
-        '<(SHARED_INTERMEDIATE_DIR)'
-      ],
-      'defines': [
-        # gtest's ASSERT macros conflict with our own.
-        'GTEST_DONT_DEFINE_ASSERT_EQ=1',
-        'GTEST_DONT_DEFINE_ASSERT_GE=1',
-        'GTEST_DONT_DEFINE_ASSERT_GT=1',
-        'GTEST_DONT_DEFINE_ASSERT_LE=1',
-        'GTEST_DONT_DEFINE_ASSERT_LT=1',
-        'GTEST_DONT_DEFINE_ASSERT_NE=1',
-        'NODE_WANT_INTERNALS=1',
-      ],
-      'sources': [
-        'test/cctest/util.cc',
-      ],
-
-      'conditions': [
-        ['v8_inspector=="true"', {
-          'defines': [
-            'HAVE_INSPECTOR=1',
-          ],
-          'dependencies': [
-            'v8_inspector_compress_protocol_json#host'
-          ],
-          'include_dirs': [
-            '<(SHARED_INTERMEDIATE_DIR)'
-          ],
-          'sources': [
-            'src/inspector_socket.cc',
-            'src/inspector_socket_server.cc',
-            'test/cctest/test_inspector_socket.cc',
-            'test/cctest/test_inspector_socket_server.cc'
-          ],
-          'conditions': [
-            [ 'node_shared_zlib=="false"', {
-              'dependencies': [
-                'deps/zlib/zlib.gyp:zlib',
-              ]
-            }],
-            [ 'node_shared_openssl=="false"', {
-              'dependencies': [
-                'deps/openssl/openssl.gyp:openssl'
-              ]
-            }],
-            [ 'node_shared_http_parser=="false"', {
-              'dependencies': [
-                'deps/http_parser/http_parser.gyp:http_parser'
-              ]
-            }],
-            [ 'node_shared_libuv=="false"', {
-              'dependencies': [
-                'deps/uv/uv.gyp:libuv'
-              ]
-            }]
-          ]
-        }],
-        [ 'node_use_v8_platform=="true"', {
-          'dependencies': [
-            'deps/v8/src/v8.gyp:v8_libplatform',
-          ],
-        }],
-        [ 'node_use_bundled_v8=="true"', {
-          'dependencies': [
-            'deps/v8/src/v8.gyp:v8',
-            'deps/v8/src/v8.gyp:v8_libplatform'
-          ],
-        }],
-      ]
     }
   ], # end targets
 
