--- ./lib/internal/module.js    2017-07-30 10:32:26.000000000 -0500
+++ ./node-v8.2.1/lib/internal/module.js        2017-07-30 10:32:50.000000000 -0500
@@ -80,7 +80,8 @@
   'assert', 'async_hooks', 'buffer', 'child_process', 'cluster', 'crypto',
   'dgram', 'dns', 'domain', 'events', 'fs', 'http', 'https', 'net', 'os',
   'path', 'punycode', 'querystring', 'readline', 'repl', 'stream',
-  'string_decoder', 'tls', 'tty', 'url', 'util', 'v8', 'vm', 'zlib'
+  'string_decoder', 'tls', 'tty', 'url', 'util', 'v8', 'vm', 'zlib',
+  'termux'
 ];

 function addBuiltinLibsToObject(object) {
