diff -u -r ../privoxy-3.0.28-stable/configure.in ./configure.in
--- ../privoxy-3.0.28-stable/configure.in	2018-12-30 18:40:13.000000000 +0000
+++ ./configure.in	2019-01-02 23:35:39.728797247 +0000
@@ -1092,7 +1092,7 @@
   pcre_dyn=yes
   AC_DEFINE(FEATURE_DYNAMIC_PCRE,1,[Define to dynamically link to pcre.])
   STATIC_PCRE_ONLY=#
-  LIBS="$LIBS -lpcre -lpcreposix"
+  LIBS="$LIBS -lpcre"
 else
   AC_MSG_WARN([You are using the static PCRE code which is out of date and scheduled for removal, for details see:
     http://sourceforge.net/mailarchive/forum.php?thread_name=20080511195555.2dc6cfdc%40fabiankeil.de&forum_name=ijbswa-developers])
