diff --git a/GNUmakefile.in b/GNUmakefile.in
index 02dd4b0..9364895 100644
--- a/GNUmakefile.in
+++ b/GNUmakefile.in
@@ -851,7 +851,7 @@ install: CONF_DEST LOG_DEST PID_DEST check_doc GROUP_T
 	@test -n "$(STRIP)" &&\
 	$(ECHO) Installing $(PROGRAM) stripped executable to $(SBIN_DEST) ||\
 	$(ECHO) Installing $(PROGRAM) executable to $(DESTDIR)$(SBIN_DEST)
-	$(INSTALL) $(INSTALL_P) $(STRIP) $(PROGRAM) $(DESTDIR)$(SBIN_DEST)
+	$(INSTALL) $(INSTALL_P) $(PROGRAM) $(DESTDIR)$(SBIN_DEST)
 
 	@# Install the DOCS and man page. install-sh only does one file at a time.
 	@# FIXME: only handles jpegs.