--- ../../build/privoxy/cache/privoxy-3.0.24-stable/jcc.c	2016-01-16 13:33:36.000000000 +0100
+++ ./jcc.c	2016-08-20 08:48:12.247371024 +0200
@@ -3655,10 +3655,10 @@
       }
       if (NULL != grp)
       {
-         if (setgroups(1, &grp->gr_gid))
-         {
-            log_error(LOG_LEVEL_FATAL, "setgroups() failed: %E");
-         }
+         //if (setgroups(1, &grp->gr_gid))
+         //{
+         //   log_error(LOG_LEVEL_FATAL, "setgroups() failed: %E");
+         //}
       }
       else if (initgroups(pw->pw_name, pw->pw_gid))
       {
