diff -u -r ../privoxy-3.0.28-stable/project.h ./project.h
--- ../privoxy-3.0.28-stable/project.h	2018-12-30 18:40:13.000000000 +0000
+++ ./project.h	2019-01-03 00:34:59.829586492 +0000
@@ -82,11 +82,7 @@
 #ifdef STATIC_PCRE
 #  include "pcreposix.h"
 #else
-#  ifdef PCRE_H_IN_SUBDIR
-#    include <pcre/pcreposix.h>
-#  else
-#    include <pcreposix.h>
-#  endif
+#  include <regex.h>
 #endif
 
 #ifdef _WIN32
