diff --git a/components/sessionsprivate/sessionsmodel.cpp b/components/sessionsprivate/sessionsmodel.cpp
index 20bda2b..0da6080 100644
--- a/components/sessionsprivate/sessionsmodel.cpp
+++ b/components/sessionsprivate/sessionsmodel.cpp
@@ -14,8 +14,6 @@
 
 #include "kscreensaversettings.h"
 
-#include "screensaver_interface.h"
-
 SessionsModel::SessionsModel(QObject *parent)
     : QAbstractListModel(parent)
     , m_screensaverInterface(
