diff --git a/components/keyboardlayout/virtualkeyboard.h b/components/keyboardlayout/virtualkeyboard.h
index 4caab11..f0356ec 100644
--- a/components/keyboardlayout/virtualkeyboard.h
+++ b/components/keyboardlayout/virtualkeyboard.h
@@ -17,7 +17,6 @@ class KwinVirtualKeyboardInterface : public OrgKdeKwinVirtualKeyboardInterface
     QML_SINGLETON
 
     Q_PROPERTY(bool active READ active WRITE setActive NOTIFY activeChanged)
-    Q_PROPERTY(int mode READ mode WRITE setMode NOTIFY modeChanged)
     Q_PROPERTY(bool visible READ visible NOTIFY visibleChanged)
     Q_PROPERTY(bool available READ available NOTIFY availableChanged)
     Q_PROPERTY(bool activeClientSupportsTextInput READ activeClientSupportsTextInput NOTIFY activeClientSupportsTextInputChanged)
