diff -uNr groff-1.22.4/Makefile.in groff-1.22.4.mod/Makefile.in
--- groff-1.22.4/Makefile.in	2021-10-02 10:46:19.717453972 +0300
+++ groff-1.22.4.mod/Makefile.in	2021-10-02 16:50:23.231359302 +0300
@@ -3350,7 +3350,7 @@
 YACC = @YACC@
 abs_builddir = @abs_builddir@
 abs_srcdir = @abs_srcdir@
-abs_top_builddir = @abs_top_builddir@
+abs_top_builddir = /home/builder/.termux-build/groff/host-build
 abs_top_srcdir = @abs_top_srcdir@
 ac_ct_AWK = @ac_ct_AWK@
 ac_ct_CC = @ac_ct_CC@
