diff -uNr groff-1.22.4/Makefile.in groff-1.22.4.mod/Makefile.in
--- groff-1.22.4/Makefile.in	2018-12-23 17:34:27.000000000 +0300
+++ groff-1.22.4.mod/Makefile.in	2021-10-01 10:47:31.635428548 +0300
@@ -11526,13 +11526,13 @@
 # fill-column: 72
 # End:
 # vim: set filetype=automake textwidth=72:
-$(hdtblexample_DATA): $(HDTBLSTRIPFILES)
-@BUILD_EXAMPLES_TRUE@contrib/hdtbl/examples/test-hdtbl.sh: $(top_builddir)/config.status \
-@BUILD_EXAMPLES_TRUE@	$(HDTBLPROCESSEDEXAMPLEFILES) $(top_srcdir)/contrib/hdtbl/examples/test-hdtbl.sh.in
-@BUILD_EXAMPLES_TRUE@	sed -e "s|[@]abs_top_builddir[@]|$(abs_top_builddir)|g" \
-@BUILD_EXAMPLES_TRUE@	    -e "s|[@]GHOSTSCRIPT[@]|$(GHOSTSCRIPT)|g" \
-@BUILD_EXAMPLES_TRUE@	    	$(top_srcdir)/contrib/hdtbl/examples/test-hdtbl.sh.in > $@
-@BUILD_EXAMPLES_TRUE@	chmod +x $@
+#$(hdtblexample_DATA): $(HDTBLSTRIPFILES)
+#@BUILD_EXAMPLES_TRUE@contrib/hdtbl/examples/test-hdtbl.sh: $(top_builddir)/config.status \
+#@BUILD_EXAMPLES_TRUE@	$(HDTBLPROCESSEDEXAMPLEFILES) $(top_srcdir)/contrib/hdtbl/examples/test-hdtbl.sh.in
+#@BUILD_EXAMPLES_TRUE@	sed -e "s|[@]abs_top_builddir[@]|$(abs_top_builddir)|g" \
+#@BUILD_EXAMPLES_TRUE@	    -e "s|[@]GHOSTSCRIPT[@]|$(GHOSTSCRIPT)|g" \
+#@BUILD_EXAMPLES_TRUE@	    	$(top_srcdir)/contrib/hdtbl/examples/test-hdtbl.sh.in > $@
+#@BUILD_EXAMPLES_TRUE@	chmod +x $@
 
 .roff.ps:
 	$(GROFF_V)$(MKDIR_P) `dirname $@` \
