--- a/External/FEXCore/include/FEXCore/Core/UContext.h
+++ b/External/FEXCore/include/FEXCore/Core/UContext.h
@@ -163,12 +163,14 @@
         val.si_errno = si_errno;
         val.si_code = si_code;
 
+#ifndef __ANDROID__
         // Host siginfo has a pad member that is set to zeros
         val.__pad0 = 0;
 
         // Copy over the union
         // The union is different sizes on 64-bit versus 32-bit
         memcpy(val._sifields._pad, _sifields.pad, std::min(sizeof(val._sifields._pad), sizeof(_sifields.pad)));
+#endif
 
         return val;
       }
