--- a/Source/Tests/LinuxSyscalls/Syscalls/Key.cpp
+++ b/Source/Tests/LinuxSyscalls/Syscalls/Key.cpp
@@ -30,6 +30,7 @@
       SYSCALL_ERRNO();
     });
 
+#ifndef __ANDROID__
     REGISTER_SYSCALL_IMPL_PASS(pkey_mprotect, [](FEXCore::Core::CpuStateFrame *Frame, void *addr, size_t len, int prot, int pkey) -> uint64_t {
       // Added in Linux 4.9
       uint64_t Result = ::pkey_mprotect(addr, len, prot, pkey);
@@ -47,5 +48,6 @@
       uint64_t Result = ::pkey_free(pkey);
       SYSCALL_ERRNO();
     });
+#endif
   }
 }
