--- a/Source/Tests/LinuxSyscalls/x64/Semaphore.cpp
+++ b/Source/Tests/LinuxSyscalls/x64/Semaphore.cpp
@@ -12,7 +12,7 @@
 
 #include <stddef.h>
 #include <stdint.h>
-#include <sys/sem.h>
+#include <linux/sem.h>
 
 namespace FEXCore::Core {
   struct CpuStateFrame;
@@ -22,6 +22,7 @@
 
 namespace FEX::HLE::x64 {
   void RegisterSemaphore() {
+#if !defined(__ANDROID__) || __ANDROID_API__ >= 26
    REGISTER_SYSCALL_IMPL_X64_PASS(semop, [](FEXCore::Core::CpuStateFrame *Frame, int semid, struct sembuf *sops, size_t nsops) -> uint64_t {
       uint64_t Result = ::semop(semid, sops, nsops);
       SYSCALL_ERRNO();
@@ -87,5 +88,6 @@
       }
       SYSCALL_ERRNO();
     });
+#endif
   }
 }
