--- a/Source/Tools/CMakeLists.txt
+++ b/Source/Tools/CMakeLists.txt
@@ -2,11 +2,11 @@
   add_subdirectory(Debugger/)
 endif()
 
-add_subdirectory(FEXConfig/)
+#add_subdirectory(FEXConfig/)
 add_subdirectory(FEXGetConfig/)
 add_subdirectory(FEXMountDaemon/)
-add_subdirectory(FEXLogServer/)
-add_subdirectory(FEXRootFSFetcher/)
+#add_subdirectory(FEXLogServer/)
+#add_subdirectory(FEXRootFSFetcher/)
 
 set(NAME Opt)
 set(SRCS Opt.cpp)
