--- a/Source/Tests/LinuxSyscalls/x64/Types.h
+++ b/Source/Tests/LinuxSyscalls/x64/Types.h
@@ -13,7 +13,9 @@
 #include <asm/posix_types.h>
 #include <asm/sembuf.h>
 #include <cstdint>
-#include <sys/sem.h>
+#include <linux/sem.h>
+#define semid_ds semid64_ds
+#include <sys/ipc.h>
 #include <sys/stat.h>
 #include <type_traits>
 
@@ -128,11 +130,15 @@
     __kernel_long_t   st_blocks;  /* Number 512-byte blocks allocated. */
 
     __kernel_ulong_t  st_atime_;
+#undef st_atime_nsec
     __kernel_ulong_t  st_atime_nsec;
     __kernel_ulong_t  st_mtime_;
+#undef st_mtime_nsec
     __kernel_ulong_t  st_mtime_nsec;
     __kernel_ulong_t  st_ctime_;
+#undef st_ctime_nsec
     __kernel_ulong_t  st_ctime_nsec;
+#undef __unused
     __kernel_long_t   __unused[3];
 
     guest_stat() = delete;
