--- a/Source/Tests/LinuxSyscalls/x64/Thread.cpp
+++ b/Source/Tests/LinuxSyscalls/x64/Thread.cpp
@@ -14,8 +14,8 @@
 #include <FEXCore/Debug/InternalThreadState.h>
 #include <FEXCore/HLE/Linux/ThreadManagement.h>
 
-#include <bits/types/siginfo_t.h>
-#include <bits/types/stack_t.h>
+#include <asm-generic/siginfo.h>
+#include <asm-generic/signal.h>
 #include <sched.h>
 #include <stddef.h>
 #include <syscall.h>
