--- a/Source/Tests/LinuxSyscalls/LinuxAllocator.cpp
+++ b/Source/Tests/LinuxSyscalls/LinuxAllocator.cpp
@@ -19,8 +19,6 @@
 class MemAllocator32Bit final : public FEX::HLE::MemAllocator {
 private:
   static constexpr uint64_t PAGE_SHIFT = 12;
-  static constexpr uint64_t PAGE_SIZE = 1 << PAGE_SHIFT;
-  static constexpr uint64_t PAGE_MASK = (1 << PAGE_SHIFT) - 1;
   static constexpr uint64_t BASE_KEY = 16;
   const uint64_t TOP_KEY = 0xFFFF'F000ULL >> PAGE_SHIFT;
   const uint64_t TOP_KEY32BIT = 0x1F'F000ULL >> PAGE_SHIFT;
