--- a/Source/Tests/LinuxSyscalls/SignalDelegator.cpp
+++ b/Source/Tests/LinuxSyscalls/SignalDelegator.cpp
@@ -24,7 +24,7 @@
 #include <exception>
 #include <functional>
 #include <linux/futex.h>
-#include <bits/types/stack_t.h>
+#include <asm-generic/signal.h>
 #include <signal.h>
 #include <syscall.h>
 #include <sys/mman.h>
@@ -42,7 +42,6 @@
   }
 #endif
 
-  constexpr static uint32_t SS_AUTODISARM = (1U << 31);
   constexpr static uint32_t X86_MINSIGSTKSZ  = 0x2000U;
 
   // We can only have one delegator per process
