--- a/crypton-x509-system/System/X509/Unix.hs	2026-09-05 18:10:05.290104441 +0530
+++ b/crypton-x509-system/System/X509/Unix.hs	2026-09-05 18:10:34.645372444 +0530
@@ -32,6 +32,7 @@
     , "/system/etc/security/cacerts/" -- android
     , "/usr/local/share/certs/" -- freebsd
     , "/etc/ssl/cert.pem" -- openbsd
+    , "@TERMUX_PREFIX@/etc/tls/cert.pem" -- termux android
     ]
 
 envPathOverride :: String
