--- xfce4-docklike-plugin-xfce4-docklike-plugin-0.4.0/src/Hotkeys.cpp	2021-09-27 04:46:28.000000000 +0000
+++ xfce4-docklike-plugin-xfce4-docklike-plugin-0.4.0.mod/src/Hotkeys.cpp	2022-09-11 11:34:36.859662256 +0000
@@ -6,9 +6,36 @@
  */
 
 #include "Hotkeys.hpp"
+#include <pthread.h>
+#include <signal.h>
 
 #define ModifierChange 85 //? this event type isn't listed in libX11
 
+#define SIG_CANCEL_SIGNAL SIGUSR1
+#define PTHREAD_CANCEL_ENABLE 1
+#define PTHREAD_CANCEL_DISABLE 0
+#define PTHREAD_CANCEL_ASYNCHRONOUS 2
+
+static int pthread_setcancelstate(int state, int *oldstate) {
+    sigset_t   new, old;
+    int ret;
+    sigemptyset (&new);
+    sigaddset (&new, SIG_CANCEL_SIGNAL);
+
+    ret = pthread_sigmask(state == PTHREAD_CANCEL_ENABLE ? SIG_BLOCK : SIG_UNBLOCK, &new , &old);
+    if(oldstate != NULL)
+    {
+        *oldstate =sigismember(&old,SIG_CANCEL_SIGNAL) == 0 ? PTHREAD_CANCEL_DISABLE : PTHREAD_CANCEL_ENABLE;
+    }
+    return ret;
+}
+
+
+static inline int pthread_cancel(pthread_t thread) {
+
+    pthread_kill(thread, SIG_CANCEL_SIGNAL);
+}
+
 namespace Hotkeys
 {
 	int mGrabbedKeys;
