diff -uNr SDL-1.2.15/src/video/x11/SDL_x11gl.c SDL-1.2.15.mod/src/video/x11/SDL_x11gl.c
--- SDL-1.2.15/src/video/x11/SDL_x11gl.c	2012-01-19 08:30:06.000000000 +0200
+++ SDL-1.2.15.mod/src/video/x11/SDL_x11gl.c	2018-10-26 23:20:45.228724684 +0300
@@ -30,7 +30,7 @@
 /* IRIX doesn't have a GL library versioning system */
 #define DEFAULT_OPENGL	"libGL.so"
 #elif defined(__MACOSX__)
-#define DEFAULT_OPENGL	"/usr/X11R6/lib/libGL.1.dylib"
+#define DEFAULT_OPENGL	"/data/data/com.termux/files/usr/X11R6/lib/libGL.1.dylib"
 #elif defined(__QNXNTO__)
 #define DEFAULT_OPENGL	"libGL.so.3"
 #elif defined(__OpenBSD__)
