diff -uNr SDL-1.2.15/src/video/SDL_yuv_sw.c SDL-1.2.15.mod/src/video/SDL_yuv_sw.c
--- SDL-1.2.15/src/video/SDL_yuv_sw.c	2012-01-19 08:30:06.000000000 +0200
+++ SDL-1.2.15.mod/src/video/SDL_yuv_sw.c	2018-10-26 23:20:45.225391322 +0300
@@ -89,6 +89,8 @@
 #include "SDL_yuvfuncs.h"
 #include "SDL_yuv_sw_c.h"
 
+#undef __OPTIMIZE__
+
 /* The functions used to manipulate software video overlays */
 static struct private_yuvhwfuncs sw_yuvfuncs = {
 	SDL_LockYUV_SW,
