diff -uNr SDL-1.2.15/src/cdrom/linux/SDL_syscdrom.c SDL-1.2.15.mod/src/cdrom/linux/SDL_syscdrom.c
--- SDL-1.2.15/src/cdrom/linux/SDL_syscdrom.c	2012-01-19 08:30:06.000000000 +0200
+++ SDL-1.2.15.mod/src/cdrom/linux/SDL_syscdrom.c	2018-10-26 23:20:45.205391150 +0300
@@ -68,12 +68,12 @@
 #ifdef MNTTAB
 #define _PATH_MNTTAB	MNTTAB
 #else
-#define _PATH_MNTTAB	"/etc/fstab"
+#define _PATH_MNTTAB	"@TERMUX_PREFIX@/etc/fstab"
 #endif
 #endif /* !_PATH_MNTTAB */
 
 #ifndef _PATH_MOUNTED
-#define _PATH_MOUNTED	"/etc/mtab"
+#define _PATH_MOUNTED	"@TERMUX_PREFIX@/etc/mtab"
 #endif /* !_PATH_MOUNTED */
 
 #ifndef MNTTYPE_CDROM
