diff -uNr SDL-1.2.15/src/video/photon/SDL_ph_gl.h SDL-1.2.15.mod/src/video/photon/SDL_ph_gl.h
--- SDL-1.2.15/src/video/photon/SDL_ph_gl.h	2012-01-19 08:30:06.000000000 +0200
+++ SDL-1.2.15.mod/src/video/photon/SDL_ph_gl.h	2018-10-26 23:20:45.215391237 +0300
@@ -26,7 +26,7 @@
 
 #include "SDL_ph_video.h"
 
-#define DEFAULT_OPENGL "/usr/lib/libGL.so"
+#define DEFAULT_OPENGL "@TERMUX_PREFIX@/lib/libGL.so"
 
 #if SDL_VIDEO_OPENGL
     void  ph_GL_SwapBuffers(_THIS);
