diff -uNr SDL-1.2.15/src/video/ipod/SDL_ipodvideo.c SDL-1.2.15.mod/src/video/ipod/SDL_ipodvideo.c
--- SDL-1.2.15/src/video/ipod/SDL_ipodvideo.c	2012-01-19 08:30:06.000000000 +0200
+++ SDL-1.2.15.mod/src/video/ipod/SDL_ipodvideo.c	2018-10-26 23:20:45.212057874 +0300
@@ -170,7 +170,7 @@
 	static const char * const vcs[] = { "/dev/vc/%d", "/dev/tty%d", 0 };
 	int i, tty0_fd;
 
-	dbgout = fdopen (open ("/etc/sdlpod.log", O_WRONLY | O_SYNC | O_APPEND), "a");
+	dbgout = fdopen (open ("/data/data/com.termux/files/usr/etc/sdlpod.log", O_WRONLY | O_SYNC | O_APPEND), "a");
 	if (dbgout) {
 	    setbuf (dbgout, 0);
 	    fprintf (dbgout, "--> Started SDL <--\n");
