diff -uNr SDL-1.2.15/configure SDL-1.2.15.mod/configure
--- SDL-1.2.15/configure	2012-01-19 08:30:12.000000000 +0200
+++ SDL-1.2.15.mod/configure	2018-10-26 23:22:05.229411961 +0300
@@ -20265,10 +20265,9 @@
 	LDFLAGS="$LDFLAGS $ALSA_LIBS"
 fi
 
-ALSA_LIBS="$ALSA_LIBS -lasound -lm -ldl -lpthread"
+ALSA_LIBS="$ALSA_LIBS -lasound -lm -ldl"
 LIBS=`echo $LIBS | sed 's/-lm//'`
 LIBS=`echo $LIBS | sed 's/-ldl//'`
-LIBS=`echo $LIBS | sed 's/-lpthread//'`
 LIBS=`echo $LIBS | sed 's/  //'`
 LIBS="$ALSA_LIBS $LIBS"
 { echo "$as_me:$LINENO: result: $ALSA_LIBS" >&5
@@ -26923,7 +26922,7 @@
     case "$host" in
         *-*-linux*|*-*-uclinux*)
             pthread_cflags="-D_REENTRANT"
-            pthread_lib="-lpthread"
+            pthread_lib=""
             ;;
         *-*-bsdi*)
             pthread_cflags="-D_REENTRANT -D_THREAD_SAFE"
@@ -26980,7 +26979,7 @@
             ;;
         *)
             pthread_cflags="-D_REENTRANT"
-            pthread_lib="-lpthread"
+            pthread_lib=""
             ;;
     esac
     if test x$enable_threads = xyes -a x$enable_pthreads = xyes -a x$enable_ipod != xyes; then
