--- ./Makefile.orig	2021-07-24 10:36:40.069413331 +0200
+++ ./Makefile	2021-07-24 10:40:29.069413167 +0200
@@ -99,7 +99,7 @@
 	cp dMagneticini.5 $(INSTALLMAN)/man5/
 	cp README.txt $(INSTALLSHARE)/dMagnetic/
 	cp LICENSE.txt $(INSTALLSHARE)/dMagnetic/
-	cp dMagnetic.ini $(INSTALLSHARE)/dMagnetic/
+	cp dMagnetic.ini $(INSTALLSHARE)/
 
 
 dMagnetic:	$(OBJ_LOADER) $(OBJ_LINEA) $(OBJ_VM68K) $(OBJ_GUI) $(OBJ_TOPLEVEL)
