--- ./Makefile.orig	2021-06-23 20:13:37.000000000 +0000
+++ ./Makefile	2021-07-10 21:09:12.768295842 +0000
@@ -106,7 +106,7 @@
 	$(LINK) $(LDFLAGS) -o $@ $(OBJ_LOADER) $(OBJ_LINEA) $(OBJ_VM68K) $(OBJ_GUI) $(OBJ_TOPLEVEL)
 
 dMagnetic.ini:	dMagnetic
-	./dMagnetic -helpini >dMagnetic.ini
+	dMagnetic -helpini >dMagnetic.ini
 
 .c.o:
 	$(CC) $(CPPFLAGS) $(CFLAGS) $(CFLAGS_EXTRA) $(INCFLAGS) -c -o $@ $<
