diff -u -r ../hexedit/Makefile.in ./Makefile.in
--- ../hexedit/Makefile.in	2010-02-04 15:20:52.000000000 +0000
+++ ./Makefile.in	2018-11-01 00:08:13.556474129 +0000
@@ -42,6 +42,6 @@
 
 install: $(PRODUCT)
 	$(INSTALL) -d -m 755 $(DESTDIR)$(bindir)
-	$(INSTALL) -s -m 755 $(PRODUCT) $(DESTDIR)$(bindir)
+	$(INSTALL) -m 755 $(PRODUCT) $(DESTDIR)$(bindir)
 	$(INSTALL) -d -m 755 $(DESTDIR)$(mandir)/man1
 	$(INSTALL) -m 644 $(PRODUCT).1 $(DESTDIR)$(mandir)/man1
