--- ext/libxml/libxml.c.org	2022-10-31 10:35:56.000000000 +0000
+++ ./ext/libxml/libxml.c	2024-04-26 05:38:31.628986684 +0000
@@ -530,7 +530,11 @@
 	return 1;
 }
 
+#if LIBXML_VERSION >= 21200
+static void _php_list_set_error_structure(const xmlError *error, const char *msg)
+#else
 static void _php_list_set_error_structure(xmlErrorPtr error, const char *msg)
+#endif
 {
 	xmlError error_copy;
 	int ret;
@@ -782,7 +786,11 @@
 	va_end(args);
 }
 
+#if LIBXML_VERSION >= 21200
+PHP_LIBXML_API void php_libxml_structured_error_handler(void *userData, const xmlError *error)
+#else
 PHP_LIBXML_API void php_libxml_structured_error_handler(void *userData, xmlErrorPtr error)
+#endif
 {
 	_php_list_set_error_structure(error, NULL);
 
@@ -1061,9 +1069,7 @@
    Retrieve last error from libxml */
 static PHP_FUNCTION(libxml_get_last_error)
 {
-	xmlErrorPtr error;
-
-	error = xmlGetLastError();
+	const xmlError *error = xmlGetLastError();
 
 	if (error) {
 		object_init_ex(return_value, libxmlerror_class_entry);
