--- a/waflib/Context.py
+++ b/waflib/Context.py
@@ -2,3 +2,3 @@
 
-import os,re,imp,sys
+import os,re,types,sys
 from waflib import Utils,Errors,Logs
@@ -106,7 +106,7 @@
 				cache[node]=True
 				self.pre_recurse(node)
 				try:
-					function_code=node.read('rU',encoding)
+					function_code=node.read('r',encoding)
 					exec(compile(function_code,node.abspath(),'exec'),self.exec_dict)
 				finally:
 					self.post_recurse(node)
@@ -346,7 +346,7 @@
 		pass
-	module=imp.new_module(WSCRIPT_FILE)
+	module=types.ModuleType(WSCRIPT_FILE)
 	try:
-		code=Utils.readf(path,m='rU',encoding=encoding)
+		code=Utils.readf(path,m='r',encoding=encoding)
 	except EnvironmentError:
 		raise Errors.WafError('Could not read the file %r'%path)
 	module_dir=os.path.dirname(path)
