--- a/SPIRV/CMakeLists.txt
+++ b/SPIRV/CMakeLists.txt
@@ -104,5 +97,5 @@ endif()
 if(GLSLANG_ENABLE_INSTALL)
     install(TARGETS SPIRV EXPORT glslang-targets)
 
-    install(FILES ${PUBLIC_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/glslang/SPIRV/)
+    install(FILES ${PUBLIC_HEADERS} ${HEADERS} ${SPIRV_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/glslang/SPIRV/)
 endif()
--- a/glslang/CMakeLists.txt
+++ b/glslang/CMakeLists.txt
@@ -256,17 +256,12 @@ if(GLSLANG_ENABLE_INSTALL)
         install(TARGETS GenericCodeGen EXPORT glslang-targets)
     endif()
 
-    set(PUBLIC_HEADERS
-        Public/ResourceLimits.h
-        Public/ShaderLang.h
-        Public/resource_limits_c.h
-        Include/glslang_c_interface.h
-        Include/glslang_c_shader_types.h
-        Include/ResourceLimits.h
-        Include/visibility.h
-        MachineIndependent/Versions.h)
+    set(ALL_HEADERS
+        ${GLSLANG_HEADERS}
+        ${MACHINEINDEPENDENT_HEADERS}
+        ${RESOURCELIMITS_HEADERS})
 
-    foreach(file ${PUBLIC_HEADERS})
+    foreach(file ${ALL_HEADERS})
         get_filename_component(dir ${file} DIRECTORY)
         install(FILES ${file} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/glslang/${dir})
     endforeach()
