diff --git a/Makefile b/Makefile
index 4843c16..6b99a1d 100644
--- a/Makefile
+++ b/Makefile
@@ -71,7 +71,7 @@ endif
 
 CPPFLAGS += $(DEFS) $(INCLUDES)
 
-BASHCOMPDIR ?= $(shell $(PKG_CONFIG) --variable=completionsdir bash-completion 2>/dev/null || echo /etc/bash_completion.d)
+BASHCOMPDIR ?= $(prefix)/etc/bash_completion.d
 
 ##############################################################################
 all: Makefile.depend whois mkpasswd pos
