diff -uNr qemu-4.2.0/linux-user/strace.c qemu-4.2.0.mod/linux-user/strace.c
--- qemu-4.2.0/linux-user/strace.c	2019-12-12 20:20:48.000000000 +0200
+++ qemu-4.2.0.mod/linux-user/strace.c	2019-12-15 00:18:22.178280369 +0200
@@ -1,7 +1,7 @@
 #include "qemu/osdep.h"
 #include <sys/ipc.h>
 #include <sys/msg.h>
-#include <sys/sem.h>
+#include <linux/sem.h>
 #include <sys/shm.h>
 #include <sys/select.h>
 #include <sys/mount.h>
