--- a/include/wx/android/config_android.h
+++ b/include/wx/android/config_android.h
@@ -22,9 +22,9 @@
 #define wxUSE_UNIX 1
 #define __UNIX__ 1
 
-#define HAVE_NANOSLEEP
+#define HAVE_NANOSLEEP 1
 #define HAVE_FCNTL 1
-#define HAVE_GCC_ATOMIC_BUILTINS
+#define HAVE_GCC_ATOMIC_BUILTINS 1
 #define HAVE_GETHOSTBYNAME 1
 #define HAVE_GETSERVBYNAME 1
 #define HAVE_GETTIMEOFDAY 1
@@ -40,15 +40,25 @@
 #define HAVE_WPRINTF 1
 
 #define SIZEOF_INT 4
+#ifdef __LP64__
+#define SIZEOF_LONG 8
+#define SIZEOF_SIZE_T 8
+#define SIZEOF_VOID_P 8
+#else
 #define SIZEOF_LONG 4
-#define SIZEOF_LONG_LONG 8
 #define SIZEOF_SIZE_T 4
 #define SIZEOF_VOID_P 4
+#endif
+#define SIZEOF_LONG_LONG 8
 #define SIZEOF_WCHAR_T 4
 
 #define wxHAVE_PTHREAD_CLEANUP 1
 #define wxNO_WOSTREAM
+#ifdef __LP64__
+#undef wxSIZE_T_IS_UINT
+#else
 #define wxSIZE_T_IS_UINT 1
+#endif
 #define wxWCHAR_T_IS_REAL_TYPE 1
 
 #define wxTYPE_SA_HANDLER int
