--- a/include/wx/android/chkconf.h
+++ b/include/wx/android/chkconf.h
@@ -12,6 +12,8 @@
 #ifndef _WX_ANDROID_CHKCONF_H_
 #define _WX_ANDROID_CHKCONF_H_
 
+#if 0
+
 // For Qt under Android, use default configuration procedure as most features
 // should be supported (see documentation for more information)
 
@@ -233,4 +235,6 @@
 
 #endif // __WXQT__
 
+#endif // 0
+
 #endif // _WX_ANDROID_CHKCONF_H_
