--- a/include/wx/wxcrtbase.h
+++ b/include/wx/wxcrtbase.h
@@ -158,7 +158,7 @@
 #define wxCRT_StrtolA    strtol
 #define wxCRT_StrtoulA   strtoul
 
-#ifdef __ANDROID__ // these functions are broken on android
+#if 0 // defined __ANDROID__ // these functions are broken on android
 
 extern double android_wcstod(const wchar_t *nptr, wchar_t **endptr);
 extern long android_wcstol(const wchar_t *nptr, wchar_t **endptr, int base);
