--- a/meson.build
+++ b/meson.build
@@ -92,7 +92,7 @@ uuid_dep = dependency('uuid', required: true, version: '>=2.19.0')
 # clients and cast do not actually require all of them, see if this can be optimised further
 if enable_clients
    boost_dep = dependency(
-      'boost', modules: ['chrono', 'filesystem', 'system', 'thread', 'program_options', 'python3'],
+      'boost', modules: ['chrono', 'filesystem', 'thread', 'program_options', 'python3'],
       required: true, version: '>=1.54'
    )
 else
