--- a/player/src/tizprogramopts.cpp
+++ b/player/src/tizprogramopts.cpp
@@ -647,7 +647,7 @@
       "    - Snap package: $HOME/snap/tizonia/current/.config/tizonia/tizonia.conf\n");
   printf (
       "\nExample configuration files may also be found at \n"
-      "    - /etc/xdg/tizonia/tizonia.conf or \n"
+      "    - @TERMUX_PREFIX@/etc/xdg/tizonia/tizonia.conf or \n"
       "    - /snap/tizonia/current/etc/xdg/tizonia/tizonia.conf.\n");
 
 //
