--- a/plugins/meson.build
+++ b/plugins/meson.build
@@ -70,10 +70,6 @@
    subdir('pcm_renderer_pa')
 endif
 
-if enabled_plugins.contains('vorbis_decoder')
-   subdir('vorbis_decoder')
-endif
-
 if enabled_plugins.contains('vp8_decoder')
    subdir('vp8_decoder')
 endif
@@ -81,19 +77,3 @@
 if enabled_plugins.contains('webm_demuxer')
    subdir('webm_demuxer')
 endif
-
-if enabled_plugins.contains('yuv_renderer')
-   subdir('yuv_renderer')
-endif
-
-if enable_libspotify
-   subdir('spotify_source')
-endif
-
-if enable_alsa
-   subdir('pcm_renderer_alsa')
-endif
-
-if enable_aac
-   subdir('aac_decoder')
-endif
