diff -uNr xfce4-settings-4.18.2/dialogs/appearance-settings/appearance-install-theme xfce4-settings-4.18.2.mod/dialogs/appearance-settings/appearance-install-theme
--- xfce4-settings-4.18.2/dialogs/appearance-settings/appearance-install-theme	2023-02-11 11:13:50.000000000 +0200
+++ xfce4-settings-4.18.2.mod/dialogs/appearance-settings/appearance-install-theme	2023-04-04 10:43:33.297227674 +0300
@@ -104,10 +104,10 @@
 # we try $XDG_CACHE_HOME because it is more likely this is on the
 # same partition, so moving the theme after extract is faster
 if test x"`which mktemp 2>/dev/null`" != x""; then
-    TMPDIR="${TMPDIR:-/tmp}"
+    TMPDIR="${TMPDIR:-@TERMUX_PREFIX@/tmp}"
     tmpdir=`TMPDIR="${XDG_CACHE_HOME:-$TMPDIR}" mktemp -d`
 else
-    tmpdir="/tmp/tmp.$$.$RANDOM"
+    tmpdir="@TERMUX_PREFIX@/tmp/tmp.$$.$RANDOM"
     mkdir "$tmpdir"
 fi
 if test ! -d "$tmpdir"; then
