diff -uNr mariadb-10.4.6/client/CMakeLists.txt mariadb-10.4.6.mod/client/CMakeLists.txt
--- mariadb-10.4.6/client/CMakeLists.txt	2019-06-18 01:00:31.000000000 +0300
+++ mariadb-10.4.6.mod/client/CMakeLists.txt	2019-07-24 17:53:17.771246533 +0300
@@ -44,10 +44,6 @@
   SET_TARGET_PROPERTIES(mysql PROPERTIES ENABLE_EXPORTS TRUE)
 ENDIF(UNIX)

-MYSQL_ADD_EXECUTABLE(mysqltest mysqltest.cc COMPONENT Test)
-SET_SOURCE_FILES_PROPERTIES(mysqltest.cc PROPERTIES COMPILE_FLAGS "-DTHREADS")
-TARGET_LINK_LIBRARIES(mysqltest  ${CLIENT_LIB} pcreposix pcre)
-SET_TARGET_PROPERTIES(mysqltest PROPERTIES ENABLE_EXPORTS TRUE)


 MYSQL_ADD_EXECUTABLE(mysqlcheck mysqlcheck.c)
@@ -92,7 +88,7 @@
 SET_TARGET_PROPERTIES (mysqlcheck mysqldump mysqlimport mysql_upgrade mysqlshow mysqlslap mysql_plugin async_example
 PROPERTIES HAS_CXX TRUE)

-FOREACH(t mysql mysqltest mysqlcheck mysqldump mysqlimport mysql_upgrade mysqlshow mysql_plugin mysqlbinlog
+FOREACH(t mysql mysqlcheck mysqldump mysqlimport mysql_upgrade mysqlshow mysql_plugin mysqlbinlog
   mysqladmin mysqlslap async_example)
   ADD_DEPENDENCIES(${t} GenError ${CLIENT_LIB})
 ENDFOREACH()
