diff -u -r ../mariadb-10.4.12/cmake/build_configurations/mysql_release.cmake ./cmake/build_configurations/mysql_release.cmake
--- ../mariadb-10.4.12/cmake/build_configurations/mysql_release.cmake	2020-01-26 20:43:49.000000000 +0000
+++ ./cmake/build_configurations/mysql_release.cmake	2020-02-15 23:22:03.546831000 +0000
@@ -124,7 +124,7 @@
 
 IF(UNIX)
   SET(WITH_EXTRA_CHARSETS all CACHE STRING "")
-  SET(PLUGIN_AUTH_PAM YES)
+  SET(PLUGIN_AUTH_PAM NO)
 
   IF(CMAKE_SYSTEM_NAME STREQUAL "Linux")
     IF(NOT IGNORE_AIO_CHECK)
