diff -u -r ../mariadb-10.3.8/CMakeLists.txt ./CMakeLists.txt
--- ../mariadb-10.3.8/CMakeLists.txt	2018-07-02 07:34:08.000000000 +0000
+++ ./CMakeLists.txt	2018-07-03 11:15:40.073729412 +0000
@@ -397,7 +397,6 @@
   OPTION (WITH_EMBEDDED_SERVER "Compile MySQL with embedded server" OFF)
   IF(WITH_EMBEDDED_SERVER) 
    ADD_SUBDIRECTORY(libmysqld)
-   ADD_SUBDIRECTORY(libmysqld/examples)
    ADD_SUBDIRECTORY(unittest/embedded)
   ENDIF(WITH_EMBEDDED_SERVER)
 
