diff -u -r ../mariadb-10.1.22/cmake/readline.cmake ./cmake/readline.cmake
--- ../mariadb-10.1.22/cmake/readline.cmake	2017-03-11 20:08:54.000000000 +0100
+++ ./cmake/readline.cmake	2017-04-07 02:43:58.242765383 +0200
@@ -180,7 +180,7 @@
     #include <readline.h>
     int main(int argc, char **argv)
     {
-      int res= (*rl_completion_entry_function)(0,0);
+      char* res= (*rl_completion_entry_function)(0,0);
       completion_matches(0,0);
     }"
     LIBEDIT_INTERFACE)
