diff -u -r ../mariadb-10.5.4/cmake/install_layout.cmake ./cmake/install_layout.cmake
--- ../mariadb-10.5.4/cmake/install_layout.cmake	2020-06-23 15:10:16.000000000 +0000
+++ ./cmake/install_layout.cmake	2020-08-08 22:17:34.254546116 +0000
@@ -175,7 +175,6 @@
 SET(INSTALL_SBINDIR_DEB                 "sbin")
 SET(INSTALL_SCRIPTDIR_DEB               "bin")
 SET(INSTALL_SYSCONFDIR_DEB              "/etc")
-SET(INSTALL_SYSCONF2DIR_DEB             "/etc/mysql/mariadb.conf.d")
 #
 SET(INSTALL_LIBDIR_DEB                  "lib/${CMAKE_CXX_LIBRARY_ARCHITECTURE}")
 SET(INSTALL_PLUGINDIR_DEB               "lib/mysql/plugin")
