diff -u -r ../mariadb-10.2.7/CMakeLists.txt ./CMakeLists.txt
--- ../mariadb-10.2.7/CMakeLists.txt	2017-07-10 10:53:26.000000000 +0000
+++ ./CMakeLists.txt	2017-08-09 23:44:24.691181837 +0000
@@ -386,7 +386,6 @@
   OPTION (WITH_EMBEDDED_SERVER "Compile MySQL with embedded server" OFF)
   IF(WITH_EMBEDDED_SERVER) 
    ADD_SUBDIRECTORY(libmysqld)
-   ADD_SUBDIRECTORY(libmysqld/examples)
   ENDIF(WITH_EMBEDDED_SERVER)
 
   IF(WITH_WSREP)
