diff -u -r ../mariadb-10.3.14/libmariadb/libmariadb/CMakeLists.txt ./libmariadb/libmariadb/CMakeLists.txt
--- ../mariadb-10.3.14/libmariadb/libmariadb/CMakeLists.txt	2019-04-01 14:03:57.000000000 +0000
+++ ./libmariadb/libmariadb/CMakeLists.txt	2019-05-22 23:50:06.105346983 +0000
@@ -302,7 +302,6 @@
   IF(ICONV_INCLUDE_DIR)
      INCLUDE_DIRECTORIES(BEFORE ${ICONV_INCLUDE_DIR}) 
   ENDIF()
-  ADD_DEFINITIONS(-DLIBICONV_PLUG)
 ENDIF()
 
 IF(ZLIB_FOUND AND WITH_EXTERNAL_ZLIB)
