diff -uNr mariadb-10.6.5.orig/libmariadb/mariadb_config/mariadb_config.c.in mariadb-10.6.5/libmariadb/mariadb_config/mariadb_config.c.in
--- mariadb-10.6.5.orig/libmariadb/mariadb_config/mariadb_config.c.in	2021-11-05 21:03:32.000000000 +0100
+++ mariadb-10.6.5/libmariadb/mariadb_config/mariadb_config.c.in	2021-11-17 13:46:21.744760378 +0100
@@ -210,7 +210,7 @@
   }
   if (!p || !p[0])
   {
-    strncpy(installation_dir, "@CMAKE_SYSROOT@@CMAKE_INSTALL_PREFIX@", PATH_MAX - 1);
+    strncpy(installation_dir, "@CMAKE_INSTALL_PREFIX@", PATH_MAX - 1);
     return;
   }
 }
