--- a/plugin/disks/information_schema_disks.cc
+++ b/plugin/disks/information_schema_disks.cc
@@ -222,8 +222,10 @@
 #endif
         // Try to keep to real storage by excluding
         // read only mounts, and mount points that aren't directories
+#if !defined __ANDROID__ || __ANDROID_API__ >= 26
         if (hasmntopt(pEnt, MNTOPT_RO) != NULL)
             continue;
+#endif
         if (stat(path, &f))
             continue;
         if (!S_ISDIR(f.st_mode))
