diff -u -r ../mariadb-10.3.8/sql/CMakeLists.txt ./sql/CMakeLists.txt
--- ../mariadb-10.3.8/sql/CMakeLists.txt	2018-07-02 07:34:11.000000000 +0000
+++ ./sql/CMakeLists.txt	2018-07-22 21:00:22.913542838 +0000
@@ -388,6 +388,7 @@
         DEPENDS
         ${CMAKE_CURRENT_BINARY_DIR}/lex_hash.h
         ${CMAKE_CURRENT_BINARY_DIR}/lex_token.h
+        ${CMAKE_CURRENT_BINARY_DIR}/sql_yacc.cc
         ${CMAKE_CURRENT_BINARY_DIR}/sql_yacc_ora.cc
 )
 
