diff -uNr mariadb-10.9.2/support-files/mysql.server.sh mariadb-10.9.2.mod/support-files/mysql.server.sh
--- mariadb-10.9.2/support-files/mysql.server.sh	2022-08-10 14:23:48.000000000 +0200
+++ mariadb-10.9.2.mod/support-files/mysql.server.sh	2022-09-07 17:44:33.263691615 +0200
@@ -185,11 +185,7 @@
 user='@MYSQLD_USER@'
 
 su_kill() {
-  if test "$USER" = "$user"; then
     kill $* >/dev/null 2>&1
-  else
-    su - $user -s /bin/sh -c "kill $*" >/dev/null 2>&1
-  fi
 }
 
 #
