diff --git a/src/core/paths.cpp b/src/core/paths.cpp
index e17c3bc..2bc6c2b 100644
--- a/src/core/paths.cpp
+++ b/src/core/paths.cpp
@@ -57,7 +57,12 @@ QDir* QsPaths::baseRunDir() {
 	if (this->baseRunState == DirState::Unknown) {
 		auto runtimeDir = qEnvironmentVariable("XDG_RUNTIME_DIR");
 		if (runtimeDir.isEmpty()) {
+                #ifdef __TERMUX__
+                        runtimeDir = QStandardPaths::writableLocation(QStandardPaths::RuntimeLocation);
+                        QDir().mkpath(runtimeDir);
+                #else
 			runtimeDir = QString("/run/user/$1").arg(getuid());
+                #endif
 			qCInfo(logPaths) << "XDG_RUNTIME_DIR was not set, defaulting to" << runtimeDir;
 		}
 
