--- a/qt6/src/platformtheme/renderer.kcfg
+++ b/qt6/src/platformtheme/renderer.kcfg
@@ -20,7 +20,7 @@
                 <choice name="software" />
                 <choice name="vulkan" />
             </choices>
-            <default>automaticbackend</default>
+            <default>software</default>
         </entry>
         <entry name="ForceGlCoreProfile" type="Bool">
             <default>false</default>
