diff -u -r ../xmlstarlet-1.6.1/src/xmlstar.h ./src/xmlstar.h
--- ../xmlstarlet-1.6.1/src/xmlstar.h	2014-03-03 00:15:08.000000000 +0000
+++ ./src/xmlstar.h	2024-05-21 12:15:09.735932324 +0000
@@ -32,7 +32,7 @@
     ErrorStop stop;
 } ErrorInfo;
 
-void reportError(void *ptr, xmlErrorPtr error);
+void reportError(void *ptr, const const xmlError * error);
 void suppressErrors(void);
 
 typedef struct _gOptions {
