diff -up openmotif-2.3.3/Makefile.am.no_demos openmotif-2.3.3/Makefile.am
--- a/Makefile.am.no_demos	2008-09-19 16:38:05.000000000 +0200
+++ b/Makefile.am	2010-03-23 13:53:13.000000000 +0100
@@ -29,7 +29,7 @@ SUBDIRS	= bindings bitmaps \
              include \
              tools \
              clients \
-             doc \
-             demos
+             doc
+
 AUTOMAKE_OPTIONS = 1.4
 ACLOCAL_AMFLAGS = -I .
