--- a/src/calibre/gui2/bars.py
+++ b/src/calibre/gui2/bars.py
@@ -167,6 +167,8 @@
         QToolBar.resizeEvent(self, a0)
         style = self.get_text_style()
         self.setToolButtonStyle(style)
+        for tb in self.findChildren(QToolButton):
+            tb.setToolButtonStyle(style)
         if self.showing_donate:
             donate_button = self.donate_button
             assert donate_button is not None
@@ -244,6 +246,8 @@
             self.all_widgets.append(ch)
         if hasattr(ch, 'setAutoRaise'):  # is a QToolButton or similar
             ch.setAutoRaise(True)
+            if hasattr(ch, 'setToolButtonStyle'):
+                ch.setToolButtonStyle(self.get_text_style())
             m = ac.menu()
             if m is not None:
                 if menu_mode is not None:
@@ -825,6 +829,8 @@
         for bar in self.bars:
             bar.setIconSize(QSize(sz, sz))
             bar.setToolButtonStyle(style)
+            for tb in bar.findChildren(QToolButton):
+                tb.setToolButtonStyle(style)
             if bar.donate_button:
                 bar.donate_button.setIconSize(bar.iconSize())
                 bar.donate_button.setToolButtonStyle(style)
