--- src/tlpkg/TeXLive/TLConfig.pm	2016-04-21 19:08:52.000000000 +0000
+++ ../TLConfig.pm	2017-05-30 13:18:48.030712610 +0000
@@ -165,13 +165,13 @@
     [ "b", 1, "postcode",
       "Run postinst code blobs" ],
   "sys_bin" =>
-    [ "p", "/usr/local/bin", "sys_bin",
+    [ "p", "@TERMUX_PREFIX@/bin", "sys_bin",
       "Destination for symlinks for binaries" ],
   "sys_info" =>
-    [ "p", "/usr/local/share/info", "sys_info",
+    [ "p", "@TERMUX_PREFIX@/share/info", "sys_info",
       "Destination for symlinks for info docs" ],
   "sys_man" =>
-    [ "p", "/usr/local/share/man", "sys_man",
+    [ "p", "@TERMUX_PREFIX@/share/man", "sys_man",
       "Destination for symlinks for man pages" ],
   "w32_multi_user" =>
     [ "b", 1, "multiuser",
