--- ../config.mk.orig	2022-01-04 16:32:56.338295489 +0100
+++ ./config.mk	2022-01-04 16:33:21.661580852 +0100
@@ -2,10 +2,10 @@
 DATADIR := .
 
 # Include terminal support. Requires ncurses
-TERMINAL := NO
+TERMINAL := YES
 
 # Include graphical support. Requires SDL2 and SDL2_image
-GRAPHICS := YES
+GRAPHICS := NO
 # Path to sdl2-config script
 SDL_CONFIG := sdl2-config
 
@@ -16,7 +16,7 @@
 DEBUG := NO
 
 # Declare this is a release build
-RELEASE := NO
+RELEASE := YES
 
 # Configure the executable to run from a macOS .app bundle (only works in graphical mode)
 MAC_APP := NO
