--- ../brogue-linux-1.7.3/brogue-1.7.3/Makefile	2013-09-11 07:38:48.000000000 +0200
+++ ./Makefile	2014-01-28 07:57:06.000000000 +0100
@@ -81,7 +81,7 @@
 .PHONY : clean both curses tcod tar
 
 bin/brogue : ${DEPENDENCIES} ${BROGUEFILES}
-	$(CC) -O2 -march=i586 -o bin/brogue ${BROGUEFILES} ${LIBRARIES} -Wl,-rpath,.
+	$(CC) -o bin/brogue ${BROGUEFILES} ${LIBRARIES} -Wl,-rpath,.
 
 clean : 
 	rm -f src/brogue/*.o src/platform/*.o bin/brogue

