diff -u -r ../brogue-1.7.4/src/platform/platformdependent.c ./src/platform/platformdependent.c
--- ../brogue-1.7.4/src/platform/platformdependent.c	2014-07-03 19:19:10.000000000 +0000
+++ ./src/platform/platformdependent.c	2017-07-07 07:09:10.116242335 +0000
@@ -21,6 +21,7 @@
  *  along with Brogue.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <ctype.h>
 #include <stdio.h>
 #include <string.h>
 #include <time.h>
