--- ./main.cpp~	2017-06-09 13:15:17.000000000 +0200
+++ ./main.cpp	2020-09-08 13:59:59.758203243 +0200
@@ -4,7 +4,6 @@
 #include <cstdio>
 #include <algorithm>
 #include <unistd.h>
-#include <bits/stdc++.h>
 #include "input.h"
 #include "snake.h"
 #include "snake_map.h"
--- ./input.cpp~	2017-06-09 13:15:17.000000000 +0200
+++ ./input.cpp	2020-09-08 14:00:10.035093940 +0200
@@ -1,7 +1,7 @@
 #include "input.h"
 #include <termios.h>
 #include <unistd.h>
-#include <bits/stdc++.h>
+#include <iostream>
 #include <string>
 #include <stdio.h>
 
