--- a/Source/bmalloc/libpas/src/libpas/pas_min_heap.h
+++ b/Source/bmalloc/libpas/src/libpas/pas_min_heap.h
@@ -260,7 +260,7 @@
         \
         size_t index = config.get_index(&element); \
         PAS_ASSERT(index); \
-        PAS_ASSERT(!bcmp(name##_get_ptr_by_index(min_heap, index), &element, sizeof(type))); \
+        PAS_ASSERT(!memcmp(name##_get_ptr_by_index(min_heap, index), &element, sizeof(type))); \
         name##_remove_by_index(min_heap, index); \
     } \
     \
