--- a/Source/ThirdParty/skia/CMakeLists.txt
+++ b/Source/ThirdParty/skia/CMakeLists.txt
@@ -833,7 +833,7 @@ else ()
     )
     target_link_libraries(Skia PRIVATE Freetype::Freetype)
 
-    if (ANDROID)
+    if (FALSE)
         target_sources(Skia PRIVATE
             src/ports/SkFontMgr_android.cpp
             src/ports/SkFontMgr_android_parser.cpp
--- a/Source/ThirdParty/skia/src/ports/SkLog_stdio.cpp
+++ b/Source/ThirdParty/skia/src/ports/SkLog_stdio.cpp
@@ -11,7 +11,7 @@
 #include "include/private/SkLoadUserConfig.h"
 #include "include/private/SkLog.h"
 
-#if !defined(SK_BUILD_FOR_WIN) && !defined(SK_BUILD_FOR_ANDROID)
+#if !defined(SK_BUILD_FOR_WIN)
 
 #include <stdarg.h>
 #include <stdio.h>
