diff -uNr unicorn-2.0.0/CMakeLists.txt unicorn-2.0.0.mod/CMakeLists.txt
--- unicorn-2.0.0/CMakeLists.txt	2022-07-07 18:49:09.000000000 +0300
+++ unicorn-2.0.0.mod/CMakeLists.txt	2022-07-21 01:09:52.344584678 +0300
@@ -1434,7 +1434,7 @@
 libdir=${CMAKE_INSTALL_FULL_LIBDIR}\n\
 includedir=${CMAKE_INSTALL_FULL_INCLUDEDIR}\n\
 Libs: -L\$\{libdir\} -lunicorn\n\
-Libs.private: -lpthread -lm\n\
+Libs.private: -lm\n\
 Cflags: -I\$\{includedir\}\n"
     )
     install(FILES ${CMAKE_BINARY_DIR}/unicorn.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
diff -uNr unicorn-2.0.0/qemu/configure unicorn-2.0.0.mod/qemu/configure
--- unicorn-2.0.0/qemu/configure	2022-07-07 18:49:09.000000000 +0300
+++ unicorn-2.0.0.mod/qemu/configure	2022-07-21 01:11:49.021654442 +0300
@@ -1245,6 +1245,7 @@
 PTHREADLIBS_LIST="-pthread -lpthread -lpthreadGC2"
 
 pthread=no
+if false; then
 cat > $TMPC << EOF
 #include <pthread.h>
 static void *f(void *p) { return NULL; }
@@ -1280,6 +1281,7 @@
   error_exit "pthread check failed" \
       "Make sure to have the pthread libs and headers installed."
 fi
+fi
 
 # check for pthread_setname_np with thread id
 pthread_setname_np_w_tid=no
diff -uNr unicorn-2.0.0/samples/Makefile unicorn-2.0.0.mod/samples/Makefile
--- unicorn-2.0.0/samples/Makefile	2022-07-07 18:49:09.000000000 +0300
+++ unicorn-2.0.0.mod/samples/Makefile	2022-07-21 01:10:30.720936535 +0300
@@ -14,12 +14,12 @@
 
 CFLAGS += -Wall -Werror -I../include
 
-LDFLAGS += -L$(LIBDIR) -lunicorn -lpthread -lm
+LDFLAGS += -L$(LIBDIR) -lunicorn -lm
 ifeq ($(UNAME_S), Linux)
 LDFLAGS += -lrt
 endif
 
-LDLIBS += -lpthread -lunicorn -lm
+LDLIBS += -lunicorn -lm
 
 ifneq ($(CROSS),)
 CC = $(CROSS)gcc
