Set to FALSE for testing with gnome-settings-daemon 49+

--- a/gnome-session/gsm-session-fill.c
+++ b/gnome-session/gsm-session-fill.c
@@ -79,7 +79,7 @@ check_required_components_helper (const char *component,
 
         if (app_path == NULL) {
                 g_warning ("Unable to find required component '%s'", component);
-                *error = TRUE;
+                *error = TRUE;
         }
 }
 
