--- a/coders/jp2.c
+++ b/coders/jp2.c
@@ -313,7 +313,7 @@
 #if MAGICK_JP2_OLD_STREAM_INTERFACE
 static int BlobRead(jas_stream_obj_t *object,char *buffer,const int length)
 #else
-static int BlobRead(jas_stream_obj_t *object,char *buffer,unsigned length)
+static ssize_t BlobRead(jas_stream_obj_t *object,char *buffer,size_t length)
 #endif
 {
   size_t
@@ -329,7 +329,7 @@
 #if MAGICK_JP2_OLD_STREAM_INTERFACE
 static int BlobWrite(jas_stream_obj_t *object,char *buffer,const int length)
 #else
-static int BlobWrite(jas_stream_obj_t *object,const char *buffer,unsigned length)
+static ssize_t BlobWrite(jas_stream_obj_t *object,const char *buffer,size_t length)
 #endif
 {
   size_t
