--- a/clang/lib/Driver/SanitizerArgs.cpp
+++ b/clang/lib/Driver/SanitizerArgs.cpp
@@ -1009,7 +1009,7 @@ SanitizerArgs::SanitizerArgs(const ToolChain &TC,
 
   SharedRuntime = Args.hasFlag(
       options::OPT_shared_libsan, options::OPT_static_libsan,
-      TC.getTriple().isAndroid() || TC.getTriple().isOSFuchsia() ||
+      TC.getTriple().isOSFuchsia() ||
           TC.getTriple().isOSDarwin() || TC.getTriple().isOSWindows());
   if (!SharedRuntime && TC.getTriple().isOSWindows()) {
     Arg *A =
