--- ../projects-compiler-rt-CMakeLists.txt.orig	2020-09-22 14:12:54.771097201 +0000
+++ ./projects/compiler-rt/CMakeLists.txt	2020-09-22 16:11:07.568559345 +0000
@@ -114,7 +114,7 @@
 endif()
 pythonize_bool(ANDROID)
 
-set(ANDROID_NDK_VERSION 18
+set(ANDROID_NDK_VERSION @TERMUX_NDK_VERSION_NUM@
     CACHE STRING "Set this to the Android NDK version that you are using")
 
 set(COMPILER_RT_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
