--- a/lldb/source/Utility/ArchSpec.cpp
+++ b/lldb/source/Utility/ArchSpec.cpp
@@ -206,7 +206,11 @@ static constexpr const CoreDefinition g_core_definitions[] = {
      ArchSpec::eCore_sparc9_generic, "sparcv9"},
 
     {eByteOrderLittle, 4, 1, 15, llvm::Triple::x86, ArchSpec::eCore_x86_32_i386,
+#ifndef __ANDROID__
      "i386"},
+#else
+     "i686"},
+#endif
     {eByteOrderLittle, 4, 1, 15, llvm::Triple::x86, ArchSpec::eCore_x86_32_i486,
      "i486"},
     {eByteOrderLittle, 4, 1, 15, llvm::Triple::x86,
