diff --git a/llvm/include/llvm/Demangle/ItaniumDemangle.h b/llvm/include/llvm/Demangle/ItaniumDemangle.h
index a398e778b037a..5953a7f8c6135 100644
--- a/llvm/include/llvm/Demangle/ItaniumDemangle.h
+++ b/llvm/include/llvm/Demangle/ItaniumDemangle.h
@@ -1725,6 +1725,10 @@ public:
   }
 };
 
+#ifdef SS
+#undef SS
+#endif
+
 inline ExpandedSpecialSubstitution::ExpandedSpecialSubstitution(
     SpecialSubstitution const *SS)
     : ExpandedSpecialSubstitution(SS->SSK) {}
