diff -u -r ../orig-src/llvm/lib/Support/Unix/Path.inc ./llvm/lib/Support/Unix/Path.inc
--- ../orig-src/llvm/lib/Support/Unix/Path.inc	2023-09-29 21:36:47.056335287 +0000
+++ ./llvm/lib/Support/Unix/Path.inc	2023-09-29 21:40:31.091417185 +0000
@@ -250,6 +250,17 @@
   // If we don't have procfs mounted, fall back to argv[0]
   if (getprogpath(exe_path, argv0) != NULL)
     return exe_path;
+#elif defined(__ANDROID__) // termux-exec linker wrapping does not work with /proc/self/exe
+  const char *termux_self_exe = std::getenv("@TERMUX_ENV__S_TERMUX_EXEC@PROC_SELF_EXE");
+  if (termux_self_exe != NULL) return std::string(termux_self_exe);
+  if (char *real_path = realpath("/proc/self/exe", nullptr)) {
+    std::string ret = std::string(real_path);
+    free(real_path);
+    return ret;
+  }
+  // Fall back to the classical detection.
+  char exe_path[PATH_MAX];
+  if (getprogpath(exe_path, argv0)) return exe_path;
 #elif defined(__linux__) || defined(__CYGWIN__) || defined(__gnu_hurd__)
   char exe_path[PATH_MAX];
   const char *aPath = "/proc/self/exe";
