diff -u -r ../llvm-3.8.0.src/tools/clang/lib/Driver/ToolChain.cpp ./tools/clang/lib/Driver/ToolChain.cpp
--- ../llvm-3.8.0.src/tools/clang/lib/Driver/ToolChain.cpp	2015-11-25 20:02:07.000000000 -0500
+++ ./tools/clang/lib/Driver/ToolChain.cpp	2016-05-02 08:50:35.526226962 -0400
@@ -611,7 +611,7 @@
     break;
 
   case ToolChain::CST_Libstdcxx:
-    CmdArgs.push_back("-lstdc++");
+    CmdArgs.push_back("-lgnustl_shared");
     break;
   }
 }
