diff -u -r ../pastebinit-1.5/pastebinit ./pastebinit
--- ../pastebinit-1.5/pastebinit	2016-03-01 00:52:32.000000000 -0500
+++ ./pastebinit	2016-08-06 16:37:25.632592458 -0400
@@ -86,8 +86,8 @@
         #  - user's overrides in ~/.pastebin.d
         # Files found later override files found earlier.
         pastebind = {}
-        for confdir in ['/usr/share/pastebin.d', '/etc/pastebin.d',
-                        '/usr/local/etc/pastebin.d',
+        for confdir in ['@TERMUX_PREFIX@/share/pastebin.d', '@TERMUX_PREFIX@/etc/pastebin.d',
+                        '@TERMUX_PREFIX@/local/etc/pastebin.d',
                         os.path.expanduser('~/.pastebin.d'),
                         os.path.join(
                             os.path.dirname(
