diff -u -r ../mosh-f30738e3256e90850e945c08624fce90b1ba78a1/src/frontend/Makefile.am ./src/frontend/Makefile.am
--- ../mosh-f30738e3256e90850e945c08624fce90b1ba78a1/src/frontend/Makefile.am	2016-04-07 20:26:08.000000000 -0400
+++ ./src/frontend/Makefile.am	2016-05-02 01:20:09.868867123 -0400
@@ -11,9 +11,15 @@
   bin_PROGRAMS += mosh-client
 endif
 
+if BUILD_CFRONT
+  bin_PROGRAMS += mosh-cfront
+endif
+
 if BUILD_SERVER
   bin_PROGRAMS += mosh-server
 endif
 
 mosh_client_SOURCES = mosh-client.cc stmclient.cc stmclient.h terminaloverlay.cc terminaloverlay.h
 mosh_server_SOURCES = mosh-server.cc
+mosh_cfront_SOURCES = mosh.cc
+mosh_cfront_LDADD = -lutil
