diff -u -r ../gnucap-20171003/lib/c_attach.cc ./lib/c_attach.cc
--- ../gnucap-20171003/lib/c_attach.cc	2017-10-04 00:50:49.000000000 +0200
+++ ./lib/c_attach.cc	2019-01-24 22:30:16.059909732 +0100
@@ -36,7 +36,7 @@
 /*--------------------------------------------------------------------------*/
 void list()
 {
-  for (std::map<std::string, void*>::iterator
+  for (std::map<const std::string, void*>::iterator
 	 ii = attach_list.begin(); ii != attach_list.end(); ++ii) {
     if (ii->second) {
       IO::mstdout << ii->first << '\n';
@@ -179,7 +179,7 @@
   void do_it(CS& cmd, CARD_LIST*)
   {
     if (CARD_LIST::card_list.is_empty()) {
-      for (std::map<std::string, void*>::iterator
+      for (std::map<const std::string, void*>::iterator
 	     ii = attach_list.begin(); ii != attach_list.end(); ++ii) {
 	void* handle = ii->second;
 	if (handle) {
