--- a/mt/Makefile.orig	2021-08-19 17:51:16 UTC
+++ b/mt/Makefile
@@ -7,7 +7,7 @@ include		$(SRCROOT)/$(RULESDIR)/rules.top
 
 INSDIR=		bin
 TARGET=		smt
-SYMLINKS=	mt
+#SYMLINKS=	mt
 CPPOPTS +=	-DUSE_REMOTE
 CPPOPTS +=	-DSCHILY_PRINT
 CPPOPTS +=	-DUSE_NLS		# Include locale support
@@ -21,7 +21,7 @@ CFILES=		mt.c
 # LIB_CAP is needed for Linux capability support in librmt.
 #
 LIBS=		-lrmt -lschily $(LIB_SOCKET) $(LIB_CAP) $(LIB_INTL)
-XMK_FILE=	Makefile.man mt.mk1
+XMK_FILE=	Makefile.man
 
 ###########################################################################
 include		$(SRCROOT)/$(RULESDIR)/rules.cmd
