--- a/rmt/Makefile.doc.orig	2021-06-04 16:12:30 UTC
+++ b/rmt/Makefile.doc
@@ -7,8 +7,8 @@ RULESDIR=	RULES
 include		$(SRCROOT)/$(RULESDIR)/rules.top
 ###########################################################################
 
-INSDIR=		share/doc/rmt
-TARGET=		default-rmt.sample
+INSDIR=		share/doc/srmt
+TARGET=		default-srmt.sample
 #XMK_FILE=	Makefile.man
 
 ###########################################################################
