--- a/sh/tests/common/test-common.orig	2022-08-19 19:43:40 UTC
+++ b/sh/tests/common/test-common
@@ -19,10 +19,10 @@ powerpc-aix-*)
 		;;
 esac
 
-if [ -f ../../../OBJ/$OARCH/sh ]; then
-	shell=${shell:=../../../OBJ/$OARCH/sh}
-elif [ -f ../../OBJ/$OARCH/sh ]; then
-	shell=${shell:=../../OBJ/$OARCH/sh}
+if [ -f ../../../OBJ/$OARCH/bosh ]; then
+	shell=${shell:=../../../OBJ/$OARCH/bosh}
+elif [ -f ../../OBJ/$OARCH/bosh ]; then
+	shell=${shell:=../../OBJ/$OARCH/bosh}
 fi
 SHELL="$shell"
 export SHELL
