diff --git a/lib/stdio-impl.h b/lib/stdio-impl.h
index 3fa94b4..2be01f3 100644
--- a/lib/stdio-impl.h
+++ b/lib/stdio-impl.h
@@ -77,6 +77,7 @@
      see <https://android.googlesource.com/platform/bionic.git/+/e78392637d5086384a5631ddfdfa8d7ec8326ee3/libc/stdio/fileext.h>
      and <https://android.googlesource.com/platform/bionic.git/+/e78392637d5086384a5631ddfdfa8d7ec8326ee3/libc/stdio/local.h>.
      After this commit, the innards of FILE are hidden.  */
+#if 0
 #  define fp_ ((struct { unsigned char *_p; \
                          int _r; \
                          int _w; \
@@ -99,7 +100,7 @@
                          fpos_t _offset; \
                          /* More fields, not relevant here.  */ \
                        } *) fp)
-# else
+#endif
 #  define fp_ fp
 # endif
 
