--- a/libgtop-sysdeps.m4
+++ b/libgtop-sysdeps.m4
@@ -33,10 +33,9 @@
 	linux*)
 	  libgtop_sysdeps_dir=linux
 	  libgtop_have_sysinfo=yes
-	  libgtop_need_server=yes
+	  libgtop_need_server=no
 	  libgtop_sysdeps_private_mountlist=yes
 	  libgtop_sysdeps_private_fsusage=yes
-	  libgtop_postinstall='chown root $(DESTDIR)$(libexecdir)/libgtop_server2 && chmod 4755 $(DESTDIR)$(libexecdir)/libgtop_server2'
 	  ;;
 	netbsd*|bsdi*)
 	  libgtop_sysdeps_dir=bsd
