--- ./src/net/if_.rs.orig	2025-03-31 16:02:54.006271135 +0530
+++ ./src/net/if_.rs	2025-03-31 16:03:54.991753288 +0530
@@ -270,6 +270,7 @@
     target_os = "fuchsia",
     target_os = "linux",
     solarish,
+    linux_android
 ))]
 mod if_nameindex {
     use super::*;
@@ -397,5 +397,6 @@
     target_os = "fuchsia",
     target_os = "linux",
     solarish,
+    linux_android
 ))]
 pub use if_nameindex::*;
