diff -uNr SDL2_mixer-2.0.4/timidity/options.h SDL2_mixer-2.0.4.mod/timidity/options.h
--- SDL2_mixer-2.0.4/timidity/options.h	2018-10-31 16:59:00.000000000 +0200
+++ SDL2_mixer-2.0.4.mod/timidity/options.h	2020-03-08 23:34:10.627495780 +0200
@@ -72,16 +72,16 @@
 /* You could specify a complete path, e.g. "/etc/timidity.cfg", and
    then specify the library directory in the configuration file. */
 #define CONFIG_FILE	"timidity.cfg"
-#define CONFIG_FILE_ETC "/etc/timidity.cfg"
-#define CONFIG_FILE_ETC_TIMIDITY_FREEPATS "/etc/timidity/freepats.cfg"
+#define CONFIG_FILE_ETC "@TERMUX_PREFIX@/etc/timidity.cfg"
+#define CONFIG_FILE_ETC_TIMIDITY_FREEPATS "@TERMUX_PREFIX@/etc/timidity/freepats.cfg"
 
 #if defined(__WIN32__) || defined(__OS2__)
 #define DEFAULT_PATH	"C:\\TIMIDITY"
 #else
-#define DEFAULT_PATH	"/etc/timidity"
-#define DEFAULT_PATH1	"/usr/share/timidity"
-#define DEFAULT_PATH2	"/usr/local/share/timidity"
-#define DEFAULT_PATH3	"/usr/local/lib/timidity"
+#define DEFAULT_PATH	"@TERMUX_PREFIX@/etc/timidity"
+#define DEFAULT_PATH1	"@TERMUX_PREFIX@/share/timidity"
+#define DEFAULT_PATH2	"@TERMUX_PREFIX@/local/share/timidity"
+#define DEFAULT_PATH3	"@TERMUX_PREFIX@/local/lib/timidity"
 #endif
 
 /* These affect general volume */
