From b6e3e12e53a4083fbbe8e6a444d1eccf9a887790 Mon Sep 17 00:00:00 2001
From: Patricio Gonzalez Vivo <patriciogonzalezvivo@gmail.com>
Date: Sun, 17 Dec 2023 16:19:40 -0500
Subject: [PATCH] suspending exiv2 lib

---
 src/CMakeLists.txt | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

--- a/deps/vera/src/CMakeLists.txt
+++ b/deps/vera/src/CMakeLists.txt
@@ -161,13 +161,13 @@ else()
     endif()
 endif()
 
-find_package(exiv2lib CONFIG NAMES exiv2)
-if (exiv2lib_FOUND)
-    link_libraries(exiv2lib)
-    target_compile_definitions(vera PUBLIC SUPPORT_EXIV2)
-    target_link_libraries(vera PRIVATE exiv2lib)
-    message(STATUS "EXIV2 FOUNDED")
-endif()
+# find_package(exiv2lib CONFIG NAMES exiv2)
+# if (exiv2lib_FOUND)
+#     link_libraries(exiv2lib)
+#     target_compile_definitions(vera PUBLIC SUPPORT_EXIV2)
+#     target_link_libraries(vera PRIVATE exiv2lib)
+#     message(STATUS "EXIV2 FOUNDED")
+# endif()
 
 if (NO_X11 OR FORCE_GBM)
     if (BROADCOM_FOUND AND NOT FORCE_GBM)
