diff -uNr bind-9.16.8/configure bind-9.16.8.mod/configure
--- bind-9.16.8/configure	2020-10-13 11:41:40.000000000 +0300
+++ bind-9.16.8.mod/configure	2020-11-09 01:10:30.546434900 +0200
@@ -22047,7 +22047,7 @@
 
 
 LDFLAGS="-Wl,--wrap,exit"
-if test "$cross_compiling" = yes; then :
+if test "$cross_compiling" = fakeyes; then :
   { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
 $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
 as_fn_error $? "cannot run test program while cross compiling
@@ -22067,7 +22067,7 @@
   return 0;
 }
 _ACEOF
-if ac_fn_c_try_run "$LINENO"; then :
+if true; then :
   LD_WRAP_TESTS=true
 
 $as_echo "#define LD_WRAP 1" >>confdefs.h
