diff -u -r ../bind-9.11.0-P5/configure ./configure
--- ../bind-9.11.0-P5/configure	2017-03-30 00:03:33.000000000 +0200
+++ ./configure	2017-05-23 00:32:54.924640433 +0200
@@ -19082,7 +19082,7 @@
 no)	;;
 yes|auto)
 	saved_LIBS="$LIBS"
-	for readline in -ledit -lreadline
+	for readline in -lreadline
 	do
 		LIBS="$readline"
 		{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for readline with $readline" >&5
