https://bugs.gentoo.org/703650

diff -u -r ../bind-9.15.7/lib/isc/Makefile.in ./lib/isc/Makefile.in
--- ../bind-9.15.7/lib/isc/Makefile.in	2019-12-13 00:00:45.000000000 +0000
+++ ./lib/isc/Makefile.in	2019-12-29 23:25:03.422518000 +0000
@@ -82,7 +82,7 @@
 		symtab.c task.c taskpool.c timer.c \
 		tm.c version.c
 
-LIBS =		${OPENSSL_LIBS} @LIBS@
+LIBS =		${ZLIB_LIBS} ${OPENSSL_LIBS} @LIBS@
 
 # Note: the order of SUBDIRS is important.
 # Attempt to disable parallel processing.
