diff -uNr bind-9.16.0/lib/isc/pthreads/thread.c bind-9.16.0.mod/lib/isc/pthreads/thread.c
--- bind-9.16.0/lib/isc/pthreads/thread.c	2020-02-12 22:03:44.000000000 +0200
+++ bind-9.16.0.mod/lib/isc/pthreads/thread.c	2020-03-08 22:29:34.847302635 +0200
@@ -95,7 +95,9 @@
 void
 isc_thread_setconcurrency(unsigned int level)
 {
+#ifndef __ANDROID__
 	(void)pthread_setconcurrency(level);
+#endif
 }
 
 void
