diff -u -r ../bind-9.11.1-P3/lib/lwres/man/Makefile.in ./lib/lwres/man/Makefile.in
--- ../bind-9.11.1-P3/lib/lwres/man/Makefile.in	2017-07-07 17:01:52.000000000 +0200
+++ ./lib/lwres/man/Makefile.in	2017-07-26 13:13:17.575587105 +0200
@@ -92,141 +92,141 @@
 install:: installdirs
 	for m in ${MANPAGES}; do ${INSTALL_DATA} ${srcdir}/$$m ${DESTDIR}${mandir}/man3; done
 	rm -f ${man3}/lwres_addr_parse.3
-	@LN@ ${man3}/lwres_resutil.3 ${man3}/lwres_addr_parse.3
+	@LN@ -s ${man3}/lwres_resutil.3 ${man3}/lwres_addr_parse.3
 	rm -f ${man3}/lwres_buffer_add.3
-	@LN@ ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_add.3
+	@LN@ -s ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_add.3
 	rm -f ${man3}/lwres_buffer_back.3
-	@LN@ ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_back.3
+	@LN@ -s ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_back.3
 	rm -f ${man3}/lwres_buffer_clear.3
-	@LN@ ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_clear.3
+	@LN@ -s ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_clear.3
 	rm -f ${man3}/lwres_buffer_first.3
-	@LN@ ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_first.3
+	@LN@ -s ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_first.3
 	rm -f ${man3}/lwres_buffer_forward.3
-	@LN@ ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_forward.3
+	@LN@ -s ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_forward.3
 	rm -f ${man3}/lwres_buffer_getmem.3
-	@LN@ ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_getmem.3
+	@LN@ -s ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_getmem.3
 	rm -f ${man3}/lwres_buffer_getuint16.3
-	@LN@ ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_getuint16.3
+	@LN@ -s ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_getuint16.3
 	rm -f ${man3}/lwres_buffer_getuint32.3
-	@LN@ ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_getuint32.3
+	@LN@ -s ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_getuint32.3
 	rm -f ${man3}/lwres_buffer_getuint8.3
-	@LN@ ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_getuint8.3
+	@LN@ -s ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_getuint8.3
 	rm -f ${man3}/lwres_buffer_init.3
-	@LN@ ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_init.3
+	@LN@ -s ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_init.3
 	rm -f ${man3}/lwres_buffer_invalidate.3
-	@LN@ ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_invalidate.3
+	@LN@ -s ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_invalidate.3
 	rm -f ${man3}/lwres_buffer_putmem.3
-	@LN@ ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_putmem.3
+	@LN@ -s ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_putmem.3
 	rm -f ${man3}/lwres_buffer_putuint16.3
-	@LN@ ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_putuint16.3
+	@LN@ -s ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_putuint16.3
 	rm -f ${man3}/lwres_buffer_putuint32.3
-	@LN@ ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_putuint32.3
+	@LN@ -s ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_putuint32.3
 	rm -f ${man3}/lwres_buffer_putuint8.3
-	@LN@ ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_putuint8.3
+	@LN@ -s ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_putuint8.3
 	rm -f ${man3}/lwres_buffer_subtract.3
-	@LN@ ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_subtract.3
+	@LN@ -s ${man3}/lwres_buffer.3 ${man3}/lwres_buffer_subtract.3
 	rm -f ${man3}/lwres_conf_clear.3
-	@LN@ ${man3}/lwres_config.3 ${man3}/lwres_conf_clear.3
+	@LN@ -s ${man3}/lwres_config.3 ${man3}/lwres_conf_clear.3
 	rm -f ${man3}/lwres_conf_get.3
-	@LN@ ${man3}/lwres_config.3 ${man3}/lwres_conf_get.3
+	@LN@ -s ${man3}/lwres_config.3 ${man3}/lwres_conf_get.3
 	rm -f ${man3}/lwres_conf_init.3
-	@LN@ ${man3}/lwres_config.3 ${man3}/lwres_conf_init.3
+	@LN@ -s ${man3}/lwres_config.3 ${man3}/lwres_conf_init.3
 	rm -f ${man3}/lwres_conf_parse.3
-	@LN@ ${man3}/lwres_config.3 ${man3}/lwres_conf_parse.3
+	@LN@ -s ${man3}/lwres_config.3 ${man3}/lwres_conf_parse.3
 	rm -f ${man3}/lwres_conf_print.3
-	@LN@ ${man3}/lwres_config.3 ${man3}/lwres_conf_print.3
+	@LN@ -s ${man3}/lwres_config.3 ${man3}/lwres_conf_print.3
 	rm -f ${man3}/lwres_context_allocmem.3
-	@LN@ ${man3}/lwres_context.3 ${man3}/lwres_context_allocmem.3
+	@LN@ -s ${man3}/lwres_context.3 ${man3}/lwres_context_allocmem.3
 	rm -f ${man3}/lwres_context_create.3
-	@LN@ ${man3}/lwres_context.3 ${man3}/lwres_context_create.3
+	@LN@ -s ${man3}/lwres_context.3 ${man3}/lwres_context_create.3
 	rm -f ${man3}/lwres_context_destroy.3
-	@LN@ ${man3}/lwres_context.3 ${man3}/lwres_context_destroy.3
+	@LN@ -s ${man3}/lwres_context.3 ${man3}/lwres_context_destroy.3
 	rm -f ${man3}/lwres_context_freemem.3
-	@LN@ ${man3}/lwres_context.3 ${man3}/lwres_context_freemem.3
+	@LN@ -s ${man3}/lwres_context.3 ${man3}/lwres_context_freemem.3
 	rm -f ${man3}/lwres_context_initserial.3
-	@LN@ ${man3}/lwres_context.3 ${man3}/lwres_context_initserial.3
+	@LN@ -s ${man3}/lwres_context.3 ${man3}/lwres_context_initserial.3
 	rm -f ${man3}/lwres_context_nextserial.3
-	@LN@ ${man3}/lwres_context.3 ${man3}/lwres_context_nextserial.3
+	@LN@ -s ${man3}/lwres_context.3 ${man3}/lwres_context_nextserial.3
 	rm -f ${man3}/lwres_context_sendrecv.3
-	@LN@ ${man3}/lwres_context.3 ${man3}/lwres_context_sendrecv.3
+	@LN@ -s ${man3}/lwres_context.3 ${man3}/lwres_context_sendrecv.3
 	rm -f ${man3}/lwres_endhostent.3
-	@LN@ ${man3}/lwres_gethostent.3 ${man3}/lwres_endhostent.3
+	@LN@ -s ${man3}/lwres_gethostent.3 ${man3}/lwres_endhostent.3
 	rm -f ${man3}/lwres_endhostent_r.3
-	@LN@ ${man3}/lwres_gethostent.3 ${man3}/lwres_endhostent_r.3
+	@LN@ -s ${man3}/lwres_gethostent.3 ${man3}/lwres_endhostent_r.3
 	rm -f ${man3}/lwres_freeaddrinfo.3
-	@LN@ ${man3}/lwres_getaddrinfo.3 ${man3}/lwres_freeaddrinfo.3
+	@LN@ -s ${man3}/lwres_getaddrinfo.3 ${man3}/lwres_freeaddrinfo.3
 	rm -f ${man3}/lwres_freehostent.3
-	@LN@ ${man3}/lwres_getipnode.3 ${man3}/lwres_freehostent.3
+	@LN@ -s ${man3}/lwres_getipnode.3 ${man3}/lwres_freehostent.3
 	rm -f ${man3}/lwres_gabnrequest_free.3
-	@LN@ ${man3}/lwres_gabn.3 ${man3}/lwres_gabnrequest_free.3
+	@LN@ -s ${man3}/lwres_gabn.3 ${man3}/lwres_gabnrequest_free.3
 	rm -f ${man3}/lwres_gabnrequest_parse.3
-	@LN@ ${man3}/lwres_gabn.3 ${man3}/lwres_gabnrequest_parse.3
+	@LN@ -s ${man3}/lwres_gabn.3 ${man3}/lwres_gabnrequest_parse.3
 	rm -f ${man3}/lwres_gabnrequest_render.3
-	@LN@ ${man3}/lwres_gabn.3 ${man3}/lwres_gabnrequest_render.3
+	@LN@ -s ${man3}/lwres_gabn.3 ${man3}/lwres_gabnrequest_render.3
 	rm -f ${man3}/lwres_gabnresponse_free.3
-	@LN@ ${man3}/lwres_gabn.3 ${man3}/lwres_gabnresponse_free.3
+	@LN@ -s ${man3}/lwres_gabn.3 ${man3}/lwres_gabnresponse_free.3
 	rm -f ${man3}/lwres_gabnresponse_parse.3
-	@LN@ ${man3}/lwres_gabn.3 ${man3}/lwres_gabnresponse_parse.3
+	@LN@ -s ${man3}/lwres_gabn.3 ${man3}/lwres_gabnresponse_parse.3
 	rm -f ${man3}/lwres_gabnresponse_render.3
-	@LN@ ${man3}/lwres_gabn.3 ${man3}/lwres_gabnresponse_render.3
+	@LN@ -s ${man3}/lwres_gabn.3 ${man3}/lwres_gabnresponse_render.3
 	rm -f ${man3}/lwres_getaddrsbyname.3
-	@LN@ ${man3}/lwres_resutil.3 ${man3}/lwres_getaddrsbyname.3
+	@LN@ -s ${man3}/lwres_resutil.3 ${man3}/lwres_getaddrsbyname.3
 	rm -f ${man3}/lwres_gethostbyaddr.3
-	@LN@ ${man3}/lwres_gethostent.3 ${man3}/lwres_gethostbyaddr.3
+	@LN@ -s ${man3}/lwres_gethostent.3 ${man3}/lwres_gethostbyaddr.3
 	rm -f ${man3}/lwres_gethostbyaddr_r.3
-	@LN@ ${man3}/lwres_gethostent.3 ${man3}/lwres_gethostbyaddr_r.3
+	@LN@ -s ${man3}/lwres_gethostent.3 ${man3}/lwres_gethostbyaddr_r.3
 	rm -f ${man3}/lwres_gethostbyname.3
-	@LN@ ${man3}/lwres_gethostent.3 ${man3}/lwres_gethostbyname.3
+	@LN@ -s ${man3}/lwres_gethostent.3 ${man3}/lwres_gethostbyname.3
 	rm -f ${man3}/lwres_gethostbyname2.3
-	@LN@ ${man3}/lwres_gethostent.3 ${man3}/lwres_gethostbyname2.3
+	@LN@ -s ${man3}/lwres_gethostent.3 ${man3}/lwres_gethostbyname2.3
 	rm -f ${man3}/lwres_gethostbyname_r.3
-	@LN@ ${man3}/lwres_gethostent.3 ${man3}/lwres_gethostbyname_r.3
+	@LN@ -s ${man3}/lwres_gethostent.3 ${man3}/lwres_gethostbyname_r.3
 	rm -f ${man3}/lwres_gethostent_r.3
-	@LN@ ${man3}/lwres_gethostent.3 ${man3}/lwres_gethostent_r.3
+	@LN@ -s ${man3}/lwres_gethostent.3 ${man3}/lwres_gethostent_r.3
 	rm -f ${man3}/lwres_getipnodebyaddr.3
-	@LN@ ${man3}/lwres_getipnode.3 ${man3}/lwres_getipnodebyaddr.3
+	@LN@ -s ${man3}/lwres_getipnode.3 ${man3}/lwres_getipnodebyaddr.3
 	rm -f ${man3}/lwres_getipnodebyname.3
-	@LN@ ${man3}/lwres_getipnode.3 ${man3}/lwres_getipnodebyname.3
+	@LN@ -s ${man3}/lwres_getipnode.3 ${man3}/lwres_getipnodebyname.3
 	rm -f ${man3}/lwres_getnamebyaddr.3
-	@LN@ ${man3}/lwres_resutil.3 ${man3}/lwres_getnamebyaddr.3
+	@LN@ -s ${man3}/lwres_resutil.3 ${man3}/lwres_getnamebyaddr.3
 	rm -f ${man3}/lwres_gnbarequest_free.3
-	@LN@ ${man3}/lwres_gnba.3 ${man3}/lwres_gnbarequest_free.3
+	@LN@ -s ${man3}/lwres_gnba.3 ${man3}/lwres_gnbarequest_free.3
 	rm -f ${man3}/lwres_gnbarequest_parse.3
-	@LN@ ${man3}/lwres_gnba.3 ${man3}/lwres_gnbarequest_parse.3
+	@LN@ -s ${man3}/lwres_gnba.3 ${man3}/lwres_gnbarequest_parse.3
 	rm -f ${man3}/lwres_gnbarequest_render.3
-	@LN@ ${man3}/lwres_gnba.3 ${man3}/lwres_gnbarequest_render.3
+	@LN@ -s ${man3}/lwres_gnba.3 ${man3}/lwres_gnbarequest_render.3
 	rm -f ${man3}/lwres_gnbaresponse_free.3
-	@LN@ ${man3}/lwres_gnba.3 ${man3}/lwres_gnbaresponse_free.3
+	@LN@ -s ${man3}/lwres_gnba.3 ${man3}/lwres_gnbaresponse_free.3
 	rm -f ${man3}/lwres_gnbaresponse_parse.3
-	@LN@ ${man3}/lwres_gnba.3 ${man3}/lwres_gnbaresponse_parse.3
+	@LN@ -s ${man3}/lwres_gnba.3 ${man3}/lwres_gnbaresponse_parse.3
 	rm -f ${man3}/lwres_gnbaresponse_render.3
-	@LN@ ${man3}/lwres_gnba.3 ${man3}/lwres_gnbaresponse_render.3
+	@LN@ -s ${man3}/lwres_gnba.3 ${man3}/lwres_gnbaresponse_render.3
 	rm -f ${man3}/lwres_herror.3
-	@LN@ ${man3}/lwres_hstrerror.3 ${man3}/lwres_herror.3
+	@LN@ -s ${man3}/lwres_hstrerror.3 ${man3}/lwres_herror.3
 	rm -f ${man3}/lwres_lwpacket_parseheader.3
-	@LN@ ${man3}/lwres_packet.3 ${man3}/lwres_lwpacket_parseheader.3
+	@LN@ -s ${man3}/lwres_packet.3 ${man3}/lwres_lwpacket_parseheader.3
 	rm -f ${man3}/lwres_lwpacket_renderheader.3
-	@LN@ ${man3}/lwres_packet.3 ${man3}/lwres_lwpacket_renderheader.3
+	@LN@ -s ${man3}/lwres_packet.3 ${man3}/lwres_lwpacket_renderheader.3
 	rm -f ${man3}/lwres_net_ntop.3
-	@LN@ ${man3}/lwres_inetntop.3 ${man3}/lwres_net_ntop.3
+	@LN@ -s ${man3}/lwres_inetntop.3 ${man3}/lwres_net_ntop.3
 	rm -f ${man3}/lwres_nooprequest_free.3
-	@LN@ ${man3}/lwres_noop.3 ${man3}/lwres_nooprequest_free.3
+	@LN@ -s ${man3}/lwres_noop.3 ${man3}/lwres_nooprequest_free.3
 	rm -f ${man3}/lwres_nooprequest_parse.3
-	@LN@ ${man3}/lwres_noop.3 ${man3}/lwres_nooprequest_parse.3
+	@LN@ -s ${man3}/lwres_noop.3 ${man3}/lwres_nooprequest_parse.3
 	rm -f ${man3}/lwres_nooprequest_render.3
-	@LN@ ${man3}/lwres_noop.3 ${man3}/lwres_nooprequest_render.3
+	@LN@ -s ${man3}/lwres_noop.3 ${man3}/lwres_nooprequest_render.3
 	rm -f ${man3}/lwres_noopresponse_free.3
-	@LN@ ${man3}/lwres_noop.3 ${man3}/lwres_noopresponse_free.3
+	@LN@ -s ${man3}/lwres_noop.3 ${man3}/lwres_noopresponse_free.3
 	rm -f ${man3}/lwres_noopresponse_parse.3
-	@LN@ ${man3}/lwres_noop.3 ${man3}/lwres_noopresponse_parse.3
+	@LN@ -s ${man3}/lwres_noop.3 ${man3}/lwres_noopresponse_parse.3
 	rm -f ${man3}/lwres_noopresponse_render.3
-	@LN@ ${man3}/lwres_noop.3 ${man3}/lwres_noopresponse_render.3
+	@LN@ -s ${man3}/lwres_noop.3 ${man3}/lwres_noopresponse_render.3
 	rm -f ${man3}/lwres_sethostent.3
-	@LN@ ${man3}/lwres_gethostent.3 ${man3}/lwres_sethostent.3
+	@LN@ -s ${man3}/lwres_gethostent.3 ${man3}/lwres_sethostent.3
 	rm -f ${man3}/lwres_sethostent_r.3
-	@LN@ ${man3}/lwres_gethostent.3 ${man3}/lwres_sethostent_r.3
+	@LN@ -s ${man3}/lwres_gethostent.3 ${man3}/lwres_sethostent_r.3
 	rm -f ${man3}/lwres_string_parse.3
-	@LN@ ${man3}/lwres_resutil.3 ${man3}/lwres_string_parse.3
+	@LN@ -s ${man3}/lwres_resutil.3 ${man3}/lwres_string_parse.3
 
 uninstall::
 	for m in ${MANPAGES}; do rm -f ${man3}/$$m ; done
