diff -uNr bind-9.13.6/lib/irs/context.c bind-9.13.6.mod/lib/irs/context.c
--- bind-9.13.6/lib/irs/context.c	2019-02-07 01:44:09.000000000 +0200
+++ bind-9.13.6.mod/lib/irs/context.c	2019-03-02 02:39:12.188552664 +0200
@@ -42,7 +42,7 @@
 
 #ifndef DNS_CONF
 /*% location of dns.conf */
-#define DNS_CONF "/etc/dns.conf"
+#define DNS_CONF "@TERMUX_PREFIX@/etc/dns.conf"
 #endif
 
 static bool thread_key_initialized = false;
