diff -uNr bind-9.11.1-P3/lib/lwres/include/lwres/lwres.h bind-9.11.1-P3.mod/lib/lwres/include/lwres/lwres.h
--- bind-9.11.1-P3/lib/lwres/include/lwres/lwres.h	2017-07-07 18:01:52.000000000 +0300
+++ bind-9.11.1-P3.mod/lib/lwres/include/lwres/lwres.h	2017-10-04 16:48:56.466109542 +0300
@@ -74,7 +74,7 @@
 #define LWRES_UDP_PORT		921	/*%< UDP Port Number */
 #define LWRES_RECVLENGTH	16384 /*%< Maximum Packet Length */
 #define LWRES_ADDR_MAXLEN	16	/*%< changing this breaks ABI */
-#define LWRES_RESOLV_CONF	"/etc/resolv.conf" /*%< Location of resolv.conf */
+#define LWRES_RESOLV_CONF	"@TERMUX_PREFIX@/etc/resolv.conf" /*%< Location of resolv.conf */
 
 /*% DNSSEC is not required (input).  Only relevant to rrset queries. */
 #define LWRES_FLAG_TRUSTNOTREQUIRED	0x00000001U
