diff -uNr ntfs-3g-2022.10.3/src/Makefile.am ntfs-3g-2022.10.3.mod/src/Makefile.am
--- ntfs-3g-2022.10.3/src/Makefile.am	2022-10-03 17:10:36.000000000 +0800
+++ ntfs-3g-2022.10.3.mod/src/Makefile.am	2022-11-26 16:38:21.634557533 +0800
@@ -67,8 +67,8 @@
 if ENABLE_MOUNT_HELPER
 install-exec-local:	install-rootbinPROGRAMS
 	$(MKDIR_P) "$(DESTDIR)/sbin"
-	$(LN_S) -f "$(rootbindir)/ntfs-3g" "$(DESTDIR)/sbin/mount.ntfs-3g"
-	$(LN_S) -f "$(rootbindir)/lowntfs-3g" "$(DESTDIR)/sbin/mount.lowntfs-3g"
+	$(LN_S) -f "$(DESTDIR)$(rootbindir)/ntfs-3g" "$(DESTDIR)/bin/mount.ntfs-3g"
+	$(LN_S) -f "$(DESTDIR)$(rootbindir)/lowntfs-3g" "$(DESTDIR)/bin/mount.lowntfs-3g"
 
 install-data-local:	install-man8
 	$(LN_S) -f ntfs-3g.8 "$(DESTDIR)$(man8dir)/mount.ntfs-3g.8"
