--- a/far2l/src/cfg/ConfigOpt.cpp
+++ b/far2l/src/cfg/ConfigOpt.cpp
@@ -226,7 +226,7 @@
 	{OST_COMMON, NSecInterface, "DecimalSeparator", &Opt.strDecimalSeparator, GetDecimalSeparatorDefaultStr(),
 		L"InterfSettings", L"The character or string used as the decimal separator" },
 
-	{OST_COMMON, NSecInterface, "OSC52ClipSet", &Opt.OSC52ClipSet, 0,
+	{OST_COMMON, NSecInterface, "OSC52ClipSet", &Opt.OSC52ClipSet, 1,
 		L"InterfSettings", L"Set the local clipboard using OSC52 escape sequences (TTY-backend only)" },
 	{OST_COMMON, NSecInterface, "TTYPaletteOverride", &Opt.TTYPaletteOverride, 1,
 		L"InterfSettings", L"Override the terminal palette (TTY-backend only)" },
@@ -418,7 +418,7 @@
 		L"SystemSettings", L"Controls how duplicate history entries are removed: 0=never remove duplicates, 1=remove by name, 2=remove by name and path" },
 	{OST_COMMON, NSecSystem, "AutoHighlightHistory", &Opt.AutoHighlightHistory, 1,
 		L"SystemSettings", L"Highlights matching entries automatically in history lists" },
-	{OST_COMMON, NSecSystem, NParamAutoSaveSetup, &Opt.AutoSaveSetup, 0,
+	{OST_COMMON, NSecSystem, NParamAutoSaveSetup, &Opt.AutoSaveSetup, 1,
 		L"SystemSettings", L"Automatically saves all (common and panels) setup options" },
 	{OST_COMMON, NSecSystem, NParamAutoSavePanels, &Opt.AutoSavePanels, 0,
 		L"SystemSettings", L"Automatically saves panel state" },
