diff --git a/NetRocks/src/Protocol/SSH/SSHConnection.cpp b/NetRocks/src/Protocol/SSH/SSHConnection.cpp
index fb1a806..e0f47eb 100644
--- a/NetRocks/src/Protocol/SSH/SSHConnection.cpp
+++ b/NetRocks/src/Protocol/SSH/SSHConnection.cpp
@@ -3,6 +3,7 @@
 #include <sys/ioctl.h>
 #include <time.h>
 #include <unistd.h>
+#include <netinet/in.h>
 #include <netinet/tcp.h>
 #include <fcntl.h>
 #include <signal.h>
