--- a/cmake/Config.cmake
+++ b/cmake/Config.cmake
@@ -21,10 +21,10 @@
         message(FATAL_ERROR "Unsupported architecture")
         return()
     endif()
-elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
+elseif(TRUE)
     set(SFML_OS_UNIX 1)
 
-    if(ANDROID)
+    if(FALSE)
         set(SFML_OS_ANDROID 1)
 
         # use the OpenGL ES implementation on Android
@@ -64,7 +64,7 @@
     set(SFML_OS_MACOSX 1)
     # don't use the OpenGL ES implementation on Mac OS X
     set(OPENGL_ES 0)
-elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Android")
+elseif(FALSE)
     set(SFML_OS_ANDROID 1)
     # use the OpenGL ES implementation on Android
     set(OPENGL_ES 1)
