--- a/cmake/Config.cmake
+++ b/cmake/Config.cmake
@@ -16,9 +16,9 @@
         message(FATAL_ERROR "Unsupported architecture")
         return()
     endif()
-elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
+elseif(TRUE)
     set(SFML_OS_UNIX 1)
-    if(ANDROID)
+    if(FALSE)
         set(SFML_OS_ANDROID 1)
         # use the OpenGL ES implementation on Android
         set(OPENGL_ES 1)
@@ -55,7 +55,7 @@
             return()
         endif()
     endif()
-elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Android")
+elseif(FALSE)
     set(SFML_OS_ANDROID 1)
 
     # use the OpenGL ES implementation on Android
