--- a/src/SFML/System/Unix/ThreadImpl.cpp
+++ b/src/SFML/System/Unix/ThreadImpl.cpp
@@ -62,7 +62,7 @@
 {
     if (m_isActive)
     {
-        #ifndef SFML_SYSTEM_ANDROID
+        #ifndef __ANDROID__
             pthread_cancel(m_thread);
         #else
             // See https://stackoverflow.com/questions/4610086/pthread-cancel-al
@@ -78,7 +78,7 @@
     // The Thread instance is stored in the user data
     Thread* owner = static_cast<Thread*>(userData);
 
-    #ifndef SFML_SYSTEM_ANDROID
+    #ifndef __ANDROID__
         // Tell the thread to handle cancel requests immediately
         pthread_setcanceltype(PTHREAD_CANCEL_ASYNCHRONOUS, NULL);
     #endif
