diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7f3f941..7c148ff 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -113,6 +113,8 @@ ELSEIF(NETBSD)
   set(CORE_LIBRARIES util resolv execinfo)
 ELSEIF(WIN32)
   set(CORE_LIBRARIES Ws2_32 Shlwapi dbghelp)
+ELSEIF(ANDROID)
+  set(CORE_LIBRARIES util)
 ELSE()
   set(CORE_LIBRARIES util resolv)
 ENDIF()
@@ -120,7 +122,11 @@ ENDIF()
 MACRO(DECORATE_TARGET TARGET_NAME)
   add_sanitizers(${TARGET_NAME})
   set_target_properties(${TARGET_NAME} PROPERTIES COTIRE_CXX_PREFIX_HEADER_INIT "src/base/Headers.hpp")
-  cotire(${TARGET_NAME})
+  IF(ANDROID)
+    # Doesn't work when cross-compiling
+  ELSE()
+    cotire(${TARGET_NAME})
+  ENDIF()
 ENDMACRO()
 
 include_directories(
@@ -129,7 +131,6 @@
   external_imported/Catch2/single_include
   external_imported/cxxopts/include
   external_imported/msgpack-c/include
-  external_imported/UniversalStacktrace/ust
   src/base
   src/terminal
   src/terminal/forwarding
