diff -u -r ../w3m-0.5.3/file.c ./file.c
--- ../w3m-0.5.3/file.c	2011-01-04 04:22:21.000000000 -0500
+++ ./file.c	2015-11-19 17:37:37.536882299 -0500
@@ -1513,6 +1513,21 @@
     return hauth->scheme ? hauth : NULL;
 }
 
+#ifdef __ANDROID__
+static char* getpass(const char* prompt) {
+	static char chars[128];
+	int len = 0;
+	while (1) {
+		char c = fgetc(stdin); 
+		if (c == '\r' || c == '\n' || c == 0) break; 
+		chars[len++] = c;
+		if (len == sizeof(chars)-1) break;
+	} 
+	chars[len] = 0;
+	return chars;
+}
+#endif
+
 static void
 getAuthCookie(struct http_auth *hauth, char *auth_header,
 	      TextList *extra_header, ParsedURL *pu, HRequest *hr,
