Fix compiling with newer libgc version.

diff -u -r ../w3m-0.5.3/main.c ./main.c
--- ../w3m-0.5.3/main.c	2011-01-04 04:42:19.000000000 -0500
+++ ./main.c	2015-11-19 17:34:47.045885249 -0500
@@ -833,7 +833,7 @@
     mySignal(SIGPIPE, SigPipe);
 #endif
 
-    orig_GC_warn_proc = GC_set_warn_proc(wrap_GC_warn_proc);
+    GC_set_warn_proc(wrap_GC_warn_proc);
     err_msg = Strnew();
     if (load_argc == 0) {
 	/* no URL specified */
