--- ../src-Makefile.in.orig	2020-04-22 20:04:10.704587245 +0200
+++ ./src/Makefile.in	2020-04-22 20:04:19.687899278 +0200
@@ -416,7 +416,7 @@
    $(LIBGNOME_CFLAGS) $(LIBNOTIFY_CFLAGS) $(XTST_CFLAGS) $(AT_SPI2_CFLAGS) $(INCLUDES)
 
 florence_LDADD = $(DEPS_LIBS) $(LIBM) $(X11_LIBS) $(LIBGNOME_LIBS) $(LIBNOTIFY_LIBS)\
-   $(XTST_LIBS) $(AT_SPI2_LIBS) $(GTK3_LIBS) -lflorence-1.0
+   $(XTST_LIBS) $(AT_SPI2_LIBS) $(GTK3_LIBS) libflorence-1.0.la
 
 libflorence_1_0_la_SOURCES = lib/florence.c
 library_includedir = $(includedir)/florence-1.0
