--- a/common/rpackagefilter.cc
+++ b/common/rpackagefilter.cc
@@ -788,7 +788,7 @@
    _enabled = conf.FindB(key + "::enabled");
    if (_enabled == true) {
       string FileName = _config->Find("Synaptic::ReducedViewFile",
-                                      "/etc/apt/metadata");
+                                      "@TERMUX_PREFIX@/etc/apt/metadata");
       if (FileExists(FileName))
          addFile(FileName);
    }
