--- a/gtk/rgtaskswin.cc
+++ b/gtk/rgtaskswin.cc
@@ -71,7 +71,7 @@
    gboolean marked = FALSE;
    gboolean activatable = FALSE;
    gchar *taskname = NULL;
-   string cmd = _config->Find("Synaptic::taskHelperProg", "/usr/bin/tasksel");
+   string cmd = _config->Find("Synaptic::taskHelperProg", "@TERMUX_PREFIX@/bin/tasksel");
    do {
       gtk_tree_model_get(model,
                          &iter,
@@ -143,7 +143,7 @@
       GTK_TREE_MODEL(me->_store), &iter, TASK_NAME_COLUMN, &str, -1);
 
    // ask tasksel about the selected task
-   string cmd = _config->Find("Synaptic::taskHelperProg", "/usr/bin/tasksel") +
+   string cmd = _config->Find("Synaptic::taskHelperProg", "@TERMUX_PREFIX@/bin/tasksel") +
                 " --task-desc " + string(str);
    string taskDescr;
    char buf[255];
@@ -235,7 +235,7 @@
    // fiel in tasks
    char buf[255];
    gchar **strArray;
-   string cmd = _config->Find("Synaptic::taskHelperProg", "/usr/bin/tasksel") +
+   string cmd = _config->Find("Synaptic::taskHelperProg", "@TERMUX_PREFIX@/bin/tasksel") +
                 " --list-tasks";
    FILE *f = popen(cmd.c_str(), "r");
 
