--- a/meson.build
+++ b/meson.build
@@ -34,7 +34,6 @@
 x11_dep = cpp.find_library('X11', required: true)
 util_dep = cpp.find_library('util', required: true)
 apt_inst_dep = cpp.find_library('apt-inst', required: false)
-dependency('polkit-gobject-1', required: true)
 
 if not cpp.has_header('apt-pkg/configuration.h', dependencies: apt_pkg_dep)
   error('You need the apt-pkg headers installed to compile synaptic.')
