diff --git a/gtk/Makefile.am b/gtk/Makefile.am
index ed83daa..6100cf9 100644
--- a/gtk/Makefile.am
+++ b/gtk/Makefile.am
@@ -23,7 +23,6 @@ synaptic_LDADD = \
 	@GTK_LIBS@ \
 	@VTE_LIBS@ @LP_LIBS@\
 	-lutil \
-	-lpthread \
 	$(LIBEPT_LIBS)
 
 synaptic_SOURCES= \
diff --git a/tests/Makefile.am b/tests/Makefile.am
index 4888444..b382f7a 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -8,7 +8,7 @@ LDADD = \
 	${top_builddir}/common/libsynaptic.a\
 	-lapt-pkg -lX11 @RPM_LIBS@ @DEB_LIBS@ \
 	@GTK_LIBS@ @VTE_LIBS@ @LP_LIBS@\
-	-lpthread $(LIBEPT_LIBS)
+	$(LIBEPT_LIBS)
 
 test_rpackage_SOURCES= test_rpackage.cc
 
diff --git a/wings/Makefile.am b/wings/Makefile.am
index f59034f..76592bf 100644
--- a/wings/Makefile.am
+++ b/wings/Makefile.am
@@ -7,7 +7,7 @@ sbin_PROGRAMS=wsynaptic
 wsynaptic_LDADD =-lExtraWINGs @WINGS_LIBS@ \
 	${top_builddir}/common/libsynaptic.a\
 	-lapt-pkg @RPM_LIBS@ @DEB_LIBS@\
-	-lpthread -L/usr/X11R6/lib $(LIBTAGCOLL_LIBS)
+	-L/usr/X11R6/lib $(LIBTAGCOLL_LIBS)
 
 
 
