--- a/wings/synaptic.cc
+++ b/wings/synaptic.cc
@@ -61,7 +61,7 @@
 {    
     setlocale(LC_ALL, "");
     
-//    bindtextdomain(PACKAGE, "/usr/local/share/locale");
+//    bindtextdomain(PACKAGE, "@TERMUX_PREFIX@/share/locale");
     textdomain(PACKAGE);
     
     WMInitializeApplication("synaptic", &argc, argv);
@@ -75,12 +75,12 @@
     //XSynchronize(dpy, 1);
     scr = WMCreateScreen(dpy, DefaultScreen(dpy));
 
-    if (getuid() != 0) {
+   /* if (getuid() != 0) {
 	WMRunAlertPanel(scr, NULL, _("Error"),
 			_("You must run this program as the root user."),
 			_("OK"), NULL, NULL);
 	exit(1);
-    }   
+    }*/
  
  
     if (!RInitConfiguration("synaptic.conf")) {
