diff -uNr twm-1.0.10/src/menus.c twm-1.0.10.mod/src/menus.c
--- twm-1.0.10/src/menus.c	2018-03-07 09:10:22.000000000 +0200
+++ twm-1.0.10.mod/src/menus.c	2018-06-17 22:56:18.428908009 +0300
@@ -2243,7 +2243,7 @@
         }
 	break;
    case F_STARTWM:
-	execlp("/bin/sh", "sh", "-c", action, (void *)NULL);
+	execlp("@TERMUX_PREFIX@/bin/sh", "sh", "-c", action, (void *)NULL);
 	fprintf (stderr, "%s:  unable to start:  %s\n", ProgramName, *Argv);
 	break;
 
