diff -uNr twm-1.0.10/src/deftwmrc.c twm-1.0.10.mod/src/deftwmrc.c
--- twm-1.0.10/src/deftwmrc.c	2018-03-07 09:10:33.000000000 +0200
+++ twm-1.0.10.mod/src/deftwmrc.c	2018-06-17 22:57:34.858908574 +0300
@@ -67,7 +67,7 @@
     (unsigned char *) "\"Show Iconmgr\"	f.showiconmgr",
     (unsigned char *) "\"Hide Iconmgr\"	f.hideiconmgr",
     (unsigned char *) "\"\"		f.nop",
-    (unsigned char *) "\"Xterm\"		f.exec \"exec xterm &\"",
+    (unsigned char *) "\"Aterm\"		f.exec \"exec aterm &\"",
     (unsigned char *) "\"\"		f.nop",
     (unsigned char *) "\"Kill\"		f.destroy",
     (unsigned char *) "\"Delete\"	f.delete",
