diff -uNr twm-1.0.10/src/system.twmrc twm-1.0.10.mod/src/system.twmrc
--- twm-1.0.10/src/system.twmrc	2018-03-07 09:10:22.000000000 +0200
+++ twm-1.0.10.mod/src/system.twmrc	2018-06-17 23:03:44.265577978 +0300
@@ -80,7 +80,7 @@
 "Show Iconmgr"	f.showiconmgr
 "Hide Iconmgr"	f.hideiconmgr
 ""		f.nop
-"Xterm"		f.exec "exec xterm &"
+"Aterm"		f.exec "exec aterm &"
 ""		f.nop
 "Kill"		f.destroy
 "Delete"	f.delete
