diff -u -r ../wol-0.7.1/lib/getline.h ./lib/getline.h
--- ../wol-0.7.1/lib/getline.h	2002-03-11 12:55:36.000000000 -0500
+++ ./lib/getline.h	2016-02-05 16:14:45.530639085 -0500
@@ -19,20 +19,4 @@
 
 # include <stdio.h>
 
-# ifndef PARAMS
-#  if defined PROTOTYPES || (defined __STDC__ && __STDC__)
-#   define PARAMS(Args) Args
-#  else
-#   define PARAMS(Args) ()
-#  endif
-# endif
-
-# if __GLIBC__ < 2
-int
-getline PARAMS ((char **_lineptr, size_t *_n, FILE *_stream));
-
-int
-getdelim PARAMS ((char **_lineptr, size_t *_n, int _delimiter, FILE *_stream));
-# endif
-
 #endif /* not GETLINE_H_ */
