From 5405d61623e82896e498c5c8342dd6f42e689115 Mon Sep 17 00:00:00 2001
From: clanmills <robin@clanmills.com>
Date: Tue, 12 Sep 2017 09:14:42 +0100
Subject: [PATCH] Fix http://dev.exiv2.org/issues/1305

---
 src/pentaxmn.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/pentaxmn.cpp b/src/pentaxmn.cpp
index 4fc38be0a..b22cb43b6 100644
--- a/src/pentaxmn.cpp
+++ b/src/pentaxmn.cpp
@@ -1167,6 +1167,8 @@ namespace Exiv2 {
 
     std::ostream& PentaxMakerNote::printShutterCount(std::ostream& os, const Value& value, const ExifData* metadata)
     {
+        if ( ! metadata ) return os << "undefined" ;
+
         ExifData::const_iterator dateIt = metadata->findKey(
                 ExifKey("Exif.PentaxDng.Date"));
         if (dateIt == metadata->end()) {
