diff -u -r ../indent-2.2.12/Makefile.in ./Makefile.in
--- ../indent-2.2.12/Makefile.in	2018-09-05 21:00:27.000000000 +0000
+++ ./Makefile.in	2018-09-08 00:11:20.094710325 +0000
@@ -378,7 +378,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = no-texinfo.tex
-SUBDIRS = intl src doc po man regression
+SUBDIRS = intl src po man
 BUILT_SOURCES = 
 DISTFILES = $(DIST_COMMON:README=README.md) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 EXTRA_DIST = README.md \
