diff -u -r ../indent-2.2.11/man/Makefile.in ./man/Makefile.in
--- ../indent-2.2.11/man/Makefile.in	2010-01-31 16:25:21.000000000 +0000
+++ ./man/Makefile.in	2018-06-07 22:52:13.810674067 +0000
@@ -575,7 +575,7 @@
 
 
 @PACKAGE@.1: ${srcdir}/@PACKAGE@.1.in  ${srcdir}/../doc/@PACKAGE@.texinfo texinfo2man  Makefile.am
-	./texinfo2man ${srcdir}/@PACKAGE@.1.in ${srcdir}/../doc/@PACKAGE@.texinfo > $@
+	@echo Skipping
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
