--- a/libselinux/src/android/android.c
+++ b/libselinux/src/android/android.c
@@ -1,9 +1,10 @@
+#include <termux_build.h>
 #include <errno.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
 
-#include <log/log.h>
+#include <android/log.h>
 #include <selinux/android.h>
 #include <selinux/label.h>
 
--- a/libselinux/src/android/android_seapp.c
+++ b/libselinux/src/android/android_seapp.c
@@ -1,3 +1,5 @@
+#include <termux_build.h>
+
 #include <ctype.h>
 #include <limits.h>
 #include <linux/magic.h>
@@ -8,7 +10,6 @@
 #include <string.h>
 #include <unistd.h>
 
-#include <private/android_filesystem_config.h>
 #include <selinux/android.h>
 #include <selinux/context.h>
 #include <selinux/selinux.h>
--- a/libselinux/src/label_file.c
+++ b/libselinux/src/label_file.c
@@ -1,3 +1,5 @@
+#include <termux_build.h>
+
 /*
  * File contexts backend for labeling system
  *
--- a/libselinux/src/regex.h
+++ b/libselinux/src/regex.h
@@ -1,6 +1,8 @@
 #ifndef SRC_REGEX_H_
 #define SRC_REGEX_H_
 
+#include <termux_build.h>
+
 #include <stdbool.h>
 #include <stdio.h>
 
