diff -uNr quickjs-2024-01-13.orig/Makefile quickjs-2024-01-13/Makefile
--- quickjs-2024-01-13.orig/Makefile	2024-01-13 15:50:39.000000000 +0530
+++ quickjs-2024-01-13/Makefile	2024-01-17 13:52:04.672637577 +0530
@@ -79,7 +79,7 @@
   CFLAGS += -Wchar-subscripts -funsigned-char
   CFLAGS += -MMD -MF $(OBJDIR)/$(@F).d
   ifdef CONFIG_DEFAULT_AR
-    AR=$(CROSS_PREFIX)ar
+    AR ?= $(CROSS_PREFIX)ar
   else
     ifdef CONFIG_LTO
       AR=$(CROSS_PREFIX)llvm-ar
@@ -106,7 +106,7 @@
     AR=$(CROSS_PREFIX)ar
   endif
 endif
-STRIP=$(CROSS_PREFIX)strip
+STRIP ?= $(CROSS_PREFIX)strip
 CFLAGS+=-fwrapv # ensure that signed overflows behave as expected
 ifdef CONFIG_WERROR
 CFLAGS+=-Werror
