diff --git a/far2l/src/base/farrtl.cpp b/far2l/src/base/farrtl.cpp
index f637a9fe6..d226ceaf6 100644
--- a/far2l/src/base/farrtl.cpp
+++ b/far2l/src/base/farrtl.cpp
@@ -30,7 +30,7 @@ wchar_t *__cdecl far_wcsncpy(wchar_t *dest, const wchar_t *src, size_t DestSize)
 	return tmpsrc;
 }
 
-#if defined(__MUSL__)
+#if defined(__MUSL__) || defined(__ANDROID__)
 struct QSortExAdapterArg
 {
 	int (*__cdecl comp)(const void *, const void *, void *);
