diff -uNr cpulimit-0.2/src/cpulimit.c cpulimit-0.2.mod/src/cpulimit.c
--- cpulimit-0.2/src/cpulimit.c	2015-06-17 09:46:59.000000000 +0300
+++ cpulimit-0.2.mod/src/cpulimit.c	2020-11-14 14:42:58.123123342 +0200
@@ -38,7 +38,7 @@
 #include <string.h>
 #include <sys/stat.h>
 #include <sys/time.h>
-#include <sys/sysctl.h>
+#include <linux/sysctl.h>
 #include <sys/resource.h>
 #include <sys/types.h>
 #include <sys/wait.h>
