diff -uNr ngspice-37/src/include/ngspice/defines.h ngspice-37.mod/src/include/ngspice/defines.h
--- ngspice-37/src/include/ngspice/defines.h    2022-05-17 18:00:52.000000000 +0800
+++ ngspice-37.mod/src/include/ngspice/defines.h        2022-06-16 20:46:02.258164585 +0800
@@ -82,8 +82,8 @@
 #define DIR_TERM            '/'
 #define DIR_CWD             "."

-#define TEMPFORMAT          "/tmp/%s%d"
-#define TEMPFORMAT2         "/tmp/%s%d_%d"
+#define TEMPFORMAT          "@TERMUX_PREFIX@/tmp/%s%d"
+#define TEMPFORMAT2         "@TERMUX_PREFIX@/tmp/%s%d_%d"
 #define SYSTEM_PLOT5LPR     "lpr -P%s -g %s"
 #define SYSTEM_PSLPR        "lpr -P%s %s"
 #define SYSTEM_MAIL         "Mail -s \"%s (%s) Bug Report\" %s"
