diff -uNr ngspice-37/src/include/ngspice/ipctiein.h ngspice-37.mod/src/include/ngspice/ipctiein.h
--- ngspice-37/src/include/ngspice/ipctiein.h   2022-05-17 18:00:52.000000000 +0800
+++ ngspice-37.mod/src/include/ngspice/ipctiein.h       2022-06-16 20:38:56.018164890 +0800
@@ -45,8 +45,8 @@
 #include "ngspice/ipcproto.h"


-#define  IPC_STDOUT_FILE_NAME  "/usr/tmp/atesse_xspice.out"
-#define  IPC_STDERR_FILE_NAME  "/usr/tmp/atesse_xspice.err"
+#define  IPC_STDOUT_FILE_NAME  "@TERMUX_PREFIX@/tmp/atesse_xspice.out"
+#define  IPC_STDERR_FILE_NAME  "@TERMUX_PREFIX@/tmp/atesse_xspice.err"

 /*
