diff -uNr ngspice-37/src/frontend/plotting/gnuplot.c ngspice-37.mod/src/frontend/plotting/gnuplot.c
--- ngspice-37/src/frontend/plotting/gnuplot.c  2022-05-17 18:00:52.000000000 +0800
+++ ngspice-37.mod/src/frontend/plotting/gnuplot.c      2022-06-16 21:26:14.926999780 +0800
@@ -388,6 +388,7 @@
 #ifndef EXT_ASC
     fprintf(file, "set encoding utf8\n");
 #endif
+    fprintf(file, "set term dumb\n");
     fprintf(file, "set termoption noenhanced\n");

     if (contours) {
