--- fdk-aac-2.0.2/libSBRdec/src/lpp_tran.cpp	2021-04-28 18:04:15.000000000 +0530
+++ fdk-aac-2.0.2.mod/libSBRdec/src/lpp_tran.cpp	2021-07-02 12:26:00.717941802 +0530
@@ -118,10 +118,6 @@
   \sa lppTransposer(), main_audio.cpp, sbr_scale.h, \ref documentationOverview
 */
 
-#ifdef __ANDROID__
-#include "log/log.h"
-#endif
-
 #include "lpp_tran.h"
 
 #include "sbr_ram.h"
@@ -336,12 +332,6 @@
       }
     }
   }
-#ifdef __ANDROID__
-  else {
-    // Safetynet logging
-    android_errorWriteLog(0x534e4554, "112160868");
-  }
-#endif
 
   /* init bwIndex for each patch */
   FDKmemclear(bwIndex, sizeof(bwIndex));
@@ -934,12 +924,6 @@
       FDKmemclear(&qmfBufferImag[i][targetStopBand], memSize);
     }
   }
-#ifdef __ANDROID__
-  else {
-    // Safetynet logging
-    android_errorWriteLog(0x534e4554, "112160868");
-  }
-#endif
 
   /*
   Calc common low band scale factor
