--- a/lib/utilities.c
+++ b/lib/utilities.c
@@ -394,12 +394,31 @@
 
 RmUserList *rm_userlist_new(void) {
     struct passwd *node = NULL;
-    struct group *grp = NULL;
+    _UNUSED struct group *grp = NULL;
 
     RmUserList *self = g_malloc0(sizeof(RmUserList));
     self->users = g_sequence_new(NULL);
     self->groups = g_sequence_new(NULL);
 
+#ifdef __ANDROID__
+    uid_t uid = getuid();
+    gid_t gid = getgid();
+    g_sequence_insert_sorted(self->users, GUINT_TO_POINTER(uid),
+                             rm_userlist_cmp_ids, NULL);
+    g_sequence_insert_sorted(self->groups, GUINT_TO_POINTER(gid),
+                             rm_userlist_cmp_ids, NULL);
+    node = getpwuid(uid);
+    if(node) {
+        g_sequence_insert_sorted(self->users, GUINT_TO_POINTER(node->pw_uid),
+                                 rm_userlist_cmp_ids, NULL);
+        g_sequence_insert_sorted(self->groups, GUINT_TO_POINTER(node->pw_gid),
+                                 rm_userlist_cmp_ids, NULL);
+    }
+    g_sequence_insert_sorted(self->users, GUINT_TO_POINTER(0),
+                             rm_userlist_cmp_ids, NULL);
+    g_sequence_insert_sorted(self->groups, GUINT_TO_POINTER(0),
+                             rm_userlist_cmp_ids, NULL);
+#else
     setpwent();
     while((node = getpwent()) != NULL) {
         g_sequence_insert_sorted(self->users, GUINT_TO_POINTER(node->pw_uid),
@@ -416,6 +435,7 @@
     }
 
     endgrent();
+#endif
     g_mutex_init(&self->lock);
     return self;
 }
@@ -432,6 +452,18 @@
                                       rm_userlist_cmp_ids, NULL);
         uid_found = g_sequence_lookup(self->users, GUINT_TO_POINTER(uid),
                                       rm_userlist_cmp_ids, NULL);
+#ifdef __ANDROID__
+        if(!uid_found && getpwuid(uid) != NULL) {
+            g_sequence_insert_sorted(self->users, GUINT_TO_POINTER(uid),
+                                     rm_userlist_cmp_ids, NULL);
+            uid_found = TRUE;
+        }
+        if(!gid_found && getgrgid(gid) != NULL) {
+            g_sequence_insert_sorted(self->groups, GUINT_TO_POINTER(gid),
+                                     rm_userlist_cmp_ids, NULL);
+            gid_found = TRUE;
+        }
+#endif
     }
     g_mutex_unlock(&self->lock);
