--- a/configure.ac
+++ b/configure.ac
@@ -2,7 +2,7 @@
 AC_INIT([jbig2enc], [0.28], [agl@imperialviolet.org], [jbig2enc-0.28],
 		[https://github.com/agl/jbig2enc])
 AC_CONFIG_MACRO_DIR([m4])
-AM_INIT_AUTOMAKE([-Wall -Werror foreign no-dependencies])
+AM_INIT_AUTOMAKE([-Wall foreign no-dependencies])
 
 # this should fix automake 1.12 build and compatible with automake 1.11
 m4_ifdef([AM_PROG_AR], [AM_PROG_AR])
@@ -73,7 +73,7 @@
 	fi],
 [libtool_patch_use_rpath=$enable_rpath])
 
-AC_CHECK_LIB([lept], [findFileFormatStream], [], [
+AC_CHECK_LIB([leptonica], [findFileFormatStream], [], [
 			echo "Error! Leptonica not detected."
 			exit -1
 			])
