--- ./Makefile.flags.orig	2021-01-01 11:52:27.000000000 +0100
+++ ./Makefile.flags	2021-08-19 16:49:06.743647568 +0200
@@ -151,9 +151,9 @@
 # fall back to using a temp file:
 CRYPT_AVAILABLE := $(shell echo 'int main(void){return 0;}' >crypttest.c; $(CC) $(CFLAGS) -lcrypt -o /dev/null crypttest.c >/dev/null 2>&1 && echo "y"; rm crypttest.c)
 ifeq ($(CRYPT_AVAILABLE),y)
-LDLIBS += m rt crypt
+LDLIBS += m crypt
 else
-LDLIBS += m rt
+LDLIBS += m
 endif
 # libm may be needed for dc, awk, ntpd
 # librt may be needed for clock_gettime()
