diff -uNr busybox-1.31.1/miscutils/crontab.c busybox-1.31.1.mod/miscutils/crontab.c
--- busybox-1.31.1/miscutils/crontab.c	2019-06-10 13:50:53.000000000 +0300
+++ busybox-1.31.1.mod/miscutils/crontab.c	2020-01-29 23:25:22.817488323 +0200
@@ -53,10 +53,12 @@
 
 	/* CHILD - change user and run editor */
 	/* initgroups, setgid, setuid */
+#ifndef __ANDROID__
 	change_identity(pas);
 	setup_environment(pas->pw_shell,
 			SETUP_ENV_CHANGEENV | SETUP_ENV_TO_TMP,
 			pas);
+#endif
 	ptr = getenv("VISUAL");
 	if (!ptr) {
 		ptr = getenv("EDITOR");
@@ -158,7 +160,6 @@
 		/* No O_EXCL: we don't want to be stuck if earlier crontabs
 		 * were killed, leaving stale temp file behind */
 		src_fd = xopen3(tmp_fname, O_RDWR|O_CREAT|O_TRUNC, 0600);
-		fchown(src_fd, pas->pw_uid, pas->pw_gid);
 		fd = open(pas->pw_name, O_RDONLY);
 		if (fd >= 0) {
 			bb_copyfd_eof(fd, src_fd);
@@ -167,7 +168,8 @@
 		}
 		close_on_exec_on(src_fd); /* don't want editor to see this fd */
 		edit_file(pas, tmp_fname);
-		/* fall through */
+		xrename(tmp_fname, pas->pw_name);
+		break;
 
 	case 0: /* Replace (no -l, -e, or -r were given) */
 		new_fname = xasprintf("%s.new", pas->pw_name);
