diff -uNr busybox-1.31.1/networking/telnetd.c busybox-1.31.1.mod/networking/telnetd.c
--- busybox-1.31.1/networking/telnetd.c	2019-11-11 22:00:22.799539737 +0200
+++ busybox-1.31.1.mod/networking/telnetd.c	2019-11-11 22:32:02.268344626 +0200
@@ -707,7 +707,7 @@
 	} else {
 		master_fd = 0;
 		if (!(opt & OPT_WAIT)) {
-			unsigned portnbr = 23;
+			unsigned portnbr = 8023;
 			if (opt & OPT_PORT)
 				portnbr = xatou16(opt_portnbr);
 			master_fd = create_and_bind_stream_or_die(opt_bindaddr, portnbr);
