diff -u -r ../busybox-1.24.1/arch/i386/Makefile ./arch/i386/Makefile
--- ../busybox-1.24.1/arch/i386/Makefile	2015-07-12 22:18:46.000000000 -0400
+++ ./arch/i386/Makefile	2015-12-25 19:05:56.660211464 -0500
@@ -4,4 +4,5 @@
 
 # -mpreferred-stack-boundary=2 is essential in preventing gcc 4.2.x
 # from aligning stack to 16 bytes. (Which is gcc's way of supporting SSE).
-CFLAGS += $(call cc-option,-march=i386 -mpreferred-stack-boundary=2,)
+# This fixes crashing e.g. running busybox sed -e "s#'#'\"'\"'#g"
+# on the x86_64 emulator.
+# CFLAGS += $(call cc-option,-march=i386 -mpreferred-stack-boundary=2,)
