diff -u -r ../busybox-1.27.1/coreutils/mktemp.c ./coreutils/mktemp.c
--- ../busybox-1.27.1/coreutils/mktemp.c	2017-07-06 15:14:57.000000000 +0000
+++ ./coreutils/mktemp.c	2017-07-24 23:44:41.384345896 +0000
@@ -52,7 +52,7 @@
 //usage:     "\n	-p DIR	Use DIR as a base directory (implies -t)"
 //usage:     "\n	-u	Do not create anything; print a name"
 //usage:     "\n"
-//usage:     "\nBase directory is: -p DIR, else $TMPDIR, else /tmp"
+//usage:     "\nBase directory is: -p DIR, else $TMPDIR, else @TERMUX_PREFIX@/tmp"
 //usage:
 //usage:#define mktemp_example_usage
 //usage:       "$ mktemp /tmp/temp.XXXXXX\n"
@@ -78,7 +78,7 @@
 
 	path = getenv("TMPDIR");
 	if (!path || path[0] == '\0')
-		path = "/tmp";
+		path = "@TERMUX_PREFIX@/tmp";
 
 	opt_complementary = "?1"; /* 1 argument max */
 	opts = getopt32(argv, "dqtp:u", &path);
