diff -u -r ../busybox-1.24.1/editors/diff.c ./editors/diff.c
--- ../busybox-1.24.1/editors/diff.c	2015-10-11 09:31:50.000000000 -0400
+++ ./editors/diff.c	2016-02-08 05:23:45.890302334 -0500
@@ -734,7 +734,7 @@
 		 * When we meet non-seekable file, we must make a temp copy.
 		 */
 		if (lseek(fd, 0, SEEK_SET) == -1 && errno == ESPIPE) {
-			char name[] = "/tmp/difXXXXXX";
+			char name[] = "@TERMUX_PREFIX@/tmp/difXXXXXX";
 			int fd_tmp = xmkstemp(name);
 
 			unlink(name);
