--- ./editors/vi.c.orig	2020-08-06 23:54:05.454464571 +0000
+++ ./editors/vi.c	2020-08-06 23:54:59.007378601 +0000
@@ -4314,6 +4314,7 @@
 #undef cur_line
 }
 
+#pragma clang optimize off
 int vi_main(int argc, char **argv) MAIN_EXTERNALLY_VISIBLE;
 int vi_main(int argc, char **argv)
 {
--- ./editors/ed.c.orig	2020-08-06 23:55:28.927889014 +0000
+++ ./editors/ed.c	2020-08-06 23:55:36.836023886 +0000
@@ -990,6 +990,7 @@
 	}
 }
 
+#pragma clang optimize off
 int ed_main(int argc, char **argv) MAIN_EXTERNALLY_VISIBLE;
 int ed_main(int argc UNUSED_PARAM, char **argv)
 {
--- ./editors/awk.c.orig	2020-08-06 23:57:12.149648346 +0000
+++ ./editors/awk.c	2020-08-06 23:57:31.737981962 +0000
@@ -3190,6 +3190,7 @@
 #undef files_happen
 }
 
+#pragma clang optimize off
 int awk_main(int argc, char **argv) MAIN_EXTERNALLY_VISIBLE;
 int awk_main(int argc UNUSED_PARAM, char **argv)
 {
--- ./editors/diff.c.orig	2020-08-06 23:58:55.919414780 +0000
+++ ./editors/diff.c	2020-08-06 23:59:09.719649530 +0000
@@ -972,6 +972,7 @@
 # define LONGOPTS
 #endif
 
+#pragma clang optimize off
 int diff_main(int argc, char **argv) MAIN_EXTERNALLY_VISIBLE;
 int diff_main(int argc UNUSED_PARAM, char **argv)
 {
