diff -u -r ../busybox-1.22.1/networking/httpd.c ./networking/httpd.c
--- ../busybox-1.22.1/networking/httpd.c	2014-01-09 19:15:44.000000000 +0100
+++ ./networking/httpd.c	2014-01-20 08:47:37.000000000 +0100
@@ -112,7 +112,7 @@
 //usage:     "\n	-i		Inetd mode"
 //usage:     "\n	-f		Don't daemonize"
 //usage:     "\n	-v[v]		Verbose"
-//usage:     "\n	-p [IP:]PORT	Bind to IP:PORT (default *:80)"
+//usage:     "\n	-p [IP:]PORT	Bind to IP:PORT (default *:8080)"
 //usage:	IF_FEATURE_HTTPD_SETUID(
 //usage:     "\n	-u USER[:GRP]	Set uid/gid after binding to port")
 //usage:	IF_FEATURE_HTTPD_BASIC_AUTH(
@@ -371,7 +371,7 @@
 	SET_PTR_TO_GLOBALS(xzalloc(sizeof(G))); \
 	IF_FEATURE_HTTPD_BASIC_AUTH(g_realm = "Web Server Authentication";) \
 	IF_FEATURE_HTTPD_RANGES(range_start = -1;) \
-	bind_addr_or_port = "80"; \
+	bind_addr_or_port = "8080"; \
 	index_page = index_html; \
 	file_size = -1; \
 } while (0)
@@ -887,7 +887,7 @@
 	if (!errno && n && n <= 0xffff)
 		n = create_and_bind_stream_or_die(NULL, n);
 	else
-		n = create_and_bind_stream_or_die(bind_addr_or_port, 80);
+		n = create_and_bind_stream_or_die(bind_addr_or_port, 8080);
 	xlisten(n, 9);
 	return n;
 }
@@ -2245,7 +2245,7 @@
 		proxy_fd = socket(AF_INET, SOCK_STREAM, 0);
 		if (proxy_fd < 0)
 			send_headers_and_exit(HTTP_INTERNAL_SERVER_ERROR);
-		lsa = host2sockaddr(proxy_entry->host_port, 80);
+		lsa = host2sockaddr(proxy_entry->host_port, 8080);
 		if (lsa == NULL)
 			send_headers_and_exit(HTTP_INTERNAL_SERVER_ERROR);
 		if (connect(proxy_fd, &lsa->u.sa, lsa->len) < 0)
