diff -u -r ../busybox-1.22.1/networking/libiproute/iplink.c ./networking/libiproute/iplink.c
--- ../busybox-1.22.1/networking/libiproute/iplink.c	2014-01-09 19:15:44.000000000 +0100
+++ ./networking/libiproute/iplink.c	2014-01-20 08:47:37.000000000 +0100
@@ -11,6 +11,31 @@
 #include <netinet/if_ether.h>
 
 #include <linux/if_vlan.h>
+#ifdef __ANDROID__
+enum {
+	IFLA_INFO_UNSPEC,
+	IFLA_INFO_KIND,
+	IFLA_INFO_DATA,
+	IFLA_INFO_XSTATS,
+	__IFLA_INFO_MAX,
+};
+
+#define IFLA_INFO_MAX   (__IFLA_INFO_MAX - 1)
+enum {
+	IFLA_VLAN_UNSPEC,
+	IFLA_VLAN_ID,
+	IFLA_VLAN_FLAGS,
+	IFLA_VLAN_EGRESS_QOS,
+	IFLA_VLAN_INGRESS_QOS,
+	IFLA_VLAN_PROTOCOL,
+	__IFLA_VLAN_MAX,
+};
+#define IFLA_VLAN_MAX   (__IFLA_VLAN_MAX - 1)
+struct ifla_vlan_flags {
+	    __u32   flags;
+		    __u32   mask;
+};
+#endif
 #include "ip_common.h"  /* #include "libbb.h" is inside */
 #include "rt_names.h"
 #include "utils.h"
