diff -u -r ../busybox-1.24.1/libbb/change_identity.c ./libbb/change_identity.c
--- ../busybox-1.24.1/libbb/change_identity.c	2015-07-12 22:18:47.000000000 -0400
+++ ./libbb/change_identity.c	2015-10-26 19:29:18.972044695 -0400
@@ -33,6 +33,7 @@
 /* Become the user and group(s) specified by PW.  */
 void FAST_FUNC change_identity(const struct passwd *pw)
 {
+#ifndef __ANDROID__
 	int res;
 
 	res = initgroups(pw->pw_name, pw->pw_gid);
@@ -57,4 +58,5 @@
 
 	xsetgid(pw->pw_gid);
 	xsetuid(pw->pw_uid);
+#endif
 }
