diff -u -r ../busybox-1.22.1/include/libbb.h ./include/libbb.h
--- ../busybox-1.22.1/include/libbb.h	2014-01-09 19:15:44.000000000 +0100
+++ ./include/libbb.h	2014-07-01 10:27:23.000000000 +0200
@@ -1785,12 +1785,12 @@
  * use bb_default_login_shell and following defines.
  * If you change LIBBB_DEFAULT_LOGIN_SHELL,
  * don't forget to change increment constant. */
-#define LIBBB_DEFAULT_LOGIN_SHELL  "-/bin/sh"
+#define LIBBB_DEFAULT_LOGIN_SHELL  "-/system/bin/sh"
 extern const char bb_default_login_shell[] ALIGN1;
 /* "/bin/sh" */
 #define DEFAULT_SHELL              (bb_default_login_shell+1)
 /* "sh" */
-#define DEFAULT_SHELL_SHORT_NAME   (bb_default_login_shell+6)
+#define DEFAULT_SHELL_SHORT_NAME   (bb_default_login_shell+13)
 
 /* The following devices are the same on all systems.  */
 #define CURRENT_TTY "/dev/tty"
