diff -u -r ../busybox-1.27.1/networking/httpd.c ./networking/httpd.c
--- ../busybox-1.27.1/networking/httpd.c	2017-07-06 15:14:57.000000000 +0000
+++ ./networking/httpd.c	2017-07-24 23:41:41.454424822 +0000
@@ -234,7 +234,7 @@
 //usage:     "\n	-i		Inetd mode"
 //usage:     "\n	-f		Don't daemonize"
 //usage:     "\n	-v[v]		Verbose"
-//usage:     "\n	-p [IP:]PORT	Bind to IP:PORT (default *:80)"
+//usage:     "\n	-p [IP:]PORT	Bind to IP:PORT (default *:8080)"
 //usage:	IF_FEATURE_HTTPD_SETUID(
 //usage:     "\n	-u USER[:GRP]	Set uid/gid after binding to port")
 //usage:	IF_FEATURE_HTTPD_BASIC_AUTH(
@@ -491,7 +491,7 @@
 	SET_PTR_TO_GLOBALS(xzalloc(sizeof(G))); \
 	IF_FEATURE_HTTPD_BASIC_AUTH(g_realm = "Web Server Authentication";) \
 	IF_FEATURE_HTTPD_RANGES(range_start = -1;) \
-	bind_addr_or_port = "80"; \
+	bind_addr_or_port = "8080"; \
 	index_page = index_html; \
 	file_size = -1; \
 } while (0)
@@ -1007,7 +1007,7 @@
 	if (!errno && n && n <= 0xffff)
 		n = create_and_bind_stream_or_die(NULL, n);
 	else
-		n = create_and_bind_stream_or_die(bind_addr_or_port, 80);
+		n = create_and_bind_stream_or_die(bind_addr_or_port, 8080);
 	xlisten(n, 9);
 	return n;
 }
@@ -2396,7 +2396,7 @@
 		int proxy_fd;
 		len_and_sockaddr *lsa;
 
-		lsa = host2sockaddr(proxy_entry->host_port, 80);
+		lsa = host2sockaddr(proxy_entry->host_port, 8080);
 		if (lsa == NULL)
 			send_headers_and_exit(HTTP_INTERNAL_SERVER_ERROR);
 		proxy_fd = socket(lsa->u.sa.sa_family, SOCK_STREAM, 0);
