diff -u -r ../busybox-1.22.1/coreutils/uname.c ./coreutils/uname.c
--- ../busybox-1.22.1/coreutils/uname.c	2014-01-09 19:15:44.000000000 +0100
+++ ./coreutils/uname.c	2014-07-01 08:21:16.000000000 +0200
@@ -139,7 +139,7 @@
 #endif
 	strcpy(uname_info.processor, unknown_str);
 	strcpy(uname_info.platform, unknown_str);
-	strcpy(uname_info.os, "GNU/Linux");
+	strcpy(uname_info.os, "Android");
 #if 0
 	/* Fedora does something like this */
 	strcpy(uname_info.processor, uname_info.name.machine);
