diff -u -r ../busybox-1.25.1/util-linux/getopt.c ./util-linux/getopt.c
--- ../busybox-1.25.1/util-linux/getopt.c	2016-10-07 10:47:47.000000000 -0400
+++ ./util-linux/getopt.c	2016-11-30 17:46:48.000185295 -0500
@@ -358,6 +358,10 @@
 	;
 #endif
 
+#ifdef __ANDROID__
+extern char* __progname;
+#endif
+
 int getopt_main(int argc, char **argv) MAIN_EXTERNALLY_VISIBLE;
 int getopt_main(int argc, char **argv)
 {
@@ -423,5 +427,8 @@
 	}
 
 	argv[n] = name ? name : argv[0];
+#ifdef __ANDROID__
+	__progname = argv[n];
+#endif
 	return generate_output(argv + n, argc - n, optstr, long_options);
 }
