--- ../cjk-gs-integrate.pl.orig	2020-05-01 17:26:10.709915553 +0200
+++ ./texmf-dist/scripts/cjk-gs-integrate/cjk-gs-integrate.pl	2020-05-01 17:28:25.023071451 +0200
@@ -1937,7 +1937,7 @@
   my $foundres = '';
   if (win32()) {
     # determine tlgs or native gs
-    chomp(my $foo = `kpsewhich -var-value=SELFAUTOPARENT`);
+    chomp(my $foo = `kpsewhich -var-value=TEXMFROOT`);
     if ( -d "$foo/tlpkg/tlgs" ) {
       # should be texlive with tlgs
       print_debug("Assuming tlgs win32 ...\n");
