--- src/texk/dvipdfm-x/pdfencrypt.c~	2016-05-04 06:24:00.000000000 +0200
+++ src/texk/dvipdfm-x/pdfencrypt.c	2016-12-20 22:38:08.739391372 +0100
@@ -604,8 +604,8 @@
       WARN("Invaid UTF-8 string for password.");
   } else {
     while (1) {
-      strncpy(input, getpass("Owner password: "), MAX_PWD_LEN);
-      retry_passwd = getpass("Re-enter owner password: ");
+      //strncpy(input, getpass("Owner password: "), MAX_PWD_LEN);
+      //retry_passwd = getpass("Re-enter owner password: ");
       if (!strncmp(input, retry_passwd, MAX_PWD_LEN))
 	break;
       fputs("Password is not identical.\nTry again.\n", stderr);
@@ -619,8 +619,8 @@
       WARN("Invalid UTF-8 string for passowrd.");
   } else {
     while (1) {
-      strncpy(input, getpass("User password: "), MAX_PWD_LEN);
-      retry_passwd = getpass("Re-enter user password: ");
+      //strncpy(input, getpass("User password: "), MAX_PWD_LEN);
+      //retry_passwd = getpass("Re-enter user password: ");
       if (!strncmp(input, retry_passwd, MAX_PWD_LEN))
 	break;
       fputs("Password is not identical.\nTry again.\n", stderr);
