--- ./texmf-dist/web2c/fmtutil.cnf.orig	2020-04-04 10:19:08.467622980 +0200
+++ ./texmf-dist/web2c/fmtutil.cnf	2020-04-04 10:24:19.200301751 +0200
@@ -43,7 +43,7 @@
 #
 # from aleph:
 aleph aleph - *aleph.ini
-lamed aleph language.dat *lambda.ini
+#! lamed aleph language.dat *lambda.ini
 #
 # from amstex:
 amstex pdftex - -translate-file=cp227.tcx *amstex.ini
@@ -68,11 +68,11 @@
 pdfcsplain luatex - -etex csplain.ini
 #
 # from eplain:
-eplain pdftex language.dat -translate-file=cp227.tcx *eplain.ini
+#! eplain pdftex language.dat -translate-file=cp227.tcx *eplain.ini
 #
 # from jadetex:
-jadetex pdftex language.dat *jadetex.ini
-pdfjadetex pdftex language.dat *pdfjadetex.ini
+#! jadetex pdftex language.dat *jadetex.ini
+#! pdfjadetex pdftex language.dat *pdfjadetex.ini
 #
 # from latex-bin:
 latex pdftex language.dat -translate-file=cp227.tcx *latex.ini
@@ -93,8 +93,8 @@
 luahbtex luahbtex language.def,language.dat.lua luatex.ini
 #
 # from luajittex:
-luajittex luajittex language.def,language.dat.lua luatex.ini
-luajithbtex luajithbtex language.def,language.dat.lua luatex.ini
+#! luajittex luajittex language.def,language.dat.lua luatex.ini
+#! luajithbtex luajithbtex language.def,language.dat.lua luatex.ini
 #
 # from luatex:
 luatex luatex language.def,language.dat.lua luatex.ini
@@ -127,12 +127,12 @@
 pdfetex pdftex language.def -translate-file=cp227.tcx *pdfetex.ini
 #
 # from platex:
-platex eptex language.dat *platex.ini
-platex-dev eptex language.dat *platex.ini
+#! platex eptex language.dat *platex.ini
+#! platex-dev eptex language.dat *platex.ini
 #
 # from ptex:
 ptex ptex - ptex.ini
-eptex eptex language.def *eptex.ini
+#! eptex eptex language.def *eptex.ini
 #
 # from tex:
 tex tex - tex.ini
@@ -141,12 +141,12 @@
 texsis pdftex - -translate-file=cp227.tcx texsis.ini
 #
 # from uplatex:
-uplatex euptex language.dat *uplatex.ini
-uplatex-dev euptex language.dat *uplatex.ini
+#! uplatex euptex language.dat *uplatex.ini
+#! uplatex-dev euptex language.dat *uplatex.ini
 #
 # from uptex:
 uptex uptex - uptex.ini
-euptex euptex language.def *euptex.ini
+#! euptex euptex language.def *euptex.ini
 #
 # from xelatex-dev:
 xelatex-dev xetex language.dat -etex xelatex.ini
@@ -156,5 +156,5 @@
 xelatex xetex language.dat -etex xelatex.ini
 #
 # from xmltex:
-xmltex pdftex language.dat *xmltex.ini
-pdfxmltex pdftex language.dat *pdfxmltex.ini
+#! xmltex pdftex language.dat *xmltex.ini
+#! pdfxmltex pdftex language.dat *pdfxmltex.ini
