--- ../texdef.pl.orig	2020-04-04 09:27:54.747454581 +0200
+++ ./texmf-dist/scripts/texdef/texdef.pl	2020-04-04 09:28:20.700734760 +0200
@@ -283,7 +283,7 @@
             $EDITOR = 'texworks "%f"';
         }
         else {
-            for my $ed (qw(/usr/bin/vim /usr/bin/emacs /usr/bin/nano)) {
+            for my $ed (qw(@TERMUX_PREFIX@/bin/vim @TERMUX_PREFIX@/bin/emacs @TERMUX_PREFIX@/bin/nano)) {
                 if (-x $ed) {
                     $EDITOR = $ed;
                     last;
