--- a/auxiliary_files/run_scripts/mumble-server-user-wrapper.in
+++ b/auxiliary_files/run_scripts/mumble-server-user-wrapper.in
@@ -6,7 +6,7 @@
 # Mumble source tree or at <https://www.mumble.info/LICENSE>.

 DIR=$HOME/mumble-server
-SYSDIR=/usr/share/doc/mumble-server/examples
+SYSDIR=@TERMUX_PREFIX@/share/doc/mumble-server/examples

 if [ $UID == 0 ] || [ $EUID == 0 ]; then
   echo "You should never run this script as root. If you want a system-wide install, see "
