--- a/meson.build
+++ b/meson.build
@@ -106,10 +106,6 @@
 introspection_dep = dependency('gobject-introspection-1.0', version: introspection_req)
 gtk_dep = dependency('gtk+-3.0', version: gtk_req, required: false)
 
-gi_docgen_dep = dependency('gi-docgen', version: '>= 2021.7',
-                           fallback: ['gi-docgen', 'dummy_dep'],
-                           native: true,
-                           required: get_option('gtk_doc'))
 gladeui_dep = dependency('gladeui-2.0', required: false)
 
 # From python 3.8 we neeed python3-embed
