--- a/ckcdeb.h
+++ b/ckcdeb.h
@@ -1308,7 +1308,7 @@
 #ifdef IKSD				/* IKSD options... */
 #ifndef IKSDCONF			/* IKSD configuration file */
 #ifdef UNIX
-#define IKSDCONF "/etc/iksd.conf"
+#define IKSDCONF "@TERMUX_PREFIX@/etc/iksd.conf"
 #else
 #ifdef OS2
 #define IKSDCONF "iksd.ksc"
@@ -1322,7 +1322,7 @@
 #define IK_LCKTRIES 16			/* How many times to try to get lock */
 #define IK_LCKSLEEP 1			/* How long to sleep between tries */
 #define IK_LOCKFILE "iksd.lck"		/* Database lockfilename */
-#define IK_DBASEDIR "/var/log/"		/* Database directory */
+#define IK_DBASEDIR "@TERMUX_PREFIX@/var/log/"		/* Database directory */
 #define IK_DBASEFIL "iksd.db"		/* Database filename */
 #else /* UNIX */
 #ifdef OS2
