--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -685,7 +685,7 @@
     # TODO(crbug.com/376278218): This causes segfault on Linux ARM builds.
     # It also causes segfault on Linux s390x:
     # https://github.com/llvm/llvm-project/issues/149511
-    if (is_linux && use_lld && current_cpu != "arm" && current_cpu != "s390x") {
+    if (is_linux && use_lld && current_cpu != "arm" && current_cpu != "s390x" && false) {
       cflags += [ "-Wa,--crel,--allow-experimental-crel" ]
     }
   }
