--- a/third_party/swiftshader/third_party/llvm-10.0/llvm/lib/ProfileData/SampleProf.cpp
+++ b/third_party/swiftshader/third_party/llvm-10.0/llvm/lib/ProfileData/SampleProf.cpp
@@ -215,8 +215,7 @@
 std::error_code ProfileSymbolList::write(raw_ostream &OS) {
   // Sort the symbols before output. If doing compression.
   // It will make the compression much more effective.
-  std::vector<StringRef> SortedList;
-  SortedList.insert(SortedList.begin(), Syms.begin(), Syms.end());
+  std::vector<StringRef> SortedList(Syms.begin(), Syms.end());
   llvm::sort(SortedList);
 
   std::string OutputString;
