--- a/third_party/blink/renderer/core/layout/geometry/box_edge.h
+++ b/third_party/blink/renderer/core/layout/geometry/box_edge.h
@@ -50,11 +50,11 @@
   constexpr bool operator==(const BoxEdge& other) const = default;
 };
 
-inline constexpr BoxEdge operator+(BoxEdge segment, LayoutUnit delta) {
+inline BoxEdge operator+(BoxEdge segment, LayoutUnit delta) {
   return BoxEdge(segment.offset + delta, segment.size);
 }
 
-inline constexpr BoxEdge operator-(BoxEdge segment, LayoutUnit delta) {
+inline BoxEdge operator-(BoxEdge segment, LayoutUnit delta) {
   return BoxEdge(segment.offset - delta, segment.size);
 }
 
