--- a/adapter/CMakeLists.txt
+++ b/adapter/CMakeLists.txt
@@ -31,8 +31,13 @@ add_custom_target(adapter
 
 set(CodelldbExe codelldb${ExeSuffix})
 set(CargoOutDir ${CMAKE_BINARY_DIR}/target/${LLVM_TRIPLE}/${CargoBuildType})
-
-if (${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
+if (${CMAKE_SYSTEM_NAME} STREQUAL "Android")
+    set(RustcFlags
+        -Csplit-debuginfo=packed
+        -Clink-arg=-fuse-ld=lld
+        -Clink-arg=-lc++
+    )
+elseif (${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
     set(RustcFlags
         -Csplit-debuginfo=packed
         -Clink-arg=-fuse-ld=lld
@@ -59,8 +59,6 @@ externalproject_add(cargo_build
     INSTALL_COMMAND ""
 )
 
-add_dependencies(cargo_build lldb)
-
 # Exe
 add_custom_target(codelldb_exe
     COMMAND ${UpdateFile} ${CargoOutDir}/${CodelldbExe} ${CMAKE_CURRENT_BINARY_DIR}
