diff -u -r ../swipl-7.3.6/src/configure ./src/configure
--- ../swipl-7.3.6/src/configure	2015-08-25 05:55:30.000000000 -0400
+++ ./src/configure	2015-09-16 17:52:41.545729451 -0400
@@ -9372,51 +9372,7 @@
 done
 
 
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for pthread support for cpu clocks" >&5
-$as_echo_n "checking for pthread support for cpu clocks... " >&6; }
-if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
-else
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-#include <stdio.h>
-#include <pthread.h>
-#include <time.h>
-#include <stdlib.h>
-#define ts2d(ts) \
-	((double)(ts).tv_sec + (double)(ts).tv_nsec/(double)1000000000.0)
-main()
-{
-  clockid_t clock_id;
-  struct timespec ts = {1,1};
-  if ( pthread_getcpuclockid(pthread_self(), &clock_id) != 0 )
-  { perror("pthread_getcpuclockid");
-    exit(1);
-  }
-  sleep(1);
-  if ( clock_gettime(clock_id, &ts) != 0 )
-  { perror("clock_gettime");
-    exit(1);
-  }
-  fprintf(stderr, "Used %f sec\n", ts2d(ts));
-  exit (ts.tv_sec == 0 ? 0 : 1);
-}
-_ACEOF
-if ac_fn_c_try_run "$LINENO"; then :
   ac_pthread_cpuclocks="yes"
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-$as_echo "yes" >&6; }
-else
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-fi
-rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
-  conftest.$ac_objext conftest.beam conftest.$ac_ext
-fi
 
 
 if test "x$ac_pthread_cpuclocks" = "xyes"; then
