--- ./src/include/gnunet_common.h	2021-12-08 19:14:43.938658667 +0530
+++ ./src/include/gnunet_common.h.mod	2021-12-08 19:35:22.328658195 +0530
@@ -143,7 +143,7 @@
  */
 
 #if __BYTE_ORDER == __LITTLE_ENDIAN
-#if defined(__linux__)
+#if defined(__linux__) && !defined(__ANDROID__)
 #define BYTE_SWAP_16(x) __bswap_16 (x)
 #define BYTE_SWAP_32(x) __bswap_32 (x)
 #define BYTE_SWAP_64(x) __bswap_64 (x)
