diff --git a/src/exit/gnunet-helper-exit.c b/src/exit/gnunet-helper-exit.c
index d460326..03beecc 100644
--- a/src/exit/gnunet-helper-exit.c
+++ b/src/exit/gnunet-helper-exit.c
@@ -85,7 +85,7 @@ static const char *sbin_sysctl;
 static const char *sbin_iptables;
 
 
-#if ! defined(_LINUX_IN6_H) && defined(__linux__)
+#if ! defined(_LINUX_IN6_H) && !defined(__ANDROID__)
 /**
  * This is in linux/include/net/ipv6.h, but not always exported.
  */
diff --git a/src/vpn/gnunet-helper-vpn.c b/src/vpn/gnunet-helper-vpn.c
index 4e35fa2..34eb018 100644
--- a/src/vpn/gnunet-helper-vpn.c
+++ b/src/vpn/gnunet-helper-vpn.c
@@ -59,7 +59,7 @@
  */
 #define MAX_SIZE 65536
 
-#ifndef _LINUX_IN6_H
+#if !defined(_LINUX_IN6_H) && !defined(__ANDROID__)
 /**
  * This is in linux/include/net/ipv6.h, but not always exported...
  */
