diff --git a/src/cadet/Makefile.am b/src/cadet/Makefile.am
index d8d81bf..cfc182b 100644
--- a/src/cadet/Makefile.am
+++ b/src/cadet/Makefile.am
@@ -72,7 +72,7 @@ gnunet_service_cadet_LDADD = \
   $(top_builddir)/src/hello/libgnunethello.la \
   $(top_builddir)/src/block/libgnunetblock.la
 if LINUX
-  gnunet_service_cadet_LDFLAGS = -lrt \
+  gnunet_service_cadet_LDFLAGS = \
     $(GN_LIBINTL)
 endif
 
