--- a/bitlbee.conf
+++ b/bitlbee.conf
@@ -108,14 +108,14 @@
 ## Specify an alternative MOTD (Message Of The Day) file. Default value depends
 ## on the --etcdir argument to configure.
 ##
-# MotdFile = /etc/bitlbee/motd.txt
+# MotdFile = @TERMUX_PREFIX@/etc/bitlbee/motd.txt
 
 ## ConfigDir
 ##
 ## Specify an alternative directory to store all the per-user configuration
 ## files. (.nicks/.accounts)
 ##
-# ConfigDir = /var/lib/bitlbee
+# ConfigDir = @TERMUX_PREFIX@/var/lib/bitlbee
 
 ## Ping settings
 ##
@@ -167,7 +167,7 @@
 ## The location of this file may be different on other distros/OSes. For
 ## example, try /etc/ssl/ca-bundle.pem on OpenSUSE.
 ##
-# CAfile = /etc/ssl/certs/ca-certificates.crt
+# CAfile = @TERMUX_PREFIX@/etc/tls/cert.pem
 
 [defaults]
 
