--- a/protocols/purple/ft-direct.c
+++ b/protocols/purple/ft-direct.c
@@ -123,7 +123,7 @@
 {
 	if (purple_xfer_get_type(xfer) == PURPLE_XFER_RECEIVE) {
 		/* This should suppress the stupid file dialog. */
-		purple_xfer_set_local_filename(xfer, "/tmp/wtf123");
+		purple_xfer_set_local_filename(xfer, "@TERMUX_PREFIX@/tmp/wtf123");
 
 		/* Sadly the xfer struct is still empty ATM so come back after
 		   the caller is done. */
