--- a/mime/CMakeLists.txt	2026-05-29 19:05:04.000000000 -0400
+++ b/mime/CMakeLists.txt	2026-06-13 13:04:42.442057972 -0400
@@ -3,7 +3,6 @@
 find_package(SharedMimeInfo REQUIRED)
 
 install(FILES palapeli-mimetypes.xml DESTINATION ${KDE_INSTALL_MIMEDIR})
-update_xdg_mimetypes(${KDE_INSTALL_MIMEDIR})
 
 # servicemenu integration
 install(PROGRAMS palapeli_servicemenu.desktop DESTINATION ${KDE_INSTALL_DATADIR}/kio/servicemenus)
