--- polybar-3.6.1/CMakeLists.txt	2022-03-05 21:29:15.000000000 +0800
+++ polybar-3.6.1.mod/CMakeLists.txt	2022-03-30 19:20:54.933488932 +0800
@@ -63,7 +63,7 @@
 
 if(BUILD_CONFIG)
   install(FILES ${CMAKE_SOURCE_DIR}/doc/config.ini
-    DESTINATION /etc/${PROJECT_NAME}
+	  DESTINATION @TERMUX_PREFIX@/etc/${PROJECT_NAME}
     COMPONENT config)
 endif()
 

--- polybar-3.6.1/cmake/02-opts.cmake	2022-03-05 21:29:15.000000000 +0800
+++ polybar-3.6.1.mod/cmake/02-opts.cmake	2022-03-30 19:21:52.283488910 +0800
@@ -1,6 +1,6 @@
 set(SETTING_ALSA_SOUNDCARD "default"
   CACHE STRING "Name of the ALSA soundcard driver")
-set(SETTING_BSPWM_SOCKET_PATH "/tmp/bspwm_0_0-socket"
+set(SETTING_BSPWM_SOCKET_PATH "@TERMUX_PREFIX@/tmp/bspwm_0_0-socket"
   CACHE STRING "Path to bspwm socket")
 set(SETTING_BSPWM_STATUS_PREFIX "W"
   CACHE STRING "Prefix prepended to the bspwm status line")
@@ -16,7 +16,7 @@
   CACHE STRING "Path to file containing cpu info")
 set(SETTING_PATH_MEMORY_INFO "/proc/meminfo"
   CACHE STRING "Path to file containing memory info")
-set(SETTING_PATH_MESSAGING_FIFO "/tmp/polybar_mqueue.%pid%"
+set(SETTING_PATH_MESSAGING_FIFO "@TERMUX_PREFIX@/tmp/polybar_mqueue.%pid%"
   CACHE STRING "Path to file containing the current temperature")
 set(SETTING_PATH_TEMPERATURE_INFO "/sys/class/thermal/thermal_zone%zone%/temp"
   CACHE STRING "Path to file containing the current temperature")
