diff -uNr polybar/src/ipc.cpp polybar.mod/src/ipc.cpp
--- polybar/src/ipc.cpp	2019-04-03 08:32:20.000000000 +0300
+++ polybar.mod/src/ipc.cpp	2019-07-10 22:14:11.763825428 +0300
@@ -12,7 +12,7 @@
 using namespace std;
 
 #ifndef IPC_CHANNEL_PREFIX
-#define IPC_CHANNEL_PREFIX "/tmp/polybar_mqueue."
+#define IPC_CHANNEL_PREFIX "@TERMUX_PREFIX@/tmp/polybar_mqueue."
 #endif
 
 void log(const string& msg) {
diff -uNr polybar/src/utils/bspwm.cpp polybar.mod/src/utils/bspwm.cpp
--- polybar/src/utils/bspwm.cpp	2019-04-03 08:32:20.000000000 +0300
+++ polybar.mod/src/utils/bspwm.cpp	2019-07-10 22:14:35.983493128 +0300
@@ -85,7 +85,7 @@
       return BSPWM_SOCKET_PATH;
     }
 
-    snprintf(sa.sun_path, sizeof(sa.sun_path), "/tmp/bspwm%s_%i_%i-socket", host, dsp, scr);
+    snprintf(sa.sun_path, sizeof(sa.sun_path), "@TERMUX_PREFIX@/tmp/bspwm%s_%i_%i-socket", host, dsp, scr);
     free(host);
 
     return sa.sun_path;
