diff --git a/crates/jit/src/profiling.rs b/crates/jit/src/profiling.rs
index d816d65..fa1f319 100644
--- a/crates/jit/src/profiling.rs
+++ b/crates/jit/src/profiling.rs
@@ -33,7 +33,7 @@ cfg_if::cfg_if! {
 cfg_if::cfg_if! {
     // Note: VTune support is disabled on windows mingw because the ittapi crate doesn't compile
     // there; see also https://github.com/bytecodealliance/wasmtime/pull/4003 for rationale.
-    if #[cfg(all(feature = "vtune", target_arch = "x86_64", not(all(target_os = "windows", target_env = "gnu"))))] {
+    if #[cfg(all(feature = "vtune", target_arch = "x86_64", not(all(any(target_os = "android", target_os = "windows", target_env = "gnu")))))] {
         mod vtune;
         pub use vtune::new as new_vtune;
     } else {
