diff --git a/crates/jit/Cargo.toml b/crates/jit/Cargo.toml
index 25cf74b..13fccaf 100644
--- a/crates/jit/Cargo.toml
+++ b/crates/jit/Cargo.toml
@@ -37,7 +37,7 @@ features = [
   "Win32_System_Diagnostics_Debug",
 ]
 
-[target.'cfg(target_arch = "x86_64")'.dependencies]
+[target.'cfg(all(target_arch = "x86_64", not(target_os = "android")))'.dependencies]
 ittapi = { version = "0.3.3", optional = true  }
 
 [features]
