diff -uNr inetutils-1.9.4/configure.orig inetutils-1.9.4/configure
--- ./configure.orig	2022-06-11 17:35:38.249779008 +0000
+++ ./configure	2022-06-11 17:37:22.492718680 +0000
@@ -17193,7 +17193,7 @@
           else
             if case "$gl_cv_have_weak" in *yes) true;; *) false;; esac; then
 
-printf "%s\n" "#define USE_POSIX_THREADS_WEAK 1" >>confdefs.h
+#printf "%s\n" "#define USE_POSIX_THREADS_WEAK 1" >>confdefs.h
 
               LIBTHREAD= LTLIBTHREAD=
             else
