--- ./metasploit-framework.gemspec	2017-08-25 15:09:02.000000000 +0000
+++ ../metasploit-framework.gemspec	2017-09-05 19:08:57.440573054 +0000
@@ -124,8 +124,6 @@
   #
   spec.add_runtime_dependency 'dnsruby'
   spec.add_runtime_dependency 'net-ssh'
-  spec.add_runtime_dependency 'rbnacl', ['< 5.0.0']
-  spec.add_runtime_dependency 'rbnacl-libsodium'
   spec.add_runtime_dependency 'bcrypt_pbkdf'
   spec.add_runtime_dependency 'ruby_smb'
 
