From e4450d8b586103ec1414e2d6245ff34a9096d97a Mon Sep 17 00:00:00 2001
From: Peter Levine <plevine457@gmail.com>
Date: Thu, 26 Sep 2019 07:24:40 +0000
Subject: [PATCH] atomic: Explicitly cast void* to unsigned long


diff --git a/src/pulsecore/atomic.h b/src/pulsecore/atomic.h
index a82ca83c5..e5c140109 100644
--- a/src/pulsecore/atomic.h
+++ b/src/pulsecore/atomic.h
@@ -117,7 +117,7 @@ static inline void* pa_atomic_ptr_load(const pa_atomic_ptr_t *a) {
 }
 
 static inline void pa_atomic_ptr_store(pa_atomic_ptr_t *a, void* p) {
-    __atomic_store_n(&a->value, p, __ATOMIC_SEQ_CST);
+    __atomic_store_n(&a->value, (unsigned long) p, __ATOMIC_SEQ_CST);
 }
 
 #else
-- 
2.23.0

