diff --git a/meson.build b/meson.build
index c5135330f..0f0c3d5ed 100644
--- a/meson.build
+++ b/meson.build
@@ -279,10 +279,6 @@ if host_machine.system() != 'windows'
   endif
 endif
 
-if cc.has_header_symbol('pthread.h', 'PTHREAD_PRIO_INHERIT')
-  cdata.set('HAVE_PTHREAD_PRIO_INHERIT', 1)
-endif
-
 # Headers which are usable
 
 check_usable_headers = [
@@ -374,7 +370,7 @@ if cc.has_function('dgettext')
     libintl_dep = cc.find_library('intl')
   endif
 else
-  libintl_dep = cc.find_library('intl')
+  libintl_dep = []
 endif
 
 # Symbols
@@ -440,6 +436,8 @@ endif
 # Core Dependencies
 
 libm_dep = cc.find_library('m', required : true)
+sles_dep = cc.find_library('OpenSLES', required : true)
+aaudio_dep = cc.find_library('aaudio', required : true)
 
 thread_dep = dependency('threads')
 foreach f : [
diff --git a/src/modules/meson.build b/src/modules/meson.build
index 1e12569dc..6112480e2 100644
--- a/src/modules/meson.build
+++ b/src/modules/meson.build
@@ -61,6 +61,9 @@ all_modules = [
   [ 'module-virtual-sink', 'module-virtual-sink.c' ],
   [ 'module-virtual-source', 'module-virtual-source.c' ],
   [ 'module-volume-restore', 'module-volume-restore.c' ],
+  [ 'module-sles-sink', 'sles/module-sles-sink.c', [], [], [sles_dep] ],
+  [ 'module-sles-source', 'sles/module-sles-source.c', [], [], [sles_dep] ],
+  [ 'module-aaudio-sink', 'aaudio/module-aaudio-sink.c', [], [], [aaudio_dep] ],
 ]
 
 if host_machine.system() == 'windows'
diff --git a/src/pulsecore/meson.build b/src/pulsecore/meson.build
index b37fec499..2cdac04e9 100644
--- a/src/pulsecore/meson.build
+++ b/src/pulsecore/meson.build
@@ -185,14 +185,7 @@ libpulsecore_simd_lib = []
 foreach simd_kwargs : simd_variants
 
   if host_machine.cpu_family() == 'arm' and 'neon' in simd_kwargs
-    if not cc.compiles('''
-        #include <arm_neon.h>
-        int main() {
-            return sizeof(uint8x8_t) + sizeof(int32x4_t) + sizeof(float32x4_t);
-        }
-        ''', name : 'neon code')
-      continue
-    endif
+    continue
   endif
 
   libpulsecore_simd = simd.check('libpulsecore_simd',
diff --git a/src/tests/meson.build b/src/tests/meson.build
index dceca55c7..04993ea1c 100644
--- a/src/tests/meson.build
+++ b/src/tests/meson.build
@@ -106,8 +106,6 @@ if get_option('daemon')
     default_tests += [
       [ 'sigbus-test', 'sigbus-test.c',
         [ check_dep, libpulse_dep, libpulsecommon_dep, libpulsecore_dep ] ],
-      [ 'usergroup-test', 'usergroup-test.c',
-        [ check_dep, libpulse_dep, libpulsecommon_dep, libpulsecore_dep ] ],
     ]
   endif
 
