diff -u -r ../libssh2-1.8.0/src/openssl.h ./src/openssl.h
--- ../libssh2-1.8.0/src/openssl.h	2016-02-17 21:59:57.000000000 +0000
+++ ./src/openssl.h	2018-09-12 04:35:42.346414320 +0000
@@ -226,10 +226,15 @@
 #define libssh2_hmac_cleanup(ctx) HMAC_cleanup(ctx)
 #endif
 
+#ifdef OPENSSL_NO_ENGINE
+#define libssh2_crypto_init() \
+  OpenSSL_add_all_algorithms()
+#else
 #define libssh2_crypto_init() \
   OpenSSL_add_all_algorithms(); \
   ENGINE_load_builtin_engines(); \
   ENGINE_register_all_complete()
+#endif
 
 #define libssh2_crypto_exit()
 
