diff -u -r ../multitail-6.2.1/mt.h ./mt.h
--- ../multitail-6.2.1/mt.h	2014-02-16 15:06:54.000000000 +0100
+++ ./mt.h	2014-06-30 16:38:06.000000000 +0200
@@ -49,8 +49,8 @@
 #endif
 
 #ifdef UTF8_SUPPORT
-	#include <ncursesw/panel.h>
-	#include <ncursesw/ncurses.h>
+	#include <panel.h>
+	#include <ncurses.h>
 #else
 	#if defined(sun) || defined(__sun) || defined(scoos) || defined(_HPUX_SOURCE) || defined(AIX) || defined(__CYGWIN__)
 		#include <ncurses/panel.h>
