--- a/src/glad/include/KHR/khrplatform.h
+++ b/src/glad/include/KHR/khrplatform.h
@@ -101,7 +101,7 @@
 #   define KHRONOS_APICALL __declspec(dllimport)
 #elif defined (__SYMBIAN32__)
 #   define KHRONOS_APICALL IMPORT_C
-#elif defined(__ANDROID__)
+#elif defined(__ANDROID__) && !defined(__TERMUX__)
 #   include <sys/cdefs.h>
 #   define KHRONOS_APICALL __attribute__((visibility("default"))) __NDK_FPABI__
 #else
