--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -188,7 +188,7 @@
 httrack_DEPENDENCIES = $(am__DEPENDENCIES_1) libhttrack.la
 httrack_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(httrack_CFLAGS) \
-	$(CFLAGS) $(httrack_LDFLAGS) $(LDFLAGS) -o $@
+	$(CFLAGS) $(httrack_LDFLAGS) $(LDFLAGS) -landroid-execinfo -o $@
 am_proxytrack_OBJECTS = proxy/proxytrack-main.$(OBJEXT) \
 	proxy/proxytrack-proxytrack.$(OBJEXT) \
 	proxy/proxytrack-store.$(OBJEXT) \
