--- httrack-3.49.2/src/proxy/store.c	2017-04-02 01:08:17.000000000 +0530
+++ store.c	2018-04-10 12:10:59.385000003 +0530
@@ -1941,7 +1941,7 @@
 
   if (t != (time_t) - 1 && t != (time_t) 0) {
     /* BSD does not have static "timezone" declared */
-#if (defined(BSD) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__) || defined(__FreeBSD_kernel__))
+#if (defined(BSD) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__) || defined(__FreeBSD_kernel__)) || defined(__ANDROID__)
     time_t now = time(NULL);
     time_t timezone = -localtime(&now)->tm_gmtoff;
 #endif
