https://github.com/termux/termux-packages/issues/15852

--- a/src/eval.c
+++ b/src/eval.c
@@ -240,8 +240,8 @@
 GType buffer_get_type (void) G_GNUC_CONST ;
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (Buffer, buffer_unref)
 GType completion_flags_get_type (void) G_GNUC_CONST ;
-static void ___lambda25__gh_func (gconstpointer key,
-                           gconstpointer value,
+static void ___lambda25__gh_func (gpointer key,
+                           gpointer value,
                            gpointer self);
 gchar* minibuf_vread_completion (const gchar* fmt,
                                  const gchar* val,
@@ -941,8 +941,8 @@
 }
 
 static void
-___lambda25__gh_func (gconstpointer key,
-                      gconstpointer value,
+___lambda25__gh_func (gpointer key,
+                      gpointer value,
                       gpointer self)
 {
 #line 87 "src/eval.vala"
--- a/src/variables.c
+++ b/src/variables.c
@@ -300,8 +300,8 @@
                   const gchar* key,
                   VarEntry* val);
 GType completion_flags_get_type (void) G_GNUC_CONST ;
-static void ___lambda91__gh_func (gconstpointer key,
-                           gconstpointer value,
+static void ___lambda91__gh_func (gpointer key,
+                           gpointer value,
                            gpointer self);
 gpointer history_ref (gpointer instance);
 void history_unref (gpointer instance);
@@ -1201,8 +1201,8 @@
 }
 
 static void
-___lambda91__gh_func (gconstpointer key,
-                      gconstpointer value,
+___lambda91__gh_func (gpointer key,
+                      gpointer value,
                       gpointer self)
 {
 #line 103 "src/variables.vala"
