diff -u -r ../libsndfile-826d5296da54c016e3cb0f7f00de3b9e295b9c4a/autogen.sh ./autogen.sh
--- ../libsndfile-826d5296da54c016e3cb0f7f00de3b9e295b9c4a/autogen.sh	2017-10-31 09:30:54.000000000 +0100
+++ ./autogen.sh	2017-11-15 23:12:23.210491716 +0100
@@ -11,18 +11,6 @@
 cd "$srcdir"
 DIE=0
 
-printf "checking for autogen ... "
-result="yes"
-(autogen --version) < /dev/null > /dev/null 2>&1 || {
-        echo
-        echo "You must have GNU autogen installed to compile $package."
-        echo "Download the appropriate package for your distribution,"
-        echo "or get the source tarball at ftp://ftp.gnu.org/pub/gnu/"
-		result="no"
-        DIE=1
-}
-echo $result
-
 printf "checking for autoconf ... "
 result="yes"
 (autoconf --version) < /dev/null > /dev/null 2>&1 || {
