diff -uNr qemu-3.0.0/util/qemu-sockets.c qemu-3.0.0.mod/util/qemu-sockets.c
--- qemu-3.0.0/util/qemu-sockets.c	2018-08-14 22:10:35.000000000 +0300
+++ qemu-3.0.0.mod/util/qemu-sockets.c	2018-10-26 16:49:45.306931999 +0300
@@ -841,7 +841,7 @@
         path = saddr->path;
     } else {
         const char *tmpdir = getenv("TMPDIR");
-        tmpdir = tmpdir ? tmpdir : "/tmp";
+        tmpdir = tmpdir ? tmpdir : "@TERMUX_PREFIX@/tmp";
         path = pathbuf = g_strdup_printf("%s/qemu-socket-XXXXXX", tmpdir);
     }
 
