diff -uNr qemu-3.0.0/include/net/net.h qemu-3.0.0.mod/include/net/net.h
--- qemu-3.0.0/include/net/net.h	2018-08-14 22:10:34.000000000 +0300
+++ qemu-3.0.0.mod/include/net/net.h	2018-10-26 16:49:45.260265074 +0300
@@ -216,8 +216,8 @@
 int net_hub_id_for_client(NetClientState *nc, int *id);
 NetClientState *net_hub_port_find(int hub_id);
 
-#define DEFAULT_NETWORK_SCRIPT "/etc/qemu-ifup"
-#define DEFAULT_NETWORK_DOWN_SCRIPT "/etc/qemu-ifdown"
+#define DEFAULT_NETWORK_SCRIPT "@TERMUX_PREFIX@/etc/qemu-ifup"
+#define DEFAULT_NETWORK_DOWN_SCRIPT "@TERMUX_PREFIX@/etc/qemu-ifdown"
 #define DEFAULT_BRIDGE_HELPER CONFIG_QEMU_HELPERDIR "/qemu-bridge-helper"
 #define DEFAULT_BRIDGE_INTERFACE "br0"
 
