diff -uNr qemu-3.0.0/net/tap.c qemu-3.0.0.mod/net/tap.c
--- qemu-3.0.0/net/tap.c	2018-08-14 22:10:34.000000000 +0300
+++ qemu-3.0.0.mod/net/tap.c	2018-10-26 16:49:45.286931887 +0300
@@ -526,7 +526,7 @@
             *parg++ = helper_cmd;
             *parg++ = NULL;
 
-            execv("/bin/sh", args);
+            execv("@TERMUX_PREFIX@/bin/sh", args);
         } else {
             /* assume helper is just the executable path name */
 
