diff -uNr qemu-3.0.0/tcg/tcg.c qemu-3.0.0.mod/tcg/tcg.c
--- qemu-3.0.0/tcg/tcg.c	2018-08-14 22:10:35.000000000 +0300
+++ qemu-3.0.0.mod/tcg/tcg.c	2018-10-26 16:49:45.303598647 +0300
@@ -3909,7 +3909,7 @@
     /* Enable this block to be able to debug the ELF image file creation.
        One can use readelf, objdump, or other inspection utilities.  */
     {
-        FILE *f = fopen("/tmp/qemu.jit", "w+b");
+        FILE *f = fopen("@TERMUX_PREFIX@/tmp/qemu.jit", "w+b");
         if (f) {
             if (fwrite(img, img_size, 1, f) != img_size) {
                 /* Avoid stupid unused return value warning for fwrite.  */
