diff -uNr qemu-3.0.0/qemu-nbd.c qemu-3.0.0.mod/qemu-nbd.c
--- qemu-3.0.0/qemu-nbd.c	2018-08-14 22:10:34.000000000 +0300
+++ qemu-3.0.0.mod/qemu-nbd.c	2018-10-26 16:49:45.290265240 +0300
@@ -43,7 +43,7 @@
 #include "trace/control.h"
 #include "qemu-version.h"
 
-#define SOCKET_PATH                "/var/lock/qemu-nbd-%s"
+#define SOCKET_PATH                "@TERMUX_PREFIX@/var/lock/qemu-nbd-%s"
 #define QEMU_NBD_OPT_CACHE         256
 #define QEMU_NBD_OPT_AIO           257
 #define QEMU_NBD_OPT_DISCARD       258
