diff -uNr qemu-2.12.1/contrib/ivshmem-server/main.c qemu-2.12.1.mod/contrib/ivshmem-server/main.c
--- qemu-2.12.1/contrib/ivshmem-server/main.c	2018-08-03 00:48:53.000000000 +0300
+++ qemu-2.12.1.mod/contrib/ivshmem-server/main.c	2018-09-17 16:27:46.974062341 +0300
@@ -14,8 +14,8 @@
 
 #define IVSHMEM_SERVER_DEFAULT_VERBOSE        0
 #define IVSHMEM_SERVER_DEFAULT_FOREGROUND     0
-#define IVSHMEM_SERVER_DEFAULT_PID_FILE       "/var/run/ivshmem-server.pid"
-#define IVSHMEM_SERVER_DEFAULT_UNIX_SOCK_PATH "/tmp/ivshmem_socket"
+#define IVSHMEM_SERVER_DEFAULT_PID_FILE       "@TERMUX_PREFIX@/var/run/ivshmem-server.pid"
+#define IVSHMEM_SERVER_DEFAULT_UNIX_SOCK_PATH "@TERMUX_PREFIX@/tmp/ivshmem_socket"
 #define IVSHMEM_SERVER_DEFAULT_SHM_PATH       "ivshmem"
 #define IVSHMEM_SERVER_DEFAULT_SHM_SIZE       (4*1024*1024)
 #define IVSHMEM_SERVER_DEFAULT_N_VECTORS      1
