diff -uNr qemu-2.12.1/linux-user/syscall_defs.h qemu-2.12.1.mod/linux-user/syscall_defs.h
--- qemu-2.12.1/linux-user/syscall_defs.h	2018-08-03 00:48:53.000000000 +0300
+++ qemu-2.12.1.mod/linux-user/syscall_defs.h	2018-09-17 16:27:47.007395873 +0300
@@ -1967,7 +1967,7 @@
        abi_ulong    target_st_mtime_nsec;
        abi_ulong    target_st_ctime;
        abi_ulong    target_st_ctime_nsec;
-       abi_long     __unused[3];
+       abi_long     __qemu_unused[3];
 };
 
 #elif defined(TARGET_SH4)
@@ -2052,7 +2052,7 @@
 	abi_ulong	target_st_ctime;
 	abi_ulong       target_st_ctime_nsec;
 
-  	abi_long	__unused[3];
+  	abi_long	__qemu_unused[3];
 };
 #elif defined(TARGET_S390X)
 struct target_stat {
@@ -2073,7 +2073,7 @@
     abi_ulong  target_st_ctime_nsec;
     abi_ulong  st_blksize;
     abi_long       st_blocks;
-    abi_ulong  __unused[3];
+    abi_ulong  __qemu_unused[3];
 };
 #elif defined(TARGET_AARCH64)
 struct target_stat {
@@ -2095,7 +2095,7 @@
     abi_ulong  target_st_mtime_nsec;
     abi_long  target_st_ctime;
     abi_ulong  target_st_ctime_nsec;
-    unsigned int __unused[2];
+    unsigned int __qemu_unused[2];
 };
 #elif defined(TARGET_XTENSA)
 struct target_stat {
