diff -uNr qemu-2.12.1/qemu-nbd.c qemu-2.12.1.mod/qemu-nbd.c
--- qemu-2.12.1/qemu-nbd.c	2018-08-03 00:48:53.000000000 +0300
+++ qemu-2.12.1.mod/qemu-nbd.c	2018-09-17 16:27:47.017395933 +0300
@@ -43,7 +43,7 @@
 #include "trace/control.h"
 #include "qemu-version.h"
 
-#define SOCKET_PATH                "/var/lock/qemu-nbd-%s"
+#define SOCKET_PATH                "@TERMUX_PREFIX@/var/lock/qemu-nbd-%s"
 #define QEMU_NBD_OPT_CACHE         256
 #define QEMU_NBD_OPT_AIO           257
 #define QEMU_NBD_OPT_DISCARD       258
