diff -uNr qemu-3.0.0/tcg/i386/tcg-target.inc.c qemu-3.0.0.mod/tcg/i386/tcg-target.inc.c
--- qemu-3.0.0/tcg/i386/tcg-target.inc.c	2018-08-14 22:10:35.000000000 +0300
+++ qemu-3.0.0.mod/tcg/i386/tcg-target.inc.c	2018-10-26 16:49:45.303598647 +0300
@@ -1869,8 +1869,12 @@
 #elif defined(__x86_64__) && defined(__linux__)
 # include <asm/prctl.h>
 # include <sys/prctl.h>
+# include <linux/unistd.h>
 
-int arch_prctl(int code, unsigned long addr);
+static int arch_prctl(int code, unsigned long addr)
+{
+    return syscall(__NR_arch_prctl, code, addr);
+}
 
 static int guest_base_flags;
 static inline void setup_guest_base_seg(void)
