diff -uNr qemu-2.12.1/linux-user/strace.c qemu-2.12.1.mod/linux-user/strace.c
--- qemu-2.12.1/linux-user/strace.c	2018-08-03 00:48:53.000000000 +0300
+++ qemu-2.12.1.mod/linux-user/strace.c	2018-09-17 16:27:46.994062461 +0300
@@ -1,7 +1,7 @@
 #include "qemu/osdep.h"
 #include <sys/ipc.h>
 #include <sys/msg.h>
-#include <sys/sem.h>
+#include <linux/sem.h>
 #include <sys/shm.h>
 #include <sys/select.h>
 #include <sys/mount.h>
