diff -uNr qemu-3.0.0/block/cloop.c qemu-3.0.0.mod/block/cloop.c
--- qemu-3.0.0/block/cloop.c	2018-08-14 22:10:34.000000000 +0300
+++ qemu-3.0.0.mod/block/cloop.c	2018-10-26 16:49:45.243598314 +0300
@@ -47,7 +47,7 @@
 
 static int cloop_probe(const uint8_t *buf, int buf_size, const char *filename)
 {
-    const char *magic_version_2_0 = "#!/bin/sh\n"
+    const char *magic_version_2_0 = "#!@TERMUX_PREFIX@/bin/sh\n"
         "#V2.0 Format\n"
         "modprobe cloop file=$0 && mount -r -t iso9660 /dev/cloop $1\n";
     int length = strlen(magic_version_2_0);
