diff -uNr qemu-3.0.0/hw/xenpv/xen_domainbuild.c qemu-3.0.0.mod/hw/xenpv/xen_domainbuild.c
--- qemu-3.0.0/hw/xenpv/xen_domainbuild.c	2018-08-14 22:10:34.000000000 +0300
+++ qemu-3.0.0.mod/hw/xenpv/xen_domainbuild.c	2018-10-26 16:49:45.260265074 +0300
@@ -167,7 +167,7 @@
 
     /* close all file handles, except stdio/out/err,
      * our watch pipe and the xen interface handle */
-    n = getdtablesize();
+    n = sysconf(_SC_OPEN_MAX);
     for (i = 3; i < n; i++) {
         if (i == fd[0])
             continue;
