diff -uNr qemu-3.0.0/linux-user/arm/signal.c qemu-3.0.0.mod/linux-user/arm/signal.c
--- qemu-3.0.0/linux-user/arm/signal.c	2018-08-14 22:10:34.000000000 +0300
+++ qemu-3.0.0.mod/linux-user/arm/signal.c	2018-10-26 16:49:45.263598425 +0300
@@ -59,7 +59,7 @@
     target_stack_t tuc_stack;
     struct target_sigcontext tuc_mcontext;
     target_sigset_t  tuc_sigmask;       /* mask last for extensibility */
-    char __unused[128 - sizeof(target_sigset_t)];
+    char __qemu_unused[128 - sizeof(target_sigset_t)];
     abi_ulong tuc_regspace[128] __attribute__((__aligned__(8)));
 };
 
