diff -uNr qemu-3.0.0/linux-user/aarch64/signal.c qemu-3.0.0.mod/linux-user/aarch64/signal.c
--- qemu-3.0.0/linux-user/aarch64/signal.c	2018-08-14 22:10:34.000000000 +0300
+++ qemu-3.0.0.mod/linux-user/aarch64/signal.c	2018-10-26 16:49:45.263598425 +0300
@@ -38,7 +38,7 @@
     target_stack_t tuc_stack;
     target_sigset_t tuc_sigmask;
     /* glibc uses a 1024-bit sigset_t */
-    char __unused[1024 / 8 - sizeof(target_sigset_t)];
+    char __qemu_unused[1024 / 8 - sizeof(target_sigset_t)];
     /* last for future expansion */
     struct target_sigcontext tuc_mcontext;
 };
