diff -u -r ../notmuch-0.25/configure ./configure
--- ../notmuch-0.25/configure	2017-07-25 13:30:24.000000000 +0200
+++ ./configure	2017-07-26 14:35:27.777600722 +0200
@@ -370,7 +370,7 @@
     return 0;
 }
 EOF
-if ${CC} ${CFLAGS} -I"$srcdir" _libversion.c -o _libversion > /dev/null 2>&1 \
+if ${CC_FOR_BUILD} -I"$srcdir" _libversion.c -o _libversion > /dev/null 2>&1 \
        && ./_libversion > _libversion.sh && . ./_libversion.sh
 then
     printf "OK.\n"
@@ -579,12 +579,7 @@
 fi
 
 printf "Checking for bash-completion (>= 1.90)... "
-if pkg-config --atleast-version=1.90 bash-completion; then
     printf "Yes.\n"
-else
-    printf "No (will not install bash completion).\n"
-    WITH_BASH=0
-fi
 
 if [ -z "${EMACSLISPDIR-}" ]; then
     EMACSLISPDIR="\$(prefix)/share/emacs/site-lisp"
