--- a/llama/llama.go
+++ b/llama/llama.go
@@ -40,8 +40,8 @@
 #cgo linux CXXFLAGS: -D_GNU_SOURCE
 #cgo linux LDFLAGS: -ldl
 #cgo linux,amd64 LDFLAGS: -L${SRCDIR}/build/linux-amd64
-#cgo linux,arm64 CFLAGS: -D__aarch64__ -D__ARM_NEON -D__ARM_FEATURE_FMA
-#cgo linux,arm64 CXXFLAGS: -D__aarch64__ -D__ARM_NEON -D__ARM_FEATURE_FMA
+#cgo linux,arm64 CFLAGS: -D__aarch64__ -D__ARM_NEON
+#cgo linux,arm64 CXXFLAGS: -D__aarch64__ -D__ARM_NEON
 #cgo linux,arm64 LDFLAGS: -L${SRCDIR}/build/linux-arm64
 #cgo linux,arm64,sve CFLAGS: -march=armv8.6-a+sve
 #cgo linux,arm64,sve CXXFLAGS: -march=armv8.6-a+sve
