--- a/ml/backend/ggml/ggml/src/mem_hip.cpp
+++ b/ml/backend/ggml/ggml/src/mem_hip.cpp
@@ -434,6 +434,20 @@
 
 } // extern "C"
 
+#elif defined(__ANDROID__)
+
+extern "C" {
+
+int ggml_hip_mgmt_init() {
+    return 0;
+}
+void ggml_hip_mgmt_release() {}
+int ggml_hip_get_device_memory(const char *id, size_t *free, size_t *total, bool is_integrated_gpu) {
+    return -1;
+}
+
+} // extern "C"
+
 #else // #ifdef _WIN32
 
 #include <fstream>
