--- a/Makefile.am	2013-04-13 09:26:53.000000000 -0400
+++ b/Makefile.am	2026-07-29 22:50:06.027235725 -0400
@@ -3,7 +3,7 @@
 ACLOCAL_AMFLAGS=-I m4
 
 # Subdirectories to scan
-SUBDIRS = build/mingw po src tests
+SUBDIRS = build/mingw po src
 
 if USE_DOCGEN
   SUBDIRS += docs
--- a/configure.ac	2016-04-04 01:59:39.000000000 -0400
+++ b/configure.ac	2026-07-29 22:51:08.891235725 -0400
@@ -36,7 +36,6 @@
   docs/Makefile
   po/Makefile.in
   src/Makefile
-  tests/Makefile
   tifiles2.pc
 ])
 
