diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 35683ef..d454e3b 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -84,11 +84,9 @@ install(
 if(NOT ANDROID)
     if(SharedMimeInfo_VERSION_STRING VERSION_LESS 2.2)
         install(FILES application-vnd-apple-pkpass.xml DESTINATION ${KDE_INSTALL_MIMEDIR})
-        update_xdg_mimetypes(${KDE_INSTALL_MIMEDIR})
     endif()
     if(SharedMimeInfo_VERSION_STRING VERSION_LESS 2.5)
         install(FILES application-vnd-apple-pkpasses.xml DESTINATION ${KDE_INSTALL_MIMEDIR})
-        update_xdg_mimetypes(${KDE_INSTALL_MIMEDIR})
     endif()
 endif()
 
