--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -86,14 +86,12 @@
             FILES application-vnd-apple-pkpass.xml
             DESTINATION ${KDE_INSTALL_MIMEDIR}
         )
-        update_xdg_mimetypes(${KDE_INSTALL_MIMEDIR})
     endif()
     if (SharedMimeInfo_VERSION_STRING VERSION_LESS 2.5)
         install(
             FILES application-vnd-apple-pkpasses.xml
             DESTINATION ${KDE_INSTALL_MIMEDIR}
         )
-        update_xdg_mimetypes(${KDE_INSTALL_MIMEDIR})
     endif()
 endif()
