--- a/src/buildtools/config/linux.pri
+++ b/src/buildtools/config/linux.pri
@@ -83,7 +83,7 @@
             gn_args += arm_use_neon=false
             # If the toolchain does not explicitly specify to use NEON instructions
             # we use arm_neon_optional for ARMv7
-            equals(MARMV, 7): gn_args += arm_optionally_use_neon=true
+            # equals(MARMV, 7): gn_args += arm_optionally_use_neon=true
         }
     }
 
