https://github.com/chromium/chromium/commit/9f43d823b6b4cdea62f0cc7563ff01f9239b8970

--- a/src/3rdparty/chromium/third_party/crc32c/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/crc32c/BUILD.gn
@@ -122,6 +122,10 @@
         "-target-feature",
         "-Xclang",
         "+crypto",
+        "-Xclang",
+        "-target-feature",
+        "-Xclang",
+        "+aes",
       ]
     } else {
       cflags = [ "-march=armv8-a+crc+crypto" ]
