--- a/src/makefile
+++ b/src/makefile
@@ -32,10 +32,10 @@
 CFLAGS += $(DEBUGFLAGS)
 
 #  Libraries and linker flags for edbrowse.
-LDFLAGS = $(STRIP) $(LINKER_LIBS) -lpthread -lm -lssl -lcrypto
+LDFLAGS = $(STRIP) $(LINKER_LIBS) -lpthread -lm -lssl -lcrypto -landroid-glob
 
 # LDFLAGS for quickjs loading.
-QUICKJS_LDFLAGS = $(QUICKJS_LIB)/libquickjs.a -ldl
+QUICKJS_LDFLAGS = $(PREFIX)/lib/quickjs/libquickjs.a -ldl
 ifeq ($(shell uname),Linux)
        QUICKJS_LDFLAGS += -latomic
 endif
