--- a/src/makefile
+++ b/src/makefile
@@ -31,10 +31,10 @@
 CFLAGS += $(DEBUGFLAGS)
 
 #  Libraries and linker flags for edbrowse.
-LDFLAGS = $(STRIP) $(LINKER_LIBS) -lpthread -lm -lssl -lcrypto
+LDFLAGS += $(STRIP) $(LINKER_LIBS) -lpthread -landroid-glob -lm -lssl -lcrypto
 
 # LDFLAGS for quickjs loading.
-QUICKJS_LDFLAGS = $(QUICKJS_DIR)/libquickjs.a -ldl
+QUICKJS_LDFLAGS = -L$(PREFIX)/lib/quickjs -lquickjs -ldl
 ifeq ($(shell uname),Linux)
 	QUICKJS_LDFLAGS += -latomic
 endif
