--- a/chntpw-presplit.c
+++ b/chntpw-presplit.c
@@ -65,7 +65,7 @@
 #include <fcntl.h>
 #include <ctype.h>
 #include <stdlib.h>
-#include <string.h>
+#include <strings.h>
 #include <unistd.h>
 #include <sys/types.h>
 #include <inttypes.h>
--- a/chntpw.c
+++ b/chntpw.c
@@ -67,7 +67,7 @@
 #include <fcntl.h>
 #include <ctype.h>
 #include <stdlib.h>
-#include <string.h>
+#include <strings.h>
 #include <unistd.h>
 #include <sys/types.h>
 #include <inttypes.h>
--- a/cpnt.c
+++ b/cpnt.c
@@ -28,7 +28,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <errno.h>
-#include <string.h>
+#include <strings.h>
 #include <unistd.h>
 #include <inttypes.h>
 
--- a/edlib.c
+++ b/edlib.c
@@ -45,7 +45,7 @@
 #include <stdio.h>
 #include <unistd.h>
 #include <stdlib.h>
-#include <string.h>
+#include <strings.h>
 #include <ctype.h>
 
 #include "ntreg.h"
--- a/hash.c
+++ b/hash.c
@@ -2,7 +2,7 @@
 
 #include <stdio.h>
 #include <ctype.h>
-#include <string.h>
+#include <strings.h>
 
 
 static const unsigned char charset2upper[] = {
--- a/libsam.c
+++ b/libsam.c
@@ -42,7 +42,7 @@
 #include <stdio.h>
 #include <sys/types.h>
 #include <stdlib.h>
-#include <string.h>
+#include <strings.h>
 
 #include "ntreg.h"
 #include "sam.h"
--- a/ntreg.c
+++ b/ntreg.c
@@ -77,7 +77,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <errno.h>
-#include <string.h>
+#include <strings.h>
 #include <unistd.h>
 #include <inttypes.h>
 #include <stdarg.h>
--- a/sampasswd.c
+++ b/sampasswd.c
@@ -31,7 +31,7 @@
 #include <stdio.h>
 #include <unistd.h>
 #include <stdlib.h>
-#include <string.h>
+#include <strings.h>
 
 #include "ntreg.h"
 #include "sam.h"
--- a/samusrgrp.c
+++ b/samusrgrp.c
@@ -36,7 +36,7 @@
 #include <stdio.h>
 #include <unistd.h>
 #include <stdlib.h>
-#include <string.h>
+#include <strings.h>
 
 #include "ntreg.h"
 #include "sam.h"
