diff -u -r ../deborphan/configure ./configure
--- ../deborphan/configure	2016-02-21 09:05:27.000000000 -0500
+++ ./configure	2016-08-11 05:52:10.144221407 -0400
@@ -2957,7 +2957,7 @@
 
 
 
-$as_echo "#define STATUS_FILE \"/var/lib/dpkg/status\"" >>confdefs.h
+$as_echo "#define STATUS_FILE \"@TERMUX_PREFIX@/var/lib/dpkg/status\"" >>confdefs.h
 
 
 $as_echo "#define REG_FLAGS REG_ICASE" >>confdefs.h
