diff --git a/src/exit.c b/src/exit.c
index e76a75e..99cfbdd 100644
--- a/src/exit.c
+++ b/src/exit.c
@@ -21,7 +21,6 @@

 #ifdef HAVE_ERRNO_H
 #  include <errno.h>
-extern int errno;
 #endif

 #ifdef ENABLE_NLS
diff --git a/src/xalloc.c b/src/xalloc.c
index 8c830e7..43cfb64 100644
--- a/src/xalloc.c
+++ b/src/xalloc.c
@@ -22,7 +22,6 @@

 #ifdef HAVE_ERRNO_H
 #  include <errno.h>
-extern int errno;
 #endif

 #ifdef STDC_HEADERS

