diff -u -r ../deborphan/src/Makefile.in ./src/Makefile.in
--- ../deborphan/src/Makefile.in	2016-02-21 09:05:27.000000000 -0500
+++ ./src/Makefile.in	2016-08-11 06:13:53.452865872 -0400
@@ -213,7 +213,7 @@
 target_alias = @target_alias@
 deborphan_SOURCES = deborphan.c exit.c libdeps.c pkginfo.c xalloc.c string.c keep.c file.c set.c
 KEEPNAME = keep
-STATEDIR = /var/lib/deborphan
+STATEDIR = @TERMUX_PREFIX@/var/lib/deborphan
 KEEPFILE = $(STATEDIR)/$(KEEPNAME)
 INCLUDES = -I../include
 all: all-am
