--- a/tools/ck-collect-session-info.c
+++ b/tools/ck-collect-session-info.c
@@ -82,7 +82,15 @@
 
         /* set the group */
         errno = 0;
+#ifdef __TERMUX__
+        if (getuid() == 0) {
+                res = setgid (pwent->pw_gid);
+        } else {
+                res = (getgid() == pwent->pw_gid) ? 0 : -1;
+        }
+#else
         res = setgid (pwent->pw_gid);
+#endif
         if (res == -1) {
                 g_warning ("Error performing setgid: %s", g_strerror (errno));
                 exit (1);
@@ -90,7 +98,15 @@
 
         /* become the user */
         errno = 0;
+#ifdef __TERMUX__
+        if (getuid() == 0) {
+                res = setuid (si->uid);
+        } else {
+                res = (getuid() == si->uid) ? 0 : -1;
+        }
+#else
         res = setuid (si->uid);
+#endif
         if (res == -1) {
                 g_warning ("Error performing setuid: %s", g_strerror (errno));
                 exit (1);
