--- a/src/sched.rs
+++ b/src/sched.rs
@@ -36,7 +36,7 @@
 }
 
 // Support a maximum CPU set of 1024 nodes
-#[cfg(all(target_arch = "x86_64", target_os = "linux"))]
+#[cfg(all(target_arch = "x86_64", any(target_os = "linux", target_os = "android")))]
 mod cpuset_attribs {
     use super::CpuMask;
     pub const CPU_SETSIZE:           usize = 1024;
@@ -53,7 +53,7 @@
     }
 }
 
-#[cfg(all(target_arch = "x86", target_os = "linux"))]
+#[cfg(all(target_arch = "x86", any(target_os = "linux", target_os = "android")))]
 mod cpuset_attribs {
     use super::CpuMask;
     pub const CPU_SETSIZE:           usize = 1024;
