--- a/src/errno.rs
+++ b/src/errno.rs
@@ -29,13 +29,13 @@
     __dfly_error()
 }
 
-#[cfg(any(target_os = "openbsd", target_os = "netbsd"))]
+#[cfg(any(target_os = "openbsd", target_os = "netbsd", target_os = "android"))]
 unsafe fn errno_location() -> *mut c_int {
     extern { fn __errno() -> *mut c_int; }
     __errno()
 }
 
-#[cfg(any(target_os = "linux", target_os = "android"))]
+#[cfg(target_os = "linux")]
 unsafe fn errno_location() -> *mut c_int {
     extern { fn __errno_location() -> *mut c_int; }
     __errno_location()
