+++ ./hw/xwayland/xwayland-shm.c
@@ -129,7 +129,30 @@
     char *name;
     int fd;
     int ret;
+#ifdef ANDROID
+    long flags;
 
+    fd = open("/dev/ashmem", O_RDWR);
+    if (fd < 0)
+        return fd;
+
+    ret = ioctl(fd, /* ASHMEM_SET_SIZE */ _IOW(0x77, 3, size_t), size);
+    if (ret < 0)
+        goto err;
+
+    flags = fcntl(fd, F_GETFD);
+    if (flags == -1)
+        goto err;
+
+    if (fcntl(fd, F_SETFD, flags | FD_CLOEXEC) == -1)
+        goto err;
+
+    return fd;
+
+err:
+    close(fd);
+    return ret;
+#endif
 #ifdef HAVE_MEMFD_CREATE
     fd = memfd_create("xwayland-shared", MFD_CLOEXEC | MFD_ALLOW_SEALING);
     if (fd >= 0) {
 
