+++ ./hw/xfree86/Makefile.am	2018-01-28 10:03:56.740173910 +0200
@@ -36,8 +36,8 @@
 
 SUBDIRS = common ddc x86emu $(INT10_SUBDIR) os-support parser \
 	  ramdac $(VGAHW_SUBDIR) loader modes $(DRI_SUBDIR) \
-	  $(DRI2_SUBDIR) . $(VBE_SUBDIR) i2c dixmods xkb \
-	  fbdevhw shadowfb exa $(XF86UTILS_SUBDIR) doc man \
+	  $(DRI2_SUBDIR) fbdevhw shadowfb . $(VBE_SUBDIR) i2c \
+	  dixmods xkb exa $(XF86UTILS_SUBDIR) doc man \
 	  $(GLAMOR_EGL_SUBDIR) drivers
 
 DIST_SUBDIRS = common ddc i2c x86emu int10 fbdevhw os-support \
@@ -45,8 +45,8 @@
                loader dixmods xkb dri dri2 exa modes \
 	       utils doc man glamor_egl drivers
 
+lib_LTLIBRARIES = libXFree86.la
 bin_PROGRAMS = Xorg
-nodist_Xorg_SOURCES = sdksyms.c
 
 AM_CFLAGS = $(DIX_CFLAGS) @XORG_CFLAGS@
 AM_CPPFLAGS = $(XORG_INCS) -I$(srcdir)/parser \
@@ -75,11 +75,35 @@
             $(top_builddir)/os/libos.la \
 	    $(top_builddir)/Xext/libXvidmode.la
 
-Xorg_LDADD = \
-            $(LOCAL_LIBS) \
-            $(XORG_SYS_LIBS) \
-            $(XSERVER_SYS_LIBS)
-Xorg_DEPENDENCIES = $(LOCAL_LIBS)
+LIBXFREE86_LIBS = \
+            $(MAIN_LIB) \
+            $(XSERVER_LIBS) \
+            loader/libloader.la \
+            common/libcommon.la \
+            os-support/libxorgos.la \
+            parser/libxf86config.la \
+            dixmods/libdixmods.la \
+            modes/libxf86modes.la \
+            ramdac/libramdac.la \
+            ddc/libddc.la \
+            i2c/libi2c.la \
+            $(XORG_LIBS) \
+            xkb/libxorgxkb.la \
+            fbdevhw/libfbdevhw.la \
+			$(top_builddir)/Xext/libXvidmode.la \
+			$(top_builddir)/fb/libfb.la \
+			$(top_builddir)/miext/shadow/libshadow.la
+
+
+libXFree86_la_LDFLAGS = $(Xorg_LDFLAGS) -module -avoid-version -Wl,--unresolved-symbols=report-all -rdynamic
+nodist_libXFree86_la_SOURCES = sdksyms.c
+libXFree86_la_SOURCES = 
+libXFree86_la_LIBADD = $(LIBXFREE86_LIBS) $(XORG_SYS_LIBS) $(XSERVER_SYS_LIBS)
+libXFree86_la_CFLAGS = $(AM_CFLAGS)
+libXFree86_la_DEPENDENCIES = $(LOCAL_LIBS)
+
+Xorg_LDADD = libXFree86.la
+Xorg_SOURCES = $(top_srcdir)/dix/stubmain.c
 
 Xorg_LDFLAGS = $(LD_EXPORT_SYMBOLS_FLAG)
 
+++ ./hw/xfree86/shadowfb/Makefile.am	2018-01-28 15:18:06.191501835 +0200
@@ -1,5 +1,5 @@
 module_LTLIBRARIES = libshadowfb.la
-libshadowfb_la_LDFLAGS = -module -avoid-version $(LD_NO_UNDEFINED_FLAG)
+libshadowfb_la_LDFLAGS = -module -avoid-version -static $(LD_NO_UNDEFINED_FLAG)
 libshadowfb_la_SOURCES = sfbmodule.c shadowfb.c
 libshadowfb_la_LIBADD = $(PIXMAN_LIBS)


+++ ./miext/shadow/Makefile.am	2018-01-28 15:24:22.836598681 +0200
@@ -6,6 +6,7 @@
 sdk_HEADERS = shadow.h
 endif
 
+libshadow_la_LDFLAGS = -module -avoid-version -static
 libshadow_la_SOURCES =		\
 	c2p_core.h		\
 	shadow.c		\

+++ ./hw/xfree86/fbdevhw/Makefile.am	2018-01-28 16:04:05.242414784 +0200
@@ -1,17 +1,11 @@
 SUBDIRS = man
 
-module_LTLIBRARIES = libfbdevhw.la
+noinst_LTLIBRARIES = libfbdevhw.la
 
-libfbdevhw_la_LDFLAGS = -module -avoid-version $(LD_NO_UNDEFINED_FLAG)
+libfbdevhw_la_LDFLAGS = -module -avoid-version -static $(LD_NO_UNDEFINED_FLAG)
-
-if FBDEVHW
 libfbdevhw_la_SOURCES = fbdevhw.c
-else
-libfbdevhw_la_SOURCES = fbdevhwstub.c
-endif
 
 AM_CPPFLAGS = $(XORG_INCS) -I$(srcdir)/../i2c -I$(srcdir)/../modes -I$(srcdir)/../ddc -I$(srcdir)/../parser
-
 AM_CFLAGS = $(DIX_CFLAGS) $(XORG_CFLAGS)
 
 sdk_HEADERS = fbdevhw.h

+++ ./hw/xfree86/dixmods/Makefile.am	2018-01-30 00:09:07.250483502 +0200
@@ -17,17 +17,17 @@
            -I$(top_srcdir)/miext/shadow \
            -I$(top_srcdir)/glx
 
-libfb_la_LDFLAGS = -module -avoid-version $(LD_NO_UNDEFINED_FLAG)
+libfb_la_LDFLAGS = -module -avoid-version -static $(LD_NO_UNDEFINED_FLAG)
 libfb_la_LIBADD = $(top_builddir)/fb/libfb.la
 libfb_la_SOURCES = fbmodule.c
 libfb_la_CFLAGS = $(AM_CFLAGS)
 
-libwfb_la_LDFLAGS = -module -avoid-version $(LD_NO_UNDEFINED_FLAG)
+libwfb_la_LDFLAGS = -module -avoid-version -static $(LD_NO_UNDEFINED_FLAG)
 libwfb_la_LIBADD = $(top_builddir)/fb/libwfb.la
 libwfb_la_SOURCES = fbmodule.c
 libwfb_la_CFLAGS = $(AM_CFLAGS) -DFB_ACCESS_WRAPPER
 
-libglx_la_LDFLAGS = -module -avoid-version $(LD_NO_UNDEFINED_FLAG)
+libglx_la_LDFLAGS = -module -avoid-version -static $(LD_NO_UNDEFINED_FLAG)
 libglx_la_LIBADD = $(top_builddir)/glx/libglx.la $(GLX_SYS_LIBS)
 if DRI2
 libglx_la_LIBADD += $(top_builddir)/glx/libglxdri.la
@@ -37,7 +37,7 @@
 endif
 libglx_la_SOURCES = glxmodule.c
 
-libshadow_la_LDFLAGS = -module -avoid-version $(LD_NO_UNDEFINED_FLAG)
+libshadow_la_LDFLAGS = -module -avoid-version -static $(LD_NO_UNDEFINED_FLAG)
 libshadow_la_LIBADD = $(top_builddir)/miext/shadow/libshadow.la
 if NO_UNDEFINED
 libshadow_la_LIBADD += libfb.la
 
+++ ./include/dix.h	2018-01-25 00:43:05.367766278 +0200
@@ -307,7 +307,7 @@
 extern _X_EXPORT void
 SetVendorString(const char *string);
 
-int
+_X_EXPORT int
 dix_main(int argc, char *argv[], char *envp[]);
 
 /* events.c */

