--- a/os/osinit.c
+++ b/os/osinit.c
@@ -72,7 +72,7 @@ SOFTWARE.
 #endif
 
 #ifndef ADMPATH
-#define ADMPATH "/usr/adm/X%smsgs"
+#define ADMPATH "@TERMUX_PREFIX@/adm/X%smsgs"
 #endif
 
 #ifdef RLIMIT_DATA
@@ -264,11 +264,6 @@ OsInit(void)
         }
 #endif /* !XQUARTZ */
 
-#if !defined(WIN32) || defined(__CYGWIN__)
-        if (getpgrp() == 0)
-            setpgid(0, 0);
-#endif
-
 #ifdef RLIMIT_DATA
         if (limitDataSpace >= 0) {
             struct rlimit rlim;
