+++ ./hw/xwayland/Makefile.am	2019-06-26 02:44:24.968986945 +0300
@@ -24,7 +24,6 @@
 Xwayland_LDADD =				\
 	$(glamor_lib)				\
 	$(XWAYLAND_LIBS)			\
-	$(top_builddir)/glx/libglxvnd.la	\
 	$(XWAYLAND_SYS_LIBS)			\
 	$(top_builddir)/Xext/libXvidmode.la	\
 	$(XSERVER_SYS_LIBS)

+++ ./hw/xwayland/Makefile.in	2019-06-26 02:46:04.650920429 +0300
@@ -640,7 +640,7 @@
 	$(top_srcdir)/mi/miinitext.c $(am__append_1) $(am__append_2) \
 	$(am__append_3)
 Xwayland_LDADD = $(glamor_lib) $(XWAYLAND_LIBS) \
-	$(top_builddir)/glx/libglxvnd.la $(XWAYLAND_SYS_LIBS) \
+	$(XWAYLAND_SYS_LIBS) \
 	$(top_builddir)/Xext/libXvidmode.la $(XSERVER_SYS_LIBS) \
 	$(am__append_5)
 Xwayland_LDFLAGS = $(LD_EXPORT_SYMBOLS_FLAG)
