--- a/configure.ac
+++ b/configure.ac
@@ -114,7 +114,7 @@
 
 
 dnl Default to -Werror, but add an easy flag to disable it.
-WERROR="-Werror -Wno-terminate"
+WERROR="-Werror -Wno-exceptions"
 
 AC_ARG_ENABLE(dynamic-backtrace,
 	AS_HELP_STRING([--enable-dynamic-backtrace], [Modify the executable so that it can generate a backtrace for uncaught exceptions.  Will double the size of the stripped binary.]),
--- a/configure
+++ b/configure
@@ -19307,7 +19307,7 @@
 
 
 
-WERROR="-Werror -Wno-terminate"
+WERROR="-Werror -Wno-exceptions"
 
 # Check whether --enable-dynamic-backtrace was given.
 if test "${enable_dynamic_backtrace+set}" = set; then :
