--- a/libwget/io.c
+++ b/libwget/io.c
@@ -413,7 +413,7 @@
 	// find out system temp directory
 	if (!(tmpdir = getenv("TMPDIR")) && !(tmpdir = getenv("TMP"))
 		&& !(tmpdir = getenv("TEMP")) && !(tmpdir = getenv("TEMPDIR")))
-		tmpdir = "/tmp";
+		tmpdir = "@TERMUX_PREFIX@/tmp";
 
 	// create a per-usr tmp file name
 #ifdef HAVE_GETUID
--- a/libwget/ssl.c
+++ b/libwget/ssl.c
@@ -29,7 +29,7 @@
 const char *wget_ssl_default_cert_dir(void)
 {
 #ifndef __OS2__
-	return "/etc/ssl/certs";
+	return "@TERMUX_PREFIX@/etc/ssl/certs";
 #else
 	return "/@unixroot/etc/ssl/certs";
 #endif
@@ -57,8 +57,8 @@
 // ssl_default_path() is only called once in tls_init().
 static const char *ssl_default_path(const char *base)
 {
-	if (access("/etc/ssl/certs", F_OK) == 0) {
-		return wget_strdup("/etc/ssl/certs");
+	if (access("@TERMUX_PREFIX@/etc/ssl/certs", F_OK) == 0) {
+		return wget_strdup("@TERMUX_PREFIX@/etc/ssl/certs");
 	}
 
 	const char *progData = getenv("ProgramData");
