diff -uNr tidy-html5/include/tidyplatform.h tidy-html5.mod/include/tidyplatform.h
--- tidy-html5/include/tidyplatform.h	2018-04-27 16:33:51.000000000 +0300
+++ tidy-html5.mod/include/tidyplatform.h	2019-01-20 18:47:23.209564431 +0200
@@ -569,7 +569,7 @@
    typedef unsigned int uint;
 #endif
 
-#if defined(HPUX_OS) || defined(CYGWIN_OS) || defined(MAC_OS) || defined(BSD_BASED_OS) || defined(_WIN32)
+#if defined(HPUX_OS) || defined(CYGWIN_OS) || defined(MAC_OS) || defined(BSD_BASED_OS) || defined(_WIN32) || defined(__ANDROID__)
 #  undef ulong
    typedef unsigned long ulong;
 #endif
