--- ./configure	2021-05-18 02:51:48.000000000 +0530
+++ ./configure.mod	2021-06-22 08:32:21.956267052 +0530
@@ -6412,8 +6412,8 @@
           LTFLAGS='--silent'
       fi
       my_libtool=`$apr_config --apr-libtool`
-      LIBTOOL="$my_libtool \$(LTFLAGS)"
-      libtoolversion=`$my_libtool --version`
+      LIBTOOL="bash $my_libtool \$(LTFLAGS)"
+      libtoolversion=`bash $my_libtool --version`
       case $libtoolversion in
           *1.[45]* | *[2-9].[0-9]*)
               SH_LIBTOOL='$(LIBTOOL)'
