--- a/src/os/unix/dir.cpp
+++ b/src/os/unix/dir.cpp
@@ -210,7 +210,7 @@ stringT pws_os::getxmldir(void)
 #ifdef __FreeBSD__
   xmldir = _T("/usr/local/share/pwsafe/xml/");
 #else
-  xmldir = _T("/usr/share/passwordsafe/xml/");
+  xmldir = _T("@TERMUX_PREFIX@/share/passwordsafe/xml/");
 #endif
   }
   return xmldir;
@@ -223,7 +223,7 @@ stringT pws_os::gethelpdir(void)
 #if defined( __FreeBSD__) || defined(__OpenBSD)
     helpdir = _T("/usr/local/share/doc/passwordsafe/help/");
 #else
-    helpdir = _T("/usr/share/passwordsafe/help/");
+    helpdir = _T("@TERMUX_PREFIX@/share/passwordsafe/help/");
 #endif
   }
   return helpdir;
--- a/src/ui/wxWidgets/PWSafeApp.cpp
+++ b/src/ui/wxWidgets/PWSafeApp.cpp
@@ -219,8 +219,8 @@ void PWSafeApp::Init()
   m_locale = new PWSLocale;
 #endif
 #if defined(__UNIX__) && !defined(__WXMAC__)
-  wxFileTranslationsLoader::AddCatalogLookupPathPrefix(L"/usr/share/locale");
-  wxFileTranslationsLoader::AddCatalogLookupPathPrefix(L"/usr/local/share/locale");
+  wxFileTranslationsLoader::AddCatalogLookupPathPrefix(L"@TERMUX_PREFIX@/share/locale");
+  wxFileTranslationsLoader::AddCatalogLookupPathPrefix(L"@TERMUX_PREFIX@/local/share/locale");
 #endif
 #if defined(_DEBUG) || defined(DEBUG)
   wxFileTranslationsLoader::AddCatalogLookupPathPrefix(L"../I18N/mos");
