--- a/csh.h
+++ b/csh.h
@@ -89,12 +89,14 @@ typedef void *ptr_t;
 #define xcalloc(n, s)	Calloc(n, s)
 #define xfree(p)	Free(p)
 
-#include <stdio.h>
-FILE *cshin, *cshout, *csherr;
+#include <bsd/stdio.h>
+extern FILE *cshin, *cshout, *csherr;
 
 #define	isdir(d)	(S_ISDIR(d.st_mode))
 
+#ifndef __bool_true_false_are_defined
 typedef int bool;
+#endif
 
 #define	eq(a, b)	(Strcmp(a, b) == 0)
 
@@ -106,56 +108,56 @@ typedef int bool;
 /*
  * Global flags
  */
-bool    chkstop;		/* Warned of stopped jobs... allow exit */
-bool    didfds;			/* Have setup i/o fd's for child */
-bool    doneinp;		/* EOF indicator after reset from readc */
-bool    exiterr;		/* Exit if error or non-zero exit status */
-bool    child;			/* Child shell ... errors cause exit */
-bool    haderr;			/* Reset was because of an error */
-bool    intty;			/* Input is a tty */
-bool    intact;			/* We are interactive... therefore prompt */
-bool    justpr;			/* Just print because of :p hist mod */
-bool    loginsh;		/* We are a loginsh -> .login/.logout */
-bool    neednote;		/* Need to pnotify() */
-bool    noexec;			/* Don't execute, just syntax check */
-bool    pjobs;			/* want to print jobs if interrupted */
-bool    setintr;		/* Set interrupts on/off -> Wait intr... */
-bool    timflg;			/* Time the next waited for command */
-bool    havhash;		/* path hashing is available */
+extern bool    chkstop;		/* Warned of stopped jobs... allow exit */
+extern bool    didfds;			/* Have setup i/o fd's for child */
+extern bool    doneinp;		/* EOF indicator after reset from readc */
+extern bool    exiterr;		/* Exit if error or non-zero exit status */
+extern bool    child;			/* Child shell ... errors cause exit */
+extern bool    haderr;			/* Reset was because of an error */
+extern bool    intty;			/* Input is a tty */
+extern bool    intact;			/* We are interactive... therefore prompt */
+extern bool    justpr;			/* Just print because of :p hist mod */
+extern bool    loginsh;		/* We are a loginsh -> .login/.logout */
+extern bool    neednote;		/* Need to pnotify() */
+extern bool    noexec;			/* Don't execute, just syntax check */
+extern bool    pjobs;			/* want to print jobs if interrupted */
+extern bool    setintr;		/* Set interrupts on/off -> Wait intr... */
+extern bool    timflg;			/* Time the next waited for command */
+extern bool    havhash;		/* path hashing is available */
 
 #ifdef FILEC
-bool    filec;			/* doing filename expansion */
+extern bool    filec;			/* doing filename expansion */
 #endif
 
 /*
  * Global i/o info
  */
-Char   *arginp;			/* Argument input for sh -c and internal `xx` */
-int     onelflg;		/* 2 -> need line for -t, 1 -> exit on read */
-Char   *ffile;			/* Name of shell file for $0 */
+extern Char   *arginp;			/* Argument input for sh -c and internal `xx` */
+extern int     onelflg;		/* 2 -> need line for -t, 1 -> exit on read */
+extern Char   *ffile;			/* Name of shell file for $0 */
 
-char   *seterr;			/* Error message from scanner/parser */
+extern char   *seterr;			/* Error message from scanner/parser */
 
 #include <sys/types.h>
 #include <sys/time.h>
 #include <sys/resource.h>
 
-struct timeval time0;		/* Time at which the shell started */
-struct rusage ru0;
+extern struct timeval time0;		/* Time at which the shell started */
+extern struct rusage ru0;
 
 /*
  * Miscellany
  */
-Char   *doldol;			/* Character pid for $$ */
-int	backpid;		/* Pid of the last background process */
-uid_t	uid, euid;		/* Invokers uid */
-gid_t	gid, egid;		/* Invokers gid */
-time_t  chktim;			/* Time mail last checked */
-pid_t	shpgrp;			/* Pgrp of shell */
-pid_t	tpgrp;			/* Terminal process group */
+extern Char   *doldol;			/* Character pid for $$ */
+extern int	backpid;		/* Pid of the last background process */
+extern uid_t	uid, euid;		/* Invokers uid */
+extern gid_t	gid, egid;		/* Invokers gid */
+extern time_t  chktim;			/* Time mail last checked */
+extern pid_t	shpgrp;			/* Pgrp of shell */
+extern pid_t	tpgrp;			/* Terminal process group */
 
 /* If tpgrp is -1, leave tty alone! */
-pid_t	opgrp;			/* Initial pgrp and tty pgrp */
+extern pid_t	opgrp;			/* Initial pgrp and tty pgrp */
 
 
 /*
@@ -165,10 +167,10 @@ pid_t	opgrp;			/* Initial pgrp and tty pgrp */
  * (this saves work) but for version 6, this is not usually possible.
  * The desired initial values for these descriptors are F{SHIN,...}.
  */
-int   SHIN;			/* Current shell input (script) */
-int   SHOUT;			/* Shell output */
-int   SHERR;			/* Diagnostic output... shell errs go here */
-int   OLDSTD;			/* Old standard input (def for cmds) */
+extern int   SHIN;			/* Current shell input (script) */
+extern int   SHOUT;			/* Shell output */
+extern int   SHERR;			/* Diagnostic output... shell errs go here */
+extern int   OLDSTD;			/* Old standard input (def for cmds) */
 
 /*
  * Error control
@@ -179,8 +181,8 @@ int   OLDSTD;			/* Old standard input (def for cmds) */
  */
 
 #include <setjmp.h>
-jmp_buf reslab;
-int exitset;
+extern jmp_buf reslab;
+extern int exitset;
 
 #define	setexit()	(setjmp(reslab))
 #define	reset()		longjmp(reslab, 1)
@@ -188,11 +190,11 @@ int exitset;
 #define	getexit(a)	memcpy((a), reslab, sizeof reslab)
 #define	resexit(a)	memcpy(reslab, (a), sizeof reslab)
 
-Char   *gointr;			/* Label for an onintr transfer */
+extern Char   *gointr;			/* Label for an onintr transfer */
 
 #include <signal.h>
-sig_t parintr;			/* Parents interrupt catch */
-sig_t parterm;			/* Parents terminate catch */
+extern sig_t parintr;			/* Parents interrupt catch */
+extern sig_t parterm;			/* Parents terminate catch */
 
 /*
  * Lexical definitions.
@@ -213,7 +215,7 @@ sig_t parterm;			/* Parents terminate catch */
 #define	TRIM		0177	/* Mask to strip quote bit */
 #endif
 
-int     AsciiOnly;		/* If set only 7 bits is expected in characters */
+extern int     AsciiOnly;		/* If set only 7 bits is expected in characters */
 
 /*
  * Each level of input has a buffered input structure.
@@ -222,7 +224,7 @@ int     AsciiOnly;		/* If set only 7 bits is expected in characters */
  * In other cases, the shell buffers enough blocks to keep all loops
  * in the buffer.
  */
-struct Bin {
+extern struct Bin {
     off_t   Bfseekp;		/* Seek pointer */
     off_t   Bfbobp;		/* Seekp of beginning of buffers */
     off_t   Bfeobp;		/* Seekp of end of buffers */
@@ -263,9 +265,9 @@ extern int aret;		/* What was the last character returned */
  * For whiles, in particular, it re-seeks to the beginning of the
  * line the while was on; hence the while placement restrictions.
  */
-struct Ain lineloc;
+extern struct Ain lineloc;
 
-bool    cantell;		/* Is current source tellable ? */
+extern bool    cantell;		/* Is current source tellable ? */
 
 /*
  * Input lines are parsed into doubly linked circular
@@ -297,7 +299,7 @@ struct wordent {
  * process id's from `$$', and modified variable values (from qualifiers
  * during expansion in sh.dol.c) here.
  */
-Char   *lap;
+extern Char   *lap;
 
 /*
  * Parser structure
@@ -395,7 +397,7 @@ extern int nsrchn;
  * source level.  Loops are implemented by seeking back in the
  * input.  For foreach (fe), the word list is attached here.
  */
-struct whyle {
+extern struct whyle {
     struct Ain   w_start;	/* Point to restart loop */
     struct Ain   w_end;		/* End of loop (0 if unknown) */
     Char  **w_fe, **w_fe0;	/* Current/initial wordlist for fe */
@@ -408,7 +410,7 @@ struct whyle {
  *
  * Aliases and variables are stored in AVL balanced binary trees.
  */
-struct varent {
+extern struct varent {
     Char  **vec;		/* Array of words which is the value */
     Char   *v_name;		/* Name of variable/alias */
     struct varent *v_link[3];	/* The links, see below */
@@ -419,7 +421,7 @@ struct varent {
 #define v_right		v_link[1]
 #define v_parent	v_link[2]
 
-struct varent *adrof1(Char *, struct varent *);
+extern struct varent *adrof1(Char *, struct varent *);
 
 #define adrof(v)	adrof1(v, &shvhed)
 #define value(v)	value1(v, &shvhed)
@@ -428,14 +430,14 @@ struct varent *adrof1(Char *, struct varent *);
  * The following are for interfacing redo substitution in
  * aliases to the lexical routines.
  */
-struct wordent *alhistp;	/* Argument list (first) */
-struct wordent *alhistt;	/* Node after last in arg list */
-Char  **alvec, *alvecp;		/* The (remnants of) alias vector */
+extern struct wordent *alhistp;	/* Argument list (first) */
+extern struct wordent *alhistt;	/* Node after last in arg list */
+extern Char  **alvec, *alvecp;		/* The (remnants of) alias vector */
 
 /*
  * Filename/command name expansion variables
  */
-int   gflag;			/* After tglob -> is globbing needed? */
+extern int   gflag;			/* After tglob -> is globbing needed? */
 
 #define MAXVARLEN 30		/* Maximum number of char in a variable name */
 
@@ -450,9 +452,9 @@ extern long gargc;		/* Number args in gargv */
  */
 extern Char **pargv;		/* Pointer to the argv list space */
 extern long pargc;		/* Count of arguments in pargv */
-Char   *pargs;			/* Pointer to start current word */
-long    pnleft;			/* Number of chars left in pargs */
-Char   *pargcp;			/* Current index into pargs */
+extern Char   *pargs;			/* Pointer to start current word */
+extern long    pnleft;			/* Number of chars left in pargs */
+extern Char   *pargcp;			/* Current index into pargs */
 
 /*
  * History list
@@ -465,19 +467,19 @@ Char   *pargcp;			/* Current index into pargs */
  * when history substitution includes modifiers, and thrown away
  * at the next discarding since their event numbers are very negative.
  */
-struct Hist {
+extern struct Hist {
     struct wordent Hlex;
     int     Hnum;
     int     Href;
     struct Hist *Hnext;
 }       Histlist;
 
-struct wordent paraml;		/* Current lexical word list */
-int     eventno;		/* Next events number */
-int     lastev;			/* Last event reference (default) */
+extern struct wordent paraml;		/* Current lexical word list */
+extern int     eventno;		/* Next events number */
+extern int     lastev;			/* Last event reference (default) */
 
-Char    HIST;			/* history invocation character */
-Char    HISTSUB;		/* auto-substitute character */
+extern Char    HIST;			/* history invocation character */
+extern Char    HISTSUB;		/* auto-substitute character */
 
 /*
  * strings.h:
@@ -518,30 +520,30 @@ Char    HISTSUB;		/* auto-substitute character */
 /*
  * setname is a macro to save space (see sh.err.c)
  */
-char   *bname;
+extern char   *bname;
 
 #define	setname(a)	(bname = (a))
 
-Char   *Vsav;
-Char   *Vdp;
-Char   *Vexpath;
-char  **Vt;
+extern Char   *Vsav;
+extern Char   *Vdp;
+extern Char   *Vexpath;
+extern char  **Vt;
 
-Char  **evalvec;
-Char   *evalp;
+extern Char  **evalvec;
+extern Char   *evalp;
 
 /* word_chars is set by default to WORD_CHARS but can be overridden by
    the worchars variable--if unset, reverts to WORD_CHARS */
 
-Char   *word_chars;
+extern Char   *word_chars;
 
 #define WORD_CHARS "*?_-.[]~="	/* default chars besides alnums in words */
 
-Char   *STR_SHELLPATH;
+extern Char   *STR_SHELLPATH;
 
 #include <paths.h>
 #ifdef _PATH_BSHELL
-Char   *STR_BSHELL;
+extern Char   *STR_BSHELL;
 #endif
-Char   *STR_WORD_CHARS;
-Char  **STR_environ;
+extern Char   *STR_WORD_CHARS;
+extern Char  **STR_environ;
