--- a/csh.c
+++ b/csh.c
@@ -37,12 +37,13 @@
 #include <fcntl.h>
 #include <errno.h>
 #include <pwd.h>
-#include <stdlib.h>
-#include <string.h>
+#include <bsd/stdlib.h>
+#include <bsd/string.h>
 #include <locale.h>
-#include <unistd.h>
-#include <vis.h>
+#include <bsd/unistd.h>
+#include <bsd/vis.h>
 #include <stdarg.h>
+#include <time.h>
 
 #include "csh.h"
 #include "proc.h"
@@ -62,20 +63,96 @@
  * June, 1991
  */
 
+struct Bin      B;
+struct whyle      *whyles;
+struct varent       shvhed, aliases;
+struct Hist       Histlist;
+bool    chkstop;		/* Warned of stopped jobs... allow exit */
+bool    didfds;			/* Have setup i/o fd's for child */
+bool    doneinp;		/* EOF indicator after reset from readc */
+bool    exiterr;		/* Exit if error or non-zero exit status */
+bool    child;			/* Child shell ... errors cause exit */
+bool    haderr;			/* Reset was because of an error */
+bool    intty;			/* Input is a tty */
+bool    intact;			/* We are interactive... therefore prompt */
+bool    justpr;			/* Just print because of :p hist mod */
+bool    loginsh;		/* We are a loginsh -> .login/.logout */
+bool    neednote;		/* Need to pnotify() */
+bool    noexec;			/* Don't execute, just syntax check */
+bool    pjobs;			/* want to print jobs if interrupted */
+bool    setintr;		/* Set interrupts on/off -> Wait intr... */
+bool    timflg;			/* Time the next waited for command */
+bool    havhash;		/* path hashing is available */
+#ifdef FILEC
+bool    filec;			/* doing filename expansion */
+#endif
+Char   *arginp;			/* Argument input for sh -c and internal `xx` */
+int     onelflg;		/* 2 -> need line for -t, 1 -> exit on read */
+Char   *ffile;			/* Name of shell file for $0 */
+struct timeval time0;		/* Time at which the shell started */
+struct rusage ru0;
+Char   *doldol;			/* Character pid for $$ */
+int	backpid;		/* Pid of the last background process */
+uid_t	uid, euid;		/* Invokers uid */
+gid_t	gid, egid;		/* Invokers gid */
+time_t  chktim;			/* Time mail last checked */
+pid_t	shpgrp;			/* Pgrp of shell */
+pid_t	tpgrp;			/* Terminal process group */
+pid_t	opgrp;			/* Initial pgrp and tty pgrp */
+int   SHIN;			/* Current shell input (script) */
+int   SHOUT;			/* Shell output */
+int   SHERR;			/* Diagnostic output... shell errs go here */
+int   OLDSTD;			/* Old standard input (def for cmds) */
+jmp_buf reslab;
+int exitset;
+Char   *gointr;			/* Label for an onintr transfer */
+sig_t parintr;			/* Parents interrupt catch */
+sig_t parterm;			/* Parents terminate catch */
+int     AsciiOnly;		/* If set only 7 bits is expected in characters */
+struct Ain lineloc;
+int    cantell;		/* Is current source tellable ? */
+Char   *lap;
+struct wordent *alhistp;	/* Argument list (first) */
+struct wordent *alhistt;	/* Node after last in arg list */
+Char  **alvec;		/* The (remnants of) alias vector */
+int   gflag;			/* After tglob -> is globbing needed? */
+Char   *pargs;			/* Pointer to start current word */
+long    pnleft;			/* Number of chars left in pargs */
+Char   *pargcp;			/* Current index into pargs */
+struct wordent paraml;		/* Current lexical word list */
+int     eventno;		/* Next events number */
+int     lastev;			/* Last event reference (default) */
+Char    HIST;			/* history invocation character */
+Char    HISTSUB;		/* auto-substitute character */
+char   *bname;
+Char   *Vsav;
+Char   *Vdp;
+Char   *Vexpath;
+char  **Vt;
+Char  **evalvec;
+Char   *evalp;
+Char   *word_chars;
+Char   *STR_SHELLPATH;
+Char   *STR_BSHELL;
+Char   *STR_WORD_CHARS;
+Char  **STR_environ;
+
 Char   *dumphist[] = {STRhistory, STRmh, 0, 0};
 Char   *loadhist[] = {STRsource, STRmh, STRtildothist, 0};
 
+FILE *cshin, *cshout, *csherr;
+
 int     nofile = 0;
-bool    reenter = 0;
-bool    nverbose = 0;
-bool    nexececho = 0;
-bool    quitit = 0;
-bool    fast = 0;
-bool    batch = 0;
-bool    mflag = 0;
-bool    prompt = 1;
-bool    enterhist = 0;
-bool    tellwhat = 0;
+int    reenter = 0;
+int    nverbose = 0;
+int    nexececho = 0;
+int    quitit = 0;
+int    fast = 0;
+int    batch = 0;
+int    mflag = 0;
+int    prompt = 1;
+int    enterhist = 0;
+int    tellwhat = 0;
 
 extern char **environ;
 
