--- a/exp.c
+++ b/exp.c
@@ -32,10 +32,10 @@
 
 #include <sys/types.h>
 #include <sys/stat.h>
-#include <stdlib.h>
-#include <unistd.h>
+#include <bsd/stdlib.h>
+#include <bsd/unistd.h>
 #ifndef SHORT_STRINGS
-#include <string.h>
+#include <bsd/string.h>
 #endif /* SHORT_STRINGS */
 #include <stdarg.h>
 
@@ -59,16 +59,16 @@
 #define EQMATCH 7
 #define NOTEQMATCH 8
 
-static int	exp1(Char ***, bool);
-static int	exp2_(Char ***, bool);
-static int	exp2a(Char ***, bool);
-static int	exp2b(Char ***, bool);
-static int	exp2c(Char ***, bool);
-static Char *	exp3(Char ***, bool);
-static Char *	exp3a(Char ***, bool);
-static Char *	exp4(Char ***, bool);
-static Char *	exp5(Char ***, bool);
-static Char *	exp6(Char ***, bool);
+static int	exp1(Char ***, int);
+static int	exp2_(Char ***, int);
+static int	exp2a(Char ***, int);
+static int	exp2b(Char ***, int);
+static int	exp2c(Char ***, int);
+static Char *	exp3(Char ***, int);
+static Char *	exp3a(Char ***, int);
+static Char *	exp4(Char ***, int);
+static Char *	exp5(Char ***, int);
+static Char *	exp6(Char ***, int);
 static void	evalav(Char **);
 static int	isa(Char *, int);
 static int	egetn(Char *);
@@ -85,7 +85,7 @@ expr(Char ***vp)
 }
 
 int
-exp0(Char ***vp, bool ignore)
+exp0(Char ***vp, int ignore)
 {
     int p1 = exp1(vp, ignore);
 
@@ -106,7 +106,7 @@ exp0(Char ***vp, bool ignore)
 }
 
 static int
-exp1(Char ***vp, bool ignore)
+exp1(Char ***vp, int ignore)
 {
     int p1 = exp2_(vp, ignore);
 
@@ -127,7 +127,7 @@ exp1(Char ***vp, bool ignore)
 }
 
 static int
-exp2_(Char ***vp, bool ignore)
+exp2_(Char ***vp, int ignore)
 {
     int p1 = exp2a(vp, ignore);
 
@@ -148,7 +148,7 @@ exp2_(Char ***vp, bool ignore)
 }
 
 static int
-exp2a(Char ***vp, bool ignore)
+exp2a(Char ***vp, int ignore)
 {
     int p1 = exp2b(vp, ignore);
 
@@ -169,7 +169,7 @@ exp2a(Char ***vp, bool ignore)
 }
 
 static int
-exp2b(Char ***vp, bool ignore)
+exp2b(Char ***vp, int ignore)
 {
     int p1 = exp2c(vp, ignore);
 
@@ -190,7 +190,7 @@ exp2b(Char ***vp, bool ignore)
 }
 
 static int
-exp2c(Char ***vp, bool ignore)
+exp2c(Char ***vp, int ignore)
 {
     Char *p1 = exp3(vp, ignore);
     Char *p2;
@@ -236,7 +236,7 @@ exp2c(Char ***vp, bool ignore)
 }
 
 static Char *
-exp3(Char ***vp, bool ignore)
+exp3(Char ***vp, int ignore)
 {
     Char *p1, *p2;
     int i;
@@ -280,7 +280,7 @@ exp3(Char ***vp, bool ignore)
 }
 
 static Char *
-exp3a(Char ***vp, bool ignore)
+exp3a(Char ***vp, int ignore)
 {
     Char *p1, *p2, *op;
     int i;
@@ -308,7 +308,7 @@ exp3a(Char ***vp, bool ignore)
 }
 
 static Char *
-exp4(Char ***vp, bool ignore)
+exp4(Char ***vp, int ignore)
 {
     Char *p1, *p2;
     int i = 0;
@@ -343,7 +343,7 @@ exp4(Char ***vp, bool ignore)
 }
 
 static Char *
-exp5(Char ***vp, bool ignore)
+exp5(Char ***vp, int ignore)
 {
     Char *p1, *p2;
     int i = 0;
@@ -388,7 +388,7 @@ exp5(Char ***vp, bool ignore)
 }
 
 static Char *
-exp6(Char ***vp, bool ignore)
+exp6(Char ***vp, int ignore)
 {
     int     ccode, i = 0;
     Char *cp, *dp, *ep;
