--- a/file.c
+++ b/file.c
@@ -38,10 +38,10 @@
 #include <termios.h>
 #include <dirent.h>
 #include <pwd.h>
-#include <stdlib.h>
-#include <unistd.h>
+#include <bsd/stdlib.h>
+#include <bsd/unistd.h>
 #ifndef SHORT_STRINGS
-#include <string.h>
+#include <bsd/string.h>
 #endif /* SHORT_STRINGS */
 #include <stdarg.h>
 
@@ -95,7 +95,7 @@ static int	 ignored(Char *);
  * completion by default.  Filec controls completion, nobeep controls
  * ringing the terminal bell on incomplete expansions.
  */
-bool    filec = 0;
+int    filec = 0;
 
 static void
 setup_tty(int on)
