--- a/func.c
+++ b/func.c
@@ -34,9 +34,9 @@
 #include <sys/stat.h>
 #include <signal.h>
 #include <locale.h>
-#include <stdlib.h>
-#include <string.h>
-#include <unistd.h>
+#include <bsd/stdlib.h>
+#include <bsd/string.h>
+#include <bsd/unistd.h>
 #include <stdarg.h>
 
 #include "csh.h"
@@ -1063,7 +1063,7 @@ doumask(Char **v, struct command *t)
     (void) umask(i);
 }
 
-typedef quad_t RLIM_TYPE;
+typedef int64_t RLIM_TYPE;
 
 static struct limits {
     int     limconst;
