--- a/lex.c
+++ b/lex.c
@@ -34,9 +34,9 @@
 #include <sys/ioctl.h>
 #include <termios.h>
 #include <errno.h>
-#include <stdlib.h>
-#include <string.h>
-#include <unistd.h>
+#include <bsd/stdlib.h>
+#include <bsd/string.h>
+#include <bsd/unistd.h>
 #include <stdarg.h>
 
 #include "csh.h"
@@ -1394,10 +1394,10 @@ reread:
 static int
 bgetc(void)
 {
-    int buf, off, c;
+    int buf, off, c, roomleft;
 
 #ifdef FILEC
-    int numleft = 0, roomleft;
+    int numleft = 0;
     Char    ttyline[BUFSIZ];
 #endif
     char    tbuf[BUFSIZ + 1];
@@ -1474,7 +1474,6 @@ again:
 		}
 #ifdef FILEC
 	    }
-#endif
 	    if (c >= 0)
 		break;
 	    if (errno == EWOULDBLOCK) {
@@ -1485,6 +1484,7 @@ again:
 	    else if (errno != EINTR)
 		break;
 	}
+#endif
 	if (c <= 0)
 	    return (-1);
 	feobp += c;
