--- a/parse.c
+++ b/parse.c
@@ -31,8 +31,8 @@
  */
 
 #include <sys/types.h>
-#include <stdlib.h>
-#include <string.h>
+#include <bsd/stdlib.h>
+#include <bsd/string.h>
 #include <stdarg.h>
 
 #include "csh.h"
@@ -140,7 +140,7 @@ asyn3(struct wordent *p1, struct wordent *p2)
 {
     struct varent *ap;
     struct wordent alout;
-    bool redid;
+    int redid;
 
     if (p1 == p2)
 	return;
@@ -466,7 +466,7 @@ syn3(struct wordent *p1, struct wordent *p2, int flags)
     int l;
     Char  **av;
     int     n, c;
-    bool    specp = 0;
+    int    specp = 0;
 
     if (p1 != p2) {
 	p = p1;
