--- a/pathnames.h
+++ b/pathnames.h
@@ -32,9 +32,9 @@
  *	@(#)pathnames.h	8.1 (Berkeley) 5/31/93
  */
 
-#define	_PATH_BIN		"/bin"
-#define	_PATH_DOTCSHRC		"/etc/csh.cshrc"
-#define	_PATH_DOTLOGIN		"/etc/csh.login"
-#define	_PATH_DOTLOGOUT		"/etc/csh.logout"
-#define	_PATH_LOGIN		"/usr/bin/login"
-#define	_PATH_USRBIN		"/usr/bin"
+#define	_PATH_BIN		"@TERMUX_PREFIX@/bin"
+#define	_PATH_DOTCSHRC		"@TERMUX_PREFIX@/etc/csh.cshrc"
+#define	_PATH_DOTLOGIN		"@TERMUX_PREFIX@/etc/csh.login"
+#define	_PATH_DOTLOGOUT		"@TERMUX_PREFIX@/etc/csh.logout"
+#define	_PATH_LOGIN		"@TERMUX_PREFIX@/bin/login"
+#define	_PATH_USRBIN		"@TERMUX_PREFIX@/bin"
