--- a/proc.c
+++ b/proc.c
@@ -33,9 +33,9 @@
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <errno.h>
-#include <unistd.h>
-#include <stdlib.h>
-#include <string.h>
+#include <bsd/unistd.h>
+#include <bsd/stdlib.h>
+#include <bsd/string.h>
 #include <stdarg.h>
 
 #include "csh.h"
@@ -45,6 +45,13 @@
 
 #define BIGINDEX	9	/* largest desirable job index */
 
+struct process proclist;	/* list head of all processes */
+bool    pnoprocesses;		/* pchild found nothing to wait for */
+struct process *pholdjob;	/* one level stack of current jobs */
+struct process *pcurrjob;	/* current job */
+struct process *pcurrent;	/* current job in table */
+struct process *pprevious;	/* previous job in table */
+int    pmaxindex;		/* current maximum job index */
 static struct rusage zru;
 
 static void	 pflushall(void);
@@ -75,13 +82,13 @@ pchild(int notused)
     int pid;
     extern int insource;
     int save_errno = errno;
-    union wait w;
+    int w;
     int     jobflags;
     struct rusage ru;
 
 loop:
     errno = 0;			/* reset, just in case */
-    pid = wait3(&w.w_status,
+    pid = wait4((pid_t) -1, &w,
        (setintr && (intty || insource) ? WNOHANG | WUNTRACED : WNOHANG), &ru);
 
     if (pid <= 0) {
@@ -103,7 +110,7 @@ found:
     pp->p_flags &= ~(PRUNNING | PSTOPPED | PREPORTED);
     if (WIFSTOPPED(w)) {
 	pp->p_flags |= PSTOPPED;
-	pp->p_reason = w.w_stopsig;
+	pp->p_reason = WSTOPSIG(w);
     }
     else {
 	if (pp->p_flags & (PTIME | PPTIME) || adrof(STRtime))
@@ -111,16 +118,16 @@ found:
 
 	pp->p_rusage = ru;
 	if (WIFSIGNALED(w)) {
-	    if (w.w_termsig == SIGINT)
+	    if (w == SIGINT)
 		pp->p_flags |= PINTERRUPTED;
 	    else
 		pp->p_flags |= PSIGNALED;
-	    if (w.w_coredump)
+	    if (w)
 		pp->p_flags |= PDUMPED;
-	    pp->p_reason = w.w_termsig;
+	    pp->p_reason = WTERMSIG(w);
 	}
 	else {
-	    pp->p_reason = w.w_retcode;
+	    pp->p_reason = WEXITSTATUS(w);
 	    if (pp->p_reason != 0)
 		pp->p_flags |= PAEXITED;
 	    else
