--- a/bin/csh/proc.c
+++ b/bin/csh/proc.c
@@ -40,6 +40,7 @@
 
 #include <sys/types.h>
 #include <sys/wait.h>
+#include <bsd/sys/time.h>
 
 #include <errno.h>
 #include <stdarg.h>
@@ -95,7 +96,7 @@
 
 loop:
     errno = 0;			/* reset, just in case */
-    pid = wait3(&w,
+    pid = wait4((pid_t)-1, &w,
        (setintr && (intty || insource) ? WNOHANG | WUNTRACED : WNOHANG), &ru);
 
     if (pid <= 0) {
@@ -846,11 +847,7 @@
 	    if (eq(*v, STRml)) {
 		flag |= FANCY | JOBDIR;
 	    } else if (eq(*v, STRmZ)) {
-		if (v[1] && v[1][0]) {
-		    setproctitle("%s", short2str(v[1]));
-		} else {
-		    setproctitle(NULL);
-		}
+		perror("-Z unavailable on Android.");
 		return;
 	    } else {
 		stderror(ERR_JOBS);
