--- a/proc.c
+++ b/proc.c
@@ -38,12 +38,25 @@
 #include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
+#include <time.h>
+#include <bsd/stdlib.h>
 
 #include "csh.h"
 #include "dir.h"
 #include "proc.h"
 #include "extern.h"
 
+#ifndef timespecsub
+#define timespecsub(a, b, result) do { \
+    (result)->tv_sec = (a)->tv_sec - (b)->tv_sec; \
+    (result)->tv_nsec = (a)->tv_nsec - (b)->tv_nsec; \
+    if ((result)->tv_nsec < 0) { --(result)->tv_sec; (result)->tv_nsec += 1000000000L; } \
+} while (0)
+#endif
+#ifndef timespeccmp
+#define timespeccmp(a, b, cmp) (((a)->tv_sec == (b)->tv_sec) ? ((a)->tv_nsec cmp (b)->tv_nsec) : ((a)->tv_sec cmp (b)->tv_sec))
+#endif
+
 #define BIGINDEX	9	/* largest desirable job index */
 
 struct process proclist;        /* list head of all processes */
@@ -93,7 +106,7 @@
 
 loop:
     errno = 0;			/* reset, just in case */
-    pid = wait3(&w,
+    pid = wait4((pid_t)-1, &w,
        (setintr && (intty || insource) ? WNOHANG | WUNTRACED : WNOHANG), &ru);
 
     if (pid <= 0) {
