--- a/set.c
+++ b/set.c
@@ -31,9 +31,9 @@
  */
 
 #include <sys/types.h>
-#include <stdlib.h>
+#include <bsd/stdlib.h>
 #ifndef SHORT_STRINGS
-#include <string.h>
+#include <bsd/string.h>
 #endif /* SHORT_STRINGS */
 #include <stdarg.h>
 
@@ -65,7 +65,7 @@ doset(Char **v, struct command *t)
     Char *p;
     Char   *vp, op;
     Char  **vecp;
-    bool    hadsub;
+    int    hadsub;
     int     subscr;
 
     v++;
@@ -209,7 +209,7 @@ dolet(Char **v, struct command *t)
 {
     Char *p;
     Char   *vp, c, op;
-    bool    hadsub;
+    int    hadsub;
     int     subscr;
 
     v++;
