--- a/bin/csh/time.c
+++ b/bin/csh/time.c
@@ -44,7 +44,7 @@
 #include "csh.h"
 #include "extern.h"
 #endif
-#include <util.h>
+#include <bsd/libutil.h>
 
 /*
  * C Shell - routines handling process timing and niceing
@@ -189,14 +189,9 @@
 		break;
 	    case 'P':		/* percent time spent running */
 		/* check if it did not run at all */
-		if (ms == 0) {
-			(void)fputs("0.0%", fp);
-		} else {
-			char pb[32];
-			(void)fputs(strpct(pb, sizeof(pb),
-			    (uintmax_t)t, (uintmax_t)ms, 1), fp);
-			(void)fputc('%', fp);
-		}
+		i = (ms == 0) ? 0 : ((long long)t * 1000 / ms);
+		/* nn.n% */
+		(void) fprintf(cshout, "%ld.%01ld%%", i / 10, i % 10);
 		break;
 	    case 'R':		/* page reclaims */
 		(void)fprintf(fp, "%ld", r1->ru_minflt - r0->ru_minflt);
