diff -u -r ../qb64-2.0.2/internal/c/parts/video/font/ttf/os/lnx/setup_build.sh ./internal/c/parts/video/font/ttf/os/lnx/setup_build.sh
--- ../qb64-2.0.2/internal/c/parts/video/font/ttf/os/lnx/setup_build.sh	2021-11-06 20:58:58.000000000 +0000
+++ ./internal/c/parts/video/font/ttf/os/lnx/setup_build.sh	2024-07-09 14:32:29.696163712 +0000
@@ -1,3 +1,3 @@
 #!/bin/sh
-g++ -s -c -w -Wall ../../src/freetypeamalgam.c -o src.o
+g++ -std=c++11 -s -c -w -Wall ../../src/freetypeamalgam.c -o src.o
 
