--- ./src/wgquick/wg-quick.c	2018-02-17 13:57:14.822065231 +0000
+++ ../wg-quick.c	2018-02-17 13:56:48.422380556 +0000
@@ -24,7 +24,7 @@
 #include <sys/param.h>
 
 #ifndef WG_CONFIG_SEARCH_PATHS
-#define WG_CONFIG_SEARCH_PATHS "/data/misc/wireguard /data/data/com.wireguard.android/files"
+#define WG_CONFIG_SEARCH_PATHS "@TERMUX_HOME@ @TERMUX_PREFIX@/etc/wireguard /data/misc/wireguard /data/data/com.wireguard.android/files"
 #endif
 
 #define _printf_(x, y) __attribute__((format(printf, x, y)))
