--- a/check-deps/Makefile
+++ b/check-deps/Makefile
@@ -19,7 +19,7 @@
 # This is needed for checking build dependency
 #
 
-CHECK_CFLAGS  = $(CFLAGS)  $(CFLAGS_$@) -Werror
+CHECK_CFLAGS  = $(CFLAGS)  $(CFLAGS_$@) $(CPPFLAGS) -Werror
 CHECK_LDFLAGS = $(LDFLAGS) $(LDFLAGS_$@)
 
 # libpython3 provides an embed verison of pkg-config file since python3.8
@@ -37,7 +37,7 @@
 CFLAGS_cc_has_mfentry = -mfentry
 LDFLAGS_cxa_demangle = -lstdc++
 LDFLAGS_have_libelf = -lelf
-CFLAGS_cc_has_mno_sse2 = -mno-sse2
+CFLAGS_cc_has_mno_sse2 = -mno-sse -mno-sse2
 LDFLAGS_have_libpython2.7 = -lpython2.7
 CFLAGS_have_libpython3 = $(shell pkg-config python3$(EMBED) --cflags 2> /dev/null)
 LDFLAGS_have_libpython3 = $(shell pkg-config python3$(EMBED) --libs 2> /dev/null)
