--- a/configure
+++ b/configure
@@ -311,11 +311,11 @@
 	@\$(MAKE) -C \$(srcdir)
 
 clean:
-	@rm -rf cmds arch libmcount libtraceevent utils misc
+	@rm -rf cmds arch libmcount libtraceevent utils misc python
 	@rm -f uftrace version.h *.o *.op
 
 prepare:
-	@mkdir -p cmds arch/\$(ARCH) libmcount libtraceevent utils misc
+	@mkdir -p cmds arch/\$(ARCH) libmcount libtraceevent utils misc python
 
 install:
 	@\$(MAKE) -C \$(srcdir) install
