diff -u -r ../brotli-1.0.4/CMakeLists.txt ./CMakeLists.txt
--- ../brotli-1.0.4/CMakeLists.txt	2018-03-29 08:37:07.000000000 +0000
+++ ./CMakeLists.txt	2018-05-27 12:33:42.384489726 +0000
@@ -183,7 +183,7 @@
 
 # Build the brotli executable
 add_executable(brotli ${BROTLI_CLI_C})
-target_link_libraries(brotli ${BROTLI_LIBRARIES_STATIC})
+target_link_libraries(brotli ${BROTLI_LIBRARIES})
 
 # Installation
 if(NOT BROTLI_BUNDLED_MODE)
