diff -uNr loqui-0.6.4/configure loqui-0.6.4.mod/configure
--- loqui-0.6.4/configure	2015-12-12 19:35:59.000000000 +0200
+++ loqui-0.6.4.mod/configure	2019-04-22 19:18:14.278023522 +0300
@@ -15550,7 +15550,7 @@
   fi
 
   if test x"$no_glib" = x ; then
-    GLIB_GENMARSHAL=`$PKG_CONFIG --variable=glib_genmarshal glib-2.0`
+    GLIB_GENMARSHAL=glib-genmarshal
     GOBJECT_QUERY=`$PKG_CONFIG --variable=gobject_query glib-2.0`
     GLIB_MKENUMS=`$PKG_CONFIG --variable=glib_mkenums glib-2.0`
     GLIB_COMPILE_RESOURCES=`$PKG_CONFIG --variable=glib_compile_resources gio-2.0`
