diff --git a/src/wayland/meta-wayland-buffer.c b/src/wayland/meta-wayland-buffer.c
index 6236579..32aedd3 100644
--- a/src/wayland/meta-wayland-buffer.c
+++ b/src/wayland/meta-wayland-buffer.c
@@ -50,7 +50,7 @@
 
 #include "wayland/meta-wayland-buffer.h"
 
-#include <drm_fourcc.h>
+#include <libdrm/drm_fourcc.h>
 
 #include "backends/meta-backend-private.h"
 #include "clutter/clutter.h"
diff --git a/src/wayland/meta-wayland-dma-buf.c b/src/wayland/meta-wayland-dma-buf.c
index ac412d2..5ea699e 100644
--- a/src/wayland/meta-wayland-dma-buf.c
+++ b/src/wayland/meta-wayland-dma-buf.c
@@ -39,7 +39,7 @@
 
 #include "wayland/meta-wayland-dma-buf.h"
 
-#include <drm_fourcc.h>
+#include <libdrm/drm_fourcc.h>
 
 #include "backends/meta-backend-private.h"
 #include "backends/meta-egl-ext.h"
