--- a/configure
+++ b/configure
@@ -22677,10 +22677,8 @@ rm -f conftest*
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for strlcpy" >&5
 $as_echo_n "checking for strlcpy... " >&6; }
 if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
+  $as_echo "yes"
+  $as_echo "#define HAVE_STRLCPY 1" >>confdefs.h
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
@@ -22719,10 +22716,8 @@ fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for strlcat" >&5
 $as_echo_n "checking for strlcat... " >&6; }
 if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
+  $as_echo "yes"
+  $as_echo "#define HAVE_STRLCAT 1" >>confdefs.h
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
@@ -22755,10 +22755,7 @@ fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for C++11 by default" >&5
 $as_echo_n "checking for C++11 by default... " >&6; }
 if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
+  $as_echo "yes"
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
