--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -2197,7 +2197,7 @@ uninstall-am: uninstall-binPROGRAMS uninstall-nodist_pkgdataDATA
 
 
 preferences: genpref$(EXEEXT)
-	$(AM_V_GEN)./genpref$(EXEEXT) -o $@ -s
+	$(AM_V_GEN)termux-proot-run ./genpref$(EXEEXT) -o $@ -s
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
