diff -u -r ../bat-0.7.1/Cargo.toml ./Cargo.toml
--- ../bat-0.7.1/Cargo.toml	2018-09-23 08:30:19.000000000 +0000
+++ ./Cargo.toml	2018-10-08 01:23:55.289414819 +0000
@@ -20,6 +20,7 @@
 console = "0.6"
 directories = "1.0"
 lazy_static = "1.0"
+termios = "0.3"
 
 [dependencies.git2]
 version = "0.7"
@@ -41,5 +42,8 @@
 default-features = false
 features = []
 
+[patch.crates-io]
+termios = { git = "https://github.com/fornwall/termios-rs", branch = "support-android" }
+
 [dev-dependencies]
 tempdir = "0.3"
