diff -uNr go-ethereum-1.10.13/accounts/usbwallet/hub.go go-ethereum-1.10.13.mod/accounts/usbwallet/hub.go
--- go-ethereum-1.10.13/accounts/usbwallet/hub.go	2021-11-24 12:44:10.000000000 +0100
+++ go-ethereum-1.10.13.mod/accounts/usbwallet/hub.go	2021-12-23 01:30:25.369778070 +0100
@@ -26,7 +26,7 @@
 	"github.com/ethereum/go-ethereum/accounts"
 	"github.com/ethereum/go-ethereum/event"
 	"github.com/ethereum/go-ethereum/log"
-	"github.com/karalabe/usb"
+	"github.com/fornwall/usb"
 )
 
 // LedgerScheme is the protocol scheme prefixing account and wallet URLs.
diff -uNr go-ethereum-1.10.13/accounts/usbwallet/wallet.go go-ethereum-1.10.13.mod/accounts/usbwallet/wallet.go
--- go-ethereum-1.10.13/accounts/usbwallet/wallet.go	2021-11-24 12:44:10.000000000 +0100
+++ go-ethereum-1.10.13.mod/accounts/usbwallet/wallet.go	2021-12-23 01:31:14.237912129 +0100
@@ -31,7 +31,7 @@
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/log"
-	"github.com/karalabe/usb"
+	"github.com/fornwall/usb"
 )
 
 // Maximum time between wallet health checks to detect USB unplugs.
diff -uNr go-ethereum-1.10.13/go.mod go-ethereum-1.10.13.mod/go.mod
--- go-ethereum-1.10.13/go.mod	2021-11-24 12:44:10.000000000 +0100
+++ go-ethereum-1.10.13.mod/go.mod	2021-12-23 01:40:39.355410092 +0100
@@ -24,6 +24,7 @@
 	github.com/edsrzf/mmap-go v1.0.0
 	github.com/fatih/color v1.7.0
 	github.com/fjl/memsize v0.0.0-20190710130421-bcb5799ab5e5
+	github.com/fornwall/usb v0.0.0-20191212001557-3a1f5af8866c
 	github.com/gballet/go-libpcsclite v0.0.0-20190607065134-2772fd86a8ff
 	github.com/go-ole/go-ole v1.2.1 // indirect
 	github.com/go-stack/stack v1.8.0
@@ -44,7 +45,6 @@
 	github.com/jackpal/go-nat-pmp v1.0.2-0.20160603034137-1fa385a6f458
 	github.com/jedisct1/go-minisign v0.0.0-20190909160543-45766022959e
 	github.com/julienschmidt/httprouter v1.2.0
-	github.com/karalabe/usb v0.0.0-20211005121534-4c5740d64559
 	github.com/kylelemons/godebug v1.1.0 // indirect
 	github.com/mattn/go-colorable v0.1.8
 	github.com/mattn/go-isatty v0.0.12
diff -uNr go-ethereum-1.10.13/go.sum go-ethereum-1.10.13.mod/go.sum
--- go-ethereum-1.10.13/go.sum	2021-11-24 12:44:10.000000000 +0100
+++ go-ethereum-1.10.13.mod/go.sum	2021-12-23 01:37:34.474954542 +0100
@@ -137,6 +137,8 @@
 github.com/fjl/memsize v0.0.0-20190710130421-bcb5799ab5e5 h1:FtmdgXiUlNeRsoNMFlKLDt+S+6hbjVMEW6RGQ7aUf7c=
 github.com/fjl/memsize v0.0.0-20190710130421-bcb5799ab5e5/go.mod h1:VvhXpOYNQvB+uIk2RvXzuaQtkQJzzIx6lSBe1xv7hi0=
 github.com/fogleman/gg v1.2.1-0.20190220221249-0403632d5b90/go.mod h1:R/bRT+9gY/C5z7JzPU0zXsXHKM4/ayA+zqcVNZzPa1k=
+github.com/fornwall/usb v0.0.0-20191212001557-3a1f5af8866c h1:UEdVa8x3R4rIeAqtLrm5Vhy65uWGrOEPTLCk+M5tfIo=
+github.com/fornwall/usb v0.0.0-20191212001557-3a1f5af8866c/go.mod h1:V2RANuPizn/vsKfOeHqQpxT2CfBd2TowKNE3owVHHoM=
 github.com/fsnotify/fsnotify v1.4.7/go.mod h1:jwhsz4b93w/PPRr/qN1Yymfu8t87LnFCMoQvtojpjFo=
 github.com/fsnotify/fsnotify v1.4.9 h1:hsms1Qyu0jgnwNXIxa+/V/PDsU6CfLf6CNO8H7IWoS4=
 github.com/fsnotify/fsnotify v1.4.9/go.mod h1:znqG4EE+3YCdAaPaxE2ZRY/06pZUdp0tY4IgpuI1SZQ=
@@ -265,8 +267,6 @@
 github.com/julienschmidt/httprouter v1.2.0/go.mod h1:SYymIcj16QtmaHHD7aYtjjsJG7VTCxuUUipMqKk8s4w=
 github.com/jung-kurt/gofpdf v1.0.3-0.20190309125859-24315acbbda5/go.mod h1:7Id9E/uU8ce6rXgefFLlgrJj/GYY22cpxn+r32jIOes=
 github.com/jwilder/encoding v0.0.0-20170811194829-b4e1701a28ef/go.mod h1:Ct9fl0F6iIOGgxJ5npU/IUOhOhqlVrGjyIZc8/MagT0=
-github.com/karalabe/usb v0.0.0-20211005121534-4c5740d64559 h1:0VWDXPNE0brOek1Q8bLfzKkvOzwbQE/snjGojlCr8CY=
-github.com/karalabe/usb v0.0.0-20211005121534-4c5740d64559/go.mod h1:Od972xHfMJowv7NGVDiWVxk2zxnWgjLlJzE+F4F7AGU=
 github.com/kisielk/errcheck v1.2.0/go.mod h1:/BMXB+zMLi60iA8Vv6Ksmxu/1UDYcXs4uQLJ+jE2L00=
 github.com/kisielk/gotool v1.0.0/go.mod h1:XhKaO+MFFWcvkIS/tQcRk01m1F5IRFswLeQ+oQHNcck=
 github.com/kkdai/bstream v0.0.0-20161212061736-f391b8402d23/go.mod h1:J+Gs4SYgM6CZQHDETBtE9HaSEkGmuNXF86RwHhHUvq4=
