diff --git a/cf-agent/cf-agent.c b/cf-agent/cf-agent.c
index 8c6341059..1b454325b 100644
--- a/cf-agent/cf-agent.c
+++ b/cf-agent/cf-agent.c
@@ -365,8 +365,7 @@ static void ConfigureBootstrap(GenericAgentConfig *config, const char *argument)

     if(IsLoopbackAddress(argument))
     {
-        Log(LOG_LEVEL_ERR, "Cannot bootstrap to a loopback address");
-        DoCleanupAndExit(EXIT_FAILURE);
+        Log(LOG_LEVEL_WARNING, "Bootstrapping to loopback interface (localhost), other hosts will not be able to bootstrap to this server");
     }

     // temporary assure that network functions are working

diff --git a/libutils/file_lib.c b/libutils/file_lib.c
index 816d719..d96a27e 100644
--- a/libntech/libutils/file_lib.c
+++ b/libntech/libutils/file_lib.c
@@ -564,6 +540,9 @@ int safe_open_create_perms(
     return open(pathname, flags, create_perms);
 #else // !__MINGW32__

+#ifdef __TERMUX__
+        return open(pathname, flags, create_perms);
+#endif
     const size_t path_bufsize = strlen(pathname) + 1;
     char path[path_bufsize];
     const size_t res_len = StringCopy(pathname, path, path_bufsize);

diff --git a/libutils/platform.h b/libutils/platform.h
index 964adaa..af6e8d7 100644
--- a/libntech/libutils/platform.h
+++ b/libntech/libutils/platform.h
@@ -178,6 +178,9 @@ struct utsname
 #endif

 #include <signal.h>
+#ifdef __ANDROID__
+#include <android/log.h>
+#endif

 #ifdef __MINGW32__
 # define LOG_LOCAL0      (16<<3)

diff --git a/libutils/known_dirs.c b/libutils/known_dirs.c
index 386f6ec..6d0955b 100644
--- a/libntech/libutils/known_dirs.c
+++ b/libntech/libutils/known_dirs.c
@@ -89,7 +89,6 @@ const char *GetDefaultDir_helper(char *dir, const char *root_dir,
     }
 }

-#endif

 #define GET_DEFAULT_DIRECTORY_DEFINE(FUNC, STATIC, GLOBAL, FOLDER)  \
 const char *GetDefault##FUNC##Dir(void)                             \
@@ -106,6 +105,8 @@ GET_DEFAULT_DIRECTORY_DEFINE(Master, master, MASTERDIR, "masterfiles")
 GET_DEFAULT_DIRECTORY_DEFINE(Input, input, INPUTDIR, "inputs")
 GET_DEFAULT_DIRECTORY_DEFINE(State, state, STATEDIR, "state")

+#endif
+
 /*******************************************************************/

 const char *GetWorkDir(void)
