diff --git a/src/cmake/pythonutils.cmake b/src/cmake/pythonutils.cmake
index efab6ea..8a42be0 100644
--- a/src/cmake/pythonutils.cmake
+++ b/src/cmake/pythonutils.cmake
@@ -47,7 +47,6 @@ macro (find_python)
                           VERSION_MIN 3.7
                           RECOMMEND_MIN 3.9
                           RECOMMEND_MIN_REASON "We don't test or support older than 3.9"
-                          COMPONENTS ${_py_components}
                           PRINT Python3_VERSION Python3_EXECUTABLE
                                 Python3_LIBRARIES
                                 Python3_Development_FOUND
