--- a/Makefile
+++ b/Makefile
@@ -48,8 +48,6 @@
 	install -m 755 reptyr $(DESTDIR)$(BINDIR)/reptyr
 	install -d -m 755 $(DESTDIR)$(MANDIR)/man1
 	install -m 644 reptyr.1 $(DESTDIR)$(MANDIR)/man1/reptyr.1
-	install -d -m 755 $(DESTDIR)$(MANDIR)/fr/man1
-	install -m 644 reptyr.fr.1 $(DESTDIR)$(MANDIR)/fr/man1/reptyr.1
 	bashcompdir=$(BASHCOMPDIR) ; \
 	test -z "$$bashcompdir" && bashcompdir=/etc/bash_completion.d ; \
 	install -d -m 755 $(DESTDIR)$$bashcompdir ; \
