--- ../Makefile.in.orig	2021-02-16 06:59:25.596423632 +0000
+++ ./src/Makefile.in	2021-02-16 06:59:45.903064002 +0000
@@ -90,10 +90,10 @@
 host_triplet = @host@
 @MINGW_TRUE@am__append_1 = -lm
 @ANDROID_TRUE@@MINGW_FALSE@am__append_2 = -ldl -lm
-@ANDROID_FALSE@@MINGW_FALSE@am__append_3 = -lpthread -ldl -lm
+@ANDROID_FALSE@@MINGW_FALSE@am__append_3 = -ldl -lm
 @MINGW_TRUE@am__append_4 = -lm
 @ANDROID_TRUE@@MINGW_FALSE@am__append_5 = -ldl -lm
-@ANDROID_FALSE@@MINGW_FALSE@am__append_6 = -lpthread -ldl -lm
+@ANDROID_FALSE@@MINGW_FALSE@am__append_6 = -ldl -lm
 subdir = src
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 am__aclocal_m4_deps = $(top_srcdir)/m4/libtool.m4 \
