diff -uNr rofi-1.5.4/configure rofi-1.5.4.mod/configure
--- rofi-1.5.4/configure	2019-07-01 15:29:12.000000000 +0300
+++ rofi-1.5.4.mod/configure	2019-07-10 16:50:01.284402244 +0300
@@ -7436,10 +7436,10 @@
   fi
 
   if test x"$no_glib" = x ; then
-    GLIB_GENMARSHAL=`$PKG_CONFIG --variable=glib_genmarshal glib-2.0`
-    GOBJECT_QUERY=`$PKG_CONFIG --variable=gobject_query glib-2.0`
-    GLIB_MKENUMS=`$PKG_CONFIG --variable=glib_mkenums glib-2.0`
-    GLIB_COMPILE_RESOURCES=`$PKG_CONFIG --variable=glib_compile_resources gio-2.0`
+    GLIB_GENMARSHAL=glib-genmarshal
+    GOBJECT_QUERY=gobject-query
+    GLIB_MKENUMS=glib-mkenums
+    GLIB_COMPILE_RESOURCES=glib-compile-resources
 
     GLIB_CFLAGS=`$PKG_CONFIG --cflags $pkg_config_args`
     GLIB_LIBS=`$PKG_CONFIG --libs $pkg_config_args`
