--- ../seafile/configure.ac	2018-04-28 11:32:08.357388789 +0200
+++ ./configure.ac	2018-04-28 11:33:03.093385829 +0200
@@ -75,7 +75,7 @@
   fi
 fi
 
-AC_CHECK_LIB(pthread, pthread_create, [echo "found library pthread"], AC_MSG_ERROR([*** Unable to find pthread library]), )
+#AC_CHECK_LIB(pthread, pthread_create, [echo "found library pthread"], AC_MSG_ERROR([*** Unable to find pthread library]), )
 AC_CHECK_LIB(sqlite3, sqlite3_open,[echo "found library sqlite3"] , AC_MSG_ERROR([*** Unable to find sqlite3 library]), )
 
 dnl Do we need to use AX_LIB_SQLITE3 to check sqlite?
@@ -134,6 +134,8 @@
   LIB_CRYPT32=
 fi
 
+LIB_RESOLV=""
+
 AC_SUBST(LIB_WS32)
 AC_SUBST(LIB_GDI32)
 AC_SUBST(LIB_RT)
