--- a/lib/SignOn/SignOn.pro
+++ b/lib/SignOn/SignOn.pro
@@ -4,4 +4,3 @@
     libsignon-qt-shared.pro \
     libsignon-qt-static.pro

-include(doc/doc.pri)

--- a/lib/plugins/plugins.pro
+++ b/lib/plugins/plugins.pro
@@ -3,5 +3,4 @@
 SUBDIRS = signon-plugins.pro signon-plugins-common

 TARGET=signon-plugins
-include(doc/doc.pri)


--- a/signon.pro
+++ b/signon.pro
@@ -7,6 +7,5 @@

 include( common-installs-config.pri )

-include( doc/doc.pri )

 # End of File
