diff -ur orig/scripts/e_gen_menu mod/scripts/e_gen_menu
--- orig/scripts/e_gen_menu	2024-06-24 12:38:21.000000000 +0200
+++ mod/scripts/e_gen_menu	2024-10-28 16:56:02.340591608 +0100
@@ -362,7 +362,7 @@
         dbg += 1
 
 # Likely  prefixes
-Prefixes  = ['/usr/local', '/usr', '/opt']
+Prefixes  = ['/usr/local', '/usr', '/opt', '@TERMUX_PREFIX@/usr']
 Prefixes += ['/opt/kde', '/opt/kde3', getenv('KDEDIR')]
 Prefixes += ['/opt/gnome']                    # SUSE
 Prefixes = MkDirList(Prefixes, ['/share'])
