diff -u -r ../glulxe/Makefile ./Makefile
--- ../glulxe/Makefile	2013-03-11 03:09:38.000000000 +0100
+++ ./Makefile	2014-02-15 13:42:39.000000000 +0100
@@ -7,9 +7,9 @@
 # sets of values appear below; uncomment one of them and change the
 # directories appropriately.
 
-GLKINCLUDEDIR = ../cheapglk
-GLKLIBDIR = ../cheapglk
-GLKMAKEFILE = Make.cheapglk
+GLKINCLUDEDIR = glkterm
+GLKLIBDIR = glkterm
+GLKMAKEFILE = Make.glktermw
 
 #GLKINCLUDEDIR = ../glkterm
 #GLKLIBDIR = ../glkterm
@@ -25,9 +25,8 @@
 
 # Pick a C compiler.
 #CC = cc
-CC = gcc
 
-OPTIONS = -g -Wall -Wmissing-prototypes -Wstrict-prototypes -Wno-unused -DOS_UNIX
+OPTIONS := $(CFLAGS) -DOS_UNIX
 
 include $(GLKINCLUDEDIR)/$(GLKMAKEFILE)
 
