diff -u -r ../gts-0.7.6/src/Makefile.in ./src/Makefile.in
--- ../gts-0.7.6/src/Makefile.in	2006-03-10 00:31:05.000000000 +0000
+++ ./src/Makefile.in	2024-05-16 11:09:10.292099773 +0000
@@ -54,7 +54,7 @@
 am__installdirs = "$(DESTDIR)$(libdir)" "$(DESTDIR)$(bindir)" "$(DESTDIR)$(m4datadir)" "$(DESTDIR)$(includedir)"
 libLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(lib_LTLIBRARIES)
-libgts_la_LIBADD =
+libgts_la_LIBADD = -lm
 am_libgts_la_OBJECTS = object.lo point.lo vertex.lo segment.lo edge.lo \
 	triangle.lo face.lo kdtree.lo bbtree.lo misc.lo predicates.lo \
 	heap.lo eheap.lo fifo.lo matrix.lo surface.lo stripe.lo \
