diff --git a/llbuild/products/llbuildSwift/BuildSystemBindings.swift b/llbuild/products/llbuildSwift/BuildSystemBindings.swift
index 33b3e88..1236069 100644
--- a/llbuild/products/llbuildSwift/BuildSystemBindings.swift
+++ b/llbuild/products/llbuildSwift/BuildSystemBindings.swift
@@ -8,7 +8,7 @@
 
 // This file contains Swift bindings for the llbuild C API.
 
-#if os(Linux)
+#if canImport(Glibc)
 import Glibc
 #elseif os(Windows)
 import MSVCRT
