diff --git a/swift-corelibs-xctest/CMakeLists.txt b/swift-corelibs-xctest/CMakeLists.txt
index 8c9800f..a29a514 100644
--- a/swift-corelibs-xctest/CMakeLists.txt
+++ b/swift-corelibs-xctest/CMakeLists.txt
@@ -55,6 +55,7 @@ add_swift_library(XCTest
 
                     # compatibility with Foundation build_script.py
                     -L${XCTEST_PATH_TO_FOUNDATION_BUILD}/Foundation
+                    -Xlinker -rpath -Xlinker "\\\$\$ORIGIN"
 
                     ${WORKAROUND_SR9138}
                     ${WORKAROUND_SR9995}
