diff -uNr feh-2.25.1/src/options.c feh-2.25.1.mod/src/options.c
--- feh-2.25.1/src/options.c	2018-03-07 18:53:33.000000000 +0200
+++ feh-2.25.1.mod/src/options.c	2018-06-21 17:31:53.803526576 +0300
@@ -62,7 +62,7 @@
 	opt.scroll_step = 20;
 	opt.menu_font = estrdup(DEFAULT_MENU_FONT);
 	opt.font = NULL;
-	opt.menu_bg = estrdup(PREFIX "/share/feh/images/menubg_default.png");
+	opt.menu_bg = estrdup("/data/data/com.termux/files/usr/share/feh/images/menubg_default.png");
 	opt.max_height = opt.max_width = UINT_MAX;
 
 	opt.start_list_at = NULL;
@@ -159,7 +159,7 @@
 
 	free(oldrcpath);
 
-	if (!fp && ((fp = fopen("/etc/feh/themes", "r")) == NULL))
+	if (!fp && ((fp = fopen("/data/data/com.termux/files/usr/etc/feh/themes", "r")) == NULL))
 		return;
 
 	/* Oooh. We have an options file :) */
