diff -uNr feh-3.0/src/Makefile feh-3.0.mod/src/Makefile
--- feh-3.0/src/Makefile	2018-11-09 18:17:58.000000000 +0200
+++ feh-3.0.mod/src/Makefile	2018-11-23 23:26:11.165786637 +0200
@@ -40,7 +40,7 @@
 I_DSTS = ${I_SRCS:.raw=.inc}
 
 feh: deps.mk ${OBJECTS} ${I_DSTS}
-	${CC} ${LDFLAGS} ${CFLAGS} -o $@ ${OBJECTS} ${LDLIBS}
+	${CC} ${LDFLAGS} ${CFLAGS} -o $@ ${OBJECTS} ${LDLIBS} -landroid-shmem
 
 include deps.mk
 
