diff -uNr feh-2.25.1/src/filelist.c feh-2.25.1.mod/src/filelist.c
--- feh-2.25.1/src/filelist.c	2018-03-07 18:53:33.000000000 +0200
+++ feh-2.25.1.mod/src/filelist.c	2018-06-21 17:31:53.786859909 +0300
@@ -159,7 +159,7 @@
 {
 	char buf[1024];
 	size_t readsize;
-	char *sfn = estrjoin("_", "/tmp/feh_stdin", "XXXXXX", NULL);
+	char *sfn = estrjoin("_", "/data/data/com.termux/files/usr/tmp/feh_stdin", "XXXXXX", NULL);
 	int fd = mkstemp(sfn);
 	FILE *outfile;
 
