diff -uNr feh-2.25.1/src/slideshow.c feh-2.25.1.mod/src/slideshow.c
--- feh-2.25.1/src/slideshow.c	2018-03-07 18:53:33.000000000 +0200
+++ feh-2.25.1.mod/src/slideshow.c	2018-06-21 17:31:53.803526576 +0300
@@ -482,7 +482,7 @@
 				if (filelist_tmppath != NULL) {
 					strncat(ret, filelist_tmppath, sizeof(ret) - strlen(ret) - 1);
 				} else {
-					filelist_tmppath = feh_unique_filename("/tmp/","filelist");
+					filelist_tmppath = feh_unique_filename("/data/data/com.termux/files/usr/tmp/","filelist");
 					feh_write_filelist(filelist, filelist_tmppath);
 					strncat(ret, filelist_tmppath, sizeof(ret) - strlen(ret) - 1);
 				}
