diff -uNr feh-3.0/src/filelist.c feh-3.0.mod/src/filelist.c
--- feh-3.0/src/filelist.c	2018-11-09 18:17:58.000000000 +0200
+++ feh-3.0.mod/src/filelist.c	2018-11-23 23:30:02.490608366 +0200
@@ -160,7 +160,7 @@
 {
 	char buf[1024];
 	size_t readsize;
-	char *sfn = estrjoin("_", "/tmp/feh_stdin", "XXXXXX", NULL);
+	char *sfn = estrjoin("_", "@TERMUX_PREFIX@/tmp/feh_stdin", "XXXXXX", NULL);
 	int fd = mkstemp(sfn);
 	FILE *outfile;
 
