--- a/gdk-pixbuf-loader/Makefile.am
+++ b/gdk-pixbuf-loader/Makefile.am
@@ -48,10 +48,7 @@
 	$(AM_LDFLAGS)
 rsvg_loader_LDADD = \
 	$(GDK_PIXBUF_LIBS)
-EXTRA_rsvg_loader_DEPENDENCIES = libpixbufloader-svg.la gdk-pixbuf-loaders
-
-gdk-pixbuf-loaders: Makefile
-	$(AM_V_GEN) ( $(GDK_PIXBUF_QUERYLOADERS) ./libpixbufloader-svg.la && GDK_PIXBUF_MODULEDIR=$(gdk_pixbuf_moduledir) $(GDK_PIXBUF_QUERYLOADERS)) > gdk-pixbuf.loaders 2>/dev/null
+EXTRA_rsvg_loader_DEPENDENCIES = libpixbufloader-svg.la
 
 if CROSS_COMPILING
 RUN_QUERY_LOADER_TEST=false
--- a/gdk-pixbuf-loader/Makefile.in
+++ b/gdk-pixbuf-loader/Makefile.in
@@ -455,7 +455,7 @@
 @ENABLE_PIXBUF_LOADER_TRUE@rsvg_loader_LDADD = \
 @ENABLE_PIXBUF_LOADER_TRUE@	$(GDK_PIXBUF_LIBS)
 
-@ENABLE_PIXBUF_LOADER_TRUE@EXTRA_rsvg_loader_DEPENDENCIES = libpixbufloader-svg.la gdk-pixbuf-loaders
+@ENABLE_PIXBUF_LOADER_TRUE@EXTRA_rsvg_loader_DEPENDENCIES = libpixbufloader-svg.la
 @CROSS_COMPILING_FALSE@@ENABLE_PIXBUF_LOADER_TRUE@RUN_QUERY_LOADER_TEST = test -z "$(DESTDIR)"
 @CROSS_COMPILING_TRUE@@ENABLE_PIXBUF_LOADER_TRUE@RUN_QUERY_LOADER_TEST = false
 @ENABLE_PIXBUF_LOADER_TRUE@thumbnailerdir = $(datadir)/thumbnailers/
@@ -865,9 +865,6 @@
 .PRECIOUS: Makefile
 
 
-@ENABLE_PIXBUF_LOADER_TRUE@gdk-pixbuf-loaders: Makefile
-@ENABLE_PIXBUF_LOADER_TRUE@	$(AM_V_GEN) ( $(GDK_PIXBUF_QUERYLOADERS) ./libpixbufloader-svg.la && GDK_PIXBUF_MODULEDIR=$(gdk_pixbuf_moduledir) $(GDK_PIXBUF_QUERYLOADERS)) > gdk-pixbuf.loaders 2>/dev/null
-
 # The following bit is stolen from gdk-pixbuf/gdk-pixbuf/Makefile.am:
 #
 # Running this if cross compiling or if DESTDIR is set is going to
