diff -u -r ../LuaJIT-2.1.0-beta2/etc/luajit.pc ./etc/luajit.pc
--- ../LuaJIT-2.1.0-beta2/etc/luajit.pc	2016-03-03 06:40:00.000000000 -0500
+++ ./etc/luajit.pc	2016-11-25 17:58:38.914012186 -0500
@@ -5,12 +5,12 @@
 version=${majver}.${minver}.${relver}-beta2
 abiver=5.1
 
-prefix=/usr/local
+prefix=@TERMUX_PREFIX@
 multilib=lib
 exec_prefix=${prefix}
 libdir=${exec_prefix}/${multilib}
 libname=luajit-${abiver}
-includedir=${prefix}/include/luajit-${majver}.${minver}
+includedir=${prefix}/include
 
 INSTALL_LMOD=${prefix}/share/lua/${abiver}
 INSTALL_CMOD=${prefix}/${multilib}/lua/${abiver}
