diff -u -r ../LuaJIT-2.1.0-beta2/src/Makefile ./src/Makefile
--- ../LuaJIT-2.1.0-beta2/src/Makefile	2016-03-03 12:40:00.000000000 +0100
+++ ./src/Makefile	2016-06-11 00:59:21.000000000 +0200
@@ -71,14 +71,14 @@
 # as dynamic mode.
 #
 # Mixed mode creates a static + dynamic library and a statically linked luajit.
-BUILDMODE= mixed
+#BUILDMODE= mixed
 #
 # Static mode creates a static library and a statically linked luajit.
 #BUILDMODE= static
 #
 # Dynamic mode creates a dynamic library and a dynamically linked luajit.
 # Note: this executable will only run when the library is installed!
-#BUILDMODE= dynamic
+BUILDMODE= dynamic
 #
 ##############################################################################
 
@@ -166,10 +166,6 @@
     HOST_SYS= Windows
     HOST_MSYS= cygwin
   endif
-  # Use Clang for OSX host.
-  ifeq (Darwin,$(HOST_SYS))
-    DEFAULT_CC= clang
-  endif
 endif
 
 ##############################################################################
