diff -u -r /home/fornwall/lib/android-ndk/platforms/android-21/arch-arm/usr/include/grp.h ./usr/include/grp.h
--- /home/fornwall/lib/android-ndk/platforms/android-21/arch-arm/usr/include/grp.h	2014-10-14 22:53:49.000000000 -0400
+++ ./usr/include/grp.h	2014-12-14 15:33:15.715243224 -0500
@@ -54,13 +54,13 @@
 struct group	*getgrgid(gid_t);
 struct group	*getgrnam(const char *);
 #if __POSIX_VISIBLE >= 200112 || __XPG_VISIBLE
-struct group	*getgrent(void);
-void		 setgrent(void);
-void		 endgrent(void);
-int		 getgrgid_r(gid_t, struct group *, char *,
-		    size_t, struct group **);
-int		 getgrnam_r(const char *, struct group *, char *,
-		    size_t, struct group **);
+static struct group	*getgrent(void) { return 0; }
+static void		 setgrent(void) {}
+static void		 endgrent(void) {}
+static int		 getgrgid_r(gid_t gid, struct group * grp, char * buf,
+		    size_t buflen, struct group ** result) { *result = 0; return 0; }
+static int		 getgrnam_r(const char * name, struct group * grp, char * buf,
+		    size_t buflen, struct group ** result) { *result = 0; return 0; }
 #endif
 
 int   getgrouplist (const char *user, gid_t group,
