--- /home/fornwall/lib/android-ndk/platforms/android-21/arch-arm64/usr/include/paths.h	2016-10-09 16:37:54.394746195 +0530
+++ ./usr/include/paths.h	2016-10-21 23:28:38.689411903 +0530
@@ -33,12 +33,12 @@
 #define	_PATHS_H_
 
 /* Default search path. */
-#define	_PATH_DEFPATH	"/usr/bin:/bin"
+#define	_PATH_DEFPATH	"@TERMUX_PREFIX@/bin:@TERMUX_PREFIX@/bin/applets"
 /* All standard utilities path. */
 #define	_PATH_STDPATH \
-	"/usr/bin:/bin:/usr/sbin:/sbin"
+	_PATH_DEFPATH
 
-#define	_PATH_BSHELL	"/system/bin/sh"
+#define	_PATH_BSHELL	"@TERMUX_PREFIX@/bin/sh"
 #define	_PATH_CONSOLE	"/dev/console"
 #define	_PATH_CSHELL	"/bin/csh"
 #define	_PATH_DEVDB	"/var/run/dev.db"
@@ -51,7 +51,7 @@
 #define	_PATH_MAN	"/usr/share/man"
 #define	_PATH_MEM	"/dev/mem"
 #define	_PATH_MNTTAB	"/etc/fstab"
-#define	_PATH_MOUNTED	"/etc/mtab"
+#define	_PATH_MOUNTED	"/proc/mounts"
 #define	_PATH_NOLOGIN	"/etc/nologin"
 #define	_PATH_PRESERVE	"/var/lib"
 #define	_PATH_RWHODIR	"/var/spool/rwho"
@@ -66,9 +66,9 @@
 
 /* Provide trailing slash, since mostly used for building pathnames. */
 #define	_PATH_DEV	"/dev/"
-#define	_PATH_TMP	"/tmp/"
-#define	_PATH_VARDB	"/var/db/"
-#define	_PATH_VARRUN	"/var/run/"
-#define	_PATH_VARTMP	"/var/tmp/"
+#define	_PATH_TMP	"@TERMUX_PREFIX@/tmp/"
+#define	_PATH_VARDB	"@TERMUX_PREFIX@/var/db/"
+#define	_PATH_VARRUN	"@TERMUX_PREFIX@/var/run/"
+#define	_PATH_VARTMP	"@TERMUX_PREFIX@/var/tmp/"
 
 #endif /* !_PATHS_H_ */
