diff -u -r /home/fornwall/lib/android-ndk/platforms/android-21/arch-arm/usr/include/paths.h ./usr/include/paths.h
--- /home/fornwall/lib/android-ndk/platforms/android-21/arch-arm/usr/include/paths.h	2016-03-03 16:54:24.000000000 -0500
+++ ./usr/include/paths.h	2016-04-11 03:56:22.460545708 -0400
@@ -66,9 +66,9 @@
 
 /* Provide trailing slash, since mostly used for building pathnames. */
 #define	_PATH_DEV	"/dev/"
-#define	_PATH_TMP	"/tmp/"
-#define	_PATH_VARDB	"/var/db/"
-#define	_PATH_VARRUN	"/var/run/"
-#define	_PATH_VARTMP	"/var/tmp/"
+#define	_PATH_TMP	"@TERMUX_PREFIX@/tmp/"
+#define	_PATH_VARDB	"@TERMUX_PREFIX@/var/db/"
+#define	_PATH_VARRUN	"@TERMUX_PREFIX@/var/run/"
+#define	_PATH_VARTMP	"@TERMUX_PREFIX@/var/tmp/"
 
 #endif /* !_PATHS_H_ */
