diff -u -r /home/fornwall/lib/android-ndk/platforms/android-21/arch-arm64/usr/include/pwd.h ./usr/include/pwd.h
--- /home/fornwall/lib/android-ndk/platforms/android-21/arch-arm64/usr/include/pwd.h	2017-02-27 17:27:10.000000000 +0100
+++ ./usr/include/pwd.h	2017-03-05 13:28:19.227658358 +0100
@@ -65,6 +65,10 @@
 #include <sys/cdefs.h>
 #include <sys/types.h>
 
+/* For access() and realpath(): */
+#include <unistd.h>
+#include <stdlib.h>
+
 #define _PATH_PASSWD        "/etc/passwd"
 #define _PATH_MASTERPASSWD  "/etc/master.passwd"
 #define _PATH_MASTERPASSWD_LOCK "/etc/ptmp"
@@ -119,7 +123,52 @@
 int getpwnam_r(const char*, struct passwd*, char*, size_t, struct passwd**);
 int getpwuid_r(uid_t, struct passwd*, char*, size_t, struct passwd**);
 
-void endpwent(void);
+static void android_setup_pwd(struct passwd* pw) {
+       static char realpath_buffer[4096/*PATH_MAX*/];
+       char* result = realpath("@TERMUX_HOME@/.termux/shell", realpath_buffer);
+       if (result == NULL || access(realpath_buffer, X_OK) == -1) {
+		char const* bash_path = "@TERMUX_PREFIX@/bin/bash";
+		if (access(bash_path, X_OK) != -1) pw->pw_shell = (char*) bash_path;
+		else pw->pw_shell = "@TERMUX_PREFIX@/bin/sh";
+       } else {
+               pw->pw_shell = realpath_buffer;
+       }
+       pw->pw_dir = "@TERMUX_HOME@";
+       pw->pw_passwd = "*";
+#ifdef __LP64__
+       pw->pw_gecos = ""; /* Avoid NULL field. */
+#endif
+}
+
+static struct passwd* android_polyfill_getpwuid(uid_t t) {
+       struct passwd* pw = getpwuid(t);
+       if (pw == NULL) return NULL;
+       android_setup_pwd(pw);
+       return pw;
+}
+
+static struct passwd* android_polyfill_getpwnam(const char* name) {
+       struct passwd* pw = getpwnam(name);
+       if (pw == NULL) return NULL;
+       android_setup_pwd(pw);
+       return pw;
+}
+
+static int android_polyfill_getpwuid_r(uid_t uid,
+                                       struct passwd *pwd,
+                                       char *buffer,
+                                       size_t bufsize,
+                                       struct passwd **result) {
+       int ret = getpwuid_r(uid, pwd, buffer, bufsize, result);
+       if (ret != 0) return ret;
+       android_setup_pwd(pwd);
+       return 0;
+}
+
+#define getpwnam android_polyfill_getpwnam
+#define getpwuid android_polyfill_getpwuid
+#define getpwuid_r android_polyfill_getpwuid_r
+static void endpwent(void) { /* Do nothing. */ }
 struct passwd* getpwent(void);
 int setpwent(void);
 
