--- /home/fornwall/lib/android-ndk/platforms/android-21/arch-arm/usr/include/sys/stat.h	2016-10-12 15:11:58.000000000 +0530
+++ ./usr/include/sys/stat.h	2016-12-03 21:17:42.104829064 +0530
@@ -136,6 +136,10 @@
 #define st_mtimensec st_mtime_nsec
 #define st_ctimensec st_ctime_nsec
 
+#define S_IREAD S_IRUSR
+#define S_IWRITE S_IWUSR
+#define S_IEXEC S_IXUSR
+
 #ifdef __USE_BSD
 /* Permission macros provided by glibc for compatibility with BSDs. */
 #define ACCESSPERMS (S_IRWXU | S_IRWXG | S_IRWXO) /* 0777 */
