diff -u -r /home/fornwall/lib/android-ndk/platforms/android-21/arch-arm64/usr/include/sys/limits.h ./usr/include/sys/limits.h
--- /home/fornwall/lib/android-ndk/platforms/android-21/arch-arm64/usr/include/sys/limits.h	2016-09-29 14:42:26.000000000 -0400
+++ ./usr/include/sys/limits.h	2017-01-07 01:23:19.004048281 -0500
@@ -116,7 +116,7 @@
 /* Bionic-specific definitions */
 
 #define  _POSIX_VERSION             200112L   /* Posix C language bindings version */
-#define  _POSIX2_VERSION            -1        /* we don't support Posix command-line tools */
+#define  _POSIX2_VERSION            200809    /* Termux change to avoid coreutils obsolete breakage */
 #define  _POSIX2_C_VERSION          _POSIX_VERSION
 #define  _XOPEN_VERSION             500       /* by Posix definition */
 #define  _XOPEN_XCU_VERSION         -1        /* we don't support command-line utilities */
