--- a/xdelta3-main.h
+++ b/xdelta3-main.h
@@ -4018,7 +4018,7 @@ main_help (void)
 
   XPR(NTR "compression options:\n");
   XPR(NTR "   -s source    source file to copy from (if any)\n");
-  XPR(NTR "   -S [lzma|djw|fgk] enable/disable secondary compression\n");
+  XPR(NTR "   -S [lzma|djw|fgk|none] enable/disable secondary compression\n");
   XPR(NTR "   -N           disable small string-matching compression\n");
   XPR(NTR "   -D           disable external decompression (encode/decode)\n");
   XPR(NTR "   -R           disable external recompression (decode)\n");
--- a/xdelta3.1
+++ b/xdelta3.1
@@ -92,7 +92,7 @@ compression options:
 .RI source
 source file to copy from (if any)
 .TP
-.BI "\-S " [djw|fgk]
+.BI "\-S " [djw|fgk|lzma|none]
 enable/disable secondary compression
 .TP
 .BI \-N
