--- ./common.gypi.orig	2024-02-18 23:11:41.173679151 +0530
+++ ./common.gypi	2024-02-18 23:11:53.907570677 +0530
@@ -237,10 +237,6 @@
               },],
             ],
           },],
-          ['OS == "android"', {
-            'cflags': [ '-fPIC', '-I<(android_ndk_path)/sources/android/cpufeatures' ],
-            'ldflags': [ '-fPIC' ]
-          }],
         ],
         'msvs_settings': {
           'VCCLCompilerTool': {
