--- ./common.gypi.orig	2024-12-17 11:40:36.782352828 +0530
+++ ./common.gypi	2024-12-17 11:41:11.245672174 +0530
@@ -240,10 +240,6 @@
               },],
             ],
           },],
-          ['OS == "android"', {
-            'cflags': [ '-fPIC', '-I<(android_ndk_path)/sources/android/cpufeatures' ],
-            'ldflags': [ '-fPIC' ]
-          }],
           ['clang==1', {
             'msbuild_toolset': 'ClangCL',
           }],
