--- ./deps/v8/src/logging/log.cc.orig	2024-02-18 22:32:25.705899829 +0530
+++ ./deps/v8/src/logging/log.cc	2024-02-18 22:49:33.993837346 +0530
@@ -367,7 +367,7 @@
   static uint64_t reference_count_;
 };
 
-const char LinuxPerfBasicLogger::kFilenameFormatString[] = "/tmp/perf-%d.map";
+const char LinuxPerfBasicLogger::kFilenameFormatString[] = "@TERMUX_PREFIX@/tmp/perf-%d.map";
 // Extra space for the PID in the filename
 const int LinuxPerfBasicLogger::kFilenameBufferPadding = 16;
 
