--- ./deps/v8/src/flags/flag-definitions.h	2021-06-03 07:15:31.000000000 +0530
+++ ./deps/v8/src/flags/flag-definitions.h.mod	2021-06-18 20:24:38.915642342 +0530
@@ -1842,7 +1842,7 @@
 #undef DEFINE_PERF_PROF_BOOL
 #undef DEFINE_PERF_PROF_IMPLICATION
 
-DEFINE_STRING(gc_fake_mmap, "/tmp/__v8_gc__",
+DEFINE_STRING(gc_fake_mmap, "@TERMUX_PREFIX@/tmp/__v8_gc__",
               "Specify the name of the file for fake gc mmap used in ll_prof")
 DEFINE_BOOL(log_internal_timer_events, false, "Time internal events.")
 DEFINE_IMPLICATION(log_internal_timer_events, prof)
