diff -uNr node-v14.0.0/deps/v8/BUILD.gn node-v14.0.0.mod/deps/v8/BUILD.gn
--- node-v14.0.0/deps/v8/BUILD.gn	2020-04-21 11:38:50.000000000 +0000
+++ node-v14.0.0.mod/deps/v8/BUILD.gn	2020-05-07 14:11:12.247429516 +0000
@@ -28,7 +28,7 @@
 
 declare_args() {
   # Print to stdout on Android.
-  v8_android_log_stdout = false
+  v8_android_log_stdout = true
 
   # Dynamically set an additional dependency from v8/custom_deps.
   v8_custom_deps = ""
diff -uNr node-v14.0.0/deps/v8/infra/mb/mb_config.pyl node-v14.0.0.mod/deps/v8/infra/mb/mb_config.pyl
--- node-v14.0.0/deps/v8/infra/mb/mb_config.pyl	2020-04-21 11:38:50.000000000 +0000
+++ node-v14.0.0.mod/deps/v8/infra/mb/mb_config.pyl	2020-05-07 14:12:44.653746359 +0000
@@ -565,7 +565,7 @@
 
   'mixins': {
     'android': {
-      'gn_args': 'target_os="android" v8_android_log_stdout=true',
+      'gn_args': 'target_os="android" v8_android_log_stdout=false',
     },
 
     'android_strip_outputs': {
diff -uNr node-v14.0.0/src/debug_utils.cc node-v14.0.0.mod/src/debug_utils.cc
--- node-v14.0.0/src/debug_utils.cc	2020-04-21 11:38:51.000000000 +0000
+++ node-v14.0.0.mod/src/debug_utils.cc	2020-05-07 14:12:00.694741330 +0000
@@ -500,7 +500,7 @@
 
   WriteConsoleW(handle, wbuf.data(), n, nullptr, nullptr);
   return;
-#elif defined(__ANDROID__)
+#elif defined(__ANDROID__) && !defined(__TERMUX__)
   if (file == stderr) {
     __android_log_print(ANDROID_LOG_ERROR, "nodejs", "%s", str.data());
     return;
