diff -u -r ../node-v6.0.0/src/cares_wrap.cc ./src/cares_wrap.cc
--- ../node-v6.0.0/src/cares_wrap.cc	2016-04-26 15:50:22.000000000 -0400
+++ ./src/cares_wrap.cc	2016-05-02 00:24:26.349502775 -0400
@@ -15,8 +15,7 @@
 #include <stdlib.h>
 #include <string.h>
 
-#if defined(__ANDROID__) || \
-    defined(__MINGW32__) || \
+#if defined(__MINGW32__) || \
     defined(__OpenBSD__) || \
     defined(_MSC_VER)
