--- ./src/node_internals.h	2021-06-03 07:15:32.000000000 +0530
+++ ./src/node_internals.h.mod	2021-06-18 20:49:41.375641769 +0530
@@ -286,7 +286,7 @@
 
 // Functions defined in node.cc that are exposed via the bootstrapper object
 
-#if defined(__POSIX__) && !defined(__ANDROID__) && !defined(__CloudABI__)
+#if defined(__POSIX__) && !defined(__CloudABI__)
 #define NODE_IMPLEMENTS_POSIX_CREDENTIALS 1
 #endif  // __POSIX__ && !defined(__ANDROID__) && !defined(__CloudABI__)
 
