--- ./test/parallel/test-process-constants-noatime.js.orig	2022-06-21 15:34:16.647265960 +0530
+++ ./test/parallel/test-process-constants-noatime.js	2022-06-21 15:34:38.957265951 +0530
@@ -4,7 +4,7 @@
 const assert = require('assert');
 const constants = require('fs').constants;
 
-if (common.isLinux) {
+if (common.isLinux || process.platform === 'android') {
   assert('O_NOATIME' in constants);
   assert.strictEqual(constants.O_NOATIME, 0x40000);
 } else {
