--- a/external/ddb_dsp_libretro/Makefile.am
+++ b/external/ddb_dsp_libretro/Makefile.am
@@ -4,5 +4,5 @@
 pkglib_LTLIBRARIES = ddb_dsp_libretro.la
 ddb_dsp_libretro_la_SOURCES = libretro.cpp ppsimd/ppsimd.h sinc_resampler.h
 ddb_dsp_libretro_la_LDFLAGS = -module -avoid-version
-ddb_dsp_libretro_la_CXXFLAGS = $(CFLAGS) -msse3 -std=c++11 -I@top_srcdir@/include
+ddb_dsp_libretro_la_CXXFLAGS = $(CFLAGS) -std=c++11 -I@top_srcdir@/include
 endif
--- a/external/ddb_dsp_libretro/Makefile.in
+++ b/external/ddb_dsp_libretro/Makefile.in
@@ -490,7 +490,7 @@
 @HAVE_LIBRETRO_TRUE@pkglib_LTLIBRARIES = ddb_dsp_libretro.la
 @HAVE_LIBRETRO_TRUE@ddb_dsp_libretro_la_SOURCES = libretro.cpp ppsimd/ppsimd.h sinc_resampler.h
 @HAVE_LIBRETRO_TRUE@ddb_dsp_libretro_la_LDFLAGS = -module -avoid-version
-@HAVE_LIBRETRO_TRUE@ddb_dsp_libretro_la_CXXFLAGS = $(CFLAGS) -msse3 -std=c++11 -I@top_srcdir@/include
+@HAVE_LIBRETRO_TRUE@ddb_dsp_libretro_la_CXXFLAGS = $(CFLAGS) -std=c++11 -I@top_srcdir@/include
 all: all-am
 
 .SUFFIXES:
