--- ./src/shims/atomic.h.orig	2020-01-18 03:08:41.044959773 +0000
+++ ./src/shims/atomic.h	2020-01-18 03:08:53.745179911 +0000
@@ -35,7 +35,7 @@
 #if defined(__cplusplus) && (defined(__FreeBSD__) || defined(_WIN32))
 #define _Bool bool
 #endif
-#include <stdatomic.h>
+#include <bits/stdatomic.h>
 
 #define memory_order_ordered    memory_order_seq_cst
 #define memory_order_dependency memory_order_acquire
