https://github.com/aosp-mirror/platform_bionic/commit/76e2b15

diff --git a/src/shims/atomic.h b/src/shims/atomic.h
index c002e72..33ad5fe 100644
--- a/src/shims/atomic.h
+++ b/src/shims/atomic.h
@@ -35,7 +35,7 @@
 #if defined(__cplusplus)
 #define _Bool bool
 #endif
-#include <stdatomic.h>
+#include <bits/stdatomic.h>
 
 #define memory_order_ordered    memory_order_seq_cst
 #define memory_order_dependency memory_order_acquire
