diff -uNr src/CMakeLists.txt src.mod/CMakeLists.txt
--- src/CMakeLists.txt	2021-05-21 09:22:54.192072000 +0000
+++ src.mod/CMakeLists.txt	2021-05-21 09:29:32.473671000 +0000
@@ -57,11 +57,11 @@
 
 # startlxqt script
 set(PREDEF_XDG_DATA_DIRS "$XDG_DATA_HOME")
-if(NOT("${LXQT_DATA_DIR}" MATCHES "^/usr(/local)?/share$"))
+if(NOT("${LXQT_DATA_DIR}" MATCHES "^@TERMUX_PREFIX@(/local)?/share$"))
     set(PREDEF_XDG_DATA_DIRS "${PREDEF_XDG_DATA_DIRS}:${LXQT_DATA_DIR}")
 endif()
-set(PREDEF_XDG_DATA_DIRS "${PREDEF_XDG_DATA_DIRS}:/usr/local/share:/usr/share")
-set(PREDEF_XDG_CONFIG_DIRS "/etc:${LXQT_ETC_XDG_DIR}:/usr/share")
+set(PREDEF_XDG_DATA_DIRS "${PREDEF_XDG_DATA_DIRS}:@TERMUX_PREFIX@/local/share:@TERMUX_PREFIX@/share")
+set(PREDEF_XDG_CONFIG_DIRS "@TERMUX_PREFIX@/etc:${LXQT_ETC_XDG_DIR}:@TERMUX_PREFIX@/share")
 configure_file(startlxqt.in startlxqt @ONLY)
 install(PROGRAMS
     "${CMAKE_CURRENT_BINARY_DIR}/startlxqt"
diff -uNr src/lxqt-config-session/autostartedit.cpp src.mod/lxqt-config-session/autostartedit.cpp
--- src/lxqt-config-session/autostartedit.cpp	2021-04-16 16:36:14.000000000 +0000
+++ src.mod/lxqt-config-session/autostartedit.cpp	2021-05-21 09:27:08.762448000 +0000
@@ -58,7 +58,7 @@
 
 void AutoStartEdit::browse()
 {
-    QString filePath = QFileDialog::getOpenFileName(this, tr("Select Application"), QSL("/usr/bin/"));
+    QString filePath = QFileDialog::getOpenFileName(this, tr("Select Application"), QSL("@TERMUX_PREFIX@/bin/"));
     if (!filePath.isEmpty())
         ui->commandEdit->setText(filePath);
 }
diff -uNr src/lxqt-config-session/sessionconfigwindow.cpp src.mod/lxqt-config-session/sessionconfigwindow.cpp
--- src/lxqt-config-session/sessionconfigwindow.cpp	2021-04-16 16:36:14.000000000 +0000
+++ src.mod/lxqt-config-session/sessionconfigwindow.cpp	2021-05-21 09:26:22.519701000 +0000
@@ -115,7 +115,7 @@
                                             const QString &prompt
                                            )
 {
-    QString fname = QFileDialog::getOpenFileName(cb, prompt, QSL("/usr/bin/"));
+    QString fname = QFileDialog::getOpenFileName(cb, prompt, QSL("@TERMUX_PREFIX@/bin/"));
     if (fname.isEmpty())
         return;
 
diff -uNr src/lxqt-session/src/wmselectdialog.cpp src.mod/lxqt-session/src/wmselectdialog.cpp
--- src/lxqt-session/src/wmselectdialog.cpp	2021-04-16 16:36:14.000000000 +0000
+++ src.mod/lxqt-session/src/wmselectdialog.cpp	2021-05-21 09:25:20.941811000 +0000
@@ -117,7 +117,7 @@
     if (item->data(1, TYPE_ROLE) != SELECT_DLG_TYPE)
         return;
 
-    QString fname = QFileDialog::getOpenFileName(this, QString(), QSL("/usr/bin/"));
+    QString fname = QFileDialog::getOpenFileName(this, QString(), QSL("@TERMUX_PREFIX@/bin/"));
     if (fname.isEmpty())
         return;
