--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -808,9 +808,9 @@
   #   lldb doesn't have the needed changes yet.
   # * Fuchsia isn't supported as zxdb doesn't support simple template names yet.
   # TODO(crbug.com/1379070): Remove if the upstream default ever changes.
-  if (is_clang && !is_nacl && !is_win && !is_apple && !is_fuchsia) {
-    cflags_cc += [ "-gsimple-template-names" ]
-  }
+  # if (is_clang && !is_nacl && !is_win && !is_apple && !is_fuchsia) {
+  #   cflags_cc += [ "-gsimple-template-names" ]
+  # }
 
   # MLGO specific flags. These flags enable an ML-based inliner trained on
   # Chrome on Android (arm32) with ThinLTO enabled, optimizing for size.
