diff -u -r ../screen-4.5.1/config.h.in ./config.h.in
--- ../screen-4.5.1/config.h.in	2017-02-25 16:36:04.156952072 +0100
+++ ./config.h.in	2017-04-24 23:19:53.515909451 +0200
@@ -192,7 +192,9 @@
  * If screen is installed with permissions to update /etc/utmp (such
  * as if it is installed set-uid root), define UTMPOK.
  */
-#define UTMPOK
+#ifndef __ANDROID__
+# define UTMPOK
+#endif
 
 /* Set LOGINDEFAULT to one (1)
  * if you want entries added to /etc/utmp by default, else set it to
