--- ../cache/screen-4.4.0/misc.c	2016-06-19 19:41:03.000000000 +0000
+++ ./misc.c	2016-12-20 03:13:04.387236649 +0000
@@ -37,7 +37,6 @@
 #ifdef SVR4
 # include <sys/resource.h>
 #endif
-
 extern struct layer *flayer;
 
 extern int eff_uid, real_uid;
@@ -379,6 +378,8 @@
 #endif /* SVR4 */
 #if defined(SYSV) && defined(NOFILE) && !defined(ISC)
   f = NOFILE;
+#elif defined(__ANDROID__)
+  f = sysconf(_SC_OPEN_MAX);
 #else /* SYSV && !ISC */
   f = getdtablesize();
 #endif /* SYSV && !ISC */
