--- ../cache/screen-4.4.0/screen.h	2016-06-19 19:41:03.000000000 +0000
+++ ./screen.h	2016-12-20 04:29:51.331170619 +0000
@@ -104,9 +104,9 @@
 #define MAXHISTHEIGHT		3000
 #define DEFAULTHISTHEIGHT	100
 #if defined(NAME_MAX) && NAME_MAX < 16
-# define DEFAULT_BUFFERFILE     "/tmp/screen-xchg"
+# define DEFAULT_BUFFERFILE     "@TERMUX_PREFIX@/tmp/screen-xchg"
 #else
-# define DEFAULT_BUFFERFILE	"/tmp/screen-exchange"
+# define DEFAULT_BUFFERFILE	"@TERMUX_PREFIX@/tmp/screen-exchange"
 #endif
 
 
