diff --git a/src/window.c b/src/window.c
index 042d152..8e1cd87 100644
--- a/window.c
+++ b/window.c
@@ -80,9 +80,9 @@ static int zmodem_parse(Window *, char *, size_t);
 
 bool VerboseCreate = false;		/* XXX move this to user.h */
 
-char DefaultShell[] = "/bin/sh";
+char DefaultShell[] = "@TERMUX_PREFIX@/bin/login";
 #ifndef HAVE_EXECVPE
-static char DefaultPath[] = ":/usr/ucb:/bin:/usr/bin";
+static char DefaultPath[] = ":@TERMUX_PREFIX@/bin";
 #endif
 
 struct NewWindow nwin_undef = {
