--- a/brut.apktool/apktool-lib/src/main/java/brut/androlib/res/AndrolibResources.java
+++ b/brut.apktool/apktool-lib/src/main/java/brut/androlib/res/AndrolibResources.java
@@ -611,7 +611,7 @@
             throws AndrolibException {
 
         String aaptPath = apkOptions.aaptPath;
-        boolean customAapt = !aaptPath.isEmpty();
+        boolean customAapt = true;
         List<String> cmd = new ArrayList<>();
 
         try {
