diff -uNr lua-5.2.4/src/luaconf.h lua-5.2.4.mod/src/luaconf.h
--- lua-5.2.4/src/luaconf.h	2013-11-21 19:26:16.000000000 +0200
+++ lua-5.2.4.mod/src/luaconf.h	2018-06-26 16:12:33.824499040 +0300
@@ -100,7 +100,7 @@
 #else			/* }{ */
 
 #define LUA_VDIR	LUA_VERSION_MAJOR "." LUA_VERSION_MINOR "/"
-#define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT	"@TERMUX_PREFIX@/"
 #define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR
 #define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR
 #define LUA_PATH_DEFAULT  \
