diff -u -r ../gtypist-2.9.4/src/cursmenu.c ./src/cursmenu.c
--- ../gtypist-2.9.4/src/cursmenu.c	2013-08-18 16:36:14.000000000 +0200
+++ ./src/cursmenu.c	2014-03-01 01:17:09.000000000 +0100
@@ -24,7 +24,7 @@
 #ifdef HAVE_PDCURSES
 #include <curses.h>
 #else
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 #endif
 
 #include "error.h"
diff -u -r ../gtypist-2.9.4/src/error.c ./src/error.c
--- ../gtypist-2.9.4/src/error.c	2013-08-18 16:36:14.000000000 +0200
+++ ./src/error.c	2014-03-01 01:17:09.000000000 +0100
@@ -25,7 +25,7 @@
 #ifdef HAVE_PDCURSES
 #include <curses.h>
 #else
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 #endif
 
 #include <stdlib.h>
diff -u -r ../gtypist-2.9.4/src/script.c ./src/script.c
--- ../gtypist-2.9.4/src/script.c	2013-08-18 16:36:14.000000000 +0200
+++ ./src/script.c	2014-03-01 01:17:09.000000000 +0100
@@ -24,7 +24,7 @@
 #ifdef HAVE_PDCURSES
 #include <curses.h>
 #else
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 #endif
 
 #include "error.h"
diff -u -r ../gtypist-2.9.4/src/utf8.c ./src/utf8.c
--- ../gtypist-2.9.4/src/utf8.c	2013-08-18 16:36:14.000000000 +0200
+++ ./src/utf8.c	2014-03-01 01:17:09.000000000 +0100
@@ -23,7 +23,7 @@
 #ifdef HAVE_PDCURSES
 #include <curses.h>
 #else
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 #endif
 
 #include <stdlib.h>
