diff --git a/src/lib/buffer/ThreadDecoder.c b/src/lib/buffer/ThreadDecoder.c
index ca80a52..3216be6 100644
--- a/src/lib/buffer/ThreadDecoder.c
+++ b/src/lib/buffer/ThreadDecoder.c
@@ -199,7 +199,7 @@ void Ppmd7T_Free(CPpmd7 *cPpmd7, ppmd_info *threadInfo, IAllocPtr allocator) {
         pthread_cond_broadcast(&tc->notEmpty);
         pthread_mutex_unlock(&tc->mutex);
 
-        pthread_cancel(tc->handle);
+        pthread_kill(tc->handle, SIGTERM);
         pthread_join(tc->handle, NULL);
         tc->finished = True;
     }
@@ -308,7 +308,7 @@ void Ppmd8T_Free(CPpmd8 *cPpmd8, ppmd_info *threadInfo, IAllocPtr allocator) {
         pthread_cond_broadcast(&tc->notEmpty);
         pthread_mutex_unlock(&tc->mutex);
 
-        pthread_cancel(tc->handle);
+        pthread_kill(tc->handle, SIGTERM);
         pthread_join(tc->handle, NULL);
         tc->finished = True;
     }
