--- a/plugins/cpu/linuxcpuplugin.cpp
+++ b/plugins/cpu/linuxcpuplugin.cpp
@@ -82,7 +82,10 @@ LinuxCpuPluginPrivate::LinuxCpuPluginPrivate(CpuPlugin *q, const QString &cpuInf
     for (QByteArray line = cpuinfo.readLine(); !line.isEmpty(); line = cpuinfo.readLine()) {
         CpuInfo info;
         // Processors are divided by empty lines
-        for (; line != "\n";  line = cpuinfo.readLine()) {
+        // but actually sometimes /proc/cpuinfo ends with a nonempty line, so in that case
+        // QIODevice::readLine() will return an empty string instead of a newline, so
+        // check for that otherwise an infinite loop will occur
+        for (; line != "\n" && line != "";  line = cpuinfo.readLine()) {
             // we are interested in processor number as identifier for /proc/stat, physical id (the
             // cpu the core belongs to) and the number of the core. However with hyperthreading
             // multiple entries will have the same combination of physical id and core id. So we just

