--- a/Makefile.in
+++ b/Makefile.in
@@ -169,7 +169,7 @@ am__installdirs = "$(DESTDIR)$(man1dir)" \
 	"$(DESTDIR)$(polkit_policydir)"
 NROFF = nroff
 MANS = $(man_MANS)
-DATA = $(Applications_DATA) $(appdata_DATA) $(polkit_policy_DATA)
+DATA = $(Applications_DATA) $(appdata_DATA)
 RECURSIVE_CLEAN_TARGETS = mostlyclean-recursive clean-recursive	\
   distclean-recursive maintainer-clean-recursive
 am__recursive_targets = \
@@ -982,7 +982,7 @@ info: info-recursive
 info-am:
 
 install-data-am: install-ApplicationsDATA install-appdataDATA \
-	install-man install-polkit_policyDATA
+	install-man
 
 install-dvi: install-dvi-recursive
 
