diff --git a/lua-shell/meson.build b/lua-shell/meson.build
index fff53fb..02c068c 100644
--- a/lua-shell/meson.build
+++ b/lua-shell/meson.build
@@ -4,7 +4,7 @@ endif
 
 config_h.set('BUILD_LUA_SHELL', '1')
 
-dep_lua = dependency('lua', version: '>= 5.4', required: false)
+dep_lua = dependency('lua55', version: '>= 5.4', required: false)
 if not dep_lua.found()
 	error('lua-shell requires Lua >= 5.4 which was not found. Or, you can use \'-Dshell-lua=false\'.')
 endif

