diff -uNr SDL2-2.0.8/include/SDL_main.h SDL2-2.0.8.mod/include/SDL_main.h
--- SDL2-2.0.8/include/SDL_main.h	2018-03-01 18:34:41.000000000 +0200
+++ SDL2-2.0.8.mod/include/SDL_main.h	2018-09-07 16:13:56.450237991 +0300
@@ -59,17 +59,6 @@
  */
 #define SDL_MAIN_NEEDED
 
-#elif defined(__ANDROID__)
-/* On Android SDL provides a Java class in SDLActivity.java that is the
-   main activity entry point.
-
-   See docs/README-android.md for more details on extending that class.
- */
-#define SDL_MAIN_NEEDED
-
-/* We need to export SDL_main so it can be launched from Java */
-#define SDLMAIN_DECLSPEC    DECLSPEC
-
 #elif defined(__NACL__)
 /* On NACL we use ppapi_simple to set up the application helper code,
    then wait for the first PSE_INSTANCE_DIDCHANGEVIEW event before 
