diff -uNr SDL2-2.0.8/src/video/khronos/vulkan/vk_platform.h SDL2-2.0.8.mod/src/video/khronos/vulkan/vk_platform.h
--- SDL2-2.0.8/src/video/khronos/vulkan/vk_platform.h	2018-03-01 18:34:42.000000000 +0200
+++ SDL2-2.0.8.mod/src/video/khronos/vulkan/vk_platform.h	2018-09-07 16:13:56.483571544 +0300
@@ -51,16 +51,6 @@
     #define VKAPI_ATTR
     #define VKAPI_CALL __stdcall
     #define VKAPI_PTR  VKAPI_CALL
-#elif defined(__ANDROID__) && defined(__ARM_ARCH) && __ARM_ARCH < 7
-    #error "Vulkan isn't supported for the 'armeabi' NDK ABI"
-#elif defined(__ANDROID__) && defined(__ARM_ARCH) && __ARM_ARCH >= 7 && defined(__ARM_32BIT_STATE)
-    // On Android 32-bit ARM targets, Vulkan functions use the "hardfloat"
-    // calling convention, i.e. float parameters are passed in registers. This
-    // is true even if the rest of the application passes floats on the stack,
-    // as it does by default when compiling for the armeabi-v7a NDK ABI.
-    #define VKAPI_ATTR __attribute__((pcs("aapcs-vfp")))
-    #define VKAPI_CALL
-    #define VKAPI_PTR  VKAPI_ATTR
 #else
     // On other platforms, use the default calling convention
     #define VKAPI_ATTR
