diff -uNr SDL2-2.0.8/include/SDL_rwops.h SDL2-2.0.8.mod/include/SDL_rwops.h
--- SDL2-2.0.8/include/SDL_rwops.h	2018-03-01 18:34:41.000000000 +0200
+++ SDL2-2.0.8.mod/include/SDL_rwops.h	2018-09-07 16:13:56.453571346 +0300
@@ -93,20 +93,7 @@
     Uint32 type;
     union
     {
-#if defined(__ANDROID__)
-        struct
-        {
-            void *fileNameRef;
-            void *inputStreamRef;
-            void *readableByteChannelRef;
-            void *readMethod;
-            void *assetFileDescriptorRef;
-            long position;
-            long size;
-            long offset;
-            int fd;
-        } androidio;
-#elif defined(__WIN32__)
+#if   defined(__WIN32__)
         struct
         {
             SDL_bool append;
