diff -uNr SDL2-2.0.8/include/SDL_platform.h SDL2-2.0.8.mod/include/SDL_platform.h
--- SDL2-2.0.8/include/SDL_platform.h	2018-03-01 18:34:41.000000000 +0200
+++ SDL2-2.0.8.mod/include/SDL_platform.h	2018-09-07 16:15:56.164360808 +0300
@@ -60,11 +60,6 @@
 #undef __LINUX__
 #define __LINUX__   1
 #endif
-#if defined(ANDROID) || defined(__ANDROID__)
-#undef __ANDROID__
-#undef __LINUX__ /* do we need to do this? */
-#define __ANDROID__ 1
-#endif
 
 #if defined(__APPLE__)
 /* lets us know what version of Mac OS X we're compiling on */
