diff -uNr SDL2-2.0.8/src/atomic/SDL_atomic.c SDL2-2.0.8.mod/src/atomic/SDL_atomic.c
--- SDL2-2.0.8/src/atomic/SDL_atomic.c	2018-03-01 18:34:42.000000000 +0200
+++ SDL2-2.0.8.mod/src/atomic/SDL_atomic.c	2018-09-07 16:13:56.456904701 +0300
@@ -41,9 +41,7 @@
 #   if __has_builtin(__atomic_load_n)
       /* !!! FIXME: this advertises as available in the NDK but uses an external symbol we don't have.
          It might be in a later NDK or we might need an extra library? --ryan. */
-#     if !defined(__ANDROID__)
 #       define HAVE_ATOMIC_LOAD_N 1
-#     endif
 #   endif
 # elif defined(__GNUC__)
 #   if (__GNUC__ >= 5)
