diff -uNr SDL2-2.0.8/src/joystick/SDL_sysjoystick.h SDL2-2.0.8.mod/src/joystick/SDL_sysjoystick.h
--- SDL2-2.0.8/src/joystick/SDL_sysjoystick.h	2018-03-01 18:34:42.000000000 +0200
+++ SDL2-2.0.8.mod/src/joystick/SDL_sysjoystick.h	2018-09-07 16:13:56.476904833 +0300
@@ -126,10 +126,6 @@
 extern SDL_bool SDL_SYS_IsXInputGamepad_DeviceIndex(int device_index);
 #endif
 
-#if defined(__ANDROID__)
-/* Function returns SDL_TRUE if this device is a DPAD (maybe a TV remote) */
-extern SDL_bool SDL_SYS_IsDPAD_DeviceIndex(int device_index);
-#endif
 
 #endif /* SDL_sysjoystick_h_ */
 
