--- a/src/NzbDrone.Common/Extensions/IEnumerableExtensions.cs
+++ b/src/NzbDrone.Common/Extensions/IEnumerableExtensions.cs
@@ -143,11 +143,19 @@
 
         public static string ConcatToString<TSource>(this IEnumerable<TSource> source, string separator = ", ")
         {
-            return string.Join(separator, source.Select(x => x.ToString()));
+            if (source == null)
+            {
+                return string.Empty;
+            }
+            return string.Join(separator, source.Select(x => x?.ToString() ?? string.Empty));
         }
 
         public static string ConcatToString<TSource>(this IEnumerable<TSource> source, Func<TSource, string> predicate, string separator = ", ")
         {
+            if (source == null)
+            {
+                return string.Empty;
+            }
             return string.Join(separator, source.Select(predicate));
         }
     }
