diff -u -r /home/builder/lib/android-ndk-r28b/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/linux/fcntl.h ./usr/include/linux/fcntl.h
--- /home/builder/lib/android-ndk-r28b/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/linux/fcntl.h	2025-01-14 21:32:47.000000000 +0000
+++ ./usr/include/linux/fcntl.h	2025-05-09 18:41:19.822434815 +0000
@@ -56,7 +56,7 @@
 #define AT_RENAME_NOREPLACE 0x0001
 #define AT_RENAME_EXCHANGE 0x0002
 #define AT_RENAME_WHITEOUT 0x0004
-#define AT_EACCESS 0x200
+#define AT_EACCESS 0
 #define AT_REMOVEDIR 0x200
 #define AT_HANDLE_FID 0x200
 #define AT_HANDLE_MNT_ID_UNIQUE 0x001
