--- ./src/CMakeLists.txt.orig	2022-09-24 19:21:35.591929038 +0000
+++ ./src/CMakeLists.txt	2022-09-24 19:21:54.019880966 +0000
@@ -49,11 +49,7 @@
   $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
 
 if(NOT WIN32)
-  find_library(LIBM_LIBRARY NAMES m)
-  if(NOT LIBM_LIBRARY)
-    message(FATAL_ERROR "Could NOT find required library LibM")
-  endif()
-  target_link_libraries(ccd "${LIBM_LIBRARY}")
+  target_link_libraries(ccd m)
   if(BUILD_SHARED_LIBS)
     set(CCD_PKGCONFIG_EXTRA_LIBS -lm PARENT_SCOPE)
   endif()
