diff --git a/data/data/bin.mt.plus/home/test/containerd-1.4.3/runtime/v2/shim/util_unix.go b/runtime/v2/shim/util_unix.go
index 2b0d0ad..9adb593 100644
--- a/data/data/bin.mt.plus/home/test/containerd-1.4.3/runtime/v2/shim/util_unix.go
+++ b/runtime/v2/shim/util_unix.go
@@ -66,7 +66,7 @@ func AdjustOOMScore(pid int) error {
 	return nil
 }
 
-const socketRoot = "/run/containerd"
+const socketRoot = "/data/docker/run/containerd"
 
 // SocketAddress returns a socket address
 func SocketAddress(ctx context.Context, socketPath, id string) (string, error) {
