--- a/magic/src/confetti.c
+++ b/magic/src/confetti.c
@@ -11,8 +11,8 @@
 #define CONFETTI_BRUSH_SIZE 8   //radius of each confetti circle
 #define CONFETTI_QUANTITY 3     //how many circles will be created every click?
 
-#ifdef __ANDROID__
-#define inline static
+#ifdef __TERMUX__
+#define inline static inline
 #endif
 
 struct confetti_rgb
--- a/magic/src/fold.c
+++ b/magic/src/fold.c
@@ -16,8 +16,8 @@
 
 #define FOLD_LEN 80
 
-#ifdef __ANDROID__
-#define inline static
+#ifdef __TERMUX__
+#define inline static inline
 #endif
 
 int right_arm_x, right_arm_y, left_arm_x, left_arm_y;
--- a/magic/src/fretwork.c
+++ b/magic/src/fretwork.c
@@ -28,8 +28,8 @@
 #define SEG_RIGHT_TOP_BOTTOM (SEG_RIGHT | SEG_TOP | SEG_BOTTOM)
 #define SEG_LEFT_RIGHT_TOP_BOTTOM (SEG_LEFT | SEG_RIGHT | SEG_TOP | SEG_BOTTOM)
 
-#ifdef __ANDROID__
-#define inline static
+#ifdef __TERMUX__
+#define inline static inline
 #endif
 
 Mix_Chunk *fretwork_snd;
--- a/magic/src/rails.c
+++ b/magic/src/rails.c
@@ -28,8 +28,8 @@
 #define SEG_RIGHT_TOP_BOTTOM (SEG_RIGHT | SEG_TOP | SEG_BOTTOM)
 #define SEG_LEFT_RIGHT_TOP_BOTTOM (SEG_LEFT | SEG_RIGHT | SEG_TOP | SEG_BOTTOM)
 
-#ifdef __ANDROID__
-#define inline static
+#ifdef __TERMUX__
+#define inline static inline
 #endif
 
 Mix_Chunk *rails_snd;
