diff -u -r ../ranger-1.7.0/ranger/ext/rifle.py ./ranger/ext/rifle.py
--- ../ranger-1.7.0/ranger/ext/rifle.py	2015-04-13 18:25:24.000000000 -0400
+++ ./ranger/ext/rifle.py	2015-04-25 04:10:19.259100887 -0400
@@ -341,7 +341,7 @@
                 if 'r' in flags:
                     prefix = ['sudo', '-E', 'su', '-mc']
                 else:
-                    prefix = ['/bin/sh', '-c']
+                    prefix = ['@TERMUX_PREFIX@/bin/sh', '-c']
 
                 cmd = prefix + [command]
                 if 't' in flags:
