--- a/crates/eww/src/paths.rs
+++ b/crates/eww/src/paths.rs
@@ -36,7 +36,7 @@
 
         let ipc_socket_file = std::env::var("XDG_RUNTIME_DIR")
             .map(std::path::PathBuf::from)
-            .unwrap_or_else(|_| std::path::PathBuf::from("/tmp"))
+            .unwrap_or_else(|_| std::path::PathBuf::from("@TERMUX_PREFIX@/tmp"))
             .join(format!("eww-server_{}", daemon_id));
 
         // 100 as the limit isn't quite 108 everywhere (i.e 104 on BSD or mac)
--- a/crates/yuck/src/config/script_var_definition.rs
+++ b/crates/yuck/src/config/script_var_definition.rs
@@ -106,6 +106,6 @@
             iter.expect_done()?;
             Ok(Self { name_span, name: VarName(name), command: script.to_string(), initial_value, command_span })
         })();
-        result.note(r#"Expected format: `(deflisten name :initial "0" "tail -f /tmp/example")`"#)
+        result.note(r#"Expected format: `(deflisten name :initial "0" "tail -f @TERMUX_PREFIX@/tmp/example")`"#)
     }
 }
