diff -u -r ../libev-4.22/Makefile.in ./Makefile.in
--- ../libev-4.22/Makefile.in	2015-12-20 16:12:45.000000000 -0500
+++ ./Makefile.in	2016-02-02 10:30:17.445084694 -0500
@@ -343,7 +343,7 @@
 	     ev.3 ev.pod Symbols.ev Symbols.event
 
 man_MANS = ev.3
-include_HEADERS = ev.h ev++.h event.h
+include_HEADERS = ev.h ev++.h
 lib_LTLIBRARIES = libev.la
 libev_la_SOURCES = ev.c event.c
 libev_la_LDFLAGS = -version-info $(VERSION_INFO)
