diff -uNr libnatpmp-20150609/Makefile libnatpmp-20150609.mod/Makefile
--- libnatpmp-20150609/Makefile	2013-12-09 22:52:28.000000000 +0200
+++ libnatpmp-20150609.mod/Makefile	2020-01-26 18:59:59.510655815 +0200
@@ -5,7 +5,7 @@
 # http://miniupnp.free.fr/libnatpmp.html
 
 OS = $(shell uname -s)
-CC = gcc
+CC ?= gcc
 INSTALL = install -p
 ARCH = $(shell uname -m | sed -e s/i.86/i686/)
 VERSION = $(shell cat VERSION)
