--- ../cache/vid.stab-1.1.0/CMakeLists.txt	2017-05-30 08:23:59.000000000 +0000
+++ ./CMakeLists.txt	2018-11-30 02:08:08.838885695 +0000
@@ -73,8 +73,8 @@
 set(PKG_EXTRA_LIBS "${PKG_EXTRA_LIBS} ${ORC_LIBRARIES}")
 endif()
 if(USE_OMP)
-target_link_libraries(vidstab gomp)
-set(PKG_EXTRA_LIBS "${PKG_EXTRA_LIBS} -lgomp")
+target_link_libraries(vidstab omp)
+set(PKG_EXTRA_LIBS "${PKG_EXTRA_LIBS} -lomp")
 endif()
 
 
