--- FreeImage/Source/LibRawLite/internal/dcraw_common.cpp	2018-06-30 02:08:06.000000000 +0530
+++ FreeImage-patch/Source/LibRawLite/internal/dcraw_common.cpp	2021-08-25 20:27:55.531999716 +0530
@@ -27,6 +27,7 @@
 #include "libraw/libraw.h"
 #include "internal/defines.h"
 #include "internal/var_defines.h"
+#include <swab.h>
 int CLASS fcol(int row, int col)
 {
   static const char filter[16][16] = {
