diff -uNr go/src/runtime/cgo/gcc_linux_386.c go.mod/src/runtime/cgo/gcc_linux_386.c
--- go/src/runtime/cgo/gcc_linux_386.c	2020-12-03 19:32:44.000000000 +0200
+++ go.mod/src/runtime/cgo/gcc_linux_386.c	2020-12-29 01:13:05.120640580 +0200
@@ -12,7 +12,11 @@
 static void (*setg_gcc)(void*);
 
 // This will be set in gcc_android.c for android-specific customization.
+#ifdef __ANDROID__
+extern void (*x_cgo_inittls)(void **tlsg, void **tlsbase);
+#else
 void (*x_cgo_inittls)(void **tlsg, void **tlsbase);
+#endif
 
 void
 x_cgo_init(G *g, void (*setg)(void*), void **tlsg, void **tlsbase)
diff -uNr go/src/runtime/cgo/gcc_linux_amd64.c go.mod/src/runtime/cgo/gcc_linux_amd64.c
--- go/src/runtime/cgo/gcc_linux_amd64.c	2020-12-03 19:32:44.000000000 +0200
+++ go.mod/src/runtime/cgo/gcc_linux_amd64.c	2020-12-29 01:11:57.624063728 +0200
@@ -14,7 +14,11 @@
 static void (*setg_gcc)(void*);
 
 // This will be set in gcc_android.c for android-specific customization.
+#ifdef __ANDROID__
+extern void (*x_cgo_inittls)(void **tlsg, void **tlsbase);
+#else
 void (*x_cgo_inittls)(void **tlsg, void **tlsbase);
+#endif
 
 void
 x_cgo_init(G *g, void (*setg)(void*), void **tlsg, void **tlsbase)
diff -uNr go/src/runtime/cgo/gcc_linux_arm64.c go.mod/src/runtime/cgo/gcc_linux_arm64.c
--- go/src/runtime/cgo/gcc_linux_arm64.c	2020-12-03 19:32:44.000000000 +0200
+++ go.mod/src/runtime/cgo/gcc_linux_arm64.c	2020-12-29 01:12:17.648234844 +0200
@@ -12,7 +12,11 @@
 
 static void *threadentry(void*);
 
+#ifdef __ANDROID__
+extern void (*x_cgo_inittls)(void **tlsg, void **tlsbase);
+#else
 void (*x_cgo_inittls)(void **tlsg, void **tlsbase);
+#endif
 static void (*setg_gcc)(void*);
 
 void
diff -uNr go/src/runtime/cgo/gcc_linux_arm.c go.mod/src/runtime/cgo/gcc_linux_arm.c
--- go/src/runtime/cgo/gcc_linux_arm.c	2020-12-03 19:32:44.000000000 +0200
+++ go.mod/src/runtime/cgo/gcc_linux_arm.c	2020-12-29 01:12:38.656414386 +0200
@@ -10,7 +10,11 @@
 
 static void *threadentry(void*);
 
+#ifdef __ANDROID__
+extern void (*x_cgo_inittls)(void **tlsg, void **tlsbase);
+#else
 void (*x_cgo_inittls)(void **tlsg, void **tlsbase);
+#endif
 static void (*setg_gcc)(void*);
 
 void
