https://github.com/golang/go/commit/7aa85e01376d840acc8bb931156d607a00b64a60

diff -uNr go/src/cmd/go/internal/cfg/cfg.go go.mod/src/cmd/go/internal/cfg/cfg.go
--- go/src/cmd/go/internal/cfg/cfg.go	2023-08-05 04:14:06.000000000 +0800
+++ go.mod/src/cmd/go/internal/cfg/cfg.go	2023-08-12 16:33:35.167682584 +0800
@@ -417,8 +417,8 @@
 	GOPPC64  = envOr("GOPPC64", fmt.Sprintf("%s%d", "power", buildcfg.GOPPC64))
 	GOWASM   = envOr("GOWASM", fmt.Sprint(buildcfg.GOWASM))
 
-	GOPROXY    = envOr("GOPROXY", "")
-	GOSUMDB    = envOr("GOSUMDB", "")
+	GOPROXY    = envOr("GOPROXY", "https://proxy.golang.org,direct")
+	GOSUMDB    = envOr("GOSUMDB", "sum.golang.org")
 	GOPRIVATE  = Getenv("GOPRIVATE")
 	GONOPROXY  = envOr("GONOPROXY", GOPRIVATE)
 	GONOSUMDB  = envOr("GONOSUMDB", GOPRIVATE)
