diff -u -r ../go/src/runtime/cgo/cgo.go ./src/runtime/cgo/cgo.go
--- ../go/src/runtime/cgo/cgo.go	2015-08-06 00:36:35.000000000 -0400
+++ ./src/runtime/cgo/cgo.go	2015-08-16 16:03:29.614948543 -0400
@@ -16,7 +16,6 @@
 #cgo darwin,arm64 LDFLAGS: -framework CoreFoundation
 #cgo dragonfly LDFLAGS: -lpthread
 #cgo freebsd LDFLAGS: -lpthread
-#cgo android LDFLAGS: -llog
 #cgo !android,linux LDFLAGS: -lpthread
 #cgo netbsd LDFLAGS: -lpthread
 #cgo openbsd LDFLAGS: -lpthread
