--- a/src/runtime/cgo/gcc_fatalf.c	2026-08-22 10:55:00.000000000 +1200
+++ b/src/runtime/cgo/gcc_fatalf.c	2026-08-22 10:55:00.000000000 +1200
@@ -21,7 +21,7 @@
 	va_end(ap);
 	fprintf(stderr, "\n");

-#ifdef __ANDROID__
+#if defined(__ANDROID__) && !defined(__TERMUX__)
 	// When running from an Android .apk, /dev/stderr and /dev/stdout
 	// redirect to /dev/null. And when running a test binary
 	// via adb shell, it's easy to miss logcat. So write to both.
