diff -uNr gitea-1.11.2/Makefile gitea-1.11.2.mod/Makefile
--- gitea-1.11.2/Makefile	2020-03-06 09:38:11.000000000 +0200
+++ gitea-1.11.2.mod/Makefile	2020-03-08 19:37:43.594425248 +0200
@@ -430,7 +430,7 @@
 
 .PHONY: generate
 generate:
-	GO111MODULE=on $(GO) generate -mod=vendor $(PACKAGES)
+	GO111MODULE=on CGO_ENABLED=0 CGO_LDFLAGS="" CGO_CFLAGS="" GOOS=linux GOARCH=amd64 $(GO) generate -mod=vendor $(PACKAGES)
 
 $(EXECUTABLE): $(GO_SOURCES)
 	GO111MODULE=on $(GO) build -mod=vendor $(GOFLAGS) $(EXTRA_GOFLAGS) -tags '$(TAGS)' -ldflags '-s -w $(LDFLAGS)' -o $@
