diff -ur orig/udocker/engine/base.py mod/udocker/engine/base.py
--- orig/udocker/engine/base.py	2024-08-28 16:01:10.000000000 +0200
+++ mod/udocker/engine/base.py	2025-05-12 15:23:35.642041791 +0200
@@ -690,4 +690,4 @@
         if not qemu_path:
             Msg().err("Warning: qemu required but not available", l=Msg.WAR)
             return ""
-        return qemu_path if return_path else qemu_filename
+        return qemu_path if return_path else qemu_path
