--- a/Makefile.am	2013-04-13 09:26:52.000000000 -0400
+++ b/Makefile.am	2026-07-29 23:12:17.215235725 -0400
@@ -3,7 +3,7 @@
 ACLOCAL_AMFLAGS=-I m4
 
 # Subdirectories to scan
-SUBDIRS = build/mingw po src tests
+SUBDIRS = build/mingw po src
 
 if USE_DOCGEN
   SUBDIRS += docs
--- a/configure.ac	2016-04-04 01:59:39.000000000 -0400
+++ b/configure.ac	2026-07-29 23:12:55.891235725 -0400
@@ -42,7 +42,6 @@
   src/romdump_9x/Makefile
   src/romdump_89t_usb/Makefile
   src/romdump_834pce_usb/Makefile
-  tests/Makefile
   ticalcs2.pc
 ])
 
