--- ../cache/bastet-0.43.2/Config.cpp	2015-08-30 07:04:24.000000000 +0000
+++ ./Config.cpp	2017-09-19 08:56:50.122235753 +0000
@@ -38,7 +38,7 @@
   const size_t HowManyHighScores=10;
   const std::string RcFileName="/.bastetrc";
   const std::string LocalHighScoresFileName="/.bastetscores";
-  const std::string GlobalHighScoresFileName="/var/games/bastet.scores2";
+  const std::string GlobalHighScoresFileName="@TERMUX_PREFIX@/var/games/bastet.scores2";
 
   bool HighScores::Qualifies(int score){
     stable_sort(begin(),end());
