--- runc-1.5.1/libcontainer/criu_linux.go	2025-01-07 04:50:22.000000000 +0530
+++ runc-1.5.1.mod/libcontainer/criu_linux.go	2025-01-17 02:05:05.490165001 +0530
@@ -167,7 +167,7 @@
 	} else {
 		// If the mentioned annotation has not been found, specify
 		// a default CRIU configuration file.
-		rpcOpts.ConfigFile = mkPtr("/etc/criu/runc.conf")
+		rpcOpts.ConfigFile = mkPtr("@TERMUX_PREFIX@/etc/criu/runc.conf")
 	}
 }
