--- runc-1.2.4/libcontainer/criu_linux.go	2025-01-07 04:50:22.000000000 +0530
+++ runc-1.2.4.mod/libcontainer/criu_linux.go	2025-01-17 02:05:05.490165001 +0530
@@ -160,7 +160,7 @@
 	} else {
 		// If the mentioned annotation has not been found, specify
 		// a default CRIU configuration file.
-		rpcOpts.ConfigFile = proto.String("/etc/criu/runc.conf")
+		rpcOpts.ConfigFile = proto.String("@TERMUX_PREFIX@/etc/criu/runc.conf")
 	}
 }
