diff -uNr runc-1.0.0-rc8/libcontainer/container_linux.go runc-1.0.0-rc8.mod/libcontainer/container_linux.go
--- runc-1.0.0-rc8/libcontainer/container_linux.go	2019-04-25 00:48:25.000000000 +0300
+++ runc-1.0.0-rc8.mod/libcontainer/container_linux.go	2019-07-14 15:24:27.714067266 +0300
@@ -899,7 +899,7 @@
 	} else {
 		// If the mentioned annotation has not been found, specify
 		// a default CRIU configuration file.
-		rpcOpts.ConfigFile = proto.String("/etc/criu/runc.conf")
+		rpcOpts.ConfigFile = proto.String("@TERMUX_PREFIX@/etc/criu/runc.conf")
 	}
 }
 
