--- a/xmlif/xmlif.c
+++ b/xmlif/xmlif.c
@@ -1151,7 +1151,7 @@
 
 static char **selections;	/* selection tokens */
 static int nselections;		/* number of selections */
-static ifsense;			/* sense of last `if' or unless seen */
+static int ifsense;			/* sense of last `if' or unless seen */
 static char *attribute;		/* last attribute scanned */
 
 struct stack_t {
@@ -2632,6 +2632,7 @@
 
 int yywrap() {exit(0);};
 
+int
 main(int argc, char *argv[])
 {
     int i;
