diff -uNr keepassxc-2.6.6/src/gui/entry/EntryView.cpp keepassxc-2.6.6.mod/src/gui/entry/EntryView.cpp
--- keepassxc-2.6.6/src/gui/entry/EntryView.cpp	2021-06-12 05:13:52.000000000 +0300
+++ keepassxc-2.6.6.mod/src/gui/entry/EntryView.cpp	2021-07-25 20:39:53.910034225 +0300
@@ -18,7 +18,7 @@
 
 #include "EntryView.h"
 
-#include <QAccessible>
+#include <QCoreApplication>
 #include <QHeaderView>
 #include <QKeyEvent>
 #include <QMenu>
@@ -150,18 +150,15 @@
 
     int last = m_model->rowCount() - 1;
     if (last > 0) {
-        QAccessibleEvent accessibleEvent(this, QAccessible::PageChanged);
         if (event->key() == Qt::Key_Up && currentIndex().row() == 0) {
             QModelIndex index = m_sortModel->mapToSource(m_sortModel->index(last, 0));
             setCurrentEntry(m_model->entryFromIndex(index));
-            QAccessible::updateAccessibility(&accessibleEvent);
             return;
         }
 
         if (event->key() == Qt::Key_Down && currentIndex().row() == last) {
             QModelIndex index = m_sortModel->mapToSource(m_sortModel->index(0, 0));
             setCurrentEntry(m_model->entryFromIndex(index));
-            QAccessible::updateAccessibility(&accessibleEvent);
             return;
         }
     }
