diff -uNr bash-5.0/lib/malloc/table.c bash-5.0.mod/lib/malloc/table.c
--- bash-5.0/lib/malloc/table.c	2016-01-15 15:47:26.000000000 +0200
+++ bash-5.0.mod/lib/malloc/table.c	2019-02-20 14:18:05.854521552 +0200
@@ -390,7 +390,7 @@
   _location_dump_table (stderr);
 }
 
-#define LOCROOT "/var/tmp/maltrace/locations."
+#define LOCROOT "@TERMUX_PREFIX@/var/tmp/maltrace/locations."
 
 void
 mlocation_write_table ()
