diff -uNr bash-5.0/lib/readline/complete.c bash-5.0.mod/lib/readline/complete.c
--- bash-5.0/lib/readline/complete.c	2017-07-05 02:43:20.000000000 +0300
+++ bash-5.0.mod/lib/readline/complete.c	2019-02-20 14:15:49.683440481 +0200
@@ -2231,7 +2231,7 @@
 char *
 rl_username_completion_function (const char *text, int state)
 {
-#if defined (__WIN32__) || defined (__OPENNT)
+#if defined (__WIN32__) || defined (__OPENNT) || defined (__ANDROID__)
   return (char *)NULL;
 #else /* !__WIN32__ && !__OPENNT) */
   static char *username = (char *)NULL;
