Fix breakage on android-21 due "conflicting type for '__errno'"

diff -u -r ../bash-4.3/y.tab.c ./y.tab.c
--- ../bash-4.3/y.tab.c	2014-02-11 10:57:47.000000000 -0500
+++ ./y.tab.c	2014-12-16 05:39:58.047338124 -0500
@@ -280,7 +280,6 @@
 extern int bash_input_fd_changed;
 #endif
 
-extern int errno;
 /* **************************************************************** */
 /*								    */
 /*		    "Forward" declarations			    */
